<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>11.4 Bonus: Efficient R | Landscape Genetic Data Analysis with R</title>
  <meta name="description" content="This is a web-interface to the teaching materials for the lab course ‘Landscape Genetic Data Analysis with R’ associated with the distributed graduate course ‘DGS Landscape Genetics’. The output format is bookdown::gitbook." />
  <meta name="generator" content="bookdown 0.37 and GitBook 2.6.7" />

  <meta property="og:title" content="11.4 Bonus: Efficient R | Landscape Genetic Data Analysis with R" />
  <meta property="og:type" content="book" />
  
  <meta property="og:description" content="This is a web-interface to the teaching materials for the lab course ‘Landscape Genetic Data Analysis with R’ associated with the distributed graduate course ‘DGS Landscape Genetics’. The output format is bookdown::gitbook." />
  <meta name="github-repo" content="hhwagner1/LandGenCourse_book" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="11.4 Bonus: Efficient R | Landscape Genetic Data Analysis with R" />
  
  <meta name="twitter:description" content="This is a web-interface to the teaching materials for the lab course ‘Landscape Genetic Data Analysis with R’ associated with the distributed graduate course ‘DGS Landscape Genetics’. The output format is bookdown::gitbook." />
  

<meta name="author" content="Editor: Helene Wagner (University of Toronto)" />


<meta name="date" content="2024-01-25" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="r-exercise-week-8.html"/>
<link rel="next" href="Week9.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>
<script src="libs/kePrint-0.0.1/kePrint.js"></script>
<link href="libs/lightable-0.0.1/lightable.css" rel="stylesheet" />
<script src="libs/htmlwidgets-1.6.4/htmlwidgets.js"></script>
<link href="libs/leaflet-1.3.1/leaflet.css" rel="stylesheet" />
<script src="libs/leaflet-1.3.1/leaflet.js"></script>
<link href="libs/leafletfix-1.0.0/leafletfix.css" rel="stylesheet" />
<script src="libs/proj4-2.6.2/proj4.min.js"></script>
<script src="libs/Proj4Leaflet-1.0.1/proj4leaflet.js"></script>
<link href="libs/rstudio_leaflet-1.3.1/rstudio_leaflet.css" rel="stylesheet" />
<script src="libs/leaflet-binding-2.2.1/leaflet.js"></script>
<script src="libs/leaflet-providers-2.0.0/leaflet-providers_2.0.0.js"></script>
<script src="libs/leaflet-providers-plugin-2.2.1/leaflet-providers-plugin.js"></script>
<script src="libs/d3-3.5.6/d3.min.js"></script>
<link href="libs/profvis-0.3.6.9000/profvis.css" rel="stylesheet" />
<script src="libs/profvis-0.3.6.9000/profvis.js"></script>
<script src="libs/profvis-0.3.6.9000/scroll.js"></script>
<link href="libs/highlight-6.2.0/textmate.css" rel="stylesheet" />
<script src="libs/highlight-6.2.0/highlight.js"></script>
<script src="libs/profvis-binding-0.3.8/profvis.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
  
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">Landscape Genetic Data Analysis with R</a></li>

<li class="divider"></li>
<li class="part"><span><b>Getting Started</b></span></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Introduction</a>
<ul>
<li class="chapter" data-level="1.1" data-path="how-to-use-this-book.html"><a href="how-to-use-this-book.html"><i class="fa fa-check"></i><b>1.1</b> How to use this Book</a>
<ul>
<li class="chapter" data-level="" data-path="how-to-use-this-book.html"><a href="how-to-use-this-book.html#book-structure"><i class="fa fa-check"></i>1. Book Structure</a></li>
<li class="chapter" data-level="" data-path="how-to-use-this-book.html"><a href="how-to-use-this-book.html#find-what-is-relevant-for-you"><i class="fa fa-check"></i>2. Find what is relevant for you</a></li>
<li class="chapter" data-level="" data-path="how-to-use-this-book.html"><a href="how-to-use-this-book.html#course-r-package-landgencourse"><i class="fa fa-check"></i>3. Course R package ‘LandGenCourse’</a></li>
</ul></li>
<li class="chapter" data-level="1.2" data-path="list-of-packages.html"><a href="list-of-packages.html"><i class="fa fa-check"></i><b>1.2</b> List of R Packages by Vignette</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="review-of-r-skills.html"><a href="review-of-r-skills.html"><i class="fa fa-check"></i><b>2</b> Review of R Skills</a>
<ul>
<li class="chapter" data-level="2.1" data-path="basic-r.html"><a href="basic-r.html"><i class="fa fa-check"></i><b>2.1</b> Basic R Programming</a>
<ul>
<li class="chapter" data-level="" data-path="basic-r.html"><a href="basic-r.html#overview"><i class="fa fa-check"></i>1. Overview</a></li>
<li class="chapter" data-level="" data-path="basic-r.html"><a href="basic-r.html#data-types"><i class="fa fa-check"></i>2. Data Types</a></li>
<li class="chapter" data-level="" data-path="basic-r.html"><a href="basic-r.html#data-containers"><i class="fa fa-check"></i>3. Data Containers</a></li>
<li class="chapter" data-level="" data-path="basic-r.html"><a href="basic-r.html#programming"><i class="fa fa-check"></i>4. Programming</a></li>
</ul></li>
<li class="chapter" data-level="2.2" data-path="r-graphics.html"><a href="r-graphics.html"><i class="fa fa-check"></i><b>2.2</b> R Graphics</a>
<ul>
<li class="chapter" data-level="" data-path="r-graphics.html"><a href="r-graphics.html#overview-1"><i class="fa fa-check"></i>1. Overview</a></li>
<li class="chapter" data-level="" data-path="r-graphics.html"><a href="r-graphics.html#univariate-plots"><i class="fa fa-check"></i>2. Univariate plots</a></li>
<li class="chapter" data-level="" data-path="r-graphics.html"><a href="r-graphics.html#bivariate-plots"><i class="fa fa-check"></i>3. Bivariate plots</a></li>
<li class="chapter" data-level="" data-path="r-graphics.html"><a href="r-graphics.html#advanced-plotting"><i class="fa fa-check"></i>4. Advanced plotting</a></li>
</ul></li>
<li class="chapter" data-level="2.3" data-path="resources.html"><a href="resources.html"><i class="fa fa-check"></i><b>2.3</b> Further R resources</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="github-for-group-projects.html"><a href="github-for-group-projects.html"><i class="fa fa-check"></i><b>3</b> GitHub for Group Projects</a>
<ul>
<li class="chapter" data-level="3.1" data-path="github.html"><a href="github.html"><i class="fa fa-check"></i><b>3.1</b> Why use GitHub?</a>
<ul>
<li class="chapter" data-level="" data-path="github.html"><a href="github.html#goals"><i class="fa fa-check"></i>1. Goals</a></li>
<li class="chapter" data-level="" data-path="github.html"><a href="github.html#version"><i class="fa fa-check"></i>2. Watch course video: Version Control 101</a></li>
<li class="chapter" data-level="" data-path="github.html"><a href="github.html#slides"><i class="fa fa-check"></i>3. Preview slides</a></li>
<li class="chapter" data-level="" data-path="github.html"><a href="github.html#next-steps"><i class="fa fa-check"></i>4. Next steps</a></li>
</ul></li>
<li class="chapter" data-level="3.2" data-path="install.html"><a href="install.html"><i class="fa fa-check"></i><b>3.2</b> Installation</a>
<ul>
<li class="chapter" data-level="" data-path="install.html"><a href="install.html#goals-1"><i class="fa fa-check"></i>1. Goals</a></li>
<li class="chapter" data-level="3.2.1" data-path="install.html"><a href="install.html#how-to-get-github-and-gitkraken-for-free"><i class="fa fa-check"></i><b>3.2.1</b> 2. How to get GitHub and GitKraken for free</a></li>
<li class="chapter" data-level="" data-path="install.html"><a href="install.html#happygit"><i class="fa fa-check"></i>3. Happy Git with R</a></li>
<li class="chapter" data-level="" data-path="install.html"><a href="install.html#gitvideos"><i class="fa fa-check"></i>4. Video tutorials: installation</a></li>
<li class="chapter" data-level="" data-path="install.html"><a href="install.html#join"><i class="fa fa-check"></i>5. Joining your group’s project on GitHub</a></li>
<li class="chapter" data-level="" data-path="install.html"><a href="install.html#repo"><i class="fa fa-check"></i>6. Setting up a group repo</a></li>
<li class="chapter" data-level="" data-path="install.html"><a href="install.html#next-steps-1"><i class="fa fa-check"></i>7. Next steps</a></li>
</ul></li>
<li class="chapter" data-level="3.3" data-path="workflow.html"><a href="workflow.html"><i class="fa fa-check"></i><b>3.3</b> Workflows</a>
<ul>
<li class="chapter" data-level="" data-path="workflow.html"><a href="workflow.html#goals-2"><i class="fa fa-check"></i>1. Goals</a></li>
<li class="chapter" data-level="" data-path="workflow.html"><a href="workflow.html#step-by-step-tutorials"><i class="fa fa-check"></i>2. Step-by-step tutorials</a></li>
<li class="chapter" data-level="" data-path="workflow.html"><a href="workflow.html#further-resources"><i class="fa fa-check"></i>3. Further resources</a></li>
</ul></li>
</ul></li>
<li class="part"><span><b>Basic Topics</b></span></li>
<li class="chapter" data-level="4" data-path="Week1.html"><a href="Week1.html"><i class="fa fa-check"></i><b>4</b> Week 1: Importing Genetic Data</a>
<ul>
<li class="chapter" data-level="4.1" data-path="video_1.html"><a href="video_1.html"><i class="fa fa-check"></i><b>4.1</b> View Course Video</a>
<ul>
<li class="chapter" data-level="" data-path="video_1.html"><a href="video_1.html#embedded-video"><i class="fa fa-check"></i>1. Embedded Video</a></li>
<li class="chapter" data-level="" data-path="video_1.html"><a href="video_1.html#preview-slides"><i class="fa fa-check"></i>2. Preview Slides</a></li>
</ul></li>
<li class="chapter" data-level="4.2" data-path="tutorial_1.html"><a href="tutorial_1.html"><i class="fa fa-check"></i><b>4.2</b> Interactive Tutorial 1</a>
<ul>
<li class="chapter" data-level="" data-path="tutorial_1.html"><a href="tutorial_1.html#list-of-r-commands-covered-this-week"><i class="fa fa-check"></i>1. List of R commands covered this week</a></li>
<li class="chapter" data-level="" data-path="tutorial_1.html"><a href="tutorial_1.html#general-instructions"><i class="fa fa-check"></i>2. General Instructions</a></li>
</ul></li>
<li class="chapter" data-level="4.3" data-path="WE_1.html"><a href="WE_1.html"><i class="fa fa-check"></i><b>4.3</b> Worked Example</a>
<ul>
<li class="chapter" data-level="" data-path="WE_1.html"><a href="WE_1.html#overview-of-worked-example"><i class="fa fa-check"></i>1. Overview of Worked Example</a></li>
<li class="chapter" data-level="" data-path="WE_1.html"><a href="WE_1.html#import-data-from-.csv-file"><i class="fa fa-check"></i>2. Import data from .csv file</a></li>
<li class="chapter" data-level="" data-path="WE_1.html"><a href="WE_1.html#create-genind-object"><i class="fa fa-check"></i>3. Create ‘genind’ Object</a></li>
<li class="chapter" data-level="" data-path="WE_1.html"><a href="WE_1.html#view-information-stored-in-genind-object"><i class="fa fa-check"></i>4. View information stored in ‘genind’ object</a></li>
<li class="chapter" data-level="" data-path="WE_1.html"><a href="WE_1.html#import-data-with-gstudio-package"><i class="fa fa-check"></i>5. Import data with ‘gstudio’ package</a></li>
<li class="chapter" data-level="" data-path="WE_1.html"><a href="WE_1.html#importing-snp-data"><i class="fa fa-check"></i>6. Importing SNP data</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="r-exercise-week-1.html"><a href="r-exercise-week-1.html"><i class="fa fa-check"></i>R exercise Week 1</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="Week2.html"><a href="Week2.html"><i class="fa fa-check"></i><b>5</b> Week 2: Spatial Data</a>
<ul>
<li class="chapter" data-level="5.1" data-path="video_2.html"><a href="video_2.html"><i class="fa fa-check"></i><b>5.1</b> View Course Video</a>
<ul>
<li class="chapter" data-level="" data-path="video_2.html"><a href="video_2.html#embedded-video-1"><i class="fa fa-check"></i>1. Embedded Video</a></li>
<li class="chapter" data-level="" data-path="video_2.html"><a href="video_2.html#preview-slides-1"><i class="fa fa-check"></i>2. Preview Slides</a></li>
</ul></li>
<li class="chapter" data-level="5.2" data-path="tutorial_2.html"><a href="tutorial_2.html"><i class="fa fa-check"></i><b>5.2</b> Interactive Tutorial 2</a>
<ul>
<li class="chapter" data-level="" data-path="tutorial_2.html"><a href="tutorial_2.html#list-of-r-commands-covered-this-week-1"><i class="fa fa-check"></i>1. List of R commands covered this week</a></li>
<li class="chapter" data-level="" data-path="tutorial_2.html"><a href="tutorial_2.html#general-instructions-1"><i class="fa fa-check"></i>2. General Instructions</a></li>
</ul></li>
<li class="chapter" data-level="5.3" data-path="WE_2.html"><a href="WE_2.html"><i class="fa fa-check"></i><b>5.3</b> Worked Example</a>
<ul>
<li class="chapter" data-level="" data-path="WE_2.html"><a href="WE_2.html#overview-of-worked-example-1"><i class="fa fa-check"></i>1. Overview of Worked Example</a></li>
<li class="chapter" data-level="" data-path="WE_2.html"><a href="WE_2.html#import-site-data-from-.csv-file"><i class="fa fa-check"></i>2. Import site data from .csv file</a></li>
<li class="chapter" data-level="" data-path="WE_2.html"><a href="WE_2.html#display-raster-data-and-overlay-sampling-locations-extract-data"><i class="fa fa-check"></i>3. Display raster data and overlay sampling locations, extract data</a></li>
<li class="chapter" data-level="" data-path="WE_2.html"><a href="WE_2.html#calculate-landscape-metrics"><i class="fa fa-check"></i>4. Calculate landscape metrics</a></li>
<li class="chapter" data-level="" data-path="WE_2.html"><a href="WE_2.html#sample-landscape-metrics-within-buffer-around-sampling-locations"><i class="fa fa-check"></i>5. Sample landscape metrics within buffer around sampling locations</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="r-exercise-week-2.html"><a href="r-exercise-week-2.html"><i class="fa fa-check"></i>R Exercise Week 2</a></li>
<li class="chapter" data-level="5.4" data-path="bonus_2a.html"><a href="bonus_2a.html"><i class="fa fa-check"></i><b>5.4</b> Bonus: ‘sf’ and ‘terra’</a>
<ul>
<li class="chapter" data-level="" data-path="bonus_2a.html"><a href="bonus_2a.html#overview-2"><i class="fa fa-check"></i>1. Overview</a></li>
<li class="chapter" data-level="" data-path="bonus_2a.html"><a href="bonus_2a.html#import-and-export-esri-shapefiles"><i class="fa fa-check"></i>2. Import and export ESRI shapefiles</a></li>
<li class="chapter" data-level="" data-path="bonus_2a.html"><a href="bonus_2a.html#compatibility-with-sp-and-raster-objects"><i class="fa fa-check"></i>3. Compatibility with <code>sp</code> and <code>raster</code> objects</a></li>
<li class="chapter" data-level="" data-path="bonus_2a.html"><a href="bonus_2a.html#plotting-spatial-data-with-tmap"><i class="fa fa-check"></i>4. Plotting spatial data with <code>tmap</code></a></li>
<li class="chapter" data-level="" data-path="bonus_2a.html"><a href="bonus_2a.html#plot-a-categorical-map-with-predefined-color-scheme"><i class="fa fa-check"></i>5. Plot a categorical map with predefined color scheme</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="6" data-path="Week3.html"><a href="Week3.html"><i class="fa fa-check"></i><b>6</b> Week 3: Genetic Diversity</a>
<ul>
<li class="chapter" data-level="6.1" data-path="video_3.html"><a href="video_3.html"><i class="fa fa-check"></i><b>6.1</b> View Course Video</a>
<ul>
<li class="chapter" data-level="" data-path="video_3.html"><a href="video_3.html#embedded-video-2"><i class="fa fa-check"></i>1. Embedded Video</a></li>
<li class="chapter" data-level="" data-path="video_3.html"><a href="video_3.html#video-part-1"><i class="fa fa-check"></i>Video, Part 1</a></li>
<li class="chapter" data-level="" data-path="video_3.html"><a href="video_3.html#video-part-2"><i class="fa fa-check"></i>Video, Part 2</a></li>
<li class="chapter" data-level="" data-path="video_3.html"><a href="video_3.html#preview-slides-2"><i class="fa fa-check"></i>Preview Slides</a></li>
</ul></li>
<li class="chapter" data-level="6.2" data-path="tutorial_3.html"><a href="tutorial_3.html"><i class="fa fa-check"></i><b>6.2</b> Interactive Tutorial 3</a>
<ul>
<li class="chapter" data-level="" data-path="tutorial_3.html"><a href="tutorial_3.html#list-of-r-commands-covered-this-week-2"><i class="fa fa-check"></i>1. List of R commands covered this week</a></li>
<li class="chapter" data-level="" data-path="tutorial_3.html"><a href="tutorial_3.html#general-instructions-2"><i class="fa fa-check"></i>2. General Instructions</a></li>
</ul></li>
<li class="chapter" data-level="6.3" data-path="WE_3.html"><a href="WE_3.html"><i class="fa fa-check"></i><b>6.3</b> Worked Example</a></li>
<li class="chapter" data-level="6.4" data-path="r-exercise-week-3.html"><a href="r-exercise-week-3.html"><i class="fa fa-check"></i><b>6.4</b> R Exercise Week 3</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="Week4.html"><a href="Week4.html"><i class="fa fa-check"></i><b>7</b> Week 4: Metapopulations</a>
<ul>
<li class="chapter" data-level="7.1" data-path="video_4.html"><a href="video_4.html"><i class="fa fa-check"></i><b>7.1</b> View Course Video</a>
<ul>
<li class="chapter" data-level="" data-path="video_4.html"><a href="video_4.html#embedded-video-3"><i class="fa fa-check"></i>1. Embedded Video</a></li>
<li class="chapter" data-level="" data-path="video_4.html"><a href="video_4.html#video-part-1-1"><i class="fa fa-check"></i>Video, Part 1</a></li>
<li class="chapter" data-level="" data-path="video_4.html"><a href="video_4.html#video-part-2-1"><i class="fa fa-check"></i>Video, Part 2</a></li>
<li class="chapter" data-level="" data-path="video_4.html"><a href="video_4.html#preview-slides-3"><i class="fa fa-check"></i>Preview Slides</a></li>
</ul></li>
<li class="chapter" data-level="7.2" data-path="tutorial_4.html"><a href="tutorial_4.html"><i class="fa fa-check"></i><b>7.2</b> Interactive Tutorial 4</a>
<ul>
<li class="chapter" data-level="" data-path="tutorial_4.html"><a href="tutorial_4.html#list-of-r-commands-covered-this-week-3"><i class="fa fa-check"></i>1. List of R commands covered this week</a></li>
<li class="chapter" data-level="" data-path="tutorial_4.html"><a href="tutorial_4.html#general-instructions-3"><i class="fa fa-check"></i>2. General Instructions</a></li>
</ul></li>
<li class="chapter" data-level="7.3" data-path="WE_4.html"><a href="WE_4.html"><i class="fa fa-check"></i><b>7.3</b> Worked Example</a>
<ul>
<li class="chapter" data-level="" data-path="WE_4.html"><a href="WE_4.html#overview-of-worked-example-3"><i class="fa fa-check"></i>1. Overview of Worked Example</a></li>
<li class="chapter" data-level="" data-path="WE_4.html"><a href="WE_4.html#spatial-distribution-of-genetic-structure"><i class="fa fa-check"></i>2. Spatial distribution of genetic structure</a></li>
<li class="chapter" data-level="" data-path="WE_4.html"><a href="WE_4.html#what-determines-genetic-differentiation-among-sites"><i class="fa fa-check"></i>3. What determines genetic differentiation among sites?</a></li>
<li class="chapter" data-level="" data-path="WE_4.html"><a href="WE_4.html#what-determines-genetic-diversity"><i class="fa fa-check"></i>4. What determines genetic diversity?</a></li>
<li class="chapter" data-level="" data-path="WE_4.html"><a href="WE_4.html#are-genetic-differentiation-and-diversity-related"><i class="fa fa-check"></i>5. Are genetic differentiation and diversity related?</a></li>
<li class="chapter" data-level="" data-path="WE_4.html"><a href="WE_4.html#effect-of-recent-extinction-events"><i class="fa fa-check"></i>6. Effect of recent extinction events</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="r-exercise-week-4.html"><a href="r-exercise-week-4.html"><i class="fa fa-check"></i>R Exercise Week 4</a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="Week5.html"><a href="Week5.html"><i class="fa fa-check"></i><b>8</b> Week 5: Spatial Statistics</a>
<ul>
<li class="chapter" data-level="8.1" data-path="video_5.html"><a href="video_5.html"><i class="fa fa-check"></i><b>8.1</b> View Course Video</a>
<ul>
<li class="chapter" data-level="" data-path="video_5.html"><a href="video_5.html#embedded-video-4"><i class="fa fa-check"></i>1. Embedded Video</a></li>
<li class="chapter" data-level="" data-path="video_5.html"><a href="video_5.html#preview-slides-4"><i class="fa fa-check"></i>Preview Slides</a></li>
</ul></li>
<li class="chapter" data-level="8.2" data-path="tutorial_5.html"><a href="tutorial_5.html"><i class="fa fa-check"></i><b>8.2</b> Interactive Tutorial 5</a>
<ul>
<li class="chapter" data-level="" data-path="tutorial_5.html"><a href="tutorial_5.html#list-of-r-commands-covered-this-week-4"><i class="fa fa-check"></i>1. List of R commands covered this week</a></li>
<li class="chapter" data-level="" data-path="tutorial_5.html"><a href="tutorial_5.html#general-instructions-4"><i class="fa fa-check"></i>2. General Instructions</a></li>
</ul></li>
<li class="chapter" data-level="8.3" data-path="WE_5.html"><a href="WE_5.html"><i class="fa fa-check"></i><b>8.3</b> Worked Example</a>
<ul>
<li class="chapter" data-level="" data-path="WE_5.html"><a href="WE_5.html#overview-of-worked-example-4"><i class="fa fa-check"></i>1. Overview of Worked Example</a></li>
<li class="chapter" data-level="" data-path="WE_5.html"><a href="WE_5.html#data-import-and-manipulation"><i class="fa fa-check"></i>2. Data import and manipulation</a></li>
<li class="chapter" data-level="" data-path="WE_5.html"><a href="WE_5.html#calculate-genetic-distances"><i class="fa fa-check"></i>3. Calculate genetic distances</a></li>
<li class="chapter" data-level="" data-path="WE_5.html"><a href="WE_5.html#perform-a-mantel-test-to-test-for-ibd"><i class="fa fa-check"></i>4. Perform a Mantel test to test for IBD</a></li>
<li class="chapter" data-level="" data-path="WE_5.html"><a href="WE_5.html#create-mantel-correlogram-for-genetic-data"><i class="fa fa-check"></i>5. Create Mantel correlogram for genetic data</a></li>
<li class="chapter" data-level="" data-path="WE_5.html"><a href="WE_5.html#specify-spatial-weights-and-calculate-morans-i"><i class="fa fa-check"></i>6. Specify spatial weights and calculate Moran’s I</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="r-exercise-week-5.html"><a href="r-exercise-week-5.html"><i class="fa fa-check"></i>R Exercise Week 5</a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="Week6.html"><a href="Week6.html"><i class="fa fa-check"></i><b>9</b> Week 6: Quantitative Genetics</a>
<ul>
<li class="chapter" data-level="9.1" data-path="video_6.html"><a href="video_6.html"><i class="fa fa-check"></i><b>9.1</b> View Course Video</a>
<ul>
<li class="chapter" data-level="" data-path="video_6.html"><a href="video_6.html#embedded-video-5"><i class="fa fa-check"></i>1. Embedded Video</a></li>
<li class="chapter" data-level="" data-path="video_6.html"><a href="video_6.html#video-part-1-2"><i class="fa fa-check"></i>Video, Part 1</a></li>
<li class="chapter" data-level="" data-path="video_6.html"><a href="video_6.html#video-part-2-2"><i class="fa fa-check"></i>Video, Part 2</a></li>
<li class="chapter" data-level="" data-path="video_6.html"><a href="video_6.html#preview-slides-5"><i class="fa fa-check"></i>Preview Slides</a></li>
</ul></li>
<li class="chapter" data-level="9.2" data-path="tutorial_6.html"><a href="tutorial_6.html"><i class="fa fa-check"></i><b>9.2</b> Interactive Tutorial 6</a>
<ul>
<li class="chapter" data-level="" data-path="tutorial_6.html"><a href="tutorial_6.html#list-of-r-commands-covered-this-week-5"><i class="fa fa-check"></i>1. List of R commands covered this week</a></li>
<li class="chapter" data-level="" data-path="tutorial_6.html"><a href="tutorial_6.html#general-instructions-5"><i class="fa fa-check"></i>2. General Instructions</a></li>
</ul></li>
<li class="chapter" data-level="9.3" data-path="WE_6.html"><a href="WE_6.html"><i class="fa fa-check"></i><b>9.3</b> Worked Example</a>
<ul>
<li class="chapter" data-level="" data-path="WE_6.html"><a href="WE_6.html#overview-of-worked-example-5"><i class="fa fa-check"></i>1. Overview of Worked Example</a></li>
<li class="chapter" data-level="" data-path="WE_6.html"><a href="WE_6.html#estimate-genetic-and-non-genetic-variance-components-from-a-common-garden-experiment"><i class="fa fa-check"></i>2. Estimate genetic and non-genetic variance components from a common garden experiment</a></li>
<li class="chapter" data-level="" data-path="WE_6.html"><a href="WE_6.html#estimate-trait-heritability"><i class="fa fa-check"></i>3. Estimate trait heritability</a></li>
<li class="chapter" data-level="" data-path="WE_6.html"><a href="WE_6.html#compare-q_st-to-f_st-measured-from-snp-data"><i class="fa fa-check"></i>4. Compare <span class="math inline">\(Q_{st}\)</span> to <span class="math inline">\(F_{st}\)</span> measured from SNP data</a></li>
<li class="chapter" data-level="" data-path="WE_6.html"><a href="WE_6.html#assess-correlation-between-trait-and-environment"><i class="fa fa-check"></i>5. Assess correlation between trait and environment</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="r-exercise-week-6.html"><a href="r-exercise-week-6.html"><i class="fa fa-check"></i>R Exercise Week 6</a></li>
</ul></li>
<li class="chapter" data-level="10" data-path="Week7.html"><a href="Week7.html"><i class="fa fa-check"></i><b>10</b> Week 7: Spatial Linear Models</a>
<ul>
<li class="chapter" data-level="10.1" data-path="video_7.html"><a href="video_7.html"><i class="fa fa-check"></i><b>10.1</b> View Course Video</a>
<ul>
<li class="chapter" data-level="" data-path="video_7.html"><a href="video_7.html#embedded-video-6"><i class="fa fa-check"></i>1. Embedded Video</a></li>
<li class="chapter" data-level="" data-path="video_7.html"><a href="video_7.html#video-part-1-3"><i class="fa fa-check"></i>Video, Part 1</a></li>
<li class="chapter" data-level="" data-path="video_7.html"><a href="video_7.html#video-part-2-3"><i class="fa fa-check"></i>Video, Part 2</a></li>
<li class="chapter" data-level="" data-path="video_7.html"><a href="video_7.html#preview-slides-6"><i class="fa fa-check"></i>Preview Slides</a></li>
</ul></li>
<li class="chapter" data-level="10.2" data-path="tutorial_7.html"><a href="tutorial_7.html"><i class="fa fa-check"></i><b>10.2</b> Interactive Tutorial 7</a>
<ul>
<li class="chapter" data-level="" data-path="tutorial_7.html"><a href="tutorial_7.html#list-of-r-commands-covered-this-week-6"><i class="fa fa-check"></i>1. List of R commands covered this week</a></li>
<li class="chapter" data-level="" data-path="tutorial_7.html"><a href="tutorial_7.html#general-instructions-6"><i class="fa fa-check"></i>2. General Instructions</a></li>
</ul></li>
<li class="chapter" data-level="10.3" data-path="WE_7.html"><a href="WE_7.html"><i class="fa fa-check"></i><b>10.3</b> Worked Example</a>
<ul>
<li class="chapter" data-level="" data-path="WE_7.html"><a href="WE_7.html#overview-of-worked-example-6"><i class="fa fa-check"></i>1. Overview of Worked Example</a></li>
<li class="chapter" data-level="" data-path="WE_7.html"><a href="WE_7.html#explore-data-set"><i class="fa fa-check"></i>2. Explore data set</a></li>
<li class="chapter" data-level="" data-path="WE_7.html"><a href="WE_7.html#test-regression-residuals-for-spatial-autocorrelation"><i class="fa fa-check"></i>3. Test regression residuals for spatial autocorrelation</a></li>
<li class="chapter" data-level="" data-path="WE_7.html"><a href="WE_7.html#fit-models-with-spatially-correlated-error-gls-with-package-nlme"><i class="fa fa-check"></i>4. Fit models with spatially correlated error (GLS) with package ‘nlme’</a></li>
<li class="chapter" data-level="" data-path="WE_7.html"><a href="WE_7.html#fit-spatial-simultaneous-autoregressive-error-models-sar"><i class="fa fa-check"></i>5. Fit spatial simultaneous autoregressive error models (SAR)</a></li>
<li class="chapter" data-level="" data-path="WE_7.html"><a href="WE_7.html#spatial-filtering-with-mem-using-package-spmoran"><i class="fa fa-check"></i>6. Spatial filtering with MEM using package ‘spmoran’</a></li>
<li class="chapter" data-level="" data-path="WE_7.html"><a href="WE_7.html#fit-spatially-varying-coefficients-model-with-package-spmoran"><i class="fa fa-check"></i>7. Fit spatially varying coefficients model with package ‘spmoran’</a></li>
<li class="chapter" data-level="" data-path="WE_7.html"><a href="WE_7.html#model-with-patchsize-and-ibr"><i class="fa fa-check"></i>Model with PatchSize and IBR</a></li>
<li class="chapter" data-level="" data-path="WE_7.html"><a href="WE_7.html#model-with-ibr-only"><i class="fa fa-check"></i>Model with IBR only</a></li>
<li class="chapter" data-level="" data-path="WE_7.html"><a href="WE_7.html#conclusions"><i class="fa fa-check"></i>8. Conclusions</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="r-exercise-week-7.html"><a href="r-exercise-week-7.html"><i class="fa fa-check"></i>R Exercise Week 7</a></li>
<li class="chapter" data-level="10.4" data-path="bonus_7a.html"><a href="bonus_7a.html"><i class="fa fa-check"></i><b>10.4</b> Bonus: Si index</a>
<ul>
<li class="chapter" data-level="" data-path="bonus_7a.html"><a href="bonus_7a.html#overview-of-bonus-materials"><i class="fa fa-check"></i>1. Overview of Bonus Materials</a></li>
<li class="chapter" data-level="" data-path="bonus_7a.html"><a href="bonus_7a.html#import-ecological-distance-matrices"><i class="fa fa-check"></i>2. Import ecological distance matrices</a></li>
<li class="chapter" data-level="" data-path="bonus_7a.html"><a href="bonus_7a.html#optimize-the-scaling-parameter-alpha"><i class="fa fa-check"></i>3. Optimize the scaling parameter alpha</a></li>
<li class="chapter" data-level="" data-path="bonus_7a.html"><a href="bonus_7a.html#calculate-hanskis-index-si-with-source-patch-parameters"><i class="fa fa-check"></i>4. Calculate Hanski’s index Si with source patch parameters</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="11" data-path="Week8.html"><a href="Week8.html"><i class="fa fa-check"></i><b>11</b> Week 8: Simulation Experiments</a>
<ul>
<li class="chapter" data-level="11.1" data-path="video_8.html"><a href="video_8.html"><i class="fa fa-check"></i><b>11.1</b> View Course Video</a>
<ul>
<li class="chapter" data-level="" data-path="video_8.html"><a href="video_8.html#embedded-video-7"><i class="fa fa-check"></i>1. Embedded Video</a></li>
<li class="chapter" data-level="" data-path="video_8.html"><a href="video_8.html#video-part-1-4"><i class="fa fa-check"></i>Video, Part 1</a></li>
<li class="chapter" data-level="" data-path="video_8.html"><a href="video_8.html#video-part-2-4"><i class="fa fa-check"></i>Video, Part 2</a></li>
<li class="chapter" data-level="" data-path="video_8.html"><a href="video_8.html#preview-slides-7"><i class="fa fa-check"></i>Preview Slides</a></li>
</ul></li>
<li class="chapter" data-level="11.2" data-path="tutorial_8.html"><a href="tutorial_8.html"><i class="fa fa-check"></i><b>11.2</b> Interactive Tutorial 8</a>
<ul>
<li class="chapter" data-level="" data-path="tutorial_8.html"><a href="tutorial_8.html#list-of-r-commands-covered-this-week-7"><i class="fa fa-check"></i>1. List of R commands covered this week</a></li>
<li class="chapter" data-level="" data-path="tutorial_8.html"><a href="tutorial_8.html#general-instructions-7"><i class="fa fa-check"></i>2. General Instructions</a></li>
</ul></li>
<li class="chapter" data-level="11.3" data-path="WE_8.html"><a href="WE_8.html"><i class="fa fa-check"></i><b>11.3</b> Worked Example</a>
<ul>
<li class="chapter" data-level="" data-path="WE_8.html"><a href="WE_8.html#overview-of-worked-example-7"><i class="fa fa-check"></i>1. Overview of Worked Example</a></li>
<li class="chapter" data-level="" data-path="WE_8.html"><a href="WE_8.html#initialize-a-landscape"><i class="fa fa-check"></i>2. Initialize a landscape</a></li>
<li class="chapter" data-level="" data-path="WE_8.html"><a href="WE_8.html#define-simulation-parameters"><i class="fa fa-check"></i>3. Define simulation parameters</a></li>
<li class="chapter" data-level="" data-path="WE_8.html"><a href="WE_8.html#run-simulations-and-analyze-results"><i class="fa fa-check"></i>4. Run simulations and analyze results</a></li>
<li class="chapter" data-level="" data-path="WE_8.html"><a href="WE_8.html#run-simulator-using-a-previously-defined-parameter-set"><i class="fa fa-check"></i>5. Run simulator using a previously defined parameter set</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="r-exercise-week-8.html"><a href="r-exercise-week-8.html"><i class="fa fa-check"></i>R Exercise Week 8</a></li>
<li class="chapter" data-level="11.4" data-path="bonus_8a.html"><a href="bonus_8a.html"><i class="fa fa-check"></i><b>11.4</b> Bonus: Efficient R</a></li>
</ul></li>
<li class="part"><span><b>Advanced Topics</b></span></li>
<li class="chapter" data-level="12" data-path="Week9.html"><a href="Week9.html"><i class="fa fa-check"></i><b>12</b> Week 9: Population Structure</a>
<ul>
<li class="chapter" data-level="12.1" data-path="WE_9.html"><a href="WE_9.html"><i class="fa fa-check"></i><b>12.1</b> Worked Example</a>
<ul>
<li class="chapter" data-level="" data-path="WE_9.html"><a href="WE_9.html#overview-of-worked-example-8"><i class="fa fa-check"></i>1. Overview of Worked Example</a></li>
<li class="chapter" data-level="" data-path="WE_9.html"><a href="WE_9.html#simulated-data-2-island-model"><i class="fa fa-check"></i>2. Simulated data: 2-island model</a></li>
<li class="chapter" data-level="" data-path="WE_9.html"><a href="WE_9.html#simulated-data-2-island-model-with-admixture"><i class="fa fa-check"></i>3. Simulated data: 2-island model with admixture</a></li>
<li class="chapter" data-level="" data-path="WE_9.html"><a href="WE_9.html#empirical-data-threespine-sticklebacks"><i class="fa fa-check"></i>4. Empirical data: Threespine sticklebacks</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="13" data-path="Week10.html"><a href="Week10.html"><i class="fa fa-check"></i><b>13</b> Week 10: Landscape Resistance</a>
<ul>
<li class="chapter" data-level="13.1" data-path="WE_10.html"><a href="WE_10.html"><i class="fa fa-check"></i><b>13.1</b> Worked Example</a>
<ul>
<li class="chapter" data-level="" data-path="WE_10.html"><a href="WE_10.html#overview-of-worked-example-9"><i class="fa fa-check"></i>1. Overview of Worked Example</a></li>
<li class="chapter" data-level="" data-path="WE_10.html"><a href="WE_10.html#explore-the-data-set"><i class="fa fa-check"></i>2. Explore the data set</a></li>
<li class="chapter" data-level="" data-path="WE_10.html"><a href="WE_10.html#setting-cost-values-and-calculating-conductance"><i class="fa fa-check"></i>3. Setting cost values and calculating conductance</a></li>
<li class="chapter" data-level="" data-path="WE_10.html"><a href="WE_10.html#convert-conductance-into-effective-distance"><i class="fa fa-check"></i>4. Convert conductance into effective distance</a></li>
<li class="chapter" data-level="" data-path="WE_10.html"><a href="WE_10.html#create-cost-distance-matrices"><i class="fa fa-check"></i>5. Create cost-distance matrices</a></li>
<li class="chapter" data-level="" data-path="WE_10.html"><a href="WE_10.html#how-does-changing-resolution-affect-these-metrics"><i class="fa fa-check"></i>6. How does changing resolution affect these metrics?</a></li>
<li class="chapter" data-level="" data-path="WE_10.html"><a href="WE_10.html#references"><i class="fa fa-check"></i>7. References</a></li>
</ul></li>
<li class="chapter" data-level="13.2" data-path="bonus-10a.html"><a href="bonus-10a.html"><i class="fa fa-check"></i><b>13.2</b> Bonus: ‘radish’ tutorial</a>
<ul>
<li class="chapter" data-level="" data-path="bonus-10a.html"><a href="bonus-10a.html#overview-of-bonus-materials-1"><i class="fa fa-check"></i>1. Overview of Bonus Materials</a></li>
<li class="chapter" data-level="" data-path="bonus-10a.html"><a href="bonus-10a.html#import-data"><i class="fa fa-check"></i>2. Import data</a></li>
<li class="chapter" data-level="" data-path="bonus-10a.html"><a href="bonus-10a.html#basic-model-fitting"><i class="fa fa-check"></i>3. Basic model fitting</a></li>
<li class="chapter" data-level="" data-path="bonus-10a.html"><a href="bonus-10a.html#advanced-modeling-topics"><i class="fa fa-check"></i>4. Advanced modeling topics</a></li>
<li class="chapter" data-level="" data-path="bonus-10a.html"><a href="bonus-10a.html#summary"><i class="fa fa-check"></i>5. Summary</a></li>
<li class="chapter" data-level="" data-path="bonus-10a.html"><a href="bonus-10a.html#references-1"><i class="fa fa-check"></i>6. References</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="14" data-path="Week11.html"><a href="Week11.html"><i class="fa fa-check"></i><b>14</b> Week 11: Detecting Adaptation</a>
<ul>
<li class="chapter" data-level="14.1" data-path="WE_11.html"><a href="WE_11.html"><i class="fa fa-check"></i><b>14.1</b> Worked Example</a>
<ul>
<li class="chapter" data-level="" data-path="WE_11.html"><a href="WE_11.html#overview-of-worked-example-10"><i class="fa fa-check"></i>1. Overview of Worked Example</a></li>
<li class="chapter" data-level="" data-path="WE_11.html"><a href="WE_11.html#import-and-prepare-the-data"><i class="fa fa-check"></i>2. Import and prepare the data</a></li>
<li class="chapter" data-level="" data-path="WE_11.html"><a href="WE_11.html#latent-factor-mixed-models-lfmm-a-univariate-gea"><i class="fa fa-check"></i>3. Latent Factor Mixed Models (LFMM): a univariate GEA</a></li>
<li class="chapter" data-level="" data-path="WE_11.html"><a href="WE_11.html#redundancy-analysis-rda-a-multivariate-gea"><i class="fa fa-check"></i>4. Redundancy Analysis (RDA): a multivariate GEA</a></li>
<li class="chapter" data-level="" data-path="WE_11.html"><a href="WE_11.html#compare-lfmm-and-rda-candidates"><i class="fa fa-check"></i>5. Compare LFMM and RDA candidates</a></li>
<li class="chapter" data-level="" data-path="WE_11.html"><a href="WE_11.html#a-quick-note-on-controlling-for-population-structure-in-rda"><i class="fa fa-check"></i>6. A quick note on controlling for population structure in RDA</a></li>
<li class="chapter" data-level="" data-path="WE_11.html"><a href="WE_11.html#references-2"><i class="fa fa-check"></i>7. References</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="15" data-path="Week12.html"><a href="Week12.html"><i class="fa fa-check"></i><b>15</b> Week 12: Model Selection</a>
<ul>
<li class="chapter" data-level="15.1" data-path="WE_12.html"><a href="WE_12.html"><i class="fa fa-check"></i><b>15.1</b> Worked Example</a>
<ul>
<li class="chapter" data-level="" data-path="WE_12.html"><a href="WE_12.html#overview-of-worked-example-11"><i class="fa fa-check"></i>1. Overview of Worked Example</a></li>
<li class="chapter" data-level="" data-path="WE_12.html"><a href="WE_12.html#fitting-candidate-models"><i class="fa fa-check"></i>2. Fitting candidate models</a></li>
<li class="chapter" data-level="" data-path="WE_12.html"><a href="WE_12.html#your-turn-to-test-additional-hypotheses"><i class="fa fa-check"></i>3. Your turn to test additional hypotheses!</a></li>
<li class="chapter" data-level="" data-path="WE_12.html"><a href="WE_12.html#nested-model-nmlpe-for-hierarchical-sampling-designs"><i class="fa fa-check"></i>4. Nested model (NMLPE) for hierarchical sampling designs</a></li>
<li class="chapter" data-level="" data-path="WE_12.html"><a href="WE_12.html#references-3"><i class="fa fa-check"></i>5. References</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="16" data-path="Week13.html"><a href="Week13.html"><i class="fa fa-check"></i><b>16</b> Week 13: Gravity Models</a>
<ul>
<li class="chapter" data-level="16.1" data-path="WE_13.html"><a href="WE_13.html"><i class="fa fa-check"></i><b>16.1</b> Worked Example</a>
<ul>
<li class="chapter" data-level="" data-path="WE_13.html"><a href="WE_13.html#overview-of-worked-example-12"><i class="fa fa-check"></i>1. Overview of Worked Example</a></li>
<li class="chapter" data-level="" data-path="WE_13.html"><a href="WE_13.html#setup"><i class="fa fa-check"></i>2. Setup</a></li>
<li class="chapter" data-level="" data-path="WE_13.html"><a href="WE_13.html#wetland-complex-data-preparation"><i class="fa fa-check"></i>3. Wetland complex data preparation</a></li>
<li class="chapter" data-level="" data-path="WE_13.html"><a href="WE_13.html#wetland-field-data-preparation"><i class="fa fa-check"></i>4. Wetland field-data preparation</a></li>
<li class="chapter" data-level="" data-path="WE_13.html"><a href="WE_13.html#saturated-graph"><i class="fa fa-check"></i>5. Saturated Graph</a></li>
<li class="chapter" data-level="" data-path="WE_13.html"><a href="WE_13.html#spatial-model-data-prepration"><i class="fa fa-check"></i>6. Spatial model data prepration</a></li>
<li class="chapter" data-level="" data-path="WE_13.html"><a href="WE_13.html#gravity-model"><i class="fa fa-check"></i>7. Gravity model</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="17" data-path="Week14.html"><a href="Week14.html"><i class="fa fa-check"></i><b>17</b> Week 14: Contemporary Gene Flow</a>
<ul>
<li class="chapter" data-level="17.1" data-path="WE_14.html"><a href="WE_14.html"><i class="fa fa-check"></i><b>17.1</b> Worked Example</a>
<ul>
<li class="chapter" data-level="" data-path="WE_14.html"><a href="WE_14.html#overview-3"><i class="fa fa-check"></i>1. Overview</a></li>
<li class="chapter" data-level="" data-path="WE_14.html"><a href="WE_14.html#import-genotypes"><i class="fa fa-check"></i>2. Import Genotypes</a></li>
<li class="chapter" data-level="" data-path="WE_14.html"><a href="WE_14.html#pollen-pool-genetic-structure"><i class="fa fa-check"></i>3. Pollen pool genetic structure</a></li>
<li class="chapter" data-level="" data-path="WE_14.html"><a href="WE_14.html#paternity-analysis"><i class="fa fa-check"></i>4. Paternity analysis</a></li>
<li class="chapter" data-level="" data-path="WE_14.html"><a href="WE_14.html#linking-paternity-to-ecological-variables"><i class="fa fa-check"></i>5. Linking paternity to ecological variables</a></li>
<li class="chapter" data-level="" data-path="WE_14.html"><a href="WE_14.html#references-4"><i class="fa fa-check"></i>6. References</a></li>
</ul></li>
</ul></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Landscape Genetic Data Analysis with R</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="bonus_8a" class="section level2 hasAnchor" number="11.4">
<h2><span class="header-section-number">11.4</span> Bonus: Efficient R<a href="bonus_8a.html#bonus_8a" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Helene Wagner</p>
<div id="overview-of-bonus-material" class="section level4 unnumbered hasAnchor">
<h4>1. Overview of Bonus Material<a href="bonus_8a.html#overview-of-bonus-material" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<div id="a.-goals-8" class="section level5 unnumbered hasAnchor">
<h5>a. Goals<a href="bonus_8a.html#a.-goals-8" class="anchor-section" aria-label="Anchor link to header"></a></h5>
<p>This Bonus Material provides some introductory worked examples for:</p>
<ul>
<li>Navigating the file system</li>
<li>Benchmarking file import and export functions</li>
<li>Profiling an R script</li>
<li>Creating and executing a Bash R script</li>
<li>Parallelizing code</li>
</ul>
</div>
<div id="b.-data-set-7" class="section level5 unnumbered hasAnchor">
<h5>b. Data set<a href="bonus_8a.html#b.-data-set-7" class="anchor-section" aria-label="Anchor link to header"></a></h5>
<p>We will use the wolf SNP data (Schweizer et al., 2016) from the Week 11 worked example. The genetic data are individual-based, and are input as allele counts (i.e. 0/1/2) for each locus. We are using a randomly sampled subset of 10,000 single nucleotide polymorphism (SNP) markers from the full data set (which contains 42,587 SNPs).</p>
</div>
<div id="c.-required-r-packages-6" class="section level5 unnumbered hasAnchor">
<h5>c. Required R packages<a href="bonus_8a.html#c.-required-r-packages-6" class="anchor-section" aria-label="Anchor link to header"></a></h5>
<div class="sourceCode" id="cb1012"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1012-1"><a href="bonus_8a.html#cb1012-1" tabindex="-1"></a><span class="fu">library</span>(LandGenCourse)</span>
<span id="cb1012-2"><a href="bonus_8a.html#cb1012-2" tabindex="-1"></a><span class="co">#library(microbenchmark)</span></span>
<span id="cb1012-3"><a href="bonus_8a.html#cb1012-3" tabindex="-1"></a><span class="co">#library(profvis)</span></span>
<span id="cb1012-4"><a href="bonus_8a.html#cb1012-4" tabindex="-1"></a><span class="co">#library(here)</span></span>
<span id="cb1012-5"><a href="bonus_8a.html#cb1012-5" tabindex="-1"></a><span class="co">#library(readr)</span></span>
<span id="cb1012-6"><a href="bonus_8a.html#cb1012-6" tabindex="-1"></a><span class="co">#library(data.table)</span></span>
<span id="cb1012-7"><a href="bonus_8a.html#cb1012-7" tabindex="-1"></a><span class="fu">library</span>(feather)</span>
<span id="cb1012-8"><a href="bonus_8a.html#cb1012-8" tabindex="-1"></a><span class="co">#library(rio)</span></span>
<span id="cb1012-9"><a href="bonus_8a.html#cb1012-9" tabindex="-1"></a><span class="co">#library(devtools)</span></span>
<span id="cb1012-10"><a href="bonus_8a.html#cb1012-10" tabindex="-1"></a><span class="co">#library(parallel)</span></span>
<span id="cb1012-11"><a href="bonus_8a.html#cb1012-11" tabindex="-1"></a><span class="co">#library(doParallel)</span></span>
<span id="cb1012-12"><a href="bonus_8a.html#cb1012-12" tabindex="-1"></a><span class="co">#library(knitr)</span></span>
<span id="cb1012-13"><a href="bonus_8a.html#cb1012-13" tabindex="-1"></a><span class="co">#library(compiler)</span></span></code></pre></div>
</div>
</div>
<div id="navigating-the-file-system" class="section level4 unnumbered hasAnchor">
<h4>2. Navigating the file system<a href="bonus_8a.html#navigating-the-file-system" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>This Bonus material assumes that you are running R in RStudio within a ‘R project’. Therefore, the default workspace when you enter commands in the console should be your R project folder. A discussed in the video, Part 2, the default location when executing code from an R Notebook is the folder where the notebook is stored. If you downloaded this Bonus material from ‘LandGenCourse’, it should be stored in a folder ‘downloads’ in your project folder.</p>
<p>More about R projects: <a href="https://support.rstudio.com/hc/en-us/articles/200526207-Using-Projects" class="uri">https://support.rstudio.com/hc/en-us/articles/200526207-Using-Projects</a>.</p>
<div id="a.-where-am-i" class="section level5 unnumbered hasAnchor">
<h5>a. Where am I?<a href="bonus_8a.html#a.-where-am-i" class="anchor-section" aria-label="Anchor link to header"></a></h5>
<p>Let’s see where we are.</p>
<div class="sourceCode" id="cb1013"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1013-1"><a href="bonus_8a.html#cb1013-1" tabindex="-1"></a><span class="fu">getwd</span>()</span></code></pre></div>
<pre><code>## [1] &quot;/Users/helene/Desktop/R_Github_projects/LandGenCourse_book/vignettes&quot;</code></pre>
<div class="sourceCode" id="cb1015"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1015-1"><a href="bonus_8a.html#cb1015-1" tabindex="-1"></a>here<span class="sc">::</span><span class="fu">here</span>()</span></code></pre></div>
<pre><code>## [1] &quot;/Users/helene/Desktop/R_Github_projects/LandGenCourse_book&quot;</code></pre>
<div class="sourceCode" id="cb1017"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1017-1"><a href="bonus_8a.html#cb1017-1" tabindex="-1"></a><span class="fu">Sys.getenv</span>(<span class="st">&quot;HOME&quot;</span>)</span></code></pre></div>
<pre><code>## [1] &quot;/Users/helene&quot;</code></pre>
<div class="sourceCode" id="cb1019"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1019-1"><a href="bonus_8a.html#cb1019-1" tabindex="-1"></a><span class="fu">R.home</span>()</span></code></pre></div>
<pre><code>## [1] &quot;/Library/Frameworks/R.framework/Resources&quot;</code></pre>
<p><strong>Question:</strong> Copy-paste the commands in the chunk above directly into the Console and run them there. Do you get the same paths?</p>
<ul>
<li><code>getwd()</code>: If you execute the chunk, or knit the notebook, this will return the locatio of the R Notebook. If you copy-paste it into the console, it will return the project folder.</li>
<li><code>here::here</code>: This should return the project folder in both cases.</li>
<li><code>Sys.getenv("HOME")</code>: This should return your home directory on the local machine. Note: “HOME” must be in all capitals.</li>
<li><code>R.home</code>: This shows the location where R is installed on your computer.</li>
</ul>
</div>
<div id="b.-accessing-a-system-file" class="section level5 unnumbered hasAnchor">
<h5>b. Accessing a system file<a href="bonus_8a.html#b.-accessing-a-system-file" class="anchor-section" aria-label="Anchor link to header"></a></h5>
<p>The example data set is available in the ‘extdata’ folder of the ‘LandGenCoures’ package. This is how we import the data in the Week 11 Worked Example:</p>
<div class="sourceCode" id="cb1021"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1021-1"><a href="bonus_8a.html#cb1021-1" tabindex="-1"></a>gen <span class="ot">&lt;-</span> <span class="fu">read.csv</span>(<span class="fu">system.file</span>(<span class="st">&quot;extdata&quot;</span>, <span class="st">&quot;wolf_geno_samp_10000.csv&quot;</span>, </span>
<span id="cb1021-2"><a href="bonus_8a.html#cb1021-2" tabindex="-1"></a>                            <span class="at">package =</span> <span class="st">&quot;LandGenCourse&quot;</span>), <span class="at">row.names=</span><span class="dv">1</span>)</span>
<span id="cb1021-3"><a href="bonus_8a.html#cb1021-3" tabindex="-1"></a><span class="fu">dim</span>(gen)</span></code></pre></div>
<pre><code>## [1]    94 10000</code></pre>
<p>Let’s unpack this code. What does the function <code>system.file</code> do? Let’s compare it to <code>R.home</code>.</p>
<div class="sourceCode" id="cb1023"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1023-1"><a href="bonus_8a.html#cb1023-1" tabindex="-1"></a><span class="fu">R.home</span>()</span></code></pre></div>
<pre><code>## [1] &quot;/Library/Frameworks/R.framework/Resources&quot;</code></pre>
<div class="sourceCode" id="cb1025"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1025-1"><a href="bonus_8a.html#cb1025-1" tabindex="-1"></a><span class="fu">system.file</span>()</span></code></pre></div>
<pre><code>## [1] &quot;/Library/Frameworks/R.framework/Resources/library/base&quot;</code></pre>
<p>So, <code>system.file</code> goes to the R home and, without additional arguments, locates the folder ‘library’ with the subfolder ‘base’.</p>
<p>What happens when we add the arguments?</p>
<div class="sourceCode" id="cb1027"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1027-1"><a href="bonus_8a.html#cb1027-1" tabindex="-1"></a><span class="fu">system.file</span>(<span class="at">package =</span> <span class="st">&quot;LandGenCourse&quot;</span>)</span></code></pre></div>
<pre><code>## [1] &quot;/Library/Frameworks/R.framework/Versions/4.3-x86_64/Resources/library/LandGenCourse&quot;</code></pre>
<div class="sourceCode" id="cb1029"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1029-1"><a href="bonus_8a.html#cb1029-1" tabindex="-1"></a><span class="fu">system.file</span>(<span class="st">&quot;extdata&quot;</span>, <span class="st">&quot;wolf_geno_samp_10000.csv&quot;</span>, </span>
<span id="cb1029-2"><a href="bonus_8a.html#cb1029-2" tabindex="-1"></a>                            <span class="at">package =</span> <span class="st">&quot;LandGenCourse&quot;</span>)</span></code></pre></div>
<pre><code>## [1] &quot;/Library/Frameworks/R.framework/Versions/4.3-x86_64/Resources/library/LandGenCourse/extdata/wolf_geno_samp_10000.csv&quot;</code></pre>
<p>Compare the paths to those from the previous chunk!</p>
<ul>
<li>When we specify the package argument, <code>system.file</code> modifies the path to the location where the package is stored.</li>
<li>The remaining, unnamed arguments are interpreted as a path and added at the end: within package ‘LandGenCourse’, go to folder ‘extdata’ and locate file “wolf_geno_samp_10000.csv”.</li>
</ul>
<p>This is pretty cool! To be honest, I have no clue where such files are stored, and the absolute path would be different anyways on my Mac and on my Windows machine.</p>
</div>
<div id="c.-file-manipulation" class="section level5 unnumbered hasAnchor">
<h5>c. File manipulation<a href="bonus_8a.html#c.-file-manipulation" class="anchor-section" aria-label="Anchor link to header"></a></h5>
<p>We can use R’s file manipulation functions to do things with the file before even importing the data. Let’s check the file size:</p>
<div class="sourceCode" id="cb1031"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1031-1"><a href="bonus_8a.html#cb1031-1" tabindex="-1"></a>myFile <span class="ot">&lt;-</span> <span class="fu">system.file</span>(<span class="st">&quot;extdata&quot;</span>, <span class="st">&quot;wolf_geno_samp_10000.csv&quot;</span>, </span>
<span id="cb1031-2"><a href="bonus_8a.html#cb1031-2" tabindex="-1"></a>                            <span class="at">package =</span> <span class="st">&quot;LandGenCourse&quot;</span>)</span>
<span id="cb1031-3"><a href="bonus_8a.html#cb1031-3" tabindex="-1"></a><span class="fu">file.size</span>(myFile)</span></code></pre></div>
<pre><code>## [1] 2078748</code></pre>
<div class="sourceCode" id="cb1033"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1033-1"><a href="bonus_8a.html#cb1033-1" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">&quot;File size: &quot;</span>, <span class="fu">file.size</span>(myFile) <span class="sc">/</span> <span class="dv">10</span><span class="sc">^</span><span class="dv">6</span>, <span class="st">&quot; MB&quot;</span>)</span></code></pre></div>
<pre><code>## File size:  2.078748  MB</code></pre>
<p>The number returned is in byte, hence we divide by 1 million (10^6) to get megabyte (MB).</p>
<p>Other useful functions include (see help file: ‘?files’):</p>
<ul>
<li><code>file.create</code>: creates file, truncates name if name already exists</li>
<li><code>file.exist</code>: checks whether a file with this name exists at location</li>
<li><code>file.remove</code>: attempts to delete file</li>
<li><code>file.rename</code>: attempts to rename file</li>
<li><code>file.append</code>: attempts to append one file to another</li>
<li><code>file.copy</code>: copies file ‘from’ ‘to’ (set ‘overwrite=TRUE’ to allow overwriting an existing file)</li>
<li><code>download.file</code>: download a file from the internet</li>
<li><code>unzip</code>: unzip a zip archive (there is also a function <code>zip</code>, also <code>untar</code> and <code>tar</code> for tar archives)</li>
<li><code>dir</code>: list content of folder (=directory)</li>
<li><code>dir.exist</code>: check whether a folder with this name exists at location</li>
<li><code>dir.create</code>: creates a folder (does not work when knitting R Notebook!)</li>
</ul>
<p>What other files are available? The function <code>dir</code> lists all files and subfolders in a specific folder. The question then is which folder to specify?</p>
<p><strong>Question:</strong> What do you think the following commands will return?</p>
<ul>
<li>dir()</li>
<li>dir(here::here())</li>
<li>dir(system.file(package = “LandGenCourse”))</li>
<li>dir(system.file(“extdata”, package = “LandGenCourse”))</li>
<li>dir(myFile)</li>
</ul>
<p>Give it a try:</p>
<div class="sourceCode" id="cb1035"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1035-1"><a href="bonus_8a.html#cb1035-1" tabindex="-1"></a><span class="fu">dir</span>()</span></code></pre></div>
<pre><code>##  [1] &quot;radish_tutorial_fig1A.png&quot; &quot;radish_tutorial_fig1B.png&quot;
##  [3] &quot;WE12_Fig1.PNG&quot;             &quot;WE13_Fig1.png&quot;            
##  [5] &quot;WE9_Fig1.png&quot;              &quot;WE9_Fig2.png&quot;             
##  [7] &quot;Week0_BasicR.Rmd&quot;          &quot;Week0_Graphics.Rmd&quot;       
##  [9] &quot;Week1_vignette.Rmd&quot;        &quot;Week10_bonus_vignette.Rmd&quot;
## [11] &quot;Week10_vignette.Rmd&quot;       &quot;Week11_vignette.Rmd&quot;      
## [13] &quot;Week12_vignette.Rmd&quot;       &quot;Week13_vignette.Rmd&quot;      
## [15] &quot;Week14_vignette.Rmd&quot;       &quot;Week2_bonus_vignette.Rmd&quot; 
## [17] &quot;Week2_vignette.Rmd&quot;        &quot;Week3_vignette.Rmd&quot;       
## [19] &quot;Week4_vignette.Rmd&quot;        &quot;Week5_vignette.Rmd&quot;       
## [21] &quot;Week6_vignette.Rmd&quot;        &quot;Week7_bonus_vignette.Rmd&quot; 
## [23] &quot;Week7_vignette.Rmd&quot;        &quot;Week8_bonus_vignette.Rmd&quot; 
## [25] &quot;Week8_vignette.Rmd&quot;        &quot;Week9_vignette.Rmd&quot;</code></pre>
<div class="sourceCode" id="cb1037"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1037-1"><a href="bonus_8a.html#cb1037-1" tabindex="-1"></a><span class="fu">dir</span>(here<span class="sc">::</span><span class="fu">here</span>())</span></code></pre></div>
<pre><code>##  [1] &quot;_book&quot;                    &quot;_bookdown_files&quot;         
##  [3] &quot;_bookdown.yml&quot;            &quot;_output.yml&quot;             
##  [5] &quot;00-GitHub_files&quot;          &quot;00-GitHub.md&quot;            
##  [7] &quot;00-GitHub.Rmd&quot;            &quot;00-ReviewR_files&quot;        
##  [9] &quot;00-ReviewR.md&quot;            &quot;00-ReviewR.Rmd&quot;          
## [11] &quot;01-Week01_files&quot;          &quot;01-Week01.md&quot;            
## [13] &quot;01-Week01.Rmd&quot;            &quot;02-Week02_files&quot;         
## [15] &quot;02-Week02.md&quot;             &quot;02-Week02.Rmd&quot;           
## [17] &quot;03-Week03_files&quot;          &quot;03-Week03.md&quot;            
## [19] &quot;03-Week03.Rmd&quot;            &quot;04-Week04_files&quot;         
## [21] &quot;04-Week04.md&quot;             &quot;04-Week04.Rmd&quot;           
## [23] &quot;05-Week05_files&quot;          &quot;05-Week05.md&quot;            
## [25] &quot;05-Week05.Rmd&quot;            &quot;06-Week06_files&quot;         
## [27] &quot;06-Week06.md&quot;             &quot;06-Week06.Rmd&quot;           
## [29] &quot;07-Week07_files&quot;          &quot;07-Week07.md&quot;            
## [31] &quot;07-Week07.Rmd&quot;            &quot;08-Week08_files&quot;         
## [33] &quot;08-Week08.Rmd&quot;            &quot;09-Week09_files&quot;         
## [35] &quot;09-Week09.Rmd&quot;            &quot;10-Week10_files&quot;         
## [37] &quot;10-Week10.Rmd&quot;            &quot;11-Week11_files&quot;         
## [39] &quot;11-Week11.Rmd&quot;            &quot;12-Week12_files&quot;         
## [41] &quot;12-Week12.Rmd&quot;            &quot;13-Week13_files&quot;         
## [43] &quot;13-Week13.Rmd&quot;            &quot;14-Week14_files&quot;         
## [45] &quot;14-Week14.Rmd&quot;            &quot;book.bib&quot;                
## [47] &quot;ChapterDepot.R&quot;           &quot;data&quot;                    
## [49] &quot;downloads&quot;                &quot;header.html&quot;             
## [51] &quot;index.md&quot;                 &quot;index.Rmd&quot;               
## [53] &quot;js&quot;                       &quot;LandGenCourse_book.rds&quot;  
## [55] &quot;LandGenCourse_book.Rproj&quot; &quot;LICENSE&quot;                 
## [57] &quot;ListOfPackages.nb.html&quot;   &quot;ListOfPackages.Rmd&quot;      
## [59] &quot;my_plot.png&quot;              &quot;output&quot;                  
## [61] &quot;README.md&quot;                &quot;render1b7c27bb24ce.rds&quot;  
## [63] &quot;render1e541582584a.rds&quot;   &quot;render23ec66f4ddd.rds&quot;   
## [65] &quot;render2fbc1ad7f46.rds&quot;    &quot;render2fd81ad75275.rds&quot;  
## [67] &quot;render33ed6170c2e4.rds&quot;   &quot;render363c1997640c.rds&quot;  
## [69] &quot;render3894c561f3.rds&quot;     &quot;render3b305196d38.rds&quot;   
## [71] &quot;render40b614e5490e.rds&quot;   &quot;render65e32e6edd81.rds&quot;  
## [73] &quot;render65ea2d72427d.rds&quot;   &quot;rendera1ac36509ec1.rds&quot;  
## [75] &quot;renderb74a25ab009f.rds&quot;   &quot;renderf9011b67ac.rds&quot;    
## [77] &quot;rsconnect&quot;                &quot;style.css&quot;               
## [79] &quot;toc.css&quot;                  &quot;tutorials&quot;               
## [81] &quot;vignettes&quot;                &quot;Week1_vignette.R&quot;        
## [83] &quot;Week2_bonus_vignette.R&quot;   &quot;Week8_bonus_vignette.R&quot;  
## [85] &quot;Workflow.nb.html&quot;         &quot;Workflow.Rmd&quot;</code></pre>
<div class="sourceCode" id="cb1039"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1039-1"><a href="bonus_8a.html#cb1039-1" tabindex="-1"></a><span class="fu">dir</span>(<span class="fu">system.file</span>(<span class="at">package =</span> <span class="st">&quot;LandGenCourse&quot;</span>))</span></code></pre></div>
<pre><code>##  [1] &quot;data&quot;        &quot;DESCRIPTION&quot; &quot;doc&quot;         &quot;extdata&quot;     &quot;help&quot;       
##  [6] &quot;html&quot;        &quot;INDEX&quot;       &quot;Meta&quot;        &quot;NAMESPACE&quot;   &quot;R&quot;          
## [11] &quot;rstudio&quot;</code></pre>
<div class="sourceCode" id="cb1041"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1041-1"><a href="bonus_8a.html#cb1041-1" tabindex="-1"></a><span class="fu">dir</span>(<span class="fu">system.file</span>(<span class="st">&quot;extdata&quot;</span>, <span class="at">package =</span> <span class="st">&quot;LandGenCourse&quot;</span>))</span></code></pre></div>
<pre><code>##  [1] &quot;BashExample.sh&quot;                   &quot;Colortable_LULC.csv&quot;             
##  [3] &quot;CSF_network.csv&quot;                  &quot;dataNm1.str&quot;                     
##  [5] &quot;dataNm10.txt&quot;                     &quot;dataNm2.txt&quot;                     
##  [7] &quot;dModels.rds&quot;                      &quot;EnvironmentalData_8pred.csv&quot;     
##  [9] &quot;ExcelTable.png&quot;                   &quot;Frogs_diversity_allpops.csv&quot;     
## [11] &quot;Index_of_functions_formatted.pdf&quot; &quot;myNotebook.Rmd&quot;                  
## [13] &quot;panel.cor.r&quot;                      &quot;Patch_XY_Dianthus.csv&quot;           
## [15] &quot;pulsatilla_genotypes.csv&quot;         &quot;pulsatilla_momVariables.csv&quot;     
## [17] &quot;pulsatilla_population.csv&quot;        &quot;ralu_coords_allpops.csv&quot;         
## [19] &quot;ralu_dc.csv&quot;                      &quot;RALU_Dps.csv&quot;                    
## [21] &quot;ralu_loci_allpops.csv&quot;            &quot;RALU_Site.csv&quot;                   
## [23] &quot;ralu.loci.csv&quot;                    &quot;ralu.rasters.rds&quot;                
## [25] &quot;RCommands.docx&quot;                   &quot;supplemental_R_functions.R&quot;      
## [27] &quot;Testfile.nb.html&quot;                 &quot;Testfile.Rmd&quot;                    
## [29] &quot;WE12_Fig1.PNG&quot;                    &quot;WE13_Fig1.png&quot;                   
## [31] &quot;WE9_Fig1.png&quot;                     &quot;WE9_Fig2.png&quot;                    
## [33] &quot;Wetlands.csv&quot;                     &quot;wolf_geno_samp_10000.csv&quot;        
## [35] &quot;WWP_environmental_data.txt&quot;       &quot;WWP_phenotype_data.txt&quot;          
## [37] &quot;WWP_SNP_genotypes.txt&quot;</code></pre>
<div class="sourceCode" id="cb1043"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1043-1"><a href="bonus_8a.html#cb1043-1" tabindex="-1"></a><span class="fu">dir</span>(myFile)</span></code></pre></div>
<pre><code>## character(0)</code></pre>
<ul>
<li><code>dir()</code>: the content of the last folder to which you navigated.</li>
<li><code>dir(here::here())</code>: the content of your project folder.</li>
<li><code>dir(system.file(package = "LandGenCourse"))</code>: the content of the package.</li>
<li><code>dir(system.file("extdata", package = "LandGenCourse"))</code>: the content of the folder ‘extdata’ in the package.</li>
<li><code>dir(myFile)</code>: nothing (this is a file, not a folder).</li>
</ul>
</div>
</div>
<div id="benchmarking-file-import-and-export-options" class="section level4 unnumbered hasAnchor">
<h4>3. Benchmarking file import and export options<a href="bonus_8a.html#benchmarking-file-import-and-export-options" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>See also Chapter 5 in “Efficient R Programming”: <a href="https://csgillespie.github.io/efficientR/input-output.html" class="uri">https://csgillespie.github.io/efficientR/input-output.html</a></p>
<div id="a.-benchmark-methods-for-importing-csv-files" class="section level5 unnumbered hasAnchor">
<h5>a. Benchmark methods for importing csv files<a href="bonus_8a.html#a.-benchmark-methods-for-importing-csv-files" class="anchor-section" aria-label="Anchor link to header"></a></h5>
<p>The file ‘myFile’ has 2MB and thus a reasonable size to compare the speed of different import and export functions.</p>
<p>Let’s benchmark the function <code>read.csv</code> used in Week 11. We use the function <code>microbenchmark</code> from the package <code>microbenchmark</code> to compare the speed of four different functions that can import a ‘csv’ file.</p>
<p>Note: Here we execute each function only once to save time, typically you would set <code>times = 10</code> or so. Also, <code>read_csv</code> will print a warning about a missing column name. This is because the first column here contains the row names and does not have a column name. We’ll ignore this here, as we can use the first column as an example to compare how character data are being imported.</p>
<div class="sourceCode" id="cb1045"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1045-1"><a href="bonus_8a.html#cb1045-1" tabindex="-1"></a>x <span class="ot">=</span> myFile</span>
<span id="cb1045-2"><a href="bonus_8a.html#cb1045-2" tabindex="-1"></a>microbenchmark<span class="sc">::</span><span class="fu">microbenchmark</span>(<span class="at">times =</span> <span class="dv">1</span>, <span class="at">unit =</span> <span class="st">&quot;ms&quot;</span>, </span>
<span id="cb1045-3"><a href="bonus_8a.html#cb1045-3" tabindex="-1"></a>          <span class="fu">read.csv</span>(x), readr<span class="sc">::</span><span class="fu">read_csv</span>(x, <span class="at">show_col_types =</span> <span class="cn">FALSE</span>), data.table<span class="sc">::</span><span class="fu">fread</span>(x),</span>
<span id="cb1045-4"><a href="bonus_8a.html#cb1045-4" tabindex="-1"></a>          rio<span class="sc">::</span><span class="fu">import</span>(x))</span></code></pre></div>
<pre><code>## Unit: milliseconds
##                                        expr       min        lq      mean
##                                 read.csv(x) 1834.0490 1834.0490 1834.0490
##  readr::read_csv(x, show_col_types = FALSE) 2818.0240 2818.0240 2818.0240
##                        data.table::fread(x)  150.3722  150.3722  150.3722
##                              rio::import(x)  100.8732  100.8732  100.8732
##     median        uq       max neval
##  1834.0490 1834.0490 1834.0490     1
##  2818.0240 2818.0240 2818.0240     1
##   150.3722  150.3722  150.3722     1
##   100.8732  100.8732  100.8732     1</code></pre>
<p>Would it be faster if we first loaded the packages so that we could call the functions directly?</p>
<p>Note: you can list several independent commands on the same line by separating the with a semi-colon ‘;’. Also, the chunk setting <code>message=FALSE</code> here suppresses the warning message from <code>read_csv</code>.</p>
<div class="sourceCode" id="cb1047"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1047-1"><a href="bonus_8a.html#cb1047-1" tabindex="-1"></a><span class="fu">library</span>(readr); <span class="fu">library</span>(data.table); <span class="fu">library</span>(rio); <span class="fu">library</span>(microbenchmark)</span>
<span id="cb1047-2"><a href="bonus_8a.html#cb1047-2" tabindex="-1"></a></span>
<span id="cb1047-3"><a href="bonus_8a.html#cb1047-3" tabindex="-1"></a><span class="fu">microbenchmark</span>(<span class="at">times =</span> <span class="dv">1</span>, <span class="at">unit =</span> <span class="st">&quot;ms&quot;</span>, </span>
<span id="cb1047-4"><a href="bonus_8a.html#cb1047-4" tabindex="-1"></a>          <span class="fu">read.csv</span>(x), <span class="fu">read_csv</span>(x, <span class="at">show_col_types =</span> <span class="cn">FALSE</span>), <span class="fu">fread</span>(x), <span class="fu">import</span>(x))</span></code></pre></div>
<pre><code>## Unit: milliseconds
##                                 expr        min         lq       mean
##                          read.csv(x) 1803.38453 1803.38453 1803.38453
##  read_csv(x, show_col_types = FALSE) 2618.14160 2618.14160 2618.14160
##                             fread(x)   86.33835   86.33835   86.33835
##                            import(x)   85.91416   85.91416   85.91416
##      median         uq        max neval
##  1803.38453 1803.38453 1803.38453     1
##  2618.14160 2618.14160 2618.14160     1
##    86.33835   86.33835   86.33835     1
##    85.91416   85.91416   85.91416     1</code></pre>
<p>Yes, the import was faster when the packages were already loaded.</p>
<p>Overall <code>fread</code> and <code>import</code> were in the order of 50 times faster than <code>read.csv</code> and <code>read_csv</code>! The two had practically the same speed, which is little surprising: for ‘csv’ files, <code>import</code> uses the function <code>fread</code>.</p>
<p>The beauty of <code>import</code> is that it can handle a wide range of file types (and the list keeps growing): csv, xls, xlsx, html, xml, json, feather, R, RData, rda, rds, psv, tsv, sas7bdat, xpt, sav, dta, xpt, por, rec, mtp, syd, dbf, arff, dif, fwf, csv.gz, CSVY, fst, mat, ods, yml, as well as Fortan files and clipboard imports (Mac and Windows). It recognizes the file type from the extension and uses an appropriate import function.</p>
<p>Note: there is also a fuction <code>Import</code> in the <code>car</code> package that similarly aims to provide an easy way to import various file formats. However, <code>car::Import</code> can be very slow (slower than ‘read.csv’), whereas <code>rio::import</code> is fast.</p>
</div>
<div id="b.-check-handling-of-character-data" class="section level5 unnumbered hasAnchor">
<h5>b. Check handling of character data<a href="bonus_8a.html#b.-check-handling-of-character-data" class="anchor-section" aria-label="Anchor link to header"></a></h5>
<p>The functions differ not only in their speed but also in how they handle text data (character or factor?), missing values etc.</p>
<p>The first column in ‘myFile’ is an ID variable that should be used as row names. Let’s compare what the four methods did with this. The following code determines, for each import method, the class of the first column (IDs), and the class (or classes) of the resulting object.</p>
<p>Note: here we use double square brackets to subset the first column. Strictly speaking, we interpret ‘gen’ as a list of vectors. With a data.frame, we could also access the first column by gen[,1]. However, this would not return what we want for ‘tbl’ of ‘data.table’ objects. Always double check.</p>
<div class="sourceCode" id="cb1049"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1049-1"><a href="bonus_8a.html#cb1049-1" tabindex="-1"></a>gen <span class="ot">&lt;-</span> <span class="fu">read.csv</span>(x); <span class="fu">c</span>(<span class="fu">class</span>(gen[[<span class="dv">1</span>]]), <span class="fu">class</span>(gen))</span></code></pre></div>
<pre><code>## [1] &quot;character&quot;  &quot;data.frame&quot;</code></pre>
<div class="sourceCode" id="cb1051"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1051-1"><a href="bonus_8a.html#cb1051-1" tabindex="-1"></a>gen <span class="ot">&lt;-</span> <span class="fu">read_csv</span>(x, <span class="at">show_col_types =</span> <span class="cn">FALSE</span>); <span class="fu">c</span>(<span class="fu">class</span>(gen[[<span class="dv">1</span>]]), <span class="fu">class</span>(gen))</span></code></pre></div>
<pre><code>## [1] &quot;character&quot;   &quot;spec_tbl_df&quot; &quot;tbl_df&quot;      &quot;tbl&quot;         &quot;data.frame&quot;</code></pre>
<div class="sourceCode" id="cb1053"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1053-1"><a href="bonus_8a.html#cb1053-1" tabindex="-1"></a>gen <span class="ot">&lt;-</span> <span class="fu">fread</span>(x); <span class="fu">c</span>(<span class="fu">class</span>(gen[[<span class="dv">1</span>]]), <span class="fu">class</span>(gen))</span></code></pre></div>
<pre><code>## [1] &quot;character&quot;  &quot;data.table&quot; &quot;data.frame&quot;</code></pre>
<div class="sourceCode" id="cb1055"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1055-1"><a href="bonus_8a.html#cb1055-1" tabindex="-1"></a>gen <span class="ot">&lt;-</span> <span class="fu">import</span>(x); <span class="fu">c</span>(<span class="fu">class</span>(gen[[<span class="dv">1</span>]]), <span class="fu">class</span>(gen))</span></code></pre></div>
<pre><code>## [1] &quot;character&quot;  &quot;data.frame&quot;</code></pre>
<ul>
<li>The function <code>read.csv</code> interprets any text as ‘factor’, the other functions use ‘character’ as default. Always double check!<br />
</li>
<li>All of these functions have optional arguments for specifying how each column how it should be interpreted (see help files).</li>
<li>With the functions <code>fread</code> and <code>import</code>, you can set the argument ‘stringsAsFactors = TRUE’ to import all text data as factors.</li>
</ul>
</div>
<div id="c.-binary-files" class="section level5 unnumbered hasAnchor">
<h5>c. Binary files<a href="bonus_8a.html#c.-binary-files" class="anchor-section" aria-label="Anchor link to header"></a></h5>
<p>Binary files are not readable by users (or other software) but provide an efficient way of storing data. Let’s compare file size and input/output speed between text files (csv) and different types of binary files (RData, rds, feather). We’ll also export the ‘csv’ file so that we have it in the same location.</p>
<p>First we make sure an <code>output</code> folder exists in the R project:</p>
<div class="sourceCode" id="cb1057"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1057-1"><a href="bonus_8a.html#cb1057-1" tabindex="-1"></a><span class="cf">if</span>(<span class="sc">!</span><span class="fu">dir.exists</span>(<span class="fu">paste0</span>(here<span class="sc">::</span><span class="fu">here</span>(),<span class="st">&quot;/output&quot;</span>))) <span class="fu">dir.create</span>(<span class="fu">paste0</span>(here<span class="sc">::</span><span class="fu">here</span>(),<span class="st">&quot;/output&quot;</span>))</span></code></pre></div>
<div class="sourceCode" id="cb1058"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1058-1"><a href="bonus_8a.html#cb1058-1" tabindex="-1"></a>gen <span class="ot">&lt;-</span> <span class="fu">import</span>(myFile)</span>
<span id="cb1058-2"><a href="bonus_8a.html#cb1058-2" tabindex="-1"></a></span>
<span id="cb1058-3"><a href="bonus_8a.html#cb1058-3" tabindex="-1"></a><span class="fu">export</span>(gen, <span class="fu">file.path</span>(here<span class="sc">::</span><span class="fu">here</span>(), <span class="st">&quot;output&quot;</span>, <span class="st">&quot;gen.csv&quot;</span>))</span>
<span id="cb1058-4"><a href="bonus_8a.html#cb1058-4" tabindex="-1"></a><span class="fu">save</span>(gen, <span class="at">file=</span><span class="fu">file.path</span>(here<span class="sc">::</span><span class="fu">here</span>(), <span class="st">&quot;output&quot;</span>, <span class="st">&quot;gen.RData&quot;</span>))</span>
<span id="cb1058-5"><a href="bonus_8a.html#cb1058-5" tabindex="-1"></a><span class="fu">saveRDS</span>(gen, <span class="at">file=</span><span class="fu">file.path</span>(here<span class="sc">::</span><span class="fu">here</span>(), <span class="st">&quot;output&quot;</span>, <span class="st">&quot;gen.rds&quot;</span>))</span>
<span id="cb1058-6"><a href="bonus_8a.html#cb1058-6" tabindex="-1"></a><span class="fu">export</span>(gen, <span class="at">file=</span><span class="fu">file.path</span>(here<span class="sc">::</span><span class="fu">here</span>(), <span class="st">&quot;output&quot;</span>, <span class="st">&quot;gen.feather&quot;</span>))</span>
<span id="cb1058-7"><a href="bonus_8a.html#cb1058-7" tabindex="-1"></a></span>
<span id="cb1058-8"><a href="bonus_8a.html#cb1058-8" tabindex="-1"></a><span class="fu">c</span>(<span class="at">csv=</span><span class="fu">file.size</span>(<span class="fu">file.path</span>(here<span class="sc">::</span><span class="fu">here</span>(), <span class="st">&quot;output&quot;</span>, <span class="st">&quot;gen.csv&quot;</span>)),</span>
<span id="cb1058-9"><a href="bonus_8a.html#cb1058-9" tabindex="-1"></a>  <span class="at">RData=</span><span class="fu">file.size</span>(<span class="fu">file.path</span>(here<span class="sc">::</span><span class="fu">here</span>(), <span class="st">&quot;output&quot;</span>, <span class="st">&quot;gen.RData&quot;</span>)),</span>
<span id="cb1058-10"><a href="bonus_8a.html#cb1058-10" tabindex="-1"></a>  <span class="at">rds=</span><span class="fu">file.size</span>(<span class="fu">file.path</span>(here<span class="sc">::</span><span class="fu">here</span>(), <span class="st">&quot;output&quot;</span>, <span class="st">&quot;gen.rds&quot;</span>)),</span>
<span id="cb1058-11"><a href="bonus_8a.html#cb1058-11" tabindex="-1"></a>  <span class="at">feather=</span><span class="fu">file.size</span>(<span class="fu">file.path</span>(here<span class="sc">::</span><span class="fu">here</span>(), <span class="st">&quot;output&quot;</span>, <span class="st">&quot;gen.feather&quot;</span>)))<span class="sc">/</span><span class="dv">10</span><span class="sc">^</span><span class="dv">6</span></span></code></pre></div>
<pre><code>##      csv    RData      rds  feather 
## 2.002491 0.351648 0.351620 3.716274</code></pre>
<ul>
<li>The ‘csv’ file is 2 MB (first row).</li>
<li>The R binary files ‘RData’ and ‘rds’ are much smaller!</li>
<li>In contrast, the ‘feather’ file (last row) is twice as large here than the ‘csv’ file, and more than 10 times larger than ‘rds’!</li>
</ul>
<p>Let’s benchmark the import again. We can use the function <code>import</code> for all of them. This is so fast that we can actually do it 10 times.</p>
<div class="sourceCode" id="cb1060"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1060-1"><a href="bonus_8a.html#cb1060-1" tabindex="-1"></a><span class="fu">microbenchmark</span>(<span class="at">times =</span> <span class="dv">10</span>, <span class="at">unit =</span> <span class="st">&quot;ms&quot;</span>, </span>
<span id="cb1060-2"><a href="bonus_8a.html#cb1060-2" tabindex="-1"></a>          <span class="at">csv=</span> <span class="fu">import</span>(<span class="fu">file.path</span>(here<span class="sc">::</span><span class="fu">here</span>(), <span class="st">&quot;output&quot;</span>, <span class="st">&quot;gen.csv&quot;</span>)),</span>
<span id="cb1060-3"><a href="bonus_8a.html#cb1060-3" tabindex="-1"></a>          <span class="at">RData=</span><span class="fu">import</span>(<span class="fu">file.path</span>(here<span class="sc">::</span><span class="fu">here</span>(), <span class="st">&quot;output&quot;</span>, <span class="st">&quot;gen.RData&quot;</span>)),</span>
<span id="cb1060-4"><a href="bonus_8a.html#cb1060-4" tabindex="-1"></a>          <span class="at">rds=</span><span class="fu">import</span>(<span class="fu">file.path</span>(here<span class="sc">::</span><span class="fu">here</span>(), <span class="st">&quot;output&quot;</span>, <span class="st">&quot;gen.rds&quot;</span>)),</span>
<span id="cb1060-5"><a href="bonus_8a.html#cb1060-5" tabindex="-1"></a>          <span class="at">feather=</span><span class="fu">import</span>(<span class="fu">file.path</span>(here<span class="sc">::</span><span class="fu">here</span>(), <span class="st">&quot;output&quot;</span>, <span class="st">&quot;gen.feather&quot;</span>)))</span></code></pre></div>
<pre><code>## Unit: milliseconds
##     expr        min         lq       mean     median         uq        max
##      csv   73.04911   74.73093   75.24703   75.03336   76.31467   77.46891
##    RData   20.85366   20.98809   21.65532   21.50691   22.46317   22.61786
##      rds   20.87919   21.36372   21.94246   22.05892   22.56156   22.77891
##  feather 5826.72931 5863.33777 5947.75843 5916.06415 5988.17160 6149.87139
##  neval cld
##     10  a 
##     10  a 
##     10  a 
##     10   b</code></pre>
<p>Look at the column ‘mean’. Importing any of the binary files was at least twice as fast as importing the ‘csv’ file with the underlying function <code>fread</code> (which was already 50 times faster than <code>read.csv</code>).</p>
<p>Here’s my recommendation for saving R objects/data efficiently:</p>
<ul>
<li>If object is not in tabular form: rds (can store any R object)</li>
<li>If storage space is most important: rds</li>
<li>If portability with Python is important: feather</li>
<li>If being able to read text file is important: csv</li>
</ul>
<p>Note: the developer of <code>feather</code> does not recommend using it for long-term data storage since its stability with future updates to R or Python can’t be guaranteed: <a href="https://github.com/wesm/feather/issues/183" class="uri">https://github.com/wesm/feather/issues/183</a></p>
<p>Why ‘rds’ and not ‘RData’? In practice, the main advantage of ‘rds’ is convenience when importing data.</p>
<ul>
<li>with ‘readRDS’, you can directly assign the imported data to an object, and thus choose the object name.</li>
<li>with ‘load’, you have to do this in two steps. When using ‘load’, the loaded object will inherit the name from the object that was stored.</li>
</ul>
<div class="sourceCode" id="cb1062"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1062-1"><a href="bonus_8a.html#cb1062-1" tabindex="-1"></a><span class="co"># Let&#39;s delete any copy of &#39;gen&#39; from the workspace:</span></span>
<span id="cb1062-2"><a href="bonus_8a.html#cb1062-2" tabindex="-1"></a><span class="fu">rm</span>(gen)</span>
<span id="cb1062-3"><a href="bonus_8a.html#cb1062-3" tabindex="-1"></a></span>
<span id="cb1062-4"><a href="bonus_8a.html#cb1062-4" tabindex="-1"></a><span class="co"># Create object &#39;myData&#39; in a single step from &#39;rds&#39; file:</span></span>
<span id="cb1062-5"><a href="bonus_8a.html#cb1062-5" tabindex="-1"></a>myData <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="fu">file.path</span>(here<span class="sc">::</span><span class="fu">here</span>(), <span class="st">&quot;output&quot;</span>, <span class="st">&quot;gen.rds&quot;</span>))</span>
<span id="cb1062-6"><a href="bonus_8a.html#cb1062-6" tabindex="-1"></a></span>
<span id="cb1062-7"><a href="bonus_8a.html#cb1062-7" tabindex="-1"></a><span class="co"># Two steps when importing &#39;RData&#39;: first, load the stored object:</span></span>
<span id="cb1062-8"><a href="bonus_8a.html#cb1062-8" tabindex="-1"></a><span class="fu">load</span>(<span class="fu">file.path</span>(here<span class="sc">::</span><span class="fu">here</span>(), <span class="st">&quot;output&quot;</span>, <span class="st">&quot;gen.RData&quot;</span>)) </span>
<span id="cb1062-9"><a href="bonus_8a.html#cb1062-9" tabindex="-1"></a><span class="co"># then assign to the new object &#39;myData&#39;:</span></span>
<span id="cb1062-10"><a href="bonus_8a.html#cb1062-10" tabindex="-1"></a>myData <span class="ot">&lt;-</span> gen</span></code></pre></div>
<p>Note that when you use <code>load</code>, the object name is NOT taken from the file name! This means that you may not know what object you are loading, if the object and file names are different.</p>
<p>Let’s test this. Here we save the object ‘gen’ in file ‘gen2.RData’, then load it.</p>
<div class="sourceCode" id="cb1063"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1063-1"><a href="bonus_8a.html#cb1063-1" tabindex="-1"></a><span class="co"># Export &#39;gen&#39; to a file with a different name &#39;gen2.RData&#39;:</span></span>
<span id="cb1063-2"><a href="bonus_8a.html#cb1063-2" tabindex="-1"></a><span class="fu">save</span>(gen, <span class="at">file=</span><span class="fu">file.path</span>(here<span class="sc">::</span><span class="fu">here</span>(), <span class="st">&quot;output&quot;</span>, <span class="st">&quot;gen2.RData&quot;</span>))</span>
<span id="cb1063-3"><a href="bonus_8a.html#cb1063-3" tabindex="-1"></a><span class="fu">rm</span>(gen)</span>
<span id="cb1063-4"><a href="bonus_8a.html#cb1063-4" tabindex="-1"></a></span>
<span id="cb1063-5"><a href="bonus_8a.html#cb1063-5" tabindex="-1"></a><span class="co"># Load:</span></span>
<span id="cb1063-6"><a href="bonus_8a.html#cb1063-6" tabindex="-1"></a><span class="fu">load</span>(<span class="fu">file.path</span>(here<span class="sc">::</span><span class="fu">here</span>(), <span class="st">&quot;output&quot;</span>, <span class="st">&quot;gen2.RData&quot;</span>)) </span>
<span id="cb1063-7"><a href="bonus_8a.html#cb1063-7" tabindex="-1"></a></span>
<span id="cb1063-8"><a href="bonus_8a.html#cb1063-8" tabindex="-1"></a><span class="co"># What is the name of the loaded object?</span></span>
<span id="cb1063-9"><a href="bonus_8a.html#cb1063-9" tabindex="-1"></a><span class="fu">exists</span>(<span class="st">&quot;gen&quot;</span>)</span></code></pre></div>
<pre><code>## [1] TRUE</code></pre>
<div class="sourceCode" id="cb1065"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1065-1"><a href="bonus_8a.html#cb1065-1" tabindex="-1"></a><span class="fu">exists</span>(<span class="st">&quot;gen2&quot;</span>)</span></code></pre></div>
<pre><code>## [1] FALSE</code></pre>
<p>We see that an object ‘gen’ exists (TRUE), but an object ‘gen2’ does not exist (FALSE). The name of the loaded object is thus ‘gen’.</p>
</div>
<div id="d.-should-you-save-your-r-workspace" class="section level5 unnumbered hasAnchor">
<h5>d. Should you save your R workspace?<a href="bonus_8a.html#d.-should-you-save-your-r-workspace" class="anchor-section" aria-label="Anchor link to header"></a></h5>
<p>When you close RStudio, you may be asked whether you want to save your workspace. What happens when you do this, and should you do so?</p>
<ul>
<li>When you save your workspace, all objects from the workspace are saved in one binary file ‘.RData’ (unless you provide a name, like ‘myWorkspace.Rdata’).</li>
<li>This may result in a large file!</li>
<li>There are other downsides: you may accidentally overwrite an object. And your code will not be portable because it depends on a copy of the workspace.</li>
<li>The general recommendation is to NOT save your workspace, but save your dataset and your R scripts (or Notebooks). This means that you can always recreate all the objects needed.</li>
</ul>
<p>Also, in the vein of reproducible research, do not save multiple copies of your data set. Instead:</p>
<ul>
<li>Save the raw data (with the original file name and extension, e.g. if you downloaded it from the internet - this will help identify the source)</li>
<li>Save all the data manipulation steps (such as checking for errors, excluding rows, recoding variables, etc.) in an R Notebook and document them (what you are doing, why and when).</li>
<li>Save the ‘clean’ dataset (result from data manipulation), preferably as a binary file (especially if it is large). Keep only one copy (you can always recreate it).</li>
<li>Save your data analysis in another R Notebook (or multiple) that starts with importing the ‘clean’ dataset. If your code runs quickly, this is sufficient to recreate your results.</li>
<li>If your code takes a lot of computation time, you may want to export the results (see above).</li>
<li>Backup your data and scripts (R Notebooks)!</li>
<li>Use version control for your script (R Notebooks)! See Chapter 0 video ‘Version Control 101’. The simplest is to include version numbers in the file name: ‘myNotebook_v7.Rmd’. A better way is to use e.g. GitHub.</li>
</ul>
</div>
<div id="e.-compile-your-functions" class="section level5 unnumbered hasAnchor">
<h5>e. Compile your functions<a href="bonus_8a.html#e.-compile-your-functions" class="anchor-section" aria-label="Anchor link to header"></a></h5>
<p>A recommended way of keeping your code tidy is to write functions for anything you will do more than once.</p>
<ul>
<li>Collect your functions in a separate file (R script), e.g. <code>myFunctions.R</code>.</li>
<li>At the beginning of your R Notebook, source the R script with the functions with <code>source("myFunctions.R")</code>.</li>
<li>This will make your code much shorter and thus easier to read.</li>
<li>If you need to change some code inside your function, you only have to change it in one place, hence there is less risk of mistakes.</li>
<li>Use some kind of version control for your functions file so that you can go back to an older version, and you always know which is the current version.</li>
</ul>
<p>To further speed up your code, you can compile your function with ‘cmpfun’:</p>
<div class="sourceCode" id="cb1067"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1067-1"><a href="bonus_8a.html#cb1067-1" tabindex="-1"></a>myFunction <span class="ot">&lt;-</span> <span class="cf">function</span>() {</span>
<span id="cb1067-2"><a href="bonus_8a.html#cb1067-2" tabindex="-1"></a>    <span class="fu">sum</span>(<span class="fu">rnorm</span>(<span class="dv">1000</span>))<span class="sc">/</span><span class="dv">1000</span></span>
<span id="cb1067-3"><a href="bonus_8a.html#cb1067-3" tabindex="-1"></a>}</span>
<span id="cb1067-4"><a href="bonus_8a.html#cb1067-4" tabindex="-1"></a>myFunction.cmp <span class="ot">&lt;-</span> compiler<span class="sc">::</span><span class="fu">cmpfun</span>(myFunction)</span>
<span id="cb1067-5"><a href="bonus_8a.html#cb1067-5" tabindex="-1"></a></span>
<span id="cb1067-6"><a href="bonus_8a.html#cb1067-6" tabindex="-1"></a>microbenchmark<span class="sc">::</span><span class="fu">microbenchmark</span>(<span class="fu">myFunction</span>(), <span class="fu">myFunction.cmp</span>())</span></code></pre></div>
<pre><code>## Unit: microseconds
##              expr    min      lq     mean  median     uq      max neval cld
##      myFunction() 46.169 46.6500 61.57885 46.9875 47.449 1459.562   100   a
##  myFunction.cmp() 46.071 46.6975 48.04005 47.0295 47.386  104.744   100   a</code></pre>
<p><strong>Question</strong>: Which of the following times are most different between the uncompiled and the compiled versions of this simple function?</p>
<ul>
<li><strong>min</strong>: minimum time across ‘neval’ replicates</li>
<li><strong>Quartiles</strong>: ‘lq’ = lower quartile (25%), ‘median’, ‘uq’ = upper quartile (75%)</li>
<li><strong>mean</strong>: mean time across ‘neval’ replicates</li>
<li><strong>max</strong>: maximum time across ‘neval’ replicates</li>
</ul>
<p>In this case, compiling mainly reduced the duration of the longest 25% runs (with longer times than the 75% quartile), which brought down the mean processing time.</p>
</div>
</div>
<div id="profiling-your-code" class="section level4 unnumbered hasAnchor">
<h4>4. Profiling your code<a href="bonus_8a.html#profiling-your-code" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<div id="a.-named-chunks" class="section level5 unnumbered hasAnchor">
<h5>a. Named chunks<a href="bonus_8a.html#a.-named-chunks" class="anchor-section" aria-label="Anchor link to header"></a></h5>
<p>An simple way to identify parts of your code that may be slow is to:</p>
<ul>
<li>Name each chunk in your R Notebook</li>
<li>Knit the notebook</li>
<li>Monitor the R markdown pane while the notebook is knitted: which chunks seem to take a lot of time?</li>
</ul>
<p>To name a chunk, click on the wheel symbol at the top right of the grey chunk area and enter a one-word name.</p>
<p>Here’s an example of a named chunk: the name ‘myChunkName’ has been added in the curly brackets <code>{r, myChunkName}</code>. You can add a name manually in the same way.</p>
<p>More generally, this is where chunk options are added in the R Notebook. Here’s a long list of chunk options: <a href="https://yihui.name/knitr/options/" class="uri">https://yihui.name/knitr/options/</a></p>
</div>
<div id="b.-profiling-some-lines-of-code" class="section level5 unnumbered hasAnchor">
<h5>b. Profiling some lines of code<a href="bonus_8a.html#b.-profiling-some-lines-of-code" class="anchor-section" aria-label="Anchor link to header"></a></h5>
<p>RStudio has a built-in menu for profiling. Check it out:</p>
<ul>
<li>Select the five lines of code below, from <code>dat &lt;- data.frame(</code> until <code>lm(y ~ x, data=dat)</code></li>
<li>In RStudio’s menu, click on ‘Profile’ &gt; ‘Profile Selected Line(s)’.</li>
</ul>
<p>You can achieve the same with a call to the funciton ‘profvis’ of the ‘profvis’ package:</p>
<div class="sourceCode" id="cb1069"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1069-1"><a href="bonus_8a.html#cb1069-1" tabindex="-1"></a>profvis<span class="sc">::</span><span class="fu">profvis</span>({</span>
<span id="cb1069-2"><a href="bonus_8a.html#cb1069-2" tabindex="-1"></a>  dat <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(</span>
<span id="cb1069-3"><a href="bonus_8a.html#cb1069-3" tabindex="-1"></a>       <span class="at">x =</span> <span class="fu">rnorm</span>(<span class="fl">5e5</span>), </span>
<span id="cb1069-4"><a href="bonus_8a.html#cb1069-4" tabindex="-1"></a>       <span class="at">y =</span> <span class="fu">rnorm</span>(<span class="fl">5e5</span>))   </span>
<span id="cb1069-5"><a href="bonus_8a.html#cb1069-5" tabindex="-1"></a>  <span class="fu">mean</span>(dat<span class="sc">$</span>x)</span>
<span id="cb1069-6"><a href="bonus_8a.html#cb1069-6" tabindex="-1"></a>  <span class="fu">with</span>(dat, <span class="fu">cor</span>(x, y))</span>
<span id="cb1069-7"><a href="bonus_8a.html#cb1069-7" tabindex="-1"></a>  <span class="fu">lm</span>(y <span class="sc">~</span> x, <span class="at">data=</span>dat)</span>
<span id="cb1069-8"><a href="bonus_8a.html#cb1069-8" tabindex="-1"></a>})</span></code></pre></div>
<div class="profvis html-widget html-fill-item" id="htmlwidget-8200f0838a0312caa364" style="width:100%;height:600px;"></div>
<script type="application/json" data-for="htmlwidget-8200f0838a0312caa364">{"x":{"message":{"prof":{"time":[1,2,3,4,5,5,5,5,5,5,5,5,5,5,5,5,5,5,5,5,5,5,5,5,5,5,5,6,6,6,6,6,6,6,6,6,6,6,6,6,6,6,6,6,6,6,6,6,6,6,6,7,7],"depth":[1,1,1,1,23,22,21,20,19,18,17,16,15,14,13,12,11,10,9,8,7,6,5,4,3,2,1,24,23,22,21,20,19,18,17,16,15,14,13,12,11,10,9,8,7,6,5,4,3,2,1,2,1],"label":["rnorm","rnorm","rnorm","rnorm","[.data.frame","na.omit.data.frame","model.frame.default","eval","eval","eval_with_user_handlers","evaluate_call","evaluate::evaluate","evaluate","eng_r","block_exec","call_block","process_group.block","process_file","knit","FUN","lapply","call_block","process_group.block","process_file","knitr::knit","<Anonymous>","local","anyDuplicated.default","[.data.frame","na.omit.data.frame","model.frame.default","eval","eval","eval_with_user_handlers","evaluate_call","evaluate::evaluate","evaluate","eng_r","block_exec","call_block","process_group.block","process_file","knit","FUN","lapply","call_block","process_group.block","process_file","knitr::knit","<Anonymous>","local","lm.fit","lm"],"filenum":[null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,1],"linenum":[null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,7],"memalloc":[162.8675003051758,162.8675003051758,166.6845855712891,166.6845855712891,180.0466079711914,180.0466079711914,180.0466079711914,180.0466079711914,180.0466079711914,180.0466079711914,180.0466079711914,180.0466079711914,180.0466079711914,180.0466079711914,180.0466079711914,180.0466079711914,180.0466079711914,180.0466079711914,180.0466079711914,180.0466079711914,180.0466079711914,180.0466079711914,180.0466079711914,180.0466079711914,180.0466079711914,180.0466079711914,180.0466079711914,203.1202621459961,203.1202621459961,203.1202621459961,203.1202621459961,203.1202621459961,203.1202621459961,203.1202621459961,203.1202621459961,203.1202621459961,203.1202621459961,203.1202621459961,203.1202621459961,203.1202621459961,203.1202621459961,203.1202621459961,203.1202621459961,203.1202621459961,203.1202621459961,203.1202621459961,203.1202621459961,203.1202621459961,203.1202621459961,203.1202621459961,203.1202621459961,231.734001159668,231.734001159668],"meminc":[0,0,3.817085266113281,0,13.36202239990234,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,23.07365417480469,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,28.61373901367188,0],"filename":[null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,"<expr>"]},"interval":10,"files":[{"filename":"<expr>","content":"profvis::profvis({\n  dat <- data.frame(\n       x = rnorm(5e5), \n       y = rnorm(5e5))   \n  mean(dat$x)\n  with(dat, cor(x, y))\n  lm(y ~ x, data=dat)\n})","normpath":"<expr>"}],"prof_output":"/var/folders/kz/fv8pzy0d3cl_mp3md294cvbh0000gn/T//RtmpPhhA1R/file491a36536425.prof","highlight":{"output":["^output\\$"],"gc":["^<GC>$"],"stacktrace":["^\\.\\.stacktraceo(n|ff)\\.\\.$"]},"split":"h"}},"evals":[],"jsHooks":[]}</script>
<p>The results will be opened in a ‘Profile’ tab. The upper part has two tabs (the lower part is less intuitive to interpret, we’ll ignore it here):</p>
<ul>
<li><strong>Flame Graph</strong>: this plots horizontal bars indicating the amount of memory and time used by each line of code, in the original order of the code.</li>
<li><strong>Data</strong>: code is sorted by resource use, with the most ‘costly’ line <code>qqnorm</code> at the top of the list. You can click on the triangle before each line to see more detail.</li>
</ul>
<p>The results may depend on the speed of your computer. Check the sample interval at the bottom of the ‘Profile’ tab: time was estimated by observing every 10 milliseconds what the computer was doing. Some lines lines must have been too fast to be recorded.</p>
</div>
<div id="c.-converting-an-r-notebook-to-an-r-script" class="section level5 unnumbered hasAnchor">
<h5>c. Converting an R Notebook to an R script<a href="bonus_8a.html#c.-converting-an-r-notebook-to-an-r-script" class="anchor-section" aria-label="Anchor link to header"></a></h5>
<p>Unfortunately, we can’t profile an entire R Notebook (as far as I know). However, we can extract the R code as a script file, then profile the script file.</p>
<p>Copy an R Notebook file (<code>Testfile.Rmd</code>) to the downloads folder. This is just so that we have an example of an <code>.Rmd</code> file to extract the code from.</p>
<div class="sourceCode" id="cb1070"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1070-1"><a href="bonus_8a.html#cb1070-1" tabindex="-1"></a><span class="fu">file.copy</span>(<span class="at">from=</span><span class="fu">system.file</span>(<span class="st">&quot;extdata&quot;</span>, <span class="st">&quot;Testfile.Rmd&quot;</span>, <span class="at">package =</span> <span class="st">&quot;LandGenCourse&quot;</span>),</span>
<span id="cb1070-2"><a href="bonus_8a.html#cb1070-2" tabindex="-1"></a>                     <span class="at">to=</span><span class="fu">file.path</span>(here<span class="sc">::</span><span class="fu">here</span>(), <span class="st">&quot;downloads&quot;</span>, <span class="st">&quot;Testfile.Rmd&quot;</span>))</span></code></pre></div>
<pre><code>## [1] FALSE</code></pre>
<p>Extract the R code from the R Notebook and save it in a script file <code>Testfile.R</code> in the downloads folder. You may adapt the infile and outfile to extract the code from any R Notebook saved in your project.</p>
<p>Let’s open the two files and compare them. Note that <code>file.show</code> opens a simple text version, whereas <code>file.edit</code> shows the colored versions commonly displayed by the RStudio editor.</p>
<p><strong>Question</strong>: Compare the two files (they should be open, check the tabs of the source pane).</p>
<ul>
<li>How are the chunks from the Notebook file divided in the new script file?</li>
<li>What happened to the titles and text?</li>
<li>What about the header information?</li>
</ul>
</div>
<div id="d.-profiling-an-r-script" class="section level5 unnumbered hasAnchor">
<h5>d. Profiling an R script<a href="bonus_8a.html#d.-profiling-an-r-script" class="anchor-section" aria-label="Anchor link to header"></a></h5>
<p>We can now use the function <code>source</code> to read and execute the code in the R script. We use the function <code>Rprof</code> to start and stop the profiling.</p>
<p>With the function <code>summaryRprof</code>, we get a summary by function. First, let’s check the total time spent executing the code, and the sample interval (in seconds). Note that profiling works in this way: as the code is executed, R checks at regular time intervals which function is being executed at that very moment. The <code>summaryRprof</code> function then tabulates the number of intervals, and thus the time used, by function.</p>
<div class="sourceCode" id="cb1072"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1072-1"><a href="bonus_8a.html#cb1072-1" tabindex="-1"></a><span class="co">#summaryRprof(here::here(&quot;downloads/Rprof.out&quot;))[c(&quot;sampling.time&quot;, &quot;sample.interval&quot;)] </span></span></code></pre></div>
<p>The attribute <code>$by.total</code> only lists the top-level functions called (not any functions called internally by that top-level function).</p>
<div class="sourceCode" id="cb1073"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1073-1"><a href="bonus_8a.html#cb1073-1" tabindex="-1"></a><span class="co">#summaryRprof(here::here(&quot;downloads/Rprof.out&quot;))$by.total</span></span></code></pre></div>
<p>Note: The functions are listed by total time, in decreasing order. The two functions <code>eval</code> and <code>source</code>, are related to evaluating (running) the code from the sourced R script file.</p>
<ul>
<li><strong>total.time</strong>: total time spent executing the function (including functions called by it)</li>
<li><strong>total.pct</strong>: (ignore this) percent time spent executing the function (including functions called by it)</li>
<li><strong>self.time</strong>: total time spent executing the function itself (excluding functions called by it)</li>
<li><strong>self.pct</strong>: percent time spent executing the function itself (excluding functions called by it)</li>
</ul>
<p>If you need to see a summary with more detail, <code>$by.self</code> lists each function used, even internal functions.</p>
<p>If you want to profile memory use rather than processing time, see here: <a href="https://developer.r-project.org/memory-profiling.html" class="uri">https://developer.r-project.org/memory-profiling.html</a></p>
</div>
</div>
<div id="creating-and-executing-a-bash-r-script" class="section level4 unnumbered hasAnchor">
<h4>5. Creating and executing a Bash R script<a href="bonus_8a.html#creating-and-executing-a-bash-r-script" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<div id="a.-run-r-script-directly-in-the-terminal" class="section level5 unnumbered hasAnchor">
<h5>a. Run R script directly in the Terminal<a href="bonus_8a.html#a.-run-r-script-directly-in-the-terminal" class="anchor-section" aria-label="Anchor link to header"></a></h5>
<p>Now that we have a stand-alone R script ‘myScript.R’, we can run it from the command line in the terminal (shell). After navigating to the correct folder, type:</p>
<p><code>Rscript myScript.R</code></p>
<p>This will source the file and execute the R code.</p>
<ul>
<li>Numerical output will be printed in the termminal. Here, a random number should be returned.</li>
<li>Best include code in your R script to export any R objects, data tables or figures that you want to retain. These will be saved in the same folder as the R script (unless you specify file paths).</li>
</ul>
</div>
<div id="b.-create-a-bash-r-script" class="section level5 unnumbered hasAnchor">
<h5>b. Create a Bash R script<a href="bonus_8a.html#b.-create-a-bash-r-script" class="anchor-section" aria-label="Anchor link to header"></a></h5>
<p>If you want to run your code on a node or cluster, you may need to take this one step further and include the R code in a bash Rscript. In a bash script, you can add bash commands that govern resource use to submit a job to a node or cluster. Bash scripts are the way of giving instructions to the scheduler of the cluster (e.g. SLURM) for how to manage input and output files.</p>
<p>To execute our R script ‘myScript.R’ as a Bash script, we need to add a few lines.</p>
<ul>
<li>The ‘shebang’ line <code>#!/bin/bash</code> that tells the computer that this is a Bash script, and where to find Bash. Note that here the hashtag symbol does NOT mean that the line is commented out (this line is Unix code, not R code).</li>
<li>The line <code>R --slave &lt;&lt; EOF</code> that declares the rest of the file (until <code>EOF</code>) as R code.</li>
<li>The end of file <code>EOF</code> marker.</li>
</ul>
<p>Let’s modify the previous code and write it into a Bash file. As an additional challenge, our code contains two figures, which won’t be written anywhere unless we change the code to write them into a file:</p>
<ul>
<li>We create a graphics file ‘my_plot.png’ that is 800 pixels wide and 400 pixels high.</li>
<li>With <code>par(mfrow=c(1,2))</code>, we specify that the two plots should be plotted side-by-side. Then we create the plots.</li>
<li>We close the graphics device (png file) with <code>dev.off</code>.</li>
</ul>
<p>Note: We use single quotes for the file name here, ‘my_plot.png’, as they are nested within a set of double quotes. R pretty much considers single and double quotes as synonyms, which allows us to nest them either way: ‘““’ or”’’“.</p>
<div class="sourceCode" id="cb1074"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1074-1"><a href="bonus_8a.html#cb1074-1" tabindex="-1"></a>myPath <span class="ot">&lt;-</span> <span class="fu">file.path</span>(here<span class="sc">::</span><span class="fu">here</span>(), <span class="st">&quot;output/myBashScript.sh&quot;</span>)</span>
<span id="cb1074-2"><a href="bonus_8a.html#cb1074-2" tabindex="-1"></a>fileConn <span class="ot">&lt;-</span> <span class="fu">file</span>(myPath)</span>
<span id="cb1074-3"><a href="bonus_8a.html#cb1074-3" tabindex="-1"></a><span class="fu">writeLines</span>(<span class="fu">c</span>(<span class="st">&quot;#!/bin/bash&quot;</span>,</span>
<span id="cb1074-4"><a href="bonus_8a.html#cb1074-4" tabindex="-1"></a>             <span class="st">&quot;R --slave &lt;&lt; EOF&quot;</span>,</span>
<span id="cb1074-5"><a href="bonus_8a.html#cb1074-5" tabindex="-1"></a>             <span class="st">&quot;x &lt;- rnorm(100)&quot;</span>,</span>
<span id="cb1074-6"><a href="bonus_8a.html#cb1074-6" tabindex="-1"></a>             <span class="st">&quot;mean(x)&quot;</span>,</span>
<span id="cb1074-7"><a href="bonus_8a.html#cb1074-7" tabindex="-1"></a>             <span class="st">&quot;png(&#39;my_plot.png&#39;, height = 400, width = 800)&quot;</span>,</span>
<span id="cb1074-8"><a href="bonus_8a.html#cb1074-8" tabindex="-1"></a>             <span class="st">&quot;par(mfrow=c(1,2))&quot;</span>, </span>
<span id="cb1074-9"><a href="bonus_8a.html#cb1074-9" tabindex="-1"></a>             <span class="st">&quot;hist(x)&quot;</span>, </span>
<span id="cb1074-10"><a href="bonus_8a.html#cb1074-10" tabindex="-1"></a>             <span class="st">&quot;qqnorm(x)&quot;</span>,</span>
<span id="cb1074-11"><a href="bonus_8a.html#cb1074-11" tabindex="-1"></a>             <span class="st">&quot;dev.off()&quot;</span>,</span>
<span id="cb1074-12"><a href="bonus_8a.html#cb1074-12" tabindex="-1"></a>             <span class="st">&quot;EOF&quot;</span>), fileConn)</span>
<span id="cb1074-13"><a href="bonus_8a.html#cb1074-13" tabindex="-1"></a><span class="fu">close</span>(fileConn)</span>
<span id="cb1074-14"><a href="bonus_8a.html#cb1074-14" tabindex="-1"></a><span class="fu">file.show</span>(myPath)</span></code></pre></div>
</div>
<div id="c.-executing-a-bash-r-script" class="section level5 unnumbered hasAnchor">
<h5>c. Executing a Bash R script<a href="bonus_8a.html#c.-executing-a-bash-r-script" class="anchor-section" aria-label="Anchor link to header"></a></h5>
<p>On Mac / Unix / Linux, this is straight-forward:</p>
<ol style="list-style-type: decimal">
<li><strong>Open terminal</strong>:
<ul>
<li>From the RStudio menu, select ‘Tools’ &gt; ‘Terminal’ &gt; ‘New Terminal’. This will open an Terminal tab in RStudio. Alternatively, you could select ‘Tools’ &gt; ‘Shell’ to open a Shell in a new window outside RStudio.</li>
<li>Check the prompt: it should start with the name of your computer, then a colon, then the name of your project folder, then your use name followed ‘$’.</li>
</ul></li>
<li><strong>Navigate to Bash file</strong>:
<ul>
<li>Enter <code>ls</code> to list the content of the project folder.</li>
<li>Enter <code>cd output</code> to change directory to the subfolder ‘output’.</li>
<li>Repeat <code>ls</code> to list the content of the ‘output’ folder. The Bash script ‘myBashScript.sh’ should be listed there.</li>
</ul></li>
<li><strong>Execute Bash file</strong>:
<ul>
<li>Enter <code>chmod +x myBashScript.sh</code> to change file permission for the script.</li>
<li>Enter <code>./myBashScript.sh</code> to execute the script.</li>
</ul></li>
<li><strong>Find output</strong>:
<ul>
<li>The output from <code>mean(x)</code> is printed in the terminal, it should look like this: <code>[1] -0.07731751</code>.</li>
<li>This is followed ‘null device’ and the number 1, which tells us that a graphics device has been closed.</li>
<li>Enter <code>ls</code> to list the content of the project folder. The graphics file ‘my_plot.png’ should now be listed.</li>
</ul></li>
</ol>
<p>Use R again to open the graphics file (the code here first checks whether the file exists):</p>
<div class="sourceCode" id="cb1075"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1075-1"><a href="bonus_8a.html#cb1075-1" tabindex="-1"></a>myPNG <span class="ot">&lt;-</span> <span class="fu">file.path</span>(here<span class="sc">::</span><span class="fu">here</span>(), <span class="st">&quot;output/my_plot.png&quot;</span>)</span>
<span id="cb1075-2"><a href="bonus_8a.html#cb1075-2" tabindex="-1"></a><span class="cf">if</span>(<span class="fu">file.exists</span>(myPNG))</span>
<span id="cb1075-3"><a href="bonus_8a.html#cb1075-3" tabindex="-1"></a>{</span>
<span id="cb1075-4"><a href="bonus_8a.html#cb1075-4" tabindex="-1"></a>  <span class="fu">file.show</span>(myPNG)</span>
<span id="cb1075-5"><a href="bonus_8a.html#cb1075-5" tabindex="-1"></a>}</span></code></pre></div>
</div>
<div id="d.-moving-to-a-node-or-cluster" class="section level5 unnumbered hasAnchor">
<h5>d. Moving to a node or cluster?<a href="bonus_8a.html#d.-moving-to-a-node-or-cluster" class="anchor-section" aria-label="Anchor link to header"></a></h5>
<p>The example bash file and advice in this section have been provided by Hossam Abdel Moniem, thanks!</p>
<p>Here’s an annotated example of a bash file that contains instructions for submitting a job to a single node (a single machine with multiple/many cores).</p>
<p>Note: A copy of the file ‘BashExample.sh’ should also have been copied into the downloads folder inside your project folder.</p>
<div class="sourceCode" id="cb1076"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1076-1"><a href="bonus_8a.html#cb1076-1" tabindex="-1"></a><span class="fu">writeLines</span>(<span class="fu">readLines</span>(<span class="fu">system.file</span>(<span class="st">&quot;extdata&quot;</span>, <span class="st">&quot;BashExample.sh&quot;</span>, <span class="at">package =</span> <span class="st">&quot;LandGenCourse&quot;</span>)))</span></code></pre></div>
<pre><code>## #!/bin/bash 
## 
## #SBATCH --nodes=1                                 # Number of Nodes 
## #SBATCH --mail-type=ALL                           # Mail events (NONE, BEGIN, END, FAIL, ALL)
## #SBATCH --mail-user=myEmail@gmail.com             # Where to send mail
## #SBATCH --ntasks=1                                # Run a single task
## #SBATCH --cpus-per-task=24                        # Number of CPU cores per task
## #SBATCH --mem-per-cpu=8000             # allocated memory for the task
## #SBATCH -p nodename                # name of cluster node
## #SBATCH --requeue                                 # Allow the job to be requeued
## #SBATCH -e myJob.err                   # File to which STDERR will be written 
## #SBATCH -o myJob.out                   # File to which STDOUT will be written 
## #SBATCH -J myJob                   # Job name 
## 
## module load R/MS3.4.1                  # call a preinstalled module(program) on the cluster
## 
## Rscript connect_calc_25.R              # Run the R script in bash 
## </code></pre>
<p>Instead of including the R code directly in the bash file, the last line here executes an R script with the <code>Rscript</code> command.</p>
<p>Notice the second-last line. Obviously, on the node, R and any relevant packages need to be pre-installed. Different users may need different configurations (different packages or versions) installed, hence each installation has a name, which needs to be specified in the bash script.</p>
<p>Note: Make sure that all packages that you need (and their dependencies), as well as the package <code>unixtools</code>, have been installed on the node or cluster (i.e., they are part of the installation you will be using). Install <code>unixtools</code> with: <code>install.packages("unixtools",,"http://rforge.net/")</code></p>
<p>Further reading:</p>
<ul>
<li>Bash script tutorial: <a href="https://ryanstutorials.net/bash-scripting-tutorial/" class="uri">https://ryanstutorials.net/bash-scripting-tutorial/</a></li>
<li>Scheduling a job with SLURM commands: <a href="https://www.rc.fas.harvard.edu/resources/documentation/convenient-slurm-commands/" class="uri">https://www.rc.fas.harvard.edu/resources/documentation/convenient-slurm-commands/</a></li>
</ul>
</div>
<div id="e.-store-your-session-info-and-package-versions" class="section level5 unnumbered hasAnchor">
<h5>e. Store your session info and package versions<a href="bonus_8a.html#e.-store-your-session-info-and-package-versions" class="anchor-section" aria-label="Anchor link to header"></a></h5>
<p>If you use any R packages, load them at the beginning of your script file with <code>library</code>. Make sure the packages are installed on the system where you will be running the Bash R script.</p>
<p>A big issue with R is that package updates may make your code break. At least at the end of any project (such as the analyses for a manuscript), save your session information.</p>
<p>Here we use the function <code>session_info</code> from the <code>devtools</code> package (preferred over the R base function <code>sessionInfo</code>). We store the information as an object, ‘Session’, of class ‘session_info’ that has two list elements:</p>
<ul>
<li><strong>platform</strong>: Information about R version, your computer’s operating system, etc.</li>
<li><strong>packages</strong>: List of all packages, including their version, installation date and repository (CRAN, Github etc.). All packages that are currently loaded will be marked with an asterisk.</li>
</ul>
<p>Platform:</p>
<div class="sourceCode" id="cb1078"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1078-1"><a href="bonus_8a.html#cb1078-1" tabindex="-1"></a>Session <span class="ot">&lt;-</span> devtools<span class="sc">::</span><span class="fu">session_info</span>()</span>
<span id="cb1078-2"><a href="bonus_8a.html#cb1078-2" tabindex="-1"></a>Session<span class="sc">$</span>platform</span></code></pre></div>
<pre><code>##  setting  value
##  version  R version 4.3.2 (2023-10-31)
##  os       macOS Sonoma 14.2.1
##  system   x86_64, darwin20
##  ui       X11
##  language (EN)
##  collate  en_US.UTF-8
##  ctype    en_US.UTF-8
##  tz       Europe/Zurich
##  date     2024-01-25
##  pandoc   3.1.1 @ /Applications/RStudio.app/Contents/Resources/app/quarto/bin/tools/ (via rmarkdown)</code></pre>
<p>Packages: here we display only the first six lines, as the list may be long.</p>
<div class="sourceCode" id="cb1080"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1080-1"><a href="bonus_8a.html#cb1080-1" tabindex="-1"></a><span class="fu">head</span>(Session<span class="sc">$</span>packages)</span></code></pre></div>
<pre><code>##  package    * version  date (UTC) lib source
##  abind        1.4-5    2016-07-21 [1] CRAN (R 4.3.0)
##  ade4         1.7-22   2023-02-06 [1] CRAN (R 4.3.0)
##  adegenet     2.1.10   2023-01-26 [1] CRAN (R 4.3.0)
##  ape          5.7-1    2023-03-13 [1] CRAN (R 4.3.0)
##  arrow        14.0.0.2 2023-12-02 [1] CRAN (R 4.3.0)
##  assertthat   0.2.1    2019-03-21 [1] CRAN (R 4.3.0)
## 
##  [1] /Library/Frameworks/R.framework/Versions/4.3-x86_64/Resources/library</code></pre>
<p>Exporting the session information is a bit tricky because ‘Session’ is not in tabular format, and what we want to export is the formatted output, not the object itself.</p>
<p>Also, we will add a time stamp to the file name. This achieves two goals: avoid overwriting earlier files, and keep a record of the date of the session information.</p>
<ul>
<li><code>capture.output</code>: captures the output of a function (here: <code>devtools::session_info()</code>)</li>
<li><code>writeLines</code>: writes the captured output into a file, here a text file.</li>
<li><code>Sys.Date</code>: returns the current date. Here we specify the format as “%Y-%m-%d”, i.e., ‘Year-month-day’.</li>
</ul>
<p>More advanced ways for handling the problem of package versions to make sure you can run your code in the future without compatibility issues include:</p>
<ul>
<li>Package ‘packrat’: bundle all your current package version. This can become quite large.</li>
<li>Package ‘checkpoint’: access daily snapshots of CRAN for any given date. Note that checkpoint only covers packages from CRAN, not other repositories like GitHub.</li>
</ul>
<p>More on the topic: <a href="https://timogrossenbacher.ch/2017/07/a-truly-reproducible-r-workflow/" class="uri">https://timogrossenbacher.ch/2017/07/a-truly-reproducible-r-workflow/</a></p>
</div>
<div id="f.-potential-windows-issues" class="section level5 unnumbered hasAnchor">
<h5>f. Potential Windows issues<a href="bonus_8a.html#f.-potential-windows-issues" class="anchor-section" aria-label="Anchor link to header"></a></h5>
<p>Check that Bash is installed and ready to use with RStudio:</p>
<ul>
<li>In the RStudio menu, select ‘Tools’ &gt; ‘Global options’ &gt; ‘Terminal’.</li>
<li>Make sure some form of Bash (e.g., Git Bash) is listed under ‘Shell: New terminals open with:’.</li>
</ul>
<p>If this does not work, try installing ‘Git for Windows’, which will also install Bash: <a href="http://neondataskills.org/setup/setup-git-bash-R" class="uri">http://neondataskills.org/setup/setup-git-bash-R</a></p>
<p>Here’s a detailed multi-part tutorial on running R from the command line, with R scripts and Bash R scripts, with some additional information on Windows:</p>
<ul>
<li><strong>Introduction</strong>: <a href="https://github.com/gastonstat/tutorial-R-noninteractive/blob/master/01-introduction.Rmd" class="uri">https://github.com/gastonstat/tutorial-R-noninteractive/blob/master/01-introduction.Rmd</a></li>
<li><strong>Batch mode</strong>: <a href="https://github.com/gastonstat/tutorial-R-noninteractive/blob/master/02-batch-mode.Rmd" class="uri">https://github.com/gastonstat/tutorial-R-noninteractive/blob/master/02-batch-mode.Rmd</a></li>
<li><strong>Executing R scripts</strong>: <a href="https://github.com/gastonstat/tutorial-R-noninteractive/blob/master/03-rscript.Rmd" class="uri">https://github.com/gastonstat/tutorial-R-noninteractive/blob/master/03-rscript.Rmd</a></li>
<li><strong>Bash R scripts</strong>: <a href="https://github.com/gastonstat/tutorial-R-noninteractive/blob/master/04-shell-script.Rmd" class="uri">https://github.com/gastonstat/tutorial-R-noninteractive/blob/master/04-shell-script.Rmd</a></li>
<li><strong>Redirection</strong>: <a href="https://github.com/gastonstat/tutorial-R-noninteractive/blob/master/05-redirection.Rmd" class="uri">https://github.com/gastonstat/tutorial-R-noninteractive/blob/master/05-redirection.Rmd</a></li>
</ul>
</div>
</div>
<div id="parallelizing-code" class="section level4 unnumbered hasAnchor">
<h4>6. Parallelizing code<a href="bonus_8a.html#parallelizing-code" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>Note that the following issues may create problems when developing Bash R scripts on Windows that you want to run e.g. on a Linux cluster or another Unix-type system;</p>
<ul>
<li>File paths are different, and system files are stored in a different place.</li>
<li>End-of-line symbols are different</li>
</ul>
<div id="a.-replace-lapply-by-mclapply-with-package-parallel" class="section level5 unnumbered hasAnchor">
<h5>a. Replace ‘lapply’ by ‘mclapply’ with package ‘parallel’<a href="bonus_8a.html#a.-replace-lapply-by-mclapply-with-package-parallel" class="anchor-section" aria-label="Anchor link to header"></a></h5>
<p>Note: while this will run on a Windows without causing an error, it will only be faster on Mac / Unix.</p>
<p>With the package ‘parallel’, it is really easy to use all cores of your local machine (as long as you are on a Mac / Unix / Linux system). Let’s check the number of cores available:</p>
<div class="sourceCode" id="cb1082"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1082-1"><a href="bonus_8a.html#cb1082-1" tabindex="-1"></a><span class="fu">library</span>(parallel)</span>
<span id="cb1082-2"><a href="bonus_8a.html#cb1082-2" tabindex="-1"></a><span class="fu">detectCores</span>()</span></code></pre></div>
<pre><code>## [1] 6</code></pre>
<p><strong>Question</strong>: How many cores does your machine have?</p>
<p>Note: Here we check whether the operating system is ‘Windows’, in which case we set <code>nCores = 1</code>. This means that we will use all cores on Mac or Linux, but only one core on Windows. This is to avoid problems on Windows machines.</p>
<div class="sourceCode" id="cb1084"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1084-1"><a href="bonus_8a.html#cb1084-1" tabindex="-1"></a>nCores <span class="ot">&lt;-</span> <span class="fu">detectCores</span>()</span>
<span id="cb1084-2"><a href="bonus_8a.html#cb1084-2" tabindex="-1"></a><span class="cf">if</span>(<span class="fu">Sys.info</span>()[[<span class="st">&#39;sysname&#39;</span>]]<span class="sc">==</span><span class="st">&quot;Windows&quot;</span>) nCores <span class="ot">=</span> <span class="dv">1</span></span>
<span id="cb1084-3"><a href="bonus_8a.html#cb1084-3" tabindex="-1"></a>nCores</span></code></pre></div>
<pre><code>## [1] 6</code></pre>
<ul>
<li>Code your analysis with <code>lapply</code> (and related functions).</li>
<li>Replace <code>lapply</code> by <code>mclapply</code> (and related functions). Use the argument <code>mc.cores=detectCores()</code> to automatically detect the number of cores in your machine.</li>
</ul>
<p>NOTE: April 2021: package build does not work with multiple cores, changing nCores to 1.</p>
<div class="sourceCode" id="cb1086"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1086-1"><a href="bonus_8a.html#cb1086-1" tabindex="-1"></a>x <span class="ot">&lt;-</span> gen[,<span class="sc">-</span><span class="dv">1</span>]</span>
<span id="cb1086-2"><a href="bonus_8a.html#cb1086-2" tabindex="-1"></a>m1 <span class="ot">&lt;-</span> <span class="fu">lapply</span>(x, mean, <span class="at">na.rm=</span><span class="cn">TRUE</span>)</span>
<span id="cb1086-3"><a href="bonus_8a.html#cb1086-3" tabindex="-1"></a><span class="co">#m2 &lt;- mclapply(x, mean, na.rm=TRUE, mc.cores=nCores)  # Use this line when running the code yourself</span></span>
<span id="cb1086-4"><a href="bonus_8a.html#cb1086-4" tabindex="-1"></a>m2 <span class="ot">&lt;-</span> <span class="fu">mclapply</span>(x, mean, <span class="at">na.rm=</span><span class="cn">TRUE</span>, <span class="at">mc.cores=</span><span class="dv">1</span>)        <span class="co"># Replace this line with the previous line</span></span></code></pre></div>
<p>Let’s benchmark four ways of calculating the mean of each column in our example data set ‘gen’ with 94 rows and 10,000 columns (SNPs):</p>
<ul>
<li><strong>Method 1</strong>: The dedicated function <code>colMeans</code> from base R.</li>
<li><strong>Method 2</strong>: A <code>for</code> loop.</li>
<li><strong>Method 3</strong>: Vectorization with <code>lapply</code></li>
<li><strong>Method 4</strong>: Multi-core with <code>mclapply</code>.</li>
</ul>
<div class="sourceCode" id="cb1087"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1087-1"><a href="bonus_8a.html#cb1087-1" tabindex="-1"></a>method1 <span class="ot">&lt;-</span> <span class="cf">function</span>(x) {<span class="fu">colMeans</span>(x, <span class="at">na.rm=</span><span class="cn">TRUE</span>)}</span>
<span id="cb1087-2"><a href="bonus_8a.html#cb1087-2" tabindex="-1"></a>method2 <span class="ot">&lt;-</span> <span class="cf">function</span>(x) {<span class="cf">for</span>(i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">ncol</span>(x)) <span class="fu">mean</span>(x[,i], <span class="at">na.rm=</span><span class="cn">TRUE</span>)}</span>
<span id="cb1087-3"><a href="bonus_8a.html#cb1087-3" tabindex="-1"></a>method3 <span class="ot">&lt;-</span> <span class="cf">function</span>(x) {<span class="fu">lapply</span>(x, mean, <span class="at">na.rm=</span><span class="cn">TRUE</span>)}</span>
<span id="cb1087-4"><a href="bonus_8a.html#cb1087-4" tabindex="-1"></a><span class="co">#method4 &lt;- function(x) {mclapply(x, mean, na.rm=TRUE, mc.cores=nCores)} # Use this line when running the code yourself</span></span>
<span id="cb1087-5"><a href="bonus_8a.html#cb1087-5" tabindex="-1"></a>method4 <span class="ot">&lt;-</span> <span class="cf">function</span>(x) {<span class="fu">mclapply</span>(x, mean, <span class="at">na.rm=</span><span class="cn">TRUE</span>, <span class="at">mc.cores=</span><span class="dv">1</span>)}  <span class="co"># Replace this line with the previous line</span></span>
<span id="cb1087-6"><a href="bonus_8a.html#cb1087-6" tabindex="-1"></a></span>
<span id="cb1087-7"><a href="bonus_8a.html#cb1087-7" tabindex="-1"></a>microbenchmark<span class="sc">::</span><span class="fu">microbenchmark</span>(<span class="at">times =</span> <span class="dv">10</span>, <span class="at">unit =</span> <span class="st">&quot;ms&quot;</span>,</span>
<span id="cb1087-8"><a href="bonus_8a.html#cb1087-8" tabindex="-1"></a>                               <span class="fu">method1</span>(x), <span class="fu">method2</span>(x), <span class="fu">method3</span>(x), <span class="fu">method4</span>(x))</span></code></pre></div>
<pre><code>## Unit: milliseconds
##        expr       min        lq      mean    median        uq       max neval
##  method1(x)  36.24154  37.78130  45.06711  38.35827  38.65051  77.02531    10
##  method2(x) 208.22822 209.65114 230.97283 219.31546 259.93541 261.70004    10
##  method3(x)  59.61928  64.77036  71.13950  68.86624  70.91237 108.48009    10
##  method4(x)  58.37850  60.10350  75.11765  66.04562  94.64458 115.26386    10
##  cld
##  a  
##   b 
##    c
##    c</code></pre>
<p><strong>Question</strong>: Compare the mean</p>
<ul>
<li>Which method was the fastest? Can you explain this?</li>
<li>Which method was slowest?</li>
<li>Was <code>mclapply</code> faster than <code>lapply</code> in this example?</li>
</ul>
<p>Note: Obviously you might only expect to see a gain in speed if nCores &gt; 1.</p>
</div>
<div id="b.-replace-for-by-foreach-with-package-doparallel" class="section level5 unnumbered hasAnchor">
<h5>b. Replace ‘for’ by ‘foreach’ with package ‘doParallel’<a href="bonus_8a.html#b.-replace-for-by-foreach-with-package-doparallel" class="anchor-section" aria-label="Anchor link to header"></a></h5>
<p>On Windows, it is easier to use the package ‘doParallel’ with the function ‘foreach’. Here’s a detailed introduction: <a href="http://127.0.0.1:26758/help/library/doParallel/doc/gettingstartedParallel.pdf" class="uri">http://127.0.0.1:26758/help/library/doParallel/doc/gettingstartedParallel.pdf</a>.</p>
<ul>
<li>Use <code>makeCluster</code> to specify the number of cores to be used. The default is half the number of cores.</li>
<li>Check number of clusters by printing <code>cl</code>.</li>
<li>Register the cluster with <code>registerDoParallel</code>. If you omit this step, the code will not use parallel computing.</li>
</ul>
<p>The following code is commented out to avoid problems when knitting the Notebook. You may uncomment and run it.</p>
<div class="sourceCode" id="cb1089"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1089-1"><a href="bonus_8a.html#cb1089-1" tabindex="-1"></a><span class="fu">library</span>(doParallel)</span></code></pre></div>
<pre><code>## Loading required package: foreach</code></pre>
<pre><code>## Loading required package: iterators</code></pre>
<div class="sourceCode" id="cb1092"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1092-1"><a href="bonus_8a.html#cb1092-1" tabindex="-1"></a><span class="co">#cl &lt;- makeCluster(2)</span></span>
<span id="cb1092-2"><a href="bonus_8a.html#cb1092-2" tabindex="-1"></a><span class="co">#cl</span></span>
<span id="cb1092-3"><a href="bonus_8a.html#cb1092-3" tabindex="-1"></a><span class="co">#registerDoParallel(cl)</span></span></code></pre></div>
<p>Now we adapt the code in two steps:</p>
<ul>
<li>The code is a little different with <code>foreach</code> than with <code>for</code>, as we use a pipe-like syntax with <code>%do%</code>, which means, for each value of <em>i</em>, do the following.</li>
<li>Thus, <code>%do%</code> is only a pipe operator, it does not result in parallelisation yet.</li>
<li>To make this parallel, replace <code>%do%</code> by <code>%dopar%</code>.</li>
</ul>
<div class="sourceCode" id="cb1093"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1093-1"><a href="bonus_8a.html#cb1093-1" tabindex="-1"></a>m1 <span class="ot">&lt;-</span> <span class="cf">for</span>(i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">ncol</span>(x)) <span class="fu">mean</span>(x[,i], <span class="at">na.rm=</span><span class="cn">TRUE</span>)</span>
<span id="cb1093-2"><a href="bonus_8a.html#cb1093-2" tabindex="-1"></a>m2 <span class="ot">&lt;-</span> <span class="fu">foreach</span>(<span class="at">i =</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">ncol</span>(x)) <span class="sc">%do%</span> (<span class="fu">mean</span>(x[,i], <span class="at">na.rm=</span><span class="cn">TRUE</span>))</span>
<span id="cb1093-3"><a href="bonus_8a.html#cb1093-3" tabindex="-1"></a><span class="co">#m3 &lt;- foreach(i = 1:ncol(x)) %dopar% (mean(x[,i], na.rm=TRUE))</span></span></code></pre></div>
<p>Let’s benchmark this again. We’ll only do 3 replicates this time (uncomment before running this code).</p>
<div class="sourceCode" id="cb1094"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1094-1"><a href="bonus_8a.html#cb1094-1" tabindex="-1"></a><span class="co">#method1 &lt;- function(x) {colMeans(x, na.rm=TRUE)}</span></span>
<span id="cb1094-2"><a href="bonus_8a.html#cb1094-2" tabindex="-1"></a><span class="co">#method2 &lt;- function(x) {for(i in 1:ncol(x)) mean(x[,i], na.rm=TRUE)}</span></span>
<span id="cb1094-3"><a href="bonus_8a.html#cb1094-3" tabindex="-1"></a><span class="co">#method3 &lt;- function(x) {lapply(x, mean, na.rm=TRUE)}</span></span>
<span id="cb1094-4"><a href="bonus_8a.html#cb1094-4" tabindex="-1"></a><span class="co">#method4 &lt;- function(x) {mclapply(x, mean, na.rm=TRUE, mc.cores=nCores)}</span></span>
<span id="cb1094-5"><a href="bonus_8a.html#cb1094-5" tabindex="-1"></a><span class="co">#method5 &lt;- function(x) {foreach(i = 1:ncol(x)) %do% (mean(x[,i], na.rm=TRUE))}</span></span>
<span id="cb1094-6"><a href="bonus_8a.html#cb1094-6" tabindex="-1"></a><span class="co">#method6 &lt;- function(x) {foreach(i = 1:ncol(x)) %dopar% (mean(x[,i], na.rm=TRUE))}</span></span>
<span id="cb1094-7"><a href="bonus_8a.html#cb1094-7" tabindex="-1"></a></span>
<span id="cb1094-8"><a href="bonus_8a.html#cb1094-8" tabindex="-1"></a><span class="co">#microbenchmark::microbenchmark(times = 3, unit = &quot;ms&quot;,method1(x), method2(x), method3(x), method4(x), method5(x), method6(x))</span></span></code></pre></div>
<p>Whether parallelisation is faster depends on the type of task and on your system. In this case, both versions that used parallelisation were actually slower than the sequential code (at least on my system).</p>
<p>Of course, this will not always be the case, it may depend on:</p>
<ul>
<li>What type and size of computational task you are running.</li>
<li>How many nodes you can use in parallel. The initial ‘cost’ of coordinating the task among cores may not be worth it if you only have two cores to use anyways, but with 20 cores, the gain will be higher.</li>
<li>Whether all cores are in the same node, or whether you distribute the work among multiple nodes (this will increase the cost of coordination, but potentially also give you access to many more cores).</li>
</ul>

</div>
</div>
</div>
<!-- </div> -->



            </section>

          </div>
        </div>
      </div>
<a href="r-exercise-week-8.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="Week9.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": true,
"facebook": false,
"twitter": false,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": false
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": null,
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "section"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.9/latest.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
