<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>10.5 Bonus: Si index | Landscape Genetic Data Analysis with R</title>
  <meta name="description" content="This is a web-interface to the teaching materials for the lab course ‘Landscape Genetic Data Analysis with R’ associated with the distributed graduate course ‘DGS Landscape Genetics.’ The output format is bookdown::gitbook." />
  <meta name="generator" content="bookdown 0.25 and GitBook 2.6.7" />

  <meta property="og:title" content="10.5 Bonus: Si index | Landscape Genetic Data Analysis with R" />
  <meta property="og:type" content="book" />
  
  <meta property="og:description" content="This is a web-interface to the teaching materials for the lab course ‘Landscape Genetic Data Analysis with R’ associated with the distributed graduate course ‘DGS Landscape Genetics.’ The output format is bookdown::gitbook." />
  <meta name="github-repo" content="hhwagner1/LandGenCourse_book" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="10.5 Bonus: Si index | Landscape Genetic Data Analysis with R" />
  
  <meta name="twitter:description" content="This is a web-interface to the teaching materials for the lab course ‘Landscape Genetic Data Analysis with R’ associated with the distributed graduate course ‘DGS Landscape Genetics.’ The output format is bookdown::gitbook." />
  

<meta name="author" content="Editor: Helene Wagner (University of Toronto)" />


<meta name="date" content="2022-04-03" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="r-exercise-week-7.html"/>
<link rel="next" href="Week8.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>
<script src="libs/kePrint-0.0.1/kePrint.js"></script>
<link href="libs/lightable-0.0.1/lightable.css" rel="stylesheet" />
<script src="libs/htmlwidgets-1.5.4/htmlwidgets.js"></script>
<script src="libs/d3-3.5.6/d3.min.js"></script>
<link href="libs/profvis-0.3.6.9000/profvis.css" rel="stylesheet" />
<script src="libs/profvis-0.3.6.9000/profvis.js"></script>
<script src="libs/profvis-0.3.6.9000/scroll.js"></script>
<link href="libs/highlight-6.2.0/textmate.css" rel="stylesheet" />
<script src="libs/highlight-6.2.0/highlight.js"></script>
<script src="libs/profvis-binding-0.3.7/profvis.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>


<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">Landscape Genetic Data Analysis with R</a></li>

<li class="divider"></li>
<li class="part"><span><b>Getting Started<span></span></b></span></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Introduction<span></span></a>
<ul>
<li class="chapter" data-level="1.1" data-path="how-to-use-this-book.html"><a href="how-to-use-this-book.html"><i class="fa fa-check"></i><b>1.1</b> How to use this Book<span></span></a>
<ul>
<li><a href="how-to-use-this-book.html#book-structure">1. Book Structure<span></span></a></li>
<li><a href="how-to-use-this-book.html#find-what-is-relevant-for-you">2. Find what is relevant for you<span></span></a></li>
<li><a href="how-to-use-this-book.html#course-r-package-landgencourse">3. Course R package ‘LandGenCourse’<span></span></a></li>
</ul></li>
<li class="chapter" data-level="1.2" data-path="list-of-packages.html"><a href="list-of-packages.html"><i class="fa fa-check"></i><b>1.2</b> List of R Packages by Vignette<span></span></a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="review-of-r-skills.html"><a href="review-of-r-skills.html"><i class="fa fa-check"></i><b>2</b> Review of R Skills<span></span></a>
<ul>
<li class="chapter" data-level="2.1" data-path="basic-r.html"><a href="basic-r.html"><i class="fa fa-check"></i><b>2.1</b> Basic R Programming<span></span></a>
<ul>
<li><a href="basic-r.html#overview">1. Overview<span></span></a></li>
<li><a href="basic-r.html#data-types">2. Data Types<span></span></a></li>
<li><a href="basic-r.html#data-containers">3. Data Containers<span></span></a></li>
<li><a href="basic-r.html#programming">4. Programming<span></span></a></li>
</ul></li>
<li class="chapter" data-level="2.2" data-path="r-graphics.html"><a href="r-graphics.html"><i class="fa fa-check"></i><b>2.2</b> R Graphics<span></span></a>
<ul>
<li><a href="r-graphics.html#overview-1">1. Overview<span></span></a></li>
<li><a href="r-graphics.html#univariate-plots">2. Univariate plots<span></span></a></li>
<li><a href="r-graphics.html#bivariate-plots">3. Bivariate plots<span></span></a></li>
<li><a href="r-graphics.html#advanced-plotting">4. Advanced plotting<span></span></a></li>
</ul></li>
<li class="chapter" data-level="2.3" data-path="resources.html"><a href="resources.html"><i class="fa fa-check"></i><b>2.3</b> Further R resources<span></span></a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="github-for-group-projects.html"><a href="github-for-group-projects.html"><i class="fa fa-check"></i><b>3</b> GitHub for Group Projects<span></span></a>
<ul>
<li class="chapter" data-level="3.1" data-path="github.html"><a href="github.html"><i class="fa fa-check"></i><b>3.1</b> Why use GitHub?<span></span></a>
<ul>
<li><a href="github.html#goals">1. Goals<span></span></a></li>
<li><a href="github.html#version">2. Watch course video: Version Control 101<span></span></a></li>
<li><a href="github.html#slides">3. Preview slides<span></span></a></li>
<li><a href="github.html#next-steps">4. Next steps<span></span></a></li>
</ul></li>
<li class="chapter" data-level="3.2" data-path="install.html"><a href="install.html"><i class="fa fa-check"></i><b>3.2</b> Installation<span></span></a>
<ul>
<li><a href="install.html#goals-1">1. Goals<span></span></a></li>
<li class="chapter" data-level="3.2.1" data-path="install.html"><a href="install.html#how-to-get-github-and-gitkraken-for-free"><i class="fa fa-check"></i><b>3.2.1</b> 2. How to get GitHub and GitKraken for free<span></span></a></li>
<li><a href="install.html#happygit">3. Happy Git with R<span></span></a></li>
<li><a href="install.html#gitvideos">4. Video tutorials: installation<span></span></a></li>
<li><a href="install.html#join">5. Joining your group’s project on GitHub<span></span></a></li>
<li><a href="install.html#repo">6. Setting up a group repo<span></span></a></li>
<li><a href="install.html#next-steps-1">7. Next steps<span></span></a></li>
</ul></li>
<li class="chapter" data-level="3.3" data-path="workflow.html"><a href="workflow.html"><i class="fa fa-check"></i><b>3.3</b> Workflows<span></span></a>
<ul>
<li><a href="workflow.html#goals-2">1. Goals<span></span></a></li>
<li><a href="workflow.html#step-by-step-tutorials">2. Step-by-step tutorials<span></span></a></li>
<li><a href="workflow.html#further-resources">3. Further resources<span></span></a></li>
</ul></li>
</ul></li>
<li class="part"><span><b>Basic Topics<span></span></b></span></li>
<li class="chapter" data-level="4" data-path="Week1.html"><a href="Week1.html"><i class="fa fa-check"></i><b>4</b> Week 1: Importing Genetic Data<span></span></a>
<ul>
<li class="chapter" data-level="4.1" data-path="video_1.html"><a href="video_1.html"><i class="fa fa-check"></i><b>4.1</b> View Course Video<span></span></a>
<ul>
<li><a href="video_1.html#embedded-video">1. Embedded Video<span></span></a></li>
<li><a href="video_1.html#preview-slides">2. Preview Slides<span></span></a></li>
</ul></li>
<li class="chapter" data-level="4.2" data-path="tutorial_1.html"><a href="tutorial_1.html"><i class="fa fa-check"></i><b>4.2</b> Interactive Tutorial 1<span></span></a>
<ul>
<li><a href="tutorial_1.html#list-of-r-commands-covered-this-week">1. List of R commands covered this week<span></span></a></li>
<li><a href="tutorial_1.html#general-instructions">2. General Instructions<span></span></a></li>
</ul></li>
<li class="chapter" data-level="4.3" data-path="WE_1.html"><a href="WE_1.html"><i class="fa fa-check"></i><b>4.3</b> Worked Example<span></span></a>
<ul>
<li><a href="WE_1.html#overview-of-worked-example">1. Overview of Worked Example<span></span></a></li>
<li><a href="WE_1.html#import-data-from-.csv-file">2. Import data from .csv file<span></span></a></li>
<li><a href="WE_1.html#create-genind-object">3. Create ‘genind’ Object<span></span></a></li>
<li><a href="WE_1.html#view-information-stored-in-genind-object">4. View information stored in ‘genind’ object<span></span></a></li>
<li><a href="WE_1.html#import-data-with-gstudio-package">5. Import data with ‘gstudio’ package<span></span></a></li>
<li><a href="WE_1.html#importing-snp-data">6. Importing SNP data<span></span></a></li>
</ul></li>
<li class="chapter" data-level="4.4" data-path="r-exercise-week-1.html"><a href="r-exercise-week-1.html"><i class="fa fa-check"></i><b>4.4</b> R exercise Week 1<span></span></a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="Week2.html"><a href="Week2.html"><i class="fa fa-check"></i><b>5</b> Week 2: Spatial Data<span></span></a>
<ul>
<li class="chapter" data-level="5.1" data-path="video_2.html"><a href="video_2.html"><i class="fa fa-check"></i><b>5.1</b> View Course Video<span></span></a>
<ul>
<li><a href="video_2.html#embedded-video-1">1. Embedded Video<span></span></a></li>
<li><a href="video_2.html#preview-slides-1">2. Preview Slides<span></span></a></li>
</ul></li>
<li class="chapter" data-level="5.2" data-path="tutorial_2.html"><a href="tutorial_2.html"><i class="fa fa-check"></i><b>5.2</b> Interactive Tutorial 2<span></span></a>
<ul>
<li><a href="tutorial_2.html#list-of-r-commands-covered-this-week-1">1. List of R commands covered this week<span></span></a></li>
<li><a href="tutorial_2.html#general-instructions-1">2. General Instructions<span></span></a></li>
</ul></li>
<li class="chapter" data-level="5.3" data-path="WE_2.html"><a href="WE_2.html"><i class="fa fa-check"></i><b>5.3</b> Worked Example<span></span></a>
<ul>
<li><a href="WE_2.html#overview-of-worked-example-1">1. Overview of Worked Example<span></span></a></li>
<li><a href="WE_2.html#import-site-data-from-.csv-file">2. Import site data from .csv file<span></span></a></li>
<li><a href="WE_2.html#display-raster-data-and-overlay-sampling-locations-extract-data">3. Display raster data and overlay sampling locations, extract data<span></span></a></li>
<li><a href="WE_2.html#calculate-landscape-metrics">4. Calculate landscape metrics<span></span></a></li>
<li><a href="WE_2.html#sample-landscape-metrics-within-buffer-around-sampling-locations">5. Sample landscape metrics within buffer around sampling locations<span></span></a></li>
</ul></li>
<li class="chapter" data-level="5.4" data-path="r-exercise-week-2.html"><a href="r-exercise-week-2.html"><i class="fa fa-check"></i><b>5.4</b> R Exercise Week 2<span></span></a></li>
<li class="chapter" data-level="5.5" data-path="bonus_2a.html"><a href="bonus_2a.html"><i class="fa fa-check"></i><b>5.5</b> Bonus: ‘sf’ and ‘terra’<span></span></a>
<ul>
<li><a href="bonus_2a.html#overview-2">1. Overview<span></span></a></li>
<li><a href="bonus_2a.html#convert-spatialpointsdataframe-to-sf-object">2. Convert ‘SpatialPointsDataFrame’ to ‘sf’ object<span></span></a></li>
<li><a href="bonus_2a.html#plot-variables-first-10-in-data-set">3. Plot variables (first 10 in data set)<span></span></a></li>
<li><a href="bonus_2a.html#export-site-data-as-esri-shapefile">4. Export site data as ESRI shapefile<span></span></a></li>
<li><a href="bonus_2a.html#use-terra-and-tmap-to-display-categorical-map-with-color-scheme">5. Use ‘terra’ and ‘tmap’ to display categorical map with color scheme<span></span></a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="6" data-path="Week3.html"><a href="Week3.html"><i class="fa fa-check"></i><b>6</b> Week 3: Genetic Diversity<span></span></a>
<ul>
<li class="chapter" data-level="6.1" data-path="video_3.html"><a href="video_3.html"><i class="fa fa-check"></i><b>6.1</b> View Course Video<span></span></a>
<ul>
<li><a href="video_3.html#embedded-video-2">1. Embedded Video<span></span></a></li>
<li><a href="video_3.html#video-part-1">Video, Part 1<span></span></a></li>
<li><a href="video_3.html#video-part-2">Video, Part 2<span></span></a></li>
<li><a href="video_3.html#preview-slides-2">Preview Slides<span></span></a></li>
</ul></li>
<li class="chapter" data-level="6.2" data-path="tutorial_3.html"><a href="tutorial_3.html"><i class="fa fa-check"></i><b>6.2</b> Interactive Tutorial 3<span></span></a>
<ul>
<li><a href="tutorial_3.html#list-of-r-commands-covered-this-week-2">1. List of R commands covered this week<span></span></a></li>
<li><a href="tutorial_3.html#general-instructions-2">2. General Instructions<span></span></a></li>
</ul></li>
<li class="chapter" data-level="6.3" data-path="WE_3.html"><a href="WE_3.html"><i class="fa fa-check"></i><b>6.3</b> Worked Example<span></span></a>
<ul>
<li><a href="WE_3.html#overview-of-worked-example-2">1. Overview of Worked Example<span></span></a></li>
<li><a href="WE_3.html#basic-checking-of-markers-and-populations">2. Basic checking of markers and populations<span></span></a></li>
<li><a href="WE_3.html#assess-genetic-diversity">3. Assess genetic diversity<span></span></a></li>
<li><a href="WE_3.html#aggregate-genetic-data-at-population-level-allele-frequencies">4. Aggregate genetic data at population level (allele frequencies)<span></span></a></li>
</ul></li>
<li class="chapter" data-level="6.4" data-path="r-exercise-week-3.html"><a href="r-exercise-week-3.html"><i class="fa fa-check"></i><b>6.4</b> R Exercise Week 3<span></span></a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="Week4.html"><a href="Week4.html"><i class="fa fa-check"></i><b>7</b> Week 4: Metapopulations<span></span></a>
<ul>
<li class="chapter" data-level="7.1" data-path="video_4.html"><a href="video_4.html"><i class="fa fa-check"></i><b>7.1</b> View Course Video<span></span></a>
<ul>
<li><a href="video_4.html#embedded-video-3">1. Embedded Video<span></span></a></li>
<li><a href="video_4.html#video-part-1-1">Video, Part 1<span></span></a></li>
<li><a href="video_4.html#video-part-2-1">Video, Part 2<span></span></a></li>
<li><a href="video_4.html#preview-slides-3">Preview Slides<span></span></a></li>
</ul></li>
<li class="chapter" data-level="7.2" data-path="tutorial_4.html"><a href="tutorial_4.html"><i class="fa fa-check"></i><b>7.2</b> Interactive Tutorial 4<span></span></a>
<ul>
<li><a href="tutorial_4.html#list-of-r-commands-covered-this-week-3">1. List of R commands covered this week<span></span></a></li>
<li><a href="tutorial_4.html#general-instructions-3">2. General Instructions<span></span></a></li>
</ul></li>
<li class="chapter" data-level="7.3" data-path="WE_4.html"><a href="WE_4.html"><i class="fa fa-check"></i><b>7.3</b> Worked Example<span></span></a>
<ul>
<li><a href="WE_4.html#overview-of-worked-example-3">1. Overview of Worked Example<span></span></a></li>
<li><a href="WE_4.html#spatial-distribution-of-genetic-structure">2. Spatial distribution of genetic structure<span></span></a></li>
<li><a href="WE_4.html#what-determines-genetic-differentiation-among-sites">3. What determines genetic differentiation among sites?<span></span></a></li>
<li><a href="WE_4.html#what-determines-genetic-diversity">4. What determines genetic diversity?<span></span></a></li>
<li><a href="WE_4.html#are-genetic-differentiation-and-diversity-related">5. Are genetic differentiation and diversity related?<span></span></a></li>
<li><a href="WE_4.html#effect-of-recent-extinction-events">6. Effect of recent extinction events<span></span></a></li>
</ul></li>
<li class="chapter" data-level="7.4" data-path="r-exercise-week-4.html"><a href="r-exercise-week-4.html"><i class="fa fa-check"></i><b>7.4</b> R Exercise Week 4<span></span></a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="Week5.html"><a href="Week5.html"><i class="fa fa-check"></i><b>8</b> Week 5: Spatial Statistics<span></span></a>
<ul>
<li class="chapter" data-level="8.1" data-path="video_5.html"><a href="video_5.html"><i class="fa fa-check"></i><b>8.1</b> View Course Video<span></span></a>
<ul>
<li><a href="video_5.html#embedded-video-4">1. Embedded Video<span></span></a></li>
<li><a href="video_5.html#preview-slides-4">Preview Slides<span></span></a></li>
</ul></li>
<li class="chapter" data-level="8.2" data-path="tutorial_5.html"><a href="tutorial_5.html"><i class="fa fa-check"></i><b>8.2</b> Interactive Tutorial 5<span></span></a>
<ul>
<li><a href="tutorial_5.html#list-of-r-commands-covered-this-week-4">1. List of R commands covered this week<span></span></a></li>
<li><a href="tutorial_5.html#general-instructions-4">2. General Instructions<span></span></a></li>
</ul></li>
<li class="chapter" data-level="8.3" data-path="WE_5.html"><a href="WE_5.html"><i class="fa fa-check"></i><b>8.3</b> Worked Example<span></span></a>
<ul>
<li><a href="WE_5.html#overview-of-worked-example-4">1. Overview of Worked Example<span></span></a></li>
<li><a href="WE_5.html#data-import-and-manipulation">2. Data import and manipulation<span></span></a></li>
<li><a href="WE_5.html#calculate-genetic-distances">3. Calculate genetic distances<span></span></a></li>
<li><a href="WE_5.html#perform-a-mantel-test-to-test-for-ibd">4. Perform a Mantel test to test for IBD<span></span></a></li>
<li><a href="WE_5.html#create-mantel-correlogram-for-genetic-data">5. Create Mantel correlogram for genetic data<span></span></a></li>
<li><a href="WE_5.html#specify-spatial-weights-and-calculate-morans-i">6. Specify spatial weights and calculate Moran’s I<span></span></a></li>
</ul></li>
<li class="chapter" data-level="8.4" data-path="r-exercise-week-5.html"><a href="r-exercise-week-5.html"><i class="fa fa-check"></i><b>8.4</b> R Exercise Week 5<span></span></a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="Week6.html"><a href="Week6.html"><i class="fa fa-check"></i><b>9</b> Week 6: Quantitative Genetics<span></span></a>
<ul>
<li class="chapter" data-level="9.1" data-path="video_6.html"><a href="video_6.html"><i class="fa fa-check"></i><b>9.1</b> View Course Video<span></span></a>
<ul>
<li><a href="video_6.html#embedded-video-5">1. Embedded Video<span></span></a></li>
<li><a href="video_6.html#video-part-1-2">Video, Part 1<span></span></a></li>
<li><a href="video_6.html#video-part-2-2">Video, Part 2<span></span></a></li>
<li><a href="video_6.html#preview-slides-5">Preview Slides<span></span></a></li>
</ul></li>
<li class="chapter" data-level="9.2" data-path="tutorial_6.html"><a href="tutorial_6.html"><i class="fa fa-check"></i><b>9.2</b> Interactive Tutorial 6<span></span></a>
<ul>
<li><a href="tutorial_6.html#list-of-r-commands-covered-this-week-5">1. List of R commands covered this week<span></span></a></li>
<li><a href="tutorial_6.html#general-instructions-5">2. General Instructions<span></span></a></li>
</ul></li>
<li class="chapter" data-level="9.3" data-path="WE_6.html"><a href="WE_6.html"><i class="fa fa-check"></i><b>9.3</b> Worked Example<span></span></a>
<ul>
<li><a href="WE_6.html#overview-of-worked-example-5">1. Overview of Worked Example<span></span></a></li>
<li><a href="WE_6.html#estimate-genetic-and-non-genetic-variance-components-from-a-common-garden-experiment">2. Estimate genetic and non-genetic variance components from a common garden experiment<span></span></a></li>
<li><a href="WE_6.html#estimate-trait-heritability">3. Estimate trait heritability<span></span></a></li>
<li><a href="WE_6.html#compare-q_st-to-f_st-measured-from-snp-data">4. Compare <span class="math inline">\(Q_{st}\)</span> to <span class="math inline">\(F_{st}\)</span> measured from SNP data<span></span></a></li>
<li><a href="WE_6.html#assess-correlation-between-trait-and-environment">5. Assess correlation between trait and environment<span></span></a></li>
</ul></li>
<li class="chapter" data-level="9.4" data-path="r-exercise-week-6.html"><a href="r-exercise-week-6.html"><i class="fa fa-check"></i><b>9.4</b> R Exercise Week 6<span></span></a></li>
</ul></li>
<li class="chapter" data-level="10" data-path="Week7.html"><a href="Week7.html"><i class="fa fa-check"></i><b>10</b> Week 7: Spatial Linear Models<span></span></a>
<ul>
<li class="chapter" data-level="10.1" data-path="video_7.html"><a href="video_7.html"><i class="fa fa-check"></i><b>10.1</b> View Course Video<span></span></a>
<ul>
<li><a href="video_7.html#embedded-video-6">1. Embedded Video<span></span></a></li>
<li><a href="video_7.html#video-part-1-3">Video, Part 1<span></span></a></li>
<li><a href="video_7.html#video-part-2-3">Video, Part 2<span></span></a></li>
<li><a href="video_7.html#preview-slides-6">Preview Slides<span></span></a></li>
</ul></li>
<li class="chapter" data-level="10.2" data-path="tutorial_7.html"><a href="tutorial_7.html"><i class="fa fa-check"></i><b>10.2</b> Interactive Tutorial 7<span></span></a>
<ul>
<li><a href="tutorial_7.html#list-of-r-commands-covered-this-week-6">1. List of R commands covered this week<span></span></a></li>
<li><a href="tutorial_7.html#general-instructions-6">2. General Instructions<span></span></a></li>
</ul></li>
<li class="chapter" data-level="10.3" data-path="WE_7.html"><a href="WE_7.html"><i class="fa fa-check"></i><b>10.3</b> Worked Example<span></span></a>
<ul>
<li><a href="WE_7.html#overview-of-worked-example-6">1. Overview of Worked Example<span></span></a></li>
<li><a href="WE_7.html#explore-data-set">2. Explore data set<span></span></a></li>
<li><a href="WE_7.html#test-regression-residuals-for-spatial-autocorrelation">3. Test regression residuals for spatial autocorrelation<span></span></a></li>
<li><a href="WE_7.html#fit-models-with-spatially-correlated-error-gls-with-package-nlme">4. Fit models with spatially correlated error (GLS) with package ‘nlme’<span></span></a></li>
<li><a href="WE_7.html#fit-spatial-simultaneous-autoregressive-error-models-sar">5. Fit spatial simultaneous autoregressive error models (SAR)<span></span></a></li>
<li><a href="WE_7.html#spatial-filtering-with-mem-using-package-spmoran">6. Spatial filtering with MEM using package ‘spmoran’<span></span></a></li>
<li><a href="WE_7.html#fit-spatially-varying-coefficients-model-with-package-spmoran">7. Fit spatially varying coefficients model with package ‘spmoran’<span></span></a></li>
<li><a href="WE_7.html#model-with-patchsize-and-ibr">Model with PatchSize and IBR<span></span></a></li>
<li><a href="WE_7.html#model-with-ibr-only">Model with IBR only<span></span></a></li>
<li><a href="WE_7.html#conclusions">8. Conclusions<span></span></a></li>
</ul></li>
<li class="chapter" data-level="10.4" data-path="r-exercise-week-7.html"><a href="r-exercise-week-7.html"><i class="fa fa-check"></i><b>10.4</b> R Exercise Week 7<span></span></a></li>
<li class="chapter" data-level="10.5" data-path="bonus_7a.html"><a href="bonus_7a.html"><i class="fa fa-check"></i><b>10.5</b> Bonus: Si index<span></span></a>
<ul>
<li><a href="bonus_7a.html#overview-of-bonus-materials">1. Overview of Bonus Materials<span></span></a></li>
<li><a href="bonus_7a.html#import-ecological-distance-matrices">2. Import ecological distance matrices<span></span></a></li>
<li><a href="bonus_7a.html#optimize-the-scaling-parameter-alpha">3. Optimize the scaling parameter alpha<span></span></a></li>
<li><a href="bonus_7a.html#calculate-hanskis-index-si-with-source-patch-parameters">4. Calculate Hanski’s index Si with source patch parameters<span></span></a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="11" data-path="Week8.html"><a href="Week8.html"><i class="fa fa-check"></i><b>11</b> Week 8: Simulation Experiments<span></span></a>
<ul>
<li class="chapter" data-level="11.1" data-path="video_8.html"><a href="video_8.html"><i class="fa fa-check"></i><b>11.1</b> View Course Video<span></span></a>
<ul>
<li><a href="video_8.html#embedded-video-7">1. Embedded Video<span></span></a></li>
<li><a href="video_8.html#video-part-1-4">Video, Part 1<span></span></a></li>
<li><a href="video_8.html#video-part-2-4">Video, Part 2<span></span></a></li>
<li><a href="video_8.html#preview-slides-7">Preview Slides<span></span></a></li>
</ul></li>
<li class="chapter" data-level="11.2" data-path="tutorial_8.html"><a href="tutorial_8.html"><i class="fa fa-check"></i><b>11.2</b> Interactive Tutorial 8<span></span></a>
<ul>
<li><a href="tutorial_8.html#list-of-r-commands-covered-this-week-7">1. List of R commands covered this week<span></span></a></li>
<li><a href="tutorial_8.html#general-instructions-7">2. General Instructions<span></span></a></li>
</ul></li>
<li class="chapter" data-level="11.3" data-path="WE_8.html"><a href="WE_8.html"><i class="fa fa-check"></i><b>11.3</b> Worked Example<span></span></a>
<ul>
<li><a href="WE_8.html#overview-of-worked-example-7">1. Overview of Worked Example<span></span></a></li>
<li><a href="WE_8.html#initialize-a-landscape">2. Initialize a landscape<span></span></a></li>
<li><a href="WE_8.html#define-simulation-parameters">3. Define simulation parameters<span></span></a></li>
<li><a href="WE_8.html#run-simulations-and-analyze-results">4. Run simulations and analyze results<span></span></a></li>
<li><a href="WE_8.html#run-simulator-using-a-previously-defined-parameter-set">5. Run simulator using a previously defined parameter set<span></span></a></li>
</ul></li>
<li class="chapter" data-level="11.4" data-path="r-exercise-week-8.html"><a href="r-exercise-week-8.html"><i class="fa fa-check"></i><b>11.4</b> R Exercise Week 8<span></span></a></li>
<li class="chapter" data-level="11.5" data-path="bonus_8a.html"><a href="bonus_8a.html"><i class="fa fa-check"></i><b>11.5</b> Bonus: Efficient R<span></span></a>
<ul>
<li><a href="bonus_8a.html#overview-of-bonus-material">1. Overview of Bonus Material<span></span></a></li>
<li><a href="bonus_8a.html#navigating-the-file-system">2. Navigating the file system<span></span></a></li>
<li><a href="bonus_8a.html#benchmarking-file-import-and-export-options">3. Benchmarking file import and export options<span></span></a></li>
<li><a href="bonus_8a.html#profiling-your-code">4. Profiling your code<span></span></a></li>
<li><a href="bonus_8a.html#creating-and-executing-a-bash-r-script">5. Creating and executing a Bash R script<span></span></a></li>
<li><a href="bonus_8a.html#parallelizing-code">6. Parallelizing code<span></span></a></li>
</ul></li>
</ul></li>
<li class="part"><span><b>Advanced Topics<span></span></b></span></li>
<li class="chapter" data-level="12" data-path="Week9.html"><a href="Week9.html"><i class="fa fa-check"></i><b>12</b> Week 9: Population Structure<span></span></a>
<ul>
<li class="chapter" data-level="12.1" data-path="WE_9.html"><a href="WE_9.html"><i class="fa fa-check"></i><b>12.1</b> Worked Example<span></span></a>
<ul>
<li><a href="WE_9.html#overview-of-worked-example-8">1. Overview of Worked Example<span></span></a></li>
<li><a href="WE_9.html#simulated-data-2-island-model">2. Simulated data: 2-island model<span></span></a></li>
<li><a href="WE_9.html#simulated-data-2-island-model-with-admixture">3. Simulated data: 2-island model with admixture<span></span></a></li>
<li><a href="WE_9.html#empirical-data-threespine-sticklebacks">4. Empirical data: Threespine sticklebacks<span></span></a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="13" data-path="Week10.html"><a href="Week10.html"><i class="fa fa-check"></i><b>13</b> Week 10: Landscape Resistance<span></span></a>
<ul>
<li class="chapter" data-level="13.1" data-path="WE_10.html"><a href="WE_10.html"><i class="fa fa-check"></i><b>13.1</b> Worked Example<span></span></a>
<ul>
<li><a href="WE_10.html#overview-of-worked-example-9">1. Overview of Worked Example<span></span></a></li>
<li><a href="WE_10.html#explore-the-data-set">2. Explore the data set<span></span></a></li>
<li><a href="WE_10.html#setting-cost-values-and-calculating-conductance">3. Setting cost values and calculating conductance<span></span></a></li>
<li><a href="WE_10.html#convert-conductance-into-effective-distance">4. Convert conductance into effective distance<span></span></a></li>
<li><a href="WE_10.html#create-cost-distance-matrices">5. Create cost-distance matrices<span></span></a></li>
<li><a href="WE_10.html#how-does-changing-resolution-affect-these-metrics">6. How does changing resolution affect these metrics?<span></span></a></li>
<li><a href="WE_10.html#references">7. References<span></span></a></li>
</ul></li>
<li class="chapter" data-level="13.2" data-path="bonus-10a.html"><a href="bonus-10a.html"><i class="fa fa-check"></i><b>13.2</b> Bonus: ‘radish’ tutorial<span></span></a>
<ul>
<li><a href="bonus-10a.html#overview-of-bonus-materials-1">1. Overview of Bonus Materials<span></span></a></li>
<li><a href="bonus-10a.html#import-data">2. Import data<span></span></a></li>
<li><a href="bonus-10a.html#basic-model-fitting">3. Basic model fitting<span></span></a></li>
<li><a href="bonus-10a.html#advanced-modeling-topics">4. Advanced modeling topics<span></span></a></li>
<li><a href="bonus-10a.html#summary">5. Summary<span></span></a></li>
<li><a href="bonus-10a.html#references-1">6. References<span></span></a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="14" data-path="Week11.html"><a href="Week11.html"><i class="fa fa-check"></i><b>14</b> Week 11: Detecting Adaptation<span></span></a>
<ul>
<li class="chapter" data-level="14.1" data-path="WE_11.html"><a href="WE_11.html"><i class="fa fa-check"></i><b>14.1</b> Worked Example<span></span></a>
<ul>
<li><a href="WE_11.html#overview-of-worked-example-10">1. Overview of Worked Example<span></span></a></li>
<li><a href="WE_11.html#import-and-prepare-the-data">2. Import and prepare the data<span></span></a></li>
<li><a href="WE_11.html#latent-factor-mixed-models-lfmm-a-univariate-gea">3. Latent Factor Mixed Models (LFMM): a univariate GEA<span></span></a></li>
<li><a href="WE_11.html#redundancy-analysis-rda-a-multivariate-gea">4. Redundancy Analysis (RDA): a multivariate GEA<span></span></a></li>
<li><a href="WE_11.html#compare-lfmm-and-rda-candidates">5. Compare LFMM and RDA candidates<span></span></a></li>
<li><a href="WE_11.html#a-quick-note-on-controlling-for-population-structure-in-rda">6. A quick note on controlling for population structure in RDA<span></span></a></li>
<li><a href="WE_11.html#references-2">7. References<span></span></a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="15" data-path="Week12.html"><a href="Week12.html"><i class="fa fa-check"></i><b>15</b> Week 12: Model Selection<span></span></a>
<ul>
<li class="chapter" data-level="15.1" data-path="WE_12.html"><a href="WE_12.html"><i class="fa fa-check"></i><b>15.1</b> Worked Example<span></span></a>
<ul>
<li><a href="WE_12.html#overview-of-worked-example-11">1. Overview of Worked Example<span></span></a></li>
<li><a href="WE_12.html#fitting-candidate-models">2. Fitting candidate models<span></span></a></li>
<li><a href="WE_12.html#your-turn-to-test-additional-hypotheses">3. Your turn to test additional hypotheses!<span></span></a></li>
<li><a href="WE_12.html#nested-model-nmlpe-for-hierarchical-sampling-designs">4. Nested model (NMLPE) for hierarchical sampling designs<span></span></a></li>
<li><a href="WE_12.html#references-3">5. References<span></span></a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="16" data-path="Week13.html"><a href="Week13.html"><i class="fa fa-check"></i><b>16</b> Week 13: Gravity Models<span></span></a>
<ul>
<li class="chapter" data-level="16.1" data-path="WE_13.html"><a href="WE_13.html"><i class="fa fa-check"></i><b>16.1</b> Worked Example<span></span></a>
<ul>
<li><a href="WE_13.html#overview-of-worked-example-12">1. Overview of Worked Example<span></span></a></li>
<li><a href="WE_13.html#setup">2. Setup<span></span></a></li>
<li><a href="WE_13.html#wetland-complex-data-preparation">3. Wetland complex data preparation<span></span></a></li>
<li><a href="WE_13.html#wetland-field-data-preparation">4. Wetland field-data preparation<span></span></a></li>
<li><a href="WE_13.html#saturated-graph">5. Saturated Graph<span></span></a></li>
<li><a href="WE_13.html#spatial-model-data-prepration">6. Spatial model data prepration<span></span></a></li>
<li><a href="WE_13.html#gravity-model">7. Gravity model<span></span></a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="17" data-path="Week14.html"><a href="Week14.html"><i class="fa fa-check"></i><b>17</b> Week 14: Contemporary Gene Flow<span></span></a>
<ul>
<li class="chapter" data-level="17.1" data-path="WE_14.html"><a href="WE_14.html"><i class="fa fa-check"></i><b>17.1</b> Worked Example<span></span></a>
<ul>
<li><a href="WE_14.html#overview-3">1. Overview<span></span></a></li>
<li><a href="WE_14.html#import-genotypes">2. Import Genotypes<span></span></a></li>
<li><a href="WE_14.html#pollen-pool-genetic-structure">3. Pollen pool genetic structure<span></span></a></li>
<li><a href="WE_14.html#paternity-analysis">4. Paternity analysis<span></span></a></li>
<li><a href="WE_14.html#linking-paternity-to-ecological-variables">5. Linking paternity to ecological variables<span></span></a></li>
<li><a href="WE_14.html#references-4">6. References<span></span></a></li>
</ul></li>
</ul></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Landscape Genetic Data Analysis with R</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="bonus_7a" class="section level2 hasAnchor" number="10.5">
<h2><span class="header-section-number">10.5</span> Bonus: Si index<a href="bonus_7a.html#bonus_7a" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Yessica Rico and Helene Wagner</p>
<div id="overview-of-bonus-materials" class="section level3 unnumbered hasAnchor">
<h3>1. Overview of Bonus Materials<a href="bonus_7a.html#overview-of-bonus-materials" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>The Week 7 Worked Example used patch-level values of an Si index as predictors. Here we show how these were calculated from distance matrices. The main steps are:</p>
<ul>
<li>Import ecological distance matrices</li>
<li>Optimize the scaling parameter alpha</li>
<li>Calculate Hanski’s index Si with source patch parameters</li>
</ul>
<p>The following data will be imported:</p>
<ul>
<li><strong>dModels.rds</strong>: a list of 5 distance matrices (stored as <code>dist</code> objects) among all 106 calcareous grassland patches in the study area. Each distance matrix represents a biological hypothesis about functional connectivity of <em>D. carthusianorum</em> (see below).</li>
<li><strong>PATCH_XY_Dianthus.csv</strong>: data frame with patch attributes for all 106 calcareous grassland patches in the study area. We will use the following attributes:
<ul>
<li><strong>Dc.89, Dc.09</strong>: binary variables indicating whether or not <em>D. carthusianorum</em> was recorded in the patch in the first survey in 1989, or the second survey in 2009, respectively.</li>
<li><strong>Ha</strong>: patch area in ha.</li>
<li><strong>pop09</strong>: population size in 2009 (categorical, 4 levels, see below).</li>
</ul></li>
</ul>
<div class="sourceCode" id="cb891"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb891-1"><a href="bonus_7a.html#cb891-1" aria-hidden="true" tabindex="-1"></a><span class="co">#library(tibble)</span></span>
<span id="cb891-2"><a href="bonus_7a.html#cb891-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(dplyr)</span>
<span id="cb891-3"><a href="bonus_7a.html#cb891-3" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(LandGenCourse)</span></code></pre></div>
</div>
<div id="import-ecological-distance-matrices" class="section level3 unnumbered hasAnchor">
<h3>2. Import ecological distance matrices<a href="bonus_7a.html#import-ecological-distance-matrices" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Each distance matrix represents one hypothesis about gene flow in this system (see Week 7 video).</p>
<ul>
<li><strong>Eu</strong>: Euclidean (geographic) distance, representing IBD</li>
<li><strong>Shecte</strong>: Connectivity by shepherding among consistently grazed patches only, distance-dependent: this counts the number of patches that sheep traverse to get from patch A to patch B.</li>
<li><strong>Sheint</strong>: Connectivity by shepherding, among consistently or intermittently grazed patches, distance-dependent: again, this counts the number of patches that sheep traverse to get from patch A to patch B.</li>
<li><strong>Shenu</strong>: Connectivity by shepherding, no distance effect. This considers only whether or not two patches are part of the same grazing system.<br />
</li>
<li><strong>Forest</strong>: This model is similar to Eu but considers forest as a barrier.</li>
</ul>
<p>Here, we import these distance matrices as a list of <code>dist</code> objects and check their names and dimensions: each has 106 rows and columns.</p>
<div class="sourceCode" id="cb892"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb892-1"><a href="bonus_7a.html#cb892-1" aria-hidden="true" tabindex="-1"></a>dModels <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="fu">system.file</span>(<span class="st">&quot;extdata&quot;</span>, <span class="st">&quot;dModels.rds&quot;</span>, </span>
<span id="cb892-2"><a href="bonus_7a.html#cb892-2" aria-hidden="true" tabindex="-1"></a>                            <span class="at">package =</span> <span class="st">&quot;LandGenCourse&quot;</span>)) </span>
<span id="cb892-3"><a href="bonus_7a.html#cb892-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb892-4"><a href="bonus_7a.html#cb892-4" aria-hidden="true" tabindex="-1"></a><span class="fu">lapply</span>(dModels, <span class="cf">function</span>(ls) <span class="fu">dim</span>(<span class="fu">as.matrix</span>(ls)))</span></code></pre></div>
<pre><code>## $Eu
## [1] 106 106
## 
## $Shecte
## [1] 106 106
## 
## $Sheint
## [1] 106 106
## 
## $Shenu
## [1] 106 106
## 
## $Forest
## [1] 106 106</code></pre>
<p>Note: the distance matrices contain 106 patches. All of these are considered as source populations. However, we will only calculate Si indices for the 65 patches included in the Dianthus dataset.</p>
</div>
<div id="optimize-the-scaling-parameter-alpha" class="section level3 unnumbered hasAnchor">
<h3>3. Optimize the scaling parameter alpha<a href="bonus_7a.html#optimize-the-scaling-parameter-alpha" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>In order to calculate Hanski’s index, we first need to optimize the value of alpha for each distance model, using presence-absence data for the two time steps of 1989 and 2009.</p>
<p>Import the patch-level data for all 106 patches in the study area.</p>
<div class="sourceCode" id="cb894"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb894-1"><a href="bonus_7a.html#cb894-1" aria-hidden="true" tabindex="-1"></a>Patches <span class="ot">&lt;-</span> <span class="fu">read.csv</span>(<span class="fu">system.file</span>(<span class="st">&quot;extdata&quot;</span>, <span class="st">&quot;PATCH_XY_Dianthus.csv&quot;</span>, </span>
<span id="cb894-2"><a href="bonus_7a.html#cb894-2" aria-hidden="true" tabindex="-1"></a>                            <span class="at">package =</span> <span class="st">&quot;LandGenCourse&quot;</span>),</span>
<span id="cb894-3"><a href="bonus_7a.html#cb894-3" aria-hidden="true" tabindex="-1"></a>         <span class="at">header=</span><span class="cn">TRUE</span>, <span class="at">row.names=</span><span class="dv">1</span>)</span>
<span id="cb894-4"><a href="bonus_7a.html#cb894-4" aria-hidden="true" tabindex="-1"></a>tibble<span class="sc">::</span><span class="fu">as_tibble</span>(Patches)</span></code></pre></div>
<pre><code>## # A tibble: 106 × 20
##    patch    id Elements2008 Elements2008.4 cat   type      Ha   area        x
##    &lt;chr&gt; &lt;int&gt;        &lt;int&gt;          &lt;int&gt; &lt;chr&gt; &lt;chr&gt;  &lt;dbl&gt;  &lt;dbl&gt;    &lt;dbl&gt;
##  1 A01       1            1              1 3     A     0.136  0.0014 4431470.
##  2 A02       2            2              2 4     B     0.182  0.0018 4431614.
##  3 A03       3            3              3 4     B     0.206  0.0021 4431320.
##  4 A04       4            0              0 4     B     0.142  0.0014 4431246.
##  5 A05       5            2              2 4     B     0.134  0.0013 4431037.
##  6 A06       6            2              2 4     B     0.201  0.002  4431027.
##  7 A07       7            0              0 3     N     0.0213 0.0002 4431284.
##  8 A07a      8            1              1 4     A     0.200  0.002  4431362.
##  9 A08       9            3              2 5     F     0.464  0.0046 4428814.
## 10 A09      10            4              3 3     F     0.0704 0.0007 4428809.
## # … with 96 more rows, and 11 more variables: y &lt;dbl&gt;, size.cat &lt;int&gt;,
## #   sizeCat &lt;chr&gt;, size.cat2 &lt;chr&gt;, pj08 &lt;dbl&gt;, pj89 &lt;dbl&gt;, pop09 &lt;int&gt;,
## #   grazing &lt;int&gt;, Dc.89 &lt;int&gt;, Dc.09 &lt;int&gt;, Sampled &lt;int&gt;</code></pre>
<p>Check that the patch IDs match between <code>Patches</code> and the distance matrices. Here we tabulate the number of rows for which the patch names match perfectly. This is the case for all 106 rows, which is great.</p>
<div class="sourceCode" id="cb896"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb896-1"><a href="bonus_7a.html#cb896-1" aria-hidden="true" tabindex="-1"></a><span class="fu">table</span>(Patches<span class="sc">$</span>patch <span class="sc">==</span> <span class="fu">rownames</span>(dModels<span class="sc">$</span>Eu))</span></code></pre></div>
<pre><code>## 
## FALSE  TRUE 
##   105     1</code></pre>
<p>The function ‘get.alphafit’ optimizes alpha for each model and stores the values in table ’table.alpha.</p>
<ul>
<li>First, we create an empty matrix (with a single column and five rows, one per model) to hold the optimized alpha value for each model (i.e., for each distance matrix).</li>
<li>We define the list of values of alpha that should be considered. Here we will use <code>nseq=100</code> values between 0.1 and 2.5.</li>
<li>Then we define the function get.alphafit that will take four arguments:
<ul>
<li><code>alpha</code>: scaling parameter of the exponential function</li>
<li><code>d</code>: distance matrix</li>
<li><code>pj</code>: whether or not the species was recorded in a patch in the second survey.</li>
<li><code>Op</code>: number of surveys (out of two) that the species was recorded in the patch.</li>
</ul></li>
</ul>
<div class="sourceCode" id="cb898"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb898-1"><a href="bonus_7a.html#cb898-1" aria-hidden="true" tabindex="-1"></a>table.alpha <span class="ot">&lt;-</span> <span class="fu">matrix</span>(<span class="cn">NA</span>, <span class="at">nrow=</span><span class="dv">5</span>)</span>
<span id="cb898-2"><a href="bonus_7a.html#cb898-2" aria-hidden="true" tabindex="-1"></a><span class="fu">dimnames</span>(table.alpha) <span class="ot">&lt;-</span> <span class="fu">list</span>(<span class="at">models=</span>(<span class="fu">names</span>(dModels)))</span>
<span id="cb898-3"><a href="bonus_7a.html#cb898-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb898-4"><a href="bonus_7a.html#cb898-4" aria-hidden="true" tabindex="-1"></a>nseq <span class="ot">=</span> <span class="dv">100</span></span>
<span id="cb898-5"><a href="bonus_7a.html#cb898-5" aria-hidden="true" tabindex="-1"></a>alpha <span class="ot">=</span> <span class="fu">seq</span>(<span class="fl">0.1</span>,<span class="fl">2.5</span>, <span class="at">length =</span> nseq)</span>
<span id="cb898-6"><a href="bonus_7a.html#cb898-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb898-7"><a href="bonus_7a.html#cb898-7" aria-hidden="true" tabindex="-1"></a>get.alphafit <span class="ot">&lt;-</span> <span class="cf">function</span>(alpha, d, pj, Op)</span>
<span id="cb898-8"><a href="bonus_7a.html#cb898-8" aria-hidden="true" tabindex="-1"></a>{</span>
<span id="cb898-9"><a href="bonus_7a.html#cb898-9" aria-hidden="true" tabindex="-1"></a>  expo<span class="ot">&lt;-</span><span class="fu">exp</span>(<span class="sc">-</span>alpha<span class="sc">*</span> d )</span>
<span id="cb898-10"><a href="bonus_7a.html#cb898-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">diag</span>(expo)<span class="ot">&lt;-</span><span class="dv">0</span></span>
<span id="cb898-11"><a href="bonus_7a.html#cb898-11" aria-hidden="true" tabindex="-1"></a>  matr<span class="ot">&lt;-</span><span class="fu">sweep</span>(expo,<span class="dv">2</span>, pj, <span class="st">&quot;*&quot;</span>)</span>
<span id="cb898-12"><a href="bonus_7a.html#cb898-12" aria-hidden="true" tabindex="-1"></a>  Si <span class="ot">&lt;-</span><span class="fu">rowSums</span>(<span class="fu">sweep</span>(matr, <span class="dv">2</span>, Op<span class="sc">/</span><span class="dv">2</span>, <span class="st">&quot;*&quot;</span>), <span class="at">na.rm=</span><span class="cn">TRUE</span>)</span>
<span id="cb898-13"><a href="bonus_7a.html#cb898-13" aria-hidden="true" tabindex="-1"></a>  mod<span class="ot">&lt;-</span> <span class="fu">glm</span>(<span class="fu">cbind</span>(Op,<span class="dv">2</span> <span class="sc">-</span>Op) <span class="sc">~</span> Si, <span class="at">family=</span>binomial)</span>
<span id="cb898-14"><a href="bonus_7a.html#cb898-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">deviance</span>(mod)</span>
<span id="cb898-15"><a href="bonus_7a.html#cb898-15" aria-hidden="true" tabindex="-1"></a>}</span></code></pre></div>
<p>Apply the function to optimize alpha.</p>
<div class="sourceCode" id="cb899"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb899-1"><a href="bonus_7a.html#cb899-1" aria-hidden="true" tabindex="-1"></a>pj <span class="ot">&lt;-</span> Patches<span class="sc">$</span>Dc<span class="fl">.09</span></span>
<span id="cb899-2"><a href="bonus_7a.html#cb899-2" aria-hidden="true" tabindex="-1"></a>Op <span class="ot">&lt;-</span> (Patches<span class="sc">$</span>Dc<span class="fl">.89</span> <span class="sc">+</span> Patches<span class="sc">$</span>Dc<span class="fl">.09</span>)</span>
<span id="cb899-3"><a href="bonus_7a.html#cb899-3" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span>(m <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(dModels))</span>
<span id="cb899-4"><a href="bonus_7a.html#cb899-4" aria-hidden="true" tabindex="-1"></a>{</span>
<span id="cb899-5"><a href="bonus_7a.html#cb899-5" aria-hidden="true" tabindex="-1"></a>  table.alpha[m] <span class="ot">&lt;-</span> (<span class="fu">optimize</span>(get.alphafit, <span class="at">interval=</span>alpha,</span>
<span id="cb899-6"><a href="bonus_7a.html#cb899-6" aria-hidden="true" tabindex="-1"></a>                              <span class="at">d=</span><span class="fu">as.matrix</span>(dModels[[m]]), pj, Op)<span class="sc">$</span>minimum)</span>
<span id="cb899-7"><a href="bonus_7a.html#cb899-7" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb899-8"><a href="bonus_7a.html#cb899-8" aria-hidden="true" tabindex="-1"></a>table.alpha</span></code></pre></div>
<pre><code>##         
## models        [,1]
##   Eu     2.1708021
##   Shecte 0.6195857
##   Sheint 1.2055182
##   Shenu  2.4999427
##   Forest 2.4999322</code></pre>
<p>These are the optimized scaling parameters alpha of the distance term (related to dispersal ability) in the incidence function <span class="math inline">\(S_{i}\)</span>, where <span class="math inline">\(p_{j}\)</span> is 1 if the species is present in source population j and 0 if it is absent, and <span class="math inline">\(d_{ij}\)</span> is the pairwise distance between source patch j and focal patch i:</p>
<p><span class="math inline">\(S_{i}\)</span> = <span class="math inline">\(\sum_{j\neq i} exp(-\alpha d_{ij})p_{j}\)</span></p>
</div>
<div id="calculate-hanskis-index-si-with-source-patch-parameters" class="section level3 unnumbered hasAnchor">
<h3>4. Calculate Hanski’s index Si with source patch parameters<a href="bonus_7a.html#calculate-hanskis-index-si-with-source-patch-parameters" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>We will consider there alternative source patch parameters</p>
<ul>
<li><span class="math inline">\(p_{j}\)</span>: whether or not the species was observed in 2009.</li>
<li><span class="math inline">\(A_{j}\)</span>: area of the patch in ha.</li>
<li><span class="math inline">\(N_{j}\)</span>: indicator of population size. This was recorded in the field in four ordinal categories:
<ul>
<li>1: &lt;4 (too few to treat as population)</li>
<li>2: 4-40 (small enough for complete sampling)</li>
<li>3: 41-100 (small enough to count)</li>
<li>4: &gt;100 (very large)</li>
</ul></li>
</ul>
<p>Note that here, <span class="math inline">\(N_{j}\)</span> is treated as a numeric variable. While this is an over-interpretation, though preliminary analyses showed that this linearized the relationships between variables.</p>
<div class="sourceCode" id="cb901"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb901-1"><a href="bonus_7a.html#cb901-1" aria-hidden="true" tabindex="-1"></a>pj <span class="ot">&lt;-</span> Patches<span class="sc">$</span>Dc<span class="fl">.09</span></span>
<span id="cb901-2"><a href="bonus_7a.html#cb901-2" aria-hidden="true" tabindex="-1"></a>Aj <span class="ot">&lt;-</span> Patches<span class="sc">$</span>Ha</span>
<span id="cb901-3"><a href="bonus_7a.html#cb901-3" aria-hidden="true" tabindex="-1"></a>Nj <span class="ot">&lt;-</span> Patches<span class="sc">$</span>pop09</span></code></pre></div>
<p>Compile three alternative source patch parameters for each patch. Note that we set the area Aj to zero of the species was not recorded in the patch. This is done by multiplication with pj.</p>
<div class="sourceCode" id="cb902"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb902-1"><a href="bonus_7a.html#cb902-1" aria-hidden="true" tabindex="-1"></a>Source<span class="ot">&lt;-</span> <span class="fu">data.matrix</span>(<span class="fu">data.frame</span>(<span class="at">pj=</span>pj, <span class="at">Aj=</span>pj<span class="sc">*</span>Aj, <span class="at">Nj=</span>Nj))</span>
<span id="cb902-2"><a href="bonus_7a.html#cb902-2" aria-hidden="true" tabindex="-1"></a>mod <span class="ot">&lt;-</span><span class="fu">rep</span>(<span class="dv">1</span><span class="sc">:</span><span class="dv">5</span>, <span class="fu">rep</span>(<span class="dv">3</span>,<span class="dv">5</span>))</span></code></pre></div>
<p>Prepare an empty table Si to hold the connectivity indices <span class="math inline">\(S_{i}\)</span>, one for each combination of focal patch <em>i</em> (rows), distance model <span class="math inline">\(d_{ij}\)</span> and source patch parameter (15 columns: 5 distances x 3 parameters).</p>
<div class="sourceCode" id="cb903"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb903-1"><a href="bonus_7a.html#cb903-1" aria-hidden="true" tabindex="-1"></a>Si <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="fu">matrix</span>(<span class="cn">NA</span>,<span class="fu">nrow</span>(Patches),<span class="at">ncol=</span><span class="dv">15</span>))</span>
<span id="cb903-2"><a href="bonus_7a.html#cb903-2" aria-hidden="true" tabindex="-1"></a>     <span class="fu">dimnames</span>(Si) <span class="ot">&lt;-</span> <span class="fu">list</span>(<span class="fu">row.names</span>(Patches),</span>
<span id="cb903-3"><a href="bonus_7a.html#cb903-3" aria-hidden="true" tabindex="-1"></a>     <span class="fu">paste</span>(<span class="fu">rep</span>(<span class="fu">names</span>(dModels), <span class="fu">rep</span>(<span class="dv">3</span>,<span class="dv">5</span>)), </span>
<span id="cb903-4"><a href="bonus_7a.html#cb903-4" aria-hidden="true" tabindex="-1"></a>           <span class="fu">rep</span>(<span class="fu">c</span>(<span class="st">&quot;pj&quot;</span>, <span class="st">&quot;Aj&quot;</span>, <span class="st">&quot;Nj&quot;</span>),<span class="dv">5</span>), <span class="at">sep=</span><span class="st">&quot;_&quot;</span>))</span></code></pre></div>
<p>Define function <code>get.Si</code> with three arguments:</p>
<ul>
<li><code>alpha</code>: the optimized alpha value from above</li>
<li><code>d</code>: the distance matrix</li>
<li><code>Ap</code>: the source patch parameter to be used.</li>
</ul>
<div class="sourceCode" id="cb904"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb904-1"><a href="bonus_7a.html#cb904-1" aria-hidden="true" tabindex="-1"></a>get.Si <span class="ot">&lt;-</span> <span class="cf">function</span>(alpha, d, Ap)</span>
<span id="cb904-2"><a href="bonus_7a.html#cb904-2" aria-hidden="true" tabindex="-1"></a>{</span>
<span id="cb904-3"><a href="bonus_7a.html#cb904-3" aria-hidden="true" tabindex="-1"></a>  expo<span class="ot">&lt;-</span><span class="fu">exp</span>(<span class="sc">-</span>alpha<span class="sc">*</span>d)</span>
<span id="cb904-4"><a href="bonus_7a.html#cb904-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">diag</span>(expo)<span class="ot">&lt;-</span><span class="dv">0</span></span>
<span id="cb904-5"><a href="bonus_7a.html#cb904-5" aria-hidden="true" tabindex="-1"></a>  matr<span class="ot">&lt;-</span><span class="fu">sweep</span>(expo,<span class="dv">2</span>, Ap, <span class="st">&quot;*&quot;</span>)</span>
<span id="cb904-6"><a href="bonus_7a.html#cb904-6" aria-hidden="true" tabindex="-1"></a>  S <span class="ot">&lt;-</span> <span class="fu">rowSums</span>(<span class="fu">sweep</span>(matr, <span class="dv">2</span>, Op<span class="sc">/</span><span class="dv">2</span>, <span class="st">&quot;*&quot;</span>), <span class="at">na.rm=</span><span class="cn">TRUE</span>)</span>
<span id="cb904-7"><a href="bonus_7a.html#cb904-7" aria-hidden="true" tabindex="-1"></a>}</span></code></pre></div>
<p>Apply function <code>get.Si</code> to calculate <code>Si</code> values. <code>sb</code> indicates which column of <code>Ap</code> to use as source patch parameter.</p>
<div class="sourceCode" id="cb905"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb905-1"><a href="bonus_7a.html#cb905-1" aria-hidden="true" tabindex="-1"></a>sb <span class="ot">&lt;-</span> <span class="fu">rep</span>(<span class="dv">1</span><span class="sc">:</span><span class="dv">3</span>,<span class="dv">5</span>)</span>
<span id="cb905-2"><a href="bonus_7a.html#cb905-2" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (n <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">ncol</span>(Si))</span>
<span id="cb905-3"><a href="bonus_7a.html#cb905-3" aria-hidden="true" tabindex="-1"></a>{</span>
<span id="cb905-4"><a href="bonus_7a.html#cb905-4" aria-hidden="true" tabindex="-1"></a>  Si[,n] <span class="ot">&lt;-</span> <span class="fu">get.Si</span>(<span class="at">alpha=</span>table.alpha[mod[n]], </span>
<span id="cb905-5"><a href="bonus_7a.html#cb905-5" aria-hidden="true" tabindex="-1"></a>                   <span class="at">d=</span><span class="fu">as.matrix</span>(dModels[[mod[n]]]),</span>
<span id="cb905-6"><a href="bonus_7a.html#cb905-6" aria-hidden="true" tabindex="-1"></a>                   <span class="at">Ap=</span>Source[,sb[n]])</span>
<span id="cb905-7"><a href="bonus_7a.html#cb905-7" aria-hidden="true" tabindex="-1"></a>}</span></code></pre></div>
<p>Table with results: values of Si. Column names are a combination of distance matrix and source patch parameter.</p>
<div class="sourceCode" id="cb906"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb906-1"><a href="bonus_7a.html#cb906-1" aria-hidden="true" tabindex="-1"></a>tibble<span class="sc">::</span><span class="fu">as_tibble</span>(Si)  </span></code></pre></div>
<pre><code>## # A tibble: 106 × 15
##    Eu_pj Eu_Aj Eu_Nj Shecte_pj Shecte_Aj Shecte_Nj Sheint_pj Sheint_Aj Sheint_Nj
##    &lt;dbl&gt; &lt;dbl&gt; &lt;dbl&gt;     &lt;dbl&gt;     &lt;dbl&gt;     &lt;dbl&gt;     &lt;dbl&gt;     &lt;dbl&gt;     &lt;dbl&gt;
##  1 0.636 1.21   1.76  8.58e-26  1.42e-25  2.47e-25  3.07e-51  5.07e-51  8.83e-51
##  2 0.626 0.616  1.58  8.52e-26  1.42e-25  2.46e-25  3.05e-51  5.06e-51  8.79e-51
##  3 0.943 0.638  2.32  8.52e-26  1.42e-25  2.46e-25  6.20e- 1  2.27e+ 0  2.05e+ 0
##  4 1.27  0.847  3.30  8.52e-26  1.42e-25  2.46e-25  5.02e- 1  7.71e- 1  1.30e+ 0
##  5 1.50  1.13   4.10  1.26e+ 0  3.41e+ 0  4.21e+ 0  5.04e- 1  4.09e- 1  1.29e+ 0
##  6 1.67  1.44   4.91  2.13e+ 0  6.27e+ 0  7.40e+ 0  7.42e- 1  1.19e+ 0  2.43e+ 0
##  7 2.12  1.98   6.50  8.58e-26  1.42e-25  2.47e-25  3.07e-51  5.07e-51  8.83e-51
##  8 1.21  1.62   3.82  2.00e+ 0  6.83e+ 0  7.11e+ 0  8.76e- 1  1.60e+ 0  2.87e+ 0
##  9 1.60  2.99   4.41  8.52e-26  1.41e-25  2.46e-25  1.00e+ 0  2.56e+ 0  3.32e+ 0
## 10 1.82  2.14   4.38  8.52e-26  1.42e-25  2.46e-25  3.28e- 1  3.53e- 1  7.79e- 1
## # … with 96 more rows, and 6 more variables: Shenu_pj &lt;dbl&gt;, Shenu_Aj &lt;dbl&gt;,
## #   Shenu_Nj &lt;dbl&gt;, Forest_pj &lt;dbl&gt;, Forest_Aj &lt;dbl&gt;, Forest_Nj &lt;dbl&gt;</code></pre>
<p>Note: the dataset <code>Dianthus</code> used in the Week 7 Worked Example only contains Si values for 65 patches with genetic data (i.e., the patches where <em>D. carthusianorum</em> was observed, and sampled, during the second survey 2009).</p>

</div>
</div>
<!-- </div> -->
            </section>

          </div>
        </div>
      </div>
<a href="r-exercise-week-7.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="Week8.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": true,
"facebook": false,
"twitter": false,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": false
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": null,
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "section"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
