<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>5.5 Bonus: ‘sf’ and ‘terra’ | Landscape Genetic Data Analysis with R</title>
  <meta name="description" content="This is a web-interface to the teaching materials for the lab course ‘Landscape Genetic Data Analysis with R’ associated with the distributed graduate course ‘DGS Landscape Genetics.’ The output format is bookdown::gitbook." />
  <meta name="generator" content="bookdown 0.25 and GitBook 2.6.7" />

  <meta property="og:title" content="5.5 Bonus: ‘sf’ and ‘terra’ | Landscape Genetic Data Analysis with R" />
  <meta property="og:type" content="book" />
  
  <meta property="og:description" content="This is a web-interface to the teaching materials for the lab course ‘Landscape Genetic Data Analysis with R’ associated with the distributed graduate course ‘DGS Landscape Genetics.’ The output format is bookdown::gitbook." />
  <meta name="github-repo" content="hhwagner1/LandGenCourse_book" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="5.5 Bonus: ‘sf’ and ‘terra’ | Landscape Genetic Data Analysis with R" />
  
  <meta name="twitter:description" content="This is a web-interface to the teaching materials for the lab course ‘Landscape Genetic Data Analysis with R’ associated with the distributed graduate course ‘DGS Landscape Genetics.’ The output format is bookdown::gitbook." />
  

<meta name="author" content="Editor: Helene Wagner (University of Toronto)" />


<meta name="date" content="2022-04-03" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="r-exercise-week-2.html"/>
<link rel="next" href="Week3.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>
<script src="libs/kePrint-0.0.1/kePrint.js"></script>
<link href="libs/lightable-0.0.1/lightable.css" rel="stylesheet" />
<script src="libs/htmlwidgets-1.5.4/htmlwidgets.js"></script>
<script src="libs/d3-3.5.6/d3.min.js"></script>
<link href="libs/profvis-0.3.6.9000/profvis.css" rel="stylesheet" />
<script src="libs/profvis-0.3.6.9000/profvis.js"></script>
<script src="libs/profvis-0.3.6.9000/scroll.js"></script>
<link href="libs/highlight-6.2.0/textmate.css" rel="stylesheet" />
<script src="libs/highlight-6.2.0/highlight.js"></script>
<script src="libs/profvis-binding-0.3.7/profvis.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>


<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">Landscape Genetic Data Analysis with R</a></li>

<li class="divider"></li>
<li class="part"><span><b>Getting Started<span></span></b></span></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Introduction<span></span></a>
<ul>
<li class="chapter" data-level="1.1" data-path="how-to-use-this-book.html"><a href="how-to-use-this-book.html"><i class="fa fa-check"></i><b>1.1</b> How to use this Book<span></span></a>
<ul>
<li><a href="how-to-use-this-book.html#book-structure">1. Book Structure<span></span></a></li>
<li><a href="how-to-use-this-book.html#find-what-is-relevant-for-you">2. Find what is relevant for you<span></span></a></li>
<li><a href="how-to-use-this-book.html#course-r-package-landgencourse">3. Course R package ‘LandGenCourse’<span></span></a></li>
</ul></li>
<li class="chapter" data-level="1.2" data-path="list-of-packages.html"><a href="list-of-packages.html"><i class="fa fa-check"></i><b>1.2</b> List of R Packages by Vignette<span></span></a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="review-of-r-skills.html"><a href="review-of-r-skills.html"><i class="fa fa-check"></i><b>2</b> Review of R Skills<span></span></a>
<ul>
<li class="chapter" data-level="2.1" data-path="basic-r.html"><a href="basic-r.html"><i class="fa fa-check"></i><b>2.1</b> Basic R Programming<span></span></a>
<ul>
<li><a href="basic-r.html#overview">1. Overview<span></span></a></li>
<li><a href="basic-r.html#data-types">2. Data Types<span></span></a></li>
<li><a href="basic-r.html#data-containers">3. Data Containers<span></span></a></li>
<li><a href="basic-r.html#programming">4. Programming<span></span></a></li>
</ul></li>
<li class="chapter" data-level="2.2" data-path="r-graphics.html"><a href="r-graphics.html"><i class="fa fa-check"></i><b>2.2</b> R Graphics<span></span></a>
<ul>
<li><a href="r-graphics.html#overview-1">1. Overview<span></span></a></li>
<li><a href="r-graphics.html#univariate-plots">2. Univariate plots<span></span></a></li>
<li><a href="r-graphics.html#bivariate-plots">3. Bivariate plots<span></span></a></li>
<li><a href="r-graphics.html#advanced-plotting">4. Advanced plotting<span></span></a></li>
</ul></li>
<li class="chapter" data-level="2.3" data-path="resources.html"><a href="resources.html"><i class="fa fa-check"></i><b>2.3</b> Further R resources<span></span></a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="github-for-group-projects.html"><a href="github-for-group-projects.html"><i class="fa fa-check"></i><b>3</b> GitHub for Group Projects<span></span></a>
<ul>
<li class="chapter" data-level="3.1" data-path="github.html"><a href="github.html"><i class="fa fa-check"></i><b>3.1</b> Why use GitHub?<span></span></a>
<ul>
<li><a href="github.html#goals">1. Goals<span></span></a></li>
<li><a href="github.html#version">2. Watch course video: Version Control 101<span></span></a></li>
<li><a href="github.html#slides">3. Preview slides<span></span></a></li>
<li><a href="github.html#next-steps">4. Next steps<span></span></a></li>
</ul></li>
<li class="chapter" data-level="3.2" data-path="install.html"><a href="install.html"><i class="fa fa-check"></i><b>3.2</b> Installation<span></span></a>
<ul>
<li><a href="install.html#goals-1">1. Goals<span></span></a></li>
<li class="chapter" data-level="3.2.1" data-path="install.html"><a href="install.html#how-to-get-github-and-gitkraken-for-free"><i class="fa fa-check"></i><b>3.2.1</b> 2. How to get GitHub and GitKraken for free<span></span></a></li>
<li><a href="install.html#happygit">3. Happy Git with R<span></span></a></li>
<li><a href="install.html#gitvideos">4. Video tutorials: installation<span></span></a></li>
<li><a href="install.html#join">5. Joining your group’s project on GitHub<span></span></a></li>
<li><a href="install.html#repo">6. Setting up a group repo<span></span></a></li>
<li><a href="install.html#next-steps-1">7. Next steps<span></span></a></li>
</ul></li>
<li class="chapter" data-level="3.3" data-path="workflow.html"><a href="workflow.html"><i class="fa fa-check"></i><b>3.3</b> Workflows<span></span></a>
<ul>
<li><a href="workflow.html#goals-2">1. Goals<span></span></a></li>
<li><a href="workflow.html#step-by-step-tutorials">2. Step-by-step tutorials<span></span></a></li>
<li><a href="workflow.html#further-resources">3. Further resources<span></span></a></li>
</ul></li>
</ul></li>
<li class="part"><span><b>Basic Topics<span></span></b></span></li>
<li class="chapter" data-level="4" data-path="Week1.html"><a href="Week1.html"><i class="fa fa-check"></i><b>4</b> Week 1: Importing Genetic Data<span></span></a>
<ul>
<li class="chapter" data-level="4.1" data-path="video_1.html"><a href="video_1.html"><i class="fa fa-check"></i><b>4.1</b> View Course Video<span></span></a>
<ul>
<li><a href="video_1.html#embedded-video">1. Embedded Video<span></span></a></li>
<li><a href="video_1.html#preview-slides">2. Preview Slides<span></span></a></li>
</ul></li>
<li class="chapter" data-level="4.2" data-path="tutorial_1.html"><a href="tutorial_1.html"><i class="fa fa-check"></i><b>4.2</b> Interactive Tutorial 1<span></span></a>
<ul>
<li><a href="tutorial_1.html#list-of-r-commands-covered-this-week">1. List of R commands covered this week<span></span></a></li>
<li><a href="tutorial_1.html#general-instructions">2. General Instructions<span></span></a></li>
</ul></li>
<li class="chapter" data-level="4.3" data-path="WE_1.html"><a href="WE_1.html"><i class="fa fa-check"></i><b>4.3</b> Worked Example<span></span></a>
<ul>
<li><a href="WE_1.html#overview-of-worked-example">1. Overview of Worked Example<span></span></a></li>
<li><a href="WE_1.html#import-data-from-.csv-file">2. Import data from .csv file<span></span></a></li>
<li><a href="WE_1.html#create-genind-object">3. Create ‘genind’ Object<span></span></a></li>
<li><a href="WE_1.html#view-information-stored-in-genind-object">4. View information stored in ‘genind’ object<span></span></a></li>
<li><a href="WE_1.html#import-data-with-gstudio-package">5. Import data with ‘gstudio’ package<span></span></a></li>
<li><a href="WE_1.html#importing-snp-data">6. Importing SNP data<span></span></a></li>
</ul></li>
<li class="chapter" data-level="4.4" data-path="r-exercise-week-1.html"><a href="r-exercise-week-1.html"><i class="fa fa-check"></i><b>4.4</b> R exercise Week 1<span></span></a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="Week2.html"><a href="Week2.html"><i class="fa fa-check"></i><b>5</b> Week 2: Spatial Data<span></span></a>
<ul>
<li class="chapter" data-level="5.1" data-path="video_2.html"><a href="video_2.html"><i class="fa fa-check"></i><b>5.1</b> View Course Video<span></span></a>
<ul>
<li><a href="video_2.html#embedded-video-1">1. Embedded Video<span></span></a></li>
<li><a href="video_2.html#preview-slides-1">2. Preview Slides<span></span></a></li>
</ul></li>
<li class="chapter" data-level="5.2" data-path="tutorial_2.html"><a href="tutorial_2.html"><i class="fa fa-check"></i><b>5.2</b> Interactive Tutorial 2<span></span></a>
<ul>
<li><a href="tutorial_2.html#list-of-r-commands-covered-this-week-1">1. List of R commands covered this week<span></span></a></li>
<li><a href="tutorial_2.html#general-instructions-1">2. General Instructions<span></span></a></li>
</ul></li>
<li class="chapter" data-level="5.3" data-path="WE_2.html"><a href="WE_2.html"><i class="fa fa-check"></i><b>5.3</b> Worked Example<span></span></a>
<ul>
<li><a href="WE_2.html#overview-of-worked-example-1">1. Overview of Worked Example<span></span></a></li>
<li><a href="WE_2.html#import-site-data-from-.csv-file">2. Import site data from .csv file<span></span></a></li>
<li><a href="WE_2.html#display-raster-data-and-overlay-sampling-locations-extract-data">3. Display raster data and overlay sampling locations, extract data<span></span></a></li>
<li><a href="WE_2.html#calculate-landscape-metrics">4. Calculate landscape metrics<span></span></a></li>
<li><a href="WE_2.html#sample-landscape-metrics-within-buffer-around-sampling-locations">5. Sample landscape metrics within buffer around sampling locations<span></span></a></li>
</ul></li>
<li class="chapter" data-level="5.4" data-path="r-exercise-week-2.html"><a href="r-exercise-week-2.html"><i class="fa fa-check"></i><b>5.4</b> R Exercise Week 2<span></span></a></li>
<li class="chapter" data-level="5.5" data-path="bonus_2a.html"><a href="bonus_2a.html"><i class="fa fa-check"></i><b>5.5</b> Bonus: ‘sf’ and ‘terra’<span></span></a>
<ul>
<li><a href="bonus_2a.html#overview-2">1. Overview<span></span></a></li>
<li><a href="bonus_2a.html#convert-spatialpointsdataframe-to-sf-object">2. Convert ‘SpatialPointsDataFrame’ to ‘sf’ object<span></span></a></li>
<li><a href="bonus_2a.html#plot-variables-first-10-in-data-set">3. Plot variables (first 10 in data set)<span></span></a></li>
<li><a href="bonus_2a.html#export-site-data-as-esri-shapefile">4. Export site data as ESRI shapefile<span></span></a></li>
<li><a href="bonus_2a.html#use-terra-and-tmap-to-display-categorical-map-with-color-scheme">5. Use ‘terra’ and ‘tmap’ to display categorical map with color scheme<span></span></a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="6" data-path="Week3.html"><a href="Week3.html"><i class="fa fa-check"></i><b>6</b> Week 3: Genetic Diversity<span></span></a>
<ul>
<li class="chapter" data-level="6.1" data-path="video_3.html"><a href="video_3.html"><i class="fa fa-check"></i><b>6.1</b> View Course Video<span></span></a>
<ul>
<li><a href="video_3.html#embedded-video-2">1. Embedded Video<span></span></a></li>
<li><a href="video_3.html#video-part-1">Video, Part 1<span></span></a></li>
<li><a href="video_3.html#video-part-2">Video, Part 2<span></span></a></li>
<li><a href="video_3.html#preview-slides-2">Preview Slides<span></span></a></li>
</ul></li>
<li class="chapter" data-level="6.2" data-path="tutorial_3.html"><a href="tutorial_3.html"><i class="fa fa-check"></i><b>6.2</b> Interactive Tutorial 3<span></span></a>
<ul>
<li><a href="tutorial_3.html#list-of-r-commands-covered-this-week-2">1. List of R commands covered this week<span></span></a></li>
<li><a href="tutorial_3.html#general-instructions-2">2. General Instructions<span></span></a></li>
</ul></li>
<li class="chapter" data-level="6.3" data-path="WE_3.html"><a href="WE_3.html"><i class="fa fa-check"></i><b>6.3</b> Worked Example<span></span></a>
<ul>
<li><a href="WE_3.html#overview-of-worked-example-2">1. Overview of Worked Example<span></span></a></li>
<li><a href="WE_3.html#basic-checking-of-markers-and-populations">2. Basic checking of markers and populations<span></span></a></li>
<li><a href="WE_3.html#assess-genetic-diversity">3. Assess genetic diversity<span></span></a></li>
<li><a href="WE_3.html#aggregate-genetic-data-at-population-level-allele-frequencies">4. Aggregate genetic data at population level (allele frequencies)<span></span></a></li>
</ul></li>
<li class="chapter" data-level="6.4" data-path="r-exercise-week-3.html"><a href="r-exercise-week-3.html"><i class="fa fa-check"></i><b>6.4</b> R Exercise Week 3<span></span></a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="Week4.html"><a href="Week4.html"><i class="fa fa-check"></i><b>7</b> Week 4: Metapopulations<span></span></a>
<ul>
<li class="chapter" data-level="7.1" data-path="video_4.html"><a href="video_4.html"><i class="fa fa-check"></i><b>7.1</b> View Course Video<span></span></a>
<ul>
<li><a href="video_4.html#embedded-video-3">1. Embedded Video<span></span></a></li>
<li><a href="video_4.html#video-part-1-1">Video, Part 1<span></span></a></li>
<li><a href="video_4.html#video-part-2-1">Video, Part 2<span></span></a></li>
<li><a href="video_4.html#preview-slides-3">Preview Slides<span></span></a></li>
</ul></li>
<li class="chapter" data-level="7.2" data-path="tutorial_4.html"><a href="tutorial_4.html"><i class="fa fa-check"></i><b>7.2</b> Interactive Tutorial 4<span></span></a>
<ul>
<li><a href="tutorial_4.html#list-of-r-commands-covered-this-week-3">1. List of R commands covered this week<span></span></a></li>
<li><a href="tutorial_4.html#general-instructions-3">2. General Instructions<span></span></a></li>
</ul></li>
<li class="chapter" data-level="7.3" data-path="WE_4.html"><a href="WE_4.html"><i class="fa fa-check"></i><b>7.3</b> Worked Example<span></span></a>
<ul>
<li><a href="WE_4.html#overview-of-worked-example-3">1. Overview of Worked Example<span></span></a></li>
<li><a href="WE_4.html#spatial-distribution-of-genetic-structure">2. Spatial distribution of genetic structure<span></span></a></li>
<li><a href="WE_4.html#what-determines-genetic-differentiation-among-sites">3. What determines genetic differentiation among sites?<span></span></a></li>
<li><a href="WE_4.html#what-determines-genetic-diversity">4. What determines genetic diversity?<span></span></a></li>
<li><a href="WE_4.html#are-genetic-differentiation-and-diversity-related">5. Are genetic differentiation and diversity related?<span></span></a></li>
<li><a href="WE_4.html#effect-of-recent-extinction-events">6. Effect of recent extinction events<span></span></a></li>
</ul></li>
<li class="chapter" data-level="7.4" data-path="r-exercise-week-4.html"><a href="r-exercise-week-4.html"><i class="fa fa-check"></i><b>7.4</b> R Exercise Week 4<span></span></a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="Week5.html"><a href="Week5.html"><i class="fa fa-check"></i><b>8</b> Week 5: Spatial Statistics<span></span></a>
<ul>
<li class="chapter" data-level="8.1" data-path="video_5.html"><a href="video_5.html"><i class="fa fa-check"></i><b>8.1</b> View Course Video<span></span></a>
<ul>
<li><a href="video_5.html#embedded-video-4">1. Embedded Video<span></span></a></li>
<li><a href="video_5.html#preview-slides-4">Preview Slides<span></span></a></li>
</ul></li>
<li class="chapter" data-level="8.2" data-path="tutorial_5.html"><a href="tutorial_5.html"><i class="fa fa-check"></i><b>8.2</b> Interactive Tutorial 5<span></span></a>
<ul>
<li><a href="tutorial_5.html#list-of-r-commands-covered-this-week-4">1. List of R commands covered this week<span></span></a></li>
<li><a href="tutorial_5.html#general-instructions-4">2. General Instructions<span></span></a></li>
</ul></li>
<li class="chapter" data-level="8.3" data-path="WE_5.html"><a href="WE_5.html"><i class="fa fa-check"></i><b>8.3</b> Worked Example<span></span></a>
<ul>
<li><a href="WE_5.html#overview-of-worked-example-4">1. Overview of Worked Example<span></span></a></li>
<li><a href="WE_5.html#data-import-and-manipulation">2. Data import and manipulation<span></span></a></li>
<li><a href="WE_5.html#calculate-genetic-distances">3. Calculate genetic distances<span></span></a></li>
<li><a href="WE_5.html#perform-a-mantel-test-to-test-for-ibd">4. Perform a Mantel test to test for IBD<span></span></a></li>
<li><a href="WE_5.html#create-mantel-correlogram-for-genetic-data">5. Create Mantel correlogram for genetic data<span></span></a></li>
<li><a href="WE_5.html#specify-spatial-weights-and-calculate-morans-i">6. Specify spatial weights and calculate Moran’s I<span></span></a></li>
</ul></li>
<li class="chapter" data-level="8.4" data-path="r-exercise-week-5.html"><a href="r-exercise-week-5.html"><i class="fa fa-check"></i><b>8.4</b> R Exercise Week 5<span></span></a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="Week6.html"><a href="Week6.html"><i class="fa fa-check"></i><b>9</b> Week 6: Quantitative Genetics<span></span></a>
<ul>
<li class="chapter" data-level="9.1" data-path="video_6.html"><a href="video_6.html"><i class="fa fa-check"></i><b>9.1</b> View Course Video<span></span></a>
<ul>
<li><a href="video_6.html#embedded-video-5">1. Embedded Video<span></span></a></li>
<li><a href="video_6.html#video-part-1-2">Video, Part 1<span></span></a></li>
<li><a href="video_6.html#video-part-2-2">Video, Part 2<span></span></a></li>
<li><a href="video_6.html#preview-slides-5">Preview Slides<span></span></a></li>
</ul></li>
<li class="chapter" data-level="9.2" data-path="tutorial_6.html"><a href="tutorial_6.html"><i class="fa fa-check"></i><b>9.2</b> Interactive Tutorial 6<span></span></a>
<ul>
<li><a href="tutorial_6.html#list-of-r-commands-covered-this-week-5">1. List of R commands covered this week<span></span></a></li>
<li><a href="tutorial_6.html#general-instructions-5">2. General Instructions<span></span></a></li>
</ul></li>
<li class="chapter" data-level="9.3" data-path="WE_6.html"><a href="WE_6.html"><i class="fa fa-check"></i><b>9.3</b> Worked Example<span></span></a>
<ul>
<li><a href="WE_6.html#overview-of-worked-example-5">1. Overview of Worked Example<span></span></a></li>
<li><a href="WE_6.html#estimate-genetic-and-non-genetic-variance-components-from-a-common-garden-experiment">2. Estimate genetic and non-genetic variance components from a common garden experiment<span></span></a></li>
<li><a href="WE_6.html#estimate-trait-heritability">3. Estimate trait heritability<span></span></a></li>
<li><a href="WE_6.html#compare-q_st-to-f_st-measured-from-snp-data">4. Compare <span class="math inline">\(Q_{st}\)</span> to <span class="math inline">\(F_{st}\)</span> measured from SNP data<span></span></a></li>
<li><a href="WE_6.html#assess-correlation-between-trait-and-environment">5. Assess correlation between trait and environment<span></span></a></li>
</ul></li>
<li class="chapter" data-level="9.4" data-path="r-exercise-week-6.html"><a href="r-exercise-week-6.html"><i class="fa fa-check"></i><b>9.4</b> R Exercise Week 6<span></span></a></li>
</ul></li>
<li class="chapter" data-level="10" data-path="Week7.html"><a href="Week7.html"><i class="fa fa-check"></i><b>10</b> Week 7: Spatial Linear Models<span></span></a>
<ul>
<li class="chapter" data-level="10.1" data-path="video_7.html"><a href="video_7.html"><i class="fa fa-check"></i><b>10.1</b> View Course Video<span></span></a>
<ul>
<li><a href="video_7.html#embedded-video-6">1. Embedded Video<span></span></a></li>
<li><a href="video_7.html#video-part-1-3">Video, Part 1<span></span></a></li>
<li><a href="video_7.html#video-part-2-3">Video, Part 2<span></span></a></li>
<li><a href="video_7.html#preview-slides-6">Preview Slides<span></span></a></li>
</ul></li>
<li class="chapter" data-level="10.2" data-path="tutorial_7.html"><a href="tutorial_7.html"><i class="fa fa-check"></i><b>10.2</b> Interactive Tutorial 7<span></span></a>
<ul>
<li><a href="tutorial_7.html#list-of-r-commands-covered-this-week-6">1. List of R commands covered this week<span></span></a></li>
<li><a href="tutorial_7.html#general-instructions-6">2. General Instructions<span></span></a></li>
</ul></li>
<li class="chapter" data-level="10.3" data-path="WE_7.html"><a href="WE_7.html"><i class="fa fa-check"></i><b>10.3</b> Worked Example<span></span></a>
<ul>
<li><a href="WE_7.html#overview-of-worked-example-6">1. Overview of Worked Example<span></span></a></li>
<li><a href="WE_7.html#explore-data-set">2. Explore data set<span></span></a></li>
<li><a href="WE_7.html#test-regression-residuals-for-spatial-autocorrelation">3. Test regression residuals for spatial autocorrelation<span></span></a></li>
<li><a href="WE_7.html#fit-models-with-spatially-correlated-error-gls-with-package-nlme">4. Fit models with spatially correlated error (GLS) with package ‘nlme’<span></span></a></li>
<li><a href="WE_7.html#fit-spatial-simultaneous-autoregressive-error-models-sar">5. Fit spatial simultaneous autoregressive error models (SAR)<span></span></a></li>
<li><a href="WE_7.html#spatial-filtering-with-mem-using-package-spmoran">6. Spatial filtering with MEM using package ‘spmoran’<span></span></a></li>
<li><a href="WE_7.html#fit-spatially-varying-coefficients-model-with-package-spmoran">7. Fit spatially varying coefficients model with package ‘spmoran’<span></span></a></li>
<li><a href="WE_7.html#model-with-patchsize-and-ibr">Model with PatchSize and IBR<span></span></a></li>
<li><a href="WE_7.html#model-with-ibr-only">Model with IBR only<span></span></a></li>
<li><a href="WE_7.html#conclusions">8. Conclusions<span></span></a></li>
</ul></li>
<li class="chapter" data-level="10.4" data-path="r-exercise-week-7.html"><a href="r-exercise-week-7.html"><i class="fa fa-check"></i><b>10.4</b> R Exercise Week 7<span></span></a></li>
<li class="chapter" data-level="10.5" data-path="bonus_7a.html"><a href="bonus_7a.html"><i class="fa fa-check"></i><b>10.5</b> Bonus: Si index<span></span></a>
<ul>
<li><a href="bonus_7a.html#overview-of-bonus-materials">1. Overview of Bonus Materials<span></span></a></li>
<li><a href="bonus_7a.html#import-ecological-distance-matrices">2. Import ecological distance matrices<span></span></a></li>
<li><a href="bonus_7a.html#optimize-the-scaling-parameter-alpha">3. Optimize the scaling parameter alpha<span></span></a></li>
<li><a href="bonus_7a.html#calculate-hanskis-index-si-with-source-patch-parameters">4. Calculate Hanski’s index Si with source patch parameters<span></span></a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="11" data-path="Week8.html"><a href="Week8.html"><i class="fa fa-check"></i><b>11</b> Week 8: Simulation Experiments<span></span></a>
<ul>
<li class="chapter" data-level="11.1" data-path="video_8.html"><a href="video_8.html"><i class="fa fa-check"></i><b>11.1</b> View Course Video<span></span></a>
<ul>
<li><a href="video_8.html#embedded-video-7">1. Embedded Video<span></span></a></li>
<li><a href="video_8.html#video-part-1-4">Video, Part 1<span></span></a></li>
<li><a href="video_8.html#video-part-2-4">Video, Part 2<span></span></a></li>
<li><a href="video_8.html#preview-slides-7">Preview Slides<span></span></a></li>
</ul></li>
<li class="chapter" data-level="11.2" data-path="tutorial_8.html"><a href="tutorial_8.html"><i class="fa fa-check"></i><b>11.2</b> Interactive Tutorial 8<span></span></a>
<ul>
<li><a href="tutorial_8.html#list-of-r-commands-covered-this-week-7">1. List of R commands covered this week<span></span></a></li>
<li><a href="tutorial_8.html#general-instructions-7">2. General Instructions<span></span></a></li>
</ul></li>
<li class="chapter" data-level="11.3" data-path="WE_8.html"><a href="WE_8.html"><i class="fa fa-check"></i><b>11.3</b> Worked Example<span></span></a>
<ul>
<li><a href="WE_8.html#overview-of-worked-example-7">1. Overview of Worked Example<span></span></a></li>
<li><a href="WE_8.html#initialize-a-landscape">2. Initialize a landscape<span></span></a></li>
<li><a href="WE_8.html#define-simulation-parameters">3. Define simulation parameters<span></span></a></li>
<li><a href="WE_8.html#run-simulations-and-analyze-results">4. Run simulations and analyze results<span></span></a></li>
<li><a href="WE_8.html#run-simulator-using-a-previously-defined-parameter-set">5. Run simulator using a previously defined parameter set<span></span></a></li>
</ul></li>
<li class="chapter" data-level="11.4" data-path="r-exercise-week-8.html"><a href="r-exercise-week-8.html"><i class="fa fa-check"></i><b>11.4</b> R Exercise Week 8<span></span></a></li>
<li class="chapter" data-level="11.5" data-path="bonus_8a.html"><a href="bonus_8a.html"><i class="fa fa-check"></i><b>11.5</b> Bonus: Efficient R<span></span></a>
<ul>
<li><a href="bonus_8a.html#overview-of-bonus-material">1. Overview of Bonus Material<span></span></a></li>
<li><a href="bonus_8a.html#navigating-the-file-system">2. Navigating the file system<span></span></a></li>
<li><a href="bonus_8a.html#benchmarking-file-import-and-export-options">3. Benchmarking file import and export options<span></span></a></li>
<li><a href="bonus_8a.html#profiling-your-code">4. Profiling your code<span></span></a></li>
<li><a href="bonus_8a.html#creating-and-executing-a-bash-r-script">5. Creating and executing a Bash R script<span></span></a></li>
<li><a href="bonus_8a.html#parallelizing-code">6. Parallelizing code<span></span></a></li>
</ul></li>
</ul></li>
<li class="part"><span><b>Advanced Topics<span></span></b></span></li>
<li class="chapter" data-level="12" data-path="Week9.html"><a href="Week9.html"><i class="fa fa-check"></i><b>12</b> Week 9: Population Structure<span></span></a>
<ul>
<li class="chapter" data-level="12.1" data-path="WE_9.html"><a href="WE_9.html"><i class="fa fa-check"></i><b>12.1</b> Worked Example<span></span></a>
<ul>
<li><a href="WE_9.html#overview-of-worked-example-8">1. Overview of Worked Example<span></span></a></li>
<li><a href="WE_9.html#simulated-data-2-island-model">2. Simulated data: 2-island model<span></span></a></li>
<li><a href="WE_9.html#simulated-data-2-island-model-with-admixture">3. Simulated data: 2-island model with admixture<span></span></a></li>
<li><a href="WE_9.html#empirical-data-threespine-sticklebacks">4. Empirical data: Threespine sticklebacks<span></span></a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="13" data-path="Week10.html"><a href="Week10.html"><i class="fa fa-check"></i><b>13</b> Week 10: Landscape Resistance<span></span></a>
<ul>
<li class="chapter" data-level="13.1" data-path="WE_10.html"><a href="WE_10.html"><i class="fa fa-check"></i><b>13.1</b> Worked Example<span></span></a>
<ul>
<li><a href="WE_10.html#overview-of-worked-example-9">1. Overview of Worked Example<span></span></a></li>
<li><a href="WE_10.html#explore-the-data-set">2. Explore the data set<span></span></a></li>
<li><a href="WE_10.html#setting-cost-values-and-calculating-conductance">3. Setting cost values and calculating conductance<span></span></a></li>
<li><a href="WE_10.html#convert-conductance-into-effective-distance">4. Convert conductance into effective distance<span></span></a></li>
<li><a href="WE_10.html#create-cost-distance-matrices">5. Create cost-distance matrices<span></span></a></li>
<li><a href="WE_10.html#how-does-changing-resolution-affect-these-metrics">6. How does changing resolution affect these metrics?<span></span></a></li>
<li><a href="WE_10.html#references">7. References<span></span></a></li>
</ul></li>
<li class="chapter" data-level="13.2" data-path="bonus-10a.html"><a href="bonus-10a.html"><i class="fa fa-check"></i><b>13.2</b> Bonus: ‘radish’ tutorial<span></span></a>
<ul>
<li><a href="bonus-10a.html#overview-of-bonus-materials-1">1. Overview of Bonus Materials<span></span></a></li>
<li><a href="bonus-10a.html#import-data">2. Import data<span></span></a></li>
<li><a href="bonus-10a.html#basic-model-fitting">3. Basic model fitting<span></span></a></li>
<li><a href="bonus-10a.html#advanced-modeling-topics">4. Advanced modeling topics<span></span></a></li>
<li><a href="bonus-10a.html#summary">5. Summary<span></span></a></li>
<li><a href="bonus-10a.html#references-1">6. References<span></span></a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="14" data-path="Week11.html"><a href="Week11.html"><i class="fa fa-check"></i><b>14</b> Week 11: Detecting Adaptation<span></span></a>
<ul>
<li class="chapter" data-level="14.1" data-path="WE_11.html"><a href="WE_11.html"><i class="fa fa-check"></i><b>14.1</b> Worked Example<span></span></a>
<ul>
<li><a href="WE_11.html#overview-of-worked-example-10">1. Overview of Worked Example<span></span></a></li>
<li><a href="WE_11.html#import-and-prepare-the-data">2. Import and prepare the data<span></span></a></li>
<li><a href="WE_11.html#latent-factor-mixed-models-lfmm-a-univariate-gea">3. Latent Factor Mixed Models (LFMM): a univariate GEA<span></span></a></li>
<li><a href="WE_11.html#redundancy-analysis-rda-a-multivariate-gea">4. Redundancy Analysis (RDA): a multivariate GEA<span></span></a></li>
<li><a href="WE_11.html#compare-lfmm-and-rda-candidates">5. Compare LFMM and RDA candidates<span></span></a></li>
<li><a href="WE_11.html#a-quick-note-on-controlling-for-population-structure-in-rda">6. A quick note on controlling for population structure in RDA<span></span></a></li>
<li><a href="WE_11.html#references-2">7. References<span></span></a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="15" data-path="Week12.html"><a href="Week12.html"><i class="fa fa-check"></i><b>15</b> Week 12: Model Selection<span></span></a>
<ul>
<li class="chapter" data-level="15.1" data-path="WE_12.html"><a href="WE_12.html"><i class="fa fa-check"></i><b>15.1</b> Worked Example<span></span></a>
<ul>
<li><a href="WE_12.html#overview-of-worked-example-11">1. Overview of Worked Example<span></span></a></li>
<li><a href="WE_12.html#fitting-candidate-models">2. Fitting candidate models<span></span></a></li>
<li><a href="WE_12.html#your-turn-to-test-additional-hypotheses">3. Your turn to test additional hypotheses!<span></span></a></li>
<li><a href="WE_12.html#nested-model-nmlpe-for-hierarchical-sampling-designs">4. Nested model (NMLPE) for hierarchical sampling designs<span></span></a></li>
<li><a href="WE_12.html#references-3">5. References<span></span></a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="16" data-path="Week13.html"><a href="Week13.html"><i class="fa fa-check"></i><b>16</b> Week 13: Gravity Models<span></span></a>
<ul>
<li class="chapter" data-level="16.1" data-path="WE_13.html"><a href="WE_13.html"><i class="fa fa-check"></i><b>16.1</b> Worked Example<span></span></a>
<ul>
<li><a href="WE_13.html#overview-of-worked-example-12">1. Overview of Worked Example<span></span></a></li>
<li><a href="WE_13.html#setup">2. Setup<span></span></a></li>
<li><a href="WE_13.html#wetland-complex-data-preparation">3. Wetland complex data preparation<span></span></a></li>
<li><a href="WE_13.html#wetland-field-data-preparation">4. Wetland field-data preparation<span></span></a></li>
<li><a href="WE_13.html#saturated-graph">5. Saturated Graph<span></span></a></li>
<li><a href="WE_13.html#spatial-model-data-prepration">6. Spatial model data prepration<span></span></a></li>
<li><a href="WE_13.html#gravity-model">7. Gravity model<span></span></a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="17" data-path="Week14.html"><a href="Week14.html"><i class="fa fa-check"></i><b>17</b> Week 14: Contemporary Gene Flow<span></span></a>
<ul>
<li class="chapter" data-level="17.1" data-path="WE_14.html"><a href="WE_14.html"><i class="fa fa-check"></i><b>17.1</b> Worked Example<span></span></a>
<ul>
<li><a href="WE_14.html#overview-3">1. Overview<span></span></a></li>
<li><a href="WE_14.html#import-genotypes">2. Import Genotypes<span></span></a></li>
<li><a href="WE_14.html#pollen-pool-genetic-structure">3. Pollen pool genetic structure<span></span></a></li>
<li><a href="WE_14.html#paternity-analysis">4. Paternity analysis<span></span></a></li>
<li><a href="WE_14.html#linking-paternity-to-ecological-variables">5. Linking paternity to ecological variables<span></span></a></li>
<li><a href="WE_14.html#references-4">6. References<span></span></a></li>
</ul></li>
</ul></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Landscape Genetic Data Analysis with R</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="bonus_2a" class="section level2 hasAnchor" number="5.5">
<h2><span class="header-section-number">5.5</span> Bonus: ‘sf’ and ‘terra’<a href="bonus_2a.html#bonus_2a" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Helene Wagner</p>
<div id="overview-2" class="section level3 unnumbered hasAnchor">
<h3>1. Overview<a href="bonus_2a.html#overview-2" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>This bonus vignette uses the new packages <code>sf</code>, <code>terra</code>, and <code>tmap</code>, yet it can only scratch the surface. For a thorough introduction to geocomputation with R, see this excellent Gitbook: <a href="https://geocompr.robinlovelace.net/index.html" class="uri">https://geocompr.robinlovelace.net/index.html</a>.</p>
<div id="a.-goals-2" class="section level4 unnumbered hasAnchor">
<h4>a. Goals<a href="bonus_2a.html#a.-goals-2" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>This bonus material expands the Worked Example to show:</p>
<ul>
<li>How to convert the site data from an <code>sp</code> object to an <code>sf</code> object, a new standard for geospatial vector data.</li>
<li>How to create static and interactive maps of an <code>sf</code> object with <code>tmap</code>.</li>
<li>How to export an <code>sf</code> object to a, ESRI shapefile.</li>
<li>How to plot a categorical raster map with a predefined color scheme, using <code>tmap</code> and a new standard for raster data (<code>terra</code> package).</li>
</ul>
<p>Try modifying the code to import your own data!</p>
</div>
<div id="b.-required-r-packages" class="section level4 unnumbered hasAnchor">
<h4>b. Required R packages<a href="bonus_2a.html#b.-required-r-packages" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<div class="sourceCode" id="cb437"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb437-1"><a href="bonus_2a.html#cb437-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(LandGenCourse)</span>
<span id="cb437-2"><a href="bonus_2a.html#cb437-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(sp)</span>
<span id="cb437-3"><a href="bonus_2a.html#cb437-3" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(sf)</span>
<span id="cb437-4"><a href="bonus_2a.html#cb437-4" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(raster)</span>
<span id="cb437-5"><a href="bonus_2a.html#cb437-5" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(GeNetIt)</span>
<span id="cb437-6"><a href="bonus_2a.html#cb437-6" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(terra)</span>
<span id="cb437-7"><a href="bonus_2a.html#cb437-7" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tmap)</span>
<span id="cb437-8"><a href="bonus_2a.html#cb437-8" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(dplyr)</span>
<span id="cb437-9"><a href="bonus_2a.html#cb437-9" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tibble)</span></code></pre></div>
</div>
</div>
<div id="convert-spatialpointsdataframe-to-sf-object" class="section level3 unnumbered hasAnchor">
<h3>2. Convert ‘SpatialPointsDataFrame’ to ‘sf’ object<a href="bonus_2a.html#convert-spatialpointsdataframe-to-sf-object" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>While ‘sp’ was a major achievement, it is being replaced now by ‘sf,’ which is short for “simple feature.” The conversion is easy. The resulting S3 object is a data frame and also an ‘sf’ object.</p>
<div class="sourceCode" id="cb438"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb438-1"><a href="bonus_2a.html#cb438-1" aria-hidden="true" tabindex="-1"></a><span class="fu">data</span>(ralu.site)</span>
<span id="cb438-2"><a href="bonus_2a.html#cb438-2" aria-hidden="true" tabindex="-1"></a>Sites.sf <span class="ot">&lt;-</span> <span class="fu">st_as_sf</span>(ralu.site)</span>
<span id="cb438-3"><a href="bonus_2a.html#cb438-3" aria-hidden="true" tabindex="-1"></a><span class="fu">class</span>(Sites.sf)</span></code></pre></div>
<pre><code>## [1] &quot;sf&quot;         &quot;data.frame&quot;</code></pre>
<p>Check whether the projection has been defined:</p>
<div class="sourceCode" id="cb440"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb440-1"><a href="bonus_2a.html#cb440-1" aria-hidden="true" tabindex="-1"></a>sf<span class="sc">::</span><span class="fu">st_crs</span>(Sites.sf)<span class="sc">$</span>input</span></code></pre></div>
<pre><code>## [1] &quot;+proj=utm +zone=11 +ellps=GRS80 +towgs84=0,0,0,0,0,0,0 +units=m +no_defs&quot;</code></pre>
</div>
<div id="plot-variables-first-10-in-data-set" class="section level3 unnumbered hasAnchor">
<h3>3. Plot variables (first 10 in data set)<a href="bonus_2a.html#plot-variables-first-10-in-data-set" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div id="a.-plot-geometry" class="section level4 unnumbered hasAnchor">
<h4>a. Plot geometry<a href="bonus_2a.html#a.-plot-geometry" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>The package sf makes a clear distinction between the geometry information (spatial coordinates: where in space) and attribute information (what’s at these locations). Hence, when using the function plot with sf objects, we need to decide what we want to plot: geometry or attributes?</p>
<p>To plot the geometry, we use function <code>st_geometry</code> to extract the geometry information from the <code>sf</code> object:</p>
<div class="sourceCode" id="cb442"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb442-1"><a href="bonus_2a.html#cb442-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(<span class="fu">st_geometry</span>(Sites.sf))</span></code></pre></div>
<p><img src="02-Week02_files/figure-html/unnamed-chunk-59-1.png" width="672" /></p>
</div>
<div id="b.-plot-attributes-in-space" class="section level4 unnumbered hasAnchor">
<h4>b. Plot attributes in space<a href="bonus_2a.html#b.-plot-attributes-in-space" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>If we don’t extract the geometry, then R will assume that we want to plot attribute data. The default is to plot the first ten attributes.</p>
<p>Here we set the point character <code>pch</code> to a filled circle, which is symbol #16. With <code>cex=2</code>, we define the symbol size.</p>
<p>For an overview of ‘pch’ symbol numbers, and colors, check: <a href="http://vis.supstat.com/2013/04/plotting-symbols-and-color-palettes/" class="uri">http://vis.supstat.com/2013/04/plotting-symbols-and-color-palettes/</a></p>
<div class="sourceCode" id="cb443"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb443-1"><a href="bonus_2a.html#cb443-1" aria-hidden="true" tabindex="-1"></a><span class="fu">par</span>(<span class="at">mar=</span><span class="fu">c</span>(<span class="dv">2</span>,<span class="dv">2</span>,<span class="dv">2</span>,<span class="dv">2</span>))</span>
<span id="cb443-2"><a href="bonus_2a.html#cb443-2" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(Sites.sf, <span class="at">pch=</span><span class="dv">16</span>, <span class="at">cex=</span><span class="dv">2</span>)</span></code></pre></div>
<pre><code>## Warning: plotting the first 10 out of 17 attributes; use max.plot = 17 to plot
## all</code></pre>
<p><img src="02-Week02_files/figure-html/unnamed-chunk-60-1.png" width="672" /></p>
<p>This is pretty cool! Let’s have a closer look at two of the variables.</p>
<ul>
<li><code>Basin</code> (left): this is a factor, and each factor level is assigned a different color.</li>
<li><code>Depth_m</code> (right): this is a quantitative variable, and R automatically uses a color ramp (from blue to pink to orange) to indicate variation in the values.</li>
</ul>
<p>Note: To learn about options for the ‘plot’ function for ‘sf’ objects, access the help file by typing ‘?plot’ and select ‘Plot sf object.’</p>
<div class="sourceCode" id="cb445"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb445-1"><a href="bonus_2a.html#cb445-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(Sites.sf[,<span class="fu">c</span>(<span class="st">&quot;Basin&quot;</span>, <span class="st">&quot;Depth_m&quot;</span>)], <span class="at">pch=</span><span class="dv">16</span>)</span></code></pre></div>
<p><img src="02-Week02_files/figure-html/unnamed-chunk-61-1.png" width="672" /></p>
</div>
<div id="c.-create-a-static-bubble-plot-with-tmap" class="section level4 unnumbered hasAnchor">
<h4>c. Create a static bubble plot with ‘tmap’<a href="bonus_2a.html#c.-create-a-static-bubble-plot-with-tmap" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>The tmap package (for plotting thematic maps) is a great new tool for plotting maps. It is based on the grammar of graphics concepts, which take a bit of getting used to. Most importantly, we need the following parts:</p>
<ul>
<li><code>tmap_mode("plot")</code>: to plot a static map (default)</li>
<li><code>tm_shape</code>: this function defines the data to be used.</li>
<li><code>tm_sf</code>: this function defines what information should be plotted and how.</li>
</ul>
<p>Note: for the second part, there are many other functions for various types of data.</p>
<p>If we use <code>tm_sf</code> without arguments (i.e., with the default settings), we get a plot of the geometry:</p>
<div class="sourceCode" id="cb446"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb446-1"><a href="bonus_2a.html#cb446-1" aria-hidden="true" tabindex="-1"></a><span class="fu">tmap_mode</span>(<span class="st">&quot;plot&quot;</span>)</span>
<span id="cb446-2"><a href="bonus_2a.html#cb446-2" aria-hidden="true" tabindex="-1"></a><span class="fu">tm_shape</span>(Sites.sf) <span class="sc">+</span> <span class="fu">tm_sf</span>()</span></code></pre></div>
<p><img src="02-Week02_files/figure-html/unnamed-chunk-62-1.png" width="672" /></p>
<p>We can indicate an attribute to plot it:</p>
<div class="sourceCode" id="cb447"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb447-1"><a href="bonus_2a.html#cb447-1" aria-hidden="true" tabindex="-1"></a><span class="fu">tmap_mode</span>(<span class="st">&quot;plot&quot;</span>)</span>
<span id="cb447-2"><a href="bonus_2a.html#cb447-2" aria-hidden="true" tabindex="-1"></a><span class="fu">tm_shape</span>(Sites.sf) <span class="sc">+</span> <span class="fu">tm_sf</span>(<span class="st">&quot;Depth_m&quot;</span>) </span></code></pre></div>
<p><img src="02-Week02_files/figure-html/unnamed-chunk-63-1.png" width="672" /></p>
<p>Let’s make the boundary box (map extent) a little larger so that the symbols are not cut off. First we extract the boundary box of <code>Sites.sf</code> and save it as <code>Bbox</code>.</p>
<div class="sourceCode" id="cb448"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb448-1"><a href="bonus_2a.html#cb448-1" aria-hidden="true" tabindex="-1"></a>Bbox <span class="ot">=</span> <span class="fu">st_bbox</span>(Sites.sf)</span>
<span id="cb448-2"><a href="bonus_2a.html#cb448-2" aria-hidden="true" tabindex="-1"></a>Bbox</span></code></pre></div>
<pre><code>##      xmin      ymin      xmax      ymax 
##  686908.5 4994089.3  690890.1 5004435.0</code></pre>
<p>Then we define the range along x and y coordinates (<code>delta.x</code>, <code>delta.y</code>), set a zoom factor (<code>Zoom</code>) and add that fraction of the range on each side. Unfortunately, there is no dedicated function for this so we do this manually:</p>
<div class="sourceCode" id="cb450"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb450-1"><a href="bonus_2a.html#cb450-1" aria-hidden="true" tabindex="-1"></a>delta.x <span class="ot">&lt;-</span> Bbox[<span class="dv">3</span>] <span class="sc">-</span> Bbox[<span class="dv">1</span>]</span>
<span id="cb450-2"><a href="bonus_2a.html#cb450-2" aria-hidden="true" tabindex="-1"></a>delta.y <span class="ot">&lt;-</span> Bbox[<span class="dv">4</span>] <span class="sc">-</span> Bbox[<span class="dv">2</span>]</span>
<span id="cb450-3"><a href="bonus_2a.html#cb450-3" aria-hidden="true" tabindex="-1"></a>Zoom <span class="ot">&lt;-</span> <span class="fl">0.1</span></span>
<span id="cb450-4"><a href="bonus_2a.html#cb450-4" aria-hidden="true" tabindex="-1"></a>Bbox2 <span class="ot">&lt;-</span> Bbox <span class="sc">+</span> <span class="fu">c</span>(<span class="sc">-</span>delta.x, <span class="sc">-</span>delta.y, delta.x, delta.y) <span class="sc">*</span> Zoom</span>
<span id="cb450-5"><a href="bonus_2a.html#cb450-5" aria-hidden="true" tabindex="-1"></a>Bbox2</span></code></pre></div>
<pre><code>##      xmin      ymin      xmax      ymax 
##  686510.4 4993054.7  691288.2 5005469.6</code></pre>
<p>Now we add the boundary box information as an argument <code>bbox</code> in function <code>tm_shape</code>.</p>
<p>To create a bubble plot, we indicate that the attribute <code>Depth_m</code> should be used to define the symbol size.</p>
<p>Note that we write the figure into an object, <code>Map1</code>, then plot the <code>Map1</code>. This will help e.g. with exporting the map, or we can later add more layers to this map object with <code>+</code>.</p>
<div class="sourceCode" id="cb452"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb452-1"><a href="bonus_2a.html#cb452-1" aria-hidden="true" tabindex="-1"></a><span class="fu">tmap_mode</span>(<span class="st">&quot;plot&quot;</span>)</span>
<span id="cb452-2"><a href="bonus_2a.html#cb452-2" aria-hidden="true" tabindex="-1"></a>Map1 <span class="ot">&lt;-</span> <span class="fu">tm_shape</span>(Sites.sf, <span class="at">bbox=</span>Bbox2) <span class="sc">+</span> <span class="fu">tm_sf</span>(<span class="at">size =</span> <span class="st">&quot;Depth_m&quot;</span>) </span>
<span id="cb452-3"><a href="bonus_2a.html#cb452-3" aria-hidden="true" tabindex="-1"></a>Map1</span></code></pre></div>
<p><img src="02-Week02_files/figure-html/unnamed-chunk-66-1.png" width="672" /></p>
</div>
<div id="d.-create-an-interactive-bubble-plot-with-tmap" class="section level4 unnumbered hasAnchor">
<h4>d. Create an interactive bubble plot with ‘tmap’<a href="bonus_2a.html#d.-create-an-interactive-bubble-plot-with-tmap" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>This is not a great plot yet. Let’s tweak it a bit:</p>
<ul>
<li><p>tmap_mode(“view”): create an interactive map with a zoomable background map</p></li>
<li><p><code>size="Depth_m"</code>: create a bubble plot where the symbol size is proportional to the value of a quantitative variable</p></li>
<li><p><code>col="Basin"</code>: use a default color ramp to show different values, here for the categorical variable Basin.</p></li>
<li><p><code>tm_basemap(server = "OpenTopoMap")</code>: define what basemap should be used from internet.</p></li>
</ul>
<p>If you omit <code>tm_basemap(server = "OpenTopoMap")</code>, R will show “Esri.WorldGrayCanvas” and include an interactive menu to toggle between “Esri.WorldGrayCanvas,” “OpenStreetMap,” and “Esri.WorldTopoMap.”</p>
<p>Uncomment the code below by removing the hashtag (#) and run it. Go ahead and play with the interactive map!</p>
<div class="sourceCode" id="cb453"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb453-1"><a href="bonus_2a.html#cb453-1" aria-hidden="true" tabindex="-1"></a><span class="co">#tmap_mode(&quot;view&quot;)</span></span>
<span id="cb453-2"><a href="bonus_2a.html#cb453-2" aria-hidden="true" tabindex="-1"></a><span class="co">#tm_shape(Sites.sf) + tm_sf(size=&quot;Depth_m&quot;, col=&quot;Depth_m&quot;, palette = &quot;Blues&quot;) </span></span></code></pre></div>
<p>Let’s make a few more changes:</p>
<ul>
<li>We change the symbolization to fixed-size, filled circles with a black border, and plot them on top of filled circles that indicate the basin. Note that although both attributes are in the same dataset <code>Sites.sf</code>, we need to include a data statement for each attribute we add to the map (i.e., map layer), using <code>tm_shape(Sites.sf)</code>.</li>
<li>We include the argument <code>bbox=Bbox2</code> for at least one of the map layers. This is not needed for the interactive map but will look better when exporting it as a static map.</li>
<li>We add a call to function <code>tm_basemap</code>, so that we can modify the list of the available basemap options. The first one listed in the argument <code>server</code> will be shown by default.</li>
</ul>
<p>Uncomment the code below by removing the hashtag (#) and run it. Go ahead and toggle between the base maps in the map below! Different base maps are suitable for different situations (data type, symbol type, size of study area) and purposes.</p>
<div class="sourceCode" id="cb454"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb454-1"><a href="bonus_2a.html#cb454-1" aria-hidden="true" tabindex="-1"></a><span class="co">#tmap_mode(&quot;view&quot;)</span></span>
<span id="cb454-2"><a href="bonus_2a.html#cb454-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb454-3"><a href="bonus_2a.html#cb454-3" aria-hidden="true" tabindex="-1"></a><span class="co">#Map2 &lt;- tm_shape(Sites.sf, bbox=Bbox2) +  tm_sf(&quot;Basin&quot;, size=2, border.col=&quot;black&quot;) +</span></span>
<span id="cb454-4"><a href="bonus_2a.html#cb454-4" aria-hidden="true" tabindex="-1"></a><span class="co">#  tm_shape(Sites.sf) + tm_sf(size=0.8, col=&quot;Depth_m&quot;, </span></span>
<span id="cb454-5"><a href="bonus_2a.html#cb454-5" aria-hidden="true" tabindex="-1"></a><span class="co">#                             palette = &quot;Blues&quot;, border.col=&quot;black&quot;) +</span></span>
<span id="cb454-6"><a href="bonus_2a.html#cb454-6" aria-hidden="true" tabindex="-1"></a><span class="co">#  tm_basemap(server = c(&quot;Esri.WorldTopoMap&quot;, &quot;Esri.WorldGrayCanvas&quot;, </span></span>
<span id="cb454-7"><a href="bonus_2a.html#cb454-7" aria-hidden="true" tabindex="-1"></a><span class="co">#                        &quot;OpenStreetMap&quot;, &quot;OpenTopoMap&quot;, </span></span>
<span id="cb454-8"><a href="bonus_2a.html#cb454-8" aria-hidden="true" tabindex="-1"></a><span class="co">#                        leaflet::providers$Stamen.Terrain, </span></span>
<span id="cb454-9"><a href="bonus_2a.html#cb454-9" aria-hidden="true" tabindex="-1"></a><span class="co">#                        leaflet::providers$Stamen.Watercolor,</span></span>
<span id="cb454-10"><a href="bonus_2a.html#cb454-10" aria-hidden="true" tabindex="-1"></a><span class="co">#                        leaflet::providers$Stamen.TonerLite))</span></span>
<span id="cb454-11"><a href="bonus_2a.html#cb454-11" aria-hidden="true" tabindex="-1"></a><span class="co">#Map2</span></span></code></pre></div>
</div>
<div id="e.-export-maps" class="section level4 unnumbered hasAnchor">
<h4>e. Export maps<a href="bonus_2a.html#e.-export-maps" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>Save the map to folder <code>output</code>:</p>
<ul>
<li>Interactive map saved to <code>html</code> file. This includes all the features of the interactive map <code>Map2</code>.</li>
<li>Static map saved to <code>png</code> file, with height = 7 inches. This drops the base map and retains only the symbolized data.</li>
</ul>
<p>Go ahead and check out the two files!</p>
<div class="sourceCode" id="cb455"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb455-1"><a href="bonus_2a.html#cb455-1" aria-hidden="true" tabindex="-1"></a><span class="fu">tmap_save</span>(Map1, <span class="fu">paste0</span>(here<span class="sc">::</span><span class="fu">here</span>(), <span class="st">&quot;/output/StaticMap.png&quot;</span>), <span class="at">height=</span><span class="dv">7</span>)</span></code></pre></div>
<pre><code>## Map saved to /Users/hhwagner1/Desktop/R_GitHub_Projects/LandGenCourse_book/output/StaticMap.png</code></pre>
<pre><code>## Resolution: 808.1867 by 2100 pixels</code></pre>
<pre><code>## Size: 2.693956 by 7 inches (300 dpi)</code></pre>
<div class="sourceCode" id="cb459"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb459-1"><a href="bonus_2a.html#cb459-1" aria-hidden="true" tabindex="-1"></a><span class="co">#tmap_save(Map2, paste0(here::here(), &quot;/output/InteractiveMap.html&quot;))</span></span></code></pre></div>
</div>
</div>
<div id="export-site-data-as-esri-shapefile" class="section level3 unnumbered hasAnchor">
<h3>4. Export site data as ESRI shapefile<a href="bonus_2a.html#export-site-data-as-esri-shapefile" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>To export your results to GIS software, you can write them into an ESRI shapefile, which is a widely used data format for sharing geospatial vector data.</p>
<p>The following code may produce a warning that column names were abbreviated. It writes the component files for the ESRI shapefile into the pre-existing folder <code>output</code> (the first line will create it if does not exist yet). Remember to remove the hashtags ‘#’ to uncomment the code before running it.</p>
<p>The argument <code>delete_dsn</code> specifies whether any existing file with the same name should be deleted first (i.e., overwritten).</p>
<div class="sourceCode" id="cb460"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb460-1"><a href="bonus_2a.html#cb460-1" aria-hidden="true" tabindex="-1"></a><span class="co">#require(here)</span></span>
<span id="cb460-2"><a href="bonus_2a.html#cb460-2" aria-hidden="true" tabindex="-1"></a><span class="co">#if(!dir.exists(paste0(here(),&quot;/output&quot;))) dir.create(paste0(here(),&quot;/output&quot;))</span></span>
<span id="cb460-3"><a href="bonus_2a.html#cb460-3" aria-hidden="true" tabindex="-1"></a><span class="co">#st_write(Sites.sf, paste0(here(),&quot;/output/Sites.shp&quot;), delete_dsn = TRUE)</span></span></code></pre></div>
</div>
<div id="use-terra-and-tmap-to-display-categorical-map-with-color-scheme" class="section level3 unnumbered hasAnchor">
<h3>5. Use ‘terra’ and ‘tmap’ to display categorical map with color scheme<a href="bonus_2a.html#use-terra-and-tmap-to-display-categorical-map-with-color-scheme" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Now to a more tricky topic. Recall that the last raster layer in the Worked Example, <code>nlcd</code>, contains categorical land cover data that are coded numerically. The <code>raster</code> package actually misinterpreted them as numeric data.</p>
<p>Similarly to <code>sp</code> being replaced by <code>sf</code>, <code>raster</code> is being replaced by <code>terra</code>. The conversion is easy, using the function <code>rast</code>.</p>
<p>Let’s again extract the categorical raster layer into a new object ‘NLCD.’</p>
<ul>
<li>The function <code>rast</code> of package <code>terra</code> converts a <code>raster</code> object (package <code>raster</code>) to a <code>SpatRaster</code> object (package <code>terra</code>).</li>
<li>We can use <code>as.factor</code> to tell R that this is a categorical raster layer.</li>
</ul>
<p>Here, we import the raster both as numeric and as categorical raster (factor).</p>
<div class="sourceCode" id="cb461"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb461-1"><a href="bonus_2a.html#cb461-1" aria-hidden="true" tabindex="-1"></a>NLCD <span class="ot">&lt;-</span> <span class="fu">raster</span>(rasters[<span class="dv">6</span>])</span>
<span id="cb461-2"><a href="bonus_2a.html#cb461-2" aria-hidden="true" tabindex="-1"></a>NLCD.num <span class="ot">&lt;-</span> terra<span class="sc">::</span><span class="fu">rast</span>(NLCD)</span>
<span id="cb461-3"><a href="bonus_2a.html#cb461-3" aria-hidden="true" tabindex="-1"></a>NLCD.cat <span class="ot">&lt;-</span> terra<span class="sc">::</span><span class="fu">as.factor</span>(terra<span class="sc">::</span><span class="fu">rast</span>(NLCD))</span></code></pre></div>
<p>The underlying code in package <code>terra</code> for creating categorical rasters seems to have changed lately. Here we will use a work-around based on the numeric map, <code>NLCD.num</code>. Check back later for an update.</p>
<p>What values occur in the raster? These are codes for cover types.</p>
<p>Here is a description of the cover types: <a href="https://www.mrlc.gov/data/legends/national-land-cover-database-2019-nlcd2019-legend" class="uri">https://www.mrlc.gov/data/legends/national-land-cover-database-2019-nlcd2019-legend</a></p>
<div class="sourceCode" id="cb462"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb462-1"><a href="bonus_2a.html#cb462-1" aria-hidden="true" tabindex="-1"></a><span class="fu">sort</span>(<span class="fu">unique</span>(<span class="fu">values</span>(NLCD.num)))</span></code></pre></div>
<pre><code>## [1] 11 12 31 42 52 71 90 95</code></pre>
<p>We will import a table with predefined colors (using hex color code) from the file <code>Colortable_LULC.csv</code> that is included with <code>LandGenCourse</code>.</p>
<p>This list has more entries (e.g., 21-24) than we need, because not all US land cover classes occur in the study area.</p>
<p>Check in the table below that the colors and cover types are stored as ‘character.’ (Luckily, since R 4.0, this is the new default for function <code>read.csv</code>). If they were coded as factors that could lead to errors later on.</p>
<div class="sourceCode" id="cb464"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb464-1"><a href="bonus_2a.html#cb464-1" aria-hidden="true" tabindex="-1"></a>ColTab <span class="ot">&lt;-</span> <span class="fu">read.csv</span>(<span class="fu">system.file</span>(<span class="st">&quot;extdata&quot;</span>, <span class="st">&quot;Colortable_LULC.csv&quot;</span>, </span>
<span id="cb464-2"><a href="bonus_2a.html#cb464-2" aria-hidden="true" tabindex="-1"></a>                            <span class="at">package =</span> <span class="st">&quot;LandGenCourse&quot;</span>), <span class="at">header=</span><span class="cn">TRUE</span>)</span>
<span id="cb464-3"><a href="bonus_2a.html#cb464-3" aria-hidden="true" tabindex="-1"></a>ColTab</span></code></pre></div>
<pre><code>##    value   color                    attribute
## 1     11 #456DA8                   Open Water
## 2     12 #E6EEF9           Perennial Ice/Snow
## 3     21 #E1CBCD        Developed, Open Space
## 4     22 #DC9786     Developed, Low Intensity
## 5     23 #F40100  Developed, Medium Intensity
## 6     24 #B00206    Developed, High Intensity
## 7     31 #B2AEA3 Barren Land (Rock/Sand/Clay)
## 8     41 #6BA95C             Deciduous Forest
## 9     42 #16692E             Evergreen Forest
## 10    43 #B9CA8F                 Mixed Forest
## 11    51 #AD9439                  Dwarf Scrub
## 12    52 #D5B883                  Shrub/Scrub
## 13    71 #EDEFCA         Grassland/Herbaceous
## 14    72 #D3D27C             Sedge/Herbaceous
## 15    73 #A5CD53                      Lichens
## 16    74 #88B8A1                         Moss
## 17    81 #DED73E                  Pasture/Hay
## 18    82 #AD722C             Cultivated Crops
## 19    90 #BED8F6               Woody Wetlands
## 20    95 #6EA5C4 Emergent Herbaceous Wetlands</code></pre>
<p>We merge this list with the list of values in <code>NLCD.num</code> to create a raster attribute table (RAT). An RAT is a table that contains attributes for each distinct value in a raster.</p>
<div class="sourceCode" id="cb466"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb466-1"><a href="bonus_2a.html#cb466-1" aria-hidden="true" tabindex="-1"></a>tmp <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">nlcd=</span><span class="fu">sort</span>(<span class="fu">unique</span>(<span class="fu">values</span>(NLCD.num))))</span>
<span id="cb466-2"><a href="bonus_2a.html#cb466-2" aria-hidden="true" tabindex="-1"></a>RAT <span class="ot">&lt;-</span> <span class="fu">merge</span>(tmp, ColTab, </span>
<span id="cb466-3"><a href="bonus_2a.html#cb466-3" aria-hidden="true" tabindex="-1"></a>             <span class="at">by.x=</span><span class="st">&quot;nlcd&quot;</span>, <span class="at">by.y=</span><span class="st">&quot;value&quot;</span>, <span class="at">all.x=</span><span class="cn">TRUE</span>, <span class="at">sort=</span><span class="cn">TRUE</span>)</span>
<span id="cb466-4"><a href="bonus_2a.html#cb466-4" aria-hidden="true" tabindex="-1"></a>RAT</span></code></pre></div>
<pre><code>##   nlcd   color                    attribute
## 1   11 #456DA8                   Open Water
## 2   12 #E6EEF9           Perennial Ice/Snow
## 3   31 #B2AEA3 Barren Land (Rock/Sand/Clay)
## 4   42 #16692E             Evergreen Forest
## 5   52 #D5B883                  Shrub/Scrub
## 6   71 #EDEFCA         Grassland/Herbaceous
## 7   90 #BED8F6               Woody Wetlands
## 8   95 #6EA5C4 Emergent Herbaceous Wetlands</code></pre>
<p>Now we can use tmp to plot the map with the predifined color palette</p>
<ul>
<li>We first define the dataset with <code>tm_shape</code></li>
<li>With <code>tm_raster</code>, we tell <code>tmap</code> that we want to plot raster values
<ul>
<li><code>style="cat"</code> tells <code>tmap</code> to interpret values as categories</li>
<li><code>palette</code> defines the colors</li>
<li><code>labels</code> defines the labels to be used for the categories</li>
<li><code>title</code> defines the legend caption for the categories</li>
</ul></li>
<li>With tm_layout, we tell tmap where to place the legend (outside to the right of map)</li>
<li>With <code>tm_grid(lines=FALSE)</code> we tell <code>tmap</code> to show the coordinates along x and y axes but to suppress grid lines that would be drawn on top of the map (change it to TRUE to see the effect)</li>
</ul>
<div class="sourceCode" id="cb468"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb468-1"><a href="bonus_2a.html#cb468-1" aria-hidden="true" tabindex="-1"></a>Map3 <span class="ot">&lt;-</span> <span class="fu">tm_shape</span>(NLCD.num) <span class="sc">+</span> </span>
<span id="cb468-2"><a href="bonus_2a.html#cb468-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">tm_raster</span>(<span class="at">style=</span><span class="st">&quot;cat&quot;</span>, <span class="at">palette=</span>RAT<span class="sc">$</span>color, <span class="at">labels=</span>RAT<span class="sc">$</span>attribute,</span>
<span id="cb468-3"><a href="bonus_2a.html#cb468-3" aria-hidden="true" tabindex="-1"></a>            <span class="at">title=</span><span class="st">&quot;Land cover&quot;</span>) <span class="sc">+</span></span>
<span id="cb468-4"><a href="bonus_2a.html#cb468-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">tm_layout</span>(<span class="at">legend.outside=</span><span class="cn">TRUE</span>, <span class="at">legend.outside.position=</span><span class="st">&quot;right&quot;</span>) <span class="sc">+</span></span>
<span id="cb468-5"><a href="bonus_2a.html#cb468-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">tm_grid</span>(<span class="at">lines=</span><span class="cn">FALSE</span>)</span>
<span id="cb468-6"><a href="bonus_2a.html#cb468-6" aria-hidden="true" tabindex="-1"></a>Map3</span></code></pre></div>
<pre><code>## Some legend labels were too wide. These labels have been resized to 0.65. Increase legend.width (argument of tm_layout) to make the legend wider and therefore the labels larger.</code></pre>
<p><img src="02-Week02_files/figure-html/unnamed-chunk-75-1.png" width="672" /></p>
<p>Let’s beef it up a bit. We can do so by adding layers to <code>Map3</code>:</p>
<ul>
<li>To add an additional layer with the sampling sites to the map, we define the data for the layer with <code>tm_shape</code>.</li>
<li>We define how the points should be symbolized with <code>tm_symbols</code>.</li>
<li>We add a North arrow with <code>tm_compass</code> , using default settings.</li>
<li>We add a scale bar with <code>tm_scale_bar</code> and set the background to a semitransparent (<code>bg.alpha=0.5</code>) light gray (<code>bg.color="lightgray"</code>).</li>
</ul>
<div class="sourceCode" id="cb470"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb470-1"><a href="bonus_2a.html#cb470-1" aria-hidden="true" tabindex="-1"></a>Map4 <span class="ot">&lt;-</span> Map3 <span class="sc">+</span> <span class="fu">tm_shape</span>(Sites.sf) <span class="sc">+</span></span>
<span id="cb470-2"><a href="bonus_2a.html#cb470-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">tm_symbols</span>(<span class="at">size=</span><span class="fl">0.4</span>, <span class="at">col=</span><span class="st">&quot;yellow&quot;</span>, <span class="at">border.col=</span><span class="st">&quot;red&quot;</span>) <span class="sc">+</span></span>
<span id="cb470-3"><a href="bonus_2a.html#cb470-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">tm_compass</span>() <span class="sc">+</span> <span class="fu">tm_scale_bar</span>(<span class="at">bg.color=</span><span class="st">&quot;lightgray&quot;</span>, <span class="at">bg.alpha=</span><span class="fl">0.5</span>)</span>
<span id="cb470-4"><a href="bonus_2a.html#cb470-4" aria-hidden="true" tabindex="-1"></a>Map4</span></code></pre></div>
<pre><code>## Some legend labels were too wide. These labels have been resized to 0.65. Increase legend.width (argument of tm_layout) to make the legend wider and therefore the labels larger.</code></pre>
<p><img src="02-Week02_files/figure-html/unnamed-chunk-76-1.png" width="672" /></p>
<p>This time, we’ll export the map to a PDF file.</p>
<div class="sourceCode" id="cb472"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb472-1"><a href="bonus_2a.html#cb472-1" aria-hidden="true" tabindex="-1"></a><span class="fu">tmap_save</span>(Map4, <span class="fu">paste0</span>(here<span class="sc">::</span><span class="fu">here</span>(), <span class="st">&quot;/output/RasterMap.pdf&quot;</span>), <span class="at">height=</span><span class="dv">6</span>, <span class="at">width=</span><span class="dv">8</span>)</span></code></pre></div>
<pre><code>## Map saved to /Users/hhwagner1/Desktop/R_GitHub_Projects/LandGenCourse_book/output/RasterMap.pdf</code></pre>
<pre><code>## Size: 8 by 6 inches</code></pre>

</div>
</div>
<!-- </div> -->
            </section>

          </div>
        </div>
      </div>
<a href="r-exercise-week-2.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="Week3.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": true,
"facebook": false,
"twitter": false,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": false
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": null,
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "section"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
