<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>5. Run simulator using a previously defined parameter set | Landscape Genetic Data Analysis with R</title>
  <meta name="description" content="This is a web-interface to the teaching materials for the lab course ‘Landscape Genetic Data Analysis with R’ associated with the distributed graduate course ‘DGS Landscape Genetics.’ The output format is bookdown::gitbook." />
  <meta name="generator" content="bookdown 0.25 and GitBook 2.6.7" />

  <meta property="og:title" content="5. Run simulator using a previously defined parameter set | Landscape Genetic Data Analysis with R" />
  <meta property="og:type" content="book" />
  
  <meta property="og:description" content="This is a web-interface to the teaching materials for the lab course ‘Landscape Genetic Data Analysis with R’ associated with the distributed graduate course ‘DGS Landscape Genetics.’ The output format is bookdown::gitbook." />
  <meta name="github-repo" content="hhwagner1/LandGenCourse_book" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="5. Run simulator using a previously defined parameter set | Landscape Genetic Data Analysis with R" />
  
  <meta name="twitter:description" content="This is a web-interface to the teaching materials for the lab course ‘Landscape Genetic Data Analysis with R’ associated with the distributed graduate course ‘DGS Landscape Genetics.’ The output format is bookdown::gitbook." />
  

<meta name="author" content="Editor: Helene Wagner (University of Toronto)" />


<meta name="date" content="2022-04-03" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="run-simulations-and-analyze-results.html"/>
<link rel="next" href="r-exercise-week-8.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>
<script src="libs/kePrint-0.0.1/kePrint.js"></script>
<link href="libs/lightable-0.0.1/lightable.css" rel="stylesheet" />
<script src="libs/htmlwidgets-1.5.4/htmlwidgets.js"></script>
<script src="libs/d3-3.5.6/d3.min.js"></script>
<link href="libs/profvis-0.3.6.9000/profvis.css" rel="stylesheet" />
<script src="libs/profvis-0.3.6.9000/profvis.js"></script>
<script src="libs/profvis-0.3.6.9000/scroll.js"></script>
<link href="libs/highlight-6.2.0/textmate.css" rel="stylesheet" />
<script src="libs/highlight-6.2.0/highlight.js"></script>
<script src="libs/profvis-binding-0.3.7/profvis.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>


<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">Landscape Genetic Data Analysis with R</a></li>

<li class="divider"></li>
<li class="part"><span><b>Getting Started<span></span></b></span></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Introduction<span></span></a>
<ul>
<li class="chapter" data-level="1.1" data-path="how-to-use-this-book.html"><a href="how-to-use-this-book.html"><i class="fa fa-check"></i><b>1.1</b> How to use this Book<span></span></a>
<ul>
<li><a href="how-to-use-this-book.html#book-structure">1. Book Structure<span></span></a></li>
<li><a href="how-to-use-this-book.html#find-what-is-relevant-for-you">2. Find what is relevant for you<span></span></a></li>
<li><a href="how-to-use-this-book.html#course-r-package-landgencourse">3. Course R package ‘LandGenCourse’<span></span></a></li>
</ul></li>
<li class="chapter" data-level="1.2" data-path="list-of-packages.html"><a href="list-of-packages.html"><i class="fa fa-check"></i><b>1.2</b> List of R Packages by Vignette<span></span></a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="review-of-r-skills.html"><a href="review-of-r-skills.html"><i class="fa fa-check"></i><b>2</b> Review of R Skills<span></span></a>
<ul>
<li class="chapter" data-level="2.1" data-path="basic-r.html"><a href="basic-r.html"><i class="fa fa-check"></i><b>2.1</b> Basic R Programming<span></span></a></li>
<li><a href="overview.html#overview">1. Overview<span></span></a></li>
<li><a href="data-types.html#data-types">2. Data Types<span></span></a>
<ul>
<li><a href="data-types.html#numeric-data-types">2.1. Numeric Data Types<span></span></a></li>
<li><a href="data-types.html#characters">2.2. Characters<span></span></a></li>
<li><a href="data-types.html#factors">2.3. Factors<span></span></a></li>
<li><a href="data-types.html#logical-types">2.4. Logical Types<span></span></a></li>
</ul></li>
<li><a href="data-containers.html#data-containers">3. Data Containers<span></span></a>
<ul>
<li><a href="data-containers.html#vectors">3.1. Vectors<span></span></a></li>
<li><a href="data-containers.html#matrices">3.2. Matrices<span></span></a></li>
<li><a href="data-containers.html#lists">3.3. Lists<span></span></a></li>
<li><a href="data-containers.html#data-frames">3.4. Data Frames<span></span></a></li>
</ul></li>
<li><a href="programming.html#programming">4. Programming<span></span></a>
<ul>
<li><a href="programming.html#function-writing">4.1. Function Writing<span></span></a></li>
<li><a href="programming.html#variable-scope">4.2. Variable Scope<span></span></a></li>
<li><a href="programming.html#decision-making">4.3. Decision Making<span></span></a></li>
<li><a href="programming.html#flow-control">4.4. Flow Control<span></span></a></li>
</ul></li>
<li class="chapter" data-level="2.2" data-path="r-graphics.html"><a href="r-graphics.html"><i class="fa fa-check"></i><b>2.2</b> R Graphics<span></span></a></li>
<li><a href="overview-1.html#overview-1">1. Overview<span></span></a></li>
<li><a href="univariate-plots.html#univariate-plots">2. Univariate plots<span></span></a>
<ul>
<li><a href="univariate-plots.html#barplots-and-histograms">2.1. Barplots and histograms<span></span></a></li>
<li><a href="univariate-plots.html#plotting-parameters">2.2. Plotting Parameters<span></span></a></li>
<li><a href="univariate-plots.html#density-plots">2.3. Density Plots<span></span></a></li>
</ul></li>
<li><a href="bivariate-plots.html#bivariate-plots">3. Bivariate plots<span></span></a>
<ul>
<li><a href="bivariate-plots.html#overlaying-plots-by-grouping">3.1. Overlaying Plots by grouping<span></span></a></li>
<li><a href="bivariate-plots.html#boxplots">3.2. Boxplots<span></span></a></li>
<li><a href="bivariate-plots.html#scatter-plots">3.3. Scatter Plots<span></span></a></li>
</ul></li>
<li><a href="advanced-plotting.html#advanced-plotting">4. Advanced plotting<span></span></a>
<ul>
<li><a href="advanced-plotting.html#multiple-plots">4.1. Multiple Plots<span></span></a></li>
<li><a href="advanced-plotting.html#color-palettes">4.2. Color Palettes<span></span></a></li>
<li><a href="advanced-plotting.html#rcolorbrewer">4.3. RColorBrewer<span></span></a></li>
<li><a href="advanced-plotting.html#saving-imagery">4.4. Saving Imagery<span></span></a></li>
<li><a href="advanced-plotting.html#interactive-graphics">4.5. Interactive Graphics<span></span></a></li>
</ul></li>
<li class="chapter" data-level="2.3" data-path="resources.html"><a href="resources.html"><i class="fa fa-check"></i><b>2.3</b> Further R resources<span></span></a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="github-for-group-projects.html"><a href="github-for-group-projects.html"><i class="fa fa-check"></i><b>3</b> GitHub for Group Projects<span></span></a>
<ul>
<li class="chapter" data-level="3.1" data-path="github.html"><a href="github.html"><i class="fa fa-check"></i><b>3.1</b> Why use GitHub?<span></span></a>
<ul>
<li><a href="github.html#goals">1. Goals<span></span></a></li>
<li><a href="github.html#version">2. Watch course video: Version Control 101<span></span></a></li>
<li><a href="github.html#slides">3. Preview slides<span></span></a></li>
<li><a href="github.html#next-steps">4. Next steps<span></span></a></li>
</ul></li>
<li class="chapter" data-level="3.2" data-path="install.html"><a href="install.html"><i class="fa fa-check"></i><b>3.2</b> Installation<span></span></a>
<ul>
<li><a href="install.html#goals-1">1. Goals<span></span></a></li>
<li class="chapter" data-level="3.2.1" data-path="install.html"><a href="install.html#how-to-get-github-and-gitkraken-for-free"><i class="fa fa-check"></i><b>3.2.1</b> 2. How to get GitHub and GitKraken for free<span></span></a></li>
<li><a href="install.html#happygit">3. Happy Git with R<span></span></a></li>
<li><a href="install.html#gitvideos">4. Video tutorials: installation<span></span></a></li>
<li><a href="install.html#join">5. Joining your group’s project on GitHub<span></span></a></li>
<li><a href="install.html#repo">6. Setting up a group repo<span></span></a></li>
<li><a href="install.html#next-steps-1">7. Next steps<span></span></a></li>
</ul></li>
<li class="chapter" data-level="3.3" data-path="workflow.html"><a href="workflow.html"><i class="fa fa-check"></i><b>3.3</b> Workflows<span></span></a>
<ul>
<li><a href="workflow.html#goals-2">1. Goals<span></span></a></li>
<li><a href="workflow.html#step-by-step-tutorials">2. Step-by-step tutorials<span></span></a></li>
<li><a href="workflow.html#further-resources">3. Further resources<span></span></a></li>
</ul></li>
</ul></li>
<li class="part"><span><b>Basic Topics<span></span></b></span></li>
<li class="chapter" data-level="4" data-path="Week1.html"><a href="Week1.html"><i class="fa fa-check"></i><b>4</b> Week 1: Importing Genetic Data<span></span></a>
<ul>
<li class="chapter" data-level="4.1" data-path="video_1.html"><a href="video_1.html"><i class="fa fa-check"></i><b>4.1</b> View Course Video<span></span></a>
<ul>
<li><a href="video_1.html#embedded-video">1. Embedded Video<span></span></a></li>
<li><a href="video_1.html#preview-slides">2. Preview Slides<span></span></a></li>
</ul></li>
<li class="chapter" data-level="4.2" data-path="tutorial_1.html"><a href="tutorial_1.html"><i class="fa fa-check"></i><b>4.2</b> Interactive Tutorial 1<span></span></a>
<ul>
<li><a href="tutorial_1.html#list-of-r-commands-covered-this-week">1. List of R commands covered this week<span></span></a></li>
<li><a href="tutorial_1.html#general-instructions">2. General Instructions<span></span></a></li>
</ul></li>
<li class="chapter" data-level="4.3" data-path="WE_1.html"><a href="WE_1.html"><i class="fa fa-check"></i><b>4.3</b> Worked Example<span></span></a></li>
<li><a href="overview-of-worked-example.html#overview-of-worked-example">1. Overview of Worked Example<span></span></a>
<ul>
<li><a href="overview-of-worked-example.html#a-goals">a) Goals<span></span></a></li>
<li><a href="overview-of-worked-example.html#b-data-set">b) Data set<span></span></a></li>
<li><a href="overview-of-worked-example.html#c-required-r-packages">c) Required R packages<span></span></a></li>
</ul></li>
<li><a href="import-data-from-.csv-file.html#import-data-from-.csv-file">2. Import data from .csv file<span></span></a>
<ul>
<li><a href="import-data-from-.csv-file.html#a-preparation">a) Preparation<span></span></a></li>
<li><a href="import-data-from-.csv-file.html#b-excel-file">b) Excel file<span></span></a></li>
<li><a href="import-data-from-.csv-file.html#c-import-with-function-read.csv">c) Import with function ‘read.csv’<span></span></a></li>
<li><a href="import-data-from-.csv-file.html#d-create-id-variable">d) Create ID variable<span></span></a></li>
<li><a href="import-data-from-.csv-file.html#e-export-with-function-write.csv">e) Export with function ‘write.csv’<span></span></a></li>
</ul></li>
<li><a href="create-genind-object.html#create-genind-object">3. Create ‘genind’ Object<span></span></a>
<ul>
<li><a href="create-genind-object.html#a-read-the-helpfile">a) Read the helpfile<span></span></a></li>
<li><a href="create-genind-object.html#b-set-all-arguments-of-function-df2genind">b) Set all arguments of function ‘df2genind’<span></span></a></li>
<li><a href="create-genind-object.html#c-check-imported-data">c) Check imported data<span></span></a></li>
<li><a href="create-genind-object.html#d-summarize-genind-object">d) Summarize ‘genind’ object<span></span></a></li>
</ul></li>
<li><a href="view-information-stored-in-genind-object.html#view-information-stored-in-genind-object">4. View information stored in ‘genind’ object<span></span></a>
<ul>
<li><a href="view-information-stored-in-genind-object.html#a-slot-tab">a) Slot <code>@tab</code><span></span></a></li>
<li><a href="view-information-stored-in-genind-object.html#b-slot-loc.n.all">b) Slot <code>@loc.n.all</code><span></span></a></li>
<li><a href="view-information-stored-in-genind-object.html#c-slot-loc.fac">c) Slot <code>@loc.fac</code><span></span></a></li>
<li><a href="view-information-stored-in-genind-object.html#d-slot-all.names">d) Slot <code>@all.names</code><span></span></a></li>
<li><a href="view-information-stored-in-genind-object.html#e-further-slots">e) Further slots<span></span></a></li>
</ul></li>
<li><a href="import-data-with-gstudio-package.html#import-data-with-gstudio-package">5. Import data with ‘gstudio’ package<span></span></a>
<ul>
<li><a href="import-data-with-gstudio-package.html#a-function-read_population">a) Function ‘read_population’<span></span></a></li>
<li><a href="import-data-with-gstudio-package.html#b-set-arguments">b) Set arguments<span></span></a></li>
<li><a href="import-data-with-gstudio-package.html#c-check-imported-data-1">c) Check imported data<span></span></a></li>
<li><a href="import-data-with-gstudio-package.html#d-add-frogid">d) Add FrogID<span></span></a></li>
<li><a href="import-data-with-gstudio-package.html#e-convert-to-genind-object">e) Convert to ‘genind’ object<span></span></a></li>
</ul></li>
<li><a href="importing-snp-data.html#importing-snp-data">6. Importing SNP data<span></span></a>
<ul>
<li><a href="importing-snp-data.html#a-import-from-text-file">a) Import from text file<span></span></a></li>
<li><a href="importing-snp-data.html#b-import-from-vcf-file-to-genind-object">b) Import from vcf file to genind object<span></span></a></li>
<li><a href="importing-snp-data.html#c-import-from-vcf-file-to-genlight-object">c) Import from vcf file to genlight object<span></span></a></li>
</ul></li>
<li><a href="r-exercise-week-1.html#r-exercise-week-1">7. R exercise Week 1<span></span></a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="Week2.html"><a href="Week2.html"><i class="fa fa-check"></i><b>5</b> Week 2: Spatial Data<span></span></a>
<ul>
<li class="chapter" data-level="5.1" data-path="video_2.html"><a href="video_2.html"><i class="fa fa-check"></i><b>5.1</b> View Course Video<span></span></a>
<ul>
<li><a href="video_2.html#embedded-video-1">1. Embedded Video<span></span></a></li>
<li><a href="video_2.html#preview-slides-1">2. Preview Slides<span></span></a></li>
</ul></li>
<li class="chapter" data-level="5.2" data-path="tutorial_2.html"><a href="tutorial_2.html"><i class="fa fa-check"></i><b>5.2</b> Interactive Tutorial 2<span></span></a>
<ul>
<li><a href="tutorial_2.html#list-of-r-commands-covered-this-week-1">1. List of R commands covered this week<span></span></a></li>
<li><a href="tutorial_2.html#general-instructions-1">2. General Instructions<span></span></a></li>
</ul></li>
<li class="chapter" data-level="5.3" data-path="WE_2.html"><a href="WE_2.html"><i class="fa fa-check"></i><b>5.3</b> Worked Example<span></span></a></li>
<li><a href="overview-of-worked-example-1.html#overview-of-worked-example-1">1. Overview of Worked Example<span></span></a>
<ul>
<li><a href="overview-of-worked-example-1.html#a-goals-1">a) Goals<span></span></a></li>
<li><a href="overview-of-worked-example-1.html#b-data-set-1">b) Data set<span></span></a></li>
<li><a href="overview-of-worked-example-1.html#c-required-r-packages-1">c) Required R packages<span></span></a></li>
</ul></li>
<li><a href="import-site-data-from-.csv-file.html#import-site-data-from-.csv-file">2. Import site data from .csv file<span></span></a>
<ul>
<li><a href="import-site-data-from-.csv-file.html#a-import-data-into-spatialpointsdataframe">a) Import data into SpatialPointsDataFrame<span></span></a></li>
<li><a href="import-site-data-from-.csv-file.html#b-add-spatial-reference-data">b) Add spatial reference data<span></span></a></li>
<li><a href="import-site-data-from-.csv-file.html#c-change-projection">c) Change projection<span></span></a></li>
<li><a href="import-site-data-from-.csv-file.html#d-access-data-in-spatialpointsdataframe">d) Access data in ‘SpatialPointsDataFrame’<span></span></a></li>
</ul></li>
<li><a href="display-raster-data-and-overlay-sampling-locations-extract-data.html#display-raster-data-and-overlay-sampling-locations-extract-data">3. Display raster data and overlay sampling locations, extract data<span></span></a>
<ul>
<li><a href="display-raster-data-and-overlay-sampling-locations-extract-data.html#a-display-raster-data">a) Display raster data<span></span></a></li>
<li><a href="display-raster-data-and-overlay-sampling-locations-extract-data.html#b-change-color-ramp-add-sampling-locations">b) Change color ramp, add sampling locations<span></span></a></li>
<li><a href="display-raster-data-and-overlay-sampling-locations-extract-data.html#c-extract-raster-values-at-sampling-locations">c) Extract raster values at sampling locations<span></span></a></li>
</ul></li>
<li><a href="calculate-landscape-metrics.html#calculate-landscape-metrics">4. Calculate landscape metrics<span></span></a>
<ul>
<li><a href="calculate-landscape-metrics.html#a-calculate-patch--class--and-landscape-level-landscape-metrics">a) Calculate patch-, class- and landscape level landscape metrics<span></span></a></li>
<li><a href="calculate-landscape-metrics.html#b-calculate-patch-level-landscape-metrics-for-evergreen-forest">b) Calculate patch-level landscape metrics for ‘Evergreen Forest’<span></span></a></li>
<li><a href="calculate-landscape-metrics.html#c-extract-forest-patch-size-at-samplig-locations">c) Extract forest patch size at samplig locations<span></span></a></li>
<li><a href="calculate-landscape-metrics.html#d-plot-a-bubble-map-of-forest-patch-size-at-sampling-locations">d) Plot a bubble map of forest patch size at sampling locations<span></span></a></li>
</ul></li>
<li><a href="sample-landscape-metrics-within-buffer-around-sampling-locations.html#sample-landscape-metrics-within-buffer-around-sampling-locations">5. Sample landscape metrics within buffer around sampling locations<span></span></a>
<ul>
<li><a href="sample-landscape-metrics-within-buffer-around-sampling-locations.html#b-extract-landscape-metric-of-choice-for-a-single-cover-type-as-vector">b) Extract landscape metric of choice for a single cover type (as vector)<span></span></a></li>
<li><a href="sample-landscape-metrics-within-buffer-around-sampling-locations.html#c-extract-landscape-metric-of-choice-for-all-cover-types-as-data-frame">c) Extract landscape metric of choice for all cover types (as data frame)<span></span></a></li>
<li><a href="sample-landscape-metrics-within-buffer-around-sampling-locations.html#d-extract-all-landscape-metrics-for-a-single-cover-type-as-data-frame">d) Extract all landscape metrics for a single cover type (as data frame)<span></span></a></li>
</ul></li>
<li><a href="r-exercise-week-2.html#r-exercise-week-2">6. R Exercise Week 2<span></span></a></li>
<li class="chapter" data-level="5.4" data-path="bonus_2a.html"><a href="bonus_2a.html"><i class="fa fa-check"></i><b>5.4</b> Bonus: ‘sf’ and ‘terra’<span></span></a></li>
<li><a href="overview-2.html#overview-2">1. Overview<span></span></a>
<ul>
<li><a href="overview-2.html#a-goals-2">a) Goals<span></span></a></li>
<li><a href="overview-2.html#b-required-r-packages">b) Required R packages<span></span></a></li>
</ul></li>
<li><a href="convert-spatialpointsdataframe-to-sf-object.html#convert-spatialpointsdataframe-to-sf-object">2. Convert ‘SpatialPointsDataFrame’ to ‘sf’ object<span></span></a></li>
<li><a href="plot-variables-first-10-in-data-set.html#plot-variables-first-10-in-data-set">3. Plot variables (first 10 in data set)<span></span></a>
<ul>
<li><a href="plot-variables-first-10-in-data-set.html#a-plot-geometry">a) Plot geometry<span></span></a></li>
<li><a href="plot-variables-first-10-in-data-set.html#b-plot-attributes-in-space">b) Plot attributes in space<span></span></a></li>
<li><a href="plot-variables-first-10-in-data-set.html#c-create-a-static-bubble-plot-with-tmap">c) Create a static bubble plot with ‘tmap’<span></span></a></li>
<li><a href="plot-variables-first-10-in-data-set.html#d-create-an-interactive-bubble-plot-with-tmap">d) Create an interactive bubble plot with ‘tmap’<span></span></a></li>
<li><a href="plot-variables-first-10-in-data-set.html#e-export-maps">e) Export maps<span></span></a></li>
</ul></li>
<li><a href="export-site-data-as-esri-shapefile.html#export-site-data-as-esri-shapefile">4. Export site data as ESRI shapefile<span></span></a></li>
<li><a href="use-terra-and-tmap-to-display-categorical-map-with-color-scheme.html#use-terra-and-tmap-to-display-categorical-map-with-color-scheme">5. Use ‘terra’ and ‘tmap’ to display categorical map with color scheme<span></span></a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="Week3.html"><a href="Week3.html"><i class="fa fa-check"></i><b>6</b> Week 3: Genetic Diversity<span></span></a>
<ul>
<li class="chapter" data-level="6.1" data-path="video_3.html"><a href="video_3.html"><i class="fa fa-check"></i><b>6.1</b> View Course Video<span></span></a>
<ul>
<li><a href="video_3.html#embedded-video-2">1. Embedded Video<span></span></a></li>
<li><a href="video_3.html#video-part-1">Video, Part 1<span></span></a></li>
<li><a href="video_3.html#video-part-2">Video, Part 2<span></span></a></li>
<li><a href="video_3.html#preview-slides-2">Preview Slides<span></span></a></li>
</ul></li>
<li class="chapter" data-level="6.2" data-path="tutorial_3.html"><a href="tutorial_3.html"><i class="fa fa-check"></i><b>6.2</b> Interactive Tutorial 3<span></span></a>
<ul>
<li><a href="tutorial_3.html#list-of-r-commands-covered-this-week-2">1. List of R commands covered this week<span></span></a></li>
<li><a href="tutorial_3.html#general-instructions-2">2. General Instructions<span></span></a></li>
</ul></li>
<li class="chapter" data-level="6.3" data-path="WE_3.html"><a href="WE_3.html"><i class="fa fa-check"></i><b>6.3</b> Worked Example<span></span></a></li>
<li><a href="overview-of-worked-example-2.html#overview-of-worked-example-2">1. Overview of Worked Example<span></span></a>
<ul>
<li><a href="overview-of-worked-example-2.html#a-goals-3">a) Goals<span></span></a></li>
<li><a href="overview-of-worked-example-2.html#b-data-set-2">b) Data set<span></span></a></li>
<li><a href="overview-of-worked-example-2.html#c-required-r-packages-2">c) Required R packages<span></span></a></li>
</ul></li>
<li><a href="basic-checking-of-markers-and-populations.html#basic-checking-of-markers-and-populations">2. Basic checking of markers and populations<span></span></a>
<ul>
<li><a href="basic-checking-of-markers-and-populations.html#a-re-create-genind-object">a) Re-create genind object<span></span></a></li>
<li><a href="basic-checking-of-markers-and-populations.html#b-check-that-markers-are-polymorphic">b) Check that markers are polymorphic<span></span></a></li>
<li><a href="basic-checking-of-markers-and-populations.html#c-check-for-deviations-from-hardy-weinberg-equilibrium-hwe">c) Check for deviations from Hardy-Weinberg equilibrium (HWE)<span></span></a></li>
<li><a href="basic-checking-of-markers-and-populations.html#d-check-for-linkage-disequilibrium-ld">d) Check for linkage disequilibrium (LD)<span></span></a></li>
<li><a href="basic-checking-of-markers-and-populations.html#e-check-for-null-alleles">e) Check for null alleles<span></span></a></li>
<li><a href="basic-checking-of-markers-and-populations.html#f-overall-interpretation">f) Overall interpretation<span></span></a></li>
</ul></li>
<li><a href="assess-genetic-diversity.html#assess-genetic-diversity">3. Assess genetic diversity<span></span></a>
<ul>
<li><a href="assess-genetic-diversity.html#a-rarefied-allelic-richness">a) Rarefied allelic richness<span></span></a></li>
<li><a href="assess-genetic-diversity.html#b-observed-and-expected-heterozygosity">b) Observed and expected heterozygosity<span></span></a></li>
<li><a href="assess-genetic-diversity.html#c-create-table-with-sitel-level-genetic-diversity-measures">c) Create table with sitel-level genetic diversity measures<span></span></a></li>
</ul></li>
<li><a href="aggregate-genetic-data-at-population-level-allele-frequencies.html#aggregate-genetic-data-at-population-level-allele-frequencies">4. Aggregate genetic data at population level (allele frequencies)<span></span></a></li>
<li><a href="r-exercise-week-3.html#r-exercise-week-3">5. R Exercise Week 3<span></span></a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="Week4.html"><a href="Week4.html"><i class="fa fa-check"></i><b>7</b> Week 4: Metapopulations<span></span></a>
<ul>
<li class="chapter" data-level="7.1" data-path="video_4.html"><a href="video_4.html"><i class="fa fa-check"></i><b>7.1</b> View Course Video<span></span></a>
<ul>
<li><a href="video_4.html#embedded-video-3">1. Embedded Video<span></span></a></li>
<li><a href="video_4.html#video-part-1-1">Video, Part 1<span></span></a></li>
<li><a href="video_4.html#video-part-2-1">Video, Part 2<span></span></a></li>
<li><a href="video_4.html#preview-slides-3">Preview Slides<span></span></a></li>
</ul></li>
<li class="chapter" data-level="7.2" data-path="tutorial_4.html"><a href="tutorial_4.html"><i class="fa fa-check"></i><b>7.2</b> Interactive Tutorial 4<span></span></a>
<ul>
<li><a href="tutorial_4.html#list-of-r-commands-covered-this-week-3">1. List of R commands covered this week<span></span></a></li>
<li><a href="tutorial_4.html#general-instructions-3">2. General Instructions<span></span></a></li>
</ul></li>
<li class="chapter" data-level="7.3" data-path="WE_4.html"><a href="WE_4.html"><i class="fa fa-check"></i><b>7.3</b> Worked Example<span></span></a></li>
<li><a href="overview-of-worked-example-3.html#overview-of-worked-example-3">1. Overview of Worked Example<span></span></a>
<ul>
<li><a href="overview-of-worked-example-3.html#a-goals-4">a) Goals<span></span></a></li>
<li><a href="overview-of-worked-example-3.html#b-data-set-3">b) Data set<span></span></a></li>
<li><a href="overview-of-worked-example-3.html#c-required-r-libraries">c) Required R libraries<span></span></a></li>
<li><a href="overview-of-worked-example-3.html#d-import-data">d) Import data<span></span></a></li>
</ul></li>
<li><a href="spatial-distribution-of-genetic-structure.html#spatial-distribution-of-genetic-structure">2. Spatial distribution of genetic structure<span></span></a>
<ul>
<li><a href="spatial-distribution-of-genetic-structure.html#a-creating-a-genind-object-with-the-hierarchical-data-set">a) Creating a ‘genind’ object with the hierarchical data set<span></span></a></li>
<li><a href="spatial-distribution-of-genetic-structure.html#b-hierarchical-amova">b) Hierarchical AMOVA<span></span></a></li>
</ul></li>
<li><a href="what-determines-genetic-differentiation-among-sites.html#what-determines-genetic-differentiation-among-sites">3. What determines genetic differentiation among sites?<span></span></a>
<ul>
<li><a href="what-determines-genetic-differentiation-among-sites.html#a-correlation-matrix">a) Correlation matrix<span></span></a></li>
<li><a href="what-determines-genetic-differentiation-among-sites.html#b-scatterplots">b) Scatterplots<span></span></a></li>
<li><a href="what-determines-genetic-differentiation-among-sites.html#c-regression-model">c) Regression model<span></span></a></li>
<li><a href="what-determines-genetic-differentiation-among-sites.html#d-which-populations-dont-fit-the-general-pattern">d) Which populations don’t fit the general pattern?<span></span></a></li>
<li><a href="what-determines-genetic-differentiation-among-sites.html#e-regression-model-without-outliers">e) Regression model without outliers<span></span></a></li>
</ul></li>
<li><a href="what-determines-genetic-diversity.html#what-determines-genetic-diversity">4. What determines genetic diversity?<span></span></a>
<ul>
<li><a href="what-determines-genetic-diversity.html#a-correlation-matrix-1">a) Correlation matrix<span></span></a></li>
<li><a href="what-determines-genetic-diversity.html#b-regression-models">b) Regression models<span></span></a></li>
</ul></li>
<li><a href="are-genetic-differentiation-and-diversity-related.html#are-genetic-differentiation-and-diversity-related">5. Are genetic differentiation and diversity related?<span></span></a></li>
<li><a href="effect-of-recent-extinction-events.html#effect-of-recent-extinction-events">6. Effect of recent extinction events<span></span></a>
<ul>
<li><a href="effect-of-recent-extinction-events.html#a-effect-of-patch-extinction-event-temporal-data-set">a) Effect of patch extinction event (temporal data set)<span></span></a></li>
<li><a href="effect-of-recent-extinction-events.html#b-power-analysis">b) Power analysis<span></span></a></li>
<li><a href="effect-of-recent-extinction-events.html#c-sample-size-calculation">c) Sample size calculation<span></span></a></li>
</ul></li>
<li><a href="r-exercise-week-4.html#r-exercise-week-4">7. R Exercise Week 4<span></span></a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="Week5.html"><a href="Week5.html"><i class="fa fa-check"></i><b>8</b> Week 5: Spatial Statistics<span></span></a>
<ul>
<li class="chapter" data-level="8.1" data-path="video_5.html"><a href="video_5.html"><i class="fa fa-check"></i><b>8.1</b> View Course Video<span></span></a>
<ul>
<li><a href="video_5.html#embedded-video-4">1. Embedded Video<span></span></a></li>
<li><a href="video_5.html#preview-slides-4">Preview Slides<span></span></a></li>
</ul></li>
<li class="chapter" data-level="8.2" data-path="tutorial_5.html"><a href="tutorial_5.html"><i class="fa fa-check"></i><b>8.2</b> Interactive Tutorial 5<span></span></a>
<ul>
<li><a href="tutorial_5.html#list-of-r-commands-covered-this-week-4">1. List of R commands covered this week<span></span></a></li>
<li><a href="tutorial_5.html#general-instructions-4">2. General Instructions<span></span></a></li>
</ul></li>
<li class="chapter" data-level="8.3" data-path="WE_5.html"><a href="WE_5.html"><i class="fa fa-check"></i><b>8.3</b> Worked Example<span></span></a></li>
<li><a href="overview-of-worked-example-4.html#overview-of-worked-example-4">1. Overview of Worked Example<span></span></a>
<ul>
<li><a href="overview-of-worked-example-4.html#a-goals-5">a) Goals<span></span></a></li>
<li><a href="overview-of-worked-example-4.html#b-data-set-4">b) Data set<span></span></a></li>
<li><a href="overview-of-worked-example-4.html#c-required-r-packages-3">c) Required R packages<span></span></a></li>
</ul></li>
<li><a href="data-import-and-manipulation.html#data-import-and-manipulation">2. Data import and manipulation<span></span></a>
<ul>
<li><a href="data-import-and-manipulation.html#a-create-an-ecogen-object">a) Create an ecogen object<span></span></a></li>
<li><a href="data-import-and-manipulation.html#b-aggregate-to-ecopop">b) Aggregate to ‘ecopop’<span></span></a></li>
<li><a href="data-import-and-manipulation.html#b-add-site-level-data">b) Add site-level data<span></span></a></li>
<li><a href="data-import-and-manipulation.html#c-export-to-adegenet-and-gstudio">c) Export to ‘adegenet’ and ‘gstudio’<span></span></a></li>
</ul></li>
<li><a href="calculate-genetic-distances.html#calculate-genetic-distances">3. Calculate genetic distances<span></span></a>
<ul>
<li><a href="calculate-genetic-distances.html#a-genetic-distances-calculated-from-genind-object">a) Genetic distances calculated from genind object<span></span></a></li>
<li><a href="calculate-genetic-distances.html#b-more-distance-matrices-with-gstudio">b) More distance matrices with ‘gstudio’<span></span></a></li>
<li><a href="calculate-genetic-distances.html#c-assemble-distance-matrices">c) Assemble distance matrices<span></span></a></li>
<li><a href="calculate-genetic-distances.html#d-export-genetic-distance-matrices">d) Export genetic distance matrices<span></span></a></li>
</ul></li>
<li><a href="perform-a-mantel-test-to-test-for-ibd.html#perform-a-mantel-test-to-test-for-ibd">4. Perform a Mantel test to test for IBD<span></span></a>
<ul>
<li><a href="perform-a-mantel-test-to-test-for-ibd.html#a-visually-check-linearity">a) Visually check linearity<span></span></a></li>
<li><a href="perform-a-mantel-test-to-test-for-ibd.html#b-perform-mantel-test">b) Perform Mantel test<span></span></a></li>
</ul></li>
<li><a href="create-mantel-correlogram-for-genetic-data.html#create-mantel-correlogram-for-genetic-data">5. Create Mantel correlogram for genetic data<span></span></a>
<ul>
<li><a href="create-mantel-correlogram-for-genetic-data.html#a-create-a-first-mantel-correlogram">a) Create a first Mantel correlogram<span></span></a></li>
<li><a href="create-mantel-correlogram-for-genetic-data.html#b-vary-distance-class-definition">b) Vary distance class definition<span></span></a></li>
<li><a href="create-mantel-correlogram-for-genetic-data.html#c-alternative-measures-of-genetic-distances">c) Alternative measures of genetic distances<span></span></a></li>
</ul></li>
<li><a href="specify-spatial-weights-and-calculate-morans-i.html#specify-spatial-weights-and-calculate-morans-i">6. Specify spatial weights and calculate Moran’s I<span></span></a>
<ul>
<li><a href="specify-spatial-weights-and-calculate-morans-i.html#a-defining-spatial-neighbors">a) Defining spatial neighbors<span></span></a></li>
<li><a href="specify-spatial-weights-and-calculate-morans-i.html#b-defining-spatial-weights">b) Defining spatial weights<span></span></a></li>
<li><a href="specify-spatial-weights-and-calculate-morans-i.html#c-calculating-and-testing-morans-i">c) Calculating and testing Moran’s I<span></span></a></li>
</ul></li>
<li><a href="r-exercise-week-5.html#r-exercise-week-5">7. R Exercise Week 5<span></span></a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="Week6.html"><a href="Week6.html"><i class="fa fa-check"></i><b>9</b> Week 6: Quantitative Genetics<span></span></a>
<ul>
<li class="chapter" data-level="9.1" data-path="video_6.html"><a href="video_6.html"><i class="fa fa-check"></i><b>9.1</b> View Course Video<span></span></a>
<ul>
<li><a href="video_6.html#embedded-video-5">1. Embedded Video<span></span></a></li>
<li><a href="video_6.html#video-part-1-2">Video, Part 1<span></span></a></li>
<li><a href="video_6.html#video-part-2-2">Video, Part 2<span></span></a></li>
<li><a href="video_6.html#preview-slides-5">Preview Slides<span></span></a></li>
</ul></li>
<li class="chapter" data-level="9.2" data-path="tutorial_6.html"><a href="tutorial_6.html"><i class="fa fa-check"></i><b>9.2</b> Interactive Tutorial 6<span></span></a>
<ul>
<li><a href="tutorial_6.html#list-of-r-commands-covered-this-week-5">1. List of R commands covered this week<span></span></a></li>
<li><a href="tutorial_6.html#general-instructions-5">2. General Instructions<span></span></a></li>
</ul></li>
<li class="chapter" data-level="9.3" data-path="WE_6.html"><a href="WE_6.html"><i class="fa fa-check"></i><b>9.3</b> Worked Example<span></span></a></li>
<li><a href="overview-of-worked-example-5.html#overview-of-worked-example-5">1. Overview of Worked Example<span></span></a>
<ul>
<li><a href="overview-of-worked-example-5.html#a-goals-6">a) Goals<span></span></a></li>
<li><a href="overview-of-worked-example-5.html#b-data-set-5">b) Data set<span></span></a></li>
<li><a href="overview-of-worked-example-5.html#c-required-r-packages-4">c) Required R packages<span></span></a></li>
</ul></li>
<li><a href="estimate-genetic-and-non-genetic-variance-components-from-a-common-garden-experiment.html#estimate-genetic-and-non-genetic-variance-components-from-a-common-garden-experiment">2. Estimate genetic and non-genetic variance components from a common garden experiment<span></span></a>
<ul>
<li><a href="estimate-genetic-and-non-genetic-variance-components-from-a-common-garden-experiment.html#a-import-phenytypic-data">a) Import phenytypic data<span></span></a></li>
<li><a href="estimate-genetic-and-non-genetic-variance-components-from-a-common-garden-experiment.html#b-fit-linear-mixed-effects-models-lmm-to-trait-data-observed-on-families">b) Fit linear mixed effects models (LMM) to trait data observed on families<span></span></a></li>
<li><a href="estimate-genetic-and-non-genetic-variance-components-from-a-common-garden-experiment.html#c-compare-model-fit">c) Compare model fit<span></span></a></li>
<li><a href="estimate-genetic-and-non-genetic-variance-components-from-a-common-garden-experiment.html#d-check-model-validity">d) Check model validity<span></span></a></li>
<li><a href="estimate-genetic-and-non-genetic-variance-components-from-a-common-garden-experiment.html#e-estimate-variance-components">e) Estimate variance components<span></span></a></li>
<li><a href="estimate-genetic-and-non-genetic-variance-components-from-a-common-garden-experiment.html#f-significance-testing">f) Significance testing<span></span></a></li>
</ul></li>
<li><a href="estimate-trait-heritability.html#estimate-trait-heritability">3. Estimate trait heritability<span></span></a>
<ul>
<li><a href="estimate-trait-heritability.html#a-estimate-heritability">a) Estimate heritability<span></span></a></li>
<li><a href="estimate-trait-heritability.html#b-estimate-trait-differentiation">b) Estimate trait differentiation<span></span></a></li>
</ul></li>
<li><a href="compare-q_st-to-f_st-measured-from-snp-data.html#compare-q_st-to-f_st-measured-from-snp-data">4. Compare <span class="math inline">\(Q_{st}\)</span> to <span class="math inline">\(F_{st}\)</span> measured from SNP data<span></span></a>
<ul>
<li><a href="compare-q_st-to-f_st-measured-from-snp-data.html#a-import-ecogen-object">a) Import ‘ecogen’ object<span></span></a></li>
<li><a href="compare-q_st-to-f_st-measured-from-snp-data.html#b-compare-q_st-to-f_st">b) Compare <span class="math inline">\(Q_{ST}\)</span> to <span class="math inline">\(F_{ST}\)</span><span></span></a></li>
</ul></li>
<li><a href="assess-correlation-between-trait-and-environment.html#assess-correlation-between-trait-and-environment">5. Assess correlation between trait and environment<span></span></a>
<ul>
<li><a href="assess-correlation-between-trait-and-environment.html#a-describe-and-extract-effects">a) Describe and extract effects<span></span></a></li>
<li><a href="assess-correlation-between-trait-and-environment.html#b-correlation-of-population-effect-with-environment">b) Correlation of population effect with environment<span></span></a></li>
</ul></li>
<li><a href="r-exercise-week-6.html#r-exercise-week-6">7. R Exercise Week 6<span></span></a></li>
</ul></li>
<li class="chapter" data-level="10" data-path="Week7.html"><a href="Week7.html"><i class="fa fa-check"></i><b>10</b> Week 7: Spatial Linear Models<span></span></a>
<ul>
<li class="chapter" data-level="10.1" data-path="video_7.html"><a href="video_7.html"><i class="fa fa-check"></i><b>10.1</b> View Course Video<span></span></a>
<ul>
<li><a href="video_7.html#embedded-video-6">1. Embedded Video<span></span></a></li>
<li><a href="video_7.html#video-part-1-3">Video, Part 1<span></span></a></li>
<li><a href="video_7.html#video-part-2-3">Video, Part 2<span></span></a></li>
<li><a href="video_7.html#preview-slides-6">Preview Slides<span></span></a></li>
</ul></li>
<li class="chapter" data-level="10.2" data-path="tutorial_7.html"><a href="tutorial_7.html"><i class="fa fa-check"></i><b>10.2</b> Interactive Tutorial 7<span></span></a>
<ul>
<li><a href="tutorial_7.html#list-of-r-commands-covered-this-week-6">1. List of R commands covered this week<span></span></a></li>
<li><a href="tutorial_7.html#general-instructions-6">2. General Instructions<span></span></a></li>
</ul></li>
<li class="chapter" data-level="10.3" data-path="WE_7.html"><a href="WE_7.html"><i class="fa fa-check"></i><b>10.3</b> Worked Example<span></span></a></li>
<li><a href="overview-of-worked-example-6.html#overview-of-worked-example-6">1. Overview of Worked Example<span></span></a>
<ul>
<li><a href="overview-of-worked-example-6.html#a-goals-7">a) Goals<span></span></a></li>
<li><a href="overview-of-worked-example-6.html#b-data-set-6">b) Data set<span></span></a></li>
<li><a href="overview-of-worked-example-6.html#c-required-r-packages-5">c) Required R packages<span></span></a></li>
</ul></li>
<li><a href="explore-data-set.html#explore-data-set">2. Explore data set<span></span></a>
<ul>
<li><a href="explore-data-set.html#a-import-data">a) Import data<span></span></a></li>
<li><a href="explore-data-set.html#b-create-a-map">b) Create a map<span></span></a></li>
<li><a href="explore-data-set.html#c-explore-correlations">c) Explore correlations<span></span></a></li>
</ul></li>
<li><a href="test-regression-residuals-for-spatial-autocorrelation.html#test-regression-residuals-for-spatial-autocorrelation">3. Test regression residuals for spatial autocorrelation<span></span></a>
<ul>
<li><a href="test-regression-residuals-for-spatial-autocorrelation.html#a-fit-regression-models">a) Fit regression models<span></span></a></li>
<li><a href="test-regression-residuals-for-spatial-autocorrelation.html#b-test-for-spatial-autocorrelation-morans-i">b) Test for spatial autocorrelation (Moran’s I)<span></span></a></li>
</ul></li>
<li><a href="fit-models-with-spatially-correlated-error-gls-with-package-nlme.html#fit-models-with-spatially-correlated-error-gls-with-package-nlme">4. Fit models with spatially correlated error (GLS) with package ‘nlme’<span></span></a>
<ul>
<li><a href="fit-models-with-spatially-correlated-error-gls-with-package-nlme.html#a-plot-empirical-variogram">a) Plot empirical variogram<span></span></a></li>
<li><a href="fit-models-with-spatially-correlated-error-gls-with-package-nlme.html#b-fit-variogram-models">b) Fit variogram models<span></span></a></li>
<li><a href="fit-models-with-spatially-correlated-error-gls-with-package-nlme.html#c-select-best-fitting-model">c) Select best-fitting model<span></span></a></li>
<li><a href="fit-models-with-spatially-correlated-error-gls-with-package-nlme.html#d-plot-fitted-variogram-model">d) Plot fitted variogram model<span></span></a></li>
<li><a href="fit-models-with-spatially-correlated-error-gls-with-package-nlme.html#e-add-random-factor">e) Add random factor<span></span></a></li>
</ul></li>
<li><a href="fit-spatial-simultaneous-autoregressive-error-models-sar.html#fit-spatial-simultaneous-autoregressive-error-models-sar">5. Fit spatial simultaneous autoregressive error models (SAR)<span></span></a>
<ul>
<li><a href="fit-spatial-simultaneous-autoregressive-error-models-sar.html#a-fit-and-compare-alternative-sar-models">a) Fit and compare alternative SAR models<span></span></a></li>
<li><a href="fit-spatial-simultaneous-autoregressive-error-models-sar.html#b-interpret-best-fitting-sar-model">b) Interpret best-fitting SAR model<span></span></a></li>
</ul></li>
<li><a href="spatial-filtering-with-mem-using-package-spmoran.html#spatial-filtering-with-mem-using-package-spmoran">6. Spatial filtering with MEM using package ‘spmoran’<span></span></a>
<ul>
<li><a href="spatial-filtering-with-mem-using-package-spmoran.html#a-default-method">a) Default method<span></span></a></li>
<li><a href="spatial-filtering-with-mem-using-package-spmoran.html#b-using-a-custom-connectivity-matrix">b) Using a custom connectivity matrix<span></span></a></li>
<li><a href="spatial-filtering-with-mem-using-package-spmoran.html#c-plot-spatial-eigenvectors">c) Plot spatial eigenvectors<span></span></a></li>
<li><a href="spatial-filtering-with-mem-using-package-spmoran.html#d-random-effect-model">d) Random effect model<span></span></a></li>
</ul></li>
<li><a href="fit-spatially-varying-coefficients-model-with-package-spmoran.html#fit-spatially-varying-coefficients-model-with-package-spmoran">7. Fit spatially varying coefficients model with package ‘spmoran’<span></span></a>
<ul>
<li><a href="fit-spatially-varying-coefficients-model-with-package-spmoran.html#model-with-patchsize-and-ibr">Model with PatchSize and IBR<span></span></a></li>
<li><a href="fit-spatially-varying-coefficients-model-with-package-spmoran.html#model-with-ibr-only">Model with IBR only<span></span></a></li>
</ul></li>
<li><a href="conclusions.html#conclusions">8. Conclusions<span></span></a></li>
<li><a href="r-exercise-week-7.html#r-exercise-week-7">9. R Exercise Week 7<span></span></a></li>
<li class="chapter" data-level="10.4" data-path="bonus_7a.html"><a href="bonus_7a.html"><i class="fa fa-check"></i><b>10.4</b> Bonus: Si index<span></span></a></li>
<li><a href="overview-of-bonus-materials.html#overview-of-bonus-materials">1. Overview of Bonus Materials<span></span></a></li>
<li><a href="import-ecological-distance-matrices.html#import-ecological-distance-matrices">2. Import ecological distance matrices<span></span></a></li>
<li><a href="optimize-the-scaling-parameter-alpha.html#optimize-the-scaling-parameter-alpha">3. Optimize the scaling parameter alpha<span></span></a></li>
<li><a href="calculate-hanskis-index-si-with-source-patch-parameters.html#calculate-hanskis-index-si-with-source-patch-parameters">4. Calculate Hanski’s index Si with source patch parameters<span></span></a></li>
</ul></li>
<li class="chapter" data-level="11" data-path="Week8.html"><a href="Week8.html"><i class="fa fa-check"></i><b>11</b> Week 8: Simulation Experiments<span></span></a>
<ul>
<li class="chapter" data-level="11.1" data-path="video_8.html"><a href="video_8.html"><i class="fa fa-check"></i><b>11.1</b> View Course Video<span></span></a>
<ul>
<li><a href="video_8.html#embedded-video-7">1. Embedded Video<span></span></a></li>
<li><a href="video_8.html#video-part-1-4">Video, Part 1<span></span></a></li>
<li><a href="video_8.html#video-part-2-4">Video, Part 2<span></span></a></li>
<li><a href="video_8.html#preview-slides-7">Preview Slides<span></span></a></li>
</ul></li>
<li class="chapter" data-level="11.2" data-path="tutorial_8.html"><a href="tutorial_8.html"><i class="fa fa-check"></i><b>11.2</b> Interactive Tutorial 8<span></span></a>
<ul>
<li><a href="tutorial_8.html#list-of-r-commands-covered-this-week-7">1. List of R commands covered this week<span></span></a></li>
<li><a href="tutorial_8.html#general-instructions-7">2. General Instructions<span></span></a></li>
</ul></li>
<li class="chapter" data-level="11.3" data-path="WE_8.html"><a href="WE_8.html"><i class="fa fa-check"></i><b>11.3</b> Worked Example<span></span></a></li>
<li><a href="overview-of-worked-example-7.html#overview-of-worked-example-7">1. Overview of Worked Example<span></span></a>
<ul>
<li><a href="overview-of-worked-example-7.html#a-goals-8">a) Goals<span></span></a></li>
<li><a href="overview-of-worked-example-7.html#b-data-set-7">b) Data set<span></span></a></li>
<li><a href="overview-of-worked-example-7.html#c-required-r-packages-6">c) Required R packages<span></span></a></li>
</ul></li>
<li><a href="initialize-a-landscape.html#initialize-a-landscape">2. Initialize a landscape<span></span></a>
<ul>
<li><a href="initialize-a-landscape.html#a-create-a-random-landscape">a) Create a random landscape<span></span></a></li>
<li><a href="initialize-a-landscape.html#b-add-populations-to-the-landscape-using-minimal-distance">b) Add populations to the landscape (using minimal distance)<span></span></a></li>
<li><a href="initialize-a-landscape.html#c-initialise-a-metapopulation">c) Initialise a metapopulation<span></span></a></li>
</ul></li>
<li><a href="define-simulation-parameters.html#define-simulation-parameters">3. Define simulation parameters<span></span></a>
<ul>
<li><a href="define-simulation-parameters.html#a-define-your-metapopulation">a) Define your metapopulation<span></span></a></li>
<li><a href="define-simulation-parameters.html#b-initialise-your-population-on-the-landscape">b) Initialise your population on the landscape<span></span></a></li>
</ul></li>
<li><a href="run-simulations-and-analyze-results.html#run-simulations-and-analyze-results">4. Run simulations and analyze results<span></span></a>
<ul>
<li><a href="run-simulations-and-analyze-results.html#a-run-your-simulation-over-multiple-time-steps-years">a) Run your simulation over multiple time steps (years)<span></span></a></li>
<li><a href="run-simulations-and-analyze-results.html#b-analyse-your-simulated-population-with-a-partial-mantel-test">b) Analyse your simulated population with a partial Mantel test<span></span></a></li>
<li><a href="run-simulations-and-analyze-results.html#c-optional-analyze-your-simulated-populations-using-sunder">c) Optional: Analyze your simulated populations using ‘Sunder’<span></span></a></li>
</ul></li>
<li><a href="run-simulator-using-a-previously-defined-parameter-set.html#run-simulator-using-a-previously-defined-parameter-set">5. Run simulator using a previously defined parameter set<span></span></a>
<ul>
<li><a href="run-simulator-using-a-previously-defined-parameter-set.html#a-specify-and-record-the-parameter-combinations">a) Specify and record the parameter combinations<span></span></a></li>
<li><a href="run-simulator-using-a-previously-defined-parameter-set.html#b-run-the-simulator-over-every-parameter-combination">b) Run the simulator over every parameter combination<span></span></a></li>
<li><a href="run-simulator-using-a-previously-defined-parameter-set.html#c-save-your-complete-simulation-run-input-and-output">c) Save your complete simulation run (input and output)<span></span></a></li>
<li><a href="run-simulator-using-a-previously-defined-parameter-set.html#d-analyze-and-synthesize-results">d) Analyze and synthesize results<span></span></a></li>
</ul></li>
<li><a href="r-exercise-week-8.html#r-exercise-week-8">6. R Exercise Week 8<span></span></a></li>
<li class="chapter" data-level="11.4" data-path="bonus_8a.html"><a href="bonus_8a.html"><i class="fa fa-check"></i><b>11.4</b> Bonus: Efficient R<span></span></a></li>
<li><a href="overview-of-bonus-material.html#overview-of-bonus-material">1. Overview of Bonus Material<span></span></a>
<ul>
<li><a href="overview-of-bonus-material.html#a-goals-9">a) Goals<span></span></a></li>
<li><a href="overview-of-bonus-material.html#b-data-set-8">b) Data set<span></span></a></li>
<li><a href="overview-of-bonus-material.html#c-required-r-packages-7">c) Required R packages<span></span></a></li>
</ul></li>
<li><a href="navigating-the-file-system.html#navigating-the-file-system">2. Navigating the file system<span></span></a>
<ul>
<li><a href="navigating-the-file-system.html#a-where-am-i">a) Where am I?<span></span></a></li>
<li><a href="navigating-the-file-system.html#b-accessing-a-system-file">b) Accessing a system file<span></span></a></li>
<li><a href="navigating-the-file-system.html#c-file-manipulation">c) File manipulation<span></span></a></li>
</ul></li>
<li><a href="benchmarking-file-import-and-export-options.html#benchmarking-file-import-and-export-options">3. Benchmarking file import and export options<span></span></a>
<ul>
<li><a href="benchmarking-file-import-and-export-options.html#a-benchmark-methods-for-importing-csv-files">a) Benchmark methods for importing csv files<span></span></a></li>
<li><a href="benchmarking-file-import-and-export-options.html#b-check-handling-of-character-data">b) Check handling of character data<span></span></a></li>
<li><a href="benchmarking-file-import-and-export-options.html#c-binary-files">c) Binary files<span></span></a></li>
<li><a href="benchmarking-file-import-and-export-options.html#d-should-you-save-your-r-workspace">d) Should you save your R workspace?<span></span></a></li>
<li><a href="benchmarking-file-import-and-export-options.html#e-compile-your-functions">e) Compile your functions<span></span></a></li>
</ul></li>
<li><a href="profiling-your-code.html#profiling-your-code">4. Profiling your code<span></span></a>
<ul>
<li><a href="profiling-your-code.html#a-named-chunks">a) Named chunks<span></span></a></li>
<li><a href="profiling-your-code.html#b-profiling-some-lines-of-code-with-profvis">b) Profiling some lines of code with ‘profvis’<span></span></a></li>
<li><a href="profiling-your-code.html#c-profiling-an-r-script-with-profvis">c) Profiling an R script with ‘profvis’<span></span></a></li>
<li><a href="profiling-your-code.html#d-converting-an-r-notebook-to-an-r-script">d) Converting an R Notebook to an R script<span></span></a></li>
</ul></li>
<li><a href="creating-and-executing-a-bash-r-script.html#creating-and-executing-a-bash-r-script">5. Creating and executing a Bash R script<span></span></a>
<ul>
<li><a href="creating-and-executing-a-bash-r-script.html#a-run-r-script-directly-in-the-terminal">a) Run R script directly in the Terminal<span></span></a></li>
<li><a href="creating-and-executing-a-bash-r-script.html#b-create-a-bash-r-script">b) Create a Bash R script<span></span></a></li>
<li><a href="creating-and-executing-a-bash-r-script.html#c-executing-a-bash-r-script">c) Executing a Bash R script<span></span></a></li>
<li><a href="creating-and-executing-a-bash-r-script.html#d-moving-to-a-node-or-cluster">d) Moving to a node or cluster?<span></span></a></li>
<li><a href="creating-and-executing-a-bash-r-script.html#e-store-your-session-info-and-package-versions">e) Store your session info and package versions<span></span></a></li>
<li><a href="creating-and-executing-a-bash-r-script.html#f-potential-windows-issues">f) Potential Windows issues<span></span></a></li>
</ul></li>
<li><a href="parallelizing-code.html#parallelizing-code">6. Parallelizing code<span></span></a>
<ul>
<li><a href="parallelizing-code.html#a-replace-lapply-by-mclapply-with-package-parallel">a) Replace ‘lapply’ by ‘mclapply’ with package ‘parallel’<span></span></a></li>
<li><a href="parallelizing-code.html#b-replace-for-by-foreach-with-package-doparallel">b) Replace ‘for’ by ‘foreach’ with package ‘doParallel’<span></span></a></li>
</ul></li>
</ul></li>
<li class="part"><span><b>Advanced Topics<span></span></b></span></li>
<li class="chapter" data-level="12" data-path="Week9.html"><a href="Week9.html"><i class="fa fa-check"></i><b>12</b> Week 9: Population Structure<span></span></a>
<ul>
<li class="chapter" data-level="12.1" data-path="WE_9.html"><a href="WE_9.html"><i class="fa fa-check"></i><b>12.1</b> Worked Example<span></span></a></li>
<li><a href="overview-of-worked-example-8.html#overview-of-worked-example-8">1. Overview of Worked Example<span></span></a>
<ul>
<li><a href="overview-of-worked-example-8.html#a-goals-10">a) Goals<span></span></a></li>
<li><a href="overview-of-worked-example-8.html#b-data-sets">b) Data sets<span></span></a></li>
<li><a href="overview-of-worked-example-8.html#c-required-r-packages-8">c) Required R packages<span></span></a></li>
</ul></li>
<li><a href="simulated-data-2-island-model.html#simulated-data-2-island-model">2. Simulated data: 2-island model<span></span></a>
<ul>
<li><a href="simulated-data-2-island-model.html#a-import-data-1">a) Import data<span></span></a></li>
<li><a href="simulated-data-2-island-model.html#b-perform-principal-components-analysis-pca">b) Perform Principal Components Analysis (PCA)<span></span></a></li>
<li><a href="simulated-data-2-island-model.html#c-create-synthetic-spatial-coordinates-xy-and-map-them">c) Create synthetic spatial coordinates (X,Y) and map them<span></span></a></li>
</ul></li>
<li><a href="simulated-data-2-island-model-with-admixture.html#simulated-data-2-island-model-with-admixture">3. Simulated data: 2-island model with admixture<span></span></a></li>
<li><a href="empirical-data-threespine-sticklebacks.html#empirical-data-threespine-sticklebacks">4. Empirical data: Threespine sticklebacks<span></span></a>
<ul>
<li><a href="empirical-data-threespine-sticklebacks.html#a-import-the-data">a) Import the data<span></span></a></li>
<li><a href="empirical-data-threespine-sticklebacks.html#b-examine-population-structure-with-pca">b) Examine population structure with PCA<span></span></a></li>
<li><a href="empirical-data-threespine-sticklebacks.html#c-clustering-with-snmf-similar-to-structure">c) Clustering with SNMF (similar to ‘STRUCTURE’)<span></span></a></li>
<li><a href="empirical-data-threespine-sticklebacks.html#d-plot-ancestry-proportions">d) Plot ancestry proportions<span></span></a></li>
<li><a href="empirical-data-threespine-sticklebacks.html#e-visualize-admixture-proportions-on-a-map">e) Visualize admixture proportions on a map<span></span></a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="13" data-path="Week10.html"><a href="Week10.html"><i class="fa fa-check"></i><b>13</b> Week 10: Landscape Resistance<span></span></a>
<ul>
<li class="chapter" data-level="13.1" data-path="WE_10.html"><a href="WE_10.html"><i class="fa fa-check"></i><b>13.1</b> Worked Example<span></span></a></li>
<li><a href="overview-of-worked-example-9.html#overview-of-worked-example-9">1. Overview of Worked Example<span></span></a>
<ul>
<li><a href="overview-of-worked-example-9.html#a-goals-11">a) Goals<span></span></a></li>
<li><a href="overview-of-worked-example-9.html#b-data-set-9">b) Data set<span></span></a></li>
<li><a href="overview-of-worked-example-9.html#c-required-r-packages-9">c) Required R packages<span></span></a></li>
<li><a href="overview-of-worked-example-9.html#d-import-rasters-gsp-err-cti-and-ffp">d) Import rasters (gsp, err, cti, and ffp)<span></span></a></li>
<li><a href="overview-of-worked-example-9.html#e-import-site-data-with-sampling-locations">e) Import site data with sampling locations<span></span></a></li>
</ul></li>
<li><a href="explore-the-data-set.html#explore-the-data-set">2. Explore the data set<span></span></a>
<ul>
<li><a href="explore-the-data-set.html#a-plot-all-rasters">a) Plot all rasters<span></span></a></li>
<li><a href="explore-the-data-set.html#b-plot-spatial-points-over-ffp-raster">b) Plot spatial points over ‘ffp’ raster<span></span></a></li>
</ul></li>
<li><a href="setting-cost-values-and-calculating-conductance.html#setting-cost-values-and-calculating-conductance">3. Setting cost values and calculating conductance<span></span></a>
<ul>
<li><a href="setting-cost-values-and-calculating-conductance.html#a-resistance-vs.-conductance-values">a) Resistance vs. conductance values<span></span></a></li>
<li><a href="setting-cost-values-and-calculating-conductance.html#b-as-needed-get-all-rasters-at-the-same-scale">b) As needed: Get all rasters at the same scale<span></span></a></li>
<li><a href="setting-cost-values-and-calculating-conductance.html#c-calculate-conductance-values">c) Calculate conductance values<span></span></a></li>
<li><a href="setting-cost-values-and-calculating-conductance.html#d-create-a-single-landscape-conductance-raster">d) Create a single landscape conductance raster<span></span></a></li>
</ul></li>
<li><a href="convert-conductance-into-effective-distance.html#convert-conductance-into-effective-distance">4. Convert conductance into effective distance<span></span></a>
<ul>
<li><a href="convert-conductance-into-effective-distance.html#a-create-a-transition-layer">a) Create a transition layer<span></span></a></li>
<li><a href="convert-conductance-into-effective-distance.html#b-visually-inspect-the-raster">b) Visually inspect the raster<span></span></a></li>
<li><a href="convert-conductance-into-effective-distance.html#c-correct-for-geometric-distortion">c) Correct for geometric distortion<span></span></a></li>
<li><a href="convert-conductance-into-effective-distance.html#d-plot-shortest-paths-in-space">d) Plot shortest paths in space<span></span></a></li>
</ul></li>
<li><a href="create-cost-distance-matrices.html#create-cost-distance-matrices">5. Create cost-distance matrices<span></span></a>
<ul>
<li><a href="create-cost-distance-matrices.html#a-least-cost-distance">a) Least cost distance<span></span></a></li>
<li><a href="create-cost-distance-matrices.html#b-cost-distance-matrix-based-on-random-paths-similar-to-circuitscape">b) Cost-distance matrix based on random paths (similar to Circuitscape)<span></span></a></li>
<li><a href="create-cost-distance-matrices.html#c-compare-cost-distances">c) Compare cost distances<span></span></a></li>
</ul></li>
<li><a href="how-does-changing-resolution-affect-these-metrics.html#how-does-changing-resolution-affect-these-metrics">6. How does changing resolution affect these metrics?<span></span></a>
<ul>
<li><a href="how-does-changing-resolution-affect-these-metrics.html#a-create-loop">a) Create loop<span></span></a></li>
<li><a href="how-does-changing-resolution-affect-these-metrics.html#b-plot-the-results">b) Plot the results<span></span></a></li>
</ul></li>
<li><a href="references.html#references">7. References<span></span></a></li>
<li class="chapter" data-level="13.2" data-path="bonus-10a.html"><a href="bonus-10a.html"><i class="fa fa-check"></i><b>13.2</b> Bonus: ‘radish’ tutorial<span></span></a></li>
<li><a href="overview-of-bonus-materials-1.html#overview-of-bonus-materials-1">1. Overview of Bonus Materials<span></span></a>
<ul>
<li><a href="overview-of-bonus-materials-1.html#a-what-is-the-package-radish">a) What is the package ‘radish?’<span></span></a></li>
<li><a href="overview-of-bonus-materials-1.html#b-data">b) Data<span></span></a></li>
<li><a href="overview-of-bonus-materials-1.html#c-required-r-packages-10">c) Required R packages<span></span></a></li>
</ul></li>
<li><a href="import-data.html#import-data">2. Import data<span></span></a>
<ul>
<li><a href="import-data.html#a-import-raster-data">a) Import raster data<span></span></a></li>
<li><a href="import-data.html#b-import-sampling-location-data">b) Import sampling location data<span></span></a></li>
<li><a href="import-data.html#c-import-genetic-data">c) Import genetic data<span></span></a></li>
<li><a href="import-data.html#d-selecting-sites">d) Selecting sites<span></span></a></li>
</ul></li>
<li><a href="basic-model-fitting.html#basic-model-fitting">3. Basic model fitting<span></span></a>
<ul>
<li><a href="basic-model-fitting.html#a-preparing-radish-objects">a) Preparing ‘radish’ objects<span></span></a></li>
<li><a href="basic-model-fitting.html#b-fitting-radish-models">b) Fitting ‘radish’ models<span></span></a></li>
<li><a href="basic-model-fitting.html#c-model-interactions">c) Model interactions<span></span></a></li>
</ul></li>
<li><a href="advanced-modeling-topics.html#advanced-modeling-topics">4. Advanced modeling topics<span></span></a>
<ul>
<li><a href="advanced-modeling-topics.html#a-isolation-by-distance">a) Isolation-by-Distance<span></span></a></li>
<li><a href="advanced-modeling-topics.html#b-visualizing-results">b) Visualizing results<span></span></a></li>
<li><a href="advanced-modeling-topics.html#c-plot-effects">c) Plot effects<span></span></a></li>
<li><a href="advanced-modeling-topics.html#d-other-measurement-models-generalized-wishart-and-least-squares">d) Other measurement models: Generalized Wishart and least squares<span></span></a></li>
<li><a href="advanced-modeling-topics.html#e-categorical-surfaces">e) Categorical surfaces<span></span></a></li>
</ul></li>
<li><a href="summary.html#summary">5. Summary<span></span></a></li>
<li><a href="references-1.html#references-1">6. References<span></span></a></li>
</ul></li>
<li class="chapter" data-level="14" data-path="Week11.html"><a href="Week11.html"><i class="fa fa-check"></i><b>14</b> Week 11: Detecting Adaptation<span></span></a>
<ul>
<li class="chapter" data-level="14.1" data-path="WE_11.html"><a href="WE_11.html"><i class="fa fa-check"></i><b>14.1</b> Worked Example<span></span></a></li>
<li><a href="overview-of-worked-example-10.html#overview-of-worked-example-10">1. Overview of Worked Example<span></span></a>
<ul>
<li><a href="overview-of-worked-example-10.html#a-goals-12">a) Goals<span></span></a></li>
<li><a href="overview-of-worked-example-10.html#b-data-set-10">b) Data set<span></span></a></li>
<li><a href="overview-of-worked-example-10.html#c-required-r-libraries-1">c) Required R libraries<span></span></a></li>
</ul></li>
<li><a href="import-and-prepare-the-data.html#import-and-prepare-the-data">2. Import and prepare the data<span></span></a>
<ul>
<li><a href="import-and-prepare-the-data.html#a-import-the-genetic-data">a) Import the genetic data<span></span></a></li>
<li><a href="import-and-prepare-the-data.html#b-import-the-environmental-data">b) Import the environmental data<span></span></a></li>
</ul></li>
<li><a href="latent-factor-mixed-models-lfmm-a-univariate-gea.html#latent-factor-mixed-models-lfmm-a-univariate-gea">3. Latent Factor Mixed Models (LFMM): a univariate GEA<span></span></a>
<ul>
<li><a href="latent-factor-mixed-models-lfmm-a-univariate-gea.html#a-determine-k">a) Determine K<span></span></a></li>
<li><a href="latent-factor-mixed-models-lfmm-a-univariate-gea.html#b-run-lfmm">b) Run LFMM<span></span></a></li>
<li><a href="latent-factor-mixed-models-lfmm-a-univariate-gea.html#c-identify-lfmm-candidates-using-false-discovery-rate">c) Identify LFMM candidates using False Discovery Rate<span></span></a></li>
</ul></li>
<li><a href="redundancy-analysis-rda-a-multivariate-gea.html#redundancy-analysis-rda-a-multivariate-gea">4. Redundancy Analysis (RDA): a multivariate GEA<span></span></a>
<ul>
<li><a href="redundancy-analysis-rda-a-multivariate-gea.html#a-run-rda">a) Run RDA<span></span></a></li>
<li><a href="redundancy-analysis-rda-a-multivariate-gea.html#b-identify-rda-candidates">b) Identify RDA candidates<span></span></a></li>
</ul></li>
<li><a href="compare-lfmm-and-rda-candidates.html#compare-lfmm-and-rda-candidates">5. Compare LFMM and RDA candidates<span></span></a></li>
<li><a href="a-quick-note-on-controlling-for-population-structure-in-rda.html#a-quick-note-on-controlling-for-population-structure-in-rda">6. A quick note on controlling for population structure in RDA<span></span></a></li>
<li><a href="references-2.html#references-2">7. References<span></span></a></li>
</ul></li>
<li class="chapter" data-level="15" data-path="Week12.html"><a href="Week12.html"><i class="fa fa-check"></i><b>15</b> Week 12: Model Selection<span></span></a>
<ul>
<li class="chapter" data-level="15.1" data-path="WE_12.html"><a href="WE_12.html"><i class="fa fa-check"></i><b>15.1</b> Worked Example<span></span></a></li>
<li><a href="overview-of-worked-example-11.html#overview-of-worked-example-11">1. Overview of Worked Example<span></span></a>
<ul>
<li><a href="overview-of-worked-example-11.html#a-goals-and-background">a) Goals and Background<span></span></a></li>
<li><a href="overview-of-worked-example-11.html#b-data-set-11">b) Data set<span></span></a></li>
<li><a href="overview-of-worked-example-11.html#c-required-packages">c) Required packages<span></span></a></li>
<li><a href="overview-of-worked-example-11.html#d-import-data-1">d) Import data<span></span></a></li>
</ul></li>
<li><a href="fitting-candidate-models.html#fitting-candidate-models">2. Fitting candidate models<span></span></a>
<ul>
<li><a href="fitting-candidate-models.html#a-alternative-hypotheses">a) Alternative hypotheses<span></span></a></li>
<li><a href="fitting-candidate-models.html#b-check-for-multicollinearity">b) Check for multicollinearity<span></span></a></li>
<li><a href="fitting-candidate-models.html#c-fit-all-five-models">c) Fit all five models<span></span></a></li>
<li><a href="fitting-candidate-models.html#d-compare-evidence-for-models">d) Compare evidence for models<span></span></a></li>
<li><a href="fitting-candidate-models.html#e-calculate-evidence-weights">e) Calculate evidence weights<span></span></a></li>
<li><a href="fitting-candidate-models.html#f-confidence-intervals-for-predictors">f) Confidence intervals for predictors<span></span></a></li>
</ul></li>
<li><a href="your-turn-to-test-additional-hypotheses.html#your-turn-to-test-additional-hypotheses">3. Your turn to test additional hypotheses!<span></span></a></li>
<li><a href="nested-model-nmlpe-for-hierarchical-sampling-designs.html#nested-model-nmlpe-for-hierarchical-sampling-designs">4. Nested model (NMLPE) for hierarchical sampling designs<span></span></a></li>
<li><a href="references-3.html#references-3">5. References<span></span></a></li>
</ul></li>
<li class="chapter" data-level="16" data-path="Week13.html"><a href="Week13.html"><i class="fa fa-check"></i><b>16</b> Week 13: Gravity Models<span></span></a>
<ul>
<li class="chapter" data-level="16.1" data-path="WE_13.html"><a href="WE_13.html"><i class="fa fa-check"></i><b>16.1</b> Worked Example<span></span></a></li>
<li><a href="overview-of-worked-example-12.html#overview-of-worked-example-12">1. Overview of Worked Example<span></span></a>
<ul>
<li><a href="overview-of-worked-example-12.html#a-background">a) Background<span></span></a></li>
<li><a href="overview-of-worked-example-12.html#b-data-set-12">b) Data set<span></span></a></li>
</ul></li>
<li><a href="setup.html#setup">2. Setup<span></span></a></li>
<li><a href="wetland-complex-data-preparation.html#wetland-complex-data-preparation">3. Wetland complex data preparation<span></span></a>
<ul>
<li><a href="wetland-complex-data-preparation.html#a-read-in-wetlands-data">a) Read in wetlands data<span></span></a></li>
<li><a href="wetland-complex-data-preparation.html#b-create-wetlands-graph">b) Create wetlands graph<span></span></a></li>
<li><a href="wetland-complex-data-preparation.html#c-graph-metrics">c) Graph metrics<span></span></a></li>
<li><a href="wetland-complex-data-preparation.html#d-plot-graph-metric">d) Plot graph metric<span></span></a></li>
</ul></li>
<li><a href="wetland-field-data-preparation.html#wetland-field-data-preparation">4. Wetland field-data preparation<span></span></a></li>
<li><a href="saturated-graph.html#saturated-graph">5. Saturated Graph<span></span></a>
<ul>
<li><a href="saturated-graph.html#a-create-graph-from-site-locations">a) Create graph from site locations<span></span></a></li>
<li><a href="saturated-graph.html#b-merge-the-graph-with-genetic-distance">b) Merge the graph with genetic distance<span></span></a></li>
</ul></li>
<li><a href="spatial-model-data-prepration.html#spatial-model-data-prepration">6. Spatial model data prepration<span></span></a>
<ul>
<li><a href="spatial-model-data-prepration.html#a-read-raster-data-using-terra">a) Read raster data using ‘terra’<span></span></a></li>
<li><a href="spatial-model-data-prepration.html#b-reclassify-wetlands">b) Reclassify wetlands<span></span></a></li>
<li><a href="spatial-model-data-prepration.html#c-calculate-the-proportion-of-the-landscape-around-sites">c) Calculate the proportion of the landscape around sites<span></span></a></li>
<li><a href="spatial-model-data-prepration.html#d-add-values-of-rasters-to-sample-sites">d) Add values of rasters to sample sites<span></span></a></li>
<li><a href="spatial-model-data-prepration.html#e-add-raster-covariates-to-graph-edges-lines">e) Add raster covariates to graph edges (lines)<span></span></a></li>
<li><a href="spatial-model-data-prepration.html#f-what-about-categorical-variables">f) What about categorical variables?<span></span></a></li>
<li><a href="spatial-model-data-prepration.html#g-evaluate-node-and-edge-correlations">g) Evaluate node and edge correlations<span></span></a></li>
<li><a href="spatial-model-data-prepration.html#h-add-node-data">h) Add node data<span></span></a></li>
</ul></li>
<li><a href="gravity-model.html#gravity-model">7. Gravity model<span></span></a>
<ul>
<li><a href="gravity-model.html#a-develop-hypothesis">a) Develop hypothesis<span></span></a></li>
<li><a href="gravity-model.html#b-compare-competing-models">b) Compare competing models<span></span></a></li>
<li><a href="gravity-model.html#c-fit-final-models-and-calculate-effect-size">c) Fit final model(s) and calculate effect size<span></span></a></li>
<li><a href="gravity-model.html#d-back-predict-global_fit-model">d) Back predict global_fit model<span></span></a></li>
<li><a href="gravity-model.html#e-aggregrate-estimates-and-plot">e) Aggregrate estimates and plot<span></span></a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="17" data-path="Week14.html"><a href="Week14.html"><i class="fa fa-check"></i><b>17</b> Week 14: Contemporary Gene Flow<span></span></a>
<ul>
<li class="chapter" data-level="17.1" data-path="WE_14.html"><a href="WE_14.html"><i class="fa fa-check"></i><b>17.1</b> Worked Example<span></span></a></li>
<li><a href="overview-3.html#overview-3">1. Overview<span></span></a>
<ul>
<li><a href="overview-3.html#a-background-and-goals">a) Background and Goals<span></span></a></li>
<li><a href="overview-3.html#b-data-set-13">b) Data set<span></span></a></li>
<li><a href="overview-3.html#c-required-packages-1">c) Required packages<span></span></a></li>
</ul></li>
<li><a href="import-genotypes.html#import-genotypes">2. Import Genotypes<span></span></a>
<ul>
<li><a href="import-genotypes.html#a-import-genotype-data">a) Import genotype data<span></span></a></li>
<li><a href="import-genotypes.html#b-plot-map-of-sampling-locations">b) Plot map of sampling locations<span></span></a></li>
</ul></li>
<li><a href="pollen-pool-genetic-structure.html#pollen-pool-genetic-structure">3. Pollen pool genetic structure<span></span></a>
<ul>
<li><a href="pollen-pool-genetic-structure.html#a-subtract-maternal-contribution-to-offspring-genotype">a) Subtract maternal contribution to offspring genotype<span></span></a></li>
<li><a href="pollen-pool-genetic-structure.html#b-twogener-analysis-amova-of-pollen-pool-data">b) ‘TwoGener’ analysis (AMOVA of pollen pool data)<span></span></a></li>
<li><a href="pollen-pool-genetic-structure.html#c-including-space-isolation-by-distance">c) Including space: Isolation by distance<span></span></a></li>
</ul></li>
<li><a href="paternity-analysis.html#paternity-analysis">4. Paternity analysis<span></span></a>
<ul>
<li><a href="paternity-analysis.html#a-exclusion-probabilities">a) Exclusion probabilities<span></span></a></li>
<li><a href="paternity-analysis.html#b-paternity-analysis">b) Paternity analysis<span></span></a></li>
<li><a href="paternity-analysis.html#c-visualize-reconstructed-pollen-flow-events-in-space">c) Visualize reconstructed pollen flow events in space<span></span></a></li>
<li><a href="paternity-analysis.html#d-visualize-dispersal-kernel">d) Visualize dispersal kernel<span></span></a></li>
</ul></li>
<li><a href="linking-paternity-to-ecological-variables.html#linking-paternity-to-ecological-variables">5. Linking paternity to ecological variables<span></span></a>
<ul>
<li><a href="linking-paternity-to-ecological-variables.html#a-explain-pollen-flow-within-populations">a) Explain pollen flow within populations<span></span></a></li>
<li><a href="linking-paternity-to-ecological-variables.html#b-explain-pollen-flow-at-the-population-level">b) Explain pollen flow at the population level<span></span></a></li>
</ul></li>
<li><a href="references-4.html#references-4">6. References<span></span></a></li>
</ul></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Landscape Genetic Data Analysis with R</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="run-simulator-using-a-previously-defined-parameter-set" class="section level2 unnumbered hasAnchor">
<h2>5. Run simulator using a previously defined parameter set<a href="run-simulator-using-a-previously-defined-parameter-set.html#run-simulator-using-a-previously-defined-parameter-set" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Once the simulator works and you are certain that you understand how the simulator needs to be set up for a single run, in almost all studies on simulations you want to be able to re-run the simulator in an automatized way. There are several reasons why you want to do that.</p>
<ol style="list-style-type: decimal">
<li>You want to perform a sensitivity analysis on a single parameter, which means, try to find how much does the output (e.g. pairwise Fst between subpopulations) change when you vary an input parameter (e.g. number of loci).</li>
<li>You want to explore the “complete” parameter space, which means, instead of changing values of a single input parameter you want to change all parameters (within certain levels) and run their combinations.</li>
<li>Another reason is that you want to create a simulated test data set that forms the backbone of your future studies.</li>
</ol>
<p>So we would like to do the following.</p>
<ul>
<li><ol style="list-style-type: lower-alpha">
<li>Specify and record all the parameter combinations that you would like to run.<br />
</li>
</ol></li>
<li><ol start="2" style="list-style-type: lower-alpha">
<li>Run the simulator with every combination</li>
</ol></li>
<li><ol start="3" style="list-style-type: lower-alpha">
<li>[Optional] save your complete simulation run (recommended, but sometimes prohibitive due to needed resources) or only a calculated summary.</li>
</ol></li>
<li><ol start="4" style="list-style-type: lower-alpha">
<li>Read in your simulations, analyse them and synthesize your results via additional statistics, tests, plots.</li>
</ol></li>
<li><ol start="5" style="list-style-type: lower-alpha">
<li>Publish an enormously important paper….</li>
</ol></li>
</ul>
<p>Admittedly there are several different approaches and as it seems every modeller has a slightly different way to achieve these steps. One approach is to create a parameter file that records all the parameter setting for each run. Another approach is to create so called scripts for every single run. The advantage here is that scripts can be easily distributed across different cores and machines and therefore this approach is easy to parallelise your runs, which sometimes is necessary. Finally the approach I will present here (also because of practical reasons) is to create an R data.frame that stores all the parameter settings and we run all combinations in serial mode instead of in parallel.</p>
<p>Okay before we start we need to think about what kind of parameters we want to explore. I would like to do the following runs:</p>
<ul>
<li>Run our simulations as above (same parameter settings) for varying time steps (say between 5 to 45 years in steps of 20). We’ll keep the number of levels and the maximum number of steps low in this example to limit computation time. Feel free to expand!</li>
<li>As output I would still like to record the Fst value, but also the full genetic data set and the parameters used to run the simulation.</li>
<li>In addition I want to repeat each run 5 times (most often you would do more repetitions) to check how much general variation there is between runs with exactly the same parameter combination.</li>
</ul>
<div id="a-specify-and-record-the-parameter-combinations" class="section level3 unnumbered hasAnchor">
<h3>a) Specify and record the parameter combinations<a href="run-simulator-using-a-previously-defined-parameter-set.html#a-specify-and-record-the-parameter-combinations" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Let’s define the varying numbers of time steps we would like to run the simulations. Here we define a sequence from 5 to 45 in steps of 20, which results in a series c(5, 25, 45). We will interpret these values as numeric, therefore we don’t convert to ‘factor.’</p>
<div class="sourceCode" id="cb965"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb965-1"><a href="run-simulator-using-a-previously-defined-parameter-set.html#cb965-1" aria-hidden="true" tabindex="-1"></a>timesteps <span class="ot">&lt;-</span> <span class="fu">seq</span>(<span class="at">from=</span><span class="dv">5</span> , <span class="at">to=</span><span class="dv">45</span>, <span class="at">by=</span><span class="dv">20</span>)</span></code></pre></div>
<p>We also specify the number of repeats (replicate simulation runs). We want to do five replicate simulation runs per for each level of ‘time,’ and we will label replicates from 1 through 5. These are essentially labels and we’ll save them as a factor:</p>
<div class="sourceCode" id="cb966"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb966-1"><a href="run-simulator-using-a-previously-defined-parameter-set.html#cb966-1" aria-hidden="true" tabindex="-1"></a>repeats <span class="ot">&lt;-</span> <span class="fu">factor</span>(<span class="dv">1</span><span class="sc">:</span><span class="dv">5</span>)</span></code></pre></div>
<p>Now we would like to have a data frame that stores all possible combinations for those two parameters. As simple way to do that in R, is to use the ‘expand.grid’ function.</p>
<div class="sourceCode" id="cb967"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb967-1"><a href="run-simulator-using-a-previously-defined-parameter-set.html#cb967-1" aria-hidden="true" tabindex="-1"></a>para.space <span class="ot">&lt;-</span> <span class="fu">expand.grid</span>(<span class="at">rep=</span>repeats, <span class="at">time=</span>timesteps)</span>
<span id="cb967-2"><a href="run-simulator-using-a-previously-defined-parameter-set.html#cb967-2" aria-hidden="true" tabindex="-1"></a>tibble<span class="sc">::</span><span class="fu">as_tibble</span>(para.space)</span></code></pre></div>
<pre><code>## # A tibble: 15 × 2
##    rep    time
##    &lt;fct&gt; &lt;dbl&gt;
##  1 1         5
##  2 2         5
##  3 3         5
##  4 4         5
##  5 5         5
##  6 1        25
##  7 2        25
##  8 3        25
##  9 4        25
## 10 5        25
## 11 1        45
## 12 2        45
## 13 3        45
## 14 4        45
## 15 5        45</code></pre>
<p>As you can see this results in 15 combinations (3 time steps x 5 repeats). The beauty of this approach is that it is very flexible and adaptable to runs over other parameter combinations, as you can provide more than two parameter variables to ‘expand.grid.’</p>
</div>
<div id="b-run-the-simulator-over-every-parameter-combination" class="section level3 unnumbered hasAnchor">
<h3>b) Run the simulator over every parameter combination<a href="run-simulator-using-a-previously-defined-parameter-set.html#b-run-the-simulator-over-every-parameter-combination" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Remember our parameters are all defined in the ‘para’ object (a list) and we want to keep them constant, except for running the simulation for different number of years. This means that we only need to modify the argument ‘steps.’</p>
<p>Summarizing the code from above, a single run of our simulator runs via:</p>
<div class="sourceCode" id="cb969"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb969-1"><a href="run-simulator-using-a-previously-defined-parameter-set.html#cb969-1" aria-hidden="true" tabindex="-1"></a><span class="co">#initialize</span></span>
<span id="cb969-2"><a href="run-simulator-using-a-previously-defined-parameter-set.html#cb969-2" aria-hidden="true" tabindex="-1"></a>simpops<span class="fl">.0</span> <span class="ot">&lt;-</span> PopGenReport<span class="sc">::</span><span class="fu">init.popgensim</span>(para<span class="sc">$</span>n.pops, para<span class="sc">$</span>n.ind,</span>
<span id="cb969-3"><a href="run-simulator-using-a-previously-defined-parameter-set.html#cb969-3" aria-hidden="true" tabindex="-1"></a>                           para<span class="sc">$</span>sex.ratio, para<span class="sc">$</span>n.loci, </span>
<span id="cb969-4"><a href="run-simulator-using-a-previously-defined-parameter-set.html#cb969-4" aria-hidden="true" tabindex="-1"></a>                           para<span class="sc">$</span>n.allels, para<span class="sc">$</span>locs, para<span class="sc">$</span>n.cov )  </span>
<span id="cb969-5"><a href="run-simulator-using-a-previously-defined-parameter-set.html#cb969-5" aria-hidden="true" tabindex="-1"></a><span class="co">#run for 20 generations</span></span>
<span id="cb969-6"><a href="run-simulator-using-a-previously-defined-parameter-set.html#cb969-6" aria-hidden="true" tabindex="-1"></a>simpops <span class="ot">&lt;-</span> PopGenReport<span class="sc">::</span><span class="fu">run.popgensim</span>(simpops<span class="fl">.0</span>, <span class="at">steps=</span><span class="dv">20</span>, cost.mat, </span>
<span id="cb969-7"><a href="run-simulator-using-a-previously-defined-parameter-set.html#cb969-7" aria-hidden="true" tabindex="-1"></a>                         <span class="at">n.offspring=</span>para<span class="sc">$</span>n.offspring, <span class="at">n.ind=</span>para<span class="sc">$</span>n.ind,</span>
<span id="cb969-8"><a href="run-simulator-using-a-previously-defined-parameter-set.html#cb969-8" aria-hidden="true" tabindex="-1"></a>                         para<span class="sc">$</span>mig.rate, para<span class="sc">$</span>disp.max, para<span class="sc">$</span>disp.rate, </span>
<span id="cb969-9"><a href="run-simulator-using-a-previously-defined-parameter-set.html#cb969-9" aria-hidden="true" tabindex="-1"></a>                         para<span class="sc">$</span>n.allels, para<span class="sc">$</span>mut.rate,</span>
<span id="cb969-10"><a href="run-simulator-using-a-previously-defined-parameter-set.html#cb969-10" aria-hidden="true" tabindex="-1"></a>                         <span class="at">n.cov=</span>para<span class="sc">$</span>n.cov, <span class="at">rec=</span><span class="st">&quot;none&quot;</span>)</span></code></pre></div>
<p>We adapt this code as follows:</p>
<ul>
<li>Create a ‘for’ loop that cycles through every row ‘i’ in ‘para.space’</li>
<li>For each value of ‘i’:
<ul>
<li>Initialize population ‘simpops.0.’</li>
<li>Run the simulation with argument ‘steps = para.space$time[i].’</li>
</ul></li>
</ul>
<p>We are not running the code just yet, hence it is commented-out with ‘#.’</p>
<div class="sourceCode" id="cb970"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb970-1"><a href="run-simulator-using-a-previously-defined-parameter-set.html#cb970-1" aria-hidden="true" tabindex="-1"></a><span class="co">#for (i in 1:nrow(para.space))</span></span>
<span id="cb970-2"><a href="run-simulator-using-a-previously-defined-parameter-set.html#cb970-2" aria-hidden="true" tabindex="-1"></a><span class="co">#{</span></span>
<span id="cb970-3"><a href="run-simulator-using-a-previously-defined-parameter-set.html#cb970-3" aria-hidden="true" tabindex="-1"></a><span class="co">#  #initialize</span></span>
<span id="cb970-4"><a href="run-simulator-using-a-previously-defined-parameter-set.html#cb970-4" aria-hidden="true" tabindex="-1"></a><span class="co">#  simpops.0 &lt;- PopGenReport::init.popgensim(para$n.pops, para$n.ind, </span></span>
<span id="cb970-5"><a href="run-simulator-using-a-previously-defined-parameter-set.html#cb970-5" aria-hidden="true" tabindex="-1"></a><span class="co">#                           para$sex.ratio, para$n.loci, para$n.allels, </span></span>
<span id="cb970-6"><a href="run-simulator-using-a-previously-defined-parameter-set.html#cb970-6" aria-hidden="true" tabindex="-1"></a><span class="co">#                           para$locs, para$n.cov )</span></span>
<span id="cb970-7"><a href="run-simulator-using-a-previously-defined-parameter-set.html#cb970-7" aria-hidden="true" tabindex="-1"></a><span class="co">#  </span></span>
<span id="cb970-8"><a href="run-simulator-using-a-previously-defined-parameter-set.html#cb970-8" aria-hidden="true" tabindex="-1"></a><span class="co">#  #run for para.space$time[i] generations</span></span>
<span id="cb970-9"><a href="run-simulator-using-a-previously-defined-parameter-set.html#cb970-9" aria-hidden="true" tabindex="-1"></a><span class="co">#  simpops &lt;- PopGenReport::run.popgensim(simpops.0, </span></span>
<span id="cb970-10"><a href="run-simulator-using-a-previously-defined-parameter-set.html#cb970-10" aria-hidden="true" tabindex="-1"></a><span class="co">#                           steps=para.space$time[i], cost.mat,</span></span>
<span id="cb970-11"><a href="run-simulator-using-a-previously-defined-parameter-set.html#cb970-11" aria-hidden="true" tabindex="-1"></a><span class="co">#                           n.offspring=para$n.offspring, n.ind=para$n.ind,</span></span>
<span id="cb970-12"><a href="run-simulator-using-a-previously-defined-parameter-set.html#cb970-12" aria-hidden="true" tabindex="-1"></a><span class="co">#                           para$mig.rate, para$disp.max, para$disp.rate, </span></span>
<span id="cb970-13"><a href="run-simulator-using-a-previously-defined-parameter-set.html#cb970-13" aria-hidden="true" tabindex="-1"></a><span class="co">#                           para$n.allels, para$mut.rate,</span></span>
<span id="cb970-14"><a href="run-simulator-using-a-previously-defined-parameter-set.html#cb970-14" aria-hidden="true" tabindex="-1"></a><span class="co">#                           n.cov=para$n.cov, rec=&quot;none&quot;)</span></span>
<span id="cb970-15"><a href="run-simulator-using-a-previously-defined-parameter-set.html#cb970-15" aria-hidden="true" tabindex="-1"></a><span class="co">#}</span></span></code></pre></div>
<p>Have a close look at the change.</p>
<p>Question: what changes between replicate runs, and what not? Consider the following aspects:</p>
<ul>
<li>Landscape</li>
<li>Population locations</li>
<li>Pairwise distances (cost, Euclidean)</li>
<li>Initial populations with initial genotypes</li>
<li>Migration and gene flow</li>
</ul>
</div>
<div id="c-save-your-complete-simulation-run-input-and-output" class="section level3 unnumbered hasAnchor">
<h3>c) Save your complete simulation run (input and output)<a href="run-simulator-using-a-previously-defined-parameter-set.html#c-save-your-complete-simulation-run-input-and-output" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Simply running the simulation 15 times (number of rows in ‘para.space’) by itself is not useful yet. We need to store the simulation runs somehow, so we can collect them afterwards to calculate summary statistics and analyse the runs.</p>
<p>How do we store the repeats seperately in a file? One approach would be to have a different file name for every repeat, but in my view, a cleaner approach is to store all simulation outputs and also store the complete parameter and input information in a file, so everything that is need is in one place. A nice way to do that in R is to create a ‘list’ object that stores all in a single object, which can be saved (and is automatically packed) and re-loaded as an R object. This is convenient as long as I only want to analyze the results in R, not export to other software.</p>
<p>Here we do the following:</p>
<ul>
<li>Create a timer with the function ‘proc.time’ so that we know roughly how long the computations take.</li>
<li>For each line ‘i’ in ‘para.space’:
<ul>
<li>Initialize simpops.0.</li>
<li>Run the simulation with ‘steps=para.space$time[i].’</li>
<li>Convert the resulting ‘simpop’ to a genind object ‘gi’ (smaller to store)</li>
<li>Create a list ‘sim’ of all simulation parameters we want to store.</li>
<li>Save the object ‘sim’ as an ‘RData’ file with a unique file name in the folder ‘output/simout’ in the project directory.</li>
<li>Print a message after each run to report progress and computation time.</li>
<li>Flush the output console to make sure it is current (only relevant for console-based versions of R).</li>
</ul></li>
</ul>
<p>The list ‘sim’ will contain the following elements. Note: the name is repeated (e.g., ‘gi = gi’) to create a named list, i.e., to specify the names of the list elements.</p>
<ul>
<li><strong>para.space</strong>: the row ‘para.space[i]’ with settings of the simulation run.</li>
<li><strong>para</strong>: a copy of the list ‘para’ that contains the other parameters that are the same for all simulation runs.</li>
<li><strong>landscape</strong>: the landscape.</li>
<li><strong>cost.mat</strong>: the matrix of pairwise cost distances</li>
<li><strong>gi</strong>: the genind object that contains the genotypes at the end of the simulation.</li>
</ul>
<p>First we make sure the folder <code>simout</code> exists within the <code>output</code> folder in the R project:</p>
<div class="sourceCode" id="cb971"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb971-1"><a href="run-simulator-using-a-previously-defined-parameter-set.html#cb971-1" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span>(<span class="sc">!</span><span class="fu">dir.exists</span>(<span class="fu">paste0</span>(here<span class="sc">::</span><span class="fu">here</span>(),<span class="st">&quot;/output&quot;</span>))) </span>
<span id="cb971-2"><a href="run-simulator-using-a-previously-defined-parameter-set.html#cb971-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">dir.create</span>(<span class="fu">paste0</span>(here<span class="sc">::</span><span class="fu">here</span>(),<span class="st">&quot;/output&quot;</span>))</span>
<span id="cb971-3"><a href="run-simulator-using-a-previously-defined-parameter-set.html#cb971-3" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span>(<span class="sc">!</span><span class="fu">dir.exists</span>(<span class="fu">paste0</span>(here<span class="sc">::</span><span class="fu">here</span>(),<span class="st">&quot;/output/simout&quot;</span>)))</span>
<span id="cb971-4"><a href="run-simulator-using-a-previously-defined-parameter-set.html#cb971-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">dir.create</span>(<span class="fu">paste0</span>(here<span class="sc">::</span><span class="fu">here</span>(),<span class="st">&quot;/output/simout&quot;</span>))</span></code></pre></div>
<div class="sourceCode" id="cb972"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb972-1"><a href="run-simulator-using-a-previously-defined-parameter-set.html#cb972-1" aria-hidden="true" tabindex="-1"></a><span class="co"># create a timer (just to know how long it will take roughly)</span></span>
<span id="cb972-2"><a href="run-simulator-using-a-previously-defined-parameter-set.html#cb972-2" aria-hidden="true" tabindex="-1"></a>timer0 <span class="ot">&lt;-</span> <span class="fu">round</span>(<span class="fu">proc.time</span>()[<span class="dv">3</span>],<span class="dv">2</span>)</span>
<span id="cb972-3"><a href="run-simulator-using-a-previously-defined-parameter-set.html#cb972-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb972-4"><a href="run-simulator-using-a-previously-defined-parameter-set.html#cb972-4" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">nrow</span>(para.space))</span>
<span id="cb972-5"><a href="run-simulator-using-a-previously-defined-parameter-set.html#cb972-5" aria-hidden="true" tabindex="-1"></a>{</span>
<span id="cb972-6"><a href="run-simulator-using-a-previously-defined-parameter-set.html#cb972-6" aria-hidden="true" tabindex="-1"></a>  <span class="co"># initialize</span></span>
<span id="cb972-7"><a href="run-simulator-using-a-previously-defined-parameter-set.html#cb972-7" aria-hidden="true" tabindex="-1"></a>  simpops<span class="fl">.0</span> <span class="ot">&lt;-</span> PopGenReport<span class="sc">::</span><span class="fu">init.popgensim</span>(para<span class="sc">$</span>n.pops, para<span class="sc">$</span>n.ind, </span>
<span id="cb972-8"><a href="run-simulator-using-a-previously-defined-parameter-set.html#cb972-8" aria-hidden="true" tabindex="-1"></a>                           para<span class="sc">$</span>sex.ratio, para<span class="sc">$</span>n.loci, para<span class="sc">$</span>n.allels, </span>
<span id="cb972-9"><a href="run-simulator-using-a-previously-defined-parameter-set.html#cb972-9" aria-hidden="true" tabindex="-1"></a>                           para<span class="sc">$</span>locs, para<span class="sc">$</span>n.cov )  </span>
<span id="cb972-10"><a href="run-simulator-using-a-previously-defined-parameter-set.html#cb972-10" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb972-11"><a href="run-simulator-using-a-previously-defined-parameter-set.html#cb972-11" aria-hidden="true" tabindex="-1"></a>  <span class="co"># run for para.space$time[i] generations</span></span>
<span id="cb972-12"><a href="run-simulator-using-a-previously-defined-parameter-set.html#cb972-12" aria-hidden="true" tabindex="-1"></a>  simpops <span class="ot">&lt;-</span> PopGenReport<span class="sc">::</span><span class="fu">run.popgensim</span>(simpops<span class="fl">.0</span>, </span>
<span id="cb972-13"><a href="run-simulator-using-a-previously-defined-parameter-set.html#cb972-13" aria-hidden="true" tabindex="-1"></a>                           <span class="at">steps=</span>para.space<span class="sc">$</span>time[i], cost.mat, </span>
<span id="cb972-14"><a href="run-simulator-using-a-previously-defined-parameter-set.html#cb972-14" aria-hidden="true" tabindex="-1"></a>                           <span class="at">n.offspring=</span>para<span class="sc">$</span>n.offspring, <span class="at">n.ind=</span>para<span class="sc">$</span>n.ind,</span>
<span id="cb972-15"><a href="run-simulator-using-a-previously-defined-parameter-set.html#cb972-15" aria-hidden="true" tabindex="-1"></a>                           para<span class="sc">$</span>mig.rate, para<span class="sc">$</span>disp.max, para<span class="sc">$</span>disp.rate, </span>
<span id="cb972-16"><a href="run-simulator-using-a-previously-defined-parameter-set.html#cb972-16" aria-hidden="true" tabindex="-1"></a>                           para<span class="sc">$</span>n.allels, para<span class="sc">$</span>mut.rate,</span>
<span id="cb972-17"><a href="run-simulator-using-a-previously-defined-parameter-set.html#cb972-17" aria-hidden="true" tabindex="-1"></a>                           <span class="at">n.cov=</span>para<span class="sc">$</span>n.cov, <span class="at">rec=</span><span class="st">&quot;none&quot;</span>)</span>
<span id="cb972-18"><a href="run-simulator-using-a-previously-defined-parameter-set.html#cb972-18" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb972-19"><a href="run-simulator-using-a-previously-defined-parameter-set.html#cb972-19" aria-hidden="true" tabindex="-1"></a>  <span class="co"># convert to genind object (smaller)</span></span>
<span id="cb972-20"><a href="run-simulator-using-a-previously-defined-parameter-set.html#cb972-20" aria-hidden="true" tabindex="-1"></a>  gi <span class="ot">&lt;-</span> PopGenReport<span class="sc">::</span><span class="fu">pops2genind</span>(simpops)</span>
<span id="cb972-21"><a href="run-simulator-using-a-previously-defined-parameter-set.html#cb972-21" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb972-22"><a href="run-simulator-using-a-previously-defined-parameter-set.html#cb972-22" aria-hidden="true" tabindex="-1"></a>  <span class="co"># create a list of all I want to collect</span></span>
<span id="cb972-23"><a href="run-simulator-using-a-previously-defined-parameter-set.html#cb972-23" aria-hidden="true" tabindex="-1"></a>  sim <span class="ot">&lt;-</span> <span class="fu">list</span>(<span class="at">para.space=</span>para.space[i,], <span class="at">para=</span>para, </span>
<span id="cb972-24"><a href="run-simulator-using-a-previously-defined-parameter-set.html#cb972-24" aria-hidden="true" tabindex="-1"></a>              <span class="at">landscape=</span>landscape, <span class="at">cost.mat=</span>cost.mat, <span class="at">gi=</span>gi)</span>
<span id="cb972-25"><a href="run-simulator-using-a-previously-defined-parameter-set.html#cb972-25" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb972-26"><a href="run-simulator-using-a-previously-defined-parameter-set.html#cb972-26" aria-hidden="true" tabindex="-1"></a>  <span class="co"># save everything in an output folder (with a consecutive number, with three leading zeros, so the file sorting is nicer)</span></span>
<span id="cb972-27"><a href="run-simulator-using-a-previously-defined-parameter-set.html#cb972-27" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb972-28"><a href="run-simulator-using-a-previously-defined-parameter-set.html#cb972-28" aria-hidden="true" tabindex="-1"></a>  <span class="fu">save</span>(sim, <span class="at">file =</span> <span class="fu">paste0</span>(here<span class="sc">::</span><span class="fu">here</span>(),<span class="st">&quot;/output/simout/sim_time5-45_rep5_&quot;</span>,</span>
<span id="cb972-29"><a href="run-simulator-using-a-previously-defined-parameter-set.html#cb972-29" aria-hidden="true" tabindex="-1"></a>                          <span class="fu">sprintf</span>(<span class="st">&quot;%03i&quot;</span>,i) ,<span class="st">&quot;.RData&quot;</span>))</span>
<span id="cb972-30"><a href="run-simulator-using-a-previously-defined-parameter-set.html#cb972-30" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb972-31"><a href="run-simulator-using-a-previously-defined-parameter-set.html#cb972-31" aria-hidden="true" tabindex="-1"></a>  <span class="fu">cat</span>(<span class="fu">paste0</span>(<span class="st">&quot;Finished run: &quot;</span>, i,<span class="st">&quot; out of &quot;</span>,<span class="fu">nrow</span>(para.space),</span>
<span id="cb972-32"><a href="run-simulator-using-a-previously-defined-parameter-set.html#cb972-32" aria-hidden="true" tabindex="-1"></a>            <span class="st">&quot;. So far, it took: &quot;</span>, <span class="fu">round</span>(<span class="fu">proc.time</span>()[<span class="dv">3</span>]<span class="sc">-</span>timer0,<span class="dv">2</span>),<span class="st">&quot; sec.</span><span class="sc">\n</span><span class="st">&quot;</span>))</span>
<span id="cb972-33"><a href="run-simulator-using-a-previously-defined-parameter-set.html#cb972-33" aria-hidden="true" tabindex="-1"></a>  <span class="fu">flush.console</span>()</span>
<span id="cb972-34"><a href="run-simulator-using-a-previously-defined-parameter-set.html#cb972-34" aria-hidden="true" tabindex="-1"></a>}</span></code></pre></div>
<pre><code>## Finished run: 1 out of 15. So far, it took: 0.49 sec.
## Finished run: 2 out of 15. So far, it took: 0.91 sec.
## Finished run: 3 out of 15. So far, it took: 1.36 sec.
## Finished run: 4 out of 15. So far, it took: 1.79 sec.
## Finished run: 5 out of 15. So far, it took: 2.23 sec.
## Finished run: 6 out of 15. So far, it took: 4.1 sec.
## Finished run: 7 out of 15. So far, it took: 5.93 sec.
## Finished run: 8 out of 15. So far, it took: 7.84 sec.
## Finished run: 9 out of 15. So far, it took: 9.69 sec.
## Finished run: 10 out of 15. So far, it took: 11.65 sec.
## Finished run: 11 out of 15. So far, it took: 15 sec.
## Finished run: 12 out of 15. So far, it took: 18.36 sec.
## Finished run: 13 out of 15. So far, it took: 21.73 sec.
## Finished run: 14 out of 15. So far, it took: 25.04 sec.
## Finished run: 15 out of 15. So far, it took: 28.43 sec.</code></pre>
</div>
<div id="d-analyze-and-synthesize-results" class="section level3 unnumbered hasAnchor">
<h3>d) Analyze and synthesize results<a href="run-simulator-using-a-previously-defined-parameter-set.html#d-analyze-and-synthesize-results" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>If you check your output folder (simout) you should see 15 files.</p>
<p>Note: File paths can be different when you execute a chunk in an R notebook compared to when you copy-paste the same line into the console! We avoid this problem by using the function ‘here’ from package ‘here.’</p>
<div class="sourceCode" id="cb974"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb974-1"><a href="run-simulator-using-a-previously-defined-parameter-set.html#cb974-1" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(<span class="fu">dir</span>(<span class="fu">paste0</span>(here<span class="sc">::</span><span class="fu">here</span>(), <span class="st">&quot;/output/simout&quot;</span>)))</span></code></pre></div>
<pre><code>## [1] &quot;sim_time5-45_rep5_001.RData&quot; &quot;sim_time5-45_rep5_002.RData&quot;
## [3] &quot;sim_time5-45_rep5_003.RData&quot; &quot;sim_time5-45_rep5_004.RData&quot;
## [5] &quot;sim_time5-45_rep5_005.RData&quot; &quot;sim_time5-45_rep5_006.RData&quot;</code></pre>
<p>Now we are at step D where we need to read in all our files one by one, calculate some summary statistics and plot our results.</p>
<p>Again, this could be easy, but be aware if you have thousands of files it could take quite some time and memory. The most convenient way is to load everyting and store it in a list, so we can access all of our simulations from memory. I will show how to do this in the example below, but be aware in larger simulations (think millions of runs, or large sample sizes) this is not possible and we would load a single simulation, calculate a statistic, store only the result in a table and free the memory for the next simulation run.</p>
<p>Let’s load our simulation runs. There is one caveat: when we load the object ‘sim’ from the ‘.RData file,’ we can’t assign it a new object name. I.e., we can’t use ‘newName &lt;- load(“sim.Rdata”). Instead, we can only type ’load(“sim.Rdata”)’ and it will create or overwrite the object ‘sim.’ Also, R takes the name from the object that was saved, not from the file name. Hence, once we load the object, any existing object of the same name will be overwritten. So if we want to keep, it we need to rename it before using ‘load.’</p>
<p>Here we do the following:</p>
<ul>
<li>Create an empty table with three columns ‘rep,’ ‘time,’ ‘fst’ and other columns to collect summary results for the 15 simulation runs.</li>
<li>Create a vector that contains all filenames. The function ‘list.files’ does just that. We specify with path=“./simout”’ that the files in folder ‘simout’ should be listed, and with ’pattern=“sim” we specify that we want all file names that contain the expression “sim” (we could also have used “time,” for example - any component of the file name that is shared by all the files we are interested in but no other files that might be in the same folder).</li>
<li>Loop through the files. For each filename ‘i’:
<ul>
<li>Load the file, which will create or overwrite the object ‘sim.’ We need to supply not only the file name but the path, hence ‘paste0(“./simout/,”filenames[i])’ (see Week 8 video for more on file paths).</li>
<li>Extract simulation parameters: Copy the ith row from ‘para.space’ (repeat number, timesteps) into the first two columns of the ith row of ‘res.’</li>
<li>Extract the genind object ‘gi’ with the final genotypes.</li>
<li>Calculate the mean of pairwise fst values and store in the third column of the ith row of ‘res.’</li>
<li>Perform partial Mantel tests with function ‘wasserman’ and store the Mantel r statistics and the p-values in the corresponding columns. Note that this is somewhat tricky because the function ‘wasserman’ ranks the models and the better fitting model is listed in the first row, so that the order of the models can vary between runs. Therefore we extract the values based on model name.</li>
<li>Perform ‘Sunder’ covariance analysis and extract which model is best supported (‘G,’ ‘E’ or ‘GE’).</li>
</ul></li>
</ul>
<div class="sourceCode" id="cb976"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb976-1"><a href="run-simulator-using-a-previously-defined-parameter-set.html#cb976-1" aria-hidden="true" tabindex="-1"></a>res <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">rep=</span><span class="cn">NA</span>, <span class="at">time=</span><span class="cn">NA</span>, <span class="at">fst=</span><span class="cn">NA</span>, <span class="at">r.Eucl=</span><span class="cn">NA</span>, <span class="at">p.Eucl=</span><span class="cn">NA</span>, </span>
<span id="cb976-2"><a href="run-simulator-using-a-previously-defined-parameter-set.html#cb976-2" aria-hidden="true" tabindex="-1"></a>                  <span class="at">r.cost=</span><span class="cn">NA</span>, <span class="at">p.cost=</span><span class="cn">NA</span>, <span class="at">Sunder=</span><span class="cn">NA</span>)</span>
<span id="cb976-3"><a href="run-simulator-using-a-previously-defined-parameter-set.html#cb976-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb976-4"><a href="run-simulator-using-a-previously-defined-parameter-set.html#cb976-4" aria-hidden="true" tabindex="-1"></a><span class="co">#load all files in the folder</span></span>
<span id="cb976-5"><a href="run-simulator-using-a-previously-defined-parameter-set.html#cb976-5" aria-hidden="true" tabindex="-1"></a>filenames <span class="ot">&lt;-</span> <span class="fu">list.files</span>(<span class="at">path=</span> <span class="fu">paste0</span>(here<span class="sc">::</span><span class="fu">here</span>(), <span class="st">&quot;/output/simout&quot;</span>), <span class="at">pattern=</span><span class="st">&quot;sim&quot;</span>)</span>
<span id="cb976-6"><a href="run-simulator-using-a-previously-defined-parameter-set.html#cb976-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb976-7"><a href="run-simulator-using-a-previously-defined-parameter-set.html#cb976-7" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(filenames))</span>
<span id="cb976-8"><a href="run-simulator-using-a-previously-defined-parameter-set.html#cb976-8" aria-hidden="true" tabindex="-1"></a>{</span>
<span id="cb976-9"><a href="run-simulator-using-a-previously-defined-parameter-set.html#cb976-9" aria-hidden="true" tabindex="-1"></a>  <span class="co">#creates a sim object</span></span>
<span id="cb976-10"><a href="run-simulator-using-a-previously-defined-parameter-set.html#cb976-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">load</span>(<span class="fu">paste0</span>(here<span class="sc">::</span><span class="fu">here</span>(), <span class="st">&quot;/output/simout/&quot;</span>,filenames[i]))</span>
<span id="cb976-11"><a href="run-simulator-using-a-previously-defined-parameter-set.html#cb976-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb976-12"><a href="run-simulator-using-a-previously-defined-parameter-set.html#cb976-12" aria-hidden="true" tabindex="-1"></a>  <span class="co">#now let us take what we need from the simulation</span></span>
<span id="cb976-13"><a href="run-simulator-using-a-previously-defined-parameter-set.html#cb976-13" aria-hidden="true" tabindex="-1"></a>  res[i,<span class="dv">1</span><span class="sc">:</span><span class="dv">2</span>] <span class="ot">&lt;-</span> sim<span class="sc">$</span>para.space</span>
<span id="cb976-14"><a href="run-simulator-using-a-previously-defined-parameter-set.html#cb976-14" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb976-15"><a href="run-simulator-using-a-previously-defined-parameter-set.html#cb976-15" aria-hidden="true" tabindex="-1"></a>  <span class="co">#calculate a summary statistic: mean of pairwise fst values</span></span>
<span id="cb976-16"><a href="run-simulator-using-a-previously-defined-parameter-set.html#cb976-16" aria-hidden="true" tabindex="-1"></a>  <span class="co"># here we only take the lower triangle of the matrix to avoid the diagonal values,</span></span>
<span id="cb976-17"><a href="run-simulator-using-a-previously-defined-parameter-set.html#cb976-17" aria-hidden="true" tabindex="-1"></a>  <span class="co"># which are zero by definition (comparing each population to itself)</span></span>
<span id="cb976-18"><a href="run-simulator-using-a-previously-defined-parameter-set.html#cb976-18" aria-hidden="true" tabindex="-1"></a>  gen.mat <span class="ot">&lt;-</span> PopGenReport<span class="sc">::</span><span class="fu">pairwise.fstb</span>(sim<span class="sc">$</span>gi)  </span>
<span id="cb976-19"><a href="run-simulator-using-a-previously-defined-parameter-set.html#cb976-19" aria-hidden="true" tabindex="-1"></a>  res [i,<span class="dv">3</span>] <span class="ot">&lt;-</span> <span class="fu">mean</span>(gen.mat[<span class="fu">lower.tri</span>(gen.mat)])</span>
<span id="cb976-20"><a href="run-simulator-using-a-previously-defined-parameter-set.html#cb976-20" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb976-21"><a href="run-simulator-using-a-previously-defined-parameter-set.html#cb976-21" aria-hidden="true" tabindex="-1"></a>  <span class="co">#partial Mantel tests</span></span>
<span id="cb976-22"><a href="run-simulator-using-a-previously-defined-parameter-set.html#cb976-22" aria-hidden="true" tabindex="-1"></a>  wass <span class="ot">&lt;-</span> PopGenReport<span class="sc">::</span><span class="fu">wassermann</span>(<span class="at">eucl.mat =</span> <span class="fu">dist</span>(sim<span class="sc">$</span>para<span class="sc">$</span>locs), </span>
<span id="cb976-23"><a href="run-simulator-using-a-previously-defined-parameter-set.html#cb976-23" aria-hidden="true" tabindex="-1"></a>                                <span class="at">cost.mats =</span> <span class="fu">list</span>(<span class="at">cost=</span>sim<span class="sc">$</span>cost.mat), </span>
<span id="cb976-24"><a href="run-simulator-using-a-previously-defined-parameter-set.html#cb976-24" aria-hidden="true" tabindex="-1"></a>                                <span class="at">gen.mat =</span> gen.mat, <span class="at">plot=</span>F)<span class="sc">$</span>mantel.tab</span>
<span id="cb976-25"><a href="run-simulator-using-a-previously-defined-parameter-set.html#cb976-25" aria-hidden="true" tabindex="-1"></a>  res[i,<span class="dv">4</span><span class="sc">:</span><span class="dv">5</span>] <span class="ot">&lt;-</span> wass[wass<span class="sc">$</span>model <span class="sc">==</span> <span class="st">&quot;Gen ~Euclidean | cost&quot;</span>, <span class="dv">2</span><span class="sc">:</span><span class="dv">3</span>]</span>
<span id="cb976-26"><a href="run-simulator-using-a-previously-defined-parameter-set.html#cb976-26" aria-hidden="true" tabindex="-1"></a>  res[i,<span class="dv">6</span><span class="sc">:</span><span class="dv">7</span>] <span class="ot">&lt;-</span> wass[wass<span class="sc">$</span>model <span class="sc">==</span> <span class="st">&quot;Gen ~cost | Euclidean&quot;</span>, <span class="dv">2</span><span class="sc">:</span><span class="dv">3</span>]</span>
<span id="cb976-27"><a href="run-simulator-using-a-previously-defined-parameter-set.html#cb976-27" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb976-28"><a href="run-simulator-using-a-previously-defined-parameter-set.html#cb976-28" aria-hidden="true" tabindex="-1"></a>  <span class="co">#Sunder</span></span>
<span id="cb976-29"><a href="run-simulator-using-a-previously-defined-parameter-set.html#cb976-29" aria-hidden="true" tabindex="-1"></a>  res[i,<span class="dv">8</span>] <span class="ot">&lt;-</span> <span class="fu">getSunder</span>(<span class="at">gen=</span>sim<span class="sc">$</span>gi, <span class="at">locs=</span>sim<span class="sc">$</span>para<span class="sc">$</span>locs, </span>
<span id="cb976-30"><a href="run-simulator-using-a-previously-defined-parameter-set.html#cb976-30" aria-hidden="true" tabindex="-1"></a>                           <span class="at">cost=</span>sim<span class="sc">$</span>cost.mat, <span class="at">nit=</span><span class="dv">10</span><span class="sc">^</span><span class="dv">2</span>) <span class="co">#if you have time, set nit=10^3</span></span>
<span id="cb976-31"><a href="run-simulator-using-a-previously-defined-parameter-set.html#cb976-31" aria-hidden="true" tabindex="-1"></a>}</span></code></pre></div>
<pre><code>## Computations for model &#39;geog+envi&#39;
## Computations for model &#39;geog&#39;
## Computations for model &#39;envi&#39;
## 
## Computations for model &#39;geog+envi&#39;
## Computations for model &#39;geog&#39;
## Computations for model &#39;envi&#39;
## 
## Computations for model &#39;geog+envi&#39;
## Computations for model &#39;geog&#39;
## Computations for model &#39;envi&#39;
## 
## Computations for model &#39;geog+envi&#39;
## Computations for model &#39;geog&#39;
## Computations for model &#39;envi&#39;
## 
## Computations for model &#39;geog+envi&#39;
## Computations for model &#39;geog&#39;
## Computations for model &#39;envi&#39;
## 
## Computations for model &#39;geog+envi&#39;
## Computations for model &#39;geog&#39;
## Computations for model &#39;envi&#39;
## 
## Computations for model &#39;geog+envi&#39;
## Computations for model &#39;geog&#39;
## Computations for model &#39;envi&#39;
## 
## Computations for model &#39;geog+envi&#39;
## Computations for model &#39;geog&#39;
## Computations for model &#39;envi&#39;
## 
## Computations for model &#39;geog+envi&#39;
## Computations for model &#39;geog&#39;
## Computations for model &#39;envi&#39;
## 
## Computations for model &#39;geog+envi&#39;
## Computations for model &#39;geog&#39;
## Computations for model &#39;envi&#39;
## 
## Computations for model &#39;geog+envi&#39;
## Computations for model &#39;geog&#39;
## Computations for model &#39;envi&#39;
## 
## Computations for model &#39;geog+envi&#39;
## Computations for model &#39;geog&#39;
## Computations for model &#39;envi&#39;
## 
## Computations for model &#39;geog+envi&#39;
## Computations for model &#39;geog&#39;
## Computations for model &#39;envi&#39;
## 
## Computations for model &#39;geog+envi&#39;
## Computations for model &#39;geog&#39;
## Computations for model &#39;envi&#39;
## 
## Computations for model &#39;geog+envi&#39;
## Computations for model &#39;geog&#39;
## Computations for model &#39;envi&#39;</code></pre>
<p>Look at the ‘res’ data frame and check the results.</p>
<div class="sourceCode" id="cb978"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb978-1"><a href="run-simulator-using-a-previously-defined-parameter-set.html#cb978-1" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(res)</span></code></pre></div>
<pre><code>##   rep time         fst  r.Eucl p.Eucl r.cost p.cost Sunder
## 1   1    5 0.009291826 -0.1677  0.818 0.3232   0.09      E
## 2   2    5 0.009304023 -0.2004  0.862 0.3199  0.067      G
## 3   3    5 0.010189370  0.1331  0.242 0.0603  0.387    G+E
## 4   4    5 0.009573448 -0.0648   0.59 0.3461  0.077      G
## 5   5    5 0.009378724   0.111  0.307   0.03  0.457    G+E
## 6   1   25 0.021535598 -0.2469   0.92 0.6393  0.028      E</code></pre>
<p>The next step would be to visualise the results (e.g. plot runs over times and color by rep). A quick way to do that is to use the function ‘ggplot’ from the ‘ggplot2’ package. Here we add a jitter to keep points from overlapping too much.</p>
<div class="sourceCode" id="cb980"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb980-1"><a href="run-simulator-using-a-previously-defined-parameter-set.html#cb980-1" aria-hidden="true" tabindex="-1"></a>ggplot2<span class="sc">::</span><span class="fu">ggplot</span>(res, ggplot2<span class="sc">::</span><span class="fu">aes</span>(<span class="at">x=</span>time, <span class="at">y=</span>fst)) <span class="sc">+</span> </span>
<span id="cb980-2"><a href="run-simulator-using-a-previously-defined-parameter-set.html#cb980-2" aria-hidden="true" tabindex="-1"></a>         ggplot2<span class="sc">::</span><span class="fu">geom_point</span>(<span class="at">position =</span> ggplot2<span class="sc">::</span><span class="fu">position_jitter</span>(<span class="at">w =</span> <span class="fl">0.5</span>))</span></code></pre></div>
<p><img src="08-Week08_files/figure-html/unnamed-chunk-46-1.png" width="672" /></p>
<p>Now it is again time for you to experiment. For example, why not set up a simulation that varies the number of loci. Or as you may have seen even after 100 generation there was no sign that the mean pairwise Fst value is levelling off. So how long do you have to run a simulation in terms of time to see this (be aware that simulation runs take longer if you increase the number of timesteps)?</p>
<p>Questions:</p>
<ul>
<li>How would you set up a simulation experiment to compare type I error rates between partial Mantel test and Sunder?</li>
<li>How about statistical power?</li>
</ul>
<p>Have fun and please give us feed back what you think about this Worked Example. <a href="mailto:bernd.gruber@canberra.edu.au">Bernd Gruber</a>, <a href="erin.landguth@gmail.com">Erin Landguth</a>, <a href="mailto:helene.wagner@utoronto.ca">Helene Wagner</a>.</p>
</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="run-simulations-and-analyze-results.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="r-exercise-week-8.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": true,
"facebook": false,
"twitter": false,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": false
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": null,
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "section"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
