<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Worked Example | Landscape Genetic Data Analysis with R</title>
  <meta name="description" content="This is a web-interface to the teaching materials for the lab course ‘Landscape Genetic Data Analysis with R’ associated with the distributed graduate course ‘DGS Landscape Genetics.’ The output format is bookdown::gitbook." />
  <meta name="generator" content="bookdown 0.22 and GitBook 2.6.7" />

  <meta property="og:title" content="Worked Example | Landscape Genetic Data Analysis with R" />
  <meta property="og:type" content="book" />
  
  
  <meta property="og:description" content="This is a web-interface to the teaching materials for the lab course ‘Landscape Genetic Data Analysis with R’ associated with the distributed graduate course ‘DGS Landscape Genetics.’ The output format is bookdown::gitbook." />
  <meta name="github-repo" content="hhwagner1/LandGenCourse_book" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Worked Example | Landscape Genetic Data Analysis with R" />
  
  <meta name="twitter:description" content="This is a web-interface to the teaching materials for the lab course ‘Landscape Genetic Data Analysis with R’ associated with the distributed graduate course ‘DGS Landscape Genetics.’ The output format is bookdown::gitbook." />
  

<meta name="author" content="Editor: Helene Wagner (University of Toronto)" />


<meta name="date" content="2021-05-04" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="Week13.html"/>
<link rel="next" href="Week14.html"/>
<script src="libs/header-attrs-2.7/header-attrs.js"></script>
<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />









<link href="libs/anchor-sections-1.0.1/anchor-sections.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.0.1/anchor-sections.js"></script>
<script src="libs/kePrint-0.0.1/kePrint.js"></script>
<link href="libs/lightable-0.0.1/lightable.css" rel="stylesheet" />

<script>
/* ========================================================================
 * Bootstrap: transition.js v3.3.7
 * http://getbootstrap.com/javascript/#transitions
 * ========================================================================
 * Copyright 2011-2016 Twitter, Inc.
 * Licensed under MIT (https://github.com/twbs/bootstrap/blob/master/LICENSE)
 * ======================================================================== */


+function ($) {
  'use strict';

  // CSS TRANSITION SUPPORT (Shoutout: http://www.modernizr.com/)
  // ============================================================

  function transitionEnd() {
    var el = document.createElement('bootstrap')

    var transEndEventNames = {
      WebkitTransition : 'webkitTransitionEnd',
      MozTransition    : 'transitionend',
      OTransition      : 'oTransitionEnd otransitionend',
      transition       : 'transitionend'
    }

    for (var name in transEndEventNames) {
      if (el.style[name] !== undefined) {
        return { end: transEndEventNames[name] }
      }
    }

    return false // explicit for ie8 (  ._.)
  }

  // http://blog.alexmaccaw.com/css-transitions
  $.fn.emulateTransitionEnd = function (duration) {
    var called = false
    var $el = this
    $(this).one('bsTransitionEnd', function () { called = true })
    var callback = function () { if (!called) $($el).trigger($.support.transition.end) }
    setTimeout(callback, duration)
    return this
  }

  $(function () {
    $.support.transition = transitionEnd()

    if (!$.support.transition) return

    $.event.special.bsTransitionEnd = {
      bindType: $.support.transition.end,
      delegateType: $.support.transition.end,
      handle: function (e) {
        if ($(e.target).is(this)) return e.handleObj.handler.apply(this, arguments)
      }
    }
  })

}(jQuery);
</script>
<script>
/* ========================================================================
 * Bootstrap: collapse.js v3.3.7
 * http://getbootstrap.com/javascript/#collapse
 * ========================================================================
 * Copyright 2011-2016 Twitter, Inc.
 * Licensed under MIT (https://github.com/twbs/bootstrap/blob/master/LICENSE)
 * ======================================================================== */

/* jshint latedef: false */

+function ($) {
  'use strict';

  // COLLAPSE PUBLIC CLASS DEFINITION
  // ================================

  var Collapse = function (element, options) {
    this.$element      = $(element)
    this.options       = $.extend({}, Collapse.DEFAULTS, options)
    this.$trigger      = $('[data-toggle="collapse"][href="#' + element.id + '"],' +
                           '[data-toggle="collapse"][data-target="#' + element.id + '"]')
    this.transitioning = null

    if (this.options.parent) {
      this.$parent = this.getParent()
    } else {
      this.addAriaAndCollapsedClass(this.$element, this.$trigger)
    }

    if (this.options.toggle) this.toggle()
  }

  Collapse.VERSION  = '3.3.7'

  Collapse.TRANSITION_DURATION = 350

  Collapse.DEFAULTS = {
    toggle: true
  }

  Collapse.prototype.dimension = function () {
    var hasWidth = this.$element.hasClass('width')
    return hasWidth ? 'width' : 'height'
  }

  Collapse.prototype.show = function () {
    if (this.transitioning || this.$element.hasClass('in')) return

    var activesData
    var actives = this.$parent && this.$parent.children('.panel').children('.in, .collapsing')

    if (actives && actives.length) {
      activesData = actives.data('bs.collapse')
      if (activesData && activesData.transitioning) return
    }

    var startEvent = $.Event('show.bs.collapse')
    this.$element.trigger(startEvent)
    if (startEvent.isDefaultPrevented()) return

    if (actives && actives.length) {
      Plugin.call(actives, 'hide')
      activesData || actives.data('bs.collapse', null)
    }

    var dimension = this.dimension()

    this.$element
      .removeClass('collapse')
      .addClass('collapsing')[dimension](0)
      .attr('aria-expanded', true)

    this.$trigger
      .removeClass('collapsed')
      .attr('aria-expanded', true)

    this.transitioning = 1

    var complete = function () {
      this.$element
        .removeClass('collapsing')
        .addClass('collapse in')[dimension]('')
      this.transitioning = 0
      this.$element
        .trigger('shown.bs.collapse')
    }

    if (!$.support.transition) return complete.call(this)

    var scrollSize = $.camelCase(['scroll', dimension].join('-'))

    this.$element
      .one('bsTransitionEnd', $.proxy(complete, this))
      .emulateTransitionEnd(Collapse.TRANSITION_DURATION)[dimension](this.$element[0][scrollSize])
  }

  Collapse.prototype.hide = function () {
    if (this.transitioning || !this.$element.hasClass('in')) return

    var startEvent = $.Event('hide.bs.collapse')
    this.$element.trigger(startEvent)
    if (startEvent.isDefaultPrevented()) return

    var dimension = this.dimension()

    this.$element[dimension](this.$element[dimension]())[0].offsetHeight

    this.$element
      .addClass('collapsing')
      .removeClass('collapse in')
      .attr('aria-expanded', false)

    this.$trigger
      .addClass('collapsed')
      .attr('aria-expanded', false)

    this.transitioning = 1

    var complete = function () {
      this.transitioning = 0
      this.$element
        .removeClass('collapsing')
        .addClass('collapse')
        .trigger('hidden.bs.collapse')
    }

    if (!$.support.transition) return complete.call(this)

    this.$element
      [dimension](0)
      .one('bsTransitionEnd', $.proxy(complete, this))
      .emulateTransitionEnd(Collapse.TRANSITION_DURATION)
  }

  Collapse.prototype.toggle = function () {
    this[this.$element.hasClass('in') ? 'hide' : 'show']()
  }

  Collapse.prototype.getParent = function () {
    return $(this.options.parent)
      .find('[data-toggle="collapse"][data-parent="' + this.options.parent + '"]')
      .each($.proxy(function (i, element) {
        var $element = $(element)
        this.addAriaAndCollapsedClass(getTargetFromTrigger($element), $element)
      }, this))
      .end()
  }

  Collapse.prototype.addAriaAndCollapsedClass = function ($element, $trigger) {
    var isOpen = $element.hasClass('in')

    $element.attr('aria-expanded', isOpen)
    $trigger
      .toggleClass('collapsed', !isOpen)
      .attr('aria-expanded', isOpen)
  }

  function getTargetFromTrigger($trigger) {
    var href
    var target = $trigger.attr('data-target')
      || (href = $trigger.attr('href')) && href.replace(/.*(?=#[^\s]+$)/, '') // strip for ie7

    return $(target)
  }


  // COLLAPSE PLUGIN DEFINITION
  // ==========================

  function Plugin(option) {
    return this.each(function () {
      var $this   = $(this)
      var data    = $this.data('bs.collapse')
      var options = $.extend({}, Collapse.DEFAULTS, $this.data(), typeof option == 'object' && option)

      if (!data && options.toggle && /show|hide/.test(option)) options.toggle = false
      if (!data) $this.data('bs.collapse', (data = new Collapse(this, options)))
      if (typeof option == 'string') data[option]()
    })
  }

  var old = $.fn.collapse

  $.fn.collapse             = Plugin
  $.fn.collapse.Constructor = Collapse


  // COLLAPSE NO CONFLICT
  // ====================

  $.fn.collapse.noConflict = function () {
    $.fn.collapse = old
    return this
  }


  // COLLAPSE DATA-API
  // =================

  $(document).on('click.bs.collapse.data-api', '[data-toggle="collapse"]', function (e) {
    var $this   = $(this)

    if (!$this.attr('data-target')) e.preventDefault()

    var $target = getTargetFromTrigger($this)
    var data    = $target.data('bs.collapse')
    var option  = data ? 'toggle' : $this.data()

    Plugin.call($target, option)
  })

}(jQuery);
</script>
<script>
window.initializeCodeFolding = function(show) {

  // handlers for show-all and hide all
  $("#rmd-show-all-code").click(function() {
    // close the dropdown menu when an option is clicked
    $("#allCodeButton").dropdown("toggle");
    $('div.r-code-collapse').each(function() {
      $(this).collapse('show');
    });
  });
  $("#rmd-hide-all-code").click(function() {
    // close the dropdown menu when an option is clicked
    $("#allCodeButton").dropdown("toggle");
    $('div.r-code-collapse').each(function() {
      $(this).collapse('hide');
    });
  });

  // index for unique code element ids
  var currentIndex = 1;

  // select all R code blocks
  var rCodeBlocks = $('pre.sourceCode, pre.r, pre.python, pre.bash, pre.sql, pre.cpp, pre.stan');
  rCodeBlocks.each(function() {

    // if code block has been labeled with class `fold-show`, show the code on init!
    var classList = $(this).attr('class').split(/\s+/);
    for (var i = 0; i < classList.length; i++) {
    if (classList[i] === 'fold-show') {
        show = true;
      }
    }

    // create a collapsable div to wrap the code in
    var div = $('<div class="collapse r-code-collapse"></div>');
    if (show)
      div.addClass('in');
    var id = 'rcode-643E0F36' + currentIndex++;
    div.attr('id', id);
    $(this).before(div);
    $(this).detach().appendTo(div);

    // add a show code button right above
    var showCodeText = $('<span>' + (show ? 'Hide' : 'Code') + '</span>');
    var showCodeButton = $('<button type="button" class="btn btn-default btn-xs code-folding-btn pull-right"></button>');
    showCodeButton.append(showCodeText);
    showCodeButton
        .attr('data-toggle', 'collapse')
        .attr('data-target', '#' + id)
        .attr('aria-expanded', show)
        .attr('aria-controls', id);

    var buttonRow = $('<div class="row"></div>');
    var buttonCol = $('<div class="col-md-12"></div>');

    buttonCol.append(showCodeButton);
    buttonRow.append(buttonCol);

    div.before(buttonRow);

    // hack: return show to false, otherwise all next codeBlocks will be shown!
    show = false;

    // update state of button on show/hide
    div.on('hidden.bs.collapse', function () {
      showCodeText.text('Code');
    });
    div.on('show.bs.collapse', function () {
      showCodeText.text('Hide');
    });
  });

}
</script>
<script>
/* ========================================================================
 * Bootstrap: dropdown.js v3.3.7
 * http://getbootstrap.com/javascript/#dropdowns
 * ========================================================================
 * Copyright 2011-2016 Twitter, Inc.
 * Licensed under MIT (https://github.com/twbs/bootstrap/blob/master/LICENSE)
 * ======================================================================== */


+function ($) {
  'use strict';

  // DROPDOWN CLASS DEFINITION
  // =========================

  var backdrop = '.dropdown-backdrop'
  var toggle   = '[data-toggle="dropdown"]'
  var Dropdown = function (element) {
    $(element).on('click.bs.dropdown', this.toggle)
  }

  Dropdown.VERSION = '3.3.7'

  function getParent($this) {
    var selector = $this.attr('data-target')

    if (!selector) {
      selector = $this.attr('href')
      selector = selector && /#[A-Za-z]/.test(selector) && selector.replace(/.*(?=#[^\s]*$)/, '') // strip for ie7
    }

    var $parent = selector && $(selector)

    return $parent && $parent.length ? $parent : $this.parent()
  }

  function clearMenus(e) {
    if (e && e.which === 3) return
    $(backdrop).remove()
    $(toggle).each(function () {
      var $this         = $(this)
      var $parent       = getParent($this)
      var relatedTarget = { relatedTarget: this }

      if (!$parent.hasClass('open')) return

      if (e && e.type == 'click' && /input|textarea/i.test(e.target.tagName) && $.contains($parent[0], e.target)) return

      $parent.trigger(e = $.Event('hide.bs.dropdown', relatedTarget))

      if (e.isDefaultPrevented()) return

      $this.attr('aria-expanded', 'false')
      $parent.removeClass('open').trigger($.Event('hidden.bs.dropdown', relatedTarget))
    })
  }

  Dropdown.prototype.toggle = function (e) {
    var $this = $(this)

    if ($this.is('.disabled, :disabled')) return

    var $parent  = getParent($this)
    var isActive = $parent.hasClass('open')

    clearMenus()

    if (!isActive) {
      if ('ontouchstart' in document.documentElement && !$parent.closest('.navbar-nav').length) {
        // if mobile we use a backdrop because click events don't delegate
        $(document.createElement('div'))
          .addClass('dropdown-backdrop')
          .insertAfter($(this))
          .on('click', clearMenus)
      }

      var relatedTarget = { relatedTarget: this }
      $parent.trigger(e = $.Event('show.bs.dropdown', relatedTarget))

      if (e.isDefaultPrevented()) return

      $this
        .trigger('focus')
        .attr('aria-expanded', 'true')

      $parent
        .toggleClass('open')
        .trigger($.Event('shown.bs.dropdown', relatedTarget))
    }

    return false
  }

  Dropdown.prototype.keydown = function (e) {
    if (!/(38|40|27|32)/.test(e.which) || /input|textarea/i.test(e.target.tagName)) return

    var $this = $(this)

    e.preventDefault()
    e.stopPropagation()

    if ($this.is('.disabled, :disabled')) return

    var $parent  = getParent($this)
    var isActive = $parent.hasClass('open')

    if (!isActive && e.which != 27 || isActive && e.which == 27) {
      if (e.which == 27) $parent.find(toggle).trigger('focus')
      return $this.trigger('click')
    }

    var desc = ' li:not(.disabled):visible a'
    var $items = $parent.find('.dropdown-menu' + desc)

    if (!$items.length) return

    var index = $items.index(e.target)

    if (e.which == 38 && index > 0)                 index--         // up
    if (e.which == 40 && index < $items.length - 1) index++         // down
    if (!~index)                                    index = 0

    $items.eq(index).trigger('focus')
  }


  // DROPDOWN PLUGIN DEFINITION
  // ==========================

  function Plugin(option) {
    return this.each(function () {
      var $this = $(this)
      var data  = $this.data('bs.dropdown')

      if (!data) $this.data('bs.dropdown', (data = new Dropdown(this)))
      if (typeof option == 'string') data[option].call($this)
    })
  }

  var old = $.fn.dropdown

  $.fn.dropdown             = Plugin
  $.fn.dropdown.Constructor = Dropdown


  // DROPDOWN NO CONFLICT
  // ====================

  $.fn.dropdown.noConflict = function () {
    $.fn.dropdown = old
    return this
  }


  // APPLY TO STANDARD DROPDOWN ELEMENTS
  // ===================================

  $(document)
    .on('click.bs.dropdown.data-api', clearMenus)
    .on('click.bs.dropdown.data-api', '.dropdown form', function (e) { e.stopPropagation() })
    .on('click.bs.dropdown.data-api', toggle, Dropdown.prototype.toggle)
    .on('keydown.bs.dropdown.data-api', toggle, Dropdown.prototype.keydown)
    .on('keydown.bs.dropdown.data-api', '.dropdown-menu', Dropdown.prototype.keydown)

}(jQuery);
</script>
<style type="text/css">
.code-folding-btn {
  margin-bottom: 4px;
}

.row { display: flex; }
.collapse { display: none; }
.in { display:block }
.pull-right > .dropdown-menu {
    right: 0;
    left: auto;
}

.dropdown-menu {
    position: absolute;
    top: 100%;
    left: 0;
    z-index: 1000;
    display: none;
    float: left;
    min-width: 160px;
    padding: 5px 0;
    margin: 2px 0 0;
    font-size: 14px;
    text-align: left;
    list-style: none;
    background-color: #fff;
    -webkit-background-clip: padding-box;
    background-clip: padding-box;
    border: 1px solid #ccc;
    border: 1px solid rgba(0,0,0,.15);
    border-radius: 4px;
    -webkit-box-shadow: 0 6px 12px rgba(0,0,0,.175);
    box-shadow: 0 6px 12px rgba(0,0,0,.175);
}

.open > .dropdown-menu {
    display: block;
    color: #ffffff;
    background-color: #ffffff;
    background-image: none;
    border-color: #92897e;
}

.dropdown-menu > li > a {
  display: block;
  padding: 3px 20px;
  clear: both;
  font-weight: 400;
  line-height: 1.42857143;
  color: #000000;
  white-space: nowrap;
}

.dropdown-menu > li > a:hover,
.dropdown-menu > li > a:focus {
  color: #ffffff;
  text-decoration: none;
  background-color: #e95420;
}

.dropdown-menu > .active > a,
.dropdown-menu > .active > a:hover,
.dropdown-menu > .active > a:focus {
  color: #ffffff;
  text-decoration: none;
  background-color: #e95420;
  outline: 0;
}
.dropdown-menu > .disabled > a,
.dropdown-menu > .disabled > a:hover,
.dropdown-menu > .disabled > a:focus {
  color: #aea79f;
}

.dropdown-menu > .disabled > a:hover,
.dropdown-menu > .disabled > a:focus {
  text-decoration: none;
  cursor: not-allowed;
  background-color: transparent;
  background-image: none;
  filter: progid:DXImageTransform.Microsoft.gradient(enabled = false);
}

.btn {
  display: inline-block;
  margin-bottom: 1;
  font-weight: normal;
  text-align: center;
  white-space: nowrap;
  vertical-align: middle;
  -ms-touch-action: manipulation;
      touch-action: manipulation;
  cursor: pointer;
  background-image: none;
  border: 1px solid transparent;
  padding: 4px 8px;
  font-size: 14px;
  line-height: 1.42857143;
  border-radius: 4px;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

.btn:focus,
.btn:active:focus,
.btn.active:focus,
.btn.focus,
.btn:active.focus,
.btn.active.focus {
  outline: 5px auto -webkit-focus-ring-color;
  outline-offset: -2px;
}
.btn:hover,
.btn:focus,
.btn.focus {
  color: #ffffff;
  text-decoration: none;
}
.btn:active,
.btn.active {
  background-image: none;
  outline: 0;
  box-shadow: inset 0 3px 5px rgba(0, 0, 0, 0.125);
}
.btn.disabled,
.btn[disabled],
fieldset[disabled] .btn {
  cursor: not-allowed;
  filter: alpha(opacity=65);
  opacity: 0.65;
  box-shadow: none;
}
a.btn.disabled,
fieldset[disabled] a.btn {
  pointer-events: none;
}
.btn-default {
  color: #ffffff;
  background-color: #aea79f; #important
  border-color: #aea79f;
}

.btn-default:focus,
.btn-default.focus {
  color: #ffffff;
  background-color: #978e83;
  border-color: #6f675e;
}

.btn-default:hover {
  color: #ffffff;
  background-color: #978e83;
  border-color: #92897e;
}
.btn-default:active,
.btn-default.active,
.btn-group > .btn:not(:first-child):not(:last-child):not(.dropdown-toggle) {
  border-radius: 0;
}
.btn-group > .btn:first-child {
  margin-left: 0;
}
.btn-group > .btn:first-child:not(:last-child):not(.dropdown-toggle) {
  border-top-right-radius: 0;
  border-bottom-right-radius: 0;
}
.btn-group > .btn:last-child:not(:first-child),
.btn-group > .dropdown-toggle:not(:first-child) {
  border-top-left-radius: 0;
  border-bottom-left-radius: 0;
}
.btn-group > .btn-group {
  float: left;
}
.btn-group > .btn-group:not(:first-child):not(:last-child) > .btn {
  border-radius: 0;
}
.btn-group > .btn-group:first-child:not(:last-child) > .btn:last-child,
.btn-group > .btn-group:first-child:not(:last-child) > .dropdown-toggle {
  border-top-right-radius: 0;
  border-bottom-right-radius: 0;
}
.btn-group > .btn-group:last-child:not(:first-child) > .btn:first-child {
  border-top-left-radius: 0;
  border-bottom-left-radius: 0;
}
.btn-group .dropdown-toggle:active,
.btn-group.open .dropdown-toggle {
  outline: 0;
}
.btn-group > .btn + .dropdown-toggle {
  padding-right: 8px;
  padding-left: 8px;
}
.btn-group > .btn-lg + .dropdown-toggle {
  padding-right: 12px;
  padding-left: 12px;
}
.btn-group.open .dropdown-toggle {
  box-shadow: inset 0 3px 5px rgba(0, 0, 0, 0.125);
}
.btn-group.open .dropdown-toggle.btn-link {
  box-shadow: none;
}

</style>
<script>
var str = '<div class="btn-group pull-right" style="position: fixed; right: 50px; top: 10px; z-index: 200"><button type="button" class="btn btn-default btn-xs dropdown-toggle" id="allCodeButton" data-toggle="dropdown" aria-haspopup="true" aria-expanded="true" data-_extension-text-contrast=""><span>Code</span> <span class="caret"></span></button><ul class="dropdown-menu" style="min-width: 50px;"><li><a id="rmd-show-all-code" href="#">Show All Code</a></li><li><a id="rmd-hide-all-code" href="#">Hide All Code</a></li></ul></div>';
document.write(str);
</script>
<script>
$(document).ready(function () {
  window.initializeCodeFolding("show" === "hide");
});
</script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>


<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">Report About Something</a></li>

<li class="divider"></li>
<li class="part"><span><b>Getting Started</b></span></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i>Introduction</a>
<ul>
<li class="chapter" data-level="" data-path="how-to-use-this-book.html"><a href="how-to-use-this-book.html"><i class="fa fa-check"></i>How to use this Book</a>
<ul>
<li class="chapter" data-level="" data-path="how-to-use-this-book.html"><a href="how-to-use-this-book.html#book-structure"><i class="fa fa-check"></i>1. Book Structure</a>
<ul>
<li class="chapter" data-level="" data-path="how-to-use-this-book.html"><a href="how-to-use-this-book.html#a-getting-started"><i class="fa fa-check"></i>a) Getting Started</a></li>
<li class="chapter" data-level="" data-path="how-to-use-this-book.html"><a href="how-to-use-this-book.html#b-basic-topics"><i class="fa fa-check"></i>b) Basic Topics</a></li>
<li class="chapter" data-level="" data-path="how-to-use-this-book.html"><a href="how-to-use-this-book.html#c-advanced-topics"><i class="fa fa-check"></i>c) Advanced Topics</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="how-to-use-this-book.html"><a href="how-to-use-this-book.html#find-what-is-relevant-for-you"><i class="fa fa-check"></i>2. Find what is relevant for you</a>
<ul>
<li class="chapter" data-level="" data-path="how-to-use-this-book.html"><a href="how-to-use-this-book.html#a-how-to-use-the-labs-weeks-1---8"><i class="fa fa-check"></i>a) How to use the labs (Weeks 1 - 8)</a></li>
<li class="chapter" data-level="" data-path="how-to-use-this-book.html"><a href="how-to-use-this-book.html#b-check-contents"><i class="fa fa-check"></i>b) Check contents</a></li>
</ul></li>
<li><a href="how-to-use-this-book.html#course-r-package-landgencourse">3. Course R package <code>LandGenCourse</code></a>
<ul>
<li class="chapter" data-level="" data-path="how-to-use-this-book.html"><a href="how-to-use-this-book.html#a-how-to-install-or-update"><i class="fa fa-check"></i>a) How to install (or update)</a></li>
<li><a href="how-to-use-this-book.html#b-how-to-use-the-r-package-landgencourse">b) How to use the R package <code>LandGenCourse</code></a></li>
<li class="chapter" data-level="" data-path="how-to-use-this-book.html"><a href="how-to-use-this-book.html#c-video-instructions-for-beginners"><i class="fa fa-check"></i>c) Video instructions for beginners</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="" data-path="list-of-packages-by-vignette.html"><a href="list-of-packages-by-vignette.html"><i class="fa fa-check"></i>List of Packages by Vignette</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="review-of-r-skills.html"><a href="review-of-r-skills.html"><i class="fa fa-check"></i>Review of R Skills</a>
<ul>
<li class="chapter" data-level="" data-path="basic-r-programming.html"><a href="basic-r-programming.html"><i class="fa fa-check"></i>Basic R Programming</a>
<ul>
<li class="chapter" data-level="" data-path="basic-r-programming.html"><a href="basic-r-programming.html#overview"><i class="fa fa-check"></i>1. Overview</a></li>
<li class="chapter" data-level="" data-path="basic-r-programming.html"><a href="basic-r-programming.html#data-types"><i class="fa fa-check"></i>2. Data Types</a>
<ul>
<li class="chapter" data-level="" data-path="basic-r-programming.html"><a href="basic-r-programming.html#numeric-data-types"><i class="fa fa-check"></i>2.1. Numeric Data Types</a></li>
<li class="chapter" data-level="" data-path="basic-r-programming.html"><a href="basic-r-programming.html#characters"><i class="fa fa-check"></i>2.2. Characters</a></li>
<li class="chapter" data-level="" data-path="basic-r-programming.html"><a href="basic-r-programming.html#factors"><i class="fa fa-check"></i>2.3. Factors</a></li>
<li class="chapter" data-level="" data-path="basic-r-programming.html"><a href="basic-r-programming.html#logical-types"><i class="fa fa-check"></i>2.4. Logical Types</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="basic-r-programming.html"><a href="basic-r-programming.html#data-containers"><i class="fa fa-check"></i>3. Data Containers</a>
<ul>
<li class="chapter" data-level="" data-path="basic-r-programming.html"><a href="basic-r-programming.html#vectors"><i class="fa fa-check"></i>3.1. Vectors</a></li>
<li class="chapter" data-level="" data-path="basic-r-programming.html"><a href="basic-r-programming.html#matrices"><i class="fa fa-check"></i>3.2. Matrices</a></li>
<li class="chapter" data-level="" data-path="basic-r-programming.html"><a href="basic-r-programming.html#lists"><i class="fa fa-check"></i>3.3. Lists</a></li>
<li class="chapter" data-level="" data-path="basic-r-programming.html"><a href="basic-r-programming.html#data-frames"><i class="fa fa-check"></i>3.4. Data Frames</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="basic-r-programming.html"><a href="basic-r-programming.html#programming"><i class="fa fa-check"></i>4. Programming</a>
<ul>
<li class="chapter" data-level="" data-path="basic-r-programming.html"><a href="basic-r-programming.html#function-writing"><i class="fa fa-check"></i>4.1. Function Writing</a></li>
<li class="chapter" data-level="" data-path="basic-r-programming.html"><a href="basic-r-programming.html#variable-scope"><i class="fa fa-check"></i>4.2. Variable Scope</a></li>
<li class="chapter" data-level="" data-path="basic-r-programming.html"><a href="basic-r-programming.html#decision-making"><i class="fa fa-check"></i>4.3. Decision Making</a></li>
<li class="chapter" data-level="" data-path="basic-r-programming.html"><a href="basic-r-programming.html#flow-control"><i class="fa fa-check"></i>4.4. Flow Control</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="" data-path="r-graphics.html"><a href="r-graphics.html"><i class="fa fa-check"></i>R Graphics</a>
<ul>
<li class="chapter" data-level="" data-path="r-graphics.html"><a href="r-graphics.html#overview-1"><i class="fa fa-check"></i>1. Overview</a></li>
<li class="chapter" data-level="" data-path="r-graphics.html"><a href="r-graphics.html#univariate-plots"><i class="fa fa-check"></i>2. Univariate plots</a>
<ul>
<li class="chapter" data-level="" data-path="r-graphics.html"><a href="r-graphics.html#barplots-and-histograms"><i class="fa fa-check"></i>2.1. Barplots and histograms</a></li>
<li class="chapter" data-level="" data-path="r-graphics.html"><a href="r-graphics.html#plotting-parameters"><i class="fa fa-check"></i>2.2. Plotting Parameters</a></li>
<li class="chapter" data-level="" data-path="r-graphics.html"><a href="r-graphics.html#density-plots"><i class="fa fa-check"></i>2.3. Density Plots</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="r-graphics.html"><a href="r-graphics.html#bivariate-plots"><i class="fa fa-check"></i>3. Bivariate plots</a>
<ul>
<li class="chapter" data-level="" data-path="r-graphics.html"><a href="r-graphics.html#overlaying-plots-by-grouping"><i class="fa fa-check"></i>3.1. Overlaying Plots by grouping</a></li>
<li class="chapter" data-level="" data-path="r-graphics.html"><a href="r-graphics.html#boxplots"><i class="fa fa-check"></i>3.2. Boxplots</a></li>
<li class="chapter" data-level="" data-path="r-graphics.html"><a href="r-graphics.html#scatter-plots"><i class="fa fa-check"></i>3.3. Scatter Plots</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="r-graphics.html"><a href="r-graphics.html#advanced-plotting"><i class="fa fa-check"></i>4. Advanced plotting</a>
<ul>
<li class="chapter" data-level="" data-path="r-graphics.html"><a href="r-graphics.html#multiple-plots"><i class="fa fa-check"></i>4.1. Multiple Plots</a></li>
<li class="chapter" data-level="" data-path="r-graphics.html"><a href="r-graphics.html#color-palettes"><i class="fa fa-check"></i>4.2. Color Palettes</a></li>
<li class="chapter" data-level="" data-path="r-graphics.html"><a href="r-graphics.html#rcolorbrewer"><i class="fa fa-check"></i>4.3. RColorBrewer</a></li>
<li class="chapter" data-level="" data-path="r-graphics.html"><a href="r-graphics.html#saving-imagery"><i class="fa fa-check"></i>4.4. Saving Imagery</a></li>
<li class="chapter" data-level="" data-path="r-graphics.html"><a href="r-graphics.html#interactive-graphics"><i class="fa fa-check"></i>4.5. Interactive Graphics</a></li>
</ul></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="" data-path="further-r-resources.html"><a href="further-r-resources.html"><i class="fa fa-check"></i>Further R Resources</a></li>
<li class="part"><span><b>Basic Topics</b></span></li>
<li class="chapter" data-level="" data-path="Week1.html"><a href="Week1.html"><i class="fa fa-check"></i>Week 1: Importing Genetic Data</a>
<ul>
<li class="chapter" data-level="" data-path="video-1.html"><a href="video-1.html"><i class="fa fa-check"></i>View Course Video</a>
<ul>
<li class="chapter" data-level="" data-path="video-1.html"><a href="video-1.html#preview-slides"><i class="fa fa-check"></i>Preview Slides</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="tutorial-1.html"><a href="tutorial-1.html"><i class="fa fa-check"></i>Interactive Tutorial 1</a>
<ul>
<li class="chapter" data-level="" data-path="tutorial-1.html"><a href="tutorial-1.html#list-of-r-commands-used"><i class="fa fa-check"></i>List of R commands used</a></li>
<li class="chapter" data-level="" data-path="tutorial-1.html"><a href="tutorial-1.html#instructions"><i class="fa fa-check"></i>Instructions</a>
<ul>
<li class="chapter" data-level="" data-path="tutorial-1.html"><a href="tutorial-1.html#a-how-to-access-tutorials"><i class="fa fa-check"></i>a) How to access tutorials:</a></li>
<li class="chapter" data-level="" data-path="tutorial-1.html"><a href="tutorial-1.html#b-how-to-complete-tutorial"><i class="fa fa-check"></i>b) How to complete tutorial:</a></li>
<li class="chapter" data-level="" data-path="tutorial-1.html"><a href="tutorial-1.html#c-how-to-submit-answers-participating-institutions-only"><i class="fa fa-check"></i>c) How to submit answers (participating institutions only):</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="" data-path="WE-1.html"><a href="WE-1.html"><i class="fa fa-check"></i>Worked Example</a>
<ul>
<li class="chapter" data-level="" data-path="WE-1.html"><a href="WE-1.html#overview-of-worked-example"><i class="fa fa-check"></i>1. Overview of Worked Example</a>
<ul>
<li class="chapter" data-level="" data-path="WE-1.html"><a href="WE-1.html#a.-goals"><i class="fa fa-check"></i>a. Goals</a></li>
<li class="chapter" data-level="" data-path="WE-1.html"><a href="WE-1.html#b.-data-set"><i class="fa fa-check"></i>b. Data set</a></li>
<li class="chapter" data-level="" data-path="WE-1.html"><a href="WE-1.html#c.-required-r-libraries"><i class="fa fa-check"></i>c. Required R libraries</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="WE-1.html"><a href="WE-1.html#import-data-from-.csv-file"><i class="fa fa-check"></i>2. Import data from .csv file</a>
<ul>
<li class="chapter" data-level="" data-path="WE-1.html"><a href="WE-1.html#a.-preparation"><i class="fa fa-check"></i>a. Preparation</a></li>
<li class="chapter" data-level="" data-path="WE-1.html"><a href="WE-1.html#b.-excel-file"><i class="fa fa-check"></i>b. Excel file</a></li>
<li class="chapter" data-level="" data-path="WE-1.html"><a href="WE-1.html#c.-import-with-function-read.csv"><i class="fa fa-check"></i>c. Import with function ‘read.csv’</a></li>
<li class="chapter" data-level="" data-path="WE-1.html"><a href="WE-1.html#d.-create-id-variable"><i class="fa fa-check"></i>d. Create ID variable</a></li>
<li class="chapter" data-level="" data-path="WE-1.html"><a href="WE-1.html#e.-export-with-function-write.csv"><i class="fa fa-check"></i>e. Export with function ‘write.csv’</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="WE-1.html"><a href="WE-1.html#create-genind-object"><i class="fa fa-check"></i>3. Create ‘genind’ Object</a>
<ul>
<li class="chapter" data-level="" data-path="WE-1.html"><a href="WE-1.html#a.-read-the-helpfile"><i class="fa fa-check"></i>a. Read the helpfile</a></li>
<li class="chapter" data-level="" data-path="WE-1.html"><a href="WE-1.html#b.-set-all-arguments-of-function-df2genind"><i class="fa fa-check"></i>b. Set all arguments of function ‘df2genind’</a></li>
<li class="chapter" data-level="" data-path="WE-1.html"><a href="WE-1.html#c.-check-imported-data"><i class="fa fa-check"></i>c. Check imported data</a></li>
<li class="chapter" data-level="" data-path="WE-1.html"><a href="WE-1.html#d.-summarize-genind-object"><i class="fa fa-check"></i>d. Summarize ‘genind’ object</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="WE-1.html"><a href="WE-1.html#view-information-stored-in-genind-object"><i class="fa fa-check"></i>4. View information stored in ‘genind’ object</a>
<ul>
<li><a href="WE-1.html#a.-slot-tab">a. Slot <code>@tab</code></a></li>
<li><a href="WE-1.html#b.-slot-loc.n.all">b. Slot <code>@loc.n.all</code></a></li>
<li><a href="WE-1.html#c.-slot-loc.fac">c. Slot <code>@loc.fac</code></a></li>
<li><a href="WE-1.html#d.-slot-all.names">d. Slot <code>@all.names</code></a></li>
<li class="chapter" data-level="" data-path="WE-1.html"><a href="WE-1.html#e.-further-slots"><i class="fa fa-check"></i>e. Further slots</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="WE-1.html"><a href="WE-1.html#import-data-with-gstudio-package"><i class="fa fa-check"></i>5. Import data with ‘gstudio’ package</a>
<ul>
<li class="chapter" data-level="" data-path="WE-1.html"><a href="WE-1.html#a.-function-read_population"><i class="fa fa-check"></i>a. Function ‘read_population’</a></li>
<li class="chapter" data-level="" data-path="WE-1.html"><a href="WE-1.html#b.-set-arguments"><i class="fa fa-check"></i>b. Set arguments</a></li>
<li class="chapter" data-level="" data-path="WE-1.html"><a href="WE-1.html#c.-check-imported-data-1"><i class="fa fa-check"></i>c. Check imported data</a></li>
<li class="chapter" data-level="" data-path="WE-1.html"><a href="WE-1.html#d.-add-frogid"><i class="fa fa-check"></i>d. Add FrogID</a></li>
<li class="chapter" data-level="" data-path="WE-1.html"><a href="WE-1.html#e.-convert-to-genind-object"><i class="fa fa-check"></i>e. Convert to ‘genind’ object</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="" data-path="r-exercise-week-1.html"><a href="r-exercise-week-1.html"><i class="fa fa-check"></i>R exercise Week 1</a></li>
<li class="chapter" data-level="" data-path="bonus-1a.html"><a href="bonus-1a.html"><i class="fa fa-check"></i>Bonus: Importing SNP data</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="Week2.html"><a href="Week2.html"><i class="fa fa-check"></i>Week 2: Spatial Data</a>
<ul>
<li class="chapter" data-level="" data-path="video-2.html"><a href="video-2.html"><i class="fa fa-check"></i>View Course Video</a>
<ul>
<li class="chapter" data-level="" data-path="video-2.html"><a href="video-2.html#preview-slides-1"><i class="fa fa-check"></i>Preview Slides</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="tutorial-2.html"><a href="tutorial-2.html"><i class="fa fa-check"></i>Interactive Tutorial 2</a>
<ul>
<li class="chapter" data-level="" data-path="tutorial-2.html"><a href="tutorial-2.html#list-of-r-commands-used-1"><i class="fa fa-check"></i>List of R commands used</a></li>
<li class="chapter" data-level="" data-path="tutorial-2.html"><a href="tutorial-2.html#instructions-1"><i class="fa fa-check"></i>Instructions</a>
<ul>
<li class="chapter" data-level="" data-path="tutorial-2.html"><a href="tutorial-2.html#a-how-to-access-tutorials-1"><i class="fa fa-check"></i>a) How to access tutorials:</a></li>
<li class="chapter" data-level="" data-path="tutorial-2.html"><a href="tutorial-2.html#b-how-to-complete-tutorial-1"><i class="fa fa-check"></i>b) How to complete tutorial:</a></li>
<li class="chapter" data-level="" data-path="tutorial-2.html"><a href="tutorial-2.html#c-how-to-submit-answers-participating-institutions-only-1"><i class="fa fa-check"></i>c) How to submit answers (participating institutions only):</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="" data-path="WE-2.html"><a href="WE-2.html"><i class="fa fa-check"></i>Worked Example</a>
<ul>
<li class="chapter" data-level="" data-path="WE-2.html"><a href="WE-2.html#overview-of-worked-example-1"><i class="fa fa-check"></i>1. Overview of Worked Example</a>
<ul>
<li class="chapter" data-level="" data-path="WE-2.html"><a href="WE-2.html#a.-goals-1"><i class="fa fa-check"></i>a. Goals</a></li>
<li class="chapter" data-level="" data-path="WE-2.html"><a href="WE-2.html#b.-data-set-1"><i class="fa fa-check"></i>b. Data set</a></li>
<li class="chapter" data-level="" data-path="WE-2.html"><a href="WE-2.html#c.-required-r-libraries-1"><i class="fa fa-check"></i>c. Required R libraries</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="WE-2.html"><a href="WE-2.html#import-site-data-from-.csv-file"><i class="fa fa-check"></i>2. Import site data from .csv file</a>
<ul>
<li class="chapter" data-level="" data-path="WE-2.html"><a href="WE-2.html#a.-import-data-into-spatialpointsdataframe"><i class="fa fa-check"></i>a. Import data into SpatialPointsDataFrame</a></li>
<li class="chapter" data-level="" data-path="WE-2.html"><a href="WE-2.html#b.-add-spatial-reference-data"><i class="fa fa-check"></i>b. Add spatial reference data</a></li>
<li class="chapter" data-level="" data-path="WE-2.html"><a href="WE-2.html#c.-change-projection"><i class="fa fa-check"></i>c. Change projection</a></li>
<li class="chapter" data-level="" data-path="WE-2.html"><a href="WE-2.html#d.-access-data-in-spatialpointsdataframe"><i class="fa fa-check"></i>d. Access data in ‘SpatialPointsDataFrame’</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="WE-2.html"><a href="WE-2.html#display-raster-data-and-overlay-sampling-locations-extract-data"><i class="fa fa-check"></i>3. Display raster data and overlay sampling locations, extract data</a>
<ul>
<li class="chapter" data-level="" data-path="WE-2.html"><a href="WE-2.html#a.-display-raster-data"><i class="fa fa-check"></i>a. Display raster data</a></li>
<li class="chapter" data-level="" data-path="WE-2.html"><a href="WE-2.html#b.-change-color-ramp-add-sampling-locations"><i class="fa fa-check"></i>b. Change color ramp, add sampling locations</a></li>
<li class="chapter" data-level="" data-path="WE-2.html"><a href="WE-2.html#c.-extract-raster-values-at-sampling-locations"><i class="fa fa-check"></i>c. Extract raster values at sampling locations</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="WE-2.html"><a href="WE-2.html#calculate-landscape-metrics"><i class="fa fa-check"></i>4. Calculate landscape metrics</a>
<ul>
<li class="chapter" data-level="" data-path="WE-2.html"><a href="WE-2.html#a.-calculate-patch--class--and-landscape-level-landscape-metrics"><i class="fa fa-check"></i>a. Calculate patch-, class- and landscape level landscape metrics</a></li>
<li class="chapter" data-level="" data-path="WE-2.html"><a href="WE-2.html#b.-calculate-patch-level-landscape-metrics-for-evergreen-forest"><i class="fa fa-check"></i>b. Calculate patch-level landscape metrics for ‘Evergreen Forest’</a></li>
<li class="chapter" data-level="" data-path="WE-2.html"><a href="WE-2.html#c.-extract-forest-patch-size-at-samplig-locations"><i class="fa fa-check"></i>c. Extract forest patch size at samplig locations</a></li>
<li class="chapter" data-level="" data-path="WE-2.html"><a href="WE-2.html#d.-plot-a-bubble-map-of-forest-patch-size-at-sampling-locations"><i class="fa fa-check"></i>d. Plot a bubble map of forest patch size at sampling locations</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="WE-2.html"><a href="WE-2.html#sample-landscape-metrics-within-buffer-around-sampling-locations"><i class="fa fa-check"></i>5. Sample landscape metrics within buffer around sampling locations</a>
<ul>
<li class="chapter" data-level="" data-path="WE-2.html"><a href="WE-2.html#b.-extract-landscape-metric-of-choice-for-a-single-cover-type-as-vector"><i class="fa fa-check"></i>b. Extract landscape metric of choice for a single cover type (as vector)</a></li>
<li class="chapter" data-level="" data-path="WE-2.html"><a href="WE-2.html#c.-extract-landscape-metric-of-choice-for-all-cover-types-as-data-frame."><i class="fa fa-check"></i>c. Extract landscape metric of choice for all cover types (as data frame).</a></li>
<li class="chapter" data-level="" data-path="WE-2.html"><a href="WE-2.html#d.-extract-all-landscape-metrics-for-a-single-cover-type-as-data-frame"><i class="fa fa-check"></i>d. Extract all landscape metrics for a single cover type (as data frame)</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="" data-path="r-exercise-week-2.html"><a href="r-exercise-week-2.html"><i class="fa fa-check"></i>R Exercise Week 2</a></li>
<li class="chapter" data-level="" data-path="bonus-2a.html"><a href="bonus-2a.html"><i class="fa fa-check"></i>Bonus: Using package sf</a></li>
<li class="chapter" data-level="" data-path="bonus-2b.html"><a href="bonus-2b.html"><i class="fa fa-check"></i>Bonus: Visualize LULC map</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="Week3.html"><a href="Week3.html"><i class="fa fa-check"></i>Week 3: Genetic Diversity</a>
<ul>
<li class="chapter" data-level="" data-path="video-3.html"><a href="video-3.html"><i class="fa fa-check"></i>View Course Video</a>
<ul>
<li class="chapter" data-level="" data-path="video-3.html"><a href="video-3.html#video-part-1"><i class="fa fa-check"></i>Video, Part 1</a></li>
<li class="chapter" data-level="" data-path="video-3.html"><a href="video-3.html#video-part-2"><i class="fa fa-check"></i>Video, Part 2</a></li>
<li class="chapter" data-level="" data-path="video-3.html"><a href="video-3.html#preview-slides-2"><i class="fa fa-check"></i>Preview Slides</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="tutorial-3.html"><a href="tutorial-3.html"><i class="fa fa-check"></i>Interactive Tutorial 3</a>
<ul>
<li class="chapter" data-level="" data-path="tutorial-3.html"><a href="tutorial-3.html#list-of-r-commands-used-2"><i class="fa fa-check"></i>List of R commands used</a></li>
<li class="chapter" data-level="" data-path="tutorial-3.html"><a href="tutorial-3.html#instructions-2"><i class="fa fa-check"></i>Instructions</a>
<ul>
<li class="chapter" data-level="" data-path="tutorial-3.html"><a href="tutorial-3.html#a-how-to-access-tutorials-2"><i class="fa fa-check"></i>a) How to access tutorials:</a></li>
<li class="chapter" data-level="" data-path="tutorial-3.html"><a href="tutorial-3.html#b-how-to-complete-tutorial-2"><i class="fa fa-check"></i>b) How to complete tutorial:</a></li>
<li class="chapter" data-level="" data-path="tutorial-3.html"><a href="tutorial-3.html#c-how-to-submit-answers-participating-institutions-only-2"><i class="fa fa-check"></i>c) How to submit answers (participating institutions only):</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="" data-path="WE-3.html"><a href="WE-3.html"><i class="fa fa-check"></i>Worked Example</a>
<ul>
<li class="chapter" data-level="" data-path="WE-3.html"><a href="WE-3.html#overview-of-worked-example-2"><i class="fa fa-check"></i>1. Overview of Worked Example</a>
<ul>
<li class="chapter" data-level="" data-path="WE-3.html"><a href="WE-3.html#a.-goals-2"><i class="fa fa-check"></i>a. Goals</a></li>
<li class="chapter" data-level="" data-path="WE-3.html"><a href="WE-3.html#b.-data-set-2"><i class="fa fa-check"></i>b. Data set</a></li>
<li class="chapter" data-level="" data-path="WE-3.html"><a href="WE-3.html#c.-required-r-libraries-2"><i class="fa fa-check"></i>c. Required R libraries</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="WE-3.html"><a href="WE-3.html#basic-checking-of-markers-and-populations"><i class="fa fa-check"></i>2. Basic checking of markers and populations</a>
<ul>
<li class="chapter" data-level="" data-path="WE-3.html"><a href="WE-3.html#a.-re-create-genind-object"><i class="fa fa-check"></i>a. Re-create genind object</a></li>
<li class="chapter" data-level="" data-path="WE-3.html"><a href="WE-3.html#b.-check-that-markers-are-polymorphic"><i class="fa fa-check"></i>b. Check that markers are polymorphic</a></li>
<li class="chapter" data-level="" data-path="WE-3.html"><a href="WE-3.html#c.-check-for-deviations-from-hardy-weinberg-equilibrium-hwe"><i class="fa fa-check"></i>c. Check for deviations from Hardy-Weinberg equilibrium (HWE)</a></li>
<li class="chapter" data-level="" data-path="WE-3.html"><a href="WE-3.html#d.-check-for-linkage-disequilibrium-ld"><i class="fa fa-check"></i>d. Check for linkage disequilibrium (LD)</a></li>
<li class="chapter" data-level="" data-path="WE-3.html"><a href="WE-3.html#e.-check-for-null-alleles"><i class="fa fa-check"></i>e. Check for null alleles</a></li>
<li class="chapter" data-level="" data-path="WE-3.html"><a href="WE-3.html#f.-overall-interpretation"><i class="fa fa-check"></i>f. Overall interpretation</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="WE-3.html"><a href="WE-3.html#assess-genetic-diversity"><i class="fa fa-check"></i>3. Assess genetic diversity</a>
<ul>
<li class="chapter" data-level="" data-path="WE-3.html"><a href="WE-3.html#a.-rarefied-allelic-richness"><i class="fa fa-check"></i>a. Rarefied allelic richness</a></li>
<li class="chapter" data-level="" data-path="WE-3.html"><a href="WE-3.html#b.-observed-and-expected-heterozygosity"><i class="fa fa-check"></i>b. Observed and expected heterozygosity</a></li>
<li class="chapter" data-level="" data-path="WE-3.html"><a href="WE-3.html#c.-create-table-with-sitel-level-genetic-diversity-measures"><i class="fa fa-check"></i>c. Create table with sitel-level genetic diversity measures</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="WE-3.html"><a href="WE-3.html#aggregate-genetic-data-at-population-level-allele-frequencies"><i class="fa fa-check"></i>4. Aggregate genetic data at population level (allele frequencies)</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="r-exercise-week-3.html"><a href="r-exercise-week-3.html"><i class="fa fa-check"></i>R Exercise Week 3</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="Week4.html"><a href="Week4.html"><i class="fa fa-check"></i>Week 4: Metapopulations</a>
<ul>
<li class="chapter" data-level="" data-path="video-4.html"><a href="video-4.html"><i class="fa fa-check"></i>View Course Video</a>
<ul>
<li class="chapter" data-level="" data-path="video-4.html"><a href="video-4.html#video-part-1-1"><i class="fa fa-check"></i>Video, Part 1</a></li>
<li class="chapter" data-level="" data-path="video-4.html"><a href="video-4.html#video-part-2-1"><i class="fa fa-check"></i>Video, Part 2</a></li>
<li class="chapter" data-level="" data-path="video-4.html"><a href="video-4.html#preview-slides-3"><i class="fa fa-check"></i>Preview Slides</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="tutorial-4.html"><a href="tutorial-4.html"><i class="fa fa-check"></i>Interactive Tutorial 4</a>
<ul>
<li class="chapter" data-level="" data-path="tutorial-4.html"><a href="tutorial-4.html#list-of-r-commands-used-3"><i class="fa fa-check"></i>List of R commands used</a></li>
<li class="chapter" data-level="" data-path="tutorial-4.html"><a href="tutorial-4.html#instructions-3"><i class="fa fa-check"></i>Instructions</a>
<ul>
<li class="chapter" data-level="" data-path="tutorial-4.html"><a href="tutorial-4.html#a-how-to-access-tutorials-3"><i class="fa fa-check"></i>a) How to access tutorials:</a></li>
<li class="chapter" data-level="" data-path="tutorial-4.html"><a href="tutorial-4.html#b-how-to-complete-tutorial-3"><i class="fa fa-check"></i>b) How to complete tutorial:</a></li>
<li class="chapter" data-level="" data-path="tutorial-4.html"><a href="tutorial-4.html#c-how-to-submit-answers-participating-institutions-only-3"><i class="fa fa-check"></i>c) How to submit answers (participating institutions only):</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="" data-path="WE-4.html"><a href="WE-4.html"><i class="fa fa-check"></i>Worked Example</a>
<ul>
<li class="chapter" data-level="" data-path="WE-4.html"><a href="WE-4.html#overview-of-worked-example-3"><i class="fa fa-check"></i>1. Overview of Worked Example</a>
<ul>
<li class="chapter" data-level="" data-path="WE-4.html"><a href="WE-4.html#a.-goals-3"><i class="fa fa-check"></i>a. Goals</a></li>
<li class="chapter" data-level="" data-path="WE-4.html"><a href="WE-4.html#b.-data-set-3"><i class="fa fa-check"></i>b. Data set</a></li>
<li class="chapter" data-level="" data-path="WE-4.html"><a href="WE-4.html#reference"><i class="fa fa-check"></i>Reference:</a></li>
<li class="chapter" data-level="" data-path="WE-4.html"><a href="WE-4.html#c.-required-r-libraries-3"><i class="fa fa-check"></i>c. Required R libraries</a></li>
<li class="chapter" data-level="" data-path="WE-4.html"><a href="WE-4.html#d.-import-data"><i class="fa fa-check"></i>d. Import data</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="WE-4.html"><a href="WE-4.html#spatial-distribution-of-genetic-structure"><i class="fa fa-check"></i>2. Spatial distribution of genetic structure</a>
<ul>
<li class="chapter" data-level="" data-path="WE-4.html"><a href="WE-4.html#a.-creating-a-genind-object-with-the-hierarchical-data-set"><i class="fa fa-check"></i>a. Creating a ‘genind’ object with the hierarchical data set</a></li>
<li class="chapter" data-level="" data-path="WE-4.html"><a href="WE-4.html#b.-hierarchical-amova"><i class="fa fa-check"></i>b. Hierarchical AMOVA</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="WE-4.html"><a href="WE-4.html#what-determines-genetic-differentiation-among-sites"><i class="fa fa-check"></i>3. What determines genetic differentiation among sites?</a>
<ul>
<li class="chapter" data-level="" data-path="WE-4.html"><a href="WE-4.html#a.-correlation-matrix"><i class="fa fa-check"></i>a. Correlation matrix</a></li>
<li class="chapter" data-level="" data-path="WE-4.html"><a href="WE-4.html#b.-scatterplots"><i class="fa fa-check"></i>b. Scatterplots</a></li>
<li class="chapter" data-level="" data-path="WE-4.html"><a href="WE-4.html#c.-regression-model"><i class="fa fa-check"></i>c. Regression model</a></li>
<li class="chapter" data-level="" data-path="WE-4.html"><a href="WE-4.html#d.-which-populations-dont-fit-the-general-pattern"><i class="fa fa-check"></i>d. Which populations don’t fit the general pattern?</a></li>
<li class="chapter" data-level="" data-path="WE-4.html"><a href="WE-4.html#e.-regression-model-without-outliers"><i class="fa fa-check"></i>e. Regression model without outliers</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="WE-4.html"><a href="WE-4.html#what-determines-genetic-diversity"><i class="fa fa-check"></i>4. What determines genetic diversity?</a>
<ul>
<li class="chapter" data-level="" data-path="WE-4.html"><a href="WE-4.html#a.-correlation-matrix-1"><i class="fa fa-check"></i>a. Correlation matrix</a></li>
<li class="chapter" data-level="" data-path="WE-4.html"><a href="WE-4.html#b.-regression-models"><i class="fa fa-check"></i>b. Regression models</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="WE-4.html"><a href="WE-4.html#are-genetic-differentiation-and-diversity-related"><i class="fa fa-check"></i>5. Are genetic differentiation and diversity related?</a></li>
<li class="chapter" data-level="" data-path="WE-4.html"><a href="WE-4.html#effect-of-recent-extinction-events"><i class="fa fa-check"></i>6. Effect of recent extinction events</a>
<ul>
<li class="chapter" data-level="" data-path="WE-4.html"><a href="WE-4.html#a.-effect-of-patch-extinction-event-temporal-data-set"><i class="fa fa-check"></i>a. Effect of patch extinction event (temporal data set)</a></li>
<li class="chapter" data-level="" data-path="WE-4.html"><a href="WE-4.html#b.-power-analysis"><i class="fa fa-check"></i>b. Power analysis</a></li>
<li class="chapter" data-level="" data-path="WE-4.html"><a href="WE-4.html#c.-sample-size-calculation"><i class="fa fa-check"></i>c. Sample size calculation</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="" data-path="r-exercise-week-4.html"><a href="r-exercise-week-4.html"><i class="fa fa-check"></i>R Exercise Week 4</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="Week5.html"><a href="Week5.html"><i class="fa fa-check"></i>Week 5: Spatial Statistics</a>
<ul>
<li class="chapter" data-level="" data-path="video-5.html"><a href="video-5.html"><i class="fa fa-check"></i>View Course Video</a>
<ul>
<li class="chapter" data-level="" data-path="video-5.html"><a href="video-5.html#preview-slides-4"><i class="fa fa-check"></i>Preview Slides</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="tutorial-5.html"><a href="tutorial-5.html"><i class="fa fa-check"></i>Interactive Tutorial 5</a>
<ul>
<li class="chapter" data-level="" data-path="tutorial-5.html"><a href="tutorial-5.html#list-of-r-commands-used-4"><i class="fa fa-check"></i>List of R commands used</a></li>
<li class="chapter" data-level="" data-path="tutorial-5.html"><a href="tutorial-5.html#instructions-4"><i class="fa fa-check"></i>Instructions</a>
<ul>
<li class="chapter" data-level="" data-path="tutorial-5.html"><a href="tutorial-5.html#a-how-to-access-tutorials-4"><i class="fa fa-check"></i>a) How to access tutorials:</a></li>
<li class="chapter" data-level="" data-path="tutorial-5.html"><a href="tutorial-5.html#b-how-to-complete-tutorial-4"><i class="fa fa-check"></i>b) How to complete tutorial:</a></li>
<li class="chapter" data-level="" data-path="tutorial-5.html"><a href="tutorial-5.html#c-how-to-submit-answers-participating-institutions-only-4"><i class="fa fa-check"></i>c) How to submit answers (participating institutions only):</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="" data-path="WE-5.html"><a href="WE-5.html"><i class="fa fa-check"></i>Worked Example</a>
<ul>
<li class="chapter" data-level="" data-path="WE-5.html"><a href="WE-5.html#overview-of-worked-example-4"><i class="fa fa-check"></i>1. Overview of Worked Example</a>
<ul>
<li class="chapter" data-level="" data-path="WE-5.html"><a href="WE-5.html#a.-goals-4"><i class="fa fa-check"></i>a. Goals</a></li>
<li class="chapter" data-level="" data-path="WE-5.html"><a href="WE-5.html#b.-data-set-4"><i class="fa fa-check"></i>b. Data set</a></li>
<li class="chapter" data-level="" data-path="WE-5.html"><a href="WE-5.html#c.-required-r-packages"><i class="fa fa-check"></i>c. Required R packages</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="WE-5.html"><a href="WE-5.html#data-import-and-manipulation"><i class="fa fa-check"></i>2. Data import and manipulation</a>
<ul>
<li class="chapter" data-level="" data-path="WE-5.html"><a href="WE-5.html#a.-create-an-ecogen-object"><i class="fa fa-check"></i>a. Create an ecogen object</a></li>
<li class="chapter" data-level="" data-path="WE-5.html"><a href="WE-5.html#b.-aggregate-to-ecopop"><i class="fa fa-check"></i>b. Aggregate to ecopop</a></li>
<li class="chapter" data-level="" data-path="WE-5.html"><a href="WE-5.html#b.-add-site-level-data"><i class="fa fa-check"></i>b. Add site-level data</a></li>
<li class="chapter" data-level="" data-path="WE-5.html"><a href="WE-5.html#c.-export-to-adegenet-and-gstudio"><i class="fa fa-check"></i>c. Export to adegenet and gstudio</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="WE-5.html"><a href="WE-5.html#calculate-genetic-distances"><i class="fa fa-check"></i>3. Calculate genetic distances</a>
<ul>
<li class="chapter" data-level="" data-path="WE-5.html"><a href="WE-5.html#a.-genetic-distances-calculated-from-genind-object"><i class="fa fa-check"></i>a. Genetic distances calculated from genind object</a></li>
<li class="chapter" data-level="" data-path="WE-5.html"><a href="WE-5.html#b.-more-distance-matrices-with-gstudio"><i class="fa fa-check"></i>b. More distance matrices with ‘gstudio’</a></li>
<li class="chapter" data-level="" data-path="WE-5.html"><a href="WE-5.html#c.-assemble-distance-matrices"><i class="fa fa-check"></i>c. Assemble distance matrices</a></li>
<li class="chapter" data-level="" data-path="WE-5.html"><a href="WE-5.html#d.-export-genetic-distance-matrices"><i class="fa fa-check"></i>d. Export genetic distance matrices</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="WE-5.html"><a href="WE-5.html#perform-a-mantel-test-to-test-for-ibd"><i class="fa fa-check"></i>4. Perform a Mantel test to test for IBD</a>
<ul>
<li class="chapter" data-level="" data-path="WE-5.html"><a href="WE-5.html#a.-visually-check-linearity"><i class="fa fa-check"></i>a. Visually check linearity</a></li>
<li class="chapter" data-level="" data-path="WE-5.html"><a href="WE-5.html#b.-perform-mantel-test"><i class="fa fa-check"></i>b. Perform Mantel test</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="WE-5.html"><a href="WE-5.html#create-mantel-correlogram-for-genetic-data"><i class="fa fa-check"></i>5. Create Mantel correlogram for genetic data</a>
<ul>
<li class="chapter" data-level="" data-path="WE-5.html"><a href="WE-5.html#a.-create-a-first-mantel-correlogram"><i class="fa fa-check"></i>a. Create a first Mantel correlogram</a></li>
<li class="chapter" data-level="" data-path="WE-5.html"><a href="WE-5.html#b.-vary-distance-class-definition"><i class="fa fa-check"></i>b. Vary distance class definition</a></li>
<li class="chapter" data-level="" data-path="WE-5.html"><a href="WE-5.html#c.-alternative-measures-of-genetic-distances"><i class="fa fa-check"></i>c. Alternative measures of genetic distances</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="WE-5.html"><a href="WE-5.html#specify-spatial-weights-and-calculate-morans-i"><i class="fa fa-check"></i>6. Specify spatial weights and calculate Moran’s I</a>
<ul>
<li class="chapter" data-level="" data-path="WE-5.html"><a href="WE-5.html#a.-defining-spatial-neighbors"><i class="fa fa-check"></i>a. Defining spatial neighbors</a></li>
<li class="chapter" data-level="" data-path="WE-5.html"><a href="WE-5.html#b.-defining-spatial-weights"><i class="fa fa-check"></i>b. Defining spatial weights</a></li>
<li class="chapter" data-level="" data-path="WE-5.html"><a href="WE-5.html#c.-calculating-and-testing-morans-i"><i class="fa fa-check"></i>c. Calculating and testing Moran’s I</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="" data-path="r-exercise-week-5.html"><a href="r-exercise-week-5.html"><i class="fa fa-check"></i>R Exercise Week 5</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="Week6.html"><a href="Week6.html"><i class="fa fa-check"></i>Week 6: Quantitative Genetics</a>
<ul>
<li class="chapter" data-level="" data-path="video-6.html"><a href="video-6.html"><i class="fa fa-check"></i>View Course Video</a>
<ul>
<li class="chapter" data-level="" data-path="video-6.html"><a href="video-6.html#video-part-1-2"><i class="fa fa-check"></i>Video, Part 1</a></li>
<li class="chapter" data-level="" data-path="video-6.html"><a href="video-6.html#video-part-2-2"><i class="fa fa-check"></i>Video, Part 2</a></li>
<li class="chapter" data-level="" data-path="video-6.html"><a href="video-6.html#preview-slides-5"><i class="fa fa-check"></i>Preview Slides</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="tutorial-6.html"><a href="tutorial-6.html"><i class="fa fa-check"></i>Interactive Tutorial 6</a>
<ul>
<li class="chapter" data-level="" data-path="tutorial-6.html"><a href="tutorial-6.html#list-of-r-commands-used-5"><i class="fa fa-check"></i>List of R commands used</a></li>
<li class="chapter" data-level="" data-path="tutorial-6.html"><a href="tutorial-6.html#instructions-5"><i class="fa fa-check"></i>Instructions</a>
<ul>
<li class="chapter" data-level="" data-path="tutorial-6.html"><a href="tutorial-6.html#a-how-to-access-tutorials-5"><i class="fa fa-check"></i>a) How to access tutorials:</a></li>
<li class="chapter" data-level="" data-path="tutorial-6.html"><a href="tutorial-6.html#b-how-to-complete-tutorial-5"><i class="fa fa-check"></i>b) How to complete tutorial:</a></li>
<li class="chapter" data-level="" data-path="tutorial-6.html"><a href="tutorial-6.html#c-how-to-submit-answers-participating-institutions-only-5"><i class="fa fa-check"></i>c) How to submit answers (participating institutions only):</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="" data-path="WE-6.html"><a href="WE-6.html"><i class="fa fa-check"></i>Worked Example</a>
<ul>
<li class="chapter" data-level="" data-path="WE-6.html"><a href="WE-6.html#overview-of-worked-example-5"><i class="fa fa-check"></i>1. Overview of Worked Example</a>
<ul>
<li class="chapter" data-level="" data-path="WE-6.html"><a href="WE-6.html#a.-goals-5"><i class="fa fa-check"></i>a. Goals</a></li>
<li class="chapter" data-level="" data-path="WE-6.html"><a href="WE-6.html#b.-data-set-5"><i class="fa fa-check"></i>b. Data set</a></li>
<li class="chapter" data-level="" data-path="WE-6.html"><a href="WE-6.html#c.-required-r-packages-1"><i class="fa fa-check"></i>c. Required R packages</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="WE-6.html"><a href="WE-6.html#estimate-genetic-and-non-genetic-variance-components-from-a-common-garden-experiment"><i class="fa fa-check"></i>2. Estimate genetic and non-genetic variance components from a common garden experiment</a>
<ul>
<li class="chapter" data-level="" data-path="WE-6.html"><a href="WE-6.html#a.-import-phenytypic-data"><i class="fa fa-check"></i>a. Import phenytypic data</a></li>
<li class="chapter" data-level="" data-path="WE-6.html"><a href="WE-6.html#b.-fit-linear-mixed-effects-models-lmm-to-trait-data-observed-on-families"><i class="fa fa-check"></i>b. Fit linear mixed effects models (LMM) to trait data observed on families</a></li>
<li class="chapter" data-level="" data-path="WE-6.html"><a href="WE-6.html#c.-compare-model-fit"><i class="fa fa-check"></i>c. Compare model fit</a></li>
<li class="chapter" data-level="" data-path="WE-6.html"><a href="WE-6.html#d.-check-model-validity"><i class="fa fa-check"></i>d. Check model validity</a></li>
<li class="chapter" data-level="" data-path="WE-6.html"><a href="WE-6.html#e.-estimate-variance-components"><i class="fa fa-check"></i>e. Estimate variance components</a></li>
<li class="chapter" data-level="" data-path="WE-6.html"><a href="WE-6.html#f.-significance-testing"><i class="fa fa-check"></i>f. Significance testing</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="WE-6.html"><a href="WE-6.html#estimate-trait-heritability"><i class="fa fa-check"></i>3. Estimate trait heritability</a>
<ul>
<li class="chapter" data-level="" data-path="WE-6.html"><a href="WE-6.html#a.-estimate-heritability"><i class="fa fa-check"></i>a. Estimate heritability</a></li>
<li class="chapter" data-level="" data-path="WE-6.html"><a href="WE-6.html#b.-estimate-trait-differentiation"><i class="fa fa-check"></i>b. Estimate trait differentiation</a></li>
</ul></li>
<li><a href="WE-6.html#compare-q_st-to-f_st-measured-from-snp-data">4. Compare <span class="math inline">\(Q_{st}\)</span> to <span class="math inline">\(F_{st}\)</span> measured from SNP data</a>
<ul>
<li class="chapter" data-level="" data-path="WE-6.html"><a href="WE-6.html#a.-import-ecogen-object"><i class="fa fa-check"></i>a. Import ‘ecogen’ object</a></li>
<li><a href="WE-6.html#b.-compare-q_st-to-f_st">b. Compare <span class="math inline">\(Q_{ST}\)</span> to <span class="math inline">\(F_{ST}\)</span></a></li>
</ul></li>
<li class="chapter" data-level="" data-path="WE-6.html"><a href="WE-6.html#assess-correlation-between-trait-and-environment"><i class="fa fa-check"></i>5. Assess correlation between trait and environment</a>
<ul>
<li class="chapter" data-level="" data-path="WE-6.html"><a href="WE-6.html#a.-describe-and-extract-effects"><i class="fa fa-check"></i>a. Describe and extract effects</a></li>
<li class="chapter" data-level="" data-path="WE-6.html"><a href="WE-6.html#b.-correlation-of-population-effect-with-environment"><i class="fa fa-check"></i>b. Correlation of population effect with environment</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="" data-path="r-exercise-week-6.html"><a href="r-exercise-week-6.html"><i class="fa fa-check"></i>R Exercise Week 6</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="Week7.html"><a href="Week7.html"><i class="fa fa-check"></i>Week 7: Spatial Linear Models</a>
<ul>
<li class="chapter" data-level="" data-path="video-7.html"><a href="video-7.html"><i class="fa fa-check"></i>View Course Video</a>
<ul>
<li class="chapter" data-level="" data-path="video-7.html"><a href="video-7.html#video-part-1-3"><i class="fa fa-check"></i>Video, Part 1</a></li>
<li class="chapter" data-level="" data-path="video-7.html"><a href="video-7.html#video-part-2-3"><i class="fa fa-check"></i>Video, Part 2</a></li>
<li class="chapter" data-level="" data-path="video-7.html"><a href="video-7.html#preview-slides-6"><i class="fa fa-check"></i>Preview Slides</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="tutorial-7.html"><a href="tutorial-7.html"><i class="fa fa-check"></i>Interactive Tutorial 7</a>
<ul>
<li class="chapter" data-level="" data-path="tutorial-7.html"><a href="tutorial-7.html#list-of-r-commands-used-6"><i class="fa fa-check"></i>List of R commands used</a></li>
<li class="chapter" data-level="" data-path="tutorial-7.html"><a href="tutorial-7.html#instructions-6"><i class="fa fa-check"></i>Instructions</a>
<ul>
<li class="chapter" data-level="" data-path="tutorial-7.html"><a href="tutorial-7.html#a-how-to-access-tutorials-6"><i class="fa fa-check"></i>a) How to access tutorials:</a></li>
<li class="chapter" data-level="" data-path="tutorial-7.html"><a href="tutorial-7.html#b-how-to-complete-tutorial-6"><i class="fa fa-check"></i>b) How to complete tutorial:</a></li>
<li class="chapter" data-level="" data-path="tutorial-7.html"><a href="tutorial-7.html#c-how-to-submit-answers-participating-institutions-only-6"><i class="fa fa-check"></i>c) How to submit answers (participating institutions only):</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="" data-path="WE-7.html"><a href="WE-7.html"><i class="fa fa-check"></i>Worked Example</a>
<ul>
<li class="chapter" data-level="" data-path="WE-7.html"><a href="WE-7.html#overview-of-worked-example-6"><i class="fa fa-check"></i>1. Overview of Worked Example</a>
<ul>
<li class="chapter" data-level="" data-path="WE-7.html"><a href="WE-7.html#a.-goals-6"><i class="fa fa-check"></i>a. Goals</a></li>
<li class="chapter" data-level="" data-path="WE-7.html"><a href="WE-7.html#b.-data-set-6"><i class="fa fa-check"></i>b. Data set</a></li>
<li class="chapter" data-level="" data-path="WE-7.html"><a href="WE-7.html#c.-required-r-packages-2"><i class="fa fa-check"></i>c. Required R packages</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="WE-7.html"><a href="WE-7.html#estimate-genetic-and-non-genetic-variance-components-from-a-common-garden-experiment-1"><i class="fa fa-check"></i>2. Estimate genetic and non-genetic variance components from a common garden experiment</a>
<ul>
<li class="chapter" data-level="" data-path="WE-7.html"><a href="WE-7.html#a.-import-phenytypic-data-1"><i class="fa fa-check"></i>a. Import phenytypic data</a></li>
<li class="chapter" data-level="" data-path="WE-7.html"><a href="WE-7.html#b.-fit-linear-mixed-effects-models-lmm-to-trait-data-observed-on-families-1"><i class="fa fa-check"></i>b. Fit linear mixed effects models (LMM) to trait data observed on families</a></li>
<li class="chapter" data-level="" data-path="WE-7.html"><a href="WE-7.html#c.-compare-model-fit-1"><i class="fa fa-check"></i>c. Compare model fit</a></li>
<li class="chapter" data-level="" data-path="WE-7.html"><a href="WE-7.html#d.-check-model-validity-1"><i class="fa fa-check"></i>d. Check model validity</a></li>
<li class="chapter" data-level="" data-path="WE-7.html"><a href="WE-7.html#e.-estimate-variance-components-1"><i class="fa fa-check"></i>e. Estimate variance components</a></li>
<li class="chapter" data-level="" data-path="WE-7.html"><a href="WE-7.html#f.-significance-testing-1"><i class="fa fa-check"></i>f. Significance testing</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="WE-7.html"><a href="WE-7.html#estimate-trait-heritability-1"><i class="fa fa-check"></i>3. Estimate trait heritability</a>
<ul>
<li class="chapter" data-level="" data-path="WE-7.html"><a href="WE-7.html#a.-estimate-heritability-1"><i class="fa fa-check"></i>a. Estimate heritability</a></li>
<li class="chapter" data-level="" data-path="WE-7.html"><a href="WE-7.html#b.-estimate-trait-differentiation-1"><i class="fa fa-check"></i>b. Estimate trait differentiation</a></li>
</ul></li>
<li><a href="WE-7.html#compare-q_st-to-f_st-measured-from-snp-data-1">4. Compare <span class="math inline">\(Q_{st}\)</span> to <span class="math inline">\(F_{st}\)</span> measured from SNP data</a>
<ul>
<li class="chapter" data-level="" data-path="WE-7.html"><a href="WE-7.html#a.-import-ecogen-object-1"><i class="fa fa-check"></i>a. Import ‘ecogen’ object</a></li>
<li><a href="WE-7.html#b.-compare-q_st-to-f_st-1">b. Compare <span class="math inline">\(Q_{ST}\)</span> to <span class="math inline">\(F_{ST}\)</span></a></li>
</ul></li>
<li class="chapter" data-level="" data-path="WE-7.html"><a href="WE-7.html#assess-correlation-between-trait-and-environment-1"><i class="fa fa-check"></i>5. Assess correlation between trait and environment</a>
<ul>
<li class="chapter" data-level="" data-path="WE-7.html"><a href="WE-7.html#a.-describe-and-extract-effects-1"><i class="fa fa-check"></i>a. Describe and extract effects</a></li>
<li class="chapter" data-level="" data-path="WE-7.html"><a href="WE-7.html#b.-correlation-of-population-effect-with-environment-1"><i class="fa fa-check"></i>b. Correlation of population effect with environment</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="" data-path="r-exercise-week-6-1.html"><a href="r-exercise-week-6-1.html"><i class="fa fa-check"></i>R Exercise Week 6</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="Week8.html"><a href="Week8.html"><i class="fa fa-check"></i>Week 8: Simulation Experiments</a>
<ul>
<li class="chapter" data-level="" data-path="video-8.html"><a href="video-8.html"><i class="fa fa-check"></i>View Course Video</a>
<ul>
<li class="chapter" data-level="" data-path="video-8.html"><a href="video-8.html#video-part-1-4"><i class="fa fa-check"></i>Video, Part 1</a></li>
<li class="chapter" data-level="" data-path="video-8.html"><a href="video-8.html#video-part-2-4"><i class="fa fa-check"></i>Video, Part 2</a></li>
<li class="chapter" data-level="" data-path="video-8.html"><a href="video-8.html#preview-slides-7"><i class="fa fa-check"></i>Preview Slides</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="tutorial-8.html"><a href="tutorial-8.html"><i class="fa fa-check"></i>Interactive Tutorial 8</a>
<ul>
<li class="chapter" data-level="" data-path="tutorial-8.html"><a href="tutorial-8.html#list-of-r-commands-used-7"><i class="fa fa-check"></i>List of R commands used</a></li>
<li class="chapter" data-level="" data-path="tutorial-8.html"><a href="tutorial-8.html#instructions-7"><i class="fa fa-check"></i>Instructions</a>
<ul>
<li class="chapter" data-level="" data-path="tutorial-8.html"><a href="tutorial-8.html#a-how-to-access-tutorials-7"><i class="fa fa-check"></i>a) How to access tutorials:</a></li>
<li class="chapter" data-level="" data-path="tutorial-8.html"><a href="tutorial-8.html#b-how-to-complete-tutorial-7"><i class="fa fa-check"></i>b) How to complete tutorial:</a></li>
<li class="chapter" data-level="" data-path="tutorial-8.html"><a href="tutorial-8.html#c-how-to-submit-answers-participating-institutions-only-7"><i class="fa fa-check"></i>c) How to submit answers (participating institutions only):</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="" data-path="WE-8.html"><a href="WE-8.html"><i class="fa fa-check"></i>Worked Example</a>
<ul>
<li class="chapter" data-level="" data-path="WE-8.html"><a href="WE-8.html#overview-of-worked-example-7"><i class="fa fa-check"></i>1. Overview of Worked Example</a>
<ul>
<li class="chapter" data-level="" data-path="WE-8.html"><a href="WE-8.html#a.-goals-7"><i class="fa fa-check"></i>a. Goals</a></li>
<li class="chapter" data-level="" data-path="WE-8.html"><a href="WE-8.html#b.-data-set-7"><i class="fa fa-check"></i>b. Data set</a></li>
<li class="chapter" data-level="" data-path="WE-8.html"><a href="WE-8.html#c.-required-r-libraries-4"><i class="fa fa-check"></i>c. Required R libraries</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="WE-8.html"><a href="WE-8.html#initialize-a-landscape"><i class="fa fa-check"></i>2. Initialize a landscape</a>
<ul>
<li class="chapter" data-level="" data-path="WE-8.html"><a href="WE-8.html#a.-create-a-random-landscape"><i class="fa fa-check"></i>a. Create a random landscape</a></li>
<li class="chapter" data-level="" data-path="WE-8.html"><a href="WE-8.html#b.-add-populations-to-the-landscape-using-minimal-distance"><i class="fa fa-check"></i>b. Add populations to the landscape (using minimal distance)</a></li>
<li class="chapter" data-level="" data-path="WE-8.html"><a href="WE-8.html#c.-initialise-a-metapopulation"><i class="fa fa-check"></i>c. Initialise a metapopulation</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="WE-8.html"><a href="WE-8.html#define-simulation-parameters"><i class="fa fa-check"></i>3. Define simulation parameters</a>
<ul>
<li class="chapter" data-level="" data-path="WE-8.html"><a href="WE-8.html#a.-define-your-metapopulation"><i class="fa fa-check"></i>a. Define your metapopulation</a></li>
<li class="chapter" data-level="" data-path="WE-8.html"><a href="WE-8.html#b.-initialise-your-population-on-the-landscape"><i class="fa fa-check"></i>b. Initialise your population on the landscape</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="WE-8.html"><a href="WE-8.html#run-simulations-and-analyze-results"><i class="fa fa-check"></i>4. Run simulations and analyze results</a>
<ul>
<li class="chapter" data-level="" data-path="WE-8.html"><a href="WE-8.html#a.-run-your-simulation-over-multiple-time-steps-years"><i class="fa fa-check"></i>a. Run your simulation over multiple time steps (years)</a></li>
<li class="chapter" data-level="" data-path="WE-8.html"><a href="WE-8.html#b.-analyse-your-simulated-population-with-a-partial-mantel-test"><i class="fa fa-check"></i>b. Analyse your simulated population with a partial Mantel test</a></li>
<li class="chapter" data-level="" data-path="WE-8.html"><a href="WE-8.html#c.-optional-analyze-your-simulated-populations-using-sunder"><i class="fa fa-check"></i>c. Optional: Analyze your simulated populations using ‘Sunder’</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="WE-8.html"><a href="WE-8.html#run-simulator-using-a-previously-defined-parameter-set"><i class="fa fa-check"></i>5. Run simulator using a previously defined parameter set</a>
<ul>
<li class="chapter" data-level="" data-path="WE-8.html"><a href="WE-8.html#a.-specify-and-record-the-parameter-combinations"><i class="fa fa-check"></i>a. Specify and record the parameter combinations</a></li>
<li class="chapter" data-level="" data-path="WE-8.html"><a href="WE-8.html#b.-run-the-simulator-over-every-parameter-combination."><i class="fa fa-check"></i>b. Run the simulator over every parameter combination.</a></li>
<li class="chapter" data-level="" data-path="WE-8.html"><a href="WE-8.html#c.-save-your-complete-simulation-run-input-and-output"><i class="fa fa-check"></i>c. Save your complete simulation run (input and output)</a></li>
<li class="chapter" data-level="" data-path="WE-8.html"><a href="WE-8.html#d.-analyze-and-synthesize-results"><i class="fa fa-check"></i>d. Analyze and synthesize results</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="" data-path="r-exercise-week-8.html"><a href="r-exercise-week-8.html"><i class="fa fa-check"></i>R Exercise Week 8</a></li>
<li class="chapter" data-level="" data-path="bonus-8a.html"><a href="bonus-8a.html"><i class="fa fa-check"></i>Bonus: Efficient R</a></li>
</ul></li>
<li class="part"><span><b>Advanced Topics</b></span></li>
<li class="chapter" data-level="" data-path="Week9.html"><a href="Week9.html"><i class="fa fa-check"></i>Week 9: Population Structure</a>
<ul>
<li class="chapter" data-level="" data-path="WE-9.html"><a href="WE-9.html"><i class="fa fa-check"></i>Worked Example</a>
<ul>
<li class="chapter" data-level="" data-path="WE-9.html"><a href="WE-9.html#overview-of-worked-example-8"><i class="fa fa-check"></i>1. Overview of Worked Example</a>
<ul>
<li class="chapter" data-level="" data-path="WE-9.html"><a href="WE-9.html#a.-goals-8"><i class="fa fa-check"></i>a. Goals</a></li>
<li class="chapter" data-level="" data-path="WE-9.html"><a href="WE-9.html#b.-data-sets"><i class="fa fa-check"></i>b. Data sets</a></li>
<li class="chapter" data-level="" data-path="WE-9.html"><a href="WE-9.html#c.-required-r-libraries-5"><i class="fa fa-check"></i>c. Required R libraries</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="WE-9.html"><a href="WE-9.html#simulated-data-2-island-model"><i class="fa fa-check"></i>2. Simulated data: 2-island model</a>
<ul>
<li class="chapter" data-level="" data-path="WE-9.html"><a href="WE-9.html#a.-import-data"><i class="fa fa-check"></i>a. Import data</a></li>
<li class="chapter" data-level="" data-path="WE-9.html"><a href="WE-9.html#b.-perform-principal-components-analysis-pca"><i class="fa fa-check"></i>b. Perform Principal Components Analysis (PCA)</a></li>
<li class="chapter" data-level="" data-path="WE-9.html"><a href="WE-9.html#c.-create-synthetic-spatial-coordinates-xy-and-map-them"><i class="fa fa-check"></i>c. Create synthetic spatial coordinates (X,Y) and map them</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="WE-9.html"><a href="WE-9.html#simulated-data-2-island-model-with-admixture"><i class="fa fa-check"></i>3. Simulated data: 2-island model with admixture</a></li>
<li class="chapter" data-level="" data-path="WE-9.html"><a href="WE-9.html#empirical-data-threespine-sticklebacks"><i class="fa fa-check"></i>4. Empirical data: Threespine sticklebacks</a>
<ul>
<li class="chapter" data-level="" data-path="WE-9.html"><a href="WE-9.html#a.-import-the-data"><i class="fa fa-check"></i>a. Import the data:</a></li>
<li class="chapter" data-level="" data-path="WE-9.html"><a href="WE-9.html#b.-examine-population-structure-with-pca"><i class="fa fa-check"></i>b. Examine population structure with PCA</a></li>
<li class="chapter" data-level="" data-path="WE-9.html"><a href="WE-9.html#c.-clustering-with-snmf-similar-to-structure"><i class="fa fa-check"></i>c. Clustering with SNMF (similar to ‘STRUCTURE’)</a></li>
<li class="chapter" data-level="" data-path="WE-9.html"><a href="WE-9.html#d.-plot-ancestry-proportions"><i class="fa fa-check"></i>d. Plot ancestry proportions</a></li>
<li class="chapter" data-level="" data-path="WE-9.html"><a href="WE-9.html#e.-visualize-admixture-proportions-on-a-map"><i class="fa fa-check"></i>e. Visualize admixture proportions on a map</a></li>
</ul></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="" data-path="Week10.html"><a href="Week10.html"><i class="fa fa-check"></i>Week 10: Landscape Resistance</a>
<ul>
<li class="chapter" data-level="" data-path="WE-10.html"><a href="WE-10.html"><i class="fa fa-check"></i>Worked Example</a>
<ul>
<li class="chapter" data-level="" data-path="WE-10.html"><a href="WE-10.html#overview-of-worked-example-9"><i class="fa fa-check"></i>1. Overview of Worked Example</a>
<ul>
<li class="chapter" data-level="" data-path="WE-10.html"><a href="WE-10.html#a.-goals-9"><i class="fa fa-check"></i>a. Goals</a></li>
<li class="chapter" data-level="" data-path="WE-10.html"><a href="WE-10.html#b.-data-set-8"><i class="fa fa-check"></i>b. Data set</a></li>
<li class="chapter" data-level="" data-path="WE-10.html"><a href="WE-10.html#c.-required-r-libraries-6"><i class="fa fa-check"></i>c. Required R libraries</a></li>
<li class="chapter" data-level="" data-path="WE-10.html"><a href="WE-10.html#d.-import-rasters-gsp-err-cti-and-ffp."><i class="fa fa-check"></i>d. Import rasters (gsp, err, cti, and ffp).</a></li>
<li class="chapter" data-level="" data-path="WE-10.html"><a href="WE-10.html#e.-import-site-data-with-sampling-locations"><i class="fa fa-check"></i>e. Import site data with sampling locations</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="WE-10.html"><a href="WE-10.html#explore-the-data-set"><i class="fa fa-check"></i>2. Explore the data set</a>
<ul>
<li class="chapter" data-level="" data-path="WE-10.html"><a href="WE-10.html#a.-plot-all-rasters"><i class="fa fa-check"></i>a. Plot all rasters</a></li>
<li class="chapter" data-level="" data-path="WE-10.html"><a href="WE-10.html#b.-plot-spatial-points-over-ffp-raster"><i class="fa fa-check"></i>b. Plot spatial points over ffp raster</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="WE-10.html"><a href="WE-10.html#setting-cost-values-and-calculating-conductance"><i class="fa fa-check"></i>3. Setting cost values and calculating conductance</a>
<ul>
<li class="chapter" data-level="" data-path="WE-10.html"><a href="WE-10.html#a.-resistance-vs.-conductance-values"><i class="fa fa-check"></i>a. Resistance vs. conductance values</a></li>
<li class="chapter" data-level="" data-path="WE-10.html"><a href="WE-10.html#b.-as-needed-get-all-rasters-at-the-same-scale"><i class="fa fa-check"></i>b. As needed: Get all rasters at the same scale</a></li>
<li class="chapter" data-level="" data-path="WE-10.html"><a href="WE-10.html#c.-calculate-conductance-values"><i class="fa fa-check"></i>c. Calculate conductance values</a></li>
<li class="chapter" data-level="" data-path="WE-10.html"><a href="WE-10.html#d.-create-a-single-landscape-conductance-raster"><i class="fa fa-check"></i>d. Create a single landscape conductance raster</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="WE-10.html"><a href="WE-10.html#convert-conductance-into-effective-distance"><i class="fa fa-check"></i>4. Convert conductance into effective distance</a>
<ul>
<li class="chapter" data-level="" data-path="WE-10.html"><a href="WE-10.html#a.-create-a-transition-layer"><i class="fa fa-check"></i>a. Create a transition layer</a></li>
<li class="chapter" data-level="" data-path="WE-10.html"><a href="WE-10.html#b.-visually-inspect-the-raster"><i class="fa fa-check"></i>b. Visually inspect the raster</a></li>
<li class="chapter" data-level="" data-path="WE-10.html"><a href="WE-10.html#c.-correct-for-geometric-distortion"><i class="fa fa-check"></i>c. Correct for geometric distortion</a></li>
<li class="chapter" data-level="" data-path="WE-10.html"><a href="WE-10.html#d.-plot-shortest-paths-in-space"><i class="fa fa-check"></i>d. Plot shortest paths in space</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="WE-10.html"><a href="WE-10.html#create-cost-distance-matrices"><i class="fa fa-check"></i>5. Create cost-distance matrices</a>
<ul>
<li class="chapter" data-level="" data-path="WE-10.html"><a href="WE-10.html#a.-least-cost-distance"><i class="fa fa-check"></i>a. Least cost distance</a></li>
<li class="chapter" data-level="" data-path="WE-10.html"><a href="WE-10.html#b.-cost-distance-matrix-based-on-random-paths-similar-to-circuitscape"><i class="fa fa-check"></i>b. Cost-distance matrix based on random paths (similar to Circuitscape)</a></li>
<li class="chapter" data-level="" data-path="WE-10.html"><a href="WE-10.html#c.-compare-cost-distances"><i class="fa fa-check"></i>c. Compare cost distances</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="WE-10.html"><a href="WE-10.html#how-does-changing-resolution-affect-these-metrics"><i class="fa fa-check"></i>6. How does changing resolution affect these metrics?</a>
<ul>
<li class="chapter" data-level="" data-path="WE-10.html"><a href="WE-10.html#a.-create-loop"><i class="fa fa-check"></i>a. Create loop</a></li>
<li class="chapter" data-level="" data-path="WE-10.html"><a href="WE-10.html#b.-plot-the-results"><i class="fa fa-check"></i>b. Plot the results</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="WE-10.html"><a href="WE-10.html#references"><i class="fa fa-check"></i>7. References</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="" data-path="Week11.html"><a href="Week11.html"><i class="fa fa-check"></i>Week 11: Detecting Adaptation</a>
<ul>
<li class="chapter" data-level="" data-path="WE-11.html"><a href="WE-11.html"><i class="fa fa-check"></i>Worked Example</a>
<ul>
<li class="chapter" data-level="" data-path="WE-11.html"><a href="WE-11.html#overview-of-worked-example-10"><i class="fa fa-check"></i>1. Overview of Worked Example</a>
<ul>
<li class="chapter" data-level="" data-path="WE-11.html"><a href="WE-11.html#a.-goals-10"><i class="fa fa-check"></i>a. Goals</a></li>
<li class="chapter" data-level="" data-path="WE-11.html"><a href="WE-11.html#b.-data-set-9"><i class="fa fa-check"></i>b. Data set</a></li>
<li class="chapter" data-level="" data-path="WE-11.html"><a href="WE-11.html#c.-required-r-libraries-7"><i class="fa fa-check"></i>c. Required R libraries</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="WE-11.html"><a href="WE-11.html#import-and-prepare-the-data"><i class="fa fa-check"></i>2. Import and prepare the data</a>
<ul>
<li class="chapter" data-level="" data-path="WE-11.html"><a href="WE-11.html#a.-import-the-genetic-data"><i class="fa fa-check"></i>a. Import the genetic data</a></li>
<li class="chapter" data-level="" data-path="WE-11.html"><a href="WE-11.html#b.-import-the-environmental-data"><i class="fa fa-check"></i>b. Import the environmental data</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="WE-11.html"><a href="WE-11.html#latent-factor-mixed-models-lfmm-a-univariate-gea"><i class="fa fa-check"></i>3. Latent Factor Mixed Models (LFMM): a univariate GEA</a>
<ul>
<li class="chapter" data-level="" data-path="WE-11.html"><a href="WE-11.html#a.-determine-k"><i class="fa fa-check"></i>a. Determine K</a></li>
<li class="chapter" data-level="" data-path="WE-11.html"><a href="WE-11.html#b.-run-lfmm"><i class="fa fa-check"></i>b. Run LFMM</a></li>
<li class="chapter" data-level="" data-path="WE-11.html"><a href="WE-11.html#c.-identify-lfmm-candidates-using-false-discovery-rate"><i class="fa fa-check"></i>c. Identify LFMM candidates using False Discovery Rate</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="WE-11.html"><a href="WE-11.html#redundancy-analysis-rda-a-multivariate-gea"><i class="fa fa-check"></i>4. Redundancy Analysis (RDA): a multivariate GEA</a>
<ul>
<li class="chapter" data-level="" data-path="WE-11.html"><a href="WE-11.html#a.-run-rda"><i class="fa fa-check"></i>a. Run RDA</a></li>
<li class="chapter" data-level="" data-path="WE-11.html"><a href="WE-11.html#b.-identify-rda-candidates"><i class="fa fa-check"></i>b. Identify RDA candidates</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="WE-11.html"><a href="WE-11.html#compare-lfmm-and-rda-candidates"><i class="fa fa-check"></i>5. Compare LFMM and RDA candidates</a></li>
<li class="chapter" data-level="" data-path="WE-11.html"><a href="WE-11.html#a-quick-note-on-controlling-for-population-structure-in-rda"><i class="fa fa-check"></i>6. A quick note on controlling for population structure in RDA</a></li>
<li class="chapter" data-level="" data-path="WE-11.html"><a href="WE-11.html#references-1"><i class="fa fa-check"></i>7. References</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="" data-path="Week12.html"><a href="Week12.html"><i class="fa fa-check"></i>Week 12: Model Selection</a>
<ul>
<li class="chapter" data-level="" data-path="WE-12.html"><a href="WE-12.html"><i class="fa fa-check"></i>Worked Example</a>
<ul>
<li class="chapter" data-level="" data-path="WE-12.html"><a href="WE-12.html#overview-of-worked-example-11"><i class="fa fa-check"></i>1. Overview of Worked Example</a>
<ul>
<li class="chapter" data-level="" data-path="WE-12.html"><a href="WE-12.html#a.-goals-and-background"><i class="fa fa-check"></i>a. Goals and Background</a></li>
<li class="chapter" data-level="" data-path="WE-12.html"><a href="WE-12.html#b.-data-set-10"><i class="fa fa-check"></i>b. Data set</a></li>
<li class="chapter" data-level="" data-path="WE-12.html"><a href="WE-12.html#c.-required-packages"><i class="fa fa-check"></i>c. Required packages</a></li>
<li class="chapter" data-level="" data-path="WE-12.html"><a href="WE-12.html#d.-import-data-1"><i class="fa fa-check"></i>d. Import data</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="WE-12.html"><a href="WE-12.html#fitting-candidate-models"><i class="fa fa-check"></i>2. Fitting candidate models</a>
<ul>
<li class="chapter" data-level="" data-path="WE-12.html"><a href="WE-12.html#a.-alternative-hypotheses"><i class="fa fa-check"></i>a. Alternative hypotheses</a></li>
<li class="chapter" data-level="" data-path="WE-12.html"><a href="WE-12.html#b.-prepare-for-model-fitting"><i class="fa fa-check"></i>b. Prepare for model fitting</a></li>
<li class="chapter" data-level="" data-path="WE-12.html"><a href="WE-12.html#c.-fit-all-five-models"><i class="fa fa-check"></i>c. Fit all five models</a></li>
<li class="chapter" data-level="" data-path="WE-12.html"><a href="WE-12.html#d.-compare-evidence-for-models"><i class="fa fa-check"></i>d. Compare evidence for models</a></li>
<li class="chapter" data-level="" data-path="WE-12.html"><a href="WE-12.html#e.-calculate-evidence-weights"><i class="fa fa-check"></i>e. Calculate evidence weights</a></li>
<li class="chapter" data-level="" data-path="WE-12.html"><a href="WE-12.html#f.-confidence-intervals-for-predictors"><i class="fa fa-check"></i>f. Confidence intervals for predictors</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="WE-12.html"><a href="WE-12.html#your-turn-to-test-additional-hypotheses"><i class="fa fa-check"></i>3. Your turn to test additional hypotheses!</a></li>
<li class="chapter" data-level="" data-path="WE-12.html"><a href="WE-12.html#references-2"><i class="fa fa-check"></i>4. References</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="" data-path="Week13.html"><a href="Week13.html"><i class="fa fa-check"></i>Week 13: Gravity Models</a>
<ul>
<li class="chapter" data-level="" data-path="WE-13.html"><a href="WE-13.html"><i class="fa fa-check"></i>Worked Example</a>
<ul>
<li class="chapter" data-level="" data-path="WE-13.html"><a href="WE-13.html#overview-of-worked-example-12"><i class="fa fa-check"></i>1. Overview of Worked Example</a>
<ul>
<li class="chapter" data-level="" data-path="WE-13.html"><a href="WE-13.html#a.-background"><i class="fa fa-check"></i>a. Background</a></li>
<li class="chapter" data-level="" data-path="WE-13.html"><a href="WE-13.html#b.-data-set-11"><i class="fa fa-check"></i>b. Data set</a></li>
<li class="chapter" data-level="" data-path="WE-13.html"><a href="WE-13.html#c.-required-r-libraries-8"><i class="fa fa-check"></i>c. Required R libraries</a></li>
<li class="chapter" data-level="" data-path="WE-13.html"><a href="WE-13.html#set-working-and-data-directories---needs-to-be-removed"><i class="fa fa-check"></i>Set working and data directories - NEEDS TO BE REMOVED</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="WE-13.html"><a href="WE-13.html#calculating-graph-metrics"><i class="fa fa-check"></i>2. Calculating graph metrics</a>
<ul>
<li class="chapter" data-level="" data-path="WE-13.html"><a href="WE-13.html#a.-read-in-make-a-spatial-object-and-plot-wetlands-wetlands.csv."><i class="fa fa-check"></i>a. Read in, make a spatial object and plot wetlands (Wetlands.csv).</a></li>
<li class="chapter" data-level="" data-path="WE-13.html"><a href="WE-13.html#b.-create-a-graph-from-the-wetlands"><i class="fa fa-check"></i>b. Create a graph from the wetlands</a></li>
<li class="chapter" data-level="" data-path="WE-13.html"><a href="WE-13.html#c.-create-adjacency-matrix"><i class="fa fa-check"></i>c. Create adjacency matrix</a></li>
<li class="chapter" data-level="" data-path="WE-13.html"><a href="WE-13.html#d.-calculate-graph-metrics"><i class="fa fa-check"></i>d. Calculate graph metrics</a></li>
<li class="chapter" data-level="" data-path="WE-13.html"><a href="WE-13.html#e.-add-graph-metrics-to-data.frame"><i class="fa fa-check"></i>e. Add graph metrics to data.frame</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="WE-13.html"><a href="WE-13.html#combine-frog-data-with-graph-metrics"><i class="fa fa-check"></i>3. Combine frog data with graph metrics</a>
<ul>
<li class="chapter" data-level="" data-path="WE-13.html"><a href="WE-13.html#a.-add-graph-node-data-to-frog-site-data"><i class="fa fa-check"></i>a. Add graph node data to frog site data</a></li>
<li class="chapter" data-level="" data-path="WE-13.html"><a href="WE-13.html#b.-check-data-types"><i class="fa fa-check"></i>b. Check data types</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="WE-13.html"><a href="WE-13.html#merge-graph-with-genetic-distance-data"><i class="fa fa-check"></i>4. Merge graph with genetic distance data</a>
<ul>
<li class="chapter" data-level="" data-path="WE-13.html"><a href="WE-13.html#a.-build-graph-from-occupied-sites"><i class="fa fa-check"></i>a. Build graph from occupied sites</a></li>
<li class="chapter" data-level="" data-path="WE-13.html"><a href="WE-13.html#b.-merge-graph-with-genetic-distance"><i class="fa fa-check"></i>b. Merge graph with genetic distance</a></li>
<li class="chapter" data-level="" data-path="WE-13.html"><a href="WE-13.html#c.-write-out-results-ascii-csv-and-lines-shapefile."><i class="fa fa-check"></i>c. Write out results (ASCII csv and lines shapefile).</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="WE-13.html"><a href="WE-13.html#preparing-raster-based-covariates"><i class="fa fa-check"></i>5. Preparing raster-based covariates</a>
<ul>
<li class="chapter" data-level="" data-path="WE-13.html"><a href="WE-13.html#a.-import-raster-data-as-stack"><i class="fa fa-check"></i>a. Import raster data as stack</a></li>
<li class="chapter" data-level="" data-path="WE-13.html"><a href="WE-13.html#b.-calculate-wetland-area-within-buffer"><i class="fa fa-check"></i>b. Calculate wetland area within buffer</a></li>
<li class="chapter" data-level="" data-path="WE-13.html"><a href="WE-13.html#c.-create-raster-of-wetland-proportion"><i class="fa fa-check"></i>c. Create raster of wetland proportion</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="WE-13.html"><a href="WE-13.html#extract-raster-stats-for-nodes-and-edges"><i class="fa fa-check"></i>6. Extract raster stats for nodes and edges</a>
<ul>
<li class="chapter" data-level="" data-path="WE-13.html"><a href="WE-13.html#a.-extract-at-site-variables-from-rasters"><i class="fa fa-check"></i>a. Extract at-site variables from rasters</a></li>
<li class="chapter" data-level="" data-path="WE-13.html"><a href="WE-13.html#b.-add-covariates-to-graph-edges."><i class="fa fa-check"></i>b. Add covariates to graph edges.</a></li>
<li class="chapter" data-level="" data-path="WE-13.html"><a href="WE-13.html#c.-calculating-stats-along-edges"><i class="fa fa-check"></i>c. Calculating stats along edges</a></li>
<li class="chapter" data-level="" data-path="WE-13.html"><a href="WE-13.html#d.-calculating-stats-for-categorical-variables"><i class="fa fa-check"></i>d. Calculating stats for categorical variables</a></li>
<li class="chapter" data-level="" data-path="WE-13.html"><a href="WE-13.html#e.-add-node-data-to-graph"><i class="fa fa-check"></i>e. Add node data to graph</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="WE-13.html"><a href="WE-13.html#gravity-models"><i class="fa fa-check"></i>7. Gravity Models</a>
<ul>
<li class="chapter" data-level="" data-path="WE-13.html"><a href="WE-13.html#a.-merge-distance-and-site-data"><i class="fa fa-check"></i>a. Merge distance and site data</a></li>
<li class="chapter" data-level="" data-path="WE-13.html"><a href="WE-13.html#b.-define-a-set-of-models"><i class="fa fa-check"></i>b. Define a set of models</a></li>
<li class="chapter" data-level="" data-path="WE-13.html"><a href="WE-13.html#c.-check-for-correlations"><i class="fa fa-check"></i>c. Check for correlations</a></li>
<li class="chapter" data-level="" data-path="WE-13.html"><a href="WE-13.html#c.-write-out-resultss"><i class="fa fa-check"></i>c. Write out resultss</a></li>
<li class="chapter" data-level="" data-path="WE-13.html"><a href="WE-13.html#e.-compare-models"><i class="fa fa-check"></i>e. Compare models</a></li>
<li class="chapter" data-level="" data-path="WE-13.html"><a href="WE-13.html#f.-diagnostic-plots"><i class="fa fa-check"></i>f. Diagnostic plots</a></li>
<li class="chapter" data-level="" data-path="WE-13.html"><a href="WE-13.html#g.-fit-final-models"><i class="fa fa-check"></i>g. Fit final model(s)</a></li>
<li class="chapter" data-level="" data-path="WE-13.html"><a href="WE-13.html#h.-effect-size"><i class="fa fa-check"></i>h. Effect size</a></li>
</ul></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="" data-path="Week14.html"><a href="Week14.html"><i class="fa fa-check"></i>Week 14: Contemporary Gene Flow</a>
<ul>
<li class="chapter" data-level="" data-path="WE-14.html"><a href="WE-14.html"><i class="fa fa-check"></i>Worked Example</a>
<ul>
<li class="chapter" data-level="" data-path="WE-14.html"><a href="WE-14.html#overview-2"><i class="fa fa-check"></i>1. Overview</a>
<ul>
<li class="chapter" data-level="" data-path="WE-14.html"><a href="WE-14.html#a.-background-and-goals"><i class="fa fa-check"></i>a. Background and Goals</a></li>
<li class="chapter" data-level="" data-path="WE-14.html"><a href="WE-14.html#b.-data-set-12"><i class="fa fa-check"></i>b. Data set</a></li>
<li class="chapter" data-level="" data-path="WE-14.html"><a href="WE-14.html#c.-required-packages-1"><i class="fa fa-check"></i>c. Required packages</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="WE-14.html"><a href="WE-14.html#import-genotypes"><i class="fa fa-check"></i>2. Import Genotypes</a>
<ul>
<li class="chapter" data-level="" data-path="WE-14.html"><a href="WE-14.html#a.-import-genotype-data"><i class="fa fa-check"></i>a. Import genotype data</a></li>
<li class="chapter" data-level="" data-path="WE-14.html"><a href="WE-14.html#b.-plot-map-of-sampling-locations"><i class="fa fa-check"></i>b. Plot map of sampling locations</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="WE-14.html"><a href="WE-14.html#pollen-pool-genetic-structure"><i class="fa fa-check"></i>3. Pollen pool genetic structure</a>
<ul>
<li class="chapter" data-level="" data-path="WE-14.html"><a href="WE-14.html#a.-subtract-maternal-contribution-to-offspring-genotype"><i class="fa fa-check"></i>a. Subtract maternal contribution to offspring genotype</a></li>
<li class="chapter" data-level="" data-path="WE-14.html"><a href="WE-14.html#b.-twogener-analysis-amova-of-pollen-pool-data"><i class="fa fa-check"></i>b. ‘TwoGener’ analysis (AMOVA of pollen pool data)</a></li>
<li class="chapter" data-level="" data-path="WE-14.html"><a href="WE-14.html#c.-including-space-isolation-by-distance"><i class="fa fa-check"></i>c. Including space: Isolation by distance</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="WE-14.html"><a href="WE-14.html#paternity-analysis"><i class="fa fa-check"></i>4. Paternity analysis</a>
<ul>
<li class="chapter" data-level="" data-path="WE-14.html"><a href="WE-14.html#a.-exclusion-probabilities"><i class="fa fa-check"></i>a. Exclusion probabilities</a></li>
<li class="chapter" data-level="" data-path="WE-14.html"><a href="WE-14.html#b.-paternity-analysis"><i class="fa fa-check"></i>b. Paternity analysis</a></li>
<li class="chapter" data-level="" data-path="WE-14.html"><a href="WE-14.html#c.-visualize-reconstructed-pollen-flow-events-in-space"><i class="fa fa-check"></i>c. Visualize reconstructed pollen flow events in space</a></li>
<li class="chapter" data-level="" data-path="WE-14.html"><a href="WE-14.html#d.-visualize-dispersal-kernel"><i class="fa fa-check"></i>d. Visualize dispersal kernel</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="WE-14.html"><a href="WE-14.html#linking-paternity-to-ecological-variables"><i class="fa fa-check"></i>5. Linking paternity to ecological variables</a>
<ul>
<li class="chapter" data-level="" data-path="WE-14.html"><a href="WE-14.html#a.-explain-pollen-flow-within-populations"><i class="fa fa-check"></i>a. Explain pollen flow within populations</a></li>
<li class="chapter" data-level="" data-path="WE-14.html"><a href="WE-14.html#b.-explain-pollen-flow-at-the-population-level"><i class="fa fa-check"></i>b. Explain pollen flow at the population level</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="WE-14.html"><a href="WE-14.html#references-3"><i class="fa fa-check"></i>6. References</a></li>
</ul></li>
</ul></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Landscape Genetic Data Analysis with R</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="WE_13" class="section level2">
<h2>Worked Example</h2>
<div id="overview-of-worked-example-12" class="section level3">
<h3>1. Overview of Worked Example</h3>
<div id="a.-background" class="section level4">
<h4>a. Background</h4>
<p>There are many ways graphs can be implemented to understand population structure and relate that structure to landscape characteristics (see Dyer and Nason 2004). In this exercise, we will calculate various graph metrics and apply graphs to fit a gravity model.</p>
<p>Gravity models are a type of inferential model that exploit graph characteristics. Gravity models include both at-site and between-site landscape data. They are a type of graph consisting of nodes and edges. These nodes and edges of landscape characteristics associated with these graph elements.</p>
</div>
<div id="b.-data-set-11" class="section level4">
<h4>b. Data set</h4>
<p>In this exercise, you will use the gravity model framework to build an empirical model of gene flow for the Columbia spotted frog dataset in central Idaho that you have used for several other exercises (Murphy et al. 2010).</p>
</div>
<div id="c.-required-r-libraries-8" class="section level4">
<h4>c. Required R libraries</h4>
<p>Install some packages needed for this worked example.</p>
<div class="sourceCode" id="cb1032"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1032-1"><a href="WE-13.html#cb1032-1" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span>(<span class="sc">!</span><span class="fu">requireNamespace</span>(<span class="st">&quot;GeNetIt&quot;</span>, <span class="at">quietly =</span> <span class="cn">TRUE</span>)) remotes<span class="sc">::</span><span class="fu">install_github</span>(<span class="st">&quot;jeffreyevans/GeNetIt&quot;</span>)</span>
<span id="cb1032-2"><a href="WE-13.html#cb1032-2" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span>(<span class="sc">!</span><span class="fu">requireNamespace</span>(<span class="st">&quot;spatialEco&quot;</span>, <span class="at">quietly =</span> <span class="cn">TRUE</span>)) remotes<span class="sc">::</span><span class="fu">install_github</span>(<span class="st">&quot;jeffreyevans/spatialEco&quot;</span>)</span></code></pre></div>
<div class="sourceCode" id="cb1033"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1033-1"><a href="WE-13.html#cb1033-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(LandGenCourse)</span>
<span id="cb1033-2"><a href="WE-13.html#cb1033-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(sp)</span>
<span id="cb1033-3"><a href="WE-13.html#cb1033-3" aria-hidden="true" tabindex="-1"></a><span class="co">#library(landscapemetrics)</span></span>
<span id="cb1033-4"><a href="WE-13.html#cb1033-4" aria-hidden="true" tabindex="-1"></a><span class="co">#library(raster)</span></span>
<span id="cb1033-5"><a href="WE-13.html#cb1033-5" aria-hidden="true" tabindex="-1"></a><span class="co">#library(rgdal)</span></span>
<span id="cb1033-6"><a href="WE-13.html#cb1033-6" aria-hidden="true" tabindex="-1"></a><span class="co">#library(GeNetIt)</span></span>
<span id="cb1033-7"><a href="WE-13.html#cb1033-7" aria-hidden="true" tabindex="-1"></a><span class="co">#library(spatialEco)</span></span>
<span id="cb1033-8"><a href="WE-13.html#cb1033-8" aria-hidden="true" tabindex="-1"></a><span class="co">#library(GeNetIt)</span></span>
<span id="cb1033-9"><a href="WE-13.html#cb1033-9" aria-hidden="true" tabindex="-1"></a><span class="co">#library(igraph)</span></span>
<span id="cb1033-10"><a href="WE-13.html#cb1033-10" aria-hidden="true" tabindex="-1"></a><span class="co">#library(deldir)</span></span></code></pre></div>
</div>
<div id="set-working-and-data-directories---needs-to-be-removed" class="section level4">
<h4>Set working and data directories - NEEDS TO BE REMOVED</h4>
<p>#wd = “C:/R/spatialR/day3/session6” # set your working directory here
#setwd(wd)
#ddir = file.path(“C:/R/spatialR/day3,” “data,” “S6”)</p>
</div>
</div>
<div id="calculating-graph-metrics" class="section level3">
<h3>2. Calculating graph metrics</h3>
<div id="a.-read-in-make-a-spatial-object-and-plot-wetlands-wetlands.csv." class="section level4">
<h4>a. Read in, make a spatial object and plot wetlands (Wetlands.csv).</h4>
<p>Read in the wetland csv file.</p>
<div class="sourceCode" id="cb1034"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1034-1"><a href="WE-13.html#cb1034-1" aria-hidden="true" tabindex="-1"></a>wetlands <span class="ot">&lt;-</span> <span class="fu">read.csv</span>(<span class="fu">system.file</span>(<span class="st">&quot;extdata&quot;</span>, <span class="st">&quot;Wetlands.csv&quot;</span>, </span>
<span id="cb1034-2"><a href="WE-13.html#cb1034-2" aria-hidden="true" tabindex="-1"></a>                            <span class="at">package =</span> <span class="st">&quot;LandGenCourse&quot;</span>), <span class="at">header =</span> <span class="cn">TRUE</span>) </span>
<span id="cb1034-3"><a href="WE-13.html#cb1034-3" aria-hidden="true" tabindex="-1"></a><span class="fu">str</span>(wetlands)</span></code></pre></div>
<pre><code>## &#39;data.frame&#39;:    121 obs. of  5 variables:
##  $ ID      : int  1 2 3 4 5 6 7 8 9 10 ...
##  $ X       : int  688835 687460 687507 687637 688850 688500 687944 687872 687150 690888 ...
##  $ Y       : int  5002939 4994400 4994314 4994117 4997750 4998900 5000006 5000041 4995850 5004126 ...
##  $ RALU    : chr  &quot;y&quot; &quot;n&quot; &quot;n&quot; &quot;n&quot; ...
##  $ SiteName: chr  &quot;AirplaneLake&quot; &quot;AlpineInletCreek&quot; &quot;AlpineInletMeadow&quot; &quot;AlpineLake&quot; ...</code></pre>
<p>Coerce to a SpatialPointsDataFrame and look at the structure of the related dataframe.</p>
<div class="sourceCode" id="cb1036"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1036-1"><a href="WE-13.html#cb1036-1" aria-hidden="true" tabindex="-1"></a>sp<span class="sc">::</span><span class="fu">coordinates</span>(wetlands) <span class="ot">&lt;-</span> <span class="er">~</span>X<span class="sc">+</span>Y</span>
<span id="cb1036-2"><a href="WE-13.html#cb1036-2" aria-hidden="true" tabindex="-1"></a><span class="fu">class</span>(wetlands)</span></code></pre></div>
<pre><code>## [1] &quot;SpatialPointsDataFrame&quot;
## attr(,&quot;package&quot;)
## [1] &quot;sp&quot;</code></pre>
<div class="sourceCode" id="cb1038"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1038-1"><a href="WE-13.html#cb1038-1" aria-hidden="true" tabindex="-1"></a><span class="fu">str</span>(wetlands)</span></code></pre></div>
<pre><code>## Formal class &#39;SpatialPointsDataFrame&#39; [package &quot;sp&quot;] with 5 slots
##   ..@ data       :&#39;data.frame&#39;:  121 obs. of  3 variables:
##   .. ..$ ID      : int [1:121] 1 2 3 4 5 6 7 8 9 10 ...
##   .. ..$ RALU    : chr [1:121] &quot;y&quot; &quot;n&quot; &quot;n&quot; &quot;n&quot; ...
##   .. ..$ SiteName: chr [1:121] &quot;AirplaneLake&quot; &quot;AlpineInletCreek&quot; &quot;AlpineInletMeadow&quot; &quot;AlpineLake&quot; ...
##   ..@ coords.nrs : int [1:2] 2 3
##   ..@ coords     : num [1:121, 1:2] 688835 687460 687507 687637 688850 ...
##   .. ..- attr(*, &quot;dimnames&quot;)=List of 2
##   .. .. ..$ : chr [1:121] &quot;1&quot; &quot;2&quot; &quot;3&quot; &quot;4&quot; ...
##   .. .. ..$ : chr [1:2] &quot;X&quot; &quot;Y&quot;
##   ..@ bbox       : num [1:2, 1:2] 686244 4993077 695699 5004317
##   .. ..- attr(*, &quot;dimnames&quot;)=List of 2
##   .. .. ..$ : chr [1:2] &quot;X&quot; &quot;Y&quot;
##   .. .. ..$ : chr [1:2] &quot;min&quot; &quot;max&quot;
##   ..@ proj4string:Formal class &#39;CRS&#39; [package &quot;sp&quot;] with 1 slot
##   .. .. ..@ projargs: chr NA</code></pre>
<p>Plot the wetlands</p>
<div class="sourceCode" id="cb1040"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1040-1"><a href="WE-13.html#cb1040-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(wetlands, <span class="at">asp=</span><span class="dv">1</span>, <span class="at">bty=</span><span class="st">&quot;n&quot;</span>, <span class="at">xlab=</span><span class="st">&quot;&quot;</span>, <span class="at">ylab=</span><span class="st">&quot;&quot;</span>, <span class="at">main =</span> <span class="st">&quot;All Wetlands&quot;</span>)</span>
<span id="cb1040-2"><a href="WE-13.html#cb1040-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">points</span>(wetlands, <span class="at">pch=</span><span class="dv">19</span>, <span class="at">cex=</span><span class="fl">0.75</span>, <span class="at">col=</span><span class="st">&quot;blue&quot;</span>)</span></code></pre></div>
<p><img src="13-Week13_files/figure-html/unnamed-chunk-5-1.png" width="672" /></p>
</div>
<div id="b.-create-a-graph-from-the-wetlands" class="section level4">
<h4>b. Create a graph from the wetlands</h4>
<p>We’ll create a graph that could represent connectivity using Delaunay triangulation, then plot the result.</p>
<p>Delaunay triangulation, create delaunay graph</p>
<div class="sourceCode" id="cb1041"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1041-1"><a href="WE-13.html#cb1041-1" aria-hidden="true" tabindex="-1"></a><span class="fu">options</span>(<span class="at">warn=</span><span class="sc">-</span><span class="dv">1</span>)</span>
<span id="cb1041-2"><a href="WE-13.html#cb1041-2" aria-hidden="true" tabindex="-1"></a>wetlandgraph <span class="ot">&lt;-</span> deldir<span class="sc">::</span><span class="fu">deldir</span>(<span class="fu">coordinates</span>(wetlands)[,<span class="dv">1</span>], </span>
<span id="cb1041-3"><a href="WE-13.html#cb1041-3" aria-hidden="true" tabindex="-1"></a>                       <span class="fu">coordinates</span>(wetlands)[,<span class="dv">2</span>], </span>
<span id="cb1041-4"><a href="WE-13.html#cb1041-4" aria-hidden="true" tabindex="-1"></a>                       <span class="at">z =</span> wetlands<span class="sc">$</span>SiteName) </span>
<span id="cb1041-5"><a href="WE-13.html#cb1041-5" aria-hidden="true" tabindex="-1"></a><span class="fu">options</span>(<span class="at">warn=</span><span class="dv">0</span>)</span></code></pre></div>
<p>Plot the graph with the wetlands</p>
<div class="sourceCode" id="cb1042"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1042-1"><a href="WE-13.html#cb1042-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(wetlands, <span class="at">asp=</span><span class="dv">1</span>, <span class="at">bty=</span><span class="st">&quot;n&quot;</span>, <span class="at">xlab=</span><span class="st">&quot;&quot;</span>, <span class="at">ylab=</span><span class="st">&quot;&quot;</span>, <span class="at">main =</span> <span class="st">&quot;All Wetlands&quot;</span>)</span>
<span id="cb1042-2"><a href="WE-13.html#cb1042-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">points</span>(wetlands, <span class="at">pch=</span><span class="dv">19</span>, <span class="at">cex=</span><span class="fl">0.75</span>, <span class="at">col=</span><span class="st">&quot;blue&quot;</span>)</span>
<span id="cb1042-3"><a href="WE-13.html#cb1042-3" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(wetlandgraph, <span class="at">wlines =</span> <span class="st">&quot;triang&quot;</span>, <span class="at">wpoints=</span><span class="st">&quot;none&quot;</span>,</span>
<span id="cb1042-4"><a href="WE-13.html#cb1042-4" aria-hidden="true" tabindex="-1"></a>     <span class="at">number=</span><span class="cn">FALSE</span>, <span class="at">add=</span><span class="cn">TRUE</span>, <span class="at">lty=</span><span class="dv">1</span>) </span></code></pre></div>
<p><img src="13-Week13_files/figure-html/unnamed-chunk-7-1.png" width="672" /></p>
<p><strong>Questions:</strong></p>
<ul>
<li>What other types of graphs could you build?<br />
</li>
<li>Can you build a different graph and use this graph in your analysis?</li>
</ul>
</div>
<div id="c.-create-adjacency-matrix" class="section level4">
<h4>c. Create adjacency matrix</h4>
<p>What wetlands are connected to each other based on the Delaunay triangulation? Create an adjacency matrix from the graph object and make an adjacency matrix.</p>
<div class="sourceCode" id="cb1043"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1043-1"><a href="WE-13.html#cb1043-1" aria-hidden="true" tabindex="-1"></a>ind <span class="ot">&lt;-</span> wetlandgraph<span class="sc">$</span>delsgs[,<span class="dv">5</span><span class="sc">:</span><span class="dv">6</span>] <span class="co">#pull out individual nodes</span></span>
<span id="cb1043-2"><a href="WE-13.html#cb1043-2" aria-hidden="true" tabindex="-1"></a>adj <span class="ot">&lt;-</span> <span class="fu">matrix</span>(<span class="dv">0</span>, <span class="fu">length</span>(wetlands<span class="sc">$</span>X), <span class="fu">length</span>(wetlands<span class="sc">$</span>Y)) </span>
<span id="cb1043-3"><a href="WE-13.html#cb1043-3" aria-hidden="true" tabindex="-1"></a>  <span class="cf">for</span> (i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">nrow</span>(ind)){ </span>
<span id="cb1043-4"><a href="WE-13.html#cb1043-4" aria-hidden="true" tabindex="-1"></a>    adj[ind[i,<span class="dv">1</span>], ind[i,<span class="dv">2</span>]] <span class="ot">&lt;-</span> <span class="dv">1</span> </span>
<span id="cb1043-5"><a href="WE-13.html#cb1043-5" aria-hidden="true" tabindex="-1"></a>    adj[ind[i,<span class="dv">2</span>], ind[i,<span class="dv">1</span>]] <span class="ot">&lt;-</span> <span class="dv">1</span> </span>
<span id="cb1043-6"><a href="WE-13.html#cb1043-6" aria-hidden="true" tabindex="-1"></a>  } </span></code></pre></div>
</div>
<div id="d.-calculate-graph-metrics" class="section level4">
<h4>d. Calculate graph metrics</h4>
<p>This graph, if it is ecologically meaningful contains information in the graph structure itself. Calculate graph metrics of degree and betweenness.</p>
<p><strong>Questions:</strong></p>
<ul>
<li>In what way(s) is the resulting graph potentially ecologically meaningful?<br />
</li>
<li>How might it not be ecologically or biologically meaningful?</li>
</ul>
<p>Make an igraph network from the matrix we just made</p>
<div class="sourceCode" id="cb1044"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1044-1"><a href="WE-13.html#cb1044-1" aria-hidden="true" tabindex="-1"></a>wetnet <span class="ot">&lt;-</span> igraph<span class="sc">::</span><span class="fu">graph_from_adjacency_matrix</span>(adj, <span class="at">weighted =</span> <span class="cn">NULL</span>, <span class="at">mode=</span><span class="st">&quot;undirected&quot;</span>) </span>
<span id="cb1044-2"><a href="WE-13.html#cb1044-2" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(wetnet)</span></code></pre></div>
<p><img src="13-Week13_files/figure-html/unnamed-chunk-9-1.png" width="672" /></p>
</div>
<div id="e.-add-graph-metrics-to-data.frame" class="section level4">
<h4>e. Add graph metrics to data.frame</h4>
<p>Calculate degree- the number of connections a node has</p>
<div class="sourceCode" id="cb1045"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1045-1"><a href="WE-13.html#cb1045-1" aria-hidden="true" tabindex="-1"></a>wetlands<span class="sc">@</span>data<span class="sc">$</span>degree <span class="ot">&lt;-</span> igraph<span class="sc">::</span><span class="fu">degree</span>(wetnet)</span>
<span id="cb1045-2"><a href="WE-13.html#cb1045-2" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(wetlands<span class="sc">@</span>data)</span></code></pre></div>
<pre><code>##   ID RALU          SiteName degree
## 1  1    y      AirplaneLake      8
## 2  2    n  AlpineInletCreek      6
## 3  3    n AlpineInletMeadow      4
## 4  4    n        AlpineLake      6
## 5  5    n   AxeHandleMeadow      9
## 6  6    y    BachelorMeadow      7</code></pre>
<p>Calculate betweenness- the number of shortest paths going through a node</p>
<div class="sourceCode" id="cb1047"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1047-1"><a href="WE-13.html#cb1047-1" aria-hidden="true" tabindex="-1"></a>wetlands<span class="sc">@</span>data<span class="sc">$</span>betweenness <span class="ot">&lt;-</span> igraph<span class="sc">::</span><span class="fu">betweenness</span>(wetnet) </span>
<span id="cb1047-2"><a href="WE-13.html#cb1047-2" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(wetlands<span class="sc">@</span>data)</span></code></pre></div>
<pre><code>##   ID RALU          SiteName degree betweenness
## 1  1    y      AirplaneLake      8   279.58405
## 2  2    n  AlpineInletCreek      6    87.97683
## 3  3    n AlpineInletMeadow      4    16.71910
## 4  4    n        AlpineLake      6   117.72297
## 5  5    n   AxeHandleMeadow      9   682.84635
## 6  6    y    BachelorMeadow      7   292.32533</code></pre>
<p><strong>Challenge:</strong></p>
<ul>
<li>Can you code your own degree or betweenness centrality function?</li>
</ul>
</div>
</div>
<div id="combine-frog-data-with-graph-metrics" class="section level3">
<h3>3. Combine frog data with graph metrics</h3>
<div id="a.-add-graph-node-data-to-frog-site-data" class="section level4">
<h4>a. Add graph node data to frog site data</h4>
<p>Using the RALU_Site data, read in the data, add the node data (betweenness and degree) and create a shape file that includes the node data. This process will mirror part 1.</p>
<p><strong>Questions:</strong> Look at the RALU_Site file.</p>
<ul>
<li>What are the fields here?<br />
</li>
<li>What data are included?</li>
</ul>
<p>Import site data</p>
<div class="sourceCode" id="cb1049"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1049-1"><a href="WE-13.html#cb1049-1" aria-hidden="true" tabindex="-1"></a>sites <span class="ot">&lt;-</span> <span class="fu">read.csv</span>(<span class="fu">system.file</span>(<span class="st">&quot;extdata&quot;</span>, <span class="st">&quot;RALU_Site.csv&quot;</span>, </span>
<span id="cb1049-2"><a href="WE-13.html#cb1049-2" aria-hidden="true" tabindex="-1"></a>                            <span class="at">package =</span> <span class="st">&quot;LandGenCourse&quot;</span>), <span class="at">header =</span> <span class="cn">TRUE</span>) </span>
<span id="cb1049-3"><a href="WE-13.html#cb1049-3" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(sites)</span></code></pre></div>
<pre><code>##   SiteID     Elev Length    Area Perim Depth  pH Dforest Drock Dshrub
## 1      1 2473.719    101  6856.1 335.7  5.70 7.2   0.000 0.900  0.100
## 2      2 2609.068    350 36915.0 916.7 12.70 6.7   0.131 0.036  0.000
## 3      3 2622.113      0  1000.0 120.0  0.25 6.1   0.200 0.000  0.000
## 4      4 2474.737    336 38912.8 921.7 16.97 7.0   0.374 0.396  0.043
## 5      5 2589.805    107  6089.4 300.5  3.89 7.1   0.647 0.176  0.000
## 6      6 2463.088     68  3444.6 220.0  2.00 7.0   0.359 0.017  0.000
##         SiteName      X       Y
## 1      Walkabout 688020 4993875
## 2    ParagonLake 687427 4994716
## 3 ParagonWetland 687170 4994770
## 4   RamshornLake 687914 4994972
## 5  Mt.WilsonLake 687645 4995325
## 6      MooseLake 688549 4995538</code></pre>
<p>Extract degree and betweenness from graph data</p>
<div class="sourceCode" id="cb1051"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1051-1"><a href="WE-13.html#cb1051-1" aria-hidden="true" tabindex="-1"></a>nodestats <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(wetlands<span class="sc">@</span>data[,<span class="dv">3</span><span class="sc">:</span><span class="dv">5</span>])</span>
<span id="cb1051-2"><a href="WE-13.html#cb1051-2" aria-hidden="true" tabindex="-1"></a>degree.betweenness <span class="ot">&lt;-</span> nodestats[<span class="fu">which</span>(nodestats<span class="sc">$</span>SiteName <span class="sc">%in%</span> sites<span class="sc">$</span>SiteName),]</span>
<span id="cb1051-3"><a href="WE-13.html#cb1051-3" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(degree.betweenness)</span></code></pre></div>
<pre><code>##          SiteName degree betweenness
## 1    AirplaneLake      8   279.58405
## 6  BachelorMeadow      7   292.32533
## 7  BarkingFoxLake      9   512.37876
## 12        BobLake      6   466.35631
## 16      CacheLake      4    11.57913
## 25        DoeLake      5    31.48064</code></pre>
<p>Add to site data</p>
<div class="sourceCode" id="cb1053"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1053-1"><a href="WE-13.html#cb1053-1" aria-hidden="true" tabindex="-1"></a>sites <span class="ot">&lt;-</span> <span class="fu">merge</span>(degree.betweenness, sites, <span class="at">by=</span> <span class="st">&quot;SiteName&quot;</span> )</span>
<span id="cb1053-2"><a href="WE-13.html#cb1053-2" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(sites)</span></code></pre></div>
<pre><code>##         SiteName degree betweenness SiteID     Elev Length    Area  Perim Depth
## 1   AirplaneLake      8   279.58405     27 2564.381    390 62582.2 1142.8 21.64
## 2 BachelorMeadow      7   292.32533     15 2591.781      0   225.0   60.0  0.40
## 3 BarkingFoxLake      9   512.37876     19 2545.275    160 12000.0  435.0  5.00
## 4        BobLake      6   466.35631     16 2649.125    143  4600.0  321.4  2.00
## 5      CacheLake      4    11.57913     10 2475.829     75  2268.8  192.0  1.86
## 6        DoeLake      5    31.48064      8 2463.006    170 13034.9  463.2  6.03
##    pH Dforest Drock Dshrub      X       Y
## 1 6.5   0.398 0.051   0.00 688835 5002939
## 2 6.1   0.000 0.000   0.20 688500 4998900
## 3 6.5   0.400 0.250   0.05 687944 5000006
## 4 7.0   0.550 0.000   0.05 690127 4999150
## 5 6.5   0.508 0.000   0.00 688777 4997264
## 6 7.6   0.254 0.000   0.00 688968 4996458</code></pre>
<p>Note: Using names is dangerous as, small changes in names can result in non-matches. In this case, the ID fields are not consistent (data were collected at different times for different purposes originally). However, names are standardized in a drop-down list of a database. So they are a matching field. My preference is do to this type of operation on a numeric field.</p>
<div class="sourceCode" id="cb1055"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1055-1"><a href="WE-13.html#cb1055-1" aria-hidden="true" tabindex="-1"></a><span class="fu">coordinates</span>(sites) <span class="ot">&lt;-</span> <span class="er">~</span>X<span class="sc">+</span>Y</span>
<span id="cb1055-2"><a href="WE-13.html#cb1055-2" aria-hidden="true" tabindex="-1"></a><span class="fu">str</span>(sites)</span></code></pre></div>
<pre><code>## Formal class &#39;SpatialPointsDataFrame&#39; [package &quot;sp&quot;] with 5 slots
##   ..@ data       :&#39;data.frame&#39;:  29 obs. of  13 variables:
##   .. ..$ SiteName   : chr [1:29] &quot;AirplaneLake&quot; &quot;BachelorMeadow&quot; &quot;BarkingFoxLake&quot; &quot;BobLake&quot; ...
##   .. ..$ degree     : num [1:29] 8 7 9 6 4 5 6 6 5 3 ...
##   .. ..$ betweenness: num [1:29] 279.6 292.3 512.4 466.4 11.6 ...
##   .. ..$ SiteID     : int [1:29] 27 15 19 16 10 8 11 31 9 12 ...
##   .. ..$ Elev       : num [1:29] 2564 2592 2545 2649 2476 ...
##   .. ..$ Length     : int [1:29] 390 0 160 143 75 170 100 0 91 65 ...
##   .. ..$ Area       : num [1:29] 62582 225 12000 4600 2269 ...
##   .. ..$ Perim      : num [1:29] 1143 60 435 321 192 ...
##   .. ..$ Depth      : num [1:29] 21.64 0.4 5 2 1.86 ...
##   .. ..$ pH         : num [1:29] 6.5 6.1 6.5 7 6.5 7.6 6.8 0 6.7 6.9 ...
##   .. ..$ Dforest    : num [1:29] 0.398 0 0.4 0.55 0.508 ...
##   .. ..$ Drock      : num [1:29] 0.051 0 0.25 0 0 0 0.163 0 0 0 ...
##   .. ..$ Dshrub     : num [1:29] 0 0.2 0.05 0.05 0 0 0 0 0 0 ...
##   ..@ coords.nrs : int [1:2] 14 15
##   ..@ coords     : num [1:29, 1:2] 688835 688500 687944 690127 688777 ...
##   .. ..- attr(*, &quot;dimnames&quot;)=List of 2
##   .. .. ..$ : chr [1:29] &quot;1&quot; &quot;2&quot; &quot;3&quot; &quot;4&quot; ...
##   .. .. ..$ : chr [1:2] &quot;X&quot; &quot;Y&quot;
##   ..@ bbox       : num [1:2, 1:2] 686850 4993875 690900 5004222
##   .. ..- attr(*, &quot;dimnames&quot;)=List of 2
##   .. .. ..$ : chr [1:2] &quot;X&quot; &quot;Y&quot;
##   .. .. ..$ : chr [1:2] &quot;min&quot; &quot;max&quot;
##   ..@ proj4string:Formal class &#39;CRS&#39; [package &quot;sp&quot;] with 1 slot
##   .. .. ..@ projargs: chr NA</code></pre>
</div>
<div id="b.-check-data-types" class="section level4">
<h4>b. Check data types</h4>
<p>Get data as proper type. Are all of the data fields reading as the correct type?</p>
<div class="sourceCode" id="cb1057"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1057-1"><a href="WE-13.html#cb1057-1" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>(sites<span class="sc">@</span>data)</span></code></pre></div>
<pre><code>##    SiteName             degree       betweenness         SiteID     
##  Length:29          Min.   :3.000   Min.   :  0.00   Min.   : 1.00  
##  Class :character   1st Qu.:4.000   1st Qu.: 11.58   1st Qu.: 8.00  
##  Mode  :character   Median :5.000   Median : 47.29   Median :15.00  
##                     Mean   :5.448   Mean   :149.86   Mean   :15.76  
##                     3rd Qu.:6.000   3rd Qu.:274.32   3rd Qu.:24.00  
##                     Max.   :9.000   Max.   :512.38   Max.   :31.00  
##       Elev          Length            Area            Perim       
##  Min.   :2390   Min.   :   0.0   Min.   :     0   Min.   :   0.0  
##  1st Qu.:2476   1st Qu.:  91.0   1st Qu.:  3526   1st Qu.: 237.7  
##  Median :2572   Median : 130.0   Median :  6556   Median : 335.7  
##  Mean   :2563   Mean   : 210.9   Mean   : 24504   Mean   : 564.5  
##  3rd Qu.:2633   3rd Qu.: 235.0   3rd Qu.: 16601   3rd Qu.: 520.2  
##  Max.   :2794   Max.   :1660.0   Max.   :353898   Max.   :4312.9  
##      Depth              pH           Dforest           Drock       
##  Min.   : 0.000   Min.   :0.000   Min.   :0.0000   Min.   :0.0000  
##  1st Qu.: 2.000   1st Qu.:6.500   1st Qu.:0.0000   1st Qu.:0.0000  
##  Median : 3.570   Median :6.900   Median :0.3590   Median :0.0470  
##  Mean   : 6.161   Mean   :6.372   Mean   :0.4785   Mean   :0.1718  
##  3rd Qu.: 6.000   3rd Qu.:7.000   3rd Qu.:0.5500   3rd Qu.:0.2130  
##  Max.   :24.300   Max.   :7.600   Max.   :4.8450   Max.   :0.9530  
##      Dshrub       
##  Min.   :0.00000  
##  1st Qu.:0.00000  
##  Median :0.00000  
##  Mean   :0.01859  
##  3rd Qu.:0.00000  
##  Max.   :0.20000</code></pre>
<div class="sourceCode" id="cb1059"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1059-1"><a href="WE-13.html#cb1059-1" aria-hidden="true" tabindex="-1"></a>sites<span class="sc">@</span>data<span class="sc">$</span>SiteName <span class="ot">&lt;-</span> <span class="fu">as.character</span>(sites<span class="sc">@</span>data<span class="sc">$</span>SiteName)</span>
<span id="cb1059-2"><a href="WE-13.html#cb1059-2" aria-hidden="true" tabindex="-1"></a><span class="fu">class</span>(sites<span class="sc">@</span>data<span class="sc">$</span>SiteName)</span></code></pre></div>
<pre><code>## [1] &quot;character&quot;</code></pre>
<div class="sourceCode" id="cb1061"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1061-1"><a href="WE-13.html#cb1061-1" aria-hidden="true" tabindex="-1"></a>sites<span class="sc">@</span>data<span class="sc">$</span>SiteID <span class="ot">&lt;-</span> <span class="fu">as.factor</span>(sites<span class="sc">@</span>data<span class="sc">$</span>SiteID)</span>
<span id="cb1061-2"><a href="WE-13.html#cb1061-2" aria-hidden="true" tabindex="-1"></a><span class="fu">class</span>(sites<span class="sc">@</span>data<span class="sc">$</span>SiteID)</span></code></pre></div>
<pre><code>## [1] &quot;factor&quot;</code></pre>
</div>
</div>
<div id="merge-graph-with-genetic-distance-data" class="section level3">
<h3>4. Merge graph with genetic distance data</h3>
<p>Create graph from site locations and merge with genetic distance data.</p>
<div id="a.-build-graph-from-occupied-sites" class="section level4">
<h4>a. Build graph from occupied sites</h4>
<p>To assess connectivity using a gravity model, we need to build a graph from the occupied frog sites. This could be any type of graph, but I generally use saturated or pruned by some maximum distance.</p>
<p>Note: make sure to use correct field here:</p>
<div class="sourceCode" id="cb1063"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1063-1"><a href="WE-13.html#cb1063-1" aria-hidden="true" tabindex="-1"></a>dist.graph <span class="ot">&lt;-</span> GeNetIt<span class="sc">::</span><span class="fu">knn.graph</span>(sites, <span class="at">row.names =</span> sites<span class="sc">@</span>data[,<span class="st">&quot;SiteID&quot;</span>])</span>
<span id="cb1063-2"><a href="WE-13.html#cb1063-2" aria-hidden="true" tabindex="-1"></a><span class="co">#dist.graph@proj4string@projargs &lt;- &quot;+proj=utm +zone=11 +ellps=GRS80 +towgs84=0,0,0,-0,-0,-0,0 +units=m +no_defs &quot;</span></span>
<span id="cb1063-3"><a href="WE-13.html#cb1063-3" aria-hidden="true" tabindex="-1"></a><span class="co">#dist.graph &lt;- GeNetIt::knn.graph(sites, row.names = sites@data[,&quot;SiteName&quot;], max.dist=5000)</span></span></code></pre></div>
<p>The “k greater than one-third of the number of data points” is a warning issued by the k nearest neighbor function. In our case, this is not a problem as we want a graph that k is equal to 1-n, that is a saturated graph.</p>
</div>
<div id="b.-merge-graph-with-genetic-distance" class="section level4">
<h4>b. Merge graph with genetic distance</h4>
<p>This involves: reading in RALU_Dps genetic distance. Read in genetic distance file (RALU_Dps), convert to flow (1-distance) and unfold into a dataframe. Then merge graph with genetic distances.</p>
<p>Read in the genetic distance matrix</p>
<div class="sourceCode" id="cb1064"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1064-1"><a href="WE-13.html#cb1064-1" aria-hidden="true" tabindex="-1"></a>gdist <span class="ot">&lt;-</span> <span class="fu">read.csv</span>(<span class="fu">system.file</span>(<span class="st">&quot;extdata&quot;</span>, <span class="st">&quot;RALU_Dps.csv&quot;</span>, </span>
<span id="cb1064-2"><a href="WE-13.html#cb1064-2" aria-hidden="true" tabindex="-1"></a>                            <span class="at">package =</span> <span class="st">&quot;LandGenCourse&quot;</span>), <span class="at">header=</span><span class="cn">TRUE</span>)</span>
<span id="cb1064-3"><a href="WE-13.html#cb1064-3" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(gdist) <span class="ot">&lt;-</span> <span class="fu">t</span>(<span class="fu">names</span>(gdist))</span>
<span id="cb1064-4"><a href="WE-13.html#cb1064-4" aria-hidden="true" tabindex="-1"></a>gdist <span class="ot">&lt;-</span> <span class="fu">as.matrix</span> (gdist)</span>
<span id="cb1064-5"><a href="WE-13.html#cb1064-5" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(gdist)</span></code></pre></div>
<pre><code>##         X1      X2      X3      X4      X5      X6      X7      X8      X9
## X1 0.00000 0.13106 0.24000 0.10717 0.11773 0.11482 0.11978 0.22386 0.31478
## X2 0.13106 0.00000 0.15344 0.10157 0.16524 0.14467 0.13044 0.25668 0.27293
## X3 0.24000 0.15344 0.00000 0.18750 0.25417 0.27656 0.22935 0.31458 0.35767
## X4 0.10717 0.10157 0.18750 0.00000 0.18065 0.15483 0.12831 0.21910 0.25948
## X5 0.11773 0.16524 0.25417 0.18065 0.00000 0.11089 0.11491 0.26057 0.31374
## X6 0.11482 0.14467 0.27656 0.15483 0.11089 0.00000 0.08381 0.23851 0.27842
##        X10     X11     X12     X13     X14     X15     X16     X18     X19
## X1 0.30506 0.26169 0.34715 0.27857 0.39477 0.36995 0.36053 0.29083 0.34259
## X2 0.30683 0.28459 0.35504 0.26397 0.37196 0.36541 0.35676 0.31283 0.31005
## X3 0.38631 0.34337 0.44137 0.34524 0.40587 0.40521 0.37579 0.42240 0.36712
## X4 0.26875 0.22184 0.33929 0.23413 0.36887 0.37743 0.32979 0.31647 0.26990
## X5 0.31012 0.29413 0.33393 0.26518 0.36171 0.36771 0.35540 0.31987 0.32218
## X6 0.27074 0.28646 0.29556 0.26167 0.36458 0.38211 0.35632 0.28228 0.33013
##        X20     X21     X23     X24     X25     X26     X27     X28     X29
## X1 0.32208 0.39705 0.41417 0.36951 0.44451 0.47939 0.35175 0.39072 0.51191
## X2 0.24993 0.45253 0.41518 0.37476 0.45304 0.46454 0.37366 0.37362 0.49434
## X3 0.25417 0.54167 0.48735 0.48620 0.51667 0.45781 0.43750 0.43818 0.50388
## X4 0.20347 0.44643 0.43088 0.39755 0.46875 0.45087 0.38889 0.42681 0.47263
## X5 0.30010 0.45017 0.35466 0.32677 0.39583 0.46230 0.34107 0.41169 0.48235
## X6 0.30192 0.40980 0.37248 0.32394 0.39645 0.46567 0.33234 0.40289 0.49531
##        X30     X31
## X1 0.50106 0.43008
## X2 0.50797 0.39241
## X3 0.53165 0.37708
## X4 0.54137 0.42292
## X5 0.44777 0.34762
## X6 0.46864 0.37533</code></pre>
<p>Convert genetic distance to flow</p>
<div class="sourceCode" id="cb1066"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1066-1"><a href="WE-13.html#cb1066-1" aria-hidden="true" tabindex="-1"></a>gdist <span class="ot">&lt;-</span> GeNetIt<span class="sc">::</span><span class="fu">flow</span>(gdist)</span>
<span id="cb1066-2"><a href="WE-13.html#cb1066-2" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(gdist)</span></code></pre></div>
<pre><code>##         X1      X2      X3      X4      X5      X6      X7      X8      X9
## X1      NA 0.86894 0.76000 0.89283 0.88227 0.88518 0.88022 0.77614 0.68522
## X2 0.86894      NA 0.84656 0.89843 0.83476 0.85533 0.86956 0.74332 0.72707
## X3 0.76000 0.84656      NA 0.81250 0.74583 0.72344 0.77065 0.68542 0.64233
## X4 0.89283 0.89843 0.81250      NA 0.81935 0.84517 0.87169 0.78090 0.74052
## X5 0.88227 0.83476 0.74583 0.81935      NA 0.88911 0.88509 0.73943 0.68626
## X6 0.88518 0.85533 0.72344 0.84517 0.88911      NA 0.91619 0.76149 0.72158
##        X10     X11     X12     X13     X14     X15     X16     X18     X19
## X1 0.69494 0.73831 0.65285 0.72143 0.60523 0.63005 0.63947 0.70917 0.65741
## X2 0.69317 0.71541 0.64496 0.73603 0.62804 0.63459 0.64324 0.68717 0.68995
## X3 0.61369 0.65663 0.55863 0.65476 0.59413 0.59479 0.62421 0.57760 0.63288
## X4 0.73125 0.77816 0.66071 0.76587 0.63113 0.62257 0.67021 0.68353 0.73010
## X5 0.68988 0.70587 0.66607 0.73482 0.63829 0.63229 0.64460 0.68013 0.67782
## X6 0.72926 0.71354 0.70444 0.73833 0.63542 0.61789 0.64368 0.71772 0.66987
##        X20     X21     X23     X24     X25     X26     X27     X28     X29
## X1 0.67792 0.60295 0.58583 0.63049 0.55549 0.52061 0.64825 0.60928 0.48809
## X2 0.75007 0.54747 0.58482 0.62524 0.54696 0.53546 0.62634 0.62638 0.50566
## X3 0.74583 0.45833 0.51265 0.51380 0.48333 0.54219 0.56250 0.56182 0.49612
## X4 0.79653 0.55357 0.56912 0.60245 0.53125 0.54913 0.61111 0.57319 0.52737
## X5 0.69990 0.54983 0.64534 0.67323 0.60417 0.53770 0.65893 0.58831 0.51765
## X6 0.69808 0.59020 0.62752 0.67606 0.60355 0.53433 0.66766 0.59711 0.50469
##        X30     X31
## X1 0.49894 0.56992
## X2 0.49203 0.60759
## X3 0.46835 0.62292
## X4 0.45863 0.57708
## X5 0.55223 0.65238
## X6 0.53136 0.62467</code></pre>
<p>Convert genetid “flow” matrix into an unfolded dataframe with site IDs</p>
<div class="sourceCode" id="cb1068"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1068-1"><a href="WE-13.html#cb1068-1" aria-hidden="true" tabindex="-1"></a>gdist <span class="ot">&lt;-</span> GeNetIt<span class="sc">::</span><span class="fu">dmatrix.df</span>(gdist)</span>
<span id="cb1068-2"><a href="WE-13.html#cb1068-2" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(gdist)</span></code></pre></div>
<pre><code>##   X1 X2 distance
## 2 X2 X1  0.86894
## 3 X3 X1  0.76000
## 4 X4 X1  0.89283
## 5 X5 X1  0.88227
## 6 X6 X1  0.88518
## 7 X7 X1  0.88022</code></pre>
<p>The default column name is “distance” as this matrix could represent multiple type of distances. We are renaming here to avoid confusion as to the type of distance here.</p>
<div class="sourceCode" id="cb1070"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1070-1"><a href="WE-13.html#cb1070-1" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(gdist)[<span class="dv">3</span>] <span class="ot">&lt;-</span> <span class="st">&quot;GDIST&quot;</span></span>
<span id="cb1070-2"><a href="WE-13.html#cb1070-2" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(gdist)</span></code></pre></div>
<pre><code>## [1] &quot;X1&quot;    &quot;X2&quot;    &quot;GDIST&quot;</code></pre>
<p>Some housekeeping here to rename colums to from, to nodes and remove “X”s added in conversion from matrix.</p>
<div class="sourceCode" id="cb1072"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1072-1"><a href="WE-13.html#cb1072-1" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(gdist)[<span class="dv">1</span>] <span class="ot">&lt;-</span> <span class="st">&quot;FROM&quot;</span></span>
<span id="cb1072-2"><a href="WE-13.html#cb1072-2" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(gdist)[<span class="dv">2</span>] <span class="ot">&lt;-</span> <span class="st">&quot;TO&quot;</span></span>
<span id="cb1072-3"><a href="WE-13.html#cb1072-3" aria-hidden="true" tabindex="-1"></a>gdist[,<span class="dv">1</span>] <span class="ot">&lt;-</span><span class="fu">sub</span>(<span class="st">&quot;X&quot;</span>, <span class="st">&quot;&quot;</span>, gdist[,<span class="dv">1</span>])</span>
<span id="cb1072-4"><a href="WE-13.html#cb1072-4" aria-hidden="true" tabindex="-1"></a>gdist[,<span class="dv">2</span>] <span class="ot">&lt;-</span><span class="fu">sub</span>(<span class="st">&quot;X&quot;</span>, <span class="st">&quot;&quot;</span>, gdist[,<span class="dv">2</span>])</span>
<span id="cb1072-5"><a href="WE-13.html#cb1072-5" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(gdist)</span></code></pre></div>
<pre><code>## [1] &quot;FROM&quot;  &quot;TO&quot;    &quot;GDIST&quot;</code></pre>
<p>Create an identifier of to from nodes to connect back to the graph and link gdist data to knn graph.</p>
<div class="sourceCode" id="cb1074"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1074-1"><a href="WE-13.html#cb1074-1" aria-hidden="true" tabindex="-1"></a>gdist <span class="ot">&lt;-</span> <span class="fu">cbind</span>(<span class="at">from.to=</span><span class="fu">paste</span>(gdist[,<span class="dv">1</span>], gdist[,<span class="dv">2</span>], <span class="at">sep=</span><span class="st">&quot;.&quot;</span>), gdist)</span>
<span id="cb1074-2"><a href="WE-13.html#cb1074-2" aria-hidden="true" tabindex="-1"></a>dist.graph<span class="sc">@</span>data<span class="sc">$</span>from.to <span class="ot">&lt;-</span> <span class="fu">paste</span>(dist.graph<span class="sc">$</span>from_ID, dist.graph<span class="sc">$</span>to_ID, <span class="at">sep=</span><span class="st">&quot;.&quot;</span>)</span>
<span id="cb1074-3"><a href="WE-13.html#cb1074-3" aria-hidden="true" tabindex="-1"></a>dist.graph <span class="ot">&lt;-</span> <span class="fu">merge</span>(dist.graph, gdist, <span class="at">by =</span> <span class="st">&quot;from.to&quot;</span>) </span>
<span id="cb1074-4"><a href="WE-13.html#cb1074-4" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(dist.graph<span class="sc">@</span>data)</span></code></pre></div>
<pre><code>##     from.to i j from_ID to_ID   length FROM TO   GDIST
## 483   27.15 1 2      27    15 4052.869   27 15 0.47934
## 486   27.19 1 3      27    19 3065.350   27 19 0.61603
## 484   27.16 1 4      27    16 4003.222   27 16 0.63321
## 478   27.10 1 5      27    10 5675.296   27 10 0.51349
## 503    27.8 1 6      27     8 6482.365   27  8 0.51932
## 479   27.11 1 7      27    11 5010.120   27 11 0.50584</code></pre>
</div>
<div id="c.-write-out-results-ascii-csv-and-lines-shapefile." class="section level4">
<h4>c. Write out results (ASCII csv and lines shapefile).</h4>
<p>Saving the results (as ASCII csv and lines shapefile) is is a really useful operation for your own data, especially if you have a large graph, so you don’t have to recreate this object.</p>
<div class="sourceCode" id="cb1076"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1076-1"><a href="WE-13.html#cb1076-1" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span>(<span class="sc">!</span><span class="fu">dir.exists</span>(<span class="fu">paste0</span>(here<span class="sc">::</span><span class="fu">here</span>(),<span class="st">&quot;/output&quot;</span>))) </span>
<span id="cb1076-2"><a href="WE-13.html#cb1076-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">dir.create</span>(<span class="fu">paste0</span>(here<span class="sc">::</span><span class="fu">here</span>(),<span class="st">&quot;/output&quot;</span>))</span>
<span id="cb1076-3"><a href="WE-13.html#cb1076-3" aria-hidden="true" tabindex="-1"></a><span class="fu">write.csv</span>(gdist, <span class="at">file=</span> <span class="fu">paste0</span>(here<span class="sc">::</span><span class="fu">here</span>(),<span class="st">&quot;/output/gdist.csv&quot;</span>))</span></code></pre></div>
<p>Uncheck the following lines to write out a shapefile. This is commented out as rgdal may not work properly, depending on your computer configuration.</p>
<div class="sourceCode" id="cb1077"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1077-1"><a href="WE-13.html#cb1077-1" aria-hidden="true" tabindex="-1"></a><span class="co">#rgdal::writeOGR(dist.graph, paste0(here::here(),&quot;/output&quot;), &quot;DistGraph&quot;, </span></span>
<span id="cb1077-2"><a href="WE-13.html#cb1077-2" aria-hidden="true" tabindex="-1"></a><span class="co">#                driver=&quot;ESRI Shapefile&quot;, check_exists=TRUE, overwrite_layer=TRUE)</span></span></code></pre></div>
</div>
</div>
<div id="preparing-raster-based-covariates" class="section level3">
<h3>5. Preparing raster-based covariates</h3>
<div id="a.-import-raster-data-as-stack" class="section level4">
<h4>a. Import raster data as stack</h4>
<p>This function will import a raster stack with nine raster maps of the study area.</p>
<div class="sourceCode" id="cb1078"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1078-1"><a href="WE-13.html#cb1078-1" aria-hidden="true" tabindex="-1"></a>xvars <span class="ot">&lt;-</span> rio<span class="sc">::</span><span class="fu">import</span>(<span class="st">&quot;https://www.dropbox.com/s/xjl9zpgqplwg1us/ralu.rasters.rds?dl=1&quot;</span>)</span>
<span id="cb1078-2"><a href="WE-13.html#cb1078-2" aria-hidden="true" tabindex="-1"></a>xvars</span></code></pre></div>
<pre><code>## class      : RasterStack 
## dimensions : 2237, 2347, 5250239, 9  (nrow, ncol, ncell, nlayers)
## resolution : 10, 10  (x, y)
## extent     : 677292.5, 700762.5, 4987413, 5009783  (xmin, xmax, ymin, ymax)
## crs        : +proj=utm +zone=11 +ellps=GRS80 +towgs84=0,0,0,-0,-0,-0,0 +units=m +no_defs 
## names      :          cti,          dd5,          ffp,          gsp,          hli,         nlcd,       pratio,      rough27,          srr 
## min values : 6.269518e-01, 2.864937e+02, 0.000000e+00, 1.966725e+02, 1.014000e+03, 1.100000e+01, 4.276079e-01, 5.816957e-04, 3.869545e-02 
## max values : 2.605896e+01, 2.143000e+03, 1.170000e+02, 3.380697e+02, 1.000000e+04, 9.500000e+01, 5.329815e-01, 1.859006e+04, 7.725910e-01</code></pre>
<div class="sourceCode" id="cb1080"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1080-1"><a href="WE-13.html#cb1080-1" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(xvars)</span></code></pre></div>
<pre><code>## [1] &quot;cti&quot;     &quot;dd5&quot;     &quot;ffp&quot;     &quot;gsp&quot;     &quot;hli&quot;     &quot;nlcd&quot;    &quot;pratio&quot; 
## [8] &quot;rough27&quot; &quot;srr&quot;</code></pre>
</div>
<div id="b.-calculate-wetland-area-within-buffer" class="section level4">
<h4>b. Calculate wetland area within buffer</h4>
<p>You want to know if areas of dense wetlands produce more frogs. Calculate the proportion of the landscape around each site that is wetland. What buffer distance will you use?</p>
<p>We can create a wetland raster from the NLCD data. Wetland classes are 11 (open water), 90 and 95.</p>
<p>Create a vector of reclassification where non-wetland is a 0, wetland is a 1.</p>
<div class="sourceCode" id="cb1082"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1082-1"><a href="WE-13.html#cb1082-1" aria-hidden="true" tabindex="-1"></a>m <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="dv">0</span>,<span class="fl">10.8</span>, <span class="dv">0</span>,<span class="fl">10.9</span>,<span class="fl">12.1</span>,<span class="dv">1</span>,<span class="fl">12.9</span>,<span class="fl">89.1</span>,<span class="dv">0</span>, <span class="fl">89.5</span>,<span class="fl">95.1</span>,<span class="dv">1</span>)</span>
<span id="cb1082-2"><a href="WE-13.html#cb1082-2" aria-hidden="true" tabindex="-1"></a>  reclass <span class="ot">&lt;-</span> <span class="fu">matrix</span>(m, <span class="at">ncol=</span><span class="dv">3</span>, <span class="at">byrow=</span><span class="cn">TRUE</span>)</span></code></pre></div>
<p>Reclassifying nlcd by the reclass matrix</p>
<div class="sourceCode" id="cb1083"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1083-1"><a href="WE-13.html#cb1083-1" aria-hidden="true" tabindex="-1"></a>wetlnd <span class="ot">&lt;-</span> raster<span class="sc">::</span><span class="fu">reclassify</span>(xvars<span class="sc">$</span>nlcd, reclass) </span></code></pre></div>
<div class="sourceCode" id="cb1084"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1084-1"><a href="WE-13.html#cb1084-1" aria-hidden="true" tabindex="-1"></a>  wetlnd<span class="sc">@</span>data<span class="sc">@</span>names <span class="ot">&lt;-</span> <span class="st">&quot;wetlnd&quot;</span></span></code></pre></div>
<div class="sourceCode" id="cb1085"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1085-1"><a href="WE-13.html#cb1085-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(wetlnd)</span></code></pre></div>
<p><img src="13-Week13_files/figure-html/unnamed-chunk-31-1.png" width="672" /></p>
<p>Adding this new parameter to our raster stack</p>
<div class="sourceCode" id="cb1086"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1086-1"><a href="WE-13.html#cb1086-1" aria-hidden="true" tabindex="-1"></a>xvars <span class="ot">&lt;-</span> raster<span class="sc">::</span><span class="fu">stack</span>(xvars, wetlnd)</span>
<span id="cb1086-2"><a href="WE-13.html#cb1086-2" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(xvars)</span></code></pre></div>
<pre><code>##  [1] &quot;cti&quot;     &quot;dd5&quot;     &quot;ffp&quot;     &quot;gsp&quot;     &quot;hli&quot;     &quot;nlcd&quot;    &quot;pratio&quot; 
##  [8] &quot;rough27&quot; &quot;srr&quot;     &quot;wetlnd&quot;</code></pre>
</div>
<div id="c.-create-raster-of-wetland-proportion" class="section level4">
<h4>c. Create raster of wetland proportion</h4>
<p>Create a raster of the proportion of landscape (PLAND) that is wetland, using a 300 m radius.</p>
<p><strong>Challenge</strong>:</p>
<ul>
<li>What happens if you change this radius?<br />
</li>
<li>What radius do you think makes the most sense ecologically?</li>
</ul>
<p>Here we adapt code from Week 2 Worked Example.</p>
<div class="sourceCode" id="cb1088"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1088-1"><a href="WE-13.html#cb1088-1" aria-hidden="true" tabindex="-1"></a>nlcd_sampled <span class="ot">&lt;-</span> landscapemetrics<span class="sc">::</span><span class="fu">sample_lsm</span>(<span class="at">landscape =</span> xvars[[<span class="st">&quot;wetlnd&quot;</span>]], </span>
<span id="cb1088-2"><a href="WE-13.html#cb1088-2" aria-hidden="true" tabindex="-1"></a>                                                   <span class="at">what =</span> <span class="st">&quot;lsm_c_pland&quot;</span>,</span>
<span id="cb1088-3"><a href="WE-13.html#cb1088-3" aria-hidden="true" tabindex="-1"></a>                                                   <span class="at">shape =</span> <span class="st">&quot;circle&quot;</span>,</span>
<span id="cb1088-4"><a href="WE-13.html#cb1088-4" aria-hidden="true" tabindex="-1"></a>                                                   <span class="at">y =</span> sites, </span>
<span id="cb1088-5"><a href="WE-13.html#cb1088-5" aria-hidden="true" tabindex="-1"></a>                                                   <span class="at">size =</span> <span class="dv">300</span>, </span>
<span id="cb1088-6"><a href="WE-13.html#cb1088-6" aria-hidden="true" tabindex="-1"></a>                                                   <span class="at">return_raster =</span> <span class="cn">FALSE</span>,</span>
<span id="cb1088-7"><a href="WE-13.html#cb1088-7" aria-hidden="true" tabindex="-1"></a>                                                   <span class="at">plot_id=</span>sites<span class="sc">@</span>data<span class="sc">$</span>SiteID)</span>
<span id="cb1088-8"><a href="WE-13.html#cb1088-8" aria-hidden="true" tabindex="-1"></a>pwetland <span class="ot">&lt;-</span> dplyr<span class="sc">::</span><span class="fu">select</span>(dplyr<span class="sc">::</span><span class="fu">filter</span>(nlcd_sampled, class <span class="sc">==</span> <span class="dv">1</span>, </span>
<span id="cb1088-9"><a href="WE-13.html#cb1088-9" aria-hidden="true" tabindex="-1"></a>                                        metric <span class="sc">==</span> <span class="st">&quot;pland&quot;</span>), plot_id, value)  </span>
<span id="cb1088-10"><a href="WE-13.html#cb1088-10" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(pwetland) <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;SiteID&quot;</span>, <span class="st">&quot;pwetland&quot;</span>)</span>
<span id="cb1088-11"><a href="WE-13.html#cb1088-11" aria-hidden="true" tabindex="-1"></a>pwetland<span class="sc">$</span>pwetland <span class="ot">&lt;-</span> pwetland<span class="sc">$</span>pwetland<span class="sc">/</span><span class="dv">100</span></span>
<span id="cb1088-12"><a href="WE-13.html#cb1088-12" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(pwetland)</span></code></pre></div>
<pre><code>## # A tibble: 6 x 2
##   SiteID pwetland
##   &lt;fct&gt;     &lt;dbl&gt;
## 1 2        0.0948
## 2 3        0.0835
## 3 4        0.102 
## 4 5        0.0609
## 5 7        0.0516
## 6 8        0.108</code></pre>
<p>Important: there could be cases with no wetlands (where prop.landscape should be zero), which will result in missing rows in <code>pwetland</code>. Here we use <code>left_join</code> to join rows withi <code>sites</code> by <code>SiteID</code>, then replace missing values by 0.</p>
<div class="sourceCode" id="cb1090"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1090-1"><a href="WE-13.html#cb1090-1" aria-hidden="true" tabindex="-1"></a>sites<span class="sc">@</span>data <span class="ot">&lt;-</span> dplyr<span class="sc">::</span><span class="fu">left_join</span>(sites<span class="sc">@</span>data, pwetland)</span></code></pre></div>
<pre><code>## Joining, by = &quot;SiteID&quot;</code></pre>
<div class="sourceCode" id="cb1092"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1092-1"><a href="WE-13.html#cb1092-1" aria-hidden="true" tabindex="-1"></a>sites<span class="sc">@</span>data<span class="sc">$</span>pwetland[<span class="fu">is.na</span>(sites<span class="sc">@</span>data<span class="sc">$</span>pwetland)] <span class="ot">&lt;-</span> <span class="dv">0</span></span>
<span id="cb1092-2"><a href="WE-13.html#cb1092-2" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(sites<span class="sc">@</span>data)</span></code></pre></div>
<pre><code>##         SiteName degree betweenness SiteID     Elev Length    Area  Perim Depth
## 1   AirplaneLake      8   279.58405     27 2564.381    390 62582.2 1142.8 21.64
## 2 BachelorMeadow      7   292.32533     15 2591.781      0   225.0   60.0  0.40
## 3 BarkingFoxLake      9   512.37876     19 2545.275    160 12000.0  435.0  5.00
## 4        BobLake      6   466.35631     16 2649.125    143  4600.0  321.4  2.00
## 5      CacheLake      4    11.57913     10 2475.829     75  2268.8  192.0  1.86
## 6        DoeLake      5    31.48064      8 2463.006    170 13034.9  463.2  6.03
##    pH Dforest Drock Dshrub    pwetland
## 1 6.5   0.398 0.051   0.00 0.195821530
## 2 6.1   0.000 0.000   0.20 0.015929204
## 3 6.5   0.400 0.250   0.05 0.051009564
## 4 7.0   0.550 0.000   0.05 0.000000000
## 5 6.5   0.508 0.000   0.00 0.004253811
## 6 7.6   0.254 0.000   0.00 0.107954545</code></pre>
<p><strong>Challenge:</strong></p>
<ul>
<li>Try creating some additional metrics of your own from these rasters.</li>
</ul>
</div>
</div>
<div id="extract-raster-stats-for-nodes-and-edges" class="section level3">
<h3>6. Extract raster stats for nodes and edges</h3>
<div id="a.-extract-at-site-variables-from-rasters" class="section level4">
<h4>a. Extract at-site variables from rasters</h4>
<p>Add characteristics of sample sites from your rasters as potential at site variables. Here we are extracting the raster values that intersect our wetlands (point data). What raster is not included in this step? Why? What at-site characteristics may impact the production of potential migrants?</p>
<div class="sourceCode" id="cb1094"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1094-1"><a href="WE-13.html#cb1094-1" aria-hidden="true" tabindex="-1"></a>sites<span class="sc">@</span>data <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(sites<span class="sc">@</span>data, raster<span class="sc">::</span><span class="fu">extract</span>(xvars, sites))</span></code></pre></div>
<div class="sourceCode" id="cb1095"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1095-1"><a href="WE-13.html#cb1095-1" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(sites<span class="sc">@</span>data)</span></code></pre></div>
<pre><code>##  [1] &quot;SiteName&quot;    &quot;degree&quot;      &quot;betweenness&quot; &quot;SiteID&quot;      &quot;Elev&quot;       
##  [6] &quot;Length&quot;      &quot;Area&quot;        &quot;Perim&quot;       &quot;Depth&quot;       &quot;pH&quot;         
## [11] &quot;Dforest&quot;     &quot;Drock&quot;       &quot;Dshrub&quot;      &quot;pwetland&quot;    &quot;cti&quot;        
## [16] &quot;dd5&quot;         &quot;ffp&quot;         &quot;gsp&quot;         &quot;hli&quot;         &quot;nlcd&quot;       
## [21] &quot;pratio&quot;      &quot;rough27&quot;     &quot;srr&quot;         &quot;wetlnd&quot;</code></pre>
</div>
<div id="b.-add-covariates-to-graph-edges." class="section level4">
<h4>b. Add covariates to graph edges.</h4>
<p>Calculating statistical moments (e.g., mean, sdev) for categorical variables makes no sense. We we will remove them here.</p>
<div class="sourceCode" id="cb1097"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1097-1"><a href="WE-13.html#cb1097-1" aria-hidden="true" tabindex="-1"></a>idx <span class="ot">&lt;-</span> <span class="fu">which</span>(<span class="fu">names</span>(xvars) <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">&quot;nlcd&quot;</span>,<span class="st">&quot;wetlnd&quot;</span>))</span></code></pre></div>
</div>
<div id="c.-calculating-stats-along-edges" class="section level4">
<h4>c. Calculating stats along edges</h4>
<p>Here we define the projection of the distance graph and then calculate statistics</p>
<p>You can calculate any statistical moment you wish from your sample of the landscape between nodes. Make sure that these moments are ecologically/biologically meaningful.</p>
<div class="sourceCode" id="cb1098"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1098-1"><a href="WE-13.html#cb1098-1" aria-hidden="true" tabindex="-1"></a>dist.graph<span class="sc">@</span>proj4string<span class="sc">@</span>projargs <span class="ot">&lt;-</span> <span class="st">&quot;+proj=utm +zone=11 +ellps=GRS80 +towgs84=0,0,0,-0,-0,-0,0 +units=m +no_defs &quot;</span></span>
<span id="cb1098-2"><a href="WE-13.html#cb1098-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1098-3"><a href="WE-13.html#cb1098-3" aria-hidden="true" tabindex="-1"></a>stats <span class="ot">&lt;-</span> GeNetIt<span class="sc">::</span><span class="fu">graph.statistics</span>(dist.graph, <span class="at">r =</span> xvars[[<span class="sc">-</span>idx]], <span class="at">buffer=</span> <span class="cn">NULL</span>, </span>
<span id="cb1098-4"><a href="WE-13.html#cb1098-4" aria-hidden="true" tabindex="-1"></a>                          <span class="at">stats =</span> <span class="fu">c</span>(<span class="st">&quot;min&quot;</span>, <span class="st">&quot;mean&quot;</span>, <span class="st">&quot;max&quot;</span>, <span class="st">&quot;var&quot;</span>, <span class="st">&quot;median&quot;</span>))</span></code></pre></div>
<pre><code>## Warning in wkt(obj): CRS object has no comment</code></pre>
<div class="sourceCode" id="cb1100"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1100-1"><a href="WE-13.html#cb1100-1" aria-hidden="true" tabindex="-1"></a>dist.graph<span class="sc">@</span>data <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(dist.graph<span class="sc">@</span>data, stats)</span>
<span id="cb1100-2"><a href="WE-13.html#cb1100-2" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(dist.graph<span class="sc">@</span>data)</span></code></pre></div>
<pre><code>##  [1] &quot;from.to&quot;        &quot;i&quot;              &quot;j&quot;              &quot;from_ID&quot;       
##  [5] &quot;to_ID&quot;          &quot;length&quot;         &quot;FROM&quot;           &quot;TO&quot;            
##  [9] &quot;GDIST&quot;          &quot;min.cti&quot;        &quot;min.dd5&quot;        &quot;min.ffp&quot;       
## [13] &quot;min.gsp&quot;        &quot;min.hli&quot;        &quot;min.pratio&quot;     &quot;min.rough27&quot;   
## [17] &quot;min.srr&quot;        &quot;mean.cti&quot;       &quot;mean.dd5&quot;       &quot;mean.ffp&quot;      
## [21] &quot;mean.gsp&quot;       &quot;mean.hli&quot;       &quot;mean.pratio&quot;    &quot;mean.rough27&quot;  
## [25] &quot;mean.srr&quot;       &quot;max.cti&quot;        &quot;max.dd5&quot;        &quot;max.ffp&quot;       
## [29] &quot;max.gsp&quot;        &quot;max.hli&quot;        &quot;max.pratio&quot;     &quot;max.rough27&quot;   
## [33] &quot;max.srr&quot;        &quot;var.cti&quot;        &quot;var.dd5&quot;        &quot;var.ffp&quot;       
## [37] &quot;var.gsp&quot;        &quot;var.hli&quot;        &quot;var.pratio&quot;     &quot;var.rough27&quot;   
## [41] &quot;var.srr&quot;        &quot;median.cti&quot;     &quot;median.dd5&quot;     &quot;median.ffp&quot;    
## [45] &quot;median.gsp&quot;     &quot;median.hli&quot;     &quot;median.pratio&quot;  &quot;median.rough27&quot;
## [49] &quot;median.srr&quot;</code></pre>
<p><strong>Bonus challenge:</strong></p>
<ul>
<li>Create a function for returning the 95th percentile.</li>
<li>How would you create this function and add it to the stats calculated?</li>
<li>What other statistical moments do you want?<br />
</li>
<li>Can you create a function?</li>
</ul>
</div>
<div id="d.-calculating-stats-for-categorical-variables" class="section level4">
<h4>d. Calculating stats for categorical variables</h4>
<p>What about categorical variables? Moments are nonsensical.</p>
<ul>
<li>Create a function for returning the % wetland between sites.</li>
<li>IMPORTANT: We do not want these values to be correlated with distance, so don’t count number pixels.<br />
</li>
<li>Then use this function to calculate an additional statistic, and then add it to the graph.</li>
<li>Are there other categorical variables that you think may be ecologically important?</li>
</ul>
<div class="sourceCode" id="cb1102"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1102-1"><a href="WE-13.html#cb1102-1" aria-hidden="true" tabindex="-1"></a>wet.pct <span class="ot">&lt;-</span> <span class="cf">function</span>(x) { </span>
<span id="cb1102-2"><a href="WE-13.html#cb1102-2" aria-hidden="true" tabindex="-1"></a>  x <span class="ot">&lt;-</span> <span class="fu">ifelse</span>( x <span class="sc">==</span> <span class="dv">11</span> <span class="sc">|</span> x <span class="sc">==</span> <span class="dv">90</span> <span class="sc">|</span> x <span class="sc">==</span> <span class="dv">95</span>, <span class="dv">1</span>, <span class="dv">0</span>)</span>
<span id="cb1102-3"><a href="WE-13.html#cb1102-3" aria-hidden="true" tabindex="-1"></a>    <span class="fu">prop.table</span>(<span class="fu">table</span>(x))[<span class="dv">2</span>] </span>
<span id="cb1102-4"><a href="WE-13.html#cb1102-4" aria-hidden="true" tabindex="-1"></a>}</span></code></pre></div>
<div class="sourceCode" id="cb1103"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1103-1"><a href="WE-13.html#cb1103-1" aria-hidden="true" tabindex="-1"></a>wetstats <span class="ot">&lt;-</span> GeNetIt<span class="sc">::</span><span class="fu">graph.statistics</span>(dist.graph, <span class="at">r=</span>xvars<span class="sc">$</span>nlcd, <span class="at">buffer=</span> <span class="cn">NULL</span>, </span>
<span id="cb1103-2"><a href="WE-13.html#cb1103-2" aria-hidden="true" tabindex="-1"></a>                        <span class="at">stats =</span> <span class="fu">c</span>(<span class="st">&quot;wet.pct&quot;</span>))</span></code></pre></div>
<pre><code>## Warning in wkt(obj): CRS object has no comment</code></pre>
<div class="sourceCode" id="cb1105"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1105-1"><a href="WE-13.html#cb1105-1" aria-hidden="true" tabindex="-1"></a>  wetstats[<span class="fu">is.na</span>(wetstats)] <span class="ot">&lt;-</span> <span class="dv">0</span></span>
<span id="cb1105-2"><a href="WE-13.html#cb1105-2" aria-hidden="true" tabindex="-1"></a>  dist.graph<span class="sc">@</span>data <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(dist.graph<span class="sc">@</span>data, wetstats)</span>
<span id="cb1105-3"><a href="WE-13.html#cb1105-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">names</span>(dist.graph<span class="sc">@</span>data)</span></code></pre></div>
<pre><code>##  [1] &quot;from.to&quot;        &quot;i&quot;              &quot;j&quot;              &quot;from_ID&quot;       
##  [5] &quot;to_ID&quot;          &quot;length&quot;         &quot;FROM&quot;           &quot;TO&quot;            
##  [9] &quot;GDIST&quot;          &quot;min.cti&quot;        &quot;min.dd5&quot;        &quot;min.ffp&quot;       
## [13] &quot;min.gsp&quot;        &quot;min.hli&quot;        &quot;min.pratio&quot;     &quot;min.rough27&quot;   
## [17] &quot;min.srr&quot;        &quot;mean.cti&quot;       &quot;mean.dd5&quot;       &quot;mean.ffp&quot;      
## [21] &quot;mean.gsp&quot;       &quot;mean.hli&quot;       &quot;mean.pratio&quot;    &quot;mean.rough27&quot;  
## [25] &quot;mean.srr&quot;       &quot;max.cti&quot;        &quot;max.dd5&quot;        &quot;max.ffp&quot;       
## [29] &quot;max.gsp&quot;        &quot;max.hli&quot;        &quot;max.pratio&quot;     &quot;max.rough27&quot;   
## [33] &quot;max.srr&quot;        &quot;var.cti&quot;        &quot;var.dd5&quot;        &quot;var.ffp&quot;       
## [37] &quot;var.gsp&quot;        &quot;var.hli&quot;        &quot;var.pratio&quot;     &quot;var.rough27&quot;   
## [41] &quot;var.srr&quot;        &quot;median.cti&quot;     &quot;median.dd5&quot;     &quot;median.ffp&quot;    
## [45] &quot;median.gsp&quot;     &quot;median.hli&quot;     &quot;median.pratio&quot;  &quot;median.rough27&quot;
## [49] &quot;median.srr&quot;     &quot;wet.pct.nlcd&quot;</code></pre>
<p><strong>Question:</strong></p>
<ul>
<li>What other categorical variables would you like to include in the analysis? Can you create a function for these variables?</li>
</ul>
</div>
<div id="e.-add-node-data-to-graph" class="section level4">
<h4>e. Add node data to graph</h4>
<p>First create a list of names for the node data</p>
<div class="sourceCode" id="cb1107"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1107-1"><a href="WE-13.html#cb1107-1" aria-hidden="true" tabindex="-1"></a>node.var <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;degree&quot;</span>, <span class="st">&quot;betweenness&quot;</span>, <span class="st">&quot;Elev&quot;</span>, <span class="st">&quot;Length&quot;</span>, <span class="st">&quot;Area&quot;</span>, <span class="st">&quot;Perim&quot;</span>, </span>
<span id="cb1107-2"><a href="WE-13.html#cb1107-2" aria-hidden="true" tabindex="-1"></a>              <span class="st">&quot;Depth&quot;</span>, <span class="st">&quot;pH&quot;</span>,<span class="st">&quot;Dforest&quot;</span>,<span class="st">&quot;Drock&quot;</span>, <span class="st">&quot;Dshrub&quot;</span>, <span class="st">&quot;pwetland&quot;</span>, <span class="st">&quot;cti&quot;</span>,</span>
<span id="cb1107-3"><a href="WE-13.html#cb1107-3" aria-hidden="true" tabindex="-1"></a>              <span class="st">&quot;dd5&quot;</span>, <span class="st">&quot;ffp&quot;</span>,<span class="st">&quot;gsp&quot;</span>,<span class="st">&quot;pratio&quot;</span>,<span class="st">&quot;hli&quot;</span>,<span class="st">&quot;rough27&quot;</span>,<span class="st">&quot;srr&quot;</span>)</span></code></pre></div>
<p>These are all at site variables. Remember that we pulled all raster varibles. We want to critically think about hypotheses and not use all of these parameters.</p>
<ul>
<li>degree - graph degree</li>
<li>betweenness - graph betweeness</li>
<li>Elev - elevation (see comments below)</li>
<li>Length - geographic distance</li>
<li>Area - wetland area (field)</li>
<li>Perim - wetland perimeter (field)</li>
<li>Depth - wetland depth (field)- highly correlated with predatory fish presence/abundance</li>
<li>pH - wetland pH (field)</li>
<li>Dforest - distance to forest (field)</li>
<li>Drock - distance to rock (field)</li>
<li>Dshrub - distance to shrub (field)</li>
<li>pwetland - proportion of wetland in X buffer (calculated above)</li>
<li>cti - compound topographic wetness index - steady-state measure of wetness based on topography (raster data)</li>
<li>dd5 - degree days &gt;5 C (sum of temp) - (raster data)</li>
<li>ffp - frost free period (raster data)</li>
<li>gsp - growing season precipitation (raster data)</li>
<li>pratio - ratio of growing season precip to annual precip (raster data) - can indicate amount of snow to rain</li>
<li>hli - heat load index - topographic measure of exposure, related to productivity (ice-off and primary productivity) in this system (raster data)</li>
<li>rough27 - unscale topographic variation at a 27 X 27 (cells) window size (raster data)</li>
<li>ssr - measure of topographic variation at a 27X27 (cells) windo size - for this system pulling out ridgelines (raster data)</li>
</ul>
<p>NOTE: we are adding elevation here as a covariate. HOWEVER - elevation does not represent
ecological processes in and of itself. I strongly encourage using the components (temp, moisture, rainfall, vegetation, accessibility, etc.) directly and not elevation as a surrogate parameter.</p>
<p>Add the node data. Remember that we will calculate the value for both the from node and the to node. in a singly constained model, we will only use one of these (constrain (“group”) by from (produciton) or to (attraction))</p>
<div class="sourceCode" id="cb1108"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1108-1"><a href="WE-13.html#cb1108-1" aria-hidden="true" tabindex="-1"></a>node <span class="ot">&lt;-</span> GeNetIt<span class="sc">::</span><span class="fu">build.node.data</span>(sites<span class="sc">@</span>data, <span class="at">group.ids =</span> <span class="st">&quot;SiteID&quot;</span>, <span class="at">from.parms =</span> node.var)</span>
<span id="cb1108-2"><a href="WE-13.html#cb1108-2" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(node)</span></code></pre></div>
<pre><code>##   SiteID from.degree from.betweenness from.Elev from.Length from.Area
## 1     27           8        279.58405  2564.381         390   62582.2
## 2     15           7        292.32533  2591.781           0     225.0
## 3     19           9        512.37876  2545.275         160   12000.0
## 4     16           6        466.35631  2649.125         143    4600.0
## 5     10           4         11.57913  2475.829          75    2268.8
## 6      8           5         31.48064  2463.006         170   13034.9
##   from.Perim from.Depth from.pH from.Dforest from.Drock from.Dshrub
## 1     1142.8      21.64     6.5        0.398      0.051        0.00
## 2       60.0       0.40     6.1        0.000      0.000        0.20
## 3      435.0       5.00     6.5        0.400      0.250        0.05
## 4      321.4       2.00     7.0        0.550      0.000        0.05
## 5      192.0       1.86     6.5        0.508      0.000        0.00
## 6      463.2       6.03     7.6        0.254      0.000        0.00
##   from.pwetland from.cti from.dd5  from.ffp from.gsp from.pratio  from.hli
## 1   0.195821530 6.018890 522.0973  8.000000 286.1415        1732 0.4485655
## 2   0.015929204 5.860837 512.0000  8.000000 286.0000        2090 0.4470909
## 3   0.051009564 2.951499 502.4061  7.285662 288.5140        3070 0.4461185
## 4   0.000000000 5.120589 457.2643  5.000000 294.8773        1527 0.4458445
## 5   0.004253811 4.110081 565.8234  9.610519 276.4601        1977 0.4521334
## 6   0.107954545 3.712187 552.0031 10.000000 278.0000        1751 0.4514005
##   from.rough27  from.srr to.degree to.betweenness  to.Elev to.Length to.Area
## 1     460.0905 0.2309995         8      279.58405 2564.381       390 62582.2
## 2     401.6590 0.4483430         7      292.32533 2591.781         0   225.0
## 3    2772.4165 0.3047715         9      512.37876 2545.275       160 12000.0
## 4     352.5726 0.5804766         6      466.35631 2649.125       143  4600.0
## 5     119.7234 0.4306586         4       11.57913 2475.829        75  2268.8
## 6     603.1037 0.3480034         5       31.48064 2463.006       170 13034.9
##   to.Perim to.Depth to.pH to.Dforest to.Drock to.Dshrub to.pwetland   to.cti
## 1   1142.8    21.64   6.5      0.398    0.051      0.00 0.195821530 6.018890
## 2     60.0     0.40   6.1      0.000    0.000      0.20 0.015929204 5.860837
## 3    435.0     5.00   6.5      0.400    0.250      0.05 0.051009564 2.951499
## 4    321.4     2.00   7.0      0.550    0.000      0.05 0.000000000 5.120589
## 5    192.0     1.86   6.5      0.508    0.000      0.00 0.004253811 4.110081
## 6    463.2     6.03   7.6      0.254    0.000      0.00 0.107954545 3.712187
##     to.dd5    to.ffp   to.gsp to.pratio    to.hli to.rough27    to.srr
## 1 522.0973  8.000000 286.1415      1732 0.4485655   460.0905 0.2309995
## 2 512.0000  8.000000 286.0000      2090 0.4470909   401.6590 0.4483430
## 3 502.4061  7.285662 288.5140      3070 0.4461185  2772.4165 0.3047715
## 4 457.2643  5.000000 294.8773      1527 0.4458445   352.5726 0.5804766
## 5 565.8234  9.610519 276.4601      1977 0.4521334   119.7234 0.4306586
## 6 552.0031 10.000000 278.0000      1751 0.4514005   603.1037 0.3480034</code></pre>
</div>
</div>
<div id="gravity-models" class="section level3">
<h3>7. Gravity Models</h3>
<p>Now we get to create gravity models!</p>
<div id="a.-merge-distance-and-site-data" class="section level4">
<h4>a. Merge distance and site data</h4>
<p>Merge edge (distance graph) and edge (site) data.</p>
<div class="sourceCode" id="cb1110"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1110-1"><a href="WE-13.html#cb1110-1" aria-hidden="true" tabindex="-1"></a>gdata <span class="ot">&lt;-</span> <span class="fu">merge</span>(dist.graph, node, <span class="at">by.x=</span><span class="st">&quot;from_ID&quot;</span>, <span class="at">by.y=</span><span class="st">&quot;SiteID&quot;</span>) </span>
<span id="cb1110-2"><a href="WE-13.html#cb1110-2" aria-hidden="true" tabindex="-1"></a>  gdata <span class="ot">&lt;-</span> gdata<span class="sc">@</span>data</span>
<span id="cb1110-3"><a href="WE-13.html#cb1110-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">names</span>(gdata)</span></code></pre></div>
<pre><code>##  [1] &quot;from_ID&quot;          &quot;from.to&quot;          &quot;i&quot;                &quot;j&quot;               
##  [5] &quot;to_ID&quot;            &quot;length&quot;           &quot;FROM&quot;             &quot;TO&quot;              
##  [9] &quot;GDIST&quot;            &quot;min.cti&quot;          &quot;min.dd5&quot;          &quot;min.ffp&quot;         
## [13] &quot;min.gsp&quot;          &quot;min.hli&quot;          &quot;min.pratio&quot;       &quot;min.rough27&quot;     
## [17] &quot;min.srr&quot;          &quot;mean.cti&quot;         &quot;mean.dd5&quot;         &quot;mean.ffp&quot;        
## [21] &quot;mean.gsp&quot;         &quot;mean.hli&quot;         &quot;mean.pratio&quot;      &quot;mean.rough27&quot;    
## [25] &quot;mean.srr&quot;         &quot;max.cti&quot;          &quot;max.dd5&quot;          &quot;max.ffp&quot;         
## [29] &quot;max.gsp&quot;          &quot;max.hli&quot;          &quot;max.pratio&quot;       &quot;max.rough27&quot;     
## [33] &quot;max.srr&quot;          &quot;var.cti&quot;          &quot;var.dd5&quot;          &quot;var.ffp&quot;         
## [37] &quot;var.gsp&quot;          &quot;var.hli&quot;          &quot;var.pratio&quot;       &quot;var.rough27&quot;     
## [41] &quot;var.srr&quot;          &quot;median.cti&quot;       &quot;median.dd5&quot;       &quot;median.ffp&quot;      
## [45] &quot;median.gsp&quot;       &quot;median.hli&quot;       &quot;median.pratio&quot;    &quot;median.rough27&quot;  
## [49] &quot;median.srr&quot;       &quot;wet.pct.nlcd&quot;     &quot;from.degree&quot;      &quot;from.betweenness&quot;
## [53] &quot;from.Elev&quot;        &quot;from.Length&quot;      &quot;from.Area&quot;        &quot;from.Perim&quot;      
## [57] &quot;from.Depth&quot;       &quot;from.pH&quot;          &quot;from.Dforest&quot;     &quot;from.Drock&quot;      
## [61] &quot;from.Dshrub&quot;      &quot;from.pwetland&quot;    &quot;from.cti&quot;         &quot;from.dd5&quot;        
## [65] &quot;from.ffp&quot;         &quot;from.gsp&quot;         &quot;from.pratio&quot;      &quot;from.hli&quot;        
## [69] &quot;from.rough27&quot;     &quot;from.srr&quot;         &quot;to.degree&quot;        &quot;to.betweenness&quot;  
## [73] &quot;to.Elev&quot;          &quot;to.Length&quot;        &quot;to.Area&quot;          &quot;to.Perim&quot;        
## [77] &quot;to.Depth&quot;         &quot;to.pH&quot;            &quot;to.Dforest&quot;       &quot;to.Drock&quot;        
## [81] &quot;to.Dshrub&quot;        &quot;to.pwetland&quot;      &quot;to.cti&quot;           &quot;to.dd5&quot;          
## [85] &quot;to.ffp&quot;           &quot;to.gsp&quot;           &quot;to.pratio&quot;        &quot;to.hli&quot;          
## [89] &quot;to.rough27&quot;       &quot;to.srr&quot;</code></pre>
</div>
<div id="b.-define-a-set-of-models" class="section level4">
<h4>b. Define a set of models</h4>
<p>Think about hypothesis and create a set of models. What type of constraint? Write out model statements. HOWEVER, need to check for correlations (#3) before settling on a final set of model. At this point, make a list of these models in a text editor, but do not run these models until you check correlations.</p>
</div>
<div id="c.-check-for-correlations" class="section level4">
<h4>c. Check for correlations</h4>
<p>You will need to do nodes and edges separately, remember that data have to be ln transformed. For zero values, a common approach is ln(x - (min(x) - 1))</p>
<div class="sourceCode" id="cb1112"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1112-1"><a href="WE-13.html#cb1112-1" aria-hidden="true" tabindex="-1"></a>nodeln <span class="ot">&lt;-</span> node[,<span class="fu">c</span>(<span class="dv">2</span><span class="sc">:</span><span class="dv">21</span>)]</span>
<span id="cb1112-2"><a href="WE-13.html#cb1112-2" aria-hidden="true" tabindex="-1"></a>  <span class="cf">for</span>(i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">ncol</span>(nodeln)) {</span>
<span id="cb1112-3"><a href="WE-13.html#cb1112-3" aria-hidden="true" tabindex="-1"></a>    nodeln[,i] <span class="ot">&lt;-</span> <span class="fu">log</span>(nodeln[,i] <span class="sc">-</span> (<span class="fu">min</span>(nodeln[,i]) <span class="sc">-</span> <span class="dv">1</span>))</span>
<span id="cb1112-4"><a href="WE-13.html#cb1112-4" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb1112-5"><a href="WE-13.html#cb1112-5" aria-hidden="true" tabindex="-1"></a>nodecor.ln <span class="ot">&lt;-</span> <span class="fu">cor</span>(nodeln, <span class="at">y =</span> <span class="cn">NULL</span>, </span>
<span id="cb1112-6"><a href="WE-13.html#cb1112-6" aria-hidden="true" tabindex="-1"></a>                  <span class="at">use =</span> <span class="st">&quot;complete.obs&quot;</span>, </span>
<span id="cb1112-7"><a href="WE-13.html#cb1112-7" aria-hidden="true" tabindex="-1"></a>                  <span class="at">method =</span> <span class="st">&quot;pearson&quot;</span>)</span>
<span id="cb1112-8"><a href="WE-13.html#cb1112-8" aria-hidden="true" tabindex="-1"></a><span class="fu">round</span>(nodecor.ln, <span class="dv">3</span>) </span></code></pre></div>
<pre><code>##                  from.degree from.betweenness from.Elev from.Length from.Area
## from.degree            1.000            0.912     0.062      -0.156    -0.078
## from.betweenness       0.912            1.000    -0.140      -0.152    -0.090
## from.Elev              0.062           -0.140     1.000      -0.402    -0.449
## from.Length           -0.156           -0.152    -0.402       1.000     0.871
## from.Area             -0.078           -0.090    -0.449       0.871     1.000
## from.Perim            -0.108           -0.111    -0.450       0.838     0.991
## from.Depth            -0.056           -0.075    -0.381       0.759     0.816
## from.pH               -0.102           -0.164    -0.189       0.292     0.474
## from.Dforest           0.061            0.141     0.158      -0.183    -0.494
## from.Drock             0.036            0.087    -0.080       0.269     0.277
## from.Dshrub            0.225            0.237     0.074      -0.297    -0.165
## from.pwetland          0.003            0.155    -0.758       0.400     0.511
## from.cti              -0.082            0.070    -0.199      -0.070    -0.131
## from.dd5              -0.111           -0.140    -0.482       0.499     0.713
## from.ffp              -0.069           -0.032    -0.622       0.368     0.457
## from.gsp              -0.006           -0.021     0.580      -0.225    -0.225
## from.pratio            0.109           -0.037     0.322       0.167     0.172
## from.hli              -0.214           -0.174    -0.640       0.376     0.338
## from.rough27           0.198            0.055     0.607      -0.117    -0.145
## from.srr               0.070           -0.031     0.530      -0.400    -0.481
##                  from.Perim from.Depth from.pH from.Dforest from.Drock
## from.degree          -0.108     -0.056  -0.102        0.061      0.036
## from.betweenness     -0.111     -0.075  -0.164        0.141      0.087
## from.Elev            -0.450     -0.381  -0.189        0.158     -0.080
## from.Length           0.838      0.759   0.292       -0.183      0.269
## from.Area             0.991      0.816   0.474       -0.494      0.277
## from.Perim            1.000      0.762   0.523       -0.545      0.234
## from.Depth            0.762      1.000   0.239       -0.257      0.370
## from.pH               0.523      0.239   1.000       -0.587     -0.067
## from.Dforest         -0.545     -0.257  -0.587        1.000     -0.310
## from.Drock            0.234      0.370  -0.067       -0.310      1.000
## from.Dshrub          -0.135     -0.158   0.101       -0.282      0.326
## from.pwetland         0.492      0.540   0.049       -0.037     -0.037
## from.cti             -0.117     -0.059  -0.211        0.239     -0.183
## from.dd5              0.739      0.422   0.655       -0.698      0.208
## from.ffp              0.456      0.332   0.451       -0.415      0.245
## from.gsp             -0.218     -0.122  -0.283        0.159     -0.050
## from.pratio           0.160      0.118   0.132       -0.281      0.194
## from.hli              0.354      0.144   0.453       -0.229     -0.066
## from.rough27         -0.158     -0.209  -0.110        0.046      0.154
## from.srr             -0.465     -0.369  -0.019       -0.065      0.152
##                  from.Dshrub from.pwetland from.cti from.dd5 from.ffp from.gsp
## from.degree            0.225         0.003   -0.082   -0.111   -0.069   -0.006
## from.betweenness       0.237         0.155    0.070   -0.140   -0.032   -0.021
## from.Elev              0.074        -0.758   -0.199   -0.482   -0.622    0.580
## from.Length           -0.297         0.400   -0.070    0.499    0.368   -0.225
## from.Area             -0.165         0.511   -0.131    0.713    0.457   -0.225
## from.Perim            -0.135         0.492   -0.117    0.739    0.456   -0.218
## from.Depth            -0.158         0.540   -0.059    0.422    0.332   -0.122
## from.pH                0.101         0.049   -0.211    0.655    0.451   -0.283
## from.Dforest          -0.282        -0.037    0.239   -0.698   -0.415    0.159
## from.Drock             0.326        -0.037   -0.183    0.208    0.245   -0.050
## from.Dshrub            1.000        -0.209   -0.111    0.089    0.143    0.025
## from.pwetland         -0.209         1.000    0.454    0.246    0.256   -0.188
## from.cti              -0.111         0.454    1.000   -0.169   -0.003   -0.034
## from.dd5               0.089         0.246   -0.169    1.000    0.850   -0.629
## from.ffp               0.143         0.256   -0.003    0.850    1.000   -0.798
## from.gsp               0.025        -0.188   -0.034   -0.629   -0.798    1.000
## from.pratio            0.219        -0.454   -0.736    0.232    0.081    0.083
## from.hli              -0.074         0.223    0.003    0.716    0.830   -0.856
## from.rough27           0.110        -0.632   -0.670   -0.156   -0.278    0.340
## from.srr               0.251        -0.771   -0.366   -0.258   -0.239    0.067
##                  from.pratio from.hli from.rough27 from.srr
## from.degree            0.109   -0.214        0.198    0.070
## from.betweenness      -0.037   -0.174        0.055   -0.031
## from.Elev              0.322   -0.640        0.607    0.530
## from.Length            0.167    0.376       -0.117   -0.400
## from.Area              0.172    0.338       -0.145   -0.481
## from.Perim             0.160    0.354       -0.158   -0.465
## from.Depth             0.118    0.144       -0.209   -0.369
## from.pH                0.132    0.453       -0.110   -0.019
## from.Dforest          -0.281   -0.229        0.046   -0.065
## from.Drock             0.194   -0.066        0.154    0.152
## from.Dshrub            0.219   -0.074        0.110    0.251
## from.pwetland         -0.454    0.223       -0.632   -0.771
## from.cti              -0.736    0.003       -0.670   -0.366
## from.dd5               0.232    0.716       -0.156   -0.258
## from.ffp               0.081    0.830       -0.278   -0.239
## from.gsp               0.083   -0.856        0.340    0.067
## from.pratio            1.000   -0.015        0.693    0.330
## from.hli              -0.015    1.000       -0.302   -0.227
## from.rough27           0.693   -0.302        1.000    0.273
## from.srr               0.330   -0.227        0.273    1.000</code></pre>
<div class="sourceCode" id="cb1114"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1114-1"><a href="WE-13.html#cb1114-1" aria-hidden="true" tabindex="-1"></a><span class="co">#pairs(nodecor.ln, pch=19, cex=0.50) </span></span></code></pre></div>
<div class="sourceCode" id="cb1115"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1115-1"><a href="WE-13.html#cb1115-1" aria-hidden="true" tabindex="-1"></a>edge.ln <span class="ot">&lt;-</span> dist.graph<span class="sc">@</span>data[,<span class="dv">10</span><span class="sc">:</span><span class="fu">length</span>(dist.graph<span class="sc">@</span>data)]</span>
<span id="cb1115-2"><a href="WE-13.html#cb1115-2" aria-hidden="true" tabindex="-1"></a>  <span class="cf">for</span>(i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">ncol</span>(edge.ln)) {</span>
<span id="cb1115-3"><a href="WE-13.html#cb1115-3" aria-hidden="true" tabindex="-1"></a>    edge.ln[,i] <span class="ot">&lt;-</span> <span class="fu">log</span>(edge.ln[,i] <span class="sc">-</span> (<span class="fu">min</span>(edge.ln[,i]) <span class="sc">-</span> <span class="dv">1</span>))</span>
<span id="cb1115-4"><a href="WE-13.html#cb1115-4" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb1115-5"><a href="WE-13.html#cb1115-5" aria-hidden="true" tabindex="-1"></a>edgecor.ln <span class="ot">&lt;-</span> <span class="fu">cor</span>(edge.ln, <span class="at">y =</span> <span class="cn">NULL</span>, </span>
<span id="cb1115-6"><a href="WE-13.html#cb1115-6" aria-hidden="true" tabindex="-1"></a>                  <span class="at">use =</span> <span class="st">&quot;complete.obs&quot;</span>, </span>
<span id="cb1115-7"><a href="WE-13.html#cb1115-7" aria-hidden="true" tabindex="-1"></a>                  <span class="at">method =</span> <span class="st">&quot;pearson&quot;</span>)</span>
<span id="cb1115-8"><a href="WE-13.html#cb1115-8" aria-hidden="true" tabindex="-1"></a><span class="fu">round</span>(edgecor.ln, <span class="dv">3</span>) </span></code></pre></div>
<pre><code>##                min.cti min.dd5 min.ffp min.gsp min.hli min.pratio min.rough27
## min.cti          1.000   0.523   0.606   0.114   0.233      0.594       0.102
## min.dd5          0.523   1.000   0.915  -0.301   0.502      0.868       0.177
## min.ffp          0.606   0.915   1.000  -0.288   0.498      0.912       0.180
## min.gsp          0.114  -0.301  -0.288   1.000  -0.001     -0.279       0.265
## min.hli          0.233   0.502   0.498  -0.001   1.000      0.520       0.347
## min.pratio       0.594   0.868   0.912  -0.279   0.520      1.000       0.191
## min.rough27      0.102   0.177   0.180   0.265   0.347      0.191       1.000
## min.srr          0.254   0.348   0.379   0.125   0.383      0.360       0.750
## mean.cti         0.264   0.106   0.127  -0.007  -0.055      0.117      -0.544
## mean.dd5         0.167   0.663   0.670  -0.671   0.362      0.637      -0.135
## mean.ffp         0.207   0.695   0.720  -0.706   0.380      0.684      -0.130
## mean.gsp        -0.356  -0.705  -0.776   0.625  -0.426     -0.801      -0.021
## mean.hli        -0.423  -0.171  -0.192  -0.045   0.150     -0.253       0.363
## mean.pratio      0.259   0.702   0.709  -0.655   0.449      0.806      -0.024
## mean.rough27    -0.570  -0.536  -0.589   0.128  -0.303     -0.686       0.141
## mean.srr         0.081   0.129   0.152  -0.159   0.023      0.099       0.343
## max.cti         -0.255  -0.275  -0.306  -0.148  -0.263     -0.314      -0.428
## max.dd5         -0.196   0.183   0.163  -0.790  -0.087      0.128      -0.384
## max.ffp         -0.215   0.188   0.162  -0.801  -0.083      0.121      -0.407
## max.gsp         -0.577  -0.760  -0.850   0.248  -0.456     -0.899      -0.150
## max.hli         -0.608  -0.391  -0.453  -0.163  -0.209     -0.507      -0.069
## max.pratio      -0.147   0.296   0.264  -0.887   0.005      0.322      -0.306
## max.rough27     -0.672  -0.540  -0.616  -0.017  -0.306     -0.687      -0.034
## max.srr         -0.268  -0.247  -0.263  -0.172  -0.208     -0.303      -0.030
## var.cti         -0.015   0.034   0.035  -0.091  -0.025     -0.027      -0.506
## var.dd5         -0.565  -0.527  -0.605  -0.314  -0.486     -0.636      -0.374
## var.ffp         -0.593  -0.565  -0.660  -0.327  -0.514     -0.666      -0.417
## var.gsp         -0.620  -0.672  -0.755  -0.174  -0.539     -0.775      -0.330
## var.hli         -0.499  -0.312  -0.346  -0.129  -0.176     -0.409       0.013
## var.pratio      -0.488  -0.572  -0.616  -0.308  -0.552     -0.610      -0.413
## var.rough27     -0.636  -0.500  -0.569  -0.040  -0.305     -0.641      -0.081
## var.srr         -0.031  -0.225  -0.240   0.028  -0.236     -0.199      -0.559
## median.cti       0.257   0.015   0.019   0.025  -0.157      0.053      -0.494
## median.dd5       0.136   0.590   0.609  -0.622   0.373      0.574      -0.114
## median.ffp       0.205   0.641   0.669  -0.659   0.388      0.641      -0.096
## median.gsp      -0.295  -0.658  -0.714   0.667  -0.405     -0.731      -0.011
## median.hli      -0.297  -0.020  -0.027  -0.085   0.285     -0.111       0.363
## median.pratio    0.236   0.650   0.655  -0.639   0.446      0.754      -0.019
## median.rough27  -0.474  -0.488  -0.525   0.193  -0.280     -0.600       0.266
## median.srr       0.099   0.138   0.160  -0.183  -0.006      0.105       0.233
## wet.pct.nlcd     0.174   0.159   0.202   0.021   0.011      0.145      -0.395
##                min.srr mean.cti mean.dd5 mean.ffp mean.gsp mean.hli mean.pratio
## min.cti          0.254    0.264    0.167    0.207   -0.356   -0.423       0.259
## min.dd5          0.348    0.106    0.663    0.695   -0.705   -0.171       0.702
## min.ffp          0.379    0.127    0.670    0.720   -0.776   -0.192       0.709
## min.gsp          0.125   -0.007   -0.671   -0.706    0.625   -0.045      -0.655
## min.hli          0.383   -0.055    0.362    0.380   -0.426    0.150       0.449
## min.pratio       0.360    0.117    0.637    0.684   -0.801   -0.253       0.806
## min.rough27      0.750   -0.544   -0.135   -0.130   -0.021    0.363      -0.024
## min.srr          1.000   -0.433    0.088    0.102   -0.200    0.242       0.129
## mean.cti        -0.433    1.000    0.141    0.162   -0.105   -0.485       0.133
## mean.dd5         0.088    0.141    1.000    0.972   -0.791   -0.078       0.848
## mean.ffp         0.102    0.162    0.972    1.000   -0.860   -0.089       0.889
## mean.gsp        -0.200   -0.105   -0.791   -0.860    1.000    0.091      -0.878
## mean.hli         0.242   -0.485   -0.078   -0.089    0.091    1.000      -0.140
## mean.pratio      0.129    0.133    0.848    0.889   -0.878   -0.140       1.000
## mean.rough27    -0.060   -0.417   -0.430   -0.469    0.576    0.562      -0.552
## mean.srr         0.560   -0.541    0.073    0.082   -0.104    0.160       0.041
## max.cti         -0.455    0.462   -0.046   -0.051    0.091    0.017      -0.090
## max.dd5         -0.282    0.163    0.659    0.672   -0.480    0.079       0.513
## max.ffp         -0.271    0.140    0.682    0.687   -0.476    0.074       0.513
## max.gsp         -0.320   -0.146   -0.577   -0.638    0.851    0.208      -0.680
## max.hli         -0.161   -0.134   -0.111   -0.120    0.219    0.538      -0.219
## max.pratio      -0.208    0.094    0.683    0.711   -0.579   -0.029       0.744
## max.rough27     -0.191   -0.270   -0.290   -0.326    0.491    0.453      -0.405
## max.srr          0.014   -0.312   -0.044   -0.062    0.153    0.151      -0.120
## var.cti         -0.432    0.757    0.151    0.162   -0.102   -0.155       0.057
## var.dd5         -0.469    0.037   -0.127   -0.144    0.339    0.154      -0.210
## var.ffp         -0.473   -0.005   -0.124   -0.156    0.403    0.082      -0.222
## var.gsp         -0.452   -0.040   -0.296   -0.327    0.517    0.170      -0.377
## var.hli         -0.058   -0.155   -0.111   -0.109    0.166    0.614      -0.208
## var.pratio      -0.450    0.011   -0.159   -0.184    0.354    0.044      -0.232
## var.rough27     -0.200   -0.149   -0.249   -0.278    0.424    0.454      -0.366
## var.srr         -0.718    0.353   -0.115   -0.126    0.165   -0.394      -0.097
## median.cti      -0.413    0.868    0.031    0.046    0.012   -0.549       0.082
## median.dd5       0.087    0.134    0.964    0.935   -0.748   -0.052       0.812
## median.ffp       0.104    0.155    0.931    0.951   -0.835   -0.091       0.868
## median.gsp      -0.192   -0.083   -0.805   -0.870    0.960    0.076      -0.886
## median.hli       0.260   -0.497    0.066    0.064   -0.022    0.741      -0.005
## median.pratio    0.121    0.126    0.836    0.872   -0.852   -0.139       0.980
## median.rough27   0.044   -0.489   -0.474   -0.515    0.549    0.619      -0.573
## median.srr       0.413   -0.455    0.079    0.100   -0.130    0.067       0.054
## wet.pct.nlcd    -0.321    0.648    0.127    0.141   -0.156   -0.346       0.060
##                mean.rough27 mean.srr max.cti max.dd5 max.ffp max.gsp max.hli
## min.cti              -0.570    0.081  -0.255  -0.196  -0.215  -0.577  -0.608
## min.dd5              -0.536    0.129  -0.275   0.183   0.188  -0.760  -0.391
## min.ffp              -0.589    0.152  -0.306   0.163   0.162  -0.850  -0.453
## min.gsp               0.128   -0.159  -0.148  -0.790  -0.801   0.248  -0.163
## min.hli              -0.303    0.023  -0.263  -0.087  -0.083  -0.456  -0.209
## min.pratio           -0.686    0.099  -0.314   0.128   0.121  -0.899  -0.507
## min.rough27           0.141    0.343  -0.428  -0.384  -0.407  -0.150  -0.069
## min.srr              -0.060    0.560  -0.455  -0.282  -0.271  -0.320  -0.161
## mean.cti             -0.417   -0.541   0.462   0.163   0.140  -0.146  -0.134
## mean.dd5             -0.430    0.073  -0.046   0.659   0.682  -0.577  -0.111
## mean.ffp             -0.469    0.082  -0.051   0.672   0.687  -0.638  -0.120
## mean.gsp              0.576   -0.104   0.091  -0.480  -0.476   0.851   0.219
## mean.hli              0.562    0.160   0.017   0.079   0.074   0.208   0.538
## mean.pratio          -0.552    0.041  -0.090   0.513   0.513  -0.680  -0.219
## mean.rough27          1.000    0.120   0.139  -0.078  -0.061   0.714   0.543
## mean.srr              0.120    1.000  -0.317  -0.046  -0.003  -0.056  -0.040
## max.cti               0.139   -0.317   1.000   0.373   0.319   0.256   0.426
## max.dd5              -0.078   -0.046   0.373   1.000   0.968  -0.156   0.301
## max.ffp              -0.061   -0.003   0.319   0.968   1.000  -0.147   0.312
## max.gsp               0.714   -0.056   0.256  -0.156  -0.147   1.000   0.441
## max.hli               0.543   -0.040   0.426   0.301   0.312   0.441   1.000
## max.pratio           -0.186    0.043   0.176   0.803   0.811  -0.222   0.124
## max.rough27           0.905    0.031   0.278   0.087   0.100   0.727   0.630
## max.srr               0.330    0.555   0.136   0.120   0.139   0.341   0.243
## var.cti              -0.109   -0.458   0.685   0.244   0.231  -0.040   0.141
## var.dd5               0.504   -0.103   0.575   0.504   0.481   0.669   0.552
## var.ffp               0.485   -0.055   0.409   0.443   0.459   0.717   0.493
## var.gsp               0.587   -0.073   0.482   0.324   0.319   0.804   0.552
## var.hli               0.565   -0.003   0.406   0.258   0.264   0.348   0.930
## var.pratio            0.379   -0.037   0.314   0.368   0.377   0.592   0.365
## var.rough27           0.802   -0.064   0.407   0.167   0.175   0.644   0.790
## var.srr              -0.023   -0.281   0.301   0.078   0.087   0.198  -0.059
## median.cti           -0.401   -0.474   0.297   0.104   0.077  -0.042  -0.186
## median.dd5           -0.395    0.062  -0.040   0.611   0.634  -0.520  -0.090
## median.ffp           -0.451    0.069  -0.041   0.606   0.618  -0.599  -0.112
## median.gsp            0.507   -0.147   0.069  -0.508  -0.508   0.727   0.176
## median.hli            0.367    0.197  -0.096   0.116   0.119   0.072   0.289
## median.pratio        -0.526    0.047  -0.076   0.502   0.500  -0.629  -0.197
## median.rough27        0.858    0.191   0.048  -0.160  -0.150   0.591   0.430
## median.srr            0.078    0.946  -0.304  -0.016   0.027  -0.084  -0.075
## wet.pct.nlcd         -0.311   -0.431   0.309   0.029   0.011  -0.223  -0.155
##                max.pratio max.rough27 max.srr var.cti var.dd5 var.ffp var.gsp
## min.cti            -0.147      -0.672  -0.268  -0.015  -0.565  -0.593  -0.620
## min.dd5             0.296      -0.540  -0.247   0.034  -0.527  -0.565  -0.672
## min.ffp             0.264      -0.616  -0.263   0.035  -0.605  -0.660  -0.755
## min.gsp            -0.887      -0.017  -0.172  -0.091  -0.314  -0.327  -0.174
## min.hli             0.005      -0.306  -0.208  -0.025  -0.486  -0.514  -0.539
## min.pratio          0.322      -0.687  -0.303  -0.027  -0.636  -0.666  -0.775
## min.rough27        -0.306      -0.034  -0.030  -0.506  -0.374  -0.417  -0.330
## min.srr            -0.208      -0.191   0.014  -0.432  -0.469  -0.473  -0.452
## mean.cti            0.094      -0.270  -0.312   0.757   0.037  -0.005  -0.040
## mean.dd5            0.683      -0.290  -0.044   0.151  -0.127  -0.124  -0.296
## mean.ffp            0.711      -0.326  -0.062   0.162  -0.144  -0.156  -0.327
## mean.gsp           -0.579       0.491   0.153  -0.102   0.339   0.403   0.517
## mean.hli           -0.029       0.453   0.151  -0.155   0.154   0.082   0.170
## mean.pratio         0.744      -0.405  -0.120   0.057  -0.210  -0.222  -0.377
## mean.rough27       -0.186       0.905   0.330  -0.109   0.504   0.485   0.587
## mean.srr            0.043       0.031   0.555  -0.458  -0.103  -0.055  -0.073
## max.cti             0.176       0.278   0.136   0.685   0.575   0.409   0.482
## max.dd5             0.803       0.087   0.120   0.244   0.504   0.443   0.324
## max.ffp             0.811       0.100   0.139   0.231   0.481   0.459   0.319
## max.gsp            -0.222       0.727   0.341  -0.040   0.669   0.717   0.804
## max.hli             0.124       0.630   0.243   0.141   0.552   0.493   0.552
## max.pratio          1.000       0.006   0.108   0.111   0.365   0.391   0.223
## max.rough27         0.006       1.000   0.388   0.009   0.625   0.613   0.683
## max.srr             0.108       0.388   1.000  -0.139   0.346   0.338   0.368
## var.cti             0.111       0.009  -0.139   1.000   0.170   0.077   0.077
## var.dd5             0.365       0.625   0.346   0.170   1.000   0.928   0.964
## var.ffp             0.391       0.613   0.338   0.077   0.928   1.000   0.955
## var.gsp             0.223       0.683   0.368   0.077   0.964   0.955   1.000
## var.hli             0.065       0.604   0.219   0.160   0.455   0.364   0.440
## var.pratio          0.385       0.472   0.234   0.056   0.777   0.888   0.828
## var.rough27         0.028       0.898   0.304   0.124   0.639   0.594   0.669
## var.srr             0.058       0.084   0.232   0.315   0.285   0.282   0.278
## median.cti          0.087      -0.259  -0.275   0.404   0.083   0.085   0.044
## median.dd5          0.626      -0.252  -0.033   0.141  -0.110  -0.112  -0.263
## median.ffp          0.656      -0.306  -0.056   0.153  -0.136  -0.156  -0.302
## median.gsp         -0.624       0.404   0.091  -0.089   0.234   0.288   0.404
## median.hli          0.015       0.259   0.127  -0.219   0.057   0.013   0.061
## median.pratio       0.716      -0.371  -0.098   0.050  -0.178  -0.191  -0.335
## median.rough27     -0.260       0.644   0.263  -0.189   0.357   0.334   0.454
## median.srr          0.074       0.000   0.494  -0.388  -0.099  -0.036  -0.072
## wet.pct.nlcd       -0.024      -0.237  -0.282   0.704  -0.164  -0.194  -0.226
##                var.hli var.pratio var.rough27 var.srr median.cti median.dd5
## min.cti         -0.499     -0.488      -0.636  -0.031      0.257      0.136
## min.dd5         -0.312     -0.572      -0.500  -0.225      0.015      0.590
## min.ffp         -0.346     -0.616      -0.569  -0.240      0.019      0.609
## min.gsp         -0.129     -0.308      -0.040   0.028      0.025     -0.622
## min.hli         -0.176     -0.552      -0.305  -0.236     -0.157      0.373
## min.pratio      -0.409     -0.610      -0.641  -0.199      0.053      0.574
## min.rough27      0.013     -0.413      -0.081  -0.559     -0.494     -0.114
## min.srr         -0.058     -0.450      -0.200  -0.718     -0.413      0.087
## mean.cti        -0.155      0.011      -0.149   0.353      0.868      0.134
## mean.dd5        -0.111     -0.159      -0.249  -0.115      0.031      0.964
## mean.ffp        -0.109     -0.184      -0.278  -0.126      0.046      0.935
## mean.gsp         0.166      0.354       0.424   0.165      0.012     -0.748
## mean.hli         0.614      0.044       0.454  -0.394     -0.549     -0.052
## mean.pratio     -0.208     -0.232      -0.366  -0.097      0.082      0.812
## mean.rough27     0.565      0.379       0.802  -0.023     -0.401     -0.395
## mean.srr        -0.003     -0.037      -0.064  -0.281     -0.474      0.062
## max.cti          0.406      0.314       0.407   0.301      0.297     -0.040
## max.dd5          0.258      0.368       0.167   0.078      0.104      0.611
## max.ffp          0.264      0.377       0.175   0.087      0.077      0.634
## max.gsp          0.348      0.592       0.644   0.198     -0.042     -0.520
## max.hli          0.930      0.365       0.790  -0.059     -0.186     -0.090
## max.pratio       0.065      0.385       0.028   0.058      0.087      0.626
## max.rough27      0.604      0.472       0.898   0.084     -0.259     -0.252
## max.srr          0.219      0.234       0.304   0.232     -0.275     -0.033
## var.cti          0.160      0.056       0.124   0.315      0.404      0.141
## var.dd5          0.455      0.777       0.639   0.285      0.083     -0.110
## var.ffp          0.364      0.888       0.594   0.282      0.085     -0.112
## var.gsp          0.440      0.828       0.669   0.278      0.044     -0.263
## var.hli          1.000      0.268       0.786  -0.110     -0.250     -0.103
## var.pratio       0.268      1.000       0.455   0.269      0.093     -0.156
## var.rough27      0.786      0.455       1.000   0.074     -0.173     -0.217
## var.srr         -0.110      0.269       0.074   1.000      0.342     -0.087
## median.cti      -0.250      0.093      -0.173   0.342      1.000      0.036
## median.dd5      -0.103     -0.156      -0.217  -0.087      0.036      1.000
## median.ffp      -0.115     -0.191      -0.261  -0.098      0.049      0.955
## median.gsp       0.136      0.271       0.349   0.158      0.019     -0.797
## median.hli       0.306     -0.021       0.240  -0.310     -0.536      0.137
## median.pratio   -0.196     -0.215      -0.336  -0.077      0.084      0.839
## median.rough27   0.468      0.299       0.578  -0.115     -0.473     -0.444
## median.srr      -0.051      0.003      -0.099  -0.142     -0.395      0.066
## wet.pct.nlcd    -0.124     -0.128      -0.130   0.257      0.391      0.111
##                median.ffp median.gsp median.hli median.pratio median.rough27
## min.cti             0.205     -0.295     -0.297         0.236         -0.474
## min.dd5             0.641     -0.658     -0.020         0.650         -0.488
## min.ffp             0.669     -0.714     -0.027         0.655         -0.525
## min.gsp            -0.659      0.667     -0.085        -0.639          0.193
## min.hli             0.388     -0.405      0.285         0.446         -0.280
## min.pratio          0.641     -0.731     -0.111         0.754         -0.600
## min.rough27        -0.096     -0.011      0.363        -0.019          0.266
## min.srr             0.104     -0.192      0.260         0.121          0.044
## mean.cti            0.155     -0.083     -0.497         0.126         -0.489
## mean.dd5            0.931     -0.805      0.066         0.836         -0.474
## mean.ffp            0.951     -0.870      0.064         0.872         -0.515
## mean.gsp           -0.835      0.960     -0.022        -0.852          0.549
## mean.hli           -0.091      0.076      0.741        -0.139          0.619
## mean.pratio         0.868     -0.886     -0.005         0.980         -0.573
## mean.rough27       -0.451      0.507      0.367        -0.526          0.858
## mean.srr            0.069     -0.147      0.197         0.047          0.191
## max.cti            -0.041      0.069     -0.096        -0.076          0.048
## max.dd5             0.606     -0.508      0.116         0.502         -0.160
## max.ffp             0.618     -0.508      0.119         0.500         -0.150
## max.gsp            -0.599      0.727      0.072        -0.629          0.591
## max.hli            -0.112      0.176      0.289        -0.197          0.430
## max.pratio          0.656     -0.624      0.015         0.716         -0.260
## max.rough27        -0.306      0.404      0.259        -0.371          0.644
## max.srr            -0.056      0.091      0.127        -0.098          0.263
## var.cti             0.153     -0.089     -0.219         0.050         -0.189
## var.dd5            -0.136      0.234      0.057        -0.178          0.357
## var.ffp            -0.156      0.288      0.013        -0.191          0.334
## var.gsp            -0.302      0.404      0.061        -0.335          0.454
## var.hli            -0.115      0.136      0.306        -0.196          0.468
## var.pratio         -0.191      0.271     -0.021        -0.215          0.299
## var.rough27        -0.261      0.349      0.240        -0.336          0.578
## var.srr            -0.098      0.158     -0.310        -0.077         -0.115
## median.cti          0.049      0.019     -0.536         0.084         -0.473
## median.dd5          0.955     -0.797      0.137         0.839         -0.444
## median.ffp          1.000     -0.884      0.084         0.895         -0.503
## median.gsp         -0.884      1.000     -0.047        -0.897          0.501
## median.hli          0.084     -0.047      1.000         0.016          0.436
## median.pratio       0.895     -0.897      0.016         1.000         -0.556
## median.rough27     -0.503      0.501      0.436        -0.556          1.000
## median.srr          0.083     -0.164      0.126         0.059          0.143
## wet.pct.nlcd        0.125     -0.116     -0.316         0.043         -0.334
##                median.srr wet.pct.nlcd
## min.cti             0.099        0.174
## min.dd5             0.138        0.159
## min.ffp             0.160        0.202
## min.gsp            -0.183        0.021
## min.hli            -0.006        0.011
## min.pratio          0.105        0.145
## min.rough27         0.233       -0.395
## min.srr             0.413       -0.321
## mean.cti           -0.455        0.648
## mean.dd5            0.079        0.127
## mean.ffp            0.100        0.141
## mean.gsp           -0.130       -0.156
## mean.hli            0.067       -0.346
## mean.pratio         0.054        0.060
## mean.rough27        0.078       -0.311
## mean.srr            0.946       -0.431
## max.cti            -0.304        0.309
## max.dd5            -0.016        0.029
## max.ffp             0.027        0.011
## max.gsp            -0.084       -0.223
## max.hli            -0.075       -0.155
## max.pratio          0.074       -0.024
## max.rough27         0.000       -0.237
## max.srr             0.494       -0.282
## var.cti            -0.388        0.704
## var.dd5            -0.099       -0.164
## var.ffp            -0.036       -0.194
## var.gsp            -0.072       -0.226
## var.hli            -0.051       -0.124
## var.pratio          0.003       -0.128
## var.rough27        -0.099       -0.130
## var.srr            -0.142        0.257
## median.cti         -0.395        0.391
## median.dd5          0.066        0.111
## median.ffp          0.083        0.125
## median.gsp         -0.164       -0.116
## median.hli          0.126       -0.316
## median.pratio       0.059        0.043
## median.rough27      0.143       -0.334
## median.srr          1.000       -0.363
## wet.pct.nlcd       -0.363        1.000</code></pre>
</div>
<div id="c.-write-out-resultss" class="section level4">
<h4>c. Write out resultss</h4>
<p>Figure as pdf file:</p>
<div class="sourceCode" id="cb1117"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1117-1"><a href="WE-13.html#cb1117-1" aria-hidden="true" tabindex="-1"></a><span class="fu">pdf</span>(<span class="at">file=</span><span class="fu">paste0</span>(here<span class="sc">::</span><span class="fu">here</span>(),<span class="st">&quot;/output/node.cor.pdf&quot;</span>), <span class="at">width=</span><span class="dv">20</span>, <span class="at">height=</span><span class="dv">20</span>)</span>
<span id="cb1117-2"><a href="WE-13.html#cb1117-2" aria-hidden="true" tabindex="-1"></a>   <span class="fu">pairs</span>(nodecor.ln, <span class="at">pch=</span><span class="dv">19</span>, <span class="at">cex=</span><span class="fl">0.50</span>)</span>
<span id="cb1117-3"><a href="WE-13.html#cb1117-3" aria-hidden="true" tabindex="-1"></a><span class="fu">dev.off</span>()</span></code></pre></div>
<pre><code>## png 
##   2</code></pre>
<p>Correlation data:</p>
<div class="sourceCode" id="cb1119"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1119-1"><a href="WE-13.html#cb1119-1" aria-hidden="true" tabindex="-1"></a><span class="fu">write.csv</span>(<span class="fu">round</span>(edgecor.ln, <span class="dv">4</span>), </span>
<span id="cb1119-2"><a href="WE-13.html#cb1119-2" aria-hidden="true" tabindex="-1"></a>          <span class="at">file =</span> <span class="fu">paste0</span>(here<span class="sc">::</span><span class="fu">here</span>(),<span class="st">&quot;/output/EdgeCorrelationsLn.csv&quot;</span>))</span>
<span id="cb1119-3"><a href="WE-13.html#cb1119-3" aria-hidden="true" tabindex="-1"></a><span class="fu">write.csv</span>(<span class="fu">round</span>(nodecor.ln, <span class="dv">4</span>), </span>
<span id="cb1119-4"><a href="WE-13.html#cb1119-4" aria-hidden="true" tabindex="-1"></a>          <span class="at">file =</span> <span class="fu">paste0</span>(here<span class="sc">::</span><span class="fu">here</span>(),<span class="st">&quot;/output/NodeCorrelationsLn.csv&quot;</span>))</span></code></pre></div>
</div>
<div id="e.-compare-models" class="section level4">
<h4>e. Compare models</h4>
<p>Run and compare models representing your hypotheses. Remember - we compete models using ML but use REML for final fit. Also remember that the null model contains distance (and all models must contain distance).</p>
<p>Null model:</p>
<div class="sourceCode" id="cb1120"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1120-1"><a href="WE-13.html#cb1120-1" aria-hidden="true" tabindex="-1"></a>( null <span class="ot">&lt;-</span> GeNetIt<span class="sc">::</span><span class="fu">gravity</span>(<span class="at">y =</span> <span class="st">&quot;GDIST&quot;</span>, <span class="at">x =</span> <span class="fu">c</span>(<span class="st">&quot;length&quot;</span>), <span class="at">d =</span> <span class="st">&quot;length&quot;</span>, <span class="at">group =</span> <span class="st">&quot;from_ID&quot;</span>, </span>
<span id="cb1120-2"><a href="WE-13.html#cb1120-2" aria-hidden="true" tabindex="-1"></a>                  <span class="at">data =</span> gdata, <span class="at">method =</span> <span class="st">&quot;ML&quot;</span>) )</span></code></pre></div>
<pre><code>## [1] &quot;Running singly-constrained gravity model&quot;</code></pre>
<pre><code>## Warning: invalid formula &quot;random = ~1 | from_ID&quot;: assignment is deprecated</code></pre>
<pre><code>## Gravity model
## 
## Linear mixed-effects model fit by maximum likelihood
##   Data: gdata 
##         AIC       BIC   logLik
##   -913.6676 -894.8696 460.8338
## 
## Random effects:
##  Formula: ~1 | from_ID
##         (Intercept)  Residual
## StdDev:  0.07766627 0.1314787
## 
## Fixed effects:  list(fmla) 
##                  Value  Std.Error  DF   t-value p-value
## (Intercept)  0.7820529 0.05054865 782  15.47129       0
## length      -0.1524624 0.00594561 782 -25.64286       0
##  Correlation: 
##        (Intr)
## length -0.954
## 
## Standardized Within-Group Residuals:
##         Min          Q1         Med          Q3         Max 
## -7.19945126 -0.52191436  0.09864931  0.60400073  3.04975798 
## 
## Number of Observations: 812
## Number of Groups: 29</code></pre>
<p>Global model (this was based on my hypotheses):</p>
<div class="sourceCode" id="cb1124"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1124-1"><a href="WE-13.html#cb1124-1" aria-hidden="true" tabindex="-1"></a>  ( global <span class="ot">&lt;-</span> GeNetIt<span class="sc">::</span><span class="fu">gravity</span>(<span class="at">y =</span> <span class="st">&quot;GDIST&quot;</span>, <span class="at">x =</span> <span class="fu">c</span>(<span class="st">&quot;length&quot;</span>, <span class="st">&quot;wet.pct.nlcd&quot;</span>, </span>
<span id="cb1124-2"><a href="WE-13.html#cb1124-2" aria-hidden="true" tabindex="-1"></a>                                         <span class="st">&quot;median.gsp&quot;</span>, <span class="st">&quot;from.Depth&quot;</span>, </span>
<span id="cb1124-3"><a href="WE-13.html#cb1124-3" aria-hidden="true" tabindex="-1"></a>                                         <span class="st">&quot;from.ffp&quot;</span>, <span class="st">&quot;from.hli&quot;</span>, <span class="st">&quot;from.pratio&quot;</span>, </span>
<span id="cb1124-4"><a href="WE-13.html#cb1124-4" aria-hidden="true" tabindex="-1"></a>                                         <span class="st">&quot;from.degree&quot;</span>, <span class="st">&quot;from.betweenness&quot;</span>, </span>
<span id="cb1124-5"><a href="WE-13.html#cb1124-5" aria-hidden="true" tabindex="-1"></a>                                         <span class="st">&quot;from.pwetland&quot;</span>, <span class="st">&quot;median.srr&quot;</span>, </span>
<span id="cb1124-6"><a href="WE-13.html#cb1124-6" aria-hidden="true" tabindex="-1"></a>                                         <span class="st">&quot;median.rough27&quot;</span>), <span class="at">d =</span> <span class="st">&quot;length&quot;</span>, </span>
<span id="cb1124-7"><a href="WE-13.html#cb1124-7" aria-hidden="true" tabindex="-1"></a>                      <span class="at">group =</span> <span class="st">&quot;from_ID&quot;</span>, <span class="at">data =</span> gdata, <span class="at">method =</span> <span class="st">&quot;ML&quot;</span>) )</span></code></pre></div>
<pre><code>## [1] &quot;Running singly-constrained gravity model&quot;</code></pre>
<pre><code>## Warning: invalid formula &quot;random = ~1 | from_ID&quot;: assignment is deprecated</code></pre>
<pre><code>## Gravity model
## 
## Linear mixed-effects model fit by maximum likelihood
##   Data: gdata 
##         AIC       BIC   logLik
##   -1007.525 -937.0325 518.7625
## 
## Random effects:
##  Formula: ~1 | from_ID
##         (Intercept)  Residual
## StdDev:  0.06165765 0.1230698
## 
## Fixed effects:  list(fmla) 
##                      Value Std.Error  DF    t-value p-value
## (Intercept)      10.531419  3.128409 778   3.366382  0.0008
## length           -0.131565  0.006892 778 -19.090228  0.0000
## wet.pct.nlcd      0.002516  0.002910 778   0.864492  0.3876
## median.gsp       -1.873423  0.248176 778  -7.548755  0.0000
## from.Depth        0.002233  0.012909  21   0.172981  0.8643
## from.ffp          0.067345  0.083928  21   0.802407  0.4313
## from.hli         -1.480776  3.332751  21  -0.444310  0.6614
## from.pratio      -0.060628  0.040288  21  -1.504873  0.1472
## from.degree       0.049009  0.097357  21   0.503400  0.6199
## from.betweenness -0.008488  0.015550  21  -0.545856  0.5909
## from.pwetland    -0.004191  0.007459  21  -0.561924  0.5801
## median.srr        0.124025  0.045645 778   2.717134  0.0067
## median.rough27   -0.018661  0.010294 778  -1.812790  0.0702
##  Correlation: 
##                  (Intr) length wt.pc. mdn.gs frm.Dp frm.ff frm.hl frm.pr frm.dg
## length           -0.040                                                        
## wet.pct.nlcd     -0.040  0.467                                                 
## median.gsp       -0.439  0.003  0.027                                          
## from.Depth        0.211 -0.004 -0.006 -0.055                                   
## from.ffp         -0.818  0.019  0.021  0.076 -0.284                            
## from.hli          0.889 -0.030 -0.026  0.008  0.192 -0.865                     
## from.pratio      -0.068  0.006 -0.026  0.055 -0.149 -0.074  0.048              
## from.degree      -0.059  0.046  0.046  0.054 -0.020  0.070 -0.011 -0.116       
## from.betweenness  0.202 -0.048 -0.037 -0.066  0.119 -0.198  0.165  0.019 -0.875
## from.pwetland    -0.041 -0.004  0.037  0.038 -0.200 -0.004 -0.014  0.170  0.106
## median.srr       -0.025  0.052 -0.062  0.131  0.032 -0.026  0.022  0.003  0.022
## median.rough27    0.265 -0.319 -0.074 -0.542  0.012 -0.045  0.035 -0.062 -0.062
##                  frm.bt frm.pw mdn.sr
## length                               
## wet.pct.nlcd                         
## median.gsp                           
## from.Depth                           
## from.ffp                             
## from.hli                             
## from.pratio                          
## from.degree                          
## from.betweenness                     
## from.pwetland    -0.155              
## median.srr       -0.032 -0.045       
## median.rough27    0.058 -0.019 -0.185
## 
## Standardized Within-Group Residuals:
##        Min         Q1        Med         Q3        Max 
## -7.5986170 -0.5245075  0.0551797  0.5560370  3.9428671 
## 
## Number of Observations: 812
## Number of Groups: 29</code></pre>
<p>Published model:</p>
<div class="sourceCode" id="cb1128"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1128-1"><a href="WE-13.html#cb1128-1" aria-hidden="true" tabindex="-1"></a>( published <span class="ot">&lt;-</span> GeNetIt<span class="sc">::</span><span class="fu">gravity</span>(<span class="at">y =</span> <span class="st">&quot;GDIST&quot;</span>, <span class="at">x =</span> <span class="fu">c</span>(<span class="st">&quot;length&quot;</span>, <span class="st">&quot;median.gsp&quot;</span>, <span class="st">&quot;from.Depth&quot;</span>, </span>
<span id="cb1128-2"><a href="WE-13.html#cb1128-2" aria-hidden="true" tabindex="-1"></a>                    <span class="st">&quot;from.hli&quot;</span>, <span class="st">&quot;median.cti&quot;</span>, <span class="st">&quot;median.srr&quot;</span>), <span class="at">d =</span> <span class="st">&quot;length&quot;</span>, </span>
<span id="cb1128-3"><a href="WE-13.html#cb1128-3" aria-hidden="true" tabindex="-1"></a>                     <span class="at">group =</span> <span class="st">&quot;from_ID&quot;</span>, <span class="at">data =</span> gdata, <span class="at">method =</span> <span class="st">&quot;ML&quot;</span>))</span></code></pre></div>
<pre><code>## [1] &quot;Running singly-constrained gravity model&quot;</code></pre>
<pre><code>## Warning: invalid formula &quot;random = ~1 | from_ID&quot;: assignment is deprecated</code></pre>
<pre><code>## Gravity model
## 
## Linear mixed-effects model fit by maximum likelihood
##   Data: gdata 
##         AIC       BIC   logLik
##   -1012.517 -970.2212 515.2583
## 
## Random effects:
##  Formula: ~1 | from_ID
##         (Intercept)  Residual
## StdDev:  0.06480527 0.1234197
## 
## Fixed effects:  list(fmla) 
##                 Value Std.Error  DF    t-value p-value
## (Intercept) 13.349985 1.6618555 779   8.033180  0.0000
## length      -0.137964 0.0057632 779 -23.938869  0.0000
## median.gsp  -2.112507 0.2093307 779 -10.091719  0.0000
## from.Depth   0.001318 0.0124098  26   0.106217  0.9162
## from.hli     0.790610 1.7086186  26   0.462719  0.6474
## median.cti   0.001195 0.0629880 779   0.018973  0.9849
## median.srr   0.109397 0.0488379 779   2.240004  0.0254
##  Correlation: 
##            (Intr) length mdn.gs frm.Dp frm.hl mdn.ct
## length      0.112                                   
## median.gsp -0.575 -0.227                            
## from.Depth -0.104 -0.002 -0.032                     
## from.hli    0.709 -0.025  0.166 -0.143              
## median.cti -0.142  0.037  0.115 -0.017 -0.010       
## median.srr -0.064  0.042  0.081  0.011 -0.006  0.408
## 
## Standardized Within-Group Residuals:
##        Min         Q1        Med         Q3        Max 
## -7.5866483 -0.5091002  0.0666032  0.5547811  3.9918695 
## 
## Number of Observations: 812
## Number of Groups: 29</code></pre>
<p>Habitat hypothesis</p>
<div class="sourceCode" id="cb1132"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1132-1"><a href="WE-13.html#cb1132-1" aria-hidden="true" tabindex="-1"></a>( habitat <span class="ot">&lt;-</span> GeNetIt<span class="sc">::</span><span class="fu">gravity</span>(<span class="at">y =</span> <span class="st">&quot;GDIST&quot;</span>, <span class="at">x =</span> <span class="fu">c</span>(<span class="st">&quot;length&quot;</span>, <span class="st">&quot;wet.pct.nlcd&quot;</span>, <span class="st">&quot;median.gsp&quot;</span>), <span class="at">d =</span> <span class="st">&quot;length&quot;</span>, </span>
<span id="cb1132-2"><a href="WE-13.html#cb1132-2" aria-hidden="true" tabindex="-1"></a>                     <span class="at">group =</span> <span class="st">&quot;from_ID&quot;</span>, <span class="at">data =</span> gdata, <span class="at">method =</span> <span class="st">&quot;ML&quot;</span>) )</span></code></pre></div>
<pre><code>## [1] &quot;Running singly-constrained gravity model&quot;</code></pre>
<pre><code>## Warning: invalid formula &quot;random = ~1 | from_ID&quot;: assignment is deprecated</code></pre>
<pre><code>## Gravity model
## 
## Linear mixed-effects model fit by maximum likelihood
##   Data: gdata 
##         AIC       BIC  logLik
##   -1013.066 -984.8689 512.533
## 
## Random effects:
##  Formula: ~1 | from_ID
##         (Intercept)  Residual
## StdDev:  0.06630675 0.1237554
## 
## Fixed effects:  list(fmla) 
##                  Value Std.Error  DF    t-value p-value
## (Intercept)  12.820217 1.1520073 780  11.128590  0.0000
## length       -0.135641 0.0065255 780 -20.786371  0.0000
## wet.pct.nlcd  0.002538 0.0028896 780   0.878297  0.3801
## median.gsp   -2.149016 0.2052203 780 -10.471750  0.0000
##  Correlation: 
##              (Intr) length wt.pc.
## length        0.171              
## wet.pct.nlcd -0.005  0.470       
## median.gsp   -0.999 -0.212 -0.011
## 
## Standardized Within-Group Residuals:
##         Min          Q1         Med          Q3         Max 
## -7.46729705 -0.49570850  0.07437762  0.55168658  3.94762511 
## 
## Number of Observations: 812
## Number of Groups: 29</code></pre>
<p>Compare models: these are the names of the hypotheses (models) I tested.</p>
<div class="sourceCode" id="cb1136"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1136-1"><a href="WE-13.html#cb1136-1" aria-hidden="true" tabindex="-1"></a><span class="co">#compare.models(null, depth, product, climate, wetlands, topo, habitat, global)</span></span>
<span id="cb1136-2"><a href="WE-13.html#cb1136-2" aria-hidden="true" tabindex="-1"></a><span class="co">#compare.models(depth, product, climate, wetlands, topo, habitat, published, global, null) </span></span>
<span id="cb1136-3"><a href="WE-13.html#cb1136-3" aria-hidden="true" tabindex="-1"></a>GeNetIt<span class="sc">::</span><span class="fu">compare.models</span>(null, habitat, global, published) <span class="co">#NOTE - global will need to be edited to match your paramters</span></span></code></pre></div>
<pre><code>##       model        AIC       BIC log.likelihood   RMSE nparms   deltaAIC
## 1      null  -913.6676 -894.8696       460.8338 0.1415      0 99.3983240
## 2   habitat -1013.0659 -984.8689       512.5330 0.1314      3  0.0000000
## 3    global -1007.5251 -937.0325       518.7625 0.1297     12  5.5408783
## 4 published -1012.5167 -970.2212       515.2583 0.1307      6  0.5492757
##   deltaBIC
## 1 89.99932
## 2  0.00000
## 3 47.83638
## 4 14.64778</code></pre>
</div>
<div id="f.-diagnostic-plots" class="section level4">
<h4>f. Diagnostic plots</h4>
<div class="sourceCode" id="cb1138"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1138-1"><a href="WE-13.html#cb1138-1" aria-hidden="true" tabindex="-1"></a><span class="fu">par</span>(<span class="at">mfrow=</span><span class="fu">c</span>(<span class="dv">2</span>,<span class="dv">3</span>))</span>
<span id="cb1138-2"><a href="WE-13.html#cb1138-2" aria-hidden="true" tabindex="-1"></a>   <span class="cf">for</span> (i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="dv">6</span>) { <span class="fu">plot</span>(global, <span class="at">type=</span>i) } </span></code></pre></div>
<pre><code>## Warning in bxp(list(stats = structure(c(-3.01594610828498, -1.23135461837275, :
## some notches went outside hinges (&#39;box&#39;): maybe set notch=FALSE</code></pre>
<p><img src="13-Week13_files/figure-html/unnamed-chunk-53-1.png" width="672" /></p>
</div>
<div id="g.-fit-final-models" class="section level4">
<h4>g. Fit final model(s)</h4>
<p>Habitat:</p>
<div class="sourceCode" id="cb1140"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1140-1"><a href="WE-13.html#cb1140-1" aria-hidden="true" tabindex="-1"></a>habitat_fit <span class="ot">&lt;-</span> GeNetIt<span class="sc">::</span><span class="fu">gravity</span>(<span class="at">y =</span> <span class="st">&quot;GDIST&quot;</span>, <span class="at">x =</span> <span class="fu">c</span>(<span class="st">&quot;length&quot;</span>, <span class="st">&quot;wet.pct.nlcd&quot;</span>, <span class="st">&quot;median.gsp&quot;</span>), </span>
<span id="cb1140-2"><a href="WE-13.html#cb1140-2" aria-hidden="true" tabindex="-1"></a>                       <span class="at">d =</span> <span class="st">&quot;length&quot;</span>, <span class="at">group =</span> <span class="st">&quot;from_ID&quot;</span>, <span class="at">data =</span> gdata, <span class="at">method =</span> <span class="st">&quot;REML&quot;</span>)</span></code></pre></div>
<pre><code>## [1] &quot;Running singly-constrained gravity model&quot;</code></pre>
<pre><code>## Warning: invalid formula &quot;random = ~1 | from_ID&quot;: assignment is deprecated</code></pre>
<p>Global:</p>
<div class="sourceCode" id="cb1143"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1143-1"><a href="WE-13.html#cb1143-1" aria-hidden="true" tabindex="-1"></a>global_fit <span class="ot">&lt;-</span> GeNetIt<span class="sc">::</span><span class="fu">gravity</span>(<span class="at">y =</span> <span class="st">&quot;GDIST&quot;</span>, <span class="at">x =</span> <span class="fu">c</span>(<span class="st">&quot;length&quot;</span>, <span class="st">&quot;wet.pct.nlcd&quot;</span>, <span class="st">&quot;median.gsp&quot;</span>, </span>
<span id="cb1143-2"><a href="WE-13.html#cb1143-2" aria-hidden="true" tabindex="-1"></a>                                         <span class="st">&quot;from.Depth&quot;</span>, <span class="st">&quot;from.ffp&quot;</span>, <span class="st">&quot;from.hli&quot;</span>, </span>
<span id="cb1143-3"><a href="WE-13.html#cb1143-3" aria-hidden="true" tabindex="-1"></a>                                         <span class="st">&quot;from.pratio&quot;</span>, <span class="st">&quot;from.degree&quot;</span>,  </span>
<span id="cb1143-4"><a href="WE-13.html#cb1143-4" aria-hidden="true" tabindex="-1"></a>                                         <span class="st">&quot;from.betweenness&quot;</span>, <span class="st">&quot;from.pwetland&quot;</span>, <span class="st">&quot;median.srr&quot;</span>, </span>
<span id="cb1143-5"><a href="WE-13.html#cb1143-5" aria-hidden="true" tabindex="-1"></a>                                         <span class="st">&quot;median.rough27&quot;</span>), </span>
<span id="cb1143-6"><a href="WE-13.html#cb1143-6" aria-hidden="true" tabindex="-1"></a>                      <span class="at">d =</span> <span class="st">&quot;length&quot;</span>, <span class="at">group =</span> <span class="st">&quot;from_ID&quot;</span>, <span class="at">data =</span> gdata, <span class="at">method =</span> <span class="st">&quot;REML&quot;</span>)</span></code></pre></div>
<pre><code>## [1] &quot;Running singly-constrained gravity model&quot;</code></pre>
<pre><code>## Warning: invalid formula &quot;random = ~1 | from_ID&quot;: assignment is deprecated</code></pre>
<p>Published:</p>
<div class="sourceCode" id="cb1146"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1146-1"><a href="WE-13.html#cb1146-1" aria-hidden="true" tabindex="-1"></a>published_fit <span class="ot">&lt;-</span> GeNetIt<span class="sc">::</span><span class="fu">gravity</span>(<span class="at">y =</span> <span class="st">&quot;GDIST&quot;</span>, <span class="at">x =</span> <span class="fu">c</span>(<span class="st">&quot;length&quot;</span>, <span class="st">&quot;median.gsp&quot;</span>, <span class="st">&quot;from.Depth&quot;</span>, </span>
<span id="cb1146-2"><a href="WE-13.html#cb1146-2" aria-hidden="true" tabindex="-1"></a>                    <span class="st">&quot;from.hli&quot;</span>, <span class="st">&quot;median.cti&quot;</span>, <span class="st">&quot;median.srr&quot;</span>), <span class="at">d =</span> <span class="st">&quot;length&quot;</span>, </span>
<span id="cb1146-3"><a href="WE-13.html#cb1146-3" aria-hidden="true" tabindex="-1"></a>                     <span class="at">group =</span> <span class="st">&quot;from_ID&quot;</span>, <span class="at">data =</span> gdata, <span class="at">method =</span> <span class="st">&quot;REML&quot;</span>) </span></code></pre></div>
<pre><code>## [1] &quot;Running singly-constrained gravity model&quot;</code></pre>
<pre><code>## Warning: invalid formula &quot;random = ~1 | from_ID&quot;: assignment is deprecated</code></pre>
<p>Compare models</p>
<div class="sourceCode" id="cb1149"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1149-1"><a href="WE-13.html#cb1149-1" aria-hidden="true" tabindex="-1"></a>GeNetIt<span class="sc">::</span><span class="fu">compare.models</span>(global_fit, habitat_fit, published_fit)</span></code></pre></div>
<pre><code>## Warning in GeNetIt::compare.models(global_fit, habitat_fit, published_fit): AIC/
## BIC not valid under REML and will not be reported</code></pre>
<pre><code>##           model log.likelihood   RMSE nparms
## 1    global_fit       484.6774 0.1296     12
## 2   habitat_fit       499.2604 0.1314      3
## 3 published_fit       500.8846 0.1307      6</code></pre>
</div>
<div id="h.-effect-size" class="section level4">
<h4>h. Effect size</h4>
<p>This effect size is NOT backwards transformed. We are working on effect sizes for backwards transformed data.</p>
<div class="sourceCode" id="cb1152"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1152-1"><a href="WE-13.html#cb1152-1" aria-hidden="true" tabindex="-1"></a>GeNetIt<span class="sc">::</span><span class="fu">gravity.es</span>(habitat_fit)</span></code></pre></div>
<pre><code>##                  t.value  df    cohen.d  p.value      low.ci     up.ci
## length       -20.7995890 780 -1.4894895 0.000000 -1.60183164 -1.377147
## wet.pct.nlcd   0.8787948 780  0.0629318 0.379783 -0.03648035  0.162344
## median.gsp   -10.4619997 780 -0.7491994 0.000000 -0.85201875 -0.646380</code></pre>
<div class="sourceCode" id="cb1154"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1154-1"><a href="WE-13.html#cb1154-1" aria-hidden="true" tabindex="-1"></a>GeNetIt<span class="sc">::</span><span class="fu">gravity.es</span>(global_fit)</span></code></pre></div>
<pre><code>##                      t.value  df     cohen.d  p.value      low.ci       up.ci
## length           -19.1893863 778 -1.37594566 0.000000 -1.48662548 -1.26526585
## wet.pct.nlcd       0.8679313 778  0.06223369 0.385700 -0.03730598  0.16177336
## median.gsp        -7.5059572 778 -0.53820321 0.000000 -0.63950664 -0.43689977
## from.Depth         0.1445143  21  0.06307123 0.886472 -0.57621575  0.70235821
## from.ffp           0.6926126  21  0.30228094 0.496141 -0.34066022  0.94522210
## from.hli          -0.3825715  21 -0.16696791 0.705882 -0.80725628  0.47332045
## from.pratio       -1.2852015  21 -0.56090791 0.212716 -1.21309035  0.09127454
## from.degree        0.4333201  21  0.18911641 0.669201 -0.45150205  0.82973488
## from.betweenness  -0.4691536  21 -0.20475543 0.643794 -0.84563159  0.43612074
## from.pwetland     -0.4765597  21 -0.20798770 0.638597 -0.84891966  0.43294427
## median.srr         2.6715256 778  0.19155767 0.007709  0.09181385  0.29130148
## median.rough27    -1.8697707 778 -0.13406905 0.061891 -0.23369648 -0.03444162</code></pre>
<div class="sourceCode" id="cb1156"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1156-1"><a href="WE-13.html#cb1156-1" aria-hidden="true" tabindex="-1"></a>GeNetIt<span class="sc">::</span><span class="fu">gravity.es</span>(published_fit)</span></code></pre></div>
<pre><code>##                 t.value  df      cohen.d  p.value      low.ci      up.ci
## length     -23.98326001 779 -1.718578929 0.000000 -1.83496961 -1.6021882
## median.gsp -10.08366298 779 -0.722569439 0.000000 -0.82521894 -0.6199199
## from.Depth   0.10013905  26  0.039277765 0.921002 -0.52888551  0.6074410
## from.hli     0.44108308  26  0.173007017 0.662799 -0.39620349  0.7422175
## median.cti   0.03131321 779  0.002243825 0.975028 -0.09720768  0.1016953
## median.srr   2.22992302 779  0.159790567 0.026037  0.06018031  0.2594008</code></pre>
<p><strong>Question:</strong> What did you learn?</p>
<ul>
<li>About Columbia spotted frogs in central Idaho?</li>
<li>About linking genetic and landscape data with graphs and gravity models?</li>
<li>About R programming?</li>
</ul>

</div>
</div>
</div>
<!-- </div> -->
            </section>

          </div>
        </div>
      </div>
<a href="Week13.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="Week14.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": true,
"facebook": false,
"twitter": false,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": false
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": null,
"toc": {
"collapse": "section"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
