<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Worked Example | Landscape Genetic Data Analysis with R</title>
  <meta name="description" content="This is a web-interface to the teaching materials for the lab course ‘Landscape Genetic Data Analysis with R’ associated with the distributed graduate course ‘DGS Landscape Genetics’. The output format is bookdown::gitbook." />
  <meta name="generator" content="bookdown 0.22 and GitBook 2.6.7" />

  <meta property="og:title" content="Worked Example | Landscape Genetic Data Analysis with R" />
  <meta property="og:type" content="book" />
  
  
  <meta property="og:description" content="This is a web-interface to the teaching materials for the lab course ‘Landscape Genetic Data Analysis with R’ associated with the distributed graduate course ‘DGS Landscape Genetics’. The output format is bookdown::gitbook." />
  <meta name="github-repo" content="hhwagner1/LandGenCourse_book" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Worked Example | Landscape Genetic Data Analysis with R" />
  
  <meta name="twitter:description" content="This is a web-interface to the teaching materials for the lab course ‘Landscape Genetic Data Analysis with R’ associated with the distributed graduate course ‘DGS Landscape Genetics’. The output format is bookdown::gitbook." />
  

<meta name="author" content="Editor: Helene Wagner (University of Toronto)" />


<meta name="date" content="2021-05-04" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="tutorial-2.html"/>
<link rel="next" href="r-exercise-week-2.html"/>
<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />









<link href="libs/anchor-sections-1.0.1/anchor-sections.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.0.1/anchor-sections.js"></script>
<script src="libs/kePrint-0.0.1/kePrint.js"></script>
<link href="libs/lightable-0.0.1/lightable.css" rel="stylesheet" />

<script>
/* ========================================================================
 * Bootstrap: transition.js v3.3.7
 * http://getbootstrap.com/javascript/#transitions
 * ========================================================================
 * Copyright 2011-2016 Twitter, Inc.
 * Licensed under MIT (https://github.com/twbs/bootstrap/blob/master/LICENSE)
 * ======================================================================== */


+function ($) {
  'use strict';

  // CSS TRANSITION SUPPORT (Shoutout: http://www.modernizr.com/)
  // ============================================================

  function transitionEnd() {
    var el = document.createElement('bootstrap')

    var transEndEventNames = {
      WebkitTransition : 'webkitTransitionEnd',
      MozTransition    : 'transitionend',
      OTransition      : 'oTransitionEnd otransitionend',
      transition       : 'transitionend'
    }

    for (var name in transEndEventNames) {
      if (el.style[name] !== undefined) {
        return { end: transEndEventNames[name] }
      }
    }

    return false // explicit for ie8 (  ._.)
  }

  // http://blog.alexmaccaw.com/css-transitions
  $.fn.emulateTransitionEnd = function (duration) {
    var called = false
    var $el = this
    $(this).one('bsTransitionEnd', function () { called = true })
    var callback = function () { if (!called) $($el).trigger($.support.transition.end) }
    setTimeout(callback, duration)
    return this
  }

  $(function () {
    $.support.transition = transitionEnd()

    if (!$.support.transition) return

    $.event.special.bsTransitionEnd = {
      bindType: $.support.transition.end,
      delegateType: $.support.transition.end,
      handle: function (e) {
        if ($(e.target).is(this)) return e.handleObj.handler.apply(this, arguments)
      }
    }
  })

}(jQuery);
</script>
<script>
/* ========================================================================
 * Bootstrap: collapse.js v3.3.7
 * http://getbootstrap.com/javascript/#collapse
 * ========================================================================
 * Copyright 2011-2016 Twitter, Inc.
 * Licensed under MIT (https://github.com/twbs/bootstrap/blob/master/LICENSE)
 * ======================================================================== */

/* jshint latedef: false */

+function ($) {
  'use strict';

  // COLLAPSE PUBLIC CLASS DEFINITION
  // ================================

  var Collapse = function (element, options) {
    this.$element      = $(element)
    this.options       = $.extend({}, Collapse.DEFAULTS, options)
    this.$trigger      = $('[data-toggle="collapse"][href="#' + element.id + '"],' +
                           '[data-toggle="collapse"][data-target="#' + element.id + '"]')
    this.transitioning = null

    if (this.options.parent) {
      this.$parent = this.getParent()
    } else {
      this.addAriaAndCollapsedClass(this.$element, this.$trigger)
    }

    if (this.options.toggle) this.toggle()
  }

  Collapse.VERSION  = '3.3.7'

  Collapse.TRANSITION_DURATION = 350

  Collapse.DEFAULTS = {
    toggle: true
  }

  Collapse.prototype.dimension = function () {
    var hasWidth = this.$element.hasClass('width')
    return hasWidth ? 'width' : 'height'
  }

  Collapse.prototype.show = function () {
    if (this.transitioning || this.$element.hasClass('in')) return

    var activesData
    var actives = this.$parent && this.$parent.children('.panel').children('.in, .collapsing')

    if (actives && actives.length) {
      activesData = actives.data('bs.collapse')
      if (activesData && activesData.transitioning) return
    }

    var startEvent = $.Event('show.bs.collapse')
    this.$element.trigger(startEvent)
    if (startEvent.isDefaultPrevented()) return

    if (actives && actives.length) {
      Plugin.call(actives, 'hide')
      activesData || actives.data('bs.collapse', null)
    }

    var dimension = this.dimension()

    this.$element
      .removeClass('collapse')
      .addClass('collapsing')[dimension](0)
      .attr('aria-expanded', true)

    this.$trigger
      .removeClass('collapsed')
      .attr('aria-expanded', true)

    this.transitioning = 1

    var complete = function () {
      this.$element
        .removeClass('collapsing')
        .addClass('collapse in')[dimension]('')
      this.transitioning = 0
      this.$element
        .trigger('shown.bs.collapse')
    }

    if (!$.support.transition) return complete.call(this)

    var scrollSize = $.camelCase(['scroll', dimension].join('-'))

    this.$element
      .one('bsTransitionEnd', $.proxy(complete, this))
      .emulateTransitionEnd(Collapse.TRANSITION_DURATION)[dimension](this.$element[0][scrollSize])
  }

  Collapse.prototype.hide = function () {
    if (this.transitioning || !this.$element.hasClass('in')) return

    var startEvent = $.Event('hide.bs.collapse')
    this.$element.trigger(startEvent)
    if (startEvent.isDefaultPrevented()) return

    var dimension = this.dimension()

    this.$element[dimension](this.$element[dimension]())[0].offsetHeight

    this.$element
      .addClass('collapsing')
      .removeClass('collapse in')
      .attr('aria-expanded', false)

    this.$trigger
      .addClass('collapsed')
      .attr('aria-expanded', false)

    this.transitioning = 1

    var complete = function () {
      this.transitioning = 0
      this.$element
        .removeClass('collapsing')
        .addClass('collapse')
        .trigger('hidden.bs.collapse')
    }

    if (!$.support.transition) return complete.call(this)

    this.$element
      [dimension](0)
      .one('bsTransitionEnd', $.proxy(complete, this))
      .emulateTransitionEnd(Collapse.TRANSITION_DURATION)
  }

  Collapse.prototype.toggle = function () {
    this[this.$element.hasClass('in') ? 'hide' : 'show']()
  }

  Collapse.prototype.getParent = function () {
    return $(this.options.parent)
      .find('[data-toggle="collapse"][data-parent="' + this.options.parent + '"]')
      .each($.proxy(function (i, element) {
        var $element = $(element)
        this.addAriaAndCollapsedClass(getTargetFromTrigger($element), $element)
      }, this))
      .end()
  }

  Collapse.prototype.addAriaAndCollapsedClass = function ($element, $trigger) {
    var isOpen = $element.hasClass('in')

    $element.attr('aria-expanded', isOpen)
    $trigger
      .toggleClass('collapsed', !isOpen)
      .attr('aria-expanded', isOpen)
  }

  function getTargetFromTrigger($trigger) {
    var href
    var target = $trigger.attr('data-target')
      || (href = $trigger.attr('href')) && href.replace(/.*(?=#[^\s]+$)/, '') // strip for ie7

    return $(target)
  }


  // COLLAPSE PLUGIN DEFINITION
  // ==========================

  function Plugin(option) {
    return this.each(function () {
      var $this   = $(this)
      var data    = $this.data('bs.collapse')
      var options = $.extend({}, Collapse.DEFAULTS, $this.data(), typeof option == 'object' && option)

      if (!data && options.toggle && /show|hide/.test(option)) options.toggle = false
      if (!data) $this.data('bs.collapse', (data = new Collapse(this, options)))
      if (typeof option == 'string') data[option]()
    })
  }

  var old = $.fn.collapse

  $.fn.collapse             = Plugin
  $.fn.collapse.Constructor = Collapse


  // COLLAPSE NO CONFLICT
  // ====================

  $.fn.collapse.noConflict = function () {
    $.fn.collapse = old
    return this
  }


  // COLLAPSE DATA-API
  // =================

  $(document).on('click.bs.collapse.data-api', '[data-toggle="collapse"]', function (e) {
    var $this   = $(this)

    if (!$this.attr('data-target')) e.preventDefault()

    var $target = getTargetFromTrigger($this)
    var data    = $target.data('bs.collapse')
    var option  = data ? 'toggle' : $this.data()

    Plugin.call($target, option)
  })

}(jQuery);
</script>
<script>
window.initializeCodeFolding = function(show) {

  // handlers for show-all and hide all
  $("#rmd-show-all-code").click(function() {
    // close the dropdown menu when an option is clicked
    $("#allCodeButton").dropdown("toggle");
    $('div.r-code-collapse').each(function() {
      $(this).collapse('show');
    });
  });
  $("#rmd-hide-all-code").click(function() {
    // close the dropdown menu when an option is clicked
    $("#allCodeButton").dropdown("toggle");
    $('div.r-code-collapse').each(function() {
      $(this).collapse('hide');
    });
  });

  // index for unique code element ids
  var currentIndex = 1;

  // select all R code blocks
  var rCodeBlocks = $('pre.sourceCode, pre.r, pre.python, pre.bash, pre.sql, pre.cpp, pre.stan');
  rCodeBlocks.each(function() {

    // if code block has been labeled with class `fold-show`, show the code on init!
    var classList = $(this).attr('class').split(/\s+/);
    for (var i = 0; i < classList.length; i++) {
    if (classList[i] === 'fold-show') {
        show = true;
      }
    }

    // create a collapsable div to wrap the code in
    var div = $('<div class="collapse r-code-collapse"></div>');
    if (show)
      div.addClass('in');
    var id = 'rcode-643E0F36' + currentIndex++;
    div.attr('id', id);
    $(this).before(div);
    $(this).detach().appendTo(div);

    // add a show code button right above
    var showCodeText = $('<span>' + (show ? 'Hide' : 'Code') + '</span>');
    var showCodeButton = $('<button type="button" class="btn btn-default btn-xs code-folding-btn pull-right"></button>');
    showCodeButton.append(showCodeText);
    showCodeButton
        .attr('data-toggle', 'collapse')
        .attr('data-target', '#' + id)
        .attr('aria-expanded', show)
        .attr('aria-controls', id);

    var buttonRow = $('<div class="row"></div>');
    var buttonCol = $('<div class="col-md-12"></div>');

    buttonCol.append(showCodeButton);
    buttonRow.append(buttonCol);

    div.before(buttonRow);

    // hack: return show to false, otherwise all next codeBlocks will be shown!
    show = false;

    // update state of button on show/hide
    div.on('hidden.bs.collapse', function () {
      showCodeText.text('Code');
    });
    div.on('show.bs.collapse', function () {
      showCodeText.text('Hide');
    });
  });

}
</script>
<script>
/* ========================================================================
 * Bootstrap: dropdown.js v3.3.7
 * http://getbootstrap.com/javascript/#dropdowns
 * ========================================================================
 * Copyright 2011-2016 Twitter, Inc.
 * Licensed under MIT (https://github.com/twbs/bootstrap/blob/master/LICENSE)
 * ======================================================================== */


+function ($) {
  'use strict';

  // DROPDOWN CLASS DEFINITION
  // =========================

  var backdrop = '.dropdown-backdrop'
  var toggle   = '[data-toggle="dropdown"]'
  var Dropdown = function (element) {
    $(element).on('click.bs.dropdown', this.toggle)
  }

  Dropdown.VERSION = '3.3.7'

  function getParent($this) {
    var selector = $this.attr('data-target')

    if (!selector) {
      selector = $this.attr('href')
      selector = selector && /#[A-Za-z]/.test(selector) && selector.replace(/.*(?=#[^\s]*$)/, '') // strip for ie7
    }

    var $parent = selector && $(selector)

    return $parent && $parent.length ? $parent : $this.parent()
  }

  function clearMenus(e) {
    if (e && e.which === 3) return
    $(backdrop).remove()
    $(toggle).each(function () {
      var $this         = $(this)
      var $parent       = getParent($this)
      var relatedTarget = { relatedTarget: this }

      if (!$parent.hasClass('open')) return

      if (e && e.type == 'click' && /input|textarea/i.test(e.target.tagName) && $.contains($parent[0], e.target)) return

      $parent.trigger(e = $.Event('hide.bs.dropdown', relatedTarget))

      if (e.isDefaultPrevented()) return

      $this.attr('aria-expanded', 'false')
      $parent.removeClass('open').trigger($.Event('hidden.bs.dropdown', relatedTarget))
    })
  }

  Dropdown.prototype.toggle = function (e) {
    var $this = $(this)

    if ($this.is('.disabled, :disabled')) return

    var $parent  = getParent($this)
    var isActive = $parent.hasClass('open')

    clearMenus()

    if (!isActive) {
      if ('ontouchstart' in document.documentElement && !$parent.closest('.navbar-nav').length) {
        // if mobile we use a backdrop because click events don't delegate
        $(document.createElement('div'))
          .addClass('dropdown-backdrop')
          .insertAfter($(this))
          .on('click', clearMenus)
      }

      var relatedTarget = { relatedTarget: this }
      $parent.trigger(e = $.Event('show.bs.dropdown', relatedTarget))

      if (e.isDefaultPrevented()) return

      $this
        .trigger('focus')
        .attr('aria-expanded', 'true')

      $parent
        .toggleClass('open')
        .trigger($.Event('shown.bs.dropdown', relatedTarget))
    }

    return false
  }

  Dropdown.prototype.keydown = function (e) {
    if (!/(38|40|27|32)/.test(e.which) || /input|textarea/i.test(e.target.tagName)) return

    var $this = $(this)

    e.preventDefault()
    e.stopPropagation()

    if ($this.is('.disabled, :disabled')) return

    var $parent  = getParent($this)
    var isActive = $parent.hasClass('open')

    if (!isActive && e.which != 27 || isActive && e.which == 27) {
      if (e.which == 27) $parent.find(toggle).trigger('focus')
      return $this.trigger('click')
    }

    var desc = ' li:not(.disabled):visible a'
    var $items = $parent.find('.dropdown-menu' + desc)

    if (!$items.length) return

    var index = $items.index(e.target)

    if (e.which == 38 && index > 0)                 index--         // up
    if (e.which == 40 && index < $items.length - 1) index++         // down
    if (!~index)                                    index = 0

    $items.eq(index).trigger('focus')
  }


  // DROPDOWN PLUGIN DEFINITION
  // ==========================

  function Plugin(option) {
    return this.each(function () {
      var $this = $(this)
      var data  = $this.data('bs.dropdown')

      if (!data) $this.data('bs.dropdown', (data = new Dropdown(this)))
      if (typeof option == 'string') data[option].call($this)
    })
  }

  var old = $.fn.dropdown

  $.fn.dropdown             = Plugin
  $.fn.dropdown.Constructor = Dropdown


  // DROPDOWN NO CONFLICT
  // ====================

  $.fn.dropdown.noConflict = function () {
    $.fn.dropdown = old
    return this
  }


  // APPLY TO STANDARD DROPDOWN ELEMENTS
  // ===================================

  $(document)
    .on('click.bs.dropdown.data-api', clearMenus)
    .on('click.bs.dropdown.data-api', '.dropdown form', function (e) { e.stopPropagation() })
    .on('click.bs.dropdown.data-api', toggle, Dropdown.prototype.toggle)
    .on('keydown.bs.dropdown.data-api', toggle, Dropdown.prototype.keydown)
    .on('keydown.bs.dropdown.data-api', '.dropdown-menu', Dropdown.prototype.keydown)

}(jQuery);
</script>
<style type="text/css">
.code-folding-btn {
  margin-bottom: 4px;
}

.row { display: flex; }
.collapse { display: none; }
.in { display:block }
.pull-right > .dropdown-menu {
    right: 0;
    left: auto;
}

.dropdown-menu {
    position: absolute;
    top: 100%;
    left: 0;
    z-index: 1000;
    display: none;
    float: left;
    min-width: 160px;
    padding: 5px 0;
    margin: 2px 0 0;
    font-size: 14px;
    text-align: left;
    list-style: none;
    background-color: #fff;
    -webkit-background-clip: padding-box;
    background-clip: padding-box;
    border: 1px solid #ccc;
    border: 1px solid rgba(0,0,0,.15);
    border-radius: 4px;
    -webkit-box-shadow: 0 6px 12px rgba(0,0,0,.175);
    box-shadow: 0 6px 12px rgba(0,0,0,.175);
}

.open > .dropdown-menu {
    display: block;
    color: #ffffff;
    background-color: #ffffff;
    background-image: none;
    border-color: #92897e;
}

.dropdown-menu > li > a {
  display: block;
  padding: 3px 20px;
  clear: both;
  font-weight: 400;
  line-height: 1.42857143;
  color: #000000;
  white-space: nowrap;
}

.dropdown-menu > li > a:hover,
.dropdown-menu > li > a:focus {
  color: #ffffff;
  text-decoration: none;
  background-color: #e95420;
}

.dropdown-menu > .active > a,
.dropdown-menu > .active > a:hover,
.dropdown-menu > .active > a:focus {
  color: #ffffff;
  text-decoration: none;
  background-color: #e95420;
  outline: 0;
}
.dropdown-menu > .disabled > a,
.dropdown-menu > .disabled > a:hover,
.dropdown-menu > .disabled > a:focus {
  color: #aea79f;
}

.dropdown-menu > .disabled > a:hover,
.dropdown-menu > .disabled > a:focus {
  text-decoration: none;
  cursor: not-allowed;
  background-color: transparent;
  background-image: none;
  filter: progid:DXImageTransform.Microsoft.gradient(enabled = false);
}

.btn {
  display: inline-block;
  margin-bottom: 1;
  font-weight: normal;
  text-align: center;
  white-space: nowrap;
  vertical-align: middle;
  -ms-touch-action: manipulation;
      touch-action: manipulation;
  cursor: pointer;
  background-image: none;
  border: 1px solid transparent;
  padding: 4px 8px;
  font-size: 14px;
  line-height: 1.42857143;
  border-radius: 4px;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

.btn:focus,
.btn:active:focus,
.btn.active:focus,
.btn.focus,
.btn:active.focus,
.btn.active.focus {
  outline: 5px auto -webkit-focus-ring-color;
  outline-offset: -2px;
}
.btn:hover,
.btn:focus,
.btn.focus {
  color: #ffffff;
  text-decoration: none;
}
.btn:active,
.btn.active {
  background-image: none;
  outline: 0;
  box-shadow: inset 0 3px 5px rgba(0, 0, 0, 0.125);
}
.btn.disabled,
.btn[disabled],
fieldset[disabled] .btn {
  cursor: not-allowed;
  filter: alpha(opacity=65);
  opacity: 0.65;
  box-shadow: none;
}
a.btn.disabled,
fieldset[disabled] a.btn {
  pointer-events: none;
}
.btn-default {
  color: #ffffff;
  background-color: #aea79f; #important
  border-color: #aea79f;
}

.btn-default:focus,
.btn-default.focus {
  color: #ffffff;
  background-color: #978e83;
  border-color: #6f675e;
}

.btn-default:hover {
  color: #ffffff;
  background-color: #978e83;
  border-color: #92897e;
}
.btn-default:active,
.btn-default.active,
.btn-group > .btn:not(:first-child):not(:last-child):not(.dropdown-toggle) {
  border-radius: 0;
}
.btn-group > .btn:first-child {
  margin-left: 0;
}
.btn-group > .btn:first-child:not(:last-child):not(.dropdown-toggle) {
  border-top-right-radius: 0;
  border-bottom-right-radius: 0;
}
.btn-group > .btn:last-child:not(:first-child),
.btn-group > .dropdown-toggle:not(:first-child) {
  border-top-left-radius: 0;
  border-bottom-left-radius: 0;
}
.btn-group > .btn-group {
  float: left;
}
.btn-group > .btn-group:not(:first-child):not(:last-child) > .btn {
  border-radius: 0;
}
.btn-group > .btn-group:first-child:not(:last-child) > .btn:last-child,
.btn-group > .btn-group:first-child:not(:last-child) > .dropdown-toggle {
  border-top-right-radius: 0;
  border-bottom-right-radius: 0;
}
.btn-group > .btn-group:last-child:not(:first-child) > .btn:first-child {
  border-top-left-radius: 0;
  border-bottom-left-radius: 0;
}
.btn-group .dropdown-toggle:active,
.btn-group.open .dropdown-toggle {
  outline: 0;
}
.btn-group > .btn + .dropdown-toggle {
  padding-right: 8px;
  padding-left: 8px;
}
.btn-group > .btn-lg + .dropdown-toggle {
  padding-right: 12px;
  padding-left: 12px;
}
.btn-group.open .dropdown-toggle {
  box-shadow: inset 0 3px 5px rgba(0, 0, 0, 0.125);
}
.btn-group.open .dropdown-toggle.btn-link {
  box-shadow: none;
}

</style>
<script>
var str = '<div class="btn-group pull-right" style="position: fixed; right: 50px; top: 10px; z-index: 200"><button type="button" class="btn btn-default btn-xs dropdown-toggle" id="allCodeButton" data-toggle="dropdown" aria-haspopup="true" aria-expanded="true" data-_extension-text-contrast=""><span>Code</span> <span class="caret"></span></button><ul class="dropdown-menu" style="min-width: 50px;"><li><a id="rmd-show-all-code" href="#">Show All Code</a></li><li><a id="rmd-hide-all-code" href="#">Hide All Code</a></li></ul></div>';
document.write(str);
</script>
<script>
$(document).ready(function () {
  window.initializeCodeFolding("show" === "hide");
});
</script>


<style type="text/css">
a.sourceLine { display: inline-block; line-height: 1.25; }
a.sourceLine { pointer-events: none; color: inherit; text-decoration: inherit; }
a.sourceLine:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode { white-space: pre; position: relative; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
code.sourceCode { white-space: pre-wrap; }
a.sourceLine { text-indent: -1em; padding-left: 1em; }
}
pre.numberSource a.sourceLine
  { position: relative; left: -4em; }
pre.numberSource a.sourceLine::before
  { content: attr(title);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; pointer-events: all; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {  }
@media screen {
a.sourceLine::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>


<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">Report About Something</a></li>

<li class="divider"></li>
<li class="part"><span><b>Getting Started</b></span></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i>Introduction</a><ul>
<li class="chapter" data-level="" data-path="how-to-use-this-book.html"><a href="how-to-use-this-book.html"><i class="fa fa-check"></i>How to use this Book</a><ul>
<li class="chapter" data-level="" data-path="how-to-use-this-book.html"><a href="how-to-use-this-book.html#book-structure"><i class="fa fa-check"></i>1. Book Structure</a><ul>
<li class="chapter" data-level="" data-path="how-to-use-this-book.html"><a href="how-to-use-this-book.html#a-getting-started"><i class="fa fa-check"></i>a) Getting Started</a></li>
<li class="chapter" data-level="" data-path="how-to-use-this-book.html"><a href="how-to-use-this-book.html#b-basic-topics"><i class="fa fa-check"></i>b) Basic Topics</a></li>
<li class="chapter" data-level="" data-path="how-to-use-this-book.html"><a href="how-to-use-this-book.html#c-advanced-topics"><i class="fa fa-check"></i>c) Advanced Topics</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="how-to-use-this-book.html"><a href="how-to-use-this-book.html#find-what-is-relevant-for-you"><i class="fa fa-check"></i>2. Find what is relevant for you</a><ul>
<li class="chapter" data-level="" data-path="how-to-use-this-book.html"><a href="how-to-use-this-book.html#a-beginner-level-users"><i class="fa fa-check"></i>a) Beginner-level users</a></li>
<li class="chapter" data-level="" data-path="how-to-use-this-book.html"><a href="how-to-use-this-book.html#b-intermediate-level-users"><i class="fa fa-check"></i>b) Intermediate-level users</a></li>
<li class="chapter" data-level="" data-path="how-to-use-this-book.html"><a href="how-to-use-this-book.html#c-advanced-level-users"><i class="fa fa-check"></i>c) Advanced-level users</a></li>
</ul></li>
<li><a href="how-to-use-this-book.html#course-r-package-landgencourse">3. Course R package <code>LandGenCourse</code></a><ul>
<li class="chapter" data-level="" data-path="how-to-use-this-book.html"><a href="how-to-use-this-book.html#a-how-to-install-or-update"><i class="fa fa-check"></i>a) How to install (or update)</a></li>
<li><a href="how-to-use-this-book.html#b-how-to-use-the-r-package-landgencourse">b) How to use the R package <code>LandGenCourse</code></a></li>
<li class="chapter" data-level="" data-path="how-to-use-this-book.html"><a href="how-to-use-this-book.html#c-video-instructions-for-beginners"><i class="fa fa-check"></i>c) Video instructions for beginners</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="" data-path="overview-of-topics.html"><a href="overview-of-topics.html"><i class="fa fa-check"></i>Overview of Topics</a></li>
<li class="chapter" data-level="" data-path="list-of-packages-by-vignette.html"><a href="list-of-packages-by-vignette.html"><i class="fa fa-check"></i>List of Packages by Vignette</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="review-of-r-skills.html"><a href="review-of-r-skills.html"><i class="fa fa-check"></i>Review of R Skills</a><ul>
<li class="chapter" data-level="" data-path="basic-r-programming.html"><a href="basic-r-programming.html"><i class="fa fa-check"></i>Basic R Programming</a><ul>
<li class="chapter" data-level="" data-path="basic-r-programming.html"><a href="basic-r-programming.html#overview"><i class="fa fa-check"></i>1. Overview</a></li>
<li class="chapter" data-level="" data-path="basic-r-programming.html"><a href="basic-r-programming.html#data-types"><i class="fa fa-check"></i>2. Data Types</a><ul>
<li class="chapter" data-level="" data-path="basic-r-programming.html"><a href="basic-r-programming.html#numeric-data-types"><i class="fa fa-check"></i>2.1. Numeric Data Types</a></li>
<li class="chapter" data-level="" data-path="basic-r-programming.html"><a href="basic-r-programming.html#characters"><i class="fa fa-check"></i>2.2. Characters</a></li>
<li class="chapter" data-level="" data-path="basic-r-programming.html"><a href="basic-r-programming.html#factors"><i class="fa fa-check"></i>2.3. Factors</a></li>
<li class="chapter" data-level="" data-path="basic-r-programming.html"><a href="basic-r-programming.html#logical-types"><i class="fa fa-check"></i>2.4. Logical Types</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="basic-r-programming.html"><a href="basic-r-programming.html#data-containers"><i class="fa fa-check"></i>3. Data Containers</a><ul>
<li class="chapter" data-level="" data-path="basic-r-programming.html"><a href="basic-r-programming.html#vectors"><i class="fa fa-check"></i>3.1. Vectors</a></li>
<li class="chapter" data-level="" data-path="basic-r-programming.html"><a href="basic-r-programming.html#matrices"><i class="fa fa-check"></i>3.2. Matrices</a></li>
<li class="chapter" data-level="" data-path="basic-r-programming.html"><a href="basic-r-programming.html#lists"><i class="fa fa-check"></i>3.3. Lists</a></li>
<li class="chapter" data-level="" data-path="basic-r-programming.html"><a href="basic-r-programming.html#data-frames"><i class="fa fa-check"></i>3.4. Data Frames</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="basic-r-programming.html"><a href="basic-r-programming.html#programming"><i class="fa fa-check"></i>4. Programming</a><ul>
<li class="chapter" data-level="" data-path="basic-r-programming.html"><a href="basic-r-programming.html#function-writing"><i class="fa fa-check"></i>4.1. Function Writing</a></li>
<li class="chapter" data-level="" data-path="basic-r-programming.html"><a href="basic-r-programming.html#variable-scope"><i class="fa fa-check"></i>4.2. Variable Scope</a></li>
<li class="chapter" data-level="" data-path="basic-r-programming.html"><a href="basic-r-programming.html#decision-making"><i class="fa fa-check"></i>4.3. Decision Making</a></li>
<li class="chapter" data-level="" data-path="basic-r-programming.html"><a href="basic-r-programming.html#flow-control"><i class="fa fa-check"></i>4.4. Flow Control</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="" data-path="r-graphics.html"><a href="r-graphics.html"><i class="fa fa-check"></i>R Graphics</a><ul>
<li class="chapter" data-level="" data-path="r-graphics.html"><a href="r-graphics.html#overview-1"><i class="fa fa-check"></i>1. Overview</a></li>
<li class="chapter" data-level="" data-path="r-graphics.html"><a href="r-graphics.html#univariate-plots"><i class="fa fa-check"></i>2. Univariate plots</a><ul>
<li class="chapter" data-level="" data-path="r-graphics.html"><a href="r-graphics.html#barplots-and-histograms"><i class="fa fa-check"></i>2.1. Barplots and histograms</a></li>
<li class="chapter" data-level="" data-path="r-graphics.html"><a href="r-graphics.html#plotting-parameters"><i class="fa fa-check"></i>2.2. Plotting Parameters</a></li>
<li class="chapter" data-level="" data-path="r-graphics.html"><a href="r-graphics.html#density-plots"><i class="fa fa-check"></i>2.3. Density Plots</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="r-graphics.html"><a href="r-graphics.html#bivariate-plots"><i class="fa fa-check"></i>3. Bivariate plots</a><ul>
<li class="chapter" data-level="" data-path="r-graphics.html"><a href="r-graphics.html#overlaying-plots-by-grouping"><i class="fa fa-check"></i>3.1. Overlaying Plots by grouping</a></li>
<li class="chapter" data-level="" data-path="r-graphics.html"><a href="r-graphics.html#boxplots"><i class="fa fa-check"></i>3.2. Boxplots</a></li>
<li class="chapter" data-level="" data-path="r-graphics.html"><a href="r-graphics.html#scatter-plots"><i class="fa fa-check"></i>3.3. Scatter Plots</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="r-graphics.html"><a href="r-graphics.html#advanced-plotting"><i class="fa fa-check"></i>4. Advanced plotting</a><ul>
<li class="chapter" data-level="" data-path="r-graphics.html"><a href="r-graphics.html#multiple-plots"><i class="fa fa-check"></i>4.1. Multiple Plots</a></li>
<li class="chapter" data-level="" data-path="r-graphics.html"><a href="r-graphics.html#color-palettes"><i class="fa fa-check"></i>4.2. Color Palettes</a></li>
<li class="chapter" data-level="" data-path="r-graphics.html"><a href="r-graphics.html#rcolorbrewer"><i class="fa fa-check"></i>4.3. RColorBrewer</a></li>
<li class="chapter" data-level="" data-path="r-graphics.html"><a href="r-graphics.html#saving-imagery"><i class="fa fa-check"></i>4.4. Saving Imagery</a></li>
<li class="chapter" data-level="" data-path="r-graphics.html"><a href="r-graphics.html#interactive-graphics"><i class="fa fa-check"></i>4.5. Interactive Graphics</a></li>
</ul></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="" data-path="further-r-resources.html"><a href="further-r-resources.html"><i class="fa fa-check"></i>Further R Resources</a></li>
<li class="part"><span><b>Basic Topics</b></span></li>
<li class="chapter" data-level="" data-path="Week1.html"><a href="Week1.html"><i class="fa fa-check"></i>Week 1: Importing Genetic Data</a><ul>
<li class="chapter" data-level="" data-path="video-1a.html"><a href="video-1a.html"><i class="fa fa-check"></i>View Course Video</a><ul>
<li class="chapter" data-level="" data-path="video-1a.html"><a href="video-1a.html#preview-slides"><i class="fa fa-check"></i>Preview Slides</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="tutorial-1.html"><a href="tutorial-1.html"><i class="fa fa-check"></i>Interactive Tutorial 1</a><ul>
<li class="chapter" data-level="" data-path="tutorial-1.html"><a href="tutorial-1.html#instructions"><i class="fa fa-check"></i>Instructions</a><ul>
<li class="chapter" data-level="" data-path="tutorial-1.html"><a href="tutorial-1.html#a-goal"><i class="fa fa-check"></i>a) Goal: …</a></li>
<li class="chapter" data-level="" data-path="tutorial-1.html"><a href="tutorial-1.html#b-how-to-access-tutorials"><i class="fa fa-check"></i>b) How to access tutorials:</a></li>
<li class="chapter" data-level="" data-path="tutorial-1.html"><a href="tutorial-1.html#c-how-to-complete-tutorial"><i class="fa fa-check"></i>c) How to complete tutorial:</a></li>
<li class="chapter" data-level="" data-path="tutorial-1.html"><a href="tutorial-1.html#d-how-to-submit-answers-participating-institutions-only"><i class="fa fa-check"></i>d) How to submit answers (participating institutions only):</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="tutorial-1.html"><a href="tutorial-1.html#list-of-r-commands-used"><i class="fa fa-check"></i>List of R commands used</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="WE-1.html"><a href="WE-1.html"><i class="fa fa-check"></i>Worked Example</a><ul>
<li class="chapter" data-level="" data-path="WE-1.html"><a href="WE-1.html#overview-of-worked-example"><i class="fa fa-check"></i>1. Overview of Worked Example</a><ul>
<li class="chapter" data-level="" data-path="WE-1.html"><a href="WE-1.html#a.-goals"><i class="fa fa-check"></i>a. Goals</a></li>
<li class="chapter" data-level="" data-path="WE-1.html"><a href="WE-1.html#b.-data-set"><i class="fa fa-check"></i>b. Data set</a></li>
<li class="chapter" data-level="" data-path="WE-1.html"><a href="WE-1.html#c.-required-r-libraries"><i class="fa fa-check"></i>c. Required R libraries</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="WE-1.html"><a href="WE-1.html#import-data-from-.csv-file"><i class="fa fa-check"></i>2. Import data from .csv file</a><ul>
<li class="chapter" data-level="" data-path="WE-1.html"><a href="WE-1.html#a.-preparation"><i class="fa fa-check"></i>a. Preparation</a></li>
<li class="chapter" data-level="" data-path="WE-1.html"><a href="WE-1.html#b.-excel-file"><i class="fa fa-check"></i>b. Excel file</a></li>
<li class="chapter" data-level="" data-path="WE-1.html"><a href="WE-1.html#c.-import-with-function-read.csv"><i class="fa fa-check"></i>c. Import with function ‘read.csv’</a></li>
<li class="chapter" data-level="" data-path="WE-1.html"><a href="WE-1.html#d.-create-id-variable"><i class="fa fa-check"></i>d. Create ID variable</a></li>
<li class="chapter" data-level="" data-path="WE-1.html"><a href="WE-1.html#e.-export-with-function-write.csv"><i class="fa fa-check"></i>e. Export with function ‘write.csv’</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="WE-1.html"><a href="WE-1.html#create-genind-object"><i class="fa fa-check"></i>3. Create ‘genind’ Object</a><ul>
<li class="chapter" data-level="" data-path="WE-1.html"><a href="WE-1.html#a.-read-the-helpfile"><i class="fa fa-check"></i>a. Read the helpfile</a></li>
<li class="chapter" data-level="" data-path="WE-1.html"><a href="WE-1.html#b.-set-all-arguments-of-function-df2genind"><i class="fa fa-check"></i>b. Set all arguments of function ‘df2genind’</a></li>
<li class="chapter" data-level="" data-path="WE-1.html"><a href="WE-1.html#c.-check-imported-data"><i class="fa fa-check"></i>c. Check imported data</a></li>
<li class="chapter" data-level="" data-path="WE-1.html"><a href="WE-1.html#d.-summarize-genind-object"><i class="fa fa-check"></i>d. Summarize ‘genind’ object</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="WE-1.html"><a href="WE-1.html#view-information-stored-in-genind-object"><i class="fa fa-check"></i>4. View information stored in ‘genind’ object</a><ul>
<li><a href="WE-1.html#a.-slot-tab">a. Slot <code>@tab</code></a></li>
<li><a href="WE-1.html#b.-slot-loc.n.all">b. Slot <code>@loc.n.all</code></a></li>
<li><a href="WE-1.html#c.-slot-loc.fac">c. Slot <code>@loc.fac</code></a></li>
<li><a href="WE-1.html#d.-slot-all.names">d. Slot <code>@all.names</code></a></li>
<li class="chapter" data-level="" data-path="WE-1.html"><a href="WE-1.html#e.-further-slots"><i class="fa fa-check"></i>e. Further slots</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="WE-1.html"><a href="WE-1.html#import-data-with-gstudio-package"><i class="fa fa-check"></i>5. Import data with ‘gstudio’ package</a><ul>
<li class="chapter" data-level="" data-path="WE-1.html"><a href="WE-1.html#a.-function-read_population"><i class="fa fa-check"></i>a. Function ‘read_population’</a></li>
<li class="chapter" data-level="" data-path="WE-1.html"><a href="WE-1.html#b.-set-arguments"><i class="fa fa-check"></i>b. Set arguments</a></li>
<li class="chapter" data-level="" data-path="WE-1.html"><a href="WE-1.html#c.-check-imported-data-1"><i class="fa fa-check"></i>c. Check imported data</a></li>
<li class="chapter" data-level="" data-path="WE-1.html"><a href="WE-1.html#d.-add-frogid"><i class="fa fa-check"></i>d. Add FrogID</a></li>
<li class="chapter" data-level="" data-path="WE-1.html"><a href="WE-1.html#e.-convert-to-genind-object"><i class="fa fa-check"></i>e. Convert to ‘genind’ object</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="" data-path="r-exercise-week-1.html"><a href="r-exercise-week-1.html"><i class="fa fa-check"></i>R exercise Week 1</a></li>
<li class="chapter" data-level="" data-path="bonus-1a.html"><a href="bonus-1a.html"><i class="fa fa-check"></i>Bonus: Importing SNP data</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="Week2.html"><a href="Week2.html"><i class="fa fa-check"></i>Week 2: Spatial Data</a><ul>
<li class="chapter" data-level="" data-path="video-2a.html"><a href="video-2a.html"><i class="fa fa-check"></i>Course Video</a><ul>
<li class="chapter" data-level="" data-path="video-2a.html"><a href="video-2a.html#preview"><i class="fa fa-check"></i>Preview</a></li>
<li class="chapter" data-level="" data-path="video-2a.html"><a href="video-2a.html#open-video-in-new-window"><i class="fa fa-check"></i>Open video in new window</a></li>
<li class="chapter" data-level="" data-path="video-2a.html"><a href="video-2a.html#download-slides"><i class="fa fa-check"></i>Download slides</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="tutorial-2.html"><a href="tutorial-2.html"><i class="fa fa-check"></i>Interactive Tutorial</a><ul>
<li class="chapter" data-level="" data-path="tutorial-2.html"><a href="tutorial-2.html#preview-1"><i class="fa fa-check"></i>Preview</a></li>
<li class="chapter" data-level="" data-path="tutorial-2.html"><a href="tutorial-2.html#list-of-r-commands-used-1"><i class="fa fa-check"></i>List of R commands used</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="WE-2.html"><a href="WE-2.html"><i class="fa fa-check"></i>Worked Example</a><ul>
<li class="chapter" data-level="" data-path="WE-2.html"><a href="WE-2.html#overview-of-worked-example-1"><i class="fa fa-check"></i>1. Overview of Worked Example</a><ul>
<li class="chapter" data-level="" data-path="WE-2.html"><a href="WE-2.html#a.-goals-1"><i class="fa fa-check"></i>a. Goals</a></li>
<li class="chapter" data-level="" data-path="WE-2.html"><a href="WE-2.html#b.-data-set-1"><i class="fa fa-check"></i>b. Data set</a></li>
<li class="chapter" data-level="" data-path="WE-2.html"><a href="WE-2.html#c.-required-r-libraries-1"><i class="fa fa-check"></i>c. Required R libraries</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="WE-2.html"><a href="WE-2.html#import-site-data-from-.csv-file"><i class="fa fa-check"></i>2. Import site data from .csv file</a><ul>
<li class="chapter" data-level="" data-path="WE-2.html"><a href="WE-2.html#a.-import-data-into-spatialpointsdataframe"><i class="fa fa-check"></i>a. Import data into SpatialPointsDataFrame</a></li>
<li class="chapter" data-level="" data-path="WE-2.html"><a href="WE-2.html#b.-add-spatial-reference-data"><i class="fa fa-check"></i>b. Add spatial reference data</a></li>
<li class="chapter" data-level="" data-path="WE-2.html"><a href="WE-2.html#c.-change-projection"><i class="fa fa-check"></i>c. Change projection</a></li>
<li class="chapter" data-level="" data-path="WE-2.html"><a href="WE-2.html#d.-access-data-in-spatialpointsdataframe"><i class="fa fa-check"></i>d. Access data in ‘SpatialPointsDataFrame’</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="WE-2.html"><a href="WE-2.html#display-raster-data-and-overlay-sampling-locations-extract-data"><i class="fa fa-check"></i>3. Display raster data and overlay sampling locations, extract data</a><ul>
<li class="chapter" data-level="" data-path="WE-2.html"><a href="WE-2.html#a.-display-raster-data"><i class="fa fa-check"></i>a. Display raster data</a></li>
<li class="chapter" data-level="" data-path="WE-2.html"><a href="WE-2.html#b.-change-color-ramp-add-sampling-locations"><i class="fa fa-check"></i>b. Change color ramp, add sampling locations</a></li>
<li class="chapter" data-level="" data-path="WE-2.html"><a href="WE-2.html#c.-extract-raster-values-at-sampling-locations"><i class="fa fa-check"></i>c. Extract raster values at sampling locations</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="WE-2.html"><a href="WE-2.html#calculate-landscape-metrics"><i class="fa fa-check"></i>4. Calculate landscape metrics</a><ul>
<li class="chapter" data-level="" data-path="WE-2.html"><a href="WE-2.html#a.-calculate-patch--class--and-landscape-level-landscape-metrics"><i class="fa fa-check"></i>a. Calculate patch-, class- and landscape level landscape metrics</a></li>
<li class="chapter" data-level="" data-path="WE-2.html"><a href="WE-2.html#b.-calculate-patch-level-landscape-metrics-for-evergreen-forest"><i class="fa fa-check"></i>b. Calculate patch-level landscape metrics for ‘Evergreen Forest’</a></li>
<li class="chapter" data-level="" data-path="WE-2.html"><a href="WE-2.html#c.-extract-forest-patch-size-at-samplig-locations"><i class="fa fa-check"></i>c. Extract forest patch size at samplig locations</a></li>
<li class="chapter" data-level="" data-path="WE-2.html"><a href="WE-2.html#d.-plot-a-bubble-map-of-forest-patch-size-at-sampling-locations"><i class="fa fa-check"></i>d. Plot a bubble map of forest patch size at sampling locations</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="WE-2.html"><a href="WE-2.html#sample-landscape-metrics-within-buffer-around-sampling-locations"><i class="fa fa-check"></i>5. Sample landscape metrics within buffer around sampling locations</a><ul>
<li class="chapter" data-level="" data-path="WE-2.html"><a href="WE-2.html#b.-extract-landscape-metric-of-choice-for-a-single-cover-type-as-vector"><i class="fa fa-check"></i>b. Extract landscape metric of choice for a single cover type (as vector)</a></li>
<li class="chapter" data-level="" data-path="WE-2.html"><a href="WE-2.html#c.-extract-landscape-metric-of-choice-for-all-cover-types-as-data-frame."><i class="fa fa-check"></i>c. Extract landscape metric of choice for all cover types (as data frame).</a></li>
<li class="chapter" data-level="" data-path="WE-2.html"><a href="WE-2.html#d.-extract-all-landscape-metrics-for-a-single-cover-type-as-data-frame"><i class="fa fa-check"></i>d. Extract all landscape metrics for a single cover type (as data frame)</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="" data-path="r-exercise-week-2.html"><a href="r-exercise-week-2.html"><i class="fa fa-check"></i>R Exercise Week 2</a></li>
<li class="chapter" data-level="" data-path="bonus-2a.html"><a href="bonus-2a.html"><i class="fa fa-check"></i>Bonus: Using package sf</a></li>
<li class="chapter" data-level="" data-path="bonus-2b.html"><a href="bonus-2b.html"><i class="fa fa-check"></i>Bonus: Visualize LULC map</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="Week3.html"><a href="Week3.html"><i class="fa fa-check"></i>Week 3: Genetic Diversity</a><ul>
<li class="chapter" data-level="" data-path="video-3a.html"><a href="video-3a.html"><i class="fa fa-check"></i>Course Video</a><ul>
<li class="chapter" data-level="" data-path="video-3a.html"><a href="video-3a.html#preview-2"><i class="fa fa-check"></i>Preview</a></li>
<li class="chapter" data-level="" data-path="video-3a.html"><a href="video-3a.html#open-video-in-new-window-1"><i class="fa fa-check"></i>Open video in new window</a></li>
<li class="chapter" data-level="" data-path="video-3a.html"><a href="video-3a.html#download-slides-1"><i class="fa fa-check"></i>Download slides</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="tutorial-3.html"><a href="tutorial-3.html"><i class="fa fa-check"></i>Interactive Tutorial</a><ul>
<li class="chapter" data-level="" data-path="tutorial-3.html"><a href="tutorial-3.html#preview-3"><i class="fa fa-check"></i>Preview</a></li>
<li class="chapter" data-level="" data-path="tutorial-3.html"><a href="tutorial-3.html#list-of-r-commands-used-2"><i class="fa fa-check"></i>List of R commands used</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="WE-3.html"><a href="WE-3.html"><i class="fa fa-check"></i>Worked Example</a><ul>
<li class="chapter" data-level="" data-path="WE-3.html"><a href="WE-3.html#overview-of-worked-example-2"><i class="fa fa-check"></i>1. Overview of Worked Example</a><ul>
<li class="chapter" data-level="" data-path="WE-3.html"><a href="WE-3.html#a.-goals-2"><i class="fa fa-check"></i>a. Goals</a></li>
<li class="chapter" data-level="" data-path="WE-3.html"><a href="WE-3.html#b.-data-set-2"><i class="fa fa-check"></i>b. Data set</a></li>
<li class="chapter" data-level="" data-path="WE-3.html"><a href="WE-3.html#c.-required-r-libraries-2"><i class="fa fa-check"></i>c. Required R libraries</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="WE-3.html"><a href="WE-3.html#basic-checking-of-markers-and-populations"><i class="fa fa-check"></i>2. Basic checking of markers and populations</a><ul>
<li class="chapter" data-level="" data-path="WE-3.html"><a href="WE-3.html#a.-re-create-genind-object"><i class="fa fa-check"></i>a. Re-create genind object</a></li>
<li class="chapter" data-level="" data-path="WE-3.html"><a href="WE-3.html#b.-check-that-markers-are-polymorphic"><i class="fa fa-check"></i>b. Check that markers are polymorphic</a></li>
<li class="chapter" data-level="" data-path="WE-3.html"><a href="WE-3.html#c.-check-for-deviations-from-hardy-weinberg-equilibrium-hwe"><i class="fa fa-check"></i>c. Check for deviations from Hardy-Weinberg equilibrium (HWE)</a></li>
<li class="chapter" data-level="" data-path="WE-3.html"><a href="WE-3.html#d.-check-for-linkage-disequilibrium-ld"><i class="fa fa-check"></i>d. Check for linkage disequilibrium (LD)</a></li>
<li class="chapter" data-level="" data-path="WE-3.html"><a href="WE-3.html#e.-check-for-null-alleles"><i class="fa fa-check"></i>e. Check for null alleles</a></li>
<li class="chapter" data-level="" data-path="WE-3.html"><a href="WE-3.html#f.-overall-interpretation"><i class="fa fa-check"></i>f. Overall interpretation</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="WE-3.html"><a href="WE-3.html#assess-genetic-diversity"><i class="fa fa-check"></i>3. Assess genetic diversity</a><ul>
<li class="chapter" data-level="" data-path="WE-3.html"><a href="WE-3.html#a.-rarefied-allelic-richness"><i class="fa fa-check"></i>a. Rarefied allelic richness</a></li>
<li class="chapter" data-level="" data-path="WE-3.html"><a href="WE-3.html#b.-observed-and-expected-heterozygosity"><i class="fa fa-check"></i>b. Observed and expected heterozygosity</a></li>
<li class="chapter" data-level="" data-path="WE-3.html"><a href="WE-3.html#c.-create-table-with-sitel-level-genetic-diversity-measures"><i class="fa fa-check"></i>c. Create table with sitel-level genetic diversity measures</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="WE-3.html"><a href="WE-3.html#aggregate-genetic-data-at-population-level-allele-frequencies"><i class="fa fa-check"></i>4. Aggregate genetic data at population level (allele frequencies)</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="r-exercise-week-3.html"><a href="r-exercise-week-3.html"><i class="fa fa-check"></i>R Exercise Week 3</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="Week4.html"><a href="Week4.html"><i class="fa fa-check"></i>Week 4: Metapopulations</a><ul>
<li class="chapter" data-level="" data-path="video-4a.html"><a href="video-4a.html"><i class="fa fa-check"></i>Course Video</a><ul>
<li class="chapter" data-level="" data-path="video-4a.html"><a href="video-4a.html#preview-4"><i class="fa fa-check"></i>Preview</a></li>
<li class="chapter" data-level="" data-path="video-4a.html"><a href="video-4a.html#open-video-in-new-window-2"><i class="fa fa-check"></i>Open video in new window</a></li>
<li class="chapter" data-level="" data-path="video-4a.html"><a href="video-4a.html#download-slides-2"><i class="fa fa-check"></i>Download slides</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="tutorial-4.html"><a href="tutorial-4.html"><i class="fa fa-check"></i>Interactive Tutorial</a><ul>
<li class="chapter" data-level="" data-path="tutorial-4.html"><a href="tutorial-4.html#preview-5"><i class="fa fa-check"></i>Preview</a></li>
<li class="chapter" data-level="" data-path="tutorial-4.html"><a href="tutorial-4.html#list-of-r-commands-used-3"><i class="fa fa-check"></i>List of R commands used</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="WE-4.html"><a href="WE-4.html"><i class="fa fa-check"></i>Worked Example</a><ul>
<li class="chapter" data-level="" data-path="WE-4.html"><a href="WE-4.html#overview-of-worked-example-3"><i class="fa fa-check"></i>1. Overview of Worked Example</a><ul>
<li class="chapter" data-level="" data-path="WE-4.html"><a href="WE-4.html#a.-goals-3"><i class="fa fa-check"></i>a. Goals</a></li>
<li class="chapter" data-level="" data-path="WE-4.html"><a href="WE-4.html#b.-data-set-3"><i class="fa fa-check"></i>b. Data set</a></li>
<li class="chapter" data-level="" data-path="WE-4.html"><a href="WE-4.html#reference"><i class="fa fa-check"></i>Reference:</a></li>
<li class="chapter" data-level="" data-path="WE-4.html"><a href="WE-4.html#c.-required-r-libraries-3"><i class="fa fa-check"></i>c. Required R libraries</a></li>
<li class="chapter" data-level="" data-path="WE-4.html"><a href="WE-4.html#d.-import-data"><i class="fa fa-check"></i>d. Import data</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="WE-4.html"><a href="WE-4.html#spatial-distribution-of-genetic-structure"><i class="fa fa-check"></i>2. Spatial distribution of genetic structure</a><ul>
<li class="chapter" data-level="" data-path="WE-4.html"><a href="WE-4.html#a.-creating-a-genind-object-with-the-hierarchical-data-set"><i class="fa fa-check"></i>a. Creating a ‘genind’ object with the hierarchical data set</a></li>
<li class="chapter" data-level="" data-path="WE-4.html"><a href="WE-4.html#b.-hierarchical-amova"><i class="fa fa-check"></i>b. Hierarchical AMOVA</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="WE-4.html"><a href="WE-4.html#what-determines-genetic-differentiation-among-sites"><i class="fa fa-check"></i>3. What determines genetic differentiation among sites?</a><ul>
<li class="chapter" data-level="" data-path="WE-4.html"><a href="WE-4.html#a.-correlation-matrix"><i class="fa fa-check"></i>a. Correlation matrix</a></li>
<li class="chapter" data-level="" data-path="WE-4.html"><a href="WE-4.html#b.-scatterplots"><i class="fa fa-check"></i>b. Scatterplots</a></li>
<li class="chapter" data-level="" data-path="WE-4.html"><a href="WE-4.html#c.-regression-model"><i class="fa fa-check"></i>c. Regression model</a></li>
<li class="chapter" data-level="" data-path="WE-4.html"><a href="WE-4.html#d.-which-populations-dont-fit-the-general-pattern"><i class="fa fa-check"></i>d. Which populations don’t fit the general pattern?</a></li>
<li class="chapter" data-level="" data-path="WE-4.html"><a href="WE-4.html#e.-regression-model-without-outliers"><i class="fa fa-check"></i>e. Regression model without outliers</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="WE-4.html"><a href="WE-4.html#what-determines-genetic-diversity"><i class="fa fa-check"></i>4. What determines genetic diversity?</a><ul>
<li class="chapter" data-level="" data-path="WE-4.html"><a href="WE-4.html#a.-correlation-matrix-1"><i class="fa fa-check"></i>a. Correlation matrix</a></li>
<li class="chapter" data-level="" data-path="WE-4.html"><a href="WE-4.html#b.-regression-models"><i class="fa fa-check"></i>b. Regression models</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="WE-4.html"><a href="WE-4.html#are-genetic-differentiation-and-diversity-related"><i class="fa fa-check"></i>5. Are genetic differentiation and diversity related?</a></li>
<li class="chapter" data-level="" data-path="WE-4.html"><a href="WE-4.html#effect-of-recent-extinction-events"><i class="fa fa-check"></i>6. Effect of recent extinction events</a><ul>
<li class="chapter" data-level="" data-path="WE-4.html"><a href="WE-4.html#a.-effect-of-patch-extinction-event-temporal-data-set"><i class="fa fa-check"></i>a. Effect of patch extinction event (temporal data set)</a></li>
<li class="chapter" data-level="" data-path="WE-4.html"><a href="WE-4.html#b.-power-analysis"><i class="fa fa-check"></i>b. Power analysis</a></li>
<li class="chapter" data-level="" data-path="WE-4.html"><a href="WE-4.html#c.-sample-size-calculation"><i class="fa fa-check"></i>c. Sample size calculation</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="" data-path="r-exercise-week-4.html"><a href="r-exercise-week-4.html"><i class="fa fa-check"></i>R Exercise Week 4</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="Week5.html"><a href="Week5.html"><i class="fa fa-check"></i>Week 5: Spatial Statistics</a><ul>
<li class="chapter" data-level="" data-path="video-5a.html"><a href="video-5a.html"><i class="fa fa-check"></i>Course Video</a><ul>
<li class="chapter" data-level="" data-path="video-5a.html"><a href="video-5a.html#preview-6"><i class="fa fa-check"></i>Preview</a></li>
<li class="chapter" data-level="" data-path="video-5a.html"><a href="video-5a.html#open-video-in-new-window-3"><i class="fa fa-check"></i>Open video in new window</a></li>
<li class="chapter" data-level="" data-path="video-5a.html"><a href="video-5a.html#download-slides-3"><i class="fa fa-check"></i>Download slides</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="tutorial-5.html"><a href="tutorial-5.html"><i class="fa fa-check"></i>Interactive Tutorial</a><ul>
<li class="chapter" data-level="" data-path="tutorial-5.html"><a href="tutorial-5.html#preview-7"><i class="fa fa-check"></i>Preview</a></li>
<li class="chapter" data-level="" data-path="tutorial-5.html"><a href="tutorial-5.html#list-of-r-commands-used-4"><i class="fa fa-check"></i>List of R commands used</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="WE-5.html"><a href="WE-5.html"><i class="fa fa-check"></i>Worked Example</a><ul>
<li class="chapter" data-level="" data-path="WE-5.html"><a href="WE-5.html#overview-of-worked-example-4"><i class="fa fa-check"></i>1. Overview of Worked Example</a><ul>
<li class="chapter" data-level="" data-path="WE-5.html"><a href="WE-5.html#a.-goals-4"><i class="fa fa-check"></i>a. Goals</a></li>
<li class="chapter" data-level="" data-path="WE-5.html"><a href="WE-5.html#b.-data-set-4"><i class="fa fa-check"></i>b. Data set</a></li>
<li class="chapter" data-level="" data-path="WE-5.html"><a href="WE-5.html#c.-required-r-packages"><i class="fa fa-check"></i>c. Required R packages</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="WE-5.html"><a href="WE-5.html#data-import-and-manipulation"><i class="fa fa-check"></i>2. Data import and manipulation</a><ul>
<li class="chapter" data-level="" data-path="WE-5.html"><a href="WE-5.html#a.-create-an-ecogen-object"><i class="fa fa-check"></i>a. Create an ecogen object</a></li>
<li class="chapter" data-level="" data-path="WE-5.html"><a href="WE-5.html#b.-aggregate-to-ecopop"><i class="fa fa-check"></i>b. Aggregate to ecopop</a></li>
<li class="chapter" data-level="" data-path="WE-5.html"><a href="WE-5.html#b.-add-site-level-data"><i class="fa fa-check"></i>b. Add site-level data</a></li>
<li class="chapter" data-level="" data-path="WE-5.html"><a href="WE-5.html#c.-export-to-adegenet-and-gstudio"><i class="fa fa-check"></i>c. Export to adegenet and gstudio</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="WE-5.html"><a href="WE-5.html#calculate-genetic-distances"><i class="fa fa-check"></i>3. Calculate genetic distances</a><ul>
<li class="chapter" data-level="" data-path="WE-5.html"><a href="WE-5.html#a.-genetic-distances-calculated-from-genind-object"><i class="fa fa-check"></i>a. Genetic distances calculated from genind object</a></li>
<li class="chapter" data-level="" data-path="WE-5.html"><a href="WE-5.html#b.-more-distance-matrices-with-gstudio"><i class="fa fa-check"></i>b. More distance matrices with ‘gstudio’</a></li>
<li class="chapter" data-level="" data-path="WE-5.html"><a href="WE-5.html#c.-assemble-distance-matrices"><i class="fa fa-check"></i>c. Assemble distance matrices</a></li>
<li class="chapter" data-level="" data-path="WE-5.html"><a href="WE-5.html#d.-export-genetic-distance-matrices"><i class="fa fa-check"></i>d. Export genetic distance matrices</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="WE-5.html"><a href="WE-5.html#perform-a-mantel-test-to-test-for-ibd"><i class="fa fa-check"></i>4. Perform a Mantel test to test for IBD</a><ul>
<li class="chapter" data-level="" data-path="WE-5.html"><a href="WE-5.html#a.-visually-check-linearity"><i class="fa fa-check"></i>a. Visually check linearity</a></li>
<li class="chapter" data-level="" data-path="WE-5.html"><a href="WE-5.html#b.-perform-mantel-test"><i class="fa fa-check"></i>b. Perform Mantel test</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="WE-5.html"><a href="WE-5.html#create-mantel-correlogram-for-genetic-data"><i class="fa fa-check"></i>5. Create Mantel correlogram for genetic data</a><ul>
<li class="chapter" data-level="" data-path="WE-5.html"><a href="WE-5.html#a.-create-a-first-mantel-correlogram"><i class="fa fa-check"></i>a. Create a first Mantel correlogram</a></li>
<li class="chapter" data-level="" data-path="WE-5.html"><a href="WE-5.html#b.-vary-distance-class-definition"><i class="fa fa-check"></i>b. Vary distance class definition</a></li>
<li class="chapter" data-level="" data-path="WE-5.html"><a href="WE-5.html#c.-alternative-measures-of-genetic-distances"><i class="fa fa-check"></i>c. Alternative measures of genetic distances</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="WE-5.html"><a href="WE-5.html#specify-spatial-weights-and-calculate-morans-i"><i class="fa fa-check"></i>6. Specify spatial weights and calculate Moran’s I</a><ul>
<li class="chapter" data-level="" data-path="WE-5.html"><a href="WE-5.html#a.-defining-spatial-neighbors"><i class="fa fa-check"></i>a. Defining spatial neighbors</a></li>
<li class="chapter" data-level="" data-path="WE-5.html"><a href="WE-5.html#b.-defining-spatial-weights"><i class="fa fa-check"></i>b. Defining spatial weights</a></li>
<li class="chapter" data-level="" data-path="WE-5.html"><a href="WE-5.html#c.-calculating-and-testing-morans-i"><i class="fa fa-check"></i>c. Calculating and testing Moran’s I</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="" data-path="r-exercise-week-5.html"><a href="r-exercise-week-5.html"><i class="fa fa-check"></i>R Exercise Week 5</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="Week6.html"><a href="Week6.html"><i class="fa fa-check"></i>Week 6: Quantitative Genetics</a><ul>
<li class="chapter" data-level="" data-path="video-6a.html"><a href="video-6a.html"><i class="fa fa-check"></i>Course Video</a><ul>
<li class="chapter" data-level="" data-path="video-6a.html"><a href="video-6a.html#preview-8"><i class="fa fa-check"></i>Preview</a></li>
<li class="chapter" data-level="" data-path="video-6a.html"><a href="video-6a.html#open-video-in-new-window-4"><i class="fa fa-check"></i>Open video in new window</a></li>
<li class="chapter" data-level="" data-path="video-6a.html"><a href="video-6a.html#download-slides-4"><i class="fa fa-check"></i>Download slides</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="tutorial-6.html"><a href="tutorial-6.html"><i class="fa fa-check"></i>Interactive Tutorial</a><ul>
<li class="chapter" data-level="" data-path="tutorial-6.html"><a href="tutorial-6.html#preview-9"><i class="fa fa-check"></i>Preview</a></li>
<li class="chapter" data-level="" data-path="tutorial-6.html"><a href="tutorial-6.html#list-of-r-commands-used-5"><i class="fa fa-check"></i>List of R commands used</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="WE-6.html"><a href="WE-6.html"><i class="fa fa-check"></i>Worked Example</a><ul>
<li class="chapter" data-level="" data-path="WE-6.html"><a href="WE-6.html#overview-of-worked-example-5"><i class="fa fa-check"></i>1. Overview of Worked Example</a><ul>
<li class="chapter" data-level="" data-path="WE-6.html"><a href="WE-6.html#a.-goals-5"><i class="fa fa-check"></i>a. Goals</a></li>
<li class="chapter" data-level="" data-path="WE-6.html"><a href="WE-6.html#b.-data-set-5"><i class="fa fa-check"></i>b. Data set</a></li>
<li class="chapter" data-level="" data-path="WE-6.html"><a href="WE-6.html#c.-required-r-packages-1"><i class="fa fa-check"></i>c. Required R packages</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="WE-6.html"><a href="WE-6.html#estimate-genetic-and-non-genetic-variance-components-from-a-common-garden-experiment"><i class="fa fa-check"></i>2. Estimate genetic and non-genetic variance components from a common garden experiment</a><ul>
<li class="chapter" data-level="" data-path="WE-6.html"><a href="WE-6.html#a.-import-phenytypic-data"><i class="fa fa-check"></i>a. Import phenytypic data</a></li>
<li class="chapter" data-level="" data-path="WE-6.html"><a href="WE-6.html#b.-fit-linear-mixed-effects-models-lmm-to-trait-data-observed-on-families"><i class="fa fa-check"></i>b. Fit linear mixed effects models (LMM) to trait data observed on families</a></li>
<li class="chapter" data-level="" data-path="WE-6.html"><a href="WE-6.html#c.-compare-model-fit"><i class="fa fa-check"></i>c. Compare model fit</a></li>
<li class="chapter" data-level="" data-path="WE-6.html"><a href="WE-6.html#d.-check-model-validity"><i class="fa fa-check"></i>d. Check model validity</a></li>
<li class="chapter" data-level="" data-path="WE-6.html"><a href="WE-6.html#e.-estimate-variance-components"><i class="fa fa-check"></i>e. Estimate variance components</a></li>
<li class="chapter" data-level="" data-path="WE-6.html"><a href="WE-6.html#f.-significance-testing"><i class="fa fa-check"></i>f. Significance testing</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="WE-6.html"><a href="WE-6.html#estimate-trait-heritability"><i class="fa fa-check"></i>3. Estimate trait heritability</a><ul>
<li class="chapter" data-level="" data-path="WE-6.html"><a href="WE-6.html#a.-estimate-heritability"><i class="fa fa-check"></i>a. Estimate heritability</a></li>
<li class="chapter" data-level="" data-path="WE-6.html"><a href="WE-6.html#b.-estimate-trait-differentiation"><i class="fa fa-check"></i>b. Estimate trait differentiation</a></li>
</ul></li>
<li><a href="WE-6.html#compare-q_st-to-f_st-measured-from-snp-data">4. Compare <span class="math inline">\(Q_{st}\)</span> to <span class="math inline">\(F_{st}\)</span> measured from SNP data</a><ul>
<li class="chapter" data-level="" data-path="WE-6.html"><a href="WE-6.html#a.-import-ecogen-object"><i class="fa fa-check"></i>a. Import ‘ecogen’ object</a></li>
<li><a href="WE-6.html#b.-compare-q_st-to-f_st">b. Compare <span class="math inline">\(Q_{ST}\)</span> to <span class="math inline">\(F_{ST}\)</span></a></li>
</ul></li>
<li class="chapter" data-level="" data-path="WE-6.html"><a href="WE-6.html#assess-correlation-between-trait-and-environment"><i class="fa fa-check"></i>5. Assess correlation between trait and environment</a><ul>
<li class="chapter" data-level="" data-path="WE-6.html"><a href="WE-6.html#a.-describe-and-extract-effects"><i class="fa fa-check"></i>a. Describe and extract effects</a></li>
<li class="chapter" data-level="" data-path="WE-6.html"><a href="WE-6.html#b.-correlation-of-population-effect-with-environment"><i class="fa fa-check"></i>b. Correlation of population effect with environment</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="" data-path="r-exercise-week-6.html"><a href="r-exercise-week-6.html"><i class="fa fa-check"></i>R Exercise Week 6</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="Week7.html"><a href="Week7.html"><i class="fa fa-check"></i>Week 7: Spatial Linear Models</a><ul>
<li class="chapter" data-level="" data-path="video-7a.html"><a href="video-7a.html"><i class="fa fa-check"></i>Course Video</a><ul>
<li class="chapter" data-level="" data-path="video-7a.html"><a href="video-7a.html#preview-10"><i class="fa fa-check"></i>Preview</a></li>
<li class="chapter" data-level="" data-path="video-7a.html"><a href="video-7a.html#open-video-in-new-window-5"><i class="fa fa-check"></i>Open video in new window</a></li>
<li class="chapter" data-level="" data-path="video-7a.html"><a href="video-7a.html#download-slides-5"><i class="fa fa-check"></i>Download slides</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="tutorial-7.html"><a href="tutorial-7.html"><i class="fa fa-check"></i>Interactive Tutorial</a><ul>
<li class="chapter" data-level="" data-path="tutorial-7.html"><a href="tutorial-7.html#preview-11"><i class="fa fa-check"></i>Preview</a></li>
<li class="chapter" data-level="" data-path="tutorial-7.html"><a href="tutorial-7.html#list-of-r-commands-used-6"><i class="fa fa-check"></i>List of R commands used</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="WE-7.html"><a href="WE-7.html"><i class="fa fa-check"></i>Worked Example</a><ul>
<li class="chapter" data-level="" data-path="WE-7.html"><a href="WE-7.html#overview-of-worked-example-6"><i class="fa fa-check"></i>1. Overview of Worked Example</a><ul>
<li class="chapter" data-level="" data-path="WE-7.html"><a href="WE-7.html#a.-goals-6"><i class="fa fa-check"></i>a. Goals</a></li>
<li class="chapter" data-level="" data-path="WE-7.html"><a href="WE-7.html#b.-data-set-6"><i class="fa fa-check"></i>b. Data set</a></li>
<li class="chapter" data-level="" data-path="WE-7.html"><a href="WE-7.html#c.-required-r-packages-2"><i class="fa fa-check"></i>c. Required R packages</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="WE-7.html"><a href="WE-7.html#estimate-genetic-and-non-genetic-variance-components-from-a-common-garden-experiment-1"><i class="fa fa-check"></i>2. Estimate genetic and non-genetic variance components from a common garden experiment</a><ul>
<li class="chapter" data-level="" data-path="WE-7.html"><a href="WE-7.html#a.-import-phenytypic-data-1"><i class="fa fa-check"></i>a. Import phenytypic data</a></li>
<li class="chapter" data-level="" data-path="WE-7.html"><a href="WE-7.html#b.-fit-linear-mixed-effects-models-lmm-to-trait-data-observed-on-families-1"><i class="fa fa-check"></i>b. Fit linear mixed effects models (LMM) to trait data observed on families</a></li>
<li class="chapter" data-level="" data-path="WE-7.html"><a href="WE-7.html#c.-compare-model-fit-1"><i class="fa fa-check"></i>c. Compare model fit</a></li>
<li class="chapter" data-level="" data-path="WE-7.html"><a href="WE-7.html#d.-check-model-validity-1"><i class="fa fa-check"></i>d. Check model validity</a></li>
<li class="chapter" data-level="" data-path="WE-7.html"><a href="WE-7.html#e.-estimate-variance-components-1"><i class="fa fa-check"></i>e. Estimate variance components</a></li>
<li class="chapter" data-level="" data-path="WE-7.html"><a href="WE-7.html#f.-significance-testing-1"><i class="fa fa-check"></i>f. Significance testing</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="WE-7.html"><a href="WE-7.html#estimate-trait-heritability-1"><i class="fa fa-check"></i>3. Estimate trait heritability</a><ul>
<li class="chapter" data-level="" data-path="WE-7.html"><a href="WE-7.html#a.-estimate-heritability-1"><i class="fa fa-check"></i>a. Estimate heritability</a></li>
<li class="chapter" data-level="" data-path="WE-7.html"><a href="WE-7.html#b.-estimate-trait-differentiation-1"><i class="fa fa-check"></i>b. Estimate trait differentiation</a></li>
</ul></li>
<li><a href="WE-7.html#compare-q_st-to-f_st-measured-from-snp-data-1">4. Compare <span class="math inline">\(Q_{st}\)</span> to <span class="math inline">\(F_{st}\)</span> measured from SNP data</a><ul>
<li class="chapter" data-level="" data-path="WE-7.html"><a href="WE-7.html#a.-import-ecogen-object-1"><i class="fa fa-check"></i>a. Import ‘ecogen’ object</a></li>
<li><a href="WE-7.html#b.-compare-q_st-to-f_st-1">b. Compare <span class="math inline">\(Q_{ST}\)</span> to <span class="math inline">\(F_{ST}\)</span></a></li>
</ul></li>
<li class="chapter" data-level="" data-path="WE-7.html"><a href="WE-7.html#assess-correlation-between-trait-and-environment-1"><i class="fa fa-check"></i>5. Assess correlation between trait and environment</a><ul>
<li class="chapter" data-level="" data-path="WE-7.html"><a href="WE-7.html#a.-describe-and-extract-effects-1"><i class="fa fa-check"></i>a. Describe and extract effects</a></li>
<li class="chapter" data-level="" data-path="WE-7.html"><a href="WE-7.html#b.-correlation-of-population-effect-with-environment-1"><i class="fa fa-check"></i>b. Correlation of population effect with environment</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="" data-path="r-exercise-week-6-1.html"><a href="r-exercise-week-6-1.html"><i class="fa fa-check"></i>R Exercise Week 6</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="Week8.html"><a href="Week8.html"><i class="fa fa-check"></i>Week 8: Simulation Experiments</a><ul>
<li class="chapter" data-level="" data-path="video-8a.html"><a href="video-8a.html"><i class="fa fa-check"></i>Course Video</a><ul>
<li class="chapter" data-level="" data-path="video-8a.html"><a href="video-8a.html#preview-12"><i class="fa fa-check"></i>Preview</a></li>
<li class="chapter" data-level="" data-path="video-8a.html"><a href="video-8a.html#open-video-in-new-window-6"><i class="fa fa-check"></i>Open video in new window</a></li>
<li class="chapter" data-level="" data-path="video-8a.html"><a href="video-8a.html#download-slides-6"><i class="fa fa-check"></i>Download slides</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="tutorial-8.html"><a href="tutorial-8.html"><i class="fa fa-check"></i>Interactive Tutorial</a><ul>
<li class="chapter" data-level="" data-path="tutorial-8.html"><a href="tutorial-8.html#preview-13"><i class="fa fa-check"></i>Preview</a></li>
<li class="chapter" data-level="" data-path="tutorial-8.html"><a href="tutorial-8.html#list-of-r-commands-used-7"><i class="fa fa-check"></i>List of R commands used</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="WE-8.html"><a href="WE-8.html"><i class="fa fa-check"></i>Worked Example</a><ul>
<li class="chapter" data-level="" data-path="WE-8.html"><a href="WE-8.html#overview-of-worked-example-7"><i class="fa fa-check"></i>1. Overview of Worked Example</a><ul>
<li class="chapter" data-level="" data-path="WE-8.html"><a href="WE-8.html#a.-goals-7"><i class="fa fa-check"></i>a. Goals</a></li>
<li class="chapter" data-level="" data-path="WE-8.html"><a href="WE-8.html#b.-data-set-7"><i class="fa fa-check"></i>b. Data set</a></li>
<li class="chapter" data-level="" data-path="WE-8.html"><a href="WE-8.html#c.-required-r-libraries-4"><i class="fa fa-check"></i>c. Required R libraries</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="WE-8.html"><a href="WE-8.html#initialize-a-landscape"><i class="fa fa-check"></i>2. Initialize a landscape</a><ul>
<li class="chapter" data-level="" data-path="WE-8.html"><a href="WE-8.html#a.-create-a-random-landscape"><i class="fa fa-check"></i>a. Create a random landscape</a></li>
<li class="chapter" data-level="" data-path="WE-8.html"><a href="WE-8.html#b.-add-populations-to-the-landscape-using-minimal-distance"><i class="fa fa-check"></i>b. Add populations to the landscape (using minimal distance)</a></li>
<li class="chapter" data-level="" data-path="WE-8.html"><a href="WE-8.html#c.-initialise-a-metapopulation"><i class="fa fa-check"></i>c. Initialise a metapopulation</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="WE-8.html"><a href="WE-8.html#define-simulation-parameters"><i class="fa fa-check"></i>3. Define simulation parameters</a><ul>
<li class="chapter" data-level="" data-path="WE-8.html"><a href="WE-8.html#a.-define-your-metapopulation"><i class="fa fa-check"></i>a. Define your metapopulation</a></li>
<li class="chapter" data-level="" data-path="WE-8.html"><a href="WE-8.html#b.-initialise-your-population-on-the-landscape"><i class="fa fa-check"></i>b. Initialise your population on the landscape</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="WE-8.html"><a href="WE-8.html#run-simulations-and-analyze-results"><i class="fa fa-check"></i>4. Run simulations and analyze results</a><ul>
<li class="chapter" data-level="" data-path="WE-8.html"><a href="WE-8.html#a.-run-your-simulation-over-multiple-time-steps-years"><i class="fa fa-check"></i>a. Run your simulation over multiple time steps (years)</a></li>
<li class="chapter" data-level="" data-path="WE-8.html"><a href="WE-8.html#b.-analyse-your-simulated-population-with-a-partial-mantel-test"><i class="fa fa-check"></i>b. Analyse your simulated population with a partial Mantel test</a></li>
<li class="chapter" data-level="" data-path="WE-8.html"><a href="WE-8.html#c.-optional-analyze-your-simulated-populations-using-sunder"><i class="fa fa-check"></i>c. Optional: Analyze your simulated populations using ‘Sunder’</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="WE-8.html"><a href="WE-8.html#run-simulator-using-a-previously-defined-parameter-set"><i class="fa fa-check"></i>5. Run simulator using a previously defined parameter set</a><ul>
<li class="chapter" data-level="" data-path="WE-8.html"><a href="WE-8.html#a.-specify-and-record-the-parameter-combinations"><i class="fa fa-check"></i>a. Specify and record the parameter combinations</a></li>
<li class="chapter" data-level="" data-path="WE-8.html"><a href="WE-8.html#b.-run-the-simulator-over-every-parameter-combination."><i class="fa fa-check"></i>b. Run the simulator over every parameter combination.</a></li>
<li class="chapter" data-level="" data-path="WE-8.html"><a href="WE-8.html#c.-save-your-complete-simulation-run-input-and-output"><i class="fa fa-check"></i>c. Save your complete simulation run (input and output)</a></li>
<li class="chapter" data-level="" data-path="WE-8.html"><a href="WE-8.html#d.-analyze-and-synthesize-results"><i class="fa fa-check"></i>d. Analyze and synthesize results</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="" data-path="r-exercise-week-8.html"><a href="r-exercise-week-8.html"><i class="fa fa-check"></i>R Exercise Week 8</a></li>
<li class="chapter" data-level="" data-path="bonus-8a.html"><a href="bonus-8a.html"><i class="fa fa-check"></i>Bonus: Efficient R</a></li>
</ul></li>
<li class="part"><span><b>Advanced Topics</b></span></li>
<li class="chapter" data-level="" data-path="Week9.html"><a href="Week9.html"><i class="fa fa-check"></i>Week 9: Population Structure</a><ul>
<li class="chapter" data-level="" data-path="WE-9.html"><a href="WE-9.html"><i class="fa fa-check"></i>Worked Example</a><ul>
<li class="chapter" data-level="" data-path="WE-9.html"><a href="WE-9.html#overview-of-worked-example-8"><i class="fa fa-check"></i>1. Overview of Worked Example</a><ul>
<li class="chapter" data-level="" data-path="WE-9.html"><a href="WE-9.html#a.-goals-8"><i class="fa fa-check"></i>a. Goals</a></li>
<li class="chapter" data-level="" data-path="WE-9.html"><a href="WE-9.html#b.-data-sets"><i class="fa fa-check"></i>b. Data sets</a></li>
<li class="chapter" data-level="" data-path="WE-9.html"><a href="WE-9.html#c.-required-r-libraries-5"><i class="fa fa-check"></i>c. Required R libraries</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="WE-9.html"><a href="WE-9.html#simulated-data-2-island-model"><i class="fa fa-check"></i>2. Simulated data: 2-island model</a><ul>
<li class="chapter" data-level="" data-path="WE-9.html"><a href="WE-9.html#a.-import-data"><i class="fa fa-check"></i>a. Import data</a></li>
<li class="chapter" data-level="" data-path="WE-9.html"><a href="WE-9.html#b.-perform-principal-components-analysis-pca"><i class="fa fa-check"></i>b. Perform Principal Components Analysis (PCA)</a></li>
<li class="chapter" data-level="" data-path="WE-9.html"><a href="WE-9.html#c.-create-synthetic-spatial-coordinates-xy-and-map-them"><i class="fa fa-check"></i>c. Create synthetic spatial coordinates (X,Y) and map them</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="WE-9.html"><a href="WE-9.html#simulated-data-2-island-model-with-admixture"><i class="fa fa-check"></i>3. Simulated data: 2-island model with admixture</a></li>
<li class="chapter" data-level="" data-path="WE-9.html"><a href="WE-9.html#empirical-data-threespine-sticklebacks"><i class="fa fa-check"></i>4. Empirical data: Threespine sticklebacks</a><ul>
<li class="chapter" data-level="" data-path="WE-9.html"><a href="WE-9.html#a.-import-the-data"><i class="fa fa-check"></i>a. Import the data:</a></li>
<li class="chapter" data-level="" data-path="WE-9.html"><a href="WE-9.html#b.-examine-population-structure-with-pca"><i class="fa fa-check"></i>b. Examine population structure with PCA</a></li>
<li class="chapter" data-level="" data-path="WE-9.html"><a href="WE-9.html#c.-clustering-with-snmf-similar-to-structure"><i class="fa fa-check"></i>c. Clustering with SNMF (similar to ‘STRUCTURE’)</a></li>
<li class="chapter" data-level="" data-path="WE-9.html"><a href="WE-9.html#d.-plot-ancestry-proportions"><i class="fa fa-check"></i>d. Plot ancestry proportions</a></li>
<li class="chapter" data-level="" data-path="WE-9.html"><a href="WE-9.html#e.-visualize-admixture-proportions-on-a-map"><i class="fa fa-check"></i>e. Visualize admixture proportions on a map</a></li>
</ul></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="" data-path="Week10.html"><a href="Week10.html"><i class="fa fa-check"></i>Week 10: Landscape Resistance</a><ul>
<li class="chapter" data-level="" data-path="WE-10.html"><a href="WE-10.html"><i class="fa fa-check"></i>Worked Example</a><ul>
<li class="chapter" data-level="" data-path="WE-10.html"><a href="WE-10.html#overview-of-worked-example-9"><i class="fa fa-check"></i>1. Overview of Worked Example</a><ul>
<li class="chapter" data-level="" data-path="WE-10.html"><a href="WE-10.html#a.-goals-9"><i class="fa fa-check"></i>a. Goals</a></li>
<li class="chapter" data-level="" data-path="WE-10.html"><a href="WE-10.html#b.-data-set-8"><i class="fa fa-check"></i>b. Data set</a></li>
<li class="chapter" data-level="" data-path="WE-10.html"><a href="WE-10.html#c.-required-r-libraries-6"><i class="fa fa-check"></i>c. Required R libraries</a></li>
<li class="chapter" data-level="" data-path="WE-10.html"><a href="WE-10.html#d.-import-rasters-gsp-err-cti-and-ffp."><i class="fa fa-check"></i>d. Import rasters (gsp, err, cti, and ffp).</a></li>
<li class="chapter" data-level="" data-path="WE-10.html"><a href="WE-10.html#e.-import-site-data-with-sampling-locations"><i class="fa fa-check"></i>e. Import site data with sampling locations</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="WE-10.html"><a href="WE-10.html#explore-the-data-set"><i class="fa fa-check"></i>2. Explore the data set</a><ul>
<li class="chapter" data-level="" data-path="WE-10.html"><a href="WE-10.html#a.-plot-all-rasters"><i class="fa fa-check"></i>a. Plot all rasters</a></li>
<li class="chapter" data-level="" data-path="WE-10.html"><a href="WE-10.html#b.-plot-spatial-points-over-ffp-raster"><i class="fa fa-check"></i>b. Plot spatial points over ffp raster</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="WE-10.html"><a href="WE-10.html#setting-cost-values-and-calculating-conductance"><i class="fa fa-check"></i>3. Setting cost values and calculating conductance</a><ul>
<li class="chapter" data-level="" data-path="WE-10.html"><a href="WE-10.html#a.-resistance-vs.-conductance-values"><i class="fa fa-check"></i>a. Resistance vs. conductance values</a></li>
<li class="chapter" data-level="" data-path="WE-10.html"><a href="WE-10.html#b.-as-needed-get-all-rasters-at-the-same-scale"><i class="fa fa-check"></i>b. As needed: Get all rasters at the same scale</a></li>
<li class="chapter" data-level="" data-path="WE-10.html"><a href="WE-10.html#c.-calculate-conductance-values"><i class="fa fa-check"></i>c. Calculate conductance values</a></li>
<li class="chapter" data-level="" data-path="WE-10.html"><a href="WE-10.html#d.-create-a-single-landscape-conductance-raster"><i class="fa fa-check"></i>d. Create a single landscape conductance raster</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="WE-10.html"><a href="WE-10.html#convert-conductance-into-effective-distance"><i class="fa fa-check"></i>4. Convert conductance into effective distance</a><ul>
<li class="chapter" data-level="" data-path="WE-10.html"><a href="WE-10.html#a.-create-a-transition-layer"><i class="fa fa-check"></i>a. Create a transition layer</a></li>
<li class="chapter" data-level="" data-path="WE-10.html"><a href="WE-10.html#b.-visually-inspect-the-raster"><i class="fa fa-check"></i>b. Visually inspect the raster</a></li>
<li class="chapter" data-level="" data-path="WE-10.html"><a href="WE-10.html#c.-correct-for-geometric-distortion"><i class="fa fa-check"></i>c. Correct for geometric distortion</a></li>
<li class="chapter" data-level="" data-path="WE-10.html"><a href="WE-10.html#d.-plot-shortest-paths-in-space"><i class="fa fa-check"></i>d. Plot shortest paths in space</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="WE-10.html"><a href="WE-10.html#create-cost-distance-matrices"><i class="fa fa-check"></i>5. Create cost-distance matrices</a><ul>
<li class="chapter" data-level="" data-path="WE-10.html"><a href="WE-10.html#a.-least-cost-distance"><i class="fa fa-check"></i>a. Least cost distance</a></li>
<li class="chapter" data-level="" data-path="WE-10.html"><a href="WE-10.html#b.-cost-distance-matrix-based-on-random-paths-similar-to-circuitscape"><i class="fa fa-check"></i>b. Cost-distance matrix based on random paths (similar to Circuitscape)</a></li>
<li class="chapter" data-level="" data-path="WE-10.html"><a href="WE-10.html#c.-compare-cost-distances"><i class="fa fa-check"></i>c. Compare cost distances</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="WE-10.html"><a href="WE-10.html#how-does-changing-resolution-affect-these-metrics"><i class="fa fa-check"></i>6. How does changing resolution affect these metrics?</a><ul>
<li class="chapter" data-level="" data-path="WE-10.html"><a href="WE-10.html#a.-create-loop"><i class="fa fa-check"></i>a. Create loop</a></li>
<li class="chapter" data-level="" data-path="WE-10.html"><a href="WE-10.html#b.-plot-the-results"><i class="fa fa-check"></i>b. Plot the results</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="WE-10.html"><a href="WE-10.html#references"><i class="fa fa-check"></i>7. References</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="" data-path="Week11.html"><a href="Week11.html"><i class="fa fa-check"></i>Week 11: Detecting Adaptation</a><ul>
<li class="chapter" data-level="" data-path="WE-11.html"><a href="WE-11.html"><i class="fa fa-check"></i>Worked Example</a><ul>
<li class="chapter" data-level="" data-path="WE-11.html"><a href="WE-11.html#overview-of-worked-example-10"><i class="fa fa-check"></i>1. Overview of Worked Example</a><ul>
<li class="chapter" data-level="" data-path="WE-11.html"><a href="WE-11.html#a.-goals-10"><i class="fa fa-check"></i>a. Goals</a></li>
<li class="chapter" data-level="" data-path="WE-11.html"><a href="WE-11.html#b.-data-set-9"><i class="fa fa-check"></i>b. Data set</a></li>
<li class="chapter" data-level="" data-path="WE-11.html"><a href="WE-11.html#c.-required-r-libraries-7"><i class="fa fa-check"></i>c. Required R libraries</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="WE-11.html"><a href="WE-11.html#import-and-prepare-the-data"><i class="fa fa-check"></i>2. Import and prepare the data</a><ul>
<li class="chapter" data-level="" data-path="WE-11.html"><a href="WE-11.html#a.-import-the-genetic-data"><i class="fa fa-check"></i>a. Import the genetic data</a></li>
<li class="chapter" data-level="" data-path="WE-11.html"><a href="WE-11.html#b.-import-the-environmental-data"><i class="fa fa-check"></i>b. Import the environmental data</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="WE-11.html"><a href="WE-11.html#latent-factor-mixed-models-lfmm-a-univariate-gea"><i class="fa fa-check"></i>3. Latent Factor Mixed Models (LFMM): a univariate GEA</a><ul>
<li class="chapter" data-level="" data-path="WE-11.html"><a href="WE-11.html#a.-determine-k"><i class="fa fa-check"></i>a. Determine K</a></li>
<li class="chapter" data-level="" data-path="WE-11.html"><a href="WE-11.html#b.-run-lfmm"><i class="fa fa-check"></i>b. Run LFMM</a></li>
<li class="chapter" data-level="" data-path="WE-11.html"><a href="WE-11.html#c.-identify-lfmm-candidates-using-false-discovery-rate"><i class="fa fa-check"></i>c. Identify LFMM candidates using False Discovery Rate</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="WE-11.html"><a href="WE-11.html#redundancy-analysis-rda-a-multivariate-gea"><i class="fa fa-check"></i>4. Redundancy Analysis (RDA): a multivariate GEA</a><ul>
<li class="chapter" data-level="" data-path="WE-11.html"><a href="WE-11.html#a.-run-rda"><i class="fa fa-check"></i>a. Run RDA</a></li>
<li class="chapter" data-level="" data-path="WE-11.html"><a href="WE-11.html#b.-identify-rda-candidates"><i class="fa fa-check"></i>b. Identify RDA candidates</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="WE-11.html"><a href="WE-11.html#compare-lfmm-and-rda-candidates"><i class="fa fa-check"></i>5. Compare LFMM and RDA candidates</a></li>
<li class="chapter" data-level="" data-path="WE-11.html"><a href="WE-11.html#a-quick-note-on-controlling-for-population-structure-in-rda"><i class="fa fa-check"></i>6. A quick note on controlling for population structure in RDA</a></li>
<li class="chapter" data-level="" data-path="WE-11.html"><a href="WE-11.html#references-1"><i class="fa fa-check"></i>7. References</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="" data-path="Week12.html"><a href="Week12.html"><i class="fa fa-check"></i>Week 12: Model Selection</a><ul>
<li class="chapter" data-level="" data-path="WE-12.html"><a href="WE-12.html"><i class="fa fa-check"></i>Worked Example</a><ul>
<li class="chapter" data-level="" data-path="WE-12.html"><a href="WE-12.html#overview-of-worked-example-11"><i class="fa fa-check"></i>1. Overview of Worked Example</a><ul>
<li class="chapter" data-level="" data-path="WE-12.html"><a href="WE-12.html#a.-goals-and-background"><i class="fa fa-check"></i>a. Goals and Background</a></li>
<li class="chapter" data-level="" data-path="WE-12.html"><a href="WE-12.html#b.-data-set-10"><i class="fa fa-check"></i>b. Data set</a></li>
<li class="chapter" data-level="" data-path="WE-12.html"><a href="WE-12.html#c.-required-packages"><i class="fa fa-check"></i>c. Required packages</a></li>
<li class="chapter" data-level="" data-path="WE-12.html"><a href="WE-12.html#d.-import-data-1"><i class="fa fa-check"></i>d. Import data</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="WE-12.html"><a href="WE-12.html#fitting-candidate-models"><i class="fa fa-check"></i>2. Fitting candidate models</a><ul>
<li class="chapter" data-level="" data-path="WE-12.html"><a href="WE-12.html#a.-alternative-hypotheses"><i class="fa fa-check"></i>a. Alternative hypotheses</a></li>
<li class="chapter" data-level="" data-path="WE-12.html"><a href="WE-12.html#b.-prepare-for-model-fitting"><i class="fa fa-check"></i>b. Prepare for model fitting</a></li>
<li class="chapter" data-level="" data-path="WE-12.html"><a href="WE-12.html#c.-fit-all-five-models"><i class="fa fa-check"></i>c. Fit all five models</a></li>
<li class="chapter" data-level="" data-path="WE-12.html"><a href="WE-12.html#d.-compare-evidence-for-models"><i class="fa fa-check"></i>d. Compare evidence for models</a></li>
<li class="chapter" data-level="" data-path="WE-12.html"><a href="WE-12.html#e.-calculate-evidence-weights"><i class="fa fa-check"></i>e. Calculate evidence weights</a></li>
<li class="chapter" data-level="" data-path="WE-12.html"><a href="WE-12.html#f.-confidence-intervals-for-predictors"><i class="fa fa-check"></i>f. Confidence intervals for predictors</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="WE-12.html"><a href="WE-12.html#your-turn-to-test-additional-hypotheses"><i class="fa fa-check"></i>3. Your turn to test additional hypotheses!</a></li>
<li class="chapter" data-level="" data-path="WE-12.html"><a href="WE-12.html#references-2"><i class="fa fa-check"></i>4. References</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="" data-path="Week13.html"><a href="Week13.html"><i class="fa fa-check"></i>Week 13: Gravity Models</a><ul>
<li class="chapter" data-level="" data-path="WE-13.html"><a href="WE-13.html"><i class="fa fa-check"></i>Worked Example</a><ul>
<li class="chapter" data-level="" data-path="WE-13.html"><a href="WE-13.html#overview-of-worked-example-12"><i class="fa fa-check"></i>1. Overview of Worked Example</a><ul>
<li class="chapter" data-level="" data-path="WE-13.html"><a href="WE-13.html#a.-background"><i class="fa fa-check"></i>a. Background</a></li>
<li class="chapter" data-level="" data-path="WE-13.html"><a href="WE-13.html#b.-data-set-11"><i class="fa fa-check"></i>b. Data set</a></li>
<li class="chapter" data-level="" data-path="WE-13.html"><a href="WE-13.html#c.-required-r-libraries-8"><i class="fa fa-check"></i>c. Required R libraries</a></li>
<li class="chapter" data-level="" data-path="WE-13.html"><a href="WE-13.html#set-working-and-data-directories---needs-to-be-removed"><i class="fa fa-check"></i>Set working and data directories - NEEDS TO BE REMOVED</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="WE-13.html"><a href="WE-13.html#calculating-graph-metrics"><i class="fa fa-check"></i>2. Calculating graph metrics</a><ul>
<li class="chapter" data-level="" data-path="WE-13.html"><a href="WE-13.html#a.-read-in-make-a-spatial-object-and-plot-wetlands-wetlands.csv."><i class="fa fa-check"></i>a. Read in, make a spatial object and plot wetlands (Wetlands.csv).</a></li>
<li class="chapter" data-level="" data-path="WE-13.html"><a href="WE-13.html#b.-create-a-graph-from-the-wetlands"><i class="fa fa-check"></i>b. Create a graph from the wetlands</a></li>
<li class="chapter" data-level="" data-path="WE-13.html"><a href="WE-13.html#c.-create-adjacency-matrix"><i class="fa fa-check"></i>c. Create adjacency matrix</a></li>
<li class="chapter" data-level="" data-path="WE-13.html"><a href="WE-13.html#d.-calculate-graph-metrics"><i class="fa fa-check"></i>d. Calculate graph metrics</a></li>
<li class="chapter" data-level="" data-path="WE-13.html"><a href="WE-13.html#e.-add-graph-metrics-to-data.frame"><i class="fa fa-check"></i>e. Add graph metrics to data.frame</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="WE-13.html"><a href="WE-13.html#combine-frog-data-with-graph-metrics"><i class="fa fa-check"></i>3. Combine frog data with graph metrics</a><ul>
<li class="chapter" data-level="" data-path="WE-13.html"><a href="WE-13.html#a.-add-graph-node-data-to-frog-site-data"><i class="fa fa-check"></i>a. Add graph node data to frog site data</a></li>
<li class="chapter" data-level="" data-path="WE-13.html"><a href="WE-13.html#b.-check-data-types"><i class="fa fa-check"></i>b. Check data types</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="WE-13.html"><a href="WE-13.html#merge-graph-with-genetic-distance-data"><i class="fa fa-check"></i>4. Merge graph with genetic distance data</a><ul>
<li class="chapter" data-level="" data-path="WE-13.html"><a href="WE-13.html#a.-build-graph-from-occupied-sites"><i class="fa fa-check"></i>a. Build graph from occupied sites</a></li>
<li class="chapter" data-level="" data-path="WE-13.html"><a href="WE-13.html#b.-merge-graph-with-genetic-distance"><i class="fa fa-check"></i>b. Merge graph with genetic distance</a></li>
<li class="chapter" data-level="" data-path="WE-13.html"><a href="WE-13.html#c.-write-out-results-ascii-csv-and-lines-shapefile."><i class="fa fa-check"></i>c. Write out results (ASCII csv and lines shapefile).</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="WE-13.html"><a href="WE-13.html#preparing-raster-based-covariates"><i class="fa fa-check"></i>5. Preparing raster-based covariates</a><ul>
<li class="chapter" data-level="" data-path="WE-13.html"><a href="WE-13.html#a.-import-raster-data-as-stack"><i class="fa fa-check"></i>a. Import raster data as stack</a></li>
<li class="chapter" data-level="" data-path="WE-13.html"><a href="WE-13.html#b.-calculate-wetland-area-within-buffer"><i class="fa fa-check"></i>b. Calculate wetland area within buffer</a></li>
<li class="chapter" data-level="" data-path="WE-13.html"><a href="WE-13.html#c.-create-raster-of-wetland-proportion"><i class="fa fa-check"></i>c. Create raster of wetland proportion</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="WE-13.html"><a href="WE-13.html#extract-raster-stats-for-nodes-and-edges"><i class="fa fa-check"></i>6. Extract raster stats for nodes and edges</a><ul>
<li class="chapter" data-level="" data-path="WE-13.html"><a href="WE-13.html#a.-extract-at-site-variables-from-rasters"><i class="fa fa-check"></i>a. Extract at-site variables from rasters</a></li>
<li class="chapter" data-level="" data-path="WE-13.html"><a href="WE-13.html#b.-add-covariates-to-graph-edges."><i class="fa fa-check"></i>b. Add covariates to graph edges.</a></li>
<li class="chapter" data-level="" data-path="WE-13.html"><a href="WE-13.html#c.-calculating-stats-along-edges"><i class="fa fa-check"></i>c. Calculating stats along edges</a></li>
<li class="chapter" data-level="" data-path="WE-13.html"><a href="WE-13.html#d.-calculating-stats-for-categorical-variables"><i class="fa fa-check"></i>d. Calculating stats for categorical variables</a></li>
<li class="chapter" data-level="" data-path="WE-13.html"><a href="WE-13.html#e.-add-node-data-to-graph"><i class="fa fa-check"></i>e. Add node data to graph</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="WE-13.html"><a href="WE-13.html#gravity-models"><i class="fa fa-check"></i>7. Gravity Models</a><ul>
<li class="chapter" data-level="" data-path="WE-13.html"><a href="WE-13.html#a.-merge-distance-and-site-data"><i class="fa fa-check"></i>a. Merge distance and site data</a></li>
<li class="chapter" data-level="" data-path="WE-13.html"><a href="WE-13.html#b.-define-a-set-of-models"><i class="fa fa-check"></i>b. Define a set of models</a></li>
<li class="chapter" data-level="" data-path="WE-13.html"><a href="WE-13.html#c.-check-for-correlations"><i class="fa fa-check"></i>c. Check for correlations</a></li>
<li class="chapter" data-level="" data-path="WE-13.html"><a href="WE-13.html#c.-write-out-resultss"><i class="fa fa-check"></i>c. Write out resultss</a></li>
<li class="chapter" data-level="" data-path="WE-13.html"><a href="WE-13.html#e.-compare-models"><i class="fa fa-check"></i>e. Compare models</a></li>
<li class="chapter" data-level="" data-path="WE-13.html"><a href="WE-13.html#f.-diagnostic-plots"><i class="fa fa-check"></i>f. Diagnostic plots</a></li>
<li class="chapter" data-level="" data-path="WE-13.html"><a href="WE-13.html#g.-fit-final-models"><i class="fa fa-check"></i>g. Fit final model(s)</a></li>
<li class="chapter" data-level="" data-path="WE-13.html"><a href="WE-13.html#h.-effect-size"><i class="fa fa-check"></i>h. Effect size</a></li>
</ul></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="" data-path="Week14.html"><a href="Week14.html"><i class="fa fa-check"></i>Week 14: Contemporary Gene Flow</a><ul>
<li class="chapter" data-level="" data-path="WE-14.html"><a href="WE-14.html"><i class="fa fa-check"></i>Worked Example</a><ul>
<li class="chapter" data-level="" data-path="WE-14.html"><a href="WE-14.html#overview-2"><i class="fa fa-check"></i>1. Overview</a><ul>
<li class="chapter" data-level="" data-path="WE-14.html"><a href="WE-14.html#a.-background-and-goals"><i class="fa fa-check"></i>a. Background and Goals</a></li>
<li class="chapter" data-level="" data-path="WE-14.html"><a href="WE-14.html#b.-data-set-12"><i class="fa fa-check"></i>b. Data set</a></li>
<li class="chapter" data-level="" data-path="WE-14.html"><a href="WE-14.html#c.-required-packages-1"><i class="fa fa-check"></i>c. Required packages</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="WE-14.html"><a href="WE-14.html#import-genotypes"><i class="fa fa-check"></i>2. Import Genotypes</a><ul>
<li class="chapter" data-level="" data-path="WE-14.html"><a href="WE-14.html#a.-import-genotype-data"><i class="fa fa-check"></i>a. Import genotype data</a></li>
<li class="chapter" data-level="" data-path="WE-14.html"><a href="WE-14.html#b.-plot-map-of-sampling-locations"><i class="fa fa-check"></i>b. Plot map of sampling locations</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="WE-14.html"><a href="WE-14.html#pollen-pool-genetic-structure"><i class="fa fa-check"></i>3. Pollen pool genetic structure</a><ul>
<li class="chapter" data-level="" data-path="WE-14.html"><a href="WE-14.html#a.-subtract-maternal-contribution-to-offspring-genotype"><i class="fa fa-check"></i>a. Subtract maternal contribution to offspring genotype</a></li>
<li class="chapter" data-level="" data-path="WE-14.html"><a href="WE-14.html#b.-twogener-analysis-amova-of-pollen-pool-data"><i class="fa fa-check"></i>b. ‘TwoGener’ analysis (AMOVA of pollen pool data)</a></li>
<li class="chapter" data-level="" data-path="WE-14.html"><a href="WE-14.html#c.-including-space-isolation-by-distance"><i class="fa fa-check"></i>c. Including space: Isolation by distance</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="WE-14.html"><a href="WE-14.html#paternity-analysis"><i class="fa fa-check"></i>4. Paternity analysis</a><ul>
<li class="chapter" data-level="" data-path="WE-14.html"><a href="WE-14.html#a.-exclusion-probabilities"><i class="fa fa-check"></i>a. Exclusion probabilities</a></li>
<li class="chapter" data-level="" data-path="WE-14.html"><a href="WE-14.html#b.-paternity-analysis"><i class="fa fa-check"></i>b. Paternity analysis</a></li>
<li class="chapter" data-level="" data-path="WE-14.html"><a href="WE-14.html#c.-visualize-reconstructed-pollen-flow-events-in-space"><i class="fa fa-check"></i>c. Visualize reconstructed pollen flow events in space</a></li>
<li class="chapter" data-level="" data-path="WE-14.html"><a href="WE-14.html#d.-visualize-dispersal-kernel"><i class="fa fa-check"></i>d. Visualize dispersal kernel</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="WE-14.html"><a href="WE-14.html#linking-paternity-to-ecological-variables"><i class="fa fa-check"></i>5. Linking paternity to ecological variables</a><ul>
<li class="chapter" data-level="" data-path="WE-14.html"><a href="WE-14.html#a.-explain-pollen-flow-within-populations"><i class="fa fa-check"></i>a. Explain pollen flow within populations</a></li>
<li class="chapter" data-level="" data-path="WE-14.html"><a href="WE-14.html#b.-explain-pollen-flow-at-the-population-level"><i class="fa fa-check"></i>b. Explain pollen flow at the population level</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="WE-14.html"><a href="WE-14.html#references-3"><i class="fa fa-check"></i>6. References</a></li>
</ul></li>
</ul></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Landscape Genetic Data Analysis with R</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="WE_2" class="section level2">
<h2>Worked Example</h2>
<div id="overview-of-worked-example-1" class="section level3">
<h3>1. Overview of Worked Example</h3>
<p>This code builds on data and code from the <code>GeNetIt</code> package by Jeff Evans and Melanie Murphy. Landscape metrics will be calculated with the <code>landscapemetrics</code> package described in: Hesselbarth et al. (2019), Ecography 42: 1648-1657.</p>
<div id="a.-goals-1" class="section level4">
<h4>a. Goals</h4>
<p>This worked example shows:</p>
<ul>
<li>How to import spatial coordinates and site attributes as spatially referenced data.<br />
</li>
<li>How to plot raster data in R and overlay sampling locations.</li>
<li>How to calculate landscape metrics.</li>
<li>How to extract landscape data at sampling locations and within a buffer around them.</li>
</ul>
<p>Try modifying the code to import your own data!</p>
</div>
<div id="b.-data-set-1" class="section level4">
<h4>b. Data set</h4>
<p>This code uses landscape data and spatial coordinates from 30 locations where Colombia spotted frogs (<em>Rana luteiventris</em>) were sampled for the full data set analyzed by Funk et al. (2005) and Murphy et al. (2010). Please see the separate introduction to the data set.</p>
<ul>
<li>ralu.site: SpatialPointsDataFrame object with UTM coordinates (zone 11) in slot <code>@coords</code> and 17 site variables in slot <code>@data</code> for 31 sites. The data are included in the ‘GeNetIt’ package, for meta data type: ?ralu.site</li>
</ul>
<p>We will extract values at sampling point locations and within a local neighborhood (buffer) from six raster maps (see Murphy et al. 2010 for definitions), which are included with the <code>GeNetIt</code> package as a SpatialPixelsDataFrame called ‘rasters’:</p>
<ul>
<li>cti: Compound Topographic Index (“wetness”)</li>
<li>err27: Elevation Relief Ratio</li>
<li>ffp: Frost Free Period</li>
<li>gsp: Growing Season Precipitation</li>
<li>hli: Heat Load Index</li>
<li>nlcd: USGS Landcover (categorical map)</li>
</ul>
</div>
<div id="c.-required-r-libraries-1" class="section level4">
<h4>c. Required R libraries</h4>
<p>Install some packages needed for this worked example.</p>
<div class="sourceCode" id="cb313"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb313-1" title="1"><span class="cf">if</span>(<span class="op">!</span><span class="kw">requireNamespace</span>(<span class="st">&quot;GeNetIt&quot;</span>, <span class="dt">quietly =</span> <span class="ot">TRUE</span>)) remotes<span class="op">::</span><span class="kw">install_github</span>(<span class="st">&quot;jeffreyevans/GeNetIt&quot;</span>)</a></code></pre></div>
<div class="sourceCode" id="cb314"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb314-1" title="1"><span class="kw">library</span>(LandGenCourse)</a>
<a class="sourceLine" id="cb314-2" title="2"><span class="kw">library</span>(landscapemetrics)</a>
<a class="sourceLine" id="cb314-3" title="3"><span class="kw">library</span>(dplyr)</a>
<a class="sourceLine" id="cb314-4" title="4"><span class="kw">library</span>(sp)</a>
<a class="sourceLine" id="cb314-5" title="5"><span class="kw">library</span>(raster)</a>
<a class="sourceLine" id="cb314-6" title="6"><span class="kw">library</span>(GeNetIt)</a>
<a class="sourceLine" id="cb314-7" title="7"><span class="kw">library</span>(tibble)</a></code></pre></div>
<p>Package <code>tmaptools</code> not automatically installed with <code>LandGenCourse</code>:</p>
<div class="sourceCode" id="cb315"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb315-1" title="1"><span class="cf">if</span>(<span class="op">!</span><span class="kw">require</span>(tmaptools)) <span class="kw">install.packages</span>(<span class="st">&quot;tmaptools&quot;</span>)</a></code></pre></div>
</div>
</div>
<div id="import-site-data-from-.csv-file" class="section level3">
<h3>2. Import site data from .csv file</h3>
<div id="a.-import-data-into-spatialpointsdataframe" class="section level4">
<h4>a. Import data into SpatialPointsDataFrame</h4>
<p>The site data are already in a SpatialPointsDataFrame named <code>ralu.site</code> that comes with the package <code>GeNetIt</code>. Use <code>data(ralu.site)</code> to load it. This will create an object <code>ralu.site</code>.</p>
<p>To demonstrate how to create a SpatialPointsDataFrame, we create a simple data frame <code>Sites</code> with the coordinates and site data.</p>
<div class="sourceCode" id="cb316"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb316-1" title="1"><span class="kw">data</span>(ralu.site)</a>
<a class="sourceLine" id="cb316-2" title="2"><span class="kw">class</span>(ralu.site)</a></code></pre></div>
<pre><code>## [1] &quot;SpatialPointsDataFrame&quot;
## attr(,&quot;package&quot;)
## [1] &quot;sp&quot;</code></pre>
<div class="sourceCode" id="cb318"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb318-1" title="1">Sites &lt;-<span class="st"> </span><span class="kw">data.frame</span>(ralu.site<span class="op">@</span>coords, ralu.site<span class="op">@</span>data)</a>
<a class="sourceLine" id="cb318-2" title="2"><span class="kw">class</span>(Sites)</a></code></pre></div>
<pre><code>## [1] &quot;data.frame&quot;</code></pre>
<div class="sourceCode" id="cb320"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb320-1" title="1"><span class="kw">head</span>(Sites)</a></code></pre></div>
<pre><code>##   coords.x1 coords.x2       SiteName        Drainage      Basin Substrate
## 1  688816.6   5003207   AirplaneLake ShipIslandCreek Sheepeater      Silt
## 2  688494.4   4999093 BachelorMeadow     WilsonCreek    Skyhigh      Silt
## 3  687938.4   5000223 BarkingFoxLake  WaterfallCreek    Terrace      Silt
## 4  689732.8   5002522   BirdbillLake      ClearCreek   Birdbill      Sand
## 5  690104.0   4999355        BobLake     WilsonCreek     Harbor      Silt
## 6  688742.5   4997481      CacheLake     WilsonCreek    Skyhigh      Silt
##                               NWI AREA_m2 PERI_m Depth_m  TDS FISH ACB   AUC
## 1                      Lacustrine 62582.2 1142.8   21.64  2.5    1   0 0.411
## 2 Riverine_Intermittent_Streambed   225.0   60.0    0.40  0.0    0   0 0.000
## 3                      Lacustrine 12000.0  435.0    5.00 13.8    1   0 0.300
## 4                      Lacustrine 12358.6  572.3    3.93  6.4    1   0 0.283
## 5                      Palustrine  4600.0  321.4    2.00 14.3    0   0 0.000
## 6                      Palustrine  2268.8  192.0    1.86 10.9    0   0 0.000
##   AUCV  AUCC   AUF AWOOD  AUFV
## 1    0 0.411 0.063 0.063 0.464
## 2    0 0.000 1.000 0.000 0.000
## 3    0 0.300 0.700 0.000 0.000
## 4    0 0.283 0.717 0.000 0.000
## 5    0 0.000 0.500 0.000 0.500
## 6    0 0.000 0.556 0.093 0.352</code></pre>
<p><strong>Question</strong>: What are the variable names for the spatial coordinates?</p>
<p>To illustrate importing spatial data from Excel, here we export the data as a csv file, import it again as a data frame, then convert it to a SpatialPointsDataFrame. First we create a folder <code>output</code> if it does not yet exist.</p>
<p>Note: to run the code, remove all the hashtags <code>#</code> at the beginning of the lines to uncomment them. This part assumes that you have writing permission on your computer. Alternatively, try setting up your R project folder on an external drive where you have writing permission.</p>
<p>The second line exports the data in <code>Sites</code> as a .csv file. The third line re-imports the .csv file to re-create data frame <code>Sites</code>.</p>
<div class="sourceCode" id="cb322"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb322-1" title="1"><span class="co">#require(here)</span></a>
<a class="sourceLine" id="cb322-2" title="2"><span class="co">#if(!dir.exists(paste0(here(),&quot;/output&quot;))) dir.create(paste0(here(),&quot;/output&quot;))</span></a>
<a class="sourceLine" id="cb322-3" title="3"><span class="co">#write.csv(Sites, file=paste0(here(),&quot;/output/ralu.site.csv&quot;), </span></a>
<a class="sourceLine" id="cb322-4" title="4"><span class="co">#          quote=FALSE, row.names=FALSE)</span></a>
<a class="sourceLine" id="cb322-5" title="5"><span class="co">#Sites &lt;- read.csv(paste0(here(),&quot;/output/ralu.site.csv&quot;), header=TRUE)</span></a></code></pre></div>
<p>The dataset <code>Sites</code> contains two columns with spatial coordinates and 17 attribute variables. So far, R treats the spatial coordinates like any other quantitative variables. To let R know this is spatial information, we import it into a spatial object type, a <code>SpatialPointsDataFrame</code> from the ‘sp’ package.</p>
<p>The conversion is done with the function <code>coordinates</code>, which takes a data frame and converts it to a spatial object of the same name. The code is not very intuitive.</p>
<p>Note: the tilde symbol <code>~</code> (here before the first coordinate) is often used in R formulas, we will see it again later. It roughly translates to “is modeled as a function of”.</p>
<div class="sourceCode" id="cb323"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb323-1" title="1">Sites.sp &lt;-<span class="st"> </span>Sites</a>
<a class="sourceLine" id="cb323-2" title="2"><span class="kw">coordinates</span>(Sites.sp) &lt;-<span class="st"> </span><span class="er">~</span>coords.x1<span class="op">+</span>coords.x2</a></code></pre></div>
<p>Now R knows these are spatial data and knows how to handle them. It does not treat the coordinates as variables anymore, hence the first column is now <code>SiteName</code>.</p>
</div>
<div id="b.-add-spatial-reference-data" class="section level4">
<h4>b. Add spatial reference data</h4>
<p>Before we can combine the sampling locations with other spatial datasets, such as raster data, we need to tell R where on earth these locations are (georeferencing). This is done by specifying the “Coordinate Reference System” (CRS) or a <code>proj4</code> string.</p>
<p>For more information on CRS, see: <a href="https://www.nceas.ucsb.edu/~frazier/RSpatialGuides/OverviewCoordinateReferenceSystems.pdf" class="uri">https://www.nceas.ucsb.edu/~frazier/RSpatialGuides/OverviewCoordinateReferenceSystems.pdf</a></p>
<p>We know that these coordinates are UTM zone 11 (Northern hemisphere) coordinates, hence we can use a helper function to find the correct <code>proj4string</code>, using function <code>get_proj4</code> from the <code>tmaptools</code> package. (For the Southern hemisphere, you would add <code>s</code> after the zone: <code>utm11s</code>). Here we call the function and the package simultaneously (this is good practice, as it helps keep track of where the functions in your code come from).</p>
<div class="sourceCode" id="cb324"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb324-1" title="1"><span class="kw">proj4string</span>(Sites.sp) &lt;-<span class="st"> </span>tmaptools<span class="op">::</span><span class="kw">get_proj4</span>(<span class="st">&quot;utm11&quot;</span>)<span class="op">$</span>proj4string</a></code></pre></div>
<pre><code>## Warning in showSRID(uprojargs, format = &quot;PROJ&quot;, multiline = &quot;NO&quot;, prefer_proj
## = prefer_proj): Discarded datum Unknown based on WGS84 ellipsoid in Proj4
## definition</code></pre>
<p>Note: this function only <strong>declares</strong> the existing projection, it does not change the coordinates to that projection.</p>
<p>If we had longitude and latitude coordinates, we would modify the command like this:
proj4string(Sites.sp) &lt;- tmaptools::get_proj4(“longlat”)$proj4string.</p>
</div>
<div id="c.-change-projection" class="section level4">
<h4>c. Change projection</h4>
<p>In case we needed to <strong>transform</strong> the projection from UTM zone 11 to longitude/latitude, we could create a new sp object <code>Sites.sp.longlat</code>. We use the function <code>spTransform</code> to change the projection from the projection of the old object <code>Sites.sp</code> to the “longlat” coordinate system, which we define by the argument <code>CRSobj</code>.</p>
<div class="sourceCode" id="cb326"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb326-1" title="1">Sites.sp.longlat &lt;-<span class="st"> </span>sp<span class="op">::</span><span class="kw">spTransform</span>(Sites.sp, </a>
<a class="sourceLine" id="cb326-2" title="2">                      <span class="dt">CRSobj =</span> tmaptools<span class="op">::</span><span class="kw">get_proj4</span>(<span class="st">&quot;longlat&quot;</span>)<span class="op">$</span>proj4string)</a>
<a class="sourceLine" id="cb326-3" title="3"><span class="kw">head</span>(Sites.sp.longlat<span class="op">@</span>coords)</a></code></pre></div>
<pre><code>##   coords.x1 coords.x2
## 1 -114.5977  45.15708
## 2 -114.6034  45.12016
## 3 -114.6100  45.13047
## 4 -114.5864  45.15067
## 5 -114.5828  45.12208
## 6 -114.6008  45.10560</code></pre>
<p><strong>Question</strong>: Where on earth is this? You can enter the coordinates from the “longlat” projection in Google maps. Note that Google expects the Latitude (Y coordinate) first, then the Longitude (X coordinate). Here, coords.x1 is the longitude (X) and coords.x2 is the latitude (Y). Thus, to locate the first site in Google maps, you will need to enter <code>45.15708, -114.5977</code>. Where is it located?</p>
</div>
<div id="d.-access-data-in-spatialpointsdataframe" class="section level4">
<h4>d. Access data in ‘SpatialPointsDataFrame’</h4>
<p>As an S4 object, Sites.sp has predefined slots. These can be accessed with the <code>@</code> symbol:</p>
<ul>
<li><code>@data</code>: the attribute data</li>
<li><code>@coords</code>: the spatial coordinates</li>
<li><code>@coords</code>.nrs: the column numbers of the input data from which the coordinates were taken (filled automatically)</li>
<li><code>@bbox</code>: bounding box, i.e., the minimum and maximum of x and y coordinates (filled automatically)</li>
<li><code>@proj4string</code>: the georeferencing information</li>
</ul>
<div class="sourceCode" id="cb328"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb328-1" title="1"><span class="kw">slotNames</span>(Sites.sp)</a></code></pre></div>
<pre><code>## [1] &quot;data&quot;        &quot;coords.nrs&quot;  &quot;coords&quot;      &quot;bbox&quot;        &quot;proj4string&quot;</code></pre>
<p>Here are the first few lines of the coordinates:</p>
<div class="sourceCode" id="cb330"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb330-1" title="1"><span class="kw">head</span>(Sites.sp<span class="op">@</span>coords)</a></code></pre></div>
<pre><code>##   coords.x1 coords.x2
## 1  688816.6   5003207
## 2  688494.4   4999093
## 3  687938.4   5000223
## 4  689732.8   5002522
## 5  690104.0   4999355
## 6  688742.5   4997481</code></pre>
<p>And the proj4 string: Let’s compare this to the proj4string of the original <code>ralu.site</code> dataset.</p>
<div class="sourceCode" id="cb332"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb332-1" title="1">Sites.sp<span class="op">@</span>proj4string</a></code></pre></div>
<pre><code>## CRS arguments:
##  +proj=utm +zone=11 +ellps=WGS84 +units=m +no_defs</code></pre>
<p>The default for <code>get_proj4("utm11")</code> results in a slightly different proj4string than the <code>ralu.site</code> dataset. The difference is in the <code>datum</code> argument (<code>WGS84</code> vs. <code>NAD83</code>):</p>
<div class="sourceCode" id="cb334"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb334-1" title="1">ralu.site<span class="op">@</span>proj4string</a></code></pre></div>
<pre><code>## CRS arguments:
##  +proj=utm +zone=11 +datum=NAD83 +units=m +no_defs +ellps=GRS80
## +towgs84=0,0,0</code></pre>
<p>Let’s go with the original information and copy it:</p>
<div class="sourceCode" id="cb336"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb336-1" title="1">Sites.sp<span class="op">@</span>proj4string &lt;-<span class="st"> </span>ralu.site<span class="op">@</span>proj4string</a></code></pre></div>
</div>
</div>
<div id="display-raster-data-and-overlay-sampling-locations-extract-data" class="section level3">
<h3>3. Display raster data and overlay sampling locations, extract data</h3>
<div id="a.-display-raster-data" class="section level4">
<h4>a. Display raster data</h4>
<p>The raster data for this project are already available in the package <code>GeNetIt</code>, under the name <code>rasters</code>, and we can load them with <code>data(rasters)</code>. They are stored as a <code>SpatialPixelsDataFrame</code>, another S4 object type from the <code>sp</code> package.</p>
<div class="sourceCode" id="cb337"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb337-1" title="1"><span class="kw">data</span>(rasters)</a>
<a class="sourceLine" id="cb337-2" title="2"><span class="kw">class</span>(rasters)</a></code></pre></div>
<pre><code>## [1] &quot;SpatialPixelsDataFrame&quot;
## attr(,&quot;package&quot;)
## [1] &quot;sp&quot;</code></pre>
<p>However, raster data are better analyzed with the package <code>raster</code>, which has an object type <code>raster</code>. Let’s convert the data to a <code>RasterStack</code> of <code>RasterLayer</code> objects (i.e. a set of raster layers with the same spatial reference information).</p>
<div class="sourceCode" id="cb339"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb339-1" title="1">RasterMaps &lt;-<span class="st"> </span><span class="kw">stack</span>(rasters)</a>
<a class="sourceLine" id="cb339-2" title="2"><span class="kw">class</span>(RasterMaps)</a></code></pre></div>
<pre><code>## [1] &quot;RasterStack&quot;
## attr(,&quot;package&quot;)
## [1] &quot;raster&quot;</code></pre>
<p>Printing the name of the raster stack displays a summary. A few explanations:</p>
<ul>
<li><strong>dimensions</strong>: number of rows (nrow), number of columns (ncol), number of cells (ncell), number of layers (nlayers). So we see there are 6 layers in the raster stack.</li>
<li><strong>resolution</strong>: cell size is 30 m both in x and y directions (typical for Landsat-derived remote sensing data)</li>
<li><strong>coord.ref</strong>: projected in UTM zone 11, though the ‘datum’ (NAD83) is different than what we used for the sampling locations.</li>
</ul>
<div class="sourceCode" id="cb341"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb341-1" title="1">RasterMaps</a></code></pre></div>
<pre><code>## class      : RasterStack 
## dimensions : 426, 358, 152508, 6  (nrow, ncol, ncell, nlayers)
## resolution : 30, 30  (x, y)
## extent     : 683282.5, 694022.5, 4992833, 5005613  (xmin, xmax, ymin, ymax)
## crs        : +proj=utm +zone=11 +datum=NAD83 +units=m +no_defs 
## names      :          cti,        err27,          ffp,          gsp,          hli,         nlcd 
## min values : 8.429851e-01, 3.906551e-02, 0.000000e+00, 2.270000e+02, 1.014000e+03, 1.100000e+01 
## max values :   23.7147598,    0.7637643,   51.0000000,  338.0696716, 9263.0000000,   95.0000000</code></pre>
<p>Now we can use <code>plot</code>, which knows what to do with a raster stack.</p>
<p>Note: layer <code>nlcd</code> is a categorical map of land cover types. See this week’s bonus materials for how to better display a categorical map in R.</p>
<div class="sourceCode" id="cb343"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb343-1" title="1"><span class="kw">plot</span>(RasterMaps)</a></code></pre></div>
<p><img src="02-Week02_files/figure-html/unnamed-chunk-18-1.png" width="768" /></p>
<p>Some layers seem to show a similar pattern. It is easy to calculate the correlation between quantitative raster layers. Here, the last layer <code>ncld</code>, is in fact categorical (land cover type), and it’s correlation here is meaningless.</p>
<div class="sourceCode" id="cb344"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb344-1" title="1"><span class="kw">layerStats</span>(RasterMaps, <span class="st">&#39;pearson&#39;</span>, <span class="dt">na.rm=</span>T)</a></code></pre></div>
<pre><code>## $`pearson correlation coefficient`
##              cti       err27         ffp         gsp         hli       nlcd
## cti    1.0000000 -0.25442672  0.12264734 -0.14029572 -0.30501483 -0.1807878
## err27 -0.2544267  1.00000000 -0.23467075  0.21403415  0.07724426  0.1256296
## ffp    0.1226473 -0.23467075  1.00000000 -0.95144256 -0.07567975 -0.3297561
## gsp   -0.1402957  0.21403415 -0.95144256  1.00000000  0.09520075  0.3765363
## hli   -0.3050148  0.07724426 -0.07567975  0.09520075  1.00000000  0.2465540
## nlcd  -0.1807878  0.12562961 -0.32975610  0.37653635  0.24655404  1.0000000
## 
## $mean
##          cti        err27          ffp          gsp          hli         nlcd 
##    5.3386441    0.4509513   11.2037444  277.2211529 1938.3644530   50.8191308</code></pre>
</div>
<div id="b.-change-color-ramp-add-sampling-locations" class="section level4">
<h4>b. Change color ramp, add sampling locations</h4>
<p>We can specify a color ramp by setting the <code>col</code> argument. The default is <code>terrain.colors(255)</code>. Here we change it to <code>rainbow(9)</code>, a rainbow colorpalette with 9 color levels.</p>
<p>Note: To learn about options for the <code>plot</code> function for <code>raster</code> objects, access the help file by typing <code>?plot</code> and select <code>Plot a Raster object</code>.</p>
<p>We can add the sampling locations (if we plot only a single raster layer). Here we use <code>rev</code> to reverse the color ramp for plotting raster layer <code>ffp</code>, and add the sites as white circles with black outlines.</p>
<div class="sourceCode" id="cb346"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb346-1" title="1"><span class="kw">par</span>(<span class="dt">mar=</span><span class="kw">c</span>(<span class="dv">3</span>,<span class="dv">3</span>,<span class="dv">1</span>,<span class="dv">2</span>))</a>
<a class="sourceLine" id="cb346-2" title="2"><span class="kw">plot</span>(<span class="kw">raster</span>(RasterMaps, <span class="dt">layer=</span><span class="st">&quot;ffp&quot;</span>), <span class="dt">col=</span><span class="kw">rev</span>(<span class="kw">rainbow</span>(<span class="dv">9</span>)))</a>
<a class="sourceLine" id="cb346-3" title="3"><span class="kw">points</span>(Sites.sp, <span class="dt">pch=</span><span class="dv">21</span>, <span class="dt">col=</span><span class="st">&quot;black&quot;</span>, <span class="dt">bg=</span><span class="st">&quot;white&quot;</span>)</a></code></pre></div>
<p><img src="02-Week02_files/figure-html/unnamed-chunk-20-1.png" width="427.2" /></p>
<p><strong>Question</strong>: Recall that ‘ffp’ stands for frost free period (in days). What do you think is the average length of the frost free period at theses sampling sites?</p>
</div>
<div id="c.-extract-raster-values-at-sampling-locations" class="section level4">
<h4>c. Extract raster values at sampling locations</h4>
<p>The following code adds six variables to the data slot of Sites.sp. Technically we combine the columns of the existing data frame <code>Sites.sp</code> with the new columns in a new data frame with the same name.</p>
<p>R notices the difference in projection (CRS) between the sampling point data and the rasters and takes care of it, providing just a warning.</p>
<div class="sourceCode" id="cb347"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb347-1" title="1">Sites.sp<span class="op">@</span>data &lt;-<span class="st"> </span><span class="kw">data.frame</span>(Sites.sp<span class="op">@</span>data, <span class="kw">extract</span>(RasterMaps, Sites.sp))</a></code></pre></div>
<p>Let’s calcualate the mean length of the frost free period for these sites:</p>
<div class="sourceCode" id="cb348"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb348-1" title="1"><span class="kw">mean</span>(Sites.sp<span class="op">@</span>data<span class="op">$</span>ffp)</a></code></pre></div>
<pre><code>## [1] 8.0963</code></pre>
<p>What land cover type is assigned to the most sampling units? Let’s tabulate them.</p>
<div class="sourceCode" id="cb350"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb350-1" title="1"><span class="kw">table</span>(Sites.sp<span class="op">@</span>data<span class="op">$</span>nlcd)</a></code></pre></div>
<pre><code>## 
## 11 12 42 52 71 90 
##  3  1 21  1  4  1</code></pre>
<p>Note: land cover types are coded by numbers. Check here what the numbers mean: <a href="https://www.mrlc.gov/data/legends/national-land-cover-database-2016-nlcd2016-legend" class="uri">https://www.mrlc.gov/data/legends/national-land-cover-database-2016-nlcd2016-legend</a></p>
<p><strong>Question</strong>: A total of 21 sites are classified as <code>42</code>. What is this most frequent land cover type?</p>
</div>
</div>
<div id="calculate-landscape-metrics" class="section level3">
<h3>4. Calculate landscape metrics</h3>
<p>We are going to use the package <a href="https://r-spatialecology.github.io/landscapemetrics/"><code>landscapemetrics</code></a>. It is an R package to calculate landscape metrics in a tidy workflow (for more information about tidy data see <a href="https://www.jstatsoft.org/article/view/v059i10/">here</a>). <code>landscapemetrics</code> is basically a reimplementation of <a href="https://www.umass.edu/landeco/research/fragstats/fragstats.html">‘FRAGSTATS’</a>, which allows an integration into larger workflows within the R environment. The core of the package are functions to calculate landscape metrics, but also several auxiliary functions exit.</p>
<p>To facilitate an integration into larger workflows, <code>landscapemetrics</code> is based on the <a href="https://CRAN.R-project.org/package=raster"><code>raster</code> package</a>. To check if a raster is suitable for <code>landscapemetrics</code>, run the <code>check_landscape()</code> function first. The function checks the coordinate reference system (and mainly if units are in meters) and if the raster values are discrete classes. If the check fails, the calculation of metrics is still possible, however, especially metrics that are based on area and distances must be used with caution.</p>
<div class="sourceCode" id="cb352"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb352-1" title="1">nlcd &lt;-<span class="st"> </span><span class="kw">raster</span>(RasterMaps, <span class="dt">layer =</span> <span class="st">&quot;nlcd&quot;</span>)</a>
<a class="sourceLine" id="cb352-2" title="2"></a>
<a class="sourceLine" id="cb352-3" title="3">landscapemetrics<span class="op">::</span><span class="kw">check_landscape</span>(nlcd)</a></code></pre></div>
<pre><code>##   layer       crs units   class n_classes OK
## 1     1 projected     m integer         8  v</code></pre>
<p>There are three different levels of landscape metrics. Firstly, metrics can be calculated for each single patch (a patch is defined as neighbouring cells of the same class). Secondly, metrics can be calculated for a certain class (i.e. all patches belonging to the same class) and lastly for the whole landscape. All these levels are implemented and easily accessible in <code>landscapemetrics</code>.</p>
<p>All functions to calculate metrics start with <code>lsm_</code> (for landscapemetrics). The second part of the name specifies the level (patch - <code>p</code>, class - <code>c</code> or landscape - <code>l</code>). Lastly, the final part of the function name is the abbreviation of the corresponding metric (e.g. <code>enn</code> for the Euclidean nearest-neighbor distance). To list all available metrics, you can use the <code>list_lsm()</code> function. The function also allows to show metrics filtered by level, type or metric name. For more information about the metrics, please see either the corresponding helpfile(s) or <a href="https://r-spatialecology.github.io/landscapemetrics/reference/index.html">https://r-spatialecology.github.io/landscapemetrics</a>.</p>
<div class="sourceCode" id="cb354"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb354-1" title="1">landscapemetrics<span class="op">::</span><span class="kw">list_lsm</span>(<span class="dt">level =</span> <span class="st">&quot;landscape&quot;</span>, <span class="dt">type =</span> <span class="st">&quot;diversity metric&quot;</span>)</a></code></pre></div>
<pre><code>## # A tibble: 9 x 5
##   metric name                             type            level    function_name
##   &lt;chr&gt;  &lt;chr&gt;                            &lt;chr&gt;           &lt;chr&gt;    &lt;chr&gt;        
## 1 msidi  modified simpson&#39;s diversity in~ diversity metr~ landsca~ lsm_l_msidi  
## 2 msiei  modified simpson&#39;s evenness ind~ diversity metr~ landsca~ lsm_l_msiei  
## 3 pr     patch richness                   diversity metr~ landsca~ lsm_l_pr     
## 4 prd    patch richness density           diversity metr~ landsca~ lsm_l_prd    
## 5 rpr    relative patch richness          diversity metr~ landsca~ lsm_l_rpr    
## 6 shdi   shannon&#39;s diversity index        diversity metr~ landsca~ lsm_l_shdi   
## 7 shei   shannon&#39;s evenness index         diversity metr~ landsca~ lsm_l_shei   
## 8 sidi   simpson&#39;s diversity index        diversity metr~ landsca~ lsm_l_sidi   
## 9 siei   simspon&#39;s evenness index         diversity metr~ landsca~ lsm_l_siei</code></pre>
<div class="sourceCode" id="cb356"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb356-1" title="1">landscapemetrics<span class="op">::</span><span class="kw">list_lsm</span>(<span class="dt">metric =</span> <span class="st">&quot;area&quot;</span>)</a></code></pre></div>
<pre><code>## # A tibble: 7 x 5
##   metric  name       type                 level     function_name
##   &lt;chr&gt;   &lt;chr&gt;      &lt;chr&gt;                &lt;chr&gt;     &lt;chr&gt;        
## 1 area    patch area area and edge metric patch     lsm_p_area   
## 2 area_cv patch area area and edge metric class     lsm_c_area_cv
## 3 area_mn patch area area and edge metric class     lsm_c_area_mn
## 4 area_sd patch area area and edge metric class     lsm_c_area_sd
## 5 area_cv patch area area and edge metric landscape lsm_l_area_cv
## 6 area_mn patch area area and edge metric landscape lsm_l_area_mn
## 7 area_sd patch area area and edge metric landscape lsm_l_area_sd</code></pre>
<div class="sourceCode" id="cb358"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb358-1" title="1">landscapemetrics<span class="op">::</span><span class="kw">list_lsm</span>(<span class="dt">level =</span> <span class="kw">c</span>(<span class="st">&quot;class&quot;</span>, <span class="st">&quot;landscape&quot;</span>), <span class="dt">type =</span> <span class="st">&quot;aggregation metric&quot;</span>, </a>
<a class="sourceLine" id="cb358-2" title="2">                           <span class="dt">simplify =</span> <span class="ot">TRUE</span>)</a></code></pre></div>
<pre><code>##  [1] &quot;lsm_c_ai&quot;       &quot;lsm_c_clumpy&quot;   &quot;lsm_c_cohesion&quot; &quot;lsm_c_division&quot;
##  [5] &quot;lsm_c_enn_cv&quot;   &quot;lsm_c_enn_mn&quot;   &quot;lsm_c_enn_sd&quot;   &quot;lsm_c_iji&quot;     
##  [9] &quot;lsm_c_lsi&quot;      &quot;lsm_c_mesh&quot;     &quot;lsm_c_nlsi&quot;     &quot;lsm_c_np&quot;      
## [13] &quot;lsm_c_pd&quot;       &quot;lsm_c_pladj&quot;    &quot;lsm_c_split&quot;    &quot;lsm_l_ai&quot;      
## [17] &quot;lsm_l_cohesion&quot; &quot;lsm_l_contag&quot;   &quot;lsm_l_division&quot; &quot;lsm_l_enn_cv&quot;  
## [21] &quot;lsm_l_enn_mn&quot;   &quot;lsm_l_enn_sd&quot;   &quot;lsm_l_iji&quot;      &quot;lsm_l_lsi&quot;     
## [25] &quot;lsm_l_mesh&quot;     &quot;lsm_l_np&quot;       &quot;lsm_l_pd&quot;       &quot;lsm_l_pladj&quot;   
## [29] &quot;lsm_l_split&quot;</code></pre>
<div id="a.-calculate-patch--class--and-landscape-level-landscape-metrics" class="section level4">
<h4>a. Calculate patch-, class- and landscape level landscape metrics</h4>
<p>Note: This section explains different ways of calculating a selection of landscape metrics from a raster map with ‘landscapemetrics’. If this seems too technical for a first go, you may jump to section 4b.</p>
<p>To calculate a single metric, just use the corresponding function. The result of all landscape metric functions is always an identically structured <code>tibble</code> (i.e. an advanced <code>data.frame</code>). The first coloumn is the layer id (only interesting for e.g. a <code>RasterStack</code>). The second coloumn specifies the level (‘patch’, ‘class’ or ‘landscape’). The third coloumn is the class ID (<code>NA</code> on landscape level) and the fourth coloumn is the patch ID (<code>NA</code> on class- and landscape level). Lastly, The fith coloumn is the abbreviation of the metric and finally the corresponding value in the last coloumn.</p>
<div class="sourceCode" id="cb360"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb360-1" title="1"><span class="co">## c.lculate percentage of landscape of class</span></a>
<a class="sourceLine" id="cb360-2" title="2">percentage_class &lt;-<span class="st"> </span><span class="kw">lsm_c_pland</span>(<span class="dt">landscape =</span> nlcd)</a>
<a class="sourceLine" id="cb360-3" title="3"></a>
<a class="sourceLine" id="cb360-4" title="4">percentage_class</a></code></pre></div>
<pre><code>## # A tibble: 8 x 6
##   layer level class    id metric   value
##   &lt;int&gt; &lt;chr&gt; &lt;int&gt; &lt;int&gt; &lt;chr&gt;    &lt;dbl&gt;
## 1     1 class    11    NA pland   0.948 
## 2     1 class    12    NA pland   0.441 
## 3     1 class    31    NA pland   0.394 
## 4     1 class    42    NA pland  59.1   
## 5     1 class    52    NA pland  11.0   
## 6     1 class    71    NA pland  28.1   
## 7     1 class    90    NA pland   0.0557
## 8     1 class    95    NA pland   0.0210</code></pre>
<p><strong>Questions</strong>:</p>
<ul>
<li>What percentage of the overall landscape (total map) is evergreen forest (class 42)?</li>
<li>What percentage of the landscape is classified as wetlands (classes 90 and 95)?</li>
</ul>
<p>Because the resulting <code>tibble</code> is type stable, you can easily row-bind (<code>rbind</code>) different metrics (even of different levels):</p>
<div class="sourceCode" id="cb362"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb362-1" title="1">metrics &lt;-<span class="st"> </span><span class="kw">rbind</span>(</a>
<a class="sourceLine" id="cb362-2" title="2">  landscapemetrics<span class="op">::</span><span class="kw">lsm_c_pladj</span>(nlcd), </a>
<a class="sourceLine" id="cb362-3" title="3">  landscapemetrics<span class="op">::</span><span class="kw">lsm_l_pr</span>(nlcd),</a>
<a class="sourceLine" id="cb362-4" title="4">  landscapemetrics<span class="op">::</span><span class="kw">lsm_l_shdi</span>(nlcd)</a>
<a class="sourceLine" id="cb362-5" title="5">  )</a>
<a class="sourceLine" id="cb362-6" title="6"></a>
<a class="sourceLine" id="cb362-7" title="7">metrics</a></code></pre></div>
<pre><code>## # A tibble: 10 x 6
##    layer level     class    id metric value
##    &lt;int&gt; &lt;chr&gt;     &lt;int&gt; &lt;int&gt; &lt;chr&gt;  &lt;dbl&gt;
##  1     1 class        11    NA pladj  77.9 
##  2     1 class        12    NA pladj  52.5 
##  3     1 class        31    NA pladj  48.8 
##  4     1 class        42    NA pladj  89.9 
##  5     1 class        52    NA pladj  58.6 
##  6     1 class        71    NA pladj  81.6 
##  7     1 class        90    NA pladj  41.8 
##  8     1 class        95    NA pladj  32.8 
##  9     1 landscape    NA    NA pr      8   
## 10     1 landscape    NA    NA shdi    1.01</code></pre>
<p>To calculate a larger set of landscape metrics, you can just use the wrapper <code>calculate_lsm()</code>. The arguments are similar to <code>list_lsm()</code>, e.g. you can specify the level or the type of metrics to calculate. Alternatively, you can also provide a vector with the function names of metrics to calculate to the <code>what</code> argument.</p>
<p>However, watch out, for large rasters and many metrics, this can be rather slow (set <code>progress = TRUE</code> to get an progress report on the console). Also, we suggest to not just calculate all available metrics, but rather think about which ones might be actually suitable for your research question.</p>
<p>Calculate all patch-level metrics using wrapper:</p>
<div class="sourceCode" id="cb364"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb364-1" title="1">nlcd_patch &lt;-<span class="st"> </span>landscapemetrics<span class="op">::</span><span class="kw">calculate_lsm</span>(<span class="dt">landscape =</span> nlcd,</a>
<a class="sourceLine" id="cb364-2" title="2">                                              <span class="dt">level =</span> <span class="st">&quot;patch&quot;</span>)</a>
<a class="sourceLine" id="cb364-3" title="3">nlcd_patch</a></code></pre></div>
<pre><code>## # A tibble: 19,776 x 6
##    layer level class    id metric value
##    &lt;int&gt; &lt;chr&gt; &lt;int&gt; &lt;int&gt; &lt;chr&gt;  &lt;dbl&gt;
##  1     1 patch    11     1 area    0.45
##  2     1 patch    11     2 area    0.45
##  3     1 patch    11     3 area   41.7 
##  4     1 patch    11     4 area    0.72
##  5     1 patch    11     5 area    6.12
##  6     1 patch    11     6 area    0.9 
##  7     1 patch    11     7 area    0.9 
##  8     1 patch    11     8 area    1.89
##  9     1 patch    11     9 area    0.09
## 10     1 patch    11    10 area    1.17
## # ... with 19,766 more rows</code></pre>
<p>Show abbreviation of all calculated metrics:</p>
<div class="sourceCode" id="cb366"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb366-1" title="1"><span class="kw">unique</span>(nlcd_patch<span class="op">$</span>metric)</a></code></pre></div>
<pre><code>##  [1] &quot;area&quot;   &quot;cai&quot;    &quot;circle&quot; &quot;contig&quot; &quot;core&quot;   &quot;enn&quot;    &quot;frac&quot;   &quot;gyrate&quot;
##  [9] &quot;ncore&quot;  &quot;para&quot;   &quot;perim&quot;  &quot;shape&quot;</code></pre>
<p>Calculate all aggregation metrics on landscape level:</p>
<div class="sourceCode" id="cb368"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb368-1" title="1">nlcd_landscape_aggr &lt;-<span class="st"> </span>landscapemetrics<span class="op">::</span><span class="kw">calculate_lsm</span>(<span class="dt">landscape =</span> nlcd, </a>
<a class="sourceLine" id="cb368-2" title="2">                                                       <span class="dt">level =</span> <span class="st">&quot;landscape&quot;</span>, </a>
<a class="sourceLine" id="cb368-3" title="3">                                                       <span class="dt">type =</span> <span class="st">&quot;aggregation metric&quot;</span>)</a>
<a class="sourceLine" id="cb368-4" title="4">nlcd_landscape_aggr</a></code></pre></div>
<pre><code>## # A tibble: 14 x 6
##    layer level     class    id metric      value
##    &lt;int&gt; &lt;chr&gt;     &lt;int&gt; &lt;int&gt; &lt;chr&gt;       &lt;dbl&gt;
##  1     1 landscape    NA    NA ai         84.1  
##  2     1 landscape    NA    NA cohesion   99.2  
##  3     1 landscape    NA    NA contag     62.7  
##  4     1 landscape    NA    NA division    0.804
##  5     1 landscape    NA    NA enn_cv    218.   
##  6     1 landscape    NA    NA enn_mn    129.   
##  7     1 landscape    NA    NA enn_sd    281.   
##  8     1 landscape    NA    NA iji        43.0  
##  9     1 landscape    NA    NA lsi        32.3  
## 10     1 landscape    NA    NA mesh     2694.   
## 11     1 landscape    NA    NA np       1648    
## 12     1 landscape    NA    NA pd         12.0  
## 13     1 landscape    NA    NA pladj      83.7  
## 14     1 landscape    NA    NA split       5.09</code></pre>
<p>Calculate specific metrics:</p>
<div class="sourceCode" id="cb370"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb370-1" title="1">nlcd_subset &lt;-<span class="st"> </span>landscapemetrics<span class="op">::</span><span class="kw">calculate_lsm</span>(<span class="dt">landscape =</span> nlcd, </a>
<a class="sourceLine" id="cb370-2" title="2">                                               <span class="dt">what =</span> <span class="kw">c</span>(<span class="st">&quot;lsm_c_pladj&quot;</span>, </a>
<a class="sourceLine" id="cb370-3" title="3">                                                        <span class="st">&quot;lsm_l_pr&quot;</span>, </a>
<a class="sourceLine" id="cb370-4" title="4">                                                        <span class="st">&quot;lsm_l_shdi&quot;</span>))</a>
<a class="sourceLine" id="cb370-5" title="5">nlcd_subset</a></code></pre></div>
<pre><code>## # A tibble: 10 x 6
##    layer level     class    id metric value
##    &lt;int&gt; &lt;chr&gt;     &lt;int&gt; &lt;int&gt; &lt;chr&gt;  &lt;dbl&gt;
##  1     1 class        11    NA pladj  77.9 
##  2     1 class        12    NA pladj  52.5 
##  3     1 class        31    NA pladj  48.8 
##  4     1 class        42    NA pladj  89.9 
##  5     1 class        52    NA pladj  58.6 
##  6     1 class        71    NA pladj  81.6 
##  7     1 class        90    NA pladj  41.8 
##  8     1 class        95    NA pladj  32.8 
##  9     1 landscape    NA    NA pr      8   
## 10     1 landscape    NA    NA shdi    1.01</code></pre>
<p>The resulting <code>tibble</code> is easy to integrate into a workflow. For example, to get the ordered patch IDs of the 5% largest patches, the following code could be used.</p>
<p>The pipe operator %&gt;% from the <code>dplyr</code> package passes the resulting object automatically to the next function as first argument.</p>
<p>Note: the last step (pulling the id variable only) could be done by adding this to the pipe: <code>%&gt;% dplyr::pull(id)</code>. Due to some package inconsistencies, this sometimes created an error. Here we extract the id variable in a separate step as a work-around.</p>
<div class="sourceCode" id="cb372"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb372-1" title="1">id_largest &lt;-<span class="st"> </span>nlcd_patch <span class="op">%&gt;%</span><span class="st"> </span><span class="co"># previously calculated patch metrics</span></a>
<a class="sourceLine" id="cb372-2" title="2"><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">filter</span>(metric <span class="op">==</span><span class="st"> &quot;area&quot;</span>) <span class="op">%&gt;%</span><span class="st"> </span><span class="co"># only patch area</span></a>
<a class="sourceLine" id="cb372-3" title="3"><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">arrange</span>(<span class="op">-</span>value) <span class="op">%&gt;%</span><span class="st"> </span><span class="co"># order by decreasing size</span></a>
<a class="sourceLine" id="cb372-4" title="4"><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">filter</span>(value <span class="op">&gt;</span><span class="st"> </span><span class="kw">quantile</span>(value, <span class="dt">probs =</span> <span class="fl">0.95</span>)) <span class="co">## g.t only patches larger than 95% quantile</span></a>
<a class="sourceLine" id="cb372-5" title="5"></a>
<a class="sourceLine" id="cb372-6" title="6">id_largest &lt;-<span class="st"> </span>id_largest<span class="op">$</span>id <span class="co">## g.t only patch id</span></a>
<a class="sourceLine" id="cb372-7" title="7">id_largest</a></code></pre></div>
<pre><code>##  [1]  206  166 1265 1459 1549 1558 1421  427 1434 1385  226    3  386 1205  589
## [16] 1441  426 1059 1206  324  433 1195  315 1225  712 1266 1377  389  753 1528
## [31]  894 1510  286  563  240 1411  478 1364  435 1336  812  640 1376 1523  786
## [46]  467  485 1559 1244  284 1537 1574  718  814  499  864  955 1015 1430  443
## [61]  393  930 1250  548  851  885 1481  928 1554   43  164  281 1608  716  514
## [76]  802 1614  729  977 1488   37  357 1353</code></pre>
<p>Because the metric names are only abbreviated, there is also a way to include the full name in the results. For the wrapper, just set <code>full_name = TRUE</code>. For the rowbinded <code>tibble</code>, you can use the provided <code>tibble</code> called <code>lsm_abbreviations_names</code> that comes with the package and use e.g. <code>dplyr::left_join()</code> to combine it with your results.</p>
<p>Add full metrics name to result:</p>
<div class="sourceCode" id="cb374"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb374-1" title="1">nlcd_subset_full_a &lt;-<span class="st"> </span>landscapemetrics<span class="op">::</span><span class="kw">calculate_lsm</span>(nlcd, </a>
<a class="sourceLine" id="cb374-2" title="2">                                                      <span class="dt">what =</span> <span class="kw">c</span>(<span class="st">&quot;lsm_c_pladj&quot;</span>, </a>
<a class="sourceLine" id="cb374-3" title="3">                                                               <span class="st">&quot;lsm_l_pr&quot;</span>, </a>
<a class="sourceLine" id="cb374-4" title="4">                                                               <span class="st">&quot;lsm_l_shdi&quot;</span>), </a>
<a class="sourceLine" id="cb374-5" title="5">                                                      <span class="dt">full_name =</span> <span class="ot">TRUE</span>)</a>
<a class="sourceLine" id="cb374-6" title="6">nlcd_subset_full_a</a></code></pre></div>
<pre><code>## # A tibble: 10 x 9
##    layer level   class    id metric value name           type      function_name
##    &lt;int&gt; &lt;chr&gt;   &lt;int&gt; &lt;int&gt; &lt;chr&gt;  &lt;dbl&gt; &lt;chr&gt;          &lt;chr&gt;     &lt;chr&gt;        
##  1     1 class      11    NA pladj  77.9  percentage of~ aggregat~ lsm_c_pladj  
##  2     1 class      12    NA pladj  52.5  percentage of~ aggregat~ lsm_c_pladj  
##  3     1 class      31    NA pladj  48.8  percentage of~ aggregat~ lsm_c_pladj  
##  4     1 class      42    NA pladj  89.9  percentage of~ aggregat~ lsm_c_pladj  
##  5     1 class      52    NA pladj  58.6  percentage of~ aggregat~ lsm_c_pladj  
##  6     1 class      71    NA pladj  81.6  percentage of~ aggregat~ lsm_c_pladj  
##  7     1 class      90    NA pladj  41.8  percentage of~ aggregat~ lsm_c_pladj  
##  8     1 class      95    NA pladj  32.8  percentage of~ aggregat~ lsm_c_pladj  
##  9     1 landsc~    NA    NA pr      8    patch richness diversit~ lsm_l_pr     
## 10     1 landsc~    NA    NA shdi    1.01 shannon&#39;s div~ diversit~ lsm_l_shdi</code></pre>
<p>Add full metrics name to results calculated previously using <code>left_join()</code>:</p>
<div class="sourceCode" id="cb376"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb376-1" title="1">nlcd_subset_full_b &lt;-<span class="st"> </span>dplyr<span class="op">::</span><span class="kw">left_join</span>(<span class="dt">x =</span> nlcd_subset,</a>
<a class="sourceLine" id="cb376-2" title="2">                                       <span class="dt">y =</span> lsm_abbreviations_names,</a>
<a class="sourceLine" id="cb376-3" title="3">                                       <span class="dt">by =</span> <span class="kw">c</span>(<span class="st">&quot;metric&quot;</span>, <span class="st">&quot;level&quot;</span>))</a>
<a class="sourceLine" id="cb376-4" title="4"></a>
<a class="sourceLine" id="cb376-5" title="5">nlcd_subset_full_b</a></code></pre></div>
<pre><code>## # A tibble: 10 x 9
##    layer level   class    id metric value name           type      function_name
##    &lt;int&gt; &lt;chr&gt;   &lt;int&gt; &lt;int&gt; &lt;chr&gt;  &lt;dbl&gt; &lt;chr&gt;          &lt;chr&gt;     &lt;chr&gt;        
##  1     1 class      11    NA pladj  77.9  percentage of~ aggregat~ lsm_c_pladj  
##  2     1 class      12    NA pladj  52.5  percentage of~ aggregat~ lsm_c_pladj  
##  3     1 class      31    NA pladj  48.8  percentage of~ aggregat~ lsm_c_pladj  
##  4     1 class      42    NA pladj  89.9  percentage of~ aggregat~ lsm_c_pladj  
##  5     1 class      52    NA pladj  58.6  percentage of~ aggregat~ lsm_c_pladj  
##  6     1 class      71    NA pladj  81.6  percentage of~ aggregat~ lsm_c_pladj  
##  7     1 class      90    NA pladj  41.8  percentage of~ aggregat~ lsm_c_pladj  
##  8     1 class      95    NA pladj  32.8  percentage of~ aggregat~ lsm_c_pladj  
##  9     1 landsc~    NA    NA pr      8    patch richness diversit~ lsm_l_pr     
## 10     1 landsc~    NA    NA shdi    1.01 shannon&#39;s div~ diversit~ lsm_l_shdi</code></pre>
</div>
<div id="b.-calculate-patch-level-landscape-metrics-for-evergreen-forest" class="section level4">
<h4>b. Calculate patch-level landscape metrics for ‘Evergreen Forest’</h4>
<p>To only get the results for class 42 (evergreen forest), you can just <code>dplyr::filter()</code> the <code>tibble</code> (or use any other subset method you prefer).</p>
<div class="sourceCode" id="cb378"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb378-1" title="1">forest_patch_metrics &lt;-<span class="st"> </span>dplyr<span class="op">::</span><span class="kw">filter</span>(nlcd_patch, class <span class="op">==</span><span class="st"> </span><span class="dv">42</span>)</a></code></pre></div>
<p>All functions make heavy use of connected components labeling to delineate patches (neighbouring cells of the same class). To get all patches of every class you can just use <code>get_patches()</code>. To get only a certain class, just specify the <code>class</code> argument and the neighbourhood rule can be chosen between 8-neighbour rule or 4-neighbour rule with the argument <code>directions</code>.</p>
<div class="sourceCode" id="cb379"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb379-1" title="1"><span class="co">## c.nnected components labeling of landscape</span></a>
<a class="sourceLine" id="cb379-2" title="2">cc_nlcd &lt;-<span class="st"> </span>landscapemetrics<span class="op">::</span><span class="kw">get_patches</span>(nlcd, <span class="dt">directions =</span> <span class="dv">8</span>)</a>
<a class="sourceLine" id="cb379-3" title="3"></a>
<a class="sourceLine" id="cb379-4" title="4"><span class="co"># show name of each class</span></a>
<a class="sourceLine" id="cb379-5" title="5"><span class="kw">sapply</span>(cc_nlcd, <span class="cf">function</span>(x) <span class="kw">names</span>(x)) </a></code></pre></div>
<pre><code>##      11      12      31      42      52      71      90      95 
## &quot;layer&quot; &quot;layer&quot; &quot;layer&quot; &quot;layer&quot; &quot;layer&quot; &quot;layer&quot; &quot;layer&quot; &quot;layer&quot;</code></pre>
<div class="sourceCode" id="cb381"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb381-1" title="1"><span class="co"># the fourth list entry is class forest</span></a>
<a class="sourceLine" id="cb381-2" title="2">cc_forest_a &lt;-<span class="st"> </span>cc_nlcd[<span class="dv">4</span>]</a>
<a class="sourceLine" id="cb381-3" title="3">cc_forest_b &lt;-<span class="st"> </span>landscapemetrics<span class="op">::</span><span class="kw">get_patches</span>(nlcd, <span class="dt">class =</span> <span class="dv">42</span>) <span class="co"># watch out: result is list with one entry</span></a>
<a class="sourceLine" id="cb381-4" title="4"></a>
<a class="sourceLine" id="cb381-5" title="5">cc_forest_a</a></code></pre></div>
<pre><code>## $`42`
## class      : RasterLayer 
## dimensions : 426, 358, 152508  (nrow, ncol, ncell)
## resolution : 30, 30  (x, y)
## extent     : 683282.5, 694022.5, 4992833, 5005613  (xmin, xmax, ymin, ymax)
## crs        : +proj=utm +zone=11 +datum=NAD83 +units=m +no_defs 
## source     : memory
## names      : layer 
## values     : 1, 222  (min, max)</code></pre>
<div class="sourceCode" id="cb383"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb383-1" title="1">cc_forest_b</a></code></pre></div>
<pre><code>## $`42`
## class      : RasterLayer 
## dimensions : 426, 358, 152508  (nrow, ncol, ncell)
## resolution : 30, 30  (x, y)
## extent     : 683282.5, 694022.5, 4992833, 5005613  (xmin, xmax, ymin, ymax)
## crs        : +proj=utm +zone=11 +datum=NAD83 +units=m +no_defs 
## source     : memory
## names      : layer 
## values     : 1, 222  (min, max)</code></pre>
<p>To plot the patches you can use the <code>show_patches()</code> function. Here we show patches of class 42 (forest) and class 52 (shrubland):</p>
<div class="sourceCode" id="cb385"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb385-1" title="1"><span class="kw">show_patches</span>(<span class="dt">landscape =</span> nlcd, <span class="dt">class =</span> <span class="kw">c</span>(<span class="dv">42</span>, <span class="dv">52</span>), <span class="dt">labels =</span> <span class="ot">FALSE</span>)</a></code></pre></div>
<p><img src="02-Week02_files/figure-html/unnamed-chunk-37-1.png" width="672" /></p>
<p>It is also possible to visualize only the core area of each patch using <code>show_cores()</code>. The core area is defined as all cells that are further away from the edge of each patch than a specified edge depth (e.g. 5 cells). Here we show core area with edge depth = 5 for class 42; try edge_depth = 1 for comparison:</p>
<div class="sourceCode" id="cb386"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb386-1" title="1"><span class="kw">show_cores</span>(<span class="dt">landscape =</span> nlcd, <span class="dt">class =</span> <span class="kw">c</span>(<span class="dv">42</span>), <span class="dt">edge_depth =</span> <span class="dv">5</span>, <span class="dt">labels =</span> <span class="ot">FALSE</span>)</a></code></pre></div>
<p><img src="02-Week02_files/figure-html/unnamed-chunk-38-1.png" width="672" /></p>
<p>Note: this may create a warning “no non-missing arguments to min; returning Inf” for each patch that does not have any core area. Here we suppressed the warnings for the chunk with the chunk option <code>warning=FALSE</code>.</p>
<p>Lastly, you can plot the map and fill each patch with the corresponding metric value, e.g. patch size, using <code>show_lsm()</code>. Notice that there are two very large patches in class 42:</p>
<div class="sourceCode" id="cb387"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb387-1" title="1"><span class="kw">show_lsm</span>(<span class="dt">landscape =</span> nlcd, <span class="dt">class =</span> <span class="kw">c</span>(<span class="dv">42</span>, <span class="dv">52</span>), <span class="dt">what =</span> <span class="st">&quot;lsm_p_area&quot;</span>, <span class="dt">labels =</span> <span class="ot">FALSE</span>)</a></code></pre></div>
<p><img src="02-Week02_files/figure-html/unnamed-chunk-39-1.png" width="672" /></p>
</div>
<div id="c.-extract-forest-patch-size-at-samplig-locations" class="section level4">
<h4>c. Extract forest patch size at samplig locations</h4>
<p>Let’s add forest patch size to the <code>Sites.sp</code> data. To extract landscape metrics of the patch in which each sampling point is located, use <code>extract_lsm()</code>. Which metrics are extracted can be specified by the <code>what</code> argument (similar to <code>calculate_lsm()</code>). However, only patch-level metrics are available. Please be aware, that the resulting <code>tibble</code> now has a new column, namely the ID of the sampling point (in the same order as the input points).</p>
<div class="sourceCode" id="cb388"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb388-1" title="1"><span class="co">## e.tract patch area of all classes:</span></a>
<a class="sourceLine" id="cb388-2" title="2">patch_size_sp &lt;-<span class="st"> </span><span class="kw">extract_lsm</span>(<span class="dt">landscape =</span> nlcd, <span class="dt">y =</span> Sites.sp, <span class="dt">what =</span> <span class="st">&quot;lsm_p_area&quot;</span>)</a></code></pre></div>
<pre><code>## Warning: Only using &#39;what&#39; argument.</code></pre>
<div class="sourceCode" id="cb390"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb390-1" title="1"><span class="co">## b.cause we are only interested in the forest patch size, we set all area of class != 42 to 0:</span></a>
<a class="sourceLine" id="cb390-2" title="2">patch_size_sp_forest &lt;-<span class="st"> </span>dplyr<span class="op">::</span><span class="kw">mutate</span>(patch_size_sp, </a>
<a class="sourceLine" id="cb390-3" title="3">                                      <span class="dt">value =</span> dplyr<span class="op">::</span><span class="kw">case_when</span>(class <span class="op">==</span><span class="st"> </span><span class="dv">42</span> <span class="op">~</span><span class="st"> </span>value, </a>
<a class="sourceLine" id="cb390-4" title="4">                                                               class <span class="op">!=</span><span class="st"> </span><span class="dv">42</span> <span class="op">~</span><span class="st"> </span><span class="dv">0</span>))</a>
<a class="sourceLine" id="cb390-5" title="5"><span class="co">## a.d data to sp object:</span></a>
<a class="sourceLine" id="cb390-6" title="6">Sites.sp<span class="op">@</span>data<span class="op">$</span>ForestPatchSize &lt;-<span class="st"> </span>patch_size_sp_forest<span class="op">$</span>value</a>
<a class="sourceLine" id="cb390-7" title="7">Sites.sp<span class="op">@</span>data<span class="op">$</span>ForestPatchSize</a></code></pre></div>
<pre><code>##  [1] 3179.88    0.00 3179.88    0.00    6.66 4539.33 4539.33 4539.33    0.00
## [10] 4539.33 4539.33   23.76   23.76    0.00    0.00 4539.33    0.00 4539.33
## [19]    3.60 3179.88    0.00 4539.33    3.87 4539.33    0.00    0.00   23.76
## [28] 3179.88    0.00 4539.33    0.09</code></pre>
</div>
<div id="d.-plot-a-bubble-map-of-forest-patch-size-at-sampling-locations" class="section level4">
<h4>d. Plot a bubble map of forest patch size at sampling locations</h4>
<div class="sourceCode" id="cb392"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb392-1" title="1"><span class="kw">par</span>(<span class="dt">mar =</span> <span class="kw">c</span>(<span class="dv">3</span>,<span class="dv">3</span>,<span class="dv">1</span>,<span class="dv">2</span>))</a>
<a class="sourceLine" id="cb392-2" title="2"><span class="kw">bubble</span>(Sites.sp, <span class="st">&quot;ForestPatchSize&quot;</span>, <span class="dt">fill =</span> <span class="ot">FALSE</span>, <span class="dt">key.entries =</span> <span class="kw">as.numeric</span>(<span class="kw">names</span>(<span class="kw">table</span>(Sites.sp<span class="op">@</span>data<span class="op">$</span>ForestPatchSize))))</a></code></pre></div>
<pre><code>## Warning in wkt(obj): CRS object has no comment

## Warning in wkt(obj): CRS object has no comment</code></pre>
<p><img src="02-Week02_files/figure-html/unnamed-chunk-41-1.png" width="427.2" /></p>
</div>
</div>
<div id="sample-landscape-metrics-within-buffer-around-sampling-locations" class="section level3">
<h3>5. Sample landscape metrics within buffer around sampling locations</h3>
<p>The package <code>landscapemetrics</code> has a built-in function <code>sample_lsm</code> to sample metrics in a buffer around sampling locations, which are provided with argument <code>y</code>. You can choose the <code>shape</code> of the buffer window (either a circle or a square) and, with the argument <code>what</code>, which metrics to sample (similar to <code>calculate_lsm()</code>).</p>
<p>The argument <code>size</code> specifies the buffer size in map units (e.g., meters): radius for circles, half of the side length for squares. Here, the value <code>size = 500</code> results in a square window of 1000 m x 1000 m centered at the sampling location.</p>
<div class="sourceCode" id="cb394"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb394-1" title="1">nlcd_sampled &lt;-<span class="st"> </span>landscapemetrics<span class="op">::</span><span class="kw">sample_lsm</span>(<span class="dt">landscape =</span> nlcd, </a>
<a class="sourceLine" id="cb394-2" title="2">                                             <span class="dt">what =</span> <span class="kw">c</span>(<span class="st">&quot;lsm_l_ta&quot;</span>, </a>
<a class="sourceLine" id="cb394-3" title="3">                                                      <span class="st">&quot;lsm_c_np&quot;</span>,</a>
<a class="sourceLine" id="cb394-4" title="4">                                                      <span class="st">&quot;lsm_c_pland&quot;</span>, </a>
<a class="sourceLine" id="cb394-5" title="5">                                                      <span class="st">&quot;lsm_c_ai&quot;</span>),</a>
<a class="sourceLine" id="cb394-6" title="6">                                             <span class="dt">shape =</span> <span class="st">&quot;square&quot;</span>,</a>
<a class="sourceLine" id="cb394-7" title="7">                                             <span class="dt">y =</span> Sites.sp, </a>
<a class="sourceLine" id="cb394-8" title="8">                                             <span class="dt">size =</span> <span class="dv">500</span>)</a>
<a class="sourceLine" id="cb394-9" title="9">nlcd_sampled</a></code></pre></div>
<pre><code>## # A tibble: 493 x 8
##    layer level class    id metric value plot_id percentage_inside
##    &lt;int&gt; &lt;chr&gt; &lt;int&gt; &lt;int&gt; &lt;chr&gt;  &lt;dbl&gt;   &lt;int&gt;             &lt;dbl&gt;
##  1     1 class    11    NA ai      88.6       1              98.0
##  2     1 class    31    NA ai      NA         1              98.0
##  3     1 class    42    NA ai      93.3       1              98.0
##  4     1 class    52    NA ai      50         1              98.0
##  5     1 class    71    NA ai      65.9       1              98.0
##  6     1 class    11    NA np       2         1              98.0
##  7     1 class    31    NA np       1         1              98.0
##  8     1 class    42    NA np       2         1              98.0
##  9     1 class    52    NA np       4         1              98.0
## 10     1 class    71    NA np       8         1              98.0
## # ... with 483 more rows</code></pre>
<p>The <code>tibble</code> now contains two additional columns. Firstly, the <code>plot_id</code> (in the same order as the input points) and secondly, the <code>percentage_inside</code>, i.e. what percentage of the buffer around the sampling location lies within the map. (In cases where the sampling location is on the edge of the landscape, the buffer around the sampling location could be only partly within the map). The value can also deviate from 100 % because the sampling locations are not necessarily in the cell center and the actually clipped cells lead to a slightly smaller or larger buffer area. A circular buffer shape increases this effect.</p>
<p>It is also possible to get the clippings of the buffer around sampling locations as a <code>RasterLayer</code>. For this, just set <code>return_raster = TRUE</code>.</p>
<div class="sourceCode" id="cb396"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb396-1" title="1"><span class="co"># sample some metrics within buffer around sample location and returning sample</span></a>
<a class="sourceLine" id="cb396-2" title="2"><span class="co"># plots as raster</span></a>
<a class="sourceLine" id="cb396-3" title="3">nlcd_sampled_plots &lt;-<span class="st"> </span>landscapemetrics<span class="op">::</span><span class="kw">sample_lsm</span>(<span class="dt">landscape =</span> nlcd, </a>
<a class="sourceLine" id="cb396-4" title="4">                                                   <span class="dt">what =</span> <span class="kw">c</span>(<span class="st">&quot;lsm_l_ta&quot;</span>,</a>
<a class="sourceLine" id="cb396-5" title="5">                                                            <span class="st">&quot;lsm_c_np&quot;</span>,</a>
<a class="sourceLine" id="cb396-6" title="6">                                                            <span class="st">&quot;lsm_c_pland&quot;</span>,</a>
<a class="sourceLine" id="cb396-7" title="7">                                                            <span class="st">&quot;lsm_c_ai&quot;</span>),</a>
<a class="sourceLine" id="cb396-8" title="8">                                                   <span class="dt">shape =</span> <span class="st">&quot;square&quot;</span>,</a>
<a class="sourceLine" id="cb396-9" title="9">                                                   <span class="dt">y =</span> Sites.sp, </a>
<a class="sourceLine" id="cb396-10" title="10">                                                   <span class="dt">size =</span> <span class="dv">500</span>, </a>
<a class="sourceLine" id="cb396-11" title="11">                                                   <span class="dt">return_raster =</span> <span class="ot">TRUE</span>)</a>
<a class="sourceLine" id="cb396-12" title="12"></a>
<a class="sourceLine" id="cb396-13" title="13">nlcd_sampled_plots</a></code></pre></div>
<pre><code>## # A tibble: 493 x 9
##    layer level class    id metric value plot_id percentage_inside
##    &lt;int&gt; &lt;chr&gt; &lt;int&gt; &lt;int&gt; &lt;chr&gt;  &lt;dbl&gt;   &lt;int&gt;             &lt;dbl&gt;
##  1     1 class    11    NA ai      88.6       1              98.0
##  2     1 class    31    NA ai      NA         1              98.0
##  3     1 class    42    NA ai      93.3       1              98.0
##  4     1 class    52    NA ai      50         1              98.0
##  5     1 class    71    NA ai      65.9       1              98.0
##  6     1 class    11    NA np       2         1              98.0
##  7     1 class    31    NA np       1         1              98.0
##  8     1 class    42    NA np       2         1              98.0
##  9     1 class    52    NA np       4         1              98.0
## 10     1 class    71    NA np       8         1              98.0
## # ... with 483 more rows, and 1 more variable: raster_sample_plots &lt;list&gt;</code></pre>
<p>The result will be a nested <code>tibble</code> containing the <code>plot_id</code>, the metrics and a <code>RasterLayer</code> with the clipped buffers (as a <code>list</code>). Attention: Because several metrics on class- and landscape-level the clipped buffers will be “repeated” several times.</p>
<p>Here we show results for the first four sampling locations:</p>
<div class="sourceCode" id="cb398"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb398-1" title="1">unique_plots &lt;-<span class="st"> </span><span class="kw">unique</span>(nlcd_sampled_plots<span class="op">$</span>raster_sample_plots)[<span class="dv">1</span><span class="op">:</span><span class="dv">4</span>]</a>
<a class="sourceLine" id="cb398-2" title="2"></a>
<a class="sourceLine" id="cb398-3" title="3"><span class="kw">par</span>(<span class="dt">mfrow =</span> <span class="kw">c</span>(<span class="dv">2</span>,<span class="dv">2</span>))</a>
<a class="sourceLine" id="cb398-4" title="4"><span class="kw">plot</span>(unique_plots[[<span class="dv">1</span>]], </a>
<a class="sourceLine" id="cb398-5" title="5">     <span class="dt">main =</span> <span class="kw">paste</span>(Sites.sp<span class="op">$</span>SiteName[<span class="dv">1</span>]), </a>
<a class="sourceLine" id="cb398-6" title="6">     <span class="dt">col =</span> <span class="kw">rev</span>(<span class="kw">rainbow</span>(<span class="dv">9</span>)))</a>
<a class="sourceLine" id="cb398-7" title="7"><span class="kw">plot</span>(unique_plots[[<span class="dv">2</span>]],</a>
<a class="sourceLine" id="cb398-8" title="8">     <span class="dt">main =</span> <span class="kw">paste</span>(Sites.sp<span class="op">$</span>SiteName[<span class="dv">2</span>]),</a>
<a class="sourceLine" id="cb398-9" title="9">     <span class="dt">col =</span> <span class="kw">rev</span>(<span class="kw">rainbow</span>(<span class="dv">9</span>)))</a>
<a class="sourceLine" id="cb398-10" title="10"><span class="kw">plot</span>(unique_plots[[<span class="dv">3</span>]],</a>
<a class="sourceLine" id="cb398-11" title="11">     <span class="dt">main =</span> <span class="kw">paste</span>(Sites.sp<span class="op">$</span>SiteName[<span class="dv">3</span>]), </a>
<a class="sourceLine" id="cb398-12" title="12">     <span class="dt">col =</span> <span class="kw">rev</span>(<span class="kw">rainbow</span>(<span class="dv">9</span>)))</a>
<a class="sourceLine" id="cb398-13" title="13"><span class="kw">plot</span>(unique_plots[[<span class="dv">4</span>]],</a>
<a class="sourceLine" id="cb398-14" title="14">     <span class="dt">main =</span> <span class="kw">paste</span>(Sites.sp<span class="op">$</span>SiteName[<span class="dv">4</span>]), </a>
<a class="sourceLine" id="cb398-15" title="15">     <span class="dt">col =</span> <span class="kw">rev</span>(<span class="kw">rainbow</span>(<span class="dv">9</span>)))</a></code></pre></div>
<p><img src="02-Week02_files/figure-html/unnamed-chunk-44-1.png" width="768" /></p>
<div class="sourceCode" id="cb399"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb399-1" title="1"><span class="kw">par</span>(<span class="dt">mfrow =</span> <span class="kw">c</span>(<span class="dv">1</span>,<span class="dv">1</span>))</a></code></pre></div>
<div id="b.-extract-landscape-metric-of-choice-for-a-single-cover-type-as-vector" class="section level4">
<h4>b. Extract landscape metric of choice for a single cover type (as vector)</h4>
<p>To extract a metrics you can just <code>dplyr::filter()</code> the resulting <code>tibble</code> and pull the <code>value</code> column.
Here we filter the results for class == 42 (forest) and metric <code>pland</code> (percentage of landscape) and pull the results as a vector:</p>
<div class="sourceCode" id="cb400"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb400-1" title="1">percentage_forest_<span class="dv">500</span>_a &lt;-<span class="st"> </span>dplyr<span class="op">::</span><span class="kw">pull</span>(dplyr<span class="op">::</span><span class="kw">filter</span>(nlcd_sampled, </a>
<a class="sourceLine" id="cb400-2" title="2">                                                     class <span class="op">==</span><span class="st"> </span><span class="dv">42</span>, </a>
<a class="sourceLine" id="cb400-3" title="3">                                                     metric <span class="op">==</span><span class="st"> &quot;pland&quot;</span>), value)</a>
<a class="sourceLine" id="cb400-4" title="4">percentage_forest_<span class="dv">500</span>_a</a></code></pre></div>
<pre><code>##  [1] 77.50230 39.57759 38.01653 31.40138 40.40404 83.30450 69.16221 86.80927
##  [9] 10.19284 72.81910 92.19467 34.04635 34.84848 27.09447 31.37255 56.23886
## [17] 59.77961 58.73440 37.64922 60.12111 46.16756 32.32323 23.35640 50.44563
## [25] 66.93405 30.57851 33.70064 26.53811 45.18717 38.05704 62.47772</code></pre>
<p>As an alternative, here’s the same workflow again, but using a pipe:</p>
<div class="sourceCode" id="cb402"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb402-1" title="1">percentage_forest_<span class="dv">500</span>_b &lt;-<span class="st"> </span>nlcd_sampled <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb402-2" title="2"><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">filter</span>(class <span class="op">==</span><span class="st"> </span><span class="dv">42</span>, </a>
<a class="sourceLine" id="cb402-3" title="3">                metric <span class="op">==</span><span class="st"> &quot;pland&quot;</span>) <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb402-4" title="4"><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">pull</span>(value)</a>
<a class="sourceLine" id="cb402-5" title="5">percentage_forest_<span class="dv">500</span>_b</a></code></pre></div>
<pre><code>##  [1] 77.50230 39.57759 38.01653 31.40138 40.40404 83.30450 69.16221 86.80927
##  [9] 10.19284 72.81910 92.19467 34.04635 34.84848 27.09447 31.37255 56.23886
## [17] 59.77961 58.73440 37.64922 60.12111 46.16756 32.32323 23.35640 50.44563
## [25] 66.93405 30.57851 33.70064 26.53811 45.18717 38.05704 62.47772</code></pre>
</div>
<div id="c.-extract-landscape-metric-of-choice-for-all-cover-types-as-data-frame." class="section level4">
<h4>c. Extract landscape metric of choice for all cover types (as data frame).</h4>
<p>To extract the landscape metric ‘prop.landscape’ for all cover types as a <code>tibble</code>, just filter <code>dplyr::filter()</code> the <code>tibble</code> again, but only use the <code>metric</code> as filter.</p>
<div class="sourceCode" id="cb404"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb404-1" title="1"><span class="co">## f.lter for percentage of landscape</span></a>
<a class="sourceLine" id="cb404-2" title="2">percentage_forest_<span class="dv">500</span>_df &lt;-<span class="st"> </span>dplyr<span class="op">::</span><span class="kw">filter</span>(nlcd_sampled,</a>
<a class="sourceLine" id="cb404-3" title="3">                                          metric <span class="op">==</span><span class="st"> &quot;pland&quot;</span>)</a>
<a class="sourceLine" id="cb404-4" title="4"></a>
<a class="sourceLine" id="cb404-5" title="5">percentage_forest_<span class="dv">500</span>_df</a></code></pre></div>
<pre><code>## # A tibble: 154 x 8
##    layer level class    id metric   value plot_id percentage_inside
##    &lt;int&gt; &lt;chr&gt; &lt;int&gt; &lt;int&gt; &lt;chr&gt;    &lt;dbl&gt;   &lt;int&gt;             &lt;dbl&gt;
##  1     1 class    11    NA pland   7.25         1              98.0
##  2     1 class    31    NA pland   0.0918       1              98.0
##  3     1 class    42    NA pland  77.5          1              98.0
##  4     1 class    52    NA pland   1.93         1              98.0
##  5     1 class    71    NA pland  13.2          1              98.0
##  6     1 class    11    NA pland   3.49         2              98.0
##  7     1 class    12    NA pland   1.01         2              98.0
##  8     1 class    31    NA pland   0.735        2              98.0
##  9     1 class    42    NA pland  39.6          2              98.0
## 10     1 class    52    NA pland   5.33         2              98.0
## # ... with 144 more rows</code></pre>
<p>The percent cover of all cover types should add up to ~ 100% (i.e., 1) for each site. We can check this with the function <code>dplyr::summarize()</code>. First, we need to group the data using the <code>plot_id</code>, then sum all percentages.</p>
<div class="sourceCode" id="cb406"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb406-1" title="1"><span class="co">## g.oup by plot_id and sum all percentages</span></a>
<a class="sourceLine" id="cb406-2" title="2">pland_sum_a &lt;-<span class="st"> </span>dplyr<span class="op">::</span><span class="kw">summarize</span>(dplyr<span class="op">::</span><span class="kw">group_by</span>(percentage_forest_<span class="dv">500</span>_df, </a>
<a class="sourceLine" id="cb406-3" title="3">                                                <span class="dt">by =</span> plot_id), </a>
<a class="sourceLine" id="cb406-4" title="4">                                <span class="dt">sum_pland =</span> <span class="kw">sum</span>(value))</a>
<a class="sourceLine" id="cb406-5" title="5">pland_sum_a</a></code></pre></div>
<pre><code>## # A tibble: 31 x 2
##       by sum_pland
##    &lt;int&gt;     &lt;dbl&gt;
##  1     1       100
##  2     2       100
##  3     3       100
##  4     4       100
##  5     5       100
##  6     6       100
##  7     7       100
##  8     8       100
##  9     9       100
## 10    10       100
## # ... with 21 more rows</code></pre>
<p>Same workflow, but using a pipe:</p>
<div class="sourceCode" id="cb408"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb408-1" title="1">pland_sum_b &lt;-<span class="st"> </span>percentage_forest_<span class="dv">500</span>_df <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb408-2" title="2"><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">group_by</span>(plot_id) <span class="op">%&gt;%</span><span class="st"> </span></a>
<a class="sourceLine" id="cb408-3" title="3"><span class="st">  </span>dplyr<span class="op">::</span><span class="kw">summarize</span>(<span class="dt">sum_pland =</span> <span class="kw">sum</span>(value))</a>
<a class="sourceLine" id="cb408-4" title="4">pland_sum_b</a></code></pre></div>
<pre><code>## # A tibble: 31 x 2
##    plot_id sum_pland
##      &lt;int&gt;     &lt;dbl&gt;
##  1       1       100
##  2       2       100
##  3       3       100
##  4       4       100
##  5       5       100
##  6       6       100
##  7       7       100
##  8       8       100
##  9       9       100
## 10      10       100
## # ... with 21 more rows</code></pre>
</div>
<div id="d.-extract-all-landscape-metrics-for-a-single-cover-type-as-data-frame" class="section level4">
<h4>d. Extract all landscape metrics for a single cover type (as data frame)</h4>
<p>Filter<code>dplyr::filter()</code> for <code>class == 42</code> and add the sites names as coloumn to the resulting <code>tibble</code>.</p>
<div class="sourceCode" id="cb410"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb410-1" title="1"><span class="co">## f.lter for class == 42 (forest)</span></a>
<a class="sourceLine" id="cb410-2" title="2">forest_<span class="dv">500</span>_df &lt;-<span class="st"> </span>dplyr<span class="op">::</span><span class="kw">filter</span>(nlcd_sampled,</a>
<a class="sourceLine" id="cb410-3" title="3">                               class <span class="op">==</span><span class="st"> </span><span class="dv">42</span>)</a>
<a class="sourceLine" id="cb410-4" title="4"></a>
<a class="sourceLine" id="cb410-5" title="5"><span class="co">## d.ta.frame with id and name of site</span></a>
<a class="sourceLine" id="cb410-6" title="6">SiteName_df &lt;-<span class="st"> </span><span class="kw">data.frame</span>(<span class="dt">id =</span> <span class="dv">1</span><span class="op">:</span><span class="kw">length</span>(Sites.sp<span class="op">$</span>SiteName), <span class="dt">site_name =</span> Sites.sp<span class="op">$</span>SiteName)</a>
<a class="sourceLine" id="cb410-7" title="7"></a>
<a class="sourceLine" id="cb410-8" title="8"><span class="co">## a.d site_name to metrics using plot_id and id of sampling sites</span></a>
<a class="sourceLine" id="cb410-9" title="9">forest_<span class="dv">500</span>_df &lt;-<span class="st"> </span>dplyr<span class="op">::</span><span class="kw">left_join</span>(forest_<span class="dv">500</span>_df, SiteName_df, <span class="dt">by =</span> <span class="kw">c</span>(<span class="st">&quot;plot_id&quot;</span> =<span class="st"> &quot;id&quot;</span>))</a>
<a class="sourceLine" id="cb410-10" title="10"></a>
<a class="sourceLine" id="cb410-11" title="11">forest_<span class="dv">500</span>_df</a></code></pre></div>
<pre><code>## # A tibble: 93 x 9
##    layer level class    id metric value plot_id percentage_inside site_name     
##    &lt;int&gt; &lt;chr&gt; &lt;int&gt; &lt;int&gt; &lt;chr&gt;  &lt;dbl&gt;   &lt;int&gt;             &lt;dbl&gt; &lt;chr&gt;         
##  1     1 class    42    NA ai      93.3       1              98.0 AirplaneLake  
##  2     1 class    42    NA np       2         1              98.0 AirplaneLake  
##  3     1 class    42    NA pland   77.5       1              98.0 AirplaneLake  
##  4     1 class    42    NA ai      89.4       2              98.0 BachelorMeadow
##  5     1 class    42    NA np       7         2              98.0 BachelorMeadow
##  6     1 class    42    NA pland   39.6       2              98.0 BachelorMeadow
##  7     1 class    42    NA ai      79.4       3              98.0 BarkingFoxLake
##  8     1 class    42    NA np      10         3              98.0 BarkingFoxLake
##  9     1 class    42    NA pland   38.0       3              98.0 BarkingFoxLake
## 10     1 class    42    NA ai      81.2       4             104.  BirdbillLake  
## # ... with 83 more rows</code></pre>
<p>Done!</p>
<p>Note: check this week’s bonus material if you want to see how to use the new ‘sf’ library for spatial data, and how to export the site data to a shapefile that you can import into a GIS.</p>
</div>
</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="tutorial-2.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="r-exercise-week-2.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": true,
"facebook": false,
"twitter": false,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": false
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": null,
"toc": {
"collapse": "section"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
