<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Worked Example | Landscape Genetic Data Analysis with R</title>
  <meta name="description" content="This is a web-interface to the teaching materials for the lab course ‘Landscape Genetic Data Analysis with R’ associated with the distributed graduate course ‘DGS Landscape Genetics.’ The output format is bookdown::gitbook." />
  <meta name="generator" content="bookdown 0.22 and GitBook 2.6.7" />

  <meta property="og:title" content="Worked Example | Landscape Genetic Data Analysis with R" />
  <meta property="og:type" content="book" />
  
  
  <meta property="og:description" content="This is a web-interface to the teaching materials for the lab course ‘Landscape Genetic Data Analysis with R’ associated with the distributed graduate course ‘DGS Landscape Genetics.’ The output format is bookdown::gitbook." />
  <meta name="github-repo" content="hhwagner1/LandGenCourse_book" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Worked Example | Landscape Genetic Data Analysis with R" />
  
  <meta name="twitter:description" content="This is a web-interface to the teaching materials for the lab course ‘Landscape Genetic Data Analysis with R’ associated with the distributed graduate course ‘DGS Landscape Genetics.’ The output format is bookdown::gitbook." />
  

<meta name="author" content="Editor: Helene Wagner (University of Toronto)" />


<meta name="date" content="2021-05-07" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="tutorial-7.html"/>
<link rel="next" href="r-exercise-week-7.html"/>
<script src="libs/header-attrs-2.7/header-attrs.js"></script>
<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />









<link href="libs/anchor-sections-1.0.1/anchor-sections.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.0.1/anchor-sections.js"></script>
<script src="libs/kePrint-0.0.1/kePrint.js"></script>
<link href="libs/lightable-0.0.1/lightable.css" rel="stylesheet" />
<script src="libs/htmlwidgets-1.5.3/htmlwidgets.js"></script>
<script src="libs/d3-3.5.6/d3.min.js"></script>
<link href="libs/profvis-0.3.6.9000/profvis.css" rel="stylesheet" />
<script src="libs/profvis-0.3.6.9000/profvis.js"></script>
<script src="libs/profvis-0.3.6.9000/scroll.js"></script>
<link href="libs/highlight-6.2.0/textmate.css" rel="stylesheet" />
<script src="libs/highlight-6.2.0/highlight.js"></script>
<script src="libs/profvis-binding-0.3.7/profvis.js"></script>

<script>
/* ========================================================================
 * Bootstrap: transition.js v3.3.7
 * http://getbootstrap.com/javascript/#transitions
 * ========================================================================
 * Copyright 2011-2016 Twitter, Inc.
 * Licensed under MIT (https://github.com/twbs/bootstrap/blob/master/LICENSE)
 * ======================================================================== */


+function ($) {
  'use strict';

  // CSS TRANSITION SUPPORT (Shoutout: http://www.modernizr.com/)
  // ============================================================

  function transitionEnd() {
    var el = document.createElement('bootstrap')

    var transEndEventNames = {
      WebkitTransition : 'webkitTransitionEnd',
      MozTransition    : 'transitionend',
      OTransition      : 'oTransitionEnd otransitionend',
      transition       : 'transitionend'
    }

    for (var name in transEndEventNames) {
      if (el.style[name] !== undefined) {
        return { end: transEndEventNames[name] }
      }
    }

    return false // explicit for ie8 (  ._.)
  }

  // http://blog.alexmaccaw.com/css-transitions
  $.fn.emulateTransitionEnd = function (duration) {
    var called = false
    var $el = this
    $(this).one('bsTransitionEnd', function () { called = true })
    var callback = function () { if (!called) $($el).trigger($.support.transition.end) }
    setTimeout(callback, duration)
    return this
  }

  $(function () {
    $.support.transition = transitionEnd()

    if (!$.support.transition) return

    $.event.special.bsTransitionEnd = {
      bindType: $.support.transition.end,
      delegateType: $.support.transition.end,
      handle: function (e) {
        if ($(e.target).is(this)) return e.handleObj.handler.apply(this, arguments)
      }
    }
  })

}(jQuery);
</script>
<script>
/* ========================================================================
 * Bootstrap: collapse.js v3.3.7
 * http://getbootstrap.com/javascript/#collapse
 * ========================================================================
 * Copyright 2011-2016 Twitter, Inc.
 * Licensed under MIT (https://github.com/twbs/bootstrap/blob/master/LICENSE)
 * ======================================================================== */

/* jshint latedef: false */

+function ($) {
  'use strict';

  // COLLAPSE PUBLIC CLASS DEFINITION
  // ================================

  var Collapse = function (element, options) {
    this.$element      = $(element)
    this.options       = $.extend({}, Collapse.DEFAULTS, options)
    this.$trigger      = $('[data-toggle="collapse"][href="#' + element.id + '"],' +
                           '[data-toggle="collapse"][data-target="#' + element.id + '"]')
    this.transitioning = null

    if (this.options.parent) {
      this.$parent = this.getParent()
    } else {
      this.addAriaAndCollapsedClass(this.$element, this.$trigger)
    }

    if (this.options.toggle) this.toggle()
  }

  Collapse.VERSION  = '3.3.7'

  Collapse.TRANSITION_DURATION = 350

  Collapse.DEFAULTS = {
    toggle: true
  }

  Collapse.prototype.dimension = function () {
    var hasWidth = this.$element.hasClass('width')
    return hasWidth ? 'width' : 'height'
  }

  Collapse.prototype.show = function () {
    if (this.transitioning || this.$element.hasClass('in')) return

    var activesData
    var actives = this.$parent && this.$parent.children('.panel').children('.in, .collapsing')

    if (actives && actives.length) {
      activesData = actives.data('bs.collapse')
      if (activesData && activesData.transitioning) return
    }

    var startEvent = $.Event('show.bs.collapse')
    this.$element.trigger(startEvent)
    if (startEvent.isDefaultPrevented()) return

    if (actives && actives.length) {
      Plugin.call(actives, 'hide')
      activesData || actives.data('bs.collapse', null)
    }

    var dimension = this.dimension()

    this.$element
      .removeClass('collapse')
      .addClass('collapsing')[dimension](0)
      .attr('aria-expanded', true)

    this.$trigger
      .removeClass('collapsed')
      .attr('aria-expanded', true)

    this.transitioning = 1

    var complete = function () {
      this.$element
        .removeClass('collapsing')
        .addClass('collapse in')[dimension]('')
      this.transitioning = 0
      this.$element
        .trigger('shown.bs.collapse')
    }

    if (!$.support.transition) return complete.call(this)

    var scrollSize = $.camelCase(['scroll', dimension].join('-'))

    this.$element
      .one('bsTransitionEnd', $.proxy(complete, this))
      .emulateTransitionEnd(Collapse.TRANSITION_DURATION)[dimension](this.$element[0][scrollSize])
  }

  Collapse.prototype.hide = function () {
    if (this.transitioning || !this.$element.hasClass('in')) return

    var startEvent = $.Event('hide.bs.collapse')
    this.$element.trigger(startEvent)
    if (startEvent.isDefaultPrevented()) return

    var dimension = this.dimension()

    this.$element[dimension](this.$element[dimension]())[0].offsetHeight

    this.$element
      .addClass('collapsing')
      .removeClass('collapse in')
      .attr('aria-expanded', false)

    this.$trigger
      .addClass('collapsed')
      .attr('aria-expanded', false)

    this.transitioning = 1

    var complete = function () {
      this.transitioning = 0
      this.$element
        .removeClass('collapsing')
        .addClass('collapse')
        .trigger('hidden.bs.collapse')
    }

    if (!$.support.transition) return complete.call(this)

    this.$element
      [dimension](0)
      .one('bsTransitionEnd', $.proxy(complete, this))
      .emulateTransitionEnd(Collapse.TRANSITION_DURATION)
  }

  Collapse.prototype.toggle = function () {
    this[this.$element.hasClass('in') ? 'hide' : 'show']()
  }

  Collapse.prototype.getParent = function () {
    return $(this.options.parent)
      .find('[data-toggle="collapse"][data-parent="' + this.options.parent + '"]')
      .each($.proxy(function (i, element) {
        var $element = $(element)
        this.addAriaAndCollapsedClass(getTargetFromTrigger($element), $element)
      }, this))
      .end()
  }

  Collapse.prototype.addAriaAndCollapsedClass = function ($element, $trigger) {
    var isOpen = $element.hasClass('in')

    $element.attr('aria-expanded', isOpen)
    $trigger
      .toggleClass('collapsed', !isOpen)
      .attr('aria-expanded', isOpen)
  }

  function getTargetFromTrigger($trigger) {
    var href
    var target = $trigger.attr('data-target')
      || (href = $trigger.attr('href')) && href.replace(/.*(?=#[^\s]+$)/, '') // strip for ie7

    return $(target)
  }


  // COLLAPSE PLUGIN DEFINITION
  // ==========================

  function Plugin(option) {
    return this.each(function () {
      var $this   = $(this)
      var data    = $this.data('bs.collapse')
      var options = $.extend({}, Collapse.DEFAULTS, $this.data(), typeof option == 'object' && option)

      if (!data && options.toggle && /show|hide/.test(option)) options.toggle = false
      if (!data) $this.data('bs.collapse', (data = new Collapse(this, options)))
      if (typeof option == 'string') data[option]()
    })
  }

  var old = $.fn.collapse

  $.fn.collapse             = Plugin
  $.fn.collapse.Constructor = Collapse


  // COLLAPSE NO CONFLICT
  // ====================

  $.fn.collapse.noConflict = function () {
    $.fn.collapse = old
    return this
  }


  // COLLAPSE DATA-API
  // =================

  $(document).on('click.bs.collapse.data-api', '[data-toggle="collapse"]', function (e) {
    var $this   = $(this)

    if (!$this.attr('data-target')) e.preventDefault()

    var $target = getTargetFromTrigger($this)
    var data    = $target.data('bs.collapse')
    var option  = data ? 'toggle' : $this.data()

    Plugin.call($target, option)
  })

}(jQuery);
</script>
<script>
window.initializeCodeFolding = function(show) {

  // handlers for show-all and hide all
  $("#rmd-show-all-code").click(function() {
    // close the dropdown menu when an option is clicked
    $("#allCodeButton").dropdown("toggle");
    $('div.r-code-collapse').each(function() {
      $(this).collapse('show');
    });
  });
  $("#rmd-hide-all-code").click(function() {
    // close the dropdown menu when an option is clicked
    $("#allCodeButton").dropdown("toggle");
    $('div.r-code-collapse').each(function() {
      $(this).collapse('hide');
    });
  });

  // index for unique code element ids
  var currentIndex = 1;

  // select all R code blocks
  var rCodeBlocks = $('pre.sourceCode, pre.r, pre.python, pre.bash, pre.sql, pre.cpp, pre.stan');
  rCodeBlocks.each(function() {

    // if code block has been labeled with class `fold-show`, show the code on init!
    var classList = $(this).attr('class').split(/\s+/);
    for (var i = 0; i < classList.length; i++) {
    if (classList[i] === 'fold-show') {
        show = true;
      }
    }

    // create a collapsable div to wrap the code in
    var div = $('<div class="collapse r-code-collapse"></div>');
    if (show)
      div.addClass('in');
    var id = 'rcode-643E0F36' + currentIndex++;
    div.attr('id', id);
    $(this).before(div);
    $(this).detach().appendTo(div);

    // add a show code button right above
    var showCodeText = $('<span>' + (show ? 'Hide' : 'Code') + '</span>');
    var showCodeButton = $('<button type="button" class="btn btn-default btn-xs code-folding-btn pull-right"></button>');
    showCodeButton.append(showCodeText);
    showCodeButton
        .attr('data-toggle', 'collapse')
        .attr('data-target', '#' + id)
        .attr('aria-expanded', show)
        .attr('aria-controls', id);

    var buttonRow = $('<div class="row"></div>');
    var buttonCol = $('<div class="col-md-12"></div>');

    buttonCol.append(showCodeButton);
    buttonRow.append(buttonCol);

    div.before(buttonRow);

    // hack: return show to false, otherwise all next codeBlocks will be shown!
    show = false;

    // update state of button on show/hide
    div.on('hidden.bs.collapse', function () {
      showCodeText.text('Code');
    });
    div.on('show.bs.collapse', function () {
      showCodeText.text('Hide');
    });
  });

}
</script>
<script>
/* ========================================================================
 * Bootstrap: dropdown.js v3.3.7
 * http://getbootstrap.com/javascript/#dropdowns
 * ========================================================================
 * Copyright 2011-2016 Twitter, Inc.
 * Licensed under MIT (https://github.com/twbs/bootstrap/blob/master/LICENSE)
 * ======================================================================== */


+function ($) {
  'use strict';

  // DROPDOWN CLASS DEFINITION
  // =========================

  var backdrop = '.dropdown-backdrop'
  var toggle   = '[data-toggle="dropdown"]'
  var Dropdown = function (element) {
    $(element).on('click.bs.dropdown', this.toggle)
  }

  Dropdown.VERSION = '3.3.7'

  function getParent($this) {
    var selector = $this.attr('data-target')

    if (!selector) {
      selector = $this.attr('href')
      selector = selector && /#[A-Za-z]/.test(selector) && selector.replace(/.*(?=#[^\s]*$)/, '') // strip for ie7
    }

    var $parent = selector && $(selector)

    return $parent && $parent.length ? $parent : $this.parent()
  }

  function clearMenus(e) {
    if (e && e.which === 3) return
    $(backdrop).remove()
    $(toggle).each(function () {
      var $this         = $(this)
      var $parent       = getParent($this)
      var relatedTarget = { relatedTarget: this }

      if (!$parent.hasClass('open')) return

      if (e && e.type == 'click' && /input|textarea/i.test(e.target.tagName) && $.contains($parent[0], e.target)) return

      $parent.trigger(e = $.Event('hide.bs.dropdown', relatedTarget))

      if (e.isDefaultPrevented()) return

      $this.attr('aria-expanded', 'false')
      $parent.removeClass('open').trigger($.Event('hidden.bs.dropdown', relatedTarget))
    })
  }

  Dropdown.prototype.toggle = function (e) {
    var $this = $(this)

    if ($this.is('.disabled, :disabled')) return

    var $parent  = getParent($this)
    var isActive = $parent.hasClass('open')

    clearMenus()

    if (!isActive) {
      if ('ontouchstart' in document.documentElement && !$parent.closest('.navbar-nav').length) {
        // if mobile we use a backdrop because click events don't delegate
        $(document.createElement('div'))
          .addClass('dropdown-backdrop')
          .insertAfter($(this))
          .on('click', clearMenus)
      }

      var relatedTarget = { relatedTarget: this }
      $parent.trigger(e = $.Event('show.bs.dropdown', relatedTarget))

      if (e.isDefaultPrevented()) return

      $this
        .trigger('focus')
        .attr('aria-expanded', 'true')

      $parent
        .toggleClass('open')
        .trigger($.Event('shown.bs.dropdown', relatedTarget))
    }

    return false
  }

  Dropdown.prototype.keydown = function (e) {
    if (!/(38|40|27|32)/.test(e.which) || /input|textarea/i.test(e.target.tagName)) return

    var $this = $(this)

    e.preventDefault()
    e.stopPropagation()

    if ($this.is('.disabled, :disabled')) return

    var $parent  = getParent($this)
    var isActive = $parent.hasClass('open')

    if (!isActive && e.which != 27 || isActive && e.which == 27) {
      if (e.which == 27) $parent.find(toggle).trigger('focus')
      return $this.trigger('click')
    }

    var desc = ' li:not(.disabled):visible a'
    var $items = $parent.find('.dropdown-menu' + desc)

    if (!$items.length) return

    var index = $items.index(e.target)

    if (e.which == 38 && index > 0)                 index--         // up
    if (e.which == 40 && index < $items.length - 1) index++         // down
    if (!~index)                                    index = 0

    $items.eq(index).trigger('focus')
  }


  // DROPDOWN PLUGIN DEFINITION
  // ==========================

  function Plugin(option) {
    return this.each(function () {
      var $this = $(this)
      var data  = $this.data('bs.dropdown')

      if (!data) $this.data('bs.dropdown', (data = new Dropdown(this)))
      if (typeof option == 'string') data[option].call($this)
    })
  }

  var old = $.fn.dropdown

  $.fn.dropdown             = Plugin
  $.fn.dropdown.Constructor = Dropdown


  // DROPDOWN NO CONFLICT
  // ====================

  $.fn.dropdown.noConflict = function () {
    $.fn.dropdown = old
    return this
  }


  // APPLY TO STANDARD DROPDOWN ELEMENTS
  // ===================================

  $(document)
    .on('click.bs.dropdown.data-api', clearMenus)
    .on('click.bs.dropdown.data-api', '.dropdown form', function (e) { e.stopPropagation() })
    .on('click.bs.dropdown.data-api', toggle, Dropdown.prototype.toggle)
    .on('keydown.bs.dropdown.data-api', toggle, Dropdown.prototype.keydown)
    .on('keydown.bs.dropdown.data-api', '.dropdown-menu', Dropdown.prototype.keydown)

}(jQuery);
</script>
<style type="text/css">
.code-folding-btn {
  margin-bottom: 4px;
}

.row { display: flex; }
.collapse { display: none; }
.in { display:block }
.pull-right > .dropdown-menu {
    right: 0;
    left: auto;
}

.dropdown-menu {
    position: absolute;
    top: 100%;
    left: 0;
    z-index: 1000;
    display: none;
    float: left;
    min-width: 160px;
    padding: 5px 0;
    margin: 2px 0 0;
    font-size: 14px;
    text-align: left;
    list-style: none;
    background-color: #fff;
    -webkit-background-clip: padding-box;
    background-clip: padding-box;
    border: 1px solid #ccc;
    border: 1px solid rgba(0,0,0,.15);
    border-radius: 4px;
    -webkit-box-shadow: 0 6px 12px rgba(0,0,0,.175);
    box-shadow: 0 6px 12px rgba(0,0,0,.175);
}

.open > .dropdown-menu {
    display: block;
    color: #ffffff;
    background-color: #ffffff;
    background-image: none;
    border-color: #92897e;
}

.dropdown-menu > li > a {
  display: block;
  padding: 3px 20px;
  clear: both;
  font-weight: 400;
  line-height: 1.42857143;
  color: #000000;
  white-space: nowrap;
}

.dropdown-menu > li > a:hover,
.dropdown-menu > li > a:focus {
  color: #ffffff;
  text-decoration: none;
  background-color: #e95420;
}

.dropdown-menu > .active > a,
.dropdown-menu > .active > a:hover,
.dropdown-menu > .active > a:focus {
  color: #ffffff;
  text-decoration: none;
  background-color: #e95420;
  outline: 0;
}
.dropdown-menu > .disabled > a,
.dropdown-menu > .disabled > a:hover,
.dropdown-menu > .disabled > a:focus {
  color: #aea79f;
}

.dropdown-menu > .disabled > a:hover,
.dropdown-menu > .disabled > a:focus {
  text-decoration: none;
  cursor: not-allowed;
  background-color: transparent;
  background-image: none;
  filter: progid:DXImageTransform.Microsoft.gradient(enabled = false);
}

.btn {
  display: inline-block;
  margin-bottom: 1;
  font-weight: normal;
  text-align: center;
  white-space: nowrap;
  vertical-align: middle;
  -ms-touch-action: manipulation;
      touch-action: manipulation;
  cursor: pointer;
  background-image: none;
  border: 1px solid transparent;
  padding: 4px 8px;
  font-size: 14px;
  line-height: 1.42857143;
  border-radius: 4px;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

.btn:focus,
.btn:active:focus,
.btn.active:focus,
.btn.focus,
.btn:active.focus,
.btn.active.focus {
  outline: 5px auto -webkit-focus-ring-color;
  outline-offset: -2px;
}
.btn:hover,
.btn:focus,
.btn.focus {
  color: #ffffff;
  text-decoration: none;
}
.btn:active,
.btn.active {
  background-image: none;
  outline: 0;
  box-shadow: inset 0 3px 5px rgba(0, 0, 0, 0.125);
}
.btn.disabled,
.btn[disabled],
fieldset[disabled] .btn {
  cursor: not-allowed;
  filter: alpha(opacity=65);
  opacity: 0.65;
  box-shadow: none;
}
a.btn.disabled,
fieldset[disabled] a.btn {
  pointer-events: none;
}
.btn-default {
  color: #ffffff;
  background-color: #aea79f; #important
  border-color: #aea79f;
}

.btn-default:focus,
.btn-default.focus {
  color: #ffffff;
  background-color: #978e83;
  border-color: #6f675e;
}

.btn-default:hover {
  color: #ffffff;
  background-color: #978e83;
  border-color: #92897e;
}
.btn-default:active,
.btn-default.active,
.btn-group > .btn:not(:first-child):not(:last-child):not(.dropdown-toggle) {
  border-radius: 0;
}
.btn-group > .btn:first-child {
  margin-left: 0;
}
.btn-group > .btn:first-child:not(:last-child):not(.dropdown-toggle) {
  border-top-right-radius: 0;
  border-bottom-right-radius: 0;
}
.btn-group > .btn:last-child:not(:first-child),
.btn-group > .dropdown-toggle:not(:first-child) {
  border-top-left-radius: 0;
  border-bottom-left-radius: 0;
}
.btn-group > .btn-group {
  float: left;
}
.btn-group > .btn-group:not(:first-child):not(:last-child) > .btn {
  border-radius: 0;
}
.btn-group > .btn-group:first-child:not(:last-child) > .btn:last-child,
.btn-group > .btn-group:first-child:not(:last-child) > .dropdown-toggle {
  border-top-right-radius: 0;
  border-bottom-right-radius: 0;
}
.btn-group > .btn-group:last-child:not(:first-child) > .btn:first-child {
  border-top-left-radius: 0;
  border-bottom-left-radius: 0;
}
.btn-group .dropdown-toggle:active,
.btn-group.open .dropdown-toggle {
  outline: 0;
}
.btn-group > .btn + .dropdown-toggle {
  padding-right: 8px;
  padding-left: 8px;
}
.btn-group > .btn-lg + .dropdown-toggle {
  padding-right: 12px;
  padding-left: 12px;
}
.btn-group.open .dropdown-toggle {
  box-shadow: inset 0 3px 5px rgba(0, 0, 0, 0.125);
}
.btn-group.open .dropdown-toggle.btn-link {
  box-shadow: none;
}

</style>
<script>
var str = '<div class="btn-group pull-right" style="position: fixed; right: 50px; top: 10px; z-index: 200"><button type="button" class="btn btn-default btn-xs dropdown-toggle" id="allCodeButton" data-toggle="dropdown" aria-haspopup="true" aria-expanded="true" data-_extension-text-contrast=""><span>Code</span> <span class="caret"></span></button><ul class="dropdown-menu" style="min-width: 50px;"><li><a id="rmd-show-all-code" href="#">Show All Code</a></li><li><a id="rmd-hide-all-code" href="#">Hide All Code</a></li></ul></div>';
document.write(str);
</script>
<script>
$(document).ready(function () {
  window.initializeCodeFolding("show" === "hide");
});
</script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>


<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">Landscape Genetic Data Analysis with R</a></li>

<li class="divider"></li>
<li class="part"><span><b>Getting Started</b></span></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i>Introduction</a>
<ul>
<li class="chapter" data-level="" data-path="how-to-use-this-book.html"><a href="how-to-use-this-book.html"><i class="fa fa-check"></i>How to use this Book</a>
<ul>
<li class="chapter" data-level="" data-path="how-to-use-this-book.html"><a href="how-to-use-this-book.html#book-structure"><i class="fa fa-check"></i>1. Book Structure</a>
<ul>
<li class="chapter" data-level="" data-path="how-to-use-this-book.html"><a href="how-to-use-this-book.html#a-getting-started"><i class="fa fa-check"></i>a) Getting Started</a></li>
<li class="chapter" data-level="" data-path="how-to-use-this-book.html"><a href="how-to-use-this-book.html#b-basic-topics"><i class="fa fa-check"></i>b) Basic Topics</a></li>
<li class="chapter" data-level="" data-path="how-to-use-this-book.html"><a href="how-to-use-this-book.html#c-advanced-topics"><i class="fa fa-check"></i>c) Advanced Topics</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="how-to-use-this-book.html"><a href="how-to-use-this-book.html#find-what-is-relevant-for-you"><i class="fa fa-check"></i>2. Find what is relevant for you</a>
<ul>
<li class="chapter" data-level="" data-path="how-to-use-this-book.html"><a href="how-to-use-this-book.html#how-to-use"><i class="fa fa-check"></i>a) How to use the labs</a></li>
<li class="chapter" data-level="" data-path="how-to-use-this-book.html"><a href="how-to-use-this-book.html#b-check-contents"><i class="fa fa-check"></i>b) Check contents</a></li>
</ul></li>
<li><a href="how-to-use-this-book.html#course-r-package-landgencourse">3. Course R package <code>LandGenCourse</code></a>
<ul>
<li class="chapter" data-level="" data-path="how-to-use-this-book.html"><a href="how-to-use-this-book.html#a-how-to-install-or-update"><i class="fa fa-check"></i>a) How to install (or update)</a></li>
<li><a href="how-to-use-this-book.html#b-how-to-use-the-r-package-landgencourse">b) How to use the R package <code>LandGenCourse</code></a></li>
<li class="chapter" data-level="" data-path="how-to-use-this-book.html"><a href="how-to-use-this-book.html#c-video-instructions-for-beginners"><i class="fa fa-check"></i>c) Video instructions for beginners</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="" data-path="list-of-packages.html"><a href="list-of-packages.html"><i class="fa fa-check"></i>List of R Packages by Vignette</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="review-of-r-skills.html"><a href="review-of-r-skills.html"><i class="fa fa-check"></i>Review of R Skills</a>
<ul>
<li class="chapter" data-level="" data-path="basic-r.html"><a href="basic-r.html"><i class="fa fa-check"></i>Basic R Programming</a>
<ul>
<li class="chapter" data-level="" data-path="basic-r.html"><a href="basic-r.html#overview"><i class="fa fa-check"></i>1. Overview</a></li>
<li class="chapter" data-level="" data-path="basic-r.html"><a href="basic-r.html#data-types"><i class="fa fa-check"></i>2. Data Types</a>
<ul>
<li class="chapter" data-level="" data-path="basic-r.html"><a href="basic-r.html#numeric-data-types"><i class="fa fa-check"></i>2.1. Numeric Data Types</a></li>
<li class="chapter" data-level="" data-path="basic-r.html"><a href="basic-r.html#characters"><i class="fa fa-check"></i>2.2. Characters</a></li>
<li class="chapter" data-level="" data-path="basic-r.html"><a href="basic-r.html#factors"><i class="fa fa-check"></i>2.3. Factors</a></li>
<li class="chapter" data-level="" data-path="basic-r.html"><a href="basic-r.html#logical-types"><i class="fa fa-check"></i>2.4. Logical Types</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="basic-r.html"><a href="basic-r.html#data-containers"><i class="fa fa-check"></i>3. Data Containers</a>
<ul>
<li class="chapter" data-level="" data-path="basic-r.html"><a href="basic-r.html#vectors"><i class="fa fa-check"></i>3.1. Vectors</a></li>
<li class="chapter" data-level="" data-path="basic-r.html"><a href="basic-r.html#matrices"><i class="fa fa-check"></i>3.2. Matrices</a></li>
<li class="chapter" data-level="" data-path="basic-r.html"><a href="basic-r.html#lists"><i class="fa fa-check"></i>3.3. Lists</a></li>
<li class="chapter" data-level="" data-path="basic-r.html"><a href="basic-r.html#data-frames"><i class="fa fa-check"></i>3.4. Data Frames</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="basic-r.html"><a href="basic-r.html#programming"><i class="fa fa-check"></i>4. Programming</a>
<ul>
<li class="chapter" data-level="" data-path="basic-r.html"><a href="basic-r.html#function-writing"><i class="fa fa-check"></i>4.1. Function Writing</a></li>
<li class="chapter" data-level="" data-path="basic-r.html"><a href="basic-r.html#variable-scope"><i class="fa fa-check"></i>4.2. Variable Scope</a></li>
<li class="chapter" data-level="" data-path="basic-r.html"><a href="basic-r.html#decision-making"><i class="fa fa-check"></i>4.3. Decision Making</a></li>
<li class="chapter" data-level="" data-path="basic-r.html"><a href="basic-r.html#flow-control"><i class="fa fa-check"></i>4.4. Flow Control</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="" data-path="r-graphics.html"><a href="r-graphics.html"><i class="fa fa-check"></i>R Graphics</a>
<ul>
<li class="chapter" data-level="" data-path="r-graphics.html"><a href="r-graphics.html#overview-1"><i class="fa fa-check"></i>1. Overview</a></li>
<li class="chapter" data-level="" data-path="r-graphics.html"><a href="r-graphics.html#univariate-plots"><i class="fa fa-check"></i>2. Univariate plots</a>
<ul>
<li class="chapter" data-level="" data-path="r-graphics.html"><a href="r-graphics.html#barplots-and-histograms"><i class="fa fa-check"></i>2.1. Barplots and histograms</a></li>
<li class="chapter" data-level="" data-path="r-graphics.html"><a href="r-graphics.html#plotting-parameters"><i class="fa fa-check"></i>2.2. Plotting Parameters</a></li>
<li class="chapter" data-level="" data-path="r-graphics.html"><a href="r-graphics.html#density-plots"><i class="fa fa-check"></i>2.3. Density Plots</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="r-graphics.html"><a href="r-graphics.html#bivariate-plots"><i class="fa fa-check"></i>3. Bivariate plots</a>
<ul>
<li class="chapter" data-level="" data-path="r-graphics.html"><a href="r-graphics.html#overlaying-plots-by-grouping"><i class="fa fa-check"></i>3.1. Overlaying Plots by grouping</a></li>
<li class="chapter" data-level="" data-path="r-graphics.html"><a href="r-graphics.html#boxplots"><i class="fa fa-check"></i>3.2. Boxplots</a></li>
<li class="chapter" data-level="" data-path="r-graphics.html"><a href="r-graphics.html#scatter-plots"><i class="fa fa-check"></i>3.3. Scatter Plots</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="r-graphics.html"><a href="r-graphics.html#advanced-plotting"><i class="fa fa-check"></i>4. Advanced plotting</a>
<ul>
<li class="chapter" data-level="" data-path="r-graphics.html"><a href="r-graphics.html#multiple-plots"><i class="fa fa-check"></i>4.1. Multiple Plots</a></li>
<li class="chapter" data-level="" data-path="r-graphics.html"><a href="r-graphics.html#color-palettes"><i class="fa fa-check"></i>4.2. Color Palettes</a></li>
<li class="chapter" data-level="" data-path="r-graphics.html"><a href="r-graphics.html#rcolorbrewer"><i class="fa fa-check"></i>4.3. RColorBrewer</a></li>
<li class="chapter" data-level="" data-path="r-graphics.html"><a href="r-graphics.html#saving-imagery"><i class="fa fa-check"></i>4.4. Saving Imagery</a></li>
<li class="chapter" data-level="" data-path="r-graphics.html"><a href="r-graphics.html#interactive-graphics"><i class="fa fa-check"></i>4.5. Interactive Graphics</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="" data-path="github.html"><a href="github.html"><i class="fa fa-check"></i>Version Control with Github</a>
<ul>
<li class="chapter" data-level="" data-path="github.html"><a href="github.html#external-github-tutorials"><i class="fa fa-check"></i>1. External Github tutorials</a></li>
<li class="chapter" data-level="" data-path="github.html"><a href="github.html#watch-course-video-version-control-101"><i class="fa fa-check"></i>2. Watch course video: Version Control 101</a></li>
<li class="chapter" data-level="" data-path="github.html"><a href="github.html#preview-slides"><i class="fa fa-check"></i>3. Preview slides</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="resources.html"><a href="resources.html"><i class="fa fa-check"></i>Further R resources</a></li>
</ul></li>
<li class="part"><span><b>Basic Topics</b></span></li>
<li class="chapter" data-level="" data-path="Week1.html"><a href="Week1.html"><i class="fa fa-check"></i>Week 1: Importing Genetic Data</a>
<ul>
<li class="chapter" data-level="" data-path="video-1.html"><a href="video-1.html"><i class="fa fa-check"></i>View Course Video</a>
<ul>
<li class="chapter" data-level="" data-path="video-1.html"><a href="video-1.html#embedded-video"><i class="fa fa-check"></i>1. Embedded Video</a></li>
<li class="chapter" data-level="" data-path="video-1.html"><a href="video-1.html#preview-slides-1"><i class="fa fa-check"></i>2. Preview Slides</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="tutorial-1.html"><a href="tutorial-1.html"><i class="fa fa-check"></i>Interactive Tutorial 1</a>
<ul>
<li class="chapter" data-level="" data-path="tutorial-1.html"><a href="tutorial-1.html#list-of-r-commands-covered-this-week"><i class="fa fa-check"></i>1. List of R commands covered this week</a></li>
<li class="chapter" data-level="" data-path="tutorial-1.html"><a href="tutorial-1.html#general-instructions"><i class="fa fa-check"></i>2. General Instructions</a>
<ul>
<li class="chapter" data-level="" data-path="tutorial-1.html"><a href="tutorial-1.html#a-how-to-access-tutorials"><i class="fa fa-check"></i>a) How to access tutorials:</a></li>
<li class="chapter" data-level="" data-path="tutorial-1.html"><a href="tutorial-1.html#b-how-to-complete-tutorial"><i class="fa fa-check"></i>b) How to complete tutorial:</a></li>
<li class="chapter" data-level="" data-path="tutorial-1.html"><a href="tutorial-1.html#c-how-to-submit-answers-participating-institutions-only"><i class="fa fa-check"></i>c) How to submit answers (participating institutions only):</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="" data-path="WE-1.html"><a href="WE-1.html"><i class="fa fa-check"></i>Worked Example</a>
<ul>
<li class="chapter" data-level="" data-path="WE-1.html"><a href="WE-1.html#overview-of-worked-example"><i class="fa fa-check"></i>1. Overview of Worked Example</a>
<ul>
<li class="chapter" data-level="" data-path="WE-1.html"><a href="WE-1.html#a.-goals"><i class="fa fa-check"></i>a. Goals</a></li>
<li class="chapter" data-level="" data-path="WE-1.html"><a href="WE-1.html#b.-data-set"><i class="fa fa-check"></i>b. Data set</a></li>
<li class="chapter" data-level="" data-path="WE-1.html"><a href="WE-1.html#c.-required-r-libraries"><i class="fa fa-check"></i>c. Required R libraries</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="WE-1.html"><a href="WE-1.html#import-data-from-.csv-file"><i class="fa fa-check"></i>2. Import data from .csv file</a>
<ul>
<li class="chapter" data-level="" data-path="WE-1.html"><a href="WE-1.html#a.-preparation"><i class="fa fa-check"></i>a. Preparation</a></li>
<li class="chapter" data-level="" data-path="WE-1.html"><a href="WE-1.html#b.-excel-file"><i class="fa fa-check"></i>b. Excel file</a></li>
<li class="chapter" data-level="" data-path="WE-1.html"><a href="WE-1.html#c.-import-with-function-read.csv"><i class="fa fa-check"></i>c. Import with function ‘read.csv’</a></li>
<li class="chapter" data-level="" data-path="WE-1.html"><a href="WE-1.html#d.-create-id-variable"><i class="fa fa-check"></i>d. Create ID variable</a></li>
<li class="chapter" data-level="" data-path="WE-1.html"><a href="WE-1.html#e.-export-with-function-write.csv"><i class="fa fa-check"></i>e. Export with function ‘write.csv’</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="WE-1.html"><a href="WE-1.html#create-genind-object"><i class="fa fa-check"></i>3. Create ‘genind’ Object</a>
<ul>
<li class="chapter" data-level="" data-path="WE-1.html"><a href="WE-1.html#a.-read-the-helpfile"><i class="fa fa-check"></i>a. Read the helpfile</a></li>
<li class="chapter" data-level="" data-path="WE-1.html"><a href="WE-1.html#b.-set-all-arguments-of-function-df2genind"><i class="fa fa-check"></i>b. Set all arguments of function ‘df2genind’</a></li>
<li class="chapter" data-level="" data-path="WE-1.html"><a href="WE-1.html#c.-check-imported-data"><i class="fa fa-check"></i>c. Check imported data</a></li>
<li class="chapter" data-level="" data-path="WE-1.html"><a href="WE-1.html#d.-summarize-genind-object"><i class="fa fa-check"></i>d. Summarize ‘genind’ object</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="WE-1.html"><a href="WE-1.html#view-information-stored-in-genind-object"><i class="fa fa-check"></i>4. View information stored in ‘genind’ object</a>
<ul>
<li><a href="WE-1.html#a.-slot-tab">a. Slot <code>@tab</code></a></li>
<li><a href="WE-1.html#b.-slot-loc.n.all">b. Slot <code>@loc.n.all</code></a></li>
<li><a href="WE-1.html#c.-slot-loc.fac">c. Slot <code>@loc.fac</code></a></li>
<li><a href="WE-1.html#d.-slot-all.names">d. Slot <code>@all.names</code></a></li>
<li class="chapter" data-level="" data-path="WE-1.html"><a href="WE-1.html#e.-further-slots"><i class="fa fa-check"></i>e. Further slots</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="WE-1.html"><a href="WE-1.html#import-data-with-gstudio-package"><i class="fa fa-check"></i>5. Import data with ‘gstudio’ package</a>
<ul>
<li class="chapter" data-level="" data-path="WE-1.html"><a href="WE-1.html#a.-function-read_population"><i class="fa fa-check"></i>a. Function ‘read_population’</a></li>
<li class="chapter" data-level="" data-path="WE-1.html"><a href="WE-1.html#b.-set-arguments"><i class="fa fa-check"></i>b. Set arguments</a></li>
<li class="chapter" data-level="" data-path="WE-1.html"><a href="WE-1.html#c.-check-imported-data-1"><i class="fa fa-check"></i>c. Check imported data</a></li>
<li class="chapter" data-level="" data-path="WE-1.html"><a href="WE-1.html#d.-add-frogid"><i class="fa fa-check"></i>d. Add FrogID</a></li>
<li class="chapter" data-level="" data-path="WE-1.html"><a href="WE-1.html#e.-convert-to-genind-object"><i class="fa fa-check"></i>e. Convert to ‘genind’ object</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="" data-path="r-exercise-week-1.html"><a href="r-exercise-week-1.html"><i class="fa fa-check"></i>R exercise Week 1</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="Week2.html"><a href="Week2.html"><i class="fa fa-check"></i>Week 2: Spatial Data</a>
<ul>
<li class="chapter" data-level="" data-path="video-2.html"><a href="video-2.html"><i class="fa fa-check"></i>View Course Video</a>
<ul>
<li class="chapter" data-level="" data-path="video-2.html"><a href="video-2.html#embedded-video-1"><i class="fa fa-check"></i>1. Embedded Video</a></li>
<li class="chapter" data-level="" data-path="video-2.html"><a href="video-2.html#preview-slides-2"><i class="fa fa-check"></i>2. Preview Slides</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="tutorial-2.html"><a href="tutorial-2.html"><i class="fa fa-check"></i>Interactive Tutorial 2</a>
<ul>
<li class="chapter" data-level="" data-path="tutorial-2.html"><a href="tutorial-2.html#list-of-r-commands-covered-this-week-1"><i class="fa fa-check"></i>1. List of R commands covered this week</a></li>
<li class="chapter" data-level="" data-path="tutorial-2.html"><a href="tutorial-2.html#general-instructions-1"><i class="fa fa-check"></i>2. General Instructions</a>
<ul>
<li class="chapter" data-level="" data-path="tutorial-2.html"><a href="tutorial-2.html#a-how-to-access-tutorials-1"><i class="fa fa-check"></i>a) How to access tutorials:</a></li>
<li class="chapter" data-level="" data-path="tutorial-2.html"><a href="tutorial-2.html#b-how-to-complete-tutorial-1"><i class="fa fa-check"></i>b) How to complete tutorial:</a></li>
<li class="chapter" data-level="" data-path="tutorial-2.html"><a href="tutorial-2.html#c-how-to-submit-answers-participating-institutions-only-1"><i class="fa fa-check"></i>c) How to submit answers (participating institutions only):</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="" data-path="WE-2.html"><a href="WE-2.html"><i class="fa fa-check"></i>Worked Example</a>
<ul>
<li class="chapter" data-level="" data-path="WE-2.html"><a href="WE-2.html#overview-of-worked-example-1"><i class="fa fa-check"></i>1. Overview of Worked Example</a>
<ul>
<li class="chapter" data-level="" data-path="WE-2.html"><a href="WE-2.html#a.-goals-1"><i class="fa fa-check"></i>a. Goals</a></li>
<li class="chapter" data-level="" data-path="WE-2.html"><a href="WE-2.html#b.-data-set-1"><i class="fa fa-check"></i>b. Data set</a></li>
<li class="chapter" data-level="" data-path="WE-2.html"><a href="WE-2.html#c.-required-r-libraries-1"><i class="fa fa-check"></i>c. Required R libraries</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="WE-2.html"><a href="WE-2.html#import-site-data-from-.csv-file"><i class="fa fa-check"></i>2. Import site data from .csv file</a>
<ul>
<li class="chapter" data-level="" data-path="WE-2.html"><a href="WE-2.html#a.-import-data-into-spatialpointsdataframe"><i class="fa fa-check"></i>a. Import data into SpatialPointsDataFrame</a></li>
<li class="chapter" data-level="" data-path="WE-2.html"><a href="WE-2.html#b.-add-spatial-reference-data"><i class="fa fa-check"></i>b. Add spatial reference data</a></li>
<li class="chapter" data-level="" data-path="WE-2.html"><a href="WE-2.html#c.-change-projection"><i class="fa fa-check"></i>c. Change projection</a></li>
<li class="chapter" data-level="" data-path="WE-2.html"><a href="WE-2.html#d.-access-data-in-spatialpointsdataframe"><i class="fa fa-check"></i>d. Access data in ‘SpatialPointsDataFrame’</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="WE-2.html"><a href="WE-2.html#display-raster-data-and-overlay-sampling-locations-extract-data"><i class="fa fa-check"></i>3. Display raster data and overlay sampling locations, extract data</a>
<ul>
<li class="chapter" data-level="" data-path="WE-2.html"><a href="WE-2.html#a.-display-raster-data"><i class="fa fa-check"></i>a. Display raster data</a></li>
<li class="chapter" data-level="" data-path="WE-2.html"><a href="WE-2.html#b.-change-color-ramp-add-sampling-locations"><i class="fa fa-check"></i>b. Change color ramp, add sampling locations</a></li>
<li class="chapter" data-level="" data-path="WE-2.html"><a href="WE-2.html#c.-extract-raster-values-at-sampling-locations"><i class="fa fa-check"></i>c. Extract raster values at sampling locations</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="WE-2.html"><a href="WE-2.html#calculate-landscape-metrics"><i class="fa fa-check"></i>4. Calculate landscape metrics</a>
<ul>
<li class="chapter" data-level="" data-path="WE-2.html"><a href="WE-2.html#a.-calculate-patch--class--and-landscape-level-landscape-metrics"><i class="fa fa-check"></i>a. Calculate patch-, class- and landscape level landscape metrics</a></li>
<li class="chapter" data-level="" data-path="WE-2.html"><a href="WE-2.html#b.-calculate-patch-level-landscape-metrics-for-evergreen-forest"><i class="fa fa-check"></i>b. Calculate patch-level landscape metrics for ‘Evergreen Forest’</a></li>
<li class="chapter" data-level="" data-path="WE-2.html"><a href="WE-2.html#c.-extract-forest-patch-size-at-samplig-locations"><i class="fa fa-check"></i>c. Extract forest patch size at samplig locations</a></li>
<li class="chapter" data-level="" data-path="WE-2.html"><a href="WE-2.html#d.-plot-a-bubble-map-of-forest-patch-size-at-sampling-locations"><i class="fa fa-check"></i>d. Plot a bubble map of forest patch size at sampling locations</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="WE-2.html"><a href="WE-2.html#sample-landscape-metrics-within-buffer-around-sampling-locations"><i class="fa fa-check"></i>5. Sample landscape metrics within buffer around sampling locations</a>
<ul>
<li class="chapter" data-level="" data-path="WE-2.html"><a href="WE-2.html#b.-extract-landscape-metric-of-choice-for-a-single-cover-type-as-vector"><i class="fa fa-check"></i>b. Extract landscape metric of choice for a single cover type (as vector)</a></li>
<li class="chapter" data-level="" data-path="WE-2.html"><a href="WE-2.html#c.-extract-landscape-metric-of-choice-for-all-cover-types-as-data-frame."><i class="fa fa-check"></i>c. Extract landscape metric of choice for all cover types (as data frame).</a></li>
<li class="chapter" data-level="" data-path="WE-2.html"><a href="WE-2.html#d.-extract-all-landscape-metrics-for-a-single-cover-type-as-data-frame"><i class="fa fa-check"></i>d. Extract all landscape metrics for a single cover type (as data frame)</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="" data-path="r-exercise-week-2.html"><a href="r-exercise-week-2.html"><i class="fa fa-check"></i>R Exercise Week 2</a></li>
<li class="chapter" data-level="" data-path="bonus-2a.html"><a href="bonus-2a.html"><i class="fa fa-check"></i>Bonus Vignette</a>
<ul>
<li class="chapter" data-level="" data-path="bonus-2a.html"><a href="bonus-2a.html#overview-2"><i class="fa fa-check"></i>1. Overview</a>
<ul>
<li class="chapter" data-level="" data-path="bonus-2a.html"><a href="bonus-2a.html#a.-goals-2"><i class="fa fa-check"></i>a. Goals</a></li>
<li class="chapter" data-level="" data-path="bonus-2a.html"><a href="bonus-2a.html#b.-required-r-libraries"><i class="fa fa-check"></i>b. Required R libraries</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="bonus-2a.html"><a href="bonus-2a.html#convert-spatialpointsdataframe-to-sf-object"><i class="fa fa-check"></i>2. Convert ‘SpatialPointsDataFrame’ to ‘sf’ object</a></li>
<li class="chapter" data-level="" data-path="bonus-2a.html"><a href="bonus-2a.html#plot-variables-first-10-in-data-set"><i class="fa fa-check"></i>3. Plot variables (first 10 in data set)</a></li>
<li class="chapter" data-level="" data-path="bonus-2a.html"><a href="bonus-2a.html#export-site-data-as-esri-shapefile"><i class="fa fa-check"></i>4. Export site data as ESRI shapefile</a></li>
<li class="chapter" data-level="" data-path="bonus-2a.html"><a href="bonus-2a.html#display-categorical-map-with-a-predefined-color-table"><i class="fa fa-check"></i>5. Display categorical map with a predefined color table</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="" data-path="Week3.html"><a href="Week3.html"><i class="fa fa-check"></i>Week 3: Genetic Diversity</a>
<ul>
<li class="chapter" data-level="" data-path="video-3.html"><a href="video-3.html"><i class="fa fa-check"></i>View Course Video</a>
<ul>
<li class="chapter" data-level="" data-path="video-3.html"><a href="video-3.html#embedded-video-2"><i class="fa fa-check"></i>1. Embedded Video</a></li>
<li class="chapter" data-level="" data-path="video-3.html"><a href="video-3.html#video-part-1"><i class="fa fa-check"></i>Video, Part 1</a></li>
<li class="chapter" data-level="" data-path="video-3.html"><a href="video-3.html#video-part-2"><i class="fa fa-check"></i>Video, Part 2</a></li>
<li class="chapter" data-level="" data-path="video-3.html"><a href="video-3.html#preview-slides-3"><i class="fa fa-check"></i>Preview Slides</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="tutorial-3.html"><a href="tutorial-3.html"><i class="fa fa-check"></i>Interactive Tutorial 3</a>
<ul>
<li class="chapter" data-level="" data-path="tutorial-3.html"><a href="tutorial-3.html#list-of-r-commands-covered-this-week-2"><i class="fa fa-check"></i>1. List of R commands covered this week</a></li>
<li class="chapter" data-level="" data-path="tutorial-3.html"><a href="tutorial-3.html#general-instructions-2"><i class="fa fa-check"></i>2. General Instructions</a>
<ul>
<li class="chapter" data-level="" data-path="tutorial-3.html"><a href="tutorial-3.html#a-how-to-access-tutorials-2"><i class="fa fa-check"></i>a) How to access tutorials:</a></li>
<li class="chapter" data-level="" data-path="tutorial-3.html"><a href="tutorial-3.html#b-how-to-complete-tutorial-2"><i class="fa fa-check"></i>b) How to complete tutorial:</a></li>
<li class="chapter" data-level="" data-path="tutorial-3.html"><a href="tutorial-3.html#c-how-to-submit-answers-participating-institutions-only-2"><i class="fa fa-check"></i>c) How to submit answers (participating institutions only):</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="" data-path="WE-3.html"><a href="WE-3.html"><i class="fa fa-check"></i>Worked Example</a>
<ul>
<li class="chapter" data-level="" data-path="WE-3.html"><a href="WE-3.html#overview-of-worked-example-2"><i class="fa fa-check"></i>1. Overview of Worked Example</a>
<ul>
<li class="chapter" data-level="" data-path="WE-3.html"><a href="WE-3.html#a.-goals-3"><i class="fa fa-check"></i>a. Goals</a></li>
<li class="chapter" data-level="" data-path="WE-3.html"><a href="WE-3.html#b.-data-set-2"><i class="fa fa-check"></i>b. Data set</a></li>
<li class="chapter" data-level="" data-path="WE-3.html"><a href="WE-3.html#c.-required-r-libraries-2"><i class="fa fa-check"></i>c. Required R libraries</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="WE-3.html"><a href="WE-3.html#basic-checking-of-markers-and-populations"><i class="fa fa-check"></i>2. Basic checking of markers and populations</a>
<ul>
<li class="chapter" data-level="" data-path="WE-3.html"><a href="WE-3.html#a.-re-create-genind-object"><i class="fa fa-check"></i>a. Re-create genind object</a></li>
<li class="chapter" data-level="" data-path="WE-3.html"><a href="WE-3.html#b.-check-that-markers-are-polymorphic"><i class="fa fa-check"></i>b. Check that markers are polymorphic</a></li>
<li class="chapter" data-level="" data-path="WE-3.html"><a href="WE-3.html#c.-check-for-deviations-from-hardy-weinberg-equilibrium-hwe"><i class="fa fa-check"></i>c. Check for deviations from Hardy-Weinberg equilibrium (HWE)</a></li>
<li class="chapter" data-level="" data-path="WE-3.html"><a href="WE-3.html#d.-check-for-linkage-disequilibrium-ld"><i class="fa fa-check"></i>d. Check for linkage disequilibrium (LD)</a></li>
<li class="chapter" data-level="" data-path="WE-3.html"><a href="WE-3.html#e.-check-for-null-alleles"><i class="fa fa-check"></i>e. Check for null alleles</a></li>
<li class="chapter" data-level="" data-path="WE-3.html"><a href="WE-3.html#f.-overall-interpretation"><i class="fa fa-check"></i>f. Overall interpretation</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="WE-3.html"><a href="WE-3.html#assess-genetic-diversity"><i class="fa fa-check"></i>3. Assess genetic diversity</a>
<ul>
<li class="chapter" data-level="" data-path="WE-3.html"><a href="WE-3.html#a.-rarefied-allelic-richness"><i class="fa fa-check"></i>a. Rarefied allelic richness</a></li>
<li class="chapter" data-level="" data-path="WE-3.html"><a href="WE-3.html#b.-observed-and-expected-heterozygosity"><i class="fa fa-check"></i>b. Observed and expected heterozygosity</a></li>
<li class="chapter" data-level="" data-path="WE-3.html"><a href="WE-3.html#c.-create-table-with-sitel-level-genetic-diversity-measures"><i class="fa fa-check"></i>c. Create table with sitel-level genetic diversity measures</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="WE-3.html"><a href="WE-3.html#aggregate-genetic-data-at-population-level-allele-frequencies"><i class="fa fa-check"></i>4. Aggregate genetic data at population level (allele frequencies)</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="r-exercise-week-3.html"><a href="r-exercise-week-3.html"><i class="fa fa-check"></i>R Exercise Week 3</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="Week4.html"><a href="Week4.html"><i class="fa fa-check"></i>Week 4: Metapopulations</a>
<ul>
<li class="chapter" data-level="" data-path="video-4.html"><a href="video-4.html"><i class="fa fa-check"></i>View Course Video</a>
<ul>
<li class="chapter" data-level="" data-path="video-4.html"><a href="video-4.html#embedded-video-3"><i class="fa fa-check"></i>1. Embedded Video</a></li>
<li class="chapter" data-level="" data-path="video-4.html"><a href="video-4.html#video-part-1-1"><i class="fa fa-check"></i>Video, Part 1</a></li>
<li class="chapter" data-level="" data-path="video-4.html"><a href="video-4.html#video-part-2-1"><i class="fa fa-check"></i>Video, Part 2</a></li>
<li class="chapter" data-level="" data-path="video-4.html"><a href="video-4.html#preview-slides-4"><i class="fa fa-check"></i>Preview Slides</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="tutorial-4.html"><a href="tutorial-4.html"><i class="fa fa-check"></i>Interactive Tutorial 4</a>
<ul>
<li class="chapter" data-level="" data-path="tutorial-4.html"><a href="tutorial-4.html#list-of-r-commands-covered-this-week-3"><i class="fa fa-check"></i>1. List of R commands covered this week</a></li>
<li class="chapter" data-level="" data-path="tutorial-4.html"><a href="tutorial-4.html#general-instructions-3"><i class="fa fa-check"></i>2. General Instructions</a>
<ul>
<li class="chapter" data-level="" data-path="tutorial-4.html"><a href="tutorial-4.html#a-how-to-access-tutorials-3"><i class="fa fa-check"></i>a) How to access tutorials:</a></li>
<li class="chapter" data-level="" data-path="tutorial-4.html"><a href="tutorial-4.html#b-how-to-complete-tutorial-3"><i class="fa fa-check"></i>b) How to complete tutorial:</a></li>
<li class="chapter" data-level="" data-path="tutorial-4.html"><a href="tutorial-4.html#c-how-to-submit-answers-participating-institutions-only-3"><i class="fa fa-check"></i>c) How to submit answers (participating institutions only):</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="" data-path="WE-4.html"><a href="WE-4.html"><i class="fa fa-check"></i>Worked Example</a>
<ul>
<li class="chapter" data-level="" data-path="WE-4.html"><a href="WE-4.html#overview-of-worked-example-3"><i class="fa fa-check"></i>1. Overview of Worked Example</a>
<ul>
<li class="chapter" data-level="" data-path="WE-4.html"><a href="WE-4.html#a.-goals-4"><i class="fa fa-check"></i>a. Goals</a></li>
<li class="chapter" data-level="" data-path="WE-4.html"><a href="WE-4.html#b.-data-set-3"><i class="fa fa-check"></i>b. Data set</a></li>
<li class="chapter" data-level="" data-path="WE-4.html"><a href="WE-4.html#reference"><i class="fa fa-check"></i>Reference:</a></li>
<li class="chapter" data-level="" data-path="WE-4.html"><a href="WE-4.html#c.-required-r-libraries-3"><i class="fa fa-check"></i>c. Required R libraries</a></li>
<li class="chapter" data-level="" data-path="WE-4.html"><a href="WE-4.html#d.-import-data"><i class="fa fa-check"></i>d. Import data</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="WE-4.html"><a href="WE-4.html#spatial-distribution-of-genetic-structure"><i class="fa fa-check"></i>2. Spatial distribution of genetic structure</a>
<ul>
<li class="chapter" data-level="" data-path="WE-4.html"><a href="WE-4.html#a.-creating-a-genind-object-with-the-hierarchical-data-set"><i class="fa fa-check"></i>a. Creating a ‘genind’ object with the hierarchical data set</a></li>
<li class="chapter" data-level="" data-path="WE-4.html"><a href="WE-4.html#b.-hierarchical-amova"><i class="fa fa-check"></i>b. Hierarchical AMOVA</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="WE-4.html"><a href="WE-4.html#what-determines-genetic-differentiation-among-sites"><i class="fa fa-check"></i>3. What determines genetic differentiation among sites?</a>
<ul>
<li class="chapter" data-level="" data-path="WE-4.html"><a href="WE-4.html#a.-correlation-matrix"><i class="fa fa-check"></i>a. Correlation matrix</a></li>
<li class="chapter" data-level="" data-path="WE-4.html"><a href="WE-4.html#b.-scatterplots"><i class="fa fa-check"></i>b. Scatterplots</a></li>
<li class="chapter" data-level="" data-path="WE-4.html"><a href="WE-4.html#c.-regression-model"><i class="fa fa-check"></i>c. Regression model</a></li>
<li class="chapter" data-level="" data-path="WE-4.html"><a href="WE-4.html#d.-which-populations-dont-fit-the-general-pattern"><i class="fa fa-check"></i>d. Which populations don’t fit the general pattern?</a></li>
<li class="chapter" data-level="" data-path="WE-4.html"><a href="WE-4.html#e.-regression-model-without-outliers"><i class="fa fa-check"></i>e. Regression model without outliers</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="WE-4.html"><a href="WE-4.html#what-determines-genetic-diversity"><i class="fa fa-check"></i>4. What determines genetic diversity?</a>
<ul>
<li class="chapter" data-level="" data-path="WE-4.html"><a href="WE-4.html#a.-correlation-matrix-1"><i class="fa fa-check"></i>a. Correlation matrix</a></li>
<li class="chapter" data-level="" data-path="WE-4.html"><a href="WE-4.html#b.-regression-models"><i class="fa fa-check"></i>b. Regression models</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="WE-4.html"><a href="WE-4.html#are-genetic-differentiation-and-diversity-related"><i class="fa fa-check"></i>5. Are genetic differentiation and diversity related?</a></li>
<li class="chapter" data-level="" data-path="WE-4.html"><a href="WE-4.html#effect-of-recent-extinction-events"><i class="fa fa-check"></i>6. Effect of recent extinction events</a>
<ul>
<li class="chapter" data-level="" data-path="WE-4.html"><a href="WE-4.html#a.-effect-of-patch-extinction-event-temporal-data-set"><i class="fa fa-check"></i>a. Effect of patch extinction event (temporal data set)</a></li>
<li class="chapter" data-level="" data-path="WE-4.html"><a href="WE-4.html#b.-power-analysis"><i class="fa fa-check"></i>b. Power analysis</a></li>
<li class="chapter" data-level="" data-path="WE-4.html"><a href="WE-4.html#c.-sample-size-calculation"><i class="fa fa-check"></i>c. Sample size calculation</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="" data-path="r-exercise-week-4.html"><a href="r-exercise-week-4.html"><i class="fa fa-check"></i>R Exercise Week 4</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="Week5.html"><a href="Week5.html"><i class="fa fa-check"></i>Week 5: Spatial Statistics</a>
<ul>
<li class="chapter" data-level="" data-path="video-5.html"><a href="video-5.html"><i class="fa fa-check"></i>View Course Video</a>
<ul>
<li class="chapter" data-level="" data-path="video-5.html"><a href="video-5.html#embedded-video-4"><i class="fa fa-check"></i>1. Embedded Video</a></li>
<li class="chapter" data-level="" data-path="video-5.html"><a href="video-5.html#preview-slides-5"><i class="fa fa-check"></i>Preview Slides</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="tutorial-5.html"><a href="tutorial-5.html"><i class="fa fa-check"></i>Interactive Tutorial 5</a>
<ul>
<li class="chapter" data-level="" data-path="tutorial-5.html"><a href="tutorial-5.html#list-of-r-commands-covered-this-week-4"><i class="fa fa-check"></i>1. List of R commands covered this week</a></li>
<li class="chapter" data-level="" data-path="tutorial-5.html"><a href="tutorial-5.html#general-instructions-4"><i class="fa fa-check"></i>2. General Instructions</a>
<ul>
<li class="chapter" data-level="" data-path="tutorial-5.html"><a href="tutorial-5.html#a-how-to-access-tutorials-4"><i class="fa fa-check"></i>a) How to access tutorials:</a></li>
<li class="chapter" data-level="" data-path="tutorial-5.html"><a href="tutorial-5.html#b-how-to-complete-tutorial-4"><i class="fa fa-check"></i>b) How to complete tutorial:</a></li>
<li class="chapter" data-level="" data-path="tutorial-5.html"><a href="tutorial-5.html#c-how-to-submit-answers-participating-institutions-only-4"><i class="fa fa-check"></i>c) How to submit answers (participating institutions only):</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="" data-path="WE-5.html"><a href="WE-5.html"><i class="fa fa-check"></i>Worked Example</a>
<ul>
<li class="chapter" data-level="" data-path="WE-5.html"><a href="WE-5.html#overview-of-worked-example-4"><i class="fa fa-check"></i>1. Overview of Worked Example</a>
<ul>
<li class="chapter" data-level="" data-path="WE-5.html"><a href="WE-5.html#a.-goals-5"><i class="fa fa-check"></i>a. Goals</a></li>
<li class="chapter" data-level="" data-path="WE-5.html"><a href="WE-5.html#b.-data-set-4"><i class="fa fa-check"></i>b. Data set</a></li>
<li class="chapter" data-level="" data-path="WE-5.html"><a href="WE-5.html#c.-required-r-packages"><i class="fa fa-check"></i>c. Required R packages</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="WE-5.html"><a href="WE-5.html#data-import-and-manipulation"><i class="fa fa-check"></i>2. Data import and manipulation</a>
<ul>
<li class="chapter" data-level="" data-path="WE-5.html"><a href="WE-5.html#a.-create-an-ecogen-object"><i class="fa fa-check"></i>a. Create an ecogen object</a></li>
<li class="chapter" data-level="" data-path="WE-5.html"><a href="WE-5.html#b.-aggregate-to-ecopop"><i class="fa fa-check"></i>b. Aggregate to ecopop</a></li>
<li class="chapter" data-level="" data-path="WE-5.html"><a href="WE-5.html#b.-add-site-level-data"><i class="fa fa-check"></i>b. Add site-level data</a></li>
<li class="chapter" data-level="" data-path="WE-5.html"><a href="WE-5.html#c.-export-to-adegenet-and-gstudio"><i class="fa fa-check"></i>c. Export to adegenet and gstudio</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="WE-5.html"><a href="WE-5.html#calculate-genetic-distances"><i class="fa fa-check"></i>3. Calculate genetic distances</a>
<ul>
<li class="chapter" data-level="" data-path="WE-5.html"><a href="WE-5.html#a.-genetic-distances-calculated-from-genind-object"><i class="fa fa-check"></i>a. Genetic distances calculated from genind object</a></li>
<li class="chapter" data-level="" data-path="WE-5.html"><a href="WE-5.html#b.-more-distance-matrices-with-gstudio"><i class="fa fa-check"></i>b. More distance matrices with ‘gstudio’</a></li>
<li class="chapter" data-level="" data-path="WE-5.html"><a href="WE-5.html#c.-assemble-distance-matrices"><i class="fa fa-check"></i>c. Assemble distance matrices</a></li>
<li class="chapter" data-level="" data-path="WE-5.html"><a href="WE-5.html#d.-export-genetic-distance-matrices"><i class="fa fa-check"></i>d. Export genetic distance matrices</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="WE-5.html"><a href="WE-5.html#perform-a-mantel-test-to-test-for-ibd"><i class="fa fa-check"></i>4. Perform a Mantel test to test for IBD</a>
<ul>
<li class="chapter" data-level="" data-path="WE-5.html"><a href="WE-5.html#a.-visually-check-linearity"><i class="fa fa-check"></i>a. Visually check linearity</a></li>
<li class="chapter" data-level="" data-path="WE-5.html"><a href="WE-5.html#b.-perform-mantel-test"><i class="fa fa-check"></i>b. Perform Mantel test</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="WE-5.html"><a href="WE-5.html#create-mantel-correlogram-for-genetic-data"><i class="fa fa-check"></i>5. Create Mantel correlogram for genetic data</a>
<ul>
<li class="chapter" data-level="" data-path="WE-5.html"><a href="WE-5.html#a.-create-a-first-mantel-correlogram"><i class="fa fa-check"></i>a. Create a first Mantel correlogram</a></li>
<li class="chapter" data-level="" data-path="WE-5.html"><a href="WE-5.html#b.-vary-distance-class-definition"><i class="fa fa-check"></i>b. Vary distance class definition</a></li>
<li class="chapter" data-level="" data-path="WE-5.html"><a href="WE-5.html#c.-alternative-measures-of-genetic-distances"><i class="fa fa-check"></i>c. Alternative measures of genetic distances</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="WE-5.html"><a href="WE-5.html#specify-spatial-weights-and-calculate-morans-i"><i class="fa fa-check"></i>6. Specify spatial weights and calculate Moran’s I</a>
<ul>
<li class="chapter" data-level="" data-path="WE-5.html"><a href="WE-5.html#a.-defining-spatial-neighbors"><i class="fa fa-check"></i>a. Defining spatial neighbors</a></li>
<li class="chapter" data-level="" data-path="WE-5.html"><a href="WE-5.html#b.-defining-spatial-weights"><i class="fa fa-check"></i>b. Defining spatial weights</a></li>
<li class="chapter" data-level="" data-path="WE-5.html"><a href="WE-5.html#c.-calculating-and-testing-morans-i"><i class="fa fa-check"></i>c. Calculating and testing Moran’s I</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="" data-path="r-exercise-week-5.html"><a href="r-exercise-week-5.html"><i class="fa fa-check"></i>R Exercise Week 5</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="Week6.html"><a href="Week6.html"><i class="fa fa-check"></i>Week 6: Quantitative Genetics</a>
<ul>
<li class="chapter" data-level="" data-path="video-6.html"><a href="video-6.html"><i class="fa fa-check"></i>View Course Video</a>
<ul>
<li class="chapter" data-level="" data-path="video-6.html"><a href="video-6.html#embedded-video-5"><i class="fa fa-check"></i>1. Embedded Video</a></li>
<li class="chapter" data-level="" data-path="video-6.html"><a href="video-6.html#video-part-1-2"><i class="fa fa-check"></i>Video, Part 1</a></li>
<li class="chapter" data-level="" data-path="video-6.html"><a href="video-6.html#video-part-2-2"><i class="fa fa-check"></i>Video, Part 2</a></li>
<li class="chapter" data-level="" data-path="video-6.html"><a href="video-6.html#preview-slides-6"><i class="fa fa-check"></i>Preview Slides</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="tutorial-6.html"><a href="tutorial-6.html"><i class="fa fa-check"></i>Interactive Tutorial 6</a>
<ul>
<li class="chapter" data-level="" data-path="tutorial-6.html"><a href="tutorial-6.html#list-of-r-commands-covered-this-week-5"><i class="fa fa-check"></i>1. List of R commands covered this week</a></li>
<li class="chapter" data-level="" data-path="tutorial-6.html"><a href="tutorial-6.html#general-instructions-5"><i class="fa fa-check"></i>2. General Instructions</a>
<ul>
<li class="chapter" data-level="" data-path="tutorial-6.html"><a href="tutorial-6.html#a-how-to-access-tutorials-5"><i class="fa fa-check"></i>a) How to access tutorials:</a></li>
<li class="chapter" data-level="" data-path="tutorial-6.html"><a href="tutorial-6.html#b-how-to-complete-tutorial-5"><i class="fa fa-check"></i>b) How to complete tutorial:</a></li>
<li class="chapter" data-level="" data-path="tutorial-6.html"><a href="tutorial-6.html#c-how-to-submit-answers-participating-institutions-only-5"><i class="fa fa-check"></i>c) How to submit answers (participating institutions only):</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="" data-path="WE-6.html"><a href="WE-6.html"><i class="fa fa-check"></i>Worked Example</a>
<ul>
<li class="chapter" data-level="" data-path="WE-6.html"><a href="WE-6.html#overview-of-worked-example-5"><i class="fa fa-check"></i>1. Overview of Worked Example</a>
<ul>
<li class="chapter" data-level="" data-path="WE-6.html"><a href="WE-6.html#a.-goals-6"><i class="fa fa-check"></i>a. Goals</a></li>
<li class="chapter" data-level="" data-path="WE-6.html"><a href="WE-6.html#b.-data-set-5"><i class="fa fa-check"></i>b. Data set</a></li>
<li class="chapter" data-level="" data-path="WE-6.html"><a href="WE-6.html#c.-required-r-packages-1"><i class="fa fa-check"></i>c. Required R packages</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="WE-6.html"><a href="WE-6.html#estimate-genetic-and-non-genetic-variance-components-from-a-common-garden-experiment"><i class="fa fa-check"></i>2. Estimate genetic and non-genetic variance components from a common garden experiment</a>
<ul>
<li class="chapter" data-level="" data-path="WE-6.html"><a href="WE-6.html#a.-import-phenytypic-data"><i class="fa fa-check"></i>a. Import phenytypic data</a></li>
<li class="chapter" data-level="" data-path="WE-6.html"><a href="WE-6.html#b.-fit-linear-mixed-effects-models-lmm-to-trait-data-observed-on-families"><i class="fa fa-check"></i>b. Fit linear mixed effects models (LMM) to trait data observed on families</a></li>
<li class="chapter" data-level="" data-path="WE-6.html"><a href="WE-6.html#c.-compare-model-fit"><i class="fa fa-check"></i>c. Compare model fit</a></li>
<li class="chapter" data-level="" data-path="WE-6.html"><a href="WE-6.html#d.-check-model-validity"><i class="fa fa-check"></i>d. Check model validity</a></li>
<li class="chapter" data-level="" data-path="WE-6.html"><a href="WE-6.html#e.-estimate-variance-components"><i class="fa fa-check"></i>e. Estimate variance components</a></li>
<li class="chapter" data-level="" data-path="WE-6.html"><a href="WE-6.html#f.-significance-testing"><i class="fa fa-check"></i>f. Significance testing</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="WE-6.html"><a href="WE-6.html#estimate-trait-heritability"><i class="fa fa-check"></i>3. Estimate trait heritability</a>
<ul>
<li class="chapter" data-level="" data-path="WE-6.html"><a href="WE-6.html#a.-estimate-heritability"><i class="fa fa-check"></i>a. Estimate heritability</a></li>
<li class="chapter" data-level="" data-path="WE-6.html"><a href="WE-6.html#b.-estimate-trait-differentiation"><i class="fa fa-check"></i>b. Estimate trait differentiation</a></li>
</ul></li>
<li><a href="WE-6.html#compare-q_st-to-f_st-measured-from-snp-data">4. Compare <span class="math inline">\(Q_{st}\)</span> to <span class="math inline">\(F_{st}\)</span> measured from SNP data</a>
<ul>
<li class="chapter" data-level="" data-path="WE-6.html"><a href="WE-6.html#a.-import-ecogen-object"><i class="fa fa-check"></i>a. Import ‘ecogen’ object</a></li>
<li><a href="WE-6.html#b.-compare-q_st-to-f_st">b. Compare <span class="math inline">\(Q_{ST}\)</span> to <span class="math inline">\(F_{ST}\)</span></a></li>
</ul></li>
<li class="chapter" data-level="" data-path="WE-6.html"><a href="WE-6.html#assess-correlation-between-trait-and-environment"><i class="fa fa-check"></i>5. Assess correlation between trait and environment</a>
<ul>
<li class="chapter" data-level="" data-path="WE-6.html"><a href="WE-6.html#a.-describe-and-extract-effects"><i class="fa fa-check"></i>a. Describe and extract effects</a></li>
<li class="chapter" data-level="" data-path="WE-6.html"><a href="WE-6.html#b.-correlation-of-population-effect-with-environment"><i class="fa fa-check"></i>b. Correlation of population effect with environment</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="" data-path="r-exercise-week-6.html"><a href="r-exercise-week-6.html"><i class="fa fa-check"></i>R Exercise Week 6</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="Week7.html"><a href="Week7.html"><i class="fa fa-check"></i>Week 7: Spatial Linear Models</a>
<ul>
<li class="chapter" data-level="" data-path="video-7.html"><a href="video-7.html"><i class="fa fa-check"></i>View Course Video</a>
<ul>
<li class="chapter" data-level="" data-path="video-7.html"><a href="video-7.html#embedded-video-6"><i class="fa fa-check"></i>1. Embedded Video</a></li>
<li class="chapter" data-level="" data-path="video-7.html"><a href="video-7.html#video-part-1-3"><i class="fa fa-check"></i>Video, Part 1</a></li>
<li class="chapter" data-level="" data-path="video-7.html"><a href="video-7.html#video-part-2-3"><i class="fa fa-check"></i>Video, Part 2</a></li>
<li class="chapter" data-level="" data-path="video-7.html"><a href="video-7.html#preview-slides-7"><i class="fa fa-check"></i>Preview Slides</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="tutorial-7.html"><a href="tutorial-7.html"><i class="fa fa-check"></i>Interactive Tutorial 7</a>
<ul>
<li class="chapter" data-level="" data-path="tutorial-7.html"><a href="tutorial-7.html#list-of-r-commands-covered-this-week-6"><i class="fa fa-check"></i>1. List of R commands covered this week</a></li>
<li class="chapter" data-level="" data-path="tutorial-7.html"><a href="tutorial-7.html#general-instructions-6"><i class="fa fa-check"></i>2. General Instructions</a>
<ul>
<li class="chapter" data-level="" data-path="tutorial-7.html"><a href="tutorial-7.html#a-how-to-access-tutorials-6"><i class="fa fa-check"></i>a) How to access tutorials:</a></li>
<li class="chapter" data-level="" data-path="tutorial-7.html"><a href="tutorial-7.html#b-how-to-complete-tutorial-6"><i class="fa fa-check"></i>b) How to complete tutorial:</a></li>
<li class="chapter" data-level="" data-path="tutorial-7.html"><a href="tutorial-7.html#c-how-to-submit-answers-participating-institutions-only-6"><i class="fa fa-check"></i>c) How to submit answers (participating institutions only):</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="" data-path="WE-7.html"><a href="WE-7.html"><i class="fa fa-check"></i>Worked Example</a>
<ul>
<li class="chapter" data-level="" data-path="WE-7.html"><a href="WE-7.html#overview-of-worked-example-6"><i class="fa fa-check"></i>1. Overview of Worked Example</a>
<ul>
<li class="chapter" data-level="" data-path="WE-7.html"><a href="WE-7.html#a.-goals-7"><i class="fa fa-check"></i>a. Goals</a></li>
<li class="chapter" data-level="" data-path="WE-7.html"><a href="WE-7.html#b.-data-set-6"><i class="fa fa-check"></i>b. Data set</a></li>
<li class="chapter" data-level="" data-path="WE-7.html"><a href="WE-7.html#c.-required-r-packages-2"><i class="fa fa-check"></i>c. Required R packages</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="WE-7.html"><a href="WE-7.html#explore-data-set"><i class="fa fa-check"></i>2. Explore data set</a>
<ul>
<li class="chapter" data-level="" data-path="WE-7.html"><a href="WE-7.html#a.-import-data"><i class="fa fa-check"></i>a. Import data</a></li>
<li class="chapter" data-level="" data-path="WE-7.html"><a href="WE-7.html#b.-create-a-map"><i class="fa fa-check"></i>b. Create a map</a></li>
<li class="chapter" data-level="" data-path="WE-7.html"><a href="WE-7.html#c.-explore-correlations"><i class="fa fa-check"></i>c. Explore correlations</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="WE-7.html"><a href="WE-7.html#test-regression-residuals-for-spatial-autocorrelation"><i class="fa fa-check"></i>3. Test regression residuals for spatial autocorrelation</a>
<ul>
<li class="chapter" data-level="" data-path="WE-7.html"><a href="WE-7.html#a.-fit-regression-models"><i class="fa fa-check"></i>a. Fit regression models</a></li>
<li class="chapter" data-level="" data-path="WE-7.html"><a href="WE-7.html#b.-test-for-spatial-autocorrelation-morans-i"><i class="fa fa-check"></i>b. Test for spatial autocorrelation (Moran’s I):</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="WE-7.html"><a href="WE-7.html#fit-models-with-spatially-correlated-error-gls-with-package-nlme"><i class="fa fa-check"></i>4. Fit models with spatially correlated error (GLS) with package ‘nlme’</a>
<ul>
<li class="chapter" data-level="" data-path="WE-7.html"><a href="WE-7.html#a.-plot-empirical-variogram"><i class="fa fa-check"></i>a. Plot empirical variogram</a></li>
<li class="chapter" data-level="" data-path="WE-7.html"><a href="WE-7.html#b.-fit-variogram-models"><i class="fa fa-check"></i>b. Fit variogram models</a></li>
<li class="chapter" data-level="" data-path="WE-7.html"><a href="WE-7.html#c.-select-best-fitting-model"><i class="fa fa-check"></i>c. Select best-fitting model</a></li>
<li class="chapter" data-level="" data-path="WE-7.html"><a href="WE-7.html#d.-plot-fitted-variogram-model"><i class="fa fa-check"></i>d. Plot fitted variogram model</a></li>
<li class="chapter" data-level="" data-path="WE-7.html"><a href="WE-7.html#e.-add-random-factor"><i class="fa fa-check"></i>e. Add random factor</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="WE-7.html"><a href="WE-7.html#fit-spatial-simultaneous-autoregressive-error-models-sar"><i class="fa fa-check"></i>5. Fit spatial simultaneous autoregressive error models (SAR)</a>
<ul>
<li class="chapter" data-level="" data-path="WE-7.html"><a href="WE-7.html#a.-fit-and-compare-alternative-sar-models"><i class="fa fa-check"></i>a. Fit and compare alternative SAR models</a></li>
<li class="chapter" data-level="" data-path="WE-7.html"><a href="WE-7.html#b.-interpret-best-fitting-sar-model"><i class="fa fa-check"></i>b. Interpret best-fitting SAR model</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="WE-7.html"><a href="WE-7.html#spatial-filtering-with-mem-using-package-spmoran"><i class="fa fa-check"></i>6. Spatial filtering with MEM using package ‘spmoran’</a>
<ul>
<li class="chapter" data-level="" data-path="WE-7.html"><a href="WE-7.html#a.-default-method"><i class="fa fa-check"></i>a. Default method</a></li>
<li class="chapter" data-level="" data-path="WE-7.html"><a href="WE-7.html#b.-using-a-custom-connectivity-matrix"><i class="fa fa-check"></i>b. Using a custom connectivity matrix</a></li>
<li class="chapter" data-level="" data-path="WE-7.html"><a href="WE-7.html#c.-plot-spatial-eigenvectors"><i class="fa fa-check"></i>c. Plot spatial eigenvectors</a></li>
<li class="chapter" data-level="" data-path="WE-7.html"><a href="WE-7.html#d.-random-effect-model"><i class="fa fa-check"></i>d. Random effect model</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="WE-7.html"><a href="WE-7.html#fit-spatially-varying-coefficients-model-with-package-spmoran"><i class="fa fa-check"></i>7. Fit spatially varying coefficients model with package ‘spmoran’</a></li>
<li class="chapter" data-level="" data-path="WE-7.html"><a href="WE-7.html#model-with-patchsize-and-ibr"><i class="fa fa-check"></i>Model with PatchSize and IBR</a></li>
<li class="chapter" data-level="" data-path="WE-7.html"><a href="WE-7.html#model-with-ibr-only"><i class="fa fa-check"></i>Model with IBR only</a></li>
<li class="chapter" data-level="" data-path="WE-7.html"><a href="WE-7.html#conclusions"><i class="fa fa-check"></i>8. Conclusions</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="r-exercise-week-7.html"><a href="r-exercise-week-7.html"><i class="fa fa-check"></i>R Exercise Week 7</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="Week8.html"><a href="Week8.html"><i class="fa fa-check"></i>Week 8: Simulation Experiments</a>
<ul>
<li class="chapter" data-level="" data-path="video-8.html"><a href="video-8.html"><i class="fa fa-check"></i>View Course Video</a>
<ul>
<li class="chapter" data-level="" data-path="video-8.html"><a href="video-8.html#embedded-video-7"><i class="fa fa-check"></i>1. Embedded Video</a></li>
<li class="chapter" data-level="" data-path="video-8.html"><a href="video-8.html#video-part-1-4"><i class="fa fa-check"></i>Video, Part 1</a></li>
<li class="chapter" data-level="" data-path="video-8.html"><a href="video-8.html#video-part-2-4"><i class="fa fa-check"></i>Video, Part 2</a></li>
<li class="chapter" data-level="" data-path="video-8.html"><a href="video-8.html#preview-slides-8"><i class="fa fa-check"></i>Preview Slides</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="tutorial-8.html"><a href="tutorial-8.html"><i class="fa fa-check"></i>Interactive Tutorial 8</a>
<ul>
<li class="chapter" data-level="" data-path="tutorial-8.html"><a href="tutorial-8.html#list-of-r-commands-covered-this-week-7"><i class="fa fa-check"></i>1. List of R commands covered this week</a></li>
<li class="chapter" data-level="" data-path="tutorial-8.html"><a href="tutorial-8.html#general-instructions-7"><i class="fa fa-check"></i>2. General Instructions</a>
<ul>
<li class="chapter" data-level="" data-path="tutorial-8.html"><a href="tutorial-8.html#a-how-to-access-tutorials-7"><i class="fa fa-check"></i>a) How to access tutorials:</a></li>
<li class="chapter" data-level="" data-path="tutorial-8.html"><a href="tutorial-8.html#b-how-to-complete-tutorial-7"><i class="fa fa-check"></i>b) How to complete tutorial:</a></li>
<li class="chapter" data-level="" data-path="tutorial-8.html"><a href="tutorial-8.html#c-how-to-submit-answers-participating-institutions-only-7"><i class="fa fa-check"></i>c) How to submit answers (participating institutions only):</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="" data-path="WE-8.html"><a href="WE-8.html"><i class="fa fa-check"></i>Worked Example</a>
<ul>
<li class="chapter" data-level="" data-path="WE-8.html"><a href="WE-8.html#overview-of-worked-example-7"><i class="fa fa-check"></i>1. Overview of Worked Example</a>
<ul>
<li class="chapter" data-level="" data-path="WE-8.html"><a href="WE-8.html#a.-goals-8"><i class="fa fa-check"></i>a. Goals</a></li>
<li class="chapter" data-level="" data-path="WE-8.html"><a href="WE-8.html#b.-data-set-7"><i class="fa fa-check"></i>b. Data set</a></li>
<li class="chapter" data-level="" data-path="WE-8.html"><a href="WE-8.html#c.-required-r-libraries-4"><i class="fa fa-check"></i>c. Required R libraries</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="WE-8.html"><a href="WE-8.html#initialize-a-landscape"><i class="fa fa-check"></i>2. Initialize a landscape</a>
<ul>
<li class="chapter" data-level="" data-path="WE-8.html"><a href="WE-8.html#a.-create-a-random-landscape"><i class="fa fa-check"></i>a. Create a random landscape</a></li>
<li class="chapter" data-level="" data-path="WE-8.html"><a href="WE-8.html#b.-add-populations-to-the-landscape-using-minimal-distance"><i class="fa fa-check"></i>b. Add populations to the landscape (using minimal distance)</a></li>
<li class="chapter" data-level="" data-path="WE-8.html"><a href="WE-8.html#c.-initialise-a-metapopulation"><i class="fa fa-check"></i>c. Initialise a metapopulation</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="WE-8.html"><a href="WE-8.html#define-simulation-parameters"><i class="fa fa-check"></i>3. Define simulation parameters</a>
<ul>
<li class="chapter" data-level="" data-path="WE-8.html"><a href="WE-8.html#a.-define-your-metapopulation"><i class="fa fa-check"></i>a. Define your metapopulation</a></li>
<li class="chapter" data-level="" data-path="WE-8.html"><a href="WE-8.html#b.-initialise-your-population-on-the-landscape"><i class="fa fa-check"></i>b. Initialise your population on the landscape</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="WE-8.html"><a href="WE-8.html#run-simulations-and-analyze-results"><i class="fa fa-check"></i>4. Run simulations and analyze results</a>
<ul>
<li class="chapter" data-level="" data-path="WE-8.html"><a href="WE-8.html#a.-run-your-simulation-over-multiple-time-steps-years"><i class="fa fa-check"></i>a. Run your simulation over multiple time steps (years)</a></li>
<li class="chapter" data-level="" data-path="WE-8.html"><a href="WE-8.html#b.-analyse-your-simulated-population-with-a-partial-mantel-test"><i class="fa fa-check"></i>b. Analyse your simulated population with a partial Mantel test</a></li>
<li class="chapter" data-level="" data-path="WE-8.html"><a href="WE-8.html#c.-optional-analyze-your-simulated-populations-using-sunder"><i class="fa fa-check"></i>c. Optional: Analyze your simulated populations using ‘Sunder’</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="WE-8.html"><a href="WE-8.html#run-simulator-using-a-previously-defined-parameter-set"><i class="fa fa-check"></i>5. Run simulator using a previously defined parameter set</a>
<ul>
<li class="chapter" data-level="" data-path="WE-8.html"><a href="WE-8.html#a.-specify-and-record-the-parameter-combinations"><i class="fa fa-check"></i>a. Specify and record the parameter combinations</a></li>
<li class="chapter" data-level="" data-path="WE-8.html"><a href="WE-8.html#b.-run-the-simulator-over-every-parameter-combination."><i class="fa fa-check"></i>b. Run the simulator over every parameter combination.</a></li>
<li class="chapter" data-level="" data-path="WE-8.html"><a href="WE-8.html#c.-save-your-complete-simulation-run-input-and-output"><i class="fa fa-check"></i>c. Save your complete simulation run (input and output)</a></li>
<li class="chapter" data-level="" data-path="WE-8.html"><a href="WE-8.html#d.-analyze-and-synthesize-results"><i class="fa fa-check"></i>d. Analyze and synthesize results</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="" data-path="r-exercise-week-8.html"><a href="r-exercise-week-8.html"><i class="fa fa-check"></i>R Exercise Week 8</a></li>
<li class="chapter" data-level="" data-path="bonus-8a.html"><a href="bonus-8a.html"><i class="fa fa-check"></i>Bonus Vignette</a>
<ul>
<li class="chapter" data-level="" data-path="bonus-8a.html"><a href="bonus-8a.html#overview-of-bonus-material"><i class="fa fa-check"></i>1. Overview of Bonus Material</a>
<ul>
<li class="chapter" data-level="" data-path="bonus-8a.html"><a href="bonus-8a.html#a.-goals-9"><i class="fa fa-check"></i>a. Goals</a></li>
<li class="chapter" data-level="" data-path="bonus-8a.html"><a href="bonus-8a.html#b.-data-set-8"><i class="fa fa-check"></i>b. Data set</a></li>
<li class="chapter" data-level="" data-path="bonus-8a.html"><a href="bonus-8a.html#c.-required-r-libraries-5"><i class="fa fa-check"></i>c. Required R libraries</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="bonus-8a.html"><a href="bonus-8a.html#navigating-the-file-system"><i class="fa fa-check"></i>2. Navigating the file system</a>
<ul>
<li class="chapter" data-level="" data-path="bonus-8a.html"><a href="bonus-8a.html#a.-where-am-i"><i class="fa fa-check"></i>a. Where am I?</a></li>
<li class="chapter" data-level="" data-path="bonus-8a.html"><a href="bonus-8a.html#b.-accessing-a-system-file"><i class="fa fa-check"></i>b. Accessing a system file</a></li>
<li class="chapter" data-level="" data-path="bonus-8a.html"><a href="bonus-8a.html#c.-file-manipulation"><i class="fa fa-check"></i>c. File manipulation</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="bonus-8a.html"><a href="bonus-8a.html#benchmarking-file-import-and-export-options"><i class="fa fa-check"></i>3. Benchmarking file import and export options</a>
<ul>
<li class="chapter" data-level="" data-path="bonus-8a.html"><a href="bonus-8a.html#a.-benchmark-methods-for-importing-csv-files"><i class="fa fa-check"></i>a. Benchmark methods for importing csv files</a></li>
<li class="chapter" data-level="" data-path="bonus-8a.html"><a href="bonus-8a.html#b.-check-handling-of-character-data"><i class="fa fa-check"></i>b. Check handling of character data</a></li>
<li class="chapter" data-level="" data-path="bonus-8a.html"><a href="bonus-8a.html#c.-binary-files"><i class="fa fa-check"></i>c. Binary files</a></li>
<li class="chapter" data-level="" data-path="bonus-8a.html"><a href="bonus-8a.html#d.-should-you-save-your-r-workspace"><i class="fa fa-check"></i>d. Should you save your R workspace?</a></li>
<li class="chapter" data-level="" data-path="bonus-8a.html"><a href="bonus-8a.html#e.-compile-your-functions"><i class="fa fa-check"></i>e. Compile your functions</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="bonus-8a.html"><a href="bonus-8a.html#profiling-your-code"><i class="fa fa-check"></i>4. Profiling your code</a>
<ul>
<li class="chapter" data-level="" data-path="bonus-8a.html"><a href="bonus-8a.html#a.-named-chunks"><i class="fa fa-check"></i>a. Named chunks</a></li>
<li class="chapter" data-level="" data-path="bonus-8a.html"><a href="bonus-8a.html#b.-profiling-some-lines-of-code-with-profvis"><i class="fa fa-check"></i>b. Profiling some lines of code with ‘profvis’</a></li>
<li class="chapter" data-level="" data-path="bonus-8a.html"><a href="bonus-8a.html#c.-profiling-an-r-script-with-profvis"><i class="fa fa-check"></i>c. Profiling an R script with ‘profvis’</a></li>
<li class="chapter" data-level="" data-path="bonus-8a.html"><a href="bonus-8a.html#d.-converting-an-r-notebook-to-an-r-script"><i class="fa fa-check"></i>d. Converting an R Notebook to an R script</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="bonus-8a.html"><a href="bonus-8a.html#creating-and-executing-a-bash-r-script"><i class="fa fa-check"></i>5. Creating and executing a Bash R script</a>
<ul>
<li class="chapter" data-level="" data-path="bonus-8a.html"><a href="bonus-8a.html#a.-run-r-script-directly-in-the-terminal"><i class="fa fa-check"></i>a. Run R script directly in the Terminal</a></li>
<li class="chapter" data-level="" data-path="bonus-8a.html"><a href="bonus-8a.html#b.-create-a-bash-r-script"><i class="fa fa-check"></i>b. Create a Bash R script</a></li>
<li class="chapter" data-level="" data-path="bonus-8a.html"><a href="bonus-8a.html#c.-executing-a-bash-r-script"><i class="fa fa-check"></i>c. Executing a Bash R script</a></li>
<li class="chapter" data-level="" data-path="bonus-8a.html"><a href="bonus-8a.html#d.-moving-to-a-node-or-cluster"><i class="fa fa-check"></i>d. Moving to a node or cluster?</a></li>
<li class="chapter" data-level="" data-path="bonus-8a.html"><a href="bonus-8a.html#e.-store-your-session-info-and-package-versions"><i class="fa fa-check"></i>e. Store your session info and package versions</a></li>
<li class="chapter" data-level="" data-path="bonus-8a.html"><a href="bonus-8a.html#f.-potential-windows-issues"><i class="fa fa-check"></i>f. Potential Windows issues</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="bonus-8a.html"><a href="bonus-8a.html#parallelizing-code"><i class="fa fa-check"></i>6. Parallelizing code</a>
<ul>
<li class="chapter" data-level="" data-path="bonus-8a.html"><a href="bonus-8a.html#a.-replace-lapply-by-mclapply-with-package-parallel"><i class="fa fa-check"></i>a. Replace ‘lapply’ by ‘mclapply’ with package ‘parallel’</a></li>
<li class="chapter" data-level="" data-path="bonus-8a.html"><a href="bonus-8a.html#b.-replace-for-by-foreach-with-package-doparallel"><i class="fa fa-check"></i>b. Replace ‘for’ by ‘foreach’ with package ‘doParallel’</a></li>
</ul></li>
</ul></li>
</ul></li>
<li class="part"><span><b>Advanced Topics</b></span></li>
<li class="chapter" data-level="" data-path="Week9.html"><a href="Week9.html"><i class="fa fa-check"></i>Week 9: Population Structure</a>
<ul>
<li class="chapter" data-level="" data-path="WE-9.html"><a href="WE-9.html"><i class="fa fa-check"></i>Worked Example</a>
<ul>
<li class="chapter" data-level="" data-path="WE-9.html"><a href="WE-9.html#overview-of-worked-example-8"><i class="fa fa-check"></i>1. Overview of Worked Example</a>
<ul>
<li class="chapter" data-level="" data-path="WE-9.html"><a href="WE-9.html#a.-goals-10"><i class="fa fa-check"></i>a. Goals</a></li>
<li class="chapter" data-level="" data-path="WE-9.html"><a href="WE-9.html#b.-data-sets"><i class="fa fa-check"></i>b. Data sets</a></li>
<li class="chapter" data-level="" data-path="WE-9.html"><a href="WE-9.html#c.-required-r-libraries-6"><i class="fa fa-check"></i>c. Required R libraries</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="WE-9.html"><a href="WE-9.html#simulated-data-2-island-model"><i class="fa fa-check"></i>2. Simulated data: 2-island model</a>
<ul>
<li class="chapter" data-level="" data-path="WE-9.html"><a href="WE-9.html#a.-import-data-1"><i class="fa fa-check"></i>a. Import data</a></li>
<li class="chapter" data-level="" data-path="WE-9.html"><a href="WE-9.html#b.-perform-principal-components-analysis-pca"><i class="fa fa-check"></i>b. Perform Principal Components Analysis (PCA)</a></li>
<li class="chapter" data-level="" data-path="WE-9.html"><a href="WE-9.html#c.-create-synthetic-spatial-coordinates-xy-and-map-them"><i class="fa fa-check"></i>c. Create synthetic spatial coordinates (X,Y) and map them</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="WE-9.html"><a href="WE-9.html#simulated-data-2-island-model-with-admixture"><i class="fa fa-check"></i>3. Simulated data: 2-island model with admixture</a></li>
<li class="chapter" data-level="" data-path="WE-9.html"><a href="WE-9.html#empirical-data-threespine-sticklebacks"><i class="fa fa-check"></i>4. Empirical data: Threespine sticklebacks</a>
<ul>
<li class="chapter" data-level="" data-path="WE-9.html"><a href="WE-9.html#a.-import-the-data"><i class="fa fa-check"></i>a. Import the data:</a></li>
<li class="chapter" data-level="" data-path="WE-9.html"><a href="WE-9.html#b.-examine-population-structure-with-pca"><i class="fa fa-check"></i>b. Examine population structure with PCA</a></li>
<li class="chapter" data-level="" data-path="WE-9.html"><a href="WE-9.html#c.-clustering-with-snmf-similar-to-structure"><i class="fa fa-check"></i>c. Clustering with SNMF (similar to ‘STRUCTURE’)</a></li>
<li class="chapter" data-level="" data-path="WE-9.html"><a href="WE-9.html#d.-plot-ancestry-proportions"><i class="fa fa-check"></i>d. Plot ancestry proportions</a></li>
<li class="chapter" data-level="" data-path="WE-9.html"><a href="WE-9.html#e.-visualize-admixture-proportions-on-a-map"><i class="fa fa-check"></i>e. Visualize admixture proportions on a map</a></li>
</ul></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="" data-path="Week10.html"><a href="Week10.html"><i class="fa fa-check"></i>Week 10: Landscape Resistance</a>
<ul>
<li class="chapter" data-level="" data-path="WE-10.html"><a href="WE-10.html"><i class="fa fa-check"></i>Worked Example</a>
<ul>
<li class="chapter" data-level="" data-path="WE-10.html"><a href="WE-10.html#overview-of-worked-example-9"><i class="fa fa-check"></i>1. Overview of Worked Example</a>
<ul>
<li class="chapter" data-level="" data-path="WE-10.html"><a href="WE-10.html#a.-goals-11"><i class="fa fa-check"></i>a. Goals</a></li>
<li class="chapter" data-level="" data-path="WE-10.html"><a href="WE-10.html#b.-data-set-9"><i class="fa fa-check"></i>b. Data set</a></li>
<li class="chapter" data-level="" data-path="WE-10.html"><a href="WE-10.html#c.-required-r-libraries-7"><i class="fa fa-check"></i>c. Required R libraries</a></li>
<li class="chapter" data-level="" data-path="WE-10.html"><a href="WE-10.html#d.-import-rasters-gsp-err-cti-and-ffp."><i class="fa fa-check"></i>d. Import rasters (gsp, err, cti, and ffp).</a></li>
<li class="chapter" data-level="" data-path="WE-10.html"><a href="WE-10.html#e.-import-site-data-with-sampling-locations"><i class="fa fa-check"></i>e. Import site data with sampling locations</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="WE-10.html"><a href="WE-10.html#explore-the-data-set"><i class="fa fa-check"></i>2. Explore the data set</a>
<ul>
<li class="chapter" data-level="" data-path="WE-10.html"><a href="WE-10.html#a.-plot-all-rasters"><i class="fa fa-check"></i>a. Plot all rasters</a></li>
<li class="chapter" data-level="" data-path="WE-10.html"><a href="WE-10.html#b.-plot-spatial-points-over-ffp-raster"><i class="fa fa-check"></i>b. Plot spatial points over ffp raster</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="WE-10.html"><a href="WE-10.html#setting-cost-values-and-calculating-conductance"><i class="fa fa-check"></i>3. Setting cost values and calculating conductance</a>
<ul>
<li class="chapter" data-level="" data-path="WE-10.html"><a href="WE-10.html#a.-resistance-vs.-conductance-values"><i class="fa fa-check"></i>a. Resistance vs. conductance values</a></li>
<li class="chapter" data-level="" data-path="WE-10.html"><a href="WE-10.html#b.-as-needed-get-all-rasters-at-the-same-scale"><i class="fa fa-check"></i>b. As needed: Get all rasters at the same scale</a></li>
<li class="chapter" data-level="" data-path="WE-10.html"><a href="WE-10.html#c.-calculate-conductance-values"><i class="fa fa-check"></i>c. Calculate conductance values</a></li>
<li class="chapter" data-level="" data-path="WE-10.html"><a href="WE-10.html#d.-create-a-single-landscape-conductance-raster"><i class="fa fa-check"></i>d. Create a single landscape conductance raster</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="WE-10.html"><a href="WE-10.html#convert-conductance-into-effective-distance"><i class="fa fa-check"></i>4. Convert conductance into effective distance</a>
<ul>
<li class="chapter" data-level="" data-path="WE-10.html"><a href="WE-10.html#a.-create-a-transition-layer"><i class="fa fa-check"></i>a. Create a transition layer</a></li>
<li class="chapter" data-level="" data-path="WE-10.html"><a href="WE-10.html#b.-visually-inspect-the-raster"><i class="fa fa-check"></i>b. Visually inspect the raster</a></li>
<li class="chapter" data-level="" data-path="WE-10.html"><a href="WE-10.html#c.-correct-for-geometric-distortion"><i class="fa fa-check"></i>c. Correct for geometric distortion</a></li>
<li class="chapter" data-level="" data-path="WE-10.html"><a href="WE-10.html#d.-plot-shortest-paths-in-space"><i class="fa fa-check"></i>d. Plot shortest paths in space</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="WE-10.html"><a href="WE-10.html#create-cost-distance-matrices"><i class="fa fa-check"></i>5. Create cost-distance matrices</a>
<ul>
<li class="chapter" data-level="" data-path="WE-10.html"><a href="WE-10.html#a.-least-cost-distance"><i class="fa fa-check"></i>a. Least cost distance</a></li>
<li class="chapter" data-level="" data-path="WE-10.html"><a href="WE-10.html#b.-cost-distance-matrix-based-on-random-paths-similar-to-circuitscape"><i class="fa fa-check"></i>b. Cost-distance matrix based on random paths (similar to Circuitscape)</a></li>
<li class="chapter" data-level="" data-path="WE-10.html"><a href="WE-10.html#c.-compare-cost-distances"><i class="fa fa-check"></i>c. Compare cost distances</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="WE-10.html"><a href="WE-10.html#how-does-changing-resolution-affect-these-metrics"><i class="fa fa-check"></i>6. How does changing resolution affect these metrics?</a>
<ul>
<li class="chapter" data-level="" data-path="WE-10.html"><a href="WE-10.html#a.-create-loop"><i class="fa fa-check"></i>a. Create loop</a></li>
<li class="chapter" data-level="" data-path="WE-10.html"><a href="WE-10.html#b.-plot-the-results"><i class="fa fa-check"></i>b. Plot the results</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="WE-10.html"><a href="WE-10.html#references"><i class="fa fa-check"></i>7. References</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="" data-path="Week11.html"><a href="Week11.html"><i class="fa fa-check"></i>Week 11: Detecting Adaptation</a>
<ul>
<li class="chapter" data-level="" data-path="WE-11.html"><a href="WE-11.html"><i class="fa fa-check"></i>Worked Example</a>
<ul>
<li class="chapter" data-level="" data-path="WE-11.html"><a href="WE-11.html#overview-of-worked-example-10"><i class="fa fa-check"></i>1. Overview of Worked Example</a>
<ul>
<li class="chapter" data-level="" data-path="WE-11.html"><a href="WE-11.html#a.-goals-12"><i class="fa fa-check"></i>a. Goals</a></li>
<li class="chapter" data-level="" data-path="WE-11.html"><a href="WE-11.html#b.-data-set-10"><i class="fa fa-check"></i>b. Data set</a></li>
<li class="chapter" data-level="" data-path="WE-11.html"><a href="WE-11.html#c.-required-r-libraries-8"><i class="fa fa-check"></i>c. Required R libraries</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="WE-11.html"><a href="WE-11.html#import-and-prepare-the-data"><i class="fa fa-check"></i>2. Import and prepare the data</a>
<ul>
<li class="chapter" data-level="" data-path="WE-11.html"><a href="WE-11.html#a.-import-the-genetic-data"><i class="fa fa-check"></i>a. Import the genetic data</a></li>
<li class="chapter" data-level="" data-path="WE-11.html"><a href="WE-11.html#b.-import-the-environmental-data"><i class="fa fa-check"></i>b. Import the environmental data</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="WE-11.html"><a href="WE-11.html#latent-factor-mixed-models-lfmm-a-univariate-gea"><i class="fa fa-check"></i>3. Latent Factor Mixed Models (LFMM): a univariate GEA</a>
<ul>
<li class="chapter" data-level="" data-path="WE-11.html"><a href="WE-11.html#a.-determine-k"><i class="fa fa-check"></i>a. Determine K</a></li>
<li class="chapter" data-level="" data-path="WE-11.html"><a href="WE-11.html#b.-run-lfmm"><i class="fa fa-check"></i>b. Run LFMM</a></li>
<li class="chapter" data-level="" data-path="WE-11.html"><a href="WE-11.html#c.-identify-lfmm-candidates-using-false-discovery-rate"><i class="fa fa-check"></i>c. Identify LFMM candidates using False Discovery Rate</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="WE-11.html"><a href="WE-11.html#redundancy-analysis-rda-a-multivariate-gea"><i class="fa fa-check"></i>4. Redundancy Analysis (RDA): a multivariate GEA</a>
<ul>
<li class="chapter" data-level="" data-path="WE-11.html"><a href="WE-11.html#a.-run-rda"><i class="fa fa-check"></i>a. Run RDA</a></li>
<li class="chapter" data-level="" data-path="WE-11.html"><a href="WE-11.html#b.-identify-rda-candidates"><i class="fa fa-check"></i>b. Identify RDA candidates</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="WE-11.html"><a href="WE-11.html#compare-lfmm-and-rda-candidates"><i class="fa fa-check"></i>5. Compare LFMM and RDA candidates</a></li>
<li class="chapter" data-level="" data-path="WE-11.html"><a href="WE-11.html#a-quick-note-on-controlling-for-population-structure-in-rda"><i class="fa fa-check"></i>6. A quick note on controlling for population structure in RDA</a></li>
<li class="chapter" data-level="" data-path="WE-11.html"><a href="WE-11.html#references-1"><i class="fa fa-check"></i>7. References</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="" data-path="Week12.html"><a href="Week12.html"><i class="fa fa-check"></i>Week 12: Model Selection</a>
<ul>
<li class="chapter" data-level="" data-path="WE-12.html"><a href="WE-12.html"><i class="fa fa-check"></i>Worked Example</a>
<ul>
<li class="chapter" data-level="" data-path="WE-12.html"><a href="WE-12.html#overview-of-worked-example-11"><i class="fa fa-check"></i>1. Overview of Worked Example</a>
<ul>
<li class="chapter" data-level="" data-path="WE-12.html"><a href="WE-12.html#a.-goals-and-background"><i class="fa fa-check"></i>a. Goals and Background</a></li>
<li class="chapter" data-level="" data-path="WE-12.html"><a href="WE-12.html#b.-data-set-11"><i class="fa fa-check"></i>b. Data set</a></li>
<li class="chapter" data-level="" data-path="WE-12.html"><a href="WE-12.html#c.-required-packages"><i class="fa fa-check"></i>c. Required packages</a></li>
<li class="chapter" data-level="" data-path="WE-12.html"><a href="WE-12.html#d.-import-data-1"><i class="fa fa-check"></i>d. Import data</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="WE-12.html"><a href="WE-12.html#fitting-candidate-models"><i class="fa fa-check"></i>2. Fitting candidate models</a>
<ul>
<li class="chapter" data-level="" data-path="WE-12.html"><a href="WE-12.html#a.-alternative-hypotheses"><i class="fa fa-check"></i>a. Alternative hypotheses</a></li>
<li class="chapter" data-level="" data-path="WE-12.html"><a href="WE-12.html#b.-prepare-for-model-fitting"><i class="fa fa-check"></i>b. Prepare for model fitting</a></li>
<li class="chapter" data-level="" data-path="WE-12.html"><a href="WE-12.html#c.-fit-all-five-models"><i class="fa fa-check"></i>c. Fit all five models</a></li>
<li class="chapter" data-level="" data-path="WE-12.html"><a href="WE-12.html#d.-compare-evidence-for-models"><i class="fa fa-check"></i>d. Compare evidence for models</a></li>
<li class="chapter" data-level="" data-path="WE-12.html"><a href="WE-12.html#e.-calculate-evidence-weights"><i class="fa fa-check"></i>e. Calculate evidence weights</a></li>
<li class="chapter" data-level="" data-path="WE-12.html"><a href="WE-12.html#f.-confidence-intervals-for-predictors"><i class="fa fa-check"></i>f. Confidence intervals for predictors</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="WE-12.html"><a href="WE-12.html#your-turn-to-test-additional-hypotheses"><i class="fa fa-check"></i>3. Your turn to test additional hypotheses!</a></li>
<li class="chapter" data-level="" data-path="WE-12.html"><a href="WE-12.html#references-2"><i class="fa fa-check"></i>4. References</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="" data-path="Week13.html"><a href="Week13.html"><i class="fa fa-check"></i>Week 13: Gravity Models</a>
<ul>
<li class="chapter" data-level="" data-path="WE-13.html"><a href="WE-13.html"><i class="fa fa-check"></i>Worked Example</a>
<ul>
<li class="chapter" data-level="" data-path="WE-13.html"><a href="WE-13.html#overview-of-worked-example-12"><i class="fa fa-check"></i>1. Overview of Worked Example</a>
<ul>
<li class="chapter" data-level="" data-path="WE-13.html"><a href="WE-13.html#a.-background"><i class="fa fa-check"></i>a. Background</a></li>
<li class="chapter" data-level="" data-path="WE-13.html"><a href="WE-13.html#b.-data-set-12"><i class="fa fa-check"></i>b. Data set</a></li>
<li class="chapter" data-level="" data-path="WE-13.html"><a href="WE-13.html#c.-required-r-libraries-9"><i class="fa fa-check"></i>c. Required R libraries</a></li>
<li class="chapter" data-level="" data-path="WE-13.html"><a href="WE-13.html#set-working-and-data-directories---needs-to-be-removed"><i class="fa fa-check"></i>Set working and data directories - NEEDS TO BE REMOVED</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="WE-13.html"><a href="WE-13.html#calculating-graph-metrics"><i class="fa fa-check"></i>2. Calculating graph metrics</a>
<ul>
<li class="chapter" data-level="" data-path="WE-13.html"><a href="WE-13.html#a.-read-in-make-a-spatial-object-and-plot-wetlands-wetlands.csv."><i class="fa fa-check"></i>a. Read in, make a spatial object and plot wetlands (Wetlands.csv).</a></li>
<li class="chapter" data-level="" data-path="WE-13.html"><a href="WE-13.html#b.-create-a-graph-from-the-wetlands"><i class="fa fa-check"></i>b. Create a graph from the wetlands</a></li>
<li class="chapter" data-level="" data-path="WE-13.html"><a href="WE-13.html#c.-create-adjacency-matrix"><i class="fa fa-check"></i>c. Create adjacency matrix</a></li>
<li class="chapter" data-level="" data-path="WE-13.html"><a href="WE-13.html#d.-calculate-graph-metrics"><i class="fa fa-check"></i>d. Calculate graph metrics</a></li>
<li class="chapter" data-level="" data-path="WE-13.html"><a href="WE-13.html#e.-add-graph-metrics-to-data.frame"><i class="fa fa-check"></i>e. Add graph metrics to data.frame</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="WE-13.html"><a href="WE-13.html#combine-frog-data-with-graph-metrics"><i class="fa fa-check"></i>3. Combine frog data with graph metrics</a>
<ul>
<li class="chapter" data-level="" data-path="WE-13.html"><a href="WE-13.html#a.-add-graph-node-data-to-frog-site-data"><i class="fa fa-check"></i>a. Add graph node data to frog site data</a></li>
<li class="chapter" data-level="" data-path="WE-13.html"><a href="WE-13.html#b.-check-data-types"><i class="fa fa-check"></i>b. Check data types</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="WE-13.html"><a href="WE-13.html#merge-graph-with-genetic-distance-data"><i class="fa fa-check"></i>4. Merge graph with genetic distance data</a>
<ul>
<li class="chapter" data-level="" data-path="WE-13.html"><a href="WE-13.html#a.-build-graph-from-occupied-sites"><i class="fa fa-check"></i>a. Build graph from occupied sites</a></li>
<li class="chapter" data-level="" data-path="WE-13.html"><a href="WE-13.html#b.-merge-graph-with-genetic-distance"><i class="fa fa-check"></i>b. Merge graph with genetic distance</a></li>
<li class="chapter" data-level="" data-path="WE-13.html"><a href="WE-13.html#c.-write-out-results-ascii-csv-and-lines-shapefile."><i class="fa fa-check"></i>c. Write out results (ASCII csv and lines shapefile).</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="WE-13.html"><a href="WE-13.html#preparing-raster-based-covariates"><i class="fa fa-check"></i>5. Preparing raster-based covariates</a>
<ul>
<li class="chapter" data-level="" data-path="WE-13.html"><a href="WE-13.html#a.-import-raster-data-as-stack"><i class="fa fa-check"></i>a. Import raster data as stack</a></li>
<li class="chapter" data-level="" data-path="WE-13.html"><a href="WE-13.html#b.-calculate-wetland-area-within-buffer"><i class="fa fa-check"></i>b. Calculate wetland area within buffer</a></li>
<li class="chapter" data-level="" data-path="WE-13.html"><a href="WE-13.html#c.-create-raster-of-wetland-proportion"><i class="fa fa-check"></i>c. Create raster of wetland proportion</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="WE-13.html"><a href="WE-13.html#extract-raster-stats-for-nodes-and-edges"><i class="fa fa-check"></i>6. Extract raster stats for nodes and edges</a>
<ul>
<li class="chapter" data-level="" data-path="WE-13.html"><a href="WE-13.html#a.-extract-at-site-variables-from-rasters"><i class="fa fa-check"></i>a. Extract at-site variables from rasters</a></li>
<li class="chapter" data-level="" data-path="WE-13.html"><a href="WE-13.html#b.-add-covariates-to-graph-edges."><i class="fa fa-check"></i>b. Add covariates to graph edges.</a></li>
<li class="chapter" data-level="" data-path="WE-13.html"><a href="WE-13.html#c.-calculating-stats-along-edges"><i class="fa fa-check"></i>c. Calculating stats along edges</a></li>
<li class="chapter" data-level="" data-path="WE-13.html"><a href="WE-13.html#d.-calculating-stats-for-categorical-variables"><i class="fa fa-check"></i>d. Calculating stats for categorical variables</a></li>
<li class="chapter" data-level="" data-path="WE-13.html"><a href="WE-13.html#e.-add-node-data-to-graph"><i class="fa fa-check"></i>e. Add node data to graph</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="WE-13.html"><a href="WE-13.html#gravity-models"><i class="fa fa-check"></i>7. Gravity Models</a>
<ul>
<li class="chapter" data-level="" data-path="WE-13.html"><a href="WE-13.html#a.-merge-distance-and-site-data"><i class="fa fa-check"></i>a. Merge distance and site data</a></li>
<li class="chapter" data-level="" data-path="WE-13.html"><a href="WE-13.html#b.-define-a-set-of-models"><i class="fa fa-check"></i>b. Define a set of models</a></li>
<li class="chapter" data-level="" data-path="WE-13.html"><a href="WE-13.html#c.-check-for-correlations"><i class="fa fa-check"></i>c. Check for correlations</a></li>
<li class="chapter" data-level="" data-path="WE-13.html"><a href="WE-13.html#c.-write-out-resultss"><i class="fa fa-check"></i>c. Write out resultss</a></li>
<li class="chapter" data-level="" data-path="WE-13.html"><a href="WE-13.html#e.-compare-models"><i class="fa fa-check"></i>e. Compare models</a></li>
<li class="chapter" data-level="" data-path="WE-13.html"><a href="WE-13.html#f.-diagnostic-plots"><i class="fa fa-check"></i>f. Diagnostic plots</a></li>
<li class="chapter" data-level="" data-path="WE-13.html"><a href="WE-13.html#g.-fit-final-models"><i class="fa fa-check"></i>g. Fit final model(s)</a></li>
<li class="chapter" data-level="" data-path="WE-13.html"><a href="WE-13.html#h.-effect-size"><i class="fa fa-check"></i>h. Effect size</a></li>
</ul></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="" data-path="Week14.html"><a href="Week14.html"><i class="fa fa-check"></i>Week 14: Contemporary Gene Flow</a>
<ul>
<li class="chapter" data-level="" data-path="WE-14.html"><a href="WE-14.html"><i class="fa fa-check"></i>Worked Example</a>
<ul>
<li class="chapter" data-level="" data-path="WE-14.html"><a href="WE-14.html#overview-3"><i class="fa fa-check"></i>1. Overview</a>
<ul>
<li class="chapter" data-level="" data-path="WE-14.html"><a href="WE-14.html#a.-background-and-goals"><i class="fa fa-check"></i>a. Background and Goals</a></li>
<li class="chapter" data-level="" data-path="WE-14.html"><a href="WE-14.html#b.-data-set-13"><i class="fa fa-check"></i>b. Data set</a></li>
<li class="chapter" data-level="" data-path="WE-14.html"><a href="WE-14.html#c.-required-packages-1"><i class="fa fa-check"></i>c. Required packages</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="WE-14.html"><a href="WE-14.html#import-genotypes"><i class="fa fa-check"></i>2. Import Genotypes</a>
<ul>
<li class="chapter" data-level="" data-path="WE-14.html"><a href="WE-14.html#a.-import-genotype-data"><i class="fa fa-check"></i>a. Import genotype data</a></li>
<li class="chapter" data-level="" data-path="WE-14.html"><a href="WE-14.html#b.-plot-map-of-sampling-locations"><i class="fa fa-check"></i>b. Plot map of sampling locations</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="WE-14.html"><a href="WE-14.html#pollen-pool-genetic-structure"><i class="fa fa-check"></i>3. Pollen pool genetic structure</a>
<ul>
<li class="chapter" data-level="" data-path="WE-14.html"><a href="WE-14.html#a.-subtract-maternal-contribution-to-offspring-genotype"><i class="fa fa-check"></i>a. Subtract maternal contribution to offspring genotype</a></li>
<li class="chapter" data-level="" data-path="WE-14.html"><a href="WE-14.html#b.-twogener-analysis-amova-of-pollen-pool-data"><i class="fa fa-check"></i>b. ‘TwoGener’ analysis (AMOVA of pollen pool data)</a></li>
<li class="chapter" data-level="" data-path="WE-14.html"><a href="WE-14.html#c.-including-space-isolation-by-distance"><i class="fa fa-check"></i>c. Including space: Isolation by distance</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="WE-14.html"><a href="WE-14.html#paternity-analysis"><i class="fa fa-check"></i>4. Paternity analysis</a>
<ul>
<li class="chapter" data-level="" data-path="WE-14.html"><a href="WE-14.html#a.-exclusion-probabilities"><i class="fa fa-check"></i>a. Exclusion probabilities</a></li>
<li class="chapter" data-level="" data-path="WE-14.html"><a href="WE-14.html#b.-paternity-analysis"><i class="fa fa-check"></i>b. Paternity analysis</a></li>
<li class="chapter" data-level="" data-path="WE-14.html"><a href="WE-14.html#c.-visualize-reconstructed-pollen-flow-events-in-space"><i class="fa fa-check"></i>c. Visualize reconstructed pollen flow events in space</a></li>
<li class="chapter" data-level="" data-path="WE-14.html"><a href="WE-14.html#d.-visualize-dispersal-kernel"><i class="fa fa-check"></i>d. Visualize dispersal kernel</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="WE-14.html"><a href="WE-14.html#linking-paternity-to-ecological-variables"><i class="fa fa-check"></i>5. Linking paternity to ecological variables</a>
<ul>
<li class="chapter" data-level="" data-path="WE-14.html"><a href="WE-14.html#a.-explain-pollen-flow-within-populations"><i class="fa fa-check"></i>a. Explain pollen flow within populations</a></li>
<li class="chapter" data-level="" data-path="WE-14.html"><a href="WE-14.html#b.-explain-pollen-flow-at-the-population-level"><i class="fa fa-check"></i>b. Explain pollen flow at the population level</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="WE-14.html"><a href="WE-14.html#references-3"><i class="fa fa-check"></i>6. References</a></li>
</ul></li>
</ul></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Landscape Genetic Data Analysis with R</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="WE_7" class="section level2">
<h2>Worked Example</h2>
<div id="overview-of-worked-example-6" class="section level3">
<h3>1. Overview of Worked Example</h3>
<div id="a.-goals-7" class="section level4">
<h4>a. Goals</h4>
<p>This worked example shows:</p>
<ul>
<li>How to test regression residuals for spatial autocorrelation.</li>
<li>How to fit a model with spatially autocorrelated errors (GLS).</li>
<li>How to fit a spatial simultaneous autoregressive error model (SAR).</li>
<li>How to perform spatial filtering with Moran eigenvector maps (MEM).</li>
<li>How to fit a spatially varying coefficients model (SVC).</li>
</ul>
</div>
<div id="b.-data-set-6" class="section level4">
<h4>b. Data set</h4>
<p>Here we analyze population-level data of the wildflower <em>Dianthus carthusianorum</em> (common name: Carthusian pink) in 65 calcareous grassland patches in the Franconian Jura, Germany (Rico et al. 2013):</p>
<ul>
<li><strong>Dianthus</strong>: Spatial point data frame with population-level data (patch characteristics, grazing regime, genetic diversity, 15 alternative connectivity indices Si), included in package ‘LandGenCourse.’ To load the data, type (without quotes): ‘data(Dianthus).’ For a definition of the variables, type: ‘?Dianthus.’</li>
</ul>
</div>
<div id="c.-required-r-packages-2" class="section level4">
<h4>c. Required R packages</h4>
<p>Note: the function ‘library’ will always load the package, even if it is already loaded, whereas ‘require’ will only load it if it is not yet loaded. Either will work.</p>
<div class="sourceCode" id="cb731"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb731-1"><a href="WE-7.html#cb731-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(LandGenCourse)</span>
<span id="cb731-2"><a href="WE-7.html#cb731-2" aria-hidden="true" tabindex="-1"></a><span class="co">#library(here)</span></span>
<span id="cb731-3"><a href="WE-7.html#cb731-3" aria-hidden="true" tabindex="-1"></a><span class="co">#libraryspdep)</span></span>
<span id="cb731-4"><a href="WE-7.html#cb731-4" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(nlme)</span>
<span id="cb731-5"><a href="WE-7.html#cb731-5" aria-hidden="true" tabindex="-1"></a><span class="co">#library(lattice)</span></span>
<span id="cb731-6"><a href="WE-7.html#cb731-6" aria-hidden="true" tabindex="-1"></a><span class="co">#library(MuMIn)</span></span>
<span id="cb731-7"><a href="WE-7.html#cb731-7" aria-hidden="true" tabindex="-1"></a><span class="co">#library(gridExtra)</span></span>
<span id="cb731-8"><a href="WE-7.html#cb731-8" aria-hidden="true" tabindex="-1"></a><span class="co">#library(dplyr)</span></span>
<span id="cb731-9"><a href="WE-7.html#cb731-9" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(spatialreg)</span>
<span id="cb731-10"><a href="WE-7.html#cb731-10" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggplot2)</span>
<span id="cb731-11"><a href="WE-7.html#cb731-11" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggmap)</span>
<span id="cb731-12"><a href="WE-7.html#cb731-12" aria-hidden="true" tabindex="-1"></a><span class="fu">source</span>(<span class="fu">system.file</span>(<span class="st">&quot;extdata&quot;</span>, <span class="st">&quot;panel.cor.r&quot;</span>, </span>
<span id="cb731-13"><a href="WE-7.html#cb731-13" aria-hidden="true" tabindex="-1"></a>                            <span class="at">package =</span> <span class="st">&quot;LandGenCourse&quot;</span>))</span></code></pre></div>
<p>Package ‘spmoran’ not automatically installed with ‘LandGenCourse’:</p>
<div class="sourceCode" id="cb732"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb732-1"><a href="WE-7.html#cb732-1" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span>(<span class="sc">!</span><span class="fu">require</span>(spmoran)) <span class="fu">install.packages</span>(<span class="st">&quot;spmoran&quot;</span>, <span class="at">repos=</span><span class="st">&#39;http://cran.us.r-project.org&#39;</span>)</span>
<span id="cb732-2"><a href="WE-7.html#cb732-2" aria-hidden="true" tabindex="-1"></a><span class="co">#require(spmoran)</span></span></code></pre></div>
</div>
</div>
<div id="explore-data-set" class="section level3">
<h3>2. Explore data set</h3>
<p>We will model allelic richness ‘A’ as a function of the following predictors:</p>
<ul>
<li><strong>IBD</strong>: connectivity index Si (‘Eu_pj’) based on Euclidean distance between source and focal patch. This represents a hypothesis of isolation by distance (IBD).</li>
<li><strong>IBR</strong>: connectivity index Si (‘Sheint_pj’) based on the number of continuously or intermittently grazed patches between source and focal patch. This represents a hypothesis of isolation by resistance (IBR). Specifically, this model assumes connectivity via sheep-mediated seed dispersal, where seeds are likely to be transported from patch to patch within the same grazing system (shepherding route). Seeds are assumed to disperse most likely to the next patch (in either direction) along the grazing route, and less likely to more remote patches along the route.</li>
<li><strong>PatchSize</strong>: Logarithm of calcareous grassland patch size in ha.</li>
</ul>
<p>The connectivity indices Si were calculated for each focal patch i, integrating over all other patches j where the species was present (potential source patches) using Hanski’s incidence function.</p>
<div id="a.-import-data" class="section level4">
<h4>a. Import data</h4>
<div class="sourceCode" id="cb733"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb733-1"><a href="WE-7.html#cb733-1" aria-hidden="true" tabindex="-1"></a><span class="fu">data</span>(Dianthus)</span></code></pre></div>
<p>Allelic richness ‘A’ was not calculate for populations with &lt; 5 individuals. Here we extract only the patches with ‘A’ values, and the variables needed, and store them in a data frame ‘Dianthus.df.’</p>
</div>
<div id="b.-create-a-map" class="section level4">
<h4>b. Create a map</h4>
<p>The functions in package ‘ggmap’ expect the projection to be Longitude/Latitude. We could start by transforming the projection of the Dianthus dataset:</p>
<div class="sourceCode" id="cb734"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb734-1"><a href="WE-7.html#cb734-1" aria-hidden="true" tabindex="-1"></a>Dianthus.longlat <span class="ot">&lt;-</span> sp<span class="sc">::</span><span class="fu">spTransform</span>(Dianthus, </span>
<span id="cb734-2"><a href="WE-7.html#cb734-2" aria-hidden="true" tabindex="-1"></a>                       <span class="at">CRSobj =</span> tmaptools<span class="sc">::</span><span class="fu">get_proj4</span>(<span class="st">&quot;longlat&quot;</span>)<span class="sc">$</span>proj4string)</span></code></pre></div>
<pre><code>## Warning in spTransform(xSP, CRSobj, ...): NULL source CRS comment, falling back
## to PROJ string</code></pre>
<pre><code>## Warning in spTransform(xSP, CRSobj, ...): +init dropped in PROJ string</code></pre>
<pre><code>## Warning in wkt(obj): CRS object has no comment

## Warning in wkt(obj): CRS object has no comment</code></pre>
<pre><code>## Warning in showSRID(uprojargs, format = &quot;PROJ&quot;, multiline = &quot;NO&quot;, prefer_proj = prefer_proj): Discarded datum Deutsches_Hauptdreiecksnetz in Proj4 definition,
##  but +towgs84= values preserved</code></pre>
<div class="sourceCode" id="cb739"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb739-1"><a href="WE-7.html#cb739-1" aria-hidden="true" tabindex="-1"></a><span class="fu">make_bbox</span>(<span class="at">lon =</span> Longitude, <span class="at">lat =</span> Latitude, <span class="at">data =</span> Dianthus<span class="sc">@</span>data, <span class="at">f =</span> <span class="fl">0.2</span>)</span></code></pre></div>
<pre><code>##     left   bottom    right      top 
## 10.91051 48.90282 11.08951 49.07073</code></pre>
<p>Here, however, we use a shortcut as the ‘<a href="mailto:Dianthus@data" class="email">Dianthus@data</a>’ dataframe already contains variables with Longitude and Latitude. We use these to define the bounding box for the map, extending the map extent by 20% (‘f=0.2’).</p>
<div class="sourceCode" id="cb741"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb741-1"><a href="WE-7.html#cb741-1" aria-hidden="true" tabindex="-1"></a>bbox <span class="ot">&lt;-</span> <span class="fu">make_bbox</span>(<span class="at">lon =</span> Longitude, <span class="at">lat =</span> Latitude, <span class="at">data =</span> Dianthus<span class="sc">@</span>data, <span class="at">f =</span> <span class="fl">0.2</span>)</span></code></pre></div>
<p>Now we can obtain a stamen map for the study area from the internet and add points for the sampling sites. Here we use a terrain map.</p>
<p>Note: here we use two types of maps from the source ‘stamen,’ which is freely available: a <code>terrain</code> map and a <code>toner</code> map. Starting in mid 2018, accessing maps from Google with ‘source = google’ requires an API key.</p>
<p>To plot the maps side-by-side, we first write each map into an object, then use the function <code>gridExtra::grid.arrange</code> to plot them on the same row. This works for a wide range of figures, including <code>ggplot2</code> or <code>trellis</code>-type plots.</p>
<div class="sourceCode" id="cb742"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb742-1"><a href="WE-7.html#cb742-1" aria-hidden="true" tabindex="-1"></a>StamenMap.terrain <span class="ot">&lt;-</span> <span class="fu">ggmap</span>(<span class="fu">get_stamenmap</span>(bbox, <span class="at">maptype =</span> <span class="st">&quot;terrain&quot;</span>, <span class="at">zoom=</span><span class="dv">12</span>)) <span class="sc">+</span> </span>
<span id="cb742-2"><a href="WE-7.html#cb742-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="fu">aes</span>(<span class="at">x =</span>  Longitude, <span class="at">y =</span> Latitude), <span class="at">data =</span> Dianthus.longlat<span class="sc">@</span>data, </span>
<span id="cb742-3"><a href="WE-7.html#cb742-3" aria-hidden="true" tabindex="-1"></a>             <span class="at">colour =</span> <span class="st">&quot;black&quot;</span>, <span class="at">size =</span> <span class="dv">2</span>)</span>
<span id="cb742-4"><a href="WE-7.html#cb742-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb742-5"><a href="WE-7.html#cb742-5" aria-hidden="true" tabindex="-1"></a>StamenMap.toner <span class="ot">&lt;-</span> ggmap<span class="sc">::</span><span class="fu">ggmap</span>(<span class="fu">get_stamenmap</span>(bbox, <span class="at">maptype =</span> <span class="st">&quot;toner&quot;</span>, <span class="at">zoom=</span><span class="dv">12</span>, <span class="at">force =</span> <span class="cn">TRUE</span>)) <span class="sc">+</span> </span>
<span id="cb742-6"><a href="WE-7.html#cb742-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="fu">aes</span>(<span class="at">x =</span>  Longitude, <span class="at">y =</span> Latitude), <span class="at">data =</span> Dianthus.longlat<span class="sc">@</span>data, </span>
<span id="cb742-7"><a href="WE-7.html#cb742-7" aria-hidden="true" tabindex="-1"></a>             <span class="at">colour =</span> <span class="st">&quot;black&quot;</span>, <span class="at">size =</span> <span class="dv">2</span>) </span>
<span id="cb742-8"><a href="WE-7.html#cb742-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb742-9"><a href="WE-7.html#cb742-9" aria-hidden="true" tabindex="-1"></a>gridExtra<span class="sc">::</span><span class="fu">grid.arrange</span>(StamenMap.terrain, StamenMap.toner, <span class="at">nrow=</span><span class="dv">1</span>)</span></code></pre></div>
<p><img src="07-Week07_files/figure-html/unnamed-chunk-9-1.png" width="672" /></p>
<p>As you can see from the terrain map, most sites lie on the steep slopes between an upper and a lower Jurassic plateau. A few sites lie at the forest edge on the upper plateau, typically in areas where the soil is too shallow to allow crop farming. With in the study area, all known sites were sampled. Additional sites are expected to be found mainly in the valley system in the Southwest.</p>
<p>There are three types of stamen maps: terrain, toner (black and white), and watercolor. Terrain and toner maps have three components each (background, labels, lines), which can be downloaded individually if needed. Hence, the argument ‘maptype’ has the following options:</p>
<ul>
<li>terrain</li>
<li>terrain-background</li>
<li>terrain-labels</li>
<li>terrain-lines</li>
<li>toner</li>
<li>toner-2010</li>
<li>toner-background</li>
<li>toner-hybrid</li>
<li>toner-labels</li>
<li>toner-lines</li>
<li>toner-lite</li>
<li>watercolor</li>
</ul>
</div>
<div id="c.-explore-correlations" class="section level4">
<h4>c. Explore correlations</h4>
<p>When fitting linear models, it is always a good idea to look at the correlations first.</p>
<div class="sourceCode" id="cb743"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb743-1"><a href="WE-7.html#cb743-1" aria-hidden="true" tabindex="-1"></a>Dianthus.df <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">A=</span>Dianthus<span class="sc">@</span>data<span class="sc">$</span>A, <span class="at">IBD=</span>Dianthus<span class="sc">@</span>data<span class="sc">$</span>Eu_pj, </span>
<span id="cb743-2"><a href="WE-7.html#cb743-2" aria-hidden="true" tabindex="-1"></a>                          <span class="at">IBR=</span>Dianthus<span class="sc">@</span>data<span class="sc">$</span>Sheint_pj,</span>
<span id="cb743-3"><a href="WE-7.html#cb743-3" aria-hidden="true" tabindex="-1"></a>                          <span class="at">PatchSize=</span><span class="fu">log</span>(Dianthus<span class="sc">@</span>data<span class="sc">$</span>Ha),</span>
<span id="cb743-4"><a href="WE-7.html#cb743-4" aria-hidden="true" tabindex="-1"></a>                          <span class="at">System=</span>Dianthus<span class="sc">@</span>data<span class="sc">$</span>System,</span>
<span id="cb743-5"><a href="WE-7.html#cb743-5" aria-hidden="true" tabindex="-1"></a>                          <span class="at">Longitude=</span>Dianthus<span class="sc">@</span>data<span class="sc">$</span>Longitude,</span>
<span id="cb743-6"><a href="WE-7.html#cb743-6" aria-hidden="true" tabindex="-1"></a>                          <span class="at">Latitude=</span>Dianthus<span class="sc">@</span>data<span class="sc">$</span>Latitude,</span>
<span id="cb743-7"><a href="WE-7.html#cb743-7" aria-hidden="true" tabindex="-1"></a>                          <span class="at">x=</span>Dianthus<span class="sc">@</span>coords[,<span class="dv">1</span>], <span class="at">y=</span>Dianthus<span class="sc">@</span>coords[,<span class="dv">2</span>])</span>
<span id="cb743-8"><a href="WE-7.html#cb743-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb743-9"><a href="WE-7.html#cb743-9" aria-hidden="true" tabindex="-1"></a><span class="co"># Define &#39;System&#39; for ungrazed patches</span></span>
<span id="cb743-10"><a href="WE-7.html#cb743-10" aria-hidden="true" tabindex="-1"></a>Dianthus.df<span class="sc">$</span>System<span class="ot">=</span><span class="fu">as.character</span>(Dianthus<span class="sc">@</span>data<span class="sc">$</span>System)</span>
<span id="cb743-11"><a href="WE-7.html#cb743-11" aria-hidden="true" tabindex="-1"></a>Dianthus.df<span class="sc">$</span>System[<span class="fu">is.na</span>(Dianthus.df<span class="sc">$</span>System)] <span class="ot">&lt;-</span> <span class="st">&quot;Ungrazed&quot;</span></span>
<span id="cb743-12"><a href="WE-7.html#cb743-12" aria-hidden="true" tabindex="-1"></a>Dianthus.df<span class="sc">$</span>System <span class="ot">&lt;-</span> <span class="fu">factor</span>(Dianthus.df<span class="sc">$</span>System, </span>
<span id="cb743-13"><a href="WE-7.html#cb743-13" aria-hidden="true" tabindex="-1"></a>                             <span class="at">levels=</span><span class="fu">c</span>(<span class="st">&quot;Ungrazed&quot;</span>, <span class="st">&quot;East&quot;</span>, <span class="st">&quot;South&quot;</span>, <span class="st">&quot;West&quot;</span>))</span>
<span id="cb743-14"><a href="WE-7.html#cb743-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb743-15"><a href="WE-7.html#cb743-15" aria-hidden="true" tabindex="-1"></a><span class="co"># Remove patches with missing values for A</span></span>
<span id="cb743-16"><a href="WE-7.html#cb743-16" aria-hidden="true" tabindex="-1"></a>Dianthus.df <span class="ot">&lt;-</span> Dianthus.df[<span class="sc">!</span><span class="fu">is.na</span>(Dianthus.df<span class="sc">$</span>A),]</span>
<span id="cb743-17"><a href="WE-7.html#cb743-17" aria-hidden="true" tabindex="-1"></a><span class="fu">dim</span>(Dianthus.df)</span></code></pre></div>
<pre><code>## [1] 59  9</code></pre>
<div class="sourceCode" id="cb745"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb745-1"><a href="WE-7.html#cb745-1" aria-hidden="true" tabindex="-1"></a><span class="fu">pairs</span>(Dianthus.df[,<span class="sc">-</span><span class="fu">c</span>(<span class="dv">5</span><span class="sc">:</span><span class="dv">7</span>)], <span class="at">lower.panel=</span>panel.smooth, </span>
<span id="cb745-2"><a href="WE-7.html#cb745-2" aria-hidden="true" tabindex="-1"></a>      <span class="at">upper.panel=</span>panel.cor, <span class="at">diag.panel=</span>panel.hist)</span></code></pre></div>
<p><img src="07-Week07_files/figure-html/unnamed-chunk-10-1.png" width="672" /></p>
<p><strong>Questions:</strong></p>
<ul>
<li>How strong is the association between ‘Eu_pj’ and ‘A?’ What does this suggest about the hypothesis of IBD?</li>
<li>How strong is the association between ‘Sheint_pj’ and ‘A?’ What does this suggest about the hypothesis of sheep-mediated gene flow (IBR)?</li>
<li>Which variable seems to be a better predictor of allelic richness: patch size ‘Ha’ or the logarithm of patch size, ‘PatchSize?’</li>
<li>Is logHa correlated with ‘IBD’ or ‘IBR?’</li>
<li>Are any of the variables correlated with the spatial coordinates x and y?</li>
</ul>
<p>Do the three grazing systems, and the ungrazed patches, differ in allelic richness A? Also, let’s check the association between patch size and population size. Here we create boxplots that show the individual values as dots. We add a horizontal jitter to avoid overlapping points.</p>
<div class="sourceCode" id="cb746"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb746-1"><a href="WE-7.html#cb746-1" aria-hidden="true" tabindex="-1"></a>Boxplot1 <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(Dianthus.df, <span class="fu">aes</span>(<span class="at">x=</span>System, <span class="at">y=</span>A)) <span class="sc">+</span> </span>
<span id="cb746-2"><a href="WE-7.html#cb746-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_boxplot</span>() <span class="sc">+</span> <span class="fu">xlab</span>(<span class="st">&quot;Grazing system&quot;</span>) <span class="sc">+</span> <span class="fu">ylab</span>(<span class="st">&quot;Allelic richness (A)&quot;</span>) <span class="sc">+</span></span>
<span id="cb746-3"><a href="WE-7.html#cb746-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_jitter</span>(<span class="at">shape=</span><span class="dv">1</span>, <span class="at">position=</span><span class="fu">position_jitter</span>(<span class="fl">0.1</span>), <span class="at">col=</span><span class="st">&quot;blue&quot;</span>)</span>
<span id="cb746-4"><a href="WE-7.html#cb746-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb746-5"><a href="WE-7.html#cb746-5" aria-hidden="true" tabindex="-1"></a>Boxplot2 <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(Dianthus<span class="sc">@</span>data, <span class="fu">aes</span>(<span class="at">x=</span><span class="fu">factor</span>(pop09), <span class="at">y=</span><span class="fu">log</span>(Ha))) <span class="sc">+</span> </span>
<span id="cb746-6"><a href="WE-7.html#cb746-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_boxplot</span>() <span class="sc">+</span> <span class="fu">xlab</span>(<span class="st">&quot;Population size class&quot;</span>) <span class="sc">+</span> <span class="fu">ylab</span>(<span class="st">&quot;PatchSize (log(Ha))&quot;</span>) <span class="sc">+</span></span>
<span id="cb746-7"><a href="WE-7.html#cb746-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_jitter</span>(<span class="at">shape=</span><span class="dv">1</span>, <span class="at">position=</span><span class="fu">position_jitter</span>(<span class="fl">0.1</span>), <span class="at">col=</span><span class="st">&quot;blue&quot;</span>)</span>
<span id="cb746-8"><a href="WE-7.html#cb746-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb746-9"><a href="WE-7.html#cb746-9" aria-hidden="true" tabindex="-1"></a>gridExtra<span class="sc">::</span><span class="fu">grid.arrange</span>(Boxplot1, Boxplot2, <span class="at">nrow=</span><span class="dv">1</span>)</span></code></pre></div>
<p><img src="07-Week07_files/figure-html/unnamed-chunk-11-1.png" width="672" /></p>
<p>Even though the population size categories were very broad, there appears to be a strong relationship between populations size (category) and (the logarithm of) patch size.</p>
<p>Despite this relationship, connectivity models Si that only considered <em>Dianthus carthusianorum</em> presence/absence (‘pj’) in source patches ‘j’ were better supported than those Si models that took into account source patch area (‘Aj’) or population size (‘Nj’).</p>
<p>We can check this by calculating the correlation of allelelic richness ‘A’ with each of the 15 connectivity models ‘Si’ in the data set.</p>
<div class="sourceCode" id="cb747"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb747-1"><a href="WE-7.html#cb747-1" aria-hidden="true" tabindex="-1"></a><span class="fu">round</span>(<span class="fu">matrix</span>(<span class="fu">cor</span>(Dianthus<span class="sc">@</span>data<span class="sc">$</span>A, Dianthus<span class="sc">@</span>data[,<span class="dv">15</span><span class="sc">:</span><span class="dv">29</span>], </span>
<span id="cb747-2"><a href="WE-7.html#cb747-2" aria-hidden="true" tabindex="-1"></a>                 <span class="at">use=</span><span class="st">&quot;pairwise.complete.obs&quot;</span>), <span class="dv">5</span>, <span class="dv">3</span>, <span class="at">byrow=</span><span class="cn">TRUE</span>, </span>
<span id="cb747-3"><a href="WE-7.html#cb747-3" aria-hidden="true" tabindex="-1"></a>           <span class="at">dimnames=</span><span class="fu">list</span>(<span class="fu">c</span>(<span class="st">&quot;Eu&quot;</span>, <span class="st">&quot;Shecte&quot;</span>, <span class="st">&quot;Sheint&quot;</span>, <span class="st">&quot;Shenu&quot;</span>, <span class="st">&quot;Forest&quot;</span>), </span>
<span id="cb747-4"><a href="WE-7.html#cb747-4" aria-hidden="true" tabindex="-1"></a>                         <span class="fu">c</span>(<span class="st">&quot;pj&quot;</span>, <span class="st">&quot;Aj&quot;</span>, <span class="st">&quot;Nj&quot;</span>))),<span class="dv">3</span>)</span></code></pre></div>
<pre><code>##           pj     Aj    Nj
## Eu     0.024  0.207 0.098
## Shecte 0.375  0.273 0.369
## Sheint 0.403  0.196 0.369
## Shenu  0.135 -0.045 0.105
## Forest 0.137  0.168 0.138</code></pre>
<ul>
<li>Correlations with ‘A’ are highest for the two ‘IBR’ models that assume seed dispersal over a limited number of patches along shepherding routes (‘Shecte’ and ‘Sheint.’ These two models include only continuously grazed, or both continuously and intermittently grazed patches, respectively.</li>
<li>Correlations for models that take in to account population size (‘Nj’) are only slightly lower, whereas those that use patch size (‘Aj’) as a proxy for the size of the seed emigrant pool had lower correlations.</li>
</ul>
</div>
</div>
<div id="test-regression-residuals-for-spatial-autocorrelation" class="section level3">
<h3>3. Test regression residuals for spatial autocorrelation</h3>
<div id="a.-fit-regression-models" class="section level4">
<h4>a. Fit regression models</h4>
<p>Here we fit three multiple regression models to explain variation in allelic richness:</p>
<ul>
<li><strong>mod.lm.IBD</strong>: IBD model of connectivity ‘Eu_pj.’</li>
<li><strong>mod.lm.IBR</strong>: IBR model shepherding connectivity ‘Sheint_pj.’</li>
<li><strong>mod.lm.PatchSize</strong>: log patch size and IBR model.</li>
</ul>
<div class="sourceCode" id="cb749"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb749-1"><a href="WE-7.html#cb749-1" aria-hidden="true" tabindex="-1"></a>mod.lm.IBD <span class="ot">&lt;-</span> <span class="fu">lm</span>(A <span class="sc">~</span> IBD, <span class="at">data =</span> Dianthus.df)</span>
<span id="cb749-2"><a href="WE-7.html#cb749-2" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>(mod.lm.IBD)</span></code></pre></div>
<pre><code>## 
## Call:
## lm(formula = A ~ IBD, data = Dianthus.df)
## 
## Residuals:
##      Min       1Q   Median       3Q      Max 
## -0.68545 -0.10220  0.03883  0.16178  0.36100 
## 
## Coefficients:
##             Estimate Std. Error t value Pr(&gt;|t|)    
## (Intercept) 4.070960   0.064061  63.549   &lt;2e-16 ***
## IBD         0.008454   0.047460   0.178    0.859    
## ---
## Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1
## 
## Residual standard error: 0.2324 on 57 degrees of freedom
## Multiple R-squared:  0.0005563,  Adjusted R-squared:  -0.01698 
## F-statistic: 0.03173 on 1 and 57 DF,  p-value: 0.8593</code></pre>
<p>This model does not fit the data at all!</p>
<div class="sourceCode" id="cb751"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb751-1"><a href="WE-7.html#cb751-1" aria-hidden="true" tabindex="-1"></a>mod.lm.IBR <span class="ot">&lt;-</span> <span class="fu">lm</span>(A <span class="sc">~</span> IBR, <span class="at">data =</span> Dianthus.df)</span>
<span id="cb751-2"><a href="WE-7.html#cb751-2" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>(mod.lm.IBR)</span></code></pre></div>
<pre><code>## 
## Call:
## lm(formula = A ~ IBR, data = Dianthus.df)
## 
## Residuals:
##      Min       1Q   Median       3Q      Max 
## -0.66844 -0.11251  0.03418  0.12219  0.41760 
## 
## Coefficients:
##             Estimate Std. Error t value Pr(&gt;|t|)    
## (Intercept)  3.92306    0.05499  71.348  &lt; 2e-16 ***
## IBR          0.25515    0.07672   3.326  0.00155 ** 
## ---
## Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1
## 
## Residual standard error: 0.2128 on 57 degrees of freedom
## Multiple R-squared:  0.1625, Adjusted R-squared:  0.1478 
## F-statistic: 11.06 on 1 and 57 DF,  p-value: 0.001547</code></pre>
<p>This model fits much better. Let’s check the residuals plots.</p>
<div class="sourceCode" id="cb753"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb753-1"><a href="WE-7.html#cb753-1" aria-hidden="true" tabindex="-1"></a><span class="fu">par</span>(<span class="at">mfrow=</span><span class="fu">c</span>(<span class="dv">2</span>,<span class="dv">2</span>), <span class="at">mar=</span><span class="fu">c</span>(<span class="dv">4</span>,<span class="dv">4</span>,<span class="dv">2</span>,<span class="dv">1</span>))</span>
<span id="cb753-2"><a href="WE-7.html#cb753-2" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(mod.lm.IBR)</span></code></pre></div>
<p><img src="07-Week07_files/figure-html/unnamed-chunk-15-1.png" width="672" /></p>
<div class="sourceCode" id="cb754"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb754-1"><a href="WE-7.html#cb754-1" aria-hidden="true" tabindex="-1"></a><span class="fu">par</span>(<span class="at">mfrow=</span><span class="fu">c</span>(<span class="dv">1</span>,<span class="dv">1</span>))</span></code></pre></div>
<p>The residuals show some deviation from a normal distribution. Specifically, the lowest values are lower than expected.</p>
<div class="sourceCode" id="cb755"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb755-1"><a href="WE-7.html#cb755-1" aria-hidden="true" tabindex="-1"></a>mod.lm.PatchSize <span class="ot">&lt;-</span> <span class="fu">lm</span>(A <span class="sc">~</span> PatchSize <span class="sc">+</span> IBR, <span class="at">data =</span> Dianthus.df)</span>
<span id="cb755-2"><a href="WE-7.html#cb755-2" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>(mod.lm.PatchSize)</span></code></pre></div>
<pre><code>## 
## Call:
## lm(formula = A ~ PatchSize + IBR, data = Dianthus.df)
## 
## Residuals:
##      Min       1Q   Median       3Q      Max 
## -0.68359 -0.08844  0.02538  0.11705  0.41705 
## 
## Coefficients:
##             Estimate Std. Error t value Pr(&gt;|t|)    
## (Intercept)  4.05158    0.07781  52.069   &lt;2e-16 ***
## PatchSize    0.04266    0.01888   2.260   0.0277 *  
## IBR          0.11338    0.09709   1.168   0.2479    
## ---
## Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1
## 
## Residual standard error: 0.2055 on 56 degrees of freedom
## Multiple R-squared:  0.2325, Adjusted R-squared:  0.2051 
## F-statistic: 8.482 on 2 and 56 DF,  p-value: 0.0006058</code></pre>
<p>This combinde model explains more variation in allelic richness than the IBR model alone. Moreover, after adding PatchSizes, the IBR term is no longer statistically significant!</p>
<p>Has the distribution of residuals improved as well?</p>
<div class="sourceCode" id="cb757"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb757-1"><a href="WE-7.html#cb757-1" aria-hidden="true" tabindex="-1"></a><span class="fu">par</span>(<span class="at">mfrow=</span><span class="fu">c</span>(<span class="dv">2</span>,<span class="dv">2</span>), <span class="at">mar=</span><span class="fu">c</span>(<span class="dv">4</span>,<span class="dv">4</span>,<span class="dv">2</span>,<span class="dv">1</span>))</span>
<span id="cb757-2"><a href="WE-7.html#cb757-2" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(mod.lm.PatchSize)</span></code></pre></div>
<p><img src="07-Week07_files/figure-html/unnamed-chunk-17-1.png" width="672" /></p>
<div class="sourceCode" id="cb758"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb758-1"><a href="WE-7.html#cb758-1" aria-hidden="true" tabindex="-1"></a><span class="fu">par</span>(<span class="at">mfrow=</span><span class="fu">c</span>(<span class="dv">1</span>,<span class="dv">1</span>))</span></code></pre></div>
<p>Not really!</p>
</div>
<div id="b.-test-for-spatial-autocorrelation-morans-i" class="section level4">
<h4>b. Test for spatial autocorrelation (Moran’s I):</h4>
<p>Before we interpret the models, let’s check whether the assumption of independent residuals is violated by spatial autocorrelation in the residuals.</p>
<p>To calculate and test Moran’s I, we first need to define neighbours and spatial weights. Here we use a Gabriel graph to define neighbours.</p>
<p>We define weights in three ways (see Week 5 video and tutorial for code explanation):</p>
<ul>
<li><strong>listw.gab</strong>: 1 = neighbour, 0 = not a neighbour.</li>
<li><strong>listw.d1</strong>: inverse distance weights: neighbour j with weight 1/dij</li>
<li><strong>listw.d2</strong>: inverse squared distance weights: neighbour j with weight 1/dij^2</li>
</ul>
<p>In each case, we row-standardize the weights with the option ‘style = “W”.’</p>
<p>Note: when using ‘graph2nb,’ make sure to use the argument ‘sym=TRUE.’ This means that if A is a neighbour of B, B is also a neighbour of A. The default is ‘sym=FALSE,’ which may result in some sites not having any neighbours assigned (though this would not be evident from the figure!).</p>
<div class="sourceCode" id="cb759"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb759-1"><a href="WE-7.html#cb759-1" aria-hidden="true" tabindex="-1"></a>xy <span class="ot">&lt;-</span> <span class="fu">data.matrix</span>(Dianthus.df[,<span class="fu">c</span>(<span class="st">&quot;x&quot;</span>, <span class="st">&quot;y&quot;</span>)])</span>
<span id="cb759-2"><a href="WE-7.html#cb759-2" aria-hidden="true" tabindex="-1"></a>nb.gab <span class="ot">&lt;-</span> spdep<span class="sc">::</span><span class="fu">graph2nb</span>(spdep<span class="sc">::</span><span class="fu">gabrielneigh</span>(xy), <span class="at">sym=</span><span class="cn">TRUE</span>)</span>
<span id="cb759-3"><a href="WE-7.html#cb759-3" aria-hidden="true" tabindex="-1"></a><span class="fu">par</span>(<span class="at">mar=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>,<span class="dv">0</span>,<span class="dv">0</span>))</span>
<span id="cb759-4"><a href="WE-7.html#cb759-4" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(nb.gab, xy)</span></code></pre></div>
<p><img src="07-Week07_files/figure-html/unnamed-chunk-18-1.png" width="672" /></p>
<div class="sourceCode" id="cb760"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb760-1"><a href="WE-7.html#cb760-1" aria-hidden="true" tabindex="-1"></a>listw.gab <span class="ot">&lt;-</span> spdep<span class="sc">::</span><span class="fu">nb2listw</span>(nb.gab)</span>
<span id="cb760-2"><a href="WE-7.html#cb760-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb760-3"><a href="WE-7.html#cb760-3" aria-hidden="true" tabindex="-1"></a>dlist <span class="ot">&lt;-</span> spdep<span class="sc">::</span><span class="fu">nbdists</span>(nb.gab, xy)</span>
<span id="cb760-4"><a href="WE-7.html#cb760-4" aria-hidden="true" tabindex="-1"></a>dlist <span class="ot">&lt;-</span> <span class="fu">lapply</span>(dlist, <span class="cf">function</span>(x) <span class="dv">1</span><span class="sc">/</span>x)</span>
<span id="cb760-5"><a href="WE-7.html#cb760-5" aria-hidden="true" tabindex="-1"></a>listw.d1 <span class="ot">&lt;-</span> spdep<span class="sc">::</span><span class="fu">nb2listw</span>(nb.gab, <span class="at">style =</span> <span class="st">&quot;W&quot;</span>, <span class="at">glist=</span>dlist)</span>
<span id="cb760-6"><a href="WE-7.html#cb760-6" aria-hidden="true" tabindex="-1"></a>dlist <span class="ot">&lt;-</span> <span class="fu">lapply</span>(dlist, <span class="cf">function</span>(x) <span class="dv">1</span><span class="sc">/</span>x<span class="sc">^</span><span class="dv">2</span>)</span>
<span id="cb760-7"><a href="WE-7.html#cb760-7" aria-hidden="true" tabindex="-1"></a>listw.d2 <span class="ot">&lt;-</span> spdep<span class="sc">::</span><span class="fu">nb2listw</span>(nb.gab, <span class="at">style =</span> <span class="st">&quot;W&quot;</span>, <span class="at">glist=</span>dlist)</span></code></pre></div>
<p>Now we can quantify and test Moran’s I for each variable to test for spatial autocorrelation in response and predictor variables. For now, we’ll take the simple weights ‘listw.gab.’</p>
<div class="sourceCode" id="cb761"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb761-1"><a href="WE-7.html#cb761-1" aria-hidden="true" tabindex="-1"></a>spdep<span class="sc">::</span><span class="fu">moran.test</span>(Dianthus.df<span class="sc">$</span>A, listw.gab)             </span></code></pre></div>
<pre><code>## 
##  Moran I test under randomisation
## 
## data:  Dianthus.df$A  
## weights: listw.gab    
## 
## Moran I statistic standard deviate = 2.881, p-value = 0.001982
## alternative hypothesis: greater
## sample estimates:
## Moran I statistic       Expectation          Variance 
##        0.29180913       -0.01724138        0.01150691</code></pre>
<div class="sourceCode" id="cb763"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb763-1"><a href="WE-7.html#cb763-1" aria-hidden="true" tabindex="-1"></a>spdep<span class="sc">::</span><span class="fu">moran.test</span>(Dianthus.df<span class="sc">$</span>IBD, listw.gab)</span></code></pre></div>
<pre><code>## 
##  Moran I test under randomisation
## 
## data:  Dianthus.df$IBD  
## weights: listw.gab    
## 
## Moran I statistic standard deviate = 5.9689, p-value = 1.194e-09
## alternative hypothesis: greater
## sample estimates:
## Moran I statistic       Expectation          Variance 
##        0.62709841       -0.01724138        0.01165320</code></pre>
<div class="sourceCode" id="cb765"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb765-1"><a href="WE-7.html#cb765-1" aria-hidden="true" tabindex="-1"></a>spdep<span class="sc">::</span><span class="fu">moran.test</span>(Dianthus.df<span class="sc">$</span>IBR, listw.gab) </span></code></pre></div>
<pre><code>## 
##  Moran I test under randomisation
## 
## data:  Dianthus.df$IBR  
## weights: listw.gab    
## 
## Moran I statistic standard deviate = 3.008, p-value = 0.001315
## alternative hypothesis: greater
## sample estimates:
## Moran I statistic       Expectation          Variance 
##        0.30918167       -0.01724138        0.01177588</code></pre>
<div class="sourceCode" id="cb767"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb767-1"><a href="WE-7.html#cb767-1" aria-hidden="true" tabindex="-1"></a>spdep<span class="sc">::</span><span class="fu">moran.test</span>(Dianthus.df<span class="sc">$</span>PatchSize, listw.gab) </span></code></pre></div>
<pre><code>## 
##  Moran I test under randomisation
## 
## data:  Dianthus.df$PatchSize  
## weights: listw.gab    
## 
## Moran I statistic standard deviate = 3.4257, p-value = 0.0003066
## alternative hypothesis: greater
## sample estimates:
## Moran I statistic       Expectation          Variance 
##        0.35490160       -0.01724138        0.01180103</code></pre>
<p><strong>Questions:</strong></p>
<ul>
<li>Which variables showed statistically signficant spatial autocorrelation?</li>
<li>Which variables showed the strongest autocorrelation? Is this surprising?</li>
</ul>
<p>Next, let’s test each model for autocorrelation in the residuals:</p>
<div class="sourceCode" id="cb769"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb769-1"><a href="WE-7.html#cb769-1" aria-hidden="true" tabindex="-1"></a>spdep<span class="sc">::</span><span class="fu">lm.morantest</span>(mod.lm.IBD, listw.gab) </span></code></pre></div>
<pre><code>## 
##  Global Moran I for regression residuals
## 
## data:  
## model: lm(formula = A ~ IBD, data = Dianthus.df)
## weights: listw.gab
## 
## Moran I statistic standard deviate = 2.9439, p-value = 0.00162
## alternative hypothesis: greater
## sample estimates:
## Observed Moran I      Expectation         Variance 
##       0.28889983      -0.02854559       0.01162751</code></pre>
<div class="sourceCode" id="cb771"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb771-1"><a href="WE-7.html#cb771-1" aria-hidden="true" tabindex="-1"></a>spdep<span class="sc">::</span><span class="fu">lm.morantest</span>(mod.lm.IBR, listw.gab)          </span></code></pre></div>
<pre><code>## 
##  Global Moran I for regression residuals
## 
## data:  
## model: lm(formula = A ~ IBR, data = Dianthus.df)
## weights: listw.gab
## 
## Moran I statistic standard deviate = 1.883, p-value = 0.02985
## alternative hypothesis: greater
## sample estimates:
## Observed Moran I      Expectation         Variance 
##       0.18032443      -0.02296810       0.01165614</code></pre>
<div class="sourceCode" id="cb773"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb773-1"><a href="WE-7.html#cb773-1" aria-hidden="true" tabindex="-1"></a>spdep<span class="sc">::</span><span class="fu">lm.morantest</span>(mod.lm.PatchSize, listw.gab)       </span></code></pre></div>
<pre><code>## 
##  Global Moran I for regression residuals
## 
## data:  
## model: lm(formula = A ~ PatchSize + IBR, data = Dianthus.df)
## weights: listw.gab
## 
## Moran I statistic standard deviate = 1.7484, p-value = 0.04019
## alternative hypothesis: greater
## sample estimates:
## Observed Moran I      Expectation         Variance 
##       0.16223745      -0.02708922       0.01172530</code></pre>
<p>Quite a bit of the spatial autocorrelation in allelic richness can be explained by the spatial structure in the predictors IBR and PatchSize. There is still statistically significant spatial autocorrelation in the residuals, though it is not strong any more.</p>
</div>
</div>
<div id="fit-models-with-spatially-correlated-error-gls-with-package-nlme" class="section level3">
<h3>4. Fit models with spatially correlated error (GLS) with package ‘nlme’</h3>
<p>One way to account for spatial autocorrelation in the residuals is to fit a Generalized Least Squares model (GLS) with a spatially autocorrelated error structure.
See also: <a href="http://rfunctions.blogspot.ca/2017/06/how-to-identify-and-remove-spatial.html" class="uri">http://rfunctions.blogspot.ca/2017/06/how-to-identify-and-remove-spatial.html</a></p>
<div id="a.-plot-empirical-variogram" class="section level4">
<h4>a. Plot empirical variogram</h4>
<p>The error structure in a GLS is defined in a geostatistical framework, based on a variogram and as a function of distance between observations. Hence we start with plotting an empirical variogram of the residuals, with a smooth line. Here we specify ’resType = “normalized,” which means that the variogram will be fitted to the normalized residuals of the model.</p>
<p>The expected value of the semivariance will be 1. Hence it would make sense to add a horizontal line at 1. However, this is cumbersome with the trellis graphics (using package ‘lattice’) used by ‘nlme.’</p>
<div class="sourceCode" id="cb775"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb775-1"><a href="WE-7.html#cb775-1" aria-hidden="true" tabindex="-1"></a>model.lm <span class="ot">&lt;-</span> nlme<span class="sc">::</span><span class="fu">gls</span>(A <span class="sc">~</span> IBR <span class="sc">+</span> PatchSize, <span class="at">data =</span> Dianthus.df, <span class="at">method=</span><span class="st">&quot;REML&quot;</span>)</span>
<span id="cb775-2"><a href="WE-7.html#cb775-2" aria-hidden="true" tabindex="-1"></a>semivario <span class="ot">&lt;-</span> nlme<span class="sc">::</span><span class="fu">Variogram</span>(model.lm, <span class="at">form =</span> <span class="sc">~</span>x  <span class="sc">+</span> y, <span class="at">resType =</span> <span class="st">&quot;normalized&quot;</span>)</span></code></pre></div>
<p>If you want to create your own figure, e.g. with ‘ggplot2,’ you can access the values stores in the data frame ‘semivario’ to plot the points, and add a smooth line yourself. Then we can add a horizontal line with ‘geom_hline.’</p>
<div class="sourceCode" id="cb776"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb776-1"><a href="WE-7.html#cb776-1" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(<span class="at">data=</span>semivario, <span class="fu">aes</span>(<span class="at">x=</span>dist, <span class="at">y=</span>variog)) <span class="sc">+</span> <span class="fu">geom_point</span>() <span class="sc">+</span> <span class="fu">geom_smooth</span>(<span class="at">se=</span><span class="cn">FALSE</span>) <span class="sc">+</span></span>
<span id="cb776-2"><a href="WE-7.html#cb776-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_hline</span>(<span class="at">yintercept=</span><span class="dv">1</span>) <span class="sc">+</span> <span class="fu">ylim</span>(<span class="fu">c</span>(<span class="dv">0</span>,<span class="fl">1.3</span>)) <span class="sc">+</span> <span class="fu">xlab</span>(<span class="st">&quot;Distance&quot;</span>) <span class="sc">+</span> <span class="fu">ylab</span>(<span class="st">&quot;Semivariance&quot;</span>)</span></code></pre></div>
<pre><code>## `geom_smooth()` using method = &#39;loess&#39; and formula &#39;y ~ x&#39;</code></pre>
<p><img src="07-Week07_files/figure-html/unnamed-chunk-27-1.png" width="672" /></p>
<p><strong>Question:</strong></p>
<ul>
<li>What do you conclude from this empirical variogram?</li>
<li>Estimate the range of the variogram from the intersection of the smooth line with the horizontal line.</li>
<li>Estimate the nugget effect from the intercept at Distance = 0.</li>
</ul>
</div>
<div id="b.-fit-variogram-models" class="section level4">
<h4>b. Fit variogram models</h4>
<p>We can ask R to fit different types of variogram models to this empirical variogram. The model family (e.g., exponential, gaussian, spherical) determines the general shape of the curve that will be fitted. With ‘nugget=T,’ we indicate that a nugget effect should be fitted.</p>
<p>Note: Here we want to compare mixed models with the same fixed effects but different random effect structures defined by <code>correlation</code>. For this, we use REML. If we wanted to compare models with the same random effects but different fixed effects (as in Week 6), we should use maximum likelihood.</p>
<ul>
<li>With function <code>lme4::lmer</code>, we can set <code>REML=TRUE</code> for REML and <code>REML=FALSE</code> for ML.</li>
<li>Here with <code>nlme::gls</code>, we set <code>method="REML"</code> for REML and <code>method="ML"</code> for ML.</li>
<li>For update to work here, you’ll need to load the library <code>nlme</code>(or use this, with three coons to access a function ‘update,’ for ‘lme’ objects, that is not ‘exported’ from the package: <code>nlme:::update.lme</code>)</li>
</ul>
<div class="sourceCode" id="cb778"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb778-1"><a href="WE-7.html#cb778-1" aria-hidden="true" tabindex="-1"></a>model.lm <span class="ot">&lt;-</span> nlme<span class="sc">::</span><span class="fu">gls</span>(A <span class="sc">~</span> IBR <span class="sc">+</span> PatchSize, <span class="at">data =</span> Dianthus.df, <span class="at">method=</span><span class="st">&quot;REML&quot;</span>)</span>
<span id="cb778-2"><a href="WE-7.html#cb778-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb778-3"><a href="WE-7.html#cb778-3" aria-hidden="true" tabindex="-1"></a>mod.corExp <span class="ot">&lt;-</span> <span class="fu">update</span>(model.lm, <span class="at">correlation =</span> nlme<span class="sc">::</span><span class="fu">corExp</span>(<span class="at">form =</span> <span class="sc">~</span> x <span class="sc">+</span> y, <span class="at">nugget=</span>T))</span>
<span id="cb778-4"><a href="WE-7.html#cb778-4" aria-hidden="true" tabindex="-1"></a>mod.corGaus <span class="ot">&lt;-</span> <span class="fu">update</span>(model.lm, <span class="at">correlation =</span> nlme<span class="sc">::</span><span class="fu">corGaus</span>(<span class="at">form =</span> <span class="sc">~</span> x <span class="sc">+</span> y, <span class="at">nugget=</span>T))</span>
<span id="cb778-5"><a href="WE-7.html#cb778-5" aria-hidden="true" tabindex="-1"></a>mod.corSpher <span class="ot">&lt;-</span> <span class="fu">update</span>(model.lm, <span class="at">correlation =</span> nlme<span class="sc">::</span><span class="fu">corSpher</span>(<span class="at">form =</span> <span class="sc">~</span> x <span class="sc">+</span> y, <span class="at">nugget=</span>T))</span>
<span id="cb778-6"><a href="WE-7.html#cb778-6" aria-hidden="true" tabindex="-1"></a>mod.corRatio <span class="ot">&lt;-</span> <span class="fu">update</span>(model.lm, <span class="at">correlation =</span> nlme<span class="sc">::</span><span class="fu">corRatio</span>(<span class="at">form =</span> <span class="sc">~</span> x <span class="sc">+</span> y, <span class="at">nugget=</span>T))</span>
<span id="cb778-7"><a href="WE-7.html#cb778-7" aria-hidden="true" tabindex="-1"></a><span class="co">#mod.corLin &lt;- update(model.lm, correlation = nlme::corLin(form = ~ x + y, nugget=T))</span></span></code></pre></div>
</div>
<div id="c.-select-best-fitting-model" class="section level4">
<h4>c. Select best-fitting model</h4>
<p>Now we compare all models for which we did not get an error message:</p>
<div class="sourceCode" id="cb779"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb779-1"><a href="WE-7.html#cb779-1" aria-hidden="true" tabindex="-1"></a>MuMIn<span class="sc">::</span><span class="fu">model.sel</span>(model.lm, mod.corExp, mod.corGaus, mod.corSpher, mod.corRatio)     </span></code></pre></div>
<pre><code>## Model selection table 
##              (Intrc)     IBR   PtchS             family            correlation
## mod.corExp     4.044 0.08468 0.04269 gaussian(identity)   nlme::corExp(~x+y,T)
## mod.corRatio   4.032 0.09704 0.04129 gaussian(identity) nlme::corRatio(~x+y,T)
## model.lm       4.052 0.11340 0.04266 gaussian(identity)                       
## mod.corGaus    4.047 0.11840 0.04247 gaussian(identity)  nlme::corGaus(~x+y,T)
## mod.corSpher   4.048 0.11710 0.04242 gaussian(identity) nlme::corSpher(~x+y,T)
##              df logLik AICc delta weight
## mod.corExp    6  6.283  1.1  0.00  0.336
## mod.corRatio  6  6.229  1.2  0.11  0.319
## model.lm      4  3.702  1.3  0.29  0.291
## mod.corGaus   6  3.763  6.1  5.04  0.027
## mod.corSpher  6  3.760  6.1  5.04  0.027
## Models ranked by AICc(x)</code></pre>
<p>The list sorts the models, with the best model on top. The last column ‘weight’ contains the model weight, which indicate how much support there is for each model, given all other models in the set (see Week 12). Here, the exponential model fitted best, though the ratio model and the model without a spatially correlated error structure fitted the data almost equally well. The top three models have delta values within 2 (in fact, close to 0).</p>
<p>We refit the best model with maximum likelihood to test the fixed effects.</p>
<div class="sourceCode" id="cb781"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb781-1"><a href="WE-7.html#cb781-1" aria-hidden="true" tabindex="-1"></a>mod.corExp.ML <span class="ot">&lt;-</span> nlme<span class="sc">::</span><span class="fu">gls</span>( A <span class="sc">~</span> PatchSize <span class="sc">+</span> IBR, <span class="at">data =</span> Dianthus.df, <span class="at">method=</span><span class="st">&quot;ML&quot;</span>,</span>
<span id="cb781-2"><a href="WE-7.html#cb781-2" aria-hidden="true" tabindex="-1"></a>                            <span class="at">correlation =</span> nlme<span class="sc">::</span><span class="fu">corExp</span>(<span class="at">form =</span> <span class="sc">~</span> x <span class="sc">+</span> y, <span class="at">nugget=</span>T))</span>
<span id="cb781-3"><a href="WE-7.html#cb781-3" aria-hidden="true" tabindex="-1"></a>car<span class="sc">::</span><span class="fu">Anova</span>(mod.corExp.ML) </span></code></pre></div>
<pre><code>## Analysis of Deviance Table (Type II tests)
## 
## Response: A
##           Df  Chisq Pr(&gt;Chisq)  
## PatchSize  1 5.4912    0.01911 *
## IBR        1 0.8710    0.35067  
## ---
## Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1</code></pre>
<p>The fitted model with the exponential error structure shows a significant effect for PatchSize but not for the IBR term.</p>
<p>We don’t get an R-squared value directly, but we can calculate a pseudo R-squared from a regression of the response ‘A’ on the fitted values (using the model fitted with REML). Let’s compare it to the R-squared from the <code>lm</code> model.</p>
<div class="sourceCode" id="cb783"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb783-1"><a href="WE-7.html#cb783-1" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>(<span class="fu">lm</span>(A <span class="sc">~</span> <span class="fu">fitted</span>(mod.corExp), <span class="at">data =</span> Dianthus.df))<span class="sc">$</span>r.squared</span></code></pre></div>
<pre><code>## [1] 0.2317629</code></pre>
<div class="sourceCode" id="cb785"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb785-1"><a href="WE-7.html#cb785-1" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>(mod.lm.PatchSize)<span class="sc">$</span>r.squared</span></code></pre></div>
<pre><code>## [1] 0.2324922</code></pre>
<p>The pseudo R-squared is almost identical to the R-squared of the non-spatial <code>lm</code> model.</p>
<p>Let’s check the residual plots:</p>
<div class="sourceCode" id="cb787"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb787-1"><a href="WE-7.html#cb787-1" aria-hidden="true" tabindex="-1"></a>predictmeans<span class="sc">::</span><span class="fu">residplot</span>(mod.corExp)</span></code></pre></div>
<pre><code>## Registered S3 methods overwritten by &#39;lme4&#39;:
##   method                          from
##   cooks.distance.influence.merMod car 
##   influence.merMod                car 
##   dfbeta.influence.merMod         car 
##   dfbetas.influence.merMod        car</code></pre>
<p>The normal probability plot still looks about the same.</p>
<p>Note that the function <code>residplot</code> recognized that we have a <code>gls</code> model and added a plot of the auto-correlation function, ACF. Here we have a value of 1 for the distance lag 0, which is the comparison of each value with itself. All other values are low. Recall that the autcorrelation is inversely related to the semivariogram (which does not report the value for lag = 0):</p>
<div class="sourceCode" id="cb789"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb789-1"><a href="WE-7.html#cb789-1" aria-hidden="true" tabindex="-1"></a>semivario <span class="ot">&lt;-</span> nlme<span class="sc">::</span><span class="fu">Variogram</span>(mod.corExp, <span class="at">form =</span> <span class="sc">~</span> x <span class="sc">+</span> y, </span>
<span id="cb789-2"><a href="WE-7.html#cb789-2" aria-hidden="true" tabindex="-1"></a>                             <span class="at">resType =</span> <span class="st">&quot;normalized&quot;</span>)</span>
<span id="cb789-3"><a href="WE-7.html#cb789-3" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(semivario, <span class="at">smooth =</span> <span class="cn">TRUE</span>)</span></code></pre></div>
<p><img src="07-Week07_files/figure-html/unnamed-chunk-33-1.png" width="672" /></p>
<p>The variogram of the residuals (after accounting for spatial autocorrelation as modeled by the variogram model) does look better!</p>
</div>
<div id="d.-plot-fitted-variogram-model" class="section level4">
<h4>d. Plot fitted variogram model</h4>
<p>How can we plot the fitted variogram? Let’s first store it in an object ‘Fitted.variog,’ then plot it. Note that the fitted variogram itself has two classes, “Variogram” and “data.frame.” The plot created by <code>plot(Fitted.variog)</code> is a “trellis” object.</p>
<div class="sourceCode" id="cb790"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb790-1"><a href="WE-7.html#cb790-1" aria-hidden="true" tabindex="-1"></a>Fitted.variog <span class="ot">&lt;-</span> nlme<span class="sc">::</span><span class="fu">Variogram</span>(mod.corExp)</span>
<span id="cb790-2"><a href="WE-7.html#cb790-2" aria-hidden="true" tabindex="-1"></a><span class="fu">class</span>(Fitted.variog)</span></code></pre></div>
<pre><code>## [1] &quot;Variogram&quot;  &quot;data.frame&quot;</code></pre>
<div class="sourceCode" id="cb792"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb792-1"><a href="WE-7.html#cb792-1" aria-hidden="true" tabindex="-1"></a><span class="fu">class</span>(<span class="fu">plot</span>(Fitted.variog))</span></code></pre></div>
<pre><code>## [1] &quot;trellis&quot;</code></pre>
<div class="sourceCode" id="cb794"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb794-1"><a href="WE-7.html#cb794-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(Fitted.variog)</span></code></pre></div>
<p><img src="07-Week07_files/figure-html/unnamed-chunk-34-1.png" width="672" /></p>
<p>That was easy. However, <code>trellis</code> plots are difficult to tweak, and we may want to create our own plot with <code>ggplot2</code>. For this, we need to access the fitted variogram values (i.e. the exponential model curve values), which is a bit more involved.</p>
<p>The object ‘Fitted.variog’ is a data frame (S3) with additional attributes. This raises a challenge, because we access attributes of S3 objects with <code>$</code>, but we also use <code>$</code> to access columns in a data frame.</p>
<p>If we just print <code>Fitted.variog</code>, we only see the data frame.</p>
<div class="sourceCode" id="cb795"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb795-1"><a href="WE-7.html#cb795-1" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(Fitted.variog)</span></code></pre></div>
<pre><code>##      variog      dist n.pairs
## 1 0.6782704  620.5202      85
## 2 0.6112605 1236.8297      86
## 3 0.7415494 1827.3367      85
## 4 0.5751311 2269.6729      86
## 5 1.0503017 2691.2494      85
## 6 0.7493939 3074.7370      86</code></pre>
<p>We can see the attributes listed by using <code>str</code>:</p>
<div class="sourceCode" id="cb797"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb797-1"><a href="WE-7.html#cb797-1" aria-hidden="true" tabindex="-1"></a><span class="fu">str</span>(Fitted.variog)</span></code></pre></div>
<pre><code>## Classes &#39;Variogram&#39; and &#39;data.frame&#39;:    20 obs. of  3 variables:
##  $ variog : num  0.678 0.611 0.742 0.575 1.05 ...
##  $ dist   : num  621 1237 1827 2270 2691 ...
##  $ n.pairs: int [1:20(1d)] 85 86 85 86 85 86 85 86 85 86 ...
##  - attr(*, &quot;modelVariog&quot;)=Classes &#39;Variogram&#39; and &#39;data.frame&#39;:  50 obs. of  2 variables:
##   ..$ variog: num [1:50] 0.598 0.637 0.673 0.705 0.734 ...
##   ..$ dist  : num [1:50] 106 385 665 945 1224 ...
##  - attr(*, &quot;collapse&quot;)= logi TRUE</code></pre>
<p>The line we are looking for is: <code>- attr(*, "modelVariog")</code>. The attribute <code>modelVariog</code> has 50 rows (obs.) and 2 variables: <code>$variog</code> and <code>$dist</code>. These are the fitted values (i.e., values of the exponential variogram model for 50 distance values).</p>
<p>The notation <code>attr(*, "modelVariog")</code> is a cryptic way of telling us how to access the attribute: use the function <code>attr</code> and provide two arguments: the object names <code>Fitted.variog</code> (represented by the asterisk), and the name of the attribute, in quotes: <code>attr(Fitted.variog, "modelVariog")</code>.</p>
<div class="sourceCode" id="cb799"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb799-1"><a href="WE-7.html#cb799-1" aria-hidden="true" tabindex="-1"></a>tibble<span class="sc">::</span><span class="fu">as_tibble</span>(<span class="fu">attr</span>(Fitted.variog, <span class="st">&quot;modelVariog&quot;</span>))</span></code></pre></div>
<pre><code>## # A tibble: 50 x 2
##    variog  dist
##     &lt;dbl&gt; &lt;dbl&gt;
##  1  0.598  106.
##  2  0.637  385.
##  3  0.673  665.
##  4  0.705  945.
##  5  0.734 1224.
##  6  0.760 1504.
##  7  0.784 1783.
##  8  0.805 2063.
##  9  0.824 2342.
## 10  0.841 2622.
## # ... with 40 more rows</code></pre>
<p>This is useful to know if you want to create your own figures, e.g. with <code>ggplot2</code>.</p>
<div class="sourceCode" id="cb801"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb801-1"><a href="WE-7.html#cb801-1" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(<span class="at">data=</span>Fitted.variog, <span class="fu">aes</span>(<span class="at">x=</span>dist, <span class="at">y=</span>variog)) <span class="sc">+</span> <span class="fu">geom_point</span>() <span class="sc">+</span> </span>
<span id="cb801-2"><a href="WE-7.html#cb801-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ylim</span>(<span class="fu">c</span>(<span class="dv">0</span>,<span class="fl">1.3</span>)) <span class="sc">+</span> <span class="fu">xlab</span>(<span class="st">&quot;Distance&quot;</span>) <span class="sc">+</span> <span class="fu">ylab</span>(<span class="st">&quot;Semivariance&quot;</span>) <span class="sc">+</span> </span>
<span id="cb801-3"><a href="WE-7.html#cb801-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_line</span>(<span class="at">data=</span><span class="fu">attr</span>(Fitted.variog, <span class="st">&quot;modelVariog&quot;</span>), <span class="fu">aes</span>(<span class="at">x=</span>dist, <span class="at">y=</span>variog), <span class="at">color=</span><span class="st">&quot;blue&quot;</span>) <span class="sc">+</span></span>
<span id="cb801-4"><a href="WE-7.html#cb801-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_hline</span>(<span class="at">yintercept=</span><span class="dv">1</span>,<span class="at">linetype=</span><span class="st">&quot;dashed&quot;</span>)</span></code></pre></div>
<p><img src="07-Week07_files/figure-html/unnamed-chunk-38-1.png" width="672" /></p>
</div>
<div id="e.-add-random-factor" class="section level4">
<h4>e. Add random factor</h4>
<p>The package <code>nlme</code> allows us also to include random factors. Here we add <code>System</code> as a random factor and test whether this would improve the model fit.</p>
<ul>
<li>Instead of function <code>nlme::gls</code>, we use the function <code>nlme::lme</code>.</li>
<li>In <code>nlme</code>, random effects are specified differently from <code>lmer::lme4</code> (Week6): <code>random = ~ 1 | System</code>.</li>
<li>The correlation structure is specified exactly as with <code>gls</code>.</li>
</ul>
<div class="sourceCode" id="cb802"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb802-1"><a href="WE-7.html#cb802-1" aria-hidden="true" tabindex="-1"></a>mod.lme.corExp <span class="ot">&lt;-</span> nlme<span class="sc">::</span><span class="fu">lme</span>( A <span class="sc">~</span> PatchSize <span class="sc">+</span> IBR, </span>
<span id="cb802-2"><a href="WE-7.html#cb802-2" aria-hidden="true" tabindex="-1"></a>                             <span class="at">random =</span> <span class="sc">~</span> <span class="dv">1</span> <span class="sc">|</span> System, <span class="at">data =</span> Dianthus.df, </span>
<span id="cb802-3"><a href="WE-7.html#cb802-3" aria-hidden="true" tabindex="-1"></a>                            <span class="at">correlation =</span> nlme<span class="sc">::</span><span class="fu">corExp</span>(<span class="at">form =</span> <span class="sc">~</span> x <span class="sc">+</span> y, <span class="at">nugget=</span>T),</span>
<span id="cb802-4"><a href="WE-7.html#cb802-4" aria-hidden="true" tabindex="-1"></a>                            <span class="at">method=</span><span class="st">&quot;REML&quot;</span>)</span>
<span id="cb802-5"><a href="WE-7.html#cb802-5" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>(mod.lme.corExp)</span></code></pre></div>
<pre><code>## Linear mixed-effects model fit by REML
##   Data: Dianthus.df 
##           AIC     BIC   logLik
##   -0.04236554 14.1351 7.021183
## 
## Random effects:
##  Formula: ~1 | System
##         (Intercept)  Residual
## StdDev:   0.0816568 0.1947089
## 
## Correlation Structure: Exponential spatial correlation
##  Formula: ~x + y | System 
##  Parameter estimate(s):
##       range      nugget 
## 385.7379751   0.1971837 
## Fixed effects:  A ~ PatchSize + IBR 
##                Value  Std.Error DF  t-value p-value
## (Intercept) 4.084490 0.08657381 53 47.17928  0.0000
## PatchSize   0.049722 0.01860186 53  2.67296  0.0100
## IBR         0.083831 0.10438753 53  0.80307  0.4255
##  Correlation: 
##           (Intr) PtchSz
## PatchSize  0.560       
## IBR       -0.782 -0.485
## 
## Standardized Within-Group Residuals:
##          Min           Q1          Med           Q3          Max 
## -3.156087747 -0.303811941  0.006632744  0.524431829  1.893761279 
## 
## Number of Observations: 59
## Number of Groups: 4</code></pre>
<p>The nature of the results did not change: PatchSize is still significant but IBR is not.</p>
<p>As in Week 6, we can obtain marginal (fixed effects) and conditional R-squared values (fixed + random):</p>
<div class="sourceCode" id="cb804"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb804-1"><a href="WE-7.html#cb804-1" aria-hidden="true" tabindex="-1"></a>MuMIn<span class="sc">::</span><span class="fu">r.squaredGLMM</span>(mod.lme.corExp)</span></code></pre></div>
<pre><code>## Warning: &#39;r.squaredGLMM&#39; now calculates a revised statistic. See the help page.</code></pre>
<pre><code>##            R2m       R2c
## [1,] 0.2294698 0.3447197</code></pre>
<p>Now we can include this model in the model comparison from above. Notes:</p>
<ul>
<li>As discussed in the Week 6 video, we should fit the mixed model with maximum likelihood (method = “ML”) to test fixed effects and to compare its AIC to the other models.</li>
<li>The Week 6 video followed the philosphy that random effect should only be fitted for factors with &gt;5 levels, whereas here, we are using a factor with 4 levels.</li>
</ul>
<div class="sourceCode" id="cb807"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb807-1"><a href="WE-7.html#cb807-1" aria-hidden="true" tabindex="-1"></a>MuMIn<span class="sc">::</span><span class="fu">model.sel</span>(model.lm, mod.corExp, mod.corRatio, mod.lme.corExp)  </span></code></pre></div>
<pre><code>## Model selection table 
##                (Intrc)     IBR   PtchS             family class
## mod.corExp       4.044 0.08468 0.04269 gaussian(identity)   gls
## mod.corRatio     4.032 0.09704 0.04129 gaussian(identity)   gls
## model.lm         4.052 0.11340 0.04266 gaussian(identity)   gls
## mod.lme.corExp   4.084 0.08383 0.04972 gaussian(identity)   lme
##                           correlation random df logLik AICc delta weight
## mod.corExp       nlme::corExp(~x+y,T)         6  6.283  1.1  0.00  0.295
## mod.corRatio   nlme::corRatio(~x+y,T)         6  6.229  1.2  0.11  0.280
## model.lm                                      4  3.702  1.3  0.29  0.256
## mod.lme.corExp   nlme::corExp(~x+y,T)      S  7  7.021  2.2  1.10  0.170
## Models ranked by AICc(x) 
## Random terms: 
## S = &#39;1 | System&#39;</code></pre>
<p><strong>Questions:</strong></p>
<ul>
<li>How may degrees of freedom (df) were used for the random effect?</li>
<li>How can you see from the table that System was fitted as a random effect, and what method was used (REML vs. ML)?</li>
<li>Was the model with the random effect ranked higher than the model without it?</li>
<li>Compare the estimates of the slope coefficients for IBR between the models. How did accounting for spatial autocorrelation affect the slope coefficient, compared to <code>model.lm</code>? How large was the difference due to using different variogram models? And how much of a difference was related to including the random effect?</li>
<li>How about the slope estimate for PatchSize?</li>
</ul>
</div>
</div>
<div id="fit-spatial-simultaneous-autoregressive-error-models-sar" class="section level3">
<h3>5. Fit spatial simultaneous autoregressive error models (SAR)</h3>
<p>An alternative way to account for spatial autocorrelation in the residuals is spatial regression with a simultaneous autoregressive error model (SAR).</p>
<div id="a.-fit-and-compare-alternative-sar-models" class="section level4">
<h4>a. Fit and compare alternative SAR models</h4>
<p>The method <code>errorsarlm</code> fits a simultaneous autoregressive model (‘sar’) to the error (‘error’) term of a ‘lm’ model.</p>
<p>This approach is based on spatial neighbours and weights. We have already defined them in three versions of a <code>listw</code> object. Let’s see which one fits the data best.</p>
<div class="sourceCode" id="cb809"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb809-1"><a href="WE-7.html#cb809-1" aria-hidden="true" tabindex="-1"></a>mod.sar.IBR.gab <span class="ot">&lt;-</span> spatialreg<span class="sc">::</span><span class="fu">errorsarlm</span>(A <span class="sc">~</span> PatchSize <span class="sc">+</span> IBR, <span class="at">data =</span> Dianthus.df, </span>
<span id="cb809-2"><a href="WE-7.html#cb809-2" aria-hidden="true" tabindex="-1"></a>                                 <span class="at">listw =</span> listw.gab)</span>
<span id="cb809-3"><a href="WE-7.html#cb809-3" aria-hidden="true" tabindex="-1"></a>mod.sar.IBR.d1 <span class="ot">&lt;-</span> spatialreg<span class="sc">::</span><span class="fu">errorsarlm</span>(A <span class="sc">~</span> PatchSize <span class="sc">+</span> IBR, <span class="at">data =</span> Dianthus.df, </span>
<span id="cb809-4"><a href="WE-7.html#cb809-4" aria-hidden="true" tabindex="-1"></a>                                 <span class="at">listw =</span> listw.d1)</span>
<span id="cb809-5"><a href="WE-7.html#cb809-5" aria-hidden="true" tabindex="-1"></a>mod.sar.IBR.d2 <span class="ot">&lt;-</span> spatialreg<span class="sc">::</span><span class="fu">errorsarlm</span>(A <span class="sc">~</span> PatchSize <span class="sc">+</span> IBR, <span class="at">data =</span> Dianthus.df, </span>
<span id="cb809-6"><a href="WE-7.html#cb809-6" aria-hidden="true" tabindex="-1"></a>                                 <span class="at">listw =</span> listw.d2)</span>
<span id="cb809-7"><a href="WE-7.html#cb809-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb809-8"><a href="WE-7.html#cb809-8" aria-hidden="true" tabindex="-1"></a>MuMIn<span class="sc">::</span><span class="fu">model.sel</span>(mod.lm.IBR, mod.sar.IBR.gab, mod.sar.IBR.d1, mod.sar.IBR.d2) </span></code></pre></div>
<pre><code>## This method assumes the response is known - see manual page
## This method assumes the response is known - see manual page
## This method assumes the response is known - see manual page</code></pre>
<pre><code>## Model selection table 
##                 (Intrc)     IBR   PtchS             family class     listw df
## mod.sar.IBR.d1    4.067 0.09058 0.04142                 () sarlm  listw.d1  5
## mod.sar.IBR.gab   4.063 0.09458 0.04081                 () sarlm listw.gab  5
## mod.sar.IBR.d2    4.055 0.10650 0.04133                 () sarlm  listw.d2  5
## mod.lm.IBR        3.923 0.25520         gaussian(identity)    lm            3
##                 logLik  AICc delta weight
## mod.sar.IBR.d1  12.500 -13.9  0.00  0.425
## mod.sar.IBR.gab 12.215 -13.3  0.57  0.320
## mod.sar.IBR.d2  11.549 -12.0  1.90  0.164
## mod.lm.IBR       8.603 -10.8  3.10  0.090
## Models ranked by AICc(x)</code></pre>
<p>The best model (‘mod.sar.IBR.d1’) is the one with (row-standardized) inverse-distance weights (‘listw.d1’). It is only slightly better than the model with the (row-standardized) binary weights (‘listw.gab’), whereas the nonspatial model and the one with (row-standardized) inverse squared distance weights have much less support.</p>
</div>
<div id="b.-interpret-best-fitting-sar-model" class="section level4">
<h4>b. Interpret best-fitting SAR model</h4>
<p>Let’s have a look at the best model. With the argument <code>Nagelkerke = TRUE</code>, we request a pseudo R-squared.</p>
<div class="sourceCode" id="cb812"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb812-1"><a href="WE-7.html#cb812-1" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>(mod.sar.IBR.d1, <span class="at">Nagelkerke =</span> <span class="cn">TRUE</span>)</span></code></pre></div>
<pre><code>## 
## Call:spatialreg::errorsarlm(formula = A ~ PatchSize + IBR, data = Dianthus.df, 
##     listw = listw.d1)
## 
## Residuals:
##       Min        1Q    Median        3Q       Max 
## -0.654613 -0.085961  0.016066  0.091687  0.389419 
## 
## Type: error 
## Coefficients: (asymptotic standard errors) 
##             Estimate Std. Error z value Pr(&gt;|z|)
## (Intercept) 4.067066   0.077809 52.2696   &lt;2e-16
## PatchSize   0.041416   0.018554  2.2322   0.0256
## IBR         0.090578   0.094946  0.9540   0.3401
## 
## Lambda: 0.22322, LR test value: 2.6442, p-value: 0.10393
## Asymptotic standard error: 0.13659
##     z-value: 1.6343, p-value: 0.1022
## Wald statistic: 2.6708, p-value: 0.1022
## 
## Log likelihood: 12.49972 for error model
## ML residual variance (sigma squared): 0.037595, (sigma: 0.19389)
## Nagelkerke pseudo-R-squared: 0.26613 
## Number of observations: 59 
## Number of parameters estimated: 5 
## AIC: -14.999, (AIC for lm: -14.355)</code></pre>
<ul>
<li>Again, PatchSize is significant but not IBR.</li>
<li>The section starting with ‘Lamba’ summarizes the fitted spatial autocorrelation term. It is not statistically significant (p-value = 0.1039 for the Likelihood Ratio test LR).</li>
</ul>
</div>
</div>
<div id="spatial-filtering-with-mem-using-package-spmoran" class="section level3">
<h3>6. Spatial filtering with MEM using package ‘spmoran’</h3>
<p>See tutorial for ‘spmoran’ package: <a href="https://arxiv.org/ftp/arxiv/papers/1703/1703.04467.pdf" class="uri">https://arxiv.org/ftp/arxiv/papers/1703/1703.04467.pdf</a></p>
<p>Both GLS and SAR fitted a spatially correlated error structure of a relatively simple form to the data. Gene flow could be more complex and for example, could create spatial autocorrelation structure that is not the same in all directions or in all parts of the study area. Moran Eigenvector Maps (MEM) allows a more flexible modeling of spatial structure in the data. In spatial filtering, we use MEM spatial eigenvectors to account for any spatial structure while fitting and testing the effect of our predictors.</p>
<div id="a.-default-method" class="section level4">
<h4>a. Default method</h4>
<p>The new package <code>spmoran</code> makes this really easy. First, we create the MEM spatial eigenvectors. This implies defining neighbors and weights, but this is well hidden in the code below. The function <code>meigen</code> here takes the coordinates, calculates a minimum spanning tree (so that each site has at least one neighbour), and finds the maximum distance ‘h’ from the spanning tree. It then calculates neighbor weights as <code>exp(-dij / h)</code>.</p>
<p>Note: if you have many sites (&gt; 200), the function <code>meigen_f</code> may be used instead of <code>meigen</code>, it should even work for &gt;1000 sites.</p>
<p>The function <code>esf</code> then performs the spatial filtering. Here it uses stepwise selection of MEM spatial eigenvectors using an R-squared criterion (<code>fn = "r2"</code>).</p>
<div class="sourceCode" id="cb814"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb814-1"><a href="WE-7.html#cb814-1" aria-hidden="true" tabindex="-1"></a><span class="co"># lm model: using truncated distance matrix (max of min spanning tree distance)</span></span>
<span id="cb814-2"><a href="WE-7.html#cb814-2" aria-hidden="true" tabindex="-1"></a>meig <span class="ot">&lt;-</span> spmoran<span class="sc">::</span><span class="fu">meigen</span>(<span class="at">coords=</span>xy)</span></code></pre></div>
<pre><code>##  9/59 eigen-pairs are extracted</code></pre>
<div class="sourceCode" id="cb816"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb816-1"><a href="WE-7.html#cb816-1" aria-hidden="true" tabindex="-1"></a>sfd.res <span class="ot">&lt;-</span> spmoran<span class="sc">::</span><span class="fu">esf</span>( <span class="at">y=</span>Dianthus.df<span class="sc">$</span>A, <span class="at">x=</span>Dianthus.df[,<span class="fu">c</span>(<span class="st">&quot;PatchSize&quot;</span>, <span class="st">&quot;IBR&quot;</span>)],</span>
<span id="cb816-2"><a href="WE-7.html#cb816-2" aria-hidden="true" tabindex="-1"></a>                       <span class="at">meig=</span>meig, <span class="at">fn =</span> <span class="st">&quot;r2&quot;</span> )</span></code></pre></div>
<pre><code>##   5/9 eigenvectors are selected</code></pre>
<p>The objects created by functions ‘meigen’ and ‘esf’ contain a lot of information:</p>
<ul>
<li><strong>meigW:</strong> a list returned by function ‘meigen,’ with the following attributes:
<ul>
<li><strong>sf:</strong> Matrix of retained spatial eigenvectors.</li>
<li><strong>ev:</strong> Eigenvalues of retained spatial eigenvectors.</li>
<li><strong>ev_full:</strong> All (n - 1) eigenvalues.</li>
</ul></li>
<li><strong>sfd.res:</strong> a list returned by function ‘esf,’ with the following attributes:
<ul>
<li><strong>b:</strong> Table with regression results for predictors X.</li>
<li><strong>r:</strong> Table with regression results the selected MEM spatial eigenvectors (based on step-wise eigenvector selection).</li>
<li><strong>e:</strong> Summary statistics for the entire model.</li>
<li><strong>vif:</strong> Variance inflation factors.</li>
<li><strong>sf:</strong> Fitted spatially dependent component (i.e., fitted value based on significant MEM spatial eigenvectors)</li>
<li><strong>pred:</strong> Fitted values.</li>
<li><strong>resid:</strong> Residuals.</li>
</ul></li>
</ul>
<p>Let’s look at the table ‘b’ with regression results for the predictors first:</p>
<div class="sourceCode" id="cb818"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb818-1"><a href="WE-7.html#cb818-1" aria-hidden="true" tabindex="-1"></a>sfd.res<span class="sc">$</span>b</span></code></pre></div>
<pre><code>##               Estimate         SE    t_value      p_value
## (Intercept) 4.08678879 0.06759658 60.4585130 3.838659e-49
## PatchSize   0.03642554 0.01673612  2.1764625 3.417603e-02
## IBR         0.04687286 0.08602602  0.5448685 5.882188e-01</code></pre>
<p>Again, PatchSize is statistically significant but not IBR.</p>
<p>Next, we look at the table ‘r’ with regression results for MEM spatial eigenvectors:</p>
<div class="sourceCode" id="cb820"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb820-1"><a href="WE-7.html#cb820-1" aria-hidden="true" tabindex="-1"></a>sfd.res<span class="sc">$</span>r</span></code></pre></div>
<pre><code>##       Estimate        SE   t_value     p_value
## sf6  0.5938646 0.1782765  3.331144 0.001613967
## sf1  0.3962440 0.1788408  2.215624 0.031207615
## sf7 -0.4065267 0.1895836 -2.144313 0.036795842
## sf3  0.3474148 0.1760289  1.973624 0.053854181
## sf5 -0.3290397 0.1757533 -1.872168 0.066922529</code></pre>
<p>Five MEM spatial eigenvectors were important enough to be included in the model. Here they are ranked by their (absolute value of) slope coefficient, and thus by the strength of their association with the response variable. Eigenvector ‘sf6’ was by far the most important.</p>
<p>Note: some eigenvectors are included despite having a p-value &gt; 0.05. This may have two reasons. First, the eigenvectors were selected without taking into account predictors X. Second, a different test was used in the stepwise eigenvector selection. The type of test can be specified with an argument <code>fn</code> (see ‘?esf’ helpfile and ‘spmoran’ tutorial).</p>
<p>Finally, let’s look at the summary results for the fitted model:</p>
<div class="sourceCode" id="cb822"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb822-1"><a href="WE-7.html#cb822-1" aria-hidden="true" tabindex="-1"></a>sfd.res<span class="sc">$</span>e</span></code></pre></div>
<pre><code>##                 stat
## resid_SE   0.1728755
## adjR2      0.4374576
## logLik    24.1369653
## AIC      -30.2739307
## BIC      -11.5760937</code></pre>
<p>Here, ‘adjR2’ is rather high (0.437), but this includes the selected MEM spatial eigenvectors!</p>
</div>
<div id="b.-using-a-custom-connectivity-matrix" class="section level4">
<h4>b. Using a custom connectivity matrix</h4>
<p>We know already that <code>listw.d1</code> fit the data well, so let’s re-run the model with our own definition of spatial weights. With the funciton ‘listw2mat,’ we convert from <code>listw</code> format to a full connnectivity matrix.</p>
<div class="sourceCode" id="cb824"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb824-1"><a href="WE-7.html#cb824-1" aria-hidden="true" tabindex="-1"></a>cmat.d1    <span class="ot">&lt;-</span> spdep<span class="sc">::</span><span class="fu">listw2mat</span>( listw.d1) </span>
<span id="cb824-2"><a href="WE-7.html#cb824-2" aria-hidden="true" tabindex="-1"></a>meigw  <span class="ot">&lt;-</span> spmoran<span class="sc">::</span><span class="fu">meigen</span>( <span class="at">cmat =</span> cmat.d1 )</span></code></pre></div>
<pre><code>##  Note:</code></pre>
<pre><code>##    cmat is symmetrized by ( cmat + t( cmat ) ) / 2</code></pre>
<pre><code>##  27/59 eigen-pairs are extracted</code></pre>
<div class="sourceCode" id="cb828"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb828-1"><a href="WE-7.html#cb828-1" aria-hidden="true" tabindex="-1"></a>sfw.res <span class="ot">&lt;-</span> spmoran<span class="sc">::</span><span class="fu">esf</span>( <span class="at">y=</span>Dianthus.df<span class="sc">$</span>A, <span class="at">x=</span>Dianthus.df[,<span class="fu">c</span>(<span class="st">&quot;PatchSize&quot;</span>, <span class="st">&quot;IBR&quot;</span>)],</span>
<span id="cb828-2"><a href="WE-7.html#cb828-2" aria-hidden="true" tabindex="-1"></a>                       <span class="at">meig=</span>meigw, <span class="at">fn =</span> <span class="st">&quot;r2&quot;</span> )</span></code></pre></div>
<pre><code>##   15/27 eigenvectors are selected</code></pre>
<div class="sourceCode" id="cb830"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb830-1"><a href="WE-7.html#cb830-1" aria-hidden="true" tabindex="-1"></a>sfw.res<span class="sc">$</span>b</span></code></pre></div>
<pre><code>##               Estimate         SE    t_value      p_value
## (Intercept) 4.05985212 0.06875531 59.0478330 2.716097e-41
## PatchSize   0.02051541 0.01679035  1.2218572 2.287404e-01
## IBR         0.06583796 0.09048375  0.7276219 4.709800e-01</code></pre>
<div class="sourceCode" id="cb832"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb832-1"><a href="WE-7.html#cb832-1" aria-hidden="true" tabindex="-1"></a>tibble<span class="sc">::</span><span class="fu">as_tibble</span>(sfw.res<span class="sc">$</span>r)</span></code></pre></div>
<pre><code>## # A tibble: 15 x 4
##    Estimate    SE t_value  p_value
##       &lt;dbl&gt; &lt;dbl&gt;   &lt;dbl&gt;    &lt;dbl&gt;
##  1   -0.712 0.172   -4.14 0.000166
##  2    0.511 0.166    3.08 0.00369 
##  3   -0.366 0.151   -2.43 0.0196  
##  4    0.284 0.152    1.87 0.0693  
##  5   -0.278 0.155   -1.79 0.0806  
##  6    0.236 0.167    1.41 0.166   
##  7    0.270 0.150    1.80 0.0787  
##  8    0.309 0.159    1.94 0.0594  
##  9   -0.289 0.160   -1.80 0.0786  
## 10    0.227 0.152    1.49 0.143   
## 11    0.233 0.151    1.55 0.130   
## 12    0.217 0.151    1.43 0.159   
## 13   -0.189 0.151   -1.25 0.217   
## 14    0.196 0.152    1.29 0.204   
## 15    0.166 0.161    1.03 0.309</code></pre>
<div class="sourceCode" id="cb834"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb834-1"><a href="WE-7.html#cb834-1" aria-hidden="true" tabindex="-1"></a>sfw.res<span class="sc">$</span>e</span></code></pre></div>
<pre><code>##                 stat
## resid_SE   0.1493144
## adjR2      0.5803455
## logLik    39.2199714
## AIC      -40.4399428
## BIC       -0.9667313</code></pre>
<p>Note: the messages tell us that ‘cmat’ has been made symmetric before analysis, that 27 out of 59 MEM spatial eigenvector (and their eigenvalues, hence ‘pairs’) were retained initially and subjected to stepwise selection, which then returned 15 statistically significant MEM eigenvectors that were included in the regression model with the predictor variables X (PatchSize and IBR).</p>
<p><strong>Questions:</strong></p>
<ul>
<li>Does this model fit the data better? Look for a lower AIC. In addition, you can compare the adjusted R-squared.</li>
<li>What could cause a difference in model performance?</li>
<li>Does this affect the results for PatchSize and IBR? Compare both parameter estimates and p-value between two two models.</li>
</ul>
</div>
<div id="c.-plot-spatial-eigenvectors" class="section level4">
<h4>c. Plot spatial eigenvectors</h4>
<p>So far, we have treated the MEM spatial eigenvectors as a black box. What kind of patterns do they represent?</p>
<p>First, we plot all the selected (significant) eigenvectors. A convenient way to do so is converting to a SpatialPointsDataFrame and then use the function ‘spplot.’</p>
<p>Here we need tweak the column names of the eigenvectors, which are called “X1,” “X2” etc., to match the row names of table ‘sf.res$r,’ which uses “sf1,” “sf2” etc.</p>
<ul>
<li>In the first line, we create a data frame ‘SF’ that combines the coordinates from ‘xy’ with the eigenvectors from ‘meigw.’ By specifying <code>sf=meigw$sf</code>, the column names will be prefaced by ‘sf.’ so that “X1” becomes “sf.1.”</li>
<li>With the function <code>gsub</code>, we substitute “.” by an empty vector "“, i.e., we remove the period from the name, so that”sf.1" becomes “sf1.” This is a bit tricky because of the special meanings of ‘.’: we need to escape (here we use “[.]”) to indicate that ‘.’ should be interpreted and matched literally as a period symbol, not as a wildcard that matches any character. Function <code>sub</code> substitutes the first occurrence of the symbol, whereas <code>gsub</code> substitutes all occurrences.</li>
</ul>
<div class="sourceCode" id="cb836"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb836-1"><a href="WE-7.html#cb836-1" aria-hidden="true" tabindex="-1"></a>SF <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(xy, <span class="at">sf=</span>meigw<span class="sc">$</span>sf)</span>
<span id="cb836-2"><a href="WE-7.html#cb836-2" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(SF) <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">&quot;[.]&quot;</span>, <span class="st">&quot;&quot;</span>, <span class="fu">names</span>(SF))</span>
<span id="cb836-3"><a href="WE-7.html#cb836-3" aria-hidden="true" tabindex="-1"></a>sp<span class="sc">::</span><span class="fu">coordinates</span>(SF) <span class="ot">&lt;-</span> <span class="er">~</span> x <span class="sc">+</span> y</span>
<span id="cb836-4"><a href="WE-7.html#cb836-4" aria-hidden="true" tabindex="-1"></a>sp<span class="sc">::</span><span class="fu">spplot</span>(SF, <span class="fu">row.names</span>(sfw.res<span class="sc">$</span>r), <span class="at">colorkey =</span> <span class="cn">TRUE</span>)</span></code></pre></div>
<p><img src="07-Week07_files/figure-html/unnamed-chunk-49-1.png" width="672" /></p>
<p>The panel starts at the bottom left, so that the most important spatial eigenvector (sf6) is plotted at the bottom left, the second most important (sf22) second from left, etc.</p>
<p>The smallest numbers are patterns with the largest spatial scale (sf1), which here shows a gradient from East to West. The most important eigenvector (sf6) shows a finer-scale pattern with the highest values (yellow) in the center, lowest values East and West, and intermediate value North and South.</p>
<p>However, these patterns individually are not meaningful. More importantly, we can plot the total spatial component in the response as a weighted sum of these component patterns, where the weights correspond to the regression coefficients in table ‘sf.res$r.’ Here we create a panel with two plots, the modeled spatial components ‘sf’ on the left and the response ‘A’ on the right (the mean has been removed to make values comparable).</p>
<div class="sourceCode" id="cb837"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb837-1"><a href="WE-7.html#cb837-1" aria-hidden="true" tabindex="-1"></a>SF<span class="sc">@</span>data<span class="sc">$</span>sf <span class="ot">&lt;-</span> sfw.res<span class="sc">$</span>sf</span>
<span id="cb837-2"><a href="WE-7.html#cb837-2" aria-hidden="true" tabindex="-1"></a>SF<span class="sc">@</span>data<span class="sc">$</span>A <span class="ot">&lt;-</span> <span class="fu">scale</span>(Dianthus.df<span class="sc">$</span>A, <span class="at">scale =</span> <span class="cn">FALSE</span>)</span>
<span id="cb837-3"><a href="WE-7.html#cb837-3" aria-hidden="true" tabindex="-1"></a>sp<span class="sc">::</span><span class="fu">spplot</span>(SF, <span class="fu">c</span>(<span class="st">&quot;sf&quot;</span>, <span class="st">&quot;A&quot;</span>), <span class="at">colorkey =</span> <span class="cn">TRUE</span>)</span></code></pre></div>
<p><img src="07-Week07_files/figure-html/unnamed-chunk-50-1.png" width="672" /></p>
<p>Obviously, a big part of the variation in allelic richness is already captured by ‘sf.’ In essence the model then tries to explain the difference between these two sets of values by the predictors “PatchSize” and “IBR.”</p>
<p>Let’s quantify the correlation of this spatial component with allelic richness, and compare the correlation between the two models:</p>
<div class="sourceCode" id="cb838"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb838-1"><a href="WE-7.html#cb838-1" aria-hidden="true" tabindex="-1"></a><span class="fu">cor</span>(Dianthus.df<span class="sc">$</span>A, <span class="fu">data.frame</span>(<span class="at">sfd=</span>sfd.res<span class="sc">$</span>sf, <span class="at">sfw=</span>sfw.res<span class="sc">$</span>sf))</span></code></pre></div>
<pre><code>##            sfd       sfw
## [1,] 0.6252594 0.8054356</code></pre>
<p>With the default method (defining neighbors based on a distance cut-off), the spatial component modeled by the significant MEM spatial eigenvectors showed a correlation of 0.625 with the response variable. Using a Gabriel graph with inverse distance weights increased this correlation to 0.772.</p>
<p>This means that the spatial eigenvectors derived from the Gabriel graph were more effective at capturing the spatial variation in allelic richness than the default method. This spatial component is then controlled for when assessing the relationship between allelic richness and the predictors (PatchSize and IBR).</p>
</div>
<div id="d.-random-effect-model" class="section level4">
<h4>d. Random effect model</h4>
<p>The previous model selected 15 MEM spatial eigenvectors, and thus fitted 15 additional models. Just like the random effects for family and population in Week 6 lab, we can save a few parameters here by fitting the set of MEM eigenvectors as a random effect. This is done by the function ‘resf.’</p>
<div class="sourceCode" id="cb840"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb840-1"><a href="WE-7.html#cb840-1" aria-hidden="true" tabindex="-1"></a>sfr.res <span class="ot">&lt;-</span> spmoran<span class="sc">::</span><span class="fu">resf</span>( <span class="at">y=</span>Dianthus.df<span class="sc">$</span>A, <span class="at">x=</span>Dianthus.df[,<span class="fu">c</span>(<span class="st">&quot;PatchSize&quot;</span>, <span class="st">&quot;IBR&quot;</span>)], </span>
<span id="cb840-2"><a href="WE-7.html#cb840-2" aria-hidden="true" tabindex="-1"></a>               <span class="at">meig =</span> meigw, <span class="at">method =</span> <span class="st">&quot;reml&quot;</span> ) </span>
<span id="cb840-3"><a href="WE-7.html#cb840-3" aria-hidden="true" tabindex="-1"></a>sfr.res<span class="sc">$</span>b</span></code></pre></div>
<pre><code>##               Estimate         SE   t_value    p_value
## (Intercept) 4.06902708 0.06645938 61.225772 0.00000000
## PatchSize   0.04247164 0.01652051  2.570843 0.01344639
## IBR         0.08489102 0.08467569  1.002543 0.32132728</code></pre>
<div class="sourceCode" id="cb842"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb842-1"><a href="WE-7.html#cb842-1" aria-hidden="true" tabindex="-1"></a>tibble<span class="sc">::</span><span class="fu">as_tibble</span>(sfr.res<span class="sc">$</span>r)</span></code></pre></div>
<pre><code>## # A tibble: 27 x 1
##      value
##      &lt;dbl&gt;
##  1 -0.191 
##  2 -0.0314
##  3 -0.0601
##  4 -0.124 
##  5  0.0640
##  6 -0.312 
##  7 -0.0576
##  8 -0.113 
##  9  0.128 
## 10  0.105 
## # ... with 17 more rows</code></pre>
<div class="sourceCode" id="cb844"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb844-1"><a href="WE-7.html#cb844-1" aria-hidden="true" tabindex="-1"></a>sfr.res<span class="sc">$</span>e</span></code></pre></div>
<pre><code>##                    stat
## resid_SE     0.15643276
## adjR2(cond)  0.50394641
## rlogLik      5.99141269
## AIC          0.01717462
## BIC         12.48239928</code></pre>
<div class="sourceCode" id="cb846"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb846-1"><a href="WE-7.html#cb846-1" aria-hidden="true" tabindex="-1"></a>sfr.res<span class="sc">$</span>s</span></code></pre></div>
<pre><code>##                      (Intercept)
## random_SE              0.1785609
## Moran.I/max(Moran.I)   0.7742903</code></pre>
<p>As in Week 6 lab, the conditional R-squared is the variance explained by the fixed effects (PatchSize and IBR) and the random effects (significant MEM spatial eigenvectors) together. It is adjusted for the number of effects that were estimated.</p>
<p>Note: we can’t compare AIC with the previous models, as the model was fitted with ‘reml.’</p>
<p>We get an additional output ‘sfr.res$s’ with two variance parameters:</p>
<ul>
<li><strong>spcomp_SE</strong>: standard error of the spatial component.</li>
<li><strong>spcomp_Moran.I/max(Moran.I)</strong>: Moran’s I of the spatial component, rescaled by the maximum possible value. From the help file: “Based on Griffith (2003), the scaled Moran’I value is interpretable as follows: 0.25-0.50:weak; 0.50-0.70:moderate; 0.70-0.90:strong; 0.90-1.00:marked.”</li>
</ul>
</div>
</div>
<div id="fit-spatially-varying-coefficients-model-with-package-spmoran" class="section level3">
<h3>7. Fit spatially varying coefficients model with package ‘spmoran’</h3>
<p>See: <a href="https://arxiv.org/ftp/arxiv/papers/1703/1703.04467.pdf" class="uri">https://arxiv.org/ftp/arxiv/papers/1703/1703.04467.pdf</a></p>
<p>Now comes the coolest part!</p>
<p>So far, we have fitted the same model for all sites. Geographically weighted regression (GWR) would allow relaxing this. Spatial filtering with MEM can be used to accomplish the same goal, and the ‘spmoran’ tutorial calls this a ‘Spatially Varying Coefficients’ model (SVC). The main advantage is that we can visualize how the slope parameter estimates, and their p-values, vary across the study area! This is a great exploratory tool that can help us better understand what is going on.</p>
</div>
<div id="model-with-patchsize-and-ibr" class="section level3">
<h3>Model with PatchSize and IBR</h3>
<p>We fit the model with ‘resf_vc.’</p>
<div class="sourceCode" id="cb848"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb848-1"><a href="WE-7.html#cb848-1" aria-hidden="true" tabindex="-1"></a>rv_res <span class="ot">&lt;-</span> spmoran<span class="sc">::</span><span class="fu">resf_vc</span>( <span class="at">y=</span>Dianthus.df<span class="sc">$</span>A, </span>
<span id="cb848-2"><a href="WE-7.html#cb848-2" aria-hidden="true" tabindex="-1"></a>                            <span class="at">x =</span> Dianthus.df[,<span class="fu">c</span>(<span class="st">&quot;PatchSize&quot;</span>, <span class="st">&quot;IBR&quot;</span>)], </span>
<span id="cb848-3"><a href="WE-7.html#cb848-3" aria-hidden="true" tabindex="-1"></a>                            <span class="at">xconst =</span> <span class="cn">NULL</span>, <span class="at">meig =</span> meigw, <span class="at">method =</span> <span class="st">&quot;reml&quot;</span>)</span></code></pre></div>
<pre><code>## [1] &quot;-------  Iteration 1  -------&quot;
## [1] &quot;1/3&quot;
## [1] &quot;2/3&quot;
## [1] &quot;3/3&quot;
## [1] &quot;BIC: 10.465&quot;
## [1] &quot;-------  Iteration 2  -------&quot;
## [1] &quot;1/3&quot;
## [1] &quot;2/3&quot;
## [1] &quot;3/3&quot;
## [1] &quot;BIC: 10.37&quot;
## [1] &quot;-------  Iteration 3  -------&quot;
## [1] &quot;1/3&quot;
## [1] &quot;2/3&quot;
## [1] &quot;3/3&quot;
## [1] &quot;BIC: 10.37&quot;
## [1] &quot;-------  Iteration 4  -------&quot;
## [1] &quot;1/3&quot;
## [1] &quot;2/3&quot;
## [1] &quot;3/3&quot;
## [1] &quot;BIC: 10.37&quot;</code></pre>
<p>Instead of one slope estimate for each predictor, we now get a different estimate for each combination of parameter and site (sounds like overfitting?). Here’s a summary of the distbiution of these estimates.</p>
<div class="sourceCode" id="cb850"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb850-1"><a href="WE-7.html#cb850-1" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>( rv_res<span class="sc">$</span>b_vc ) </span></code></pre></div>
<pre><code>##   (Intercept)      PatchSize            IBR         
##  Min.   :3.877   Min.   :0.04717   Min.   :0.06883  
##  1st Qu.:4.049   1st Qu.:0.04717   1st Qu.:0.06883  
##  Median :4.106   Median :0.04717   Median :0.06883  
##  Mean   :4.083   Mean   :0.04717   Mean   :0.06883  
##  3rd Qu.:4.128   3rd Qu.:0.04717   3rd Qu.:0.06883  
##  Max.   :4.189   Max.   :0.04717   Max.   :0.06883</code></pre>
<p>The slope estimate for PatchSize varied between 0 and 0.098, with a mean of 0.046. The slope estimate for the ‘IBR’ term varied between -0.81 and 0.34, with a mean very close to 0! That is an astounding range of variation. Keep in mind that we really expect a positive relationship, there is no biological explanation for a ngeative relationship.</p>
<p>Here is a similar summary of the p-values:</p>
<div class="sourceCode" id="cb852"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb852-1"><a href="WE-7.html#cb852-1" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>( rv_res<span class="sc">$</span>p_vc )</span></code></pre></div>
<pre><code>##   (Intercept)   PatchSize             IBR        
##  Min.   :0    Min.   :0.005485   Min.   :0.4135  
##  1st Qu.:0    1st Qu.:0.005485   1st Qu.:0.4135  
##  Median :0    Median :0.005485   Median :0.4135  
##  Mean   :0    Mean   :0.005485   Mean   :0.4135  
##  3rd Qu.:0    3rd Qu.:0.005485   3rd Qu.:0.4135  
##  Max.   :0    Max.   :0.005485   Max.   :0.4135</code></pre>
<p>For both variables, most sites do not show a significant effect (i.e., only few sites show a p-value &lt; 0.05).</p>
<p>We could print these results by site (type <code>rv_res$b_vc</code> or <code>rv_res$p_vc</code>). Even better, we can plot them in space. We start with combining the data (‘Dianthus.df’) and the results into one data frame ‘Results.’ By specifying <code>b=rv_res$b_vc</code> and <code>p=rv_res$p_vc</code>, R will create column names that start with ‘b’ or ‘p,’ respectively.</p>
<div class="sourceCode" id="cb854"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb854-1"><a href="WE-7.html#cb854-1" aria-hidden="true" tabindex="-1"></a>Result <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(Dianthus.df, <span class="at">b=</span>rv_res<span class="sc">$</span>b_vc, <span class="at">p=</span>rv_res<span class="sc">$</span>p_vc)</span>
<span id="cb854-2"><a href="WE-7.html#cb854-2" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(Result)</span></code></pre></div>
<pre><code>##  [1] &quot;A&quot;             &quot;IBD&quot;           &quot;IBR&quot;           &quot;PatchSize&quot;    
##  [5] &quot;System&quot;        &quot;Longitude&quot;     &quot;Latitude&quot;      &quot;x&quot;            
##  [9] &quot;y&quot;             &quot;b..Intercept.&quot; &quot;b.PatchSize&quot;   &quot;b.IBR&quot;        
## [13] &quot;p..Intercept.&quot; &quot;p.PatchSize&quot;   &quot;p.IBR&quot;</code></pre>
<p>Let’s start with PatchSize. Here, we first plot PatchSize in space, with symbol size as a function of patch size. In a second plot, we color sites by statistical significance and the size of the symbols represents the parameter estimate of the regression slope coefficient for Patch Size. The layer ‘coord_fixed’ keeps controls the aspect ratio between x- and y-axes.</p>
<div class="sourceCode" id="cb856"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb856-1"><a href="WE-7.html#cb856-1" aria-hidden="true" tabindex="-1"></a><span class="fu">require</span>(ggplot2)</span>
<span id="cb856-2"><a href="WE-7.html#cb856-2" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(<span class="fu">as.data.frame</span>(Result), <span class="fu">aes</span>(x, y, <span class="at">size=</span>PatchSize)) <span class="sc">+</span></span>
<span id="cb856-3"><a href="WE-7.html#cb856-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">color=</span><span class="st">&quot;darkblue&quot;</span>) <span class="sc">+</span> <span class="fu">coord_fixed</span>()</span>
<span id="cb856-4"><a href="WE-7.html#cb856-4" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(<span class="fu">as.data.frame</span>(Result), <span class="fu">aes</span>(x, y, <span class="at">col=</span>p.PatchSize <span class="sc">&lt;</span> <span class="fl">0.05</span>, <span class="at">size=</span>b.PatchSize)) <span class="sc">+</span></span>
<span id="cb856-5"><a href="WE-7.html#cb856-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>() <span class="sc">+</span> <span class="fu">coord_fixed</span>()</span></code></pre></div>
<p><img src="07-Week07_files/figure-html/unnamed-chunk-57-1.png" width="672" /><img src="07-Week07_files/figure-html/unnamed-chunk-57-2.png" width="672" /></p>
<p>Let’s do the same for ‘IBR’:</p>
<div class="sourceCode" id="cb857"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb857-1"><a href="WE-7.html#cb857-1" aria-hidden="true" tabindex="-1"></a><span class="fu">require</span>(ggplot2)</span>
<span id="cb857-2"><a href="WE-7.html#cb857-2" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(<span class="fu">as.data.frame</span>(Result), <span class="fu">aes</span>(x, y, <span class="at">size=</span>IBR)) <span class="sc">+</span></span>
<span id="cb857-3"><a href="WE-7.html#cb857-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">color=</span><span class="st">&quot;darkgreen&quot;</span>) <span class="sc">+</span> <span class="fu">coord_fixed</span>()</span>
<span id="cb857-4"><a href="WE-7.html#cb857-4" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(<span class="fu">as.data.frame</span>(Result), <span class="fu">aes</span>(x, y, <span class="at">col=</span>p.IBR <span class="sc">&lt;</span> <span class="fl">0.05</span>, <span class="at">size=</span>b.IBR)) <span class="sc">+</span></span>
<span id="cb857-5"><a href="WE-7.html#cb857-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>() <span class="sc">+</span> <span class="fu">coord_fixed</span>()</span></code></pre></div>
<p><img src="07-Week07_files/figure-html/unnamed-chunk-58-1.png" width="672" /><img src="07-Week07_files/figure-html/unnamed-chunk-58-2.png" width="672" /></p>
<ul>
<li>The very small dots in the first map are the ungrazed patches.</li>
<li>From the second map, it looks like the significant values were the one with negative slope estimates, for which we don’t have a biological interpretation.</li>
</ul>
</div>
<div id="model-with-ibr-only" class="section level3">
<h3>Model with IBR only</h3>
<p>Keep in mind that ‘IBR’ and ‘PatchSize’ showed a strong correlation. The parameter estimates could therefore depend quite a bit on the other variables. To help with the interpretation, let’s repeat the last analysis just with ‘IBR,’ without ‘PatchSize.’</p>
<div class="sourceCode" id="cb858"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb858-1"><a href="WE-7.html#cb858-1" aria-hidden="true" tabindex="-1"></a>rv_res <span class="ot">&lt;-</span> spmoran<span class="sc">::</span><span class="fu">resf_vc</span>( <span class="at">y=</span>Dianthus.df<span class="sc">$</span>A, </span>
<span id="cb858-2"><a href="WE-7.html#cb858-2" aria-hidden="true" tabindex="-1"></a>                            <span class="at">x =</span> Dianthus.df[,<span class="fu">c</span>(<span class="st">&quot;IBR&quot;</span>)], </span>
<span id="cb858-3"><a href="WE-7.html#cb858-3" aria-hidden="true" tabindex="-1"></a>                            <span class="at">xconst =</span> <span class="cn">NULL</span>, <span class="at">meig =</span> meigw, <span class="at">method =</span> <span class="st">&quot;reml&quot;</span>)</span></code></pre></div>
<pre><code>## [1] &quot;-------  Iteration 1  -------&quot;
## [1] &quot;1/2&quot;
## [1] &quot;2/2&quot;
## [1] &quot;BIC: 5.181&quot;
## [1] &quot;-------  Iteration 2  -------&quot;
## [1] &quot;1/2&quot;
## [1] &quot;2/2&quot;
## [1] &quot;BIC: 5.226&quot;
## [1] &quot;-------  Iteration 3  -------&quot;
## [1] &quot;1/2&quot;
## [1] &quot;2/2&quot;
## [1] &quot;BIC: 5.226&quot;
## [1] &quot;-------  Iteration 4  -------&quot;
## [1] &quot;1/2&quot;
## [1] &quot;2/2&quot;
## [1] &quot;BIC: 5.226&quot;</code></pre>
<div class="sourceCode" id="cb860"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb860-1"><a href="WE-7.html#cb860-1" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>( rv_res<span class="sc">$</span>b_vc ) </span></code></pre></div>
<pre><code>##   (Intercept)          V1        
##  Min.   :3.737   Min.   :0.2264  
##  1st Qu.:3.902   1st Qu.:0.2264  
##  Median :3.961   Median :0.2264  
##  Mean   :3.941   Mean   :0.2264  
##  3rd Qu.:3.994   3rd Qu.:0.2264  
##  Max.   :4.064   Max.   :0.2264</code></pre>
<p>Now the range of slope estimates is smaller, most sites have a positive estimate, and the mean is approx. 0.21.</p>
<div class="sourceCode" id="cb862"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb862-1"><a href="WE-7.html#cb862-1" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>( rv_res<span class="sc">$</span>p_vc ) </span></code></pre></div>
<pre><code>##   (Intercept)       V1          
##  Min.   :0    Min.   :0.001875  
##  1st Qu.:0    1st Qu.:0.001875  
##  Median :0    Median :0.001875  
##  Mean   :0    Mean   :0.001875  
##  3rd Qu.:0    3rd Qu.:0.001875  
##  Max.   :0    Max.   :0.001875</code></pre>
<p>Also, a larger proportion of sites nows has p-values &lt; 0.05.</p>
<p>Let’s plot the results onto a gray-scale, stamen terrain map to facilitate interpretation. Note: here the zoom level <code>zoom = 12</code> covers the entire study area, whereas the default value would actually cut off a large number of sites. We use the argument <code>force=TRUE</code> to force the map to be downloaded again (otherwise the argument <code>color="bw"</code> may not have an effect if we already downloaded the terrain map in color).</p>
<div class="sourceCode" id="cb864"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb864-1"><a href="WE-7.html#cb864-1" aria-hidden="true" tabindex="-1"></a>Result <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(Dianthus.df, <span class="at">b=</span>rv_res<span class="sc">$</span>b_vc, <span class="at">p=</span>rv_res<span class="sc">$</span>p_vc, <span class="at">resid=</span>rv_res<span class="sc">$</span>resid)</span>
<span id="cb864-2"><a href="WE-7.html#cb864-2" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(Result)</span></code></pre></div>
<pre><code>##  [1] &quot;A&quot;             &quot;IBD&quot;           &quot;IBR&quot;           &quot;PatchSize&quot;    
##  [5] &quot;System&quot;        &quot;Longitude&quot;     &quot;Latitude&quot;      &quot;x&quot;            
##  [9] &quot;y&quot;             &quot;b..Intercept.&quot; &quot;b.V1&quot;          &quot;p..Intercept.&quot;
## [13] &quot;p.V1&quot;          &quot;resid&quot;</code></pre>
<div class="sourceCode" id="cb866"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb866-1"><a href="WE-7.html#cb866-1" aria-hidden="true" tabindex="-1"></a>ggmap<span class="sc">::</span><span class="fu">ggmap</span>(<span class="fu">get_stamenmap</span>(bbox, <span class="at">maptype =</span> <span class="st">&quot;terrain&quot;</span>, <span class="at">color=</span><span class="st">&quot;bw&quot;</span>, <span class="at">force=</span><span class="cn">TRUE</span>, <span class="at">zoom=</span><span class="dv">12</span>)) <span class="sc">+</span> </span>
<span id="cb866-2"><a href="WE-7.html#cb866-2" aria-hidden="true" tabindex="-1"></a>   <span class="fu">geom_point</span>(<span class="fu">aes</span>(<span class="at">x =</span>  Longitude, <span class="at">y =</span> Latitude, <span class="at">col=</span>p.V1 <span class="sc">&lt;</span> <span class="fl">0.05</span>, <span class="at">size=</span>b.V1), <span class="at">data =</span> Result) </span></code></pre></div>
<p><img src="07-Week07_files/figure-html/unnamed-chunk-61-1.png" width="672" /></p>
<p>This is a very different map of results!</p>
<ul>
<li>Most sites now show significant effects.</li>
<li>The sites with larger positive estimates show significant effects, whereas those with small or negative estimates show non-significant effects.</li>
<li>There are 3 - 4 clusters of sites where the IBR models is not effective at explaining variation in allelic richness: in the very East, in the South-East, and one area in the South-West.</li>
<li>Knowing the study area, these are distinct regions (e.g. valleys) that may suggest further biological explanations.</li>
</ul>
</div>
<div id="conclusions" class="section level3">
<h3>8. Conclusions</h3>
<ul>
<li>We moved from pair-wise distance matrices (link-based) to node-based analysis by integrating the explanatory distance matrices for IBD and IBR into patch-level connectivity indices Si (neighborhood analysis).</li>
<li>We found no support for the IBD model, and strong support for the IBR model when tested without additional predictors.</li>
<li>The site-level predictors ‘PatchSize’ (log(‘Ha’)) was strongly correlated with our IBR model, and when PatchSize was added to the model, ‘IBR’ was no long statistically significant and its slope estimate changed considerably.</li>
<li>The MEM analogue to spatially weighted regression showed very different patterns for ‘IBR’ depending on whether or not ‘PatchSize’ was included in the model. Withouth ‘PatchSize,’ ‘IBR’ showed significant positive correlation with allelic richness across the study area, except for three sub-areas.</li>
<li>In practical terms, this may suggest that the management strategy of maintaining plant functional connectivity through shepherding seems to be working for this species overall, though there are three parts of the study area where this may not be sufficient to maintain gene flow.</li>
<li>The evidence is not conclusive, however, the observed patterns could also be explained by population size, which in this species seems to be associated with patch size. This makes sense if smaller patches contain smaller populations with higher rates of genetic drift.</li>
</ul>
</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="tutorial-7.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="r-exercise-week-7.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": true,
"facebook": false,
"twitter": false,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": false
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": null,
"toc": {
"collapse": "section"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
