<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>15.1 Worked Example | Landscape Genetic Data Analysis with R</title>
  <meta name="description" content="This is a web-interface to the teaching materials for the lab course ‘Landscape Genetic Data Analysis with R’ associated with the distributed graduate course ‘DGS Landscape Genetics’. The output format is bookdown::gitbook." />
  <meta name="generator" content="bookdown 0.37 and GitBook 2.6.7" />

  <meta property="og:title" content="15.1 Worked Example | Landscape Genetic Data Analysis with R" />
  <meta property="og:type" content="book" />
  
  <meta property="og:description" content="This is a web-interface to the teaching materials for the lab course ‘Landscape Genetic Data Analysis with R’ associated with the distributed graduate course ‘DGS Landscape Genetics’. The output format is bookdown::gitbook." />
  <meta name="github-repo" content="hhwagner1/LandGenCourse_book" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="15.1 Worked Example | Landscape Genetic Data Analysis with R" />
  
  <meta name="twitter:description" content="This is a web-interface to the teaching materials for the lab course ‘Landscape Genetic Data Analysis with R’ associated with the distributed graduate course ‘DGS Landscape Genetics’. The output format is bookdown::gitbook." />
  

<meta name="author" content="Editor: Helene Wagner (University of Toronto)" />


<meta name="date" content="2024-01-25" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="Week12.html"/>
<link rel="next" href="Week13.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>
<script src="libs/kePrint-0.0.1/kePrint.js"></script>
<link href="libs/lightable-0.0.1/lightable.css" rel="stylesheet" />
<script src="libs/htmlwidgets-1.6.4/htmlwidgets.js"></script>
<link href="libs/leaflet-1.3.1/leaflet.css" rel="stylesheet" />
<script src="libs/leaflet-1.3.1/leaflet.js"></script>
<link href="libs/leafletfix-1.0.0/leafletfix.css" rel="stylesheet" />
<script src="libs/proj4-2.6.2/proj4.min.js"></script>
<script src="libs/Proj4Leaflet-1.0.1/proj4leaflet.js"></script>
<link href="libs/rstudio_leaflet-1.3.1/rstudio_leaflet.css" rel="stylesheet" />
<script src="libs/leaflet-binding-2.2.1/leaflet.js"></script>
<script src="libs/leaflet-providers-2.0.0/leaflet-providers_2.0.0.js"></script>
<script src="libs/leaflet-providers-plugin-2.2.1/leaflet-providers-plugin.js"></script>
<script src="libs/d3-3.5.6/d3.min.js"></script>
<link href="libs/profvis-0.3.6.9000/profvis.css" rel="stylesheet" />
<script src="libs/profvis-0.3.6.9000/profvis.js"></script>
<script src="libs/profvis-0.3.6.9000/scroll.js"></script>
<link href="libs/highlight-6.2.0/textmate.css" rel="stylesheet" />
<script src="libs/highlight-6.2.0/highlight.js"></script>
<script src="libs/profvis-binding-0.3.8/profvis.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
  
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">Landscape Genetic Data Analysis with R</a></li>

<li class="divider"></li>
<li class="part"><span><b>Getting Started</b></span></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Introduction</a>
<ul>
<li class="chapter" data-level="1.1" data-path="how-to-use-this-book.html"><a href="how-to-use-this-book.html"><i class="fa fa-check"></i><b>1.1</b> How to use this Book</a>
<ul>
<li class="chapter" data-level="" data-path="how-to-use-this-book.html"><a href="how-to-use-this-book.html#book-structure"><i class="fa fa-check"></i>1. Book Structure</a></li>
<li class="chapter" data-level="" data-path="how-to-use-this-book.html"><a href="how-to-use-this-book.html#find-what-is-relevant-for-you"><i class="fa fa-check"></i>2. Find what is relevant for you</a></li>
<li class="chapter" data-level="" data-path="how-to-use-this-book.html"><a href="how-to-use-this-book.html#course-r-package-landgencourse"><i class="fa fa-check"></i>3. Course R package ‘LandGenCourse’</a></li>
</ul></li>
<li class="chapter" data-level="1.2" data-path="list-of-packages.html"><a href="list-of-packages.html"><i class="fa fa-check"></i><b>1.2</b> List of R Packages by Vignette</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="review-of-r-skills.html"><a href="review-of-r-skills.html"><i class="fa fa-check"></i><b>2</b> Review of R Skills</a>
<ul>
<li class="chapter" data-level="2.1" data-path="basic-r.html"><a href="basic-r.html"><i class="fa fa-check"></i><b>2.1</b> Basic R Programming</a>
<ul>
<li class="chapter" data-level="" data-path="basic-r.html"><a href="basic-r.html#overview"><i class="fa fa-check"></i>1. Overview</a></li>
<li class="chapter" data-level="" data-path="basic-r.html"><a href="basic-r.html#data-types"><i class="fa fa-check"></i>2. Data Types</a></li>
<li class="chapter" data-level="" data-path="basic-r.html"><a href="basic-r.html#data-containers"><i class="fa fa-check"></i>3. Data Containers</a></li>
<li class="chapter" data-level="" data-path="basic-r.html"><a href="basic-r.html#programming"><i class="fa fa-check"></i>4. Programming</a></li>
</ul></li>
<li class="chapter" data-level="2.2" data-path="r-graphics.html"><a href="r-graphics.html"><i class="fa fa-check"></i><b>2.2</b> R Graphics</a>
<ul>
<li class="chapter" data-level="" data-path="r-graphics.html"><a href="r-graphics.html#overview-1"><i class="fa fa-check"></i>1. Overview</a></li>
<li class="chapter" data-level="" data-path="r-graphics.html"><a href="r-graphics.html#univariate-plots"><i class="fa fa-check"></i>2. Univariate plots</a></li>
<li class="chapter" data-level="" data-path="r-graphics.html"><a href="r-graphics.html#bivariate-plots"><i class="fa fa-check"></i>3. Bivariate plots</a></li>
<li class="chapter" data-level="" data-path="r-graphics.html"><a href="r-graphics.html#advanced-plotting"><i class="fa fa-check"></i>4. Advanced plotting</a></li>
</ul></li>
<li class="chapter" data-level="2.3" data-path="resources.html"><a href="resources.html"><i class="fa fa-check"></i><b>2.3</b> Further R resources</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="github-for-group-projects.html"><a href="github-for-group-projects.html"><i class="fa fa-check"></i><b>3</b> GitHub for Group Projects</a>
<ul>
<li class="chapter" data-level="3.1" data-path="github.html"><a href="github.html"><i class="fa fa-check"></i><b>3.1</b> Why use GitHub?</a>
<ul>
<li class="chapter" data-level="" data-path="github.html"><a href="github.html#goals"><i class="fa fa-check"></i>1. Goals</a></li>
<li class="chapter" data-level="" data-path="github.html"><a href="github.html#version"><i class="fa fa-check"></i>2. Watch course video: Version Control 101</a></li>
<li class="chapter" data-level="" data-path="github.html"><a href="github.html#slides"><i class="fa fa-check"></i>3. Preview slides</a></li>
<li class="chapter" data-level="" data-path="github.html"><a href="github.html#next-steps"><i class="fa fa-check"></i>4. Next steps</a></li>
</ul></li>
<li class="chapter" data-level="3.2" data-path="install.html"><a href="install.html"><i class="fa fa-check"></i><b>3.2</b> Installation</a>
<ul>
<li class="chapter" data-level="" data-path="install.html"><a href="install.html#goals-1"><i class="fa fa-check"></i>1. Goals</a></li>
<li class="chapter" data-level="3.2.1" data-path="install.html"><a href="install.html#how-to-get-github-and-gitkraken-for-free"><i class="fa fa-check"></i><b>3.2.1</b> 2. How to get GitHub and GitKraken for free</a></li>
<li class="chapter" data-level="" data-path="install.html"><a href="install.html#happygit"><i class="fa fa-check"></i>3. Happy Git with R</a></li>
<li class="chapter" data-level="" data-path="install.html"><a href="install.html#gitvideos"><i class="fa fa-check"></i>4. Video tutorials: installation</a></li>
<li class="chapter" data-level="" data-path="install.html"><a href="install.html#join"><i class="fa fa-check"></i>5. Joining your group’s project on GitHub</a></li>
<li class="chapter" data-level="" data-path="install.html"><a href="install.html#repo"><i class="fa fa-check"></i>6. Setting up a group repo</a></li>
<li class="chapter" data-level="" data-path="install.html"><a href="install.html#next-steps-1"><i class="fa fa-check"></i>7. Next steps</a></li>
</ul></li>
<li class="chapter" data-level="3.3" data-path="workflow.html"><a href="workflow.html"><i class="fa fa-check"></i><b>3.3</b> Workflows</a>
<ul>
<li class="chapter" data-level="" data-path="workflow.html"><a href="workflow.html#goals-2"><i class="fa fa-check"></i>1. Goals</a></li>
<li class="chapter" data-level="" data-path="workflow.html"><a href="workflow.html#step-by-step-tutorials"><i class="fa fa-check"></i>2. Step-by-step tutorials</a></li>
<li class="chapter" data-level="" data-path="workflow.html"><a href="workflow.html#further-resources"><i class="fa fa-check"></i>3. Further resources</a></li>
</ul></li>
</ul></li>
<li class="part"><span><b>Basic Topics</b></span></li>
<li class="chapter" data-level="4" data-path="Week1.html"><a href="Week1.html"><i class="fa fa-check"></i><b>4</b> Week 1: Importing Genetic Data</a>
<ul>
<li class="chapter" data-level="4.1" data-path="video_1.html"><a href="video_1.html"><i class="fa fa-check"></i><b>4.1</b> View Course Video</a>
<ul>
<li class="chapter" data-level="" data-path="video_1.html"><a href="video_1.html#embedded-video"><i class="fa fa-check"></i>1. Embedded Video</a></li>
<li class="chapter" data-level="" data-path="video_1.html"><a href="video_1.html#preview-slides"><i class="fa fa-check"></i>2. Preview Slides</a></li>
</ul></li>
<li class="chapter" data-level="4.2" data-path="tutorial_1.html"><a href="tutorial_1.html"><i class="fa fa-check"></i><b>4.2</b> Interactive Tutorial 1</a>
<ul>
<li class="chapter" data-level="" data-path="tutorial_1.html"><a href="tutorial_1.html#list-of-r-commands-covered-this-week"><i class="fa fa-check"></i>1. List of R commands covered this week</a></li>
<li class="chapter" data-level="" data-path="tutorial_1.html"><a href="tutorial_1.html#general-instructions"><i class="fa fa-check"></i>2. General Instructions</a></li>
</ul></li>
<li class="chapter" data-level="4.3" data-path="WE_1.html"><a href="WE_1.html"><i class="fa fa-check"></i><b>4.3</b> Worked Example</a>
<ul>
<li class="chapter" data-level="" data-path="WE_1.html"><a href="WE_1.html#overview-of-worked-example"><i class="fa fa-check"></i>1. Overview of Worked Example</a></li>
<li class="chapter" data-level="" data-path="WE_1.html"><a href="WE_1.html#import-data-from-.csv-file"><i class="fa fa-check"></i>2. Import data from .csv file</a></li>
<li class="chapter" data-level="" data-path="WE_1.html"><a href="WE_1.html#create-genind-object"><i class="fa fa-check"></i>3. Create ‘genind’ Object</a></li>
<li class="chapter" data-level="" data-path="WE_1.html"><a href="WE_1.html#view-information-stored-in-genind-object"><i class="fa fa-check"></i>4. View information stored in ‘genind’ object</a></li>
<li class="chapter" data-level="" data-path="WE_1.html"><a href="WE_1.html#import-data-with-gstudio-package"><i class="fa fa-check"></i>5. Import data with ‘gstudio’ package</a></li>
<li class="chapter" data-level="" data-path="WE_1.html"><a href="WE_1.html#importing-snp-data"><i class="fa fa-check"></i>6. Importing SNP data</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="r-exercise-week-1.html"><a href="r-exercise-week-1.html"><i class="fa fa-check"></i>R exercise Week 1</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="Week2.html"><a href="Week2.html"><i class="fa fa-check"></i><b>5</b> Week 2: Spatial Data</a>
<ul>
<li class="chapter" data-level="5.1" data-path="video_2.html"><a href="video_2.html"><i class="fa fa-check"></i><b>5.1</b> View Course Video</a>
<ul>
<li class="chapter" data-level="" data-path="video_2.html"><a href="video_2.html#embedded-video-1"><i class="fa fa-check"></i>1. Embedded Video</a></li>
<li class="chapter" data-level="" data-path="video_2.html"><a href="video_2.html#preview-slides-1"><i class="fa fa-check"></i>2. Preview Slides</a></li>
</ul></li>
<li class="chapter" data-level="5.2" data-path="tutorial_2.html"><a href="tutorial_2.html"><i class="fa fa-check"></i><b>5.2</b> Interactive Tutorial 2</a>
<ul>
<li class="chapter" data-level="" data-path="tutorial_2.html"><a href="tutorial_2.html#list-of-r-commands-covered-this-week-1"><i class="fa fa-check"></i>1. List of R commands covered this week</a></li>
<li class="chapter" data-level="" data-path="tutorial_2.html"><a href="tutorial_2.html#general-instructions-1"><i class="fa fa-check"></i>2. General Instructions</a></li>
</ul></li>
<li class="chapter" data-level="5.3" data-path="WE_2.html"><a href="WE_2.html"><i class="fa fa-check"></i><b>5.3</b> Worked Example</a>
<ul>
<li class="chapter" data-level="" data-path="WE_2.html"><a href="WE_2.html#overview-of-worked-example-1"><i class="fa fa-check"></i>1. Overview of Worked Example</a></li>
<li class="chapter" data-level="" data-path="WE_2.html"><a href="WE_2.html#import-site-data-from-.csv-file"><i class="fa fa-check"></i>2. Import site data from .csv file</a></li>
<li class="chapter" data-level="" data-path="WE_2.html"><a href="WE_2.html#display-raster-data-and-overlay-sampling-locations-extract-data"><i class="fa fa-check"></i>3. Display raster data and overlay sampling locations, extract data</a></li>
<li class="chapter" data-level="" data-path="WE_2.html"><a href="WE_2.html#calculate-landscape-metrics"><i class="fa fa-check"></i>4. Calculate landscape metrics</a></li>
<li class="chapter" data-level="" data-path="WE_2.html"><a href="WE_2.html#sample-landscape-metrics-within-buffer-around-sampling-locations"><i class="fa fa-check"></i>5. Sample landscape metrics within buffer around sampling locations</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="r-exercise-week-2.html"><a href="r-exercise-week-2.html"><i class="fa fa-check"></i>R Exercise Week 2</a></li>
<li class="chapter" data-level="5.4" data-path="bonus_2a.html"><a href="bonus_2a.html"><i class="fa fa-check"></i><b>5.4</b> Bonus: ‘sf’ and ‘terra’</a>
<ul>
<li class="chapter" data-level="" data-path="bonus_2a.html"><a href="bonus_2a.html#overview-2"><i class="fa fa-check"></i>1. Overview</a></li>
<li class="chapter" data-level="" data-path="bonus_2a.html"><a href="bonus_2a.html#import-and-export-esri-shapefiles"><i class="fa fa-check"></i>2. Import and export ESRI shapefiles</a></li>
<li class="chapter" data-level="" data-path="bonus_2a.html"><a href="bonus_2a.html#compatibility-with-sp-and-raster-objects"><i class="fa fa-check"></i>3. Compatibility with <code>sp</code> and <code>raster</code> objects</a></li>
<li class="chapter" data-level="" data-path="bonus_2a.html"><a href="bonus_2a.html#plotting-spatial-data-with-tmap"><i class="fa fa-check"></i>4. Plotting spatial data with <code>tmap</code></a></li>
<li class="chapter" data-level="" data-path="bonus_2a.html"><a href="bonus_2a.html#plot-a-categorical-map-with-predefined-color-scheme"><i class="fa fa-check"></i>5. Plot a categorical map with predefined color scheme</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="6" data-path="Week3.html"><a href="Week3.html"><i class="fa fa-check"></i><b>6</b> Week 3: Genetic Diversity</a>
<ul>
<li class="chapter" data-level="6.1" data-path="video_3.html"><a href="video_3.html"><i class="fa fa-check"></i><b>6.1</b> View Course Video</a>
<ul>
<li class="chapter" data-level="" data-path="video_3.html"><a href="video_3.html#embedded-video-2"><i class="fa fa-check"></i>1. Embedded Video</a></li>
<li class="chapter" data-level="" data-path="video_3.html"><a href="video_3.html#video-part-1"><i class="fa fa-check"></i>Video, Part 1</a></li>
<li class="chapter" data-level="" data-path="video_3.html"><a href="video_3.html#video-part-2"><i class="fa fa-check"></i>Video, Part 2</a></li>
<li class="chapter" data-level="" data-path="video_3.html"><a href="video_3.html#preview-slides-2"><i class="fa fa-check"></i>Preview Slides</a></li>
</ul></li>
<li class="chapter" data-level="6.2" data-path="tutorial_3.html"><a href="tutorial_3.html"><i class="fa fa-check"></i><b>6.2</b> Interactive Tutorial 3</a>
<ul>
<li class="chapter" data-level="" data-path="tutorial_3.html"><a href="tutorial_3.html#list-of-r-commands-covered-this-week-2"><i class="fa fa-check"></i>1. List of R commands covered this week</a></li>
<li class="chapter" data-level="" data-path="tutorial_3.html"><a href="tutorial_3.html#general-instructions-2"><i class="fa fa-check"></i>2. General Instructions</a></li>
</ul></li>
<li class="chapter" data-level="6.3" data-path="WE_3.html"><a href="WE_3.html"><i class="fa fa-check"></i><b>6.3</b> Worked Example</a></li>
<li class="chapter" data-level="6.4" data-path="r-exercise-week-3.html"><a href="r-exercise-week-3.html"><i class="fa fa-check"></i><b>6.4</b> R Exercise Week 3</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="Week4.html"><a href="Week4.html"><i class="fa fa-check"></i><b>7</b> Week 4: Metapopulations</a>
<ul>
<li class="chapter" data-level="7.1" data-path="video_4.html"><a href="video_4.html"><i class="fa fa-check"></i><b>7.1</b> View Course Video</a>
<ul>
<li class="chapter" data-level="" data-path="video_4.html"><a href="video_4.html#embedded-video-3"><i class="fa fa-check"></i>1. Embedded Video</a></li>
<li class="chapter" data-level="" data-path="video_4.html"><a href="video_4.html#video-part-1-1"><i class="fa fa-check"></i>Video, Part 1</a></li>
<li class="chapter" data-level="" data-path="video_4.html"><a href="video_4.html#video-part-2-1"><i class="fa fa-check"></i>Video, Part 2</a></li>
<li class="chapter" data-level="" data-path="video_4.html"><a href="video_4.html#preview-slides-3"><i class="fa fa-check"></i>Preview Slides</a></li>
</ul></li>
<li class="chapter" data-level="7.2" data-path="tutorial_4.html"><a href="tutorial_4.html"><i class="fa fa-check"></i><b>7.2</b> Interactive Tutorial 4</a>
<ul>
<li class="chapter" data-level="" data-path="tutorial_4.html"><a href="tutorial_4.html#list-of-r-commands-covered-this-week-3"><i class="fa fa-check"></i>1. List of R commands covered this week</a></li>
<li class="chapter" data-level="" data-path="tutorial_4.html"><a href="tutorial_4.html#general-instructions-3"><i class="fa fa-check"></i>2. General Instructions</a></li>
</ul></li>
<li class="chapter" data-level="7.3" data-path="WE_4.html"><a href="WE_4.html"><i class="fa fa-check"></i><b>7.3</b> Worked Example</a>
<ul>
<li class="chapter" data-level="" data-path="WE_4.html"><a href="WE_4.html#overview-of-worked-example-3"><i class="fa fa-check"></i>1. Overview of Worked Example</a></li>
<li class="chapter" data-level="" data-path="WE_4.html"><a href="WE_4.html#spatial-distribution-of-genetic-structure"><i class="fa fa-check"></i>2. Spatial distribution of genetic structure</a></li>
<li class="chapter" data-level="" data-path="WE_4.html"><a href="WE_4.html#what-determines-genetic-differentiation-among-sites"><i class="fa fa-check"></i>3. What determines genetic differentiation among sites?</a></li>
<li class="chapter" data-level="" data-path="WE_4.html"><a href="WE_4.html#what-determines-genetic-diversity"><i class="fa fa-check"></i>4. What determines genetic diversity?</a></li>
<li class="chapter" data-level="" data-path="WE_4.html"><a href="WE_4.html#are-genetic-differentiation-and-diversity-related"><i class="fa fa-check"></i>5. Are genetic differentiation and diversity related?</a></li>
<li class="chapter" data-level="" data-path="WE_4.html"><a href="WE_4.html#effect-of-recent-extinction-events"><i class="fa fa-check"></i>6. Effect of recent extinction events</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="r-exercise-week-4.html"><a href="r-exercise-week-4.html"><i class="fa fa-check"></i>R Exercise Week 4</a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="Week5.html"><a href="Week5.html"><i class="fa fa-check"></i><b>8</b> Week 5: Spatial Statistics</a>
<ul>
<li class="chapter" data-level="8.1" data-path="video_5.html"><a href="video_5.html"><i class="fa fa-check"></i><b>8.1</b> View Course Video</a>
<ul>
<li class="chapter" data-level="" data-path="video_5.html"><a href="video_5.html#embedded-video-4"><i class="fa fa-check"></i>1. Embedded Video</a></li>
<li class="chapter" data-level="" data-path="video_5.html"><a href="video_5.html#preview-slides-4"><i class="fa fa-check"></i>Preview Slides</a></li>
</ul></li>
<li class="chapter" data-level="8.2" data-path="tutorial_5.html"><a href="tutorial_5.html"><i class="fa fa-check"></i><b>8.2</b> Interactive Tutorial 5</a>
<ul>
<li class="chapter" data-level="" data-path="tutorial_5.html"><a href="tutorial_5.html#list-of-r-commands-covered-this-week-4"><i class="fa fa-check"></i>1. List of R commands covered this week</a></li>
<li class="chapter" data-level="" data-path="tutorial_5.html"><a href="tutorial_5.html#general-instructions-4"><i class="fa fa-check"></i>2. General Instructions</a></li>
</ul></li>
<li class="chapter" data-level="8.3" data-path="WE_5.html"><a href="WE_5.html"><i class="fa fa-check"></i><b>8.3</b> Worked Example</a>
<ul>
<li class="chapter" data-level="" data-path="WE_5.html"><a href="WE_5.html#overview-of-worked-example-4"><i class="fa fa-check"></i>1. Overview of Worked Example</a></li>
<li class="chapter" data-level="" data-path="WE_5.html"><a href="WE_5.html#data-import-and-manipulation"><i class="fa fa-check"></i>2. Data import and manipulation</a></li>
<li class="chapter" data-level="" data-path="WE_5.html"><a href="WE_5.html#calculate-genetic-distances"><i class="fa fa-check"></i>3. Calculate genetic distances</a></li>
<li class="chapter" data-level="" data-path="WE_5.html"><a href="WE_5.html#perform-a-mantel-test-to-test-for-ibd"><i class="fa fa-check"></i>4. Perform a Mantel test to test for IBD</a></li>
<li class="chapter" data-level="" data-path="WE_5.html"><a href="WE_5.html#create-mantel-correlogram-for-genetic-data"><i class="fa fa-check"></i>5. Create Mantel correlogram for genetic data</a></li>
<li class="chapter" data-level="" data-path="WE_5.html"><a href="WE_5.html#specify-spatial-weights-and-calculate-morans-i"><i class="fa fa-check"></i>6. Specify spatial weights and calculate Moran’s I</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="r-exercise-week-5.html"><a href="r-exercise-week-5.html"><i class="fa fa-check"></i>R Exercise Week 5</a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="Week6.html"><a href="Week6.html"><i class="fa fa-check"></i><b>9</b> Week 6: Quantitative Genetics</a>
<ul>
<li class="chapter" data-level="9.1" data-path="video_6.html"><a href="video_6.html"><i class="fa fa-check"></i><b>9.1</b> View Course Video</a>
<ul>
<li class="chapter" data-level="" data-path="video_6.html"><a href="video_6.html#embedded-video-5"><i class="fa fa-check"></i>1. Embedded Video</a></li>
<li class="chapter" data-level="" data-path="video_6.html"><a href="video_6.html#video-part-1-2"><i class="fa fa-check"></i>Video, Part 1</a></li>
<li class="chapter" data-level="" data-path="video_6.html"><a href="video_6.html#video-part-2-2"><i class="fa fa-check"></i>Video, Part 2</a></li>
<li class="chapter" data-level="" data-path="video_6.html"><a href="video_6.html#preview-slides-5"><i class="fa fa-check"></i>Preview Slides</a></li>
</ul></li>
<li class="chapter" data-level="9.2" data-path="tutorial_6.html"><a href="tutorial_6.html"><i class="fa fa-check"></i><b>9.2</b> Interactive Tutorial 6</a>
<ul>
<li class="chapter" data-level="" data-path="tutorial_6.html"><a href="tutorial_6.html#list-of-r-commands-covered-this-week-5"><i class="fa fa-check"></i>1. List of R commands covered this week</a></li>
<li class="chapter" data-level="" data-path="tutorial_6.html"><a href="tutorial_6.html#general-instructions-5"><i class="fa fa-check"></i>2. General Instructions</a></li>
</ul></li>
<li class="chapter" data-level="9.3" data-path="WE_6.html"><a href="WE_6.html"><i class="fa fa-check"></i><b>9.3</b> Worked Example</a>
<ul>
<li class="chapter" data-level="" data-path="WE_6.html"><a href="WE_6.html#overview-of-worked-example-5"><i class="fa fa-check"></i>1. Overview of Worked Example</a></li>
<li class="chapter" data-level="" data-path="WE_6.html"><a href="WE_6.html#estimate-genetic-and-non-genetic-variance-components-from-a-common-garden-experiment"><i class="fa fa-check"></i>2. Estimate genetic and non-genetic variance components from a common garden experiment</a></li>
<li class="chapter" data-level="" data-path="WE_6.html"><a href="WE_6.html#estimate-trait-heritability"><i class="fa fa-check"></i>3. Estimate trait heritability</a></li>
<li class="chapter" data-level="" data-path="WE_6.html"><a href="WE_6.html#compare-q_st-to-f_st-measured-from-snp-data"><i class="fa fa-check"></i>4. Compare <span class="math inline">\(Q_{st}\)</span> to <span class="math inline">\(F_{st}\)</span> measured from SNP data</a></li>
<li class="chapter" data-level="" data-path="WE_6.html"><a href="WE_6.html#assess-correlation-between-trait-and-environment"><i class="fa fa-check"></i>5. Assess correlation between trait and environment</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="r-exercise-week-6.html"><a href="r-exercise-week-6.html"><i class="fa fa-check"></i>R Exercise Week 6</a></li>
</ul></li>
<li class="chapter" data-level="10" data-path="Week7.html"><a href="Week7.html"><i class="fa fa-check"></i><b>10</b> Week 7: Spatial Linear Models</a>
<ul>
<li class="chapter" data-level="10.1" data-path="video_7.html"><a href="video_7.html"><i class="fa fa-check"></i><b>10.1</b> View Course Video</a>
<ul>
<li class="chapter" data-level="" data-path="video_7.html"><a href="video_7.html#embedded-video-6"><i class="fa fa-check"></i>1. Embedded Video</a></li>
<li class="chapter" data-level="" data-path="video_7.html"><a href="video_7.html#video-part-1-3"><i class="fa fa-check"></i>Video, Part 1</a></li>
<li class="chapter" data-level="" data-path="video_7.html"><a href="video_7.html#video-part-2-3"><i class="fa fa-check"></i>Video, Part 2</a></li>
<li class="chapter" data-level="" data-path="video_7.html"><a href="video_7.html#preview-slides-6"><i class="fa fa-check"></i>Preview Slides</a></li>
</ul></li>
<li class="chapter" data-level="10.2" data-path="tutorial_7.html"><a href="tutorial_7.html"><i class="fa fa-check"></i><b>10.2</b> Interactive Tutorial 7</a>
<ul>
<li class="chapter" data-level="" data-path="tutorial_7.html"><a href="tutorial_7.html#list-of-r-commands-covered-this-week-6"><i class="fa fa-check"></i>1. List of R commands covered this week</a></li>
<li class="chapter" data-level="" data-path="tutorial_7.html"><a href="tutorial_7.html#general-instructions-6"><i class="fa fa-check"></i>2. General Instructions</a></li>
</ul></li>
<li class="chapter" data-level="10.3" data-path="WE_7.html"><a href="WE_7.html"><i class="fa fa-check"></i><b>10.3</b> Worked Example</a>
<ul>
<li class="chapter" data-level="" data-path="WE_7.html"><a href="WE_7.html#overview-of-worked-example-6"><i class="fa fa-check"></i>1. Overview of Worked Example</a></li>
<li class="chapter" data-level="" data-path="WE_7.html"><a href="WE_7.html#explore-data-set"><i class="fa fa-check"></i>2. Explore data set</a></li>
<li class="chapter" data-level="" data-path="WE_7.html"><a href="WE_7.html#test-regression-residuals-for-spatial-autocorrelation"><i class="fa fa-check"></i>3. Test regression residuals for spatial autocorrelation</a></li>
<li class="chapter" data-level="" data-path="WE_7.html"><a href="WE_7.html#fit-models-with-spatially-correlated-error-gls-with-package-nlme"><i class="fa fa-check"></i>4. Fit models with spatially correlated error (GLS) with package ‘nlme’</a></li>
<li class="chapter" data-level="" data-path="WE_7.html"><a href="WE_7.html#fit-spatial-simultaneous-autoregressive-error-models-sar"><i class="fa fa-check"></i>5. Fit spatial simultaneous autoregressive error models (SAR)</a></li>
<li class="chapter" data-level="" data-path="WE_7.html"><a href="WE_7.html#spatial-filtering-with-mem-using-package-spmoran"><i class="fa fa-check"></i>6. Spatial filtering with MEM using package ‘spmoran’</a></li>
<li class="chapter" data-level="" data-path="WE_7.html"><a href="WE_7.html#fit-spatially-varying-coefficients-model-with-package-spmoran"><i class="fa fa-check"></i>7. Fit spatially varying coefficients model with package ‘spmoran’</a></li>
<li class="chapter" data-level="" data-path="WE_7.html"><a href="WE_7.html#model-with-patchsize-and-ibr"><i class="fa fa-check"></i>Model with PatchSize and IBR</a></li>
<li class="chapter" data-level="" data-path="WE_7.html"><a href="WE_7.html#model-with-ibr-only"><i class="fa fa-check"></i>Model with IBR only</a></li>
<li class="chapter" data-level="" data-path="WE_7.html"><a href="WE_7.html#conclusions"><i class="fa fa-check"></i>8. Conclusions</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="r-exercise-week-7.html"><a href="r-exercise-week-7.html"><i class="fa fa-check"></i>R Exercise Week 7</a></li>
<li class="chapter" data-level="10.4" data-path="bonus_7a.html"><a href="bonus_7a.html"><i class="fa fa-check"></i><b>10.4</b> Bonus: Si index</a>
<ul>
<li class="chapter" data-level="" data-path="bonus_7a.html"><a href="bonus_7a.html#overview-of-bonus-materials"><i class="fa fa-check"></i>1. Overview of Bonus Materials</a></li>
<li class="chapter" data-level="" data-path="bonus_7a.html"><a href="bonus_7a.html#import-ecological-distance-matrices"><i class="fa fa-check"></i>2. Import ecological distance matrices</a></li>
<li class="chapter" data-level="" data-path="bonus_7a.html"><a href="bonus_7a.html#optimize-the-scaling-parameter-alpha"><i class="fa fa-check"></i>3. Optimize the scaling parameter alpha</a></li>
<li class="chapter" data-level="" data-path="bonus_7a.html"><a href="bonus_7a.html#calculate-hanskis-index-si-with-source-patch-parameters"><i class="fa fa-check"></i>4. Calculate Hanski’s index Si with source patch parameters</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="11" data-path="Week8.html"><a href="Week8.html"><i class="fa fa-check"></i><b>11</b> Week 8: Simulation Experiments</a>
<ul>
<li class="chapter" data-level="11.1" data-path="video_8.html"><a href="video_8.html"><i class="fa fa-check"></i><b>11.1</b> View Course Video</a>
<ul>
<li class="chapter" data-level="" data-path="video_8.html"><a href="video_8.html#embedded-video-7"><i class="fa fa-check"></i>1. Embedded Video</a></li>
<li class="chapter" data-level="" data-path="video_8.html"><a href="video_8.html#video-part-1-4"><i class="fa fa-check"></i>Video, Part 1</a></li>
<li class="chapter" data-level="" data-path="video_8.html"><a href="video_8.html#video-part-2-4"><i class="fa fa-check"></i>Video, Part 2</a></li>
<li class="chapter" data-level="" data-path="video_8.html"><a href="video_8.html#preview-slides-7"><i class="fa fa-check"></i>Preview Slides</a></li>
</ul></li>
<li class="chapter" data-level="11.2" data-path="tutorial_8.html"><a href="tutorial_8.html"><i class="fa fa-check"></i><b>11.2</b> Interactive Tutorial 8</a>
<ul>
<li class="chapter" data-level="" data-path="tutorial_8.html"><a href="tutorial_8.html#list-of-r-commands-covered-this-week-7"><i class="fa fa-check"></i>1. List of R commands covered this week</a></li>
<li class="chapter" data-level="" data-path="tutorial_8.html"><a href="tutorial_8.html#general-instructions-7"><i class="fa fa-check"></i>2. General Instructions</a></li>
</ul></li>
<li class="chapter" data-level="11.3" data-path="WE_8.html"><a href="WE_8.html"><i class="fa fa-check"></i><b>11.3</b> Worked Example</a>
<ul>
<li class="chapter" data-level="" data-path="WE_8.html"><a href="WE_8.html#overview-of-worked-example-7"><i class="fa fa-check"></i>1. Overview of Worked Example</a></li>
<li class="chapter" data-level="" data-path="WE_8.html"><a href="WE_8.html#initialize-a-landscape"><i class="fa fa-check"></i>2. Initialize a landscape</a></li>
<li class="chapter" data-level="" data-path="WE_8.html"><a href="WE_8.html#define-simulation-parameters"><i class="fa fa-check"></i>3. Define simulation parameters</a></li>
<li class="chapter" data-level="" data-path="WE_8.html"><a href="WE_8.html#run-simulations-and-analyze-results"><i class="fa fa-check"></i>4. Run simulations and analyze results</a></li>
<li class="chapter" data-level="" data-path="WE_8.html"><a href="WE_8.html#run-simulator-using-a-previously-defined-parameter-set"><i class="fa fa-check"></i>5. Run simulator using a previously defined parameter set</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="r-exercise-week-8.html"><a href="r-exercise-week-8.html"><i class="fa fa-check"></i>R Exercise Week 8</a></li>
<li class="chapter" data-level="11.4" data-path="bonus_8a.html"><a href="bonus_8a.html"><i class="fa fa-check"></i><b>11.4</b> Bonus: Efficient R</a></li>
</ul></li>
<li class="part"><span><b>Advanced Topics</b></span></li>
<li class="chapter" data-level="12" data-path="Week9.html"><a href="Week9.html"><i class="fa fa-check"></i><b>12</b> Week 9: Population Structure</a>
<ul>
<li class="chapter" data-level="12.1" data-path="WE_9.html"><a href="WE_9.html"><i class="fa fa-check"></i><b>12.1</b> Worked Example</a>
<ul>
<li class="chapter" data-level="" data-path="WE_9.html"><a href="WE_9.html#overview-of-worked-example-8"><i class="fa fa-check"></i>1. Overview of Worked Example</a></li>
<li class="chapter" data-level="" data-path="WE_9.html"><a href="WE_9.html#simulated-data-2-island-model"><i class="fa fa-check"></i>2. Simulated data: 2-island model</a></li>
<li class="chapter" data-level="" data-path="WE_9.html"><a href="WE_9.html#simulated-data-2-island-model-with-admixture"><i class="fa fa-check"></i>3. Simulated data: 2-island model with admixture</a></li>
<li class="chapter" data-level="" data-path="WE_9.html"><a href="WE_9.html#empirical-data-threespine-sticklebacks"><i class="fa fa-check"></i>4. Empirical data: Threespine sticklebacks</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="13" data-path="Week10.html"><a href="Week10.html"><i class="fa fa-check"></i><b>13</b> Week 10: Landscape Resistance</a>
<ul>
<li class="chapter" data-level="13.1" data-path="WE_10.html"><a href="WE_10.html"><i class="fa fa-check"></i><b>13.1</b> Worked Example</a>
<ul>
<li class="chapter" data-level="" data-path="WE_10.html"><a href="WE_10.html#overview-of-worked-example-9"><i class="fa fa-check"></i>1. Overview of Worked Example</a></li>
<li class="chapter" data-level="" data-path="WE_10.html"><a href="WE_10.html#explore-the-data-set"><i class="fa fa-check"></i>2. Explore the data set</a></li>
<li class="chapter" data-level="" data-path="WE_10.html"><a href="WE_10.html#setting-cost-values-and-calculating-conductance"><i class="fa fa-check"></i>3. Setting cost values and calculating conductance</a></li>
<li class="chapter" data-level="" data-path="WE_10.html"><a href="WE_10.html#convert-conductance-into-effective-distance"><i class="fa fa-check"></i>4. Convert conductance into effective distance</a></li>
<li class="chapter" data-level="" data-path="WE_10.html"><a href="WE_10.html#create-cost-distance-matrices"><i class="fa fa-check"></i>5. Create cost-distance matrices</a></li>
<li class="chapter" data-level="" data-path="WE_10.html"><a href="WE_10.html#how-does-changing-resolution-affect-these-metrics"><i class="fa fa-check"></i>6. How does changing resolution affect these metrics?</a></li>
<li class="chapter" data-level="" data-path="WE_10.html"><a href="WE_10.html#references"><i class="fa fa-check"></i>7. References</a></li>
</ul></li>
<li class="chapter" data-level="13.2" data-path="bonus-10a.html"><a href="bonus-10a.html"><i class="fa fa-check"></i><b>13.2</b> Bonus: ‘radish’ tutorial</a>
<ul>
<li class="chapter" data-level="" data-path="bonus-10a.html"><a href="bonus-10a.html#overview-of-bonus-materials-1"><i class="fa fa-check"></i>1. Overview of Bonus Materials</a></li>
<li class="chapter" data-level="" data-path="bonus-10a.html"><a href="bonus-10a.html#import-data"><i class="fa fa-check"></i>2. Import data</a></li>
<li class="chapter" data-level="" data-path="bonus-10a.html"><a href="bonus-10a.html#basic-model-fitting"><i class="fa fa-check"></i>3. Basic model fitting</a></li>
<li class="chapter" data-level="" data-path="bonus-10a.html"><a href="bonus-10a.html#advanced-modeling-topics"><i class="fa fa-check"></i>4. Advanced modeling topics</a></li>
<li class="chapter" data-level="" data-path="bonus-10a.html"><a href="bonus-10a.html#summary"><i class="fa fa-check"></i>5. Summary</a></li>
<li class="chapter" data-level="" data-path="bonus-10a.html"><a href="bonus-10a.html#references-1"><i class="fa fa-check"></i>6. References</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="14" data-path="Week11.html"><a href="Week11.html"><i class="fa fa-check"></i><b>14</b> Week 11: Detecting Adaptation</a>
<ul>
<li class="chapter" data-level="14.1" data-path="WE_11.html"><a href="WE_11.html"><i class="fa fa-check"></i><b>14.1</b> Worked Example</a>
<ul>
<li class="chapter" data-level="" data-path="WE_11.html"><a href="WE_11.html#overview-of-worked-example-10"><i class="fa fa-check"></i>1. Overview of Worked Example</a></li>
<li class="chapter" data-level="" data-path="WE_11.html"><a href="WE_11.html#import-and-prepare-the-data"><i class="fa fa-check"></i>2. Import and prepare the data</a></li>
<li class="chapter" data-level="" data-path="WE_11.html"><a href="WE_11.html#latent-factor-mixed-models-lfmm-a-univariate-gea"><i class="fa fa-check"></i>3. Latent Factor Mixed Models (LFMM): a univariate GEA</a></li>
<li class="chapter" data-level="" data-path="WE_11.html"><a href="WE_11.html#redundancy-analysis-rda-a-multivariate-gea"><i class="fa fa-check"></i>4. Redundancy Analysis (RDA): a multivariate GEA</a></li>
<li class="chapter" data-level="" data-path="WE_11.html"><a href="WE_11.html#compare-lfmm-and-rda-candidates"><i class="fa fa-check"></i>5. Compare LFMM and RDA candidates</a></li>
<li class="chapter" data-level="" data-path="WE_11.html"><a href="WE_11.html#a-quick-note-on-controlling-for-population-structure-in-rda"><i class="fa fa-check"></i>6. A quick note on controlling for population structure in RDA</a></li>
<li class="chapter" data-level="" data-path="WE_11.html"><a href="WE_11.html#references-2"><i class="fa fa-check"></i>7. References</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="15" data-path="Week12.html"><a href="Week12.html"><i class="fa fa-check"></i><b>15</b> Week 12: Model Selection</a>
<ul>
<li class="chapter" data-level="15.1" data-path="WE_12.html"><a href="WE_12.html"><i class="fa fa-check"></i><b>15.1</b> Worked Example</a>
<ul>
<li class="chapter" data-level="" data-path="WE_12.html"><a href="WE_12.html#overview-of-worked-example-11"><i class="fa fa-check"></i>1. Overview of Worked Example</a></li>
<li class="chapter" data-level="" data-path="WE_12.html"><a href="WE_12.html#fitting-candidate-models"><i class="fa fa-check"></i>2. Fitting candidate models</a></li>
<li class="chapter" data-level="" data-path="WE_12.html"><a href="WE_12.html#your-turn-to-test-additional-hypotheses"><i class="fa fa-check"></i>3. Your turn to test additional hypotheses!</a></li>
<li class="chapter" data-level="" data-path="WE_12.html"><a href="WE_12.html#nested-model-nmlpe-for-hierarchical-sampling-designs"><i class="fa fa-check"></i>4. Nested model (NMLPE) for hierarchical sampling designs</a></li>
<li class="chapter" data-level="" data-path="WE_12.html"><a href="WE_12.html#references-3"><i class="fa fa-check"></i>5. References</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="16" data-path="Week13.html"><a href="Week13.html"><i class="fa fa-check"></i><b>16</b> Week 13: Gravity Models</a>
<ul>
<li class="chapter" data-level="16.1" data-path="WE_13.html"><a href="WE_13.html"><i class="fa fa-check"></i><b>16.1</b> Worked Example</a>
<ul>
<li class="chapter" data-level="" data-path="WE_13.html"><a href="WE_13.html#overview-of-worked-example-12"><i class="fa fa-check"></i>1. Overview of Worked Example</a></li>
<li class="chapter" data-level="" data-path="WE_13.html"><a href="WE_13.html#setup"><i class="fa fa-check"></i>2. Setup</a></li>
<li class="chapter" data-level="" data-path="WE_13.html"><a href="WE_13.html#wetland-complex-data-preparation"><i class="fa fa-check"></i>3. Wetland complex data preparation</a></li>
<li class="chapter" data-level="" data-path="WE_13.html"><a href="WE_13.html#wetland-field-data-preparation"><i class="fa fa-check"></i>4. Wetland field-data preparation</a></li>
<li class="chapter" data-level="" data-path="WE_13.html"><a href="WE_13.html#saturated-graph"><i class="fa fa-check"></i>5. Saturated Graph</a></li>
<li class="chapter" data-level="" data-path="WE_13.html"><a href="WE_13.html#spatial-model-data-prepration"><i class="fa fa-check"></i>6. Spatial model data prepration</a></li>
<li class="chapter" data-level="" data-path="WE_13.html"><a href="WE_13.html#gravity-model"><i class="fa fa-check"></i>7. Gravity model</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="17" data-path="Week14.html"><a href="Week14.html"><i class="fa fa-check"></i><b>17</b> Week 14: Contemporary Gene Flow</a>
<ul>
<li class="chapter" data-level="17.1" data-path="WE_14.html"><a href="WE_14.html"><i class="fa fa-check"></i><b>17.1</b> Worked Example</a>
<ul>
<li class="chapter" data-level="" data-path="WE_14.html"><a href="WE_14.html#overview-3"><i class="fa fa-check"></i>1. Overview</a></li>
<li class="chapter" data-level="" data-path="WE_14.html"><a href="WE_14.html#import-genotypes"><i class="fa fa-check"></i>2. Import Genotypes</a></li>
<li class="chapter" data-level="" data-path="WE_14.html"><a href="WE_14.html#pollen-pool-genetic-structure"><i class="fa fa-check"></i>3. Pollen pool genetic structure</a></li>
<li class="chapter" data-level="" data-path="WE_14.html"><a href="WE_14.html#paternity-analysis"><i class="fa fa-check"></i>4. Paternity analysis</a></li>
<li class="chapter" data-level="" data-path="WE_14.html"><a href="WE_14.html#linking-paternity-to-ecological-variables"><i class="fa fa-check"></i>5. Linking paternity to ecological variables</a></li>
<li class="chapter" data-level="" data-path="WE_14.html"><a href="WE_14.html#references-4"><i class="fa fa-check"></i>6. References</a></li>
</ul></li>
</ul></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Landscape Genetic Data Analysis with R</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="WE_12" class="section level2 hasAnchor" number="15.1">
<h2><span class="header-section-number">15.1</span> Worked Example<a href="WE_12.html#WE_12" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Caren Goldberg and Helene Wagner</p>
<div id="overview-of-worked-example-11" class="section level3 unnumbered hasAnchor">
<h3>1. Overview of Worked Example<a href="WE_12.html#overview-of-worked-example-11" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div id="a.-goals-and-background" class="section level4 unnumbered hasAnchor">
<h4>a. Goals and Background<a href="WE_12.html#a.-goals-and-background" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p><strong>Goals</strong>: The goal of this worked example is for you to become familiar with creating and analyzing landscape genetic hypotheses using maximum likelihood population effect models (MLPE; Van Strien et al. 2012) and information theory. With this approach, we focus on evaluating evidence for the influence of the (intervening) landscape on link-based metrics (i.e. genetic distance). Incorporating node-based effects (i.e. pond size, etc.) will be covered in the gravity modeling lab.</p>
<p>We will re-analyze a dataset from Goldberg and Waits (2010) using a maximum likelihood population effect model (MLPE).</p>
<p>Here’s a map of the study area. The lines represent those pair-wise links between sampling locations that we will be considering here (graph model: Delaunay triangulation).</p>
<div class="float">
<img src="vignettes/WE12_Fig1.PNG" alt="Map of study area" />
<div class="figcaption">Map of study area</div>
</div>
</div>
<div id="b.-data-set-10" class="section level4 unnumbered hasAnchor">
<h4>b. Data set<a href="WE_12.html#b.-data-set-10" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>In previous labs, you have created genetic distance matrices, so here we are going to start with those data already complete, as are the landscape data. For MLPE, two vectors representing the nodes at the ends of each link must be created. These are already included in the data set (<code>pop1</code> and <code>pop2</code>).</p>
<p>Each row in the file ‘CSF_network.csv’ represents a link between two pairs of sampling locations. The columns contain the following variables:</p>
<ul>
<li><strong>Id</strong>: Link ID.</li>
<li><strong>Name</strong>: Link name (combines the two site names).</li>
<li><strong>logDc.km</strong>: Genetic distance per geographic distance, log transformed (response variable).</li>
<li><strong>LENGTH</strong>: Euclidean distance.</li>
<li><strong>slope</strong>: Average slope along the link.</li>
<li><strong>solarinso</strong>: Average solar insolation along the link.</li>
<li><strong>soils</strong>: Dominant soil type along the link (categorical).</li>
<li><strong>ag</strong>: Proportion of agriculture along the link.</li>
<li><strong>grass</strong>: Proportion of grassland along the link.</li>
<li><strong>shrub</strong>: Proportion of shrubland along the link.</li>
<li><strong>hi</strong>: Proportion of high density forest along the link.</li>
<li><strong>lo</strong>: Proportion of low density forest along the link.</li>
<li><strong>dev</strong>: Proportion of development (buildings) along the link.</li>
<li><strong>forest</strong>: Proportion of forest (the sum of hi and lo) along the link.</li>
<li><strong>pop1</strong>: ‘From’ population.</li>
<li><strong>pop2</strong>: ‘To’ population.</li>
</ul>
</div>
<div id="c.-required-packages" class="section level4 unnumbered hasAnchor">
<h4>c. Required packages<a href="WE_12.html#c.-required-packages" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>Install corMLPE if it is not yet installed.</p>
<div class="sourceCode" id="cb1290"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1290-1"><a href="WE_12.html#cb1290-1" tabindex="-1"></a><span class="cf">if</span>(<span class="sc">!</span><span class="fu">requireNamespace</span>(<span class="st">&quot;corMLPE&quot;</span>, <span class="at">quietly =</span> <span class="cn">TRUE</span>)) </span>
<span id="cb1290-2"><a href="WE_12.html#cb1290-2" tabindex="-1"></a>  remotes<span class="sc">::</span><span class="fu">install_github</span>(<span class="st">&quot;nspope/corMLPE&quot;</span>)</span></code></pre></div>
<div class="sourceCode" id="cb1291"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1291-1"><a href="WE_12.html#cb1291-1" tabindex="-1"></a><span class="fu">library</span>(LandGenCourse)</span>
<span id="cb1291-2"><a href="WE_12.html#cb1291-2" tabindex="-1"></a><span class="fu">library</span>(ggplot2)</span>
<span id="cb1291-3"><a href="WE_12.html#cb1291-3" tabindex="-1"></a><span class="co">#library(nlme)</span></span>
<span id="cb1291-4"><a href="WE_12.html#cb1291-4" tabindex="-1"></a><span class="co">#library(usdm)</span></span>
<span id="cb1291-5"><a href="WE_12.html#cb1291-5" tabindex="-1"></a><span class="co">#library(corMLPE)</span></span></code></pre></div>
</div>
<div id="d.-import-data-1" class="section level4 unnumbered hasAnchor">
<h4>d. Import data<a href="WE_12.html#d.-import-data-1" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>First, read in the dataset. It is named CSF (Columbia spotted frog) to differentiate it from the RALU dataset you have used in previous labs (same species, very different landscapes).</p>
<div class="sourceCode" id="cb1292"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1292-1"><a href="WE_12.html#cb1292-1" tabindex="-1"></a>CSFdata <span class="ot">&lt;-</span> <span class="fu">read.csv</span>(<span class="fu">system.file</span>(<span class="st">&quot;extdata&quot;</span>, <span class="st">&quot;CSF_network.csv&quot;</span>, </span>
<span id="cb1292-2"><a href="WE_12.html#cb1292-2" tabindex="-1"></a>                                        <span class="at">package =</span> <span class="st">&quot;LandGenCourse&quot;</span>))</span>
<span id="cb1292-3"><a href="WE_12.html#cb1292-3" tabindex="-1"></a><span class="fu">head</span>(CSFdata) </span></code></pre></div>
<pre><code>##   Id         Name  logDc.km   LENGTH     slope solarinso    soils         ag
## 1  1 CeGr to FeCr -5.664895 5449.643 11.328851  16332.75 thirteen 0.00000000
## 2  2 CeGr to FlCr -5.786712 8352.643 12.822890  16142.01      one 0.01351351
## 3 43 CeGr to Krum -5.466544 7232.601 15.852762  16208.67      one 0.00000000
## 4  3 FlCr to Krum -5.473660 6196.521 14.156965  15871.14   twelve 0.00000000
## 5 44  Krum to P37 -5.088242 4470.826 15.381000  15883.31 thirteen 0.00000000
## 6 46 CeGr to LaTr -5.997768 8766.482  9.709769  16298.75     one5 0.00000000
##        grass      shrub        hi        lo        dev    forest pop1 pop2
## 1 0.00000000 0.55319149 0.2553191 0.1914894 0.00000000 0.4468085    1    2
## 2 0.01351351 0.10810811 0.6216216 0.2432432 0.00000000 0.8648649    1    3
## 3 0.00000000 0.19696970 0.6212121 0.1818182 0.00000000 0.8030303    4    1
## 4 0.00000000 0.05084746 0.5932203 0.3559322 0.00000000 0.9491525    3    4
## 5 0.05000000 0.00000000 0.3250000 0.6250000 0.00000000 0.9500000   12    4
## 6 0.20000000 0.21333333 0.2666667 0.3066667 0.01333333 0.5733333    5    1</code></pre>
<p>The variable solarinso is on a very different scale, which can create problems with model fitting with some methods. To avoid this, we scale it.</p>
<p>Note: we could use the scale function without specifying the arguments ‘center’ or ‘scale’, as both are ‘TRUE’ by default. The argument ‘center=TRUE’ centers the variable by subtracting the mean, the ‘scale=TRUE’ argument rescales the variables to unit variance by dividing each value by the standard deviation. The resulting variable has a mean of zero and a standard deviation and variance of one.</p>
<div class="sourceCode" id="cb1294"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1294-1"><a href="WE_12.html#cb1294-1" tabindex="-1"></a>CSFdata<span class="sc">$</span>solarinsoz <span class="ot">&lt;-</span> <span class="fu">scale</span>(CSFdata<span class="sc">$</span>solarinso, <span class="at">center=</span><span class="cn">TRUE</span>, <span class="at">scale=</span><span class="cn">TRUE</span>)</span></code></pre></div>
<p>Take a look at the column names and make sure you and R agree on what everything is called, and on the data types (e.g., factor vs. character).</p>
<div class="sourceCode" id="cb1295"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1295-1"><a href="WE_12.html#cb1295-1" tabindex="-1"></a><span class="fu">str</span>(CSFdata)</span></code></pre></div>
<pre><code>## &#39;data.frame&#39;:    48 obs. of  17 variables:
##  $ Id        : int  1 2 43 3 44 46 4 6 7 31 ...
##  $ Name      : chr  &quot;CeGr to FeCr&quot; &quot;CeGr to FlCr&quot; &quot;CeGr to Krum&quot; &quot;FlCr to Krum&quot; ...
##  $ logDc.km  : num  -5.66 -5.79 -5.47 -5.47 -5.09 ...
##  $ LENGTH    : num  5450 8353 7233 6197 4471 ...
##  $ slope     : num  11.3 12.8 15.9 14.2 15.4 ...
##  $ solarinso : num  16333 16142 16209 15871 15883 ...
##  $ soils     : chr  &quot;thirteen&quot; &quot;one&quot; &quot;one&quot; &quot;twelve&quot; ...
##  $ ag        : num  0 0.0135 0 0 0 ...
##  $ grass     : num  0 0.0135 0 0 0.05 ...
##  $ shrub     : num  0.5532 0.1081 0.197 0.0508 0 ...
##  $ hi        : num  0.255 0.622 0.621 0.593 0.325 ...
##  $ lo        : num  0.191 0.243 0.182 0.356 0.625 ...
##  $ dev       : num  0 0 0 0 0 ...
##  $ forest    : num  0.447 0.865 0.803 0.949 0.95 ...
##  $ pop1      : int  1 1 4 3 12 5 2 1 2 4 ...
##  $ pop2      : int  2 3 1 4 4 1 5 6 6 7 ...
##  $ solarinsoz: num [1:48, 1] 0.796 -0.28 0.096 -1.808 -1.74 ...
##   ..- attr(*, &quot;scaled:center&quot;)= num 16192
##   ..- attr(*, &quot;scaled:scale&quot;)= num 177</code></pre>
<ul>
<li><code>Name</code> and <code>soils</code> are interpreted as factors,</li>
<li><code>Id</code>, <code>pop1</code> and <code>pop2</code> are vectors of integers, and</li>
<li>everything else is a numeric vector, i.e., continuous variable.</li>
</ul>
<p>The response Y in our MLPE models will be <code>logDc.km</code>, the genetic distance per geographic distance, log transformed to meet normality assumptions. This was used in the paper as a way to include IBD as the base assumption without increasing k. The landscape variables were then also estimated so that they would not necessarily increase with increasing distance between sites (e.g., proportion of forest between sites).</p>
<p><strong>This is a pruned network, where the links are only those neighbors in a Delauney trangulation. &lt;Capture.PNG&gt;</strong></p>
<p>For this exercise, you will test 5 alternative hypotheses. At the end, you will be invited to create your own hypotheses from these variables and test support for them.</p>
</div>
</div>
<div id="fitting-candidate-models" class="section level3 unnumbered hasAnchor">
<h3>2. Fitting candidate models<a href="WE_12.html#fitting-candidate-models" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div id="a.-alternative-hypotheses" class="section level4 unnumbered hasAnchor">
<h4>a. Alternative hypotheses<a href="WE_12.html#a.-alternative-hypotheses" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>In information theory, we evaluate evidence for a candidate set of hypotheses that we develop from the literature and theory. Although it is mathematically possible (and often practiced) to fit all possible combinations of input variables, that approach is not consistent with the methodology (see Burnham and Anderson 2002 Chapter 8, the reading for this unit, for more detail). Here, your “a priori”” set of hypotheses (as in the Week 12 Conceptual Exercise) is as follows:</p>
<ol style="list-style-type: decimal">
<li><strong>Full model</strong>: solarinso, forest, ag, shrub, dev</li>
<li><strong>Landcover model</strong>: ag, shrub, forest, and dev</li>
<li><strong>Human footprint model</strong>: ag, dev</li>
<li><strong>Energy conservation model</strong>: slope, shrub, dev</li>
<li><strong>Historical model</strong>: soils, slope, solarinso</li>
</ol>
</div>
<div id="b.-check-for-multicollinearity" class="section level4 unnumbered hasAnchor">
<h4>b. Check for multicollinearity<a href="WE_12.html#b.-check-for-multicollinearity" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>Before we add predictor variables to a model, we should make sure that these variables do not have issues with multicollinearity, i.e., that they are not highly correlated among themselves. To this end, we calculate variance inflation factors, VIF.</p>
<p>Make a dataframe of just the variables to test (note, you cannot use factors here):</p>
<div class="sourceCode" id="cb1297"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1297-1"><a href="WE_12.html#cb1297-1" tabindex="-1"></a>CSF.df <span class="ot">&lt;-</span> <span class="fu">with</span>(CSFdata, <span class="fu">data.frame</span>(solarinsoz, forest, dev, shrub, ag))</span>
<span id="cb1297-2"><a href="WE_12.html#cb1297-2" tabindex="-1"></a>usdm<span class="sc">::</span><span class="fu">vif</span>(CSF.df)</span></code></pre></div>
<pre><code>##    Variables      VIF
## 1 solarinsoz 1.479273
## 2     forest 3.127104
## 3        dev 1.800218
## 4      shrub 1.523341
## 5         ag 2.532198</code></pre>
<p>If you get an error that there is no package called ‘usdm’, use <code>install.packages("usdm")</code>.</p>
<p>VIF values less than 10, or 3, or 4, depending on who you ask, are considered to not be collinear enough to affect model outcomes. So we are good to go here. What would happen if we added grass to this?</p>
<div class="sourceCode" id="cb1299"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1299-1"><a href="WE_12.html#cb1299-1" tabindex="-1"></a>usdm<span class="sc">::</span><span class="fu">vif</span>(<span class="fu">cbind</span>(CSF.df, <span class="at">grass=</span>CSFdata<span class="sc">$</span>grass))</span></code></pre></div>
<pre><code>##    Variables       VIF
## 1 solarinsoz  1.541954
## 2     forest 97.167730
## 3        dev 55.768172
## 4      shrub 11.657666
## 5         ag 68.852254
## 6      grass 45.970721</code></pre>
<p><strong>Question 1</strong>: Why would adding in the last land cover cause a high amount of collinearity? Consider how these data were calculated.</p>
<p>The variables ‘forest’, ‘dev’, ‘shrub’, ‘ag’ and ‘grass’ together cover almost all land cover types in the study area. Each is measured as percent area, and together, they make up 100% or close to 100% of the land cover types along each link. Thus if we know e.g. that all ‘forest’, ‘dev’, ‘shrub’ and ‘ag’ together cover 80% along a link, then it is a good bet that ‘grass’ will cover the remaining 20%.</p>
<p>With such sets of compositional data, we need to omit one variable to avoid multi-collinearity (where a variable is a linear combination of other variables), even though the pairwise correlations may be low. For example, there the variable ‘grass’ showed relatively low correlations with the other variables:</p>
<div class="sourceCode" id="cb1301"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1301-1"><a href="WE_12.html#cb1301-1" tabindex="-1"></a><span class="fu">cor</span>(CSF.df, CSFdata<span class="sc">$</span>grass)</span></code></pre></div>
<pre><code>##                   [,1]
## solarinsoz  0.10394523
## forest     -0.41341968
## dev        -0.23617929
## shrub      -0.29807027
## ag          0.01060842</code></pre>
</div>
<div id="c.-fit-all-five-models" class="section level4 unnumbered hasAnchor">
<h4>c. Fit all five models<a href="WE_12.html#c.-fit-all-five-models" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>The R package <code>corMLPE</code> makes it easy to fit an MLPE model to our dataset. We use the function <code>gls</code> from package <code>nlme</code> to fit a “generalized least squares” model to the data.</p>
<p>This is similar to a spatial regression model we’ve seen in Week 7. The main difference is how we define the correlation structure of the errors: instead of using a function of the spatial coordinates, we use a function of the population indicators: <code>correlation=corMLPE(form=~pop1+pop2)</code>.</p>
<p>Note: for linear mixed models, see Week 6 videos. For gls models, see Week 7 videos.</p>
<div class="sourceCode" id="cb1303"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1303-1"><a href="WE_12.html#cb1303-1" tabindex="-1"></a>mod1 <span class="ot">&lt;-</span> nlme<span class="sc">::</span><span class="fu">gls</span>(logDc.km <span class="sc">~</span> solarinsoz <span class="sc">+</span> forest <span class="sc">+</span> ag <span class="sc">+</span> shrub <span class="sc">+</span> dev, </span>
<span id="cb1303-2"><a href="WE_12.html#cb1303-2" tabindex="-1"></a>                  <span class="at">correlation=</span>corMLPE<span class="sc">::</span><span class="fu">corMLPE</span>(<span class="at">form=</span><span class="sc">~</span>pop1<span class="sc">+</span>pop2), </span>
<span id="cb1303-3"><a href="WE_12.html#cb1303-3" tabindex="-1"></a>                  <span class="at">data=</span>CSFdata, <span class="at">method=</span><span class="st">&quot;REML&quot;</span>)</span></code></pre></div>
<p>This is the closest to a full model in our dataset (although our models are not completely nested), so we will take a look at the residuals.</p>
<div class="sourceCode" id="cb1304"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1304-1"><a href="WE_12.html#cb1304-1" tabindex="-1"></a><span class="fu">plot</span>(mod1, <span class="at">abline=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>)) </span></code></pre></div>
<p><img src="12-Week12_files/figure-html/unnamed-chunk-13-1.png" width="672" /></p>
<p>Residuals are centered around zero and do not show large groupings or patterns, although there is some increase in variation at larger numbers (heteroscedasticity: the plot thickens). Here, this means that the model is having a more difficult time predicting larger genetic distances per km. We will keep that in mind as we move on.</p>
<p>Next we will check to make sure the residuals are normally distributed. This is an important assumption for models fitted with (Restricted) Maximum Likelihood.</p>
<p>We’ll check this with two plots:</p>
<ul>
<li>Histogram (left): we want to see a symmetric, bell-shaped distribution.</li>
<li>Normal probability plot (right): we want to see the points fall on a straight line, without systematic deviations.</li>
</ul>
<div class="sourceCode" id="cb1305"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1305-1"><a href="WE_12.html#cb1305-1" tabindex="-1"></a><span class="fu">par</span>(<span class="at">mfrow=</span><span class="fu">c</span>(<span class="dv">1</span>,<span class="dv">2</span>))</span>
<span id="cb1305-2"><a href="WE_12.html#cb1305-2" tabindex="-1"></a><span class="fu">hist</span>(<span class="fu">residuals</span>(mod1)) </span>
<span id="cb1305-3"><a href="WE_12.html#cb1305-3" tabindex="-1"></a><span class="fu">qqnorm</span>(<span class="fu">residuals</span>(mod1))</span></code></pre></div>
<p><img src="12-Week12_files/figure-html/unnamed-chunk-14-1.png" width="672" /></p>
<div class="sourceCode" id="cb1306"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1306-1"><a href="WE_12.html#cb1306-1" tabindex="-1"></a><span class="fu">par</span>(<span class="at">mfrow=</span><span class="fu">c</span>(<span class="dv">1</span>,<span class="dv">1</span>))</span></code></pre></div>
<p>These residuals look okay so we will move on (but if this were a publication we would do a closer examination of the shape of the data at the ends of the distribution).</p>
<p>Recall that mod1 had the following formula:
<code>logDc.km ~ solarinsoz + forest + ag + shrub + dev</code>.</p>
<p>Here we’ll use the function <code>update</code> to change the list of predictors in the model (i.e., the right-hand side of the formula, starting with the tilde symbol <code>~</code>).</p>
<div class="sourceCode" id="cb1307"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1307-1"><a href="WE_12.html#cb1307-1" tabindex="-1"></a>mod2 <span class="ot">&lt;-</span> <span class="fu">update</span>(mod1, <span class="sc">~</span> forest <span class="sc">+</span> ag <span class="sc">+</span> shrub <span class="sc">+</span> dev)</span>
<span id="cb1307-2"><a href="WE_12.html#cb1307-2" tabindex="-1"></a>mod3 <span class="ot">&lt;-</span> <span class="fu">update</span>(mod1, <span class="sc">~</span> ag <span class="sc">+</span> dev)</span>
<span id="cb1307-3"><a href="WE_12.html#cb1307-3" tabindex="-1"></a>mod4 <span class="ot">&lt;-</span> <span class="fu">update</span>(mod1, <span class="sc">~</span> slope <span class="sc">+</span> shrub <span class="sc">+</span> dev)</span>
<span id="cb1307-4"><a href="WE_12.html#cb1307-4" tabindex="-1"></a>mod5 <span class="ot">&lt;-</span> <span class="fu">update</span>(mod1, <span class="sc">~</span> soils <span class="sc">+</span> slope <span class="sc">+</span> solarinsoz)</span></code></pre></div>
</div>
<div id="d.-compare-evidence-for-models" class="section level4 unnumbered hasAnchor">
<h4>d. Compare evidence for models<a href="WE_12.html#d.-compare-evidence-for-models" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>Our goal here is to compare five models that all have the same random effects (the population effects modeled by the correlation structure) but differ in the fixed effects. Recall from the Week 6 videos:</p>
<ul>
<li>Use REML to test random effects, and to get AIC to compare models with different random effects.</li>
<li>Use ML to test fixed effects, and to get AIC to compare models with different fixed effects.</li>
<li>Use REML for parameter estimates of fixed effects.</li>
</ul>
<p>We can use the function update again, this time to change the fitting method from <code>method="REML"</code> to <code>method="ML"</code>.</p>
<div class="sourceCode" id="cb1308"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1308-1"><a href="WE_12.html#cb1308-1" tabindex="-1"></a>mod1noREML <span class="ot">&lt;-</span> <span class="fu">update</span>(mod1, <span class="at">method=</span><span class="st">&quot;ML&quot;</span>)</span>
<span id="cb1308-2"><a href="WE_12.html#cb1308-2" tabindex="-1"></a>mod2noREML <span class="ot">&lt;-</span> <span class="fu">update</span>(mod2, <span class="at">method=</span><span class="st">&quot;ML&quot;</span>)</span>
<span id="cb1308-3"><a href="WE_12.html#cb1308-3" tabindex="-1"></a>mod3noREML <span class="ot">&lt;-</span> <span class="fu">update</span>(mod3, <span class="at">method=</span><span class="st">&quot;ML&quot;</span>)</span>
<span id="cb1308-4"><a href="WE_12.html#cb1308-4" tabindex="-1"></a>mod4noREML <span class="ot">&lt;-</span> <span class="fu">update</span>(mod4, <span class="at">method=</span><span class="st">&quot;ML&quot;</span>)</span>
<span id="cb1308-5"><a href="WE_12.html#cb1308-5" tabindex="-1"></a>mod5noREML <span class="ot">&lt;-</span> <span class="fu">update</span>(mod5, <span class="at">method=</span><span class="st">&quot;ML&quot;</span>)</span></code></pre></div>
<p>In information theory, we use information criteria (AIC, BIC) to evaluate the relative distance of our hypothesis from truth. For more information, see Burnham and Anderson (2002).</p>
<p>Note: package MuMIn will do this for you, but for this exercise it is useful to see all the pieces of what goes into your evaluation.</p>
<div class="sourceCode" id="cb1309"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1309-1"><a href="WE_12.html#cb1309-1" tabindex="-1"></a>Models <span class="ot">&lt;-</span> <span class="fu">list</span>(<span class="at">Full=</span>mod1noREML, <span class="at">Landcover=</span>mod2noREML, <span class="at">HumanFootprint=</span>mod3noREML, </span>
<span id="cb1309-2"><a href="WE_12.html#cb1309-2" tabindex="-1"></a>               <span class="at">EnergyConservation=</span>mod4noREML, <span class="at">Historical=</span>mod5noREML)</span>
<span id="cb1309-3"><a href="WE_12.html#cb1309-3" tabindex="-1"></a>CSF.IC <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">AIC =</span> <span class="fu">sapply</span>(Models, AIC),</span>
<span id="cb1309-4"><a href="WE_12.html#cb1309-4" tabindex="-1"></a>                     <span class="at">BIC =</span> <span class="fu">sapply</span>(Models, BIC)) </span>
<span id="cb1309-5"><a href="WE_12.html#cb1309-5" tabindex="-1"></a>CSF.IC</span></code></pre></div>
<pre><code>##                         AIC      BIC
## Full               115.9620 130.9316
## Landcover          113.9757 127.0742
## HumanFootprint     113.4053 122.7613
## EnergyConservation 114.4910 125.7182
## Historical         122.4089 137.3785</code></pre>
<p>We now have some results, great!</p>
<p>Now we will work with these a bit. First, because we do not have an infinite number of samples, we will convert AIC to AICc (which adds a small-sample correction to AIC).</p>
<p>First, find the k parameters used in the model and add them to the table.</p>
<div class="sourceCode" id="cb1311"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1311-1"><a href="WE_12.html#cb1311-1" tabindex="-1"></a>CSF.IC <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(CSF.IC, <span class="at">k =</span> <span class="fu">sapply</span>(Models, <span class="cf">function</span>(ls) <span class="fu">attr</span>(<span class="fu">logLik</span>(ls), <span class="st">&quot;df&quot;</span>)))</span>
<span id="cb1311-2"><a href="WE_12.html#cb1311-2" tabindex="-1"></a>CSF.IC</span></code></pre></div>
<pre><code>##                         AIC      BIC k
## Full               115.9620 130.9316 8
## Landcover          113.9757 127.0742 7
## HumanFootprint     113.4053 122.7613 5
## EnergyConservation 114.4910 125.7182 6
## Historical         122.4089 137.3785 8</code></pre>
<p><strong>Question 2</strong>: How does k relate to the number of parameters in each model?</p>
<p><strong>Disclaimer</strong>:</p>
<ul>
<li>More research is needed to establish how to best account for sample size in MLPE. Here we use N = number of pairs.</li>
<li>More generally, more research is needed to establish whether delta values and evidence weights calculated from MLPE are indeed valid.</li>
</ul>
<p>Calculate AICc and add it to the dataframe (experimental, see disclaimer above):</p>
<div class="sourceCode" id="cb1313"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1313-1"><a href="WE_12.html#cb1313-1" tabindex="-1"></a>N <span class="ot">=</span> <span class="fu">nrow</span>(CSFdata)  <span class="co"># Number of unique pairs</span></span>
<span id="cb1313-2"><a href="WE_12.html#cb1313-2" tabindex="-1"></a>CSF.IC<span class="sc">$</span>AICc <span class="ot">&lt;-</span> CSF.IC<span class="sc">$</span>AIC <span class="sc">+</span> <span class="dv">2</span><span class="sc">*</span>CSF.IC<span class="sc">$</span>k<span class="sc">*</span>(CSF.IC<span class="sc">$</span>k<span class="sc">+</span><span class="dv">1</span>)<span class="sc">/</span>(N<span class="sc">-</span>CSF.IC<span class="sc">$</span>k<span class="dv">-1</span>)</span>
<span id="cb1313-3"><a href="WE_12.html#cb1313-3" tabindex="-1"></a>CSF.IC</span></code></pre></div>
<pre><code>##                         AIC      BIC k     AICc
## Full               115.9620 130.9316 8 119.6543
## Landcover          113.9757 127.0742 7 116.7757
## HumanFootprint     113.4053 122.7613 5 114.8339
## EnergyConservation 114.4910 125.7182 6 116.5398
## Historical         122.4089 137.3785 8 126.1012</code></pre>
</div>
<div id="e.-calculate-evidence-weights" class="section level4 unnumbered hasAnchor">
<h4>e. Calculate evidence weights<a href="WE_12.html#e.-calculate-evidence-weights" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>Next we calculate evidence weights for each model based on AICc and BIC. If all assumptions are met, these can be interpreted as the probability that the model is the closest to truth in the candidate model set.</p>
<p>Calculate model weights for AICc (experimental, see disclaimer above):</p>
<div class="sourceCode" id="cb1315"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1315-1"><a href="WE_12.html#cb1315-1" tabindex="-1"></a>AICcmin <span class="ot">&lt;-</span> <span class="fu">min</span>(CSF.IC<span class="sc">$</span>AICc)</span>
<span id="cb1315-2"><a href="WE_12.html#cb1315-2" tabindex="-1"></a>RL <span class="ot">&lt;-</span> <span class="fu">exp</span>(<span class="sc">-</span><span class="fl">0.5</span><span class="sc">*</span>(CSF.IC<span class="sc">$</span>AICc <span class="sc">-</span> AICcmin))</span>
<span id="cb1315-3"><a href="WE_12.html#cb1315-3" tabindex="-1"></a>sumRL <span class="ot">&lt;-</span> <span class="fu">sum</span>(RL)</span>
<span id="cb1315-4"><a href="WE_12.html#cb1315-4" tabindex="-1"></a>CSF.IC<span class="sc">$</span>AICcmin <span class="ot">&lt;-</span> RL<span class="sc">/</span>sumRL</span></code></pre></div>
<p>Calculate model weights for BIC (experimental, see disclaimer above):</p>
<div class="sourceCode" id="cb1316"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1316-1"><a href="WE_12.html#cb1316-1" tabindex="-1"></a>BICmin <span class="ot">&lt;-</span> <span class="fu">min</span>(CSF.IC<span class="sc">$</span>BIC)</span>
<span id="cb1316-2"><a href="WE_12.html#cb1316-2" tabindex="-1"></a>RL.B <span class="ot">&lt;-</span> <span class="fu">exp</span>(<span class="sc">-</span><span class="fl">0.5</span><span class="sc">*</span>(CSF.IC<span class="sc">$</span>BIC <span class="sc">-</span> BICmin))</span>
<span id="cb1316-3"><a href="WE_12.html#cb1316-3" tabindex="-1"></a>sumRL.B <span class="ot">&lt;-</span> <span class="fu">sum</span>(RL.B)</span>
<span id="cb1316-4"><a href="WE_12.html#cb1316-4" tabindex="-1"></a>CSF.IC<span class="sc">$</span>BICew <span class="ot">&lt;-</span> RL.B<span class="sc">/</span>sumRL.B</span>
<span id="cb1316-5"><a href="WE_12.html#cb1316-5" tabindex="-1"></a><span class="fu">round</span>(CSF.IC,<span class="dv">3</span>)</span></code></pre></div>
<pre><code>##                        AIC     BIC k    AICc AICcmin BICew
## Full               115.962 130.932 8 119.654   0.047 0.012
## Landcover          113.976 127.074 7 116.776   0.200 0.085
## HumanFootprint     113.405 122.761 5 114.834   0.527 0.735
## EnergyConservation 114.491 125.718 6 116.540   0.224 0.167
## Historical         122.409 137.378 8 126.101   0.002 0.000</code></pre>
<p><strong>Question 3</strong>: What did using AICc (rather than AIC) do to inference from these results?</p>
<p>In the multi-model inference approach, we can look across all the models in the dataset and their evidence weights to better understand the system. Here we see that models 1 and 5 have very little evidence of being close to the truth, that models 2 and 4 have more, and that model 3 has the most evidence of being the closest to truth, although how much more evidence there is for model 3 over 2 and 4 depends on which criterion you are looking at. BIC imposes a larger penalty for additional parameters, so it has a larger distinction between these. Let’s review what these models are:</p>
<ol style="list-style-type: decimal">
<li><strong>Full model</strong>: solarinso, forest, ag, shrub, dev</li>
<li><strong>Landcover model</strong>: ag, shrub, forest, and dev</li>
<li><strong>Human footprint model</strong>: ag, dev</li>
<li><strong>Energy conservation model</strong>: slope, shrub, dev</li>
<li><strong>Historical model</strong>: soils, slope, solarinso</li>
</ol>
</div>
<div id="f.-confidence-intervals-for-predictors" class="section level4 unnumbered hasAnchor">
<h4>f. Confidence intervals for predictors<a href="WE_12.html#f.-confidence-intervals-for-predictors" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>The <code>summary</code> function for the model fitted with “ML” provides a model summary that includes a table with t-tests for each fixed effect in a Frequentist approach (see Week 6 Worked Example). Here we stay within the information-theoretic approach and use a different approach than significance testing to identify the most important predictors.</p>
<p>According to Row et al. (2017), we can use the confidence interval from the variables in the models to identify those that are contributing to landscape resistance. Looking at models 2 and 3, we have some evidence that shrub and forest cover matter, but it’s not as strong as the evidence for ag and development. Let’s look at the parameter estimates to understand more. Note that for parameter estimation we use the REML estimates.</p>
<div class="sourceCode" id="cb1318"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1318-1"><a href="WE_12.html#cb1318-1" tabindex="-1"></a>ModelsREML <span class="ot">&lt;-</span> <span class="fu">list</span>(<span class="at">Full=</span>mod1, <span class="at">Landcover=</span>mod2, <span class="at">HumanFootprint=</span>mod3,</span>
<span id="cb1318-2"><a href="WE_12.html#cb1318-2" tabindex="-1"></a>               <span class="at">EnergyConservation=</span>mod4, <span class="at">Historical=</span>mod5)</span></code></pre></div>
<p>Now we’ll estimate the 95% confidence interval for these estimates:</p>
<div class="sourceCode" id="cb1319"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1319-1"><a href="WE_12.html#cb1319-1" tabindex="-1"></a><span class="fu">confint</span>(ModelsREML<span class="sc">$</span>Landcover, <span class="at">level =</span> <span class="fl">0.95</span>, <span class="at">method =</span> <span class="st">&quot;Wald&quot;</span>)</span></code></pre></div>
<pre><code>##                 2.5 %     97.5 %
## (Intercept) -5.412493 -3.9372220
## forest      -1.234534  0.8980824
## ag          -2.767849 -0.2789160
## shrub       -4.335188  0.4391631
## dev         -1.077595  1.4292393</code></pre>
<div class="sourceCode" id="cb1321"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1321-1"><a href="WE_12.html#cb1321-1" tabindex="-1"></a><span class="fu">confint</span>(ModelsREML<span class="sc">$</span>HumanFootprint, <span class="at">level =</span> <span class="fl">0.95</span>, <span class="at">method =</span> <span class="st">&quot;Wald&quot;</span>)</span></code></pre></div>
<pre><code>##                  2.5 %     97.5 %
## (Intercept) -5.3193590 -4.5159303
## ag          -2.0215023 -0.1997056
## dev         -0.6843463  1.5082812</code></pre>
<div class="sourceCode" id="cb1323"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1323-1"><a href="WE_12.html#cb1323-1" tabindex="-1"></a><span class="fu">confint</span>(ModelsREML<span class="sc">$</span>EnergyConservation, <span class="at">level =</span> <span class="fl">0.95</span>, <span class="at">method =</span> <span class="st">&quot;Wald&quot;</span>)</span></code></pre></div>
<pre><code>##                   2.5 %     97.5 %
## (Intercept) -6.83962997 -5.3057702
## slope        0.02028322  0.1725869
## shrub       -3.83624990  1.0650554
## dev          0.22109749  2.6485991</code></pre>
<p>In essence, we use the confidence intervals to assess whether we can differentiate the effect of each variable from zero (i.e. statistical significance). The confidence intervals that overlap 0 do not have as much evidence of being important to gene flow, based on simulations in Row et al. (2017). So in this case, there is strong evidence that agriculture influences the rate of gene flow. Note that because the metric is genetic distance, negative values indicate more gene flow.</p>
<p><strong>Question 4</strong>: What evidence is there that shrub and forest matter to gene flow from these analyses? What about slope? Would you include them in a conclusion about landscape resistance in this system?</p>
<p><strong>Question 5</strong>: What evidence is there that development influences gene flow from these analyses? Would you include it in a conclusion about landscape resistance in this system?</p>
<p><strong>Question 6</strong>: We excluded grass from our models because of multicollinearity. How could we investigate the importance of this variable?</p>
</div>
</div>
<div id="your-turn-to-test-additional-hypotheses" class="section level3 unnumbered hasAnchor">
<h3>3. Your turn to test additional hypotheses!<a href="WE_12.html#your-turn-to-test-additional-hypotheses" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Create your own (small) set of hypotheses to rank for evidence using the dataset provided. Modify the code above to complete the following steps:</p>
<ol style="list-style-type: decimal">
<li>Define your hypotheses.</li>
<li>Calculate the VIF table for full model.</li>
<li>Make a residual plot for full model.</li>
<li>Create table of AICc and BIC weights.</li>
<li>Create confidence intervals from models with high evidence weights.</li>
</ol>
<p>What can you infer from your analysis about what influences gene flow of this species?</p>
</div>
<div id="nested-model-nmlpe-for-hierarchical-sampling-designs" class="section level3 unnumbered hasAnchor">
<h3>4. Nested model (NMLPE) for hierarchical sampling designs<a href="WE_12.html#nested-model-nmlpe-for-hierarchical-sampling-designs" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Jaffe et al. (2019) proposed nested MLPE (NMLPE) to account for an additional level of hierarchical sampling. In their example, they had sampled multiple sites (clusters), and within each site, multiple colonies (populations).</p>
<p>When they fitted a MLPE model that did not account for the hierarchical level of sites, there was unaccounted autocorrelation in the residuals. They tested for this by sorting the pairwise distances by site and then calculating an autocorrelation function (ACF) as a function of lag distance along rows (i.e., the lag represented how many rows apart to values were in the dataset). The following pseudo-code is based on code in the supplementary material of Jaffe et al. (2019).</p>
<p>Sort data frame with pairwise data by site, then by pop (pseudo-code). Note: this is only needed for the ACF analysis.</p>
<div class="sourceCode" id="cb1325"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1325-1"><a href="WE_12.html#cb1325-1" tabindex="-1"></a><span class="do">## d. &lt;- df[order(df$site, df$pop1, df$pop2),]</span></span></code></pre></div>
<p>MLPE model (pseudo-code):</p>
<div class="sourceCode" id="cb1326"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1326-1"><a href="WE_12.html#cb1326-1" tabindex="-1"></a><span class="co"># m1 &lt;- gls(Genetic.distance ~ Predictor.distance, </span></span>
<span id="cb1326-2"><a href="WE_12.html#cb1326-2" tabindex="-1"></a><span class="co">#           correlation = corMLPE(form = ~ pop1+pop2), data = df)</span></span>
<span id="cb1326-3"><a href="WE_12.html#cb1326-3" tabindex="-1"></a><span class="do">## a.f(resid(m1, type=&#39;normalized&#39;)) ## Autocorrelation function</span></span></code></pre></div>
<p>The package corMLPE contains another function, corNMLPE2, where we can specify cluster as an additional hierarchical level of non-independence in the data. The ACF of this model did not show any autocorrelation anymore.</p>
<p>NMLPE model (pseudo-code)</p>
<div class="sourceCode" id="cb1327"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1327-1"><a href="WE_12.html#cb1327-1" tabindex="-1"></a><span class="co"># m2 &lt;- gls(Genetic.distance ~ Predictor.distance, </span></span>
<span id="cb1327-2"><a href="WE_12.html#cb1327-2" tabindex="-1"></a><span class="co">#           correlation = corMLPE(form = ~ pop1+pop2, clusters = site), data = df)</span></span>
<span id="cb1327-3"><a href="WE_12.html#cb1327-3" tabindex="-1"></a><span class="do">## a.f(resid(m2, type=&#39;normalized&#39;)) ## Autocorrelation function</span></span></code></pre></div>
<p>Note that this is a method to account for hierarchical sampling designs. It does not explicitly account for spatial autocorrelation - indeed, no spatial information is included, unless Euclidean distance is included as a predictor in the model.</p>
</div>
<div id="references-3" class="section level3 unnumbered hasAnchor">
<h3>5. References<a href="WE_12.html#references-3" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<ul>
<li><p>Burnham KP and DR Anderson. 2002. Model Selection and Multimodel Inference: A Practical Information Theoretic Approach. Chapter 8, p. 437-454. Springer-Verlag, New York.</p></li>
<li><p>Goldberg CS and LP Waits (2010). Comparative landscape genetics of two pond-breeding amphibian species in a highly modified agricultural landscape. Molecular Ecology 19: 3650-3663.</p></li>
<li><p>Jaffé, R, Veiga, JC, Pope, NS, et al. (2019). Landscape genomics to the rescue of a tropical bee threatened by habitat loss and climate change. Evol Appl. 12: 1164– 1177. <a href="https://doi-org.myaccess.library.utoronto.ca/10.1111/eva.12794" class="uri">https://doi-org.myaccess.library.utoronto.ca/10.1111/eva.12794</a></p></li>
<li><p>Row JR, ST Knick, SJ Oyler-McCance, SC Lougheed and BC Fedy (2017). Developing approaches for linear mixed modeling in landscape genetics through landscape-directed dispersal simulations. Ecology &amp; Evolution 7: 3751–3761.</p></li>
<li><p>Van Strien MJ, D Keller and R Holderegger (2012). A new analytical approach to landscape genetic modelling: least‐cost transect analysis and linear mixed models. Molecular Ecology 21: 4010-4023.</p></li>
</ul>

</div>
</div>
<!-- </div> -->
            </section>

          </div>
        </div>
      </div>
<a href="Week12.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="Week13.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": true,
"facebook": false,
"twitter": false,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": false
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": null,
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "section"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.9/latest.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
