<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>16.1 Worked Example | Landscape Genetic Data Analysis with R</title>
  <meta name="description" content="This is a web-interface to the teaching materials for the lab course ‘Landscape Genetic Data Analysis with R’ associated with the distributed graduate course ‘DGS Landscape Genetics’. The output format is bookdown::gitbook." />
  <meta name="generator" content="bookdown 0.37 and GitBook 2.6.7" />

  <meta property="og:title" content="16.1 Worked Example | Landscape Genetic Data Analysis with R" />
  <meta property="og:type" content="book" />
  
  <meta property="og:description" content="This is a web-interface to the teaching materials for the lab course ‘Landscape Genetic Data Analysis with R’ associated with the distributed graduate course ‘DGS Landscape Genetics’. The output format is bookdown::gitbook." />
  <meta name="github-repo" content="hhwagner1/LandGenCourse_book" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="16.1 Worked Example | Landscape Genetic Data Analysis with R" />
  
  <meta name="twitter:description" content="This is a web-interface to the teaching materials for the lab course ‘Landscape Genetic Data Analysis with R’ associated with the distributed graduate course ‘DGS Landscape Genetics’. The output format is bookdown::gitbook." />
  

<meta name="author" content="Editor: Helene Wagner (University of Toronto)" />


<meta name="date" content="2024-01-25" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="Week13.html"/>
<link rel="next" href="Week14.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>
<script src="libs/kePrint-0.0.1/kePrint.js"></script>
<link href="libs/lightable-0.0.1/lightable.css" rel="stylesheet" />
<script src="libs/htmlwidgets-1.6.4/htmlwidgets.js"></script>
<link href="libs/leaflet-1.3.1/leaflet.css" rel="stylesheet" />
<script src="libs/leaflet-1.3.1/leaflet.js"></script>
<link href="libs/leafletfix-1.0.0/leafletfix.css" rel="stylesheet" />
<script src="libs/proj4-2.6.2/proj4.min.js"></script>
<script src="libs/Proj4Leaflet-1.0.1/proj4leaflet.js"></script>
<link href="libs/rstudio_leaflet-1.3.1/rstudio_leaflet.css" rel="stylesheet" />
<script src="libs/leaflet-binding-2.2.1/leaflet.js"></script>
<script src="libs/leaflet-providers-2.0.0/leaflet-providers_2.0.0.js"></script>
<script src="libs/leaflet-providers-plugin-2.2.1/leaflet-providers-plugin.js"></script>
<script src="libs/d3-3.5.6/d3.min.js"></script>
<link href="libs/profvis-0.3.6.9000/profvis.css" rel="stylesheet" />
<script src="libs/profvis-0.3.6.9000/profvis.js"></script>
<script src="libs/profvis-0.3.6.9000/scroll.js"></script>
<link href="libs/highlight-6.2.0/textmate.css" rel="stylesheet" />
<script src="libs/highlight-6.2.0/highlight.js"></script>
<script src="libs/profvis-binding-0.3.8/profvis.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
  
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">Landscape Genetic Data Analysis with R</a></li>

<li class="divider"></li>
<li class="part"><span><b>Getting Started</b></span></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Introduction</a>
<ul>
<li class="chapter" data-level="1.1" data-path="how-to-use-this-book.html"><a href="how-to-use-this-book.html"><i class="fa fa-check"></i><b>1.1</b> How to use this Book</a>
<ul>
<li class="chapter" data-level="" data-path="how-to-use-this-book.html"><a href="how-to-use-this-book.html#book-structure"><i class="fa fa-check"></i>1. Book Structure</a></li>
<li class="chapter" data-level="" data-path="how-to-use-this-book.html"><a href="how-to-use-this-book.html#find-what-is-relevant-for-you"><i class="fa fa-check"></i>2. Find what is relevant for you</a></li>
<li class="chapter" data-level="" data-path="how-to-use-this-book.html"><a href="how-to-use-this-book.html#course-r-package-landgencourse"><i class="fa fa-check"></i>3. Course R package ‘LandGenCourse’</a></li>
</ul></li>
<li class="chapter" data-level="1.2" data-path="list-of-packages.html"><a href="list-of-packages.html"><i class="fa fa-check"></i><b>1.2</b> List of R Packages by Vignette</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="review-of-r-skills.html"><a href="review-of-r-skills.html"><i class="fa fa-check"></i><b>2</b> Review of R Skills</a>
<ul>
<li class="chapter" data-level="2.1" data-path="basic-r.html"><a href="basic-r.html"><i class="fa fa-check"></i><b>2.1</b> Basic R Programming</a>
<ul>
<li class="chapter" data-level="" data-path="basic-r.html"><a href="basic-r.html#overview"><i class="fa fa-check"></i>1. Overview</a></li>
<li class="chapter" data-level="" data-path="basic-r.html"><a href="basic-r.html#data-types"><i class="fa fa-check"></i>2. Data Types</a></li>
<li class="chapter" data-level="" data-path="basic-r.html"><a href="basic-r.html#data-containers"><i class="fa fa-check"></i>3. Data Containers</a></li>
<li class="chapter" data-level="" data-path="basic-r.html"><a href="basic-r.html#programming"><i class="fa fa-check"></i>4. Programming</a></li>
</ul></li>
<li class="chapter" data-level="2.2" data-path="r-graphics.html"><a href="r-graphics.html"><i class="fa fa-check"></i><b>2.2</b> R Graphics</a>
<ul>
<li class="chapter" data-level="" data-path="r-graphics.html"><a href="r-graphics.html#overview-1"><i class="fa fa-check"></i>1. Overview</a></li>
<li class="chapter" data-level="" data-path="r-graphics.html"><a href="r-graphics.html#univariate-plots"><i class="fa fa-check"></i>2. Univariate plots</a></li>
<li class="chapter" data-level="" data-path="r-graphics.html"><a href="r-graphics.html#bivariate-plots"><i class="fa fa-check"></i>3. Bivariate plots</a></li>
<li class="chapter" data-level="" data-path="r-graphics.html"><a href="r-graphics.html#advanced-plotting"><i class="fa fa-check"></i>4. Advanced plotting</a></li>
</ul></li>
<li class="chapter" data-level="2.3" data-path="resources.html"><a href="resources.html"><i class="fa fa-check"></i><b>2.3</b> Further R resources</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="github-for-group-projects.html"><a href="github-for-group-projects.html"><i class="fa fa-check"></i><b>3</b> GitHub for Group Projects</a>
<ul>
<li class="chapter" data-level="3.1" data-path="github.html"><a href="github.html"><i class="fa fa-check"></i><b>3.1</b> Why use GitHub?</a>
<ul>
<li class="chapter" data-level="" data-path="github.html"><a href="github.html#goals"><i class="fa fa-check"></i>1. Goals</a></li>
<li class="chapter" data-level="" data-path="github.html"><a href="github.html#version"><i class="fa fa-check"></i>2. Watch course video: Version Control 101</a></li>
<li class="chapter" data-level="" data-path="github.html"><a href="github.html#slides"><i class="fa fa-check"></i>3. Preview slides</a></li>
<li class="chapter" data-level="" data-path="github.html"><a href="github.html#next-steps"><i class="fa fa-check"></i>4. Next steps</a></li>
</ul></li>
<li class="chapter" data-level="3.2" data-path="install.html"><a href="install.html"><i class="fa fa-check"></i><b>3.2</b> Installation</a>
<ul>
<li class="chapter" data-level="" data-path="install.html"><a href="install.html#goals-1"><i class="fa fa-check"></i>1. Goals</a></li>
<li class="chapter" data-level="3.2.1" data-path="install.html"><a href="install.html#how-to-get-github-and-gitkraken-for-free"><i class="fa fa-check"></i><b>3.2.1</b> 2. How to get GitHub and GitKraken for free</a></li>
<li class="chapter" data-level="" data-path="install.html"><a href="install.html#happygit"><i class="fa fa-check"></i>3. Happy Git with R</a></li>
<li class="chapter" data-level="" data-path="install.html"><a href="install.html#gitvideos"><i class="fa fa-check"></i>4. Video tutorials: installation</a></li>
<li class="chapter" data-level="" data-path="install.html"><a href="install.html#join"><i class="fa fa-check"></i>5. Joining your group’s project on GitHub</a></li>
<li class="chapter" data-level="" data-path="install.html"><a href="install.html#repo"><i class="fa fa-check"></i>6. Setting up a group repo</a></li>
<li class="chapter" data-level="" data-path="install.html"><a href="install.html#next-steps-1"><i class="fa fa-check"></i>7. Next steps</a></li>
</ul></li>
<li class="chapter" data-level="3.3" data-path="workflow.html"><a href="workflow.html"><i class="fa fa-check"></i><b>3.3</b> Workflows</a>
<ul>
<li class="chapter" data-level="" data-path="workflow.html"><a href="workflow.html#goals-2"><i class="fa fa-check"></i>1. Goals</a></li>
<li class="chapter" data-level="" data-path="workflow.html"><a href="workflow.html#step-by-step-tutorials"><i class="fa fa-check"></i>2. Step-by-step tutorials</a></li>
<li class="chapter" data-level="" data-path="workflow.html"><a href="workflow.html#further-resources"><i class="fa fa-check"></i>3. Further resources</a></li>
</ul></li>
</ul></li>
<li class="part"><span><b>Basic Topics</b></span></li>
<li class="chapter" data-level="4" data-path="Week1.html"><a href="Week1.html"><i class="fa fa-check"></i><b>4</b> Week 1: Importing Genetic Data</a>
<ul>
<li class="chapter" data-level="4.1" data-path="video_1.html"><a href="video_1.html"><i class="fa fa-check"></i><b>4.1</b> View Course Video</a>
<ul>
<li class="chapter" data-level="" data-path="video_1.html"><a href="video_1.html#embedded-video"><i class="fa fa-check"></i>1. Embedded Video</a></li>
<li class="chapter" data-level="" data-path="video_1.html"><a href="video_1.html#preview-slides"><i class="fa fa-check"></i>2. Preview Slides</a></li>
</ul></li>
<li class="chapter" data-level="4.2" data-path="tutorial_1.html"><a href="tutorial_1.html"><i class="fa fa-check"></i><b>4.2</b> Interactive Tutorial 1</a>
<ul>
<li class="chapter" data-level="" data-path="tutorial_1.html"><a href="tutorial_1.html#list-of-r-commands-covered-this-week"><i class="fa fa-check"></i>1. List of R commands covered this week</a></li>
<li class="chapter" data-level="" data-path="tutorial_1.html"><a href="tutorial_1.html#general-instructions"><i class="fa fa-check"></i>2. General Instructions</a></li>
</ul></li>
<li class="chapter" data-level="4.3" data-path="WE_1.html"><a href="WE_1.html"><i class="fa fa-check"></i><b>4.3</b> Worked Example</a>
<ul>
<li class="chapter" data-level="" data-path="WE_1.html"><a href="WE_1.html#overview-of-worked-example"><i class="fa fa-check"></i>1. Overview of Worked Example</a></li>
<li class="chapter" data-level="" data-path="WE_1.html"><a href="WE_1.html#import-data-from-.csv-file"><i class="fa fa-check"></i>2. Import data from .csv file</a></li>
<li class="chapter" data-level="" data-path="WE_1.html"><a href="WE_1.html#create-genind-object"><i class="fa fa-check"></i>3. Create ‘genind’ Object</a></li>
<li class="chapter" data-level="" data-path="WE_1.html"><a href="WE_1.html#view-information-stored-in-genind-object"><i class="fa fa-check"></i>4. View information stored in ‘genind’ object</a></li>
<li class="chapter" data-level="" data-path="WE_1.html"><a href="WE_1.html#import-data-with-gstudio-package"><i class="fa fa-check"></i>5. Import data with ‘gstudio’ package</a></li>
<li class="chapter" data-level="" data-path="WE_1.html"><a href="WE_1.html#importing-snp-data"><i class="fa fa-check"></i>6. Importing SNP data</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="r-exercise-week-1.html"><a href="r-exercise-week-1.html"><i class="fa fa-check"></i>R exercise Week 1</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="Week2.html"><a href="Week2.html"><i class="fa fa-check"></i><b>5</b> Week 2: Spatial Data</a>
<ul>
<li class="chapter" data-level="5.1" data-path="video_2.html"><a href="video_2.html"><i class="fa fa-check"></i><b>5.1</b> View Course Video</a>
<ul>
<li class="chapter" data-level="" data-path="video_2.html"><a href="video_2.html#embedded-video-1"><i class="fa fa-check"></i>1. Embedded Video</a></li>
<li class="chapter" data-level="" data-path="video_2.html"><a href="video_2.html#preview-slides-1"><i class="fa fa-check"></i>2. Preview Slides</a></li>
</ul></li>
<li class="chapter" data-level="5.2" data-path="tutorial_2.html"><a href="tutorial_2.html"><i class="fa fa-check"></i><b>5.2</b> Interactive Tutorial 2</a>
<ul>
<li class="chapter" data-level="" data-path="tutorial_2.html"><a href="tutorial_2.html#list-of-r-commands-covered-this-week-1"><i class="fa fa-check"></i>1. List of R commands covered this week</a></li>
<li class="chapter" data-level="" data-path="tutorial_2.html"><a href="tutorial_2.html#general-instructions-1"><i class="fa fa-check"></i>2. General Instructions</a></li>
</ul></li>
<li class="chapter" data-level="5.3" data-path="WE_2.html"><a href="WE_2.html"><i class="fa fa-check"></i><b>5.3</b> Worked Example</a>
<ul>
<li class="chapter" data-level="" data-path="WE_2.html"><a href="WE_2.html#overview-of-worked-example-1"><i class="fa fa-check"></i>1. Overview of Worked Example</a></li>
<li class="chapter" data-level="" data-path="WE_2.html"><a href="WE_2.html#import-site-data-from-.csv-file"><i class="fa fa-check"></i>2. Import site data from .csv file</a></li>
<li class="chapter" data-level="" data-path="WE_2.html"><a href="WE_2.html#display-raster-data-and-overlay-sampling-locations-extract-data"><i class="fa fa-check"></i>3. Display raster data and overlay sampling locations, extract data</a></li>
<li class="chapter" data-level="" data-path="WE_2.html"><a href="WE_2.html#calculate-landscape-metrics"><i class="fa fa-check"></i>4. Calculate landscape metrics</a></li>
<li class="chapter" data-level="" data-path="WE_2.html"><a href="WE_2.html#sample-landscape-metrics-within-buffer-around-sampling-locations"><i class="fa fa-check"></i>5. Sample landscape metrics within buffer around sampling locations</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="r-exercise-week-2.html"><a href="r-exercise-week-2.html"><i class="fa fa-check"></i>R Exercise Week 2</a></li>
<li class="chapter" data-level="5.4" data-path="bonus_2a.html"><a href="bonus_2a.html"><i class="fa fa-check"></i><b>5.4</b> Bonus: ‘sf’ and ‘terra’</a>
<ul>
<li class="chapter" data-level="" data-path="bonus_2a.html"><a href="bonus_2a.html#overview-2"><i class="fa fa-check"></i>1. Overview</a></li>
<li class="chapter" data-level="" data-path="bonus_2a.html"><a href="bonus_2a.html#import-and-export-esri-shapefiles"><i class="fa fa-check"></i>2. Import and export ESRI shapefiles</a></li>
<li class="chapter" data-level="" data-path="bonus_2a.html"><a href="bonus_2a.html#compatibility-with-sp-and-raster-objects"><i class="fa fa-check"></i>3. Compatibility with <code>sp</code> and <code>raster</code> objects</a></li>
<li class="chapter" data-level="" data-path="bonus_2a.html"><a href="bonus_2a.html#plotting-spatial-data-with-tmap"><i class="fa fa-check"></i>4. Plotting spatial data with <code>tmap</code></a></li>
<li class="chapter" data-level="" data-path="bonus_2a.html"><a href="bonus_2a.html#plot-a-categorical-map-with-predefined-color-scheme"><i class="fa fa-check"></i>5. Plot a categorical map with predefined color scheme</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="6" data-path="Week3.html"><a href="Week3.html"><i class="fa fa-check"></i><b>6</b> Week 3: Genetic Diversity</a>
<ul>
<li class="chapter" data-level="6.1" data-path="video_3.html"><a href="video_3.html"><i class="fa fa-check"></i><b>6.1</b> View Course Video</a>
<ul>
<li class="chapter" data-level="" data-path="video_3.html"><a href="video_3.html#embedded-video-2"><i class="fa fa-check"></i>1. Embedded Video</a></li>
<li class="chapter" data-level="" data-path="video_3.html"><a href="video_3.html#video-part-1"><i class="fa fa-check"></i>Video, Part 1</a></li>
<li class="chapter" data-level="" data-path="video_3.html"><a href="video_3.html#video-part-2"><i class="fa fa-check"></i>Video, Part 2</a></li>
<li class="chapter" data-level="" data-path="video_3.html"><a href="video_3.html#preview-slides-2"><i class="fa fa-check"></i>Preview Slides</a></li>
</ul></li>
<li class="chapter" data-level="6.2" data-path="tutorial_3.html"><a href="tutorial_3.html"><i class="fa fa-check"></i><b>6.2</b> Interactive Tutorial 3</a>
<ul>
<li class="chapter" data-level="" data-path="tutorial_3.html"><a href="tutorial_3.html#list-of-r-commands-covered-this-week-2"><i class="fa fa-check"></i>1. List of R commands covered this week</a></li>
<li class="chapter" data-level="" data-path="tutorial_3.html"><a href="tutorial_3.html#general-instructions-2"><i class="fa fa-check"></i>2. General Instructions</a></li>
</ul></li>
<li class="chapter" data-level="6.3" data-path="WE_3.html"><a href="WE_3.html"><i class="fa fa-check"></i><b>6.3</b> Worked Example</a></li>
<li class="chapter" data-level="6.4" data-path="r-exercise-week-3.html"><a href="r-exercise-week-3.html"><i class="fa fa-check"></i><b>6.4</b> R Exercise Week 3</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="Week4.html"><a href="Week4.html"><i class="fa fa-check"></i><b>7</b> Week 4: Metapopulations</a>
<ul>
<li class="chapter" data-level="7.1" data-path="video_4.html"><a href="video_4.html"><i class="fa fa-check"></i><b>7.1</b> View Course Video</a>
<ul>
<li class="chapter" data-level="" data-path="video_4.html"><a href="video_4.html#embedded-video-3"><i class="fa fa-check"></i>1. Embedded Video</a></li>
<li class="chapter" data-level="" data-path="video_4.html"><a href="video_4.html#video-part-1-1"><i class="fa fa-check"></i>Video, Part 1</a></li>
<li class="chapter" data-level="" data-path="video_4.html"><a href="video_4.html#video-part-2-1"><i class="fa fa-check"></i>Video, Part 2</a></li>
<li class="chapter" data-level="" data-path="video_4.html"><a href="video_4.html#preview-slides-3"><i class="fa fa-check"></i>Preview Slides</a></li>
</ul></li>
<li class="chapter" data-level="7.2" data-path="tutorial_4.html"><a href="tutorial_4.html"><i class="fa fa-check"></i><b>7.2</b> Interactive Tutorial 4</a>
<ul>
<li class="chapter" data-level="" data-path="tutorial_4.html"><a href="tutorial_4.html#list-of-r-commands-covered-this-week-3"><i class="fa fa-check"></i>1. List of R commands covered this week</a></li>
<li class="chapter" data-level="" data-path="tutorial_4.html"><a href="tutorial_4.html#general-instructions-3"><i class="fa fa-check"></i>2. General Instructions</a></li>
</ul></li>
<li class="chapter" data-level="7.3" data-path="WE_4.html"><a href="WE_4.html"><i class="fa fa-check"></i><b>7.3</b> Worked Example</a>
<ul>
<li class="chapter" data-level="" data-path="WE_4.html"><a href="WE_4.html#overview-of-worked-example-3"><i class="fa fa-check"></i>1. Overview of Worked Example</a></li>
<li class="chapter" data-level="" data-path="WE_4.html"><a href="WE_4.html#spatial-distribution-of-genetic-structure"><i class="fa fa-check"></i>2. Spatial distribution of genetic structure</a></li>
<li class="chapter" data-level="" data-path="WE_4.html"><a href="WE_4.html#what-determines-genetic-differentiation-among-sites"><i class="fa fa-check"></i>3. What determines genetic differentiation among sites?</a></li>
<li class="chapter" data-level="" data-path="WE_4.html"><a href="WE_4.html#what-determines-genetic-diversity"><i class="fa fa-check"></i>4. What determines genetic diversity?</a></li>
<li class="chapter" data-level="" data-path="WE_4.html"><a href="WE_4.html#are-genetic-differentiation-and-diversity-related"><i class="fa fa-check"></i>5. Are genetic differentiation and diversity related?</a></li>
<li class="chapter" data-level="" data-path="WE_4.html"><a href="WE_4.html#effect-of-recent-extinction-events"><i class="fa fa-check"></i>6. Effect of recent extinction events</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="r-exercise-week-4.html"><a href="r-exercise-week-4.html"><i class="fa fa-check"></i>R Exercise Week 4</a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="Week5.html"><a href="Week5.html"><i class="fa fa-check"></i><b>8</b> Week 5: Spatial Statistics</a>
<ul>
<li class="chapter" data-level="8.1" data-path="video_5.html"><a href="video_5.html"><i class="fa fa-check"></i><b>8.1</b> View Course Video</a>
<ul>
<li class="chapter" data-level="" data-path="video_5.html"><a href="video_5.html#embedded-video-4"><i class="fa fa-check"></i>1. Embedded Video</a></li>
<li class="chapter" data-level="" data-path="video_5.html"><a href="video_5.html#preview-slides-4"><i class="fa fa-check"></i>Preview Slides</a></li>
</ul></li>
<li class="chapter" data-level="8.2" data-path="tutorial_5.html"><a href="tutorial_5.html"><i class="fa fa-check"></i><b>8.2</b> Interactive Tutorial 5</a>
<ul>
<li class="chapter" data-level="" data-path="tutorial_5.html"><a href="tutorial_5.html#list-of-r-commands-covered-this-week-4"><i class="fa fa-check"></i>1. List of R commands covered this week</a></li>
<li class="chapter" data-level="" data-path="tutorial_5.html"><a href="tutorial_5.html#general-instructions-4"><i class="fa fa-check"></i>2. General Instructions</a></li>
</ul></li>
<li class="chapter" data-level="8.3" data-path="WE_5.html"><a href="WE_5.html"><i class="fa fa-check"></i><b>8.3</b> Worked Example</a>
<ul>
<li class="chapter" data-level="" data-path="WE_5.html"><a href="WE_5.html#overview-of-worked-example-4"><i class="fa fa-check"></i>1. Overview of Worked Example</a></li>
<li class="chapter" data-level="" data-path="WE_5.html"><a href="WE_5.html#data-import-and-manipulation"><i class="fa fa-check"></i>2. Data import and manipulation</a></li>
<li class="chapter" data-level="" data-path="WE_5.html"><a href="WE_5.html#calculate-genetic-distances"><i class="fa fa-check"></i>3. Calculate genetic distances</a></li>
<li class="chapter" data-level="" data-path="WE_5.html"><a href="WE_5.html#perform-a-mantel-test-to-test-for-ibd"><i class="fa fa-check"></i>4. Perform a Mantel test to test for IBD</a></li>
<li class="chapter" data-level="" data-path="WE_5.html"><a href="WE_5.html#create-mantel-correlogram-for-genetic-data"><i class="fa fa-check"></i>5. Create Mantel correlogram for genetic data</a></li>
<li class="chapter" data-level="" data-path="WE_5.html"><a href="WE_5.html#specify-spatial-weights-and-calculate-morans-i"><i class="fa fa-check"></i>6. Specify spatial weights and calculate Moran’s I</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="r-exercise-week-5.html"><a href="r-exercise-week-5.html"><i class="fa fa-check"></i>R Exercise Week 5</a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="Week6.html"><a href="Week6.html"><i class="fa fa-check"></i><b>9</b> Week 6: Quantitative Genetics</a>
<ul>
<li class="chapter" data-level="9.1" data-path="video_6.html"><a href="video_6.html"><i class="fa fa-check"></i><b>9.1</b> View Course Video</a>
<ul>
<li class="chapter" data-level="" data-path="video_6.html"><a href="video_6.html#embedded-video-5"><i class="fa fa-check"></i>1. Embedded Video</a></li>
<li class="chapter" data-level="" data-path="video_6.html"><a href="video_6.html#video-part-1-2"><i class="fa fa-check"></i>Video, Part 1</a></li>
<li class="chapter" data-level="" data-path="video_6.html"><a href="video_6.html#video-part-2-2"><i class="fa fa-check"></i>Video, Part 2</a></li>
<li class="chapter" data-level="" data-path="video_6.html"><a href="video_6.html#preview-slides-5"><i class="fa fa-check"></i>Preview Slides</a></li>
</ul></li>
<li class="chapter" data-level="9.2" data-path="tutorial_6.html"><a href="tutorial_6.html"><i class="fa fa-check"></i><b>9.2</b> Interactive Tutorial 6</a>
<ul>
<li class="chapter" data-level="" data-path="tutorial_6.html"><a href="tutorial_6.html#list-of-r-commands-covered-this-week-5"><i class="fa fa-check"></i>1. List of R commands covered this week</a></li>
<li class="chapter" data-level="" data-path="tutorial_6.html"><a href="tutorial_6.html#general-instructions-5"><i class="fa fa-check"></i>2. General Instructions</a></li>
</ul></li>
<li class="chapter" data-level="9.3" data-path="WE_6.html"><a href="WE_6.html"><i class="fa fa-check"></i><b>9.3</b> Worked Example</a>
<ul>
<li class="chapter" data-level="" data-path="WE_6.html"><a href="WE_6.html#overview-of-worked-example-5"><i class="fa fa-check"></i>1. Overview of Worked Example</a></li>
<li class="chapter" data-level="" data-path="WE_6.html"><a href="WE_6.html#estimate-genetic-and-non-genetic-variance-components-from-a-common-garden-experiment"><i class="fa fa-check"></i>2. Estimate genetic and non-genetic variance components from a common garden experiment</a></li>
<li class="chapter" data-level="" data-path="WE_6.html"><a href="WE_6.html#estimate-trait-heritability"><i class="fa fa-check"></i>3. Estimate trait heritability</a></li>
<li class="chapter" data-level="" data-path="WE_6.html"><a href="WE_6.html#compare-q_st-to-f_st-measured-from-snp-data"><i class="fa fa-check"></i>4. Compare <span class="math inline">\(Q_{st}\)</span> to <span class="math inline">\(F_{st}\)</span> measured from SNP data</a></li>
<li class="chapter" data-level="" data-path="WE_6.html"><a href="WE_6.html#assess-correlation-between-trait-and-environment"><i class="fa fa-check"></i>5. Assess correlation between trait and environment</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="r-exercise-week-6.html"><a href="r-exercise-week-6.html"><i class="fa fa-check"></i>R Exercise Week 6</a></li>
</ul></li>
<li class="chapter" data-level="10" data-path="Week7.html"><a href="Week7.html"><i class="fa fa-check"></i><b>10</b> Week 7: Spatial Linear Models</a>
<ul>
<li class="chapter" data-level="10.1" data-path="video_7.html"><a href="video_7.html"><i class="fa fa-check"></i><b>10.1</b> View Course Video</a>
<ul>
<li class="chapter" data-level="" data-path="video_7.html"><a href="video_7.html#embedded-video-6"><i class="fa fa-check"></i>1. Embedded Video</a></li>
<li class="chapter" data-level="" data-path="video_7.html"><a href="video_7.html#video-part-1-3"><i class="fa fa-check"></i>Video, Part 1</a></li>
<li class="chapter" data-level="" data-path="video_7.html"><a href="video_7.html#video-part-2-3"><i class="fa fa-check"></i>Video, Part 2</a></li>
<li class="chapter" data-level="" data-path="video_7.html"><a href="video_7.html#preview-slides-6"><i class="fa fa-check"></i>Preview Slides</a></li>
</ul></li>
<li class="chapter" data-level="10.2" data-path="tutorial_7.html"><a href="tutorial_7.html"><i class="fa fa-check"></i><b>10.2</b> Interactive Tutorial 7</a>
<ul>
<li class="chapter" data-level="" data-path="tutorial_7.html"><a href="tutorial_7.html#list-of-r-commands-covered-this-week-6"><i class="fa fa-check"></i>1. List of R commands covered this week</a></li>
<li class="chapter" data-level="" data-path="tutorial_7.html"><a href="tutorial_7.html#general-instructions-6"><i class="fa fa-check"></i>2. General Instructions</a></li>
</ul></li>
<li class="chapter" data-level="10.3" data-path="WE_7.html"><a href="WE_7.html"><i class="fa fa-check"></i><b>10.3</b> Worked Example</a>
<ul>
<li class="chapter" data-level="" data-path="WE_7.html"><a href="WE_7.html#overview-of-worked-example-6"><i class="fa fa-check"></i>1. Overview of Worked Example</a></li>
<li class="chapter" data-level="" data-path="WE_7.html"><a href="WE_7.html#explore-data-set"><i class="fa fa-check"></i>2. Explore data set</a></li>
<li class="chapter" data-level="" data-path="WE_7.html"><a href="WE_7.html#test-regression-residuals-for-spatial-autocorrelation"><i class="fa fa-check"></i>3. Test regression residuals for spatial autocorrelation</a></li>
<li class="chapter" data-level="" data-path="WE_7.html"><a href="WE_7.html#fit-models-with-spatially-correlated-error-gls-with-package-nlme"><i class="fa fa-check"></i>4. Fit models with spatially correlated error (GLS) with package ‘nlme’</a></li>
<li class="chapter" data-level="" data-path="WE_7.html"><a href="WE_7.html#fit-spatial-simultaneous-autoregressive-error-models-sar"><i class="fa fa-check"></i>5. Fit spatial simultaneous autoregressive error models (SAR)</a></li>
<li class="chapter" data-level="" data-path="WE_7.html"><a href="WE_7.html#spatial-filtering-with-mem-using-package-spmoran"><i class="fa fa-check"></i>6. Spatial filtering with MEM using package ‘spmoran’</a></li>
<li class="chapter" data-level="" data-path="WE_7.html"><a href="WE_7.html#fit-spatially-varying-coefficients-model-with-package-spmoran"><i class="fa fa-check"></i>7. Fit spatially varying coefficients model with package ‘spmoran’</a></li>
<li class="chapter" data-level="" data-path="WE_7.html"><a href="WE_7.html#model-with-patchsize-and-ibr"><i class="fa fa-check"></i>Model with PatchSize and IBR</a></li>
<li class="chapter" data-level="" data-path="WE_7.html"><a href="WE_7.html#model-with-ibr-only"><i class="fa fa-check"></i>Model with IBR only</a></li>
<li class="chapter" data-level="" data-path="WE_7.html"><a href="WE_7.html#conclusions"><i class="fa fa-check"></i>8. Conclusions</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="r-exercise-week-7.html"><a href="r-exercise-week-7.html"><i class="fa fa-check"></i>R Exercise Week 7</a></li>
<li class="chapter" data-level="10.4" data-path="bonus_7a.html"><a href="bonus_7a.html"><i class="fa fa-check"></i><b>10.4</b> Bonus: Si index</a>
<ul>
<li class="chapter" data-level="" data-path="bonus_7a.html"><a href="bonus_7a.html#overview-of-bonus-materials"><i class="fa fa-check"></i>1. Overview of Bonus Materials</a></li>
<li class="chapter" data-level="" data-path="bonus_7a.html"><a href="bonus_7a.html#import-ecological-distance-matrices"><i class="fa fa-check"></i>2. Import ecological distance matrices</a></li>
<li class="chapter" data-level="" data-path="bonus_7a.html"><a href="bonus_7a.html#optimize-the-scaling-parameter-alpha"><i class="fa fa-check"></i>3. Optimize the scaling parameter alpha</a></li>
<li class="chapter" data-level="" data-path="bonus_7a.html"><a href="bonus_7a.html#calculate-hanskis-index-si-with-source-patch-parameters"><i class="fa fa-check"></i>4. Calculate Hanski’s index Si with source patch parameters</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="11" data-path="Week8.html"><a href="Week8.html"><i class="fa fa-check"></i><b>11</b> Week 8: Simulation Experiments</a>
<ul>
<li class="chapter" data-level="11.1" data-path="video_8.html"><a href="video_8.html"><i class="fa fa-check"></i><b>11.1</b> View Course Video</a>
<ul>
<li class="chapter" data-level="" data-path="video_8.html"><a href="video_8.html#embedded-video-7"><i class="fa fa-check"></i>1. Embedded Video</a></li>
<li class="chapter" data-level="" data-path="video_8.html"><a href="video_8.html#video-part-1-4"><i class="fa fa-check"></i>Video, Part 1</a></li>
<li class="chapter" data-level="" data-path="video_8.html"><a href="video_8.html#video-part-2-4"><i class="fa fa-check"></i>Video, Part 2</a></li>
<li class="chapter" data-level="" data-path="video_8.html"><a href="video_8.html#preview-slides-7"><i class="fa fa-check"></i>Preview Slides</a></li>
</ul></li>
<li class="chapter" data-level="11.2" data-path="tutorial_8.html"><a href="tutorial_8.html"><i class="fa fa-check"></i><b>11.2</b> Interactive Tutorial 8</a>
<ul>
<li class="chapter" data-level="" data-path="tutorial_8.html"><a href="tutorial_8.html#list-of-r-commands-covered-this-week-7"><i class="fa fa-check"></i>1. List of R commands covered this week</a></li>
<li class="chapter" data-level="" data-path="tutorial_8.html"><a href="tutorial_8.html#general-instructions-7"><i class="fa fa-check"></i>2. General Instructions</a></li>
</ul></li>
<li class="chapter" data-level="11.3" data-path="WE_8.html"><a href="WE_8.html"><i class="fa fa-check"></i><b>11.3</b> Worked Example</a>
<ul>
<li class="chapter" data-level="" data-path="WE_8.html"><a href="WE_8.html#overview-of-worked-example-7"><i class="fa fa-check"></i>1. Overview of Worked Example</a></li>
<li class="chapter" data-level="" data-path="WE_8.html"><a href="WE_8.html#initialize-a-landscape"><i class="fa fa-check"></i>2. Initialize a landscape</a></li>
<li class="chapter" data-level="" data-path="WE_8.html"><a href="WE_8.html#define-simulation-parameters"><i class="fa fa-check"></i>3. Define simulation parameters</a></li>
<li class="chapter" data-level="" data-path="WE_8.html"><a href="WE_8.html#run-simulations-and-analyze-results"><i class="fa fa-check"></i>4. Run simulations and analyze results</a></li>
<li class="chapter" data-level="" data-path="WE_8.html"><a href="WE_8.html#run-simulator-using-a-previously-defined-parameter-set"><i class="fa fa-check"></i>5. Run simulator using a previously defined parameter set</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="r-exercise-week-8.html"><a href="r-exercise-week-8.html"><i class="fa fa-check"></i>R Exercise Week 8</a></li>
<li class="chapter" data-level="11.4" data-path="bonus_8a.html"><a href="bonus_8a.html"><i class="fa fa-check"></i><b>11.4</b> Bonus: Efficient R</a></li>
</ul></li>
<li class="part"><span><b>Advanced Topics</b></span></li>
<li class="chapter" data-level="12" data-path="Week9.html"><a href="Week9.html"><i class="fa fa-check"></i><b>12</b> Week 9: Population Structure</a>
<ul>
<li class="chapter" data-level="12.1" data-path="WE_9.html"><a href="WE_9.html"><i class="fa fa-check"></i><b>12.1</b> Worked Example</a>
<ul>
<li class="chapter" data-level="" data-path="WE_9.html"><a href="WE_9.html#overview-of-worked-example-8"><i class="fa fa-check"></i>1. Overview of Worked Example</a></li>
<li class="chapter" data-level="" data-path="WE_9.html"><a href="WE_9.html#simulated-data-2-island-model"><i class="fa fa-check"></i>2. Simulated data: 2-island model</a></li>
<li class="chapter" data-level="" data-path="WE_9.html"><a href="WE_9.html#simulated-data-2-island-model-with-admixture"><i class="fa fa-check"></i>3. Simulated data: 2-island model with admixture</a></li>
<li class="chapter" data-level="" data-path="WE_9.html"><a href="WE_9.html#empirical-data-threespine-sticklebacks"><i class="fa fa-check"></i>4. Empirical data: Threespine sticklebacks</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="13" data-path="Week10.html"><a href="Week10.html"><i class="fa fa-check"></i><b>13</b> Week 10: Landscape Resistance</a>
<ul>
<li class="chapter" data-level="13.1" data-path="WE_10.html"><a href="WE_10.html"><i class="fa fa-check"></i><b>13.1</b> Worked Example</a>
<ul>
<li class="chapter" data-level="" data-path="WE_10.html"><a href="WE_10.html#overview-of-worked-example-9"><i class="fa fa-check"></i>1. Overview of Worked Example</a></li>
<li class="chapter" data-level="" data-path="WE_10.html"><a href="WE_10.html#explore-the-data-set"><i class="fa fa-check"></i>2. Explore the data set</a></li>
<li class="chapter" data-level="" data-path="WE_10.html"><a href="WE_10.html#setting-cost-values-and-calculating-conductance"><i class="fa fa-check"></i>3. Setting cost values and calculating conductance</a></li>
<li class="chapter" data-level="" data-path="WE_10.html"><a href="WE_10.html#convert-conductance-into-effective-distance"><i class="fa fa-check"></i>4. Convert conductance into effective distance</a></li>
<li class="chapter" data-level="" data-path="WE_10.html"><a href="WE_10.html#create-cost-distance-matrices"><i class="fa fa-check"></i>5. Create cost-distance matrices</a></li>
<li class="chapter" data-level="" data-path="WE_10.html"><a href="WE_10.html#how-does-changing-resolution-affect-these-metrics"><i class="fa fa-check"></i>6. How does changing resolution affect these metrics?</a></li>
<li class="chapter" data-level="" data-path="WE_10.html"><a href="WE_10.html#references"><i class="fa fa-check"></i>7. References</a></li>
</ul></li>
<li class="chapter" data-level="13.2" data-path="bonus-10a.html"><a href="bonus-10a.html"><i class="fa fa-check"></i><b>13.2</b> Bonus: ‘radish’ tutorial</a>
<ul>
<li class="chapter" data-level="" data-path="bonus-10a.html"><a href="bonus-10a.html#overview-of-bonus-materials-1"><i class="fa fa-check"></i>1. Overview of Bonus Materials</a></li>
<li class="chapter" data-level="" data-path="bonus-10a.html"><a href="bonus-10a.html#import-data"><i class="fa fa-check"></i>2. Import data</a></li>
<li class="chapter" data-level="" data-path="bonus-10a.html"><a href="bonus-10a.html#basic-model-fitting"><i class="fa fa-check"></i>3. Basic model fitting</a></li>
<li class="chapter" data-level="" data-path="bonus-10a.html"><a href="bonus-10a.html#advanced-modeling-topics"><i class="fa fa-check"></i>4. Advanced modeling topics</a></li>
<li class="chapter" data-level="" data-path="bonus-10a.html"><a href="bonus-10a.html#summary"><i class="fa fa-check"></i>5. Summary</a></li>
<li class="chapter" data-level="" data-path="bonus-10a.html"><a href="bonus-10a.html#references-1"><i class="fa fa-check"></i>6. References</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="14" data-path="Week11.html"><a href="Week11.html"><i class="fa fa-check"></i><b>14</b> Week 11: Detecting Adaptation</a>
<ul>
<li class="chapter" data-level="14.1" data-path="WE_11.html"><a href="WE_11.html"><i class="fa fa-check"></i><b>14.1</b> Worked Example</a>
<ul>
<li class="chapter" data-level="" data-path="WE_11.html"><a href="WE_11.html#overview-of-worked-example-10"><i class="fa fa-check"></i>1. Overview of Worked Example</a></li>
<li class="chapter" data-level="" data-path="WE_11.html"><a href="WE_11.html#import-and-prepare-the-data"><i class="fa fa-check"></i>2. Import and prepare the data</a></li>
<li class="chapter" data-level="" data-path="WE_11.html"><a href="WE_11.html#latent-factor-mixed-models-lfmm-a-univariate-gea"><i class="fa fa-check"></i>3. Latent Factor Mixed Models (LFMM): a univariate GEA</a></li>
<li class="chapter" data-level="" data-path="WE_11.html"><a href="WE_11.html#redundancy-analysis-rda-a-multivariate-gea"><i class="fa fa-check"></i>4. Redundancy Analysis (RDA): a multivariate GEA</a></li>
<li class="chapter" data-level="" data-path="WE_11.html"><a href="WE_11.html#compare-lfmm-and-rda-candidates"><i class="fa fa-check"></i>5. Compare LFMM and RDA candidates</a></li>
<li class="chapter" data-level="" data-path="WE_11.html"><a href="WE_11.html#a-quick-note-on-controlling-for-population-structure-in-rda"><i class="fa fa-check"></i>6. A quick note on controlling for population structure in RDA</a></li>
<li class="chapter" data-level="" data-path="WE_11.html"><a href="WE_11.html#references-2"><i class="fa fa-check"></i>7. References</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="15" data-path="Week12.html"><a href="Week12.html"><i class="fa fa-check"></i><b>15</b> Week 12: Model Selection</a>
<ul>
<li class="chapter" data-level="15.1" data-path="WE_12.html"><a href="WE_12.html"><i class="fa fa-check"></i><b>15.1</b> Worked Example</a>
<ul>
<li class="chapter" data-level="" data-path="WE_12.html"><a href="WE_12.html#overview-of-worked-example-11"><i class="fa fa-check"></i>1. Overview of Worked Example</a></li>
<li class="chapter" data-level="" data-path="WE_12.html"><a href="WE_12.html#fitting-candidate-models"><i class="fa fa-check"></i>2. Fitting candidate models</a></li>
<li class="chapter" data-level="" data-path="WE_12.html"><a href="WE_12.html#your-turn-to-test-additional-hypotheses"><i class="fa fa-check"></i>3. Your turn to test additional hypotheses!</a></li>
<li class="chapter" data-level="" data-path="WE_12.html"><a href="WE_12.html#nested-model-nmlpe-for-hierarchical-sampling-designs"><i class="fa fa-check"></i>4. Nested model (NMLPE) for hierarchical sampling designs</a></li>
<li class="chapter" data-level="" data-path="WE_12.html"><a href="WE_12.html#references-3"><i class="fa fa-check"></i>5. References</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="16" data-path="Week13.html"><a href="Week13.html"><i class="fa fa-check"></i><b>16</b> Week 13: Gravity Models</a>
<ul>
<li class="chapter" data-level="16.1" data-path="WE_13.html"><a href="WE_13.html"><i class="fa fa-check"></i><b>16.1</b> Worked Example</a>
<ul>
<li class="chapter" data-level="" data-path="WE_13.html"><a href="WE_13.html#overview-of-worked-example-12"><i class="fa fa-check"></i>1. Overview of Worked Example</a></li>
<li class="chapter" data-level="" data-path="WE_13.html"><a href="WE_13.html#setup"><i class="fa fa-check"></i>2. Setup</a></li>
<li class="chapter" data-level="" data-path="WE_13.html"><a href="WE_13.html#wetland-complex-data-preparation"><i class="fa fa-check"></i>3. Wetland complex data preparation</a></li>
<li class="chapter" data-level="" data-path="WE_13.html"><a href="WE_13.html#wetland-field-data-preparation"><i class="fa fa-check"></i>4. Wetland field-data preparation</a></li>
<li class="chapter" data-level="" data-path="WE_13.html"><a href="WE_13.html#saturated-graph"><i class="fa fa-check"></i>5. Saturated Graph</a></li>
<li class="chapter" data-level="" data-path="WE_13.html"><a href="WE_13.html#spatial-model-data-prepration"><i class="fa fa-check"></i>6. Spatial model data prepration</a></li>
<li class="chapter" data-level="" data-path="WE_13.html"><a href="WE_13.html#gravity-model"><i class="fa fa-check"></i>7. Gravity model</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="17" data-path="Week14.html"><a href="Week14.html"><i class="fa fa-check"></i><b>17</b> Week 14: Contemporary Gene Flow</a>
<ul>
<li class="chapter" data-level="17.1" data-path="WE_14.html"><a href="WE_14.html"><i class="fa fa-check"></i><b>17.1</b> Worked Example</a>
<ul>
<li class="chapter" data-level="" data-path="WE_14.html"><a href="WE_14.html#overview-3"><i class="fa fa-check"></i>1. Overview</a></li>
<li class="chapter" data-level="" data-path="WE_14.html"><a href="WE_14.html#import-genotypes"><i class="fa fa-check"></i>2. Import Genotypes</a></li>
<li class="chapter" data-level="" data-path="WE_14.html"><a href="WE_14.html#pollen-pool-genetic-structure"><i class="fa fa-check"></i>3. Pollen pool genetic structure</a></li>
<li class="chapter" data-level="" data-path="WE_14.html"><a href="WE_14.html#paternity-analysis"><i class="fa fa-check"></i>4. Paternity analysis</a></li>
<li class="chapter" data-level="" data-path="WE_14.html"><a href="WE_14.html#linking-paternity-to-ecological-variables"><i class="fa fa-check"></i>5. Linking paternity to ecological variables</a></li>
<li class="chapter" data-level="" data-path="WE_14.html"><a href="WE_14.html#references-4"><i class="fa fa-check"></i>6. References</a></li>
</ul></li>
</ul></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Landscape Genetic Data Analysis with R</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="WE_13" class="section level2 hasAnchor" number="16.1">
<h2><span class="header-section-number">16.1</span> Worked Example<a href="WE_13.html#WE_13" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Melanie A. Murphy &amp; Jeffrey S. Evans</p>
<div id="overview-of-worked-example-12" class="section level3 unnumbered hasAnchor">
<h3>1. Overview of Worked Example<a href="WE_13.html#overview-of-worked-example-12" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div id="a.-background" class="section level4 unnumbered hasAnchor">
<h4>a. Background<a href="WE_13.html#a.-background" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>There are many ways graphs can be implemented to understand population structure and relate that structure to landscape characteristics (see Dyer and Nason 2004). In this exercise, we will calculate various graph metrics and apply graphs to fit a gravity model.</p>
<p>Gravity models are a type of inferential model that exploit graph characteristics. Gravity models include both at-site and between-site landscape data. They are a type of graph consisting of nodes and edges. These nodes and edges represent landscape characteristics associated with these graph elements.</p>
</div>
<div id="b.-data-set-11" class="section level4 unnumbered hasAnchor">
<h4>b. Data set<a href="WE_13.html#b.-data-set-11" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>In this exercise, you will use the gravity model framework to build an empirical model of gene flow for the Columbia spotted frog dataset in central Idaho that you have used for several other exercises (Murphy et al. 2010).</p>
</div>
</div>
<div id="setup" class="section level3 unnumbered hasAnchor">
<h3>2. Setup<a href="WE_13.html#setup" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div id="add-required-packages" class="section level4 unnumbered hasAnchor">
<h4>Add required packages<a href="WE_13.html#add-required-packages" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>This code checks of all required packages are installed, and if so, loads them. If any are missing, it will return a message that identifies the packages that need to be installed. If that happens, install the packages and run the code again.</p>
<p>The package then checks if your version of the <code>GeNetIt</code> package is up-to-date (at least 0.1-5), and if not, installs the latest version.</p>
<div class="sourceCode" id="cb1328"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1328-1"><a href="WE_13.html#cb1328-1" tabindex="-1"></a>p <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;raster&quot;</span>, <span class="st">&quot;igraph&quot;</span>, <span class="st">&quot;sp&quot;</span>, <span class="st">&quot;GeNetIt&quot;</span>, <span class="st">&quot;spatialEco&quot;</span>, <span class="st">&quot;leaflet&quot;</span>,</span>
<span id="cb1328-2"><a href="WE_13.html#cb1328-2" tabindex="-1"></a>       <span class="st">&quot;sf&quot;</span>, <span class="st">&quot;terra&quot;</span>, <span class="st">&quot;sfnetworks&quot;</span>, <span class="st">&quot;spdep&quot;</span>, <span class="st">&quot;dplyr&quot;</span>, <span class="st">&quot;tmap&quot;</span>, <span class="st">&quot;devtools&quot;</span>) </span>
<span id="cb1328-3"><a href="WE_13.html#cb1328-3" tabindex="-1"></a>  <span class="cf">if</span>(<span class="fu">any</span>(<span class="sc">!</span><span class="fu">unlist</span>(<span class="fu">lapply</span>(p, requireNamespace, <span class="at">quietly=</span><span class="cn">TRUE</span>)))) { </span>
<span id="cb1328-4"><a href="WE_13.html#cb1328-4" tabindex="-1"></a>    m <span class="ot">=</span> <span class="fu">which</span>(<span class="sc">!</span><span class="fu">unlist</span>(<span class="fu">lapply</span>(p, requireNamespace, <span class="at">quietly=</span><span class="cn">TRUE</span>)))</span>
<span id="cb1328-5"><a href="WE_13.html#cb1328-5" tabindex="-1"></a>      <span class="fu">suppressMessages</span>(<span class="fu">invisible</span>(<span class="fu">lapply</span>(p[<span class="sc">-</span>m], require,    </span>
<span id="cb1328-6"><a href="WE_13.html#cb1328-6" tabindex="-1"></a>                       <span class="at">character.only=</span><span class="cn">TRUE</span>)))</span>
<span id="cb1328-7"><a href="WE_13.html#cb1328-7" tabindex="-1"></a>    <span class="fu">stop</span>(<span class="st">&quot;Missing library, please install &quot;</span>, <span class="fu">paste</span>(p[m], <span class="at">collapse =</span> <span class="st">&quot; &quot;</span>))</span>
<span id="cb1328-8"><a href="WE_13.html#cb1328-8" tabindex="-1"></a>  } <span class="cf">else</span> {</span>
<span id="cb1328-9"><a href="WE_13.html#cb1328-9" tabindex="-1"></a>    <span class="cf">if</span>(<span class="fu">packageVersion</span>(<span class="st">&quot;GeNetIt&quot;</span>) <span class="sc">&lt;</span> <span class="st">&quot;0.1-5&quot;</span>) {</span>
<span id="cb1328-10"><a href="WE_13.html#cb1328-10" tabindex="-1"></a>      remotes<span class="sc">::</span><span class="fu">install_github</span>(<span class="st">&quot;jeffreyevans/GeNetIt&quot;</span>)</span>
<span id="cb1328-11"><a href="WE_13.html#cb1328-11" tabindex="-1"></a>    } </span>
<span id="cb1328-12"><a href="WE_13.html#cb1328-12" tabindex="-1"></a>    <span class="fu">suppressMessages</span>(<span class="fu">invisible</span>(<span class="fu">lapply</span>(p, require, <span class="at">character.only=</span><span class="cn">TRUE</span>)))</span>
<span id="cb1328-13"><a href="WE_13.html#cb1328-13" tabindex="-1"></a>  }</span></code></pre></div>
</div>
<div id="prepare-work-environment" class="section level4 unnumbered hasAnchor">
<h4>Prepare work environment<a href="WE_13.html#prepare-work-environment" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<div class="sourceCode" id="cb1329"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1329-1"><a href="WE_13.html#cb1329-1" tabindex="-1"></a><span class="co"># Get the path to your project folder</span></span>
<span id="cb1329-2"><a href="WE_13.html#cb1329-2" tabindex="-1"></a>Path <span class="ot">&lt;-</span> here<span class="sc">::</span><span class="fu">here</span>(<span class="st">&quot;downloads&quot;</span>)</span>
<span id="cb1329-3"><a href="WE_13.html#cb1329-3" tabindex="-1"></a></span>
<span id="cb1329-4"><a href="WE_13.html#cb1329-4" tabindex="-1"></a><span class="do">## i.dicates UTM 11 NAD83 projection</span></span>
<span id="cb1329-5"><a href="WE_13.html#cb1329-5" tabindex="-1"></a>prj <span class="ot">=</span> <span class="dv">32611</span> </span>
<span id="cb1329-6"><a href="WE_13.html#cb1329-6" tabindex="-1"></a></span>
<span id="cb1329-7"><a href="WE_13.html#cb1329-7" tabindex="-1"></a><span class="co"># Some needed functions</span></span>
<span id="cb1329-8"><a href="WE_13.html#cb1329-8" tabindex="-1"></a>back.transform <span class="ot">&lt;-</span> <span class="cf">function</span>(y) <span class="fu">exp</span>(y <span class="sc">+</span> <span class="fl">0.5</span> <span class="sc">*</span> stats<span class="sc">::</span><span class="fu">var</span>(y))</span>
<span id="cb1329-9"><a href="WE_13.html#cb1329-9" tabindex="-1"></a>rmse <span class="ot">=</span> <span class="cf">function</span>(p, o){ <span class="fu">sqrt</span>(<span class="fu">mean</span>((p <span class="sc">-</span> o)<span class="sc">^</span><span class="dv">2</span>)) }</span></code></pre></div>
</div>
</div>
<div id="wetland-complex-data-preparation" class="section level3 unnumbered hasAnchor">
<h3>3. Wetland complex data preparation<a href="WE_13.html#wetland-complex-data-preparation" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>In this sections, we will read in all wetland locations in the study area and calculate a few graph-based metrics to assign to wetland sites that data was collected at. This allows us to put our samples into the context of the larger wetland system thus, accounting for proximity and juxtaposition.</p>
<div id="a.-read-in-wetlands-data" class="section level4 unnumbered hasAnchor">
<h4>a. Read in wetlands data<a href="WE_13.html#a.-read-in-wetlands-data" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>Import file “Wetlands.csv”.</p>
<div class="sourceCode" id="cb1330"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1330-1"><a href="WE_13.html#cb1330-1" tabindex="-1"></a>wetlands <span class="ot">&lt;-</span> <span class="fu">read.csv</span>(<span class="fu">system.file</span>(<span class="st">&quot;extdata/Wetlands.csv&quot;</span>, <span class="at">package=</span><span class="st">&quot;LandGenCourse&quot;</span>), </span>
<span id="cb1330-2"><a href="WE_13.html#cb1330-2" tabindex="-1"></a>                    <span class="at">header =</span> <span class="cn">TRUE</span>)</span>
<span id="cb1330-3"><a href="WE_13.html#cb1330-3" tabindex="-1"></a><span class="fu">head</span>(wetlands)</span></code></pre></div>
<pre><code>##   ID      X       Y RALU          SiteName
## 1  1 688835 5002939    y      AirplaneLake
## 2  2 687460 4994400    n  AlpineInletCreek
## 3  3 687507 4994314    n AlpineInletMeadow
## 4  4 687637 4994117    n        AlpineLake
## 5  5 688850 4997750    n   AxeHandleMeadow
## 6  6 688500 4998900    y    BachelorMeadow</code></pre>
<p>Make it a spatial object.</p>
<div class="sourceCode" id="cb1332"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1332-1"><a href="WE_13.html#cb1332-1" tabindex="-1"></a>wetlands <span class="ot">&lt;-</span> <span class="fu">st_as_sf</span>(wetlands, <span class="at">coords =</span> <span class="fu">c</span>(<span class="st">&quot;X&quot;</span>, <span class="st">&quot;Y&quot;</span>), </span>
<span id="cb1332-2"><a href="WE_13.html#cb1332-2" tabindex="-1"></a>                     <span class="at">crs =</span> <span class="dv">32611</span>, <span class="at">agr =</span> <span class="st">&quot;constant&quot;</span>) </span>
<span id="cb1332-3"><a href="WE_13.html#cb1332-3" tabindex="-1"></a><span class="fu">str</span>(wetlands)</span></code></pre></div>
<pre><code>## Classes &#39;sf&#39; and &#39;data.frame&#39;:   121 obs. of  4 variables:
##  $ ID      : int  1 2 3 4 5 6 7 8 9 10 ...
##  $ RALU    : chr  &quot;y&quot; &quot;n&quot; &quot;n&quot; &quot;n&quot; ...
##  $ SiteName: chr  &quot;AirplaneLake&quot; &quot;AlpineInletCreek&quot; &quot;AlpineInletMeadow&quot; &quot;AlpineLake&quot; ...
##  $ geometry:sfc_POINT of length 121; first list element:  &#39;XY&#39; num  688835 5002939
##  - attr(*, &quot;sf_column&quot;)= chr &quot;geometry&quot;
##  - attr(*, &quot;agr&quot;)= Factor w/ 3 levels &quot;constant&quot;,&quot;aggregate&quot;,..: 1 1 1
##   ..- attr(*, &quot;names&quot;)= chr [1:3] &quot;ID&quot; &quot;RALU&quot; &quot;SiteName&quot;</code></pre>
</div>
<div id="b.-create-wetlands-graph" class="section level4 unnumbered hasAnchor">
<h4>b. Create wetlands graph<a href="WE_13.html#b.-create-wetlands-graph" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>Create Gabriel graph from the wetlands to represent a “realization” of connectivity and spatial arrangement.</p>
<p>Derive Gabriel graph</p>
<div class="sourceCode" id="cb1334"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1334-1"><a href="WE_13.html#cb1334-1" tabindex="-1"></a>gg <span class="ot">&lt;-</span> <span class="fu">graph2nb</span>(<span class="fu">gabrielneigh</span>(<span class="fu">st_coordinates</span>(wetlands)),<span class="at">sym=</span><span class="cn">TRUE</span>)</span>
<span id="cb1334-2"><a href="WE_13.html#cb1334-2" tabindex="-1"></a></span>
<span id="cb1334-3"><a href="WE_13.html#cb1334-3" tabindex="-1"></a><span class="fu">plot</span>(gg, <span class="at">coords=</span><span class="fu">st_coordinates</span>(wetlands))</span></code></pre></div>
<p><img src="13-Week13_files/figure-html/gabriel-1.png" width="672" /></p>
<p><strong>Questions</strong>:</p>
<ul>
<li>This graph may or may not be the best representation of wetland connectivity. What other types of graphs could you build?</li>
<li>What wetlands are connected to each other based on the graph?</li>
</ul>
</div>
<div id="c.-graph-metrics" class="section level4 unnumbered hasAnchor">
<h4>c. Graph metrics<a href="WE_13.html#c.-graph-metrics" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>To calculate graph metrics, we need to do a few steps:</p>
<p>Coerce to sf line object (will be used to create igraph object)</p>
<div class="sourceCode" id="cb1335"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1335-1"><a href="WE_13.html#cb1335-1" tabindex="-1"></a>gg <span class="ot">&lt;-</span> <span class="fu">nb2lines</span>(gg, <span class="at">coords =</span> sf<span class="sc">::</span><span class="fu">st_coordinates</span>(wetlands), </span>
<span id="cb1335-2"><a href="WE_13.html#cb1335-2" tabindex="-1"></a>                 <span class="at">proj4string =</span> prj, <span class="at">as_sf=</span><span class="cn">TRUE</span>)</span></code></pre></div>
<p>Coerce to a <code>sfnetwork</code>, which is an <code>igraph</code> object:</p>
<div class="sourceCode" id="cb1336"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1336-1"><a href="WE_13.html#cb1336-1" tabindex="-1"></a>wg <span class="ot">&lt;-</span> <span class="fu">as_sfnetwork</span>(gg, <span class="at">edges=</span>gg, <span class="at">nodes=</span>wetlands, <span class="at">directed =</span> <span class="cn">FALSE</span>,</span>
<span id="cb1336-2"><a href="WE_13.html#cb1336-2" tabindex="-1"></a>                  <span class="at">node_key =</span> <span class="st">&quot;SiteName&quot;</span>, <span class="at">length_as_weight =</span> <span class="cn">TRUE</span>, </span>
<span id="cb1336-3"><a href="WE_13.html#cb1336-3" tabindex="-1"></a>                          <span class="at">edges_as_lines =</span> <span class="cn">TRUE</span>)</span></code></pre></div>
<p>Calculate weights</p>
<div class="sourceCode" id="cb1337"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1337-1"><a href="WE_13.html#cb1337-1" tabindex="-1"></a>w <span class="ot">&lt;-</span> wg <span class="sc">%&gt;%</span> <span class="fu">activate</span>(<span class="st">&quot;edges&quot;</span>) <span class="sc">%&gt;%</span> <span class="fu">pull</span>(weight) <span class="sc">%&gt;%</span> <span class="fu">as.numeric</span>()</span>
<span id="cb1337-2"><a href="WE_13.html#cb1337-2" tabindex="-1"></a>w[w <span class="sc">&lt;=</span> <span class="dv">0</span>] <span class="ot">&lt;-</span> <span class="dv">1</span> </span>
<span id="cb1337-3"><a href="WE_13.html#cb1337-3" tabindex="-1"></a>w <span class="ot">=</span> w <span class="sc">/</span> <span class="fu">sum</span>(w)</span></code></pre></div>
<p>Calculate graph metrics of betweenness and closeness with weights and degree. We’ll add these as attributes (columns) to the wetlands <code>sf</code> object.</p>
<ul>
<li><strong>degree</strong> - the number of connections a node has</li>
<li><strong>betweenness</strong> - the number of shortest paths going through a node</li>
<li><strong>closensess</strong> - the average of the shortest path length from the node to every other node in the network</li>
</ul>
<div class="sourceCode" id="cb1338"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1338-1"><a href="WE_13.html#cb1338-1" tabindex="-1"></a>wetlands<span class="sc">$</span>betweenness <span class="ot">&lt;-</span> igraph<span class="sc">::</span><span class="fu">betweenness</span>(wg, <span class="at">directed=</span><span class="cn">FALSE</span>, <span class="at">weights=</span>w)</span>
<span id="cb1338-2"><a href="WE_13.html#cb1338-2" tabindex="-1"></a>wetlands<span class="sc">$</span>degree <span class="ot">&lt;-</span> igraph<span class="sc">::</span><span class="fu">degree</span>(wg)</span>
<span id="cb1338-3"><a href="WE_13.html#cb1338-3" tabindex="-1"></a>wetlands<span class="sc">$</span>closeness <span class="ot">&lt;-</span> igraph<span class="sc">::</span><span class="fu">closeness</span>(wg, <span class="at">weights=</span>w)</span>
<span id="cb1338-4"><a href="WE_13.html#cb1338-4" tabindex="-1"></a>wetlands</span></code></pre></div>
<pre><code>## Simple feature collection with 121 features and 6 fields
## Attribute-geometry relationships: constant (3), NA&#39;s (3)
## Geometry type: POINT
## Dimension:     XY
## Bounding box:  xmin: 686244 ymin: 4993077 xmax: 695699 ymax: 5004317
## Projected CRS: WGS 84 / UTM zone 11N
## First 10 features:
##    ID RALU          SiteName               geometry betweenness degree
## 1   1    y      AirplaneLake POINT (688835 5002939)         288      8
## 2   2    n  AlpineInletCreek POINT (687460 4994400)         443      6
## 3   3    n AlpineInletMeadow POINT (687507 4994314)         332      4
## 4   4    n        AlpineLake POINT (687637 4994117)         220      6
## 5   5    n   AxeHandleMeadow POINT (688850 4997750)        1690      6
## 6   6    y    BachelorMeadow POINT (688500 4998900)         195      4
## 7   7    y    BarkingFoxLake POINT (687944 5000006)        1323      6
## 8   8    n BarkingFoxWetland POINT (687872 5000041)        1291     10
## 9   9    n          BartPond POINT (687150 4995850)           0      6
## 10 10    n      BigClearLake POINT (690888 5004126)           0      4
##    closeness
## 1  0.3538381
## 2  0.3407043
## 3  0.3352415
## 4  0.3226762
## 5  0.5426490
## 6  0.5334932
## 7  0.5094841
## 8  0.5066511
## 9  0.3783181
## 10 0.2706522</code></pre>
</div>
<div id="d.-plot-graph-metric" class="section level4 unnumbered hasAnchor">
<h4>d. Plot graph metric<a href="WE_13.html#d.-plot-graph-metric" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>Plot results using the graph edges and wetlands points with the attributes “betweenness”, “closeness” and “degree”.</p>
<p>Plot betweenness</p>
<div class="sourceCode" id="cb1340"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1340-1"><a href="WE_13.html#cb1340-1" tabindex="-1"></a><span class="fu">plot</span>(<span class="fu">st_geometry</span>(gg), <span class="at">col=</span><span class="st">&quot;grey&quot;</span>)</span>
<span id="cb1340-2"><a href="WE_13.html#cb1340-2" tabindex="-1"></a>  <span class="fu">plot</span>(wetlands[<span class="st">&quot;betweenness&quot;</span>], <span class="at">pch=</span><span class="dv">19</span>,  </span>
<span id="cb1340-3"><a href="WE_13.html#cb1340-3" tabindex="-1"></a>       <span class="at">cex=</span><span class="fl">0.75</span>, <span class="at">add=</span><span class="cn">TRUE</span>)</span>
<span id="cb1340-4"><a href="WE_13.html#cb1340-4" tabindex="-1"></a>     <span class="fu">box</span>()</span>
<span id="cb1340-5"><a href="WE_13.html#cb1340-5" tabindex="-1"></a>     <span class="fu">title</span>(<span class="st">&quot;Wetlands Gabriel graph betweenness&quot;</span>)</span></code></pre></div>
<p><img src="13-Week13_files/figure-html/plot_btwn-1.png" width="672" /></p>
<p>Plot closeness</p>
<div class="sourceCode" id="cb1341"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1341-1"><a href="WE_13.html#cb1341-1" tabindex="-1"></a><span class="fu">plot</span>(<span class="fu">st_geometry</span>(gg), <span class="at">col=</span><span class="st">&quot;grey&quot;</span>)</span>
<span id="cb1341-2"><a href="WE_13.html#cb1341-2" tabindex="-1"></a><span class="fu">plot</span>(wetlands[<span class="st">&quot;closeness&quot;</span>], <span class="at">pch=</span><span class="dv">19</span>,  </span>
<span id="cb1341-3"><a href="WE_13.html#cb1341-3" tabindex="-1"></a>     <span class="at">cex=</span><span class="fl">0.75</span>, <span class="at">add=</span><span class="cn">TRUE</span>)</span>
<span id="cb1341-4"><a href="WE_13.html#cb1341-4" tabindex="-1"></a><span class="fu">box</span>()</span>
<span id="cb1341-5"><a href="WE_13.html#cb1341-5" tabindex="-1"></a><span class="fu">title</span>(<span class="st">&quot;Wetlands Gabriel graph closeness&quot;</span>)</span></code></pre></div>
<p><img src="13-Week13_files/figure-html/plot_clsn-1.png" width="672" /></p>
<p>Plot degree</p>
<div class="sourceCode" id="cb1342"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1342-1"><a href="WE_13.html#cb1342-1" tabindex="-1"></a><span class="fu">plot</span>(<span class="fu">st_geometry</span>(gg), <span class="at">col=</span><span class="st">&quot;grey&quot;</span>)</span>
<span id="cb1342-2"><a href="WE_13.html#cb1342-2" tabindex="-1"></a><span class="fu">plot</span>(wetlands[<span class="st">&quot;degree&quot;</span>], <span class="at">pch=</span><span class="dv">19</span>,  </span>
<span id="cb1342-3"><a href="WE_13.html#cb1342-3" tabindex="-1"></a>     <span class="at">cex=</span><span class="fl">0.75</span>, <span class="at">add=</span><span class="cn">TRUE</span>)</span>
<span id="cb1342-4"><a href="WE_13.html#cb1342-4" tabindex="-1"></a><span class="fu">box</span>()</span>
<span id="cb1342-5"><a href="WE_13.html#cb1342-5" tabindex="-1"></a><span class="fu">title</span>(<span class="st">&quot;Wetlands Gabriel graph degree&quot;</span>)</span></code></pre></div>
<p><img src="13-Week13_files/figure-html/plot_dgr-1.png" width="672" /></p>
<p><strong>Questions:</strong> Consider the three figures above and how the graph was constructed.</p>
<ul>
<li>In what way(s) is the resulting graph potentially ecologically meaningful?</li>
<li>How might it not be ecologically or biologically meaningful?</li>
</ul>
</div>
</div>
<div id="wetland-field-data-preparation" class="section level3 unnumbered hasAnchor">
<h3>4. Wetland field-data preparation<a href="WE_13.html#wetland-field-data-preparation" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>In this section we will read the field data and add the node metrics we just calculated.</p>
<p>Using <code>RALU_Site.csv</code>, we read in the data, add the node data (betweenness and degree), create a spatial object that includes the node data.</p>
<p>Read in site data</p>
<div class="sourceCode" id="cb1343"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1343-1"><a href="WE_13.html#cb1343-1" tabindex="-1"></a>sites <span class="ot">&lt;-</span> <span class="fu">read.csv</span>(<span class="fu">system.file</span>(<span class="st">&quot;extdata/RALU_Site.csv&quot;</span>, <span class="at">package=</span><span class="st">&quot;LandGenCourse&quot;</span>), </span>
<span id="cb1343-2"><a href="WE_13.html#cb1343-2" tabindex="-1"></a>                  <span class="at">header =</span> <span class="cn">TRUE</span>)</span>
<span id="cb1343-3"><a href="WE_13.html#cb1343-3" tabindex="-1"></a>sites<span class="sc">$</span>SiteID <span class="ot">&lt;-</span> <span class="fu">as.character</span>(sites<span class="sc">$</span>SiteID)</span></code></pre></div>
<p>Add the node data.</p>
<p>Note: using names is dangerous as, small changes in names can result in non-matches. In this case, the ID fields are not consistent (data were collected at different times for different purposes originally). However, names are standardized in a drop-down list of a database. So they are a matching field. My recommendation to you all is to do to this type of operation on a numeric field.</p>
<div class="sourceCode" id="cb1344"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1344-1"><a href="WE_13.html#cb1344-1" tabindex="-1"></a>nodestats <span class="ot">&lt;-</span> <span class="fu">st_drop_geometry</span>(wetlands[,<span class="fu">c</span>(<span class="dv">3</span>,<span class="dv">5</span><span class="sc">:</span><span class="dv">7</span>)])</span>
<span id="cb1344-2"><a href="WE_13.html#cb1344-2" tabindex="-1"></a>  nodestats <span class="ot">&lt;-</span> nodestats[<span class="fu">which</span>(nodestats<span class="sc">$</span>SiteName <span class="sc">%in%</span> sites<span class="sc">$</span>SiteName),] </span>
<span id="cb1344-3"><a href="WE_13.html#cb1344-3" tabindex="-1"></a>sites <span class="ot">&lt;-</span> <span class="fu">merge</span>(nodestats, sites, <span class="at">by=</span><span class="st">&quot;SiteName&quot;</span>)</span></code></pre></div>
<p>Convert data frame to <code>sf</code> object.</p>
<div class="sourceCode" id="cb1345"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1345-1"><a href="WE_13.html#cb1345-1" tabindex="-1"></a>sites <span class="ot">&lt;-</span> <span class="fu">st_as_sf</span>(sites, <span class="at">coords =</span> <span class="fu">c</span>(<span class="st">&quot;X&quot;</span>, <span class="st">&quot;Y&quot;</span>), </span>
<span id="cb1345-2"><a href="WE_13.html#cb1345-2" tabindex="-1"></a>                 <span class="at">crs =</span> prj, <span class="at">agr =</span> <span class="st">&quot;constant&quot;</span>) </span>
<span id="cb1345-3"><a href="WE_13.html#cb1345-3" tabindex="-1"></a><span class="fu">head</span>(sites)</span></code></pre></div>
<pre><code>## Simple feature collection with 6 features and 14 fields
## Attribute-geometry relationships: constant (14)
## Geometry type: POINT
## Dimension:     XY
## Bounding box:  xmin: 687944 ymin: 4996458 xmax: 690127 ymax: 5002939
## Projected CRS: WGS 84 / UTM zone 11N
##         SiteName betweenness degree closeness SiteID     Elev Length    Area
## 1   AirplaneLake         288      8 0.3538381     27 2564.381    390 62582.2
## 2 BachelorMeadow         195      4 0.5334932     15 2591.781      0   225.0
## 3 BarkingFoxLake        1323      6 0.5094841     19 2545.275    160 12000.0
## 4        BobLake         116      4 0.4652124     16 2649.125    143  4600.0
## 5      CacheLake         673      4 0.5189996     10 2475.829     75  2268.8
## 6        DoeLake         758      6 0.4710215      8 2463.006    170 13034.9
##    Perim Depth  pH Dforest Drock Dshrub               geometry
## 1 1142.8 21.64 6.5   0.398 0.051   0.00 POINT (688835 5002939)
## 2   60.0  0.40 6.1   0.000 0.000   0.20 POINT (688500 4998900)
## 3  435.0  5.00 6.5   0.400 0.250   0.05 POINT (687944 5000006)
## 4  321.4  2.00 7.0   0.550 0.000   0.05 POINT (690127 4999150)
## 5  192.0  1.86 6.5   0.508 0.000   0.00 POINT (688777 4997264)
## 6  463.2  6.03 7.6   0.254 0.000   0.00 POINT (688968 4996458)</code></pre>
<p><strong>Question</strong>:</p>
<ul>
<li>What are the fields here? What data are included in <code>sites</code>?</li>
</ul>
</div>
<div id="saturated-graph" class="section level3 unnumbered hasAnchor">
<h3>5. Saturated Graph<a href="WE_13.html#saturated-graph" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div id="a.-create-graph-from-site-locations" class="section level4 unnumbered hasAnchor">
<h4>a. Create graph from site locations<a href="WE_13.html#a.-create-graph-from-site-locations" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>To assess connectivity using a gravity model, we need to build a graph from the occupied frog sites create a graph. This could be any type of graph, but I generally use saturated (essentially a full genetic distance matrix) or pruned by some maximum distance.</p>
<div class="sourceCode" id="cb1347"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1347-1"><a href="WE_13.html#cb1347-1" tabindex="-1"></a>dist.graph <span class="ot">&lt;-</span> <span class="fu">knn.graph</span>(sites, <span class="at">row.names =</span> sites<span class="sc">$</span>SiteName)</span>
<span id="cb1347-2"><a href="WE_13.html#cb1347-2" tabindex="-1"></a>dist.graph <span class="ot">&lt;-</span> <span class="fu">merge</span>(dist.graph, <span class="fu">st_drop_geometry</span>(sites), </span>
<span id="cb1347-3"><a href="WE_13.html#cb1347-3" tabindex="-1"></a>                      <span class="at">by.y=</span><span class="st">&quot;SiteName&quot;</span>, <span class="at">by.x=</span><span class="st">&quot;from_ID&quot;</span>)</span>
<span id="cb1347-4"><a href="WE_13.html#cb1347-4" tabindex="-1"></a>dist.graph <span class="ot">&lt;-</span> dist.graph[,<span class="sc">-</span><span class="fu">c</span>(<span class="dv">11</span><span class="sc">:</span><span class="dv">19</span>)] <span class="do">## d.op extra columns</span></span></code></pre></div>
<p>Note: Can create a distance-constrained graph with max.dist arg (not run)</p>
<div class="sourceCode" id="cb1348"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1348-1"><a href="WE_13.html#cb1348-1" tabindex="-1"></a><span class="do">## d.st.graph &lt;- knn.graph(sites, row.names = sites$SiteName, max.dist=5000)</span></span></code></pre></div>
</div>
<div id="b.-merge-the-graph-with-genetic-distance" class="section level4 unnumbered hasAnchor">
<h4>b. Merge the graph with genetic distance<a href="WE_13.html#b.-merge-the-graph-with-genetic-distance" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>Read in the genetic distance data and make a matrix, gdist, then unfold data</p>
<div class="sourceCode" id="cb1349"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1349-1"><a href="WE_13.html#cb1349-1" tabindex="-1"></a>gdist <span class="ot">&lt;-</span> <span class="fu">read.csv</span>(<span class="fu">system.file</span>(<span class="st">&quot;extdata/RALU_Dps.csv&quot;</span>, <span class="at">package=</span><span class="st">&quot;LandGenCourse&quot;</span>), <span class="at">header=</span><span class="cn">TRUE</span>) </span>
<span id="cb1349-2"><a href="WE_13.html#cb1349-2" tabindex="-1"></a>  <span class="fu">rownames</span>(gdist) <span class="ot">&lt;-</span> <span class="fu">t</span>(<span class="fu">names</span>(gdist))</span>
<span id="cb1349-3"><a href="WE_13.html#cb1349-3" tabindex="-1"></a>gdist <span class="ot">&lt;-</span> <span class="fu">dmatrix.df</span>(<span class="fu">as.matrix</span>(gdist)) </span>
<span id="cb1349-4"><a href="WE_13.html#cb1349-4" tabindex="-1"></a><span class="fu">names</span>(gdist) <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;FROM&quot;</span>, <span class="st">&quot;TO&quot;</span>, <span class="st">&quot;GDIST&quot;</span>) <span class="co">#unfold the file</span></span>
<span id="cb1349-5"><a href="WE_13.html#cb1349-5" tabindex="-1"></a>gdist <span class="ot">&lt;-</span> gdist[<span class="sc">!</span>gdist<span class="sc">$</span>FROM <span class="sc">==</span> gdist<span class="sc">$</span>TO ,]</span>
<span id="cb1349-6"><a href="WE_13.html#cb1349-6" tabindex="-1"></a>gdist[,<span class="dv">1</span>] <span class="ot">&lt;-</span><span class="fu">sub</span>(<span class="st">&quot;X&quot;</span>, <span class="st">&quot;&quot;</span>, gdist[,<span class="dv">1</span>])</span>
<span id="cb1349-7"><a href="WE_13.html#cb1349-7" tabindex="-1"></a>gdist[,<span class="dv">2</span>] <span class="ot">&lt;-</span><span class="fu">sub</span>(<span class="st">&quot;X&quot;</span>, <span class="st">&quot;&quot;</span>, gdist[,<span class="dv">2</span>])</span>
<span id="cb1349-8"><a href="WE_13.html#cb1349-8" tabindex="-1"></a>gdist <span class="ot">&lt;-</span> <span class="fu">cbind</span>(<span class="at">from.to=</span><span class="fu">paste</span>(gdist[,<span class="dv">1</span>], gdist[,<span class="dv">2</span>], <span class="at">sep=</span><span class="st">&quot;.&quot;</span>), gdist)</span></code></pre></div>
<p>Transform genetic distance to genetic flow (1-distance)</p>
<div class="sourceCode" id="cb1350"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1350-1"><a href="WE_13.html#cb1350-1" tabindex="-1"></a>gdist<span class="sc">$</span>GDIST <span class="ot">&lt;-</span> <span class="fu">flow</span>(gdist<span class="sc">$</span>GDIST)</span></code></pre></div>
<p>Merge graph with genetic distances, based on from node - to node combination</p>
<div class="sourceCode" id="cb1351"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1351-1"><a href="WE_13.html#cb1351-1" tabindex="-1"></a>dist.graph<span class="sc">$</span>from.to <span class="ot">&lt;-</span> <span class="fu">paste</span>(dist.graph<span class="sc">$</span>i, dist.graph<span class="sc">$</span>j, <span class="at">sep=</span><span class="st">&quot;.&quot;</span>)</span>
<span id="cb1351-2"><a href="WE_13.html#cb1351-2" tabindex="-1"></a>dist.graph <span class="ot">&lt;-</span> <span class="fu">merge</span>(dist.graph, gdist, <span class="at">by =</span> <span class="st">&quot;from.to&quot;</span>)</span></code></pre></div>
<p><strong>Question</strong>:</p>
<ul>
<li>What is in the resulting object?</li>
</ul>
</div>
</div>
<div id="spatial-model-data-prepration" class="section level3 unnumbered hasAnchor">
<h3>6. Spatial model data prepration<a href="WE_13.html#spatial-model-data-prepration" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div id="a.-read-raster-data-using-terra" class="section level4 unnumbered hasAnchor">
<h4>a. Read raster data using ‘terra’<a href="WE_13.html#a.-read-raster-data-using-terra" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<div class="sourceCode" id="cb1352"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1352-1"><a href="WE_13.html#cb1352-1" tabindex="-1"></a>xvars <span class="ot">&lt;-</span> terra<span class="sc">::</span><span class="fu">rast</span>(<span class="fu">system.file</span>(<span class="st">&quot;extdata/covariates.tif&quot;</span>, <span class="at">package=</span><span class="st">&quot;GeNetIt&quot;</span>))</span></code></pre></div>
</div>
<div id="b.-reclassify-wetlands" class="section level4 unnumbered hasAnchor">
<h4>b. Reclassify wetlands<a href="WE_13.html#b.-reclassify-wetlands" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>NLCD is land cover data. The wetland classes are 11 and 90-95 (for this system and vintage NLCD data)</p>
<div class="sourceCode" id="cb1353"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1353-1"><a href="WE_13.html#cb1353-1" tabindex="-1"></a>m <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="dv">0</span>,<span class="fl">10.8</span>, <span class="dv">0</span>,<span class="fl">10.9</span>,<span class="fl">12.1</span>,<span class="dv">1</span>,<span class="fl">12.9</span>,<span class="fl">89.5</span>,<span class="dv">0</span>, <span class="fl">89.1</span>,<span class="fl">95.1</span>,<span class="dv">1</span>,<span class="fl">95.9</span>,<span class="dv">100</span>,<span class="dv">0</span> )</span>
<span id="cb1353-2"><a href="WE_13.html#cb1353-2" tabindex="-1"></a>  reclass <span class="ot">&lt;-</span> <span class="fu">matrix</span>(m, <span class="at">ncol=</span><span class="dv">3</span>, <span class="at">byrow=</span><span class="cn">TRUE</span>)</span>
<span id="cb1353-3"><a href="WE_13.html#cb1353-3" tabindex="-1"></a>  </span>
<span id="cb1353-4"><a href="WE_13.html#cb1353-4" tabindex="-1"></a>wetlnd <span class="ot">&lt;-</span> <span class="fu">classify</span>(xvars[[<span class="st">&quot;nlcd&quot;</span>]], reclass)</span>
<span id="cb1353-5"><a href="WE_13.html#cb1353-5" tabindex="-1"></a>  <span class="fu">names</span>(wetlnd) <span class="ot">&lt;-</span> <span class="st">&quot;wetlnd&quot;</span></span>
<span id="cb1353-6"><a href="WE_13.html#cb1353-6" tabindex="-1"></a>    xvars <span class="ot">&lt;-</span> <span class="fu">c</span>(xvars, wetlnd)</span></code></pre></div>
</div>
<div id="c.-calculate-the-proportion-of-the-landscape-around-sites" class="section level4 unnumbered hasAnchor">
<h4>c. Calculate the proportion of the landscape around sites<a href="WE_13.html#c.-calculate-the-proportion-of-the-landscape-around-sites" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>Assign proportion of landcover that is wetland to sites as <code>pwetland</code>. You could create a binary raster for any cover type of interest and calculate this parameter.</p>
<p><strong>Question</strong>:</p>
<ul>
<li>You want to know if areas of dense wetlands produce more frogs. What buffer distance will you use?</li>
</ul>
<p>Create function to extract the proportion of wetland.</p>
<div class="sourceCode" id="cb1354"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1354-1"><a href="WE_13.html#cb1354-1" tabindex="-1"></a><span class="do">## method 1 (can result in Inf if all zero)</span></span>
<span id="cb1354-2"><a href="WE_13.html#cb1354-2" tabindex="-1"></a><span class="co">#  prop.land &lt;- function(x) {</span></span>
<span id="cb1354-3"><a href="WE_13.html#cb1354-3" tabindex="-1"></a><span class="co">#   length(x[x==1]) / length(x)  </span></span>
<span id="cb1354-4"><a href="WE_13.html#cb1354-4" tabindex="-1"></a><span class="co">#  }</span></span>
<span id="cb1354-5"><a href="WE_13.html#cb1354-5" tabindex="-1"></a></span>
<span id="cb1354-6"><a href="WE_13.html#cb1354-6" tabindex="-1"></a><span class="do">## method 2 (no divide by zero error)</span></span>
<span id="cb1354-7"><a href="WE_13.html#cb1354-7" tabindex="-1"></a>prop.land <span class="ot">&lt;-</span> <span class="cf">function</span>(x) {</span>
<span id="cb1354-8"><a href="WE_13.html#cb1354-8" tabindex="-1"></a><span class="fu">prop.table</span>(<span class="fu">table</span>(<span class="fu">factor</span>(x, <span class="at">levels=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">1</span>))))[<span class="dv">2</span>]</span>
<span id="cb1354-9"><a href="WE_13.html#cb1354-9" tabindex="-1"></a>}</span></code></pre></div>
<p>Apply the function to extract the proportion of wetland within a buffer (here: 300 m).</p>
<div class="sourceCode" id="cb1355"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1355-1"><a href="WE_13.html#cb1355-1" tabindex="-1"></a>b <span class="ot">&lt;-</span> <span class="fu">st_buffer</span>(sites, <span class="dv">300</span>)</span>
<span id="cb1355-2"><a href="WE_13.html#cb1355-2" tabindex="-1"></a>pwetland <span class="ot">&lt;-</span> <span class="fu">extract</span>(wetlnd, <span class="fu">vect</span>(b))</span></code></pre></div>
<pre><code>## Warning: [extract] transforming vector data to the CRS of the raster</code></pre>
<div class="sourceCode" id="cb1357"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1357-1"><a href="WE_13.html#cb1357-1" tabindex="-1"></a>pwetland <span class="ot">&lt;-</span> <span class="fu">tapply</span>(pwetland[,<span class="dv">2</span>], pwetland[,<span class="dv">1</span>], prop.land)</span></code></pre></div>
<p>Add the proportion of wetland back to the dataframe.</p>
<div class="sourceCode" id="cb1358"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1358-1"><a href="WE_13.html#cb1358-1" tabindex="-1"></a>sites<span class="sc">$</span>pwetland <span class="ot">&lt;-</span> <span class="fu">as.numeric</span>(pwetland)</span>
<span id="cb1358-2"><a href="WE_13.html#cb1358-2" tabindex="-1"></a><span class="fu">head</span>(sites<span class="sc">$</span>pwetland)</span></code></pre></div>
<pre><code>## [1] 0.205047319 0.018987342 0.047923323 0.000000000 0.003174603 0.115755627</code></pre>
<p><strong>Challenge</strong>:</p>
<ul>
<li>What happens if you change this radius?</li>
<li>What radius do you think makes the most sense ecologically?</li>
</ul>
<p>Alternatively, you can use the <code>landscapemetrics</code> package (see Week 2) to calculate a broader set of landscape metrics. Here is an example for calculating <code>pland</code>.</p>
<p>Note: as per help file, the argument <code>y</code> of the function <code>sample_lsm</code> can accept an <code>sf</code> points object (e.g., <code>sites</code>). This returned an error though, hence we first convert <code>sites</code> to an <code>sp</code> object, using the function <code>as_Spatial</code>.</p>
<div class="sourceCode" id="cb1360"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1360-1"><a href="WE_13.html#cb1360-1" tabindex="-1"></a>nlcd_sampled <span class="ot">&lt;-</span> landscapemetrics<span class="sc">::</span><span class="fu">sample_lsm</span>(<span class="at">landscape =</span> xvars[[<span class="st">&quot;wetlnd&quot;</span>]], </span>
<span id="cb1360-2"><a href="WE_13.html#cb1360-2" tabindex="-1"></a>                                             <span class="at">what =</span> <span class="st">&quot;lsm_c_pland&quot;</span>,</span>
<span id="cb1360-3"><a href="WE_13.html#cb1360-3" tabindex="-1"></a>                                             <span class="at">shape =</span> <span class="st">&quot;circle&quot;</span>,</span>
<span id="cb1360-4"><a href="WE_13.html#cb1360-4" tabindex="-1"></a>                                             <span class="at">y =</span> sf<span class="sc">::</span><span class="fu">st_coordinates</span>(sites), </span>
<span id="cb1360-5"><a href="WE_13.html#cb1360-5" tabindex="-1"></a>                                             <span class="at">size =</span> <span class="dv">300</span>, </span>
<span id="cb1360-6"><a href="WE_13.html#cb1360-6" tabindex="-1"></a>                                             <span class="at">return_raster =</span> <span class="cn">FALSE</span>,</span>
<span id="cb1360-7"><a href="WE_13.html#cb1360-7" tabindex="-1"></a>                                             <span class="at">plot_id=</span>sites<span class="sc">$</span>SiteID)</span>
<span id="cb1360-8"><a href="WE_13.html#cb1360-8" tabindex="-1"></a>pwetland <span class="ot">&lt;-</span> dplyr<span class="sc">::</span><span class="fu">select</span>(dplyr<span class="sc">::</span><span class="fu">filter</span>(nlcd_sampled, class <span class="sc">==</span> <span class="dv">1</span>, </span>
<span id="cb1360-9"><a href="WE_13.html#cb1360-9" tabindex="-1"></a>                                        metric <span class="sc">==</span> <span class="st">&quot;pland&quot;</span>), plot_id, value)  </span>
<span id="cb1360-10"><a href="WE_13.html#cb1360-10" tabindex="-1"></a><span class="fu">names</span>(pwetland) <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;SiteID&quot;</span>, <span class="st">&quot;pwetland&quot;</span>)</span>
<span id="cb1360-11"><a href="WE_13.html#cb1360-11" tabindex="-1"></a>pwetland<span class="sc">$</span>pwetland <span class="ot">&lt;-</span> pwetland<span class="sc">$</span>pwetland<span class="sc">/</span><span class="dv">100</span></span>
<span id="cb1360-12"><a href="WE_13.html#cb1360-12" tabindex="-1"></a></span>
<span id="cb1360-13"><a href="WE_13.html#cb1360-13" tabindex="-1"></a><span class="fu">head</span>(pwetland)</span></code></pre></div>
<p>Note: these values are sorted by <code>SiteID</code> and are not in the same order as the <code>sf</code> object <code>sites</code>, that’s why the first six values may not be the same.</p>
</div>
<div id="d.-add-values-of-rasters-to-sample-sites" class="section level4 unnumbered hasAnchor">
<h4>d. Add values of rasters to sample sites<a href="WE_13.html#d.-add-values-of-rasters-to-sample-sites" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>This adds potential “at site” variables, keep as sf POINT class object. We are removing raster 6 as knowing the cover class from NLCD that intersects the same points is not very useful information.</p>
<div class="sourceCode" id="cb1361"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1361-1"><a href="WE_13.html#cb1361-1" tabindex="-1"></a>stats <span class="ot">&lt;-</span> <span class="fu">extract</span>(xvars[[<span class="sc">-</span><span class="dv">6</span>]], <span class="fu">vect</span>(sites))</span></code></pre></div>
<pre><code>## Warning: [extract] transforming vector data to the CRS of the raster</code></pre>
<div class="sourceCode" id="cb1363"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1363-1"><a href="WE_13.html#cb1363-1" tabindex="-1"></a>sites <span class="ot">&lt;-</span> <span class="fu">st_sf</span>(<span class="fu">data.frame</span>(<span class="fu">as.data.frame</span>(sites), stats), </span>
<span id="cb1363-2"><a href="WE_13.html#cb1363-2" tabindex="-1"></a>                 <span class="at">geometry=</span>sites<span class="sc">$</span>geometry)</span></code></pre></div>
</div>
<div id="e.-add-raster-covariates-to-graph-edges-lines" class="section level4 unnumbered hasAnchor">
<h4>e. Add raster covariates to graph edges (lines)<a href="WE_13.html#e.-add-raster-covariates-to-graph-edges-lines" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>Remove <code>nlcd</code> and <code>wetlnd</code> rasters before calculating statistics, as calculating statistics on categorical data is nonsensical.</p>
<p>Calculating stats - this will take a while!</p>
<div class="sourceCode" id="cb1364"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1364-1"><a href="WE_13.html#cb1364-1" tabindex="-1"></a>idx <span class="ot">&lt;-</span> <span class="fu">which</span>(<span class="fu">names</span>(xvars) <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">&quot;nlcd&quot;</span>,<span class="st">&quot;wetlnd&quot;</span>))</span>
<span id="cb1364-2"><a href="WE_13.html#cb1364-2" tabindex="-1"></a><span class="fu">suppressWarnings</span>(</span>
<span id="cb1364-3"><a href="WE_13.html#cb1364-3" tabindex="-1"></a>  stats <span class="ot">&lt;-</span> <span class="fu">graph.statistics</span>(dist.graph, <span class="at">r =</span> xvars[[<span class="sc">-</span>idx]], </span>
<span id="cb1364-4"><a href="WE_13.html#cb1364-4" tabindex="-1"></a>                            <span class="at">buffer=</span> <span class="cn">NULL</span>, <span class="at">stats =</span> <span class="fu">c</span>(<span class="st">&quot;min&quot;</span>,         </span>
<span id="cb1364-5"><a href="WE_13.html#cb1364-5" tabindex="-1"></a>                            <span class="st">&quot;mean&quot;</span>,<span class="st">&quot;max&quot;</span>, <span class="st">&quot;var&quot;</span>, <span class="st">&quot;median&quot;</span>)))</span></code></pre></div>
<p>Add statistics to graph.</p>
<div class="sourceCode" id="cb1365"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1365-1"><a href="WE_13.html#cb1365-1" tabindex="-1"></a>dist.graph <span class="ot">&lt;-</span> <span class="fu">st_sf</span>(<span class="fu">data.frame</span>(<span class="fu">as.data.frame</span>(dist.graph), stats), </span>
<span id="cb1365-2"><a href="WE_13.html#cb1365-2" tabindex="-1"></a>                    <span class="at">geometry=</span>dist.graph<span class="sc">$</span>geometry)</span></code></pre></div>
</div>
<div id="f.-what-about-categorical-variables" class="section level4 unnumbered hasAnchor">
<h4>f. What about categorical variables?<a href="WE_13.html#f.-what-about-categorical-variables" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>Statistical moments (mean, variance, etc.) are nonsensical for categorical variables. Here we create a function for returning the percent wetland between sites. Then we use it to calculate an additional statistic and add the result to the graph.</p>
<p><strong>Question</strong>:</p>
<ul>
<li>Are there other categorical variables that you think may be ecologically important?</li>
</ul>
<p>Function to calculate percent wetland between pairs of sites:</p>
<div class="sourceCode" id="cb1366"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1366-1"><a href="WE_13.html#cb1366-1" tabindex="-1"></a>wet.pct <span class="ot">&lt;-</span> <span class="cf">function</span>(x) {</span>
<span id="cb1366-2"><a href="WE_13.html#cb1366-2" tabindex="-1"></a>  x <span class="ot">&lt;-</span> <span class="fu">ifelse</span>(x <span class="sc">==</span> <span class="dv">11</span> <span class="sc">|</span> x <span class="sc">==</span> <span class="dv">12</span> <span class="sc">|</span> x <span class="sc">==</span> <span class="dv">90</span> <span class="sc">|</span> x <span class="sc">==</span> <span class="dv">95</span>, <span class="dv">1</span>, <span class="dv">0</span>)</span>
<span id="cb1366-3"><a href="WE_13.html#cb1366-3" tabindex="-1"></a>    <span class="fu">prop.table</span>(<span class="fu">table</span>(<span class="fu">factor</span>(x, <span class="at">levels=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">1</span>))))[<span class="dv">2</span>] </span>
<span id="cb1366-4"><a href="WE_13.html#cb1366-4" tabindex="-1"></a>} </span></code></pre></div>
<p>Calculate statistic and add it to graph.</p>
<div class="sourceCode" id="cb1367"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1367-1"><a href="WE_13.html#cb1367-1" tabindex="-1"></a><span class="fu">suppressWarnings</span>(</span>
<span id="cb1367-2"><a href="WE_13.html#cb1367-2" tabindex="-1"></a>  wetstats <span class="ot">&lt;-</span> <span class="fu">graph.statistics</span>(dist.graph, <span class="at">r=</span>xvars[[<span class="st">&quot;nlcd&quot;</span>]], </span>
<span id="cb1367-3"><a href="WE_13.html#cb1367-3" tabindex="-1"></a>                               <span class="at">buffer=</span> <span class="cn">NULL</span>, <span class="at">stats =</span> <span class="fu">c</span>(<span class="st">&quot;wet.pct&quot;</span>)) )</span>
<span id="cb1367-4"><a href="WE_13.html#cb1367-4" tabindex="-1"></a>dist.graph<span class="sc">$</span>wet.pct.nlcd <span class="ot">&lt;-</span> <span class="fu">as.numeric</span>(wetstats[,<span class="dv">1</span>]) </span></code></pre></div>
</div>
<div id="g.-evaluate-node-and-edge-correlations" class="section level4 unnumbered hasAnchor">
<h4>g. Evaluate node and edge correlations<a href="WE_13.html#g.-evaluate-node-and-edge-correlations" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>We need to evaluate correlations in the data to avoid overdispersion in our models (remember lab from Week 12, where we calculated the variance inflation factor, VIF).</p>
<p>Note, we are not going to actually remove the correlated variables but, just go through a few methods of evaluating them. The code to remove colinear variables is commented out for reference. We do have to log transform the data as to evaluate the actual model structure.</p>
<p>Create data frame with the node variables to be evaluated.</p>
<div class="sourceCode" id="cb1368"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1368-1"><a href="WE_13.html#cb1368-1" tabindex="-1"></a>node.var <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;degree&quot;</span>, <span class="st">&quot;betweenness&quot;</span>, <span class="st">&quot;Elev&quot;</span>, <span class="st">&quot;Length&quot;</span>, <span class="st">&quot;Area&quot;</span>, <span class="st">&quot;Perim&quot;</span>, </span>
<span id="cb1368-2"><a href="WE_13.html#cb1368-2" tabindex="-1"></a>              <span class="st">&quot;Depth&quot;</span>, <span class="st">&quot;pH&quot;</span>,<span class="st">&quot;Dforest&quot;</span>,<span class="st">&quot;Drock&quot;</span>, <span class="st">&quot;Dshrub&quot;</span>, <span class="st">&quot;pwetland&quot;</span>, <span class="st">&quot;cti&quot;</span>,</span>
<span id="cb1368-3"><a href="WE_13.html#cb1368-3" tabindex="-1"></a>              <span class="st">&quot;dd5&quot;</span>, <span class="st">&quot;ffp&quot;</span>,<span class="st">&quot;gsp&quot;</span>,<span class="st">&quot;pratio&quot;</span>,<span class="st">&quot;hli&quot;</span>,<span class="st">&quot;rough27&quot;</span>,<span class="st">&quot;srr&quot;</span>)</span>
<span id="cb1368-4"><a href="WE_13.html#cb1368-4" tabindex="-1"></a>s <span class="ot">&lt;-</span> <span class="fu">st_drop_geometry</span>(sites) <span class="sc">%&gt;%</span> <span class="fu">select</span>(<span class="st">&quot;degree&quot;</span>, <span class="st">&quot;betweenness&quot;</span>, <span class="st">&quot;Elev&quot;</span>, <span class="st">&quot;Length&quot;</span>, <span class="st">&quot;Area&quot;</span>, <span class="st">&quot;Perim&quot;</span>, </span>
<span id="cb1368-5"><a href="WE_13.html#cb1368-5" tabindex="-1"></a>              <span class="st">&quot;Depth&quot;</span>, <span class="st">&quot;pH&quot;</span>,<span class="st">&quot;Dforest&quot;</span>,<span class="st">&quot;Drock&quot;</span>, <span class="st">&quot;Dshrub&quot;</span>, <span class="st">&quot;pwetland&quot;</span>, <span class="st">&quot;cti&quot;</span>, <span class="st">&quot;ffp&quot;</span>,<span class="st">&quot;gsp&quot;</span>)</span></code></pre></div>
<p>Log-transform values, set log of negative values or zero (where log is not defined) to 0.00001.</p>
<div class="sourceCode" id="cb1369"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1369-1"><a href="WE_13.html#cb1369-1" tabindex="-1"></a><span class="cf">for</span>(i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">ncol</span>(s)) </span>
<span id="cb1369-2"><a href="WE_13.html#cb1369-2" tabindex="-1"></a>{</span>
<span id="cb1369-3"><a href="WE_13.html#cb1369-3" tabindex="-1"></a>  s[,i] <span class="ot">&lt;-</span> <span class="fu">ifelse</span>(s[,i] <span class="sc">&lt;=</span> <span class="dv">0</span>, <span class="fl">0.00001</span>, <span class="fu">log</span>(s[,i]))</span>
<span id="cb1369-4"><a href="WE_13.html#cb1369-4" tabindex="-1"></a>}</span></code></pre></div>
<p>Site correlations:</p>
<div class="sourceCode" id="cb1370"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1370-1"><a href="WE_13.html#cb1370-1" tabindex="-1"></a>p <span class="ot">=</span> <span class="fl">0.8</span> <span class="co"># Set upper limit of for collinearity</span></span>
<span id="cb1370-2"><a href="WE_13.html#cb1370-2" tabindex="-1"></a>site.cor <span class="ot">&lt;-</span> <span class="fu">cor</span>(s, <span class="at">y =</span> <span class="cn">NULL</span>, </span>
<span id="cb1370-3"><a href="WE_13.html#cb1370-3" tabindex="-1"></a>                <span class="at">use =</span> <span class="st">&quot;complete.obs&quot;</span>, </span>
<span id="cb1370-4"><a href="WE_13.html#cb1370-4" tabindex="-1"></a>                <span class="at">method =</span> <span class="st">&quot;pearson&quot;</span>)</span>
<span id="cb1370-5"><a href="WE_13.html#cb1370-5" tabindex="-1"></a><span class="fu">diag</span>(site.cor) <span class="ot">&lt;-</span> <span class="dv">0</span>                 </span>
<span id="cb1370-6"><a href="WE_13.html#cb1370-6" tabindex="-1"></a></span>
<span id="cb1370-7"><a href="WE_13.html#cb1370-7" tabindex="-1"></a>cor.idx <span class="ot">&lt;-</span> <span class="fu">which</span>(site.cor <span class="sc">&gt;</span> p <span class="sc">|</span> site.cor <span class="sc">&lt;</span> <span class="sc">-</span>p, <span class="at">arr.ind =</span> <span class="cn">TRUE</span>)</span>
<span id="cb1370-8"><a href="WE_13.html#cb1370-8" tabindex="-1"></a>cor.names <span class="ot">&lt;-</span> <span class="fu">vector</span>()</span>
<span id="cb1370-9"><a href="WE_13.html#cb1370-9" tabindex="-1"></a>cor.p <span class="ot">&lt;-</span> <span class="fu">vector</span>()</span>
<span id="cb1370-10"><a href="WE_13.html#cb1370-10" tabindex="-1"></a></span>
<span id="cb1370-11"><a href="WE_13.html#cb1370-11" tabindex="-1"></a><span class="cf">for</span>(i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">nrow</span>(cor.idx)) </span>
<span id="cb1370-12"><a href="WE_13.html#cb1370-12" tabindex="-1"></a>{</span>
<span id="cb1370-13"><a href="WE_13.html#cb1370-13" tabindex="-1"></a>     cor.p[i] <span class="ot">&lt;-</span> site.cor[cor.idx[i,][<span class="dv">1</span>], cor.idx[i,][<span class="dv">2</span>]]</span>
<span id="cb1370-14"><a href="WE_13.html#cb1370-14" tabindex="-1"></a>  cor.names [i] <span class="ot">&lt;-</span> <span class="fu">paste</span>(<span class="fu">rownames</span>(site.cor)[cor.idx[i,][<span class="dv">1</span>]],</span>
<span id="cb1370-15"><a href="WE_13.html#cb1370-15" tabindex="-1"></a>                         <span class="fu">colnames</span>(site.cor)[cor.idx[i,][<span class="dv">2</span>]], <span class="at">sep=</span><span class="st">&quot;_&quot;</span>)</span>
<span id="cb1370-16"><a href="WE_13.html#cb1370-16" tabindex="-1"></a>}   </span>
<span id="cb1370-17"><a href="WE_13.html#cb1370-17" tabindex="-1"></a><span class="fu">data.frame</span>(<span class="at">parm=</span>cor.names, <span class="at">p=</span>cor.p)</span></code></pre></div>
<pre><code>##            parm          p
## 1      ffp_Elev -0.9016287
## 2      gsp_Elev  0.8934291
## 3   Area_Length  0.8715535
## 4  Perim_Length  0.8392532
## 5   Length_Area  0.8715535
## 6    Perim_Area  0.9912114
## 7  Length_Perim  0.8392532
## 8    Area_Perim  0.9912114
## 9      Elev_ffp -0.9016287
## 10      gsp_ffp -0.9656032
## 11     Elev_gsp  0.8934291
## 12      ffp_gsp -0.9656032</code></pre>
<p>This returns a list of pairwise correlations that are than the threshold p. Instead of doing this manually, we can use the function collinear to check this. It will make a suggestions which variable to drop.</p>
<div class="sourceCode" id="cb1372"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1372-1"><a href="WE_13.html#cb1372-1" tabindex="-1"></a>node.cor <span class="ot">&lt;-</span> spatialEco<span class="sc">::</span><span class="fu">collinear</span>(s, <span class="at">p=</span>p)</span></code></pre></div>
<pre><code>## Collinearity between Area and Perim correlation = 0.9912</code></pre>
<pre><code>##   Correlation means: 0.446 vs 0.276</code></pre>
<pre><code>##    recommend dropping Area</code></pre>
<pre><code>## Collinearity between Area and Length correlation = 0.8716</code></pre>
<pre><code>##   Correlation means: 0.446 vs 0.278</code></pre>
<pre><code>##    recommend dropping Area</code></pre>
<pre><code>## Collinearity between Perim and Length correlation = 0.8393</code></pre>
<pre><code>##   Correlation means: 0.441 vs 0.278</code></pre>
<pre><code>##    recommend dropping Perim</code></pre>
<pre><code>## Collinearity between ffp and Elev correlation = 0.9016</code></pre>
<pre><code>##   Correlation means: 0.341 vs 0.283</code></pre>
<pre><code>##    recommend dropping ffp</code></pre>
<pre><code>## Collinearity between ffp and gsp correlation = 0.9656</code></pre>
<pre><code>##   Correlation means: 0.341 vs 0.284</code></pre>
<pre><code>##    recommend dropping ffp</code></pre>
<pre><code>## Collinearity between Elev and gsp correlation = 0.8934</code></pre>
<pre><code>##   Correlation means: 0.336 vs 0.284</code></pre>
<pre><code>##    recommend dropping Elev</code></pre>
<p>It also returns a vector of correlated variables:</p>
<div class="sourceCode" id="cb1391"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1391-1"><a href="WE_13.html#cb1391-1" tabindex="-1"></a>node.cor</span></code></pre></div>
<pre><code>## [1] &quot;Elev&quot;  &quot;Area&quot;  &quot;Perim&quot; &quot;ffp&quot;</code></pre>
</div>
<div id="h.-add-node-data" class="section level4 unnumbered hasAnchor">
<h4>h. Add node data<a href="WE_13.html#h.-add-node-data" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>Build and add node (at-site) level data to graph, then merge edge (distance graph) and edge (between-site) data.</p>
<div class="sourceCode" id="cb1393"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1393-1"><a href="WE_13.html#cb1393-1" tabindex="-1"></a>node <span class="ot">&lt;-</span> <span class="fu">build.node.data</span>(<span class="fu">st_drop_geometry</span>(sites), <span class="at">group.ids =</span> <span class="st">&quot;SiteID&quot;</span>, </span>
<span id="cb1393-2"><a href="WE_13.html#cb1393-2" tabindex="-1"></a>                        <span class="at">from.parms =</span> <span class="fu">names</span>(s))</span></code></pre></div>
<p>Merge node and edges for model data.frame</p>
<div class="sourceCode" id="cb1394"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1394-1"><a href="WE_13.html#cb1394-1" tabindex="-1"></a>gdata <span class="ot">&lt;-</span> <span class="fu">merge</span>(<span class="fu">st_drop_geometry</span>(dist.graph)[<span class="fu">c</span>(<span class="dv">1</span>,<span class="dv">2</span>,<span class="dv">5</span>,<span class="dv">11</span>,<span class="dv">14</span>,<span class="dv">7</span>)], node, </span>
<span id="cb1394-2"><a href="WE_13.html#cb1394-2" tabindex="-1"></a>               <span class="at">by.x=</span><span class="st">&quot;SiteID&quot;</span>, <span class="at">by.y=</span><span class="st">&quot;SiteID&quot;</span>)</span>
<span id="cb1394-3"><a href="WE_13.html#cb1394-3" tabindex="-1"></a>gdata <span class="ot">&lt;-</span> <span class="fu">merge</span>(gdata, <span class="fu">st_drop_geometry</span>(dist.graph)[<span class="fu">c</span>(<span class="dv">11</span>, <span class="dv">8</span><span class="sc">:</span><span class="dv">10</span>, <span class="dv">15</span><span class="sc">:</span><span class="dv">40</span>)], </span>
<span id="cb1394-4"><a href="WE_13.html#cb1394-4" tabindex="-1"></a>                   <span class="at">by.x=</span><span class="st">&quot;SiteID&quot;</span>, <span class="at">by.y=</span><span class="st">&quot;SiteID&quot;</span>) </span>
<span id="cb1394-5"><a href="WE_13.html#cb1394-5" tabindex="-1"></a><span class="co"># log transform matrix</span></span>
<span id="cb1394-6"><a href="WE_13.html#cb1394-6" tabindex="-1"></a><span class="cf">for</span>(i <span class="cf">in</span> <span class="dv">5</span><span class="sc">:</span><span class="fu">ncol</span>(gdata)) </span>
<span id="cb1394-7"><a href="WE_13.html#cb1394-7" tabindex="-1"></a>{</span>
<span id="cb1394-8"><a href="WE_13.html#cb1394-8" tabindex="-1"></a>  gdata[,i] <span class="ot">&lt;-</span> <span class="fu">ifelse</span>(gdata[,i] <span class="sc">&lt;=</span> <span class="dv">0</span>, <span class="fl">0.00001</span>, <span class="fu">log</span>(gdata[,i]))</span>
<span id="cb1394-9"><a href="WE_13.html#cb1394-9" tabindex="-1"></a>}</span></code></pre></div>
</div>
</div>
<div id="gravity-model" class="section level3 unnumbered hasAnchor">
<h3>7. Gravity model<a href="WE_13.html#gravity-model" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div id="a.-develop-hypothesis" class="section level4 unnumbered hasAnchor">
<h4>a. Develop hypothesis<a href="WE_13.html#a.-develop-hypothesis" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>What type of gravity model do you wish to run (production or attraction constraint)? Think about what hypotheses that you want to test. Write out model statements that group parameters into hypotheses and run models. Remember to run a NULL that is just distance.</p>
<p><strong>At-site (node) potential parameters.</strong> These are all at-site variables. Remember that we pulled all raster variables. We want to critically think about hypotheses and not use all of these parameters.</p>
<ul>
<li><code>degree</code> - graph degree</li>
<li><code>betweenness</code> - graph betweeness</li>
<li><code>Elev</code> - elevation (see comments below)</li>
<li><code>Length</code> - geographic distance</li>
<li><code>Area</code> - wetland area (field)</li>
<li><code>Perim</code> - wetland perimeter (field)</li>
<li><code>Depth</code> - wetland depth (field)- highly correlated with predatory fish presence/abundance</li>
<li><code>pH</code> - wetland pH (field)</li>
<li><code>Dforest</code> - distance to forest (field)</li>
<li><code>Drock</code> - distance to rock (field)</li>
<li><code>Dshrub</code> - distance to shrub (field)</li>
<li><code>pwetland</code> - proportion of wetland in X buffer (calculated above)</li>
<li><code>cti</code> - compound topographic wetness index - steady-state measure of wetness based on topography (raster data)</li>
<li><code>dd5</code> - degree days &gt;5 C (sum of temp) - (raster data)</li>
<li><code>ffp</code> - frost free period (raster data)</li>
<li><code>gsp</code> - growing season precipitation (raster data)</li>
<li><code>pratio</code> - ratio of growing season precip to annual precip (raster data) - can indicate amount of snow to rain</li>
<li><code>hli</code> - heat load index - topographic measure of exposure, related to productivity (ice-off and primary productivity) in this system (raster data)</li>
<li><code>rough27</code> - unscale topographic variation at a 27 X 27 (cells) window size (raster data)</li>
<li><code>ssr</code> - measure of topographic variation at a 27X27 (cells) windo size - for this system pulling out ridgelines (raster data)</li>
</ul>
<p>NOTE: we are adding elevation here as a covariate. HOWEVER - elevation does not represent ecological processes in and of itself. I strongly encourage using the components (temp, moisture, rainfall, vegetation, accessibility, etc.) directly and not elevation as a surrogate parameter.</p>
<p><strong>Between site (edge) potential parameters include:</strong></p>
<ul>
<li><code>cti</code>, <code>dd5</code>, <code>ffp</code>, <code>gsp</code>, <code>pratio</code>, <code>hli</code>, <code>rough27</code>, <code>ssr</code> (min, mean, max, var, median for each)</li>
<li><code>wet.pct.nlcd</code> - percent of cells that are wetland class</li>
</ul>
<div class="sourceCode" id="cb1395"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1395-1"><a href="WE_13.html#cb1395-1" tabindex="-1"></a><span class="co"># null model (under Maximum Likelihood) </span></span>
<span id="cb1395-2"><a href="WE_13.html#cb1395-2" tabindex="-1"></a>( null <span class="ot">&lt;-</span> <span class="fu">gravity</span>(<span class="at">y =</span> <span class="st">&quot;GDIST&quot;</span>, <span class="at">x =</span> <span class="fu">c</span>(<span class="st">&quot;length&quot;</span>), <span class="at">d =</span> <span class="st">&quot;length&quot;</span>, <span class="at">group =</span> <span class="st">&quot;from_ID&quot;</span>, </span>
<span id="cb1395-3"><a href="WE_13.html#cb1395-3" tabindex="-1"></a>                  <span class="at">data =</span> gdata, <span class="at">fit.method =</span> <span class="st">&quot;ML&quot;</span>) )</span></code></pre></div>
<pre><code>## [1] &quot;Running singly-constrained gravity model&quot;</code></pre>
<pre><code>## Gravity model
## 
## Linear mixed-effects model fit by maximum likelihood
##   Data: gdata 
##        AIC      BIC    logLik
##   20554.37 20585.62 -10273.18
## 
## Random effects:
##  Formula: GDIST ~ 1 | from_ID
##         (Intercept)  Residual
## StdDev:   0.2009152 0.4232432
## 
## Fixed effects:  stats::as.formula(paste(paste(y, &quot;~&quot;, sep = &quot;&quot;), paste(x, collapse = &quot;+&quot;))) 
##                  Value  Std.Error    DF    t-value p-value
## (Intercept) -1.2019984 0.07467247 18224 -16.096941       0
## length       0.1418157 0.03059252 18224   4.635633       0
##  Correlation: 
##        (Intr)
## length -0.854
## 
## Standardized Within-Group Residuals:
##        Min         Q1        Med         Q3        Max 
## -3.2285925 -0.5168847  0.1184083  0.6182159  2.4848654 
## 
## Number of Observations: 18252
## Number of Groups: 27</code></pre>
<div class="sourceCode" id="cb1398"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1398-1"><a href="WE_13.html#cb1398-1" tabindex="-1"></a><span class="co"># Fish hypothesis (under Maximum Likelihood) </span></span>
<span id="cb1398-2"><a href="WE_13.html#cb1398-2" tabindex="-1"></a>( depth <span class="ot">&lt;-</span> <span class="fu">gravity</span>(<span class="at">y =</span> <span class="st">&quot;GDIST&quot;</span>, <span class="at">x =</span> <span class="fu">c</span>(<span class="st">&quot;length&quot;</span>,<span class="st">&quot;from.Depth&quot;</span>), <span class="at">d =</span> <span class="st">&quot;length&quot;</span>, </span>
<span id="cb1398-3"><a href="WE_13.html#cb1398-3" tabindex="-1"></a>                   <span class="at">group =</span> <span class="st">&quot;from_ID&quot;</span>, <span class="at">data =</span> gdata, <span class="at">fit.method =</span> <span class="st">&quot;ML&quot;</span>, <span class="at">ln =</span> <span class="cn">FALSE</span>) )</span></code></pre></div>
<pre><code>## [1] &quot;Running singly-constrained gravity model&quot;</code></pre>
<pre><code>## Gravity model
## 
## Linear mixed-effects model fit by maximum likelihood
##   Data: gdata 
##         AIC       BIC  logLik
##   -13067.22 -13028.16 6538.61
## 
## Random effects:
##  Formula: GDIST ~ 1 | from_ID
##         (Intercept) Residual
## StdDev:  0.08102556 0.168483
## 
## Fixed effects:  stats::as.formula(paste(paste(y, &quot;~&quot;, sep = &quot;&quot;), paste(x, collapse = &quot;+&quot;))) 
##                  Value   Std.Error    DF    t-value p-value
## (Intercept) -0.4101360 0.028357541 18224 -14.463032  0.0000
## length      -0.0002202 0.001624886 18224  -0.135530  0.8922
## from.Depth  -0.0200540 0.014527936    25  -1.380377  0.1797
##  Correlation: 
##            (Intr) length
## length     -0.462       
## from.Depth -0.693 -0.002
## 
## Standardized Within-Group Residuals:
##         Min          Q1         Med          Q3         Max 
## -5.07683320 -0.58959052  0.02162005  0.65913788  3.21875035 
## 
## Number of Observations: 18252
## Number of Groups: 27</code></pre>
<div class="sourceCode" id="cb1401"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1401-1"><a href="WE_13.html#cb1401-1" tabindex="-1"></a><span class="co"># Productivity hypothesis (under Maximum Likelihood) </span></span>
<span id="cb1401-2"><a href="WE_13.html#cb1401-2" tabindex="-1"></a><span class="co">#( production &lt;- gravity(y = &quot;GDIST&quot;, x = c(&quot;length&quot;, &quot;from.ffp&quot;, &quot;from.hli&quot;), </span></span>
<span id="cb1401-3"><a href="WE_13.html#cb1401-3" tabindex="-1"></a><span class="co">#                    d = &quot;length&quot;,  group = &quot;from_ID&quot;, data = gdata, </span></span>
<span id="cb1401-4"><a href="WE_13.html#cb1401-4" tabindex="-1"></a><span class="co">#                    fit.method = &quot;ML&quot;, ln = FALSE) )</span></span>
<span id="cb1401-5"><a href="WE_13.html#cb1401-5" tabindex="-1"></a></span>
<span id="cb1401-6"><a href="WE_13.html#cb1401-6" tabindex="-1"></a><span class="co"># Climate hypothesis (under Maximum Likelihood) </span></span>
<span id="cb1401-7"><a href="WE_13.html#cb1401-7" tabindex="-1"></a><span class="co">#( climate &lt;- gravity(y = &quot;GDIST&quot;, x = c(&quot;length&quot;, &quot;median.ffp&quot;, &quot;median.pratio&quot;), </span></span>
<span id="cb1401-8"><a href="WE_13.html#cb1401-8" tabindex="-1"></a><span class="co">#                     d = &quot;length&quot;, group = &quot;from_ID&quot;, data = gdata, </span></span>
<span id="cb1401-9"><a href="WE_13.html#cb1401-9" tabindex="-1"></a><span class="co">#                    fit.method = &quot;ML&quot;,  ln = FALSE) )</span></span>
<span id="cb1401-10"><a href="WE_13.html#cb1401-10" tabindex="-1"></a></span>
<span id="cb1401-11"><a href="WE_13.html#cb1401-11" tabindex="-1"></a><span class="co"># Wetlands hypothesis (under Maximum Likelihood) </span></span>
<span id="cb1401-12"><a href="WE_13.html#cb1401-12" tabindex="-1"></a>( wetlands <span class="ot">&lt;-</span> <span class="fu">gravity</span>(<span class="at">y =</span> <span class="st">&quot;GDIST&quot;</span>, <span class="at">x =</span> <span class="fu">c</span>(<span class="st">&quot;length&quot;</span>, <span class="st">&quot;from.degree&quot;</span>, <span class="st">&quot;from.betweenness&quot;</span>, <span class="st">&quot;from.pwetland&quot;</span>), </span>
<span id="cb1401-13"><a href="WE_13.html#cb1401-13" tabindex="-1"></a>                      <span class="at">d =</span> <span class="st">&quot;length&quot;</span>, <span class="at">group =</span> <span class="st">&quot;from_ID&quot;</span>, <span class="at">data =</span> gdata, <span class="at">fit.method =</span> <span class="st">&quot;ML&quot;</span>,</span>
<span id="cb1401-14"><a href="WE_13.html#cb1401-14" tabindex="-1"></a>                      <span class="at">ln =</span> <span class="cn">FALSE</span>) )</span></code></pre></div>
<pre><code>## [1] &quot;Running singly-constrained gravity model&quot;</code></pre>
<pre><code>## Gravity model
## 
## Linear mixed-effects model fit by maximum likelihood
##   Data: gdata 
##        AIC       BIC   logLik
##   -13061.9 -13007.22 6537.952
## 
## Random effects:
##  Formula: GDIST ~ 1 | from_ID
##         (Intercept) Residual
## StdDev:  0.08303671 0.168483
## 
## Fixed effects:  stats::as.formula(paste(paste(y, &quot;~&quot;, sep = &quot;&quot;), paste(x, collapse = &quot;+&quot;))) 
##                       Value  Std.Error    DF   t-value p-value
## (Intercept)      -0.3844698 0.10679706 18224 -3.600004  0.0003
## length           -0.0002183 0.00162517 18224 -0.134338  0.8931
## from.degree       0.0035277 0.01270448    23  0.277677  0.7837
## from.betweenness -0.0318164 0.05937320    23 -0.535872  0.5972
## from.pwetland     0.0069227 0.01136835    23  0.608941  0.5485
##  Correlation: 
##                  (Intr) length frm.dg frm.bt
## length           -0.126                     
## from.degree      -0.163  0.014              
## from.betweenness -0.782 -0.007 -0.429       
## from.pwetland     0.218  0.004  0.443 -0.286
## 
## Standardized Within-Group Residuals:
##         Min          Q1         Med          Q3         Max 
## -5.07691616 -0.58884894  0.02146133  0.65939630  3.21854105 
## 
## Number of Observations: 18252
## Number of Groups: 27</code></pre>
<div class="sourceCode" id="cb1404"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1404-1"><a href="WE_13.html#cb1404-1" tabindex="-1"></a><span class="co"># Topography hypothesis (under Maximum Likelihood) </span></span>
<span id="cb1404-2"><a href="WE_13.html#cb1404-2" tabindex="-1"></a><span class="co">#( topo &lt;- gravity(y = &quot;GDIST&quot;, x = c(&quot;length&quot;, &quot;median.srr&quot;, &quot;median.rough27&quot;), d = &quot;length&quot;, </span></span>
<span id="cb1404-3"><a href="WE_13.html#cb1404-3" tabindex="-1"></a><span class="co">#                  group = &quot;from_ID&quot;, data = gdata, fit.method = &quot;ML&quot;,</span></span>
<span id="cb1404-4"><a href="WE_13.html#cb1404-4" tabindex="-1"></a><span class="co">#                 ln = FALSE) )</span></span>
<span id="cb1404-5"><a href="WE_13.html#cb1404-5" tabindex="-1"></a></span>
<span id="cb1404-6"><a href="WE_13.html#cb1404-6" tabindex="-1"></a><span class="co"># Habitat hypothesis (under Maximum Likelihood) </span></span>
<span id="cb1404-7"><a href="WE_13.html#cb1404-7" tabindex="-1"></a>( habitat <span class="ot">&lt;-</span> <span class="fu">gravity</span>(<span class="at">y =</span> <span class="st">&quot;GDIST&quot;</span>, <span class="at">x =</span> <span class="fu">c</span>(<span class="st">&quot;length&quot;</span>, <span class="st">&quot;wet.pct.nlcd&quot;</span>, <span class="st">&quot;median.gsp&quot;</span>), </span>
<span id="cb1404-8"><a href="WE_13.html#cb1404-8" tabindex="-1"></a>                     <span class="at">d =</span> <span class="st">&quot;length&quot;</span>, <span class="at">group =</span> <span class="st">&quot;from_ID&quot;</span>, <span class="at">data =</span> gdata, <span class="at">fit.method =</span> <span class="st">&quot;ML&quot;</span>,</span>
<span id="cb1404-9"><a href="WE_13.html#cb1404-9" tabindex="-1"></a>                     <span class="at">ln =</span> <span class="cn">FALSE</span>, <span class="at">method=</span><span class="st">&quot;ML&quot;</span>) )</span></code></pre></div>
<pre><code>## [1] &quot;Running singly-constrained gravity model&quot;</code></pre>
<pre><code>## Gravity model
## 
## Linear mixed-effects model fit by maximum likelihood
##   Data: gdata 
##         AIC       BIC   logLik
##   -13063.38 -13016.51 6537.689
## 
## Random effects:
##  Formula: GDIST ~ 1 | from_ID
##         (Intercept) Residual
## StdDev:    0.083851 0.168483
## 
## Fixed effects:  stats::as.formula(paste(paste(y, &quot;~&quot;, sep = &quot;&quot;), paste(x, collapse = &quot;+&quot;))) 
##                   Value Std.Error    DF    t-value p-value
## (Intercept)  -0.4356243 0.3357628 18222 -1.2974167  0.1945
## length       -0.0002220 0.0016250 18222 -0.1365924  0.8914
## wet.pct.nlcd  0.0000254 0.0008624 18222  0.0294094  0.9765
## median.gsp   -0.0002840 0.0592239 18222 -0.0047961  0.9962
##  Correlation: 
##              (Intr) length wt.pc.
## length       -0.039              
## wet.pct.nlcd -0.129  0.000       
## median.gsp   -0.998  0.000  0.135
## 
## Standardized Within-Group Residuals:
##         Min          Q1         Med          Q3         Max 
## -5.07738165 -0.58923728  0.02085918  0.65955597  3.21896871 
## 
## Number of Observations: 18252
## Number of Groups: 27</code></pre>
<div class="sourceCode" id="cb1407"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1407-1"><a href="WE_13.html#cb1407-1" tabindex="-1"></a><span class="co"># Global model (under Maximum Likelihood) </span></span>
<span id="cb1407-2"><a href="WE_13.html#cb1407-2" tabindex="-1"></a></span>
<span id="cb1407-3"><a href="WE_13.html#cb1407-3" tabindex="-1"></a><span class="co">#( global &lt;- gravity(y = &quot;GDIST&quot;, x = c(&quot;length&quot;, &quot;wet.pct.nlcd&quot;, &quot;median.gsp&quot;, </span></span>
<span id="cb1407-4"><a href="WE_13.html#cb1407-4" tabindex="-1"></a><span class="co">#                    &quot;from.Depth&quot;, &quot;from.ffp&quot;, &quot;from.hli&quot;, &quot;from.pratio&quot;, &quot;from.degree&quot;, </span></span>
<span id="cb1407-5"><a href="WE_13.html#cb1407-5" tabindex="-1"></a><span class="co">#                   &quot;from.betweenness&quot;, &quot;from.pwetland&quot;, &quot;median.srr&quot;, &quot;median.rough27&quot;), </span></span>
<span id="cb1407-6"><a href="WE_13.html#cb1407-6" tabindex="-1"></a><span class="co">#                   d = &quot;length&quot;, group = &quot;from_ID&quot;, data = gdata, fit.method = &quot;ML&quot;,</span></span>
<span id="cb1407-7"><a href="WE_13.html#cb1407-7" tabindex="-1"></a><span class="co">#                   ln = FALSE) )</span></span>
<span id="cb1407-8"><a href="WE_13.html#cb1407-8" tabindex="-1"></a>( global <span class="ot">&lt;-</span> <span class="fu">gravity</span>(<span class="at">y =</span> <span class="st">&quot;GDIST&quot;</span>, <span class="at">x =</span> <span class="fu">c</span>(<span class="st">&quot;length&quot;</span>, <span class="st">&quot;wet.pct.nlcd&quot;</span>, <span class="st">&quot;median.gsp&quot;</span>, </span>
<span id="cb1407-9"><a href="WE_13.html#cb1407-9" tabindex="-1"></a>                    <span class="st">&quot;from.Depth&quot;</span>, <span class="st">&quot;from.ffp&quot;</span>,  </span>
<span id="cb1407-10"><a href="WE_13.html#cb1407-10" tabindex="-1"></a>                    <span class="st">&quot;from.betweenness&quot;</span>, <span class="st">&quot;from.pwetland&quot;</span>), </span>
<span id="cb1407-11"><a href="WE_13.html#cb1407-11" tabindex="-1"></a>                    <span class="at">d =</span> <span class="st">&quot;length&quot;</span>, <span class="at">group =</span> <span class="st">&quot;from_ID&quot;</span>, <span class="at">data =</span> gdata, <span class="at">fit.method =</span> <span class="st">&quot;ML&quot;</span>,</span>
<span id="cb1407-12"><a href="WE_13.html#cb1407-12" tabindex="-1"></a>                    <span class="at">ln =</span> <span class="cn">FALSE</span>) )</span></code></pre></div>
<pre><code>## [1] &quot;Running singly-constrained gravity model&quot;</code></pre>
<pre><code>## Gravity model
## 
## Linear mixed-effects model fit by maximum likelihood
##   Data: gdata 
##         AIC       BIC   logLik
##   -13058.71 -12980.59 6539.355
## 
## Random effects:
##  Formula: GDIST ~ 1 | from_ID
##         (Intercept) Residual
## StdDev:   0.0788026 0.168483
## 
## Fixed effects:  stats::as.formula(paste(paste(y, &quot;~&quot;, sep = &quot;&quot;), paste(x, collapse = &quot;+&quot;))) 
##                       Value Std.Error    DF    t-value p-value
## (Intercept)      -0.5297034 0.3719807 18222 -1.4240077  0.1545
## length           -0.0002215 0.0016251 18222 -0.1363155  0.8916
## wet.pct.nlcd      0.0000281 0.0008625 18222  0.0325393  0.9740
## median.gsp        0.0013770 0.0592685 18222  0.0232332  0.9815
## from.Depth       -0.0307357 0.0179440    22 -1.7128674  0.1008
## from.ffp          0.0442308 0.0548190    22  0.8068514  0.4284
## from.betweenness  0.0291035 0.0606375    22  0.4799595  0.6360
## from.pwetland     0.0071061 0.0099232    22  0.7161115  0.4815
##  Correlation: 
##                  (Intr) length wt.pc. mdn.gs frm.Dp frm.ff frm.bt
## length           -0.036                                          
## wet.pct.nlcd     -0.117  0.000                                   
## median.gsp       -0.909  0.000  0.135                            
## from.Depth        0.176 -0.002  0.001 -0.004                     
## from.ffp         -0.340  0.004  0.003  0.038 -0.288              
## from.betweenness -0.273  0.000  0.001 -0.012 -0.536  0.068       
## from.pwetland     0.112 -0.004  0.001  0.001 -0.128 -0.137 -0.012
## 
## Standardized Within-Group Residuals:
##         Min          Q1         Med          Q3         Max 
## -5.07768386 -0.58928064  0.02237781  0.65817829  3.21958216 
## 
## Number of Observations: 18252
## Number of Groups: 27</code></pre>
</div>
<div id="b.-compare-competing-models" class="section level4 unnumbered hasAnchor">
<h4>b. Compare competing models<a href="WE_13.html#b.-compare-competing-models" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>Should you use ML or REML? Create diagnostic plots.</p>
<p>Can you directly compare ML and REML? Why not?</p>
<div class="sourceCode" id="cb1410"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1410-1"><a href="WE_13.html#cb1410-1" tabindex="-1"></a><span class="co">#compare.models(null, depth, production, climate, wetlands, </span></span>
<span id="cb1410-2"><a href="WE_13.html#cb1410-2" tabindex="-1"></a><span class="co">#               topo, habitat, global)</span></span>
<span id="cb1410-3"><a href="WE_13.html#cb1410-3" tabindex="-1"></a><span class="fu">compare.models</span>(null, depth, wetlands, </span>
<span id="cb1410-4"><a href="WE_13.html#cb1410-4" tabindex="-1"></a>               habitat, global)</span></code></pre></div>
<pre><code>##      model       AIC       BIC log.likelihood   RMSE nparms fit.method
## 1     null  20554.37  20585.62     -10273.185 0.2894      2         ML
## 2    depth -13067.22 -13028.16       6538.610 0.1213      3         ML
## 3 wetlands -13061.90 -13007.22       6537.952 0.1217      5         ML
## 4  habitat -13063.38 -13016.51       6537.689 0.1220      4         ML
## 5   global -13058.71 -12980.59       6539.355 0.1207      8         ML
##       deltaAIC    deltaBIC
## 1 33621.588595 33613.77657
## 2     0.000000     0.00000
## 3     5.315719    20.93978
## 4     3.840564    11.65259
## 5     8.509063    47.56921</code></pre>
<div class="sourceCode" id="cb1412"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1412-1"><a href="WE_13.html#cb1412-1" tabindex="-1"></a><span class="fu">par</span>(<span class="at">mfrow=</span><span class="fu">c</span>(<span class="dv">2</span>,<span class="dv">3</span>))</span>
<span id="cb1412-2"><a href="WE_13.html#cb1412-2" tabindex="-1"></a>   <span class="cf">for</span> (i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="dv">6</span>) { <span class="fu">plot</span>(null, <span class="at">type=</span>i) } </span></code></pre></div>
<p><img src="13-Week13_files/figure-html/null-1.png" width="672" /></p>
</div>
<div id="c.-fit-final-models-and-calculate-effect-size" class="section level4 unnumbered hasAnchor">
<h4>c. Fit final model(s) and calculate effect size<a href="WE_13.html#c.-fit-final-models-and-calculate-effect-size" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>To calculate effect size, we refit the models with REML.</p>
<div class="sourceCode" id="cb1413"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1413-1"><a href="WE_13.html#cb1413-1" tabindex="-1"></a><span class="co"># productivity fit (under REML)</span></span>
<span id="cb1413-2"><a href="WE_13.html#cb1413-2" tabindex="-1"></a><span class="co">#h &lt;- c(&quot;length&quot;, &quot;from.ffp&quot;, &quot;from.hli&quot;)</span></span>
<span id="cb1413-3"><a href="WE_13.html#cb1413-3" tabindex="-1"></a><span class="co">#production_fit &lt;- gravity(y = &quot;GDIST&quot;, x = h, d = &quot;length&quot;, group = &quot;from_ID&quot;,</span></span>
<span id="cb1413-4"><a href="WE_13.html#cb1413-4" tabindex="-1"></a><span class="co">#                      data = gdata, ln=FALSE)</span></span>
<span id="cb1413-5"><a href="WE_13.html#cb1413-5" tabindex="-1"></a></span>
<span id="cb1413-6"><a href="WE_13.html#cb1413-6" tabindex="-1"></a><span class="do">## g.obal fit (under REML)</span></span>
<span id="cb1413-7"><a href="WE_13.html#cb1413-7" tabindex="-1"></a><span class="co">#g &lt;-  c(&quot;length&quot;, &quot;wet.pct.nlcd&quot;, &quot;median.gsp&quot;, &quot;from.Depth&quot;, &quot;from.ffp&quot;,</span></span>
<span id="cb1413-8"><a href="WE_13.html#cb1413-8" tabindex="-1"></a><span class="co">#        &quot;from.hli&quot;, &quot;from.pratio&quot;,  &quot;from.degree&quot;, &quot;from.betweenness&quot;,  </span></span>
<span id="cb1413-9"><a href="WE_13.html#cb1413-9" tabindex="-1"></a><span class="co">#        &quot;from.pwetland&quot;, &quot;median.srr&quot;,  &quot;median.rough27&quot;)</span></span>
<span id="cb1413-10"><a href="WE_13.html#cb1413-10" tabindex="-1"></a>g <span class="ot">&lt;-</span>  <span class="fu">c</span>(<span class="st">&quot;length&quot;</span>, <span class="st">&quot;wet.pct.nlcd&quot;</span>, <span class="st">&quot;median.gsp&quot;</span>, <span class="st">&quot;from.Depth&quot;</span>, <span class="st">&quot;from.ffp&quot;</span>,</span>
<span id="cb1413-11"><a href="WE_13.html#cb1413-11" tabindex="-1"></a>        <span class="st">&quot;from.betweenness&quot;</span>,  <span class="st">&quot;from.pwetland&quot;</span>)</span>
<span id="cb1413-12"><a href="WE_13.html#cb1413-12" tabindex="-1"></a>global_fit <span class="ot">&lt;-</span> <span class="fu">gravity</span>(<span class="at">y =</span> <span class="st">&quot;GDIST&quot;</span>, <span class="at">x =</span> g, <span class="at">d =</span> <span class="st">&quot;length&quot;</span>, </span>
<span id="cb1413-13"><a href="WE_13.html#cb1413-13" tabindex="-1"></a>                      <span class="at">group =</span> <span class="st">&quot;from_ID&quot;</span>, <span class="at">data =</span> gdata, <span class="at">ln=</span><span class="cn">FALSE</span>)</span></code></pre></div>
<pre><code>## [1] &quot;Running singly-constrained gravity model&quot;</code></pre>
<p>Effect size calculation: global model</p>
<div class="sourceCode" id="cb1415"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1415-1"><a href="WE_13.html#cb1415-1" tabindex="-1"></a><span class="fu">gravity.es</span>(global_fit)</span></code></pre></div>
<pre><code>##                      t.value    df       cohen.d  p.value      low.ci
## length           -0.13233104 18222 -0.0019606212 0.894724 -0.02249547
## wet.pct.nlcd      0.02652092 18222  0.0003929348 0.978842 -0.02014191
## median.gsp        0.01893690 18222  0.0002805698 0.984892 -0.02025427
## from.Depth       -1.54640025    22 -0.6593872813 0.136273 -1.29966538
## from.ffp          0.72841095    22  0.3105954724 0.474045 -0.31611957
## from.betweenness  0.43335537    22  0.1847833506 0.668976 -0.43940100
## from.pwetland     0.64649121    22  0.2756647800 0.524649 -0.35021986
##                        up.ci
## length            0.01857423
## wet.pct.nlcd      0.02092778
## median.gsp        0.02081541
## from.Depth       -0.01910919
## from.ffp          0.93731051
## from.betweenness  0.80896770
## from.pwetland     0.90154942</code></pre>
<div class="sourceCode" id="cb1417"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1417-1"><a href="WE_13.html#cb1417-1" tabindex="-1"></a><span class="fu">par</span>(<span class="at">mfrow=</span><span class="fu">c</span>(<span class="dv">2</span>,<span class="dv">3</span>))</span>
<span id="cb1417-2"><a href="WE_13.html#cb1417-2" tabindex="-1"></a>   <span class="cf">for</span> (i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="dv">6</span>) { <span class="fu">plot</span>(global_fit, <span class="at">type=</span>i) } </span></code></pre></div>
<p><img src="13-Week13_files/figure-html/global_fit-1.png" width="672" /></p>
<p>Production model:</p>
<div class="sourceCode" id="cb1418"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1418-1"><a href="WE_13.html#cb1418-1" tabindex="-1"></a><span class="co">#gravity.es(production_fit)</span></span>
<span id="cb1418-2"><a href="WE_13.html#cb1418-2" tabindex="-1"></a><span class="co">#par(mfrow=c(2,3))</span></span>
<span id="cb1418-3"><a href="WE_13.html#cb1418-3" tabindex="-1"></a><span class="co">#   for (i in 1:6) { plot(production_fit, type=i) } </span></span></code></pre></div>
</div>
<div id="d.-back-predict-global_fit-model" class="section level4 unnumbered hasAnchor">
<h4>d. Back predict global_fit model<a href="WE_13.html#d.-back-predict-global_fit-model" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>Feel free to add other top models.</p>
<div class="sourceCode" id="cb1419"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1419-1"><a href="WE_13.html#cb1419-1" tabindex="-1"></a>gd <span class="ot">&lt;-</span> <span class="fu">back.transform</span>(gdata<span class="sc">$</span>GDIST)</span>
<span id="cb1419-2"><a href="WE_13.html#cb1419-2" tabindex="-1"></a></span>
<span id="cb1419-3"><a href="WE_13.html#cb1419-3" tabindex="-1"></a><span class="co"># Make individual-level (group) predictions (per slope) and show RMSE </span></span>
<span id="cb1419-4"><a href="WE_13.html#cb1419-4" tabindex="-1"></a>global.p <span class="ot">&lt;-</span> <span class="fu">predict</span>(global_fit, <span class="at">y =</span> <span class="st">&quot;GDIST&quot;</span>, <span class="at">x =</span> g,  </span>
<span id="cb1419-5"><a href="WE_13.html#cb1419-5" tabindex="-1"></a>                    <span class="at">newdata=</span>gdata, <span class="at">groups =</span> gdata<span class="sc">$</span>from_ID,</span>
<span id="cb1419-6"><a href="WE_13.html#cb1419-6" tabindex="-1"></a>                    <span class="at">back.transform =</span> <span class="st">&quot;simple&quot;</span>)</span></code></pre></div>
<pre><code>## Making individual-level (per-slope group) 
##        constrained predictions</code></pre>
<pre><code>## Back-transforming exp(y-hat)*0.5*variance(y-hat), 
##        assumes normally distributed errors</code></pre>
<div class="sourceCode" id="cb1422"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1422-1"><a href="WE_13.html#cb1422-1" tabindex="-1"></a><span class="co">#production.p &lt;- predict(production_fit, y = &quot;GDIST&quot;, x = h,  </span></span>
<span id="cb1422-2"><a href="WE_13.html#cb1422-2" tabindex="-1"></a><span class="co">#                     newdata=gdata, groups = gdata$from_ID,</span></span>
<span id="cb1422-3"><a href="WE_13.html#cb1422-3" tabindex="-1"></a><span class="co">#                    back.transform = &quot;simple&quot;)</span></span>
<span id="cb1422-4"><a href="WE_13.html#cb1422-4" tabindex="-1"></a></span>
<span id="cb1422-5"><a href="WE_13.html#cb1422-5" tabindex="-1"></a><span class="fu">cat</span>(<span class="st">&quot;RMSE of global&quot;</span>, <span class="fu">rmse</span>(global.p, gd), <span class="st">&quot;</span><span class="sc">\n</span><span class="st">&quot;</span>)</span></code></pre></div>
<pre><code>## RMSE of global 0.1104107</code></pre>
<div class="sourceCode" id="cb1424"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1424-1"><a href="WE_13.html#cb1424-1" tabindex="-1"></a><span class="co">#cat(&quot;RMSE of production&quot;, rmse(production.p, gd), &quot;\n&quot;)</span></span></code></pre></div>
</div>
<div id="e.-aggregrate-estimates-and-plot" class="section level4 unnumbered hasAnchor">
<h4>e. Aggregrate estimates and plot<a href="WE_13.html#e.-aggregrate-estimates-and-plot" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>We can aggregrate estimates back to the edges and nodes. An interactive map can be created using the <code>tmap</code> package (see Week 2 Bonus vignette).</p>
<p>Aggregate estimates to graph and node.</p>
<div class="sourceCode" id="cb1425"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1425-1"><a href="WE_13.html#cb1425-1" tabindex="-1"></a>global.p <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">EID =</span> gdata<span class="sc">$</span>from.to,</span>
<span id="cb1425-2"><a href="WE_13.html#cb1425-2" tabindex="-1"></a>                       <span class="at">NID =</span> gdata<span class="sc">$</span>from_ID,  </span>
<span id="cb1425-3"><a href="WE_13.html#cb1425-3" tabindex="-1"></a>                       <span class="at">p=</span>global.p)</span>
<span id="cb1425-4"><a href="WE_13.html#cb1425-4" tabindex="-1"></a></span>
<span id="cb1425-5"><a href="WE_13.html#cb1425-5" tabindex="-1"></a>edge.p <span class="ot">&lt;-</span> <span class="fu">tapply</span>(global.p<span class="sc">$</span>p, global.p<span class="sc">$</span>EID, mean)</span>
<span id="cb1425-6"><a href="WE_13.html#cb1425-6" tabindex="-1"></a>  dist.graph<span class="sc">$</span>global.flow <span class="ot">&lt;-</span> edge.p</span>
<span id="cb1425-7"><a href="WE_13.html#cb1425-7" tabindex="-1"></a>  </span>
<span id="cb1425-8"><a href="WE_13.html#cb1425-8" tabindex="-1"></a>node.p <span class="ot">&lt;-</span> <span class="fu">tapply</span>(global.p<span class="sc">$</span>p, global.p<span class="sc">$</span>NID, mean)</span>
<span id="cb1425-9"><a href="WE_13.html#cb1425-9" tabindex="-1"></a>node.var <span class="ot">&lt;-</span> <span class="fu">tapply</span>(global.p<span class="sc">$</span>p, global.p<span class="sc">$</span>NID, var)</span>
<span id="cb1425-10"><a href="WE_13.html#cb1425-10" tabindex="-1"></a></span>
<span id="cb1425-11"><a href="WE_13.html#cb1425-11" tabindex="-1"></a>idx <span class="ot">&lt;-</span> <span class="fu">which</span>(sites<span class="sc">$</span>SiteName <span class="sc">%in%</span> <span class="fu">names</span>(node.p))</span>
<span id="cb1425-12"><a href="WE_13.html#cb1425-12" tabindex="-1"></a>sites<span class="sc">$</span>global.flow[idx] <span class="ot">&lt;-</span> node.p</span>
<span id="cb1425-13"><a href="WE_13.html#cb1425-13" tabindex="-1"></a>sites<span class="sc">$</span>global.var[idx] <span class="ot">&lt;-</span> node.var</span></code></pre></div>
<p>Define the map and store it in object <code>Map</code>.</p>
<div class="sourceCode" id="cb1426"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1426-1"><a href="WE_13.html#cb1426-1" tabindex="-1"></a>pal <span class="ot">&lt;-</span> <span class="fu">colorRampPalette</span>(<span class="fu">rev</span>(<span class="fu">c</span>(<span class="st">&quot;red&quot;</span>,<span class="st">&quot;orange&quot;</span>,<span class="st">&quot;blue&quot;</span>)), <span class="at">bias=</span><span class="fl">0.15</span>)</span>
<span id="cb1426-2"><a href="WE_13.html#cb1426-2" tabindex="-1"></a></span>
<span id="cb1426-3"><a href="WE_13.html#cb1426-3" tabindex="-1"></a>Map <span class="ot">&lt;-</span> </span>
<span id="cb1426-4"><a href="WE_13.html#cb1426-4" tabindex="-1"></a>  <span class="fu">tm_shape</span>(dist.graph) <span class="sc">+</span></span>
<span id="cb1426-5"><a href="WE_13.html#cb1426-5" tabindex="-1"></a>  <span class="fu">tm_lines</span>(<span class="st">&quot;global.flow&quot;</span>, <span class="at">palette=</span><span class="fu">pal</span>(<span class="dv">10</span>), <span class="at">title.col=</span><span class="st">&quot;Edges: global.flow&quot;</span>) <span class="sc">+</span></span>
<span id="cb1426-6"><a href="WE_13.html#cb1426-6" tabindex="-1"></a>  <span class="fu">tm_shape</span>(sites) <span class="sc">+</span></span>
<span id="cb1426-7"><a href="WE_13.html#cb1426-7" tabindex="-1"></a>  <span class="fu">tm_symbols</span>(<span class="at">col =</span> <span class="st">&quot;global.flow&quot;</span>, <span class="at">size =</span> <span class="st">&quot;global.var&quot;</span>, </span>
<span id="cb1426-8"><a href="WE_13.html#cb1426-8" tabindex="-1"></a>               <span class="at">shape =</span> <span class="dv">20</span>, <span class="at">scale =</span> <span class="fl">0.75</span>, <span class="at">palette=</span><span class="fu">pal</span>(<span class="dv">10</span>), </span>
<span id="cb1426-9"><a href="WE_13.html#cb1426-9" tabindex="-1"></a>             <span class="at">title.col=</span><span class="st">&quot;Nodes: global.flow&quot;</span>, <span class="at">title.size=</span><span class="st">&quot;Nodes: global.var&quot;</span>) </span></code></pre></div>
<p>Plot static map: here we tell R to plot the legend outside of the map.</p>
<div class="sourceCode" id="cb1427"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1427-1"><a href="WE_13.html#cb1427-1" tabindex="-1"></a><span class="fu">tmap_mode</span>(<span class="fu">c</span>(<span class="st">&quot;plot&quot;</span>, <span class="st">&quot;view&quot;</span>)[<span class="dv">1</span>])</span></code></pre></div>
<pre><code>## tmap mode set to plotting</code></pre>
<div class="sourceCode" id="cb1429"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1429-1"><a href="WE_13.html#cb1429-1" tabindex="-1"></a>Map <span class="sc">+</span> <span class="fu">tm_layout</span>(<span class="at">legend.outside=</span><span class="cn">TRUE</span>, <span class="at">legend.position =</span> <span class="fu">c</span>(<span class="st">&quot;right&quot;</span>, <span class="st">&quot;top&quot;</span>))</span></code></pre></div>
<pre><code>## Legend labels were too wide. Therefore, legend.text.size has been set to 0.49. Increase legend.width (argument of tm_layout) to make the legend wider and therefore the labels larger.</code></pre>
<p><img src="13-Week13_files/figure-html/static-1.png" width="672" /></p>
<p>Plot interactive map: this may not render e.g. in Bookdown or when you knit your notebook to output formats other than <code>.html</code>.</p>
<div class="sourceCode" id="cb1431"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1431-1"><a href="WE_13.html#cb1431-1" tabindex="-1"></a><span class="co">#tmap_mode(c(&quot;plot&quot;, &quot;view&quot;)[2])</span></span>
<span id="cb1431-2"><a href="WE_13.html#cb1431-2" tabindex="-1"></a><span class="co">#Map </span></span></code></pre></div>

</div>
</div>
</div>
<!-- </div> -->
            </section>

          </div>
        </div>
      </div>
<a href="Week13.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="Week14.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": true,
"facebook": false,
"twitter": false,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": false
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": null,
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "section"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.9/latest.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
