<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>17.1 Worked Example | Landscape Genetic Data Analysis with R</title>
  <meta name="description" content="This is a web-interface to the teaching materials for the lab course ‘Landscape Genetic Data Analysis with R’ associated with the distributed graduate course ‘DGS Landscape Genetics.’ The output format is bookdown::gitbook." />
  <meta name="generator" content="bookdown 0.25 and GitBook 2.6.7" />

  <meta property="og:title" content="17.1 Worked Example | Landscape Genetic Data Analysis with R" />
  <meta property="og:type" content="book" />
  
  <meta property="og:description" content="This is a web-interface to the teaching materials for the lab course ‘Landscape Genetic Data Analysis with R’ associated with the distributed graduate course ‘DGS Landscape Genetics.’ The output format is bookdown::gitbook." />
  <meta name="github-repo" content="hhwagner1/LandGenCourse_book" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="17.1 Worked Example | Landscape Genetic Data Analysis with R" />
  
  <meta name="twitter:description" content="This is a web-interface to the teaching materials for the lab course ‘Landscape Genetic Data Analysis with R’ associated with the distributed graduate course ‘DGS Landscape Genetics.’ The output format is bookdown::gitbook." />
  

<meta name="author" content="Editor: Helene Wagner (University of Toronto)" />


<meta name="date" content="2022-04-03" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="Week14.html"/>

<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>
<script src="libs/kePrint-0.0.1/kePrint.js"></script>
<link href="libs/lightable-0.0.1/lightable.css" rel="stylesheet" />
<script src="libs/htmlwidgets-1.5.4/htmlwidgets.js"></script>
<script src="libs/d3-3.5.6/d3.min.js"></script>
<link href="libs/profvis-0.3.6.9000/profvis.css" rel="stylesheet" />
<script src="libs/profvis-0.3.6.9000/profvis.js"></script>
<script src="libs/profvis-0.3.6.9000/scroll.js"></script>
<link href="libs/highlight-6.2.0/textmate.css" rel="stylesheet" />
<script src="libs/highlight-6.2.0/highlight.js"></script>
<script src="libs/profvis-binding-0.3.7/profvis.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>


<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">Landscape Genetic Data Analysis with R</a></li>

<li class="divider"></li>
<li class="part"><span><b>Getting Started<span></span></b></span></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Introduction<span></span></a>
<ul>
<li class="chapter" data-level="1.1" data-path="how-to-use-this-book.html"><a href="how-to-use-this-book.html"><i class="fa fa-check"></i><b>1.1</b> How to use this Book<span></span></a>
<ul>
<li><a href="how-to-use-this-book.html#book-structure">1. Book Structure<span></span></a></li>
<li><a href="how-to-use-this-book.html#find-what-is-relevant-for-you">2. Find what is relevant for you<span></span></a></li>
<li><a href="how-to-use-this-book.html#course-r-package-landgencourse">3. Course R package ‘LandGenCourse’<span></span></a></li>
</ul></li>
<li class="chapter" data-level="1.2" data-path="list-of-packages.html"><a href="list-of-packages.html"><i class="fa fa-check"></i><b>1.2</b> List of R Packages by Vignette<span></span></a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="review-of-r-skills.html"><a href="review-of-r-skills.html"><i class="fa fa-check"></i><b>2</b> Review of R Skills<span></span></a>
<ul>
<li class="chapter" data-level="2.1" data-path="basic-r.html"><a href="basic-r.html"><i class="fa fa-check"></i><b>2.1</b> Basic R Programming<span></span></a>
<ul>
<li><a href="basic-r.html#overview">1. Overview<span></span></a></li>
<li><a href="basic-r.html#data-types">2. Data Types<span></span></a></li>
<li><a href="basic-r.html#data-containers">3. Data Containers<span></span></a></li>
<li><a href="basic-r.html#programming">4. Programming<span></span></a></li>
</ul></li>
<li class="chapter" data-level="2.2" data-path="r-graphics.html"><a href="r-graphics.html"><i class="fa fa-check"></i><b>2.2</b> R Graphics<span></span></a>
<ul>
<li><a href="r-graphics.html#overview-1">1. Overview<span></span></a></li>
<li><a href="r-graphics.html#univariate-plots">2. Univariate plots<span></span></a></li>
<li><a href="r-graphics.html#bivariate-plots">3. Bivariate plots<span></span></a></li>
<li><a href="r-graphics.html#advanced-plotting">4. Advanced plotting<span></span></a></li>
</ul></li>
<li class="chapter" data-level="2.3" data-path="resources.html"><a href="resources.html"><i class="fa fa-check"></i><b>2.3</b> Further R resources<span></span></a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="github-for-group-projects.html"><a href="github-for-group-projects.html"><i class="fa fa-check"></i><b>3</b> GitHub for Group Projects<span></span></a>
<ul>
<li class="chapter" data-level="3.1" data-path="github.html"><a href="github.html"><i class="fa fa-check"></i><b>3.1</b> Why use GitHub?<span></span></a>
<ul>
<li><a href="github.html#goals">1. Goals<span></span></a></li>
<li><a href="github.html#version">2. Watch course video: Version Control 101<span></span></a></li>
<li><a href="github.html#slides">3. Preview slides<span></span></a></li>
<li><a href="github.html#next-steps">4. Next steps<span></span></a></li>
</ul></li>
<li class="chapter" data-level="3.2" data-path="install.html"><a href="install.html"><i class="fa fa-check"></i><b>3.2</b> Installation<span></span></a>
<ul>
<li><a href="install.html#goals-1">1. Goals<span></span></a></li>
<li class="chapter" data-level="3.2.1" data-path="install.html"><a href="install.html#how-to-get-github-and-gitkraken-for-free"><i class="fa fa-check"></i><b>3.2.1</b> 2. How to get GitHub and GitKraken for free<span></span></a></li>
<li><a href="install.html#happygit">3. Happy Git with R<span></span></a></li>
<li><a href="install.html#gitvideos">4. Video tutorials: installation<span></span></a></li>
<li><a href="install.html#join">5. Joining your group’s project on GitHub<span></span></a></li>
<li><a href="install.html#repo">6. Setting up a group repo<span></span></a></li>
<li><a href="install.html#next-steps-1">7. Next steps<span></span></a></li>
</ul></li>
<li class="chapter" data-level="3.3" data-path="workflow.html"><a href="workflow.html"><i class="fa fa-check"></i><b>3.3</b> Workflows<span></span></a>
<ul>
<li><a href="workflow.html#goals-2">1. Goals<span></span></a></li>
<li><a href="workflow.html#step-by-step-tutorials">2. Step-by-step tutorials<span></span></a></li>
<li><a href="workflow.html#further-resources">3. Further resources<span></span></a></li>
</ul></li>
</ul></li>
<li class="part"><span><b>Basic Topics<span></span></b></span></li>
<li class="chapter" data-level="4" data-path="Week1.html"><a href="Week1.html"><i class="fa fa-check"></i><b>4</b> Week 1: Importing Genetic Data<span></span></a>
<ul>
<li class="chapter" data-level="4.1" data-path="video_1.html"><a href="video_1.html"><i class="fa fa-check"></i><b>4.1</b> View Course Video<span></span></a>
<ul>
<li><a href="video_1.html#embedded-video">1. Embedded Video<span></span></a></li>
<li><a href="video_1.html#preview-slides">2. Preview Slides<span></span></a></li>
</ul></li>
<li class="chapter" data-level="4.2" data-path="tutorial_1.html"><a href="tutorial_1.html"><i class="fa fa-check"></i><b>4.2</b> Interactive Tutorial 1<span></span></a>
<ul>
<li><a href="tutorial_1.html#list-of-r-commands-covered-this-week">1. List of R commands covered this week<span></span></a></li>
<li><a href="tutorial_1.html#general-instructions">2. General Instructions<span></span></a></li>
</ul></li>
<li class="chapter" data-level="4.3" data-path="WE_1.html"><a href="WE_1.html"><i class="fa fa-check"></i><b>4.3</b> Worked Example<span></span></a>
<ul>
<li><a href="WE_1.html#overview-of-worked-example">1. Overview of Worked Example<span></span></a></li>
<li><a href="WE_1.html#import-data-from-.csv-file">2. Import data from .csv file<span></span></a></li>
<li><a href="WE_1.html#create-genind-object">3. Create ‘genind’ Object<span></span></a></li>
<li><a href="WE_1.html#view-information-stored-in-genind-object">4. View information stored in ‘genind’ object<span></span></a></li>
<li><a href="WE_1.html#import-data-with-gstudio-package">5. Import data with ‘gstudio’ package<span></span></a></li>
<li><a href="WE_1.html#importing-snp-data">6. Importing SNP data<span></span></a></li>
</ul></li>
<li class="chapter" data-level="4.4" data-path="r-exercise-week-1.html"><a href="r-exercise-week-1.html"><i class="fa fa-check"></i><b>4.4</b> R exercise Week 1<span></span></a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="Week2.html"><a href="Week2.html"><i class="fa fa-check"></i><b>5</b> Week 2: Spatial Data<span></span></a>
<ul>
<li class="chapter" data-level="5.1" data-path="video_2.html"><a href="video_2.html"><i class="fa fa-check"></i><b>5.1</b> View Course Video<span></span></a>
<ul>
<li><a href="video_2.html#embedded-video-1">1. Embedded Video<span></span></a></li>
<li><a href="video_2.html#preview-slides-1">2. Preview Slides<span></span></a></li>
</ul></li>
<li class="chapter" data-level="5.2" data-path="tutorial_2.html"><a href="tutorial_2.html"><i class="fa fa-check"></i><b>5.2</b> Interactive Tutorial 2<span></span></a>
<ul>
<li><a href="tutorial_2.html#list-of-r-commands-covered-this-week-1">1. List of R commands covered this week<span></span></a></li>
<li><a href="tutorial_2.html#general-instructions-1">2. General Instructions<span></span></a></li>
</ul></li>
<li class="chapter" data-level="5.3" data-path="WE_2.html"><a href="WE_2.html"><i class="fa fa-check"></i><b>5.3</b> Worked Example<span></span></a>
<ul>
<li><a href="WE_2.html#overview-of-worked-example-1">1. Overview of Worked Example<span></span></a></li>
<li><a href="WE_2.html#import-site-data-from-.csv-file">2. Import site data from .csv file<span></span></a></li>
<li><a href="WE_2.html#display-raster-data-and-overlay-sampling-locations-extract-data">3. Display raster data and overlay sampling locations, extract data<span></span></a></li>
<li><a href="WE_2.html#calculate-landscape-metrics">4. Calculate landscape metrics<span></span></a></li>
<li><a href="WE_2.html#sample-landscape-metrics-within-buffer-around-sampling-locations">5. Sample landscape metrics within buffer around sampling locations<span></span></a></li>
</ul></li>
<li class="chapter" data-level="5.4" data-path="r-exercise-week-2.html"><a href="r-exercise-week-2.html"><i class="fa fa-check"></i><b>5.4</b> R Exercise Week 2<span></span></a></li>
<li class="chapter" data-level="5.5" data-path="bonus_2a.html"><a href="bonus_2a.html"><i class="fa fa-check"></i><b>5.5</b> Bonus: ‘sf’ and ‘terra’<span></span></a>
<ul>
<li><a href="bonus_2a.html#overview-2">1. Overview<span></span></a></li>
<li><a href="bonus_2a.html#convert-spatialpointsdataframe-to-sf-object">2. Convert ‘SpatialPointsDataFrame’ to ‘sf’ object<span></span></a></li>
<li><a href="bonus_2a.html#plot-variables-first-10-in-data-set">3. Plot variables (first 10 in data set)<span></span></a></li>
<li><a href="bonus_2a.html#export-site-data-as-esri-shapefile">4. Export site data as ESRI shapefile<span></span></a></li>
<li><a href="bonus_2a.html#use-terra-and-tmap-to-display-categorical-map-with-color-scheme">5. Use ‘terra’ and ‘tmap’ to display categorical map with color scheme<span></span></a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="6" data-path="Week3.html"><a href="Week3.html"><i class="fa fa-check"></i><b>6</b> Week 3: Genetic Diversity<span></span></a>
<ul>
<li class="chapter" data-level="6.1" data-path="video_3.html"><a href="video_3.html"><i class="fa fa-check"></i><b>6.1</b> View Course Video<span></span></a>
<ul>
<li><a href="video_3.html#embedded-video-2">1. Embedded Video<span></span></a></li>
<li><a href="video_3.html#video-part-1">Video, Part 1<span></span></a></li>
<li><a href="video_3.html#video-part-2">Video, Part 2<span></span></a></li>
<li><a href="video_3.html#preview-slides-2">Preview Slides<span></span></a></li>
</ul></li>
<li class="chapter" data-level="6.2" data-path="tutorial_3.html"><a href="tutorial_3.html"><i class="fa fa-check"></i><b>6.2</b> Interactive Tutorial 3<span></span></a>
<ul>
<li><a href="tutorial_3.html#list-of-r-commands-covered-this-week-2">1. List of R commands covered this week<span></span></a></li>
<li><a href="tutorial_3.html#general-instructions-2">2. General Instructions<span></span></a></li>
</ul></li>
<li class="chapter" data-level="6.3" data-path="WE_3.html"><a href="WE_3.html"><i class="fa fa-check"></i><b>6.3</b> Worked Example<span></span></a>
<ul>
<li><a href="WE_3.html#overview-of-worked-example-2">1. Overview of Worked Example<span></span></a></li>
<li><a href="WE_3.html#basic-checking-of-markers-and-populations">2. Basic checking of markers and populations<span></span></a></li>
<li><a href="WE_3.html#assess-genetic-diversity">3. Assess genetic diversity<span></span></a></li>
<li><a href="WE_3.html#aggregate-genetic-data-at-population-level-allele-frequencies">4. Aggregate genetic data at population level (allele frequencies)<span></span></a></li>
</ul></li>
<li class="chapter" data-level="6.4" data-path="r-exercise-week-3.html"><a href="r-exercise-week-3.html"><i class="fa fa-check"></i><b>6.4</b> R Exercise Week 3<span></span></a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="Week4.html"><a href="Week4.html"><i class="fa fa-check"></i><b>7</b> Week 4: Metapopulations<span></span></a>
<ul>
<li class="chapter" data-level="7.1" data-path="video_4.html"><a href="video_4.html"><i class="fa fa-check"></i><b>7.1</b> View Course Video<span></span></a>
<ul>
<li><a href="video_4.html#embedded-video-3">1. Embedded Video<span></span></a></li>
<li><a href="video_4.html#video-part-1-1">Video, Part 1<span></span></a></li>
<li><a href="video_4.html#video-part-2-1">Video, Part 2<span></span></a></li>
<li><a href="video_4.html#preview-slides-3">Preview Slides<span></span></a></li>
</ul></li>
<li class="chapter" data-level="7.2" data-path="tutorial_4.html"><a href="tutorial_4.html"><i class="fa fa-check"></i><b>7.2</b> Interactive Tutorial 4<span></span></a>
<ul>
<li><a href="tutorial_4.html#list-of-r-commands-covered-this-week-3">1. List of R commands covered this week<span></span></a></li>
<li><a href="tutorial_4.html#general-instructions-3">2. General Instructions<span></span></a></li>
</ul></li>
<li class="chapter" data-level="7.3" data-path="WE_4.html"><a href="WE_4.html"><i class="fa fa-check"></i><b>7.3</b> Worked Example<span></span></a>
<ul>
<li><a href="WE_4.html#overview-of-worked-example-3">1. Overview of Worked Example<span></span></a></li>
<li><a href="WE_4.html#spatial-distribution-of-genetic-structure">2. Spatial distribution of genetic structure<span></span></a></li>
<li><a href="WE_4.html#what-determines-genetic-differentiation-among-sites">3. What determines genetic differentiation among sites?<span></span></a></li>
<li><a href="WE_4.html#what-determines-genetic-diversity">4. What determines genetic diversity?<span></span></a></li>
<li><a href="WE_4.html#are-genetic-differentiation-and-diversity-related">5. Are genetic differentiation and diversity related?<span></span></a></li>
<li><a href="WE_4.html#effect-of-recent-extinction-events">6. Effect of recent extinction events<span></span></a></li>
</ul></li>
<li class="chapter" data-level="7.4" data-path="r-exercise-week-4.html"><a href="r-exercise-week-4.html"><i class="fa fa-check"></i><b>7.4</b> R Exercise Week 4<span></span></a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="Week5.html"><a href="Week5.html"><i class="fa fa-check"></i><b>8</b> Week 5: Spatial Statistics<span></span></a>
<ul>
<li class="chapter" data-level="8.1" data-path="video_5.html"><a href="video_5.html"><i class="fa fa-check"></i><b>8.1</b> View Course Video<span></span></a>
<ul>
<li><a href="video_5.html#embedded-video-4">1. Embedded Video<span></span></a></li>
<li><a href="video_5.html#preview-slides-4">Preview Slides<span></span></a></li>
</ul></li>
<li class="chapter" data-level="8.2" data-path="tutorial_5.html"><a href="tutorial_5.html"><i class="fa fa-check"></i><b>8.2</b> Interactive Tutorial 5<span></span></a>
<ul>
<li><a href="tutorial_5.html#list-of-r-commands-covered-this-week-4">1. List of R commands covered this week<span></span></a></li>
<li><a href="tutorial_5.html#general-instructions-4">2. General Instructions<span></span></a></li>
</ul></li>
<li class="chapter" data-level="8.3" data-path="WE_5.html"><a href="WE_5.html"><i class="fa fa-check"></i><b>8.3</b> Worked Example<span></span></a>
<ul>
<li><a href="WE_5.html#overview-of-worked-example-4">1. Overview of Worked Example<span></span></a></li>
<li><a href="WE_5.html#data-import-and-manipulation">2. Data import and manipulation<span></span></a></li>
<li><a href="WE_5.html#calculate-genetic-distances">3. Calculate genetic distances<span></span></a></li>
<li><a href="WE_5.html#perform-a-mantel-test-to-test-for-ibd">4. Perform a Mantel test to test for IBD<span></span></a></li>
<li><a href="WE_5.html#create-mantel-correlogram-for-genetic-data">5. Create Mantel correlogram for genetic data<span></span></a></li>
<li><a href="WE_5.html#specify-spatial-weights-and-calculate-morans-i">6. Specify spatial weights and calculate Moran’s I<span></span></a></li>
</ul></li>
<li class="chapter" data-level="8.4" data-path="r-exercise-week-5.html"><a href="r-exercise-week-5.html"><i class="fa fa-check"></i><b>8.4</b> R Exercise Week 5<span></span></a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="Week6.html"><a href="Week6.html"><i class="fa fa-check"></i><b>9</b> Week 6: Quantitative Genetics<span></span></a>
<ul>
<li class="chapter" data-level="9.1" data-path="video_6.html"><a href="video_6.html"><i class="fa fa-check"></i><b>9.1</b> View Course Video<span></span></a>
<ul>
<li><a href="video_6.html#embedded-video-5">1. Embedded Video<span></span></a></li>
<li><a href="video_6.html#video-part-1-2">Video, Part 1<span></span></a></li>
<li><a href="video_6.html#video-part-2-2">Video, Part 2<span></span></a></li>
<li><a href="video_6.html#preview-slides-5">Preview Slides<span></span></a></li>
</ul></li>
<li class="chapter" data-level="9.2" data-path="tutorial_6.html"><a href="tutorial_6.html"><i class="fa fa-check"></i><b>9.2</b> Interactive Tutorial 6<span></span></a>
<ul>
<li><a href="tutorial_6.html#list-of-r-commands-covered-this-week-5">1. List of R commands covered this week<span></span></a></li>
<li><a href="tutorial_6.html#general-instructions-5">2. General Instructions<span></span></a></li>
</ul></li>
<li class="chapter" data-level="9.3" data-path="WE_6.html"><a href="WE_6.html"><i class="fa fa-check"></i><b>9.3</b> Worked Example<span></span></a>
<ul>
<li><a href="WE_6.html#overview-of-worked-example-5">1. Overview of Worked Example<span></span></a></li>
<li><a href="WE_6.html#estimate-genetic-and-non-genetic-variance-components-from-a-common-garden-experiment">2. Estimate genetic and non-genetic variance components from a common garden experiment<span></span></a></li>
<li><a href="WE_6.html#estimate-trait-heritability">3. Estimate trait heritability<span></span></a></li>
<li><a href="WE_6.html#compare-q_st-to-f_st-measured-from-snp-data">4. Compare <span class="math inline">\(Q_{st}\)</span> to <span class="math inline">\(F_{st}\)</span> measured from SNP data<span></span></a></li>
<li><a href="WE_6.html#assess-correlation-between-trait-and-environment">5. Assess correlation between trait and environment<span></span></a></li>
</ul></li>
<li class="chapter" data-level="9.4" data-path="r-exercise-week-6.html"><a href="r-exercise-week-6.html"><i class="fa fa-check"></i><b>9.4</b> R Exercise Week 6<span></span></a></li>
</ul></li>
<li class="chapter" data-level="10" data-path="Week7.html"><a href="Week7.html"><i class="fa fa-check"></i><b>10</b> Week 7: Spatial Linear Models<span></span></a>
<ul>
<li class="chapter" data-level="10.1" data-path="video_7.html"><a href="video_7.html"><i class="fa fa-check"></i><b>10.1</b> View Course Video<span></span></a>
<ul>
<li><a href="video_7.html#embedded-video-6">1. Embedded Video<span></span></a></li>
<li><a href="video_7.html#video-part-1-3">Video, Part 1<span></span></a></li>
<li><a href="video_7.html#video-part-2-3">Video, Part 2<span></span></a></li>
<li><a href="video_7.html#preview-slides-6">Preview Slides<span></span></a></li>
</ul></li>
<li class="chapter" data-level="10.2" data-path="tutorial_7.html"><a href="tutorial_7.html"><i class="fa fa-check"></i><b>10.2</b> Interactive Tutorial 7<span></span></a>
<ul>
<li><a href="tutorial_7.html#list-of-r-commands-covered-this-week-6">1. List of R commands covered this week<span></span></a></li>
<li><a href="tutorial_7.html#general-instructions-6">2. General Instructions<span></span></a></li>
</ul></li>
<li class="chapter" data-level="10.3" data-path="WE_7.html"><a href="WE_7.html"><i class="fa fa-check"></i><b>10.3</b> Worked Example<span></span></a>
<ul>
<li><a href="WE_7.html#overview-of-worked-example-6">1. Overview of Worked Example<span></span></a></li>
<li><a href="WE_7.html#explore-data-set">2. Explore data set<span></span></a></li>
<li><a href="WE_7.html#test-regression-residuals-for-spatial-autocorrelation">3. Test regression residuals for spatial autocorrelation<span></span></a></li>
<li><a href="WE_7.html#fit-models-with-spatially-correlated-error-gls-with-package-nlme">4. Fit models with spatially correlated error (GLS) with package ‘nlme’<span></span></a></li>
<li><a href="WE_7.html#fit-spatial-simultaneous-autoregressive-error-models-sar">5. Fit spatial simultaneous autoregressive error models (SAR)<span></span></a></li>
<li><a href="WE_7.html#spatial-filtering-with-mem-using-package-spmoran">6. Spatial filtering with MEM using package ‘spmoran’<span></span></a></li>
<li><a href="WE_7.html#fit-spatially-varying-coefficients-model-with-package-spmoran">7. Fit spatially varying coefficients model with package ‘spmoran’<span></span></a></li>
<li><a href="WE_7.html#model-with-patchsize-and-ibr">Model with PatchSize and IBR<span></span></a></li>
<li><a href="WE_7.html#model-with-ibr-only">Model with IBR only<span></span></a></li>
<li><a href="WE_7.html#conclusions">8. Conclusions<span></span></a></li>
</ul></li>
<li class="chapter" data-level="10.4" data-path="r-exercise-week-7.html"><a href="r-exercise-week-7.html"><i class="fa fa-check"></i><b>10.4</b> R Exercise Week 7<span></span></a></li>
<li class="chapter" data-level="10.5" data-path="bonus_7a.html"><a href="bonus_7a.html"><i class="fa fa-check"></i><b>10.5</b> Bonus: Si index<span></span></a>
<ul>
<li><a href="bonus_7a.html#overview-of-bonus-materials">1. Overview of Bonus Materials<span></span></a></li>
<li><a href="bonus_7a.html#import-ecological-distance-matrices">2. Import ecological distance matrices<span></span></a></li>
<li><a href="bonus_7a.html#optimize-the-scaling-parameter-alpha">3. Optimize the scaling parameter alpha<span></span></a></li>
<li><a href="bonus_7a.html#calculate-hanskis-index-si-with-source-patch-parameters">4. Calculate Hanski’s index Si with source patch parameters<span></span></a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="11" data-path="Week8.html"><a href="Week8.html"><i class="fa fa-check"></i><b>11</b> Week 8: Simulation Experiments<span></span></a>
<ul>
<li class="chapter" data-level="11.1" data-path="video_8.html"><a href="video_8.html"><i class="fa fa-check"></i><b>11.1</b> View Course Video<span></span></a>
<ul>
<li><a href="video_8.html#embedded-video-7">1. Embedded Video<span></span></a></li>
<li><a href="video_8.html#video-part-1-4">Video, Part 1<span></span></a></li>
<li><a href="video_8.html#video-part-2-4">Video, Part 2<span></span></a></li>
<li><a href="video_8.html#preview-slides-7">Preview Slides<span></span></a></li>
</ul></li>
<li class="chapter" data-level="11.2" data-path="tutorial_8.html"><a href="tutorial_8.html"><i class="fa fa-check"></i><b>11.2</b> Interactive Tutorial 8<span></span></a>
<ul>
<li><a href="tutorial_8.html#list-of-r-commands-covered-this-week-7">1. List of R commands covered this week<span></span></a></li>
<li><a href="tutorial_8.html#general-instructions-7">2. General Instructions<span></span></a></li>
</ul></li>
<li class="chapter" data-level="11.3" data-path="WE_8.html"><a href="WE_8.html"><i class="fa fa-check"></i><b>11.3</b> Worked Example<span></span></a>
<ul>
<li><a href="WE_8.html#overview-of-worked-example-7">1. Overview of Worked Example<span></span></a></li>
<li><a href="WE_8.html#initialize-a-landscape">2. Initialize a landscape<span></span></a></li>
<li><a href="WE_8.html#define-simulation-parameters">3. Define simulation parameters<span></span></a></li>
<li><a href="WE_8.html#run-simulations-and-analyze-results">4. Run simulations and analyze results<span></span></a></li>
<li><a href="WE_8.html#run-simulator-using-a-previously-defined-parameter-set">5. Run simulator using a previously defined parameter set<span></span></a></li>
</ul></li>
<li class="chapter" data-level="11.4" data-path="r-exercise-week-8.html"><a href="r-exercise-week-8.html"><i class="fa fa-check"></i><b>11.4</b> R Exercise Week 8<span></span></a></li>
<li class="chapter" data-level="11.5" data-path="bonus_8a.html"><a href="bonus_8a.html"><i class="fa fa-check"></i><b>11.5</b> Bonus: Efficient R<span></span></a>
<ul>
<li><a href="bonus_8a.html#overview-of-bonus-material">1. Overview of Bonus Material<span></span></a></li>
<li><a href="bonus_8a.html#navigating-the-file-system">2. Navigating the file system<span></span></a></li>
<li><a href="bonus_8a.html#benchmarking-file-import-and-export-options">3. Benchmarking file import and export options<span></span></a></li>
<li><a href="bonus_8a.html#profiling-your-code">4. Profiling your code<span></span></a></li>
<li><a href="bonus_8a.html#creating-and-executing-a-bash-r-script">5. Creating and executing a Bash R script<span></span></a></li>
<li><a href="bonus_8a.html#parallelizing-code">6. Parallelizing code<span></span></a></li>
</ul></li>
</ul></li>
<li class="part"><span><b>Advanced Topics<span></span></b></span></li>
<li class="chapter" data-level="12" data-path="Week9.html"><a href="Week9.html"><i class="fa fa-check"></i><b>12</b> Week 9: Population Structure<span></span></a>
<ul>
<li class="chapter" data-level="12.1" data-path="WE_9.html"><a href="WE_9.html"><i class="fa fa-check"></i><b>12.1</b> Worked Example<span></span></a>
<ul>
<li><a href="WE_9.html#overview-of-worked-example-8">1. Overview of Worked Example<span></span></a></li>
<li><a href="WE_9.html#simulated-data-2-island-model">2. Simulated data: 2-island model<span></span></a></li>
<li><a href="WE_9.html#simulated-data-2-island-model-with-admixture">3. Simulated data: 2-island model with admixture<span></span></a></li>
<li><a href="WE_9.html#empirical-data-threespine-sticklebacks">4. Empirical data: Threespine sticklebacks<span></span></a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="13" data-path="Week10.html"><a href="Week10.html"><i class="fa fa-check"></i><b>13</b> Week 10: Landscape Resistance<span></span></a>
<ul>
<li class="chapter" data-level="13.1" data-path="WE_10.html"><a href="WE_10.html"><i class="fa fa-check"></i><b>13.1</b> Worked Example<span></span></a>
<ul>
<li><a href="WE_10.html#overview-of-worked-example-9">1. Overview of Worked Example<span></span></a></li>
<li><a href="WE_10.html#explore-the-data-set">2. Explore the data set<span></span></a></li>
<li><a href="WE_10.html#setting-cost-values-and-calculating-conductance">3. Setting cost values and calculating conductance<span></span></a></li>
<li><a href="WE_10.html#convert-conductance-into-effective-distance">4. Convert conductance into effective distance<span></span></a></li>
<li><a href="WE_10.html#create-cost-distance-matrices">5. Create cost-distance matrices<span></span></a></li>
<li><a href="WE_10.html#how-does-changing-resolution-affect-these-metrics">6. How does changing resolution affect these metrics?<span></span></a></li>
<li><a href="WE_10.html#references">7. References<span></span></a></li>
</ul></li>
<li class="chapter" data-level="13.2" data-path="bonus-10a.html"><a href="bonus-10a.html"><i class="fa fa-check"></i><b>13.2</b> Bonus: ‘radish’ tutorial<span></span></a>
<ul>
<li><a href="bonus-10a.html#overview-of-bonus-materials-1">1. Overview of Bonus Materials<span></span></a></li>
<li><a href="bonus-10a.html#import-data">2. Import data<span></span></a></li>
<li><a href="bonus-10a.html#basic-model-fitting">3. Basic model fitting<span></span></a></li>
<li><a href="bonus-10a.html#advanced-modeling-topics">4. Advanced modeling topics<span></span></a></li>
<li><a href="bonus-10a.html#summary">5. Summary<span></span></a></li>
<li><a href="bonus-10a.html#references-1">6. References<span></span></a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="14" data-path="Week11.html"><a href="Week11.html"><i class="fa fa-check"></i><b>14</b> Week 11: Detecting Adaptation<span></span></a>
<ul>
<li class="chapter" data-level="14.1" data-path="WE_11.html"><a href="WE_11.html"><i class="fa fa-check"></i><b>14.1</b> Worked Example<span></span></a>
<ul>
<li><a href="WE_11.html#overview-of-worked-example-10">1. Overview of Worked Example<span></span></a></li>
<li><a href="WE_11.html#import-and-prepare-the-data">2. Import and prepare the data<span></span></a></li>
<li><a href="WE_11.html#latent-factor-mixed-models-lfmm-a-univariate-gea">3. Latent Factor Mixed Models (LFMM): a univariate GEA<span></span></a></li>
<li><a href="WE_11.html#redundancy-analysis-rda-a-multivariate-gea">4. Redundancy Analysis (RDA): a multivariate GEA<span></span></a></li>
<li><a href="WE_11.html#compare-lfmm-and-rda-candidates">5. Compare LFMM and RDA candidates<span></span></a></li>
<li><a href="WE_11.html#a-quick-note-on-controlling-for-population-structure-in-rda">6. A quick note on controlling for population structure in RDA<span></span></a></li>
<li><a href="WE_11.html#references-2">7. References<span></span></a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="15" data-path="Week12.html"><a href="Week12.html"><i class="fa fa-check"></i><b>15</b> Week 12: Model Selection<span></span></a>
<ul>
<li class="chapter" data-level="15.1" data-path="WE_12.html"><a href="WE_12.html"><i class="fa fa-check"></i><b>15.1</b> Worked Example<span></span></a>
<ul>
<li><a href="WE_12.html#overview-of-worked-example-11">1. Overview of Worked Example<span></span></a></li>
<li><a href="WE_12.html#fitting-candidate-models">2. Fitting candidate models<span></span></a></li>
<li><a href="WE_12.html#your-turn-to-test-additional-hypotheses">3. Your turn to test additional hypotheses!<span></span></a></li>
<li><a href="WE_12.html#nested-model-nmlpe-for-hierarchical-sampling-designs">4. Nested model (NMLPE) for hierarchical sampling designs<span></span></a></li>
<li><a href="WE_12.html#references-3">5. References<span></span></a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="16" data-path="Week13.html"><a href="Week13.html"><i class="fa fa-check"></i><b>16</b> Week 13: Gravity Models<span></span></a>
<ul>
<li class="chapter" data-level="16.1" data-path="WE_13.html"><a href="WE_13.html"><i class="fa fa-check"></i><b>16.1</b> Worked Example<span></span></a>
<ul>
<li><a href="WE_13.html#overview-of-worked-example-12">1. Overview of Worked Example<span></span></a></li>
<li><a href="WE_13.html#setup">2. Setup<span></span></a></li>
<li><a href="WE_13.html#wetland-complex-data-preparation">3. Wetland complex data preparation<span></span></a></li>
<li><a href="WE_13.html#wetland-field-data-preparation">4. Wetland field-data preparation<span></span></a></li>
<li><a href="WE_13.html#saturated-graph">5. Saturated Graph<span></span></a></li>
<li><a href="WE_13.html#spatial-model-data-prepration">6. Spatial model data prepration<span></span></a></li>
<li><a href="WE_13.html#gravity-model">7. Gravity model<span></span></a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="17" data-path="Week14.html"><a href="Week14.html"><i class="fa fa-check"></i><b>17</b> Week 14: Contemporary Gene Flow<span></span></a>
<ul>
<li class="chapter" data-level="17.1" data-path="WE_14.html"><a href="WE_14.html"><i class="fa fa-check"></i><b>17.1</b> Worked Example<span></span></a>
<ul>
<li><a href="WE_14.html#overview-3">1. Overview<span></span></a></li>
<li><a href="WE_14.html#import-genotypes">2. Import Genotypes<span></span></a></li>
<li><a href="WE_14.html#pollen-pool-genetic-structure">3. Pollen pool genetic structure<span></span></a></li>
<li><a href="WE_14.html#paternity-analysis">4. Paternity analysis<span></span></a></li>
<li><a href="WE_14.html#linking-paternity-to-ecological-variables">5. Linking paternity to ecological variables<span></span></a></li>
<li><a href="WE_14.html#references-4">6. References<span></span></a></li>
</ul></li>
</ul></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Landscape Genetic Data Analysis with R</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="WE_14" class="section level2 hasAnchor" number="17.1">
<h2><span class="header-section-number">17.1</span> Worked Example<a href="WE_14.html#WE_14" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Michelle DiLeo</p>
<div id="overview-3" class="section level3 unnumbered hasAnchor">
<h3>1. Overview<a href="WE_14.html#overview-3" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div id="a.-background-and-goals" class="section level4 unnumbered hasAnchor">
<h4>a. Background and Goals<a href="WE_14.html#a.-background-and-goals" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>In this lab, you will use the <code>gstudio</code> package to analyse parent-offspring data for the wildflower <em>Pulsatilla vulgaris</em>. You will learn:</p>
<ul>
<li>how to conduct landscape genetic analyses on pollen pools,</li>
<li>how to conduct paternity analysis to quantify contemporary pollen flow, and</li>
<li>how to relate realized pollen flow events to landscape variables.</li>
</ul>
<p>The data are from DiLeo et al. (in press). The data include genotypes from seeds, mothers, and all potential fathers from seven focal populations. The study area in the Franconian Jura in Germany is the same as for the <em>Dianthus carthusianorum</em> data from Week 7 (see introductory video by Yessica Rico for the Dianthus data set to learn more about the system).</p>
<p>Here, we will recreate one analysis from DiLeo et al. (in press), looking at the ecological factors that best explain pollen immigration rates of the seven populations.</p>
</div>
<div id="b.-data-set-13" class="section level4 unnumbered hasAnchor">
<h4>b. Data set<a href="WE_14.html#b.-data-set-13" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>There are three data files associated with this lab:</p>
<ul>
<li>“pulsatilla_genotypes.csv,” which contains the genotypes of offspring, mothers, and potential fathers from seven populations,</li>
<li>“pulsatilla_momVariables.csv,” which contains ecological variables collect for each mother, and</li>
<li>“pulsatilla_population.csv,” which contains landscape and population size information for each of the seven populations.</li>
</ul>
</div>
<div id="c.-required-packages-1" class="section level4 unnumbered hasAnchor">
<h4>c. Required packages<a href="WE_14.html#c.-required-packages-1" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>Install some packages needed for this worked example.</p>
<div class="sourceCode" id="cb1429"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1429-1"><a href="WE_14.html#cb1429-1" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span>(<span class="sc">!</span><span class="fu">requireNamespace</span>(<span class="st">&quot;popgraph&quot;</span>, <span class="at">quietly =</span> <span class="cn">TRUE</span>))</span>
<span id="cb1429-2"><a href="WE_14.html#cb1429-2" aria-hidden="true" tabindex="-1"></a>{</span>
<span id="cb1429-3"><a href="WE_14.html#cb1429-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">install.packages</span>(<span class="fu">c</span>(<span class="st">&quot;RgoogleMaps&quot;</span>, <span class="st">&quot;geosphere&quot;</span>, <span class="st">&quot;proto&quot;</span>, <span class="st">&quot;sampling&quot;</span>, </span>
<span id="cb1429-4"><a href="WE_14.html#cb1429-4" aria-hidden="true" tabindex="-1"></a>                      <span class="st">&quot;seqinr&quot;</span>, <span class="st">&quot;spacetime&quot;</span>, <span class="st">&quot;spdep&quot;</span>), <span class="at">dependencies=</span><span class="cn">TRUE</span>)</span>
<span id="cb1429-5"><a href="WE_14.html#cb1429-5" aria-hidden="true" tabindex="-1"></a>  remotes<span class="sc">::</span><span class="fu">install_github</span>(<span class="st">&quot;dyerlab/popgraph&quot;</span>)</span>
<span id="cb1429-6"><a href="WE_14.html#cb1429-6" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb1429-7"><a href="WE_14.html#cb1429-7" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span>(<span class="sc">!</span><span class="fu">requireNamespace</span>(<span class="st">&quot;gstudio&quot;</span>, <span class="at">quietly =</span> <span class="cn">TRUE</span>)) remotes<span class="sc">::</span><span class="fu">install_github</span>(<span class="st">&quot;dyerlab/gstudio&quot;</span>)</span></code></pre></div>
<div class="sourceCode" id="cb1430"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1430-1"><a href="WE_14.html#cb1430-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(LandGenCourse)</span>
<span id="cb1430-2"><a href="WE_14.html#cb1430-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(dplyr)</span>
<span id="cb1430-3"><a href="WE_14.html#cb1430-3" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggplot2)</span>
<span id="cb1430-4"><a href="WE_14.html#cb1430-4" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggmap)</span>
<span id="cb1430-5"><a href="WE_14.html#cb1430-5" aria-hidden="true" tabindex="-1"></a><span class="co">#library(gstudio)</span></span>
<span id="cb1430-6"><a href="WE_14.html#cb1430-6" aria-hidden="true" tabindex="-1"></a><span class="co">#library(pegas)</span></span>
<span id="cb1430-7"><a href="WE_14.html#cb1430-7" aria-hidden="true" tabindex="-1"></a><span class="co">#library(vegan)</span></span>
<span id="cb1430-8"><a href="WE_14.html#cb1430-8" aria-hidden="true" tabindex="-1"></a><span class="co">#library(purrr)</span></span>
<span id="cb1430-9"><a href="WE_14.html#cb1430-9" aria-hidden="true" tabindex="-1"></a><span class="co">#library(MuMIn)</span></span>
<span id="cb1430-10"><a href="WE_14.html#cb1430-10" aria-hidden="true" tabindex="-1"></a><span class="co">#library(lme4)</span></span>
<span id="cb1430-11"><a href="WE_14.html#cb1430-11" aria-hidden="true" tabindex="-1"></a><span class="co">#library(cowplot)</span></span></code></pre></div>
</div>
</div>
<div id="import-genotypes" class="section level3 unnumbered hasAnchor">
<h3>2. Import Genotypes<a href="WE_14.html#import-genotypes" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>The package <code>gstudio</code> makes it very easy to import genotype data (see Week 1). Typically, we store our genotype data in a spreadsheet with one line per individual and two columns per locus. Using the <code>read_population</code> function, we tell <code>gstudio</code> which columns contain our loci. The argument <code>type="column"</code> indicates that our raw text file stores genotypes in two columns per locus.</p>
<div id="a.-import-genotype-data" class="section level4 unnumbered hasAnchor">
<h4>a. Import genotype data<a href="WE_14.html#a.-import-genotype-data" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<div class="sourceCode" id="cb1431"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1431-1"><a href="WE_14.html#cb1431-1" aria-hidden="true" tabindex="-1"></a>dat <span class="ot">&lt;-</span> gstudio<span class="sc">::</span><span class="fu">read_population</span>(<span class="fu">system.file</span>(<span class="st">&quot;extdata&quot;</span>,              </span>
<span id="cb1431-2"><a href="WE_14.html#cb1431-2" aria-hidden="true" tabindex="-1"></a>        <span class="st">&quot;pulsatilla_genotypes.csv&quot;</span>, <span class="at">package=</span><span class="st">&quot;LandGenCourse&quot;</span>), </span>
<span id="cb1431-3"><a href="WE_14.html#cb1431-3" aria-hidden="true" tabindex="-1"></a>        <span class="at">type =</span> <span class="st">&quot;column&quot;</span>, <span class="at">locus.columns =</span> <span class="dv">6</span><span class="sc">:</span><span class="dv">19</span>)</span>
<span id="cb1431-4"><a href="WE_14.html#cb1431-4" aria-hidden="true" tabindex="-1"></a>dat<span class="sc">$</span>ID <span class="ot">&lt;-</span> <span class="fu">as.factor</span>(dat<span class="sc">$</span>ID)</span></code></pre></div>
<p>Let’s take a look at the data:</p>
<div class="sourceCode" id="cb1432"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1432-1"><a href="WE_14.html#cb1432-1" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(dat)</span></code></pre></div>
<pre><code>##   ID OffID Population       X       Y  loc1_a  loc2_a  loc3_a  loc4_a  loc5_a
## 1 62     0        A21 4426941 5427173 340:340 422:422 413:425 446:448 121:121
## 2 64     0        A21 4426933 5427178 334:334 424:424 417:434 444:449 122:133
## 3 65     0        A21 4426936 5427173 338:340 417:422 417:419 446:452 135:137
## 4 66     0        A21 4426937 5427174 340:344 422:422 411:431 446:448 122:126
## 5 68     0        A21 4426934 5427171 336:342 417:422 423:438 448:448 119:123
## 6 69     0        A21 4426933 5427166 336:346 422:422 417:421 444:452 122:124
##    loc6_a  loc7_a
## 1 155:157 242:244
## 2 155:161 242:252
## 3 153:157 244:246
## 4 157:159 242:252
## 5 155:155 244:256
## 6 155:161 256:256</code></pre>
<p>There are 12 columns:</p>
<ul>
<li><strong>ID</strong>: A unique identifier for every adult.</li>
<li><strong>OffID</strong>: Zero if the sample is an adult and &gt; 0 if the sample is an offspring.</li>
<li><strong>Population</strong>: The population of origin.</li>
<li><strong>2 coordinate columns, X &amp; Y</strong>: The geographic coordinates of the sample (Euclidean coordinate system Gauss-Krueger GK4 with CRS(“+init=epsg:31468”)).</li>
<li><strong>7 locus columns</strong>: For each locus, the microsatellite genotype is stored in a single column with alleles separated by “:”</li>
</ul>
</div>
<div id="b.-plot-map-of-sampling-locations" class="section level4 unnumbered hasAnchor">
<h4>b. Plot map of sampling locations<a href="WE_14.html#b.-plot-map-of-sampling-locations" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>To create a map with <code>qmplot</code> (see Week 4), here we do the following:</p>
<ul>
<li>Use functions from <code>dplyr</code> to get mean coordinates per population</li>
<li>Convert the coordinates to lat-lon coordinates with <code>sf_project</code> from the <code>sf</code> package. We use the EPSG reference numbers (<a href="http://epsg.io/" class="uri">http://epsg.io/</a>) to specify the ‘from’ and ‘to’ coordinate systems.</li>
<li>Download a “terrain” map from “stamen,” with zoom level 12.</li>
<li>Add population labels with <code>geom_text</code>. We specify an offset along the x-axis,<code>nudge_x</code>, and for sites ‘A25’ and ‘A26’ also an offset along the y-axis to prevent labels from overlapping.</li>
</ul>
<div class="sourceCode" id="cb1434"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1434-1"><a href="WE_14.html#cb1434-1" aria-hidden="true" tabindex="-1"></a>Coords <span class="ot">&lt;-</span> dat <span class="sc">%&gt;%</span> <span class="fu">group_by</span>(Population) <span class="sc">%&gt;%</span> </span>
<span id="cb1434-2"><a href="WE_14.html#cb1434-2" aria-hidden="true" tabindex="-1"></a>          <span class="fu">summarize</span>(<span class="at">X =</span> <span class="fu">mean</span>(X), <span class="at">Y =</span> <span class="fu">mean</span>(Y))</span>
<span id="cb1434-3"><a href="WE_14.html#cb1434-3" aria-hidden="true" tabindex="-1"></a>Coords <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(Coords, sf<span class="sc">::</span><span class="fu">sf_project</span>(<span class="at">from =</span> <span class="st">&quot;+init=epsg:31468&quot;</span>, </span>
<span id="cb1434-4"><a href="WE_14.html#cb1434-4" aria-hidden="true" tabindex="-1"></a>          <span class="at">to =</span> <span class="st">&quot;+init=epsg:4326&quot;</span>, <span class="at">pts =</span> Coords[<span class="fu">c</span>(<span class="st">&quot;X&quot;</span>, <span class="st">&quot;Y&quot;</span>)]))</span>
<span id="cb1434-5"><a href="WE_14.html#cb1434-5" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(Coords)[<span class="dv">4</span><span class="sc">:</span><span class="dv">5</span>] <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;Longitude&quot;</span>, <span class="st">&quot;Latitude&quot;</span>)</span>
<span id="cb1434-6"><a href="WE_14.html#cb1434-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1434-7"><a href="WE_14.html#cb1434-7" aria-hidden="true" tabindex="-1"></a>bbox <span class="ot">&lt;-</span> <span class="fu">make_bbox</span>(<span class="at">lon =</span> Longitude, <span class="at">lat =</span> Latitude, <span class="at">data =</span> Coords, <span class="at">f=</span><span class="fl">0.2</span>)</span>
<span id="cb1434-8"><a href="WE_14.html#cb1434-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1434-9"><a href="WE_14.html#cb1434-9" aria-hidden="true" tabindex="-1"></a>MyMap <span class="ot">&lt;-</span> <span class="fu">ggmap</span>(<span class="fu">get_stamenmap</span>(bbox, <span class="at">maptype=</span><span class="st">&quot;terrain&quot;</span>, <span class="at">zoom=</span><span class="dv">12</span>, <span class="at">force=</span><span class="cn">TRUE</span>)) </span></code></pre></div>
<pre><code>## Source : http://tile.stamen.com/terrain/12/2172/1405.png</code></pre>
<pre><code>## Source : http://tile.stamen.com/terrain/12/2173/1405.png</code></pre>
<pre><code>## Source : http://tile.stamen.com/terrain/12/2174/1405.png</code></pre>
<pre><code>## Source : http://tile.stamen.com/terrain/12/2172/1406.png</code></pre>
<pre><code>## Source : http://tile.stamen.com/terrain/12/2173/1406.png</code></pre>
<pre><code>## Source : http://tile.stamen.com/terrain/12/2174/1406.png</code></pre>
<pre><code>## Source : http://tile.stamen.com/terrain/12/2172/1407.png</code></pre>
<pre><code>## Source : http://tile.stamen.com/terrain/12/2173/1407.png</code></pre>
<pre><code>## Source : http://tile.stamen.com/terrain/12/2174/1407.png</code></pre>
<pre><code>## Source : http://tile.stamen.com/terrain/12/2172/1408.png</code></pre>
<pre><code>## Source : http://tile.stamen.com/terrain/12/2173/1408.png</code></pre>
<pre><code>## Source : http://tile.stamen.com/terrain/12/2174/1408.png</code></pre>
<div class="sourceCode" id="cb1447"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1447-1"><a href="WE_14.html#cb1447-1" aria-hidden="true" tabindex="-1"></a>MyMap <span class="sc">+</span> <span class="fu">geom_text</span>(<span class="at">data =</span> Coords, <span class="at">mapping =</span> <span class="fu">aes</span>(Longitude, Latitude, </span>
<span id="cb1447-2"><a href="WE_14.html#cb1447-2" aria-hidden="true" tabindex="-1"></a>              <span class="at">label =</span> Population), <span class="at">size =</span> <span class="dv">4</span>, <span class="at">col =</span> <span class="st">&quot;black&quot;</span>, <span class="at">hjust =</span> <span class="dv">0</span>, </span>
<span id="cb1447-3"><a href="WE_14.html#cb1447-3" aria-hidden="true" tabindex="-1"></a>              <span class="at">nudge_x =</span> <span class="fl">0.005</span>, <span class="at">nudge_y =</span> <span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>,<span class="fl">0.002</span>,<span class="sc">-</span><span class="fl">0.001</span>,<span class="dv">0</span>,<span class="dv">0</span>,<span class="dv">0</span>)) </span></code></pre></div>
<p><img src="14-Week14_files/figure-html/unnamed-chunk-4-1.png" width="672" /></p>
</div>
</div>
<div id="pollen-pool-genetic-structure" class="section level3 unnumbered hasAnchor">
<h3>3. Pollen pool genetic structure<a href="WE_14.html#pollen-pool-genetic-structure" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div id="a.-subtract-maternal-contribution-to-offspring-genotype" class="section level4 unnumbered hasAnchor">
<h4>a. Subtract maternal contribution to offspring genotype<a href="WE_14.html#a.-subtract-maternal-contribution-to-offspring-genotype" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>Let’s look at the data for the mother with the ID=3083 and her associated offspring:</p>
<div class="sourceCode" id="cb1448"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1448-1"><a href="WE_14.html#cb1448-1" aria-hidden="true" tabindex="-1"></a>dat[dat<span class="sc">$</span>ID <span class="sc">==</span> <span class="st">&quot;3083&quot;</span>,]</span></code></pre></div>
<pre><code>##       ID OffID Population       X       Y  loc1_a  loc2_a  loc3_a  loc4_a
## 207 3083     0        A25 4422659 5425370 338:340 417:424 392:392 452:452
## 208 3083     1        A25 4422659 5425370 338:340 417:424 392:409 446:452
## 209 3083     3        A25 4422659 5425370 338:338 417:424         452:452
## 210 3083     4        A25 4422659 5425370 340:340 417:422 392:411 448:452
## 211 3083     5        A25 4422659 5425370 340:340 417:424 392:392 452:452
## 212 3083     6        A25 4422659 5425370 338:338 422:424 392:411 452:452
## 213 3083     7        A25 4422659 5425370 338:340 424:424 392:392 452:452
## 214 3083     8        A25 4422659 5425370 340:340 417:417 392:417 452:452
##      loc5_a  loc6_a  loc7_a
## 207 121:121 157:157 244:252
## 208 121:121 157:157 240:244
## 209 121:121 157:157        
## 210 121:121 155:157 236:252
## 211 121:121 157:157 244:244
## 212 121:121 157:159 242:244
## 213 121:121 157:157 244:244
## 214 121:121 155:157 242:244</code></pre>
<p>The first row is the mother and the subsequent rows are her offspring with OffIDs 1-8.</p>
<p>Now, use the <code>minus_mom</code> function to remove the mother’s contribution to the offspring genotypes. You will be left with just the paternal contribution. The allele frequecies of all paternal contributions associated with a single mother is called a “pollen pool.”</p>
<p>In some cases, the pollen contribution is ambiguous when the offspring has both alleles in common with the mother (e.g. see OffID 1, loc1_a and loc2_a). In such cases, both alleles are retained.</p>
<div class="sourceCode" id="cb1450"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1450-1"><a href="WE_14.html#cb1450-1" aria-hidden="true" tabindex="-1"></a>pollen <span class="ot">&lt;-</span> gstudio<span class="sc">::</span><span class="fu">minus_mom</span>(dat, <span class="at">MomCol =</span> <span class="st">&quot;ID&quot;</span>, <span class="at">OffCol =</span> <span class="st">&quot;OffID&quot;</span>)</span>
<span id="cb1450-2"><a href="WE_14.html#cb1450-2" aria-hidden="true" tabindex="-1"></a>pollen[pollen<span class="sc">$</span>ID <span class="sc">==</span> <span class="st">&quot;3083&quot;</span>,]</span></code></pre></div>
<pre><code>##       ID OffID Population       X       Y  loc1_a  loc2_a loc3_a loc4_a loc5_a
## 208 3083     1        A25 4422659 5425370 338:340 417:424    409    446    121
## 209 3083     3        A25 4422659 5425370     338 417:424           452    121
## 210 3083     4        A25 4422659 5425370     340     422    411    448    121
## 211 3083     5        A25 4422659 5425370     340 417:424    392    452    121
## 212 3083     6        A25 4422659 5425370     338     422    411    452    121
## 213 3083     7        A25 4422659 5425370 338:340     424    392    452    121
## 214 3083     8        A25 4422659 5425370     340     417    417    452    121
##     loc6_a loc7_a
## 208    157    240
## 209    157       
## 210    155    236
## 211    157    244
## 212    159    242
## 213    157    244
## 214    155    242</code></pre>
<p>The pollen pool data can be analysed as any other population genetic data.</p>
<p>Note: There are some missing genotypes for the offspring that trigger warnings, which you may ignore. Here we suppressed the warnings to reduce output.</p>
</div>
<div id="b.-twogener-analysis-amova-of-pollen-pool-data" class="section level4 unnumbered hasAnchor">
<h4>b. ‘TwoGener’ analysis (AMOVA of pollen pool data)<a href="WE_14.html#b.-twogener-analysis-amova-of-pollen-pool-data" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>For example, we can conduct an AMOVA (see Week 4) to test if pollen pools belonging to different mothers are signficantly differentiated from one another. When an AMOVA is applied to pollen pools it is called a ‘TwoGener’ analysis (Two-generational analysis; Smouse et al. 2001, Dyer et al. 2004).</p>
<p>Here we use the <code>amova</code> function from the <code>pegas</code> package to perform a hierarchical analysis of variance on genetic distance matrices (AMOVA), with moms nested within patches (this may take a while).</p>
<div class="sourceCode" id="cb1452"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1452-1"><a href="WE_14.html#cb1452-1" aria-hidden="true" tabindex="-1"></a>D <span class="ot">&lt;-</span> gstudio<span class="sc">::</span><span class="fu">genetic_distance</span>(pollen,<span class="at">mode=</span><span class="st">&quot;amova&quot;</span>)</span>
<span id="cb1452-2"><a href="WE_14.html#cb1452-2" aria-hidden="true" tabindex="-1"></a>D <span class="ot">&lt;-</span> <span class="fu">as.dist</span>(D)</span>
<span id="cb1452-3"><a href="WE_14.html#cb1452-3" aria-hidden="true" tabindex="-1"></a>Moms <span class="ot">&lt;-</span> pollen<span class="sc">$</span>ID</span>
<span id="cb1452-4"><a href="WE_14.html#cb1452-4" aria-hidden="true" tabindex="-1"></a>Populations <span class="ot">&lt;-</span> <span class="fu">as.factor</span>(pollen<span class="sc">$</span>Population) </span>
<span id="cb1452-5"><a href="WE_14.html#cb1452-5" aria-hidden="true" tabindex="-1"></a>amova.result <span class="ot">&lt;-</span> pegas<span class="sc">::</span><span class="fu">amova</span>(D <span class="sc">~</span> Populations<span class="sc">/</span>Moms, <span class="at">nperm=</span><span class="dv">500</span>)</span>
<span id="cb1452-6"><a href="WE_14.html#cb1452-6" aria-hidden="true" tabindex="-1"></a>amova.result</span></code></pre></div>
<pre><code>## 
##  Analysis of Molecular Variance
## 
## Call: pegas::amova(formula = D ~ Populations/Moms, nperm = 500)
## 
##                   SSD        MSD  df
## Populations  664.3079 110.717976   6
## Moms        1252.7296  36.844988  34
## Error       1895.1503   6.916607 274
## Total       3812.1878  12.140725 314
## 
## Variance components:
##             sigma2 P.value
## Populations 1.6747       0
## Moms        3.9278       0
## Error       6.9166        
## 
## Phi-statistics:
## Populations.in.GLOBAL (Phi_CT)        Moms.in.GLOBAL (Phi_ST) 
##                      0.1337729                      0.4475164 
##   Moms.in.Populations (Phi_SC) 
##                      0.3621954 
## 
## Variance coefficients:
##         a         b         c 
##  7.619648  7.920406 43.405291</code></pre>
<p>Calculate phi from the variance components:</p>
<div class="sourceCode" id="cb1454"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1454-1"><a href="WE_14.html#cb1454-1" aria-hidden="true" tabindex="-1"></a>phi <span class="ot">&lt;-</span> amova.result<span class="sc">$</span>varcomp[<span class="dv">1</span>]<span class="sc">/</span><span class="fu">sum</span>(amova.result<span class="sc">$</span>varcomp[<span class="dv">1</span>])</span>
<span id="cb1454-2"><a href="WE_14.html#cb1454-2" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(phi) <span class="ot">&lt;-</span> <span class="st">&quot;phi&quot;</span></span>
<span id="cb1454-3"><a href="WE_14.html#cb1454-3" aria-hidden="true" tabindex="-1"></a>phi</span></code></pre></div>
<pre><code>##                   phi
## Populations 0.1337729
## Moms        0.3137434
## Error       0.5524836</code></pre>
<p>Looking at the variance components and the phi-statistics (analogue to F-statistics), what can be said about differentiation among pollen pools?</p>
<p><strong>Questions:</strong></p>
<ul>
<li>Are pollen pools more differentiated among populations or among mothers within populations?</li>
<li>What does this mean biologically?</li>
</ul>
</div>
<div id="c.-including-space-isolation-by-distance" class="section level4 unnumbered hasAnchor">
<h4>c. Including space: Isolation by distance<a href="WE_14.html#c.-including-space-isolation-by-distance" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>The ‘TwoGener’ analysis gives a summary statistic describing overall differentiation, but this does not tell us anything about <strong>why</strong> the pollen pools are differentiated. In order to better understand <strong>why</strong>, we can analyse the data spatially just like any other landscape genetic dataset.</p>
<p>For example, we can test a basic hypothesis of isolation by distance by asking: are pollen pools from mothers that are geographically close more similar than those that are far? We test this by calculating pairwise genetic similarity (proportion of shared alleles) and relate this to geographic distance. For a more sophistocated analysis, the data could also be analysed as a “pollination graph” (Dyer et al. 2012), which is similar to population graphs that will be included in the Bonus Material for Week 13, but we will not do this here.</p>
<p>First, calculate matrix of pairwise genetic similarity among pollen pools.</p>
<div class="sourceCode" id="cb1456"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1456-1"><a href="WE_14.html#cb1456-1" aria-hidden="true" tabindex="-1"></a>dps <span class="ot">&lt;-</span> <span class="dv">1</span> <span class="sc">-</span> gstudio<span class="sc">::</span><span class="fu">genetic_distance</span>(pollen, <span class="at">stratum =</span> <span class="st">&quot;ID&quot;</span>, <span class="at">mode =</span> <span class="st">&quot;Dps&quot;</span>)</span></code></pre></div>
<pre><code>## Bray distance will be assumed to be entirely additive across loci.</code></pre>
<p>Note: the proportion of shared alleles is a measure of genetic similarity, hence we expect it to decrease with geographic distance. We convert this into a measure of genetic distance by subtracting it from 1 (see Week 5).</p>
<p>Calculate matrix of pairwise geographic distance among pollen pools:</p>
<div class="sourceCode" id="cb1458"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1458-1"><a href="WE_14.html#cb1458-1" aria-hidden="true" tabindex="-1"></a>xy <span class="ot">&lt;-</span> <span class="fu">unique</span>(<span class="fu">data.frame</span>(pollen<span class="sc">$</span>X, pollen<span class="sc">$</span>Y))</span>
<span id="cb1458-2"><a href="WE_14.html#cb1458-2" aria-hidden="true" tabindex="-1"></a>xy.dist <span class="ot">&lt;-</span> <span class="fu">as.matrix</span>(<span class="fu">dist</span>(xy))</span></code></pre></div>
<p>Plot ‘dps’ against geographic distance.</p>
<div class="sourceCode" id="cb1459"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1459-1"><a href="WE_14.html#cb1459-1" aria-hidden="true" tabindex="-1"></a><span class="fu">par</span>(<span class="at">mar=</span><span class="fu">c</span>(<span class="dv">4</span>,<span class="dv">4</span>,<span class="dv">0</span>,<span class="dv">0</span>))</span>
<span id="cb1459-2"><a href="WE_14.html#cb1459-2" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(xy.dist[<span class="fu">lower.tri</span>(xy.dist)], dps[<span class="fu">lower.tri</span>(dps)], </span>
<span id="cb1459-3"><a href="WE_14.html#cb1459-3" aria-hidden="true" tabindex="-1"></a>    <span class="at">xlab =</span> <span class="st">&quot;Geographic distance (m)&quot;</span>, </span>
<span id="cb1459-4"><a href="WE_14.html#cb1459-4" aria-hidden="true" tabindex="-1"></a>     <span class="at">ylab =</span> <span class="st">&quot;1 - Proportion of shared alleles (dps)&quot;</span>)</span>
<span id="cb1459-5"><a href="WE_14.html#cb1459-5" aria-hidden="true" tabindex="-1"></a><span class="fu">abline</span>(<span class="fu">lm</span>(dps[<span class="fu">lower.tri</span>(dps)]<span class="sc">~</span>xy.dist[<span class="fu">lower.tri</span>(xy.dist)]))</span></code></pre></div>
<p><img src="14-Week14_files/figure-html/plot-1.png" width="672" /></p>
<p>We see that pollen pools belonging to mothers that are spatially close are more similar. Let’s quantify the strength of this relationship using a Mantel test.</p>
<div class="sourceCode" id="cb1460"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1460-1"><a href="WE_14.html#cb1460-1" aria-hidden="true" tabindex="-1"></a>vegan<span class="sc">::</span><span class="fu">mantel</span>(xy.dist, dps)</span></code></pre></div>
<pre><code>## 
## Mantel statistic based on Pearson&#39;s product-moment correlation 
## 
## Call:
## vegan::mantel(xdis = xy.dist, ydis = dps) 
## 
## Mantel statistic r: 0.4432 
##       Significance: 0.001 
## 
## Upper quantiles of permutations (null model):
##   90%   95% 97.5%   99% 
## 0.108 0.140 0.177 0.201 
## Permutation: free
## Number of permutations: 999</code></pre>
<p>Note: it seems that the <code>mantel</code> function in <code>vegan</code> automatically tests a one-sided alternative that the Mantel correlation is greater than zero. This means that if we had used ‘Dps’ as a measure of genetic similarity, the default setting would have resulted in a p-value of 0.999. Unfortunately, for now, this is not discussed in the help file and there is no option to change the alternative.</p>
</div>
</div>
<div id="paternity-analysis" class="section level3 unnumbered hasAnchor">
<h3>4. Paternity analysis<a href="WE_14.html#paternity-analysis" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>‘TwoGener’ and related methods are sensitive to inbreeding and strong genetic structure of adult populations, although methods exist to correct for some of these things (e.g. Dyer et al. 2004). However, it is often the best (or only) option for quantifying pollen flow for high density species where other analyses such as paternity assignment cannot be conducted.</p>
<p>In our case, we have complete sampling of potential fathers within the seven populations (i.e., patches) from which mothers and their offspring were sampled. Therefore, we are able to conduct a paternity analysis, which will give us more detailed information about contemporary pollen flow.</p>
<p>Note that we do not have complete sampling of all potential fathers within the larger study region, and so we assume that any seed that is not assigned to a father is a pollen immigration event. In this section, we will conduct a paternity analysis using the <code>gstudio</code> package and relate the results to landscape variables.</p>
<div id="a.-exclusion-probabilities" class="section level4 unnumbered hasAnchor">
<h4>a. Exclusion probabilities<a href="WE_14.html#a.-exclusion-probabilities" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>First let’s check if our genetic markers have sufficient information (genetic resolution) to discriminate among alternative fathers by calculating exclusion probabilites.</p>
<div class="sourceCode" id="cb1462"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1462-1"><a href="WE_14.html#cb1462-1" aria-hidden="true" tabindex="-1"></a><span class="do">## e.clusion probabilities</span></span>
<span id="cb1462-2"><a href="WE_14.html#cb1462-2" aria-hidden="true" tabindex="-1"></a>pollen.freqs <span class="ot">&lt;-</span> gstudio<span class="sc">::</span><span class="fu">frequencies</span>(dat)</span>
<span id="cb1462-3"><a href="WE_14.html#cb1462-3" aria-hidden="true" tabindex="-1"></a>p.excl <span class="ot">&lt;-</span> gstudio<span class="sc">::</span><span class="fu">exclusion_probability</span>( pollen.freqs )</span>
<span id="cb1462-4"><a href="WE_14.html#cb1462-4" aria-hidden="true" tabindex="-1"></a>p.excl</span></code></pre></div>
<pre><code>##    Locus     Pexcl  PexclMax  Fraction
## 1 loc1_a 0.6817515 0.8926727 0.7637194
## 2 loc2_a 0.3359125 0.7434082 0.4518547
## 3 loc3_a 0.7614844 0.9218755 0.8260165
## 4 loc4_a 0.5451972 0.7718336 0.7063663
## 5 loc5_a 0.5984514 0.8981062 0.6663481
## 6 loc6_a 0.5998659 0.8537849 0.7025960
## 7 loc7_a 0.7002772 0.8424075 0.8312808</code></pre>
<p>The second column, ‘Pexcl,’ gives us the information we need. We can see that certain loci have poor ability to exclude alternative potential fathers (e.g. ‘loc2’) and others are much more informative (e.g. ‘loc3’). Luckily, we have multiple loci to use and the multilocus exclusion probability can be calculated by multiplying across loci.</p>
<p>The code below does the following:</p>
<ul>
<li><code>1 - unlist(p.excl$Pexcl)</code>: extract the elements of <code>p.excl.Pexcl</code> (the exclusion probabilities for all loci) as a vector and calculage for each locus, the complement of the exclusion probability.<br />
</li>
<li><code>1 - prod()</code>: multiplies the probabilities, and take the complement to get the multilocus exculsion probability.</li>
</ul>
<div class="sourceCode" id="cb1464"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1464-1"><a href="WE_14.html#cb1464-1" aria-hidden="true" tabindex="-1"></a><span class="dv">1</span><span class="sc">-</span> <span class="fu">prod</span>((<span class="dv">1</span><span class="sc">-</span><span class="fu">unlist</span>(p.excl<span class="sc">$</span>Pexcl)))</span></code></pre></div>
<pre><code>## [1] 0.9988959</code></pre>
<p>We see that when using all seven loci, we achieve a very high exlusion probability, and that on average we should be able to exclude 99.9 of alternative fathers.</p>
</div>
<div id="b.-paternity-analysis" class="section level4 unnumbered hasAnchor">
<h4>b. Paternity analysis<a href="WE_14.html#b.-paternity-analysis" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>We will use the <code>gstudio</code> function <code>paternity</code>, which conducts a fractional-allocation paternity analyis. This approach is useful when full exclusion cannot be achieved. In case multiple potential fathers cannot be exluded for a particular offspring, a likelihood of paternity is calculated for each non-excluded father based on Mendelian transition probabilites.</p>
<p>Note that in the original paper from which these data originate, we used a more sophisticated approach implemented in the program ‘COLONY2.’ ‘COLONY2’ can account for genotyping errors, whereas the <code>gstudio</code> function <code>paternity</code> approach cannot and so we had to throw out some of our data here.</p>
<p>Now let’s conduct a paternity analysis for the seeds from mother 3083:</p>
<div class="sourceCode" id="cb1466"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1466-1"><a href="WE_14.html#cb1466-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Select all offspring of mom 3083:</span></span>
<span id="cb1466-2"><a href="WE_14.html#cb1466-2" aria-hidden="true" tabindex="-1"></a>offspring<span class="fl">.3083</span> <span class="ot">&lt;-</span> <span class="fu">subset</span>(dat, OffID<span class="sc">!=</span><span class="dv">0</span> <span class="sc">&amp;</span> ID <span class="sc">==</span> <span class="st">&quot;3083&quot;</span>)</span>
<span id="cb1466-3"><a href="WE_14.html#cb1466-3" aria-hidden="true" tabindex="-1"></a><span class="co"># Select mom 3083:</span></span>
<span id="cb1466-4"><a href="WE_14.html#cb1466-4" aria-hidden="true" tabindex="-1"></a>mother<span class="fl">.3083</span> <span class="ot">&lt;-</span> <span class="fu">subset</span>(dat, ID <span class="sc">==</span> <span class="st">&quot;3083&quot;</span> <span class="sc">&amp;</span> dat<span class="sc">$</span>OffID <span class="sc">==</span> <span class="dv">0</span> )</span>
<span id="cb1466-5"><a href="WE_14.html#cb1466-5" aria-hidden="true" tabindex="-1"></a><span class="co"># Select all potential fathers in the same population as mom 3083:</span></span>
<span id="cb1466-6"><a href="WE_14.html#cb1466-6" aria-hidden="true" tabindex="-1"></a>fathers<span class="fl">.3083</span> <span class="ot">&lt;-</span> <span class="fu">subset</span>(dat, OffID <span class="sc">==</span> <span class="dv">0</span> <span class="sc">&amp;</span> Population <span class="sc">%in%</span> offspring<span class="fl">.3083</span><span class="sc">$</span>Population)</span>
<span id="cb1466-7"><a href="WE_14.html#cb1466-7" aria-hidden="true" tabindex="-1"></a><span class="co"># Paternity analysis: </span></span>
<span id="cb1466-8"><a href="WE_14.html#cb1466-8" aria-hidden="true" tabindex="-1"></a>pat<span class="fl">.3083</span> <span class="ot">&lt;-</span> gstudio<span class="sc">::</span><span class="fu">paternity</span>(offspring<span class="fl">.3083</span>, mother<span class="fl">.3083</span>, fathers<span class="fl">.3083</span> )</span>
<span id="cb1466-9"><a href="WE_14.html#cb1466-9" aria-hidden="true" tabindex="-1"></a>pat<span class="fl">.3083</span></span></code></pre></div>
<pre><code>##    MomID OffID DadID        Fij
## 1   3083     1   212 1.00000000
## 2   3083     3  3083 0.55172414
## 3   3083     3   218 0.13793103
## 4   3083     3   230 0.13793103
## 5   3083     3  3084 0.06896552
## 6   3083     3   255 0.03448276
## 7   3083     3   261 0.03448276
## 8   3083     3  3081 0.03448276
## 9   3083     5  3083 0.72727273
## 10  3083     5   234 0.18181818
## 11  3083     5   219 0.04545455
## 12  3083     5   239 0.04545455
## 13  3083     6   215 1.00000000
## 14  3083     7  3083 0.88888889
## 15  3083     7   219 0.05555556
## 16  3083     7   239 0.05555556
## 17  3083     8   212 1.00000000</code></pre>
<p>The output shows the probability of paternity (Fij) for each offspring-father pair, which is calcualted from Mendelian inheritance transition probabilities. Some offspring have multiple potential fathers and so appear in the table more than once (e.g. Offsprings 3 and 5). A higher ‘Fij’ value indcates that the Offspring-father pair is more likely.</p>
<p>Now let’s do the paternity analysis for the entire data set. First, we define a function <code>get.parentage</code> that does the paternity analysis for the offspring of one mom (i.e., one family).</p>
<div class="sourceCode" id="cb1468"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1468-1"><a href="WE_14.html#cb1468-1" aria-hidden="true" tabindex="-1"></a><span class="co"># make a dataframe just for the offspring:</span></span>
<span id="cb1468-2"><a href="WE_14.html#cb1468-2" aria-hidden="true" tabindex="-1"></a>offspring <span class="ot">&lt;-</span> <span class="fu">subset</span>(dat, OffID<span class="sc">!=</span><span class="dv">0</span>)</span>
<span id="cb1468-3"><a href="WE_14.html#cb1468-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1468-4"><a href="WE_14.html#cb1468-4" aria-hidden="true" tabindex="-1"></a><span class="do">## h.re is the function that we will apply to all mothers:</span></span>
<span id="cb1468-5"><a href="WE_14.html#cb1468-5" aria-hidden="true" tabindex="-1"></a>get.parentage <span class="ot">&lt;-</span> <span class="cf">function</span>(x){</span>
<span id="cb1468-6"><a href="WE_14.html#cb1468-6" aria-hidden="true" tabindex="-1"></a>  tmp.offspring <span class="ot">&lt;-</span> <span class="fu">subset</span>(offspring, ID <span class="sc">==</span> x)</span>
<span id="cb1468-7"><a href="WE_14.html#cb1468-7" aria-hidden="true" tabindex="-1"></a>  tmp.mother <span class="ot">&lt;-</span> <span class="fu">subset</span>(dat, ID <span class="sc">==</span> x <span class="sc">&amp;</span> OffID <span class="sc">==</span> <span class="dv">0</span>)</span>
<span id="cb1468-8"><a href="WE_14.html#cb1468-8" aria-hidden="true" tabindex="-1"></a>  tmp.fathers <span class="ot">&lt;-</span> <span class="fu">subset</span>(dat, OffID <span class="sc">==</span> <span class="dv">0</span> <span class="sc">&amp;</span> Population <span class="sc">%in%</span> tmp.offspring<span class="sc">$</span>Population)</span>
<span id="cb1468-9"><a href="WE_14.html#cb1468-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span>(gstudio<span class="sc">::</span><span class="fu">paternity</span>(tmp.offspring, tmp.mother, tmp.fathers ))</span>
<span id="cb1468-10"><a href="WE_14.html#cb1468-10" aria-hidden="true" tabindex="-1"></a>}</span></code></pre></div>
<p>We could use <code>lapply</code> to apply <code>get.parentage</code> to each mom and her offspring. Here, we use an alternative function <code>map</code> from package <code>purr</code> (which is part of the new ‘tidyverse’ set of packages by Hadley Wickham). This allows us to address the following issue: Because we have not sampled all potential fathers in the larger study region, <code>paternity</code> will return an error when all offspring of a certain mother cannot be assigned to any father.</p>
<p>To get around this, we will use the <code>possibly</code> function from the library <code>purrr</code>. When an error occurs, it will simply return <code>NA</code> instead of stopping our function. The function <code>map</code> knows how to apply a ‘purrr-ified’ function to an object.</p>
<p>This will take about a minute to run:</p>
<div class="sourceCode" id="cb1469"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1469-1"><a href="WE_14.html#cb1469-1" aria-hidden="true" tabindex="-1"></a><span class="co"># purrr-ify the function so that NA is returned when an error pops up:</span></span>
<span id="cb1469-2"><a href="WE_14.html#cb1469-2" aria-hidden="true" tabindex="-1"></a>possible_pat <span class="ot">&lt;-</span> purrr<span class="sc">::</span><span class="fu">possibly</span>(get.parentage, <span class="at">otherwise =</span> <span class="cn">NA_real_</span>)</span>
<span id="cb1469-3"><a href="WE_14.html#cb1469-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1469-4"><a href="WE_14.html#cb1469-4" aria-hidden="true" tabindex="-1"></a><span class="co"># run the function and store the output:</span></span>
<span id="cb1469-5"><a href="WE_14.html#cb1469-5" aria-hidden="true" tabindex="-1"></a><span class="co"># list of results for each mother:</span></span>
<span id="cb1469-6"><a href="WE_14.html#cb1469-6" aria-hidden="true" tabindex="-1"></a>pat.all <span class="ot">&lt;-</span> purrr<span class="sc">::</span><span class="fu">map</span>(<span class="fu">unique</span>(offspring<span class="sc">$</span>ID), possible_pat)</span>
<span id="cb1469-7"><a href="WE_14.html#cb1469-7" aria-hidden="true" tabindex="-1"></a><span class="do">## c.nvert the list to a dataframe:</span></span>
<span id="cb1469-8"><a href="WE_14.html#cb1469-8" aria-hidden="true" tabindex="-1"></a>pat.all <span class="ot">&lt;-</span> <span class="fu">do.call</span>(rbind, pat.all[<span class="sc">!</span><span class="fu">is.na</span>(pat.all)]) </span></code></pre></div>
<p>For the purposes of this tutorial, we will keep only the highest-probability father for each offspring. With your own data, you should choose some sort of threshold (e.g. keep only those with probability &gt; 0.9) or you can conduct analyses using the probability as weights.</p>
<div class="sourceCode" id="cb1470"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1470-1"><a href="WE_14.html#cb1470-1" aria-hidden="true" tabindex="-1"></a><span class="do">## c.eate a temporary ID that combines the MomID and the OffID</span></span>
<span id="cb1470-2"><a href="WE_14.html#cb1470-2" aria-hidden="true" tabindex="-1"></a>pat.all<span class="sc">$</span>tmpID <span class="ot">&lt;-</span> <span class="fu">paste</span>(pat.all<span class="sc">$</span>MomID, pat.all<span class="sc">$</span>OffID, <span class="at">sep=</span><span class="st">&quot;_&quot;</span>)</span>
<span id="cb1470-3"><a href="WE_14.html#cb1470-3" aria-hidden="true" tabindex="-1"></a><span class="do">## g.t rid of all rows with duplicated tmpIDs, leaving just the first entry for each</span></span>
<span id="cb1470-4"><a href="WE_14.html#cb1470-4" aria-hidden="true" tabindex="-1"></a>pat.sub <span class="ot">&lt;-</span> pat.all[<span class="sc">!</span><span class="fu">duplicated</span>(pat.all<span class="sc">$</span>tmpID),]</span>
<span id="cb1470-5"><a href="WE_14.html#cb1470-5" aria-hidden="true" tabindex="-1"></a><span class="do">## g.t rid of the tmpID column</span></span>
<span id="cb1470-6"><a href="WE_14.html#cb1470-6" aria-hidden="true" tabindex="-1"></a>pat.sub <span class="ot">&lt;-</span> pat.sub[,<span class="dv">1</span><span class="sc">:</span><span class="dv">4</span>] <span class="do">## g.t rid of the tmp_ID column</span></span></code></pre></div>
</div>
<div id="c.-visualize-reconstructed-pollen-flow-events-in-space" class="section level4 unnumbered hasAnchor">
<h4>c. Visualize reconstructed pollen flow events in space<a href="WE_14.html#c.-visualize-reconstructed-pollen-flow-events-in-space" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>We can use the <code>spiderplot_data</code> function from <code>gstudio</code> to visualize the paternity assignments. <code>spiderplot_data</code> takes the output from <code>paternity</code> and combines it with spatial XY coordinate information. The output includes the coordinates of both mother (X, Y) and father (Xend, Yend) for each offspring. Here we will also add in the population IDs so that we can visualize each of the seven populations separately. Note that the function asks for “longitude” and “latitude” but any form of spatial XY coordinates will do.</p>
<div class="sourceCode" id="cb1471"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1471-1"><a href="WE_14.html#cb1471-1" aria-hidden="true" tabindex="-1"></a>pat.sub <span class="ot">&lt;-</span> gstudio<span class="sc">::</span><span class="fu">spiderplot_data</span>(pat.sub, dat, <span class="at">longitude =</span> <span class="st">&quot;X&quot;</span>, <span class="at">latitude =</span> <span class="st">&quot;Y&quot;</span>)</span>
<span id="cb1471-2"><a href="WE_14.html#cb1471-2" aria-hidden="true" tabindex="-1"></a><span class="co"># Join data to add population IDs</span></span>
<span id="cb1471-3"><a href="WE_14.html#cb1471-3" aria-hidden="true" tabindex="-1"></a>pat.sub <span class="ot">&lt;-</span> <span class="fu">merge</span>(pat.sub, dat[, <span class="fu">c</span>(<span class="st">&quot;Population&quot;</span>, <span class="st">&quot;ID&quot;</span> ,<span class="st">&quot;OffID&quot;</span>)],</span>
<span id="cb1471-4"><a href="WE_14.html#cb1471-4" aria-hidden="true" tabindex="-1"></a>                 <span class="at">by.x=</span><span class="fu">c</span>(<span class="st">&quot;MomID&quot;</span>, <span class="st">&quot;OffID&quot;</span>), <span class="at">by.y=</span><span class="fu">c</span>(<span class="st">&quot;ID&quot;</span>, <span class="st">&quot;OffID&quot;</span>), <span class="at">all.x=</span>T)</span>
<span id="cb1471-5"><a href="WE_14.html#cb1471-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1471-6"><a href="WE_14.html#cb1471-6" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(pat.sub)</span></code></pre></div>
<pre><code>##   MomID OffID DadID Fij       X    Xend       Y    Yend Population
## 1  3080     1   211   1 4422657 4422658 5425371 5425371        A25
## 2  3080    10   211   1 4422657 4422658 5425371 5425371        A25
## 3  3080     2   214   1 4422657 4422658 5425371 5425370        A25
## 4  3080     3   214   1 4422657 4422658 5425371 5425370        A25
## 5  3080     4   211   1 4422657 4422658 5425371 5425371        A25
## 6  3080     5   211   1 4422657 4422658 5425371 5425371        A25</code></pre>
<p>Use <code>ggplot2</code> to visualize the reconstructed pollen flow events. Let’s look at population ‘A25’ first. The arrows point to mothers, and arrows without lines indicate selfing events. The darkness of the lines is proportional to the number of pollen flow events between a particular mother and father.</p>
<div class="sourceCode" id="cb1473"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1473-1"><a href="WE_14.html#cb1473-1" aria-hidden="true" tabindex="-1"></a>pop <span class="ot">&lt;-</span> <span class="st">&quot;A25&quot;</span></span>
<span id="cb1473-2"><a href="WE_14.html#cb1473-2" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>() <span class="sc">+</span></span>
<span id="cb1473-3"><a href="WE_14.html#cb1473-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">data=</span>dat[dat<span class="sc">$</span>Population<span class="sc">==</span>pop,], </span>
<span id="cb1473-4"><a href="WE_14.html#cb1473-4" aria-hidden="true" tabindex="-1"></a>             <span class="fu">aes</span>(X,Y),<span class="at">size=</span><span class="dv">3</span>, <span class="at">color=</span><span class="st">&quot;red&quot;</span>) <span class="sc">+</span></span>
<span id="cb1473-5"><a href="WE_14.html#cb1473-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_segment</span>(<span class="at">data=</span>pat.sub[pat.sub<span class="sc">$</span>Population<span class="sc">==</span>pop,], </span>
<span id="cb1473-6"><a href="WE_14.html#cb1473-6" aria-hidden="true" tabindex="-1"></a>       <span class="fu">aes</span>(<span class="at">x=</span>X, <span class="at">y=</span>Y, <span class="at">xend=</span>Xend, <span class="at">yend=</span>Yend), <span class="at">size=</span><span class="fl">0.5</span>, <span class="at">alpha=</span><span class="fl">0.2</span>, </span>
<span id="cb1473-7"><a href="WE_14.html#cb1473-7" aria-hidden="true" tabindex="-1"></a>       <span class="at">arrow =</span> <span class="fu">arrow</span>(<span class="at">ends =</span> <span class="st">&quot;first&quot;</span>, <span class="at">length =</span> <span class="fu">unit</span>(<span class="fl">0.3</span>, <span class="st">&quot;cm&quot;</span>))) <span class="sc">+</span></span>
<span id="cb1473-8"><a href="WE_14.html#cb1473-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="st">&quot;none&quot;</span>)</span></code></pre></div>
<p><img src="14-Week14_files/figure-html/plot_pollenFlow-1.png" width="672" /></p>
<p>Try modifying the code to look at patterns from other populations.</p>
<p><strong>Questions:</strong></p>
<ul>
<li>Do pollen flow events appear to be amongst near-neigbours?</li>
<li>Are there clear differences in pollen flow patterns among populations?</li>
</ul>
</div>
<div id="d.-visualize-dispersal-kernel" class="section level4 unnumbered hasAnchor">
<h4>d. Visualize dispersal kernel<a href="WE_14.html#d.-visualize-dispersal-kernel" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>We can calculate the distance of each pollen flow event, and then plot the distribution to visualize the dispersal kernel.</p>
<p>Calculate distance for each event:</p>
<div class="sourceCode" id="cb1474"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1474-1"><a href="WE_14.html#cb1474-1" aria-hidden="true" tabindex="-1"></a>pat.sub<span class="sc">$</span>pollen.dist <span class="ot">&lt;-</span> <span class="fu">unlist</span>(<span class="fu">lapply</span>(<span class="dv">1</span><span class="sc">:</span><span class="fu">nrow</span>(pat.sub),</span>
<span id="cb1474-2"><a href="WE_14.html#cb1474-2" aria-hidden="true" tabindex="-1"></a>    <span class="cf">function</span>(x) <span class="fu">dist</span>(<span class="fu">rbind</span>(<span class="fu">c</span>(pat.sub<span class="sc">$</span>X[x], pat.sub<span class="sc">$</span>Y[x]), </span>
<span id="cb1474-3"><a href="WE_14.html#cb1474-3" aria-hidden="true" tabindex="-1"></a>                <span class="fu">c</span>(pat.sub<span class="sc">$</span>Xend[x], pat.sub<span class="sc">$</span>Yend[x]))) ))</span></code></pre></div>
<p>Plot the distribution of pollination events that are greater than 0 m (i.e. excluding selfing):</p>
<div class="sourceCode" id="cb1475"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1475-1"><a href="WE_14.html#cb1475-1" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(pat.sub[pat.sub<span class="sc">$</span>pollen.dist <span class="sc">&gt;</span><span class="dv">0</span>,]) <span class="sc">+</span></span>
<span id="cb1475-2"><a href="WE_14.html#cb1475-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_histogram</span>( <span class="fu">aes</span>(<span class="at">x=</span>pollen.dist),  <span class="at">bins=</span><span class="dv">20</span>) <span class="sc">+</span></span>
<span id="cb1475-3"><a href="WE_14.html#cb1475-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">xlab</span>(<span class="st">&quot;Distance from pollen source (m)&quot;</span>)  <span class="sc">+</span></span>
<span id="cb1475-4"><a href="WE_14.html#cb1475-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ylab</span>(<span class="st">&quot;Number of pollen flow events&quot;</span>)</span></code></pre></div>
<p><img src="14-Week14_files/figure-html/dispersal_kernel-1.png" width="672" /></p>
</div>
</div>
<div id="linking-paternity-to-ecological-variables" class="section level3 unnumbered hasAnchor">
<h3>5. Linking paternity to ecological variables<a href="WE_14.html#linking-paternity-to-ecological-variables" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Now that we have conducted our paternity analysis, we can ask which ecological factors explain the patterns that we see.</p>
<ul>
<li>First we will look within populations: what factors explain pollen flow distances within populations?</li>
<li>Second, we will look at the population level: what factors explain pollen immigration rates?</li>
</ul>
<div id="a.-explain-pollen-flow-within-populations" class="section level4 unnumbered hasAnchor">
<h4>a. Explain pollen flow within populations<a href="WE_14.html#a.-explain-pollen-flow-within-populations" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>We have collected some information about mothers, which we now add to the ‘pat.sub’ dataframe. Specifically, we have measured:</p>
<ul>
<li><strong>Mother isolation</strong>: how far a mom is from other conspecifics, and</li>
<li><strong>Local floral density</strong>: how many flowers are within 2 m of mother plants.</li>
</ul>
<p>We are only interested in outcrossed pollen events, so we make a new data frame that excludes selfing.</p>
<div class="sourceCode" id="cb1476"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1476-1"><a href="WE_14.html#cb1476-1" aria-hidden="true" tabindex="-1"></a><span class="co"># read in the data</span></span>
<span id="cb1476-2"><a href="WE_14.html#cb1476-2" aria-hidden="true" tabindex="-1"></a>mom.vars <span class="ot">&lt;-</span> <span class="fu">read.csv</span>(<span class="fu">system.file</span>(<span class="st">&quot;extdata&quot;</span>,                 </span>
<span id="cb1476-3"><a href="WE_14.html#cb1476-3" aria-hidden="true" tabindex="-1"></a> <span class="st">&quot;pulsatilla_momVariables.csv&quot;</span>, <span class="at">package=</span><span class="st">&quot;LandGenCourse&quot;</span>))</span>
<span id="cb1476-4"><a href="WE_14.html#cb1476-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1476-5"><a href="WE_14.html#cb1476-5" aria-hidden="true" tabindex="-1"></a><span class="do">## e.clude selfing</span></span>
<span id="cb1476-6"><a href="WE_14.html#cb1476-6" aria-hidden="true" tabindex="-1"></a>pat.outcrossed <span class="ot">&lt;-</span> <span class="fu">subset</span>(pat.sub, MomID <span class="sc">!=</span> DadID)</span>
<span id="cb1476-7"><a href="WE_14.html#cb1476-7" aria-hidden="true" tabindex="-1"></a><span class="do">## a.d mom variables to pat.outcrossed</span></span>
<span id="cb1476-8"><a href="WE_14.html#cb1476-8" aria-hidden="true" tabindex="-1"></a>pat.outcrossed <span class="ot">&lt;-</span> <span class="fu">merge</span>(pat.outcrossed, mom.vars, <span class="at">by.x =</span> <span class="st">&quot;MomID&quot;</span>, </span>
<span id="cb1476-9"><a href="WE_14.html#cb1476-9" aria-hidden="true" tabindex="-1"></a>                        <span class="at">by.y =</span> <span class="st">&quot;ID&quot;</span>, <span class="at">all.x=</span>T)</span>
<span id="cb1476-10"><a href="WE_14.html#cb1476-10" aria-hidden="true" tabindex="-1"></a><span class="co"># look at the data</span></span>
<span id="cb1476-11"><a href="WE_14.html#cb1476-11" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(pat.outcrossed)</span></code></pre></div>
<pre><code>##   MomID OffID DadID Fij       X    Xend       Y    Yend Population pollen.dist
## 1  3080     1   211   1 4422657 4422658 5425371 5425371        A25   0.8632989
## 2  3080    10   211   1 4422657 4422658 5425371 5425371        A25   0.8632989
## 3  3080     2   214   1 4422657 4422658 5425371 5425370        A25   1.3829147
## 4  3080     3   214   1 4422657 4422658 5425371 5425370        A25   1.3829147
## 5  3080     4   211   1 4422657 4422658 5425371 5425371        A25   0.8632989
## 6  3080     5   211   1 4422657 4422658 5425371 5425371        A25   0.8632989
##   mom.isolation flower.density
## 1      10.97917             11
## 2      10.97917             11
## 3      10.97917             11
## 4      10.97917             11
## 5      10.97917             11
## 6      10.97917             11</code></pre>
<p>Let’s run some models to test if <code>mom.isolation</code> or <code>flower.density</code> explain pollen flow distances. We used mixed models with population and mother ID as random effects to control for multiple sampling from the same mom and populations (see Week 6 videos for linear mixed models).</p>
<p>Note: for model selection with varying fixed effects and constant random effects, we fit the models with maximum likelihood ML, hence we set <code>REML=F</code> (see Week 12).</p>
<div class="sourceCode" id="cb1478"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1478-1"><a href="WE_14.html#cb1478-1" aria-hidden="true" tabindex="-1"></a><span class="co"># specify the model</span></span>
<span id="cb1478-2"><a href="WE_14.html#cb1478-2" aria-hidden="true" tabindex="-1"></a>mod <span class="ot">&lt;-</span> lme4<span class="sc">::</span><span class="fu">lmer</span>(<span class="fu">log</span>(pollen.dist) <span class="sc">~</span> <span class="fu">scale</span>(<span class="fu">log</span>(flower.density)) <span class="sc">+</span> </span>
<span id="cb1478-3"><a href="WE_14.html#cb1478-3" aria-hidden="true" tabindex="-1"></a>            <span class="fu">scale</span>(<span class="fu">log</span>(mom.isolation)) <span class="sc">+</span> (<span class="dv">1</span><span class="sc">|</span>Population<span class="sc">/</span>MomID),</span>
<span id="cb1478-4"><a href="WE_14.html#cb1478-4" aria-hidden="true" tabindex="-1"></a>            <span class="at">data=</span>pat.outcrossed, <span class="at">na.action =</span> <span class="st">&quot;na.fail&quot;</span>, <span class="at">REML=</span>F)</span></code></pre></div>
<pre><code>## boundary (singular) fit: see help(&#39;isSingular&#39;)</code></pre>
<p>Here we use the function <code>dredge</code> from package <code>MuMIn</code> to compare all nested submodels and select the ‘best’ model. This is not generally recommended (see Week 12), as submodels may not be biologically meaningful. In this case, there is good reason to hypothesize that either mom isolation or local flower density, or both, would affect pollination distance.</p>
<div class="sourceCode" id="cb1480"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1480-1"><a href="WE_14.html#cb1480-1" aria-hidden="true" tabindex="-1"></a>MuMIn<span class="sc">::</span><span class="fu">dredge</span>(mod)</span></code></pre></div>
<pre><code>## Fixed term is &quot;(Intercept)&quot;</code></pre>
<pre><code>## boundary (singular) fit: see help(&#39;isSingular&#39;)
## boundary (singular) fit: see help(&#39;isSingular&#39;)
## boundary (singular) fit: see help(&#39;isSingular&#39;)</code></pre>
<pre><code>## Global model call: lme4::lmer(formula = log(pollen.dist) ~ scale(log(flower.density)) + 
##     scale(log(mom.isolation)) + (1 | Population/MomID), data = pat.outcrossed, 
##     REML = F, na.action = &quot;na.fail&quot;)
## ---
## Model selection table 
##   (Int) scl(log(flw.dns)) scl(log(mom.isl)) df   logLik  AICc delta weight
## 4 1.075           -0.3609            0.3493  6 -102.581 218.0  0.00  0.698
## 2 1.076           -0.5151                    5 -105.193 221.0  2.99  0.157
## 3 1.094                              0.5977  5 -105.280 221.1  3.16  0.144
## 1 1.081                                      4 -110.975 230.3 12.35  0.001
## Models ranked by AICc(x) 
## Random terms (all models): 
##   1 | Population/MomID</code></pre>
<p>We see that the best model includes both flower density and mom isolation (for model selection, see Week 12 worked example and Week 7 video).</p>
<p>Note: we should follow up with residual analysis etc., which we’ll skip here to keep it short.</p>
<p>Let’s plot the relationships:</p>
<div class="sourceCode" id="cb1484"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1484-1"><a href="WE_14.html#cb1484-1" aria-hidden="true" tabindex="-1"></a>Mom.isolation.plot <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(pat.outcrossed, </span>
<span id="cb1484-2"><a href="WE_14.html#cb1484-2" aria-hidden="true" tabindex="-1"></a>       <span class="fu">aes</span>(<span class="at">x=</span><span class="fu">log</span>(mom.isolation), <span class="at">y=</span><span class="fu">log</span>(pollen.dist))) <span class="sc">+</span> </span>
<span id="cb1484-3"><a href="WE_14.html#cb1484-3" aria-hidden="true" tabindex="-1"></a>       <span class="fu">geom_point</span>() <span class="sc">+</span> <span class="fu">stat_smooth</span>(<span class="at">method=</span><span class="st">&quot;lm&quot;</span>, <span class="at">se=</span>F) <span class="sc">+</span> </span>
<span id="cb1484-4"><a href="WE_14.html#cb1484-4" aria-hidden="true" tabindex="-1"></a>       <span class="fu">xlab</span>(<span class="st">&quot;Log(mom isolation)&quot;</span>) <span class="sc">+</span></span>
<span id="cb1484-5"><a href="WE_14.html#cb1484-5" aria-hidden="true" tabindex="-1"></a>       <span class="fu">ylab</span>(<span class="st">&quot;Log(pollen flow distance)&quot;</span>)</span>
<span id="cb1484-6"><a href="WE_14.html#cb1484-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1484-7"><a href="WE_14.html#cb1484-7" aria-hidden="true" tabindex="-1"></a>Flower.density.plot <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(pat.outcrossed, </span>
<span id="cb1484-8"><a href="WE_14.html#cb1484-8" aria-hidden="true" tabindex="-1"></a>       <span class="fu">aes</span>(<span class="at">x=</span><span class="fu">log</span>(flower.density), </span>
<span id="cb1484-9"><a href="WE_14.html#cb1484-9" aria-hidden="true" tabindex="-1"></a>       <span class="at">y=</span><span class="fu">log</span>(pollen.dist))) <span class="sc">+</span> <span class="fu">geom_point</span>() <span class="sc">+</span></span>
<span id="cb1484-10"><a href="WE_14.html#cb1484-10" aria-hidden="true" tabindex="-1"></a>       <span class="fu">stat_smooth</span>(<span class="at">method=</span><span class="st">&quot;lm&quot;</span>, <span class="at">se=</span>F) <span class="sc">+</span> </span>
<span id="cb1484-11"><a href="WE_14.html#cb1484-11" aria-hidden="true" tabindex="-1"></a>       <span class="fu">xlab</span>(<span class="st">&quot;Log(flower density)&quot;</span>) <span class="sc">+</span></span>
<span id="cb1484-12"><a href="WE_14.html#cb1484-12" aria-hidden="true" tabindex="-1"></a>       <span class="fu">ylab</span>(<span class="st">&quot;Log(pollen flow distance)&quot;</span>)</span>
<span id="cb1484-13"><a href="WE_14.html#cb1484-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1484-14"><a href="WE_14.html#cb1484-14" aria-hidden="true" tabindex="-1"></a>cowplot<span class="sc">::</span><span class="fu">plot_grid</span>(Mom.isolation.plot, Flower.density.plot, </span>
<span id="cb1484-15"><a href="WE_14.html#cb1484-15" aria-hidden="true" tabindex="-1"></a>                   <span class="at">labels =</span> <span class="fu">c</span>(<span class="st">&quot;A&quot;</span>, <span class="st">&quot;B&quot;</span>))</span></code></pre></div>
<pre><code>## `geom_smooth()` using formula &#39;y ~ x&#39;
## `geom_smooth()` using formula &#39;y ~ x&#39;</code></pre>
<p><img src="14-Week14_files/figure-html/unnamed-chunk-6-1.png" width="672" /></p>
<p><strong>Questions:</strong></p>
<ul>
<li>Which moms (isolated/non-isolated, surrounded by more/less flowers) tend to receive pollen from further away?</li>
<li>What biological processes might explain these patterns?</li>
</ul>
</div>
<div id="b.-explain-pollen-flow-at-the-population-level" class="section level4 unnumbered hasAnchor">
<h4>b. Explain pollen flow at the population level<a href="WE_14.html#b.-explain-pollen-flow-at-the-population-level" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>Now let’s look at the population level. First we need to calculate how many offspring were unable to be assigned a father. We do this by adding the first four columns of the paternity analysis output (which contains only successful assignments) to the offspring dataframe (which includes all offspring, regardless of assignment). If there is no assigment, ‘Fij’ will be NA in the newly merged data frame.</p>
<div class="sourceCode" id="cb1486"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1486-1"><a href="WE_14.html#cb1486-1" aria-hidden="true" tabindex="-1"></a>offspring <span class="ot">&lt;-</span> <span class="fu">merge</span>(offspring, pat.sub[,<span class="dv">1</span><span class="sc">:</span><span class="dv">4</span>], <span class="at">by.x=</span><span class="fu">c</span>(<span class="st">&quot;ID&quot;</span>, <span class="st">&quot;OffID&quot;</span>), </span>
<span id="cb1486-2"><a href="WE_14.html#cb1486-2" aria-hidden="true" tabindex="-1"></a>                   <span class="at">by.y=</span><span class="fu">c</span>(<span class="st">&quot;MomID&quot;</span>, <span class="st">&quot;OffID&quot;</span>), <span class="at">all.x=</span>T)</span>
<span id="cb1486-3"><a href="WE_14.html#cb1486-3" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(offspring)</span></code></pre></div>
<pre><code>##     ID OffID Population       X       Y  loc1_a  loc2_a  loc3_a  loc4_a  loc5_a
## 1 3080     1        A25 4422657 5425371 334:358 417:424 392:411 446:448 121:121
## 2 3080     2        A25 4422657 5425371 334:358 422:424 392:434 442:446 121:121
## 3 3080     3        A25 4422657 5425371 334:354 422:424 392:434 446:446 118:121
## 4 3080     4        A25 4422657 5425371 334:354 422:424 392:392 446:446 121:121
## 5 3080     5        A25 4422657 5425371 352:358 417:424 392:411 446:446 121:121
## 6 3080     6        A25 4422657 5425371 352:354 417:424 392:392 446:446 121:121
##    loc6_a  loc7_a DadID Fij
## 1 157:165 242:258   211   1
## 2 157:157 252:258   214   1
## 3 157:157 236:258   214   1
## 4 153:157 242:258   211   1
## 5 157:165 240:258   211   1
## 6 153:157 240:256   211   1</code></pre>
<p>Calculate the number of outside pollination events per population by counting the number of NAs in the ‘Fij’ column:</p>
<div class="sourceCode" id="cb1488"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1488-1"><a href="WE_14.html#cb1488-1" aria-hidden="true" tabindex="-1"></a>num.out <span class="ot">&lt;-</span> <span class="fu">sapply</span>(<span class="fu">split</span>(offspring<span class="sc">$</span>Fij, offspring<span class="sc">$</span>Population),</span>
<span id="cb1488-2"><a href="WE_14.html#cb1488-2" aria-hidden="true" tabindex="-1"></a>                  <span class="cf">function</span>(x) <span class="fu">sum</span>(<span class="fu">is.na</span>(x)))</span></code></pre></div>
<p>Total number of pollination events per population (should be same as the number of seeds sampled):</p>
<div class="sourceCode" id="cb1489"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1489-1"><a href="WE_14.html#cb1489-1" aria-hidden="true" tabindex="-1"></a>num.tot <span class="ot">&lt;-</span> <span class="fu">table</span>(offspring<span class="sc">$</span>Population)</span></code></pre></div>
<p>Let’s put this information into a data frame and add ecological information that we have gathered about each population.</p>
<p>Specifically we have measured the proportion of forest cover within radii of 50, 100, 250, 500, and 1000 metres from population centroids, and have measured population size as the total number of flowering <em>P. vulgaris</em> plants per population.</p>
<div class="sourceCode" id="cb1490"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1490-1"><a href="WE_14.html#cb1490-1" aria-hidden="true" tabindex="-1"></a><span class="co"># turn it into a dataframe:</span></span>
<span id="cb1490-2"><a href="WE_14.html#cb1490-2" aria-hidden="true" tabindex="-1"></a>pop.df <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">Population=</span><span class="fu">names</span>(num.out), </span>
<span id="cb1490-3"><a href="WE_14.html#cb1490-3" aria-hidden="true" tabindex="-1"></a>    <span class="at">num.out=</span><span class="fu">as.vector</span>(num.out), <span class="at">num.tot=</span><span class="fu">as.vector</span>(num.tot))</span>
<span id="cb1490-4"><a href="WE_14.html#cb1490-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1490-5"><a href="WE_14.html#cb1490-5" aria-hidden="true" tabindex="-1"></a><span class="co"># read in the population variable data:</span></span>
<span id="cb1490-6"><a href="WE_14.html#cb1490-6" aria-hidden="true" tabindex="-1"></a>pop.vars <span class="ot">&lt;-</span> <span class="fu">read.csv</span>(<span class="fu">system.file</span>(<span class="st">&quot;extdata&quot;</span>, </span>
<span id="cb1490-7"><a href="WE_14.html#cb1490-7" aria-hidden="true" tabindex="-1"></a>  <span class="st">&quot;pulsatilla_population.csv&quot;</span>, <span class="at">package=</span><span class="st">&quot;LandGenCourse&quot;</span>))</span>
<span id="cb1490-8"><a href="WE_14.html#cb1490-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1490-9"><a href="WE_14.html#cb1490-9" aria-hidden="true" tabindex="-1"></a><span class="do">## a.d the population variables to our outside pollination data:</span></span>
<span id="cb1490-10"><a href="WE_14.html#cb1490-10" aria-hidden="true" tabindex="-1"></a>pop.df <span class="ot">&lt;-</span> <span class="fu">merge</span>(pop.df, pop.vars, <span class="at">by=</span><span class="st">&quot;Population&quot;</span>)</span>
<span id="cb1490-11"><a href="WE_14.html#cb1490-11" aria-hidden="true" tabindex="-1"></a>pop.df</span></code></pre></div>
<pre><code>##   Population num.out num.tot forest.50 forest.100 forest.250 forest.500
## 1        A03       2      13 0.3531695  0.5305529  0.8116160  0.8809752
## 2        A21      17      48 0.6036269  0.5291262  0.5581914  0.6131979
## 3        A25      22      72 0.3622251  0.2200647  0.2445019  0.2904836
## 4        A26      13      57 0.4948187  0.4838188  0.5212385  0.4778212
## 5        A41      11      57 0.6843467  0.8159172  0.9087349  0.7966062
## 6        A45      15      53 0.3824289  0.4297280  0.4747971  0.4027923
## 7       G05a       4      15 0.6152850  0.4102232  0.3046323  0.2119176
##   forest.1000 population.size
## 1   0.8723172              44
## 2   0.4553216              21
## 3   0.2106881              57
## 4   0.2314417              22
## 5   0.6717830              15
## 6   0.2988908              22
## 7   0.2810409              46</code></pre>
<p>Now we can run a model to see which variables best explain the proportion of immigrated pollen per population. Because we only have n=7 populations, we limit the models to a single explanatory variable.</p>
<p>Our response variable is a proportion, so we use a glm (generalized linear model) with binomial error distribution. This model does not include any random effects.</p>
<p>To automatize this, we first fit a ‘full’ model with all six potential predictors. We won’t interpret this model as it would be overfitted, but we’ll use it as input for the next step.</p>
<div class="sourceCode" id="cb1492"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1492-1"><a href="WE_14.html#cb1492-1" aria-hidden="true" tabindex="-1"></a><span class="co"># specify the model</span></span>
<span id="cb1492-2"><a href="WE_14.html#cb1492-2" aria-hidden="true" tabindex="-1"></a>mod2 <span class="ot">&lt;-</span> <span class="fu">glm</span>(<span class="fu">cbind</span>(num.out, num.tot<span class="sc">-</span>num.out) <span class="sc">~</span> forest<span class="fl">.50</span> <span class="sc">+</span> forest<span class="fl">.100</span> <span class="sc">+</span></span>
<span id="cb1492-3"><a href="WE_14.html#cb1492-3" aria-hidden="true" tabindex="-1"></a>              forest<span class="fl">.250</span> <span class="sc">+</span> forest<span class="fl">.500</span> <span class="sc">+</span> forest<span class="fl">.1000</span> <span class="sc">+</span> population.size,</span>
<span id="cb1492-4"><a href="WE_14.html#cb1492-4" aria-hidden="true" tabindex="-1"></a>              <span class="at">family =</span> <span class="fu">binomial</span>(<span class="at">link =</span> <span class="st">&quot;logit&quot;</span>), <span class="at">data =</span> pop.df, </span>
<span id="cb1492-5"><a href="WE_14.html#cb1492-5" aria-hidden="true" tabindex="-1"></a>              <span class="at">na.action =</span> <span class="st">&quot;na.fail&quot;</span>)</span></code></pre></div>
<p>We use the function <code>dredge</code> again to select the best model among those with only a single predictors (plus the intercept-only model). This is specified with the argument <code>m.lim=c(0,1)</code>.</p>
<div class="sourceCode" id="cb1493"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1493-1"><a href="WE_14.html#cb1493-1" aria-hidden="true" tabindex="-1"></a>MuMIn<span class="sc">::</span><span class="fu">dredge</span>(mod2,<span class="at">m.lim=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">1</span>))</span></code></pre></div>
<pre><code>## Fixed term is &quot;(Intercept)&quot;</code></pre>
<pre><code>## Global model call: glm(formula = cbind(num.out, num.tot - num.out) ~ forest.50 + 
##     forest.100 + forest.250 + forest.500 + forest.1000 + population.size, 
##     family = binomial(link = &quot;logit&quot;), data = pop.df, na.action = &quot;na.fail&quot;)
## ---
## Model selection table 
##      (Int) frs.100 frs.1000 frs.250  frs.50 frs.500 ppl.siz df  logLik AICc
## 1  -1.0120                                                   1 -16.061 34.9
## 5  -0.5516                  -0.8899                          2 -14.866 36.7
## 2  -0.5758 -0.9236                                           2 -15.118 37.2
## 17 -0.6049                                  -0.8182          2 -15.268 37.5
## 3  -0.7071          -0.8159                                  2 -15.318 37.6
## 33 -1.1830                                          0.00555  2 -15.807 38.6
## 9  -0.7084                          -0.6136                  2 -15.877 38.8
##    delta weight
## 1   0.00  0.392
## 5   1.81  0.158
## 2   2.31  0.123
## 17  2.62  0.106
## 3   2.71  0.101
## 33  3.69  0.062
## 9   3.83  0.058
## Models ranked by AICc(x)</code></pre>
<p>We see that the best model is the intercept-only (null) model, suggesting that none of the predictors fit the data very well. However, the next-best model includes ‘forest.250’ with a delta AICc of less than 2, and we take this model to be equally likely.</p>
<p>Note: Interestingly, in our original paper with found that forest.500 was the best predictor, followed by population size. This suggests that the choice of paternity analysis method really can make a difference.</p>
<p>Let’s plot the model including forest.250:</p>
<div class="sourceCode" id="cb1496"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1496-1"><a href="WE_14.html#cb1496-1" aria-hidden="true" tabindex="-1"></a>forest.<span class="fl">250.</span>mod <span class="ot">&lt;-</span> <span class="fu">glm</span>(<span class="fu">cbind</span>(num.out, num.tot<span class="sc">-</span>num.out) <span class="sc">~</span> forest<span class="fl">.250</span>,</span>
<span id="cb1496-2"><a href="WE_14.html#cb1496-2" aria-hidden="true" tabindex="-1"></a>                <span class="at">family=</span><span class="fu">binomial</span>(<span class="at">link=</span><span class="st">&quot;logit&quot;</span>), <span class="at">data=</span>pop.df)</span>
<span id="cb1496-3"><a href="WE_14.html#cb1496-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1496-4"><a href="WE_14.html#cb1496-4" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(pop.df, <span class="fu">aes</span>(<span class="at">x=</span>forest<span class="fl">.250</span>, <span class="at">y=</span>num.out<span class="sc">/</span>num.tot)) <span class="sc">+</span> <span class="fu">geom_point</span>() <span class="sc">+</span></span>
<span id="cb1496-5"><a href="WE_14.html#cb1496-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_line</span>(<span class="fu">aes</span>(<span class="at">x=</span>forest<span class="fl">.250</span>, <span class="at">y=</span><span class="fu">predict</span>(forest.<span class="fl">250.</span>mod, <span class="at">type=</span><span class="st">&quot;response&quot;</span>))) <span class="sc">+</span></span>
<span id="cb1496-6"><a href="WE_14.html#cb1496-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">xlab</span>(<span class="st">&quot;Proportion of forest at 250 m&quot;</span>) <span class="sc">+</span></span>
<span id="cb1496-7"><a href="WE_14.html#cb1496-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ylab</span>(<span class="st">&quot;Proportion of immigrant pollen&quot;</span>)</span></code></pre></div>
<p><img src="14-Week14_files/figure-html/plot_forest.250-1.png" width="672" /></p>
<p>We see that populations surrounded by more forest receive less outside pollen, although it is apparent that there is one outlier (population A21) with a high proportion of immigration pollen but intermediate levels of surrounding forest. This is likely also a product of our choice of paternity anlaysis - in our original paper (DiLeo et al. in press) we were able to assign a much higher proportion of fathers in popualtion A21, giving us lower pollen immigration rates.</p>
</div>
</div>
<div id="references-4" class="section level3 unnumbered hasAnchor">
<h3>6. References<a href="WE_14.html#references-4" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<ul>
<li><p>DiLeo MF, R Holderegger and HH Wagner (<em>in press</em>). Contemporary pollen flow as a multiscale process: evidence from the insect-pollinated herb <em>Pulsatilla vulgaris</em>. Journal of Ecology. <a href="https://doi.org/10.1111/1365-2745.12992" class="uri">https://doi.org/10.1111/1365-2745.12992</a></p></li>
<li><p>Dyer RJ, RD Westfall, VL Sork VL and PE Smouse (2004). Two-generation analysis of pollen flow across a landscape V: a stepwise approach for extracting factors contributing to pollen structure. Heredity 92: 204-211.</p></li>
<li><p>Dyer RJ, Chan DM, Gardiakos VA, Meadows CA (2012) Pollination graphs: quantifying pollen pool covariance networks and the influence of intervening landscape on genetic connectivity in the North American understory tree, Cornus florida L. Landscape ecology 27:239-51.</p></li>
<li><p>Smouse PE, RJ Dyer, RD Westfall and VL Sork (2001). Two‐generation analysis of pollen flow across a landscape. I. Male gamete heterogeneity among females. Evolution 55: 260–271.</p></li>
</ul>

</div>
</div>
<!-- </div> -->






            </section>

          </div>
        </div>
      </div>
<a href="Week14.html" class="navigation navigation-prev navigation-unique" aria-label="Previous page"><i class="fa fa-angle-left"></i></a>

    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": true,
"facebook": false,
"twitter": false,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": false
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": null,
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "section"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
