<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>13.1 Worked Example | Landscape Genetic Data Analysis with R</title>
  <meta name="description" content="This is a web-interface to the teaching materials for the lab course ‘Landscape Genetic Data Analysis with R’ associated with the distributed graduate course ‘DGS Landscape Genetics.’ The output format is bookdown::gitbook." />
  <meta name="generator" content="bookdown 0.25 and GitBook 2.6.7" />

  <meta property="og:title" content="13.1 Worked Example | Landscape Genetic Data Analysis with R" />
  <meta property="og:type" content="book" />
  
  <meta property="og:description" content="This is a web-interface to the teaching materials for the lab course ‘Landscape Genetic Data Analysis with R’ associated with the distributed graduate course ‘DGS Landscape Genetics.’ The output format is bookdown::gitbook." />
  <meta name="github-repo" content="hhwagner1/LandGenCourse_book" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="13.1 Worked Example | Landscape Genetic Data Analysis with R" />
  
  <meta name="twitter:description" content="This is a web-interface to the teaching materials for the lab course ‘Landscape Genetic Data Analysis with R’ associated with the distributed graduate course ‘DGS Landscape Genetics.’ The output format is bookdown::gitbook." />
  

<meta name="author" content="Editor: Helene Wagner (University of Toronto)" />


<meta name="date" content="2022-04-03" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="Week10.html"/>
<link rel="next" href="bonus-10a.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>
<script src="libs/kePrint-0.0.1/kePrint.js"></script>
<link href="libs/lightable-0.0.1/lightable.css" rel="stylesheet" />
<script src="libs/htmlwidgets-1.5.4/htmlwidgets.js"></script>
<script src="libs/d3-3.5.6/d3.min.js"></script>
<link href="libs/profvis-0.3.6.9000/profvis.css" rel="stylesheet" />
<script src="libs/profvis-0.3.6.9000/profvis.js"></script>
<script src="libs/profvis-0.3.6.9000/scroll.js"></script>
<link href="libs/highlight-6.2.0/textmate.css" rel="stylesheet" />
<script src="libs/highlight-6.2.0/highlight.js"></script>
<script src="libs/profvis-binding-0.3.7/profvis.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>


<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">Landscape Genetic Data Analysis with R</a></li>

<li class="divider"></li>
<li class="part"><span><b>Getting Started<span></span></b></span></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Introduction<span></span></a>
<ul>
<li class="chapter" data-level="1.1" data-path="how-to-use-this-book.html"><a href="how-to-use-this-book.html"><i class="fa fa-check"></i><b>1.1</b> How to use this Book<span></span></a>
<ul>
<li><a href="how-to-use-this-book.html#book-structure">1. Book Structure<span></span></a></li>
<li><a href="how-to-use-this-book.html#find-what-is-relevant-for-you">2. Find what is relevant for you<span></span></a></li>
<li><a href="how-to-use-this-book.html#course-r-package-landgencourse">3. Course R package ‘LandGenCourse’<span></span></a></li>
</ul></li>
<li class="chapter" data-level="1.2" data-path="list-of-packages.html"><a href="list-of-packages.html"><i class="fa fa-check"></i><b>1.2</b> List of R Packages by Vignette<span></span></a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="review-of-r-skills.html"><a href="review-of-r-skills.html"><i class="fa fa-check"></i><b>2</b> Review of R Skills<span></span></a>
<ul>
<li class="chapter" data-level="2.1" data-path="basic-r.html"><a href="basic-r.html"><i class="fa fa-check"></i><b>2.1</b> Basic R Programming<span></span></a>
<ul>
<li><a href="basic-r.html#overview">1. Overview<span></span></a></li>
<li><a href="basic-r.html#data-types">2. Data Types<span></span></a></li>
<li><a href="basic-r.html#data-containers">3. Data Containers<span></span></a></li>
<li><a href="basic-r.html#programming">4. Programming<span></span></a></li>
</ul></li>
<li class="chapter" data-level="2.2" data-path="r-graphics.html"><a href="r-graphics.html"><i class="fa fa-check"></i><b>2.2</b> R Graphics<span></span></a>
<ul>
<li><a href="r-graphics.html#overview-1">1. Overview<span></span></a></li>
<li><a href="r-graphics.html#univariate-plots">2. Univariate plots<span></span></a></li>
<li><a href="r-graphics.html#bivariate-plots">3. Bivariate plots<span></span></a></li>
<li><a href="r-graphics.html#advanced-plotting">4. Advanced plotting<span></span></a></li>
</ul></li>
<li class="chapter" data-level="2.3" data-path="resources.html"><a href="resources.html"><i class="fa fa-check"></i><b>2.3</b> Further R resources<span></span></a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="github-for-group-projects.html"><a href="github-for-group-projects.html"><i class="fa fa-check"></i><b>3</b> GitHub for Group Projects<span></span></a>
<ul>
<li class="chapter" data-level="3.1" data-path="github.html"><a href="github.html"><i class="fa fa-check"></i><b>3.1</b> Why use GitHub?<span></span></a>
<ul>
<li><a href="github.html#goals">1. Goals<span></span></a></li>
<li><a href="github.html#version">2. Watch course video: Version Control 101<span></span></a></li>
<li><a href="github.html#slides">3. Preview slides<span></span></a></li>
<li><a href="github.html#next-steps">4. Next steps<span></span></a></li>
</ul></li>
<li class="chapter" data-level="3.2" data-path="install.html"><a href="install.html"><i class="fa fa-check"></i><b>3.2</b> Installation<span></span></a>
<ul>
<li><a href="install.html#goals-1">1. Goals<span></span></a></li>
<li class="chapter" data-level="3.2.1" data-path="install.html"><a href="install.html#how-to-get-github-and-gitkraken-for-free"><i class="fa fa-check"></i><b>3.2.1</b> 2. How to get GitHub and GitKraken for free<span></span></a></li>
<li><a href="install.html#happygit">3. Happy Git with R<span></span></a></li>
<li><a href="install.html#gitvideos">4. Video tutorials: installation<span></span></a></li>
<li><a href="install.html#join">5. Joining your group’s project on GitHub<span></span></a></li>
<li><a href="install.html#repo">6. Setting up a group repo<span></span></a></li>
<li><a href="install.html#next-steps-1">7. Next steps<span></span></a></li>
</ul></li>
<li class="chapter" data-level="3.3" data-path="workflow.html"><a href="workflow.html"><i class="fa fa-check"></i><b>3.3</b> Workflows<span></span></a>
<ul>
<li><a href="workflow.html#goals-2">1. Goals<span></span></a></li>
<li><a href="workflow.html#step-by-step-tutorials">2. Step-by-step tutorials<span></span></a></li>
<li><a href="workflow.html#further-resources">3. Further resources<span></span></a></li>
</ul></li>
</ul></li>
<li class="part"><span><b>Basic Topics<span></span></b></span></li>
<li class="chapter" data-level="4" data-path="Week1.html"><a href="Week1.html"><i class="fa fa-check"></i><b>4</b> Week 1: Importing Genetic Data<span></span></a>
<ul>
<li class="chapter" data-level="4.1" data-path="video_1.html"><a href="video_1.html"><i class="fa fa-check"></i><b>4.1</b> View Course Video<span></span></a>
<ul>
<li><a href="video_1.html#embedded-video">1. Embedded Video<span></span></a></li>
<li><a href="video_1.html#preview-slides">2. Preview Slides<span></span></a></li>
</ul></li>
<li class="chapter" data-level="4.2" data-path="tutorial_1.html"><a href="tutorial_1.html"><i class="fa fa-check"></i><b>4.2</b> Interactive Tutorial 1<span></span></a>
<ul>
<li><a href="tutorial_1.html#list-of-r-commands-covered-this-week">1. List of R commands covered this week<span></span></a></li>
<li><a href="tutorial_1.html#general-instructions">2. General Instructions<span></span></a></li>
</ul></li>
<li class="chapter" data-level="4.3" data-path="WE_1.html"><a href="WE_1.html"><i class="fa fa-check"></i><b>4.3</b> Worked Example<span></span></a>
<ul>
<li><a href="WE_1.html#overview-of-worked-example">1. Overview of Worked Example<span></span></a></li>
<li><a href="WE_1.html#import-data-from-.csv-file">2. Import data from .csv file<span></span></a></li>
<li><a href="WE_1.html#create-genind-object">3. Create ‘genind’ Object<span></span></a></li>
<li><a href="WE_1.html#view-information-stored-in-genind-object">4. View information stored in ‘genind’ object<span></span></a></li>
<li><a href="WE_1.html#import-data-with-gstudio-package">5. Import data with ‘gstudio’ package<span></span></a></li>
<li><a href="WE_1.html#importing-snp-data">6. Importing SNP data<span></span></a></li>
</ul></li>
<li class="chapter" data-level="4.4" data-path="r-exercise-week-1.html"><a href="r-exercise-week-1.html"><i class="fa fa-check"></i><b>4.4</b> R exercise Week 1<span></span></a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="Week2.html"><a href="Week2.html"><i class="fa fa-check"></i><b>5</b> Week 2: Spatial Data<span></span></a>
<ul>
<li class="chapter" data-level="5.1" data-path="video_2.html"><a href="video_2.html"><i class="fa fa-check"></i><b>5.1</b> View Course Video<span></span></a>
<ul>
<li><a href="video_2.html#embedded-video-1">1. Embedded Video<span></span></a></li>
<li><a href="video_2.html#preview-slides-1">2. Preview Slides<span></span></a></li>
</ul></li>
<li class="chapter" data-level="5.2" data-path="tutorial_2.html"><a href="tutorial_2.html"><i class="fa fa-check"></i><b>5.2</b> Interactive Tutorial 2<span></span></a>
<ul>
<li><a href="tutorial_2.html#list-of-r-commands-covered-this-week-1">1. List of R commands covered this week<span></span></a></li>
<li><a href="tutorial_2.html#general-instructions-1">2. General Instructions<span></span></a></li>
</ul></li>
<li class="chapter" data-level="5.3" data-path="WE_2.html"><a href="WE_2.html"><i class="fa fa-check"></i><b>5.3</b> Worked Example<span></span></a>
<ul>
<li><a href="WE_2.html#overview-of-worked-example-1">1. Overview of Worked Example<span></span></a></li>
<li><a href="WE_2.html#import-site-data-from-.csv-file">2. Import site data from .csv file<span></span></a></li>
<li><a href="WE_2.html#display-raster-data-and-overlay-sampling-locations-extract-data">3. Display raster data and overlay sampling locations, extract data<span></span></a></li>
<li><a href="WE_2.html#calculate-landscape-metrics">4. Calculate landscape metrics<span></span></a></li>
<li><a href="WE_2.html#sample-landscape-metrics-within-buffer-around-sampling-locations">5. Sample landscape metrics within buffer around sampling locations<span></span></a></li>
</ul></li>
<li class="chapter" data-level="5.4" data-path="r-exercise-week-2.html"><a href="r-exercise-week-2.html"><i class="fa fa-check"></i><b>5.4</b> R Exercise Week 2<span></span></a></li>
<li class="chapter" data-level="5.5" data-path="bonus_2a.html"><a href="bonus_2a.html"><i class="fa fa-check"></i><b>5.5</b> Bonus: ‘sf’ and ‘terra’<span></span></a>
<ul>
<li><a href="bonus_2a.html#overview-2">1. Overview<span></span></a></li>
<li><a href="bonus_2a.html#convert-spatialpointsdataframe-to-sf-object">2. Convert ‘SpatialPointsDataFrame’ to ‘sf’ object<span></span></a></li>
<li><a href="bonus_2a.html#plot-variables-first-10-in-data-set">3. Plot variables (first 10 in data set)<span></span></a></li>
<li><a href="bonus_2a.html#export-site-data-as-esri-shapefile">4. Export site data as ESRI shapefile<span></span></a></li>
<li><a href="bonus_2a.html#use-terra-and-tmap-to-display-categorical-map-with-color-scheme">5. Use ‘terra’ and ‘tmap’ to display categorical map with color scheme<span></span></a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="6" data-path="Week3.html"><a href="Week3.html"><i class="fa fa-check"></i><b>6</b> Week 3: Genetic Diversity<span></span></a>
<ul>
<li class="chapter" data-level="6.1" data-path="video_3.html"><a href="video_3.html"><i class="fa fa-check"></i><b>6.1</b> View Course Video<span></span></a>
<ul>
<li><a href="video_3.html#embedded-video-2">1. Embedded Video<span></span></a></li>
<li><a href="video_3.html#video-part-1">Video, Part 1<span></span></a></li>
<li><a href="video_3.html#video-part-2">Video, Part 2<span></span></a></li>
<li><a href="video_3.html#preview-slides-2">Preview Slides<span></span></a></li>
</ul></li>
<li class="chapter" data-level="6.2" data-path="tutorial_3.html"><a href="tutorial_3.html"><i class="fa fa-check"></i><b>6.2</b> Interactive Tutorial 3<span></span></a>
<ul>
<li><a href="tutorial_3.html#list-of-r-commands-covered-this-week-2">1. List of R commands covered this week<span></span></a></li>
<li><a href="tutorial_3.html#general-instructions-2">2. General Instructions<span></span></a></li>
</ul></li>
<li class="chapter" data-level="6.3" data-path="WE_3.html"><a href="WE_3.html"><i class="fa fa-check"></i><b>6.3</b> Worked Example<span></span></a>
<ul>
<li><a href="WE_3.html#overview-of-worked-example-2">1. Overview of Worked Example<span></span></a></li>
<li><a href="WE_3.html#basic-checking-of-markers-and-populations">2. Basic checking of markers and populations<span></span></a></li>
<li><a href="WE_3.html#assess-genetic-diversity">3. Assess genetic diversity<span></span></a></li>
<li><a href="WE_3.html#aggregate-genetic-data-at-population-level-allele-frequencies">4. Aggregate genetic data at population level (allele frequencies)<span></span></a></li>
</ul></li>
<li class="chapter" data-level="6.4" data-path="r-exercise-week-3.html"><a href="r-exercise-week-3.html"><i class="fa fa-check"></i><b>6.4</b> R Exercise Week 3<span></span></a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="Week4.html"><a href="Week4.html"><i class="fa fa-check"></i><b>7</b> Week 4: Metapopulations<span></span></a>
<ul>
<li class="chapter" data-level="7.1" data-path="video_4.html"><a href="video_4.html"><i class="fa fa-check"></i><b>7.1</b> View Course Video<span></span></a>
<ul>
<li><a href="video_4.html#embedded-video-3">1. Embedded Video<span></span></a></li>
<li><a href="video_4.html#video-part-1-1">Video, Part 1<span></span></a></li>
<li><a href="video_4.html#video-part-2-1">Video, Part 2<span></span></a></li>
<li><a href="video_4.html#preview-slides-3">Preview Slides<span></span></a></li>
</ul></li>
<li class="chapter" data-level="7.2" data-path="tutorial_4.html"><a href="tutorial_4.html"><i class="fa fa-check"></i><b>7.2</b> Interactive Tutorial 4<span></span></a>
<ul>
<li><a href="tutorial_4.html#list-of-r-commands-covered-this-week-3">1. List of R commands covered this week<span></span></a></li>
<li><a href="tutorial_4.html#general-instructions-3">2. General Instructions<span></span></a></li>
</ul></li>
<li class="chapter" data-level="7.3" data-path="WE_4.html"><a href="WE_4.html"><i class="fa fa-check"></i><b>7.3</b> Worked Example<span></span></a>
<ul>
<li><a href="WE_4.html#overview-of-worked-example-3">1. Overview of Worked Example<span></span></a></li>
<li><a href="WE_4.html#spatial-distribution-of-genetic-structure">2. Spatial distribution of genetic structure<span></span></a></li>
<li><a href="WE_4.html#what-determines-genetic-differentiation-among-sites">3. What determines genetic differentiation among sites?<span></span></a></li>
<li><a href="WE_4.html#what-determines-genetic-diversity">4. What determines genetic diversity?<span></span></a></li>
<li><a href="WE_4.html#are-genetic-differentiation-and-diversity-related">5. Are genetic differentiation and diversity related?<span></span></a></li>
<li><a href="WE_4.html#effect-of-recent-extinction-events">6. Effect of recent extinction events<span></span></a></li>
</ul></li>
<li class="chapter" data-level="7.4" data-path="r-exercise-week-4.html"><a href="r-exercise-week-4.html"><i class="fa fa-check"></i><b>7.4</b> R Exercise Week 4<span></span></a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="Week5.html"><a href="Week5.html"><i class="fa fa-check"></i><b>8</b> Week 5: Spatial Statistics<span></span></a>
<ul>
<li class="chapter" data-level="8.1" data-path="video_5.html"><a href="video_5.html"><i class="fa fa-check"></i><b>8.1</b> View Course Video<span></span></a>
<ul>
<li><a href="video_5.html#embedded-video-4">1. Embedded Video<span></span></a></li>
<li><a href="video_5.html#preview-slides-4">Preview Slides<span></span></a></li>
</ul></li>
<li class="chapter" data-level="8.2" data-path="tutorial_5.html"><a href="tutorial_5.html"><i class="fa fa-check"></i><b>8.2</b> Interactive Tutorial 5<span></span></a>
<ul>
<li><a href="tutorial_5.html#list-of-r-commands-covered-this-week-4">1. List of R commands covered this week<span></span></a></li>
<li><a href="tutorial_5.html#general-instructions-4">2. General Instructions<span></span></a></li>
</ul></li>
<li class="chapter" data-level="8.3" data-path="WE_5.html"><a href="WE_5.html"><i class="fa fa-check"></i><b>8.3</b> Worked Example<span></span></a>
<ul>
<li><a href="WE_5.html#overview-of-worked-example-4">1. Overview of Worked Example<span></span></a></li>
<li><a href="WE_5.html#data-import-and-manipulation">2. Data import and manipulation<span></span></a></li>
<li><a href="WE_5.html#calculate-genetic-distances">3. Calculate genetic distances<span></span></a></li>
<li><a href="WE_5.html#perform-a-mantel-test-to-test-for-ibd">4. Perform a Mantel test to test for IBD<span></span></a></li>
<li><a href="WE_5.html#create-mantel-correlogram-for-genetic-data">5. Create Mantel correlogram for genetic data<span></span></a></li>
<li><a href="WE_5.html#specify-spatial-weights-and-calculate-morans-i">6. Specify spatial weights and calculate Moran’s I<span></span></a></li>
</ul></li>
<li class="chapter" data-level="8.4" data-path="r-exercise-week-5.html"><a href="r-exercise-week-5.html"><i class="fa fa-check"></i><b>8.4</b> R Exercise Week 5<span></span></a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="Week6.html"><a href="Week6.html"><i class="fa fa-check"></i><b>9</b> Week 6: Quantitative Genetics<span></span></a>
<ul>
<li class="chapter" data-level="9.1" data-path="video_6.html"><a href="video_6.html"><i class="fa fa-check"></i><b>9.1</b> View Course Video<span></span></a>
<ul>
<li><a href="video_6.html#embedded-video-5">1. Embedded Video<span></span></a></li>
<li><a href="video_6.html#video-part-1-2">Video, Part 1<span></span></a></li>
<li><a href="video_6.html#video-part-2-2">Video, Part 2<span></span></a></li>
<li><a href="video_6.html#preview-slides-5">Preview Slides<span></span></a></li>
</ul></li>
<li class="chapter" data-level="9.2" data-path="tutorial_6.html"><a href="tutorial_6.html"><i class="fa fa-check"></i><b>9.2</b> Interactive Tutorial 6<span></span></a>
<ul>
<li><a href="tutorial_6.html#list-of-r-commands-covered-this-week-5">1. List of R commands covered this week<span></span></a></li>
<li><a href="tutorial_6.html#general-instructions-5">2. General Instructions<span></span></a></li>
</ul></li>
<li class="chapter" data-level="9.3" data-path="WE_6.html"><a href="WE_6.html"><i class="fa fa-check"></i><b>9.3</b> Worked Example<span></span></a>
<ul>
<li><a href="WE_6.html#overview-of-worked-example-5">1. Overview of Worked Example<span></span></a></li>
<li><a href="WE_6.html#estimate-genetic-and-non-genetic-variance-components-from-a-common-garden-experiment">2. Estimate genetic and non-genetic variance components from a common garden experiment<span></span></a></li>
<li><a href="WE_6.html#estimate-trait-heritability">3. Estimate trait heritability<span></span></a></li>
<li><a href="WE_6.html#compare-q_st-to-f_st-measured-from-snp-data">4. Compare <span class="math inline">\(Q_{st}\)</span> to <span class="math inline">\(F_{st}\)</span> measured from SNP data<span></span></a></li>
<li><a href="WE_6.html#assess-correlation-between-trait-and-environment">5. Assess correlation between trait and environment<span></span></a></li>
</ul></li>
<li class="chapter" data-level="9.4" data-path="r-exercise-week-6.html"><a href="r-exercise-week-6.html"><i class="fa fa-check"></i><b>9.4</b> R Exercise Week 6<span></span></a></li>
</ul></li>
<li class="chapter" data-level="10" data-path="Week7.html"><a href="Week7.html"><i class="fa fa-check"></i><b>10</b> Week 7: Spatial Linear Models<span></span></a>
<ul>
<li class="chapter" data-level="10.1" data-path="video_7.html"><a href="video_7.html"><i class="fa fa-check"></i><b>10.1</b> View Course Video<span></span></a>
<ul>
<li><a href="video_7.html#embedded-video-6">1. Embedded Video<span></span></a></li>
<li><a href="video_7.html#video-part-1-3">Video, Part 1<span></span></a></li>
<li><a href="video_7.html#video-part-2-3">Video, Part 2<span></span></a></li>
<li><a href="video_7.html#preview-slides-6">Preview Slides<span></span></a></li>
</ul></li>
<li class="chapter" data-level="10.2" data-path="tutorial_7.html"><a href="tutorial_7.html"><i class="fa fa-check"></i><b>10.2</b> Interactive Tutorial 7<span></span></a>
<ul>
<li><a href="tutorial_7.html#list-of-r-commands-covered-this-week-6">1. List of R commands covered this week<span></span></a></li>
<li><a href="tutorial_7.html#general-instructions-6">2. General Instructions<span></span></a></li>
</ul></li>
<li class="chapter" data-level="10.3" data-path="WE_7.html"><a href="WE_7.html"><i class="fa fa-check"></i><b>10.3</b> Worked Example<span></span></a>
<ul>
<li><a href="WE_7.html#overview-of-worked-example-6">1. Overview of Worked Example<span></span></a></li>
<li><a href="WE_7.html#explore-data-set">2. Explore data set<span></span></a></li>
<li><a href="WE_7.html#test-regression-residuals-for-spatial-autocorrelation">3. Test regression residuals for spatial autocorrelation<span></span></a></li>
<li><a href="WE_7.html#fit-models-with-spatially-correlated-error-gls-with-package-nlme">4. Fit models with spatially correlated error (GLS) with package ‘nlme’<span></span></a></li>
<li><a href="WE_7.html#fit-spatial-simultaneous-autoregressive-error-models-sar">5. Fit spatial simultaneous autoregressive error models (SAR)<span></span></a></li>
<li><a href="WE_7.html#spatial-filtering-with-mem-using-package-spmoran">6. Spatial filtering with MEM using package ‘spmoran’<span></span></a></li>
<li><a href="WE_7.html#fit-spatially-varying-coefficients-model-with-package-spmoran">7. Fit spatially varying coefficients model with package ‘spmoran’<span></span></a></li>
<li><a href="WE_7.html#model-with-patchsize-and-ibr">Model with PatchSize and IBR<span></span></a></li>
<li><a href="WE_7.html#model-with-ibr-only">Model with IBR only<span></span></a></li>
<li><a href="WE_7.html#conclusions">8. Conclusions<span></span></a></li>
</ul></li>
<li class="chapter" data-level="10.4" data-path="r-exercise-week-7.html"><a href="r-exercise-week-7.html"><i class="fa fa-check"></i><b>10.4</b> R Exercise Week 7<span></span></a></li>
<li class="chapter" data-level="10.5" data-path="bonus_7a.html"><a href="bonus_7a.html"><i class="fa fa-check"></i><b>10.5</b> Bonus: Si index<span></span></a>
<ul>
<li><a href="bonus_7a.html#overview-of-bonus-materials">1. Overview of Bonus Materials<span></span></a></li>
<li><a href="bonus_7a.html#import-ecological-distance-matrices">2. Import ecological distance matrices<span></span></a></li>
<li><a href="bonus_7a.html#optimize-the-scaling-parameter-alpha">3. Optimize the scaling parameter alpha<span></span></a></li>
<li><a href="bonus_7a.html#calculate-hanskis-index-si-with-source-patch-parameters">4. Calculate Hanski’s index Si with source patch parameters<span></span></a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="11" data-path="Week8.html"><a href="Week8.html"><i class="fa fa-check"></i><b>11</b> Week 8: Simulation Experiments<span></span></a>
<ul>
<li class="chapter" data-level="11.1" data-path="video_8.html"><a href="video_8.html"><i class="fa fa-check"></i><b>11.1</b> View Course Video<span></span></a>
<ul>
<li><a href="video_8.html#embedded-video-7">1. Embedded Video<span></span></a></li>
<li><a href="video_8.html#video-part-1-4">Video, Part 1<span></span></a></li>
<li><a href="video_8.html#video-part-2-4">Video, Part 2<span></span></a></li>
<li><a href="video_8.html#preview-slides-7">Preview Slides<span></span></a></li>
</ul></li>
<li class="chapter" data-level="11.2" data-path="tutorial_8.html"><a href="tutorial_8.html"><i class="fa fa-check"></i><b>11.2</b> Interactive Tutorial 8<span></span></a>
<ul>
<li><a href="tutorial_8.html#list-of-r-commands-covered-this-week-7">1. List of R commands covered this week<span></span></a></li>
<li><a href="tutorial_8.html#general-instructions-7">2. General Instructions<span></span></a></li>
</ul></li>
<li class="chapter" data-level="11.3" data-path="WE_8.html"><a href="WE_8.html"><i class="fa fa-check"></i><b>11.3</b> Worked Example<span></span></a>
<ul>
<li><a href="WE_8.html#overview-of-worked-example-7">1. Overview of Worked Example<span></span></a></li>
<li><a href="WE_8.html#initialize-a-landscape">2. Initialize a landscape<span></span></a></li>
<li><a href="WE_8.html#define-simulation-parameters">3. Define simulation parameters<span></span></a></li>
<li><a href="WE_8.html#run-simulations-and-analyze-results">4. Run simulations and analyze results<span></span></a></li>
<li><a href="WE_8.html#run-simulator-using-a-previously-defined-parameter-set">5. Run simulator using a previously defined parameter set<span></span></a></li>
</ul></li>
<li class="chapter" data-level="11.4" data-path="r-exercise-week-8.html"><a href="r-exercise-week-8.html"><i class="fa fa-check"></i><b>11.4</b> R Exercise Week 8<span></span></a></li>
<li class="chapter" data-level="11.5" data-path="bonus_8a.html"><a href="bonus_8a.html"><i class="fa fa-check"></i><b>11.5</b> Bonus: Efficient R<span></span></a>
<ul>
<li><a href="bonus_8a.html#overview-of-bonus-material">1. Overview of Bonus Material<span></span></a></li>
<li><a href="bonus_8a.html#navigating-the-file-system">2. Navigating the file system<span></span></a></li>
<li><a href="bonus_8a.html#benchmarking-file-import-and-export-options">3. Benchmarking file import and export options<span></span></a></li>
<li><a href="bonus_8a.html#profiling-your-code">4. Profiling your code<span></span></a></li>
<li><a href="bonus_8a.html#creating-and-executing-a-bash-r-script">5. Creating and executing a Bash R script<span></span></a></li>
<li><a href="bonus_8a.html#parallelizing-code">6. Parallelizing code<span></span></a></li>
</ul></li>
</ul></li>
<li class="part"><span><b>Advanced Topics<span></span></b></span></li>
<li class="chapter" data-level="12" data-path="Week9.html"><a href="Week9.html"><i class="fa fa-check"></i><b>12</b> Week 9: Population Structure<span></span></a>
<ul>
<li class="chapter" data-level="12.1" data-path="WE_9.html"><a href="WE_9.html"><i class="fa fa-check"></i><b>12.1</b> Worked Example<span></span></a>
<ul>
<li><a href="WE_9.html#overview-of-worked-example-8">1. Overview of Worked Example<span></span></a></li>
<li><a href="WE_9.html#simulated-data-2-island-model">2. Simulated data: 2-island model<span></span></a></li>
<li><a href="WE_9.html#simulated-data-2-island-model-with-admixture">3. Simulated data: 2-island model with admixture<span></span></a></li>
<li><a href="WE_9.html#empirical-data-threespine-sticklebacks">4. Empirical data: Threespine sticklebacks<span></span></a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="13" data-path="Week10.html"><a href="Week10.html"><i class="fa fa-check"></i><b>13</b> Week 10: Landscape Resistance<span></span></a>
<ul>
<li class="chapter" data-level="13.1" data-path="WE_10.html"><a href="WE_10.html"><i class="fa fa-check"></i><b>13.1</b> Worked Example<span></span></a>
<ul>
<li><a href="WE_10.html#overview-of-worked-example-9">1. Overview of Worked Example<span></span></a></li>
<li><a href="WE_10.html#explore-the-data-set">2. Explore the data set<span></span></a></li>
<li><a href="WE_10.html#setting-cost-values-and-calculating-conductance">3. Setting cost values and calculating conductance<span></span></a></li>
<li><a href="WE_10.html#convert-conductance-into-effective-distance">4. Convert conductance into effective distance<span></span></a></li>
<li><a href="WE_10.html#create-cost-distance-matrices">5. Create cost-distance matrices<span></span></a></li>
<li><a href="WE_10.html#how-does-changing-resolution-affect-these-metrics">6. How does changing resolution affect these metrics?<span></span></a></li>
<li><a href="WE_10.html#references">7. References<span></span></a></li>
</ul></li>
<li class="chapter" data-level="13.2" data-path="bonus-10a.html"><a href="bonus-10a.html"><i class="fa fa-check"></i><b>13.2</b> Bonus: ‘radish’ tutorial<span></span></a>
<ul>
<li><a href="bonus-10a.html#overview-of-bonus-materials-1">1. Overview of Bonus Materials<span></span></a></li>
<li><a href="bonus-10a.html#import-data">2. Import data<span></span></a></li>
<li><a href="bonus-10a.html#basic-model-fitting">3. Basic model fitting<span></span></a></li>
<li><a href="bonus-10a.html#advanced-modeling-topics">4. Advanced modeling topics<span></span></a></li>
<li><a href="bonus-10a.html#summary">5. Summary<span></span></a></li>
<li><a href="bonus-10a.html#references-1">6. References<span></span></a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="14" data-path="Week11.html"><a href="Week11.html"><i class="fa fa-check"></i><b>14</b> Week 11: Detecting Adaptation<span></span></a>
<ul>
<li class="chapter" data-level="14.1" data-path="WE_11.html"><a href="WE_11.html"><i class="fa fa-check"></i><b>14.1</b> Worked Example<span></span></a>
<ul>
<li><a href="WE_11.html#overview-of-worked-example-10">1. Overview of Worked Example<span></span></a></li>
<li><a href="WE_11.html#import-and-prepare-the-data">2. Import and prepare the data<span></span></a></li>
<li><a href="WE_11.html#latent-factor-mixed-models-lfmm-a-univariate-gea">3. Latent Factor Mixed Models (LFMM): a univariate GEA<span></span></a></li>
<li><a href="WE_11.html#redundancy-analysis-rda-a-multivariate-gea">4. Redundancy Analysis (RDA): a multivariate GEA<span></span></a></li>
<li><a href="WE_11.html#compare-lfmm-and-rda-candidates">5. Compare LFMM and RDA candidates<span></span></a></li>
<li><a href="WE_11.html#a-quick-note-on-controlling-for-population-structure-in-rda">6. A quick note on controlling for population structure in RDA<span></span></a></li>
<li><a href="WE_11.html#references-2">7. References<span></span></a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="15" data-path="Week12.html"><a href="Week12.html"><i class="fa fa-check"></i><b>15</b> Week 12: Model Selection<span></span></a>
<ul>
<li class="chapter" data-level="15.1" data-path="WE_12.html"><a href="WE_12.html"><i class="fa fa-check"></i><b>15.1</b> Worked Example<span></span></a>
<ul>
<li><a href="WE_12.html#overview-of-worked-example-11">1. Overview of Worked Example<span></span></a></li>
<li><a href="WE_12.html#fitting-candidate-models">2. Fitting candidate models<span></span></a></li>
<li><a href="WE_12.html#your-turn-to-test-additional-hypotheses">3. Your turn to test additional hypotheses!<span></span></a></li>
<li><a href="WE_12.html#nested-model-nmlpe-for-hierarchical-sampling-designs">4. Nested model (NMLPE) for hierarchical sampling designs<span></span></a></li>
<li><a href="WE_12.html#references-3">5. References<span></span></a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="16" data-path="Week13.html"><a href="Week13.html"><i class="fa fa-check"></i><b>16</b> Week 13: Gravity Models<span></span></a>
<ul>
<li class="chapter" data-level="16.1" data-path="WE_13.html"><a href="WE_13.html"><i class="fa fa-check"></i><b>16.1</b> Worked Example<span></span></a>
<ul>
<li><a href="WE_13.html#overview-of-worked-example-12">1. Overview of Worked Example<span></span></a></li>
<li><a href="WE_13.html#setup">2. Setup<span></span></a></li>
<li><a href="WE_13.html#wetland-complex-data-preparation">3. Wetland complex data preparation<span></span></a></li>
<li><a href="WE_13.html#wetland-field-data-preparation">4. Wetland field-data preparation<span></span></a></li>
<li><a href="WE_13.html#saturated-graph">5. Saturated Graph<span></span></a></li>
<li><a href="WE_13.html#spatial-model-data-prepration">6. Spatial model data prepration<span></span></a></li>
<li><a href="WE_13.html#gravity-model">7. Gravity model<span></span></a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="17" data-path="Week14.html"><a href="Week14.html"><i class="fa fa-check"></i><b>17</b> Week 14: Contemporary Gene Flow<span></span></a>
<ul>
<li class="chapter" data-level="17.1" data-path="WE_14.html"><a href="WE_14.html"><i class="fa fa-check"></i><b>17.1</b> Worked Example<span></span></a>
<ul>
<li><a href="WE_14.html#overview-3">1. Overview<span></span></a></li>
<li><a href="WE_14.html#import-genotypes">2. Import Genotypes<span></span></a></li>
<li><a href="WE_14.html#pollen-pool-genetic-structure">3. Pollen pool genetic structure<span></span></a></li>
<li><a href="WE_14.html#paternity-analysis">4. Paternity analysis<span></span></a></li>
<li><a href="WE_14.html#linking-paternity-to-ecological-variables">5. Linking paternity to ecological variables<span></span></a></li>
<li><a href="WE_14.html#references-4">6. References<span></span></a></li>
</ul></li>
</ul></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Landscape Genetic Data Analysis with R</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="WE_10" class="section level2 hasAnchor" number="13.1">
<h2><span class="header-section-number">13.1</span> Worked Example<a href="WE_10.html#WE_10" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Melanie Murphy, Jeff Bowman, Robby Marrotte and Helene Wagner</p>
<div id="overview-of-worked-example-9" class="section level3 unnumbered hasAnchor">
<h3>1. Overview of Worked Example<a href="WE_10.html#overview-of-worked-example-9" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div id="a.-goals-11" class="section level4 unnumbered hasAnchor">
<h4>a. Goals<a href="WE_10.html#a.-goals-11" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>This worked example shows how to:</p>
<ul>
<li>Convert spatial data into weights (“costs”) based on different weighting approaches.</li>
<li>Calculate landscape resistance using least-cost and commute time (analogous to circuit theory) approaches.</li>
</ul>
<p><strong>Bonus Materials</strong>:</p>
<p>The Week 10 bonus vignette is a tutorial for the R package <code>radish</code>, which provides a user-friendly interface for optimizing landscape resistance models.</p>
</div>
<div id="b.-data-set-9" class="section level4 unnumbered hasAnchor">
<h4>b. Data set<a href="WE_10.html#b.-data-set-9" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>This code builds on the data from the ‘GeNetIt’ package by Jeff Evans and Melanie Murphy that we’ve used already in the Week 2 Worked Example. We will use spatial data from central Idaho for Columbia spotted frogs (Murphy et al. 2010).</p>
<p><strong>Raster data</strong>: A raster is essentially a georeferenced matrix (i.e., a matrix with geographical co-ordinates). Here we’ll analyze raster data that are available in the ‘GeNetIt’ packages in the form of a SpatialPixelsDataFrame called ‘rasters’ (for meta data, type: ?GeNetIt::rasters). These data include classified Landsat land-cover data (NLCD), spline-based climate predictions (Rehfeldt et al. 2006), and topographically derived variables (Moore et al. 1993, Evans 1972). Landscape variables were selected based on knowledge of species’ ecology and previous research (Murphy et al. 2010, Pilliod et al. 2002, Funk et al. 2005):</p>
<ul>
<li>cti: Compound Topographic Index (“wetness”)</li>
<li>err27: Elevation Relief Ratio</li>
<li>ffp: Frost Free Period</li>
<li>gsp: Growing Season Precipitation</li>
<li>hli: Heat Load Index</li>
<li>nlcd: USGS Landcover (categorical map)</li>
</ul>
<p><strong>Sampling locations</strong>: We will model landscape resistance between the ponds from which Colombia spotted frogs were sampled. The spatial coordinates and site data are available in the ‘GeNEtIt’ package in the form of a SpatialPointsDataFrame ‘ralu.site.’ The slot <code>@coords</code> contains UTM coordinates (zone 11), and the slot <code>@data</code> contains 17 site variables for 31 sites. For meta data type: ?GeNetIt::ralu.site.</p>
</div>
<div id="c.-required-r-packages-9" class="section level4 unnumbered hasAnchor">
<h4>c. Required R packages<a href="WE_10.html#c.-required-r-packages-9" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>Install some packages needed for this worked example.</p>
<div class="sourceCode" id="cb1105"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1105-1"><a href="WE_10.html#cb1105-1" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span>(<span class="sc">!</span><span class="fu">requireNamespace</span>(<span class="st">&quot;GeNetIt&quot;</span>, <span class="at">quietly =</span> <span class="cn">TRUE</span>)) remotes<span class="sc">::</span><span class="fu">install_github</span>(<span class="st">&quot;jeffreyevans/GeNetIt&quot;</span>)</span></code></pre></div>
<div class="sourceCode" id="cb1106"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1106-1"><a href="WE_10.html#cb1106-1" aria-hidden="true" tabindex="-1"></a><span class="fu">require</span>(LandGenCourse)</span>
<span id="cb1106-2"><a href="WE_10.html#cb1106-2" aria-hidden="true" tabindex="-1"></a><span class="co">#require(GeNetIt)</span></span>
<span id="cb1106-3"><a href="WE_10.html#cb1106-3" aria-hidden="true" tabindex="-1"></a><span class="co">#require(raster)</span></span>
<span id="cb1106-4"><a href="WE_10.html#cb1106-4" aria-hidden="true" tabindex="-1"></a><span class="co">#require(gdistance)</span></span></code></pre></div>
</div>
<div id="d.-import-rasters-gsp-err-cti-and-ffp" class="section level4 unnumbered hasAnchor">
<h4>d. Import rasters (gsp, err, cti, and ffp)<a href="WE_10.html#d.-import-rasters-gsp-err-cti-and-ffp" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>Here we load the raster data as a SpatialPixelsDataFrame and convert them in to a raster stack.</p>
<div class="sourceCode" id="cb1107"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1107-1"><a href="WE_10.html#cb1107-1" aria-hidden="true" tabindex="-1"></a><span class="fu">data</span>(rasters, <span class="at">package=</span><span class="st">&quot;GeNetIt&quot;</span>)</span>
<span id="cb1107-2"><a href="WE_10.html#cb1107-2" aria-hidden="true" tabindex="-1"></a>RasterMaps <span class="ot">&lt;-</span> raster<span class="sc">::</span><span class="fu">stack</span>(rasters)</span></code></pre></div>
</div>
<div id="e.-import-site-data-with-sampling-locations" class="section level4 unnumbered hasAnchor">
<h4>e. Import site data with sampling locations<a href="WE_10.html#e.-import-site-data-with-sampling-locations" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>In addition to spatial data, sample locations are also needed. Sample locations are located in the data folder (RALU_UTM.csv). Read in site locations (wetlands with Columbia spotted frogs).</p>
<div class="sourceCode" id="cb1108"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1108-1"><a href="WE_10.html#cb1108-1" aria-hidden="true" tabindex="-1"></a><span class="fu">data</span>(ralu.site, <span class="at">package=</span><span class="st">&quot;GeNetIt&quot;</span>)</span>
<span id="cb1108-2"><a href="WE_10.html#cb1108-2" aria-hidden="true" tabindex="-1"></a>sites <span class="ot">&lt;-</span> ralu.site</span></code></pre></div>
<p><strong>Question 1</strong>: What are UTMs and why might it be important to work in UTMs (as opposed to latitude and longitude)?</p>
</div>
</div>
<div id="explore-the-data-set" class="section level3 unnumbered hasAnchor">
<h3>2. Explore the data set<a href="WE_10.html#explore-the-data-set" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Explore these rasters by plotting them. As you go through this exercise, use the plot function at each step to make sure your outputs “make sense.”</p>
<div id="a.-plot-all-rasters" class="section level4 unnumbered hasAnchor">
<h4>a. Plot all rasters<a href="WE_10.html#a.-plot-all-rasters" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<div class="sourceCode" id="cb1109"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1109-1"><a href="WE_10.html#cb1109-1" aria-hidden="true" tabindex="-1"></a>raster<span class="sc">::</span><span class="fu">plot</span>(RasterMaps)</span></code></pre></div>
<p><img src="10-Week10_files/figure-html/unnamed-chunk-9-1.png" width="768" /></p>
</div>
<div id="b.-plot-spatial-points-over-ffp-raster" class="section level4 unnumbered hasAnchor">
<h4>b. Plot spatial points over ‘ffp’ raster<a href="WE_10.html#b.-plot-spatial-points-over-ffp-raster" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>These are the locations of the ponds where Columbia spotted frogs were sampled within the study area.</p>
<div class="sourceCode" id="cb1110"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1110-1"><a href="WE_10.html#cb1110-1" aria-hidden="true" tabindex="-1"></a><span class="fu">par</span>(<span class="at">mar=</span><span class="fu">c</span>(<span class="dv">2</span>,<span class="dv">2</span>,<span class="dv">1</span>,<span class="dv">1</span>))</span>
<span id="cb1110-2"><a href="WE_10.html#cb1110-2" aria-hidden="true" tabindex="-1"></a>raster<span class="sc">::</span><span class="fu">plot</span>(RasterMaps<span class="sc">$</span>ffp)</span>
<span id="cb1110-3"><a href="WE_10.html#cb1110-3" aria-hidden="true" tabindex="-1"></a><span class="fu">points</span>(sites, <span class="at">pch=</span><span class="dv">3</span>)</span></code></pre></div>
<p><img src="10-Week10_files/figure-html/unnamed-chunk-10-1.png" width="672" /></p>
</div>
</div>
<div id="setting-cost-values-and-calculating-conductance" class="section level3 unnumbered hasAnchor">
<h3>3. Setting cost values and calculating conductance<a href="WE_10.html#setting-cost-values-and-calculating-conductance" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div id="a.-resistance-vs.-conductance-values" class="section level4 unnumbered hasAnchor">
<h4>a. Resistance vs. conductance values<a href="WE_10.html#a.-resistance-vs.-conductance-values" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>The next step involves creating cost categories from the continuous data. For the purposes of this exercise, a single “landscape resistance” cost surface will be created, based on merging the 4 variables, and employing expert opinion.</p>
<p>Rescaling the grids will take some processor time (depending on your computer).</p>
<p>First, use a relative ranking of the costs of the landscape variables based on expert opinion. Higher rank indicates more resistance as follows.</p>
<p>Landscape resistances:</p>
<pre><code>1) err27, 2) ffp, 3) gsp, 4) cti</code></pre>
<p>Important considerations:</p>
<ol style="list-style-type: decimal">
<li>The transition matrix in package ‘gdistance’ is based on conductance and not resistance.</li>
<li>If we are going to create a single landscape resistance (i.e., add the costs together to create one synthetic landscape variable), costs need to represent relative importance of the variables.</li>
</ol>
<p>Keep in mind there are a variety of approaches for creating landscape resistance values. This exercise implements one simplistic approach.</p>
<p>To create relative conductance values, reverse the rank order of the resistance values:</p>
<pre><code>1) cti, 2) gsp, 3) ffp, 4) err27 </code></pre>
</div>
<div id="b.-as-needed-get-all-rasters-at-the-same-scale" class="section level4 unnumbered hasAnchor">
<h4>b. As needed: Get all rasters at the same scale<a href="WE_10.html#b.-as-needed-get-all-rasters-at-the-same-scale" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>Note: this step is not needed here as all rasters do already have the same resolution.</p>
<p>Originally, the topographic variables were calculated off a 10 m DEM. The climate variables are at a 30 m resolution. In order to calculate costs, all of the rasters need to have exactly the same resolution, dimensions, and co-ordinate locations (i.e., the cells need to match up perfectly. The method of resampling (bilinear vs nearest-neighbor) depends on the type of data. Nearest-neighbor is for categorical data whereas bilinear interpolation is for continuous data.</p>
<p>Here is some sample code to show how you would resample a raster ‘cti’ (assumed to have 10 m resolution) to the dimensions of a raster ‘gsp’ (with 30 m resolution):</p>
<div class="sourceCode" id="cb1113"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1113-1"><a href="WE_10.html#cb1113-1" aria-hidden="true" tabindex="-1"></a><span class="co">#cti &lt;- raster::resample(cti, gsp, method= &quot;bilinear&quot;)</span></span></code></pre></div>
</div>
<div id="c.-calculate-conductance-values" class="section level4 unnumbered hasAnchor">
<h4>c. Calculate conductance values<a href="WE_10.html#c.-calculate-conductance-values" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>Functions (e.g., 1/err, ffp/5) rescale the raw data from each raster to represent costs that correspond to perceived differences among variables arising from expert opinion. A wide variety of functions could be used to achieve a variety of cost structures in order to test alternative hypotheses.
The err function is inverted compared to the other variables because the raw err data have low values relative to the other variables (&lt; 1).</p>
<p>To use the below tools, conductance values must be calculated. First, look at the range of the variable in a given raster. Then apply a function to get the desired
relative conductance values. Make sure to use the plot function and visually inspect your conductance surfaces.</p>
<div class="sourceCode" id="cb1114"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1114-1"><a href="WE_10.html#cb1114-1" aria-hidden="true" tabindex="-1"></a>RasterMaps<span class="sc">$</span>err27</span></code></pre></div>
<pre><code>## class      : RasterLayer 
## dimensions : 426, 358, 152508  (nrow, ncol, ncell)
## resolution : 30, 30  (x, y)
## extent     : 683282.5, 694022.5, 4992833, 5005613  (xmin, xmax, ymin, ymax)
## crs        : +proj=utm +zone=11 +ellps=GRS80 +towgs84=0,0,0,0,0,0,0 +units=m +no_defs 
## source     : memory
## names      : err27 
## values     : 0.03906551, 0.7637643  (min, max)</code></pre>
<p>Elevation relief ratio (‘err27’) is identifying major topographic features. In this first case, the goal is for ‘err27’ to have the lowest conductance values compared to our other landscape variables. Greater ‘err27’ means more change in topography in a given area. So, the higher the value, the more resistance to a Columbian spotted frog.
Note that the function below gives a linear relationship between the raw value and the cost. Any type of functional relationship could be used however.</p>
<div class="sourceCode" id="cb1116"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1116-1"><a href="WE_10.html#cb1116-1" aria-hidden="true" tabindex="-1"></a>err.cost <span class="ot">&lt;-</span> (<span class="dv">1</span><span class="sc">/</span>RasterMaps<span class="sc">$</span>err27)</span>
<span id="cb1116-2"><a href="WE_10.html#cb1116-2" aria-hidden="true" tabindex="-1"></a>err.cost</span></code></pre></div>
<pre><code>## class      : RasterLayer 
## dimensions : 426, 358, 152508  (nrow, ncol, ncell)
## resolution : 30, 30  (x, y)
## extent     : 683282.5, 694022.5, 4992833, 5005613  (xmin, xmax, ymin, ymax)
## crs        : +proj=utm +zone=11 +ellps=GRS80 +towgs84=0,0,0,0,0,0,0 +units=m +no_defs 
## source     : memory
## names      : err27 
## values     : 1.309305, 25.59803  (min, max)</code></pre>
<div class="sourceCode" id="cb1118"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1118-1"><a href="WE_10.html#cb1118-1" aria-hidden="true" tabindex="-1"></a>RasterMaps<span class="sc">$</span>ffp</span></code></pre></div>
<pre><code>## class      : RasterLayer 
## dimensions : 426, 358, 152508  (nrow, ncol, ncell)
## resolution : 30, 30  (x, y)
## extent     : 683282.5, 694022.5, 4992833, 5005613  (xmin, xmax, ymin, ymax)
## crs        : +proj=utm +zone=11 +ellps=GRS80 +towgs84=0,0,0,0,0,0,0 +units=m +no_defs 
## source     : memory
## names      : ffp 
## values     : 0, 51  (min, max)</code></pre>
<div class="sourceCode" id="cb1120"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1120-1"><a href="WE_10.html#cb1120-1" aria-hidden="true" tabindex="-1"></a>ffp.cost <span class="ot">&lt;-</span> (RasterMaps<span class="sc">$</span>ffp<span class="sc">/</span><span class="dv">5</span>)</span>
<span id="cb1120-2"><a href="WE_10.html#cb1120-2" aria-hidden="true" tabindex="-1"></a>ffp.cost</span></code></pre></div>
<pre><code>## class      : RasterLayer 
## dimensions : 426, 358, 152508  (nrow, ncol, ncell)
## resolution : 30, 30  (x, y)
## extent     : 683282.5, 694022.5, 4992833, 5005613  (xmin, xmax, ymin, ymax)
## crs        : +proj=utm +zone=11 +ellps=GRS80 +towgs84=0,0,0,0,0,0,0 +units=m +no_defs 
## source     : memory
## names      : ffp 
## values     : 0, 10.2  (min, max)</code></pre>
<div class="sourceCode" id="cb1122"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1122-1"><a href="WE_10.html#cb1122-1" aria-hidden="true" tabindex="-1"></a>RasterMaps<span class="sc">$</span>gsp</span></code></pre></div>
<pre><code>## class      : RasterLayer 
## dimensions : 426, 358, 152508  (nrow, ncol, ncell)
## resolution : 30, 30  (x, y)
## extent     : 683282.5, 694022.5, 4992833, 5005613  (xmin, xmax, ymin, ymax)
## crs        : +proj=utm +zone=11 +ellps=GRS80 +towgs84=0,0,0,0,0,0,0 +units=m +no_defs 
## source     : memory
## names      : gsp 
## values     : 227, 338.0697  (min, max)</code></pre>
<div class="sourceCode" id="cb1124"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1124-1"><a href="WE_10.html#cb1124-1" aria-hidden="true" tabindex="-1"></a>gsp.cost <span class="ot">&lt;-</span> (RasterMaps<span class="sc">$</span>gsp<span class="dv">-196</span>)<span class="sc">/</span><span class="dv">15</span></span>
<span id="cb1124-2"><a href="WE_10.html#cb1124-2" aria-hidden="true" tabindex="-1"></a>gsp.cost</span></code></pre></div>
<pre><code>## class      : RasterLayer 
## dimensions : 426, 358, 152508  (nrow, ncol, ncell)
## resolution : 30, 30  (x, y)
## extent     : 683282.5, 694022.5, 4992833, 5005613  (xmin, xmax, ymin, ymax)
## crs        : +proj=utm +zone=11 +ellps=GRS80 +towgs84=0,0,0,0,0,0,0 +units=m +no_defs 
## source     : memory
## names      : gsp 
## values     : 2.066667, 9.471311  (min, max)</code></pre>
<div class="sourceCode" id="cb1126"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1126-1"><a href="WE_10.html#cb1126-1" aria-hidden="true" tabindex="-1"></a>RasterMaps<span class="sc">$</span>cti</span></code></pre></div>
<pre><code>## class      : RasterLayer 
## dimensions : 426, 358, 152508  (nrow, ncol, ncell)
## resolution : 30, 30  (x, y)
## extent     : 683282.5, 694022.5, 4992833, 5005613  (xmin, xmax, ymin, ymax)
## crs        : +proj=utm +zone=11 +ellps=GRS80 +towgs84=0,0,0,0,0,0,0 +units=m +no_defs 
## source     : memory
## names      : cti 
## values     : 0.8429851, 23.71476  (min, max)</code></pre>
<div class="sourceCode" id="cb1128"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1128-1"><a href="WE_10.html#cb1128-1" aria-hidden="true" tabindex="-1"></a>cti.cost <span class="ot">&lt;-</span> RasterMaps<span class="sc">$</span>cti<span class="sc">/</span><span class="dv">5</span></span>
<span id="cb1128-2"><a href="WE_10.html#cb1128-2" aria-hidden="true" tabindex="-1"></a>cti.cost</span></code></pre></div>
<pre><code>## class      : RasterLayer 
## dimensions : 426, 358, 152508  (nrow, ncol, ncell)
## resolution : 30, 30  (x, y)
## extent     : 683282.5, 694022.5, 4992833, 5005613  (xmin, xmax, ymin, ymax)
## crs        : +proj=utm +zone=11 +ellps=GRS80 +towgs84=0,0,0,0,0,0,0 +units=m +no_defs 
## source     : memory
## names      : cti 
## values     : 0.168597, 4.742952  (min, max)</code></pre>
</div>
<div id="d.-create-a-single-landscape-conductance-raster" class="section level4 unnumbered hasAnchor">
<h4>d. Create a single landscape conductance raster<a href="WE_10.html#d.-create-a-single-landscape-conductance-raster" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>Testing variables independently may be more appropriate depending on the research question but for the purposes of today’s lab, a single conductance raster will be employed.</p>
<div class="sourceCode" id="cb1130"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1130-1"><a href="WE_10.html#cb1130-1" aria-hidden="true" tabindex="-1"></a>cost1 <span class="ot">&lt;-</span> (gsp.cost <span class="sc">+</span> cti.cost <span class="sc">+</span> err.cost <span class="sc">+</span> ffp.cost)</span>
<span id="cb1130-2"><a href="WE_10.html#cb1130-2" aria-hidden="true" tabindex="-1"></a>cost1</span></code></pre></div>
<pre><code>## class      : RasterLayer 
## dimensions : 426, 358, 152508  (nrow, ncol, ncell)
## resolution : 30, 30  (x, y)
## extent     : 683282.5, 694022.5, 4992833, 5005613  (xmin, xmax, ymin, ymax)
## crs        : +proj=utm +zone=11 +ellps=GRS80 +towgs84=0,0,0,0,0,0,0 +units=m +no_defs 
## source     : memory
## names      : layer 
## values     : 9.012874, 35.90095  (min, max)</code></pre>
<p><strong>Question 2</strong>: Plot your cost surface with your sample locations on top. What does this tell you?</p>
</div>
</div>
<div id="convert-conductance-into-effective-distance" class="section level3 unnumbered hasAnchor">
<h3>4. Convert conductance into effective distance<a href="WE_10.html#convert-conductance-into-effective-distance" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>The higher the conductance, the lower the cost or resistance of a cell, and vice versa. We want to integrate conductance across cells to derive some measure of effective (or ecological) distance.</p>
<div id="a.-create-a-transition-layer" class="section level4 unnumbered hasAnchor">
<h4>a. Create a transition layer<a href="WE_10.html#a.-create-a-transition-layer" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>Transition layers are constructed from a raster, and they take the geographic references (projection, resolution, extent) from the original raster object. They also contain a matrix of probability of movement between cells which can be interpreted as “conductance.” Each cell in the matrix represents a cell in the original raster object.</p>
<p>The first step is to construct a transition object based on ‘cost1’ (which is a conductance layer as calculated). This step is computationally intensive and make take a few minutes to run.</p>
<p>Connections can be set based on 4, 8, or 16 neighbor rules. A value of 8 connects all adjacent cells in 8 directions.</p>
<div class="sourceCode" id="cb1132"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1132-1"><a href="WE_10.html#cb1132-1" aria-hidden="true" tabindex="-1"></a>tr.cost1 <span class="ot">&lt;-</span> gdistance<span class="sc">::</span><span class="fu">transition</span>(cost1, <span class="at">transitionFunction=</span>mean, <span class="at">directions=</span><span class="dv">8</span>) </span>
<span id="cb1132-2"><a href="WE_10.html#cb1132-2" aria-hidden="true" tabindex="-1"></a>tr.cost1</span></code></pre></div>
<pre><code>## class      : TransitionLayer 
## dimensions : 426, 358, 152508  (nrow, ncol, ncell)
## resolution : 30, 30  (x, y)
## extent     : 683282.5, 694022.5, 4992833, 5005613  (xmin, xmax, ymin, ymax)
## crs        : +proj=utm +zone=11 +ellps=GRS80 +towgs84=0,0,0,0,0,0,0 +units=m +no_defs 
## values      : conductance 
## matrix class: dsCMatrix</code></pre>
</div>
<div id="b.-visually-inspect-the-raster" class="section level4 unnumbered hasAnchor">
<h4>b. Visually inspect the raster<a href="WE_10.html#b.-visually-inspect-the-raster" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<div class="sourceCode" id="cb1134"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1134-1"><a href="WE_10.html#cb1134-1" aria-hidden="true" tabindex="-1"></a><span class="fu">par</span>(<span class="at">mar=</span><span class="fu">c</span>(<span class="dv">2</span>,<span class="dv">2</span>,<span class="dv">1</span>,<span class="dv">1</span>))</span>
<span id="cb1134-2"><a href="WE_10.html#cb1134-2" aria-hidden="true" tabindex="-1"></a>raster<span class="sc">::</span><span class="fu">plot</span>(raster<span class="sc">::</span><span class="fu">raster</span>(tr.cost1))</span></code></pre></div>
<p><img src="10-Week10_files/figure-html/unnamed-chunk-22-1.png" width="672" /></p>
</div>
<div id="c.-correct-for-geometric-distortion" class="section level4 unnumbered hasAnchor">
<h4>c. Correct for geometric distortion<a href="WE_10.html#c.-correct-for-geometric-distortion" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>Transition values are calculated based on values of adjacent cells in the cost raster. However, we used an 8 neighbor rule and the center of diagonally connected
raster cells are farther apart from each other than the orthogonally connected cells. We are using UTM co-ordinates; however, in lat-long projections, cell sizes become smaller as you move poleward.</p>
<p>Values of the matrix have to be corrected for the first type of distortion for our analysis (and we would need to correct for the second type of distortion if we were using lat-longs).</p>
<div class="sourceCode" id="cb1135"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1135-1"><a href="WE_10.html#cb1135-1" aria-hidden="true" tabindex="-1"></a>tr.cost1 <span class="ot">&lt;-</span> gdistance<span class="sc">::</span><span class="fu">geoCorrection</span>(tr.cost1,<span class="at">type =</span> <span class="st">&quot;c&quot;</span>,<span class="at">multpl=</span><span class="cn">FALSE</span>)</span></code></pre></div>
</div>
<div id="d.-plot-shortest-paths-in-space" class="section level4 unnumbered hasAnchor">
<h4>d. Plot shortest paths in space<a href="WE_10.html#d.-plot-shortest-paths-in-space" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>Here we do the following:</p>
<ul>
<li>Plot site locations on top of conductance raster ‘tr.cost1’</li>
<li>Calculate ‘AtoB’ as the shortest path (least cost path) between sites 1 and 2</li>
<li>Plot the shortest path onto the map</li>
</ul>
<p>Note: the path is stored as a ‘SpatialLines’ object (package ‘spdep’). A path is a sequence of coordinate pairs, which is stored in the <code>@lines</code> slot. Like a ‘SpatialPointsDataFrame,’ a ‘SpatialLines’ object has a ‘proje4string’ attribute that stores the projection information. This means that we can easily plot it on top of a raster map.</p>
<div class="sourceCode" id="cb1136"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1136-1"><a href="WE_10.html#cb1136-1" aria-hidden="true" tabindex="-1"></a><span class="fu">par</span>(<span class="at">mar=</span><span class="fu">c</span>(<span class="dv">2</span>,<span class="dv">2</span>,<span class="dv">1</span>,<span class="dv">2</span>))</span>
<span id="cb1136-2"><a href="WE_10.html#cb1136-2" aria-hidden="true" tabindex="-1"></a>AtoB <span class="ot">&lt;-</span> gdistance<span class="sc">::</span><span class="fu">shortestPath</span>(tr.cost1, <span class="at">origin=</span>sites[<span class="dv">1</span>,], </span>
<span id="cb1136-3"><a href="WE_10.html#cb1136-3" aria-hidden="true" tabindex="-1"></a>                                <span class="at">goal=</span>sites[<span class="dv">2</span>,], <span class="at">output=</span><span class="st">&quot;SpatialLines&quot;</span>)</span>
<span id="cb1136-4"><a href="WE_10.html#cb1136-4" aria-hidden="true" tabindex="-1"></a>raster<span class="sc">::</span><span class="fu">plot</span>(raster<span class="sc">::</span><span class="fu">raster</span>(tr.cost1), <span class="at">xlab=</span><span class="st">&quot;x coordinate (m)&quot;</span>, </span>
<span id="cb1136-5"><a href="WE_10.html#cb1136-5" aria-hidden="true" tabindex="-1"></a>             <span class="at">ylab=</span><span class="st">&quot;y coordinate (m)&quot;</span>,<span class="at">legend.lab=</span><span class="st">&quot;Conductance&quot;</span>)</span>
<span id="cb1136-6"><a href="WE_10.html#cb1136-6" aria-hidden="true" tabindex="-1"></a><span class="fu">lines</span>(AtoB, <span class="at">col=</span><span class="st">&quot;red&quot;</span>, <span class="at">lwd=</span><span class="dv">2</span>)</span>
<span id="cb1136-7"><a href="WE_10.html#cb1136-7" aria-hidden="true" tabindex="-1"></a><span class="fu">points</span>(sites[<span class="dv">1</span><span class="sc">:</span><span class="dv">2</span>,])</span></code></pre></div>
<p><img src="10-Week10_files/figure-html/unnamed-chunk-24-1.png" width="672" /></p>
<p>That’s nice, but it shows only the shortest path between sites 1 and 2. It would not be very useful to plot all pairwise shortest paths. Here we do the following:</p>
<ul>
<li>Plot site locations on top of conductance raster ‘tr.cost1’</li>
<li>Use Delaunay triangulation to define neighbors (we want to retain only those paths that won’t cross each other, though they may merge)</li>
<li>Plot grey lines between neighbours</li>
<li>For each unique pair of neighbours, calculate ‘AtoB’ as the shortest path<br />
</li>
<li>Plot the shortest paths onto the map in red</li>
</ul>
<p>Note: this may take a while. If it does not show up nicely in the notebook, copy and paste the code to the console and watch how the plot is evolving in the Plots tab (nothing beats being able to actually watch R do the work for you).</p>
<div class="sourceCode" id="cb1137"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1137-1"><a href="WE_10.html#cb1137-1" aria-hidden="true" tabindex="-1"></a><span class="fu">par</span>(<span class="at">mar=</span><span class="fu">c</span>(<span class="dv">2</span>,<span class="dv">2</span>,<span class="dv">1</span>,<span class="dv">2</span>))</span>
<span id="cb1137-2"><a href="WE_10.html#cb1137-2" aria-hidden="true" tabindex="-1"></a>raster<span class="sc">::</span><span class="fu">plot</span>(raster<span class="sc">::</span><span class="fu">raster</span>(tr.cost1), <span class="at">xlab=</span><span class="st">&quot;x coordinate (m)&quot;</span>, </span>
<span id="cb1137-3"><a href="WE_10.html#cb1137-3" aria-hidden="true" tabindex="-1"></a>             <span class="at">ylab=</span><span class="st">&quot;y coordinate (m)&quot;</span>, <span class="at">legend.lab=</span><span class="st">&quot;Conductance&quot;</span>)</span>
<span id="cb1137-4"><a href="WE_10.html#cb1137-4" aria-hidden="true" tabindex="-1"></a><span class="fu">points</span>(sites)</span>
<span id="cb1137-5"><a href="WE_10.html#cb1137-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1137-6"><a href="WE_10.html#cb1137-6" aria-hidden="true" tabindex="-1"></a>Neighbours <span class="ot">&lt;-</span> spdep<span class="sc">::</span><span class="fu">tri2nb</span>(sites<span class="sc">@</span>coords, <span class="at">row.names =</span> sites<span class="sc">$</span>SiteName)</span>
<span id="cb1137-7"><a href="WE_10.html#cb1137-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1137-8"><a href="WE_10.html#cb1137-8" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(Neighbours, sites<span class="sc">@</span>coords, <span class="at">col=</span><span class="st">&quot;darkgrey&quot;</span>, <span class="at">add=</span><span class="cn">TRUE</span>)</span>
<span id="cb1137-9"><a href="WE_10.html#cb1137-9" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span>(i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(Neighbours))</span>
<span id="cb1137-10"><a href="WE_10.html#cb1137-10" aria-hidden="true" tabindex="-1"></a>{</span>
<span id="cb1137-11"><a href="WE_10.html#cb1137-11" aria-hidden="true" tabindex="-1"></a>  <span class="cf">for</span>(j <span class="cf">in</span> Neighbours[[i]][Neighbours[[i]] <span class="sc">&gt;</span> i])</span>
<span id="cb1137-12"><a href="WE_10.html#cb1137-12" aria-hidden="true" tabindex="-1"></a>  {</span>
<span id="cb1137-13"><a href="WE_10.html#cb1137-13" aria-hidden="true" tabindex="-1"></a>    AtoB <span class="ot">&lt;-</span> gdistance<span class="sc">::</span><span class="fu">shortestPath</span>(tr.cost1, <span class="at">origin=</span>sites[i,], </span>
<span id="cb1137-14"><a href="WE_10.html#cb1137-14" aria-hidden="true" tabindex="-1"></a>                                <span class="at">goal=</span>sites[j,], <span class="at">output=</span><span class="st">&quot;SpatialLines&quot;</span>)</span>
<span id="cb1137-15"><a href="WE_10.html#cb1137-15" aria-hidden="true" tabindex="-1"></a>    <span class="fu">lines</span>(AtoB, <span class="at">col=</span><span class="st">&quot;red&quot;</span>, <span class="at">lwd=</span><span class="fl">1.5</span>)</span>
<span id="cb1137-16"><a href="WE_10.html#cb1137-16" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb1137-17"><a href="WE_10.html#cb1137-17" aria-hidden="true" tabindex="-1"></a>}</span></code></pre></div>
<p><img src="10-Week10_files/figure-html/unnamed-chunk-25-1.png" width="672" /></p>
<p><strong>Question 3</strong>: Do the shortest paths (red) deviate from the Euclidean distances (grey)? Why?</p>
<p><strong>Question 4</strong>: Are there paths that merge? How would you explain this?</p>
</div>
</div>
<div id="create-cost-distance-matrices" class="section level3 unnumbered hasAnchor">
<h3>5. Create cost-distance matrices<a href="WE_10.html#create-cost-distance-matrices" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div id="a.-least-cost-distance" class="section level4 unnumbered hasAnchor">
<h4>a. Least cost distance<a href="WE_10.html#a.-least-cost-distance" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>The cost-distance matrix is based on the corrected transition layer and the site locations (here we supply a ‘SpatialPointsDataFrame’ object). The ‘costDistance’ function requires conductance values, even though cost distance will be 1/conductance (i.e., resistance). The cost distance is a function of the transition (tr.cost1, a transition object) and spatial locations (sites, a spatial object).
This is a single least-cost path (accumulated cost units as opposed to length units) between each pair of sites..</p>
<div class="sourceCode" id="cb1138"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1138-1"><a href="WE_10.html#cb1138-1" aria-hidden="true" tabindex="-1"></a>cost1.dist <span class="ot">&lt;-</span> gdistance<span class="sc">::</span><span class="fu">costDistance</span>(tr.cost1,sites)</span></code></pre></div>
</div>
<div id="b.-cost-distance-matrix-based-on-random-paths-similar-to-circuitscape" class="section level4 unnumbered hasAnchor">
<h4>b. Cost-distance matrix based on random paths (similar to Circuitscape)<a href="WE_10.html#b.-cost-distance-matrix-based-on-random-paths-similar-to-circuitscape" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>Commute-time distance is an alternative to effective resistance from circuit theory. Commute-time is the expected time it takes for a random walk between nodes, and has been shown to correlate highly with effective resistance (Marrotte and Bowman 2017).</p>
<div class="sourceCode" id="cb1139"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1139-1"><a href="WE_10.html#cb1139-1" aria-hidden="true" tabindex="-1"></a>comm1.dist <span class="ot">&lt;-</span> gdistance<span class="sc">::</span><span class="fu">commuteDistance</span>(<span class="at">x =</span> tr.cost1, <span class="at">coords =</span> sites)</span></code></pre></div>
</div>
<div id="c.-compare-cost-distances" class="section level4 unnumbered hasAnchor">
<h4>c. Compare cost distances<a href="WE_10.html#c.-compare-cost-distances" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>Create a distance table to compare different cost distance estimates:</p>
<div class="sourceCode" id="cb1140"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1140-1"><a href="WE_10.html#cb1140-1" aria-hidden="true" tabindex="-1"></a>dist_df <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="st">&quot;cost1.dist&quot;</span><span class="ot">=</span><span class="fu">as.numeric</span>(cost1.dist),</span>
<span id="cb1140-2"><a href="WE_10.html#cb1140-2" aria-hidden="true" tabindex="-1"></a>                      <span class="st">&quot;comm1.dist&quot;</span><span class="ot">=</span><span class="fu">as.numeric</span>(comm1.dist))</span></code></pre></div>
<p>Look at the correlation between the two different cost distances:</p>
<div class="sourceCode" id="cb1141"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1141-1"><a href="WE_10.html#cb1141-1" aria-hidden="true" tabindex="-1"></a>corr.LCD.comm <span class="ot">&lt;-</span> <span class="fu">cor</span>(dist_df<span class="sc">$</span>cost1.dist, dist_df<span class="sc">$</span>comm1.dist, <span class="at">method =</span> <span class="st">&quot;spearman&quot;</span>)</span>
<span id="cb1141-2"><a href="WE_10.html#cb1141-2" aria-hidden="true" tabindex="-1"></a>corr.LCD.comm</span></code></pre></div>
<pre><code>## [1] 0.9519704</code></pre>
<div class="sourceCode" id="cb1143"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1143-1"><a href="WE_10.html#cb1143-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(cost1.dist<span class="sc">~</span>comm1.dist)</span></code></pre></div>
<p><img src="10-Week10_files/figure-html/unnamed-chunk-29-1.png" width="672" /></p>
<p><strong>Question 5</strong>: Are the distance measures similar or different? Why?</p>
<p><strong>Question 6</strong>: What is the major difference? What are the implications of this difference? When might you use one or other of the methods?</p>
</div>
</div>
<div id="how-does-changing-resolution-affect-these-metrics" class="section level3 unnumbered hasAnchor">
<h3>6. How does changing resolution affect these metrics?<a href="WE_10.html#how-does-changing-resolution-affect-these-metrics" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Create a loop that runs through a sequential coarsening of the raster and calculate
these metrics and compare them to the finer resolution raster metric using a Spearman correlation. The loop will take a few minutes to run.</p>
<div id="a.-create-loop" class="section level4 unnumbered hasAnchor">
<h4>a. Create loop<a href="WE_10.html#a.-create-loop" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<div class="sourceCode" id="cb1144"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1144-1"><a href="WE_10.html#cb1144-1" aria-hidden="true" tabindex="-1"></a>cor_cost <span class="ot">&lt;-</span> <span class="fu">c</span>()</span>
<span id="cb1144-2"><a href="WE_10.html#cb1144-2" aria-hidden="true" tabindex="-1"></a>cor_comm <span class="ot">&lt;-</span> <span class="fu">c</span>()</span>
<span id="cb1144-3"><a href="WE_10.html#cb1144-3" aria-hidden="true" tabindex="-1"></a>res_fact <span class="ot">&lt;-</span> <span class="fu">seq</span>(<span class="dv">2</span>,<span class="dv">20</span>,<span class="dv">2</span>)</span>
<span id="cb1144-4"><a href="WE_10.html#cb1144-4" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span>(fac <span class="cf">in</span> res_fact){</span>
<span id="cb1144-5"><a href="WE_10.html#cb1144-5" aria-hidden="true" tabindex="-1"></a>  cost1_agg <span class="ot">&lt;-</span> raster<span class="sc">::</span><span class="fu">aggregate</span>(cost1, <span class="at">fact =</span> fac)</span>
<span id="cb1144-6"><a href="WE_10.html#cb1144-6" aria-hidden="true" tabindex="-1"></a>  tr.cost_agg <span class="ot">&lt;-</span> gdistance<span class="sc">::</span><span class="fu">transition</span>(cost1_agg, </span>
<span id="cb1144-7"><a href="WE_10.html#cb1144-7" aria-hidden="true" tabindex="-1"></a>                 <span class="at">transitionFunction=</span>mean, <span class="at">directions=</span><span class="dv">8</span>)</span>
<span id="cb1144-8"><a href="WE_10.html#cb1144-8" aria-hidden="true" tabindex="-1"></a>  tr.cost_agg <span class="ot">&lt;-</span> gdistance<span class="sc">::</span><span class="fu">geoCorrection</span>(tr.cost_agg,<span class="at">type =</span> <span class="st">&quot;c&quot;</span>,<span class="at">multpl=</span><span class="cn">FALSE</span>)</span>
<span id="cb1144-9"><a href="WE_10.html#cb1144-9" aria-hidden="true" tabindex="-1"></a>  cost.dist_agg <span class="ot">&lt;-</span> gdistance<span class="sc">::</span><span class="fu">costDistance</span>(tr.cost_agg,sites)</span>
<span id="cb1144-10"><a href="WE_10.html#cb1144-10" aria-hidden="true" tabindex="-1"></a>  comm.dist_agg <span class="ot">&lt;-</span> gdistance<span class="sc">::</span><span class="fu">commuteDistance</span>(<span class="at">x =</span> tr.cost_agg, <span class="at">coords =</span> sites)</span>
<span id="cb1144-11"><a href="WE_10.html#cb1144-11" aria-hidden="true" tabindex="-1"></a>  cost.dist_agg <span class="ot">&lt;-</span> <span class="fu">as.numeric</span>(cost.dist_agg)</span>
<span id="cb1144-12"><a href="WE_10.html#cb1144-12" aria-hidden="true" tabindex="-1"></a>  comm.dist_agg <span class="ot">&lt;-</span> <span class="fu">as.numeric</span>(comm.dist_agg)</span>
<span id="cb1144-13"><a href="WE_10.html#cb1144-13" aria-hidden="true" tabindex="-1"></a>  cor_cost <span class="ot">&lt;-</span> <span class="fu">c</span>(cor_cost,<span class="fu">cor</span>(dist_df<span class="sc">$</span>cost1.dist, cost.dist_agg, </span>
<span id="cb1144-14"><a href="WE_10.html#cb1144-14" aria-hidden="true" tabindex="-1"></a>                             <span class="at">method =</span> <span class="st">&quot;spearman&quot;</span>))</span>
<span id="cb1144-15"><a href="WE_10.html#cb1144-15" aria-hidden="true" tabindex="-1"></a>  cor_comm <span class="ot">&lt;-</span> <span class="fu">c</span>(cor_comm,<span class="fu">cor</span>(dist_df<span class="sc">$</span>comm1.dist, comm.dist_agg, </span>
<span id="cb1144-16"><a href="WE_10.html#cb1144-16" aria-hidden="true" tabindex="-1"></a>                             <span class="at">method =</span> <span class="st">&quot;spearman&quot;</span>))</span>
<span id="cb1144-17"><a href="WE_10.html#cb1144-17" aria-hidden="true" tabindex="-1"></a>}</span></code></pre></div>
</div>
<div id="b.-plot-the-results" class="section level4 unnumbered hasAnchor">
<h4>b. Plot the results<a href="WE_10.html#b.-plot-the-results" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<div class="sourceCode" id="cb1145"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1145-1"><a href="WE_10.html#cb1145-1" aria-hidden="true" tabindex="-1"></a><span class="fu">par</span>(<span class="at">mar=</span><span class="fu">c</span>(<span class="dv">4</span>,<span class="dv">4</span>,<span class="dv">1</span>,<span class="dv">1</span>))</span>
<span id="cb1145-2"><a href="WE_10.html#cb1145-2" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(<span class="at">y =</span> cor_cost, <span class="at">x =</span> res_fact, <span class="at">col =</span> <span class="st">&quot;red&quot;</span>, <span class="at">pch =</span> <span class="dv">19</span>, </span>
<span id="cb1145-3"><a href="WE_10.html#cb1145-3" aria-hidden="true" tabindex="-1"></a>     <span class="at">ylim =</span> <span class="fu">c</span>(<span class="fl">0.9</span>,<span class="dv">1</span>), <span class="at">xlab =</span> <span class="st">&quot;Aggregation factor&quot;</span>, <span class="at">ylab =</span> <span class="st">&quot;Spearman correlation&quot;</span>)</span>
<span id="cb1145-4"><a href="WE_10.html#cb1145-4" aria-hidden="true" tabindex="-1"></a><span class="fu">points</span>(<span class="at">y =</span> cor_comm, <span class="at">x =</span> res_fact, <span class="at">col =</span> <span class="st">&quot;blue&quot;</span>, <span class="at">pch =</span> <span class="dv">19</span>)</span>
<span id="cb1145-5"><a href="WE_10.html#cb1145-5" aria-hidden="true" tabindex="-1"></a><span class="fu">legend</span>(<span class="st">&quot;bottomleft&quot;</span>, <span class="at">legend =</span> <span class="fu">c</span>(<span class="st">&quot;Costdist&quot;</span>,<span class="st">&quot;Commdist&quot;</span>), </span>
<span id="cb1145-6"><a href="WE_10.html#cb1145-6" aria-hidden="true" tabindex="-1"></a>       <span class="at">pch =</span> <span class="dv">19</span>, <span class="at">col =</span> <span class="fu">c</span>(<span class="st">&quot;red&quot;</span>, <span class="st">&quot;blue&quot;</span>))</span></code></pre></div>
<p><img src="10-Week10_files/figure-html/unnamed-chunk-31-1.png" width="672" /></p>
<p><strong>Question 7</strong>: What effect does changing resolution have on these metrics?</p>
</div>
</div>
<div id="references" class="section level3 unnumbered hasAnchor">
<h3>7. References<a href="WE_10.html#references" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Evans IS (1972) General geomorphometry, derivatives of altitude, and descriptive statistics. In: Chorley RJ (ed) Spatial analysis in geomorphology. Harper &amp; Row, New York.</p>
<p>Funk WC, Blouin MS, Corn PS, Maxell BA, Pilliod DS, Amish S, Allendorf FW (2005) Population structure of Columbia spotted frogs (Rana luteiventris) is strongly affected by the landscape. Mol Ecol 14(2): 483-496</p>
<p>Marrotte RR, Bowman J (2017) The relationship between least-cost and resistance distance. PLOS ONE 12(3): e0174212</p>
<p>McRae BH, Dickson BG, Keitt TH, Shah VB (2008) Using circuit theory to model connectivity in ecology, evolution, and conservation. Ecology 89 (10):2712-2724</p>
<p>Moore I, Gessler P, Nielsen G, Petersen G (eds) (1993) Terrain attributes and estimation methods and scale effects. Modeling change in environmental systems. Wiley, London.</p>
<p>Murphy MA, Dezanni RJ, Pilliod D, Storfer A (2010) Landscape genetics of high mountain frog populations. Mol Ecol 19:3634-3649.</p>
<p>Pilliod DS, Peterson CR, Ritson PI (2002) Seasonal migration of Columbia spotted frogs (Rana luteiventris) among complementary resources in a high mountain basin. Can J Zool 80: 1849-1862</p>
<p>Rehfeldt GE, Crookston NL, Warwell MV, Evans JS (2006) Empirical analysis of plan-climate relationships for western United States. International Journal of Plan Sciences 167: 1123-1150</p>
</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="Week10.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="bonus-10a.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": true,
"facebook": false,
"twitter": false,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": false
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": null,
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "section"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
