<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Worked Example | Landscape Genetic Data Analysis with R</title>
  <meta name="description" content="This is a web-interface to the teaching materials for the lab course ‘Landscape Genetic Data Analysis with R’ associated with the distributed graduate course ‘DGS Landscape Genetics.’ The output format is bookdown::gitbook." />
  <meta name="generator" content="bookdown 0.24 and GitBook 2.6.7" />

  <meta property="og:title" content="Worked Example | Landscape Genetic Data Analysis with R" />
  <meta property="og:type" content="book" />
  
  
  <meta property="og:description" content="This is a web-interface to the teaching materials for the lab course ‘Landscape Genetic Data Analysis with R’ associated with the distributed graduate course ‘DGS Landscape Genetics.’ The output format is bookdown::gitbook." />
  <meta name="github-repo" content="hhwagner1/LandGenCourse_book" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Worked Example | Landscape Genetic Data Analysis with R" />
  
  <meta name="twitter:description" content="This is a web-interface to the teaching materials for the lab course ‘Landscape Genetic Data Analysis with R’ associated with the distributed graduate course ‘DGS Landscape Genetics.’ The output format is bookdown::gitbook." />
  

<meta name="author" content="Editor: Helene Wagner (University of Toronto)" />


<meta name="date" content="2022-01-08" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="Week10.html"/>
<link rel="next" href="Week11.html"/>
<script src="libs/header-attrs-2.11/header-attrs.js"></script>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.0.1/anchor-sections.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.0.1/anchor-sections.js"></script>
<script src="libs/kePrint-0.0.1/kePrint.js"></script>
<link href="libs/lightable-0.0.1/lightable.css" rel="stylesheet" />
<script src="libs/htmlwidgets-1.5.4/htmlwidgets.js"></script>
<script src="libs/d3-3.5.6/d3.min.js"></script>
<link href="libs/profvis-0.3.6.9000/profvis.css" rel="stylesheet" />
<script src="libs/profvis-0.3.6.9000/profvis.js"></script>
<script src="libs/profvis-0.3.6.9000/scroll.js"></script>
<link href="libs/highlight-6.2.0/textmate.css" rel="stylesheet" />
<script src="libs/highlight-6.2.0/highlight.js"></script>
<script src="libs/profvis-binding-0.3.7/profvis.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>


<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">Landscape Genetic Data Analysis with R</a></li>

<li class="divider"></li>
<li class="part"><span><b>Getting Started</b></span></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i>Introduction</a>
<ul>
<li class="chapter" data-level="" data-path="how-to-use-this-book.html"><a href="how-to-use-this-book.html"><i class="fa fa-check"></i>How to use this Book</a>
<ul>
<li class="chapter" data-level="" data-path="how-to-use-this-book.html"><a href="how-to-use-this-book.html#book-structure"><i class="fa fa-check"></i>1. Book Structure</a>
<ul>
<li class="chapter" data-level="" data-path="how-to-use-this-book.html"><a href="how-to-use-this-book.html#a-getting-started"><i class="fa fa-check"></i>a) Getting Started</a></li>
<li class="chapter" data-level="" data-path="how-to-use-this-book.html"><a href="how-to-use-this-book.html#b-basic-topics"><i class="fa fa-check"></i>b) Basic Topics</a></li>
<li class="chapter" data-level="" data-path="how-to-use-this-book.html"><a href="how-to-use-this-book.html#c-advanced-topics"><i class="fa fa-check"></i>c) Advanced Topics</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="how-to-use-this-book.html"><a href="how-to-use-this-book.html#find-what-is-relevant-for-you"><i class="fa fa-check"></i>2. Find what is relevant for you</a>
<ul>
<li class="chapter" data-level="" data-path="how-to-use-this-book.html"><a href="how-to-use-this-book.html#how-to-use"><i class="fa fa-check"></i>a) How to use the labs</a></li>
<li class="chapter" data-level="" data-path="how-to-use-this-book.html"><a href="how-to-use-this-book.html#b-check-contents"><i class="fa fa-check"></i>b) Check contents</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="how-to-use-this-book.html"><a href="how-to-use-this-book.html#course-r-package-landgencourse"><i class="fa fa-check"></i>3. Course R package ‘LandGenCourse’</a>
<ul>
<li class="chapter" data-level="" data-path="how-to-use-this-book.html"><a href="how-to-use-this-book.html#a-practice-good-r-hygiene"><i class="fa fa-check"></i>a) Practice good R hygiene</a></li>
<li class="chapter" data-level="" data-path="how-to-use-this-book.html"><a href="how-to-use-this-book.html#b-how-to-install-or-update-landgencourse"><i class="fa fa-check"></i>b) How to install (or update) ‘LandGenCourse’</a></li>
<li class="chapter" data-level="" data-path="how-to-use-this-book.html"><a href="how-to-use-this-book.html#c-how-to-install-dependencies"><i class="fa fa-check"></i>c) How to install dependencies</a></li>
<li><a href="how-to-use-this-book.html#d-how-to-use-the-r-package-landgencourse">d) How to use the R package <code>LandGenCourse</code></a></li>
<li class="chapter" data-level="" data-path="how-to-use-this-book.html"><a href="how-to-use-this-book.html#e-video-instructions-for-beginners"><i class="fa fa-check"></i>e) Video instructions for beginners</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="" data-path="list-of-packages.html"><a href="list-of-packages.html"><i class="fa fa-check"></i>List of R Packages by Vignette</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="review-of-r-skills.html"><a href="review-of-r-skills.html"><i class="fa fa-check"></i>Review of R Skills</a>
<ul>
<li class="chapter" data-level="" data-path="basic-r.html"><a href="basic-r.html"><i class="fa fa-check"></i>Basic R Programming</a>
<ul>
<li class="chapter" data-level="" data-path="basic-r.html"><a href="basic-r.html#overview"><i class="fa fa-check"></i>1. Overview</a></li>
<li class="chapter" data-level="" data-path="basic-r.html"><a href="basic-r.html#data-types"><i class="fa fa-check"></i>2. Data Types</a>
<ul>
<li class="chapter" data-level="" data-path="basic-r.html"><a href="basic-r.html#numeric-data-types"><i class="fa fa-check"></i>2.1. Numeric Data Types</a></li>
<li class="chapter" data-level="" data-path="basic-r.html"><a href="basic-r.html#characters"><i class="fa fa-check"></i>2.2. Characters</a></li>
<li class="chapter" data-level="" data-path="basic-r.html"><a href="basic-r.html#factors"><i class="fa fa-check"></i>2.3. Factors</a></li>
<li class="chapter" data-level="" data-path="basic-r.html"><a href="basic-r.html#logical-types"><i class="fa fa-check"></i>2.4. Logical Types</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="basic-r.html"><a href="basic-r.html#data-containers"><i class="fa fa-check"></i>3. Data Containers</a>
<ul>
<li class="chapter" data-level="" data-path="basic-r.html"><a href="basic-r.html#vectors"><i class="fa fa-check"></i>3.1. Vectors</a></li>
<li class="chapter" data-level="" data-path="basic-r.html"><a href="basic-r.html#matrices"><i class="fa fa-check"></i>3.2. Matrices</a></li>
<li class="chapter" data-level="" data-path="basic-r.html"><a href="basic-r.html#lists"><i class="fa fa-check"></i>3.3. Lists</a></li>
<li class="chapter" data-level="" data-path="basic-r.html"><a href="basic-r.html#data-frames"><i class="fa fa-check"></i>3.4. Data Frames</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="basic-r.html"><a href="basic-r.html#programming"><i class="fa fa-check"></i>4. Programming</a>
<ul>
<li class="chapter" data-level="" data-path="basic-r.html"><a href="basic-r.html#function-writing"><i class="fa fa-check"></i>4.1. Function Writing</a></li>
<li class="chapter" data-level="" data-path="basic-r.html"><a href="basic-r.html#variable-scope"><i class="fa fa-check"></i>4.2. Variable Scope</a></li>
<li class="chapter" data-level="" data-path="basic-r.html"><a href="basic-r.html#decision-making"><i class="fa fa-check"></i>4.3. Decision Making</a></li>
<li class="chapter" data-level="" data-path="basic-r.html"><a href="basic-r.html#flow-control"><i class="fa fa-check"></i>4.4. Flow Control</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="" data-path="r-graphics.html"><a href="r-graphics.html"><i class="fa fa-check"></i>R Graphics</a>
<ul>
<li class="chapter" data-level="" data-path="r-graphics.html"><a href="r-graphics.html#overview-1"><i class="fa fa-check"></i>1. Overview</a></li>
<li class="chapter" data-level="" data-path="r-graphics.html"><a href="r-graphics.html#univariate-plots"><i class="fa fa-check"></i>2. Univariate plots</a>
<ul>
<li class="chapter" data-level="" data-path="r-graphics.html"><a href="r-graphics.html#barplots-and-histograms"><i class="fa fa-check"></i>2.1. Barplots and histograms</a></li>
<li class="chapter" data-level="" data-path="r-graphics.html"><a href="r-graphics.html#plotting-parameters"><i class="fa fa-check"></i>2.2. Plotting Parameters</a></li>
<li class="chapter" data-level="" data-path="r-graphics.html"><a href="r-graphics.html#density-plots"><i class="fa fa-check"></i>2.3. Density Plots</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="r-graphics.html"><a href="r-graphics.html#bivariate-plots"><i class="fa fa-check"></i>3. Bivariate plots</a>
<ul>
<li class="chapter" data-level="" data-path="r-graphics.html"><a href="r-graphics.html#overlaying-plots-by-grouping"><i class="fa fa-check"></i>3.1. Overlaying Plots by grouping</a></li>
<li class="chapter" data-level="" data-path="r-graphics.html"><a href="r-graphics.html#boxplots"><i class="fa fa-check"></i>3.2. Boxplots</a></li>
<li class="chapter" data-level="" data-path="r-graphics.html"><a href="r-graphics.html#scatter-plots"><i class="fa fa-check"></i>3.3. Scatter Plots</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="r-graphics.html"><a href="r-graphics.html#advanced-plotting"><i class="fa fa-check"></i>4. Advanced plotting</a>
<ul>
<li class="chapter" data-level="" data-path="r-graphics.html"><a href="r-graphics.html#multiple-plots"><i class="fa fa-check"></i>4.1. Multiple Plots</a></li>
<li class="chapter" data-level="" data-path="r-graphics.html"><a href="r-graphics.html#color-palettes"><i class="fa fa-check"></i>4.2. Color Palettes</a></li>
<li class="chapter" data-level="" data-path="r-graphics.html"><a href="r-graphics.html#rcolorbrewer"><i class="fa fa-check"></i>4.3. RColorBrewer</a></li>
<li class="chapter" data-level="" data-path="r-graphics.html"><a href="r-graphics.html#saving-imagery"><i class="fa fa-check"></i>4.4. Saving Imagery</a></li>
<li class="chapter" data-level="" data-path="r-graphics.html"><a href="r-graphics.html#interactive-graphics"><i class="fa fa-check"></i>4.5. Interactive Graphics</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="" data-path="github.html"><a href="github.html"><i class="fa fa-check"></i>Version Control with Github</a>
<ul>
<li class="chapter" data-level="" data-path="github.html"><a href="github.html#external-github-tutorials"><i class="fa fa-check"></i>1. External Github tutorials</a></li>
<li class="chapter" data-level="" data-path="github.html"><a href="github.html#watch-course-video-version-control-101"><i class="fa fa-check"></i>2. Watch course video: Version Control 101</a></li>
<li class="chapter" data-level="" data-path="github.html"><a href="github.html#preview-slides"><i class="fa fa-check"></i>3. Preview slides</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="resources.html"><a href="resources.html"><i class="fa fa-check"></i>Further R resources</a></li>
</ul></li>
<li class="part"><span><b>Basic Topics</b></span></li>
<li class="chapter" data-level="" data-path="Week1.html"><a href="Week1.html"><i class="fa fa-check"></i>Week 1: Importing Genetic Data</a>
<ul>
<li class="chapter" data-level="" data-path="video_1.html"><a href="video_1.html"><i class="fa fa-check"></i>View Course Video</a>
<ul>
<li class="chapter" data-level="" data-path="video_1.html"><a href="video_1.html#embedded-video"><i class="fa fa-check"></i>1. Embedded Video</a></li>
<li class="chapter" data-level="" data-path="video_1.html"><a href="video_1.html#preview-slides-1"><i class="fa fa-check"></i>2. Preview Slides</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="tutorial_1.html"><a href="tutorial_1.html"><i class="fa fa-check"></i>Interactive Tutorial 1</a>
<ul>
<li class="chapter" data-level="" data-path="tutorial_1.html"><a href="tutorial_1.html#list-of-r-commands-covered-this-week"><i class="fa fa-check"></i>1. List of R commands covered this week</a></li>
<li class="chapter" data-level="" data-path="tutorial_1.html"><a href="tutorial_1.html#general-instructions"><i class="fa fa-check"></i>2. General Instructions</a>
<ul>
<li class="chapter" data-level="" data-path="tutorial_1.html"><a href="tutorial_1.html#a-how-to-access-tutorials"><i class="fa fa-check"></i>a) How to access tutorials:</a></li>
<li class="chapter" data-level="" data-path="tutorial_1.html"><a href="tutorial_1.html#b-how-to-complete-tutorial"><i class="fa fa-check"></i>b) How to complete tutorial:</a></li>
<li class="chapter" data-level="" data-path="tutorial_1.html"><a href="tutorial_1.html#c-how-to-submit-answers-participating-institutions-only"><i class="fa fa-check"></i>c) How to submit answers (participating institutions only):</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="" data-path="WE_1.html"><a href="WE_1.html"><i class="fa fa-check"></i>Worked Example</a>
<ul>
<li class="chapter" data-level="" data-path="WE_1.html"><a href="WE_1.html#overview-of-worked-example"><i class="fa fa-check"></i>1. Overview of Worked Example</a>
<ul>
<li class="chapter" data-level="" data-path="WE_1.html"><a href="WE_1.html#a.-goals"><i class="fa fa-check"></i>a. Goals</a></li>
<li class="chapter" data-level="" data-path="WE_1.html"><a href="WE_1.html#b.-data-set"><i class="fa fa-check"></i>b. Data set</a></li>
<li class="chapter" data-level="" data-path="WE_1.html"><a href="WE_1.html#c.-required-r-libraries"><i class="fa fa-check"></i>c. Required R libraries</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="WE_1.html"><a href="WE_1.html#import-data-from-.csv-file"><i class="fa fa-check"></i>2. Import data from .csv file</a>
<ul>
<li class="chapter" data-level="" data-path="WE_1.html"><a href="WE_1.html#a.-preparation"><i class="fa fa-check"></i>a. Preparation</a></li>
<li class="chapter" data-level="" data-path="WE_1.html"><a href="WE_1.html#b.-excel-file"><i class="fa fa-check"></i>b. Excel file</a></li>
<li class="chapter" data-level="" data-path="WE_1.html"><a href="WE_1.html#c.-import-with-function-read.csv"><i class="fa fa-check"></i>c. Import with function ‘read.csv’</a></li>
<li class="chapter" data-level="" data-path="WE_1.html"><a href="WE_1.html#d.-create-id-variable"><i class="fa fa-check"></i>d. Create ID variable</a></li>
<li class="chapter" data-level="" data-path="WE_1.html"><a href="WE_1.html#e.-export-with-function-write.csv"><i class="fa fa-check"></i>e. Export with function ‘write.csv’</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="WE_1.html"><a href="WE_1.html#create-genind-object"><i class="fa fa-check"></i>3. Create ‘genind’ Object</a>
<ul>
<li class="chapter" data-level="" data-path="WE_1.html"><a href="WE_1.html#a.-read-the-helpfile"><i class="fa fa-check"></i>a. Read the helpfile</a></li>
<li class="chapter" data-level="" data-path="WE_1.html"><a href="WE_1.html#b.-set-all-arguments-of-function-df2genind"><i class="fa fa-check"></i>b. Set all arguments of function ‘df2genind’</a></li>
<li class="chapter" data-level="" data-path="WE_1.html"><a href="WE_1.html#c.-check-imported-data"><i class="fa fa-check"></i>c. Check imported data</a></li>
<li class="chapter" data-level="" data-path="WE_1.html"><a href="WE_1.html#d.-summarize-genind-object"><i class="fa fa-check"></i>d. Summarize ‘genind’ object</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="WE_1.html"><a href="WE_1.html#view-information-stored-in-genind-object"><i class="fa fa-check"></i>4. View information stored in ‘genind’ object</a>
<ul>
<li><a href="WE_1.html#a.-slot-tab">a. Slot <code>@tab</code></a></li>
<li><a href="WE_1.html#b.-slot-loc.n.all">b. Slot <code>@loc.n.all</code></a></li>
<li><a href="WE_1.html#c.-slot-loc.fac">c. Slot <code>@loc.fac</code></a></li>
<li><a href="WE_1.html#d.-slot-all.names">d. Slot <code>@all.names</code></a></li>
<li class="chapter" data-level="" data-path="WE_1.html"><a href="WE_1.html#e.-further-slots"><i class="fa fa-check"></i>e. Further slots</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="WE_1.html"><a href="WE_1.html#import-data-with-gstudio-package"><i class="fa fa-check"></i>5. Import data with ‘gstudio’ package</a>
<ul>
<li class="chapter" data-level="" data-path="WE_1.html"><a href="WE_1.html#a.-function-read_population"><i class="fa fa-check"></i>a. Function ‘read_population’</a></li>
<li class="chapter" data-level="" data-path="WE_1.html"><a href="WE_1.html#b.-set-arguments"><i class="fa fa-check"></i>b. Set arguments</a></li>
<li class="chapter" data-level="" data-path="WE_1.html"><a href="WE_1.html#c.-check-imported-data-1"><i class="fa fa-check"></i>c. Check imported data</a></li>
<li class="chapter" data-level="" data-path="WE_1.html"><a href="WE_1.html#d.-add-frogid"><i class="fa fa-check"></i>d. Add FrogID</a></li>
<li class="chapter" data-level="" data-path="WE_1.html"><a href="WE_1.html#e.-convert-to-genind-object"><i class="fa fa-check"></i>e. Convert to ‘genind’ object</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="WE_1.html"><a href="WE_1.html#importing-snp-data"><i class="fa fa-check"></i>6. Importing SNP data</a>
<ul>
<li class="chapter" data-level="" data-path="WE_1.html"><a href="WE_1.html#a.-import-from-text-file"><i class="fa fa-check"></i>a. Import from text file</a></li>
<li class="chapter" data-level="" data-path="WE_1.html"><a href="WE_1.html#b.-import-from-vcf-file-to-genind-object"><i class="fa fa-check"></i>b. Import from vcf file to genind object</a></li>
<li class="chapter" data-level="" data-path="WE_1.html"><a href="WE_1.html#c.-import-from-vcf-file-to-genlight-object"><i class="fa fa-check"></i>c. Import from vcf file to genlight object</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="" data-path="r-exercise-week-1.html"><a href="r-exercise-week-1.html"><i class="fa fa-check"></i>R exercise Week 1</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="Week2.html"><a href="Week2.html"><i class="fa fa-check"></i>Week 2: Spatial Data</a>
<ul>
<li class="chapter" data-level="" data-path="video_2.html"><a href="video_2.html"><i class="fa fa-check"></i>View Course Video</a>
<ul>
<li class="chapter" data-level="" data-path="video_2.html"><a href="video_2.html#embedded-video-1"><i class="fa fa-check"></i>1. Embedded Video</a></li>
<li class="chapter" data-level="" data-path="video_2.html"><a href="video_2.html#preview-slides-2"><i class="fa fa-check"></i>2. Preview Slides</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="tutorial_2.html"><a href="tutorial_2.html"><i class="fa fa-check"></i>Interactive Tutorial 2</a>
<ul>
<li class="chapter" data-level="" data-path="tutorial_2.html"><a href="tutorial_2.html#list-of-r-commands-covered-this-week-1"><i class="fa fa-check"></i>1. List of R commands covered this week</a></li>
<li class="chapter" data-level="" data-path="tutorial_2.html"><a href="tutorial_2.html#general-instructions-1"><i class="fa fa-check"></i>2. General Instructions</a>
<ul>
<li class="chapter" data-level="" data-path="tutorial_2.html"><a href="tutorial_2.html#a-how-to-access-tutorials-1"><i class="fa fa-check"></i>a) How to access tutorials:</a></li>
<li class="chapter" data-level="" data-path="tutorial_2.html"><a href="tutorial_2.html#b-how-to-complete-tutorial-1"><i class="fa fa-check"></i>b) How to complete tutorial:</a></li>
<li class="chapter" data-level="" data-path="tutorial_2.html"><a href="tutorial_2.html#c-how-to-submit-answers-participating-institutions-only-1"><i class="fa fa-check"></i>c) How to submit answers (participating institutions only):</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="" data-path="WE_2.html"><a href="WE_2.html"><i class="fa fa-check"></i>Worked Example</a>
<ul>
<li class="chapter" data-level="" data-path="WE_2.html"><a href="WE_2.html#overview-of-worked-example-1"><i class="fa fa-check"></i>1. Overview of Worked Example</a>
<ul>
<li class="chapter" data-level="" data-path="WE_2.html"><a href="WE_2.html#a.-goals-1"><i class="fa fa-check"></i>a. Goals</a></li>
<li class="chapter" data-level="" data-path="WE_2.html"><a href="WE_2.html#b.-data-set-1"><i class="fa fa-check"></i>b. Data set</a></li>
<li class="chapter" data-level="" data-path="WE_2.html"><a href="WE_2.html#c.-required-r-libraries-1"><i class="fa fa-check"></i>c. Required R libraries</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="WE_2.html"><a href="WE_2.html#import-site-data-from-.csv-file"><i class="fa fa-check"></i>2. Import site data from .csv file</a>
<ul>
<li class="chapter" data-level="" data-path="WE_2.html"><a href="WE_2.html#a.-import-data-into-spatialpointsdataframe"><i class="fa fa-check"></i>a. Import data into SpatialPointsDataFrame</a></li>
<li class="chapter" data-level="" data-path="WE_2.html"><a href="WE_2.html#b.-add-spatial-reference-data"><i class="fa fa-check"></i>b. Add spatial reference data</a></li>
<li class="chapter" data-level="" data-path="WE_2.html"><a href="WE_2.html#c.-change-projection"><i class="fa fa-check"></i>c. Change projection</a></li>
<li class="chapter" data-level="" data-path="WE_2.html"><a href="WE_2.html#d.-access-data-in-spatialpointsdataframe"><i class="fa fa-check"></i>d. Access data in ‘SpatialPointsDataFrame’</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="WE_2.html"><a href="WE_2.html#display-raster-data-and-overlay-sampling-locations-extract-data"><i class="fa fa-check"></i>3. Display raster data and overlay sampling locations, extract data</a>
<ul>
<li class="chapter" data-level="" data-path="WE_2.html"><a href="WE_2.html#a.-display-raster-data"><i class="fa fa-check"></i>a. Display raster data</a></li>
<li class="chapter" data-level="" data-path="WE_2.html"><a href="WE_2.html#b.-change-color-ramp-add-sampling-locations"><i class="fa fa-check"></i>b. Change color ramp, add sampling locations</a></li>
<li class="chapter" data-level="" data-path="WE_2.html"><a href="WE_2.html#c.-extract-raster-values-at-sampling-locations"><i class="fa fa-check"></i>c. Extract raster values at sampling locations</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="WE_2.html"><a href="WE_2.html#calculate-landscape-metrics"><i class="fa fa-check"></i>4. Calculate landscape metrics</a>
<ul>
<li class="chapter" data-level="" data-path="WE_2.html"><a href="WE_2.html#a.-calculate-patch--class--and-landscape-level-landscape-metrics"><i class="fa fa-check"></i>a. Calculate patch-, class- and landscape level landscape metrics</a></li>
<li class="chapter" data-level="" data-path="WE_2.html"><a href="WE_2.html#b.-calculate-patch-level-landscape-metrics-for-evergreen-forest"><i class="fa fa-check"></i>b. Calculate patch-level landscape metrics for ‘Evergreen Forest’</a></li>
<li class="chapter" data-level="" data-path="WE_2.html"><a href="WE_2.html#c.-extract-forest-patch-size-at-samplig-locations"><i class="fa fa-check"></i>c. Extract forest patch size at samplig locations</a></li>
<li class="chapter" data-level="" data-path="WE_2.html"><a href="WE_2.html#d.-plot-a-bubble-map-of-forest-patch-size-at-sampling-locations"><i class="fa fa-check"></i>d. Plot a bubble map of forest patch size at sampling locations</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="WE_2.html"><a href="WE_2.html#sample-landscape-metrics-within-buffer-around-sampling-locations"><i class="fa fa-check"></i>5. Sample landscape metrics within buffer around sampling locations</a>
<ul>
<li class="chapter" data-level="" data-path="WE_2.html"><a href="WE_2.html#b.-extract-landscape-metric-of-choice-for-a-single-cover-type-as-vector"><i class="fa fa-check"></i>b. Extract landscape metric of choice for a single cover type (as vector)</a></li>
<li class="chapter" data-level="" data-path="WE_2.html"><a href="WE_2.html#c.-extract-landscape-metric-of-choice-for-all-cover-types-as-data-frame."><i class="fa fa-check"></i>c. Extract landscape metric of choice for all cover types (as data frame).</a></li>
<li class="chapter" data-level="" data-path="WE_2.html"><a href="WE_2.html#d.-extract-all-landscape-metrics-for-a-single-cover-type-as-data-frame"><i class="fa fa-check"></i>d. Extract all landscape metrics for a single cover type (as data frame)</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="" data-path="r-exercise-week-2.html"><a href="r-exercise-week-2.html"><i class="fa fa-check"></i>R Exercise Week 2</a></li>
<li class="chapter" data-level="" data-path="bonus_2a.html"><a href="bonus_2a.html"><i class="fa fa-check"></i>Bonus Vignette</a>
<ul>
<li class="chapter" data-level="" data-path="bonus_2a.html"><a href="bonus_2a.html#overview-2"><i class="fa fa-check"></i>1. Overview</a>
<ul>
<li class="chapter" data-level="" data-path="bonus_2a.html"><a href="bonus_2a.html#a.-goals-2"><i class="fa fa-check"></i>a. Goals</a></li>
<li class="chapter" data-level="" data-path="bonus_2a.html"><a href="bonus_2a.html#b.-required-r-libraries"><i class="fa fa-check"></i>b. Required R libraries</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="bonus_2a.html"><a href="bonus_2a.html#convert-spatialpointsdataframe-to-sf-object"><i class="fa fa-check"></i>2. Convert ‘SpatialPointsDataFrame’ to ‘sf’ object</a></li>
<li class="chapter" data-level="" data-path="bonus_2a.html"><a href="bonus_2a.html#plot-variables-first-10-in-data-set"><i class="fa fa-check"></i>3. Plot variables (first 10 in data set)</a></li>
<li class="chapter" data-level="" data-path="bonus_2a.html"><a href="bonus_2a.html#export-site-data-as-esri-shapefile"><i class="fa fa-check"></i>4. Export site data as ESRI shapefile</a></li>
<li class="chapter" data-level="" data-path="bonus_2a.html"><a href="bonus_2a.html#usterra-and-tmap-to-display-categorical-map-with-color-scheme"><i class="fa fa-check"></i>5. Us’terra’ and ‘tmap’ to display categorical map with color scheme</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="" data-path="Week3.html"><a href="Week3.html"><i class="fa fa-check"></i>Week 3: Genetic Diversity</a>
<ul>
<li class="chapter" data-level="" data-path="video_3.html"><a href="video_3.html"><i class="fa fa-check"></i>View Course Video</a>
<ul>
<li class="chapter" data-level="" data-path="video_3.html"><a href="video_3.html#embedded-video-2"><i class="fa fa-check"></i>1. Embedded Video</a></li>
<li class="chapter" data-level="" data-path="video_3.html"><a href="video_3.html#video-part-1"><i class="fa fa-check"></i>Video, Part 1</a></li>
<li class="chapter" data-level="" data-path="video_3.html"><a href="video_3.html#video-part-2"><i class="fa fa-check"></i>Video, Part 2</a></li>
<li class="chapter" data-level="" data-path="video_3.html"><a href="video_3.html#preview-slides-3"><i class="fa fa-check"></i>Preview Slides</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="tutorial_3.html"><a href="tutorial_3.html"><i class="fa fa-check"></i>Interactive Tutorial 3</a>
<ul>
<li class="chapter" data-level="" data-path="tutorial_3.html"><a href="tutorial_3.html#list-of-r-commands-covered-this-week-2"><i class="fa fa-check"></i>1. List of R commands covered this week</a></li>
<li class="chapter" data-level="" data-path="tutorial_3.html"><a href="tutorial_3.html#general-instructions-2"><i class="fa fa-check"></i>2. General Instructions</a>
<ul>
<li class="chapter" data-level="" data-path="tutorial_3.html"><a href="tutorial_3.html#a-how-to-access-tutorials-2"><i class="fa fa-check"></i>a) How to access tutorials:</a></li>
<li class="chapter" data-level="" data-path="tutorial_3.html"><a href="tutorial_3.html#b-how-to-complete-tutorial-2"><i class="fa fa-check"></i>b) How to complete tutorial:</a></li>
<li class="chapter" data-level="" data-path="tutorial_3.html"><a href="tutorial_3.html#c-how-to-submit-answers-participating-institutions-only-2"><i class="fa fa-check"></i>c) How to submit answers (participating institutions only):</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="" data-path="WE_3.html"><a href="WE_3.html"><i class="fa fa-check"></i>Worked Example</a>
<ul>
<li class="chapter" data-level="" data-path="WE_3.html"><a href="WE_3.html#overview-of-worked-example-2"><i class="fa fa-check"></i>1. Overview of Worked Example</a>
<ul>
<li class="chapter" data-level="" data-path="WE_3.html"><a href="WE_3.html#a.-goals-3"><i class="fa fa-check"></i>a. Goals</a></li>
<li class="chapter" data-level="" data-path="WE_3.html"><a href="WE_3.html#b.-data-set-2"><i class="fa fa-check"></i>b. Data set</a></li>
<li class="chapter" data-level="" data-path="WE_3.html"><a href="WE_3.html#c.-required-r-libraries-2"><i class="fa fa-check"></i>c. Required R libraries</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="WE_3.html"><a href="WE_3.html#basic-checking-of-markers-and-populations"><i class="fa fa-check"></i>2. Basic checking of markers and populations</a>
<ul>
<li class="chapter" data-level="" data-path="WE_3.html"><a href="WE_3.html#a.-re-create-genind-object"><i class="fa fa-check"></i>a. Re-create genind object</a></li>
<li class="chapter" data-level="" data-path="WE_3.html"><a href="WE_3.html#b.-check-that-markers-are-polymorphic"><i class="fa fa-check"></i>b. Check that markers are polymorphic</a></li>
<li class="chapter" data-level="" data-path="WE_3.html"><a href="WE_3.html#c.-check-for-deviations-from-hardy-weinberg-equilibrium-hwe"><i class="fa fa-check"></i>c. Check for deviations from Hardy-Weinberg equilibrium (HWE)</a></li>
<li class="chapter" data-level="" data-path="WE_3.html"><a href="WE_3.html#d.-check-for-linkage-disequilibrium-ld"><i class="fa fa-check"></i>d. Check for linkage disequilibrium (LD)</a></li>
<li class="chapter" data-level="" data-path="WE_3.html"><a href="WE_3.html#e.-check-for-null-alleles"><i class="fa fa-check"></i>e. Check for null alleles</a></li>
<li class="chapter" data-level="" data-path="WE_3.html"><a href="WE_3.html#f.-overall-interpretation"><i class="fa fa-check"></i>f. Overall interpretation</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="WE_3.html"><a href="WE_3.html#assess-genetic-diversity"><i class="fa fa-check"></i>3. Assess genetic diversity</a>
<ul>
<li class="chapter" data-level="" data-path="WE_3.html"><a href="WE_3.html#a.-rarefied-allelic-richness"><i class="fa fa-check"></i>a. Rarefied allelic richness</a></li>
<li class="chapter" data-level="" data-path="WE_3.html"><a href="WE_3.html#b.-observed-and-expected-heterozygosity"><i class="fa fa-check"></i>b. Observed and expected heterozygosity</a></li>
<li class="chapter" data-level="" data-path="WE_3.html"><a href="WE_3.html#c.-create-table-with-sitel-level-genetic-diversity-measures"><i class="fa fa-check"></i>c. Create table with sitel-level genetic diversity measures</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="WE_3.html"><a href="WE_3.html#aggregate-genetic-data-at-population-level-allele-frequencies"><i class="fa fa-check"></i>4. Aggregate genetic data at population level (allele frequencies)</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="r-exercise-week-3.html"><a href="r-exercise-week-3.html"><i class="fa fa-check"></i>R Exercise Week 3</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="Week4.html"><a href="Week4.html"><i class="fa fa-check"></i>Week 4: Metapopulations</a>
<ul>
<li class="chapter" data-level="" data-path="video_4.html"><a href="video_4.html"><i class="fa fa-check"></i>View Course Video</a>
<ul>
<li class="chapter" data-level="" data-path="video_4.html"><a href="video_4.html#embedded-video-3"><i class="fa fa-check"></i>1. Embedded Video</a></li>
<li class="chapter" data-level="" data-path="video_4.html"><a href="video_4.html#video-part-1-1"><i class="fa fa-check"></i>Video, Part 1</a></li>
<li class="chapter" data-level="" data-path="video_4.html"><a href="video_4.html#video-part-2-1"><i class="fa fa-check"></i>Video, Part 2</a></li>
<li class="chapter" data-level="" data-path="video_4.html"><a href="video_4.html#preview-slides-4"><i class="fa fa-check"></i>Preview Slides</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="tutorial_4.html"><a href="tutorial_4.html"><i class="fa fa-check"></i>Interactive Tutorial 4</a>
<ul>
<li class="chapter" data-level="" data-path="tutorial_4.html"><a href="tutorial_4.html#list-of-r-commands-covered-this-week-3"><i class="fa fa-check"></i>1. List of R commands covered this week</a></li>
<li class="chapter" data-level="" data-path="tutorial_4.html"><a href="tutorial_4.html#general-instructions-3"><i class="fa fa-check"></i>2. General Instructions</a>
<ul>
<li class="chapter" data-level="" data-path="tutorial_4.html"><a href="tutorial_4.html#a-how-to-access-tutorials-3"><i class="fa fa-check"></i>a) How to access tutorials:</a></li>
<li class="chapter" data-level="" data-path="tutorial_4.html"><a href="tutorial_4.html#b-how-to-complete-tutorial-3"><i class="fa fa-check"></i>b) How to complete tutorial:</a></li>
<li class="chapter" data-level="" data-path="tutorial_4.html"><a href="tutorial_4.html#c-how-to-submit-answers-participating-institutions-only-3"><i class="fa fa-check"></i>c) How to submit answers (participating institutions only):</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="" data-path="WE_4.html"><a href="WE_4.html"><i class="fa fa-check"></i>Worked Example</a>
<ul>
<li class="chapter" data-level="" data-path="WE_4.html"><a href="WE_4.html#overview-of-worked-example-3"><i class="fa fa-check"></i>1. Overview of Worked Example</a>
<ul>
<li class="chapter" data-level="" data-path="WE_4.html"><a href="WE_4.html#a.-goals-4"><i class="fa fa-check"></i>a. Goals</a></li>
<li class="chapter" data-level="" data-path="WE_4.html"><a href="WE_4.html#b.-data-set-3"><i class="fa fa-check"></i>b. Data set</a></li>
<li class="chapter" data-level="" data-path="WE_4.html"><a href="WE_4.html#reference"><i class="fa fa-check"></i>Reference:</a></li>
<li class="chapter" data-level="" data-path="WE_4.html"><a href="WE_4.html#c.-required-r-libraries-3"><i class="fa fa-check"></i>c. Required R libraries</a></li>
<li class="chapter" data-level="" data-path="WE_4.html"><a href="WE_4.html#d.-import-data"><i class="fa fa-check"></i>d. Import data</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="WE_4.html"><a href="WE_4.html#spatial-distribution-of-genetic-structure"><i class="fa fa-check"></i>2. Spatial distribution of genetic structure</a>
<ul>
<li class="chapter" data-level="" data-path="WE_4.html"><a href="WE_4.html#a.-creating-a-genind-object-with-the-hierarchical-data-set"><i class="fa fa-check"></i>a. Creating a ‘genind’ object with the hierarchical data set</a></li>
<li class="chapter" data-level="" data-path="WE_4.html"><a href="WE_4.html#b.-hierarchical-amova"><i class="fa fa-check"></i>b. Hierarchical AMOVA</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="WE_4.html"><a href="WE_4.html#what-determines-genetic-differentiation-among-sites"><i class="fa fa-check"></i>3. What determines genetic differentiation among sites?</a>
<ul>
<li class="chapter" data-level="" data-path="WE_4.html"><a href="WE_4.html#a.-correlation-matrix"><i class="fa fa-check"></i>a. Correlation matrix</a></li>
<li class="chapter" data-level="" data-path="WE_4.html"><a href="WE_4.html#b.-scatterplots"><i class="fa fa-check"></i>b. Scatterplots</a></li>
<li class="chapter" data-level="" data-path="WE_4.html"><a href="WE_4.html#c.-regression-model"><i class="fa fa-check"></i>c. Regression model</a></li>
<li class="chapter" data-level="" data-path="WE_4.html"><a href="WE_4.html#d.-which-populations-dont-fit-the-general-pattern"><i class="fa fa-check"></i>d. Which populations don’t fit the general pattern?</a></li>
<li class="chapter" data-level="" data-path="WE_4.html"><a href="WE_4.html#e.-regression-model-without-outliers"><i class="fa fa-check"></i>e. Regression model without outliers</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="WE_4.html"><a href="WE_4.html#what-determines-genetic-diversity"><i class="fa fa-check"></i>4. What determines genetic diversity?</a>
<ul>
<li class="chapter" data-level="" data-path="WE_4.html"><a href="WE_4.html#a.-correlation-matrix-1"><i class="fa fa-check"></i>a. Correlation matrix</a></li>
<li class="chapter" data-level="" data-path="WE_4.html"><a href="WE_4.html#b.-regression-models"><i class="fa fa-check"></i>b. Regression models</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="WE_4.html"><a href="WE_4.html#are-genetic-differentiation-and-diversity-related"><i class="fa fa-check"></i>5. Are genetic differentiation and diversity related?</a></li>
<li class="chapter" data-level="" data-path="WE_4.html"><a href="WE_4.html#effect-of-recent-extinction-events"><i class="fa fa-check"></i>6. Effect of recent extinction events</a>
<ul>
<li class="chapter" data-level="" data-path="WE_4.html"><a href="WE_4.html#a.-effect-of-patch-extinction-event-temporal-data-set"><i class="fa fa-check"></i>a. Effect of patch extinction event (temporal data set)</a></li>
<li class="chapter" data-level="" data-path="WE_4.html"><a href="WE_4.html#b.-power-analysis"><i class="fa fa-check"></i>b. Power analysis</a></li>
<li class="chapter" data-level="" data-path="WE_4.html"><a href="WE_4.html#c.-sample-size-calculation"><i class="fa fa-check"></i>c. Sample size calculation</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="" data-path="r-exercise-week-4.html"><a href="r-exercise-week-4.html"><i class="fa fa-check"></i>R Exercise Week 4</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="Week5.html"><a href="Week5.html"><i class="fa fa-check"></i>Week 5: Spatial Statistics</a>
<ul>
<li class="chapter" data-level="" data-path="video_5.html"><a href="video_5.html"><i class="fa fa-check"></i>View Course Video</a>
<ul>
<li class="chapter" data-level="" data-path="video_5.html"><a href="video_5.html#embedded-video-4"><i class="fa fa-check"></i>1. Embedded Video</a></li>
<li class="chapter" data-level="" data-path="video_5.html"><a href="video_5.html#preview-slides-5"><i class="fa fa-check"></i>Preview Slides</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="tutorial_5.html"><a href="tutorial_5.html"><i class="fa fa-check"></i>Interactive Tutorial 5</a>
<ul>
<li class="chapter" data-level="" data-path="tutorial_5.html"><a href="tutorial_5.html#list-of-r-commands-covered-this-week-4"><i class="fa fa-check"></i>1. List of R commands covered this week</a></li>
<li class="chapter" data-level="" data-path="tutorial_5.html"><a href="tutorial_5.html#general-instructions-4"><i class="fa fa-check"></i>2. General Instructions</a>
<ul>
<li class="chapter" data-level="" data-path="tutorial_5.html"><a href="tutorial_5.html#a-how-to-access-tutorials-4"><i class="fa fa-check"></i>a) How to access tutorials:</a></li>
<li class="chapter" data-level="" data-path="tutorial_5.html"><a href="tutorial_5.html#b-how-to-complete-tutorial-4"><i class="fa fa-check"></i>b) How to complete tutorial:</a></li>
<li class="chapter" data-level="" data-path="tutorial_5.html"><a href="tutorial_5.html#c-how-to-submit-answers-participating-institutions-only-4"><i class="fa fa-check"></i>c) How to submit answers (participating institutions only):</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="" data-path="WE_5.html"><a href="WE_5.html"><i class="fa fa-check"></i>Worked Example</a>
<ul>
<li class="chapter" data-level="" data-path="WE_5.html"><a href="WE_5.html#overview-of-worked-example-4"><i class="fa fa-check"></i>1. Overview of Worked Example</a>
<ul>
<li class="chapter" data-level="" data-path="WE_5.html"><a href="WE_5.html#a.-goals-5"><i class="fa fa-check"></i>a. Goals</a></li>
<li class="chapter" data-level="" data-path="WE_5.html"><a href="WE_5.html#b.-data-set-4"><i class="fa fa-check"></i>b. Data set</a></li>
<li class="chapter" data-level="" data-path="WE_5.html"><a href="WE_5.html#c.-required-r-packages"><i class="fa fa-check"></i>c. Required R packages</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="WE_5.html"><a href="WE_5.html#data-import-and-manipulation"><i class="fa fa-check"></i>2. Data import and manipulation</a>
<ul>
<li class="chapter" data-level="" data-path="WE_5.html"><a href="WE_5.html#a.-create-an-ecogen-object"><i class="fa fa-check"></i>a. Create an ecogen object</a></li>
<li class="chapter" data-level="" data-path="WE_5.html"><a href="WE_5.html#b.-aggregate-to-ecopop"><i class="fa fa-check"></i>b. Aggregate to ecopop</a></li>
<li class="chapter" data-level="" data-path="WE_5.html"><a href="WE_5.html#b.-add-site-level-data"><i class="fa fa-check"></i>b. Add site-level data</a></li>
<li class="chapter" data-level="" data-path="WE_5.html"><a href="WE_5.html#c.-export-to-adegenet-and-gstudio"><i class="fa fa-check"></i>c. Export to adegenet and gstudio</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="WE_5.html"><a href="WE_5.html#calculate-genetic-distances"><i class="fa fa-check"></i>3. Calculate genetic distances</a>
<ul>
<li class="chapter" data-level="" data-path="WE_5.html"><a href="WE_5.html#a.-genetic-distances-calculated-from-genind-object"><i class="fa fa-check"></i>a. Genetic distances calculated from genind object</a></li>
<li class="chapter" data-level="" data-path="WE_5.html"><a href="WE_5.html#b.-more-distance-matrices-with-gstudio"><i class="fa fa-check"></i>b. More distance matrices with ‘gstudio’</a></li>
<li class="chapter" data-level="" data-path="WE_5.html"><a href="WE_5.html#c.-assemble-distance-matrices"><i class="fa fa-check"></i>c. Assemble distance matrices</a></li>
<li class="chapter" data-level="" data-path="WE_5.html"><a href="WE_5.html#d.-export-genetic-distance-matrices"><i class="fa fa-check"></i>d. Export genetic distance matrices</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="WE_5.html"><a href="WE_5.html#perform-a-mantel-test-to-test-for-ibd"><i class="fa fa-check"></i>4. Perform a Mantel test to test for IBD</a>
<ul>
<li class="chapter" data-level="" data-path="WE_5.html"><a href="WE_5.html#a.-visually-check-linearity"><i class="fa fa-check"></i>a. Visually check linearity</a></li>
<li class="chapter" data-level="" data-path="WE_5.html"><a href="WE_5.html#b.-perform-mantel-test"><i class="fa fa-check"></i>b. Perform Mantel test</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="WE_5.html"><a href="WE_5.html#create-mantel-correlogram-for-genetic-data"><i class="fa fa-check"></i>5. Create Mantel correlogram for genetic data</a>
<ul>
<li class="chapter" data-level="" data-path="WE_5.html"><a href="WE_5.html#a.-create-a-first-mantel-correlogram"><i class="fa fa-check"></i>a. Create a first Mantel correlogram</a></li>
<li class="chapter" data-level="" data-path="WE_5.html"><a href="WE_5.html#b.-vary-distance-class-definition"><i class="fa fa-check"></i>b. Vary distance class definition</a></li>
<li class="chapter" data-level="" data-path="WE_5.html"><a href="WE_5.html#c.-alternative-measures-of-genetic-distances"><i class="fa fa-check"></i>c. Alternative measures of genetic distances</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="WE_5.html"><a href="WE_5.html#specify-spatial-weights-and-calculate-morans-i"><i class="fa fa-check"></i>6. Specify spatial weights and calculate Moran’s I</a>
<ul>
<li class="chapter" data-level="" data-path="WE_5.html"><a href="WE_5.html#a.-defining-spatial-neighbors"><i class="fa fa-check"></i>a. Defining spatial neighbors</a></li>
<li class="chapter" data-level="" data-path="WE_5.html"><a href="WE_5.html#b.-defining-spatial-weights"><i class="fa fa-check"></i>b. Defining spatial weights</a></li>
<li class="chapter" data-level="" data-path="WE_5.html"><a href="WE_5.html#c.-calculating-and-testing-morans-i"><i class="fa fa-check"></i>c. Calculating and testing Moran’s I</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="" data-path="r-exercise-week-5.html"><a href="r-exercise-week-5.html"><i class="fa fa-check"></i>R Exercise Week 5</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="Week6.html"><a href="Week6.html"><i class="fa fa-check"></i>Week 6: Quantitative Genetics</a>
<ul>
<li class="chapter" data-level="" data-path="video_6.html"><a href="video_6.html"><i class="fa fa-check"></i>View Course Video</a>
<ul>
<li class="chapter" data-level="" data-path="video_6.html"><a href="video_6.html#embedded-video-5"><i class="fa fa-check"></i>1. Embedded Video</a></li>
<li class="chapter" data-level="" data-path="video_6.html"><a href="video_6.html#video-part-1-2"><i class="fa fa-check"></i>Video, Part 1</a></li>
<li class="chapter" data-level="" data-path="video_6.html"><a href="video_6.html#video-part-2-2"><i class="fa fa-check"></i>Video, Part 2</a></li>
<li class="chapter" data-level="" data-path="video_6.html"><a href="video_6.html#preview-slides-6"><i class="fa fa-check"></i>Preview Slides</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="tutorial_6.html"><a href="tutorial_6.html"><i class="fa fa-check"></i>Interactive Tutorial 6</a>
<ul>
<li class="chapter" data-level="" data-path="tutorial_6.html"><a href="tutorial_6.html#list-of-r-commands-covered-this-week-5"><i class="fa fa-check"></i>1. List of R commands covered this week</a></li>
<li class="chapter" data-level="" data-path="tutorial_6.html"><a href="tutorial_6.html#general-instructions-5"><i class="fa fa-check"></i>2. General Instructions</a>
<ul>
<li class="chapter" data-level="" data-path="tutorial_6.html"><a href="tutorial_6.html#a-how-to-access-tutorials-5"><i class="fa fa-check"></i>a) How to access tutorials:</a></li>
<li class="chapter" data-level="" data-path="tutorial_6.html"><a href="tutorial_6.html#b-how-to-complete-tutorial-5"><i class="fa fa-check"></i>b) How to complete tutorial:</a></li>
<li class="chapter" data-level="" data-path="tutorial_6.html"><a href="tutorial_6.html#c-how-to-submit-answers-participating-institutions-only-5"><i class="fa fa-check"></i>c) How to submit answers (participating institutions only):</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="" data-path="WE_6.html"><a href="WE_6.html"><i class="fa fa-check"></i>Worked Example</a>
<ul>
<li class="chapter" data-level="" data-path="WE_6.html"><a href="WE_6.html#overview-of-worked-example-5"><i class="fa fa-check"></i>1. Overview of Worked Example</a>
<ul>
<li class="chapter" data-level="" data-path="WE_6.html"><a href="WE_6.html#a.-goals-6"><i class="fa fa-check"></i>a. Goals</a></li>
<li class="chapter" data-level="" data-path="WE_6.html"><a href="WE_6.html#b.-data-set-5"><i class="fa fa-check"></i>b. Data set</a></li>
<li class="chapter" data-level="" data-path="WE_6.html"><a href="WE_6.html#c.-required-r-packages-1"><i class="fa fa-check"></i>c. Required R packages</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="WE_6.html"><a href="WE_6.html#estimate-genetic-and-non-genetic-variance-components-from-a-common-garden-experiment"><i class="fa fa-check"></i>2. Estimate genetic and non-genetic variance components from a common garden experiment</a>
<ul>
<li class="chapter" data-level="" data-path="WE_6.html"><a href="WE_6.html#a.-import-phenytypic-data"><i class="fa fa-check"></i>a. Import phenytypic data</a></li>
<li class="chapter" data-level="" data-path="WE_6.html"><a href="WE_6.html#b.-fit-linear-mixed-effects-models-lmm-to-trait-data-observed-on-families"><i class="fa fa-check"></i>b. Fit linear mixed effects models (LMM) to trait data observed on families</a></li>
<li class="chapter" data-level="" data-path="WE_6.html"><a href="WE_6.html#c.-compare-model-fit"><i class="fa fa-check"></i>c. Compare model fit</a></li>
<li class="chapter" data-level="" data-path="WE_6.html"><a href="WE_6.html#d.-check-model-validity"><i class="fa fa-check"></i>d. Check model validity</a></li>
<li class="chapter" data-level="" data-path="WE_6.html"><a href="WE_6.html#e.-estimate-variance-components"><i class="fa fa-check"></i>e. Estimate variance components</a></li>
<li class="chapter" data-level="" data-path="WE_6.html"><a href="WE_6.html#f.-significance-testing"><i class="fa fa-check"></i>f. Significance testing</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="WE_6.html"><a href="WE_6.html#estimate-trait-heritability"><i class="fa fa-check"></i>3. Estimate trait heritability</a>
<ul>
<li class="chapter" data-level="" data-path="WE_6.html"><a href="WE_6.html#a.-estimate-heritability"><i class="fa fa-check"></i>a. Estimate heritability</a></li>
<li class="chapter" data-level="" data-path="WE_6.html"><a href="WE_6.html#b.-estimate-trait-differentiation"><i class="fa fa-check"></i>b. Estimate trait differentiation</a></li>
</ul></li>
<li><a href="WE_6.html#compare-q_st-to-f_st-measured-from-snp-data">4. Compare <span class="math inline">\(Q_{st}\)</span> to <span class="math inline">\(F_{st}\)</span> measured from SNP data</a>
<ul>
<li class="chapter" data-level="" data-path="WE_6.html"><a href="WE_6.html#a.-import-ecogen-object"><i class="fa fa-check"></i>a. Import ‘ecogen’ object</a></li>
<li><a href="WE_6.html#b.-compare-q_st-to-f_st">b. Compare <span class="math inline">\(Q_{ST}\)</span> to <span class="math inline">\(F_{ST}\)</span></a></li>
</ul></li>
<li class="chapter" data-level="" data-path="WE_6.html"><a href="WE_6.html#assess-correlation-between-trait-and-environment"><i class="fa fa-check"></i>5. Assess correlation between trait and environment</a>
<ul>
<li class="chapter" data-level="" data-path="WE_6.html"><a href="WE_6.html#a.-describe-and-extract-effects"><i class="fa fa-check"></i>a. Describe and extract effects</a></li>
<li class="chapter" data-level="" data-path="WE_6.html"><a href="WE_6.html#b.-correlation-of-population-effect-with-environment"><i class="fa fa-check"></i>b. Correlation of population effect with environment</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="" data-path="r-exercise-week-6.html"><a href="r-exercise-week-6.html"><i class="fa fa-check"></i>R Exercise Week 6</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="Week7.html"><a href="Week7.html"><i class="fa fa-check"></i>Week 7: Spatial Linear Models</a>
<ul>
<li class="chapter" data-level="" data-path="video_7.html"><a href="video_7.html"><i class="fa fa-check"></i>View Course Video</a>
<ul>
<li class="chapter" data-level="" data-path="video_7.html"><a href="video_7.html#embedded-video-6"><i class="fa fa-check"></i>1. Embedded Video</a></li>
<li class="chapter" data-level="" data-path="video_7.html"><a href="video_7.html#video-part-1-3"><i class="fa fa-check"></i>Video, Part 1</a></li>
<li class="chapter" data-level="" data-path="video_7.html"><a href="video_7.html#video-part-2-3"><i class="fa fa-check"></i>Video, Part 2</a></li>
<li class="chapter" data-level="" data-path="video_7.html"><a href="video_7.html#preview-slides-7"><i class="fa fa-check"></i>Preview Slides</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="tutorial_7.html"><a href="tutorial_7.html"><i class="fa fa-check"></i>Interactive Tutorial 7</a>
<ul>
<li class="chapter" data-level="" data-path="tutorial_7.html"><a href="tutorial_7.html#list-of-r-commands-covered-this-week-6"><i class="fa fa-check"></i>1. List of R commands covered this week</a></li>
<li class="chapter" data-level="" data-path="tutorial_7.html"><a href="tutorial_7.html#general-instructions-6"><i class="fa fa-check"></i>2. General Instructions</a>
<ul>
<li class="chapter" data-level="" data-path="tutorial_7.html"><a href="tutorial_7.html#a-how-to-access-tutorials-6"><i class="fa fa-check"></i>a) How to access tutorials:</a></li>
<li class="chapter" data-level="" data-path="tutorial_7.html"><a href="tutorial_7.html#b-how-to-complete-tutorial-6"><i class="fa fa-check"></i>b) How to complete tutorial:</a></li>
<li class="chapter" data-level="" data-path="tutorial_7.html"><a href="tutorial_7.html#c-how-to-submit-answers-participating-institutions-only-6"><i class="fa fa-check"></i>c) How to submit answers (participating institutions only):</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="" data-path="WE_7.html"><a href="WE_7.html"><i class="fa fa-check"></i>Worked Example</a>
<ul>
<li class="chapter" data-level="" data-path="WE_7.html"><a href="WE_7.html#overview-of-worked-example-6"><i class="fa fa-check"></i>1. Overview of Worked Example</a>
<ul>
<li class="chapter" data-level="" data-path="WE_7.html"><a href="WE_7.html#a.-goals-7"><i class="fa fa-check"></i>a. Goals</a></li>
<li class="chapter" data-level="" data-path="WE_7.html"><a href="WE_7.html#b.-data-set-6"><i class="fa fa-check"></i>b. Data set</a></li>
<li class="chapter" data-level="" data-path="WE_7.html"><a href="WE_7.html#c.-required-r-packages-2"><i class="fa fa-check"></i>c. Required R packages</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="WE_7.html"><a href="WE_7.html#explore-data-set"><i class="fa fa-check"></i>2. Explore data set</a>
<ul>
<li class="chapter" data-level="" data-path="WE_7.html"><a href="WE_7.html#a.-import-data"><i class="fa fa-check"></i>a. Import data</a></li>
<li class="chapter" data-level="" data-path="WE_7.html"><a href="WE_7.html#b.-create-a-map"><i class="fa fa-check"></i>b. Create a map</a></li>
<li class="chapter" data-level="" data-path="WE_7.html"><a href="WE_7.html#c.-explore-correlations"><i class="fa fa-check"></i>c. Explore correlations</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="WE_7.html"><a href="WE_7.html#test-regression-residuals-for-spatial-autocorrelation"><i class="fa fa-check"></i>3. Test regression residuals for spatial autocorrelation</a>
<ul>
<li class="chapter" data-level="" data-path="WE_7.html"><a href="WE_7.html#a.-fit-regression-models"><i class="fa fa-check"></i>a. Fit regression models</a></li>
<li class="chapter" data-level="" data-path="WE_7.html"><a href="WE_7.html#b.-test-for-spatial-autocorrelation-morans-i"><i class="fa fa-check"></i>b. Test for spatial autocorrelation (Moran’s I):</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="WE_7.html"><a href="WE_7.html#fit-models-with-spatially-correlated-error-gls-with-package-nlme"><i class="fa fa-check"></i>4. Fit models with spatially correlated error (GLS) with package ‘nlme’</a>
<ul>
<li class="chapter" data-level="" data-path="WE_7.html"><a href="WE_7.html#a.-plot-empirical-variogram"><i class="fa fa-check"></i>a. Plot empirical variogram</a></li>
<li class="chapter" data-level="" data-path="WE_7.html"><a href="WE_7.html#b.-fit-variogram-models"><i class="fa fa-check"></i>b. Fit variogram models</a></li>
<li class="chapter" data-level="" data-path="WE_7.html"><a href="WE_7.html#c.-select-best-fitting-model"><i class="fa fa-check"></i>c. Select best-fitting model</a></li>
<li class="chapter" data-level="" data-path="WE_7.html"><a href="WE_7.html#d.-plot-fitted-variogram-model"><i class="fa fa-check"></i>d. Plot fitted variogram model</a></li>
<li class="chapter" data-level="" data-path="WE_7.html"><a href="WE_7.html#e.-add-random-factor"><i class="fa fa-check"></i>e. Add random factor</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="WE_7.html"><a href="WE_7.html#fit-spatial-simultaneous-autoregressive-error-models-sar"><i class="fa fa-check"></i>5. Fit spatial simultaneous autoregressive error models (SAR)</a>
<ul>
<li class="chapter" data-level="" data-path="WE_7.html"><a href="WE_7.html#a.-fit-and-compare-alternative-sar-models"><i class="fa fa-check"></i>a. Fit and compare alternative SAR models</a></li>
<li class="chapter" data-level="" data-path="WE_7.html"><a href="WE_7.html#b.-interpret-best-fitting-sar-model"><i class="fa fa-check"></i>b. Interpret best-fitting SAR model</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="WE_7.html"><a href="WE_7.html#spatial-filtering-with-mem-using-package-spmoran"><i class="fa fa-check"></i>6. Spatial filtering with MEM using package ‘spmoran’</a>
<ul>
<li class="chapter" data-level="" data-path="WE_7.html"><a href="WE_7.html#a.-default-method"><i class="fa fa-check"></i>a. Default method</a></li>
<li class="chapter" data-level="" data-path="WE_7.html"><a href="WE_7.html#b.-using-a-custom-connectivity-matrix"><i class="fa fa-check"></i>b. Using a custom connectivity matrix</a></li>
<li class="chapter" data-level="" data-path="WE_7.html"><a href="WE_7.html#c.-plot-spatial-eigenvectors"><i class="fa fa-check"></i>c. Plot spatial eigenvectors</a></li>
<li class="chapter" data-level="" data-path="WE_7.html"><a href="WE_7.html#d.-random-effect-model"><i class="fa fa-check"></i>d. Random effect model</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="WE_7.html"><a href="WE_7.html#fit-spatially-varying-coefficients-model-with-package-spmoran"><i class="fa fa-check"></i>7. Fit spatially varying coefficients model with package ‘spmoran’</a></li>
<li class="chapter" data-level="" data-path="WE_7.html"><a href="WE_7.html#model-with-patchsize-and-ibr"><i class="fa fa-check"></i>Model with PatchSize and IBR</a></li>
<li class="chapter" data-level="" data-path="WE_7.html"><a href="WE_7.html#model-with-ibr-only"><i class="fa fa-check"></i>Model with IBR only</a></li>
<li class="chapter" data-level="" data-path="WE_7.html"><a href="WE_7.html#conclusions"><i class="fa fa-check"></i>8. Conclusions</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="r-exercise-week-7.html"><a href="r-exercise-week-7.html"><i class="fa fa-check"></i>R Exercise Week 7</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="Week8.html"><a href="Week8.html"><i class="fa fa-check"></i>Week 8: Simulation Experiments</a>
<ul>
<li class="chapter" data-level="" data-path="video_8.html"><a href="video_8.html"><i class="fa fa-check"></i>View Course Video</a>
<ul>
<li class="chapter" data-level="" data-path="video_8.html"><a href="video_8.html#embedded-video-7"><i class="fa fa-check"></i>1. Embedded Video</a></li>
<li class="chapter" data-level="" data-path="video_8.html"><a href="video_8.html#video-part-1-4"><i class="fa fa-check"></i>Video, Part 1</a></li>
<li class="chapter" data-level="" data-path="video_8.html"><a href="video_8.html#video-part-2-4"><i class="fa fa-check"></i>Video, Part 2</a></li>
<li class="chapter" data-level="" data-path="video_8.html"><a href="video_8.html#preview-slides-8"><i class="fa fa-check"></i>Preview Slides</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="tutorial_8.html"><a href="tutorial_8.html"><i class="fa fa-check"></i>Interactive Tutorial 8</a>
<ul>
<li class="chapter" data-level="" data-path="tutorial_8.html"><a href="tutorial_8.html#list-of-r-commands-covered-this-week-7"><i class="fa fa-check"></i>1. List of R commands covered this week</a></li>
<li class="chapter" data-level="" data-path="tutorial_8.html"><a href="tutorial_8.html#general-instructions-7"><i class="fa fa-check"></i>2. General Instructions</a>
<ul>
<li class="chapter" data-level="" data-path="tutorial_8.html"><a href="tutorial_8.html#a-how-to-access-tutorials-7"><i class="fa fa-check"></i>a) How to access tutorials:</a></li>
<li class="chapter" data-level="" data-path="tutorial_8.html"><a href="tutorial_8.html#b-how-to-complete-tutorial-7"><i class="fa fa-check"></i>b) How to complete tutorial:</a></li>
<li class="chapter" data-level="" data-path="tutorial_8.html"><a href="tutorial_8.html#c-how-to-submit-answers-participating-institutions-only-7"><i class="fa fa-check"></i>c) How to submit answers (participating institutions only):</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="" data-path="WE_8.html"><a href="WE_8.html"><i class="fa fa-check"></i>Worked Example</a>
<ul>
<li class="chapter" data-level="" data-path="WE_8.html"><a href="WE_8.html#overview-of-worked-example-7"><i class="fa fa-check"></i>1. Overview of Worked Example</a>
<ul>
<li class="chapter" data-level="" data-path="WE_8.html"><a href="WE_8.html#a.-goals-8"><i class="fa fa-check"></i>a. Goals</a></li>
<li class="chapter" data-level="" data-path="WE_8.html"><a href="WE_8.html#b.-data-set-7"><i class="fa fa-check"></i>b. Data set</a></li>
<li class="chapter" data-level="" data-path="WE_8.html"><a href="WE_8.html#c.-required-r-libraries-4"><i class="fa fa-check"></i>c. Required R libraries</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="WE_8.html"><a href="WE_8.html#initialize-a-landscape"><i class="fa fa-check"></i>2. Initialize a landscape</a>
<ul>
<li class="chapter" data-level="" data-path="WE_8.html"><a href="WE_8.html#a.-create-a-random-landscape"><i class="fa fa-check"></i>a. Create a random landscape</a></li>
<li class="chapter" data-level="" data-path="WE_8.html"><a href="WE_8.html#b.-add-populations-to-the-landscape-using-minimal-distance"><i class="fa fa-check"></i>b. Add populations to the landscape (using minimal distance)</a></li>
<li class="chapter" data-level="" data-path="WE_8.html"><a href="WE_8.html#c.-initialise-a-metapopulation"><i class="fa fa-check"></i>c. Initialise a metapopulation</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="WE_8.html"><a href="WE_8.html#define-simulation-parameters"><i class="fa fa-check"></i>3. Define simulation parameters</a>
<ul>
<li class="chapter" data-level="" data-path="WE_8.html"><a href="WE_8.html#a.-define-your-metapopulation"><i class="fa fa-check"></i>a. Define your metapopulation</a></li>
<li class="chapter" data-level="" data-path="WE_8.html"><a href="WE_8.html#b.-initialise-your-population-on-the-landscape"><i class="fa fa-check"></i>b. Initialise your population on the landscape</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="WE_8.html"><a href="WE_8.html#run-simulations-and-analyze-results"><i class="fa fa-check"></i>4. Run simulations and analyze results</a>
<ul>
<li class="chapter" data-level="" data-path="WE_8.html"><a href="WE_8.html#a.-run-your-simulation-over-multiple-time-steps-years"><i class="fa fa-check"></i>a. Run your simulation over multiple time steps (years)</a></li>
<li class="chapter" data-level="" data-path="WE_8.html"><a href="WE_8.html#b.-analyse-your-simulated-population-with-a-partial-mantel-test"><i class="fa fa-check"></i>b. Analyse your simulated population with a partial Mantel test</a></li>
<li class="chapter" data-level="" data-path="WE_8.html"><a href="WE_8.html#c.-optional-analyze-your-simulated-populations-using-sunder"><i class="fa fa-check"></i>c. Optional: Analyze your simulated populations using ‘Sunder’</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="WE_8.html"><a href="WE_8.html#run-simulator-using-a-previously-defined-parameter-set"><i class="fa fa-check"></i>5. Run simulator using a previously defined parameter set</a>
<ul>
<li class="chapter" data-level="" data-path="WE_8.html"><a href="WE_8.html#a.-specify-and-record-the-parameter-combinations"><i class="fa fa-check"></i>a. Specify and record the parameter combinations</a></li>
<li class="chapter" data-level="" data-path="WE_8.html"><a href="WE_8.html#b.-run-the-simulator-over-every-parameter-combination."><i class="fa fa-check"></i>b. Run the simulator over every parameter combination.</a></li>
<li class="chapter" data-level="" data-path="WE_8.html"><a href="WE_8.html#c.-save-your-complete-simulation-run-input-and-output"><i class="fa fa-check"></i>c. Save your complete simulation run (input and output)</a></li>
<li class="chapter" data-level="" data-path="WE_8.html"><a href="WE_8.html#d.-analyze-and-synthesize-results"><i class="fa fa-check"></i>d. Analyze and synthesize results</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="" data-path="r-exercise-week-8.html"><a href="r-exercise-week-8.html"><i class="fa fa-check"></i>R Exercise Week 8</a></li>
<li class="chapter" data-level="" data-path="bonus_8a.html"><a href="bonus_8a.html"><i class="fa fa-check"></i>Bonus Vignette</a>
<ul>
<li class="chapter" data-level="" data-path="bonus_8a.html"><a href="bonus_8a.html#overview-of-bonus-material"><i class="fa fa-check"></i>1. Overview of Bonus Material</a>
<ul>
<li class="chapter" data-level="" data-path="bonus_8a.html"><a href="bonus_8a.html#a.-goals-9"><i class="fa fa-check"></i>a. Goals</a></li>
<li class="chapter" data-level="" data-path="bonus_8a.html"><a href="bonus_8a.html#b.-data-set-8"><i class="fa fa-check"></i>b. Data set</a></li>
<li class="chapter" data-level="" data-path="bonus_8a.html"><a href="bonus_8a.html#c.-required-r-libraries-5"><i class="fa fa-check"></i>c. Required R libraries</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="bonus_8a.html"><a href="bonus_8a.html#navigating-the-file-system"><i class="fa fa-check"></i>2. Navigating the file system</a>
<ul>
<li class="chapter" data-level="" data-path="bonus_8a.html"><a href="bonus_8a.html#a.-where-am-i"><i class="fa fa-check"></i>a. Where am I?</a></li>
<li class="chapter" data-level="" data-path="bonus_8a.html"><a href="bonus_8a.html#b.-accessing-a-system-file"><i class="fa fa-check"></i>b. Accessing a system file</a></li>
<li class="chapter" data-level="" data-path="bonus_8a.html"><a href="bonus_8a.html#c.-file-manipulation"><i class="fa fa-check"></i>c. File manipulation</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="bonus_8a.html"><a href="bonus_8a.html#benchmarking-file-import-and-export-options"><i class="fa fa-check"></i>3. Benchmarking file import and export options</a>
<ul>
<li class="chapter" data-level="" data-path="bonus_8a.html"><a href="bonus_8a.html#a.-benchmark-methods-for-importing-csv-files"><i class="fa fa-check"></i>a. Benchmark methods for importing csv files</a></li>
<li class="chapter" data-level="" data-path="bonus_8a.html"><a href="bonus_8a.html#b.-check-handling-of-character-data"><i class="fa fa-check"></i>b. Check handling of character data</a></li>
<li class="chapter" data-level="" data-path="bonus_8a.html"><a href="bonus_8a.html#c.-binary-files"><i class="fa fa-check"></i>c. Binary files</a></li>
<li class="chapter" data-level="" data-path="bonus_8a.html"><a href="bonus_8a.html#d.-should-you-save-your-r-workspace"><i class="fa fa-check"></i>d. Should you save your R workspace?</a></li>
<li class="chapter" data-level="" data-path="bonus_8a.html"><a href="bonus_8a.html#e.-compile-your-functions"><i class="fa fa-check"></i>e. Compile your functions</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="bonus_8a.html"><a href="bonus_8a.html#profiling-your-code"><i class="fa fa-check"></i>4. Profiling your code</a>
<ul>
<li class="chapter" data-level="" data-path="bonus_8a.html"><a href="bonus_8a.html#a.-named-chunks"><i class="fa fa-check"></i>a. Named chunks</a></li>
<li class="chapter" data-level="" data-path="bonus_8a.html"><a href="bonus_8a.html#b.-profiling-some-lines-of-code-with-profvis"><i class="fa fa-check"></i>b. Profiling some lines of code with ‘profvis’</a></li>
<li class="chapter" data-level="" data-path="bonus_8a.html"><a href="bonus_8a.html#c.-profiling-an-r-script-with-profvis"><i class="fa fa-check"></i>c. Profiling an R script with ‘profvis’</a></li>
<li class="chapter" data-level="" data-path="bonus_8a.html"><a href="bonus_8a.html#d.-converting-an-r-notebook-to-an-r-script"><i class="fa fa-check"></i>d. Converting an R Notebook to an R script</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="bonus_8a.html"><a href="bonus_8a.html#creating-and-executing-a-bash-r-script"><i class="fa fa-check"></i>5. Creating and executing a Bash R script</a>
<ul>
<li class="chapter" data-level="" data-path="bonus_8a.html"><a href="bonus_8a.html#a.-run-r-script-directly-in-the-terminal"><i class="fa fa-check"></i>a. Run R script directly in the Terminal</a></li>
<li class="chapter" data-level="" data-path="bonus_8a.html"><a href="bonus_8a.html#b.-create-a-bash-r-script"><i class="fa fa-check"></i>b. Create a Bash R script</a></li>
<li class="chapter" data-level="" data-path="bonus_8a.html"><a href="bonus_8a.html#c.-executing-a-bash-r-script"><i class="fa fa-check"></i>c. Executing a Bash R script</a></li>
<li class="chapter" data-level="" data-path="bonus_8a.html"><a href="bonus_8a.html#d.-moving-to-a-node-or-cluster"><i class="fa fa-check"></i>d. Moving to a node or cluster?</a></li>
<li class="chapter" data-level="" data-path="bonus_8a.html"><a href="bonus_8a.html#e.-store-your-session-info-and-package-versions"><i class="fa fa-check"></i>e. Store your session info and package versions</a></li>
<li class="chapter" data-level="" data-path="bonus_8a.html"><a href="bonus_8a.html#f.-potential-windows-issues"><i class="fa fa-check"></i>f. Potential Windows issues</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="bonus_8a.html"><a href="bonus_8a.html#parallelizing-code"><i class="fa fa-check"></i>6. Parallelizing code</a>
<ul>
<li class="chapter" data-level="" data-path="bonus_8a.html"><a href="bonus_8a.html#a.-replace-lapply-by-mclapply-with-package-parallel"><i class="fa fa-check"></i>a. Replace ‘lapply’ by ‘mclapply’ with package ‘parallel’</a></li>
<li class="chapter" data-level="" data-path="bonus_8a.html"><a href="bonus_8a.html#b.-replace-for-by-foreach-with-package-doparallel"><i class="fa fa-check"></i>b. Replace ‘for’ by ‘foreach’ with package ‘doParallel’</a></li>
</ul></li>
</ul></li>
</ul></li>
<li class="part"><span><b>Advanced Topics</b></span></li>
<li class="chapter" data-level="" data-path="Week9.html"><a href="Week9.html"><i class="fa fa-check"></i>Week 9: Population Structure</a>
<ul>
<li class="chapter" data-level="" data-path="WE_9.html"><a href="WE_9.html"><i class="fa fa-check"></i>Worked Example</a>
<ul>
<li class="chapter" data-level="" data-path="WE_9.html"><a href="WE_9.html#overview-of-worked-example-8"><i class="fa fa-check"></i>1. Overview of Worked Example</a>
<ul>
<li class="chapter" data-level="" data-path="WE_9.html"><a href="WE_9.html#a.-goals-10"><i class="fa fa-check"></i>a. Goals</a></li>
<li class="chapter" data-level="" data-path="WE_9.html"><a href="WE_9.html#b.-data-sets"><i class="fa fa-check"></i>b. Data sets</a></li>
<li class="chapter" data-level="" data-path="WE_9.html"><a href="WE_9.html#c.-required-r-libraries-6"><i class="fa fa-check"></i>c. Required R libraries</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="WE_9.html"><a href="WE_9.html#simulated-data-2-island-model"><i class="fa fa-check"></i>2. Simulated data: 2-island model</a>
<ul>
<li class="chapter" data-level="" data-path="WE_9.html"><a href="WE_9.html#a.-import-data-1"><i class="fa fa-check"></i>a. Import data</a></li>
<li class="chapter" data-level="" data-path="WE_9.html"><a href="WE_9.html#b.-perform-principal-components-analysis-pca"><i class="fa fa-check"></i>b. Perform Principal Components Analysis (PCA)</a></li>
<li class="chapter" data-level="" data-path="WE_9.html"><a href="WE_9.html#c.-create-synthetic-spatial-coordinates-xy-and-map-them"><i class="fa fa-check"></i>c. Create synthetic spatial coordinates (X,Y) and map them</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="WE_9.html"><a href="WE_9.html#simulated-data-2-island-model-with-admixture"><i class="fa fa-check"></i>3. Simulated data: 2-island model with admixture</a></li>
<li class="chapter" data-level="" data-path="WE_9.html"><a href="WE_9.html#empirical-data-threespine-sticklebacks"><i class="fa fa-check"></i>4. Empirical data: Threespine sticklebacks</a>
<ul>
<li class="chapter" data-level="" data-path="WE_9.html"><a href="WE_9.html#a.-import-the-data"><i class="fa fa-check"></i>a. Import the data:</a></li>
<li class="chapter" data-level="" data-path="WE_9.html"><a href="WE_9.html#b.-examine-population-structure-with-pca"><i class="fa fa-check"></i>b. Examine population structure with PCA</a></li>
<li class="chapter" data-level="" data-path="WE_9.html"><a href="WE_9.html#c.-clustering-with-snmf-similar-to-structure"><i class="fa fa-check"></i>c. Clustering with SNMF (similar to ‘STRUCTURE’)</a></li>
<li class="chapter" data-level="" data-path="WE_9.html"><a href="WE_9.html#d.-plot-ancestry-proportions"><i class="fa fa-check"></i>d. Plot ancestry proportions</a></li>
<li class="chapter" data-level="" data-path="WE_9.html"><a href="WE_9.html#e.-visualize-admixture-proportions-on-a-map"><i class="fa fa-check"></i>e. Visualize admixture proportions on a map</a></li>
</ul></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="" data-path="Week10.html"><a href="Week10.html"><i class="fa fa-check"></i>Week 10: Landscape Resistance</a>
<ul>
<li class="chapter" data-level="" data-path="WE_10.html"><a href="WE_10.html"><i class="fa fa-check"></i>Worked Example</a>
<ul>
<li class="chapter" data-level="" data-path="WE_10.html"><a href="WE_10.html#overview-of-worked-example-9"><i class="fa fa-check"></i>1. Overview of Worked Example</a>
<ul>
<li class="chapter" data-level="" data-path="WE_10.html"><a href="WE_10.html#a.-goals-11"><i class="fa fa-check"></i>a. Goals</a></li>
<li class="chapter" data-level="" data-path="WE_10.html"><a href="WE_10.html#b.-data-set-9"><i class="fa fa-check"></i>b. Data set</a></li>
<li class="chapter" data-level="" data-path="WE_10.html"><a href="WE_10.html#c.-required-r-libraries-7"><i class="fa fa-check"></i>c. Required R libraries</a></li>
<li class="chapter" data-level="" data-path="WE_10.html"><a href="WE_10.html#d.-import-rasters-gsp-err-cti-and-ffp."><i class="fa fa-check"></i>d. Import rasters (gsp, err, cti, and ffp).</a></li>
<li class="chapter" data-level="" data-path="WE_10.html"><a href="WE_10.html#e.-import-site-data-with-sampling-locations"><i class="fa fa-check"></i>e. Import site data with sampling locations</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="WE_10.html"><a href="WE_10.html#explore-the-data-set"><i class="fa fa-check"></i>2. Explore the data set</a>
<ul>
<li class="chapter" data-level="" data-path="WE_10.html"><a href="WE_10.html#a.-plot-all-rasters"><i class="fa fa-check"></i>a. Plot all rasters</a></li>
<li class="chapter" data-level="" data-path="WE_10.html"><a href="WE_10.html#b.-plot-spatial-points-over-ffp-raster"><i class="fa fa-check"></i>b. Plot spatial points over ffp raster</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="WE_10.html"><a href="WE_10.html#setting-cost-values-and-calculating-conductance"><i class="fa fa-check"></i>3. Setting cost values and calculating conductance</a>
<ul>
<li class="chapter" data-level="" data-path="WE_10.html"><a href="WE_10.html#a.-resistance-vs.-conductance-values"><i class="fa fa-check"></i>a. Resistance vs. conductance values</a></li>
<li class="chapter" data-level="" data-path="WE_10.html"><a href="WE_10.html#b.-as-needed-get-all-rasters-at-the-same-scale"><i class="fa fa-check"></i>b. As needed: Get all rasters at the same scale</a></li>
<li class="chapter" data-level="" data-path="WE_10.html"><a href="WE_10.html#c.-calculate-conductance-values"><i class="fa fa-check"></i>c. Calculate conductance values</a></li>
<li class="chapter" data-level="" data-path="WE_10.html"><a href="WE_10.html#d.-create-a-single-landscape-conductance-raster"><i class="fa fa-check"></i>d. Create a single landscape conductance raster</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="WE_10.html"><a href="WE_10.html#convert-conductance-into-effective-distance"><i class="fa fa-check"></i>4. Convert conductance into effective distance</a>
<ul>
<li class="chapter" data-level="" data-path="WE_10.html"><a href="WE_10.html#a.-create-a-transition-layer"><i class="fa fa-check"></i>a. Create a transition layer</a></li>
<li class="chapter" data-level="" data-path="WE_10.html"><a href="WE_10.html#b.-visually-inspect-the-raster"><i class="fa fa-check"></i>b. Visually inspect the raster</a></li>
<li class="chapter" data-level="" data-path="WE_10.html"><a href="WE_10.html#c.-correct-for-geometric-distortion"><i class="fa fa-check"></i>c. Correct for geometric distortion</a></li>
<li class="chapter" data-level="" data-path="WE_10.html"><a href="WE_10.html#d.-plot-shortest-paths-in-space"><i class="fa fa-check"></i>d. Plot shortest paths in space</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="WE_10.html"><a href="WE_10.html#create-cost-distance-matrices"><i class="fa fa-check"></i>5. Create cost-distance matrices</a>
<ul>
<li class="chapter" data-level="" data-path="WE_10.html"><a href="WE_10.html#a.-least-cost-distance"><i class="fa fa-check"></i>a. Least cost distance</a></li>
<li class="chapter" data-level="" data-path="WE_10.html"><a href="WE_10.html#b.-cost-distance-matrix-based-on-random-paths-similar-to-circuitscape"><i class="fa fa-check"></i>b. Cost-distance matrix based on random paths (similar to Circuitscape)</a></li>
<li class="chapter" data-level="" data-path="WE_10.html"><a href="WE_10.html#c.-compare-cost-distances"><i class="fa fa-check"></i>c. Compare cost distances</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="WE_10.html"><a href="WE_10.html#how-does-changing-resolution-affect-these-metrics"><i class="fa fa-check"></i>6. How does changing resolution affect these metrics?</a>
<ul>
<li class="chapter" data-level="" data-path="WE_10.html"><a href="WE_10.html#a.-create-loop"><i class="fa fa-check"></i>a. Create loop</a></li>
<li class="chapter" data-level="" data-path="WE_10.html"><a href="WE_10.html#b.-plot-the-results"><i class="fa fa-check"></i>b. Plot the results</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="WE_10.html"><a href="WE_10.html#references"><i class="fa fa-check"></i>7. References</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="" data-path="Week11.html"><a href="Week11.html"><i class="fa fa-check"></i>Week 11: Detecting Adaptation</a>
<ul>
<li class="chapter" data-level="" data-path="WE_11.html"><a href="WE_11.html"><i class="fa fa-check"></i>Worked Example</a>
<ul>
<li class="chapter" data-level="" data-path="WE_11.html"><a href="WE_11.html#overview-of-worked-example-10"><i class="fa fa-check"></i>1. Overview of Worked Example</a>
<ul>
<li class="chapter" data-level="" data-path="WE_11.html"><a href="WE_11.html#a.-goals-12"><i class="fa fa-check"></i>a. Goals</a></li>
<li class="chapter" data-level="" data-path="WE_11.html"><a href="WE_11.html#b.-data-set-10"><i class="fa fa-check"></i>b. Data set</a></li>
<li class="chapter" data-level="" data-path="WE_11.html"><a href="WE_11.html#c.-required-r-libraries-8"><i class="fa fa-check"></i>c. Required R libraries</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="WE_11.html"><a href="WE_11.html#import-and-prepare-the-data"><i class="fa fa-check"></i>2. Import and prepare the data</a>
<ul>
<li class="chapter" data-level="" data-path="WE_11.html"><a href="WE_11.html#a.-import-the-genetic-data"><i class="fa fa-check"></i>a. Import the genetic data</a></li>
<li class="chapter" data-level="" data-path="WE_11.html"><a href="WE_11.html#b.-import-the-environmental-data"><i class="fa fa-check"></i>b. Import the environmental data</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="WE_11.html"><a href="WE_11.html#latent-factor-mixed-models-lfmm-a-univariate-gea"><i class="fa fa-check"></i>3. Latent Factor Mixed Models (LFMM): a univariate GEA</a>
<ul>
<li class="chapter" data-level="" data-path="WE_11.html"><a href="WE_11.html#a.-determine-k"><i class="fa fa-check"></i>a. Determine K</a></li>
<li class="chapter" data-level="" data-path="WE_11.html"><a href="WE_11.html#b.-run-lfmm"><i class="fa fa-check"></i>b. Run LFMM</a></li>
<li class="chapter" data-level="" data-path="WE_11.html"><a href="WE_11.html#c.-identify-lfmm-candidates-using-false-discovery-rate"><i class="fa fa-check"></i>c. Identify LFMM candidates using False Discovery Rate</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="WE_11.html"><a href="WE_11.html#redundancy-analysis-rda-a-multivariate-gea"><i class="fa fa-check"></i>4. Redundancy Analysis (RDA): a multivariate GEA</a>
<ul>
<li class="chapter" data-level="" data-path="WE_11.html"><a href="WE_11.html#a.-run-rda"><i class="fa fa-check"></i>a. Run RDA</a></li>
<li class="chapter" data-level="" data-path="WE_11.html"><a href="WE_11.html#b.-identify-rda-candidates"><i class="fa fa-check"></i>b. Identify RDA candidates</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="WE_11.html"><a href="WE_11.html#compare-lfmm-and-rda-candidates"><i class="fa fa-check"></i>5. Compare LFMM and RDA candidates</a></li>
<li class="chapter" data-level="" data-path="WE_11.html"><a href="WE_11.html#a-quick-note-on-controlling-for-population-structure-in-rda"><i class="fa fa-check"></i>6. A quick note on controlling for population structure in RDA</a></li>
<li class="chapter" data-level="" data-path="WE_11.html"><a href="WE_11.html#references-1"><i class="fa fa-check"></i>7. References</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="" data-path="Week12.html"><a href="Week12.html"><i class="fa fa-check"></i>Week 12: Model Selection</a>
<ul>
<li class="chapter" data-level="" data-path="WE_12.html"><a href="WE_12.html"><i class="fa fa-check"></i>Worked Example</a>
<ul>
<li class="chapter" data-level="" data-path="WE_12.html"><a href="WE_12.html#overview-of-worked-example-11"><i class="fa fa-check"></i>1. Overview of Worked Example</a>
<ul>
<li class="chapter" data-level="" data-path="WE_12.html"><a href="WE_12.html#a.-goals-and-background"><i class="fa fa-check"></i>a. Goals and Background</a></li>
<li class="chapter" data-level="" data-path="WE_12.html"><a href="WE_12.html#b.-data-set-11"><i class="fa fa-check"></i>b. Data set</a></li>
<li class="chapter" data-level="" data-path="WE_12.html"><a href="WE_12.html#c.-required-packages"><i class="fa fa-check"></i>c. Required packages</a></li>
<li class="chapter" data-level="" data-path="WE_12.html"><a href="WE_12.html#d.-import-data-1"><i class="fa fa-check"></i>d. Import data</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="WE_12.html"><a href="WE_12.html#fitting-candidate-models"><i class="fa fa-check"></i>2. Fitting candidate models</a>
<ul>
<li class="chapter" data-level="" data-path="WE_12.html"><a href="WE_12.html#a.-alternative-hypotheses"><i class="fa fa-check"></i>a. Alternative hypotheses</a></li>
<li class="chapter" data-level="" data-path="WE_12.html"><a href="WE_12.html#b.-prepare-for-model-fitting"><i class="fa fa-check"></i>b. Prepare for model fitting</a></li>
<li class="chapter" data-level="" data-path="WE_12.html"><a href="WE_12.html#c.-fit-all-five-models"><i class="fa fa-check"></i>c. Fit all five models</a></li>
<li class="chapter" data-level="" data-path="WE_12.html"><a href="WE_12.html#d.-compare-evidence-for-models"><i class="fa fa-check"></i>d. Compare evidence for models</a></li>
<li class="chapter" data-level="" data-path="WE_12.html"><a href="WE_12.html#e.-calculate-evidence-weights"><i class="fa fa-check"></i>e. Calculate evidence weights</a></li>
<li class="chapter" data-level="" data-path="WE_12.html"><a href="WE_12.html#f.-confidence-intervals-for-predictors"><i class="fa fa-check"></i>f. Confidence intervals for predictors</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="WE_12.html"><a href="WE_12.html#your-turn-to-test-additional-hypotheses"><i class="fa fa-check"></i>3. Your turn to test additional hypotheses!</a></li>
<li class="chapter" data-level="" data-path="WE_12.html"><a href="WE_12.html#references-2"><i class="fa fa-check"></i>4. References</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="" data-path="Week13.html"><a href="Week13.html"><i class="fa fa-check"></i>Week 13: Gravity Models</a>
<ul>
<li class="chapter" data-level="" data-path="WE_13.html"><a href="WE_13.html"><i class="fa fa-check"></i>Worked Example</a>
<ul>
<li class="chapter" data-level="" data-path="WE_13.html"><a href="WE_13.html#overview-of-worked-example-12"><i class="fa fa-check"></i>1. Overview of Worked Example</a>
<ul>
<li class="chapter" data-level="" data-path="WE_13.html"><a href="WE_13.html#a.-background"><i class="fa fa-check"></i>a. Background</a></li>
<li class="chapter" data-level="" data-path="WE_13.html"><a href="WE_13.html#b.-data-set-12"><i class="fa fa-check"></i>b. Data set</a></li>
<li class="chapter" data-level="" data-path="WE_13.html"><a href="WE_13.html#c.-required-r-libraries-9"><i class="fa fa-check"></i>c. Required R libraries</a></li>
<li class="chapter" data-level="" data-path="WE_13.html"><a href="WE_13.html#set-working-and-data-directories---needs-to-be-removed"><i class="fa fa-check"></i>Set working and data directories - NEEDS TO BE REMOVED</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="WE_13.html"><a href="WE_13.html#calculating-graph-metrics"><i class="fa fa-check"></i>2. Calculating graph metrics</a>
<ul>
<li class="chapter" data-level="" data-path="WE_13.html"><a href="WE_13.html#a.-read-in-make-a-spatial-object-and-plot-wetlands-wetlands.csv."><i class="fa fa-check"></i>a. Read in, make a spatial object and plot wetlands (Wetlands.csv).</a></li>
<li class="chapter" data-level="" data-path="WE_13.html"><a href="WE_13.html#b.-create-a-graph-from-the-wetlands"><i class="fa fa-check"></i>b. Create a graph from the wetlands</a></li>
<li class="chapter" data-level="" data-path="WE_13.html"><a href="WE_13.html#c.-create-adjacency-matrix"><i class="fa fa-check"></i>c. Create adjacency matrix</a></li>
<li class="chapter" data-level="" data-path="WE_13.html"><a href="WE_13.html#d.-calculate-graph-metrics"><i class="fa fa-check"></i>d. Calculate graph metrics</a></li>
<li class="chapter" data-level="" data-path="WE_13.html"><a href="WE_13.html#e.-add-graph-metrics-to-data.frame"><i class="fa fa-check"></i>e. Add graph metrics to data.frame</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="WE_13.html"><a href="WE_13.html#combine-frog-data-with-graph-metrics"><i class="fa fa-check"></i>3. Combine frog data with graph metrics</a>
<ul>
<li class="chapter" data-level="" data-path="WE_13.html"><a href="WE_13.html#a.-add-graph-node-data-to-frog-site-data"><i class="fa fa-check"></i>a. Add graph node data to frog site data</a></li>
<li class="chapter" data-level="" data-path="WE_13.html"><a href="WE_13.html#b.-check-data-types"><i class="fa fa-check"></i>b. Check data types</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="WE_13.html"><a href="WE_13.html#merge-graph-with-genetic-distance-data"><i class="fa fa-check"></i>4. Merge graph with genetic distance data</a>
<ul>
<li class="chapter" data-level="" data-path="WE_13.html"><a href="WE_13.html#a.-build-graph-from-occupied-sites"><i class="fa fa-check"></i>a. Build graph from occupied sites</a></li>
<li class="chapter" data-level="" data-path="WE_13.html"><a href="WE_13.html#b.-merge-graph-with-genetic-distance"><i class="fa fa-check"></i>b. Merge graph with genetic distance</a></li>
<li class="chapter" data-level="" data-path="WE_13.html"><a href="WE_13.html#c.-write-out-results-ascii-csv-and-lines-shapefile."><i class="fa fa-check"></i>c. Write out results (ASCII csv and lines shapefile).</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="WE_13.html"><a href="WE_13.html#preparing-raster-based-covariates"><i class="fa fa-check"></i>5. Preparing raster-based covariates</a>
<ul>
<li class="chapter" data-level="" data-path="WE_13.html"><a href="WE_13.html#a.-import-raster-data-as-stack"><i class="fa fa-check"></i>a. Import raster data as stack</a></li>
<li class="chapter" data-level="" data-path="WE_13.html"><a href="WE_13.html#b.-calculate-wetland-area-within-buffer"><i class="fa fa-check"></i>b. Calculate wetland area within buffer</a></li>
<li class="chapter" data-level="" data-path="WE_13.html"><a href="WE_13.html#c.-create-raster-of-wetland-proportion"><i class="fa fa-check"></i>c. Create raster of wetland proportion</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="WE_13.html"><a href="WE_13.html#extract-raster-stats-for-nodes-and-edges"><i class="fa fa-check"></i>6. Extract raster stats for nodes and edges</a>
<ul>
<li class="chapter" data-level="" data-path="WE_13.html"><a href="WE_13.html#a.-extract-at-site-variables-from-rasters"><i class="fa fa-check"></i>a. Extract at-site variables from rasters</a></li>
<li class="chapter" data-level="" data-path="WE_13.html"><a href="WE_13.html#b.-add-covariates-to-graph-edges."><i class="fa fa-check"></i>b. Add covariates to graph edges.</a></li>
<li class="chapter" data-level="" data-path="WE_13.html"><a href="WE_13.html#c.-calculating-stats-along-edges"><i class="fa fa-check"></i>c. Calculating stats along edges</a></li>
<li class="chapter" data-level="" data-path="WE_13.html"><a href="WE_13.html#d.-calculating-stats-for-categorical-variables"><i class="fa fa-check"></i>d. Calculating stats for categorical variables</a></li>
<li class="chapter" data-level="" data-path="WE_13.html"><a href="WE_13.html#e.-add-node-data-to-graph"><i class="fa fa-check"></i>e. Add node data to graph</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="WE_13.html"><a href="WE_13.html#gravity-models"><i class="fa fa-check"></i>7. Gravity Models</a>
<ul>
<li class="chapter" data-level="" data-path="WE_13.html"><a href="WE_13.html#a.-merge-distance-and-site-data"><i class="fa fa-check"></i>a. Merge distance and site data</a></li>
<li class="chapter" data-level="" data-path="WE_13.html"><a href="WE_13.html#b.-define-a-set-of-models"><i class="fa fa-check"></i>b. Define a set of models</a></li>
<li class="chapter" data-level="" data-path="WE_13.html"><a href="WE_13.html#c.-check-for-correlations"><i class="fa fa-check"></i>c. Check for correlations</a></li>
<li class="chapter" data-level="" data-path="WE_13.html"><a href="WE_13.html#c.-write-out-resultss"><i class="fa fa-check"></i>c. Write out resultss</a></li>
<li class="chapter" data-level="" data-path="WE_13.html"><a href="WE_13.html#e.-compare-models"><i class="fa fa-check"></i>e. Compare models</a></li>
<li class="chapter" data-level="" data-path="WE_13.html"><a href="WE_13.html#f.-diagnostic-plots"><i class="fa fa-check"></i>f. Diagnostic plots</a></li>
<li class="chapter" data-level="" data-path="WE_13.html"><a href="WE_13.html#g.-fit-final-models"><i class="fa fa-check"></i>g. Fit final model(s)</a></li>
<li class="chapter" data-level="" data-path="WE_13.html"><a href="WE_13.html#h.-effect-size"><i class="fa fa-check"></i>h. Effect size</a></li>
</ul></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="" data-path="Week14.html"><a href="Week14.html"><i class="fa fa-check"></i>Week 14: Contemporary Gene Flow</a>
<ul>
<li class="chapter" data-level="" data-path="WE_14.html"><a href="WE_14.html"><i class="fa fa-check"></i>Worked Example</a>
<ul>
<li class="chapter" data-level="" data-path="WE_14.html"><a href="WE_14.html#overview-3"><i class="fa fa-check"></i>1. Overview</a>
<ul>
<li class="chapter" data-level="" data-path="WE_14.html"><a href="WE_14.html#a.-background-and-goals"><i class="fa fa-check"></i>a. Background and Goals</a></li>
<li class="chapter" data-level="" data-path="WE_14.html"><a href="WE_14.html#b.-data-set-13"><i class="fa fa-check"></i>b. Data set</a></li>
<li class="chapter" data-level="" data-path="WE_14.html"><a href="WE_14.html#c.-required-packages-1"><i class="fa fa-check"></i>c. Required packages</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="WE_14.html"><a href="WE_14.html#import-genotypes"><i class="fa fa-check"></i>2. Import Genotypes</a>
<ul>
<li class="chapter" data-level="" data-path="WE_14.html"><a href="WE_14.html#a.-import-genotype-data"><i class="fa fa-check"></i>a. Import genotype data</a></li>
<li class="chapter" data-level="" data-path="WE_14.html"><a href="WE_14.html#b.-plot-map-of-sampling-locations"><i class="fa fa-check"></i>b. Plot map of sampling locations</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="WE_14.html"><a href="WE_14.html#pollen-pool-genetic-structure"><i class="fa fa-check"></i>3. Pollen pool genetic structure</a>
<ul>
<li class="chapter" data-level="" data-path="WE_14.html"><a href="WE_14.html#a.-subtract-maternal-contribution-to-offspring-genotype"><i class="fa fa-check"></i>a. Subtract maternal contribution to offspring genotype</a></li>
<li class="chapter" data-level="" data-path="WE_14.html"><a href="WE_14.html#b.-twogener-analysis-amova-of-pollen-pool-data"><i class="fa fa-check"></i>b. ‘TwoGener’ analysis (AMOVA of pollen pool data)</a></li>
<li class="chapter" data-level="" data-path="WE_14.html"><a href="WE_14.html#c.-including-space-isolation-by-distance"><i class="fa fa-check"></i>c. Including space: Isolation by distance</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="WE_14.html"><a href="WE_14.html#paternity-analysis"><i class="fa fa-check"></i>4. Paternity analysis</a>
<ul>
<li class="chapter" data-level="" data-path="WE_14.html"><a href="WE_14.html#a.-exclusion-probabilities"><i class="fa fa-check"></i>a. Exclusion probabilities</a></li>
<li class="chapter" data-level="" data-path="WE_14.html"><a href="WE_14.html#b.-paternity-analysis"><i class="fa fa-check"></i>b. Paternity analysis</a></li>
<li class="chapter" data-level="" data-path="WE_14.html"><a href="WE_14.html#c.-visualize-reconstructed-pollen-flow-events-in-space"><i class="fa fa-check"></i>c. Visualize reconstructed pollen flow events in space</a></li>
<li class="chapter" data-level="" data-path="WE_14.html"><a href="WE_14.html#d.-visualize-dispersal-kernel"><i class="fa fa-check"></i>d. Visualize dispersal kernel</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="WE_14.html"><a href="WE_14.html#linking-paternity-to-ecological-variables"><i class="fa fa-check"></i>5. Linking paternity to ecological variables</a>
<ul>
<li class="chapter" data-level="" data-path="WE_14.html"><a href="WE_14.html#a.-explain-pollen-flow-within-populations"><i class="fa fa-check"></i>a. Explain pollen flow within populations</a></li>
<li class="chapter" data-level="" data-path="WE_14.html"><a href="WE_14.html#b.-explain-pollen-flow-at-the-population-level"><i class="fa fa-check"></i>b. Explain pollen flow at the population level</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="WE_14.html"><a href="WE_14.html#references-3"><i class="fa fa-check"></i>6. References</a></li>
</ul></li>
</ul></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Landscape Genetic Data Analysis with R</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="WE_10" class="section level2">
<h2>Worked Example</h2>
<p>Melanie Murphy, Jeff Bowman, Robby Marrotte and Helene Wagner</p>
<div id="overview-of-worked-example-9" class="section level3">
<h3>1. Overview of Worked Example</h3>
<div id="a.-goals-11" class="section level4">
<h4>a. Goals</h4>
<p>This worked example shows how to:</p>
<ul>
<li>Convert spatial data into weights (“costs”) based on different weighting approaches.</li>
<li>Calculate landscape resistance using least-cost and commute time (analogous to circuit theory) approaches.</li>
</ul>
</div>
<div id="b.-data-set-9" class="section level4">
<h4>b. Data set</h4>
<p>This code builds on the data from the ‘GeNetIt’ package by Jeff Evans and Melanie Murphy that we’ve used already in the Week 2 Worked Example. We will use spatial data from central Idaho for Columbia spotted frogs (Murphy et al. 2010).</p>
<p><strong>Raster data</strong>: A raster is essentially a georeferenced matrix (i.e., a matrix with geographical co-ordinates). Here we’ll analyze raster data that are available in the ‘GeNetIt’ packages in the form of a SpatialPixelsDataFrame called ‘rasters’ (for meta data, type: ?GeNetIt::rasters). These data include classified Landsat land-cover data (NLCD), spline-based climate predictions (Rehfeldt et al. 2006), and topographically derived variables (Moore et al. 1993, Evans 1972). Landscape variables were selected based on knowledge of species’ ecology and previous research (Murphy et al. 2010, Pilliod et al. 2002, Funk et al. 2005):</p>
<ul>
<li>cti: Compound Topographic Index (“wetness”)</li>
<li>err27: Elevation Relief Ratio</li>
<li>ffp: Frost Free Period</li>
<li>gsp: Growing Season Precipitation</li>
<li>hli: Heat Load Index</li>
<li>nlcd: USGS Landcover (categorical map)</li>
</ul>
<p><strong>Sampling locations</strong>: We will model landscape resistance between the ponds from which Colombia spotted frogs were sampled. The spatial coordinates and site data are available in the ‘GeNEtIt’ package in the form of a SpatialPointsDataFrame ‘ralu.site.’ The slot <code>@coords</code> contains UTM coordinates (zone 11), and the slot <code>@data</code> contains 17 site variables for 31 sites. For meta data type: ?GeNetIt::ralu.site.</p>
</div>
<div id="c.-required-r-libraries-7" class="section level4">
<h4>c. Required R libraries</h4>
<p>Install some packages needed for this worked example.</p>
<div class="sourceCode" id="cb39"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb39-1"><a href="WE_10.html#cb39-1" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span>(<span class="sc">!</span><span class="fu">requireNamespace</span>(<span class="st">&quot;GeNetIt&quot;</span>, <span class="at">quietly =</span> <span class="cn">TRUE</span>)) remotes<span class="sc">::</span><span class="fu">install_github</span>(<span class="st">&quot;jeffreyevans/GeNetIt&quot;</span>)</span></code></pre></div>
<div class="sourceCode" id="cb40"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb40-1"><a href="WE_10.html#cb40-1" aria-hidden="true" tabindex="-1"></a><span class="fu">require</span>(LandGenCourse)</span>
<span id="cb40-2"><a href="WE_10.html#cb40-2" aria-hidden="true" tabindex="-1"></a><span class="co">#require(GeNetIt)</span></span>
<span id="cb40-3"><a href="WE_10.html#cb40-3" aria-hidden="true" tabindex="-1"></a><span class="co">#require(raster)</span></span>
<span id="cb40-4"><a href="WE_10.html#cb40-4" aria-hidden="true" tabindex="-1"></a><span class="co">#require(gdistance)</span></span></code></pre></div>
</div>
<div id="d.-import-rasters-gsp-err-cti-and-ffp." class="section level4">
<h4>d. Import rasters (gsp, err, cti, and ffp).</h4>
<p>Here we load the raster data as a SpatialPixelsDataFrame and convert them in to a raster stack.</p>
<div class="sourceCode" id="cb41"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb41-1"><a href="WE_10.html#cb41-1" aria-hidden="true" tabindex="-1"></a><span class="fu">data</span>(rasters, <span class="at">package=</span><span class="st">&quot;GeNetIt&quot;</span>)</span>
<span id="cb41-2"><a href="WE_10.html#cb41-2" aria-hidden="true" tabindex="-1"></a>RasterMaps <span class="ot">&lt;-</span> raster<span class="sc">::</span><span class="fu">stack</span>(rasters)</span></code></pre></div>
</div>
<div id="e.-import-site-data-with-sampling-locations" class="section level4">
<h4>e. Import site data with sampling locations</h4>
<p>In addition to spatial data, sample locations are also needed. Sample locations are located in the data folder (RALU_UTM.csv). Read in site locations (wetlands with Columbia spotted frogs).</p>
<div class="sourceCode" id="cb42"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb42-1"><a href="WE_10.html#cb42-1" aria-hidden="true" tabindex="-1"></a><span class="fu">data</span>(ralu.site, <span class="at">package=</span><span class="st">&quot;GeNetIt&quot;</span>)</span>
<span id="cb42-2"><a href="WE_10.html#cb42-2" aria-hidden="true" tabindex="-1"></a>sites <span class="ot">&lt;-</span> ralu.site</span></code></pre></div>
<p><strong>Question 1</strong>: What are UTMs and why might it be important to work in UTMs (as opposed to latitude and longitude)?</p>
</div>
</div>
<div id="explore-the-data-set" class="section level3">
<h3>2. Explore the data set</h3>
<p>Explore these rasters by plotting them. As you go through this exercise, use the plot function at each step to make sure your outputs “make sense.”</p>
<div id="a.-plot-all-rasters" class="section level4">
<h4>a. Plot all rasters</h4>
<div class="sourceCode" id="cb43"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb43-1"><a href="WE_10.html#cb43-1" aria-hidden="true" tabindex="-1"></a>raster<span class="sc">::</span><span class="fu">plot</span>(RasterMaps)</span></code></pre></div>
<p><img src="10-Week10_files/figure-html/unnamed-chunk-7-1.png" width="768" /></p>
</div>
<div id="b.-plot-spatial-points-over-ffp-raster" class="section level4">
<h4>b. Plot spatial points over ffp raster</h4>
<p>These are the locations of the ponds where Columbia spotted frogs were sampled within the study area.</p>
<div class="sourceCode" id="cb44"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb44-1"><a href="WE_10.html#cb44-1" aria-hidden="true" tabindex="-1"></a><span class="fu">par</span>(<span class="at">mar=</span><span class="fu">c</span>(<span class="dv">2</span>,<span class="dv">2</span>,<span class="dv">1</span>,<span class="dv">1</span>))</span>
<span id="cb44-2"><a href="WE_10.html#cb44-2" aria-hidden="true" tabindex="-1"></a>raster<span class="sc">::</span><span class="fu">plot</span>(RasterMaps<span class="sc">$</span>ffp)</span>
<span id="cb44-3"><a href="WE_10.html#cb44-3" aria-hidden="true" tabindex="-1"></a><span class="fu">points</span>(sites, <span class="at">pch=</span><span class="dv">3</span>)</span></code></pre></div>
<p><img src="10-Week10_files/figure-html/unnamed-chunk-8-1.png" width="672" /></p>
</div>
</div>
<div id="setting-cost-values-and-calculating-conductance" class="section level3">
<h3>3. Setting cost values and calculating conductance</h3>
<div id="a.-resistance-vs.-conductance-values" class="section level4">
<h4>a. Resistance vs. conductance values</h4>
<p>The next step involves creating cost categories from the continuous data. For the purposes of this exercise, a single “landscape resistance” cost surface will be created, based on merging the 4 variables, and employing expert opinion.</p>
<p>Rescaling the grids will take some processor time (depending on your computer).</p>
<p>First, use a relative ranking of the costs of the landscape variables based on expert opinion. Higher rank indicates more resistance as follows.</p>
<p>Landscape resistances:</p>
<pre><code>1) err27, 2) ffp, 3) gsp, 4) cti</code></pre>
<p>Important considerations:</p>
<ol style="list-style-type: decimal">
<li>The transition matrix in package ‘gdistance’ is based on conductance and not resistance.</li>
<li>If we are going to create a single landscape resistance (i.e., add the costs together to create one synthetic landscape variable), costs need to represent relative importance of the variables.</li>
</ol>
<p>Keep in mind there are a variety of approaches for creating landscape resistance values. This exercise implements one simplistic approach.</p>
<p>To create relative conductance values, reverse the rank order of the resistance values:</p>
<pre><code>1) cti, 2) gsp, 3) ffp, 4) err27 </code></pre>
</div>
<div id="b.-as-needed-get-all-rasters-at-the-same-scale" class="section level4">
<h4>b. As needed: Get all rasters at the same scale</h4>
<p>Note: this step is not needed here as all rasters do already have the same resolution.</p>
<p>Originally, the topographic variables were calculated off a 10 m DEM. The climate variables are at a 30 m resolution. In order to calculate costs, all of the rasters need to have exactly the same resolution, dimensions, and co-ordinate locations (i.e., the cells need to match up perfectly. The method of resampling (bilinear vs nearest-neighbor) depends on the type of data. Nearest-neighbor is for categorical data whereas bilinear interpolation is for continuous data.</p>
<p>Here is some sample code to show how you would resample a raster ‘cti’ (assumed to have 10 m resolution) to the dimensions of a raster ‘gsp’ (with 30 m resolution):</p>
<div class="sourceCode" id="cb47"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb47-1"><a href="WE_10.html#cb47-1" aria-hidden="true" tabindex="-1"></a><span class="co">#cti &lt;- raster::resample(cti, gsp, method= &quot;bilinear&quot;)</span></span></code></pre></div>
</div>
<div id="c.-calculate-conductance-values" class="section level4">
<h4>c. Calculate conductance values</h4>
<p>Functions (e.g., 1/err, ffp/5) rescale the raw data from each raster to represent costs that correspond to perceived differences among variables arising from expert opinion. A wide variety of functions could be used to achieve a variety of cost structures in order to test alternative hypotheses.
The err function is inverted compared to the other variables because the raw err data have low values relative to the other variables (&lt; 1).</p>
<p>To use the below tools, conductance values must be calculated. First, look at the range of the variable in a given raster. Then apply a function to get the desired
relative conductance values. Make sure to use the plot function and visually inspect your conductance surfaces.</p>
<div class="sourceCode" id="cb48"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb48-1"><a href="WE_10.html#cb48-1" aria-hidden="true" tabindex="-1"></a>RasterMaps<span class="sc">$</span>err27</span></code></pre></div>
<pre><code>## class      : RasterLayer 
## dimensions : 426, 358, 152508  (nrow, ncol, ncell)
## resolution : 30, 30  (x, y)
## extent     : 683282.5, 694022.5, 4992833, 5005613  (xmin, xmax, ymin, ymax)
## crs        : +proj=utm +zone=11 +datum=NAD83 +units=m +no_defs 
## source     : memory
## names      : err27 
## values     : 0.03906551, 0.7637643  (min, max)</code></pre>
<p>Elevation relief ratio (‘err27’) is identifying major topographic features. In this first case, the goal is for ‘err27’ to have the lowest conductance values compared to our other landscape variables. Greater ‘err27’ means more change in topography in a given area. So, the higher the value, the more resistance to a Columbian spotted frog.
Note that the function below gives a linear relationship between the raw value and the cost. Any type of functional relationship could be used however.</p>
<div class="sourceCode" id="cb50"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb50-1"><a href="WE_10.html#cb50-1" aria-hidden="true" tabindex="-1"></a>err.cost <span class="ot">&lt;-</span> (<span class="dv">1</span><span class="sc">/</span>RasterMaps<span class="sc">$</span>err27)</span>
<span id="cb50-2"><a href="WE_10.html#cb50-2" aria-hidden="true" tabindex="-1"></a>err.cost</span></code></pre></div>
<pre><code>## class      : RasterLayer 
## dimensions : 426, 358, 152508  (nrow, ncol, ncell)
## resolution : 30, 30  (x, y)
## extent     : 683282.5, 694022.5, 4992833, 5005613  (xmin, xmax, ymin, ymax)
## crs        : +proj=utm +zone=11 +datum=NAD83 +units=m +no_defs 
## source     : memory
## names      : err27 
## values     : 1.309305, 25.59803  (min, max)</code></pre>
<div class="sourceCode" id="cb52"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb52-1"><a href="WE_10.html#cb52-1" aria-hidden="true" tabindex="-1"></a>RasterMaps<span class="sc">$</span>ffp</span></code></pre></div>
<pre><code>## class      : RasterLayer 
## dimensions : 426, 358, 152508  (nrow, ncol, ncell)
## resolution : 30, 30  (x, y)
## extent     : 683282.5, 694022.5, 4992833, 5005613  (xmin, xmax, ymin, ymax)
## crs        : +proj=utm +zone=11 +datum=NAD83 +units=m +no_defs 
## source     : memory
## names      : ffp 
## values     : 0, 51  (min, max)</code></pre>
<div class="sourceCode" id="cb54"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb54-1"><a href="WE_10.html#cb54-1" aria-hidden="true" tabindex="-1"></a>ffp.cost <span class="ot">&lt;-</span> (RasterMaps<span class="sc">$</span>ffp<span class="sc">/</span><span class="dv">5</span>)</span>
<span id="cb54-2"><a href="WE_10.html#cb54-2" aria-hidden="true" tabindex="-1"></a>ffp.cost</span></code></pre></div>
<pre><code>## class      : RasterLayer 
## dimensions : 426, 358, 152508  (nrow, ncol, ncell)
## resolution : 30, 30  (x, y)
## extent     : 683282.5, 694022.5, 4992833, 5005613  (xmin, xmax, ymin, ymax)
## crs        : +proj=utm +zone=11 +datum=NAD83 +units=m +no_defs 
## source     : memory
## names      : ffp 
## values     : 0, 10.2  (min, max)</code></pre>
<div class="sourceCode" id="cb56"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb56-1"><a href="WE_10.html#cb56-1" aria-hidden="true" tabindex="-1"></a>RasterMaps<span class="sc">$</span>gsp</span></code></pre></div>
<pre><code>## class      : RasterLayer 
## dimensions : 426, 358, 152508  (nrow, ncol, ncell)
## resolution : 30, 30  (x, y)
## extent     : 683282.5, 694022.5, 4992833, 5005613  (xmin, xmax, ymin, ymax)
## crs        : +proj=utm +zone=11 +datum=NAD83 +units=m +no_defs 
## source     : memory
## names      : gsp 
## values     : 227, 338.0697  (min, max)</code></pre>
<div class="sourceCode" id="cb58"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb58-1"><a href="WE_10.html#cb58-1" aria-hidden="true" tabindex="-1"></a>gsp.cost <span class="ot">&lt;-</span> (RasterMaps<span class="sc">$</span>gsp<span class="dv">-196</span>)<span class="sc">/</span><span class="dv">15</span></span>
<span id="cb58-2"><a href="WE_10.html#cb58-2" aria-hidden="true" tabindex="-1"></a>gsp.cost</span></code></pre></div>
<pre><code>## class      : RasterLayer 
## dimensions : 426, 358, 152508  (nrow, ncol, ncell)
## resolution : 30, 30  (x, y)
## extent     : 683282.5, 694022.5, 4992833, 5005613  (xmin, xmax, ymin, ymax)
## crs        : +proj=utm +zone=11 +datum=NAD83 +units=m +no_defs 
## source     : memory
## names      : gsp 
## values     : 2.066667, 9.471311  (min, max)</code></pre>
<div class="sourceCode" id="cb60"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb60-1"><a href="WE_10.html#cb60-1" aria-hidden="true" tabindex="-1"></a>RasterMaps<span class="sc">$</span>cti</span></code></pre></div>
<pre><code>## class      : RasterLayer 
## dimensions : 426, 358, 152508  (nrow, ncol, ncell)
## resolution : 30, 30  (x, y)
## extent     : 683282.5, 694022.5, 4992833, 5005613  (xmin, xmax, ymin, ymax)
## crs        : +proj=utm +zone=11 +datum=NAD83 +units=m +no_defs 
## source     : memory
## names      : cti 
## values     : 0.8429851, 23.71476  (min, max)</code></pre>
<div class="sourceCode" id="cb62"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb62-1"><a href="WE_10.html#cb62-1" aria-hidden="true" tabindex="-1"></a>cti.cost <span class="ot">&lt;-</span> RasterMaps<span class="sc">$</span>cti<span class="sc">/</span><span class="dv">5</span></span>
<span id="cb62-2"><a href="WE_10.html#cb62-2" aria-hidden="true" tabindex="-1"></a>cti.cost</span></code></pre></div>
<pre><code>## class      : RasterLayer 
## dimensions : 426, 358, 152508  (nrow, ncol, ncell)
## resolution : 30, 30  (x, y)
## extent     : 683282.5, 694022.5, 4992833, 5005613  (xmin, xmax, ymin, ymax)
## crs        : +proj=utm +zone=11 +datum=NAD83 +units=m +no_defs 
## source     : memory
## names      : cti 
## values     : 0.168597, 4.742952  (min, max)</code></pre>
</div>
<div id="d.-create-a-single-landscape-conductance-raster" class="section level4">
<h4>d. Create a single landscape conductance raster</h4>
<p>Testing variables independently may be more appropriate depending on the research question but for the purposes of today’s lab, a single conductance raster will be employed.</p>
<div class="sourceCode" id="cb64"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb64-1"><a href="WE_10.html#cb64-1" aria-hidden="true" tabindex="-1"></a>cost1 <span class="ot">&lt;-</span> (gsp.cost <span class="sc">+</span> cti.cost <span class="sc">+</span> err.cost <span class="sc">+</span> ffp.cost)</span>
<span id="cb64-2"><a href="WE_10.html#cb64-2" aria-hidden="true" tabindex="-1"></a>cost1</span></code></pre></div>
<pre><code>## class      : RasterLayer 
## dimensions : 426, 358, 152508  (nrow, ncol, ncell)
## resolution : 30, 30  (x, y)
## extent     : 683282.5, 694022.5, 4992833, 5005613  (xmin, xmax, ymin, ymax)
## crs        : +proj=utm +zone=11 +datum=NAD83 +units=m +no_defs 
## source     : memory
## names      : layer 
## values     : 9.012874, 35.90095  (min, max)</code></pre>
<p><strong>Question 2</strong>: Plot your cost surface with your sample locations on top. What does this tell you?</p>
</div>
</div>
<div id="convert-conductance-into-effective-distance" class="section level3">
<h3>4. Convert conductance into effective distance</h3>
<p>The higher the conductance, the lower the cost or resistance of a cell, and vice versa. We want to integrate conductance across cells to derive some measure of effective (or ecological) distance.</p>
<div id="a.-create-a-transition-layer" class="section level4">
<h4>a. Create a transition layer</h4>
<p>Transition layers are constructed from a raster, and they take the geographic references (projection, resolution, extent) from the original raster object. They also contain a matrix of probability of movement between cells which can be interpreted as “conductance.” Each cell in the matrix represents a cell in the original raster object.</p>
<p>The first step is to construct a transition object based on ‘cost1’ (which is a conductance layer as calculated). This step is computationally intensive and make take a few minutes to run.</p>
<p>Connections can be set based on 4, 8, or 16 neighbor rules. A value of 8 connects all adjacent cells in 8 directions.</p>
<div class="sourceCode" id="cb66"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb66-1"><a href="WE_10.html#cb66-1" aria-hidden="true" tabindex="-1"></a>tr.cost1 <span class="ot">&lt;-</span> gdistance<span class="sc">::</span><span class="fu">transition</span>(cost1, <span class="at">transitionFunction=</span>mean, <span class="at">directions=</span><span class="dv">8</span>) </span>
<span id="cb66-2"><a href="WE_10.html#cb66-2" aria-hidden="true" tabindex="-1"></a>tr.cost1</span></code></pre></div>
<pre><code>## class      : TransitionLayer 
## dimensions : 426, 358, 152508  (nrow, ncol, ncell)
## resolution : 30, 30  (x, y)
## extent     : 683282.5, 694022.5, 4992833, 5005613  (xmin, xmax, ymin, ymax)
## crs        : +proj=utm +zone=11 +datum=NAD83 +units=m +no_defs 
## values      : conductance 
## matrix class: dsCMatrix</code></pre>
</div>
<div id="b.-visually-inspect-the-raster" class="section level4">
<h4>b. Visually inspect the raster</h4>
<div class="sourceCode" id="cb68"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb68-1"><a href="WE_10.html#cb68-1" aria-hidden="true" tabindex="-1"></a><span class="fu">par</span>(<span class="at">mar=</span><span class="fu">c</span>(<span class="dv">2</span>,<span class="dv">2</span>,<span class="dv">1</span>,<span class="dv">1</span>))</span>
<span id="cb68-2"><a href="WE_10.html#cb68-2" aria-hidden="true" tabindex="-1"></a>raster<span class="sc">::</span><span class="fu">plot</span>(raster<span class="sc">::</span><span class="fu">raster</span>(tr.cost1))</span></code></pre></div>
<p><img src="10-Week10_files/figure-html/unnamed-chunk-20-1.png" width="672" /></p>
</div>
<div id="c.-correct-for-geometric-distortion" class="section level4">
<h4>c. Correct for geometric distortion</h4>
<p>Transition values are calculated based on values of adjacent cells in the cost raster. However, we used an 8 neighbor rule and the center of diagonally connected
raster cells are farther apart from each other than the orthogonally connected cells. We are using UTM co-ordinates; however, in lat-long projections, cell sizes become smaller as you move poleward.</p>
<p>Values of the matrix have to be corrected for the first type of distortion for our analysis (and we would need to correct for the second type of distortion if we were using lat-longs).</p>
<div class="sourceCode" id="cb69"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb69-1"><a href="WE_10.html#cb69-1" aria-hidden="true" tabindex="-1"></a>tr.cost1 <span class="ot">&lt;-</span> gdistance<span class="sc">::</span><span class="fu">geoCorrection</span>(tr.cost1,<span class="at">type =</span> <span class="st">&quot;c&quot;</span>,<span class="at">multpl=</span><span class="cn">FALSE</span>)</span></code></pre></div>
</div>
<div id="d.-plot-shortest-paths-in-space" class="section level4">
<h4>d. Plot shortest paths in space</h4>
<p>Here we do the following:</p>
<ul>
<li>Plot site locations on top of conductance raster ‘tr.cost1’</li>
<li>Calculate ‘AtoB’ as the shortest path (least cost path) between sites 1 and 2</li>
<li>Plot the shortest path onto the map</li>
</ul>
<p>Note: the path is stored as a ‘SpatialLines’ object (package ‘spdep’). A path is a sequence of coordinate pairs, which is stored in the <code>@lines</code> slot. Like a ‘SpatialPointsDataFrame,’ a ‘SpatialLines’ object has a ‘proje4string’ attribute that stores the projection information. This means that we can easily plot it on top of a raster map.</p>
<div class="sourceCode" id="cb70"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb70-1"><a href="WE_10.html#cb70-1" aria-hidden="true" tabindex="-1"></a><span class="fu">par</span>(<span class="at">mar=</span><span class="fu">c</span>(<span class="dv">2</span>,<span class="dv">2</span>,<span class="dv">1</span>,<span class="dv">2</span>))</span>
<span id="cb70-2"><a href="WE_10.html#cb70-2" aria-hidden="true" tabindex="-1"></a>AtoB <span class="ot">&lt;-</span> gdistance<span class="sc">::</span><span class="fu">shortestPath</span>(tr.cost1, <span class="at">origin=</span>sites[<span class="dv">1</span>,], </span>
<span id="cb70-3"><a href="WE_10.html#cb70-3" aria-hidden="true" tabindex="-1"></a>                                <span class="at">goal=</span>sites[<span class="dv">2</span>,], <span class="at">output=</span><span class="st">&quot;SpatialLines&quot;</span>)</span>
<span id="cb70-4"><a href="WE_10.html#cb70-4" aria-hidden="true" tabindex="-1"></a>raster<span class="sc">::</span><span class="fu">plot</span>(raster<span class="sc">::</span><span class="fu">raster</span>(tr.cost1), <span class="at">xlab=</span><span class="st">&quot;x coordinate (m)&quot;</span>, </span>
<span id="cb70-5"><a href="WE_10.html#cb70-5" aria-hidden="true" tabindex="-1"></a>             <span class="at">ylab=</span><span class="st">&quot;y coordinate (m)&quot;</span>,<span class="at">legend.lab=</span><span class="st">&quot;Conductance&quot;</span>)</span>
<span id="cb70-6"><a href="WE_10.html#cb70-6" aria-hidden="true" tabindex="-1"></a><span class="fu">lines</span>(AtoB, <span class="at">col=</span><span class="st">&quot;red&quot;</span>, <span class="at">lwd=</span><span class="dv">2</span>)</span>
<span id="cb70-7"><a href="WE_10.html#cb70-7" aria-hidden="true" tabindex="-1"></a><span class="fu">points</span>(sites[<span class="dv">1</span><span class="sc">:</span><span class="dv">2</span>,])</span></code></pre></div>
<p><img src="10-Week10_files/figure-html/unnamed-chunk-22-1.png" width="672" /></p>
<p>That’s nice, but it shows only the shortest path between sites 1 and 2. It would not be very useful to plot all pairwise shortest paths. Here we do the following:</p>
<ul>
<li>Plot site locations on top of conductance raster ‘tr.cost1’</li>
<li>Use Delaunay triangulation to define neighbors (we want to retain only those paths that won’t cross each other, though they may merge)</li>
<li>Plot grey lines between neighbours</li>
<li>For each unique pair of neighbours, calculate ‘AtoB’ as the shortest path<br />
</li>
<li>Plot the shortest paths onto the map in red</li>
</ul>
<p>Note: this may take a while. If it does not show up nicely in the notebook, copy and paste the code to the console and watch how the plot is evolving in the Plots tab (nothing beats being able to actually watch R do the work for you).</p>
<div class="sourceCode" id="cb71"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb71-1"><a href="WE_10.html#cb71-1" aria-hidden="true" tabindex="-1"></a><span class="fu">par</span>(<span class="at">mar=</span><span class="fu">c</span>(<span class="dv">2</span>,<span class="dv">2</span>,<span class="dv">1</span>,<span class="dv">2</span>))</span>
<span id="cb71-2"><a href="WE_10.html#cb71-2" aria-hidden="true" tabindex="-1"></a>raster<span class="sc">::</span><span class="fu">plot</span>(raster<span class="sc">::</span><span class="fu">raster</span>(tr.cost1), <span class="at">xlab=</span><span class="st">&quot;x coordinate (m)&quot;</span>, </span>
<span id="cb71-3"><a href="WE_10.html#cb71-3" aria-hidden="true" tabindex="-1"></a>             <span class="at">ylab=</span><span class="st">&quot;y coordinate (m)&quot;</span>, <span class="at">legend.lab=</span><span class="st">&quot;Conductance&quot;</span>)</span>
<span id="cb71-4"><a href="WE_10.html#cb71-4" aria-hidden="true" tabindex="-1"></a><span class="fu">points</span>(sites)</span>
<span id="cb71-5"><a href="WE_10.html#cb71-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb71-6"><a href="WE_10.html#cb71-6" aria-hidden="true" tabindex="-1"></a>Neighbours <span class="ot">&lt;-</span> spdep<span class="sc">::</span><span class="fu">tri2nb</span>(sites<span class="sc">@</span>coords, <span class="at">row.names =</span> sites<span class="sc">$</span>SiteName)</span>
<span id="cb71-7"><a href="WE_10.html#cb71-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb71-8"><a href="WE_10.html#cb71-8" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(Neighbours, sites<span class="sc">@</span>coords, <span class="at">col=</span><span class="st">&quot;darkgrey&quot;</span>, <span class="at">add=</span><span class="cn">TRUE</span>)</span>
<span id="cb71-9"><a href="WE_10.html#cb71-9" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span>(i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(Neighbours))</span>
<span id="cb71-10"><a href="WE_10.html#cb71-10" aria-hidden="true" tabindex="-1"></a>{</span>
<span id="cb71-11"><a href="WE_10.html#cb71-11" aria-hidden="true" tabindex="-1"></a>  <span class="cf">for</span>(j <span class="cf">in</span> Neighbours[[i]][Neighbours[[i]] <span class="sc">&gt;</span> i])</span>
<span id="cb71-12"><a href="WE_10.html#cb71-12" aria-hidden="true" tabindex="-1"></a>  {</span>
<span id="cb71-13"><a href="WE_10.html#cb71-13" aria-hidden="true" tabindex="-1"></a>    AtoB <span class="ot">&lt;-</span> gdistance<span class="sc">::</span><span class="fu">shortestPath</span>(tr.cost1, <span class="at">origin=</span>sites[i,], </span>
<span id="cb71-14"><a href="WE_10.html#cb71-14" aria-hidden="true" tabindex="-1"></a>                                <span class="at">goal=</span>sites[j,], <span class="at">output=</span><span class="st">&quot;SpatialLines&quot;</span>)</span>
<span id="cb71-15"><a href="WE_10.html#cb71-15" aria-hidden="true" tabindex="-1"></a>    <span class="fu">lines</span>(AtoB, <span class="at">col=</span><span class="st">&quot;red&quot;</span>, <span class="at">lwd=</span><span class="fl">1.5</span>)</span>
<span id="cb71-16"><a href="WE_10.html#cb71-16" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb71-17"><a href="WE_10.html#cb71-17" aria-hidden="true" tabindex="-1"></a>}</span></code></pre></div>
<p><img src="10-Week10_files/figure-html/unnamed-chunk-23-1.png" width="672" /></p>
<p><strong>Question 3</strong>: Do the shortest paths (red) deviate from the Euclidean distances (grey)? Why?</p>
<p><strong>Question 4</strong>: Are there paths that merge? How would you explain this?</p>
</div>
</div>
<div id="create-cost-distance-matrices" class="section level3">
<h3>5. Create cost-distance matrices</h3>
<div id="a.-least-cost-distance" class="section level4">
<h4>a. Least cost distance</h4>
<p>The cost-distance matrix is based on the corrected transition layer and the site locations (here we supply a ‘SpatialPointsDataFrame’ object). The ‘costDistance’ function requires conductance values, even though cost distance will be 1/conductance (i.e., resistance). The cost distance is a function of the transition (tr.cost1, a transition object) and spatial locations (sites, a spatial object).
This is a single least-cost path (accumulated cost units as opposed to length units) between each pair of sites..</p>
<div class="sourceCode" id="cb72"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb72-1"><a href="WE_10.html#cb72-1" aria-hidden="true" tabindex="-1"></a>cost1.dist <span class="ot">&lt;-</span> gdistance<span class="sc">::</span><span class="fu">costDistance</span>(tr.cost1,sites)</span></code></pre></div>
</div>
<div id="b.-cost-distance-matrix-based-on-random-paths-similar-to-circuitscape" class="section level4">
<h4>b. Cost-distance matrix based on random paths (similar to Circuitscape)</h4>
<p>Commute-time distance is an alternative to effective resistance from circuit theory. Commute-time is the expected time it takes for a random walk between nodes, and has been shown to correlate highly with effective resistance (Marrotte and Bowman 2017).</p>
<div class="sourceCode" id="cb73"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb73-1"><a href="WE_10.html#cb73-1" aria-hidden="true" tabindex="-1"></a>comm1.dist <span class="ot">&lt;-</span> gdistance<span class="sc">::</span><span class="fu">commuteDistance</span>(<span class="at">x =</span> tr.cost1, <span class="at">coords =</span> sites)</span></code></pre></div>
</div>
<div id="c.-compare-cost-distances" class="section level4">
<h4>c. Compare cost distances</h4>
<p>Create a distance table to compare different cost distance estimates:</p>
<div class="sourceCode" id="cb74"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb74-1"><a href="WE_10.html#cb74-1" aria-hidden="true" tabindex="-1"></a>dist_df <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="st">&quot;cost1.dist&quot;</span><span class="ot">=</span><span class="fu">as.numeric</span>(cost1.dist),</span>
<span id="cb74-2"><a href="WE_10.html#cb74-2" aria-hidden="true" tabindex="-1"></a>                      <span class="st">&quot;comm1.dist&quot;</span><span class="ot">=</span><span class="fu">as.numeric</span>(comm1.dist))</span></code></pre></div>
<p>Look at the correlation between the two different cost distances:</p>
<div class="sourceCode" id="cb75"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb75-1"><a href="WE_10.html#cb75-1" aria-hidden="true" tabindex="-1"></a>corr.LCD.comm <span class="ot">&lt;-</span> <span class="fu">cor</span>(dist_df<span class="sc">$</span>cost1.dist, dist_df<span class="sc">$</span>comm1.dist, <span class="at">method =</span> <span class="st">&quot;spearman&quot;</span>)</span>
<span id="cb75-2"><a href="WE_10.html#cb75-2" aria-hidden="true" tabindex="-1"></a>corr.LCD.comm</span></code></pre></div>
<pre><code>## [1] 0.9519704</code></pre>
<div class="sourceCode" id="cb77"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb77-1"><a href="WE_10.html#cb77-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(cost1.dist<span class="sc">~</span>comm1.dist)</span></code></pre></div>
<p><img src="10-Week10_files/figure-html/unnamed-chunk-27-1.png" width="672" /></p>
<p><strong>Question 5</strong>: Are the distance measures similar or different? Why?</p>
<p><strong>Question 6</strong>: What is the major difference? What are the implications of this difference? When might you use one or other of the methods?</p>
</div>
</div>
<div id="how-does-changing-resolution-affect-these-metrics" class="section level3">
<h3>6. How does changing resolution affect these metrics?</h3>
<p>Create a loop that runs through a sequential coarsening of the raster and calculate
these metrics and compare them to the finer resolution raster metric using a Spearman correlation. The loop will take a few minutes to run.</p>
<div id="a.-create-loop" class="section level4">
<h4>a. Create loop</h4>
<div class="sourceCode" id="cb78"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb78-1"><a href="WE_10.html#cb78-1" aria-hidden="true" tabindex="-1"></a>cor_cost <span class="ot">&lt;-</span> <span class="fu">c</span>()</span>
<span id="cb78-2"><a href="WE_10.html#cb78-2" aria-hidden="true" tabindex="-1"></a>cor_comm <span class="ot">&lt;-</span> <span class="fu">c</span>()</span>
<span id="cb78-3"><a href="WE_10.html#cb78-3" aria-hidden="true" tabindex="-1"></a>res_fact <span class="ot">&lt;-</span> <span class="fu">seq</span>(<span class="dv">2</span>,<span class="dv">20</span>,<span class="dv">2</span>)</span>
<span id="cb78-4"><a href="WE_10.html#cb78-4" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span>(fac <span class="cf">in</span> res_fact){</span>
<span id="cb78-5"><a href="WE_10.html#cb78-5" aria-hidden="true" tabindex="-1"></a>  cost1_agg <span class="ot">&lt;-</span> raster<span class="sc">::</span><span class="fu">aggregate</span>(cost1, <span class="at">fact =</span> fac)</span>
<span id="cb78-6"><a href="WE_10.html#cb78-6" aria-hidden="true" tabindex="-1"></a>  tr.cost_agg <span class="ot">&lt;-</span> gdistance<span class="sc">::</span><span class="fu">transition</span>(cost1_agg, </span>
<span id="cb78-7"><a href="WE_10.html#cb78-7" aria-hidden="true" tabindex="-1"></a>                 <span class="at">transitionFunction=</span>mean, <span class="at">directions=</span><span class="dv">8</span>)</span>
<span id="cb78-8"><a href="WE_10.html#cb78-8" aria-hidden="true" tabindex="-1"></a>  tr.cost_agg <span class="ot">&lt;-</span> gdistance<span class="sc">::</span><span class="fu">geoCorrection</span>(tr.cost_agg,<span class="at">type =</span> <span class="st">&quot;c&quot;</span>,<span class="at">multpl=</span><span class="cn">FALSE</span>)</span>
<span id="cb78-9"><a href="WE_10.html#cb78-9" aria-hidden="true" tabindex="-1"></a>  cost.dist_agg <span class="ot">&lt;-</span> gdistance<span class="sc">::</span><span class="fu">costDistance</span>(tr.cost_agg,sites)</span>
<span id="cb78-10"><a href="WE_10.html#cb78-10" aria-hidden="true" tabindex="-1"></a>  comm.dist_agg <span class="ot">&lt;-</span> gdistance<span class="sc">::</span><span class="fu">commuteDistance</span>(<span class="at">x =</span> tr.cost_agg, <span class="at">coords =</span> sites)</span>
<span id="cb78-11"><a href="WE_10.html#cb78-11" aria-hidden="true" tabindex="-1"></a>  cost.dist_agg <span class="ot">&lt;-</span> <span class="fu">as.numeric</span>(cost.dist_agg)</span>
<span id="cb78-12"><a href="WE_10.html#cb78-12" aria-hidden="true" tabindex="-1"></a>  comm.dist_agg <span class="ot">&lt;-</span> <span class="fu">as.numeric</span>(comm.dist_agg)</span>
<span id="cb78-13"><a href="WE_10.html#cb78-13" aria-hidden="true" tabindex="-1"></a>  cor_cost <span class="ot">&lt;-</span> <span class="fu">c</span>(cor_cost,<span class="fu">cor</span>(dist_df<span class="sc">$</span>cost1.dist, cost.dist_agg, </span>
<span id="cb78-14"><a href="WE_10.html#cb78-14" aria-hidden="true" tabindex="-1"></a>                             <span class="at">method =</span> <span class="st">&quot;spearman&quot;</span>))</span>
<span id="cb78-15"><a href="WE_10.html#cb78-15" aria-hidden="true" tabindex="-1"></a>  cor_comm <span class="ot">&lt;-</span> <span class="fu">c</span>(cor_comm,<span class="fu">cor</span>(dist_df<span class="sc">$</span>comm1.dist, comm.dist_agg, </span>
<span id="cb78-16"><a href="WE_10.html#cb78-16" aria-hidden="true" tabindex="-1"></a>                             <span class="at">method =</span> <span class="st">&quot;spearman&quot;</span>))</span>
<span id="cb78-17"><a href="WE_10.html#cb78-17" aria-hidden="true" tabindex="-1"></a>}</span></code></pre></div>
</div>
<div id="b.-plot-the-results" class="section level4">
<h4>b. Plot the results</h4>
<div class="sourceCode" id="cb79"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb79-1"><a href="WE_10.html#cb79-1" aria-hidden="true" tabindex="-1"></a><span class="fu">par</span>(<span class="at">mar=</span><span class="fu">c</span>(<span class="dv">4</span>,<span class="dv">4</span>,<span class="dv">1</span>,<span class="dv">1</span>))</span>
<span id="cb79-2"><a href="WE_10.html#cb79-2" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(<span class="at">y =</span> cor_cost, <span class="at">x =</span> res_fact, <span class="at">col =</span> <span class="st">&quot;red&quot;</span>, <span class="at">pch =</span> <span class="dv">19</span>, </span>
<span id="cb79-3"><a href="WE_10.html#cb79-3" aria-hidden="true" tabindex="-1"></a>     <span class="at">ylim =</span> <span class="fu">c</span>(<span class="fl">0.9</span>,<span class="dv">1</span>), <span class="at">xlab =</span> <span class="st">&quot;Aggregation factor&quot;</span>, <span class="at">ylab =</span> <span class="st">&quot;Spearman correlation&quot;</span>)</span>
<span id="cb79-4"><a href="WE_10.html#cb79-4" aria-hidden="true" tabindex="-1"></a><span class="fu">points</span>(<span class="at">y =</span> cor_comm, <span class="at">x =</span> res_fact, <span class="at">col =</span> <span class="st">&quot;blue&quot;</span>, <span class="at">pch =</span> <span class="dv">19</span>)</span>
<span id="cb79-5"><a href="WE_10.html#cb79-5" aria-hidden="true" tabindex="-1"></a><span class="fu">legend</span>(<span class="st">&quot;bottomleft&quot;</span>, <span class="at">legend =</span> <span class="fu">c</span>(<span class="st">&quot;Costdist&quot;</span>,<span class="st">&quot;Commdist&quot;</span>), </span>
<span id="cb79-6"><a href="WE_10.html#cb79-6" aria-hidden="true" tabindex="-1"></a>       <span class="at">pch =</span> <span class="dv">19</span>, <span class="at">col =</span> <span class="fu">c</span>(<span class="st">&quot;red&quot;</span>, <span class="st">&quot;blue&quot;</span>))</span></code></pre></div>
<p><img src="10-Week10_files/figure-html/unnamed-chunk-29-1.png" width="672" /></p>
<p><strong>Question 7</strong>: What effect does changing resolution have on these metrics?</p>
</div>
</div>
<div id="references" class="section level3">
<h3>7. References</h3>
<p>Evans IS (1972) General geomorphometry, derivatives of altitude, and descriptive statistics. In: Chorley RJ (ed) Spatial analysis in geomorphology. Harper &amp; Row, New York.</p>
<p>Funk WC, Blouin MS, Corn PS, Maxell BA, Pilliod DS, Amish S, Allendorf FW (2005) Population structure of Columbia spotted frogs (Rana luteiventris) is strongly affected by the landscape. Mol Ecol 14(2): 483-496</p>
<p>Marrotte RR, Bowman J (2017) The relationship between least-cost and resistance distance. PLOS ONE 12(3): e0174212</p>
<p>McRae BH, Dickson BG, Keitt TH, Shah VB (2008) Using circuit theory to model connectivity in ecology, evolution, and conservation. Ecology 89 (10):2712-2724</p>
<p>Moore I, Gessler P, Nielsen G, Petersen G (eds) (1993) Terrain attributes and estimation methods and scale effects. Modeling change in environmental systems. Wiley, London.</p>
<p>Murphy MA, Dezanni RJ, Pilliod D, Storfer A (2010) Landscape genetics of high mountain frog populations. Mol Ecol 19:3634-3649.</p>
<p>Pilliod DS, Peterson CR, Ritson PI (2002) Seasonal migration of Columbia spotted frogs (Rana luteiventris) among complementary resources in a high mountain basin. Can J Zool 80: 1849-1862</p>
<p>Rehfeldt GE, Crookston NL, Warwell MV, Evans JS (2006) Empirical analysis of plan-climate relationships for western United States. International Journal of Plan Sciences 167: 1123-1150</p>

</div>
</div>
<!-- </div> -->
            </section>

          </div>
        </div>
      </div>
<a href="Week10.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="Week11.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": true,
"facebook": false,
"twitter": false,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": false
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": null,
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "section"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
