<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>8.3 Worked Example | Landscape Genetic Data Analysis with R</title>
  <meta name="description" content="This is a web-interface to the teaching materials for the lab course ‘Landscape Genetic Data Analysis with R’ associated with the distributed graduate course ‘DGS Landscape Genetics.’ The output format is bookdown::gitbook." />
  <meta name="generator" content="bookdown 0.25 and GitBook 2.6.7" />

  <meta property="og:title" content="8.3 Worked Example | Landscape Genetic Data Analysis with R" />
  <meta property="og:type" content="book" />
  
  <meta property="og:description" content="This is a web-interface to the teaching materials for the lab course ‘Landscape Genetic Data Analysis with R’ associated with the distributed graduate course ‘DGS Landscape Genetics.’ The output format is bookdown::gitbook." />
  <meta name="github-repo" content="hhwagner1/LandGenCourse_book" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="8.3 Worked Example | Landscape Genetic Data Analysis with R" />
  
  <meta name="twitter:description" content="This is a web-interface to the teaching materials for the lab course ‘Landscape Genetic Data Analysis with R’ associated with the distributed graduate course ‘DGS Landscape Genetics.’ The output format is bookdown::gitbook." />
  

<meta name="author" content="Editor: Helene Wagner (University of Toronto)" />


<meta name="date" content="2022-04-03" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="tutorial_5.html"/>
<link rel="next" href="r-exercise-week-5.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>
<script src="libs/kePrint-0.0.1/kePrint.js"></script>
<link href="libs/lightable-0.0.1/lightable.css" rel="stylesheet" />
<script src="libs/htmlwidgets-1.5.4/htmlwidgets.js"></script>
<script src="libs/d3-3.5.6/d3.min.js"></script>
<link href="libs/profvis-0.3.6.9000/profvis.css" rel="stylesheet" />
<script src="libs/profvis-0.3.6.9000/profvis.js"></script>
<script src="libs/profvis-0.3.6.9000/scroll.js"></script>
<link href="libs/highlight-6.2.0/textmate.css" rel="stylesheet" />
<script src="libs/highlight-6.2.0/highlight.js"></script>
<script src="libs/profvis-binding-0.3.7/profvis.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>


<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">Landscape Genetic Data Analysis with R</a></li>

<li class="divider"></li>
<li class="part"><span><b>Getting Started<span></span></b></span></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Introduction<span></span></a>
<ul>
<li class="chapter" data-level="1.1" data-path="how-to-use-this-book.html"><a href="how-to-use-this-book.html"><i class="fa fa-check"></i><b>1.1</b> How to use this Book<span></span></a>
<ul>
<li><a href="how-to-use-this-book.html#book-structure">1. Book Structure<span></span></a></li>
<li><a href="how-to-use-this-book.html#find-what-is-relevant-for-you">2. Find what is relevant for you<span></span></a></li>
<li><a href="how-to-use-this-book.html#course-r-package-landgencourse">3. Course R package ‘LandGenCourse’<span></span></a></li>
</ul></li>
<li class="chapter" data-level="1.2" data-path="list-of-packages.html"><a href="list-of-packages.html"><i class="fa fa-check"></i><b>1.2</b> List of R Packages by Vignette<span></span></a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="review-of-r-skills.html"><a href="review-of-r-skills.html"><i class="fa fa-check"></i><b>2</b> Review of R Skills<span></span></a>
<ul>
<li class="chapter" data-level="2.1" data-path="basic-r.html"><a href="basic-r.html"><i class="fa fa-check"></i><b>2.1</b> Basic R Programming<span></span></a>
<ul>
<li><a href="basic-r.html#overview">1. Overview<span></span></a></li>
<li><a href="basic-r.html#data-types">2. Data Types<span></span></a></li>
<li><a href="basic-r.html#data-containers">3. Data Containers<span></span></a></li>
<li><a href="basic-r.html#programming">4. Programming<span></span></a></li>
</ul></li>
<li class="chapter" data-level="2.2" data-path="r-graphics.html"><a href="r-graphics.html"><i class="fa fa-check"></i><b>2.2</b> R Graphics<span></span></a>
<ul>
<li><a href="r-graphics.html#overview-1">1. Overview<span></span></a></li>
<li><a href="r-graphics.html#univariate-plots">2. Univariate plots<span></span></a></li>
<li><a href="r-graphics.html#bivariate-plots">3. Bivariate plots<span></span></a></li>
<li><a href="r-graphics.html#advanced-plotting">4. Advanced plotting<span></span></a></li>
</ul></li>
<li class="chapter" data-level="2.3" data-path="resources.html"><a href="resources.html"><i class="fa fa-check"></i><b>2.3</b> Further R resources<span></span></a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="github-for-group-projects.html"><a href="github-for-group-projects.html"><i class="fa fa-check"></i><b>3</b> GitHub for Group Projects<span></span></a>
<ul>
<li class="chapter" data-level="3.1" data-path="github.html"><a href="github.html"><i class="fa fa-check"></i><b>3.1</b> Why use GitHub?<span></span></a>
<ul>
<li><a href="github.html#goals">1. Goals<span></span></a></li>
<li><a href="github.html#version">2. Watch course video: Version Control 101<span></span></a></li>
<li><a href="github.html#slides">3. Preview slides<span></span></a></li>
<li><a href="github.html#next-steps">4. Next steps<span></span></a></li>
</ul></li>
<li class="chapter" data-level="3.2" data-path="install.html"><a href="install.html"><i class="fa fa-check"></i><b>3.2</b> Installation<span></span></a>
<ul>
<li><a href="install.html#goals-1">1. Goals<span></span></a></li>
<li class="chapter" data-level="3.2.1" data-path="install.html"><a href="install.html#how-to-get-github-and-gitkraken-for-free"><i class="fa fa-check"></i><b>3.2.1</b> 2. How to get GitHub and GitKraken for free<span></span></a></li>
<li><a href="install.html#happygit">3. Happy Git with R<span></span></a></li>
<li><a href="install.html#gitvideos">4. Video tutorials: installation<span></span></a></li>
<li><a href="install.html#join">5. Joining your group’s project on GitHub<span></span></a></li>
<li><a href="install.html#repo">6. Setting up a group repo<span></span></a></li>
<li><a href="install.html#next-steps-1">7. Next steps<span></span></a></li>
</ul></li>
<li class="chapter" data-level="3.3" data-path="workflow.html"><a href="workflow.html"><i class="fa fa-check"></i><b>3.3</b> Workflows<span></span></a>
<ul>
<li><a href="workflow.html#goals-2">1. Goals<span></span></a></li>
<li><a href="workflow.html#step-by-step-tutorials">2. Step-by-step tutorials<span></span></a></li>
<li><a href="workflow.html#further-resources">3. Further resources<span></span></a></li>
</ul></li>
</ul></li>
<li class="part"><span><b>Basic Topics<span></span></b></span></li>
<li class="chapter" data-level="4" data-path="Week1.html"><a href="Week1.html"><i class="fa fa-check"></i><b>4</b> Week 1: Importing Genetic Data<span></span></a>
<ul>
<li class="chapter" data-level="4.1" data-path="video_1.html"><a href="video_1.html"><i class="fa fa-check"></i><b>4.1</b> View Course Video<span></span></a>
<ul>
<li><a href="video_1.html#embedded-video">1. Embedded Video<span></span></a></li>
<li><a href="video_1.html#preview-slides">2. Preview Slides<span></span></a></li>
</ul></li>
<li class="chapter" data-level="4.2" data-path="tutorial_1.html"><a href="tutorial_1.html"><i class="fa fa-check"></i><b>4.2</b> Interactive Tutorial 1<span></span></a>
<ul>
<li><a href="tutorial_1.html#list-of-r-commands-covered-this-week">1. List of R commands covered this week<span></span></a></li>
<li><a href="tutorial_1.html#general-instructions">2. General Instructions<span></span></a></li>
</ul></li>
<li class="chapter" data-level="4.3" data-path="WE_1.html"><a href="WE_1.html"><i class="fa fa-check"></i><b>4.3</b> Worked Example<span></span></a>
<ul>
<li><a href="WE_1.html#overview-of-worked-example">1. Overview of Worked Example<span></span></a></li>
<li><a href="WE_1.html#import-data-from-.csv-file">2. Import data from .csv file<span></span></a></li>
<li><a href="WE_1.html#create-genind-object">3. Create ‘genind’ Object<span></span></a></li>
<li><a href="WE_1.html#view-information-stored-in-genind-object">4. View information stored in ‘genind’ object<span></span></a></li>
<li><a href="WE_1.html#import-data-with-gstudio-package">5. Import data with ‘gstudio’ package<span></span></a></li>
<li><a href="WE_1.html#importing-snp-data">6. Importing SNP data<span></span></a></li>
</ul></li>
<li class="chapter" data-level="4.4" data-path="r-exercise-week-1.html"><a href="r-exercise-week-1.html"><i class="fa fa-check"></i><b>4.4</b> R exercise Week 1<span></span></a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="Week2.html"><a href="Week2.html"><i class="fa fa-check"></i><b>5</b> Week 2: Spatial Data<span></span></a>
<ul>
<li class="chapter" data-level="5.1" data-path="video_2.html"><a href="video_2.html"><i class="fa fa-check"></i><b>5.1</b> View Course Video<span></span></a>
<ul>
<li><a href="video_2.html#embedded-video-1">1. Embedded Video<span></span></a></li>
<li><a href="video_2.html#preview-slides-1">2. Preview Slides<span></span></a></li>
</ul></li>
<li class="chapter" data-level="5.2" data-path="tutorial_2.html"><a href="tutorial_2.html"><i class="fa fa-check"></i><b>5.2</b> Interactive Tutorial 2<span></span></a>
<ul>
<li><a href="tutorial_2.html#list-of-r-commands-covered-this-week-1">1. List of R commands covered this week<span></span></a></li>
<li><a href="tutorial_2.html#general-instructions-1">2. General Instructions<span></span></a></li>
</ul></li>
<li class="chapter" data-level="5.3" data-path="WE_2.html"><a href="WE_2.html"><i class="fa fa-check"></i><b>5.3</b> Worked Example<span></span></a>
<ul>
<li><a href="WE_2.html#overview-of-worked-example-1">1. Overview of Worked Example<span></span></a></li>
<li><a href="WE_2.html#import-site-data-from-.csv-file">2. Import site data from .csv file<span></span></a></li>
<li><a href="WE_2.html#display-raster-data-and-overlay-sampling-locations-extract-data">3. Display raster data and overlay sampling locations, extract data<span></span></a></li>
<li><a href="WE_2.html#calculate-landscape-metrics">4. Calculate landscape metrics<span></span></a></li>
<li><a href="WE_2.html#sample-landscape-metrics-within-buffer-around-sampling-locations">5. Sample landscape metrics within buffer around sampling locations<span></span></a></li>
</ul></li>
<li class="chapter" data-level="5.4" data-path="r-exercise-week-2.html"><a href="r-exercise-week-2.html"><i class="fa fa-check"></i><b>5.4</b> R Exercise Week 2<span></span></a></li>
<li class="chapter" data-level="5.5" data-path="bonus_2a.html"><a href="bonus_2a.html"><i class="fa fa-check"></i><b>5.5</b> Bonus: ‘sf’ and ‘terra’<span></span></a>
<ul>
<li><a href="bonus_2a.html#overview-2">1. Overview<span></span></a></li>
<li><a href="bonus_2a.html#convert-spatialpointsdataframe-to-sf-object">2. Convert ‘SpatialPointsDataFrame’ to ‘sf’ object<span></span></a></li>
<li><a href="bonus_2a.html#plot-variables-first-10-in-data-set">3. Plot variables (first 10 in data set)<span></span></a></li>
<li><a href="bonus_2a.html#export-site-data-as-esri-shapefile">4. Export site data as ESRI shapefile<span></span></a></li>
<li><a href="bonus_2a.html#use-terra-and-tmap-to-display-categorical-map-with-color-scheme">5. Use ‘terra’ and ‘tmap’ to display categorical map with color scheme<span></span></a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="6" data-path="Week3.html"><a href="Week3.html"><i class="fa fa-check"></i><b>6</b> Week 3: Genetic Diversity<span></span></a>
<ul>
<li class="chapter" data-level="6.1" data-path="video_3.html"><a href="video_3.html"><i class="fa fa-check"></i><b>6.1</b> View Course Video<span></span></a>
<ul>
<li><a href="video_3.html#embedded-video-2">1. Embedded Video<span></span></a></li>
<li><a href="video_3.html#video-part-1">Video, Part 1<span></span></a></li>
<li><a href="video_3.html#video-part-2">Video, Part 2<span></span></a></li>
<li><a href="video_3.html#preview-slides-2">Preview Slides<span></span></a></li>
</ul></li>
<li class="chapter" data-level="6.2" data-path="tutorial_3.html"><a href="tutorial_3.html"><i class="fa fa-check"></i><b>6.2</b> Interactive Tutorial 3<span></span></a>
<ul>
<li><a href="tutorial_3.html#list-of-r-commands-covered-this-week-2">1. List of R commands covered this week<span></span></a></li>
<li><a href="tutorial_3.html#general-instructions-2">2. General Instructions<span></span></a></li>
</ul></li>
<li class="chapter" data-level="6.3" data-path="WE_3.html"><a href="WE_3.html"><i class="fa fa-check"></i><b>6.3</b> Worked Example<span></span></a>
<ul>
<li><a href="WE_3.html#overview-of-worked-example-2">1. Overview of Worked Example<span></span></a></li>
<li><a href="WE_3.html#basic-checking-of-markers-and-populations">2. Basic checking of markers and populations<span></span></a></li>
<li><a href="WE_3.html#assess-genetic-diversity">3. Assess genetic diversity<span></span></a></li>
<li><a href="WE_3.html#aggregate-genetic-data-at-population-level-allele-frequencies">4. Aggregate genetic data at population level (allele frequencies)<span></span></a></li>
</ul></li>
<li class="chapter" data-level="6.4" data-path="r-exercise-week-3.html"><a href="r-exercise-week-3.html"><i class="fa fa-check"></i><b>6.4</b> R Exercise Week 3<span></span></a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="Week4.html"><a href="Week4.html"><i class="fa fa-check"></i><b>7</b> Week 4: Metapopulations<span></span></a>
<ul>
<li class="chapter" data-level="7.1" data-path="video_4.html"><a href="video_4.html"><i class="fa fa-check"></i><b>7.1</b> View Course Video<span></span></a>
<ul>
<li><a href="video_4.html#embedded-video-3">1. Embedded Video<span></span></a></li>
<li><a href="video_4.html#video-part-1-1">Video, Part 1<span></span></a></li>
<li><a href="video_4.html#video-part-2-1">Video, Part 2<span></span></a></li>
<li><a href="video_4.html#preview-slides-3">Preview Slides<span></span></a></li>
</ul></li>
<li class="chapter" data-level="7.2" data-path="tutorial_4.html"><a href="tutorial_4.html"><i class="fa fa-check"></i><b>7.2</b> Interactive Tutorial 4<span></span></a>
<ul>
<li><a href="tutorial_4.html#list-of-r-commands-covered-this-week-3">1. List of R commands covered this week<span></span></a></li>
<li><a href="tutorial_4.html#general-instructions-3">2. General Instructions<span></span></a></li>
</ul></li>
<li class="chapter" data-level="7.3" data-path="WE_4.html"><a href="WE_4.html"><i class="fa fa-check"></i><b>7.3</b> Worked Example<span></span></a>
<ul>
<li><a href="WE_4.html#overview-of-worked-example-3">1. Overview of Worked Example<span></span></a></li>
<li><a href="WE_4.html#spatial-distribution-of-genetic-structure">2. Spatial distribution of genetic structure<span></span></a></li>
<li><a href="WE_4.html#what-determines-genetic-differentiation-among-sites">3. What determines genetic differentiation among sites?<span></span></a></li>
<li><a href="WE_4.html#what-determines-genetic-diversity">4. What determines genetic diversity?<span></span></a></li>
<li><a href="WE_4.html#are-genetic-differentiation-and-diversity-related">5. Are genetic differentiation and diversity related?<span></span></a></li>
<li><a href="WE_4.html#effect-of-recent-extinction-events">6. Effect of recent extinction events<span></span></a></li>
</ul></li>
<li class="chapter" data-level="7.4" data-path="r-exercise-week-4.html"><a href="r-exercise-week-4.html"><i class="fa fa-check"></i><b>7.4</b> R Exercise Week 4<span></span></a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="Week5.html"><a href="Week5.html"><i class="fa fa-check"></i><b>8</b> Week 5: Spatial Statistics<span></span></a>
<ul>
<li class="chapter" data-level="8.1" data-path="video_5.html"><a href="video_5.html"><i class="fa fa-check"></i><b>8.1</b> View Course Video<span></span></a>
<ul>
<li><a href="video_5.html#embedded-video-4">1. Embedded Video<span></span></a></li>
<li><a href="video_5.html#preview-slides-4">Preview Slides<span></span></a></li>
</ul></li>
<li class="chapter" data-level="8.2" data-path="tutorial_5.html"><a href="tutorial_5.html"><i class="fa fa-check"></i><b>8.2</b> Interactive Tutorial 5<span></span></a>
<ul>
<li><a href="tutorial_5.html#list-of-r-commands-covered-this-week-4">1. List of R commands covered this week<span></span></a></li>
<li><a href="tutorial_5.html#general-instructions-4">2. General Instructions<span></span></a></li>
</ul></li>
<li class="chapter" data-level="8.3" data-path="WE_5.html"><a href="WE_5.html"><i class="fa fa-check"></i><b>8.3</b> Worked Example<span></span></a>
<ul>
<li><a href="WE_5.html#overview-of-worked-example-4">1. Overview of Worked Example<span></span></a></li>
<li><a href="WE_5.html#data-import-and-manipulation">2. Data import and manipulation<span></span></a></li>
<li><a href="WE_5.html#calculate-genetic-distances">3. Calculate genetic distances<span></span></a></li>
<li><a href="WE_5.html#perform-a-mantel-test-to-test-for-ibd">4. Perform a Mantel test to test for IBD<span></span></a></li>
<li><a href="WE_5.html#create-mantel-correlogram-for-genetic-data">5. Create Mantel correlogram for genetic data<span></span></a></li>
<li><a href="WE_5.html#specify-spatial-weights-and-calculate-morans-i">6. Specify spatial weights and calculate Moran’s I<span></span></a></li>
</ul></li>
<li class="chapter" data-level="8.4" data-path="r-exercise-week-5.html"><a href="r-exercise-week-5.html"><i class="fa fa-check"></i><b>8.4</b> R Exercise Week 5<span></span></a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="Week6.html"><a href="Week6.html"><i class="fa fa-check"></i><b>9</b> Week 6: Quantitative Genetics<span></span></a>
<ul>
<li class="chapter" data-level="9.1" data-path="video_6.html"><a href="video_6.html"><i class="fa fa-check"></i><b>9.1</b> View Course Video<span></span></a>
<ul>
<li><a href="video_6.html#embedded-video-5">1. Embedded Video<span></span></a></li>
<li><a href="video_6.html#video-part-1-2">Video, Part 1<span></span></a></li>
<li><a href="video_6.html#video-part-2-2">Video, Part 2<span></span></a></li>
<li><a href="video_6.html#preview-slides-5">Preview Slides<span></span></a></li>
</ul></li>
<li class="chapter" data-level="9.2" data-path="tutorial_6.html"><a href="tutorial_6.html"><i class="fa fa-check"></i><b>9.2</b> Interactive Tutorial 6<span></span></a>
<ul>
<li><a href="tutorial_6.html#list-of-r-commands-covered-this-week-5">1. List of R commands covered this week<span></span></a></li>
<li><a href="tutorial_6.html#general-instructions-5">2. General Instructions<span></span></a></li>
</ul></li>
<li class="chapter" data-level="9.3" data-path="WE_6.html"><a href="WE_6.html"><i class="fa fa-check"></i><b>9.3</b> Worked Example<span></span></a>
<ul>
<li><a href="WE_6.html#overview-of-worked-example-5">1. Overview of Worked Example<span></span></a></li>
<li><a href="WE_6.html#estimate-genetic-and-non-genetic-variance-components-from-a-common-garden-experiment">2. Estimate genetic and non-genetic variance components from a common garden experiment<span></span></a></li>
<li><a href="WE_6.html#estimate-trait-heritability">3. Estimate trait heritability<span></span></a></li>
<li><a href="WE_6.html#compare-q_st-to-f_st-measured-from-snp-data">4. Compare <span class="math inline">\(Q_{st}\)</span> to <span class="math inline">\(F_{st}\)</span> measured from SNP data<span></span></a></li>
<li><a href="WE_6.html#assess-correlation-between-trait-and-environment">5. Assess correlation between trait and environment<span></span></a></li>
</ul></li>
<li class="chapter" data-level="9.4" data-path="r-exercise-week-6.html"><a href="r-exercise-week-6.html"><i class="fa fa-check"></i><b>9.4</b> R Exercise Week 6<span></span></a></li>
</ul></li>
<li class="chapter" data-level="10" data-path="Week7.html"><a href="Week7.html"><i class="fa fa-check"></i><b>10</b> Week 7: Spatial Linear Models<span></span></a>
<ul>
<li class="chapter" data-level="10.1" data-path="video_7.html"><a href="video_7.html"><i class="fa fa-check"></i><b>10.1</b> View Course Video<span></span></a>
<ul>
<li><a href="video_7.html#embedded-video-6">1. Embedded Video<span></span></a></li>
<li><a href="video_7.html#video-part-1-3">Video, Part 1<span></span></a></li>
<li><a href="video_7.html#video-part-2-3">Video, Part 2<span></span></a></li>
<li><a href="video_7.html#preview-slides-6">Preview Slides<span></span></a></li>
</ul></li>
<li class="chapter" data-level="10.2" data-path="tutorial_7.html"><a href="tutorial_7.html"><i class="fa fa-check"></i><b>10.2</b> Interactive Tutorial 7<span></span></a>
<ul>
<li><a href="tutorial_7.html#list-of-r-commands-covered-this-week-6">1. List of R commands covered this week<span></span></a></li>
<li><a href="tutorial_7.html#general-instructions-6">2. General Instructions<span></span></a></li>
</ul></li>
<li class="chapter" data-level="10.3" data-path="WE_7.html"><a href="WE_7.html"><i class="fa fa-check"></i><b>10.3</b> Worked Example<span></span></a>
<ul>
<li><a href="WE_7.html#overview-of-worked-example-6">1. Overview of Worked Example<span></span></a></li>
<li><a href="WE_7.html#explore-data-set">2. Explore data set<span></span></a></li>
<li><a href="WE_7.html#test-regression-residuals-for-spatial-autocorrelation">3. Test regression residuals for spatial autocorrelation<span></span></a></li>
<li><a href="WE_7.html#fit-models-with-spatially-correlated-error-gls-with-package-nlme">4. Fit models with spatially correlated error (GLS) with package ‘nlme’<span></span></a></li>
<li><a href="WE_7.html#fit-spatial-simultaneous-autoregressive-error-models-sar">5. Fit spatial simultaneous autoregressive error models (SAR)<span></span></a></li>
<li><a href="WE_7.html#spatial-filtering-with-mem-using-package-spmoran">6. Spatial filtering with MEM using package ‘spmoran’<span></span></a></li>
<li><a href="WE_7.html#fit-spatially-varying-coefficients-model-with-package-spmoran">7. Fit spatially varying coefficients model with package ‘spmoran’<span></span></a></li>
<li><a href="WE_7.html#model-with-patchsize-and-ibr">Model with PatchSize and IBR<span></span></a></li>
<li><a href="WE_7.html#model-with-ibr-only">Model with IBR only<span></span></a></li>
<li><a href="WE_7.html#conclusions">8. Conclusions<span></span></a></li>
</ul></li>
<li class="chapter" data-level="10.4" data-path="r-exercise-week-7.html"><a href="r-exercise-week-7.html"><i class="fa fa-check"></i><b>10.4</b> R Exercise Week 7<span></span></a></li>
<li class="chapter" data-level="10.5" data-path="bonus_7a.html"><a href="bonus_7a.html"><i class="fa fa-check"></i><b>10.5</b> Bonus: Si index<span></span></a>
<ul>
<li><a href="bonus_7a.html#overview-of-bonus-materials">1. Overview of Bonus Materials<span></span></a></li>
<li><a href="bonus_7a.html#import-ecological-distance-matrices">2. Import ecological distance matrices<span></span></a></li>
<li><a href="bonus_7a.html#optimize-the-scaling-parameter-alpha">3. Optimize the scaling parameter alpha<span></span></a></li>
<li><a href="bonus_7a.html#calculate-hanskis-index-si-with-source-patch-parameters">4. Calculate Hanski’s index Si with source patch parameters<span></span></a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="11" data-path="Week8.html"><a href="Week8.html"><i class="fa fa-check"></i><b>11</b> Week 8: Simulation Experiments<span></span></a>
<ul>
<li class="chapter" data-level="11.1" data-path="video_8.html"><a href="video_8.html"><i class="fa fa-check"></i><b>11.1</b> View Course Video<span></span></a>
<ul>
<li><a href="video_8.html#embedded-video-7">1. Embedded Video<span></span></a></li>
<li><a href="video_8.html#video-part-1-4">Video, Part 1<span></span></a></li>
<li><a href="video_8.html#video-part-2-4">Video, Part 2<span></span></a></li>
<li><a href="video_8.html#preview-slides-7">Preview Slides<span></span></a></li>
</ul></li>
<li class="chapter" data-level="11.2" data-path="tutorial_8.html"><a href="tutorial_8.html"><i class="fa fa-check"></i><b>11.2</b> Interactive Tutorial 8<span></span></a>
<ul>
<li><a href="tutorial_8.html#list-of-r-commands-covered-this-week-7">1. List of R commands covered this week<span></span></a></li>
<li><a href="tutorial_8.html#general-instructions-7">2. General Instructions<span></span></a></li>
</ul></li>
<li class="chapter" data-level="11.3" data-path="WE_8.html"><a href="WE_8.html"><i class="fa fa-check"></i><b>11.3</b> Worked Example<span></span></a>
<ul>
<li><a href="WE_8.html#overview-of-worked-example-7">1. Overview of Worked Example<span></span></a></li>
<li><a href="WE_8.html#initialize-a-landscape">2. Initialize a landscape<span></span></a></li>
<li><a href="WE_8.html#define-simulation-parameters">3. Define simulation parameters<span></span></a></li>
<li><a href="WE_8.html#run-simulations-and-analyze-results">4. Run simulations and analyze results<span></span></a></li>
<li><a href="WE_8.html#run-simulator-using-a-previously-defined-parameter-set">5. Run simulator using a previously defined parameter set<span></span></a></li>
</ul></li>
<li class="chapter" data-level="11.4" data-path="r-exercise-week-8.html"><a href="r-exercise-week-8.html"><i class="fa fa-check"></i><b>11.4</b> R Exercise Week 8<span></span></a></li>
<li class="chapter" data-level="11.5" data-path="bonus_8a.html"><a href="bonus_8a.html"><i class="fa fa-check"></i><b>11.5</b> Bonus: Efficient R<span></span></a>
<ul>
<li><a href="bonus_8a.html#overview-of-bonus-material">1. Overview of Bonus Material<span></span></a></li>
<li><a href="bonus_8a.html#navigating-the-file-system">2. Navigating the file system<span></span></a></li>
<li><a href="bonus_8a.html#benchmarking-file-import-and-export-options">3. Benchmarking file import and export options<span></span></a></li>
<li><a href="bonus_8a.html#profiling-your-code">4. Profiling your code<span></span></a></li>
<li><a href="bonus_8a.html#creating-and-executing-a-bash-r-script">5. Creating and executing a Bash R script<span></span></a></li>
<li><a href="bonus_8a.html#parallelizing-code">6. Parallelizing code<span></span></a></li>
</ul></li>
</ul></li>
<li class="part"><span><b>Advanced Topics<span></span></b></span></li>
<li class="chapter" data-level="12" data-path="Week9.html"><a href="Week9.html"><i class="fa fa-check"></i><b>12</b> Week 9: Population Structure<span></span></a>
<ul>
<li class="chapter" data-level="12.1" data-path="WE_9.html"><a href="WE_9.html"><i class="fa fa-check"></i><b>12.1</b> Worked Example<span></span></a>
<ul>
<li><a href="WE_9.html#overview-of-worked-example-8">1. Overview of Worked Example<span></span></a></li>
<li><a href="WE_9.html#simulated-data-2-island-model">2. Simulated data: 2-island model<span></span></a></li>
<li><a href="WE_9.html#simulated-data-2-island-model-with-admixture">3. Simulated data: 2-island model with admixture<span></span></a></li>
<li><a href="WE_9.html#empirical-data-threespine-sticklebacks">4. Empirical data: Threespine sticklebacks<span></span></a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="13" data-path="Week10.html"><a href="Week10.html"><i class="fa fa-check"></i><b>13</b> Week 10: Landscape Resistance<span></span></a>
<ul>
<li class="chapter" data-level="13.1" data-path="WE_10.html"><a href="WE_10.html"><i class="fa fa-check"></i><b>13.1</b> Worked Example<span></span></a>
<ul>
<li><a href="WE_10.html#overview-of-worked-example-9">1. Overview of Worked Example<span></span></a></li>
<li><a href="WE_10.html#explore-the-data-set">2. Explore the data set<span></span></a></li>
<li><a href="WE_10.html#setting-cost-values-and-calculating-conductance">3. Setting cost values and calculating conductance<span></span></a></li>
<li><a href="WE_10.html#convert-conductance-into-effective-distance">4. Convert conductance into effective distance<span></span></a></li>
<li><a href="WE_10.html#create-cost-distance-matrices">5. Create cost-distance matrices<span></span></a></li>
<li><a href="WE_10.html#how-does-changing-resolution-affect-these-metrics">6. How does changing resolution affect these metrics?<span></span></a></li>
<li><a href="WE_10.html#references">7. References<span></span></a></li>
</ul></li>
<li class="chapter" data-level="13.2" data-path="bonus-10a.html"><a href="bonus-10a.html"><i class="fa fa-check"></i><b>13.2</b> Bonus: ‘radish’ tutorial<span></span></a>
<ul>
<li><a href="bonus-10a.html#overview-of-bonus-materials-1">1. Overview of Bonus Materials<span></span></a></li>
<li><a href="bonus-10a.html#import-data">2. Import data<span></span></a></li>
<li><a href="bonus-10a.html#basic-model-fitting">3. Basic model fitting<span></span></a></li>
<li><a href="bonus-10a.html#advanced-modeling-topics">4. Advanced modeling topics<span></span></a></li>
<li><a href="bonus-10a.html#summary">5. Summary<span></span></a></li>
<li><a href="bonus-10a.html#references-1">6. References<span></span></a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="14" data-path="Week11.html"><a href="Week11.html"><i class="fa fa-check"></i><b>14</b> Week 11: Detecting Adaptation<span></span></a>
<ul>
<li class="chapter" data-level="14.1" data-path="WE_11.html"><a href="WE_11.html"><i class="fa fa-check"></i><b>14.1</b> Worked Example<span></span></a>
<ul>
<li><a href="WE_11.html#overview-of-worked-example-10">1. Overview of Worked Example<span></span></a></li>
<li><a href="WE_11.html#import-and-prepare-the-data">2. Import and prepare the data<span></span></a></li>
<li><a href="WE_11.html#latent-factor-mixed-models-lfmm-a-univariate-gea">3. Latent Factor Mixed Models (LFMM): a univariate GEA<span></span></a></li>
<li><a href="WE_11.html#redundancy-analysis-rda-a-multivariate-gea">4. Redundancy Analysis (RDA): a multivariate GEA<span></span></a></li>
<li><a href="WE_11.html#compare-lfmm-and-rda-candidates">5. Compare LFMM and RDA candidates<span></span></a></li>
<li><a href="WE_11.html#a-quick-note-on-controlling-for-population-structure-in-rda">6. A quick note on controlling for population structure in RDA<span></span></a></li>
<li><a href="WE_11.html#references-2">7. References<span></span></a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="15" data-path="Week12.html"><a href="Week12.html"><i class="fa fa-check"></i><b>15</b> Week 12: Model Selection<span></span></a>
<ul>
<li class="chapter" data-level="15.1" data-path="WE_12.html"><a href="WE_12.html"><i class="fa fa-check"></i><b>15.1</b> Worked Example<span></span></a>
<ul>
<li><a href="WE_12.html#overview-of-worked-example-11">1. Overview of Worked Example<span></span></a></li>
<li><a href="WE_12.html#fitting-candidate-models">2. Fitting candidate models<span></span></a></li>
<li><a href="WE_12.html#your-turn-to-test-additional-hypotheses">3. Your turn to test additional hypotheses!<span></span></a></li>
<li><a href="WE_12.html#nested-model-nmlpe-for-hierarchical-sampling-designs">4. Nested model (NMLPE) for hierarchical sampling designs<span></span></a></li>
<li><a href="WE_12.html#references-3">5. References<span></span></a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="16" data-path="Week13.html"><a href="Week13.html"><i class="fa fa-check"></i><b>16</b> Week 13: Gravity Models<span></span></a>
<ul>
<li class="chapter" data-level="16.1" data-path="WE_13.html"><a href="WE_13.html"><i class="fa fa-check"></i><b>16.1</b> Worked Example<span></span></a>
<ul>
<li><a href="WE_13.html#overview-of-worked-example-12">1. Overview of Worked Example<span></span></a></li>
<li><a href="WE_13.html#setup">2. Setup<span></span></a></li>
<li><a href="WE_13.html#wetland-complex-data-preparation">3. Wetland complex data preparation<span></span></a></li>
<li><a href="WE_13.html#wetland-field-data-preparation">4. Wetland field-data preparation<span></span></a></li>
<li><a href="WE_13.html#saturated-graph">5. Saturated Graph<span></span></a></li>
<li><a href="WE_13.html#spatial-model-data-prepration">6. Spatial model data prepration<span></span></a></li>
<li><a href="WE_13.html#gravity-model">7. Gravity model<span></span></a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="17" data-path="Week14.html"><a href="Week14.html"><i class="fa fa-check"></i><b>17</b> Week 14: Contemporary Gene Flow<span></span></a>
<ul>
<li class="chapter" data-level="17.1" data-path="WE_14.html"><a href="WE_14.html"><i class="fa fa-check"></i><b>17.1</b> Worked Example<span></span></a>
<ul>
<li><a href="WE_14.html#overview-3">1. Overview<span></span></a></li>
<li><a href="WE_14.html#import-genotypes">2. Import Genotypes<span></span></a></li>
<li><a href="WE_14.html#pollen-pool-genetic-structure">3. Pollen pool genetic structure<span></span></a></li>
<li><a href="WE_14.html#paternity-analysis">4. Paternity analysis<span></span></a></li>
<li><a href="WE_14.html#linking-paternity-to-ecological-variables">5. Linking paternity to ecological variables<span></span></a></li>
<li><a href="WE_14.html#references-4">6. References<span></span></a></li>
</ul></li>
</ul></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Landscape Genetic Data Analysis with R</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="WE_5" class="section level2 hasAnchor" number="8.3">
<h2><span class="header-section-number">8.3</span> Worked Example<a href="WE_5.html#WE_5" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Helene Wagner</p>
<div id="overview-of-worked-example-4" class="section level3 unnumbered hasAnchor">
<h3>1. Overview of Worked Example<a href="WE_5.html#overview-of-worked-example-4" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div id="a.-goals-5" class="section level4 unnumbered hasAnchor">
<h4>a. Goals<a href="WE_5.html#a.-goals-5" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>This worked example shows how to:</p>
<ul>
<li>Calculate genetic distance at population level.</li>
<li>Perform a Mantel test to test for IBD.</li>
<li>Create Mantel correlograms for the genetic data.</li>
<li>Calculate and test Moran’s I for population-level genetic diversity data.</li>
</ul>
</div>
<div id="b.-data-set-4" class="section level4 unnumbered hasAnchor">
<h4>b. Data set<a href="WE_5.html#b.-data-set-4" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>This is a larger version of the frog data set used in Weeks 1 - 3.</p>
<p>Here, we will analyze microsatellite data from Funk et al. (2005) and Murphy et al. (2010) for 418 individuals of Colombia spotted frogs (Rana luteiventris) from 30 populations, together with site-level spatial coordinates. Please see the introductory video on the DGS course website, R lab page.</p>
<ul>
<li><strong>ralu_loci_allpops.csv</strong>: Data frame with populations and genetic data (406 rows x 9 columns). Included as external file in package <code>LandGenCourse</code>.</li>
<li><strong>Frogs_diversity_allpops.csv</strong>: Table with genetic diversity variables similar to Week 3.</li>
<li><strong>ralu_coords_allpops.csv</strong>: Spatial coordinates (both in UTM and LongLat format), and basin.</li>
</ul>
</div>
<div id="c.-required-r-packages-3" class="section level4 unnumbered hasAnchor">
<h4>c. Required R packages<a href="WE_5.html#c.-required-r-packages-3" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>Install some packages (not on CRAN) that are needed for this worked example.</p>
<div class="sourceCode" id="cb602"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb602-1"><a href="WE_5.html#cb602-1" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span>(<span class="sc">!</span><span class="fu">requireNamespace</span>(<span class="st">&quot;EcoGenetics&quot;</span>, <span class="at">quietly =</span> <span class="cn">TRUE</span>)) </span>
<span id="cb602-2"><a href="WE_5.html#cb602-2" aria-hidden="true" tabindex="-1"></a>  remotes<span class="sc">::</span><span class="fu">install_github</span>(<span class="st">&quot;leandroroser/EcoGenetics-devel&quot;</span>)</span>
<span id="cb602-3"><a href="WE_5.html#cb602-3" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span>(<span class="sc">!</span><span class="fu">requireNamespace</span>(<span class="st">&quot;GeNetIt&quot;</span>, <span class="at">quietly =</span> <span class="cn">TRUE</span>)) remotes<span class="sc">::</span><span class="fu">install_github</span>(<span class="st">&quot;jeffreyevans/GeNetIt&quot;</span>)</span>
<span id="cb602-4"><a href="WE_5.html#cb602-4" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span>(<span class="sc">!</span><span class="fu">requireNamespace</span>(<span class="st">&quot;popgraph&quot;</span>, <span class="at">quietly =</span> <span class="cn">TRUE</span>))</span>
<span id="cb602-5"><a href="WE_5.html#cb602-5" aria-hidden="true" tabindex="-1"></a>{</span>
<span id="cb602-6"><a href="WE_5.html#cb602-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">install.packages</span>(<span class="fu">c</span>(<span class="st">&quot;RgoogleMaps&quot;</span>, <span class="st">&quot;geosphere&quot;</span>, <span class="st">&quot;proto&quot;</span>, <span class="st">&quot;sampling&quot;</span>, </span>
<span id="cb602-7"><a href="WE_5.html#cb602-7" aria-hidden="true" tabindex="-1"></a>                      <span class="st">&quot;seqinr&quot;</span>, <span class="st">&quot;spacetime&quot;</span>, <span class="st">&quot;spdep&quot;</span>), <span class="at">dependencies=</span><span class="cn">TRUE</span>)</span>
<span id="cb602-8"><a href="WE_5.html#cb602-8" aria-hidden="true" tabindex="-1"></a>  remotes<span class="sc">::</span><span class="fu">install_github</span>(<span class="st">&quot;dyerlab/popgraph&quot;</span>)</span>
<span id="cb602-9"><a href="WE_5.html#cb602-9" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb602-10"><a href="WE_5.html#cb602-10" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span>(<span class="sc">!</span><span class="fu">requireNamespace</span>(<span class="st">&quot;gstudio&quot;</span>, <span class="at">quietly =</span> <span class="cn">TRUE</span>)) remotes<span class="sc">::</span><span class="fu">install_github</span>(<span class="st">&quot;dyerlab/gstudio&quot;</span>)</span></code></pre></div>
<div class="sourceCode" id="cb603"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb603-1"><a href="WE_5.html#cb603-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(LandGenCourse)</span>
<span id="cb603-2"><a href="WE_5.html#cb603-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(GeNetIt)</span>
<span id="cb603-3"><a href="WE_5.html#cb603-3" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(dplyr)</span>
<span id="cb603-4"><a href="WE_5.html#cb603-4" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(EcoGenetics)</span>
<span id="cb603-5"><a href="WE_5.html#cb603-5" aria-hidden="true" tabindex="-1"></a><span class="co">#require(adegenet)</span></span>
<span id="cb603-6"><a href="WE_5.html#cb603-6" aria-hidden="true" tabindex="-1"></a><span class="co">#require(tibble)</span></span>
<span id="cb603-7"><a href="WE_5.html#cb603-7" aria-hidden="true" tabindex="-1"></a><span class="co">#require(gstudio)</span></span>
<span id="cb603-8"><a href="WE_5.html#cb603-8" aria-hidden="true" tabindex="-1"></a><span class="co">#require(hierfstat)</span></span>
<span id="cb603-9"><a href="WE_5.html#cb603-9" aria-hidden="true" tabindex="-1"></a><span class="co">#require(PopGenReport)</span></span>
<span id="cb603-10"><a href="WE_5.html#cb603-10" aria-hidden="true" tabindex="-1"></a><span class="co">#require(mmod)</span></span>
<span id="cb603-11"><a href="WE_5.html#cb603-11" aria-hidden="true" tabindex="-1"></a><span class="co">#require(spdep)</span></span></code></pre></div>
</div>
</div>
<div id="data-import-and-manipulation" class="section level3 unnumbered hasAnchor">
<h3>2. Data import and manipulation<a href="WE_5.html#data-import-and-manipulation" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div id="a.-create-an-ecogen-object" class="section level4 unnumbered hasAnchor">
<h4>a. Create an ecogen object<a href="WE_5.html#a.-create-an-ecogen-object" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>Import the site data that we used before in Week 2.</p>
<div class="sourceCode" id="cb604"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb604-1"><a href="WE_5.html#cb604-1" aria-hidden="true" tabindex="-1"></a>Frogs.coords <span class="ot">&lt;-</span> <span class="fu">read.csv</span>(<span class="fu">system.file</span>(<span class="st">&quot;extdata&quot;</span>, <span class="st">&quot;ralu_coords_allpops.csv&quot;</span>, </span>
<span id="cb604-2"><a href="WE_5.html#cb604-2" aria-hidden="true" tabindex="-1"></a>                                     <span class="at">package =</span> <span class="st">&quot;LandGenCourse&quot;</span>))</span></code></pre></div>
<p>In Week 3, we calculated population-level genetic data as a table <code>Frogs.diversity</code>. Here we import a table with the same diversity measures for all 29 sites from a system file in <code>LandGenCourse</code>.</p>
<div class="sourceCode" id="cb605"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb605-1"><a href="WE_5.html#cb605-1" aria-hidden="true" tabindex="-1"></a>Frogs.diversity <span class="ot">&lt;-</span> <span class="fu">read.csv</span>(<span class="fu">system.file</span>(<span class="st">&quot;extdata&quot;</span>, <span class="st">&quot;Frogs_diversity_allpops.csv&quot;</span>, </span>
<span id="cb605-2"><a href="WE_5.html#cb605-2" aria-hidden="true" tabindex="-1"></a>                                     <span class="at">package =</span> <span class="st">&quot;LandGenCourse&quot;</span>))</span></code></pre></div>
<p>Import the genetic data for 29 sites:</p>
<div class="sourceCode" id="cb606"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb606-1"><a href="WE_5.html#cb606-1" aria-hidden="true" tabindex="-1"></a>Frogs.loci <span class="ot">&lt;-</span> <span class="fu">read.csv</span>(<span class="fu">system.file</span>(<span class="st">&quot;extdata&quot;</span>, <span class="st">&quot;ralu_loci_allpops.csv&quot;</span>, </span>
<span id="cb606-2"><a href="WE_5.html#cb606-2" aria-hidden="true" tabindex="-1"></a>                                     <span class="at">package =</span> <span class="st">&quot;LandGenCourse&quot;</span>))</span></code></pre></div>
<p>We start building an <code>ecogen</code> object by assigning the genetic data to the <code>G</code> slot, and the structure (i.e., hierarchical sampling information, here site names) to the <code>S</code> slot. For the genetic data, we need to specify the type of data and coding.</p>
<p>Note the tweak using <code>data.frame</code> when specifying <code>data.frame(ralu.loci[,1:2])</code>. This is necessary to import the multiple columns correctly and with their original names.</p>
<div class="sourceCode" id="cb607"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb607-1"><a href="WE_5.html#cb607-1" aria-hidden="true" tabindex="-1"></a>Frogs.ecogen <span class="ot">&lt;-</span> <span class="fu">ecogen</span>(<span class="at">G =</span> Frogs.loci[,<span class="sc">-</span><span class="fu">c</span>(<span class="dv">1</span><span class="sc">:</span><span class="dv">2</span>)], <span class="at">ploidy =</span> <span class="dv">2</span>, </span>
<span id="cb607-2"><a href="WE_5.html#cb607-2" aria-hidden="true" tabindex="-1"></a>                       <span class="at">type =</span> <span class="st">&quot;codominant&quot;</span>, <span class="at">sep =</span> <span class="st">&quot;:&quot;</span>,</span>
<span id="cb607-3"><a href="WE_5.html#cb607-3" aria-hidden="true" tabindex="-1"></a>                       <span class="at">S =</span> <span class="fu">data.frame</span>(Frogs.loci[,<span class="dv">1</span><span class="sc">:</span><span class="dv">2</span>]))</span>
<span id="cb607-4"><a href="WE_5.html#cb607-4" aria-hidden="true" tabindex="-1"></a>Frogs.ecogen</span></code></pre></div>
<pre><code>## 
##                    || ECOGEN CLASS OBJECT ||
## ----------------------------------------------------------------------------
##  Access to slots: &lt;ecoslot.&gt; + &lt;name of the slot&gt; + &lt;(name of the object)&gt; 
##  See: help(&quot;EcoGenetics accessors&quot;)
## ----------------------------------------------------------------------------
## 
##  | slot XY:    | --&gt; 0 x 0          coordinates
##  | slot P:     | --&gt; 0 x 0          phenotypic variables
##  | slot G:     | --&gt; 413 x 8        loci         &gt;&gt; ploidy: 2 || codominant
##  | slot A:     | --&gt; 413 x 43       alleles
##  | slot E:     | --&gt; 0 x 0          environmental variables
##  | slot S:     | --&gt; 413 x 2        structures   &gt;&gt; 2 structures found
##  | slot C:     | --&gt; 0 x 0          variables
##  | slot OUT:   | --&gt; 0              results
## ----------------------------------------------------------------------------</code></pre>
<p>The summary confirms that we now have data in the <code>S</code> and <code>G</code> slots. In addition, EcoGenetics created a table of absolute frequencies (i.e. counts) of alleles for each individual in slot <code>A</code>.</p>
</div>
<div id="b.-aggregate-to-ecopop" class="section level4 unnumbered hasAnchor">
<h4>b. Aggregate to ‘ecopop’<a href="WE_5.html#b.-aggregate-to-ecopop" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>Most of our analysis for this lab will be at the population level. The function <code>ecogen2ecopop</code> aggregates from individual to population-level data.</p>
<div class="sourceCode" id="cb609"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb609-1"><a href="WE_5.html#cb609-1" aria-hidden="true" tabindex="-1"></a>Frogs.ecopop <span class="ot">&lt;-</span> <span class="fu">ecogen2ecopop</span>(Frogs.ecogen, <span class="at">hier =</span> <span class="st">&quot;SiteName&quot;</span>)</span>
<span id="cb609-2"><a href="WE_5.html#cb609-2" aria-hidden="true" tabindex="-1"></a>Frogs.ecopop</span></code></pre></div>
<pre><code>## 
##                    &lt;&lt; ECOPOP CLASS OBJECT &gt;&gt;
## ****************************************************************************
##  Access to slots: &lt;ecoslot.&gt; + &lt;name of the slot&gt; + &lt;(name of the object)&gt; 
##  See: help(&quot;EcoGenetics accessors&quot;)
## ****************************************************************************
## 
##  # slot XY:    # =&gt;  0 x 0          coordinates
##  # slot P:     # =&gt;  0 x 0          phenotypic variables
##  # slot AF:    # =&gt;  30 x 43        loci         &gt;&gt; ploidy: 2 || codominant
##  # slot E:     # =&gt;  0 x 0          environmental variables
##  # slot S:     # =&gt;  30 x 1         population found
##  # slot C:     # =&gt;  0 x 0          variables
## ****************************************************************************</code></pre>
<p>Instead of 413 individuals, we now have data for 30 populations. Slot <code>AF</code> contains the population-level absolute frequencies (counts) of alleles.</p>
</div>
<div id="b.-add-site-level-data" class="section level4 unnumbered hasAnchor">
<h4>b. Add site-level data<a href="WE_5.html#b.-add-site-level-data" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>Before importing the genetic diversity, spatial coordinates and site data into the <code>ecopop</code> object, we need to match the rows and extract the data for the sampled populations.</p>
<p>Unfortunately, the datasets use different versions of the site name, both were include in the file <code>ralu_loci_allpops.csv</code> and thus in the <code>S</code> slot of <code>Frogs.ecogen</code>.</p>
<ul>
<li>Here, we use <code>group_by</code> and <code>summarize</code> to get the unique set of site names in both versions. We save it as a data frame Subset.</li>
<li>Then we join the genetic diversity data. Because the two data frames share a common variable, we dont’ need to specify the argument <code>by</code>. R will confirm what ID variable it used to join.</li>
</ul>
<div class="sourceCode" id="cb611"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb611-1"><a href="WE_5.html#cb611-1" aria-hidden="true" tabindex="-1"></a>Subset <span class="ot">&lt;-</span> <span class="fu">ecoslot.S</span>(Frogs.ecogen) <span class="sc">%&gt;%</span> <span class="fu">group_by</span>(SiteName, Pop) <span class="sc">%&gt;%</span> <span class="fu">summarize</span>()</span></code></pre></div>
<pre><code>## `summarise()` has grouped output by &#39;SiteName&#39;. You can override using the
## `.groups` argument.</code></pre>
<div class="sourceCode" id="cb613"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb613-1"><a href="WE_5.html#cb613-1" aria-hidden="true" tabindex="-1"></a>Subset <span class="ot">&lt;-</span> <span class="fu">left_join</span>(Subset, Frogs.diversity)</span></code></pre></div>
<pre><code>## Joining, by = &quot;Pop&quot;</code></pre>
<p>R tells us that it used the shared ID variable <code>Pop</code> to join the data.</p>
<p>Now we can join the site data. Note: with <code>as.data.frame</code>, we combine the <code>@data</code> and <code>@coords</code> slots of the SpatialPointsDataFrame <code>ralu.site</code> to a single data frame.</p>
<div class="sourceCode" id="cb615"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb615-1"><a href="WE_5.html#cb615-1" aria-hidden="true" tabindex="-1"></a>Subset <span class="ot">&lt;-</span> <span class="fu">left_join</span>(Subset, <span class="fu">as.data.frame</span>(Frogs.coords))</span></code></pre></div>
<pre><code>## Joining, by = &quot;SiteName&quot;</code></pre>
<p>This time, R used the shared ID variable <code>SiteName</code> to join the data.</p>
<p>Now we have all site-level data in the data frame <code>Subset</code> that has the same row names as <code>Frogs.ecopop</code>. We check the names of the variables to decide which ones to put where. Then we assign them to their respective slots (<code>@C</code> for the custom data, here genetic diversity, <code>@E</code> for the environmental data, and <code>@XY</code> for the coordinates).</p>
<p>The argument pop specifies the matching ID variable in the <code>@S</code> slot. When we aggregated from <code>Frogs.ecogen</code> to <code>Frogs.ecopop</code>, unfortunately EcoGenetics renamed the variable <code>SiteName</code> to <code>pop</code>. The argument <code>pop_levels</code> identifies the corresponding ID variable in <code>Subset</code>.</p>
<div class="sourceCode" id="cb617"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb617-1"><a href="WE_5.html#cb617-1" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(Subset)</span></code></pre></div>
<pre><code>##  [1] &quot;SiteName&quot;  &quot;Pop&quot;       &quot;n&quot;         &quot;Hobs&quot;      &quot;Hexp&quot;      &quot;Ar&quot;       
##  [7] &quot;UTM83_E&quot;   &quot;UTM83_N&quot;   &quot;Longitude&quot; &quot;Latitude&quot;  &quot;Basin&quot;</code></pre>
<div class="sourceCode" id="cb619"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb619-1"><a href="WE_5.html#cb619-1" aria-hidden="true" tabindex="-1"></a>Frogs.ecopop <span class="ot">&lt;-</span> EcoGenetics<span class="sc">::</span><span class="fu">eco.fill_ecogen_with_df</span>(Frogs.ecopop,</span>
<span id="cb619-2"><a href="WE_5.html#cb619-2" aria-hidden="true" tabindex="-1"></a>             <span class="at">pop =</span><span class="st">&quot;pop&quot;</span>, <span class="at">pop_levels =</span> Subset<span class="sc">$</span>SiteName, </span>
<span id="cb619-3"><a href="WE_5.html#cb619-3" aria-hidden="true" tabindex="-1"></a>             <span class="at">C =</span> Subset[,<span class="dv">3</span><span class="sc">:</span><span class="dv">6</span>],</span>
<span id="cb619-4"><a href="WE_5.html#cb619-4" aria-hidden="true" tabindex="-1"></a>             <span class="at">XY =</span> Subset[,<span class="dv">7</span><span class="sc">:</span><span class="dv">10</span>])</span>
<span id="cb619-5"><a href="WE_5.html#cb619-5" aria-hidden="true" tabindex="-1"></a>Frogs.ecopop</span></code></pre></div>
<pre><code>## 
##                    &lt;&lt; ECOPOP CLASS OBJECT &gt;&gt;
## ****************************************************************************
##  Access to slots: &lt;ecoslot.&gt; + &lt;name of the slot&gt; + &lt;(name of the object)&gt; 
##  See: help(&quot;EcoGenetics accessors&quot;)
## ****************************************************************************
## 
##  # slot XY:    # =&gt;  30 x 4         coordinates
##  # slot P:     # =&gt;  0 x 0          phenotypic variables
##  # slot AF:    # =&gt;  30 x 43        loci         &gt;&gt; ploidy: 2 || codominant
##  # slot E:     # =&gt;  0 x 0          environmental variables
##  # slot S:     # =&gt;  30 x 1         population found
##  # slot C:     # =&gt;  30 x 4         variables
## ****************************************************************************</code></pre>
</div>
<div id="c.-export-to-adegenet-and-gstudio" class="section level4 unnumbered hasAnchor">
<h4>c. Export to ‘adegenet’ and ‘gstudio’<a href="WE_5.html#c.-export-to-adegenet-and-gstudio" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>‘EcoGenetics’ provides convenient functions for converting genetic data to and from other packages.</p>
<p>Import into ‘genind’ object (package ‘adegenet’): there is a dedicated function, but we need to separately declare the variable that represents the populations and write it into the <code>@pop</code> slot of the ‘genind’ object.</p>
<div class="sourceCode" id="cb621"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb621-1"><a href="WE_5.html#cb621-1" aria-hidden="true" tabindex="-1"></a>Frogs.genind <span class="ot">&lt;-</span> EcoGenetics<span class="sc">::</span><span class="fu">ecogen2genind</span>(Frogs.ecogen)</span></code></pre></div>
<pre><code>## Loading required package: adegenet</code></pre>
<pre><code>## Loading required package: ade4</code></pre>
<pre><code>## 
##    /// adegenet 2.1.5 is loaded ////////////
## 
##    &gt; overview: &#39;?adegenet&#39;
##    &gt; tutorials/doc/questions: &#39;adegenetWeb()&#39; 
##    &gt; bug reports/feature requests: adegenetIssues()</code></pre>
<div class="sourceCode" id="cb625"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb625-1"><a href="WE_5.html#cb625-1" aria-hidden="true" tabindex="-1"></a>Frogs.genind<span class="sc">@</span>pop <span class="ot">&lt;-</span> <span class="fu">ecoslot.S</span>(Frogs.ecogen)<span class="sc">$</span>SiteName</span></code></pre></div>
<p>For calculating population-level genetic distances, we aggregate the individual-level data to a <code>genpop</code> object with population-level allele frequencies.</p>
<div class="sourceCode" id="cb626"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb626-1"><a href="WE_5.html#cb626-1" aria-hidden="true" tabindex="-1"></a>Frogs.genpop <span class="ot">&lt;-</span> adegenet<span class="sc">::</span><span class="fu">genind2genpop</span>(Frogs.genind)</span></code></pre></div>
<pre><code>## 
##  Converting data from a genind to a genpop object... 
## 
## ...done.</code></pre>
<div class="sourceCode" id="cb628"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb628-1"><a href="WE_5.html#cb628-1" aria-hidden="true" tabindex="-1"></a>Frogs.genpop</span></code></pre></div>
<pre><code>## /// GENPOP OBJECT /////////
## 
##  // 30 populations; 8 loci; 43 alleles; size: 18.2 Kb
## 
##  // Basic content
##    @tab:  30 x 43 matrix of allele counts
##    @loc.n.all: number of alleles per locus (range: 3-9)
##    @loc.fac: locus factor for the 43 columns of @tab
##    @all.names: list of allele names for each locus
##    @ploidy: ploidy of each individual  (range: 2-2)
##    @type:  codom
##    @call: adegenet::genind2genpop(x = Frogs.genind)
## 
##  // Optional content
##    - empty -</code></pre>
<p>Note: Alternatively, we could directly import the <code>ecopop</code> object into a <code>genpop</code> object (<code>adegenet</code>) with <code>EcoGenetics::ecopop2genpop(Frogs.ecopop)</code>. However, this can create warnings later on when calculating genetic distances.</p>
<p>The object ‘Frogs.genpop’ has 30 rows, each representing a population.</p>
<p>We will also use some functions from the package <code>gstudio</code>, hence we import the individual-level genetic data into <code>gstudio</code>. This should be easy with the function <code>EcoGenetics::ecogen2gstudio</code>. However, there seems to be an issue. The following chunk contains code adapted from within the <code>ecogen2gstudio</code> function, tweaked to work for our data. (You can view the original function with: <code>findMethods(ecogen2gstudio)</code>.)</p>
<div class="sourceCode" id="cb630"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb630-1"><a href="WE_5.html#cb630-1" aria-hidden="true" tabindex="-1"></a><span class="co">#Frogs.gstudio &lt;- EcoGenetics::ecogen2gstudio(Frogs.ecogen, type=&quot;codominant&quot;)</span></span>
<span id="cb630-2"><a href="WE_5.html#cb630-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb630-3"><a href="WE_5.html#cb630-3" aria-hidden="true" tabindex="-1"></a>  dat <span class="ot">&lt;-</span> <span class="fu">eco.convert</span>(Frogs.ecogen<span class="sc">@</span>G, <span class="st">&quot;matrix&quot;</span>, <span class="at">sep.in =</span> <span class="st">&quot;:&quot;</span>, <span class="at">sep.out =</span> <span class="st">&quot;:&quot;</span>)</span>
<span id="cb630-4"><a href="WE_5.html#cb630-4" aria-hidden="true" tabindex="-1"></a>  dat <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(dat, <span class="at">stringsAsFactors =</span> <span class="cn">FALSE</span>)</span>
<span id="cb630-5"><a href="WE_5.html#cb630-5" aria-hidden="true" tabindex="-1"></a>  <span class="cf">for</span> (i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">ncol</span>(dat)) {</span>
<span id="cb630-6"><a href="WE_5.html#cb630-6" aria-hidden="true" tabindex="-1"></a>     <span class="fu">class</span>(dat[, i]) <span class="ot">&lt;-</span> <span class="st">&quot;locus&quot;</span></span>
<span id="cb630-7"><a href="WE_5.html#cb630-7" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb630-8"><a href="WE_5.html#cb630-8" aria-hidden="true" tabindex="-1"></a>  dat[<span class="fu">is.na</span>(dat)] <span class="ot">&lt;-</span> gstudio<span class="sc">::</span><span class="fu">locus</span>(<span class="cn">NA</span>)</span>
<span id="cb630-9"><a href="WE_5.html#cb630-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">colnames</span>(dat) <span class="ot">&lt;-</span> <span class="fu">colnames</span>(Frogs.ecogen<span class="sc">@</span>G)</span>
<span id="cb630-10"><a href="WE_5.html#cb630-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb630-11"><a href="WE_5.html#cb630-11" aria-hidden="true" tabindex="-1"></a>Frogs.gstudio <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="fu">ecoslot.S</span>(Frogs.ecogen), dat)</span>
<span id="cb630-12"><a href="WE_5.html#cb630-12" aria-hidden="true" tabindex="-1"></a>tibble<span class="sc">::</span><span class="fu">as_tibble</span>(Frogs.gstudio)        </span></code></pre></div>
<pre><code>## # A tibble: 413 × 10
##    SiteName     Pop      A       B       C       D       E     F     G     H    
##    &lt;fct&gt;        &lt;fct&gt;    &lt;locus&gt; &lt;locus&gt; &lt;locus&gt; &lt;locus&gt; &lt;loc&gt; &lt;loc&gt; &lt;loc&gt; &lt;loc&gt;
##  1 AirplaneLake Airplane 1:1     1:1     1:1     1:1     1:2   1:1   1:1   4:5  
##  2 AirplaneLake Airplane 2:2     1:1             1:1     1:1         2:2        
##  3 AirplaneLake Airplane 1:1     1:1     1:1     1:1     3:3   1:1   1:1   3:3  
##  4 AirplaneLake Airplane 1:1     1:1             2:2     1:2                    
##  5 AirplaneLake Airplane 1:2     1:3     1:1     1:1     1:2   1:1   1:2   4:5  
##  6 AirplaneLake Airplane 1:2     1:1     1:1     1:3     1:1   1:1   1:2   4:5  
##  7 AirplaneLake Airplane 2:2     1:3     1:1     1:1     3:3   1:1   1:1   2:3  
##  8 AirplaneLake Airplane 2:2     1:3     1:1     1:1     3:3   1:1   1:1   2:3  
##  9 AirplaneLake Airplane 1:3     1:1     1:1     1:1     1:7   1:1   1:1   3:5  
## 10 AirplaneLake Airplane 2:2     1:3     1:1     1:1     3:7   1:1   1:1   3:3  
## # … with 403 more rows</code></pre>
</div>
</div>
<div id="calculate-genetic-distances" class="section level3 unnumbered hasAnchor">
<h3>3. Calculate genetic distances<a href="WE_5.html#calculate-genetic-distances" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Here, we’ll calculate a number of different measures of genetic distance, using functions from several packages. Adding the package name to each distance matrix name helps keeping track of methods used. Normally you would not calculate all of them for your own data, and some are redundant, as we will see.</p>
<p>Note: Some functions provide an option <code>linearized = TRUE</code> to linearize distances <code>d</code> by calculating <code>d/(1-d)</code>. This should result in more linear relationships when plotted or correlated against geographic distance.Here we don’t linearize, we can do so later manually.</p>
<div id="a.-genetic-distances-calculated-from-genind-object" class="section level4 unnumbered hasAnchor">
<h4>a. Genetic distances calculated from genind object<a href="WE_5.html#a.-genetic-distances-calculated-from-genind-object" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>Pairwise Fst with package <code>hierfstat</code>:</p>
<div class="sourceCode" id="cb632"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb632-1"><a href="WE_5.html#cb632-1" aria-hidden="true" tabindex="-1"></a>GD.pop.PairwiseFst.hierfstat <span class="ot">&lt;-</span> <span class="fu">as.dist</span>(hierfstat<span class="sc">::</span><span class="fu">pairwise.neifst</span>(hierfstat<span class="sc">::</span><span class="fu">genind2hierfstat</span>(Frogs.genind)))</span></code></pre></div>
<p>Proportion of shared alleles with package ‘PopGenReport’:</p>
<div class="sourceCode" id="cb633"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb633-1"><a href="WE_5.html#cb633-1" aria-hidden="true" tabindex="-1"></a>GD.pop.propShared <span class="ot">&lt;-</span> PopGenReport<span class="sc">::</span><span class="fu">pairwise.propShared</span>(Frogs.genind)</span></code></pre></div>
<pre><code>## Registered S3 method overwritten by &#39;pegas&#39;:
##   method      from
##   print.amova ade4</code></pre>
<pre><code>## Registered S3 method overwritten by &#39;GGally&#39;:
##   method from   
##   +.gg   ggplot2</code></pre>
<pre><code>## Registered S3 methods overwritten by &#39;genetics&#39;:
##   method      from   
##   print.locus gstudio
##   [.haplotype pegas</code></pre>
<p>Several distance matrices with package ‘adegenet’:</p>
<div class="sourceCode" id="cb637"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb637-1"><a href="WE_5.html#cb637-1" aria-hidden="true" tabindex="-1"></a>GD.pop.Nei <span class="ot">&lt;-</span> adegenet<span class="sc">::</span><span class="fu">dist.genpop</span>(Frogs.genpop, <span class="at">method=</span><span class="dv">1</span>)   </span>
<span id="cb637-2"><a href="WE_5.html#cb637-2" aria-hidden="true" tabindex="-1"></a>GD.pop.Edwards <span class="ot">&lt;-</span> adegenet<span class="sc">::</span><span class="fu">dist.genpop</span>(Frogs.genpop, <span class="at">method=</span><span class="dv">2</span>)</span>
<span id="cb637-3"><a href="WE_5.html#cb637-3" aria-hidden="true" tabindex="-1"></a>GD.pop.Reynolds <span class="ot">&lt;-</span> adegenet<span class="sc">::</span><span class="fu">dist.genpop</span>(Frogs.genpop, <span class="at">method=</span><span class="dv">3</span>)  </span>
<span id="cb637-4"><a href="WE_5.html#cb637-4" aria-hidden="true" tabindex="-1"></a>GD.pop.Rogers <span class="ot">&lt;-</span> adegenet<span class="sc">::</span><span class="fu">dist.genpop</span>(Frogs.genpop, <span class="at">method=</span><span class="dv">4</span>)  </span>
<span id="cb637-5"><a href="WE_5.html#cb637-5" aria-hidden="true" tabindex="-1"></a>GD.pop.Provesti <span class="ot">&lt;-</span> adegenet<span class="sc">::</span><span class="fu">dist.genpop</span>(Frogs.genpop, <span class="at">method=</span><span class="dv">5</span>)</span></code></pre></div>
<p>Additional distance matrices with package ‘mmod’:</p>
<div class="sourceCode" id="cb638"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb638-1"><a href="WE_5.html#cb638-1" aria-hidden="true" tabindex="-1"></a>GD.pop.Joost <span class="ot">&lt;-</span> mmod<span class="sc">::</span><span class="fu">pairwise_D</span>(Frogs.genind, <span class="at">linearized =</span> <span class="cn">FALSE</span>)</span>
<span id="cb638-2"><a href="WE_5.html#cb638-2" aria-hidden="true" tabindex="-1"></a>GD.pop.Hedrick <span class="ot">&lt;-</span> mmod<span class="sc">::</span><span class="fu">pairwise_Gst_Hedrick</span>(Frogs.genind, <span class="at">linearized =</span> <span class="cn">FALSE</span>)</span>
<span id="cb638-3"><a href="WE_5.html#cb638-3" aria-hidden="true" tabindex="-1"></a>GD.pop.NeiGst <span class="ot">&lt;-</span> mmod<span class="sc">::</span><span class="fu">pairwise_Gst_Nei</span>(Frogs.genind, <span class="at">linearized =</span> <span class="cn">FALSE</span>)</span></code></pre></div>
</div>
<div id="b.-more-distance-matrices-with-gstudio" class="section level4 unnumbered hasAnchor">
<h4>b. More distance matrices with ‘gstudio’<a href="WE_5.html#b.-more-distance-matrices-with-gstudio" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<div class="sourceCode" id="cb639"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb639-1"><a href="WE_5.html#cb639-1" aria-hidden="true" tabindex="-1"></a>GD.pop.Euclidean.gstudio <span class="ot">&lt;-</span>gstudio<span class="sc">::</span><span class="fu">genetic_distance</span>(Frogs.gstudio, <span class="at">mode =</span> <span class="st">&quot;Euclidean&quot;</span>, <span class="at">stratum=</span><span class="st">&quot;SiteName&quot;</span>)</span></code></pre></div>
<pre><code>## Multilous estimates of Euclidean distance are assumed to be additive.</code></pre>
<div class="sourceCode" id="cb641"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb641-1"><a href="WE_5.html#cb641-1" aria-hidden="true" tabindex="-1"></a>GD.pop.cGD.gstudio <span class="ot">&lt;-</span>gstudio<span class="sc">::</span><span class="fu">genetic_distance</span>(Frogs.gstudio, <span class="at">mode =</span> <span class="st">&quot;cGD&quot;</span>, <span class="at">stratum=</span><span class="st">&quot;SiteName&quot;</span>)</span>
<span id="cb641-2"><a href="WE_5.html#cb641-2" aria-hidden="true" tabindex="-1"></a>GD.pop.Nei.gstudio <span class="ot">&lt;-</span>gstudio<span class="sc">::</span><span class="fu">genetic_distance</span>(Frogs.gstudio, <span class="at">mode =</span> <span class="st">&quot;Nei&quot;</span>, <span class="at">stratum=</span><span class="st">&quot;SiteName&quot;</span>)</span>
<span id="cb641-3"><a href="WE_5.html#cb641-3" aria-hidden="true" tabindex="-1"></a>GD.pop.Dps.gstudio <span class="ot">&lt;-</span>gstudio<span class="sc">::</span><span class="fu">genetic_distance</span>(Frogs.gstudio, <span class="at">mode =</span> <span class="st">&quot;Dps&quot;</span>, <span class="at">stratum=</span><span class="st">&quot;SiteName&quot;</span>)</span></code></pre></div>
<pre><code>## Bray distance will be assumed to be entirely additive across loci.</code></pre>
<div class="sourceCode" id="cb643"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb643-1"><a href="WE_5.html#cb643-1" aria-hidden="true" tabindex="-1"></a>GD.pop.Jaccard.gstudio <span class="ot">&lt;-</span>gstudio<span class="sc">::</span><span class="fu">genetic_distance</span>(Frogs.gstudio, <span class="at">mode =</span> <span class="st">&quot;Jaccard&quot;</span>, <span class="at">stratum=</span><span class="st">&quot;SiteName&quot;</span>)</span></code></pre></div>
<pre><code>## Jaccard distance will be assumed to be entirely additive across loci.</code></pre>
</div>
<div id="c.-assemble-distance-matrices" class="section level4 unnumbered hasAnchor">
<h4>c. Assemble distance matrices<a href="WE_5.html#c.-assemble-distance-matrices" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>We’ll store the population-level genetic distances in a list ‘GD.pop.’</p>
<p>Note: a few measures return similarities (scaled between 0 and 1) instead of distances. For instance, ‘proporition of shared alleles’ is 1 if the alleles are identical, and zero of no alleles are shared. We convert these values to distances by subtracting them from 1.</p>
<div class="sourceCode" id="cb645"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb645-1"><a href="WE_5.html#cb645-1" aria-hidden="true" tabindex="-1"></a>GD.pop <span class="ot">&lt;-</span> <span class="fu">list</span>(<span class="at">pairwiseFst.hierfstat =</span> GD.pop.PairwiseFst.hierfstat,</span>
<span id="cb645-2"><a href="WE_5.html#cb645-2" aria-hidden="true" tabindex="-1"></a>               <span class="at">propShared.PopGenReport =</span> <span class="dv">1</span> <span class="sc">-</span> GD.pop.propShared,</span>
<span id="cb645-3"><a href="WE_5.html#cb645-3" aria-hidden="true" tabindex="-1"></a>               <span class="at">Nei.adegenet =</span> GD.pop.Nei,</span>
<span id="cb645-4"><a href="WE_5.html#cb645-4" aria-hidden="true" tabindex="-1"></a>               <span class="at">Edwards.adegenet =</span> GD.pop.Edwards,</span>
<span id="cb645-5"><a href="WE_5.html#cb645-5" aria-hidden="true" tabindex="-1"></a>               <span class="at">Reynolds.adegenet =</span> GD.pop.Reynolds,</span>
<span id="cb645-6"><a href="WE_5.html#cb645-6" aria-hidden="true" tabindex="-1"></a>               <span class="at">Rogers.adegenet =</span> GD.pop.Rogers,</span>
<span id="cb645-7"><a href="WE_5.html#cb645-7" aria-hidden="true" tabindex="-1"></a>               <span class="at">Provesti.adegenet =</span> GD.pop.Provesti,</span>
<span id="cb645-8"><a href="WE_5.html#cb645-8" aria-hidden="true" tabindex="-1"></a>               <span class="at">Joost.mmod =</span> GD.pop.Joost,</span>
<span id="cb645-9"><a href="WE_5.html#cb645-9" aria-hidden="true" tabindex="-1"></a>               <span class="at">Hedrick.mmod =</span> GD.pop.Hedrick,</span>
<span id="cb645-10"><a href="WE_5.html#cb645-10" aria-hidden="true" tabindex="-1"></a>               <span class="at">Nei.mmod =</span> GD.pop.NeiGst,</span>
<span id="cb645-11"><a href="WE_5.html#cb645-11" aria-hidden="true" tabindex="-1"></a>               <span class="at">Euclidean.gstudio =</span> <span class="fu">as.dist</span>(GD.pop.Euclidean.gstudio),</span>
<span id="cb645-12"><a href="WE_5.html#cb645-12" aria-hidden="true" tabindex="-1"></a>               <span class="at">cGD.gstudio =</span> <span class="fu">as.dist</span>(GD.pop.cGD.gstudio),</span>
<span id="cb645-13"><a href="WE_5.html#cb645-13" aria-hidden="true" tabindex="-1"></a>               <span class="at">Nei.gstudio =</span> <span class="fu">as.dist</span>(GD.pop.Nei.gstudio),</span>
<span id="cb645-14"><a href="WE_5.html#cb645-14" aria-hidden="true" tabindex="-1"></a>               <span class="at">Dps.gstudio =</span> <span class="fu">as.dist</span>(<span class="dv">1</span> <span class="sc">-</span> GD.pop.Dps.gstudio),</span>
<span id="cb645-15"><a href="WE_5.html#cb645-15" aria-hidden="true" tabindex="-1"></a>               <span class="at">Jaccard.gstudio =</span> <span class="fu">as.dist</span>(<span class="dv">1</span> <span class="sc">-</span> GD.pop.Jaccard.gstudio))</span>
<span id="cb645-16"><a href="WE_5.html#cb645-16" aria-hidden="true" tabindex="-1"></a><span class="fu">round</span>(<span class="fu">cor</span>(<span class="fu">sapply</span>(GD.pop, <span class="cf">function</span>(ls) <span class="fu">as.vector</span>(ls))),<span class="dv">2</span>)[,<span class="dv">1</span><span class="sc">:</span><span class="dv">2</span>]</span></code></pre></div>
<pre><code>##                         pairwiseFst.hierfstat propShared.PopGenReport
## pairwiseFst.hierfstat                    1.00                    0.79
## propShared.PopGenReport                  0.79                    1.00
## Nei.adegenet                             0.85                    0.96
## Edwards.adegenet                         0.83                    0.93
## Reynolds.adegenet                        0.97                    0.82
## Rogers.adegenet                          0.83                    0.99
## Provesti.adegenet                        0.79                    1.00
## Joost.mmod                               0.90                    0.95
## Hedrick.mmod                             0.98                    0.88
## Nei.mmod                                 1.00                    0.77
## Euclidean.gstudio                        0.93                    0.95
## cGD.gstudio                              0.50                    0.62
## Nei.gstudio                              0.85                    0.96
## Dps.gstudio                              0.71                    0.90
## Jaccard.gstudio                          0.71                    0.90</code></pre>
<p>Consider the correlations printed above (only the first two columns of the correlation matrix are shown).</p>
<ul>
<li>Correlations are high in general, except for conditional genetic distance (cGD) (see Week 13).</li>
<li>There are some duplicate measures (with correlation = 1).</li>
</ul>
<p>Note: the following functions calculate distance matrices at the individual level:</p>
<ul>
<li>PopGenReport::gd.smouse()</li>
<li>adegenet::propShared()</li>
<li>gstudio::genetic_distance(mode = “AMOVA”)</li>
</ul>
</div>
<div id="d.-export-genetic-distance-matrices" class="section level4 unnumbered hasAnchor">
<h4>d. Export genetic distance matrices<a href="WE_5.html#d.-export-genetic-distance-matrices" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>Optional: You can use <code>save</code> to save an R object to your file system, and <code>load</code> to read it in again. Note: the default setting is that <code>save</code> will overwrite existing files with the same name.</p>
<p>The code is commented out with <code>#</code>. To run it, remove the <code>#</code>. The first part creates a folder <code>output</code> in your project folder if it does not yet exist. The function <code>save</code> writes the list <code>GD.pop</code> into a file “GD.pop.RData,” and the function <code>load</code> imports it again.</p>
<div class="sourceCode" id="cb647"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb647-1"><a href="WE_5.html#cb647-1" aria-hidden="true" tabindex="-1"></a><span class="co">#require(here)</span></span>
<span id="cb647-2"><a href="WE_5.html#cb647-2" aria-hidden="true" tabindex="-1"></a><span class="co">#if(!dir.exists(paste0(here::here(),&quot;/output&quot;))) dir.create(paste0(here::here(),&quot;/output&quot;))</span></span>
<span id="cb647-3"><a href="WE_5.html#cb647-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb647-4"><a href="WE_5.html#cb647-4" aria-hidden="true" tabindex="-1"></a><span class="co">#save(GD.pop, file = paste0(here::here(),&quot;/output/GD.pop.RData&quot;))</span></span>
<span id="cb647-5"><a href="WE_5.html#cb647-5" aria-hidden="true" tabindex="-1"></a><span class="co">#load(paste0(here::here(),&quot;/output/GD.pop.RData&quot;))</span></span></code></pre></div>
</div>
</div>
<div id="perform-a-mantel-test-to-test-for-ibd" class="section level3 unnumbered hasAnchor">
<h3>4. Perform a Mantel test to test for IBD<a href="WE_5.html#perform-a-mantel-test-to-test-for-ibd" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>First, we calculate geographic (Euclidean) distances <code>Dgeo</code> with the <code>dist</code> function, using the spatial coordinates that we imported from ralu.site. These are UTM coordinates and thus metric.</p>
<div class="sourceCode" id="cb648"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb648-1"><a href="WE_5.html#cb648-1" aria-hidden="true" tabindex="-1"></a>Dgeo <span class="ot">&lt;-</span> <span class="fu">as.vector</span>(<span class="fu">dist</span>(<span class="fu">ecoslot.XY</span>(Frogs.ecopop)[,<span class="dv">1</span><span class="sc">:</span><span class="dv">2</span>]))</span></code></pre></div>
<div id="a.-visually-check-linearity" class="section level4 unnumbered hasAnchor">
<h4>a. Visually check linearity<a href="WE_5.html#a.-visually-check-linearity" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>Before we quantify the linear relationship between genetic and geographic distances, let’s check visually whether the relationship is indeed linear. To start, we will define genetic distance <code>Dgen</code> based on proportion of shared alleles.</p>
<div class="sourceCode" id="cb649"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb649-1"><a href="WE_5.html#cb649-1" aria-hidden="true" tabindex="-1"></a><span class="fu">par</span>(<span class="at">mar=</span><span class="fu">c</span>(<span class="dv">4</span>,<span class="dv">4</span>,<span class="dv">0</span>,<span class="dv">0</span>))</span>
<span id="cb649-2"><a href="WE_5.html#cb649-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb649-3"><a href="WE_5.html#cb649-3" aria-hidden="true" tabindex="-1"></a>Dgen <span class="ot">&lt;-</span> <span class="fu">as.vector</span>(GD.pop<span class="sc">$</span>propShared.PopGenReport)</span>
<span id="cb649-4"><a href="WE_5.html#cb649-4" aria-hidden="true" tabindex="-1"></a>dens <span class="ot">&lt;-</span> MASS<span class="sc">::</span><span class="fu">kde2d</span>(Dgeo, Dgen, <span class="at">n=</span><span class="dv">300</span>)</span>
<span id="cb649-5"><a href="WE_5.html#cb649-5" aria-hidden="true" tabindex="-1"></a>myPal <span class="ot">&lt;-</span> <span class="fu">colorRampPalette</span>(<span class="fu">c</span>(<span class="st">&quot;white&quot;</span>,<span class="st">&quot;blue&quot;</span>,<span class="st">&quot;gold&quot;</span>,<span class="st">&quot;orange&quot;</span>,<span class="st">&quot;red&quot;</span>))</span>
<span id="cb649-6"><a href="WE_5.html#cb649-6" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(Dgeo, Dgen, <span class="at">pch=</span><span class="dv">20</span>, <span class="at">cex=</span><span class="fl">0.5</span>,  </span>
<span id="cb649-7"><a href="WE_5.html#cb649-7" aria-hidden="true" tabindex="-1"></a>    <span class="at">xlab=</span><span class="st">&quot;Geographic Distance&quot;</span>, <span class="at">ylab=</span><span class="st">&quot;Genetic Distance&quot;</span>)</span>
<span id="cb649-8"><a href="WE_5.html#cb649-8" aria-hidden="true" tabindex="-1"></a><span class="fu">image</span>(dens, <span class="at">col=</span><span class="fu">transp</span>(<span class="fu">myPal</span>(<span class="dv">300</span>), <span class="fl">0.7</span>), <span class="at">add=</span><span class="cn">TRUE</span>)</span>
<span id="cb649-9"><a href="WE_5.html#cb649-9" aria-hidden="true" tabindex="-1"></a><span class="fu">abline</span>(<span class="fu">lm</span>(Dgen <span class="sc">~</span> Dgeo))</span>
<span id="cb649-10"><a href="WE_5.html#cb649-10" aria-hidden="true" tabindex="-1"></a><span class="fu">lines</span>(<span class="fu">loess.smooth</span>(Dgeo, Dgen), <span class="at">col=</span><span class="st">&quot;red&quot;</span>)</span></code></pre></div>
<p><img src="05-Week05_files/figure-html/unnamed-chunk-26-1.png" width="672" /></p>
<p>There is clearly an increase of genetic distance with geographic distance. However, the red line, which is a smoothed local mean, indicates that the relationship is not linear.</p>
<p>Let’s take the natural logarithm of geographic distance:</p>
<div class="sourceCode" id="cb650"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb650-1"><a href="WE_5.html#cb650-1" aria-hidden="true" tabindex="-1"></a><span class="fu">par</span>(<span class="at">mar=</span><span class="fu">c</span>(<span class="dv">4</span>,<span class="dv">4</span>,<span class="dv">0</span>,<span class="dv">0</span>))</span>
<span id="cb650-2"><a href="WE_5.html#cb650-2" aria-hidden="true" tabindex="-1"></a>dens <span class="ot">&lt;-</span> MASS<span class="sc">::</span><span class="fu">kde2d</span>(<span class="fu">log</span>(Dgeo), Dgen, <span class="at">n=</span><span class="dv">300</span>)</span>
<span id="cb650-3"><a href="WE_5.html#cb650-3" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(<span class="fu">log</span>(Dgeo), Dgen, <span class="at">pch=</span><span class="dv">20</span>, <span class="at">cex=</span><span class="fl">0.5</span>,  </span>
<span id="cb650-4"><a href="WE_5.html#cb650-4" aria-hidden="true" tabindex="-1"></a>    <span class="at">xlab=</span><span class="st">&quot;Geographic Distance&quot;</span>, <span class="at">ylab=</span><span class="st">&quot;Genetic Distance&quot;</span>)</span>
<span id="cb650-5"><a href="WE_5.html#cb650-5" aria-hidden="true" tabindex="-1"></a><span class="fu">image</span>(dens, <span class="at">col=</span><span class="fu">transp</span>(<span class="fu">myPal</span>(<span class="dv">300</span>), <span class="fl">0.7</span>), <span class="at">add=</span><span class="cn">TRUE</span>)</span>
<span id="cb650-6"><a href="WE_5.html#cb650-6" aria-hidden="true" tabindex="-1"></a><span class="fu">abline</span>(<span class="fu">lm</span>(Dgen <span class="sc">~</span> <span class="fu">log</span>(Dgeo)))</span>
<span id="cb650-7"><a href="WE_5.html#cb650-7" aria-hidden="true" tabindex="-1"></a><span class="fu">lines</span>(<span class="fu">loess.smooth</span>(<span class="fu">log</span>(Dgeo), Dgen), <span class="at">col=</span><span class="st">&quot;red&quot;</span>)</span></code></pre></div>
<p><img src="05-Week05_files/figure-html/unnamed-chunk-27-1.png" width="672" /></p>
<p><strong>Questions:</strong></p>
<ul>
<li>Is the relationship between Dgen and Dgeo more linear than before the transformation?</li>
<li>What happened to the spread of points along the x axis (geographic distance)?</li>
<li>What do the units of the x-axis represent, after the transformation?</li>
</ul>
</div>
<div id="b.-perform-mantel-test" class="section level4 unnumbered hasAnchor">
<h4>b. Perform Mantel test<a href="WE_5.html#b.-perform-mantel-test" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>Next, we perform a Mantel test with the function <code>mantel</code> from the <code>vegan</code> package.</p>
<p>We define Dgen and Dgeo anew, as we need them in ‘dist’ format this time, not as vectors.</p>
<div class="sourceCode" id="cb651"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb651-1"><a href="WE_5.html#cb651-1" aria-hidden="true" tabindex="-1"></a>Dgen <span class="ot">&lt;-</span> GD.pop<span class="sc">$</span>propShared.PopGenReport</span>
<span id="cb651-2"><a href="WE_5.html#cb651-2" aria-hidden="true" tabindex="-1"></a>Dgeo <span class="ot">&lt;-</span> <span class="fu">dist</span>(<span class="fu">ecoslot.XY</span>(Frogs.ecopop)[,<span class="dv">1</span><span class="sc">:</span><span class="dv">2</span>])</span>
<span id="cb651-3"><a href="WE_5.html#cb651-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb651-4"><a href="WE_5.html#cb651-4" aria-hidden="true" tabindex="-1"></a>IBD <span class="ot">&lt;-</span> vegan<span class="sc">::</span><span class="fu">mantel</span>(Dgen,Dgeo, <span class="at">method=</span><span class="st">&quot;pearson&quot;</span>)</span>
<span id="cb651-5"><a href="WE_5.html#cb651-5" aria-hidden="true" tabindex="-1"></a>IBD</span></code></pre></div>
<pre><code>## 
## Mantel statistic based on Pearson&#39;s product-moment correlation 
## 
## Call:
## vegan::mantel(xdis = Dgen, ydis = Dgeo, method = &quot;pearson&quot;) 
## 
## Mantel statistic r: 0.6211 
##       Significance: 0.001 
## 
## Upper quantiles of permutations (null model):
##    90%    95%  97.5%    99% 
## 0.0750 0.0994 0.1327 0.1612 
## Permutation: free
## Number of permutations: 999</code></pre>
<p>What happens if we take the log of geographic distance, which we saw above helps linearize the relationship for these data?</p>
<div class="sourceCode" id="cb653"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb653-1"><a href="WE_5.html#cb653-1" aria-hidden="true" tabindex="-1"></a>IBD <span class="ot">&lt;-</span> vegan<span class="sc">::</span><span class="fu">mantel</span>(Dgen,<span class="fu">log</span>(Dgeo), <span class="at">method=</span><span class="st">&quot;pearson&quot;</span>)</span>
<span id="cb653-2"><a href="WE_5.html#cb653-2" aria-hidden="true" tabindex="-1"></a>IBD</span></code></pre></div>
<pre><code>## 
## Mantel statistic based on Pearson&#39;s product-moment correlation 
## 
## Call:
## vegan::mantel(xdis = Dgen, ydis = log(Dgeo), method = &quot;pearson&quot;) 
## 
## Mantel statistic r: 0.6839 
##       Significance: 0.001 
## 
## Upper quantiles of permutations (null model):
##    90%    95%  97.5%    99% 
## 0.0742 0.0949 0.1238 0.1660 
## Permutation: free
## Number of permutations: 999</code></pre>
<p>The statistical significance (p-value) didn’t really change, but the strength of the Mantel correlation increased from r = 0.62 to r = 0.68.</p>
<p>Instead of transforming variables, we could use Spearman rank correlation to quantify the strength of the association. Rank correlations can be used to quantify and test the strength of curvilinear relationships.</p>
<div class="sourceCode" id="cb655"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb655-1"><a href="WE_5.html#cb655-1" aria-hidden="true" tabindex="-1"></a>IBD <span class="ot">&lt;-</span> vegan<span class="sc">::</span><span class="fu">mantel</span>(Dgen,Dgeo, <span class="at">method=</span><span class="st">&quot;spearman&quot;</span>)</span>
<span id="cb655-2"><a href="WE_5.html#cb655-2" aria-hidden="true" tabindex="-1"></a>IBD</span></code></pre></div>
<pre><code>## 
## Mantel statistic based on Spearman&#39;s rank correlation rho 
## 
## Call:
## vegan::mantel(xdis = Dgen, ydis = Dgeo, method = &quot;spearman&quot;) 
## 
## Mantel statistic r: 0.6512 
##       Significance: 0.001 
## 
## Upper quantiles of permutations (null model):
##    90%    95%  97.5%    99% 
## 0.0766 0.1039 0.1304 0.1545 
## Permutation: free
## Number of permutations: 999</code></pre>
<p>For this measure of genetic diversity, the Mantel correlation was significant, quite strong, and the relationship not linear, hence the transformed data or the rank correlation performed better. What about the other measures?</p>
<p>Here we use <code>lapply</code> to apply the function <code>mantel</code> to each genetic distance matrix in <code>GD.pop</code>. Then we use <code>sapply</code> to extract two values for each distance matrix: <code>statistic</code> is the Mantel r statistic (here: Pearson linear correlation), and <code>signif</code> is the p-value. We can find these names with the function <code>attributes</code> (see above).</p>
<div class="sourceCode" id="cb657"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb657-1"><a href="WE_5.html#cb657-1" aria-hidden="true" tabindex="-1"></a><span class="fu">attributes</span>(IBD)</span></code></pre></div>
<pre><code>## $names
## [1] &quot;call&quot;         &quot;method&quot;       &quot;statistic&quot;    &quot;signif&quot;       &quot;perm&quot;        
## [6] &quot;permutations&quot; &quot;control&quot;     
## 
## $class
## [1] &quot;mantel&quot;</code></pre>
<div class="sourceCode" id="cb659"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb659-1"><a href="WE_5.html#cb659-1" aria-hidden="true" tabindex="-1"></a>Mantel.test <span class="ot">&lt;-</span> <span class="fu">lapply</span>(GD.pop, <span class="cf">function</span>(x) vegan<span class="sc">::</span><span class="fu">mantel</span>(x,Dgeo, <span class="at">method=</span><span class="st">&quot;pearson&quot;</span>))</span>
<span id="cb659-2"><a href="WE_5.html#cb659-2" aria-hidden="true" tabindex="-1"></a><span class="fu">data.frame</span>(<span class="at">Mantel.r =</span> <span class="fu">sapply</span>(Mantel.test, <span class="cf">function</span>(x) x<span class="sc">$</span>statistic),</span>
<span id="cb659-3"><a href="WE_5.html#cb659-3" aria-hidden="true" tabindex="-1"></a>           <span class="at">p.value =</span> <span class="fu">sapply</span>(Mantel.test, <span class="cf">function</span>(x) x<span class="sc">$</span>signif))</span></code></pre></div>
<pre><code>##                          Mantel.r p.value
## pairwiseFst.hierfstat   0.6238025   0.001
## propShared.PopGenReport 0.6211335   0.001
## Nei.adegenet            0.6316224   0.001
## Edwards.adegenet        0.6639963   0.001
## Reynolds.adegenet       0.6162305   0.001
## Rogers.adegenet         0.6188092   0.001
## Provesti.adegenet       0.6211335   0.001
## Joost.mmod              0.6556870   0.001
## Hedrick.mmod            0.6520614   0.001
## Nei.mmod                0.6147866   0.001
## Euclidean.gstudio       0.6591863   0.001
## cGD.gstudio             0.5667087   0.001
## Nei.gstudio             0.6317948   0.001
## Dps.gstudio             0.5371591   0.001
## Jaccard.gstudio         0.5357096   0.001</code></pre>
<p>The nature of the result did not depend on the measure of genetic diversity used. Let’s repeat the analysis with log(Dgeo).</p>
<div class="sourceCode" id="cb661"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb661-1"><a href="WE_5.html#cb661-1" aria-hidden="true" tabindex="-1"></a>Mantel.test <span class="ot">&lt;-</span> <span class="fu">lapply</span>(GD.pop, <span class="cf">function</span>(x) ade4<span class="sc">::</span><span class="fu">mantel.randtest</span>(x,<span class="fu">log</span>(Dgeo)))</span>
<span id="cb661-2"><a href="WE_5.html#cb661-2" aria-hidden="true" tabindex="-1"></a><span class="fu">data.frame</span>(<span class="at">Mantel.r =</span> <span class="fu">sapply</span>(Mantel.test, <span class="cf">function</span>(x) x<span class="sc">$</span>obs),</span>
<span id="cb661-3"><a href="WE_5.html#cb661-3" aria-hidden="true" tabindex="-1"></a>           <span class="at">p.value =</span> <span class="fu">sapply</span>(Mantel.test, <span class="cf">function</span>(x) x<span class="sc">$</span>pvalue))</span></code></pre></div>
<pre><code>##                          Mantel.r p.value
## pairwiseFst.hierfstat   0.6218519   0.001
## propShared.PopGenReport 0.6839092   0.001
## Nei.adegenet            0.6547320   0.001
## Edwards.adegenet        0.7122499   0.001
## Reynolds.adegenet       0.6437788   0.001
## Rogers.adegenet         0.6718605   0.001
## Provesti.adegenet       0.6839092   0.001
## Joost.mmod              0.6748485   0.001
## Hedrick.mmod            0.6673508   0.001
## Nei.mmod                0.6034555   0.001
## Euclidean.gstudio       0.6967457   0.001
## cGD.gstudio             0.6008983   0.001
## Nei.gstudio             0.6548413   0.001
## Dps.gstudio             0.5911350   0.001
## Jaccard.gstudio         0.5880305   0.001</code></pre>
<p><strong>Questions:</strong>
- Did the transformation lead to higher Mantel correlations for all measures of genetic distance used here?
- Why was it higher?</p>
</div>
</div>
<div id="create-mantel-correlogram-for-genetic-data" class="section level3 unnumbered hasAnchor">
<h3>5. Create Mantel correlogram for genetic data<a href="WE_5.html#create-mantel-correlogram-for-genetic-data" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Let’s look at the relationship between genetic distance and geographic distance in a different way, with a Mantel correlogram. Note that this method does not make the assumption of linearity.</p>
<div id="a.-create-a-first-mantel-correlogram" class="section level4 unnumbered hasAnchor">
<h4>a. Create a first Mantel correlogram<a href="WE_5.html#a.-create-a-first-mantel-correlogram" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>Here, we’ll create a population-level Mantel correlogram with the proportion of shared alleles.</p>
<p>Note: The function <code>eco.cormantel</code> has an option <code>latlon=TRUE</code> that takes care of the distance calculation from lat-lon coordinates. To use this option, the coordinates must be in a matrix or data frame with the longitude in the first column and the latitude in the second. Here, we can set we can set <code>latlon=FALSE</code> because the spatial coordinates are in UTM projection.</p>
<p>The biological hypothesis of isolation-by-distance postulates that genetic distance increases with geographic distance. Hence it makes sense to use a one-sided alternative. Somewhat counter-intutitively, we use ‘alternative=“less”’ to test for positive spatial autocorrelation.</p>
<div class="sourceCode" id="cb663"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb663-1"><a href="WE_5.html#cb663-1" aria-hidden="true" tabindex="-1"></a>corm <span class="ot">&lt;-</span> EcoGenetics<span class="sc">::</span><span class="fu">eco.cormantel</span>(<span class="at">M =</span> GD.pop<span class="sc">$</span>propShared.PopGenReport, </span>
<span id="cb663-2"><a href="WE_5.html#cb663-2" aria-hidden="true" tabindex="-1"></a>        <span class="at">XY =</span> <span class="fu">ecoslot.XY</span>(Frogs.ecopop)[,<span class="dv">1</span><span class="sc">:</span><span class="dv">2</span>],  <span class="at">nsim =</span> <span class="dv">199</span>, <span class="at">latlon=</span><span class="cn">FALSE</span>, </span>
<span id="cb663-3"><a href="WE_5.html#cb663-3" aria-hidden="true" tabindex="-1"></a>        <span class="at">alternative=</span><span class="st">&quot;less&quot;</span>, <span class="at">method =</span> <span class="st">&quot;pearson&quot;</span>)</span></code></pre></div>
<pre><code>## 
 interval 0 / 10 completed
 interval 1 / 10 completed
 interval 2 / 10 completed
 interval 3 / 10 completed
 interval 4 / 10 completed
 interval 5 / 10 completed
 interval 6 / 10 completed
 interval 7 / 10 completed
 interval 8 / 10 completed
 interval 9 / 10 completed
 interval 10 / 10 completed</code></pre>
<div class="sourceCode" id="cb665"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb665-1"><a href="WE_5.html#cb665-1" aria-hidden="true" tabindex="-1"></a>corm</span></code></pre></div>
<pre><code>## 
##  ############################ 
##    Mantel statistic 
##  ############################ 
## 
##  &gt; Correlation coefficient used --&gt;  Pearson 
##   &gt; Number of simulations --&gt;  199 
##   &gt; Random test --&gt;  permutation 
##   &gt; P-adjust method --&gt;  holm -sequential: TRUE 
## 
##   &gt;  ecoslot.OUT(x): results --&gt;  
## 
## [[1]]
##                           d.mean     obs     exp p.val cardinal
## d= 0 - 1055.243          697.776  0.4815  0.0085 0.005       45
## d= 1055.243 - 2110.485  1583.490  0.2708 -0.0021 0.010       67
## d= 2110.485 - 3165.728  2622.371  0.2043 -0.0057 0.015       64
## d= 3165.728 - 4220.97   3710.591 -0.1302  0.0015 1.000       64
## d= 4220.97 - 5276.213   4716.029 -0.1719 -0.0053 1.000       54
## d= 5276.213 - 6331.455  5824.450 -0.2021  0.0010 1.000       46
## d= 6331.455 - 7386.698  6856.457 -0.1496  0.0009 1.000       27
## d= 7386.698 - 8441.941  7883.679 -0.2297  0.0038 1.000       33
## d= 8441.941 - 9497.183  8882.536 -0.1909  0.0005 1.000       25
## d= 9497.183 - 10552.426 9847.000 -0.0807 -0.0062 1.000        9
## 
## 
##  Results table(s) in ecoslot.OUT(x) 
## --------------------------------------------------------------------------
##  Access to slots: &lt;ecoslot.&gt; + &lt;name of the slot&gt; + &lt;(name of the object)&gt; 
##  See: help(&quot;EcoGenetics accessors&quot;)</code></pre>
<p>The table shows:</p>
<ul>
<li>Breaks of distance lag ‘d’: here in meters (default for lag definition: Sturges rule)</li>
<li>Mean distance ‘d.mean’: mean distance of pairs in each lag.</li>
<li>obs: observed value of the statistic</li>
<li>Expected value ‘exp’: expected if there is no autocorrelation.</li>
<li>P-value ‘p.val’: default uses a two-sided permutation test with sequential Holm-Bonferroni adjustement of p-values.</li>
<li>Number of pairs ‘cardinal’: number of unique pairs per lag.</li>
</ul>
<p>The result <code>corm</code> is an object of class <code>eco.correlog</code> (package: EcoGenetics). A safe way to access thet table is <code>ecoslot.OUT(corm)</code>.</p>
<p>Let’s plot the correlogram:</p>
<div class="sourceCode" id="cb667"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb667-1"><a href="WE_5.html#cb667-1" aria-hidden="true" tabindex="-1"></a>EcoGenetics<span class="sc">::</span><span class="fu">eco.plotCorrelog</span>(corm)</span></code></pre></div>
<p>You can hover over individual points of the correlogram to see the test statistic and mean distance.</p>
</div>
<div id="b.-vary-distance-class-definition" class="section level4 unnumbered hasAnchor">
<h4>b. Vary distance class definition<a href="WE_5.html#b.-vary-distance-class-definition" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>Under IBD, at least the first distance lag should show positive spatial autocorrelation. Here, it is the first two classes, as indicated by the red symbols.</p>
<p>To what degree does this result depend on the following:</p>
<ul>
<li>The distance lag definition?</li>
<li>The measure of genetic distance?</li>
<li>Non-linear relationship between genetic and geographic distances?</li>
</ul>
<p>There are several options of the <code>eco.cormantel</code> function to modify the definition of distance classes:</p>
<ul>
<li><code>int</code>: distance interval in the units of XY</li>
<li><code>smin</code>: minimum class distance in the units of XY</li>
<li><code>smax</code>: maximum class distance in the units of XY</li>
<li><code>nclass</code>: number of classes</li>
<li><code>seqvec</code>: vector with breaks in the unites of XY</li>
<li><code>size</code>: number of individuals per class</li>
<li><code>bin</code>: rule for constructing intervals if no other parameters provided (default: Sturge’s rule)</li>
</ul>
<p>The easiest ones to modify are either <code>nclass</code> or <code>size</code>. Here we use <code>size</code> to specify that there should be at least 50 or 100 pairs in each distance class. (Note: for a reliable analysis, this should be 100 or more).</p>
<div class="sourceCode" id="cb668"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb668-1"><a href="WE_5.html#cb668-1" aria-hidden="true" tabindex="-1"></a>corm<span class="fl">.50</span> <span class="ot">&lt;-</span> EcoGenetics<span class="sc">::</span><span class="fu">eco.cormantel</span>(<span class="at">M =</span> GD.pop<span class="sc">$</span>propShared.PopGenReport, </span>
<span id="cb668-2"><a href="WE_5.html#cb668-2" aria-hidden="true" tabindex="-1"></a>           <span class="at">XY =</span> <span class="fu">ecoslot.XY</span>(Frogs.ecopop)[,<span class="dv">1</span><span class="sc">:</span><span class="dv">2</span>],  <span class="at">nsim =</span> <span class="dv">199</span>, <span class="at">latlon=</span><span class="cn">FALSE</span>,</span>
<span id="cb668-3"><a href="WE_5.html#cb668-3" aria-hidden="true" tabindex="-1"></a>           <span class="at">alternative=</span><span class="st">&quot;less&quot;</span>, <span class="at">size=</span><span class="dv">50</span>)</span></code></pre></div>
<pre><code>## 
 interval 0 / 8 completed
 interval 1 / 8 completed
 interval 2 / 8 completed
 interval 3 / 8 completed
 interval 4 / 8 completed
 interval 5 / 8 completed
 interval 6 / 8 completed
 interval 7 / 8 completed
 interval 8 / 8 completed</code></pre>
<div class="sourceCode" id="cb670"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb670-1"><a href="WE_5.html#cb670-1" aria-hidden="true" tabindex="-1"></a>EcoGenetics<span class="sc">::</span><span class="fu">ecoslot.OUT</span>(corm<span class="fl">.50</span>)</span></code></pre></div>
<pre><code>## [[1]]
##                          d.mean     obs     exp p.val cardinal
## d= 0 - 1152.337         739.966  0.4872 -0.0076 0.005       50
## d= 1152.337 - 1821.843 1536.052  0.2256  0.0029 0.010       50
## d= 1821.843 - 2769.341 2286.134  0.2307  0.0047 0.015       50
## d= 2769.341 - 3526.318 3169.188  0.0411 -0.0019 0.185       50
## d= 3526.318 - 4408.262 3992.072 -0.1714 -0.0057 1.000       50
## d= 4408.262 - 5339.102 4863.176 -0.1579 -0.0077 1.000       50
## d= 5339.102 - 6668.986 6011.873 -0.2404 -0.0041 1.000       50
## d= 6668.986 - 8434.883 7615.126 -0.2429 -0.0127 1.000       50</code></pre>
<div class="sourceCode" id="cb672"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb672-1"><a href="WE_5.html#cb672-1" aria-hidden="true" tabindex="-1"></a>corm<span class="fl">.100</span> <span class="ot">&lt;-</span> EcoGenetics<span class="sc">::</span><span class="fu">eco.cormantel</span>(<span class="at">M =</span> GD.pop<span class="sc">$</span>propShared.PopGenReport, </span>
<span id="cb672-2"><a href="WE_5.html#cb672-2" aria-hidden="true" tabindex="-1"></a>           <span class="at">XY =</span> <span class="fu">ecoslot.XY</span>(Frogs.ecopop)[,<span class="dv">1</span><span class="sc">:</span><span class="dv">2</span>],  <span class="at">nsim =</span> <span class="dv">199</span>, <span class="at">latlon=</span><span class="cn">FALSE</span>,</span>
<span id="cb672-3"><a href="WE_5.html#cb672-3" aria-hidden="true" tabindex="-1"></a>           <span class="at">alternative=</span><span class="st">&quot;less&quot;</span>, <span class="at">size=</span><span class="dv">100</span>)</span></code></pre></div>
<pre><code>## 
 interval 0 / 4 completed
 interval 1 / 4 completed
 interval 2 / 4 completed
 interval 3 / 4 completed
 interval 4 / 4 completed</code></pre>
<div class="sourceCode" id="cb674"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb674-1"><a href="WE_5.html#cb674-1" aria-hidden="true" tabindex="-1"></a>EcoGenetics<span class="sc">::</span><span class="fu">ecoslot.OUT</span>(corm<span class="fl">.100</span>)</span></code></pre></div>
<pre><code>## [[1]]
##                          d.mean     obs     exp p.val cardinal
## d= 0 - 1821.843        1138.009  0.5403 -0.0006 0.005      100
## d= 1821.843 - 3526.318 2727.661  0.2060  0.0011 0.010      100
## d= 3526.318 - 5339.102 4427.624 -0.2496  0.0130 1.000      100
## d= 5339.102 - 8434.883 6813.499 -0.3664  0.0033 1.000      100</code></pre>
<div class="sourceCode" id="cb676"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb676-1"><a href="WE_5.html#cb676-1" aria-hidden="true" tabindex="-1"></a>EcoGenetics<span class="sc">::</span><span class="fu">eco.plotCorrelog</span>(corm<span class="fl">.50</span>)</span>
<span id="cb676-2"><a href="WE_5.html#cb676-2" aria-hidden="true" tabindex="-1"></a>EcoGenetics<span class="sc">::</span><span class="fu">eco.plotCorrelog</span>(corm<span class="fl">.100</span>)</span></code></pre></div>
<p>Let’s compare the observed Mantel r statistic, p-value, number of pairs in the first distance class and their mean distance, as well as the definition of the first lag interval. We can get all of this by extracting the first line from each object.</p>
<p>The lag intervals are stored only in the row names, and we need to extract those separately and add them as a colum.</p>
<div class="sourceCode" id="cb677"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb677-1"><a href="WE_5.html#cb677-1" aria-hidden="true" tabindex="-1"></a>Lag1.def <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="fu">rbind</span>(<span class="at">Sturge =</span> EcoGenetics<span class="sc">::</span><span class="fu">ecoslot.OUT</span>(corm)[[<span class="dv">1</span>]][<span class="dv">1</span>,],</span>
<span id="cb677-2"><a href="WE_5.html#cb677-2" aria-hidden="true" tabindex="-1"></a>      <span class="at">size.50 =</span> EcoGenetics<span class="sc">::</span><span class="fu">ecoslot.OUT</span>(corm<span class="fl">.50</span>)[[<span class="dv">1</span>]][<span class="dv">1</span>,],</span>
<span id="cb677-3"><a href="WE_5.html#cb677-3" aria-hidden="true" tabindex="-1"></a>      <span class="at">size.100 =</span> EcoGenetics<span class="sc">::</span><span class="fu">ecoslot.OUT</span>(corm<span class="fl">.100</span>)[[<span class="dv">1</span>]][<span class="dv">1</span>,]))</span>
<span id="cb677-4"><a href="WE_5.html#cb677-4" aria-hidden="true" tabindex="-1"></a>Lag1.def<span class="sc">$</span>bin <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="fu">row.names</span>(EcoGenetics<span class="sc">::</span><span class="fu">ecoslot.OUT</span>(corm)[[<span class="dv">1</span>]])[<span class="dv">1</span>],</span>
<span id="cb677-5"><a href="WE_5.html#cb677-5" aria-hidden="true" tabindex="-1"></a>              <span class="fu">row.names</span>(EcoGenetics<span class="sc">::</span><span class="fu">ecoslot.OUT</span>(corm<span class="fl">.50</span>)[[<span class="dv">1</span>]])[<span class="dv">1</span>],</span>
<span id="cb677-6"><a href="WE_5.html#cb677-6" aria-hidden="true" tabindex="-1"></a>              <span class="fu">row.names</span>(EcoGenetics<span class="sc">::</span><span class="fu">ecoslot.OUT</span>(corm<span class="fl">.100</span>)[[<span class="dv">1</span>]])[<span class="dv">1</span>])</span>
<span id="cb677-7"><a href="WE_5.html#cb677-7" aria-hidden="true" tabindex="-1"></a>Lag1.def</span></code></pre></div>
<pre><code>##            d.mean    obs     exp p.val cardinal             bin
## Sturge    697.776 0.4815  0.0085 0.005       45 d= 0 - 1055.243
## size.50   739.966 0.4872 -0.0076 0.005       50 d= 0 - 1152.337
## size.100 1138.009 0.5403 -0.0006 0.005      100 d= 0 - 1821.843</code></pre>
<ul>
<li>All three distance class definitions (Sturge’s rule, size = 50, size = 100) resulted in statistically significant p-values.</li>
<li>The Mantel correlation in the first distance lag was strongest for size.100.</li>
<li>For the first distance class, Sturge’s rule resulted in 45 pairs and a mean distance similar to size.50 (for this specific dataset).</li>
</ul>
<p>Overall, Sturge’s rule to define distance classes seems to be a good compromise. What is the trade-off, i.e., what happens if distance lags are defined too narrowly or too widely?</p>
<p>It can be really helpful to plot the distribution of distances among the pairs and compare it to the distance intervals:</p>
<div class="sourceCode" id="cb679"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb679-1"><a href="WE_5.html#cb679-1" aria-hidden="true" tabindex="-1"></a><span class="fu">par</span>(<span class="at">mfrow=</span><span class="fu">c</span>(<span class="dv">3</span>,<span class="dv">1</span>))</span>
<span id="cb679-2"><a href="WE_5.html#cb679-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb679-3"><a href="WE_5.html#cb679-3" aria-hidden="true" tabindex="-1"></a><span class="fu">hist</span>(Dgeo, <span class="at">nclass=</span><span class="dv">50</span>, <span class="at">main=</span><span class="st">&quot;Sturge&#39;s rule&quot;</span>, <span class="at">axes=</span>F, <span class="at">xlab=</span><span class="st">&quot;&quot;</span>, <span class="at">ylab=</span><span class="st">&quot;&quot;</span>)</span>
<span id="cb679-4"><a href="WE_5.html#cb679-4" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span>(i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(EcoGenetics<span class="sc">::</span><span class="fu">ecoslot.BREAKS</span>(corm))){</span>
<span id="cb679-5"><a href="WE_5.html#cb679-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">lines</span>(<span class="fu">rep</span>(EcoGenetics<span class="sc">::</span><span class="fu">ecoslot.BREAKS</span>(corm)[i], <span class="dv">2</span>), <span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">50</span>), <span class="at">col=</span><span class="st">&quot;blue&quot;</span>)}</span>
<span id="cb679-6"><a href="WE_5.html#cb679-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb679-7"><a href="WE_5.html#cb679-7" aria-hidden="true" tabindex="-1"></a><span class="fu">hist</span>(Dgeo, <span class="at">nclass=</span><span class="dv">50</span>, <span class="at">main =</span> <span class="st">&quot;50 pairs per lag&quot;</span>, <span class="at">axes=</span>F)</span>
<span id="cb679-8"><a href="WE_5.html#cb679-8" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span>(i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(EcoGenetics<span class="sc">::</span><span class="fu">ecoslot.BREAKS</span>(corm<span class="fl">.50</span>))){</span>
<span id="cb679-9"><a href="WE_5.html#cb679-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">lines</span>(<span class="fu">rep</span>(EcoGenetics<span class="sc">::</span><span class="fu">ecoslot.BREAKS</span>(corm<span class="fl">.50</span>)[i], <span class="dv">2</span>), <span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">50</span>), <span class="at">col=</span><span class="st">&quot;blue&quot;</span>)}</span>
<span id="cb679-10"><a href="WE_5.html#cb679-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb679-11"><a href="WE_5.html#cb679-11" aria-hidden="true" tabindex="-1"></a><span class="fu">hist</span>(Dgeo, <span class="at">nclass=</span><span class="dv">50</span>, <span class="at">main =</span> <span class="st">&quot;100 pairs per lag&quot;</span>, <span class="at">axes=</span>F)</span>
<span id="cb679-12"><a href="WE_5.html#cb679-12" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span>(i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(EcoGenetics<span class="sc">::</span><span class="fu">ecoslot.BREAKS</span>(corm<span class="fl">.100</span>))){</span>
<span id="cb679-13"><a href="WE_5.html#cb679-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">lines</span>(<span class="fu">rep</span>(EcoGenetics<span class="sc">::</span><span class="fu">ecoslot.BREAKS</span>(corm<span class="fl">.100</span>)[i], <span class="dv">2</span>), <span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">50</span>), <span class="at">col=</span><span class="st">&quot;blue&quot;</span>)}</span></code></pre></div>
<p><img src="05-Week05_files/figure-html/unnamed-chunk-38-1.png" width="768" /></p>
<p><strong>Question:</strong> Compare what happens at larger distances. Do you think Sturge’s rule does a good job for these as well?</p>
<p>Unlike a Mantel test, where all pairs are considered, in geostatistics we typically interpret only values for distances up to a certain threshold, e.g. half the maximum distance, for two reasons:</p>
<ul>
<li>There are few pairs in of these bins, making estimates highly variable.</li>
<li>Not all pairs contribute (equally) to the largest distance classes (those in the center of the study area are not involved in very large distances).</li>
</ul>
</div>
<div id="c.-alternative-measures-of-genetic-distances" class="section level4 unnumbered hasAnchor">
<h4>c. Alternative measures of genetic distances<a href="WE_5.html#c.-alternative-measures-of-genetic-distances" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>Which measure of genetic distance would provide the strongest Mantel correlation in the first distance class for this data set?</p>
<p>Here we will cycle through all genetic distance matrices in <code>GD.pop</code> and calculate a Mantel correlogram with Sturge’s rule (not linearized, method=“pearson”). This may take a while.</p>
<p>Note: the code that calculates <code>corm.GD.pop</code> is included here twice, first commented out, then with the option <code>include=FALSE</code>. This avoids printing out a lot of unnecessary output while still showing the (commented out) code in the .html version of the file.</p>
<div class="sourceCode" id="cb680"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb680-1"><a href="WE_5.html#cb680-1" aria-hidden="true" tabindex="-1"></a><span class="co">#corm.GD.pop &lt;- lapply(GD.pop, function(x) EcoGenetics::eco.cormantel(M = x, </span></span>
<span id="cb680-2"><a href="WE_5.html#cb680-2" aria-hidden="true" tabindex="-1"></a><span class="co">#                      XY = ecoslot.XY(Frogs.ecopop)[,1:2],  nsim = 199, latlon=FALSE,</span></span>
<span id="cb680-3"><a href="WE_5.html#cb680-3" aria-hidden="true" tabindex="-1"></a><span class="co">#                      alternative=&quot;less&quot;))</span></span></code></pre></div>
<p>Next, we extract for each genetic distance matrix the observed value of the Mantel correlation for the first distance class and its p-value.</p>
<div class="sourceCode" id="cb681"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb681-1"><a href="WE_5.html#cb681-1" aria-hidden="true" tabindex="-1"></a><span class="fu">t</span>(<span class="fu">sapply</span>(corm.GD.pop, <span class="cf">function</span>(x) EcoGenetics<span class="sc">::</span><span class="fu">ecoslot.OUT</span>(x)[[<span class="dv">1</span>]][<span class="dv">1</span>,<span class="fu">c</span>(<span class="dv">2</span>,<span class="dv">4</span>)]))</span></code></pre></div>
<pre><code>##                            obs p.val
## pairwiseFst.hierfstat   0.3908 0.005
## propShared.PopGenReport 0.4815 0.005
## Nei.adegenet            0.4175 0.005
## Edwards.adegenet        0.4782 0.005
## Reynolds.adegenet       0.4337 0.005
## Rogers.adegenet         0.4704 0.005
## Provesti.adegenet       0.4815 0.005
## Joost.mmod              0.4331 0.005
## Hedrick.mmod            0.4331 0.005
## Nei.mmod                0.3698 0.005
## Euclidean.gstudio       0.4721 0.005
## cGD.gstudio             0.3647 0.005
## Nei.gstudio             0.4175 0.005
## Dps.gstudio             0.4074 0.005
## Jaccard.gstudio         0.4031 0.005</code></pre>
<p>Compare the p-values: for this dataset, all genetic distance measures resulted in significant spatial autocorrelation (indicating IBD).</p>
<p>Let’s plot the Mantel correlogram for <code>Nei.adegenet</code>. Statistically significant lags are shown in a different color than non-significant ones.</p>
<div class="sourceCode" id="cb683"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb683-1"><a href="WE_5.html#cb683-1" aria-hidden="true" tabindex="-1"></a>EcoGenetics<span class="sc">::</span><span class="fu">eco.plotCorrelog</span>(corm.GD.pop<span class="sc">$</span>Nei.adegenet)</span></code></pre></div>
<p><strong>Questions:</strong></p>
<ul>
<li>Starting with the first distance class, how many consecutive distance classes showed significant spatial autocorrelation?</li>
<li>At what distance (range), roughly, did the positive autocorrelation disappear?</li>
</ul>
</div>
</div>
<div id="specify-spatial-weights-and-calculate-morans-i" class="section level3 unnumbered hasAnchor">
<h3>6. Specify spatial weights and calculate Moran’s I<a href="WE_5.html#specify-spatial-weights-and-calculate-morans-i" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>In this part, we’ll quantify and test Moran’s I for the genetic diversity data as calculated in Week 3 lab.</p>
<p>Note: Above, we used a distance lag approach from geostatistics, here we use spatial neighbours and weights (neighbor topology). Either approach could be used with either type of data.</p>
<p>For a detailed tutorial on defining spatial neighbors and weights, see: <a href="https://cran.r-project.org/web/packages/adespatial/vignettes/tutorial.html#irregular-samplings" class="uri">https://cran.r-project.org/web/packages/adespatial/vignettes/tutorial.html#irregular-samplings</a></p>
<div id="a.-defining-spatial-neighbors" class="section level4 unnumbered hasAnchor">
<h4>a. Defining spatial neighbors<a href="WE_5.html#a.-defining-spatial-neighbors" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>The function <code>chooseCN</code> (package: <code>adegenet</code>) provides an interface for choosing a connection network, i.e., for defining spatial neighbors. The underlying functions are defined in package <code>spdep</code> (for defining spatial dependences). It can return the following graph types:</p>
<ul>
<li>Delaunay triangulation (type 1)</li>
<li>Gabriel graph (type 2)</li>
<li>Relative neighbours (type 3)</li>
<li>Minimum spanning tree (type 4)</li>
<li>Neighbourhood by distance (type 5)</li>
<li>K nearests neighbours (type 6)</li>
<li>Inverse distances (type 7)</li>
</ul>
<p>Here we use types 1 - 6 to define neighbors in different ways. Then we plot each graph in geographic space. Lines indicate pairs of sites classified as neighbors.</p>
<p>Note: this function expects metric spatial coordinates (e.g., UTM).</p>
<div class="sourceCode" id="cb684"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb684-1"><a href="WE_5.html#cb684-1" aria-hidden="true" tabindex="-1"></a>nb.del <span class="ot">&lt;-</span> adegenet<span class="sc">::</span><span class="fu">chooseCN</span>(<span class="at">xy =</span> <span class="fu">ecoslot.XY</span>(Frogs.ecopop)[,<span class="dv">1</span><span class="sc">:</span><span class="dv">2</span>], </span>
<span id="cb684-2"><a href="WE_5.html#cb684-2" aria-hidden="true" tabindex="-1"></a>          <span class="at">result.type =</span> <span class="st">&quot;nb&quot;</span>, <span class="at">plot.nb =</span> <span class="cn">FALSE</span>, <span class="at">type =</span> <span class="dv">1</span>)</span></code></pre></div>
<pre><code>## Registered S3 method overwritten by &#39;spdep&#39;:
##   method   from
##   plot.mst ape</code></pre>
<div class="sourceCode" id="cb686"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb686-1"><a href="WE_5.html#cb686-1" aria-hidden="true" tabindex="-1"></a>nb.gab <span class="ot">&lt;-</span> adegenet<span class="sc">::</span><span class="fu">chooseCN</span>(<span class="at">xy =</span> <span class="fu">ecoslot.XY</span>(Frogs.ecopop)[,<span class="dv">1</span><span class="sc">:</span><span class="dv">2</span>], </span>
<span id="cb686-2"><a href="WE_5.html#cb686-2" aria-hidden="true" tabindex="-1"></a>          <span class="at">result.type =</span> <span class="st">&quot;nb&quot;</span>, <span class="at">plot.nb =</span> <span class="cn">FALSE</span>, <span class="at">type =</span> <span class="dv">2</span>)</span>
<span id="cb686-3"><a href="WE_5.html#cb686-3" aria-hidden="true" tabindex="-1"></a>nb.rel <span class="ot">&lt;-</span> adegenet<span class="sc">::</span><span class="fu">chooseCN</span>(<span class="at">xy =</span> <span class="fu">ecoslot.XY</span>(Frogs.ecopop)[,<span class="dv">1</span><span class="sc">:</span><span class="dv">2</span>], </span>
<span id="cb686-4"><a href="WE_5.html#cb686-4" aria-hidden="true" tabindex="-1"></a>          <span class="at">result.type =</span> <span class="st">&quot;nb&quot;</span>, <span class="at">plot.nb =</span> <span class="cn">FALSE</span>, <span class="at">type =</span> <span class="dv">3</span>)</span>
<span id="cb686-5"><a href="WE_5.html#cb686-5" aria-hidden="true" tabindex="-1"></a>nb.mst <span class="ot">&lt;-</span> adegenet<span class="sc">::</span><span class="fu">chooseCN</span>(<span class="at">xy =</span> <span class="fu">ecoslot.XY</span>(Frogs.ecopop)[,<span class="dv">1</span><span class="sc">:</span><span class="dv">2</span>], </span>
<span id="cb686-6"><a href="WE_5.html#cb686-6" aria-hidden="true" tabindex="-1"></a>          <span class="at">result.type =</span> <span class="st">&quot;nb&quot;</span>, <span class="at">plot.nb =</span> <span class="cn">FALSE</span>, <span class="at">type =</span> <span class="dv">4</span>)</span>
<span id="cb686-7"><a href="WE_5.html#cb686-7" aria-hidden="true" tabindex="-1"></a>nb.nbd <span class="ot">&lt;-</span> adegenet<span class="sc">::</span><span class="fu">chooseCN</span>(<span class="at">xy =</span> <span class="fu">ecoslot.XY</span>(Frogs.ecopop)[,<span class="dv">1</span><span class="sc">:</span><span class="dv">2</span>], </span>
<span id="cb686-8"><a href="WE_5.html#cb686-8" aria-hidden="true" tabindex="-1"></a>          <span class="at">result.type =</span> <span class="st">&quot;nb&quot;</span>, <span class="at">plot.nb =</span> <span class="cn">FALSE</span>, <span class="at">type =</span> <span class="dv">5</span>, <span class="at">d1=</span><span class="dv">100</span>, <span class="at">d2=</span><span class="dv">15000</span>)</span>
<span id="cb686-9"><a href="WE_5.html#cb686-9" aria-hidden="true" tabindex="-1"></a>nb<span class="fl">.4</span>nn <span class="ot">&lt;-</span> adegenet<span class="sc">::</span><span class="fu">chooseCN</span>(<span class="at">xy =</span> <span class="fu">ecoslot.XY</span>(Frogs.ecopop)[,<span class="dv">1</span><span class="sc">:</span><span class="dv">2</span>], </span>
<span id="cb686-10"><a href="WE_5.html#cb686-10" aria-hidden="true" tabindex="-1"></a>          <span class="at">result.type =</span> <span class="st">&quot;nb&quot;</span>, <span class="at">plot.nb =</span> <span class="cn">FALSE</span>, <span class="at">type =</span> <span class="dv">6</span>, <span class="at">k =</span> <span class="dv">4</span>)</span>
<span id="cb686-11"><a href="WE_5.html#cb686-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb686-12"><a href="WE_5.html#cb686-12" aria-hidden="true" tabindex="-1"></a><span class="fu">par</span>(<span class="at">mfrow=</span><span class="fu">c</span>(<span class="dv">2</span>,<span class="dv">3</span>), <span class="at">mai=</span><span class="fu">c</span>(<span class="fl">0.1</span>,<span class="fl">0.1</span>,<span class="fl">0.1</span>, <span class="fl">0.1</span>))</span>
<span id="cb686-13"><a href="WE_5.html#cb686-13" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(nb.del, <span class="at">coords=</span><span class="fu">ecoslot.XY</span>(Frogs.ecopop)); <span class="fu">title</span>(<span class="at">main=</span><span class="st">&quot;Delaunay&quot;</span>)</span>
<span id="cb686-14"><a href="WE_5.html#cb686-14" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(nb.gab, <span class="at">coords=</span><span class="fu">ecoslot.XY</span>(Frogs.ecopop)); <span class="fu">title</span>(<span class="at">main=</span><span class="st">&quot;Gabriel&quot;</span>)</span>
<span id="cb686-15"><a href="WE_5.html#cb686-15" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(nb.rel, <span class="at">coords=</span><span class="fu">ecoslot.XY</span>(Frogs.ecopop)); <span class="fu">title</span>(<span class="at">main=</span> <span class="st">&quot;Rel. neighbors&quot;</span>)</span>
<span id="cb686-16"><a href="WE_5.html#cb686-16" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(nb.mst, <span class="at">coords=</span><span class="fu">ecoslot.XY</span>(Frogs.ecopop)); <span class="fu">title</span>(<span class="at">main=</span> <span class="st">&quot;Min spanning tree&quot;</span>)</span>
<span id="cb686-17"><a href="WE_5.html#cb686-17" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(nb.nbd, <span class="at">coords=</span><span class="fu">ecoslot.XY</span>(Frogs.ecopop)); <span class="fu">title</span>(<span class="at">main =</span> <span class="st">&quot;Neighbor distance&quot;</span>)</span>
<span id="cb686-18"><a href="WE_5.html#cb686-18" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(nb<span class="fl">.4</span>nn, <span class="at">coords=</span><span class="fu">ecoslot.XY</span>(Frogs.ecopop)); <span class="fu">title</span>(<span class="at">main =</span> <span class="st">&quot;4 nearest neighbors&quot;</span>)</span>
<span id="cb686-19"><a href="WE_5.html#cb686-19" aria-hidden="true" tabindex="-1"></a><span class="fu">par</span>(<span class="at">mfrow=</span><span class="fu">c</span>(<span class="dv">1</span>,<span class="dv">1</span>))</span></code></pre></div>
<p><img src="05-Week05_files/figure-html/unnamed-chunk-43-1.png" width="768" /></p>
<p>For spatial statistics, spatial neighbors are used to calculate a local mean. We want each site to have multiple neighbors, but they should be nearby only. Gabriel graph (type = 2) is often a good option, and we’ll use it for the rest of this worked example.</p>
</div>
<div id="b.-defining-spatial-weights" class="section level4 unnumbered hasAnchor">
<h4>b. Defining spatial weights<a href="WE_5.html#b.-defining-spatial-weights" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>By default, <code>chooseCN</code> returns row-standardized weights, so that for each site, the weights of its neighbors sum to 1. This means that a local mean can be calculated as a weighted mean of the other sites (non-neighboring sites have a weight of 0).</p>
<p>With the function <code>nb2mat</code> we can convert the neighbor object to a matrix of spatial weights. Let’s look at the first five lines and columns:</p>
<div class="sourceCode" id="cb687"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb687-1"><a href="WE_5.html#cb687-1" aria-hidden="true" tabindex="-1"></a>spdep<span class="sc">::</span><span class="fu">nb2mat</span>(nb.gab)[<span class="dv">1</span><span class="sc">:</span><span class="dv">5</span>,<span class="dv">1</span><span class="sc">:</span><span class="dv">5</span>]</span></code></pre></div>
<pre><code>##   [,1]      [,2]      [,3]      [,4] [,5]
## 1    0 0.0000000 0.0000000 0.0000000    0
## 2    0 0.0000000 0.1666667 0.1666667    0
## 3    0 0.2500000 0.0000000 0.0000000    0
## 4    0 0.3333333 0.0000000 0.0000000    0
## 5    0 0.0000000 0.0000000 0.0000000    0</code></pre>
<p>Each row contains the weights for the neighbors of one site. We see that the third site is a neighbor of the second site and vice versa. However, the weights are not the same. It seems that site 2 has six neighbors, so each has a weight of 0.167, whereas site 3 has four neighbors, each with a weight of 0.25.</p>
</div>
<div id="c.-calculating-and-testing-morans-i" class="section level4 unnumbered hasAnchor">
<h4>c. Calculating and testing Moran’s I<a href="WE_5.html#c.-calculating-and-testing-morans-i" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>Let’s see whether rarefied allelic richness, Ar, shows spatial autocorrelation among these sites.</p>
<div class="sourceCode" id="cb689"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb689-1"><a href="WE_5.html#cb689-1" aria-hidden="true" tabindex="-1"></a>spdep<span class="sc">::</span><span class="fu">moran.test</span>(<span class="fu">ecoslot.C</span>(Frogs.ecopop)<span class="sc">$</span>Ar, spdep<span class="sc">::</span><span class="fu">nb2listw</span>(nb.gab),</span>
<span id="cb689-2"><a href="WE_5.html#cb689-2" aria-hidden="true" tabindex="-1"></a>                  <span class="at">alternative=</span><span class="st">&quot;greater&quot;</span>)</span></code></pre></div>
<pre><code>## 
##  Moran I test under randomisation
## 
## data:  ecoslot.C(Frogs.ecopop)$Ar  
## weights: spdep::nb2listw(nb.gab)    
## 
## Moran I statistic standard deviate = 4.648, p-value = 1.676e-06
## alternative hypothesis: greater
## sample estimates:
## Moran I statistic       Expectation          Variance 
##        0.60990344       -0.03448276        0.01921995</code></pre>
<p>The test statistic is 0.61, and the p-value for a one-sided alternative “greater” (i.e., positive spatial autocorrelation) is <em>p</em> &lt; 0.0001. Thus, Ar showed strong and statistially significant spatial autocorrelation.</p>
<p>Let’s do this for all genetic diversity variables and extract the value of the Moran I statistics (for first neighbors) and its p-value.</p>
<div class="sourceCode" id="cb691"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb691-1"><a href="WE_5.html#cb691-1" aria-hidden="true" tabindex="-1"></a>Frogs.moran <span class="ot">&lt;-</span> <span class="fu">lapply</span>(<span class="fu">ecoslot.C</span>(Frogs.ecopop), </span>
<span id="cb691-2"><a href="WE_5.html#cb691-2" aria-hidden="true" tabindex="-1"></a>                       <span class="cf">function</span>(x) spdep<span class="sc">::</span><span class="fu">moran.test</span>(x,</span>
<span id="cb691-3"><a href="WE_5.html#cb691-3" aria-hidden="true" tabindex="-1"></a>                       spdep<span class="sc">::</span><span class="fu">nb2listw</span>(nb.gab), <span class="at">alternative=</span><span class="st">&quot;two.sided&quot;</span>))</span>
<span id="cb691-4"><a href="WE_5.html#cb691-4" aria-hidden="true" tabindex="-1"></a><span class="fu">round</span>(<span class="fu">data.frame</span>(<span class="at">obs =</span> <span class="fu">sapply</span>(Frogs.moran, <span class="cf">function</span>(x) <span class="fu">as.vector</span>(x<span class="sc">$</span>estimate[<span class="dv">1</span>])),  </span>
<span id="cb691-5"><a href="WE_5.html#cb691-5" aria-hidden="true" tabindex="-1"></a><span class="at">p.value =</span> <span class="fu">sapply</span>(Frogs.moran, <span class="cf">function</span>(x) x<span class="sc">$</span>p.value)),<span class="dv">3</span>)</span></code></pre></div>
<pre><code>##         obs p.value
## n    -0.109   0.606
## Hobs  0.507   0.000
## Hexp  0.551   0.000
## Ar    0.610   0.000</code></pre>
<p><strong>Questions:</strong></p>
<ul>
<li>Which measure of genetic diversity showed the strongest autocorrelation?</li>
<li>Sample size n had a negative value of Moran’s I (obs). What does this mean?</li>
<li>Did all four variables show statistically significant spatial autocorrelation?</li>
</ul>
</div>
</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="tutorial_5.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="r-exercise-week-5.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": true,
"facebook": false,
"twitter": false,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": false
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": null,
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "section"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
